{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import shutil\n",
    "from shutil import copy\n",
    "import time\n",
    "from collections import defaultdict\n",
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "os.chdir(\"/master/nplatt/pathogen_probes/\")\n",
    "\n",
    "\n",
    "def wait_on_running_jobs():\n",
    "   \n",
    "    num_jobs = 1\n",
    "    \n",
    "    while num_jobs > 0:\n",
    "        num_jobs = len(subprocess.check_output('qstat', shell=True).split(\"\\n\")) - 2\n",
    "        time.sleep(60)\n",
    "        print(\".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Apicomplexa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'apicomplexa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Ascogregarina_taiwanensis_GCA_000172235\",\n",
    "               \"Babesia_bigemina_GCF_000981445\",\n",
    "               \"Babesia_bovis_GCF_000165395\",\n",
    "               \"Babesia_divergens_GCA_001077455\",\n",
    "               \"Babesia_microti_GCF_000691945\",\n",
    "               \"Babesia_ovata_GCF_002897235\",\n",
    "               \"Babesia_sp_GCF_002095265\",\n",
    "               \"Besnoitia_besnoiti_GCF_002563875\",\n",
    "               \"Cryptospridium_andersoni_GCA_001865355\",\n",
    "               \"Cryptospridium_baileyi_GCA_001593455\",\n",
    "               \"Cryptospridium_bovis_GCA_009768925\",\n",
    "               \"Cryptospridium_cuniculus_GCA_004337835\",\n",
    "               \"Cryptospridium_hominis_GCA_000006425\",\n",
    "               \"Cryptospridium_meleagridis_GCA_001593445\",\n",
    "               \"Cryptospridium_muris_GCF_000006515\",\n",
    "               \"Cryptospridium_parvum_GCF_000165345\",\n",
    "               \"Cryptospridium_ryanae_GCA_009792415\",\n",
    "               \"Cryptospridium_sp_GCA_004936735\",\n",
    "               \"Cryptospridium_tyzzeri_GCA_007210665\",\n",
    "               \"Cryptospridium_ubiquitum_GCF_001865345\",\n",
    "               \"Cryptospridium_viatorum_GCA_004337795\",\n",
    "               \"Cyclospra_cayetanensis_GCF_002999335\",\n",
    "               \"Cystoisospra_suis_GCA_002600585\",\n",
    "               \"Eimeria_acervulina_GCF_000499425\",\n",
    "               \"Eimeria_brunetti_GCA_000499725\",\n",
    "               \"Eimeria_falciformis_GCA_002271815\",\n",
    "               \"Eimeria_maxima_GCF_000499605\",\n",
    "               \"Eimeria_mitis_GCA_000499745\",\n",
    "               \"Eimeria_necatrix_GCF_000499385\",\n",
    "               \"Eimeria_nieschulzi_GCA_000826945\",\n",
    "               \"Eimeria_praecox_GCA_000499445\",\n",
    "               \"Eimeria_tenella_GCF_000499545\",\n",
    "               \"Gregarina_niphandrodes_GCF_000223845\",\n",
    "               \"Haemoproteus_tartakovskyi_GCA_001625125\",\n",
    "               \"Hammondia_hammondi_GCA_000447165\",\n",
    "               \"Neospra_caninum_GCF_000208865\",\n",
    "               \"Nephromyces_sp_GCA_004523865\",\n",
    "               \"Plasmodium_berghei_GCA_900002375\",\n",
    "               \"Plasmodium_brasilianum_GCA_001885115\",\n",
    "               \"Plasmodium_chabaudi_GCA_900002335\",\n",
    "               \"Plasmodium_coatneyi_GCF_001680005\",\n",
    "               \"Plasmodium_cynomolgi_GCF_000321355\",\n",
    "               \"Plasmodium_fragile_GCF_000956335\",\n",
    "               \"Plasmodium_gaboni_GCF_001602025\",\n",
    "               \"Plasmodium_gallinaceum_GCF_900005855\",\n",
    "               \"Plasmodium_gonderi_GCF_002157705\",\n",
    "               \"Plasmodium_inui_GCF_000524495\",\n",
    "               \"Plasmodium_knowlesi_GCF_000006355\",\n",
    "               \"Plasmodium_malariae_GCF_900090045\",\n",
    "               \"Plasmodium_ovale_GCA_900090025\",\n",
    "               \"Plasmodium_reichenowi_GCF_001601855\",\n",
    "               \"Plasmodium_relictum_GCF_900005765\",\n",
    "               \"Plasmodium_sp_GCF_900097015\",\n",
    "               \"Plasmodium_vinckei_GCF_000709005\",\n",
    "               \"Plasmodium_vivax_GCF_000002415\",\n",
    "               \"Plasmodium_yoelii_GCA_900002385\",\n",
    "               \"Sarcocystis_neurona_GCA_000727475\",\n",
    "               \"Theileria_annulata_GCF_000003225\",\n",
    "               \"Theileria_equi_GCF_000342415\",\n",
    "               \"Theileria_orientalis_GCF_000740895\",\n",
    "               \"Theileria_parva_GCF_000165365\",\n",
    "               \"Toxoplasma_gondii_GCF_000006565\" ]\n",
    "                    \n",
    "reference_taxon = \"Plasmodium_falciparum_GCA_000002765\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000172235.1_ASM17223v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1911069 bytes  764444.40 bytes/sec\n",
      "total size is 1910492  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000981445.1_Bbig001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 4109992 bytes  1644013.60 bytes/sec\n",
      "total size is 4108882  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000165395.1_ASM16539v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2513750 bytes  1675861.33 bytes/sec\n",
      "total size is 2513029  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001077455.2_ASM107745v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2790058 bytes  1116040.00 bytes/sec\n",
      "total size is 2789264  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000691945.2_ASM69194v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2032965 bytes  1355338.00 bytes/sec\n",
      "total size is 2032356  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002897235.1_Bovata_7.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 4311003 bytes  2874030.00 bytes/sec\n",
      "total size is 4309842  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002095265.1_B_xinjiang1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2598453 bytes  1732330.00 bytes/sec\n",
      "total size is 2597705  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002563875.1_Bbes1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 18412682 bytes  7365089.60 bytes/sec\n",
      "total size is 18408083  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001865355.1_ASM186535v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2853226 bytes  1141307.20 bytes/sec\n",
      "total size is 2852416  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001593455.1_CryBaiTAMU-09Q1-1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2624008 bytes  1049620.00 bytes/sec\n",
      "total size is 2623246  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_009768925.1_ASM976892v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2893793 bytes  1157534.00 bytes/sec\n",
      "total size is 2892975  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004337835.1_UKCU2.v0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2384552 bytes  953837.60 bytes/sec\n",
      "total size is 2383865  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006425.2_ASM642v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2810052 bytes  1124037.60 bytes/sec\n",
      "total size is 2809261  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001593445.1_CryMelUKMEL1-1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2827708 bytes  1885166.67 bytes/sec\n",
      "total size is 2826901  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006515.1_JCVI_cmg_v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2888744 bytes  1155514.40 bytes/sec\n",
      "total size is 2887924  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000165345.1_ASM16534v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2849519 bytes  1139824.40 bytes/sec\n",
      "total size is 2848718  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009792415.1_ASM979241v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2882270 bytes  1921541.33 bytes/sec\n",
      "total size is 2881460  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004936735.1_ASM493673v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2849616 bytes  1139863.20 bytes/sec\n",
      "total size is 2848814  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007210665.1_Ctyz_UGA_55_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2801251 bytes  1120517.20 bytes/sec\n",
      "total size is 2800457  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001865345.1_ASM186534v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2821328 bytes  1880913.33 bytes/sec\n",
      "total size is 2820526  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004337795.1_ABER_CVIA_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2937753 bytes  1958530.00 bytes/sec\n",
      "total size is 2936925  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002999335.1_CcayRef3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 14143402 bytes  5657377.60 bytes/sec\n",
      "total size is 14139842  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002600585.1_ASM260058v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 26643909 bytes  5920878.00 bytes/sec\n",
      "total size is 26637298  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499425.1_EAH001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 14018126 bytes  5607267.20 bytes/sec\n",
      "total size is 14014600  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499725.1_EBH001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 19434463 bytes  4318778.89 bytes/sec\n",
      "total size is 19429617  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002271815.1_ASM227181v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 13292992 bytes  5317213.60 bytes/sec\n",
      "total size is 13289637  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499605.1_EMW001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 13935720 bytes  5574304.80 bytes/sec\n",
      "total size is 13932210  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499745.2_EMH001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 19588153 bytes  5596627.14 bytes/sec\n",
      "total size is 19583267  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499385.1_ENH001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16620870 bytes  4748832.00 bytes/sec\n",
      "total size is 16616704  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000826945.1_ASM82694v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20543548 bytes  5869597.14 bytes/sec\n",
      "total size is 20538426  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499445.1_EPH001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16589787 bytes  4739951.14 bytes/sec\n",
      "total size is 16585629  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499545.1_ETH001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15348007 bytes  6139219.60 bytes/sec\n",
      "total size is 15344153  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000223845.4_GNI3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 4126417 bytes  2750972.67 bytes/sec\n",
      "total size is 4125310  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001625125.1_ASM162512v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7082052 bytes  2023455.43 bytes/sec\n",
      "total size is 7080210  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000447165.1_Hammondia_hammondi_HhCatGer041_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 17069642 bytes  4877052.57 bytes/sec\n",
      "total size is 17065348  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000208865.2_ASM20886v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 18262194 bytes  5217781.71 bytes/sec\n",
      "total size is 18257624  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004523865.1_ASM452386v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 181449886 bytes  9808104.22 bytes/sec\n",
      "total size is 181405483  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900002375.2_GCA_900002375_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5772979 bytes  3848680.67 bytes/sec\n",
      "total size is 5771455  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001885115.2_Pbrasilianum_v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9289093 bytes  3715654.00 bytes/sec\n",
      "total size is 9286710  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900002335.2_GCA_900002335_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5916509 bytes  2366620.40 bytes/sec\n",
      "total size is 5914953  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001680005.1_ASM168000v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8850316 bytes  2528673.71 bytes/sec\n",
      "total size is 8848041  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000321355.1_PcynB_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8134353 bytes  3253758.00 bytes/sec\n",
      "total size is 8132257  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002765.3_GCA_000002765_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6674088 bytes  2669652.00 bytes/sec\n",
      "total size is 6672348  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000956335.1_Plas_frag_nilgiri_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7428007 bytes  2971219.60 bytes/sec\n",
      "total size is 7426076  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001602025.1_ASM160202v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5758622 bytes  2303465.60 bytes/sec\n",
      "total size is 5757108  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900005855.1_PGAL8A_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7011679 bytes  2804688.40 bytes/sec\n",
      "total size is 7009866  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002157705.1_Pgonderi_assembly01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10178427 bytes  4071387.60 bytes/sec\n",
      "total size is 10175824  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000524495.1_Plas_inui_San_Antonio_1_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8484883 bytes  2424264.29 bytes/sec\n",
      "total size is 8482689  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006355.1_ASM635v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7162133 bytes  2864870.00 bytes/sec\n",
      "total size is 7160278  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900090045.1_PmUG01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10267373 bytes  2933547.14 bytes/sec\n",
      "total size is 10264759  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900090025.2_PowCR01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10430557 bytes  2980171.14 bytes/sec\n",
      "total size is 10427902  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001601855.1_ASM160185v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5695450 bytes  2278196.80 bytes/sec\n",
      "total size is 5693952  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900005765.1_PRELSG_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6650499 bytes  1900154.57 bytes/sec\n",
      "total size is 6648774  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900097015.1_PADLG01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6429670 bytes  2571884.80 bytes/sec\n",
      "total size is 6427992  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000709005.1_Plas_vinc_vinckei_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5574903 bytes  2229978.00 bytes/sec\n",
      "total size is 5573420  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002415.2_ASM241v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8568852 bytes  3427557.60 bytes/sec\n",
      "total size is 8566652  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900002385.2_GCA_900002385_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6848912 bytes  2739581.60 bytes/sec\n",
      "total size is 6847132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000727475.1_ASM72747v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 39352240 bytes  6054197.23 bytes/sec\n",
      "total size is 39342526  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000003225.1_ASM322v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2577588 bytes  1031052.00 bytes/sec\n",
      "total size is 2576853  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000342415.1_JCVI-bewag-v1.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3636075 bytes  1454446.80 bytes/sec\n",
      "total size is 3635077  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000740895.1_ASM74089v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2831236 bytes  1887518.67 bytes/sec\n",
      "total size is 2830435  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000165365.1_ASM16536v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2589579 bytes  1035848.40 bytes/sec\n",
      "total size is 2588834  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006565.2_TGA4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20517776 bytes  5862233.71 bytes/sec\n",
      "total size is 20512668  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484306 (\"sim_Ascogregarina_taiwanensis_GCA_000172235\") has been submitted\n",
      "Your job 5484307 (\"sim_Babesia_bigemina_GCF_000981445\") has been submitted\n",
      "Your job 5484308 (\"sim_Babesia_bovis_GCF_000165395\") has been submitted\n",
      "Your job 5484309 (\"sim_Babesia_divergens_GCA_001077455\") has been submitted\n",
      "Your job 5484310 (\"sim_Babesia_microti_GCF_000691945\") has been submitted\n",
      "Your job 5484311 (\"sim_Babesia_ovata_GCF_002897235\") has been submitted\n",
      "Your job 5484312 (\"sim_Babesia_sp_GCF_002095265\") has been submitted\n",
      "Your job 5484313 (\"sim_Besnoitia_besnoiti_GCF_002563875\") has been submitted\n",
      "Your job 5484314 (\"sim_Cryptospridium_andersoni_GCA_001865355\") has been submitted\n",
      "Your job 5484315 (\"sim_Cryptospridium_baileyi_GCA_001593455\") has been submitted\n",
      "Your job 5484316 (\"sim_Cryptospridium_bovis_GCA_009768925\") has been submitted\n",
      "Your job 5484317 (\"sim_Cryptospridium_cuniculus_GCA_004337835\") has been submitted\n",
      "Your job 5484318 (\"sim_Cryptospridium_hominis_GCA_000006425\") has been submitted\n",
      "Your job 5484319 (\"sim_Cryptospridium_meleagridis_GCA_001593445\") has been submitted\n",
      "Your job 5484320 (\"sim_Cryptospridium_muris_GCF_000006515\") has been submitted\n",
      "Your job 5484321 (\"sim_Cryptospridium_parvum_GCF_000165345\") has been submitted\n",
      "Your job 5484322 (\"sim_Cryptospridium_ryanae_GCA_009792415\") has been submitted\n",
      "Your job 5484323 (\"sim_Cryptospridium_sp_GCA_004936735\") has been submitted\n",
      "Your job 5484324 (\"sim_Cryptospridium_tyzzeri_GCA_007210665\") has been submitted\n",
      "Your job 5484325 (\"sim_Cryptospridium_ubiquitum_GCF_001865345\") has been submitted\n",
      "Your job 5484326 (\"sim_Cryptospridium_viatorum_GCA_004337795\") has been submitted\n",
      "Your job 5484327 (\"sim_Cyclospra_cayetanensis_GCF_002999335\") has been submitted\n",
      "Your job 5484328 (\"sim_Cystoisospra_suis_GCA_002600585\") has been submitted\n",
      "Your job 5484329 (\"sim_Eimeria_acervulina_GCF_000499425\") has been submitted\n",
      "Your job 5484330 (\"sim_Eimeria_brunetti_GCA_000499725\") has been submitted\n",
      "Your job 5484331 (\"sim_Eimeria_falciformis_GCA_002271815\") has been submitted\n",
      "Your job 5484332 (\"sim_Eimeria_maxima_GCF_000499605\") has been submitted\n",
      "Your job 5484333 (\"sim_Eimeria_mitis_GCA_000499745\") has been submitted\n",
      "Your job 5484334 (\"sim_Eimeria_necatrix_GCF_000499385\") has been submitted\n",
      "Your job 5484335 (\"sim_Eimeria_nieschulzi_GCA_000826945\") has been submitted\n",
      "Your job 5484336 (\"sim_Eimeria_praecox_GCA_000499445\") has been submitted\n",
      "Your job 5484337 (\"sim_Eimeria_tenella_GCF_000499545\") has been submitted\n",
      "Your job 5484338 (\"sim_Gregarina_niphandrodes_GCF_000223845\") has been submitted\n",
      "Your job 5484339 (\"sim_Haemoproteus_tartakovskyi_GCA_001625125\") has been submitted\n",
      "Your job 5484340 (\"sim_Hammondia_hammondi_GCA_000447165\") has been submitted\n",
      "Your job 5484341 (\"sim_Neospra_caninum_GCF_000208865\") has been submitted\n",
      "Your job 5484342 (\"sim_Nephromyces_sp_GCA_004523865\") has been submitted\n",
      "Your job 5484343 (\"sim_Plasmodium_berghei_GCA_900002375\") has been submitted\n",
      "Your job 5484344 (\"sim_Plasmodium_brasilianum_GCA_001885115\") has been submitted\n",
      "Your job 5484345 (\"sim_Plasmodium_chabaudi_GCA_900002335\") has been submitted\n",
      "Your job 5484346 (\"sim_Plasmodium_coatneyi_GCF_001680005\") has been submitted\n",
      "Your job 5484347 (\"sim_Plasmodium_cynomolgi_GCF_000321355\") has been submitted\n",
      "Your job 5484348 (\"sim_Plasmodium_fragile_GCF_000956335\") has been submitted\n",
      "Your job 5484349 (\"sim_Plasmodium_gaboni_GCF_001602025\") has been submitted\n",
      "Your job 5484350 (\"sim_Plasmodium_gallinaceum_GCF_900005855\") has been submitted\n",
      "Your job 5484351 (\"sim_Plasmodium_gonderi_GCF_002157705\") has been submitted\n",
      "Your job 5484352 (\"sim_Plasmodium_inui_GCF_000524495\") has been submitted\n",
      "Your job 5484353 (\"sim_Plasmodium_knowlesi_GCF_000006355\") has been submitted\n",
      "Your job 5484354 (\"sim_Plasmodium_malariae_GCF_900090045\") has been submitted\n",
      "Your job 5484355 (\"sim_Plasmodium_ovale_GCA_900090025\") has been submitted\n",
      "Your job 5484356 (\"sim_Plasmodium_reichenowi_GCF_001601855\") has been submitted\n",
      "Your job 5484357 (\"sim_Plasmodium_relictum_GCF_900005765\") has been submitted\n",
      "Your job 5484358 (\"sim_Plasmodium_sp_GCF_900097015\") has been submitted\n",
      "Your job 5484359 (\"sim_Plasmodium_vinckei_GCF_000709005\") has been submitted\n",
      "Your job 5484360 (\"sim_Plasmodium_vivax_GCF_000002415\") has been submitted\n",
      "Your job 5484361 (\"sim_Plasmodium_yoelii_GCA_900002385\") has been submitted\n",
      "Your job 5484362 (\"sim_Sarcocystis_neurona_GCA_000727475\") has been submitted\n",
      "Your job 5484363 (\"sim_Theileria_annulata_GCF_000003225\") has been submitted\n",
      "Your job 5484364 (\"sim_Theileria_equi_GCF_000342415\") has been submitted\n",
      "Your job 5484365 (\"sim_Theileria_orientalis_GCF_000740895\") has been submitted\n",
      "Your job 5484366 (\"sim_Theileria_parva_GCF_000165365\") has been submitted\n",
      "Your job 5484367 (\"sim_Toxoplasma_gondii_GCF_000006565\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/apicomplexa/cleaned_genomes/Plasmodium_falciparum_GCA_000002765_formatted.fas path=results/phyluce/apicomplexa/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484368 (\"map_Ascogregarina_taiwanensis_GCA_000172235\") has been submitted\n",
      "Your job 5484369 (\"map_Babesia_bigemina_GCF_000981445\") has been submitted\n",
      "Your job 5484370 (\"map_Babesia_bovis_GCF_000165395\") has been submitted\n",
      "Your job 5484371 (\"map_Babesia_divergens_GCA_001077455\") has been submitted\n",
      "Your job 5484372 (\"map_Babesia_microti_GCF_000691945\") has been submitted\n",
      "Your job 5484373 (\"map_Babesia_ovata_GCF_002897235\") has been submitted\n",
      "Your job 5484374 (\"map_Babesia_sp_GCF_002095265\") has been submitted\n",
      "Your job 5484375 (\"map_Besnoitia_besnoiti_GCF_002563875\") has been submitted\n",
      "Your job 5484376 (\"map_Cryptospridium_andersoni_GCA_001865355\") has been submitted\n",
      "Your job 5484377 (\"map_Cryptospridium_baileyi_GCA_001593455\") has been submitted\n",
      "Your job 5484378 (\"map_Cryptospridium_bovis_GCA_009768925\") has been submitted\n",
      "Your job 5484379 (\"map_Cryptospridium_cuniculus_GCA_004337835\") has been submitted\n",
      "Your job 5484380 (\"map_Cryptospridium_hominis_GCA_000006425\") has been submitted\n",
      "Your job 5484381 (\"map_Cryptospridium_meleagridis_GCA_001593445\") has been submitted\n",
      "Your job 5484382 (\"map_Cryptospridium_muris_GCF_000006515\") has been submitted\n",
      "Your job 5484383 (\"map_Cryptospridium_parvum_GCF_000165345\") has been submitted\n",
      "Your job 5484384 (\"map_Cryptospridium_ryanae_GCA_009792415\") has been submitted\n",
      "Your job 5484385 (\"map_Cryptospridium_sp_GCA_004936735\") has been submitted\n",
      "Your job 5484386 (\"map_Cryptospridium_tyzzeri_GCA_007210665\") has been submitted\n",
      "Your job 5484387 (\"map_Cryptospridium_ubiquitum_GCF_001865345\") has been submitted\n",
      "Your job 5484388 (\"map_Cryptospridium_viatorum_GCA_004337795\") has been submitted\n",
      "Your job 5484389 (\"map_Cyclospra_cayetanensis_GCF_002999335\") has been submitted\n",
      "Your job 5484390 (\"map_Cystoisospra_suis_GCA_002600585\") has been submitted\n",
      "Your job 5484391 (\"map_Eimeria_acervulina_GCF_000499425\") has been submitted\n",
      "Your job 5484392 (\"map_Eimeria_brunetti_GCA_000499725\") has been submitted\n",
      "Your job 5484393 (\"map_Eimeria_falciformis_GCA_002271815\") has been submitted\n",
      "Your job 5484394 (\"map_Eimeria_maxima_GCF_000499605\") has been submitted\n",
      "Your job 5484395 (\"map_Eimeria_mitis_GCA_000499745\") has been submitted\n",
      "Your job 5484396 (\"map_Eimeria_necatrix_GCF_000499385\") has been submitted\n",
      "Your job 5484397 (\"map_Eimeria_nieschulzi_GCA_000826945\") has been submitted\n",
      "Your job 5484398 (\"map_Eimeria_praecox_GCA_000499445\") has been submitted\n",
      "Your job 5484399 (\"map_Eimeria_tenella_GCF_000499545\") has been submitted\n",
      "Your job 5484400 (\"map_Gregarina_niphandrodes_GCF_000223845\") has been submitted\n",
      "Your job 5484401 (\"map_Haemoproteus_tartakovskyi_GCA_001625125\") has been submitted\n",
      "Your job 5484402 (\"map_Hammondia_hammondi_GCA_000447165\") has been submitted\n",
      "Your job 5484403 (\"map_Neospra_caninum_GCF_000208865\") has been submitted\n",
      "Your job 5484404 (\"map_Nephromyces_sp_GCA_004523865\") has been submitted\n",
      "Your job 5484405 (\"map_Plasmodium_berghei_GCA_900002375\") has been submitted\n",
      "Your job 5484406 (\"map_Plasmodium_brasilianum_GCA_001885115\") has been submitted\n",
      "Your job 5484407 (\"map_Plasmodium_chabaudi_GCA_900002335\") has been submitted\n",
      "Your job 5484408 (\"map_Plasmodium_coatneyi_GCF_001680005\") has been submitted\n",
      "Your job 5484409 (\"map_Plasmodium_cynomolgi_GCF_000321355\") has been submitted\n",
      "Your job 5484410 (\"map_Plasmodium_fragile_GCF_000956335\") has been submitted\n",
      "Your job 5484411 (\"map_Plasmodium_gaboni_GCF_001602025\") has been submitted\n",
      "Your job 5484412 (\"map_Plasmodium_gallinaceum_GCF_900005855\") has been submitted\n",
      "Your job 5484413 (\"map_Plasmodium_gonderi_GCF_002157705\") has been submitted\n",
      "Your job 5484414 (\"map_Plasmodium_inui_GCF_000524495\") has been submitted\n",
      "Your job 5484415 (\"map_Plasmodium_knowlesi_GCF_000006355\") has been submitted\n",
      "Your job 5484416 (\"map_Plasmodium_malariae_GCF_900090045\") has been submitted\n",
      "Your job 5484417 (\"map_Plasmodium_ovale_GCA_900090025\") has been submitted\n",
      "Your job 5484418 (\"map_Plasmodium_reichenowi_GCF_001601855\") has been submitted\n",
      "Your job 5484419 (\"map_Plasmodium_relictum_GCF_900005765\") has been submitted\n",
      "Your job 5484420 (\"map_Plasmodium_sp_GCF_900097015\") has been submitted\n",
      "Your job 5484421 (\"map_Plasmodium_vinckei_GCF_000709005\") has been submitted\n",
      "Your job 5484422 (\"map_Plasmodium_vivax_GCF_000002415\") has been submitted\n",
      "Your job 5484423 (\"map_Plasmodium_yoelii_GCA_900002385\") has been submitted\n",
      "Your job 5484424 (\"map_Sarcocystis_neurona_GCA_000727475\") has been submitted\n",
      "Your job 5484425 (\"map_Theileria_annulata_GCF_000003225\") has been submitted\n",
      "Your job 5484426 (\"map_Theileria_equi_GCF_000342415\") has been submitted\n",
      "Your job 5484427 (\"map_Theileria_orientalis_GCF_000740895\") has been submitted\n",
      "Your job 5484428 (\"map_Theileria_parva_GCF_000165365\") has been submitted\n",
      "Your job 5484429 (\"map_Toxoplasma_gondii_GCF_000006565\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484430 (\"merge_Ascogregarina_taiwanensis_GCA_000172235\") has been submitted\n",
      "Your job 5484431 (\"merge_Babesia_bigemina_GCF_000981445\") has been submitted\n",
      "Your job 5484432 (\"merge_Babesia_bovis_GCF_000165395\") has been submitted\n",
      "Your job 5484433 (\"merge_Babesia_divergens_GCA_001077455\") has been submitted\n",
      "Your job 5484434 (\"merge_Babesia_microti_GCF_000691945\") has been submitted\n",
      "Your job 5484435 (\"merge_Babesia_ovata_GCF_002897235\") has been submitted\n",
      "Your job 5484436 (\"merge_Babesia_sp_GCF_002095265\") has been submitted\n",
      "Your job 5484437 (\"merge_Besnoitia_besnoiti_GCF_002563875\") has been submitted\n",
      "Your job 5484438 (\"merge_Cryptospridium_andersoni_GCA_001865355\") has been submitted\n",
      "Your job 5484439 (\"merge_Cryptospridium_baileyi_GCA_001593455\") has been submitted\n",
      "Your job 5484440 (\"merge_Cryptospridium_bovis_GCA_009768925\") has been submitted\n",
      "Your job 5484441 (\"merge_Cryptospridium_cuniculus_GCA_004337835\") has been submitted\n",
      "Your job 5484442 (\"merge_Cryptospridium_hominis_GCA_000006425\") has been submitted\n",
      "Your job 5484443 (\"merge_Cryptospridium_meleagridis_GCA_001593445\") has been submitted\n",
      "Your job 5484444 (\"merge_Cryptospridium_muris_GCF_000006515\") has been submitted\n",
      "Your job 5484445 (\"merge_Cryptospridium_parvum_GCF_000165345\") has been submitted\n",
      "Your job 5484446 (\"merge_Cryptospridium_ryanae_GCA_009792415\") has been submitted\n",
      "Your job 5484447 (\"merge_Cryptospridium_sp_GCA_004936735\") has been submitted\n",
      "Your job 5484448 (\"merge_Cryptospridium_tyzzeri_GCA_007210665\") has been submitted\n",
      "Your job 5484449 (\"merge_Cryptospridium_ubiquitum_GCF_001865345\") has been submitted\n",
      "Your job 5484450 (\"merge_Cryptospridium_viatorum_GCA_004337795\") has been submitted\n",
      "Your job 5484451 (\"merge_Cyclospra_cayetanensis_GCF_002999335\") has been submitted\n",
      "Your job 5484452 (\"merge_Cystoisospra_suis_GCA_002600585\") has been submitted\n",
      "Your job 5484453 (\"merge_Eimeria_acervulina_GCF_000499425\") has been submitted\n",
      "Your job 5484454 (\"merge_Eimeria_brunetti_GCA_000499725\") has been submitted\n",
      "Your job 5484455 (\"merge_Eimeria_falciformis_GCA_002271815\") has been submitted\n",
      "Your job 5484456 (\"merge_Eimeria_maxima_GCF_000499605\") has been submitted\n",
      "Your job 5484457 (\"merge_Eimeria_mitis_GCA_000499745\") has been submitted\n",
      "Your job 5484458 (\"merge_Eimeria_necatrix_GCF_000499385\") has been submitted\n",
      "Your job 5484459 (\"merge_Eimeria_nieschulzi_GCA_000826945\") has been submitted\n",
      "Your job 5484460 (\"merge_Eimeria_praecox_GCA_000499445\") has been submitted\n",
      "Your job 5484461 (\"merge_Eimeria_tenella_GCF_000499545\") has been submitted\n",
      "Your job 5484462 (\"merge_Gregarina_niphandrodes_GCF_000223845\") has been submitted\n",
      "Your job 5484463 (\"merge_Haemoproteus_tartakovskyi_GCA_001625125\") has been submitted\n",
      "Your job 5484464 (\"merge_Hammondia_hammondi_GCA_000447165\") has been submitted\n",
      "Your job 5484465 (\"merge_Neospra_caninum_GCF_000208865\") has been submitted\n",
      "Your job 5484466 (\"merge_Nephromyces_sp_GCA_004523865\") has been submitted\n",
      "Your job 5484467 (\"merge_Plasmodium_berghei_GCA_900002375\") has been submitted\n",
      "Your job 5484468 (\"merge_Plasmodium_brasilianum_GCA_001885115\") has been submitted\n",
      "Your job 5484469 (\"merge_Plasmodium_chabaudi_GCA_900002335\") has been submitted\n",
      "Your job 5484470 (\"merge_Plasmodium_coatneyi_GCF_001680005\") has been submitted\n",
      "Your job 5484471 (\"merge_Plasmodium_cynomolgi_GCF_000321355\") has been submitted\n",
      "Your job 5484472 (\"merge_Plasmodium_fragile_GCF_000956335\") has been submitted\n",
      "Your job 5484473 (\"merge_Plasmodium_gaboni_GCF_001602025\") has been submitted\n",
      "Your job 5484474 (\"merge_Plasmodium_gallinaceum_GCF_900005855\") has been submitted\n",
      "Your job 5484475 (\"merge_Plasmodium_gonderi_GCF_002157705\") has been submitted\n",
      "Your job 5484476 (\"merge_Plasmodium_inui_GCF_000524495\") has been submitted\n",
      "Your job 5484477 (\"merge_Plasmodium_knowlesi_GCF_000006355\") has been submitted\n",
      "Your job 5484478 (\"merge_Plasmodium_malariae_GCF_900090045\") has been submitted\n",
      "Your job 5484479 (\"merge_Plasmodium_ovale_GCA_900090025\") has been submitted\n",
      "Your job 5484480 (\"merge_Plasmodium_reichenowi_GCF_001601855\") has been submitted\n",
      "Your job 5484481 (\"merge_Plasmodium_relictum_GCF_900005765\") has been submitted\n",
      "Your job 5484482 (\"merge_Plasmodium_sp_GCF_900097015\") has been submitted\n",
      "Your job 5484483 (\"merge_Plasmodium_vinckei_GCF_000709005\") has been submitted\n",
      "Your job 5484484 (\"merge_Plasmodium_vivax_GCF_000002415\") has been submitted\n",
      "Your job 5484485 (\"merge_Plasmodium_yoelii_GCA_900002385\") has been submitted\n",
      "Your job 5484486 (\"merge_Sarcocystis_neurona_GCA_000727475\") has been submitted\n",
      "Your job 5484487 (\"merge_Theileria_annulata_GCF_000003225\") has been submitted\n",
      "Your job 5484488 (\"merge_Theileria_equi_GCF_000342415\") has been submitted\n",
      "Your job 5484489 (\"merge_Theileria_orientalis_GCF_000740895\") has been submitted\n",
      "Your job 5484490 (\"merge_Theileria_parva_GCF_000165365\") has been submitted\n",
      "Your job 5484491 (\"merge_Toxoplasma_gondii_GCF_000006565\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 15 sequences from Ascogregarina_taiwanensis_GCA_000172235_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 23 sequences from Babesia_bigemina_GCF_000981445_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 28 sequences from Babesia_bovis_GCF_000165395_merged.bed.  Filtered 24 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 21 sequences from Babesia_divergens_GCA_001077455_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 14 sequences from Babesia_microti_GCF_000691945_merged.bed.  Filtered 11 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 19 sequences from Babesia_ovata_GCF_002897235_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 31 sequences from Babesia_sp_GCF_002095265_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 141 sequences from Besnoitia_besnoiti_GCF_002563875_merged.bed.  Filtered 137 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 31 sequences from Cryptospridium_andersoni_GCA_001865355_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 147 sequences from Cryptospridium_baileyi_GCA_001593455_merged.bed.  Filtered 141 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 46 sequences from Cryptospridium_bovis_GCA_009768925_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 28 sequences from Cryptospridium_cuniculus_GCA_004337835_merged.bed.  Filtered 24 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 50 sequences from Cryptospridium_hominis_GCA_000006425_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 25 sequences from Cryptospridium_meleagridis_GCA_001593445_merged.bed.  Filtered 22 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 63 sequences from Cryptospridium_muris_GCF_000006515_merged.bed.  Filtered 59 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 38 sequences from Cryptospridium_parvum_GCF_000165345_merged.bed.  Filtered 35 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 29 sequences from Cryptospridium_ryanae_GCA_009792415_merged.bed.  Filtered 24 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 36 sequences from Cryptospridium_sp_GCA_004936735_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 34 sequences from Cryptospridium_tyzzeri_GCA_007210665_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 46 sequences from Cryptospridium_ubiquitum_GCF_001865345_merged.bed.  Filtered 42 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 25 sequences from Cryptospridium_viatorum_GCA_004337795_merged.bed.  Filtered 22 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 78 sequences from Cyclospra_cayetanensis_GCF_002999335_merged.bed.  Filtered 75 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 52 sequences from Cystoisospra_suis_GCA_002600585_merged.bed.  Filtered 48 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 332 sequences from Eimeria_acervulina_GCF_000499425_merged.bed.  Filtered 331 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1.\n",
      "Screened 544 sequences from Eimeria_brunetti_GCA_000499725_merged.bed.  Filtered 542 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 122 sequences from Eimeria_falciformis_GCA_002271815_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 222 sequences from Eimeria_maxima_GCF_000499605_merged.bed.  Filtered 218 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 515 sequences from Eimeria_mitis_GCA_000499745_merged.bed.  Filtered 513 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 339 sequences from Eimeria_necatrix_GCF_000499385_merged.bed.  Filtered 338 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1.\n",
      "Screened 140 sequences from Eimeria_nieschulzi_GCA_000826945_merged.bed.  Filtered 136 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 513 sequences from Eimeria_praecox_GCA_000499445_merged.bed.  Filtered 508 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 191 sequences from Eimeria_tenella_GCF_000499545_merged.bed.  Filtered 187 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 8 sequences from Gregarina_niphandrodes_GCF_000223845_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 995 sequences from Haemoproteus_tartakovskyi_GCA_001625125_merged.bed.  Filtered 925 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 70.\n",
      "Screened 14 sequences from Hammondia_hammondi_GCA_000447165_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 141 sequences from Neospra_caninum_GCF_000208865_merged.bed.  Filtered 140 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1.\n",
      "Screened 1174 sequences from Nephromyces_sp_GCA_004523865_merged.bed.  Filtered 1165 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 2763 sequences from Plasmodium_berghei_GCA_900002375_merged.bed.  Filtered 2409 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 354.\n",
      "Screened 3403 sequences from Plasmodium_brasilianum_GCA_001885115_merged.bed.  Filtered 2989 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 414.\n",
      "Screened 2826 sequences from Plasmodium_chabaudi_GCA_900002335_merged.bed.  Filtered 2439 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 387.\n",
      "Screened 338 sequences from Plasmodium_coatneyi_GCF_001680005_merged.bed.  Filtered 294 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 44.\n",
      "Screened 342 sequences from Plasmodium_cynomolgi_GCF_000321355_merged.bed.  Filtered 300 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 42.\n",
      "Screened 448 sequences from Plasmodium_fragile_GCF_000956335_merged.bed.  Filtered 408 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 40.\n",
      "Screened 38081 sequences from Plasmodium_gaboni_GCF_001602025_merged.bed.  Filtered 36812 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1269.\n",
      "Screened 4385 sequences from Plasmodium_gallinaceum_GCF_900005855_merged.bed.  Filtered 3833 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 552.\n",
      "Screened 2137 sequences from Plasmodium_gonderi_GCF_002157705_merged.bed.  Filtered 1897 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 240.\n",
      "Screened 289 sequences from Plasmodium_inui_GCF_000524495_merged.bed.  Filtered 260 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 29.\n",
      "Screened 472 sequences from Plasmodium_knowlesi_GCF_000006355_merged.bed.  Filtered 431 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 41.\n",
      "Screened 3558 sequences from Plasmodium_malariae_GCF_900090045_merged.bed.  Filtered 3152 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 406.\n",
      "Screened 2147 sequences from Plasmodium_ovale_GCA_900090025_merged.bed.  Filtered 1883 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 264.\n",
      "Screened 17153 sequences from Plasmodium_reichenowi_GCF_001601855_merged.bed.  Filtered 16925 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 228.\n",
      "Screened 4180 sequences from Plasmodium_relictum_GCF_900005765_merged.bed.  Filtered 3686 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 494.\n",
      "Screened 39816 sequences from Plasmodium_sp_GCF_900097015_merged.bed.  Filtered 38463 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1353.\n",
      "Screened 2691 sequences from Plasmodium_vinckei_GCF_000709005_merged.bed.  Filtered 2297 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 394.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 314 sequences from Plasmodium_vivax_GCF_000002415_merged.bed.  Filtered 271 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 3160 sequences from Plasmodium_yoelii_GCA_900002385_merged.bed.  Filtered 2738 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 422.\n",
      "Screened 62 sequences from Sarcocystis_neurona_GCA_000727475_merged.bed.  Filtered 60 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 41 sequences from Theileria_annulata_GCF_000003225_merged.bed.  Filtered 37 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 30 sequences from Theileria_equi_GCF_000342415_merged.bed.  Filtered 26 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 36 sequences from Theileria_orientalis_GCF_000740895_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 25 sequences from Theileria_parva_GCF_000165365_merged.bed.  Filtered 21 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 142 sequences from Toxoplasma_gondii_GCF_000006565_merged.bed.  Filtered 138 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/apicomplexa/initial_intervals/Toxoplasma_gondii_GCF_000006565_merged.bed --twobit results/phyluce/apicomplexa/cleaned_genomes/Plasmodium_falciparum_GCA_000002765_formatted.2bit --output results/phyluce/apicomplexa/initial_intervals/Toxoplasma_gondii_GCF_000006565_stripped.bed;\n",
      "ascogregarina_taiwanensis_gca_000172235.\n",
      "babesia_bigemina_gcf_000981445.\n",
      "babesia_bovis_gcf_000165395.\n",
      "babesia_divergens_gca_001077455.\n",
      "babesia_microti_gcf_000691945.\n",
      "babesia_ovata_gcf_002897235.\n",
      "babesia_sp_gcf_002095265.\n",
      "besnoitia_besnoiti_gcf_002563875.\n",
      "cryptospridium_andersoni_gca_001865355.\n",
      "cryptospridium_baileyi_gca_001593455.\n",
      "cryptospridium_bovis_gca_009768925.\n",
      "cryptospridium_cuniculus_gca_004337835.\n",
      "cryptospridium_hominis_gca_000006425.\n",
      "cryptospridium_meleagridis_gca_001593445.\n",
      "cryptospridium_muris_gcf_000006515.\n",
      "cryptospridium_parvum_gcf_000165345.\n",
      "cryptospridium_ryanae_gca_009792415.\n",
      "cryptospridium_sp_gca_004936735.\n",
      "cryptospridium_tyzzeri_gca_007210665.\n",
      "cryptospridium_ubiquitum_gcf_001865345.\n",
      "cryptospridium_viatorum_gca_004337795.\n",
      "cyclospra_cayetanensis_gcf_002999335.\n",
      "cystoisospra_suis_gca_002600585.\n",
      "eimeria_acervulina_gcf_000499425.\n",
      "eimeria_brunetti_gca_000499725.\n",
      "eimeria_falciformis_gca_002271815.\n",
      "eimeria_maxima_gcf_000499605.\n",
      "eimeria_mitis_gca_000499745.\n",
      "eimeria_necatrix_gcf_000499385.\n",
      "eimeria_nieschulzi_gca_000826945.\n",
      "eimeria_praecox_gca_000499445.\n",
      "eimeria_tenella_gcf_000499545.\n",
      "gregarina_niphandrodes_gcf_000223845.\n",
      "haemoproteus_tartakovskyi_gca_001625125.\n",
      "hammondia_hammondi_gca_000447165.\n",
      "neospra_caninum_gcf_000208865.\n",
      "nephromyces_sp_gca_004523865.\n",
      "plasmodium_berghei_gca_900002375.\n",
      "plasmodium_brasilianum_gca_001885115.\n",
      "plasmodium_chabaudi_gca_900002335.\n",
      "plasmodium_coatneyi_gcf_001680005.\n",
      "plasmodium_cynomolgi_gcf_000321355.\n",
      "plasmodium_fragile_gcf_000956335.\n",
      "plasmodium_gaboni_gcf_001602025..\n",
      "plasmodium_gallinaceum_gcf_900005855.\n",
      "plasmodium_gonderi_gcf_002157705.\n",
      "plasmodium_inui_gcf_000524495.\n",
      "plasmodium_knowlesi_gcf_000006355.\n",
      "plasmodium_malariae_gcf_900090045.\n",
      "plasmodium_ovale_gca_900090025.\n",
      "plasmodium_reichenowi_gcf_001601855.\n",
      "plasmodium_relictum_gcf_900005765.\n",
      "plasmodium_sp_gcf_900097015..\n",
      "plasmodium_vinckei_gcf_000709005.\n",
      "plasmodium_vivax_gcf_000002415.\n",
      "plasmodium_yoelii_gca_900002385.\n",
      "sarcocystis_neurona_gca_000727475.\n",
      "theileria_annulata_gcf_000003225.\n",
      "theileria_equi_gcf_000342415.\n",
      "theileria_orientalis_gcf_000740895.\n",
      "theileria_parva_gcf_000165365.\n",
      "toxoplasma_gondii_gcf_000006565.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/apicomplexa/initial_intervals/apicomplexa-to-Plasmodium_falciparum_GCA_000002765.sqlite --base-taxon Plasmodium_falciparum_GCA_000002765\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 0 taxa:\t2,585.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 1 taxa:\t2,585.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 2 taxa:\t1,695.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 3 taxa:\t836.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 4 taxa:\t604.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 5 taxa:\t461.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 6 taxa:\t356.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 7 taxa:\t278.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 8 taxa:\t210.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 9 taxa:\t166.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 10 taxa:\t135.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 11 taxa:\t92.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 12 taxa:\t60.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 13 taxa:\t45.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 14 taxa:\t33.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 15 taxa:\t23.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 16 taxa:\t18.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 17 taxa:\t14.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 18 taxa:\t12.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 19 taxa:\t7.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 20 taxa:\t7.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 21 taxa:\t6.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 22 taxa:\t5.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 23 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 24 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 25 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 26 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 27 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 28 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 29 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 30 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 31 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 32 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 33 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 34 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 35 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 36 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 37 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 38 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 39 taxa:\t4.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 40 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 41 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 42 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 43 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 44 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 45 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 46 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 47 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 48 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 49 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 50 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 51 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 52 taxa:\t3.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 53 taxa:\t2.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 54 taxa:\t1.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 55 taxa:\t1.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 56 taxa:\t1.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 57 taxa:\t1.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 58 taxa:\t0.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 59 taxa:\t0.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 60 taxa:\t0.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 61 taxa:\t0.0\n",
      "Loci shared by Plasmodium_falciparum_GCA_000002765 + 62 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/apicomplexa/initial_intervals/apicomplexa-to-Plasmodium_falciparum_GCA_000002765.sqlite --base-taxon Plasmodium_falciparum_GCA_000002765 --output results/phyluce/apicomplexa/initial_intervals/Plasmodium_falciparum_GCA_000002765_+2.bed --specific-counts 2;\n",
      "Counter({'plasmodium_sp_gcf_900097015': 1239, 'plasmodium_gaboni_gcf_001602025': 1150, 'plasmodium_gallinaceum_gcf_900005855': 490, 'plasmodium_relictum_gcf_900005765': 471, 'plasmodium_yoelii_gca_900002385': 428, 'plasmodium_malariae_gcf_900090045': 414, 'plasmodium_vinckei_gcf_000709005': 409, 'plasmodium_brasilianum_gca_001885115': 404, 'plasmodium_chabaudi_gca_900002335': 387, 'plasmodium_berghei_gca_900002375': 341, 'plasmodium_ovale_gca_900090025': 257, 'plasmodium_gonderi_gcf_002157705': 238, 'plasmodium_reichenowi_gcf_001601855': 180, 'haemoproteus_tartakovskyi_gca_001625125': 66, 'plasmodium_coatneyi_gcf_001680005': 47, 'plasmodium_vivax_gcf_000002415': 45, 'plasmodium_cynomolgi_gcf_000321355': 44, 'plasmodium_fragile_gcf_000956335': 43, 'plasmodium_knowlesi_gcf_000006355': 43, 'plasmodium_inui_gcf_000524495': 31, 'nephromyces_sp_gca_004523865': 7, 'cryptospridium_baileyi_gca_001593455': 6, 'theileria_parva_gcf_000165365': 5, 'babesia_bigemina_gcf_000981445': 5, 'toxoplasma_gondii_gcf_000006565': 5, 'theileria_equi_gcf_000342415': 5, 'cryptospridium_muris_gcf_000006515': 4, 'babesia_ovata_gcf_002897235': 4, 'babesia_bovis_gcf_000165395': 4, 'hammondia_hammondi_gca_000447165': 4, 'cryptospridium_ubiquitum_gcf_001865345': 4, 'theileria_orientalis_gcf_000740895': 4, 'cryptospridium_sp_gca_004936735': 4, 'theileria_annulata_gcf_000003225': 4, 'cystoisospra_suis_gca_002600585': 4, 'eimeria_maxima_gcf_000499605': 4, 'eimeria_nieschulzi_gca_000826945': 4, 'eimeria_tenella_gcf_000499545': 4, 'cryptospridium_cuniculus_gca_004337835': 4, 'cryptospridium_hominis_gca_000006425': 4, 'cryptospridium_ryanae_gca_009792415': 4, 'besnoitia_besnoiti_gcf_002563875': 4, 'cryptospridium_meleagridis_gca_001593445': 3, 'cryptospridium_viatorum_gca_004337795': 3, 'eimeria_praecox_gca_000499445': 3, 'cryptospridium_andersoni_gca_001865355': 3, 'babesia_microti_gcf_000691945': 3, 'cyclospra_cayetanensis_gcf_002999335': 3, 'cryptospridium_bovis_gca_009768925': 3, 'cryptospridium_parvum_gcf_000165345': 3, 'babesia_sp_gcf_002095265': 3, 'babesia_divergens_gca_001077455': 3, 'ascogregarina_taiwanensis_gca_000172235': 2, 'cryptospridium_tyzzeri_gca_007210665': 2, 'gregarina_niphandrodes_gcf_000223845': 2, 'eimeria_mitis_gca_000499745': 2, 'sarcocystis_neurona_gca_000727475': 2, 'eimeria_falciformis_gca_002271815': 2, 'eimeria_brunetti_gca_000499725': 2, 'neospra_caninum_gcf_000208865': 1, 'eimeria_necatrix_gcf_000499385': 1, 'eimeria_acervulina_gcf_000499425': 1})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 2\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/apicomplexa/initial_intervals/Plasmodium_falciparum_GCA_000002765_+2.bed --twobit results/phyluce/apicomplexa/cleaned_genomes/Plasmodium_falciparum_GCA_000002765_formatted.2bit --buffer-to 160 --output results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2.fasta;\n",
      "Screened 1695 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 1695.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2.fasta --probe-prefix uce_apicomplexa_ --design apicomplexa_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 938\n",
      "Probe Count = 1484\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas --query results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes_vself.lastz;\n",
      "Started:  Mon Feb 10, 2020  08:31:36\n",
      "Ended:  Mon Feb 10, 2020  08:31:39\n",
      "Time for execution:  0.0589553833008 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas                        --lastz results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes_vself.lastz                       --probe-prefix=uce_apicomplexa_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 1483 fasta sequences.  Filtered 117 duplicates. Kept 1277.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/apicomplexa/validate_intervals/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas --scaffoldlist Ascogregarina_taiwanensis_GCA_000172235 Babesia_bigemina_GCF_000981445 Babesia_bovis_GCF_000165395 Babesia_divergens_GCA_001077455 Babesia_microti_GCF_000691945 Babesia_ovata_GCF_002897235 Babesia_sp_GCF_002095265 Besnoitia_besnoiti_GCF_002563875 Cryptospridium_andersoni_GCA_001865355 Cryptospridium_baileyi_GCA_001593455 Cryptospridium_bovis_GCA_009768925 Cryptospridium_cuniculus_GCA_004337835 Cryptospridium_hominis_GCA_000006425 Cryptospridium_meleagridis_GCA_001593445 Cryptospridium_muris_GCF_000006515 Cryptospridium_parvum_GCF_000165345 Cryptospridium_ryanae_GCA_009792415 Cryptospridium_sp_GCA_004936735 Cryptospridium_tyzzeri_GCA_007210665 Cryptospridium_ubiquitum_GCF_001865345 Cryptospridium_viatorum_GCA_004337795 Cyclospra_cayetanensis_GCF_002999335 Cystoisospra_suis_GCA_002600585 Eimeria_acervulina_GCF_000499425 Eimeria_brunetti_GCA_000499725 Eimeria_falciformis_GCA_002271815 Eimeria_maxima_GCF_000499605 Eimeria_mitis_GCA_000499745 Eimeria_necatrix_GCF_000499385 Eimeria_nieschulzi_GCA_000826945 Eimeria_praecox_GCA_000499445 Eimeria_tenella_GCF_000499545 Gregarina_niphandrodes_GCF_000223845 Haemoproteus_tartakovskyi_GCA_001625125 Hammondia_hammondi_GCA_000447165 Neospra_caninum_GCF_000208865 Nephromyces_sp_GCA_004523865 Plasmodium_berghei_GCA_900002375 Plasmodium_brasilianum_GCA_001885115 Plasmodium_chabaudi_GCA_900002335 Plasmodium_coatneyi_GCF_001680005 Plasmodium_cynomolgi_GCF_000321355 Plasmodium_fragile_GCF_000956335 Plasmodium_gaboni_GCF_001602025 Plasmodium_gallinaceum_GCF_900005855 Plasmodium_gonderi_GCF_002157705 Plasmodium_inui_GCF_000524495 Plasmodium_knowlesi_GCF_000006355 Plasmodium_malariae_GCF_900090045 Plasmodium_ovale_GCA_900090025 Plasmodium_reichenowi_GCF_001601855 Plasmodium_relictum_GCF_900005765 Plasmodium_sp_GCF_900097015 Plasmodium_vinckei_GCF_000709005 Plasmodium_vivax_GCF_000002415 Plasmodium_yoelii_GCA_900002385 Sarcocystis_neurona_GCA_000727475 Theileria_annulata_GCF_000003225 Theileria_equi_GCF_000342415 Theileria_orientalis_GCF_000740895 Theileria_parva_GCF_000165365 Toxoplasma_gondii_GCF_000006565 Plasmodium_falciparum_GCA_000002765 --genome-base-path results/phyluce/apicomplexa/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/apicomplexa/validate_intervals/apicomplexa-to-Plasmodium_falciparum_GCA_000002765.sqlite --output results/phyluce/apicomplexa/validate_intervals/lastz/;\n",
      "\n",
      "Running against Ascogregarina_taiwanensis_GCA_000172235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjvxs2t.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl4RAf5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Ascogregarina_taiwanensis_GCA_000172235.lastz\n",
      "Creating Ascogregarina_taiwanensis_GCA_000172235 table\n",
      "Inserting data to Ascogregarina_taiwanensis_GCA_000172235 table\n",
      "\n",
      "Running against Babesia_bigemina_GCF_000981445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpcFAQv6.fasta\n",
      "\t/tmp/tmpdI7mt2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptxJmMN.lastz\n",
      "\t/tmp/tmpZaOHPL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Babesia_bigemina_GCF_000981445.lastz\n",
      "Creating Babesia_bigemina_GCF_000981445 table\n",
      "Inserting data to Babesia_bigemina_GCF_000981445 table\n",
      "\n",
      "Running against Babesia_bovis_GCF_000165395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpO0V4m2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyG_F8Z.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Babesia_bovis_GCF_000165395.lastz\n",
      "Creating Babesia_bovis_GCF_000165395 table\n",
      "Inserting data to Babesia_bovis_GCF_000165395 table\n",
      "\n",
      "Running against Babesia_divergens_GCA_001077455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpR85Egc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2_mQtX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Babesia_divergens_GCA_001077455.lastz\n",
      "Creating Babesia_divergens_GCA_001077455 table\n",
      "Inserting data to Babesia_divergens_GCA_001077455 table\n",
      "\n",
      "Running against Babesia_microti_GCF_000691945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1RlGM6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpieGWWZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Babesia_microti_GCF_000691945.lastz\n",
      "Creating Babesia_microti_GCF_000691945 table\n",
      "Inserting data to Babesia_microti_GCF_000691945 table\n",
      "\n",
      "Running against Babesia_ovata_GCF_002897235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmphLEZeV.fasta\n",
      "\t/tmp/tmplzAPYS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHEJJB5.lastz\n",
      "\t/tmp/tmpqlPK0g.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Babesia_ovata_GCF_002897235.lastz\n",
      "Creating Babesia_ovata_GCF_002897235 table\n",
      "Inserting data to Babesia_ovata_GCF_002897235 table\n",
      "\n",
      "Running against Babesia_sp_GCF_002095265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp47TaM2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvvQjSR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Babesia_sp_GCF_002095265.lastz\n",
      "Creating Babesia_sp_GCF_002095265 table\n",
      "Inserting data to Babesia_sp_GCF_002095265 table\n",
      "\n",
      "Running against Besnoitia_besnoiti_GCF_002563875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpJapiSD.fasta\n",
      "\t/tmp/tmpZYEdeN.fasta\n",
      "\t/tmp/tmpbP535x.fasta\n",
      "\t/tmp/tmpYJXpu5.fasta\n",
      "\t/tmp/tmpDVAPu8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQMFdoa.lastz\n",
      "\t/tmp/tmpL9UYQX.lastz\n",
      "\t/tmp/tmpVrH3Hu.lastz\n",
      "\t/tmp/tmp3EHGWs.lastz\n",
      "\t/tmp/tmp1pk2lp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Besnoitia_besnoiti_GCF_002563875.lastz\n",
      "Creating Besnoitia_besnoiti_GCF_002563875 table\n",
      "Inserting data to Besnoitia_besnoiti_GCF_002563875 table\n",
      "\n",
      "Running against Cryptospridium_andersoni_GCA_001865355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5ns1gm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSRS8qx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_andersoni_GCA_001865355.lastz\n",
      "Creating Cryptospridium_andersoni_GCA_001865355 table\n",
      "Inserting data to Cryptospridium_andersoni_GCA_001865355 table\n",
      "\n",
      "Running against Cryptospridium_baileyi_GCA_001593455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdX6Qso.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCcyvWZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_baileyi_GCA_001593455.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Cryptospridium_baileyi_GCA_001593455 table\n",
      "Inserting data to Cryptospridium_baileyi_GCA_001593455 table\n",
      "\n",
      "Running against Cryptospridium_bovis_GCA_009768925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpl4qUO5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4koWzP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_bovis_GCA_009768925.lastz\n",
      "Creating Cryptospridium_bovis_GCA_009768925 table\n",
      "Inserting data to Cryptospridium_bovis_GCA_009768925 table\n",
      "\n",
      "Running against Cryptospridium_cuniculus_GCA_004337835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_zQ2Ke.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXY1DDd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_cuniculus_GCA_004337835.lastz\n",
      "Creating Cryptospridium_cuniculus_GCA_004337835 table\n",
      "Inserting data to Cryptospridium_cuniculus_GCA_004337835 table\n",
      "\n",
      "Running against Cryptospridium_hominis_GCA_000006425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplg3otK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwRr0_Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_hominis_GCA_000006425.lastz\n",
      "Creating Cryptospridium_hominis_GCA_000006425 table\n",
      "Inserting data to Cryptospridium_hominis_GCA_000006425 table\n",
      "\n",
      "Running against Cryptospridium_meleagridis_GCA_001593445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpT5QcEE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWZBCBB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_meleagridis_GCA_001593445.lastz\n",
      "Creating Cryptospridium_meleagridis_GCA_001593445 table\n",
      "Inserting data to Cryptospridium_meleagridis_GCA_001593445 table\n",
      "\n",
      "Running against Cryptospridium_muris_GCF_000006515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpl4Tgqc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRBIX85.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_muris_GCF_000006515.lastz\n",
      "Creating Cryptospridium_muris_GCF_000006515 table\n",
      "Inserting data to Cryptospridium_muris_GCF_000006515 table\n",
      "\n",
      "Running against Cryptospridium_parvum_GCF_000165345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyAL__P.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6dpBJQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_parvum_GCF_000165345.lastz\n",
      "Creating Cryptospridium_parvum_GCF_000165345 table\n",
      "Inserting data to Cryptospridium_parvum_GCF_000165345 table\n",
      "\n",
      "Running against Cryptospridium_ryanae_GCA_009792415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLnXuis.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpk78n7_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_ryanae_GCA_009792415.lastz\n",
      "Creating Cryptospridium_ryanae_GCA_009792415 table\n",
      "Inserting data to Cryptospridium_ryanae_GCA_009792415 table\n",
      "\n",
      "Running against Cryptospridium_sp_GCA_004936735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxsPhYc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnzFwg_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_sp_GCA_004936735.lastz\n",
      "Creating Cryptospridium_sp_GCA_004936735 table\n",
      "Inserting data to Cryptospridium_sp_GCA_004936735 table\n",
      "\n",
      "Running against Cryptospridium_tyzzeri_GCA_007210665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp02uD9N.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNOg3bz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_tyzzeri_GCA_007210665.lastz\n",
      "Creating Cryptospridium_tyzzeri_GCA_007210665 table\n",
      "Inserting data to Cryptospridium_tyzzeri_GCA_007210665 table\n",
      "\n",
      "Running against Cryptospridium_ubiquitum_GCF_001865345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJqgOdm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBqnhN2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_ubiquitum_GCF_001865345.lastz\n",
      "Creating Cryptospridium_ubiquitum_GCF_001865345 table\n",
      "Inserting data to Cryptospridium_ubiquitum_GCF_001865345 table\n",
      "\n",
      "Running against Cryptospridium_viatorum_GCA_004337795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPUFmjI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa5zFbX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cryptospridium_viatorum_GCA_004337795.lastz\n",
      "Creating Cryptospridium_viatorum_GCA_004337795 table\n",
      "Inserting data to Cryptospridium_viatorum_GCA_004337795 table\n",
      "\n",
      "Running against Cyclospra_cayetanensis_GCF_002999335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpsHmRHF.fasta\n",
      "\t/tmp/tmppJnYQo.fasta\n",
      "\t/tmp/tmp37mEI_.fasta\n",
      "\t/tmp/tmpOD8OYY.fasta\n",
      "\t/tmp/tmp0v7bZt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFnc8qF.lastz\n",
      "\t/tmp/tmpNcVdS9.lastz\n",
      "\t/tmp/tmpKTtqS4.lastz\n",
      "\t/tmp/tmpdogoym.lastz\n",
      "\t/tmp/tmpMyToi_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cyclospra_cayetanensis_GCF_002999335.lastz\n",
      "Creating Cyclospra_cayetanensis_GCF_002999335 table\n",
      "Inserting data to Cyclospra_cayetanensis_GCF_002999335 table\n",
      "\n",
      "Running against Cystoisospra_suis_GCA_002600585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpilaoo1.fasta\n",
      "\t/tmp/tmpXdKf2y.fasta\n",
      "\t/tmp/tmp6h2ZTv.fasta\n",
      "\t/tmp/tmpOpVrTR.fasta\n",
      "\t/tmp/tmp_WmqDS.fasta\n",
      "\t/tmp/tmpA6b9iE.fasta\n",
      "\t/tmp/tmpX6fwuR.fasta\n",
      "\t/tmp/tmpi5O2cY.fasta\n",
      "\t/tmp/tmpfXfnVZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMm1887.lastz\n",
      "\t/tmp/tmp_iseXk.lastz\n",
      "\t/tmp/tmpp6BHIl.lastz\n",
      "\t/tmp/tmpvq6IfZ.lastz\n",
      "\t/tmp/tmpEA7TA1.lastz\n",
      "\t/tmp/tmp61ZQwn.lastz\n",
      "\t/tmp/tmpMAUOBX.lastz\n",
      "\t/tmp/tmpvAK7VG.lastz\n",
      "\t/tmp/tmpf2La_w.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Cystoisospra_suis_GCA_002600585.lastz\n",
      "Creating Cystoisospra_suis_GCA_002600585 table\n",
      "Inserting data to Cystoisospra_suis_GCA_002600585 table\n",
      "\n",
      "Running against Eimeria_acervulina_GCF_000499425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpywt0HR.fasta\n",
      "\t/tmp/tmpP_basr.fasta\n",
      "\t/tmp/tmpLIKqXn.fasta\n",
      "\t/tmp/tmpBMykat.fasta\n",
      "\t/tmp/tmpHwjqO5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCCxX_q.lastz\n",
      "\t/tmp/tmpVE_Hw7.lastz\n",
      "\t/tmp/tmpD7Pb_T.lastz\n",
      "\t/tmp/tmpwEmPWf.lastz\n",
      "\t/tmp/tmpXF6pYE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_acervulina_GCF_000499425.lastz\n",
      "Creating Eimeria_acervulina_GCF_000499425 table\n",
      "Inserting data to Eimeria_acervulina_GCF_000499425 table\n",
      "\n",
      "Running against Eimeria_brunetti_GCA_000499725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpMx9z_E.fasta\n",
      "\t/tmp/tmpnSTqGP.fasta\n",
      "\t/tmp/tmpIp0k9_.fasta\n",
      "\t/tmp/tmp3XXICS.fasta\n",
      "\t/tmp/tmpsTNCxJ.fasta\n",
      "\t/tmp/tmptxCEEI.fasta\n",
      "\t/tmp/tmplBBZKT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphg_SYO.lastz\n",
      "\t/tmp/tmpyEA8sH.lastz\n",
      "\t/tmp/tmp9x0TYe.lastz\n",
      "\t/tmp/tmp4AuSbM.lastz\n",
      "\t/tmp/tmpDY0W1g.lastz\n",
      "\t/tmp/tmp7ueFxO.lastz\n",
      "\t/tmp/tmpE9hFs4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_brunetti_GCA_000499725.lastz\n",
      "Creating Eimeria_brunetti_GCA_000499725 table\n",
      "Inserting data to Eimeria_brunetti_GCA_000499725 table\n",
      "\n",
      "Running against Eimeria_falciformis_GCA_002271815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpWco6is.fasta\n",
      "\t/tmp/tmpABNKtR.fasta\n",
      "\t/tmp/tmp7f3Jdx.fasta\n",
      "\t/tmp/tmp7l2ieb.fasta\n",
      "\t/tmp/tmpHxAZmF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEEzYoa.lastz\n",
      "\t/tmp/tmpPOAYuG.lastz\n",
      "\t/tmp/tmpf_QN5U.lastz\n",
      "\t/tmp/tmpUnJt4s.lastz\n",
      "\t/tmp/tmpwql0eb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_falciformis_GCA_002271815.lastz\n",
      "Creating Eimeria_falciformis_GCA_002271815 table\n",
      "Inserting data to Eimeria_falciformis_GCA_002271815 table\n",
      "\n",
      "Running against Eimeria_maxima_GCF_000499605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmppBrmzK.fasta\n",
      "\t/tmp/tmphC8FdA.fasta\n",
      "\t/tmp/tmp4MM4PX.fasta\n",
      "\t/tmp/tmp6rsyrM.fasta\n",
      "\t/tmp/tmprlwbH0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNC7ljW.lastz\n",
      "\t/tmp/tmpQizS_X.lastz\n",
      "\t/tmp/tmpQ8A5ru.lastz\n",
      "\t/tmp/tmpcdug5a.lastz\n",
      "\t/tmp/tmp296L0Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_maxima_GCF_000499605.lastz\n",
      "Creating Eimeria_maxima_GCF_000499605 table\n",
      "Inserting data to Eimeria_maxima_GCF_000499605 table\n",
      "\n",
      "Running against Eimeria_mitis_GCA_000499745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmpOLav20.fasta\n",
      "\t/tmp/tmptKr5Lz.fasta\n",
      "\t/tmp/tmpmhlJhY.fasta\n",
      "\t/tmp/tmplZU1BO.fasta\n",
      "\t/tmp/tmpIENWsB.fasta\n",
      "\t/tmp/tmpO_C0Nt.fasta\n",
      "\t/tmp/tmp5CXmxA.fasta\n",
      "\t/tmp/tmpFwy8in.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfApF2D.lastz\n",
      "\t/tmp/tmpxwIsXX.lastz\n",
      "\t/tmp/tmpAokMAh.lastz\n",
      "\t/tmp/tmpWGjEfX.lastz\n",
      "\t/tmp/tmpWazsc8.lastz\n",
      "\t/tmp/tmpuKx_lv.lastz\n",
      "\t/tmp/tmpvSFCxM.lastz\n",
      "\t/tmp/tmplHOqK9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_mitis_GCA_000499745.lastz\n",
      "Creating Eimeria_mitis_GCA_000499745 table\n",
      "Inserting data to Eimeria_mitis_GCA_000499745 table\n",
      "\n",
      "Running against Eimeria_necatrix_GCF_000499385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpE4VTLl.fasta\n",
      "\t/tmp/tmptMn9LQ.fasta\n",
      "\t/tmp/tmpWMvpd9.fasta\n",
      "\t/tmp/tmpAl96tF.fasta\n",
      "\t/tmp/tmp4O2AjC.fasta\n",
      "\t/tmp/tmpQjF7xW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNw2h6E.lastz\n",
      "\t/tmp/tmpV020oC.lastz\n",
      "\t/tmp/tmpvdv0Ec.lastz\n",
      "\t/tmp/tmpczagWr.lastz\n",
      "\t/tmp/tmpg0gBoF.lastz\n",
      "\t/tmp/tmpzI0Kmn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_necatrix_GCF_000499385.lastz\n",
      "Creating Eimeria_necatrix_GCF_000499385 table\n",
      "Inserting data to Eimeria_necatrix_GCF_000499385 table\n",
      "\n",
      "Running against Eimeria_nieschulzi_GCA_000826945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpoUlWUC.fasta\n",
      "\t/tmp/tmpEBeyhn.fasta\n",
      "\t/tmp/tmpXTWYnd.fasta\n",
      "\t/tmp/tmpHjpxfW.fasta\n",
      "\t/tmp/tmpOJi3R0.fasta\n",
      "\t/tmp/tmpSev7NS.fasta\n",
      "\t/tmp/tmprrm7g3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKYnf3M.lastz\n",
      "\t/tmp/tmptuqsNX.lastz\n",
      "\t/tmp/tmp7xNwRM.lastz\n",
      "\t/tmp/tmpGqxXQn.lastz\n",
      "\t/tmp/tmpV7XiEk.lastz\n",
      "\t/tmp/tmpv2m4ra.lastz\n",
      "\t/tmp/tmpDh7VTP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_nieschulzi_GCA_000826945.lastz\n",
      "Creating Eimeria_nieschulzi_GCA_000826945 table\n",
      "Inserting data to Eimeria_nieschulzi_GCA_000826945 table\n",
      "\n",
      "Running against Eimeria_praecox_GCA_000499445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpegWIWz.fasta\n",
      "\t/tmp/tmpnGDP4I.fasta\n",
      "\t/tmp/tmp9NTqWH.fasta\n",
      "\t/tmp/tmpz6GkHm.fasta\n",
      "\t/tmp/tmpuNLq26.fasta\n",
      "\t/tmp/tmpBh_3g1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_B9TnW.lastz\n",
      "\t/tmp/tmpX2K2ix.lastz\n",
      "\t/tmp/tmpfJy82v.lastz\n",
      "\t/tmp/tmpomD320.lastz\n",
      "\t/tmp/tmpxdX2nI.lastz\n",
      "\t/tmp/tmpAYRAkn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_praecox_GCA_000499445.lastz\n",
      "Creating Eimeria_praecox_GCA_000499445 table\n",
      "Inserting data to Eimeria_praecox_GCA_000499445 table\n",
      "\n",
      "Running against Eimeria_tenella_GCF_000499545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpNVJrjM.fasta\n",
      "\t/tmp/tmpNSjEgX.fasta\n",
      "\t/tmp/tmpQsJ2KJ.fasta\n",
      "\t/tmp/tmpq34k9j.fasta\n",
      "\t/tmp/tmpYupX3z.fasta\n",
      "\t/tmp/tmp7Wk_X6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6B3PkT.lastz\n",
      "\t/tmp/tmpFC46sh.lastz\n",
      "\t/tmp/tmp6PXGQt.lastz\n",
      "\t/tmp/tmp5J83u2.lastz\n",
      "\t/tmp/tmpygzRX8.lastz\n",
      "\t/tmp/tmpC12_bm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Eimeria_tenella_GCF_000499545.lastz\n",
      "Creating Eimeria_tenella_GCF_000499545 table\n",
      "Inserting data to Eimeria_tenella_GCF_000499545 table\n",
      "\n",
      "Running against Gregarina_niphandrodes_GCF_000223845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpy1TaUs.fasta\n",
      "\t/tmp/tmpKNsbpk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXXWMKW.lastz\n",
      "\t/tmp/tmpAcvNYa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Gregarina_niphandrodes_GCF_000223845.lastz\n",
      "Creating Gregarina_niphandrodes_GCF_000223845 table\n",
      "Inserting data to Gregarina_niphandrodes_GCF_000223845 table\n",
      "\n",
      "Running against Haemoproteus_tartakovskyi_GCA_001625125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpPSHZ89.fasta\n",
      "\t/tmp/tmpWkUDhe.fasta\n",
      "\t/tmp/tmps96PYy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFVQovM.lastz\n",
      "\t/tmp/tmpvuzWMh.lastz\n",
      "\t/tmp/tmp8ipoCJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Haemoproteus_tartakovskyi_GCA_001625125.lastz\n",
      "Creating Haemoproteus_tartakovskyi_GCA_001625125 table\n",
      "Inserting data to Haemoproteus_tartakovskyi_GCA_001625125 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Hammondia_hammondi_GCA_000447165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpyhceI_.fasta\n",
      "\t/tmp/tmptoOM3S.fasta\n",
      "\t/tmp/tmpWw6lym.fasta\n",
      "\t/tmp/tmpVyWe3a.fasta\n",
      "\t/tmp/tmpTjySHt.fasta\n",
      "\t/tmp/tmpBncjq8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3ClmzQ.lastz\n",
      "\t/tmp/tmprjzB7H.lastz\n",
      "\t/tmp/tmp04LXPs.lastz\n",
      "\t/tmp/tmpqBy3v9.lastz\n",
      "\t/tmp/tmptaIndL.lastz\n",
      "\t/tmp/tmp2oVDYe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Hammondia_hammondi_GCA_000447165.lastz\n",
      "Creating Hammondia_hammondi_GCA_000447165 table\n",
      "Inserting data to Hammondia_hammondi_GCA_000447165 table\n",
      "\n",
      "Running against Neospra_caninum_GCF_000208865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpQ3V5N0.fasta\n",
      "\t/tmp/tmpjVtAfe.fasta\n",
      "\t/tmp/tmp_qVOYY.fasta\n",
      "\t/tmp/tmpNjJYWO.fasta\n",
      "\t/tmp/tmpBQxEnO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQsLtPy.lastz\n",
      "\t/tmp/tmp9fF5gl.lastz\n",
      "\t/tmp/tmprVbxor.lastz\n",
      "\t/tmp/tmpLjDQko.lastz\n",
      "\t/tmp/tmpSnnSzE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Neospra_caninum_GCF_000208865.lastz\n",
      "Creating Neospra_caninum_GCF_000208865 table\n",
      "Inserting data to Neospra_caninum_GCF_000208865 table\n",
      "\n",
      "Running against Nephromyces_sp_GCA_004523865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 57 queries...\n",
      "\t/tmp/tmpQmDn9c.fasta\n",
      "\t/tmp/tmpIhJ6fd.fasta\n",
      "\t/tmp/tmp9c7Zbt.fasta\n",
      "\t/tmp/tmpCa78Hg.fasta\n",
      "\t/tmp/tmp1swjDR.fasta\n",
      "\t/tmp/tmpVGLCHp.fasta\n",
      "\t/tmp/tmpThQzOX.fasta\n",
      "\t/tmp/tmpaJeRYI.fasta\n",
      "\t/tmp/tmpoWOiIN.fasta\n",
      "\t/tmp/tmpz1Badg.fasta\n",
      "\t/tmp/tmpUJcaA6.fasta\n",
      "\t/tmp/tmpfo26Xw.fasta\n",
      "\t/tmp/tmphH3zg3.fasta\n",
      "\t/tmp/tmpYUIgcH.fasta\n",
      "\t/tmp/tmpIDQZ9B.fasta\n",
      "\t/tmp/tmpoNcqn8.fasta\n",
      "\t/tmp/tmpiGCv7M.fasta\n",
      "\t/tmp/tmphkW6DV.fasta\n",
      "\t/tmp/tmpnFVE4j.fasta\n",
      "\t/tmp/tmpNMMMAF.fasta\n",
      "\t/tmp/tmpdDV6zY.fasta\n",
      "\t/tmp/tmpGcINsN.fasta\n",
      "\t/tmp/tmpfJ27jG.fasta\n",
      "\t/tmp/tmpxCm8mP.fasta\n",
      "\t/tmp/tmpBBsIac.fasta\n",
      "\t/tmp/tmpOGXHCi.fasta\n",
      "\t/tmp/tmpYe26Qn.fasta\n",
      "\t/tmp/tmpG0CYD9.fasta\n",
      "\t/tmp/tmpEH584G.fasta\n",
      "\t/tmp/tmp4sypVv.fasta\n",
      "\t/tmp/tmppPoCBd.fasta\n",
      "\t/tmp/tmpOna49T.fasta\n",
      "\t/tmp/tmp5Knkzh.fasta\n",
      "\t/tmp/tmpEfhfCl.fasta\n",
      "\t/tmp/tmppAZCZo.fasta\n",
      "\t/tmp/tmpC4yR33.fasta\n",
      "\t/tmp/tmpEbHLrC.fasta\n",
      "\t/tmp/tmpcJKQJG.fasta\n",
      "\t/tmp/tmp80JHth.fasta\n",
      "\t/tmp/tmpnOhAFt.fasta\n",
      "\t/tmp/tmp8Snfiq.fasta\n",
      "\t/tmp/tmpnkmSfT.fasta\n",
      "\t/tmp/tmpUwv41k.fasta\n",
      "\t/tmp/tmpEaH2hC.fasta\n",
      "\t/tmp/tmp7MPxHE.fasta\n",
      "\t/tmp/tmpJ1sYmq.fasta\n",
      "\t/tmp/tmpBx8kEd.fasta\n",
      "\t/tmp/tmpxmXKUc.fasta\n",
      "\t/tmp/tmpQo9jt6.fasta\n",
      "\t/tmp/tmpHJFXx3.fasta\n",
      "\t/tmp/tmpyhttUl.fasta\n",
      "\t/tmp/tmpATzVXV.fasta\n",
      "\t/tmp/tmpOQRXnV.fasta\n",
      "\t/tmp/tmpryDvkT.fasta\n",
      "\t/tmp/tmpKTJojm.fasta\n",
      "\t/tmp/tmpTTpmI1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgX_0gN.lastz\n",
      "\t/tmp/tmpm635o3.lastz\n",
      "\t/tmp/tmp1rCwKY.lastz\n",
      "\t/tmp/tmpWS455I.lastz\n",
      "\t/tmp/tmpPNmZq4.lastz\n",
      "\t/tmp/tmpfApt0T.lastz\n",
      "\t/tmp/tmptCB9Wv.lastz\n",
      "\t/tmp/tmpaj0LLL.lastz\n",
      "\t/tmp/tmpBMBudt.lastz\n",
      "\t/tmp/tmptQZa_P.lastz\n",
      "\t/tmp/tmplfM3A1.lastz\n",
      "\t/tmp/tmp0nYKK6.lastz\n",
      "\t/tmp/tmphLpCuI.lastz\n",
      "\t/tmp/tmpO_AIEY.lastz\n",
      "\t/tmp/tmpP8oFcm.lastz\n",
      "\t/tmp/tmpHk7dd9.lastz\n",
      "\t/tmp/tmpojgoon.lastz\n",
      "\t/tmp/tmpj1QhGF.lastz\n",
      "\t/tmp/tmprU8agc.lastz\n",
      "\t/tmp/tmpxPlGGN.lastz\n",
      "\t/tmp/tmpXaHPu9.lastz\n",
      "\t/tmp/tmpyrAHl7.lastz\n",
      "\t/tmp/tmpf5Peyk.lastz\n",
      "\t/tmp/tmpf5NsRD.lastz\n",
      "\t/tmp/tmptCEMdE.lastz\n",
      "\t/tmp/tmp0DrQpM.lastz\n",
      "\t/tmp/tmpBHU4D7.lastz\n",
      "\t/tmp/tmpHms6rE.lastz\n",
      "\t/tmp/tmpo8vJAG.lastz\n",
      "\t/tmp/tmpbrwy4J.lastz\n",
      "\t/tmp/tmpG32VXe.lastz\n",
      "\t/tmp/tmpEuSYsA.lastz\n",
      "\t/tmp/tmpXLr7AQ.lastz\n",
      "\t/tmp/tmp3pzNww.lastz\n",
      "\t/tmp/tmpVALyad.lastz\n",
      "\t/tmp/tmpCh4VFc.lastz\n",
      "\t/tmp/tmp2aSfNF.lastz\n",
      "\t/tmp/tmp8epIMb.lastz\n",
      "\t/tmp/tmpCNPlQc.lastz\n",
      "\t/tmp/tmpvysqNY.lastz\n",
      "\t/tmp/tmpzctFtQ.lastz\n",
      "\t/tmp/tmpAqxUQU.lastz\n",
      "\t/tmp/tmpg9WeCh.lastz\n",
      "\t/tmp/tmp_50ss8.lastz\n",
      "\t/tmp/tmpp0zsAt.lastz\n",
      "\t/tmp/tmp0kYtul.lastz\n",
      "\t/tmp/tmpN29trq.lastz\n",
      "\t/tmp/tmpDd2eVG.lastz\n",
      "\t/tmp/tmpsN2nf5.lastz\n",
      "\t/tmp/tmpwz4FBH.lastz\n",
      "\t/tmp/tmp_d5P3D.lastz\n",
      "\t/tmp/tmpqzUa_n.lastz\n",
      "\t/tmp/tmpwbXeLm.lastz\n",
      "\t/tmp/tmplrYxju.lastz\n",
      "\t/tmp/tmpgT_xBx.lastz\n",
      "\t/tmp/tmpbH4wbD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Nephromyces_sp_GCA_004523865.lastz\n",
      "Creating Nephromyces_sp_GCA_004523865 table\n",
      "Inserting data to Nephromyces_sp_GCA_004523865 table\n",
      "\n",
      "Running against Plasmodium_berghei_GCA_900002375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp3mZWAS.fasta\n",
      "\t/tmp/tmpFhdBAN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoGI448.lastz\n",
      "\t/tmp/tmpg5Ymt6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_berghei_GCA_900002375.lastz\n",
      "Creating Plasmodium_berghei_GCA_900002375 table\n",
      "Inserting data to Plasmodium_berghei_GCA_900002375 table\n",
      "\n",
      "Running against Plasmodium_brasilianum_GCA_001885115.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpSYd5TZ.fasta\n",
      "\t/tmp/tmpnAbmfA.fasta\n",
      "\t/tmp/tmpZHlyFh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwd9AuF.lastz\n",
      "\t/tmp/tmpM0kIow.lastz\n",
      "\t/tmp/tmpc74DOc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_brasilianum_GCA_001885115.lastz\n",
      "Creating Plasmodium_brasilianum_GCA_001885115 table\n",
      "Inserting data to Plasmodium_brasilianum_GCA_001885115 table\n",
      "\n",
      "Running against Plasmodium_chabaudi_GCA_900002335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp4XJsb_.fasta\n",
      "\t/tmp/tmppxuf2T.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd3D8lB.lastz\n",
      "\t/tmp/tmpBs7JSz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_chabaudi_GCA_900002335.lastz\n",
      "Creating Plasmodium_chabaudi_GCA_900002335 table\n",
      "Inserting data to Plasmodium_chabaudi_GCA_900002335 table\n",
      "\n",
      "Running against Plasmodium_coatneyi_GCF_001680005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpr5M2xb.fasta\n",
      "\t/tmp/tmpiefQtw.fasta\n",
      "\t/tmp/tmp0wd6K5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphsdMNZ.lastz\n",
      "\t/tmp/tmpcYdZSh.lastz\n",
      "\t/tmp/tmpb_oAgv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_coatneyi_GCF_001680005.lastz\n",
      "Creating Plasmodium_coatneyi_GCF_001680005 table\n",
      "Inserting data to Plasmodium_coatneyi_GCF_001680005 table\n",
      "\n",
      "Running against Plasmodium_cynomolgi_GCF_000321355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmptUJ0QC.fasta\n",
      "\t/tmp/tmpu1nbgc.fasta\n",
      "\t/tmp/tmpnX_Odn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphna7Ue.lastz\n",
      "\t/tmp/tmpBwB3MU.lastz\n",
      "\t/tmp/tmpmtmDqk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_cynomolgi_GCF_000321355.lastz\n",
      "Creating Plasmodium_cynomolgi_GCF_000321355 table\n",
      "Inserting data to Plasmodium_cynomolgi_GCF_000321355 table\n",
      "\n",
      "Running against Plasmodium_fragile_GCF_000956335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpK413KO.fasta\n",
      "\t/tmp/tmpIW2PCA.fasta\n",
      "\t/tmp/tmpHnZJmQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbi6d4A.lastz\n",
      "\t/tmp/tmpN_GL4G.lastz\n",
      "\t/tmp/tmpnkg0y6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_fragile_GCF_000956335.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Plasmodium_fragile_GCF_000956335 table\n",
      "Inserting data to Plasmodium_fragile_GCF_000956335 table\n",
      "\n",
      "Running against Plasmodium_gaboni_GCF_001602025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpjDm1by.fasta\n",
      "\t/tmp/tmpHZmECQ.fasta\n",
      "\t/tmp/tmpUTMR7c.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyssxrD.lastz\n",
      "\t/tmp/tmphFeXjt.lastz\n",
      "\t/tmp/tmpIUBquQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_gaboni_GCF_001602025.lastz\n",
      "Creating Plasmodium_gaboni_GCF_001602025 table\n",
      "Inserting data to Plasmodium_gaboni_GCF_001602025 table\n",
      "\n",
      "Running against Plasmodium_gallinaceum_GCF_900005855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpJXCDmj.fasta\n",
      "\t/tmp/tmpBnnn6z.fasta\n",
      "\t/tmp/tmpLa6w3R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpF_dccd.lastz\n",
      "\t/tmp/tmpXF9GYp.lastz\n",
      "\t/tmp/tmpIIDHE1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_gallinaceum_GCF_900005855.lastz\n",
      "Creating Plasmodium_gallinaceum_GCF_900005855 table\n",
      "Inserting data to Plasmodium_gallinaceum_GCF_900005855 table\n",
      "\n",
      "Running against Plasmodium_gonderi_GCF_002157705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp5HfVJd.fasta\n",
      "\t/tmp/tmpwZGgYi.fasta\n",
      "\t/tmp/tmpzADWKC.fasta\n",
      "\t/tmp/tmpYIABjT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpddNRbI.lastz\n",
      "\t/tmp/tmpeEX4U6.lastz\n",
      "\t/tmp/tmpKAeD7b.lastz\n",
      "\t/tmp/tmpfMf5x3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_gonderi_GCF_002157705.lastz\n",
      "Creating Plasmodium_gonderi_GCF_002157705 table\n",
      "Inserting data to Plasmodium_gonderi_GCF_002157705 table\n",
      "\n",
      "Running against Plasmodium_inui_GCF_000524495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp1XoF43.fasta\n",
      "\t/tmp/tmpyaRa5k.fasta\n",
      "\t/tmp/tmpMMnYqa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpig9pS7.lastz\n",
      "\t/tmp/tmpCfF97O.lastz\n",
      "\t/tmp/tmp1rsVjq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_inui_GCF_000524495.lastz\n",
      "Creating Plasmodium_inui_GCF_000524495 table\n",
      "Inserting data to Plasmodium_inui_GCF_000524495 table\n",
      "\n",
      "Running against Plasmodium_knowlesi_GCF_000006355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpjYW_gv.fasta\n",
      "\t/tmp/tmpybIbTZ.fasta\n",
      "\t/tmp/tmpN7DzVD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpv5CXWn.lastz\n",
      "\t/tmp/tmpoxeuNY.lastz\n",
      "\t/tmp/tmpIRRf26.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_knowlesi_GCF_000006355.lastz\n",
      "Creating Plasmodium_knowlesi_GCF_000006355 table\n",
      "Inserting data to Plasmodium_knowlesi_GCF_000006355 table\n",
      "\n",
      "Running against Plasmodium_malariae_GCF_900090045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpsYCAZf.fasta\n",
      "\t/tmp/tmpOLzFsc.fasta\n",
      "\t/tmp/tmpXKPgxV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkV77dR.lastz\n",
      "\t/tmp/tmpni62nM.lastz\n",
      "\t/tmp/tmpDpyi6E.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_malariae_GCF_900090045.lastz\n",
      "Creating Plasmodium_malariae_GCF_900090045 table\n",
      "Inserting data to Plasmodium_malariae_GCF_900090045 table\n",
      "\n",
      "Running against Plasmodium_ovale_GCA_900090025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpNRtzmU.fasta\n",
      "\t/tmp/tmpRexdTu.fasta\n",
      "\t/tmp/tmpPtSlGs.fasta\n",
      "\t/tmp/tmp294HEc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWqw2le.lastz\n",
      "\t/tmp/tmpEbi4Ec.lastz\n",
      "\t/tmp/tmpOcJjwt.lastz\n",
      "\t/tmp/tmpCYMMle.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_ovale_GCA_900090025.lastz\n",
      "Creating Plasmodium_ovale_GCA_900090025 table\n",
      "Inserting data to Plasmodium_ovale_GCA_900090025 table\n",
      "\n",
      "Running against Plasmodium_reichenowi_GCF_001601855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpTcw7RA.fasta\n",
      "\t/tmp/tmpekSxzz.fasta\n",
      "\t/tmp/tmptbAIoX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPl0vMG.lastz\n",
      "\t/tmp/tmpiM6fqC.lastz\n",
      "\t/tmp/tmp3yWPjU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_reichenowi_GCF_001601855.lastz\n",
      "Creating Plasmodium_reichenowi_GCF_001601855 table\n",
      "Inserting data to Plasmodium_reichenowi_GCF_001601855 table\n",
      "\n",
      "Running against Plasmodium_relictum_GCF_900005765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpIEsKUR.fasta\n",
      "\t/tmp/tmprMw1RM.fasta\n",
      "\t/tmp/tmprhP0Xi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp71MPR1.lastz\n",
      "\t/tmp/tmpZMD8jk.lastz\n",
      "\t/tmp/tmpWWUc35.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_relictum_GCF_900005765.lastz\n",
      "Creating Plasmodium_relictum_GCF_900005765 table\n",
      "Inserting data to Plasmodium_relictum_GCF_900005765 table\n",
      "\n",
      "Running against Plasmodium_sp_GCF_900097015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpROOlWT.fasta\n",
      "\t/tmp/tmploR48q.fasta\n",
      "\t/tmp/tmpk9GGvT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRuz6Fu.lastz\n",
      "\t/tmp/tmpImiOoj.lastz\n",
      "\t/tmp/tmpxmWBcM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_sp_GCF_900097015.lastz\n",
      "Creating Plasmodium_sp_GCF_900097015 table\n",
      "Inserting data to Plasmodium_sp_GCF_900097015 table\n",
      "\n",
      "Running against Plasmodium_vinckei_GCF_000709005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp0VCMAN.fasta\n",
      "\t/tmp/tmpEtVauj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsVZ47v.lastz\n",
      "\t/tmp/tmpns1YKR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_vinckei_GCF_000709005.lastz\n",
      "Creating Plasmodium_vinckei_GCF_000709005 table\n",
      "Inserting data to Plasmodium_vinckei_GCF_000709005 table\n",
      "\n",
      "Running against Plasmodium_vivax_GCF_000002415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpl3TNkK.fasta\n",
      "\t/tmp/tmpktUqe3.fasta\n",
      "\t/tmp/tmpEggXQG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkHasjo.lastz\n",
      "\t/tmp/tmpOGkm4b.lastz\n",
      "\t/tmp/tmpw1SxRp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_vivax_GCF_000002415.lastz\n",
      "Creating Plasmodium_vivax_GCF_000002415 table\n",
      "Inserting data to Plasmodium_vivax_GCF_000002415 table\n",
      "\n",
      "Running against Plasmodium_yoelii_GCA_900002385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpbHAM93.fasta\n",
      "\t/tmp/tmpibZfmu.fasta\n",
      "\t/tmp/tmpc9wkvI.fasta\n",
      "\n",
      "Writing the results file...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpiEfxAJ.lastz\n",
      "\t/tmp/tmpN80c3H.lastz\n",
      "\t/tmp/tmpxNKQAL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_yoelii_GCA_900002385.lastz\n",
      "Creating Plasmodium_yoelii_GCA_900002385 table\n",
      "Inserting data to Plasmodium_yoelii_GCA_900002385 table\n",
      "\n",
      "Running against Sarcocystis_neurona_GCA_000727475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmpJSkzBj.fasta\n",
      "\t/tmp/tmpVi2MzA.fasta\n",
      "\t/tmp/tmpRJuqUu.fasta\n",
      "\t/tmp/tmp1nlll1.fasta\n",
      "\t/tmp/tmpMz08OI.fasta\n",
      "\t/tmp/tmpoRecPW.fasta\n",
      "\t/tmp/tmpAtGhDv.fasta\n",
      "\t/tmp/tmpDaUMA0.fasta\n",
      "\t/tmp/tmphDl4uh.fasta\n",
      "\t/tmp/tmpZf58GE.fasta\n",
      "\t/tmp/tmpiiJXbv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuqJcqa.lastz\n",
      "\t/tmp/tmpgo4c2Y.lastz\n",
      "\t/tmp/tmp0vidyL.lastz\n",
      "\t/tmp/tmpjguxoL.lastz\n",
      "\t/tmp/tmpOUMYTv.lastz\n",
      "\t/tmp/tmprQwaE6.lastz\n",
      "\t/tmp/tmp4Rdirm.lastz\n",
      "\t/tmp/tmppL6Sjl.lastz\n",
      "\t/tmp/tmpuNg5PU.lastz\n",
      "\t/tmp/tmp1ibSPH.lastz\n",
      "\t/tmp/tmpvtqKWR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Sarcocystis_neurona_GCA_000727475.lastz\n",
      "Creating Sarcocystis_neurona_GCA_000727475 table\n",
      "Inserting data to Sarcocystis_neurona_GCA_000727475 table\n",
      "\n",
      "Running against Theileria_annulata_GCF_000003225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBdJ1FP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxehYw7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Theileria_annulata_GCF_000003225.lastz\n",
      "Creating Theileria_annulata_GCF_000003225 table\n",
      "Inserting data to Theileria_annulata_GCF_000003225 table\n",
      "\n",
      "Running against Theileria_equi_GCF_000342415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp1XGjom.fasta\n",
      "\t/tmp/tmpto7n0o.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFedqNE.lastz\n",
      "\t/tmp/tmpVFds55.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Theileria_equi_GCF_000342415.lastz\n",
      "Creating Theileria_equi_GCF_000342415 table\n",
      "Inserting data to Theileria_equi_GCF_000342415 table\n",
      "\n",
      "Running against Theileria_orientalis_GCF_000740895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqFByDN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5VhoEx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Theileria_orientalis_GCF_000740895.lastz\n",
      "Creating Theileria_orientalis_GCF_000740895 table\n",
      "Inserting data to Theileria_orientalis_GCF_000740895 table\n",
      "\n",
      "Running against Theileria_parva_GCF_000165365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkOfuwi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaupolN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Theileria_parva_GCF_000165365.lastz\n",
      "Creating Theileria_parva_GCF_000165365 table\n",
      "Inserting data to Theileria_parva_GCF_000165365 table\n",
      "\n",
      "Running against Toxoplasma_gondii_GCF_000006565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpaUy25c.fasta\n",
      "\t/tmp/tmp_2JP03.fasta\n",
      "\t/tmp/tmpzICl5P.fasta\n",
      "\t/tmp/tmpGLU8Re.fasta\n",
      "\t/tmp/tmpeByjll.fasta\n",
      "\t/tmp/tmpUgWd8M.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpju3Lxk.lastz\n",
      "\t/tmp/tmpvwCzwp.lastz\n",
      "\t/tmp/tmp942sCL.lastz\n",
      "\t/tmp/tmpFFgakj.lastz\n",
      "\t/tmp/tmpC_vBs6.lastz\n",
      "\t/tmp/tmpZKxHgT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Toxoplasma_gondii_GCF_000006565.lastz\n",
      "Creating Toxoplasma_gondii_GCF_000006565 table\n",
      "Inserting data to Toxoplasma_gondii_GCF_000006565 table\n",
      "\n",
      "Running against Plasmodium_falciparum_GCA_000002765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpOlePOO.fasta\n",
      "\t/tmp/tmp7wDFiU.fasta\n",
      "\t/tmp/tmpwDyHoz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNAPGit.lastz\n",
      "\t/tmp/tmpCqamCl.lastz\n",
      "\t/tmp/tmprS9mP3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/apicomplexa/validate_intervals/lastz/Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_Plasmodium_falciparum_GCA_000002765.lastz\n",
      "Creating Plasmodium_falciparum_GCA_000002765 table\n",
      "Inserting data to Plasmodium_falciparum_GCA_000002765 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/apicomplexa/extract_probes_from_group/apicomplexa_genome.conf --lastz results/phyluce/apicomplexa/validate_intervals/lastz --probes 120 --probe-prefix uce_apicomplexa_ --name-pattern \"Plasmodium_falciparum_GCA_000002765_+2_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/apicomplexa/extract_probes_from_group/probe_fasta;\n",
      "2020-02-10 09:02:58,563 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-10 09:02:58,570 - Phyluce - INFO - --- Working on Ascogregarina_taiwanensis_GCA_000172235 genome ---\n",
      "2020-02-10 09:02:58,584 - Phyluce - INFO - Reading Ascogregarina_taiwanensis_GCA_000172235 genome\n",
      "2020-02-10 09:03:01,175 - Phyluce - INFO - Ascogregarina_taiwanensis_GCA_000172235: 110 uces, 35 dupes, 75 non-dupes, 0 orient drop, 0 length drop, 74 written\n",
      "2020-02-10 09:03:01,176 - Phyluce - INFO - -------- Working on Babesia_bigemina_GCF_000981445 genome -------\n",
      "2020-02-10 09:03:01,177 - Phyluce - INFO - Reading Babesia_bigemina_GCF_000981445 genome\n",
      "2020-02-10 09:03:01,927 - Phyluce - INFO - Babesia_bigemina_GCF_000981445: 222 uces, 46 dupes, 176 non-dupes, 8 orient drop, 5 length drop, 163 written\n",
      "2020-02-10 09:03:01,928 - Phyluce - INFO - --------- Working on Babesia_bovis_GCF_000165395 genome ---------\n",
      "2020-02-10 09:03:01,929 - Phyluce - INFO - Reading Babesia_bovis_GCF_000165395 genome\n",
      "2020-02-10 09:03:03,106 - Phyluce - INFO - Babesia_bovis_GCF_000165395: 329 uces, 76 dupes, 253 non-dupes, 8 orient drop, 3 length drop, 242 written\n",
      "2020-02-10 09:03:03,106 - Phyluce - INFO - ------- Working on Babesia_divergens_GCA_001077455 genome -------\n",
      "2020-02-10 09:03:03,112 - Phyluce - INFO - Reading Babesia_divergens_GCA_001077455 genome\n",
      "2020-02-10 09:03:04,055 - Phyluce - INFO - Babesia_divergens_GCA_001077455: 295 uces, 45 dupes, 250 non-dupes, 4 orient drop, 6 length drop, 240 written\n",
      "2020-02-10 09:03:04,055 - Phyluce - INFO - -------- Working on Babesia_microti_GCF_000691945 genome --------\n",
      "2020-02-10 09:03:04,068 - Phyluce - INFO - Reading Babesia_microti_GCF_000691945 genome\n",
      "2020-02-10 09:03:05,188 - Phyluce - INFO - Babesia_microti_GCF_000691945: 331 uces, 51 dupes, 280 non-dupes, 30 orient drop, 5 length drop, 245 written\n",
      "2020-02-10 09:03:05,188 - Phyluce - INFO - --------- Working on Babesia_ovata_GCF_002897235 genome ---------\n",
      "2020-02-10 09:03:05,189 - Phyluce - INFO - Reading Babesia_ovata_GCF_002897235 genome\n",
      "2020-02-10 09:03:06,094 - Phyluce - INFO - Babesia_ovata_GCF_002897235: 258 uces, 51 dupes, 207 non-dupes, 10 orient drop, 7 length drop, 190 written\n",
      "2020-02-10 09:03:06,094 - Phyluce - INFO - ----------- Working on Babesia_sp_GCF_002095265 genome ----------\n",
      "2020-02-10 09:03:06,095 - Phyluce - INFO - Reading Babesia_sp_GCF_002095265 genome\n",
      "2020-02-10 09:03:07,141 - Phyluce - INFO - Babesia_sp_GCF_002095265: 315 uces, 59 dupes, 256 non-dupes, 28 orient drop, 2 length drop, 226 written\n",
      "2020-02-10 09:03:07,141 - Phyluce - INFO - ------- Working on Besnoitia_besnoiti_GCF_002563875 genome ------\n",
      "2020-02-10 09:03:07,194 - Phyluce - INFO - Reading Besnoitia_besnoiti_GCF_002563875 genome\n",
      "2020-02-10 09:03:08,295 - Phyluce - INFO - Besnoitia_besnoiti_GCF_002563875: 190 uces, 45 dupes, 145 non-dupes, 2 orient drop, 14 length drop, 129 written\n",
      "2020-02-10 09:03:08,295 - Phyluce - INFO - ---- Working on Cryptospridium_andersoni_GCA_001865355 genome ---\n",
      "2020-02-10 09:03:08,516 - Phyluce - INFO - Reading Cryptospridium_andersoni_GCA_001865355 genome\n",
      "2020-02-10 09:03:09,768 - Phyluce - INFO - Cryptospridium_andersoni_GCA_001865355: 352 uces, 77 dupes, 275 non-dupes, 1 orient drop, 0 length drop, 274 written\n",
      "2020-02-10 09:03:09,768 - Phyluce - INFO - ----- Working on Cryptospridium_baileyi_GCA_001593455 genome ----\n",
      "2020-02-10 09:03:09,769 - Phyluce - INFO - Reading Cryptospridium_baileyi_GCA_001593455 genome\n",
      "2020-02-10 09:03:11,756 - Phyluce - INFO - Cryptospridium_baileyi_GCA_001593455: 344 uces, 106 dupes, 238 non-dupes, 0 orient drop, 0 length drop, 238 written\n",
      "2020-02-10 09:03:11,756 - Phyluce - INFO - ------ Working on Cryptospridium_bovis_GCA_009768925 genome -----\n",
      "2020-02-10 09:03:11,767 - Phyluce - INFO - Reading Cryptospridium_bovis_GCA_009768925 genome\n",
      "2020-02-10 09:03:13,127 - Phyluce - INFO - Cryptospridium_bovis_GCA_009768925: 317 uces, 58 dupes, 259 non-dupes, 0 orient drop, 0 length drop, 259 written\n",
      "2020-02-10 09:03:13,128 - Phyluce - INFO - ---- Working on Cryptospridium_cuniculus_GCA_004337835 genome ---\n",
      "2020-02-10 09:03:13,143 - Phyluce - INFO - Reading Cryptospridium_cuniculus_GCA_004337835 genome\n",
      "2020-02-10 09:03:14,657 - Phyluce - INFO - Cryptospridium_cuniculus_GCA_004337835: 341 uces, 61 dupes, 280 non-dupes, 0 orient drop, 2 length drop, 278 written\n",
      "2020-02-10 09:03:14,658 - Phyluce - INFO - ----- Working on Cryptospridium_hominis_GCA_000006425 genome ----\n",
      "2020-02-10 09:03:14,662 - Phyluce - INFO - Reading Cryptospridium_hominis_GCA_000006425 genome\n",
      "2020-02-10 09:03:16,360 - Phyluce - INFO - Cryptospridium_hominis_GCA_000006425: 337 uces, 96 dupes, 241 non-dupes, 0 orient drop, 0 length drop, 241 written\n",
      "2020-02-10 09:03:16,360 - Phyluce - INFO - --- Working on Cryptospridium_meleagridis_GCA_001593445 genome --\n",
      "2020-02-10 09:03:16,362 - Phyluce - INFO - Reading Cryptospridium_meleagridis_GCA_001593445 genome\n",
      "2020-02-10 09:03:17,747 - Phyluce - INFO - Cryptospridium_meleagridis_GCA_001593445: 335 uces, 62 dupes, 273 non-dupes, 0 orient drop, 0 length drop, 273 written\n",
      "2020-02-10 09:03:17,747 - Phyluce - INFO - ------ Working on Cryptospridium_muris_GCF_000006515 genome -----\n",
      "2020-02-10 09:03:17,756 - Phyluce - INFO - Reading Cryptospridium_muris_GCF_000006515 genome\n",
      "2020-02-10 09:03:19,535 - Phyluce - INFO - Cryptospridium_muris_GCF_000006515: 352 uces, 109 dupes, 243 non-dupes, 1 orient drop, 2 length drop, 240 written\n",
      "2020-02-10 09:03:19,535 - Phyluce - INFO - ----- Working on Cryptospridium_parvum_GCF_000165345 genome -----\n",
      "2020-02-10 09:03:19,536 - Phyluce - INFO - Reading Cryptospridium_parvum_GCF_000165345 genome\n",
      "2020-02-10 09:03:21,099 - Phyluce - INFO - Cryptospridium_parvum_GCF_000165345: 340 uces, 74 dupes, 266 non-dupes, 0 orient drop, 4 length drop, 262 written\n",
      "2020-02-10 09:03:21,099 - Phyluce - INFO - ----- Working on Cryptospridium_ryanae_GCA_009792415 genome -----\n",
      "2020-02-10 09:03:21,100 - Phyluce - INFO - Reading Cryptospridium_ryanae_GCA_009792415 genome\n",
      "2020-02-10 09:03:22,535 - Phyluce - INFO - Cryptospridium_ryanae_GCA_009792415: 322 uces, 62 dupes, 260 non-dupes, 0 orient drop, 0 length drop, 260 written\n",
      "2020-02-10 09:03:22,535 - Phyluce - INFO - ------- Working on Cryptospridium_sp_GCA_004936735 genome -------\n",
      "2020-02-10 09:03:22,691 - Phyluce - INFO - Reading Cryptospridium_sp_GCA_004936735 genome\n",
      "2020-02-10 09:03:24,051 - Phyluce - INFO - Cryptospridium_sp_GCA_004936735: 332 uces, 65 dupes, 267 non-dupes, 0 orient drop, 0 length drop, 267 written\n",
      "2020-02-10 09:03:24,051 - Phyluce - INFO - ----- Working on Cryptospridium_tyzzeri_GCA_007210665 genome ----\n",
      "2020-02-10 09:03:24,052 - Phyluce - INFO - Reading Cryptospridium_tyzzeri_GCA_007210665 genome\n",
      "2020-02-10 09:03:25,521 - Phyluce - INFO - Cryptospridium_tyzzeri_GCA_007210665: 333 uces, 66 dupes, 267 non-dupes, 0 orient drop, 3 length drop, 264 written\n",
      "2020-02-10 09:03:25,521 - Phyluce - INFO - ---- Working on Cryptospridium_ubiquitum_GCF_001865345 genome ---\n",
      "2020-02-10 09:03:25,522 - Phyluce - INFO - Reading Cryptospridium_ubiquitum_GCF_001865345 genome\n",
      "2020-02-10 09:03:27,049 - Phyluce - INFO - Cryptospridium_ubiquitum_GCF_001865345: 332 uces, 63 dupes, 269 non-dupes, 0 orient drop, 1 length drop, 268 written\n",
      "2020-02-10 09:03:27,050 - Phyluce - INFO - ---- Working on Cryptospridium_viatorum_GCA_004337795 genome ----\n",
      "2020-02-10 09:03:27,057 - Phyluce - INFO - Reading Cryptospridium_viatorum_GCA_004337795 genome\n",
      "2020-02-10 09:03:28,412 - Phyluce - INFO - Cryptospridium_viatorum_GCA_004337795: 323 uces, 63 dupes, 260 non-dupes, 0 orient drop, 0 length drop, 260 written\n",
      "2020-02-10 09:03:28,413 - Phyluce - INFO - ----- Working on Cyclospra_cayetanensis_GCF_002999335 genome ----\n",
      "2020-02-10 09:03:28,493 - Phyluce - INFO - Reading Cyclospra_cayetanensis_GCF_002999335 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 09:03:29,204 - Phyluce - INFO - Cyclospra_cayetanensis_GCF_002999335: 205 uces, 42 dupes, 163 non-dupes, 0 orient drop, 0 length drop, 163 written\n",
      "2020-02-10 09:03:29,204 - Phyluce - INFO - ------- Working on Cystoisospra_suis_GCA_002600585 genome -------\n",
      "2020-02-10 09:03:29,209 - Phyluce - INFO - Reading Cystoisospra_suis_GCA_002600585 genome\n",
      "2020-02-10 09:03:30,287 - Phyluce - INFO - Cystoisospra_suis_GCA_002600585: 243 uces, 70 dupes, 173 non-dupes, 0 orient drop, 3 length drop, 170 written\n",
      "2020-02-10 09:03:30,287 - Phyluce - INFO - ------- Working on Eimeria_acervulina_GCF_000499425 genome ------\n",
      "2020-02-10 09:03:30,337 - Phyluce - INFO - Reading Eimeria_acervulina_GCF_000499425 genome\n",
      "2020-02-10 09:03:31,018 - Phyluce - INFO - Eimeria_acervulina_GCF_000499425: 211 uces, 42 dupes, 169 non-dupes, 0 orient drop, 1 length drop, 168 written\n",
      "2020-02-10 09:03:31,018 - Phyluce - INFO - -------- Working on Eimeria_brunetti_GCA_000499725 genome -------\n",
      "2020-02-10 09:03:31,019 - Phyluce - INFO - Reading Eimeria_brunetti_GCA_000499725 genome\n",
      "2020-02-10 09:03:32,560 - Phyluce - INFO - Eimeria_brunetti_GCA_000499725: 205 uces, 56 dupes, 149 non-dupes, 0 orient drop, 0 length drop, 149 written\n",
      "2020-02-10 09:03:32,560 - Phyluce - INFO - ------ Working on Eimeria_falciformis_GCA_002271815 genome ------\n",
      "2020-02-10 09:03:32,572 - Phyluce - INFO - Reading Eimeria_falciformis_GCA_002271815 genome\n",
      "2020-02-10 09:03:33,345 - Phyluce - INFO - Eimeria_falciformis_GCA_002271815: 212 uces, 52 dupes, 160 non-dupes, 0 orient drop, 0 length drop, 160 written\n",
      "2020-02-10 09:03:33,345 - Phyluce - INFO - --------- Working on Eimeria_maxima_GCF_000499605 genome --------\n",
      "2020-02-10 09:03:33,346 - Phyluce - INFO - Reading Eimeria_maxima_GCF_000499605 genome\n",
      "2020-02-10 09:03:34,488 - Phyluce - INFO - Eimeria_maxima_GCF_000499605: 236 uces, 53 dupes, 183 non-dupes, 0 orient drop, 0 length drop, 183 written\n",
      "2020-02-10 09:03:34,488 - Phyluce - INFO - --------- Working on Eimeria_mitis_GCA_000499745 genome ---------\n",
      "2020-02-10 09:03:34,489 - Phyluce - INFO - Reading Eimeria_mitis_GCA_000499745 genome\n",
      "2020-02-10 09:03:35,342 - Phyluce - INFO - Eimeria_mitis_GCA_000499745: 190 uces, 43 dupes, 147 non-dupes, 0 orient drop, 0 length drop, 147 written\n",
      "2020-02-10 09:03:35,342 - Phyluce - INFO - -------- Working on Eimeria_necatrix_GCF_000499385 genome -------\n",
      "2020-02-10 09:03:35,343 - Phyluce - INFO - Reading Eimeria_necatrix_GCF_000499385 genome\n",
      "2020-02-10 09:03:36,319 - Phyluce - INFO - Eimeria_necatrix_GCF_000499385: 176 uces, 44 dupes, 132 non-dupes, 1 orient drop, 0 length drop, 130 written\n",
      "2020-02-10 09:03:36,319 - Phyluce - INFO - ------- Working on Eimeria_nieschulzi_GCA_000826945 genome ------\n",
      "2020-02-10 09:03:36,349 - Phyluce - INFO - Reading Eimeria_nieschulzi_GCA_000826945 genome\n",
      "2020-02-10 09:03:37,715 - Phyluce - INFO - Eimeria_nieschulzi_GCA_000826945: 217 uces, 78 dupes, 139 non-dupes, 0 orient drop, 0 length drop, 139 written\n",
      "2020-02-10 09:03:37,715 - Phyluce - INFO - -------- Working on Eimeria_praecox_GCA_000499445 genome --------\n",
      "2020-02-10 09:03:37,763 - Phyluce - INFO - Reading Eimeria_praecox_GCA_000499445 genome\n",
      "2020-02-10 09:03:38,784 - Phyluce - INFO - Eimeria_praecox_GCA_000499445: 214 uces, 49 dupes, 165 non-dupes, 0 orient drop, 0 length drop, 165 written\n",
      "2020-02-10 09:03:38,784 - Phyluce - INFO - -------- Working on Eimeria_tenella_GCF_000499545 genome --------\n",
      "2020-02-10 09:03:38,809 - Phyluce - INFO - Reading Eimeria_tenella_GCF_000499545 genome\n",
      "2020-02-10 09:03:40,572 - Phyluce - INFO - Eimeria_tenella_GCF_000499545: 171 uces, 55 dupes, 116 non-dupes, 1 orient drop, 0 length drop, 115 written\n",
      "2020-02-10 09:03:40,572 - Phyluce - INFO - ----- Working on Gregarina_niphandrodes_GCF_000223845 genome ----\n",
      "2020-02-10 09:03:40,628 - Phyluce - INFO - Reading Gregarina_niphandrodes_GCF_000223845 genome\n",
      "2020-02-10 09:03:41,469 - Phyluce - INFO - Gregarina_niphandrodes_GCF_000223845: 196 uces, 55 dupes, 141 non-dupes, 0 orient drop, 4 length drop, 137 written\n",
      "2020-02-10 09:03:41,469 - Phyluce - INFO - --- Working on Haemoproteus_tartakovskyi_GCA_001625125 genome ---\n",
      "2020-02-10 09:03:41,470 - Phyluce - INFO - Reading Haemoproteus_tartakovskyi_GCA_001625125 genome\n",
      "2020-02-10 09:03:49,240 - Phyluce - INFO - Haemoproteus_tartakovskyi_GCA_001625125: 658 uces, 106 dupes, 552 non-dupes, 1 orient drop, 2 length drop, 549 written\n",
      "2020-02-10 09:03:49,240 - Phyluce - INFO - ------- Working on Hammondia_hammondi_GCA_000447165 genome ------\n",
      "2020-02-10 09:03:49,241 - Phyluce - INFO - Reading Hammondia_hammondi_GCA_000447165 genome\n",
      "2020-02-10 09:03:50,317 - Phyluce - INFO - Hammondia_hammondi_GCA_000447165: 173 uces, 33 dupes, 140 non-dupes, 0 orient drop, 0 length drop, 140 written\n",
      "2020-02-10 09:03:50,317 - Phyluce - INFO - -------- Working on Neospra_caninum_GCF_000208865 genome --------\n",
      "2020-02-10 09:03:50,444 - Phyluce - INFO - Reading Neospra_caninum_GCF_000208865 genome\n",
      "2020-02-10 09:03:51,239 - Phyluce - INFO - Neospra_caninum_GCF_000208865: 194 uces, 28 dupes, 166 non-dupes, 1 orient drop, 4 length drop, 161 written\n",
      "2020-02-10 09:03:51,240 - Phyluce - INFO - --------- Working on Nephromyces_sp_GCA_004523865 genome --------\n",
      "2020-02-10 09:03:51,255 - Phyluce - INFO - Reading Nephromyces_sp_GCA_004523865 genome\n",
      "2020-02-10 09:04:13,685 - Phyluce - INFO - Nephromyces_sp_GCA_004523865: 466 uces, 390 dupes, 76 non-dupes, 0 orient drop, 0 length drop, 74 written\n",
      "2020-02-10 09:04:13,685 - Phyluce - INFO - ------- Working on Plasmodium_berghei_GCA_900002375 genome ------\n",
      "2020-02-10 09:04:13,689 - Phyluce - INFO - Reading Plasmodium_berghei_GCA_900002375 genome\n",
      "2020-02-10 09:04:17,692 - Phyluce - INFO - Plasmodium_berghei_GCA_900002375: 714 uces, 135 dupes, 579 non-dupes, 7 orient drop, 4 length drop, 568 written\n",
      "2020-02-10 09:04:17,693 - Phyluce - INFO - ----- Working on Plasmodium_brasilianum_GCA_001885115 genome ----\n",
      "2020-02-10 09:04:17,761 - Phyluce - INFO - Reading Plasmodium_brasilianum_GCA_001885115 genome\n",
      "2020-02-10 09:04:26,372 - Phyluce - INFO - Plasmodium_brasilianum_GCA_001885115: 751 uces, 151 dupes, 600 non-dupes, 0 orient drop, 3 length drop, 597 written\n",
      "2020-02-10 09:04:26,372 - Phyluce - INFO - ------ Working on Plasmodium_chabaudi_GCA_900002335 genome ------\n",
      "2020-02-10 09:04:26,388 - Phyluce - INFO - Reading Plasmodium_chabaudi_GCA_900002335 genome\n",
      "2020-02-10 09:04:30,247 - Phyluce - INFO - Plasmodium_chabaudi_GCA_900002335: 705 uces, 138 dupes, 567 non-dupes, 7 orient drop, 6 length drop, 554 written\n",
      "2020-02-10 09:04:30,247 - Phyluce - INFO - ------ Working on Plasmodium_coatneyi_GCF_001680005 genome ------\n",
      "2020-02-10 09:04:30,261 - Phyluce - INFO - Reading Plasmodium_coatneyi_GCF_001680005 genome\n",
      "2020-02-10 09:04:33,354 - Phyluce - INFO - Plasmodium_coatneyi_GCF_001680005: 728 uces, 113 dupes, 615 non-dupes, 11 orient drop, 7 length drop, 597 written\n",
      "2020-02-10 09:04:33,354 - Phyluce - INFO - ------ Working on Plasmodium_cynomolgi_GCF_000321355 genome -----\n",
      "2020-02-10 09:04:33,370 - Phyluce - INFO - Reading Plasmodium_cynomolgi_GCF_000321355 genome\n",
      "2020-02-10 09:04:37,949 - Phyluce - INFO - Plasmodium_cynomolgi_GCF_000321355: 738 uces, 114 dupes, 624 non-dupes, 0 orient drop, 8 length drop, 616 written\n",
      "2020-02-10 09:04:37,949 - Phyluce - INFO - ------- Working on Plasmodium_fragile_GCF_000956335 genome ------\n",
      "2020-02-10 09:04:37,950 - Phyluce - INFO - Reading Plasmodium_fragile_GCF_000956335 genome\n",
      "2020-02-10 09:04:40,785 - Phyluce - INFO - Plasmodium_fragile_GCF_000956335: 723 uces, 117 dupes, 606 non-dupes, 7 orient drop, 14 length drop, 585 written\n",
      "2020-02-10 09:04:40,785 - Phyluce - INFO - ------- Working on Plasmodium_gaboni_GCF_001602025 genome -------\n",
      "2020-02-10 09:04:40,787 - Phyluce - INFO - Reading Plasmodium_gaboni_GCF_001602025 genome\n",
      "2020-02-10 09:04:46,257 - Phyluce - INFO - Plasmodium_gaboni_GCF_001602025: 923 uces, 160 dupes, 763 non-dupes, 3 orient drop, 10 length drop, 750 written\n",
      "2020-02-10 09:04:46,258 - Phyluce - INFO - ----- Working on Plasmodium_gallinaceum_GCF_900005855 genome ----\n",
      "2020-02-10 09:04:46,278 - Phyluce - INFO - Reading Plasmodium_gallinaceum_GCF_900005855 genome\n",
      "2020-02-10 09:04:50,028 - Phyluce - INFO - Plasmodium_gallinaceum_GCF_900005855: 747 uces, 136 dupes, 611 non-dupes, 5 orient drop, 4 length drop, 602 written\n",
      "2020-02-10 09:04:50,028 - Phyluce - INFO - ------- Working on Plasmodium_gonderi_GCF_002157705 genome ------\n",
      "2020-02-10 09:04:50,029 - Phyluce - INFO - Reading Plasmodium_gonderi_GCF_002157705 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 09:04:57,746 - Phyluce - INFO - Plasmodium_gonderi_GCF_002157705: 742 uces, 124 dupes, 618 non-dupes, 6 orient drop, 7 length drop, 605 written\n",
      "2020-02-10 09:04:57,746 - Phyluce - INFO - -------- Working on Plasmodium_inui_GCF_000524495 genome --------\n",
      "2020-02-10 09:04:57,749 - Phyluce - INFO - Reading Plasmodium_inui_GCF_000524495 genome\n",
      "2020-02-10 09:05:01,333 - Phyluce - INFO - Plasmodium_inui_GCF_000524495: 733 uces, 108 dupes, 625 non-dupes, 6 orient drop, 6 length drop, 613 written\n",
      "2020-02-10 09:05:01,333 - Phyluce - INFO - ------ Working on Plasmodium_knowlesi_GCF_000006355 genome ------\n",
      "2020-02-10 09:05:01,379 - Phyluce - INFO - Reading Plasmodium_knowlesi_GCF_000006355 genome\n",
      "2020-02-10 09:05:04,899 - Phyluce - INFO - Plasmodium_knowlesi_GCF_000006355: 726 uces, 114 dupes, 612 non-dupes, 6 orient drop, 6 length drop, 600 written\n",
      "2020-02-10 09:05:04,900 - Phyluce - INFO - ------ Working on Plasmodium_malariae_GCF_900090045 genome ------\n",
      "2020-02-10 09:05:04,902 - Phyluce - INFO - Reading Plasmodium_malariae_GCF_900090045 genome\n",
      "2020-02-10 09:05:14,417 - Phyluce - INFO - Plasmodium_malariae_GCF_900090045: 769 uces, 153 dupes, 616 non-dupes, 6 orient drop, 10 length drop, 600 written\n",
      "2020-02-10 09:05:14,417 - Phyluce - INFO - -------- Working on Plasmodium_ovale_GCA_900090025 genome -------\n",
      "2020-02-10 09:05:14,445 - Phyluce - INFO - Reading Plasmodium_ovale_GCA_900090025 genome\n",
      "2020-02-10 09:05:23,765 - Phyluce - INFO - Plasmodium_ovale_GCA_900090025: 742 uces, 148 dupes, 594 non-dupes, 0 orient drop, 6 length drop, 588 written\n",
      "2020-02-10 09:05:23,765 - Phyluce - INFO - ----- Working on Plasmodium_reichenowi_GCF_001601855 genome -----\n",
      "2020-02-10 09:05:24,125 - Phyluce - INFO - Reading Plasmodium_reichenowi_GCF_001601855 genome\n",
      "2020-02-10 09:05:29,969 - Phyluce - INFO - Plasmodium_reichenowi_GCF_001601855: 890 uces, 139 dupes, 751 non-dupes, 1 orient drop, 11 length drop, 739 written\n",
      "2020-02-10 09:05:29,969 - Phyluce - INFO - ------ Working on Plasmodium_relictum_GCF_900005765 genome ------\n",
      "2020-02-10 09:05:29,983 - Phyluce - INFO - Reading Plasmodium_relictum_GCF_900005765 genome\n",
      "2020-02-10 09:05:33,916 - Phyluce - INFO - Plasmodium_relictum_GCF_900005765: 767 uces, 136 dupes, 631 non-dupes, 4 orient drop, 6 length drop, 621 written\n",
      "2020-02-10 09:05:33,916 - Phyluce - INFO - --------- Working on Plasmodium_sp_GCF_900097015 genome ---------\n",
      "2020-02-10 09:05:33,961 - Phyluce - INFO - Reading Plasmodium_sp_GCF_900097015 genome\n",
      "2020-02-10 09:05:40,769 - Phyluce - INFO - Plasmodium_sp_GCF_900097015: 922 uces, 184 dupes, 738 non-dupes, 3 orient drop, 8 length drop, 727 written\n",
      "2020-02-10 09:05:40,770 - Phyluce - INFO - ------- Working on Plasmodium_vinckei_GCF_000709005 genome ------\n",
      "2020-02-10 09:05:40,772 - Phyluce - INFO - Reading Plasmodium_vinckei_GCF_000709005 genome\n",
      "2020-02-10 09:05:44,338 - Phyluce - INFO - Plasmodium_vinckei_GCF_000709005: 693 uces, 140 dupes, 553 non-dupes, 7 orient drop, 5 length drop, 541 written\n",
      "2020-02-10 09:05:44,338 - Phyluce - INFO - -------- Working on Plasmodium_vivax_GCF_000002415 genome -------\n",
      "2020-02-10 09:05:44,339 - Phyluce - INFO - Reading Plasmodium_vivax_GCF_000002415 genome\n",
      "2020-02-10 09:05:48,660 - Phyluce - INFO - Plasmodium_vivax_GCF_000002415: 719 uces, 130 dupes, 589 non-dupes, 0 orient drop, 9 length drop, 580 written\n",
      "2020-02-10 09:05:48,660 - Phyluce - INFO - ------- Working on Plasmodium_yoelii_GCA_900002385 genome -------\n",
      "2020-02-10 09:05:48,661 - Phyluce - INFO - Reading Plasmodium_yoelii_GCA_900002385 genome\n",
      "2020-02-10 09:05:55,402 - Phyluce - INFO - Plasmodium_yoelii_GCA_900002385: 729 uces, 144 dupes, 585 non-dupes, 8 orient drop, 5 length drop, 572 written\n",
      "2020-02-10 09:05:55,402 - Phyluce - INFO - ------ Working on Sarcocystis_neurona_GCA_000727475 genome ------\n",
      "2020-02-10 09:05:55,441 - Phyluce - INFO - Reading Sarcocystis_neurona_GCA_000727475 genome\n",
      "2020-02-10 09:05:56,207 - Phyluce - INFO - Sarcocystis_neurona_GCA_000727475: 203 uces, 38 dupes, 165 non-dupes, 0 orient drop, 4 length drop, 161 written\n",
      "2020-02-10 09:05:56,207 - Phyluce - INFO - ------- Working on Theileria_annulata_GCF_000003225 genome ------\n",
      "2020-02-10 09:05:56,221 - Phyluce - INFO - Reading Theileria_annulata_GCF_000003225 genome\n",
      "2020-02-10 09:05:57,537 - Phyluce - INFO - Theileria_annulata_GCF_000003225: 324 uces, 80 dupes, 244 non-dupes, 10 orient drop, 3 length drop, 231 written\n",
      "2020-02-10 09:05:57,537 - Phyluce - INFO - --------- Working on Theileria_equi_GCF_000342415 genome --------\n",
      "2020-02-10 09:05:57,538 - Phyluce - INFO - Reading Theileria_equi_GCF_000342415 genome\n",
      "2020-02-10 09:05:58,811 - Phyluce - INFO - Theileria_equi_GCF_000342415: 361 uces, 93 dupes, 268 non-dupes, 7 orient drop, 4 length drop, 257 written\n",
      "2020-02-10 09:05:58,811 - Phyluce - INFO - ------ Working on Theileria_orientalis_GCF_000740895 genome -----\n",
      "2020-02-10 09:05:58,817 - Phyluce - INFO - Reading Theileria_orientalis_GCF_000740895 genome\n",
      "2020-02-10 09:05:59,711 - Phyluce - INFO - Theileria_orientalis_GCF_000740895: 235 uces, 60 dupes, 175 non-dupes, 5 orient drop, 3 length drop, 167 written\n",
      "2020-02-10 09:05:59,711 - Phyluce - INFO - -------- Working on Theileria_parva_GCF_000165365 genome --------\n",
      "2020-02-10 09:05:59,712 - Phyluce - INFO - Reading Theileria_parva_GCF_000165365 genome\n",
      "2020-02-10 09:06:00,875 - Phyluce - INFO - Theileria_parva_GCF_000165365: 313 uces, 81 dupes, 232 non-dupes, 4 orient drop, 5 length drop, 223 written\n",
      "2020-02-10 09:06:00,875 - Phyluce - INFO - ------- Working on Toxoplasma_gondii_GCF_000006565 genome -------\n",
      "2020-02-10 09:06:00,876 - Phyluce - INFO - Reading Toxoplasma_gondii_GCF_000006565 genome\n",
      "2020-02-10 09:06:01,684 - Phyluce - INFO - Toxoplasma_gondii_GCF_000006565: 177 uces, 44 dupes, 133 non-dupes, 1 orient drop, 4 length drop, 128 written\n",
      "2020-02-10 09:06:01,684 - Phyluce - INFO - ----- Working on Plasmodium_falciparum_GCA_000002765 genome -----\n",
      "2020-02-10 09:06:01,698 - Phyluce - INFO - Reading Plasmodium_falciparum_GCA_000002765 genome\n",
      "2020-02-10 09:06:11,672 - Phyluce - INFO - Plasmodium_falciparum_GCA_000002765: 938 uces, 180 dupes, 758 non-dupes, 8 orient drop, 12 length drop, 738 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/apicomplexa/extract_probes_from_group/probe_fasta --output results/phyluce/apicomplexa/extract_probes_from_group/multifastas.sqlite --base-taxon Plasmodium_falciparum_GCA_000002765;\n",
      "ascogregarina_taiwanensis_gca_000172235.\n",
      "babesia_bigemina_gcf_000981445.\n",
      "babesia_bovis_gcf_000165395.\n",
      "babesia_divergens_gca_001077455.\n",
      "babesia_microti_gcf_000691945.\n",
      "babesia_ovata_gcf_002897235.\n",
      "babesia_sp_gcf_002095265.\n",
      "besnoitia_besnoiti_gcf_002563875.\n",
      "cryptospridium_andersoni_gca_001865355.\n",
      "cryptospridium_baileyi_gca_001593455.\n",
      "cryptospridium_bovis_gca_009768925.\n",
      "cryptospridium_cuniculus_gca_004337835.\n",
      "cryptospridium_hominis_gca_000006425.\n",
      "cryptospridium_meleagridis_gca_001593445.\n",
      "cryptospridium_muris_gcf_000006515.\n",
      "cryptospridium_parvum_gcf_000165345.\n",
      "cryptospridium_ryanae_gca_009792415.\n",
      "cryptospridium_sp_gca_004936735.\n",
      "cryptospridium_tyzzeri_gca_007210665.\n",
      "cryptospridium_ubiquitum_gcf_001865345.\n",
      "cryptospridium_viatorum_gca_004337795.\n",
      "cyclospra_cayetanensis_gcf_002999335.\n",
      "cystoisospra_suis_gca_002600585.\n",
      "eimeria_acervulina_gcf_000499425.\n",
      "eimeria_brunetti_gca_000499725.\n",
      "eimeria_falciformis_gca_002271815.\n",
      "eimeria_maxima_gcf_000499605.\n",
      "eimeria_mitis_gca_000499745.\n",
      "eimeria_necatrix_gcf_000499385.\n",
      "eimeria_nieschulzi_gca_000826945.\n",
      "eimeria_praecox_gca_000499445.\n",
      "eimeria_tenella_gcf_000499545.\n",
      "gregarina_niphandrodes_gcf_000223845.\n",
      "haemoproteus_tartakovskyi_gca_001625125.\n",
      "hammondia_hammondi_gca_000447165.\n",
      "neospra_caninum_gcf_000208865.\n",
      "nephromyces_sp_gca_004523865.\n",
      "plasmodium_berghei_gca_900002375.\n",
      "plasmodium_brasilianum_gca_001885115.\n",
      "plasmodium_chabaudi_gca_900002335.\n",
      "plasmodium_coatneyi_gcf_001680005.\n",
      "plasmodium_cynomolgi_gcf_000321355.\n",
      "plasmodium_fragile_gcf_000956335.\n",
      "plasmodium_gaboni_gcf_001602025.\n",
      "plasmodium_gallinaceum_gcf_900005855.\n",
      "plasmodium_gonderi_gcf_002157705.\n",
      "plasmodium_inui_gcf_000524495.\n",
      "plasmodium_knowlesi_gcf_000006355.\n",
      "plasmodium_malariae_gcf_900090045.\n",
      "plasmodium_ovale_gca_900090025.\n",
      "plasmodium_reichenowi_gcf_001601855.\n",
      "plasmodium_relictum_gcf_900005765.\n",
      "plasmodium_sp_gcf_900097015.\n",
      "plasmodium_vinckei_gcf_000709005.\n",
      "plasmodium_vivax_gcf_000002415.\n",
      "plasmodium_yoelii_gca_900002385.\n",
      "sarcocystis_neurona_gca_000727475.\n",
      "theileria_annulata_gcf_000003225.\n",
      "theileria_equi_gcf_000342415.\n",
      "theileria_orientalis_gcf_000740895.\n",
      "theileria_parva_gcf_000165365.\n",
      "toxoplasma_gondii_gcf_000006565.\n",
      "plasmodium_falciparum_gca_000002765.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/apicomplexa/extract_probes_from_group/multifastas.sqlite --base-taxon Plasmodium_falciparum_GCA_000002765;\n",
      "Loci shared by 0 taxa:\t934.0\n",
      "Loci shared by 1 taxa:\t934.0\n",
      "Loci shared by 2 taxa:\t928.0\n",
      "Loci shared by 3 taxa:\t921.0\n",
      "Loci shared by 4 taxa:\t895.0\n",
      "Loci shared by 5 taxa:\t806.0\n",
      "Loci shared by 6 taxa:\t792.0\n",
      "Loci shared by 7 taxa:\t770.0\n",
      "Loci shared by 8 taxa:\t755.0\n",
      "Loci shared by 9 taxa:\t743.0\n",
      "Loci shared by 10 taxa:\t730.0\n",
      "Loci shared by 11 taxa:\t718.0\n",
      "Loci shared by 12 taxa:\t703.0\n",
      "Loci shared by 13 taxa:\t692.0\n",
      "Loci shared by 14 taxa:\t674.0\n",
      "Loci shared by 15 taxa:\t652.0\n",
      "Loci shared by 16 taxa:\t638.0\n",
      "Loci shared by 17 taxa:\t623.0\n",
      "Loci shared by 18 taxa:\t597.0\n",
      "Loci shared by 19 taxa:\t575.0\n",
      "Loci shared by 20 taxa:\t551.0\n",
      "Loci shared by 21 taxa:\t499.0\n",
      "Loci shared by 22 taxa:\t423.0\n",
      "Loci shared by 23 taxa:\t390.0\n",
      "Loci shared by 24 taxa:\t364.0\n",
      "Loci shared by 25 taxa:\t349.0\n",
      "Loci shared by 26 taxa:\t339.0\n",
      "Loci shared by 27 taxa:\t324.0\n",
      "Loci shared by 28 taxa:\t301.0\n",
      "Loci shared by 29 taxa:\t283.0\n",
      "Loci shared by 30 taxa:\t263.0\n",
      "Loci shared by 31 taxa:\t253.0\n",
      "Loci shared by 32 taxa:\t236.0\n",
      "Loci shared by 33 taxa:\t223.0\n",
      "Loci shared by 34 taxa:\t208.0\n",
      "Loci shared by 35 taxa:\t193.0\n",
      "Loci shared by 36 taxa:\t184.0\n",
      "Loci shared by 37 taxa:\t174.0\n",
      "Loci shared by 38 taxa:\t165.0\n",
      "Loci shared by 39 taxa:\t150.0\n",
      "Loci shared by 40 taxa:\t144.0\n",
      "Loci shared by 41 taxa:\t130.0\n",
      "Loci shared by 42 taxa:\t121.0\n",
      "Loci shared by 43 taxa:\t108.0\n",
      "Loci shared by 44 taxa:\t100.0\n",
      "Loci shared by 45 taxa:\t86.0\n",
      "Loci shared by 46 taxa:\t74.0\n",
      "Loci shared by 47 taxa:\t66.0\n",
      "Loci shared by 48 taxa:\t58.0\n",
      "Loci shared by 49 taxa:\t51.0\n",
      "Loci shared by 50 taxa:\t47.0\n",
      "Loci shared by 51 taxa:\t40.0\n",
      "Loci shared by 52 taxa:\t33.0\n",
      "Loci shared by 53 taxa:\t31.0\n",
      "Loci shared by 54 taxa:\t25.0\n",
      "Loci shared by 55 taxa:\t19.0\n",
      "Loci shared by 56 taxa:\t18.0\n",
      "Loci shared by 57 taxa:\t14.0\n",
      "Loci shared by 58 taxa:\t11.0\n",
      "Loci shared by 59 taxa:\t6.0\n",
      "Loci shared by 60 taxa:\t2.0\n",
      "Loci shared by 61 taxa:\t1.0\n",
      "Loci shared by 62 taxa:\t0.0\n",
      "Loci shared by 63 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/apicomplexa/extract_probes_from_group/multifastas.sqlite  --base-taxon Plasmodium_falciparum_GCA_000002765 --output results/phyluce/apicomplexa/extract_probes_from_group/Plasmodium_falciparum_GCA_000002765+2-back-to-44.conf --specific-counts 44;\n",
      "Counter({'plasmodium_sp_gcf_900097015': 100, 'plasmodium_coatneyi_gcf_001680005': 100, 'plasmodium_inui_gcf_000524495': 100, 'plasmodium_gonderi_gcf_002157705': 99, 'plasmodium_vinckei_gcf_000709005': 99, 'plasmodium_knowlesi_gcf_000006355': 99, 'plasmodium_relictum_gcf_900005765': 98, 'plasmodium_reichenowi_gcf_001601855': 98, 'plasmodium_falciparum_gca_000002765': 98, 'plasmodium_cynomolgi_gcf_000321355': 98, 'plasmodium_chabaudi_gca_900002335': 97, 'plasmodium_malariae_gcf_900090045': 97, 'plasmodium_yoelii_gca_900002385': 97, 'plasmodium_gaboni_gcf_001602025': 97, 'plasmodium_fragile_gcf_000956335': 97, 'plasmodium_vivax_gcf_000002415': 97, 'plasmodium_ovale_gca_900090025': 97, 'plasmodium_gallinaceum_gcf_900005855': 96, 'plasmodium_berghei_gca_900002375': 96, 'haemoproteus_tartakovskyi_gca_001625125': 96, 'plasmodium_brasilianum_gca_001885115': 94, 'theileria_equi_gcf_000342415': 94, 'cryptospridium_meleagridis_gca_001593445': 92, 'cryptospridium_hominis_gca_000006425': 92, 'cryptospridium_cuniculus_gca_004337835': 92, 'cryptospridium_tyzzeri_gca_007210665': 90, 'cryptospridium_sp_gca_004936735': 90, 'cryptospridium_viatorum_gca_004337795': 89, 'babesia_microti_gcf_000691945': 89, 'cryptospridium_parvum_gcf_000165345': 89, 'cryptospridium_muris_gcf_000006515': 88, 'cryptospridium_andersoni_gca_001865355': 88, 'cryptospridium_ubiquitum_gcf_001865345': 88, 'theileria_annulata_gcf_000003225': 87, 'cryptospridium_bovis_gca_009768925': 87, 'cryptospridium_ryanae_gca_009792415': 86, 'cryptospridium_baileyi_gca_001593455': 85, 'babesia_bovis_gcf_000165395': 84, 'theileria_parva_gcf_000165365': 82, 'babesia_divergens_gca_001077455': 78, 'babesia_sp_gcf_002095265': 76, 'theileria_orientalis_gcf_000740895': 68, 'eimeria_maxima_gcf_000499605': 68, 'eimeria_praecox_gca_000499445': 67, 'neospra_caninum_gcf_000208865': 66, 'eimeria_falciformis_gca_002271815': 65, 'babesia_ovata_gcf_002897235': 64, 'cystoisospra_suis_gca_002600585': 63, 'eimeria_acervulina_gcf_000499425': 62, 'eimeria_brunetti_gca_000499725': 62, 'cyclospra_cayetanensis_gcf_002999335': 61, 'toxoplasma_gondii_gcf_000006565': 59, 'babesia_bigemina_gcf_000981445': 58, 'sarcocystis_neurona_gca_000727475': 55, 'eimeria_mitis_gca_000499745': 54, 'besnoitia_besnoiti_gcf_002563875': 52, 'eimeria_necatrix_gcf_000499385': 50, 'gregarina_niphandrodes_gcf_000223845': 49, 'eimeria_nieschulzi_gca_000826945': 48, 'eimeria_tenella_gcf_000499545': 45, 'hammondia_hammondi_gca_000447165': 43, 'ascogregarina_taiwanensis_gca_000172235': 19, 'nephromyces_sp_gca_004523865': 8})\n",
      "Total loci = 100\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(44)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/apicomplexa/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/apicomplexa/extract_probes_from_group/Plasmodium_falciparum_GCA_000002765+2-back-to-44.conf --probe-prefix uce_apicomplexa_ --designer rnplattii --design apicomplexa_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNNNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 100\n",
      "Probe Count = 9289\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list.fasta --query results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Mon Feb 10, 2020  09:09:19\n",
      "Ended:  Mon Feb 10, 2020  09:09:46\n",
      "Time for execution:  0.454900900523 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list.fasta --lastz results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_apicomplexa_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 9288 fasta sequences.  Filtered 0 duplicates. Kept 9289.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/apicomplexa/final_probe_design/apicomplexa_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Mon Feb 10 09:19:36 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 9289\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 743120\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3949185\n",
      "Max number of word counting entries: 88896166\n",
      "\n",
      "# comparing sequences from          0  to       1548\n",
      ".---------- new table with     1265 representatives\n",
      "# comparing sequences from       1548  to       2838\n",
      "----------    264 remaining sequences to the next cycle\n",
      "---------- new table with      838 representatives\n",
      "# comparing sequences from       2574  to       3693\n",
      "----------    360 remaining sequences to the next cycle\n",
      "---------- new table with      592 representatives\n",
      "# comparing sequences from       3333  to       4325\n",
      "----------    363 remaining sequences to the next cycle\n",
      "---------- new table with      494 representatives\n",
      "# comparing sequences from       3962  to       4849\n",
      "----------    260 remaining sequences to the next cycle\n",
      "---------- new table with      512 representatives\n",
      "# comparing sequences from       4589  to       5372\n",
      "----------    314 remaining sequences to the next cycle\n",
      "---------- new table with      379 representatives\n",
      "# comparing sequences from       5058  to       5763\n",
      "----------    211 remaining sequences to the next cycle\n",
      "---------- new table with      379 representatives\n",
      "# comparing sequences from       5552  to       6174\n",
      "----------    252 remaining sequences to the next cycle\n",
      "---------- new table with      296 representatives\n",
      "# comparing sequences from       5922  to       6483\n",
      "----------    199 remaining sequences to the next cycle\n",
      "---------- new table with      299 representatives\n",
      "# comparing sequences from       6284  to       6784\n",
      "----------     69 remaining sequences to the next cycle\n",
      "---------- new table with      345 representatives\n",
      "# comparing sequences from       6715  to       7144\n",
      "----------     39 remaining sequences to the next cycle\n",
      "---------- new table with      267 representatives\n",
      "# comparing sequences from       7105  to       7469\n",
      "----------     33 remaining sequences to the next cycle\n",
      "---------- new table with      279 representatives\n",
      "# comparing sequences from       7436  to       7744\n",
      "----------    108 remaining sequences to the next cycle\n",
      "---------- new table with      158 representatives\n",
      "# comparing sequences from       7636  to       7911\n",
      "----------     94 remaining sequences to the next cycle\n",
      "---------- new table with      150 representatives\n",
      "# comparing sequences from       7817  to       8062\n",
      "----------    100 remaining sequences to the next cycle\n",
      "---------- new table with      126 representatives\n",
      "# comparing sequences from       7962  to       8183\n",
      "----------     93 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       8090  to       8289\n",
      "100.0%---------- new table with      159 representatives\n",
      "# comparing sequences from       8289  to       9289\n",
      ".....................---------- new table with      747 representatives\n",
      "\n",
      "     9289  finished       7385  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 90M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.75\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Cestodes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'cestoda'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Dibothriocephalus_latus_GCA_900617775\",\n",
    "               \"Echinococcus_canadensis_GCA_900004735\",\n",
    "               \"Echinococcus_granulosus_GCF_000524195\",\n",
    "               \"Echinococcus_multilocularis_GCA_000469725\",\n",
    "               \"Echinococcus_oligarthrus_GCA_900683695\",\n",
    "               \"Hydatigera_taeniaeformis_GCA_900622495\",\n",
    "               \"Hymenolepis_diminuta_GCA_900708905\",\n",
    "               \"Hymenolepis_microstoma_GCA_000469805\",\n",
    "               \"Mesocestoides_corti_GCA_900604375\",\n",
    "               \"Rodentolepis_nana_GCA_900617975\",\n",
    "               \"Schistocephalus_solidus_GCA_900618435\",\n",
    "               \"Sparganum_proliferum_GCA_902702955\",\n",
    "               \"Spirometra_erinaceieuropaei_GCA_000951995\",\n",
    "               \"Taenia_asiatica_GCA_001693035\",\n",
    "               \"Taenia_saginata_GCA_001693075\",\n",
    "               \"Taenia_solium_GCA_001870725\" ]\n",
    "\n",
    "reference_taxon = \"Taenia_multiceps_GCA_001923025\"\n",
    "\n",
    "all_taxa = group_taxa + [reference_taxon]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617775.1_D_latum_Geneva_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 167490987 bytes  9570915.94 bytes/sec\n",
      "total size is 167449980  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900004735.1_ECANG7_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 37000967 bytes  6727456.18 bytes/sec\n",
      "total size is 36991833  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000524195.1_ASM52419v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 35014944 bytes  6366361.09 bytes/sec\n",
      "total size is 35006286  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000469725.3_EMULTI002_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 35132467 bytes  789494.58 bytes/sec\n",
      "total size is 35123786  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900683695.1_ASM90068369v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 34988655 bytes  7775266.00 bytes/sec\n",
      "total size is 34980002  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900622495.1_H_taeniaeformis_Canary_Islands_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 32546197 bytes  7232497.56 bytes/sec\n",
      "total size is 32538118  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900708905.1_H._diminuta_assembly_from_ERS3052629-ERS3052634_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 54677536 bytes  8411935.08 bytes/sec\n",
      "total size is 54664041  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000469805.3_HMN_v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 49800907 bytes  9054718.00 bytes/sec\n",
      "total size is 49788645  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900604375.1_M_corti_Specht_Voge_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 37581989 bytes  6833096.55 bytes/sec\n",
      "total size is 37572689  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617975.1_H_nana_Japan_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 50677417 bytes  4406735.57 bytes/sec\n",
      "total size is 50664924  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_900618435.1_S_solidus_NST_G2_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 171382592 bytes  9263926.16 bytes/sec\n",
      "total size is 171340639  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902702955.1_SprV_v2_2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 197599344 bytes  8782194.93 bytes/sec\n",
      "total size is 197551007  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000951995.1_S_erinaceieuropaei_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 383362792 bytes  10503091.34 bytes/sec\n",
      "total size is 383269102  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001693035.2_Taenia_asiatica_TASYD01_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 51858448 bytes  6914465.33 bytes/sec\n",
      "total size is 51845662  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001923025.3_ASM192302v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 65784391 bytes  7739345.06 bytes/sec\n",
      "total size is 65768220  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001693075.2_ASM169307v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 52265391 bytes  4977660.29 bytes/sec\n",
      "total size is 52252524  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001870725.1_MEX_genome_complete.1-6-13_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 41597472 bytes  7563184.36 bytes/sec\n",
      "total size is 41587190  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484492 (\"sim_Dibothriocephalus_latus_GCA_900617775\") has been submitted\n",
      "Your job 5484493 (\"sim_Echinococcus_canadensis_GCA_900004735\") has been submitted\n",
      "Your job 5484494 (\"sim_Echinococcus_granulosus_GCF_000524195\") has been submitted\n",
      "Your job 5484495 (\"sim_Echinococcus_multilocularis_GCA_000469725\") has been submitted\n",
      "Your job 5484496 (\"sim_Echinococcus_oligarthrus_GCA_900683695\") has been submitted\n",
      "Your job 5484497 (\"sim_Hydatigera_taeniaeformis_GCA_900622495\") has been submitted\n",
      "Your job 5484498 (\"sim_Hymenolepis_diminuta_GCA_900708905\") has been submitted\n",
      "Your job 5484499 (\"sim_Hymenolepis_microstoma_GCA_000469805\") has been submitted\n",
      "Your job 5484500 (\"sim_Mesocestoides_corti_GCA_900604375\") has been submitted\n",
      "Your job 5484501 (\"sim_Rodentolepis_nana_GCA_900617975\") has been submitted\n",
      "Your job 5484502 (\"sim_Schistocephalus_solidus_GCA_900618435\") has been submitted\n",
      "Your job 5484503 (\"sim_Sparganum_proliferum_GCA_902702955\") has been submitted\n",
      "Your job 5484504 (\"sim_Spirometra_erinaceieuropaei_GCA_000951995\") has been submitted\n",
      "Your job 5484505 (\"sim_Taenia_asiatica_GCA_001693035\") has been submitted\n",
      "Your job 5484506 (\"sim_Taenia_saginata_GCA_001693075\") has been submitted\n",
      "Your job 5484507 (\"sim_Taenia_solium_GCA_001870725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/cestoda/cleaned_genomes/Taenia_multiceps_GCA_001923025_formatted.fas path=results/phyluce/cestoda/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484508 (\"map_Dibothriocephalus_latus_GCA_900617775\") has been submitted\n",
      "Your job 5484509 (\"map_Echinococcus_canadensis_GCA_900004735\") has been submitted\n",
      "Your job 5484510 (\"map_Echinococcus_granulosus_GCF_000524195\") has been submitted\n",
      "Your job 5484511 (\"map_Echinococcus_multilocularis_GCA_000469725\") has been submitted\n",
      "Your job 5484512 (\"map_Echinococcus_oligarthrus_GCA_900683695\") has been submitted\n",
      "Your job 5484513 (\"map_Hydatigera_taeniaeformis_GCA_900622495\") has been submitted\n",
      "Your job 5484514 (\"map_Hymenolepis_diminuta_GCA_900708905\") has been submitted\n",
      "Your job 5484515 (\"map_Hymenolepis_microstoma_GCA_000469805\") has been submitted\n",
      "Your job 5484516 (\"map_Mesocestoides_corti_GCA_900604375\") has been submitted\n",
      "Your job 5484517 (\"map_Rodentolepis_nana_GCA_900617975\") has been submitted\n",
      "Your job 5484518 (\"map_Schistocephalus_solidus_GCA_900618435\") has been submitted\n",
      "Your job 5484519 (\"map_Sparganum_proliferum_GCA_902702955\") has been submitted\n",
      "Your job 5484520 (\"map_Spirometra_erinaceieuropaei_GCA_000951995\") has been submitted\n",
      "Your job 5484521 (\"map_Taenia_asiatica_GCA_001693035\") has been submitted\n",
      "Your job 5484522 (\"map_Taenia_saginata_GCA_001693075\") has been submitted\n",
      "Your job 5484523 (\"map_Taenia_solium_GCA_001870725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5488019 (\"merge_Dibothriocephalus_latus_GCA_900617775\") has been submitted\n",
      "Your job 5488020 (\"merge_Echinococcus_canadensis_GCA_900004735\") has been submitted\n",
      "Your job 5488021 (\"merge_Echinococcus_granulosus_GCF_000524195\") has been submitted\n",
      "Your job 5488022 (\"merge_Echinococcus_multilocularis_GCA_000469725\") has been submitted\n",
      "Your job 5488023 (\"merge_Echinococcus_oligarthrus_GCA_900683695\") has been submitted\n",
      "Your job 5488024 (\"merge_Hydatigera_taeniaeformis_GCA_900622495\") has been submitted\n",
      "Your job 5488025 (\"merge_Hymenolepis_diminuta_GCA_900708905\") has been submitted\n",
      "Your job 5488026 (\"merge_Hymenolepis_microstoma_GCA_000469805\") has been submitted\n",
      "Your job 5488027 (\"merge_Mesocestoides_corti_GCA_900604375\") has been submitted\n",
      "Your job 5488028 (\"merge_Rodentolepis_nana_GCA_900617975\") has been submitted\n",
      "Your job 5488029 (\"merge_Schistocephalus_solidus_GCA_900618435\") has been submitted\n",
      "Your job 5488030 (\"merge_Sparganum_proliferum_GCA_902702955\") has been submitted\n",
      "Your job 5488031 (\"merge_Spirometra_erinaceieuropaei_GCA_000951995\") has been submitted\n",
      "Your job 5488032 (\"merge_Taenia_asiatica_GCA_001693035\") has been submitted\n",
      "Your job 5488033 (\"merge_Taenia_saginata_GCA_001693075\") has been submitted\n",
      "Your job 5488034 (\"merge_Taenia_solium_GCA_001870725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 231 sequences from Dibothriocephalus_latus_GCA_900617775_merged.bed.  Filtered 225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 47288 sequences from Echinococcus_canadensis_GCA_900004735_merged.bed.  Filtered 26600 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 20688.\n",
      "Screened 46393 sequences from Echinococcus_granulosus_GCF_000524195_merged.bed.  Filtered 26290 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 20103.\n",
      "Screened 43636 sequences from Echinococcus_multilocularis_GCA_000469725_merged.bed.  Filtered 25669 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 17967.\n",
      "Screened 26722 sequences from Echinococcus_oligarthrus_GCA_900683695_merged.bed.  Filtered 18324 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8398.\n",
      "Screened 34187 sequences from Hydatigera_taeniaeformis_GCA_900622495_merged.bed.  Filtered 21464 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12723.\n",
      "Screened 203 sequences from Hymenolepis_diminuta_GCA_900708905_merged.bed.  Filtered 192 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 525 sequences from Hymenolepis_microstoma_GCA_000469805_merged.bed.  Filtered 510 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15.\n",
      "Screened 404 sequences from Mesocestoides_corti_GCA_900604375_merged.bed.  Filtered 365 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 39.\n",
      "Screened 327 sequences from Rodentolepis_nana_GCA_900617975_merged.bed.  Filtered 314 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 343 sequences from Schistocephalus_solidus_GCA_900618435_merged.bed.  Filtered 336 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 410 sequences from Sparganum_proliferum_GCA_902702955_merged.bed.  Filtered 400 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 300 sequences from Spirometra_erinaceieuropaei_GCA_000951995_merged.bed.  Filtered 292 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 192033 sequences from Taenia_asiatica_GCA_001693035_merged.bed.  Filtered 91654 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 100379.\n",
      "Screened 196874 sequences from Taenia_saginata_GCA_001693075_merged.bed.  Filtered 94191 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102683.\n",
      "Screened 216554 sequences from Taenia_solium_GCA_001870725_merged.bed.  Filtered 92294 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 124260.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/cestoda/initial_intervals/Taenia_solium_GCA_001870725_merged.bed --twobit results/phyluce/cestoda/cleaned_genomes/Taenia_multiceps_GCA_001923025_formatted.2bit --output results/phyluce/cestoda/initial_intervals/Taenia_solium_GCA_001870725_stripped.bed;\n",
      "dibothriocephalus_latus_gca_900617775.\n",
      "echinococcus_canadensis_gca_900004735.....................\n",
      "echinococcus_granulosus_gcf_000524195.....................\n",
      "echinococcus_multilocularis_gca_000469725..................\n",
      "echinococcus_oligarthrus_gca_900683695.........\n",
      "hydatigera_taeniaeformis_gca_900622495.............\n",
      "hymenolepis_diminuta_gca_900708905.\n",
      "hymenolepis_microstoma_gca_000469805.\n",
      "mesocestoides_corti_gca_900604375.\n",
      "rodentolepis_nana_gca_900617975.\n",
      "schistocephalus_solidus_gca_900618435.\n",
      "sparganum_proliferum_gca_902702955.\n",
      "spirometra_erinaceieuropaei_gca_000951995.\n",
      "taenia_asiatica_gca_001693035.....................................................................................................\n",
      "taenia_saginata_gca_001693075.......................................................................................................\n",
      "taenia_solium_gca_001870725.............................................................................................................................\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/cestoda/initial_intervals/cestoda-to-Taenia_multiceps_GCA_001923025.sqlite --base-taxon Taenia_multiceps_GCA_001923025\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 0 taxa:\t172,255.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 1 taxa:\t172,255.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 2 taxa:\t111,926.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 3 taxa:\t60,355.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 4 taxa:\t26,818.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 5 taxa:\t21,044.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 6 taxa:\t17,362.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 7 taxa:\t10,785.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 8 taxa:\t3,350.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 9 taxa:\t28.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 10 taxa:\t12.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 11 taxa:\t9.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 12 taxa:\t6.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 13 taxa:\t3.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 14 taxa:\t2.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 15 taxa:\t2.0\n",
      "Loci shared by Taenia_multiceps_GCA_001923025 + 16 taxa:\t2.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/cestoda/initial_intervals/cestoda-to-Taenia_multiceps_GCA_001923025.sqlite --base-taxon Taenia_multiceps_GCA_001923025 --output results/phyluce/cestoda/initial_intervals/Taenia_multiceps_GCA_001923025_+8.bed --specific-counts 8;\n",
      "Counter({'echinococcus_multilocularis_gca_000469725': 3350, 'echinococcus_canadensis_gca_900004735': 3350, 'taenia_saginata_gca_001693075': 3349, 'taenia_solium_gca_001870725': 3349, 'taenia_asiatica_gca_001693035': 3349, 'echinococcus_granulosus_gcf_000524195': 3348, 'hydatigera_taeniaeformis_gca_900622495': 3344, 'echinococcus_oligarthrus_gca_900683695': 3326, 'mesocestoides_corti_gca_900604375': 33, 'hymenolepis_diminuta_gca_900708905': 12, 'hymenolepis_microstoma_gca_000469805': 12, 'rodentolepis_nana_gca_900617975': 12, 'sparganum_proliferum_gca_902702955': 9, 'spirometra_erinaceieuropaei_gca_000951995': 9, 'dibothriocephalus_latus_gca_900617775': 6, 'schistocephalus_solidus_gca_900618435': 6})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 8\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/cestoda/initial_intervals/Taenia_multiceps_GCA_001923025_+8.bed --twobit results/phyluce/cestoda/cleaned_genomes/Taenia_multiceps_GCA_001923025_formatted.2bit --buffer-to 160 --output results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8.fasta;\n",
      "Screened 3350 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 3350.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8.fasta --probe-prefix uce_cestoda_ --design cestoda_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 3350\n",
      "Probe Count = 6696\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas --query results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  14:23:34\n",
      "Ended:  Tue Feb 11, 2020  14:23:37\n",
      "Time for execution:  0.0525618354479 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas                        --lastz results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes_vself.lastz                       --probe-prefix=uce_cestoda_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6695 fasta sequences.  Filtered 175 duplicates. Kept 6346.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/cestoda/validate_intervals/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas --scaffoldlist Dibothriocephalus_latus_GCA_900617775 Echinococcus_canadensis_GCA_900004735 Echinococcus_granulosus_GCF_000524195 Echinococcus_multilocularis_GCA_000469725 Echinococcus_oligarthrus_GCA_900683695 Hydatigera_taeniaeformis_GCA_900622495 Hymenolepis_diminuta_GCA_900708905 Hymenolepis_microstoma_GCA_000469805 Mesocestoides_corti_GCA_900604375 Rodentolepis_nana_GCA_900617975 Schistocephalus_solidus_GCA_900618435 Sparganum_proliferum_GCA_902702955 Spirometra_erinaceieuropaei_GCA_000951995 Taenia_asiatica_GCA_001693035 Taenia_saginata_GCA_001693075 Taenia_solium_GCA_001870725 Taenia_multiceps_GCA_001923025 --genome-base-path results/phyluce/cestoda/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/cestoda/validate_intervals/cestoda-to-Taenia_multiceps_GCA_001923025.sqlite --output results/phyluce/cestoda/validate_intervals/lastz/;\n",
      "\n",
      "Running against Dibothriocephalus_latus_GCA_900617775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 54 queries...\n",
      "\t/tmp/tmpmlmLt3.fasta\n",
      "\t/tmp/tmpUoGdjL.fasta\n",
      "\t/tmp/tmpCABJ6c.fasta\n",
      "\t/tmp/tmpu6RE3u.fasta\n",
      "\t/tmp/tmpHNCAYM.fasta\n",
      "\t/tmp/tmp99JNPR.fasta\n",
      "\t/tmp/tmp28zVqC.fasta\n",
      "\t/tmp/tmpgBG145.fasta\n",
      "\t/tmp/tmpA1zpuj.fasta\n",
      "\t/tmp/tmp5hbNaY.fasta\n",
      "\t/tmp/tmpyVyF25.fasta\n",
      "\t/tmp/tmp5CcJtE.fasta\n",
      "\t/tmp/tmpd5Rdcr.fasta\n",
      "\t/tmp/tmpGh2E58.fasta\n",
      "\t/tmp/tmpfh2lTk.fasta\n",
      "\t/tmp/tmpuVu_DC.fasta\n",
      "\t/tmp/tmpIh09vv.fasta\n",
      "\t/tmp/tmpCKy2zQ.fasta\n",
      "\t/tmp/tmp_zuvBH.fasta\n",
      "\t/tmp/tmpacQ5Un.fasta\n",
      "\t/tmp/tmpoGJBnd.fasta\n",
      "\t/tmp/tmpgEKIMq.fasta\n",
      "\t/tmp/tmpCzln5Q.fasta\n",
      "\t/tmp/tmp9zUMcm.fasta\n",
      "\t/tmp/tmp4K0YW2.fasta\n",
      "\t/tmp/tmp2lfRhM.fasta\n",
      "\t/tmp/tmp0ryXEK.fasta\n",
      "\t/tmp/tmpnEH7vs.fasta\n",
      "\t/tmp/tmpvozH0D.fasta\n",
      "\t/tmp/tmppf3b68.fasta\n",
      "\t/tmp/tmpfH68yq.fasta\n",
      "\t/tmp/tmpilHGMH.fasta\n",
      "\t/tmp/tmpPENy0M.fasta\n",
      "\t/tmp/tmpmJqWTU.fasta\n",
      "\t/tmp/tmp_uRzzT.fasta\n",
      "\t/tmp/tmpPKRoiy.fasta\n",
      "\t/tmp/tmp4kDu6o.fasta\n",
      "\t/tmp/tmpk7nfLD.fasta\n",
      "\t/tmp/tmpsNMjjO.fasta\n",
      "\t/tmp/tmpyBqD6o.fasta\n",
      "\t/tmp/tmpXhLdks.fasta\n",
      "\t/tmp/tmpw7iMPL.fasta\n",
      "\t/tmp/tmpJ2FQ17.fasta\n",
      "\t/tmp/tmp9awxG2.fasta\n",
      "\t/tmp/tmpv9A44s.fasta\n",
      "\t/tmp/tmpcmVWCc.fasta\n",
      "\t/tmp/tmpXeYXUV.fasta\n",
      "\t/tmp/tmpzX0ye3.fasta\n",
      "\t/tmp/tmpxSWOei.fasta\n",
      "\t/tmp/tmppAOCXA.fasta\n",
      "\t/tmp/tmpIDLOhR.fasta\n",
      "\t/tmp/tmpAvZkDd.fasta\n",
      "\t/tmp/tmpXf_f6_.fasta\n",
      "\t/tmp/tmpjEuIjk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfL72Vj.lastz\n",
      "\t/tmp/tmpMcBZL8.lastz\n",
      "\t/tmp/tmpsxWPLe.lastz\n",
      "\t/tmp/tmpZyGaZq.lastz\n",
      "\t/tmp/tmpYMwlFe.lastz\n",
      "\t/tmp/tmpbPboo7.lastz\n",
      "\t/tmp/tmp1bgaQ8.lastz\n",
      "\t/tmp/tmp8KdVU1.lastz\n",
      "\t/tmp/tmpgN7yVT.lastz\n",
      "\t/tmp/tmpVoTD8b.lastz\n",
      "\t/tmp/tmpsY1kr_.lastz\n",
      "\t/tmp/tmpo9Wv3Y.lastz\n",
      "\t/tmp/tmp3NX7_9.lastz\n",
      "\t/tmp/tmp7JQ9KV.lastz\n",
      "\t/tmp/tmpeUlTML.lastz\n",
      "\t/tmp/tmpIlROBJ.lastz\n",
      "\t/tmp/tmpNj5zFY.lastz\n",
      "\t/tmp/tmpgI4bQY.lastz\n",
      "\t/tmp/tmpRlNF4p.lastz\n",
      "\t/tmp/tmpWrcjQP.lastz\n",
      "\t/tmp/tmp9Og6nT.lastz\n",
      "\t/tmp/tmpGT1Lh8.lastz\n",
      "\t/tmp/tmp2tlsxF.lastz\n",
      "\t/tmp/tmp7EUWej.lastz\n",
      "\t/tmp/tmpFpDw2q.lastz\n",
      "\t/tmp/tmp6XUIn7.lastz\n",
      "\t/tmp/tmpKSTRzR.lastz\n",
      "\t/tmp/tmpw3det5.lastz\n",
      "\t/tmp/tmp6P5lsy.lastz\n",
      "\t/tmp/tmp_hBvzU.lastz\n",
      "\t/tmp/tmpcFsYJh.lastz\n",
      "\t/tmp/tmpCgV6Mm.lastz\n",
      "\t/tmp/tmpMom1gc.lastz\n",
      "\t/tmp/tmpa9d4YR.lastz\n",
      "\t/tmp/tmpzCFv7v.lastz\n",
      "\t/tmp/tmpEQZrxP.lastz\n",
      "\t/tmp/tmp7qYLeC.lastz\n",
      "\t/tmp/tmpRsd8y6.lastz\n",
      "\t/tmp/tmpWk_fsW.lastz\n",
      "\t/tmp/tmppC9ROu.lastz\n",
      "\t/tmp/tmptk2ctD.lastz\n",
      "\t/tmp/tmpwfrnoD.lastz\n",
      "\t/tmp/tmpPWyB6q.lastz\n",
      "\t/tmp/tmpsnPxjj.lastz\n",
      "\t/tmp/tmp8RR_Q9.lastz\n",
      "\t/tmp/tmpECtbGJ.lastz\n",
      "\t/tmp/tmpo4PzHJ.lastz\n",
      "\t/tmp/tmpp4Nu2J.lastz\n",
      "\t/tmp/tmpisSNnM.lastz\n",
      "\t/tmp/tmpAL9Kt7.lastz\n",
      "\t/tmp/tmpIIUjs5.lastz\n",
      "\t/tmp/tmpTqEYqC.lastz\n",
      "\t/tmp/tmpM2MT7G.lastz\n",
      "\t/tmp/tmpjmzzKF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Dibothriocephalus_latus_GCA_900617775.lastz\n",
      "Creating Dibothriocephalus_latus_GCA_900617775 table\n",
      "Inserting data to Dibothriocephalus_latus_GCA_900617775 table\n",
      "\n",
      "Running against Echinococcus_canadensis_GCA_900004735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 12 queries...\n",
      "\t/tmp/tmpCGXpey.fasta\n",
      "\t/tmp/tmpXuBZod.fasta\n",
      "\t/tmp/tmpjV5k6Q.fasta\n",
      "\t/tmp/tmpHtcxk2.fasta\n",
      "\t/tmp/tmpbYdGCU.fasta\n",
      "\t/tmp/tmpeDOajv.fasta\n",
      "\t/tmp/tmp6JUpwu.fasta\n",
      "\t/tmp/tmpnG7Ipc.fasta\n",
      "\t/tmp/tmpCAMNXQ.fasta\n",
      "\t/tmp/tmpS95fKj.fasta\n",
      "\t/tmp/tmptZUMqj.fasta\n",
      "\t/tmp/tmpkGTSol.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcD37Nd.lastz\n",
      "\t/tmp/tmpBbzNDS.lastz\n",
      "\t/tmp/tmp4TVUDF.lastz\n",
      "\t/tmp/tmp2PNHTd.lastz\n",
      "\t/tmp/tmpXscwlH.lastz\n",
      "\t/tmp/tmpuPjWCM.lastz\n",
      "\t/tmp/tmpWhUKlo.lastz\n",
      "\t/tmp/tmpLRuY8P.lastz\n",
      "\t/tmp/tmpl_QHG3.lastz\n",
      "\t/tmp/tmpvVJP5i.lastz\n",
      "\t/tmp/tmpe7RMLM.lastz\n",
      "\t/tmp/tmpmwNikm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Echinococcus_canadensis_GCA_900004735.lastz\n",
      "Creating Echinococcus_canadensis_GCA_900004735 table\n",
      "Inserting data to Echinococcus_canadensis_GCA_900004735 table\n",
      "\n",
      "Running against Echinococcus_granulosus_GCF_000524195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmpMUb2fF.fasta\n",
      "\t/tmp/tmpGNLBqI.fasta\n",
      "\t/tmp/tmpdYr7gB.fasta\n",
      "\t/tmp/tmp5Nkbdq.fasta\n",
      "\t/tmp/tmpAmfL2x.fasta\n",
      "\t/tmp/tmpg0B6ih.fasta\n",
      "\t/tmp/tmpkGFe8U.fasta\n",
      "\t/tmp/tmpl5x4fC.fasta\n",
      "\t/tmp/tmp5VPIAi.fasta\n",
      "\t/tmp/tmpjVK4mJ.fasta\n",
      "\t/tmp/tmpBtujAC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_kuAvu.lastz\n",
      "\t/tmp/tmpq6SyA_.lastz\n",
      "\t/tmp/tmppXiN5E.lastz\n",
      "\t/tmp/tmpgPTq_Z.lastz\n",
      "\t/tmp/tmp0lI0k3.lastz\n",
      "\t/tmp/tmp766ZAm.lastz\n",
      "\t/tmp/tmpXMBQdD.lastz\n",
      "\t/tmp/tmp_WRj2t.lastz\n",
      "\t/tmp/tmphyj08e.lastz\n",
      "\t/tmp/tmpfV3ZYT.lastz\n",
      "\t/tmp/tmpR5WB8b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Echinococcus_granulosus_GCF_000524195.lastz\n",
      "Creating Echinococcus_granulosus_GCF_000524195 table\n",
      "Inserting data to Echinococcus_granulosus_GCF_000524195 table\n",
      "\n",
      "Running against Echinococcus_multilocularis_GCA_000469725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpsD8kW3.fasta\n",
      "\t/tmp/tmpA846r8.fasta\n",
      "\t/tmp/tmpoT9k9g.fasta\n",
      "\t/tmp/tmpDZV2DN.fasta\n",
      "\t/tmp/tmp9RTV6H.fasta\n",
      "\t/tmp/tmptyMog1.fasta\n",
      "\t/tmp/tmpPU2HlY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSQTaUA.lastz\n",
      "\t/tmp/tmpVji3VG.lastz\n",
      "\t/tmp/tmpcKcroZ.lastz\n",
      "\t/tmp/tmp0CcGvF.lastz\n",
      "\t/tmp/tmpSkiLY9.lastz\n",
      "\t/tmp/tmp0COS3U.lastz\n",
      "\t/tmp/tmpZGUByC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Echinococcus_multilocularis_GCA_000469725.lastz\n",
      "Creating Echinococcus_multilocularis_GCA_000469725 table\n",
      "Inserting data to Echinococcus_multilocularis_GCA_000469725 table\n",
      "\n",
      "Running against Echinococcus_oligarthrus_GCA_900683695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmpgKfJ2G.fasta\n",
      "\t/tmp/tmp5flZm4.fasta\n",
      "\t/tmp/tmpW6KZ5l.fasta\n",
      "\t/tmp/tmppXtFJA.fasta\n",
      "\t/tmp/tmp8oSxhF.fasta\n",
      "\t/tmp/tmpUormO6.fasta\n",
      "\t/tmp/tmpLfTk3Z.fasta\n",
      "\t/tmp/tmpQrdfl5.fasta\n",
      "\t/tmp/tmp05GdaA.fasta\n",
      "\t/tmp/tmp1K5_PP.fasta\n",
      "\t/tmp/tmpAynwjf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpK2Ue3I.lastz\n",
      "\t/tmp/tmpFtsQRM.lastz\n",
      "\t/tmp/tmpWMsyVj.lastz\n",
      "\t/tmp/tmpkwT1bN.lastz\n",
      "\t/tmp/tmpmhqQde.lastz\n",
      "\t/tmp/tmpZG04F4.lastz\n",
      "\t/tmp/tmpiiLJjj.lastz\n",
      "\t/tmp/tmpB5AVKv.lastz\n",
      "\t/tmp/tmpL6GwkP.lastz\n",
      "\t/tmp/tmpQx2vgx.lastz\n",
      "\t/tmp/tmpEjGkO3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Echinococcus_oligarthrus_GCA_900683695.lastz\n",
      "Creating Echinococcus_oligarthrus_GCA_900683695 table\n",
      "Inserting data to Echinococcus_oligarthrus_GCA_900683695 table\n",
      "\n",
      "Running against Hydatigera_taeniaeformis_GCA_900622495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmp96csdD.fasta\n",
      "\t/tmp/tmp_Ew5sk.fasta\n",
      "\t/tmp/tmpS5VTTA.fasta\n",
      "\t/tmp/tmpOK7gYj.fasta\n",
      "\t/tmp/tmpks8Zau.fasta\n",
      "\t/tmp/tmp8_UX7N.fasta\n",
      "\t/tmp/tmp5j7AXq.fasta\n",
      "\t/tmp/tmp0QJh_U.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpKUkawo.fasta\n",
      "\t/tmp/tmpdPFjml.fasta\n",
      "\t/tmp/tmpvYVO_i.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEAo7Pj.lastz\n",
      "\t/tmp/tmpbZnU61.lastz\n",
      "\t/tmp/tmpaJr9iH.lastz\n",
      "\t/tmp/tmp_ZWgnl.lastz\n",
      "\t/tmp/tmpv34SIc.lastz\n",
      "\t/tmp/tmpes_zpi.lastz\n",
      "\t/tmp/tmpIXYYjS.lastz\n",
      "\t/tmp/tmpw6k8So.lastz\n",
      "\t/tmp/tmp4LHPLK.lastz\n",
      "\t/tmp/tmpsPXZeK.lastz\n",
      "\t/tmp/tmpgtYQrT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Hydatigera_taeniaeformis_GCA_900622495.lastz\n",
      "Creating Hydatigera_taeniaeformis_GCA_900622495 table\n",
      "Inserting data to Hydatigera_taeniaeformis_GCA_900622495 table\n",
      "\n",
      "Running against Hymenolepis_diminuta_GCA_900708905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 16 queries...\n",
      "\t/tmp/tmpSxImsi.fasta\n",
      "\t/tmp/tmpODIMXA.fasta\n",
      "\t/tmp/tmpMDIlHf.fasta\n",
      "\t/tmp/tmphkaBYA.fasta\n",
      "\t/tmp/tmpeVIfSe.fasta\n",
      "\t/tmp/tmpvx9dTj.fasta\n",
      "\t/tmp/tmpxfH_5_.fasta\n",
      "\t/tmp/tmpOjIvDy.fasta\n",
      "\t/tmp/tmpnbOhzA.fasta\n",
      "\t/tmp/tmp45gBUu.fasta\n",
      "\t/tmp/tmpUimXTe.fasta\n",
      "\t/tmp/tmp5R_2rJ.fasta\n",
      "\t/tmp/tmpPu15IY.fasta\n",
      "\t/tmp/tmplD9srr.fasta\n",
      "\t/tmp/tmpogowrw.fasta\n",
      "\t/tmp/tmpXW3ZjB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRVjwTL.lastz\n",
      "\t/tmp/tmpzz88xO.lastz\n",
      "\t/tmp/tmpOSl1W3.lastz\n",
      "\t/tmp/tmpWyKWMV.lastz\n",
      "\t/tmp/tmphiDZS5.lastz\n",
      "\t/tmp/tmpbtBjB0.lastz\n",
      "\t/tmp/tmppr_Opr.lastz\n",
      "\t/tmp/tmprihToQ.lastz\n",
      "\t/tmp/tmpLC7Fny.lastz\n",
      "\t/tmp/tmpqGkNYY.lastz\n",
      "\t/tmp/tmpSrTM0E.lastz\n",
      "\t/tmp/tmpy3_D2q.lastz\n",
      "\t/tmp/tmp8FhGt5.lastz\n",
      "\t/tmp/tmpc2ChE7.lastz\n",
      "\t/tmp/tmp974Zlg.lastz\n",
      "\t/tmp/tmptx1Sp3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Hymenolepis_diminuta_GCA_900708905.lastz\n",
      "Creating Hymenolepis_diminuta_GCA_900708905 table\n",
      "Inserting data to Hymenolepis_diminuta_GCA_900708905 table\n",
      "\n",
      "Running against Hymenolepis_microstoma_GCA_000469805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpcJRu6A.fasta\n",
      "\t/tmp/tmpntULXl.fasta\n",
      "\t/tmp/tmp3KK3YJ.fasta\n",
      "\t/tmp/tmpzcaepT.fasta\n",
      "\t/tmp/tmpurVU9b.fasta\n",
      "\t/tmp/tmpwBSa49.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeNsLQL.lastz\n",
      "\t/tmp/tmphwG1P2.lastz\n",
      "\t/tmp/tmpEpC2xF.lastz\n",
      "\t/tmp/tmpxtWFMi.lastz\n",
      "\t/tmp/tmp3kLOTx.lastz\n",
      "\t/tmp/tmpXvC1d_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Hymenolepis_microstoma_GCA_000469805.lastz\n",
      "Creating Hymenolepis_microstoma_GCA_000469805 table\n",
      "Inserting data to Hymenolepis_microstoma_GCA_000469805 table\n",
      "\n",
      "Running against Mesocestoides_corti_GCA_900604375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 12 queries...\n",
      "\t/tmp/tmpmmQjOj.fasta\n",
      "\t/tmp/tmp7j0wtv.fasta\n",
      "\t/tmp/tmpugRZEj.fasta\n",
      "\t/tmp/tmpvIZCOJ.fasta\n",
      "\t/tmp/tmplnUi3b.fasta\n",
      "\t/tmp/tmpVXZYdt.fasta\n",
      "\t/tmp/tmp802iWP.fasta\n",
      "\t/tmp/tmpQDnWuY.fasta\n",
      "\t/tmp/tmpEF7cUh.fasta\n",
      "\t/tmp/tmpuwT2ET.fasta\n",
      "\t/tmp/tmp_sXHU1.fasta\n",
      "\t/tmp/tmp6ExoV9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkMght0.lastz\n",
      "\t/tmp/tmpxDPnwD.lastz\n",
      "\t/tmp/tmpsP9Qg2.lastz\n",
      "\t/tmp/tmpik5C5b.lastz\n",
      "\t/tmp/tmpt0umF2.lastz\n",
      "\t/tmp/tmpxKIGl8.lastz\n",
      "\t/tmp/tmpi_91KL.lastz\n",
      "\t/tmp/tmpVH3oCr.lastz\n",
      "\t/tmp/tmpoBZowi.lastz\n",
      "\t/tmp/tmpSGDvFq.lastz\n",
      "\t/tmp/tmpMarFZk.lastz\n",
      "\t/tmp/tmpYGxyio.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Mesocestoides_corti_GCA_900604375.lastz\n",
      "Creating Mesocestoides_corti_GCA_900604375 table\n",
      "Inserting data to Mesocestoides_corti_GCA_900604375 table\n",
      "\n",
      "Running against Rodentolepis_nana_GCA_900617975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 17 queries...\n",
      "\t/tmp/tmp3E1PBC.fasta\n",
      "\t/tmp/tmpzBUdkS.fasta\n",
      "\t/tmp/tmpyVzZ85.fasta\n",
      "\t/tmp/tmpXbePOD.fasta\n",
      "\t/tmp/tmpwy49oj.fasta\n",
      "\t/tmp/tmpDpPYZx.fasta\n",
      "\t/tmp/tmpBXKPCq.fasta\n",
      "\t/tmp/tmpAmekhk.fasta\n",
      "\t/tmp/tmpDkgsAO.fasta\n",
      "\t/tmp/tmpT72GG0.fasta\n",
      "\t/tmp/tmpSI9viq.fasta\n",
      "\t/tmp/tmpPJxdaC.fasta\n",
      "\t/tmp/tmpLWBrkl.fasta\n",
      "\t/tmp/tmpf2dOWi.fasta\n",
      "\t/tmp/tmp_mTZo1.fasta\n",
      "\t/tmp/tmpDUrkj3.fasta\n",
      "\t/tmp/tmpc1j_Uz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZZ1Fz_.lastz\n",
      "\t/tmp/tmpAaePt1.lastz\n",
      "\t/tmp/tmp_PRVtd.lastz\n",
      "\t/tmp/tmplfMtYx.lastz\n",
      "\t/tmp/tmpbOLBQt.lastz\n",
      "\t/tmp/tmp15zN_m.lastz\n",
      "\t/tmp/tmpqdr2be.lastz\n",
      "\t/tmp/tmpSYU1dS.lastz\n",
      "\t/tmp/tmp7Ju6BX.lastz\n",
      "\t/tmp/tmprT4dRa.lastz\n",
      "\t/tmp/tmpBWLtii.lastz\n",
      "\t/tmp/tmpIMZGuI.lastz\n",
      "\t/tmp/tmphjrPPj.lastz\n",
      "\t/tmp/tmp4tpztx.lastz\n",
      "\t/tmp/tmpvhcYfD.lastz\n",
      "\t/tmp/tmpE3TTEN.lastz\n",
      "\t/tmp/tmpI7ngue.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Rodentolepis_nana_GCA_900617975.lastz\n",
      "Creating Rodentolepis_nana_GCA_900617975 table\n",
      "Inserting data to Rodentolepis_nana_GCA_900617975 table\n",
      "\n",
      "Running against Schistocephalus_solidus_GCA_900618435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 54 queries...\n",
      "\t/tmp/tmplLeNvg.fasta\n",
      "\t/tmp/tmpFZoTxI.fasta\n",
      "\t/tmp/tmpfU6yQ_.fasta\n",
      "\t/tmp/tmpuabjSl.fasta\n",
      "\t/tmp/tmpw0b5FN.fasta\n",
      "\t/tmp/tmp4XjR7n.fasta\n",
      "\t/tmp/tmpIu4luv.fasta\n",
      "\t/tmp/tmpRlQsBy.fasta\n",
      "\t/tmp/tmpr5vSiv.fasta\n",
      "\t/tmp/tmpNLaZrv.fasta\n",
      "\t/tmp/tmpisiD0T.fasta\n",
      "\t/tmp/tmpmQuDqR.fasta\n",
      "\t/tmp/tmpxN4njx.fasta\n",
      "\t/tmp/tmpzWzeLd.fasta\n",
      "\t/tmp/tmpz1xroR.fasta\n",
      "\t/tmp/tmpLo95L8.fasta\n",
      "\t/tmp/tmperPbFn.fasta\n",
      "\t/tmp/tmpL4NKhJ.fasta\n",
      "\t/tmp/tmpAU90gN.fasta\n",
      "\t/tmp/tmpBDmhnK.fasta\n",
      "\t/tmp/tmp39jVcM.fasta\n",
      "\t/tmp/tmpPy_V7m.fasta\n",
      "\t/tmp/tmp7xrLjs.fasta\n",
      "\t/tmp/tmpJ12fHq.fasta\n",
      "\t/tmp/tmppCHb_o.fasta\n",
      "\t/tmp/tmpw0WEv5.fasta\n",
      "\t/tmp/tmpdIj681.fasta\n",
      "\t/tmp/tmpYskBKM.fasta\n",
      "\t/tmp/tmpZQozgg.fasta\n",
      "\t/tmp/tmpEObuQP.fasta\n",
      "\t/tmp/tmpe46Ktw.fasta\n",
      "\t/tmp/tmpt6nw7j.fasta\n",
      "\t/tmp/tmpDFBdik.fasta\n",
      "\t/tmp/tmpcx6ucE.fasta\n",
      "\t/tmp/tmprr98XT.fasta\n",
      "\t/tmp/tmpEEILxJ.fasta\n",
      "\t/tmp/tmpAFMt6i.fasta\n",
      "\t/tmp/tmphvHxMP.fasta\n",
      "\t/tmp/tmpngb0Ww.fasta\n",
      "\t/tmp/tmpwxaIRf.fasta\n",
      "\t/tmp/tmp8fCZz9.fasta\n",
      "\t/tmp/tmpkBzaLX.fasta\n",
      "\t/tmp/tmpoWJdUP.fasta\n",
      "\t/tmp/tmp5DH855.fasta\n",
      "\t/tmp/tmpAstwnH.fasta\n",
      "\t/tmp/tmp7pTpBY.fasta\n",
      "\t/tmp/tmpdkfxjq.fasta\n",
      "\t/tmp/tmpbYrvQa.fasta\n",
      "\t/tmp/tmpaS8MUi.fasta\n",
      "\t/tmp/tmpkJHR60.fasta\n",
      "\t/tmp/tmpEWKYIh.fasta\n",
      "\t/tmp/tmpDq6cOy.fasta\n",
      "\t/tmp/tmpTRv_BL.fasta\n",
      "\t/tmp/tmpEvdlpt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr0byKO.lastz\n",
      "\t/tmp/tmpCYEDqb.lastz\n",
      "\t/tmp/tmpsiHprT.lastz\n",
      "\t/tmp/tmp2vms0E.lastz\n",
      "\t/tmp/tmp6_qeP0.lastz\n",
      "\t/tmp/tmpXTDQKM.lastz\n",
      "\t/tmp/tmpowbGQ1.lastz\n",
      "\t/tmp/tmpF8W6zN.lastz\n",
      "\t/tmp/tmp_N3QkS.lastz\n",
      "\t/tmp/tmpGCr3zZ.lastz\n",
      "\t/tmp/tmpyEobEt.lastz\n",
      "\t/tmp/tmpMtCW24.lastz\n",
      "\t/tmp/tmp1Ng91z.lastz\n",
      "\t/tmp/tmpVdvUGa.lastz\n",
      "\t/tmp/tmpScoB8o.lastz\n",
      "\t/tmp/tmpRISgrs.lastz\n",
      "\t/tmp/tmpP3fJNO.lastz\n",
      "\t/tmp/tmpY56jEO.lastz\n",
      "\t/tmp/tmpLaIrYb.lastz\n",
      "\t/tmp/tmpRfEytN.lastz\n",
      "\t/tmp/tmpBuk0c7.lastz\n",
      "\t/tmp/tmpgzISAq.lastz\n",
      "\t/tmp/tmpz_DBAQ.lastz\n",
      "\t/tmp/tmpU8L3_a.lastz\n",
      "\t/tmp/tmpFMIME3.lastz\n",
      "\t/tmp/tmpTopbAE.lastz\n",
      "\t/tmp/tmpKptnBI.lastz\n",
      "\t/tmp/tmp6r0eN8.lastz\n",
      "\t/tmp/tmpwAXOQN.lastz\n",
      "\t/tmp/tmpM30Hke.lastz\n",
      "\t/tmp/tmpRytW9y.lastz\n",
      "\t/tmp/tmp_XlJsU.lastz\n",
      "\t/tmp/tmp_mex1V.lastz\n",
      "\t/tmp/tmpaO4Equ.lastz\n",
      "\t/tmp/tmppw4_aG.lastz\n",
      "\t/tmp/tmpyY0gwp.lastz\n",
      "\t/tmp/tmpdiqwJb.lastz\n",
      "\t/tmp/tmpWsdY41.lastz\n",
      "\t/tmp/tmp0zGFh1.lastz\n",
      "\t/tmp/tmpOCkW0D.lastz\n",
      "\t/tmp/tmpC7Zvwp.lastz\n",
      "\t/tmp/tmpItCs0r.lastz\n",
      "\t/tmp/tmpfT76JD.lastz\n",
      "\t/tmp/tmp2x5_0V.lastz\n",
      "\t/tmp/tmp_O6bXH.lastz\n",
      "\t/tmp/tmpwGJNZr.lastz\n",
      "\t/tmp/tmp7ehy8P.lastz\n",
      "\t/tmp/tmp3XrxXM.lastz\n",
      "\t/tmp/tmpzT7_LN.lastz\n",
      "\t/tmp/tmpQia4Sd.lastz\n",
      "\t/tmp/tmpCtjnJZ.lastz\n",
      "\t/tmp/tmpNQn_jl.lastz\n",
      "\t/tmp/tmplDBct8.lastz\n",
      "\t/tmp/tmp3WUBkN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Schistocephalus_solidus_GCA_900618435.lastz\n",
      "Creating Schistocephalus_solidus_GCA_900618435 table\n",
      "Inserting data to Schistocephalus_solidus_GCA_900618435 table\n",
      "\n",
      "Running against Sparganum_proliferum_GCA_902702955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 61 queries...\n",
      "\t/tmp/tmp9b_R0L.fasta\n",
      "\t/tmp/tmpcgSn6v.fasta\n",
      "\t/tmp/tmpGIJRaZ.fasta\n",
      "\t/tmp/tmpSWyhl6.fasta\n",
      "\t/tmp/tmpshirPK.fasta\n",
      "\t/tmp/tmphVhISy.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpqpcT8o.fasta\n",
      "\t/tmp/tmpcKouOZ.fasta\n",
      "\t/tmp/tmp786ygv.fasta\n",
      "\t/tmp/tmpy26qT_.fasta\n",
      "\t/tmp/tmpUaQ4rz.fasta\n",
      "\t/tmp/tmp__3Ju6.fasta\n",
      "\t/tmp/tmppUmNH6.fasta\n",
      "\t/tmp/tmpxKewCG.fasta\n",
      "\t/tmp/tmpGvqOYM.fasta\n",
      "\t/tmp/tmpwbfQBH.fasta\n",
      "\t/tmp/tmpFceksi.fasta\n",
      "\t/tmp/tmpprKbMO.fasta\n",
      "\t/tmp/tmpQdShzD.fasta\n",
      "\t/tmp/tmpOL1awv.fasta\n",
      "\t/tmp/tmpNaCRJH.fasta\n",
      "\t/tmp/tmpuLJOgz.fasta\n",
      "\t/tmp/tmpZZ9BNr.fasta\n",
      "\t/tmp/tmpcSejsf.fasta\n",
      "\t/tmp/tmpr_BwCn.fasta\n",
      "\t/tmp/tmpteNs6N.fasta\n",
      "\t/tmp/tmp4wmaWN.fasta\n",
      "\t/tmp/tmp94ZwZ1.fasta\n",
      "\t/tmp/tmp7OPs65.fasta\n",
      "\t/tmp/tmpkeLZS2.fasta\n",
      "\t/tmp/tmpomGaDH.fasta\n",
      "\t/tmp/tmpGS9lQG.fasta\n",
      "\t/tmp/tmpKJCZRK.fasta\n",
      "\t/tmp/tmpgxgz93.fasta\n",
      "\t/tmp/tmpk3c3vK.fasta\n",
      "\t/tmp/tmpscP9vy.fasta\n",
      "\t/tmp/tmpiSh907.fasta\n",
      "\t/tmp/tmpn8gFNd.fasta\n",
      "\t/tmp/tmpF5uEea.fasta\n",
      "\t/tmp/tmpa3q5kJ.fasta\n",
      "\t/tmp/tmpCwW4yA.fasta\n",
      "\t/tmp/tmp2e3i97.fasta\n",
      "\t/tmp/tmpnAtcoV.fasta\n",
      "\t/tmp/tmpYB63QV.fasta\n",
      "\t/tmp/tmpcHgsUt.fasta\n",
      "\t/tmp/tmpbHa_3T.fasta\n",
      "\t/tmp/tmplprHFg.fasta\n",
      "\t/tmp/tmpQguMdg.fasta\n",
      "\t/tmp/tmpzI9tYs.fasta\n",
      "\t/tmp/tmpcbXY4b.fasta\n",
      "\t/tmp/tmpqtAJro.fasta\n",
      "\t/tmp/tmpe10KPH.fasta\n",
      "\t/tmp/tmpfuSpDR.fasta\n",
      "\t/tmp/tmp2PKKFl.fasta\n",
      "\t/tmp/tmpAV0_D0.fasta\n",
      "\t/tmp/tmpGzIPPF.fasta\n",
      "\t/tmp/tmpST6cAa.fasta\n",
      "\t/tmp/tmpqbY2g_.fasta\n",
      "\t/tmp/tmp5S6OQs.fasta\n",
      "\t/tmp/tmp6gRrR3.fasta\n",
      "\t/tmp/tmp1Y0hZU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKlcRP6.lastz\n",
      "\t/tmp/tmp7tW4Tf.lastz\n",
      "\t/tmp/tmpmfQKGR.lastz\n",
      "\t/tmp/tmpuuLI9O.lastz\n",
      "\t/tmp/tmp1McAI5.lastz\n",
      "\t/tmp/tmp_eXUUg.lastz\n",
      "\t/tmp/tmprBym4N.lastz\n",
      "\t/tmp/tmpF5q1zh.lastz\n",
      "\t/tmp/tmp8Ry0G2.lastz\n",
      "\t/tmp/tmpFBm15L.lastz\n",
      "\t/tmp/tmpELOYKq.lastz\n",
      "\t/tmp/tmp8R5gq8.lastz\n",
      "\t/tmp/tmpXyRyMZ.lastz\n",
      "\t/tmp/tmpifQipM.lastz\n",
      "\t/tmp/tmpVztK3D.lastz\n",
      "\t/tmp/tmp8DXD5N.lastz\n",
      "\t/tmp/tmpAdBROS.lastz\n",
      "\t/tmp/tmpRunLfl.lastz\n",
      "\t/tmp/tmpVosXzZ.lastz\n",
      "\t/tmp/tmpmToSM5.lastz\n",
      "\t/tmp/tmpq7D_86.lastz\n",
      "\t/tmp/tmpZ82E6j.lastz\n",
      "\t/tmp/tmpzOvkz4.lastz\n",
      "\t/tmp/tmpIxfVyP.lastz\n",
      "\t/tmp/tmpKo0LVF.lastz\n",
      "\t/tmp/tmpBXdCDm.lastz\n",
      "\t/tmp/tmp83BNc5.lastz\n",
      "\t/tmp/tmpEAA3nZ.lastz\n",
      "\t/tmp/tmpWSqwNx.lastz\n",
      "\t/tmp/tmpt7HZVz.lastz\n",
      "\t/tmp/tmpNSgeho.lastz\n",
      "\t/tmp/tmp8YnCED.lastz\n",
      "\t/tmp/tmpF1KavI.lastz\n",
      "\t/tmp/tmpYDmIih.lastz\n",
      "\t/tmp/tmpMNJyWg.lastz\n",
      "\t/tmp/tmpcAPx0a.lastz\n",
      "\t/tmp/tmpQiZvgK.lastz\n",
      "\t/tmp/tmp8nhWw4.lastz\n",
      "\t/tmp/tmpEeebSP.lastz\n",
      "\t/tmp/tmpg_lWy1.lastz\n",
      "\t/tmp/tmpXrkdv9.lastz\n",
      "\t/tmp/tmpeP6tPI.lastz\n",
      "\t/tmp/tmpxzhIVd.lastz\n",
      "\t/tmp/tmpij1L_t.lastz\n",
      "\t/tmp/tmpv7V9gW.lastz\n",
      "\t/tmp/tmp0eRjhj.lastz\n",
      "\t/tmp/tmpdtGPqp.lastz\n",
      "\t/tmp/tmpYut8gO.lastz\n",
      "\t/tmp/tmplZOxui.lastz\n",
      "\t/tmp/tmpC5CfUT.lastz\n",
      "\t/tmp/tmp5TC0id.lastz\n",
      "\t/tmp/tmp8w0410.lastz\n",
      "\t/tmp/tmpaSC4Bh.lastz\n",
      "\t/tmp/tmpejc1pE.lastz\n",
      "\t/tmp/tmphYHPw4.lastz\n",
      "\t/tmp/tmpA8OiIs.lastz\n",
      "\t/tmp/tmpFhV32k.lastz\n",
      "\t/tmp/tmpSjjNda.lastz\n",
      "\t/tmp/tmpOzBxnx.lastz\n",
      "\t/tmp/tmpZxzxhO.lastz\n",
      "\t/tmp/tmpqp1P8X.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Sparganum_proliferum_GCA_902702955.lastz\n",
      "Creating Sparganum_proliferum_GCA_902702955 table\n",
      "Inserting data to Sparganum_proliferum_GCA_902702955 table\n",
      "\n",
      "Running against Spirometra_erinaceieuropaei_GCA_000951995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 126 queries...\n",
      "\t/tmp/tmp3Bva2S.fasta\n",
      "\t/tmp/tmpFHq_Um.fasta\n",
      "\t/tmp/tmpEl_n95.fasta\n",
      "\t/tmp/tmpOQXFx1.fasta\n",
      "\t/tmp/tmpPW3zjs.fasta\n",
      "\t/tmp/tmpT91bes.fasta\n",
      "\t/tmp/tmpJHCLgt.fasta\n",
      "\t/tmp/tmpEMRA2Z.fasta\n",
      "\t/tmp/tmpXZfqFy.fasta\n",
      "\t/tmp/tmpxwKQEb.fasta\n",
      "\t/tmp/tmpICtp2l.fasta\n",
      "\t/tmp/tmpwAKgvB.fasta\n",
      "\t/tmp/tmp8LJGZO.fasta\n",
      "\t/tmp/tmpRlzWq8.fasta\n",
      "\t/tmp/tmpN2EZ6k.fasta\n",
      "\t/tmp/tmpAz3uqc.fasta\n",
      "\t/tmp/tmpdilL9i.fasta\n",
      "\t/tmp/tmpl_0r9r.fasta\n",
      "\t/tmp/tmp0C9Fum.fasta\n",
      "\t/tmp/tmpV5uMAy.fasta\n",
      "\t/tmp/tmpUfYKXB.fasta\n",
      "\t/tmp/tmpFJyaN1.fasta\n",
      "\t/tmp/tmpKFM3d5.fasta\n",
      "\t/tmp/tmpZ1Ljg2.fasta\n",
      "\t/tmp/tmpwfdChN.fasta\n",
      "\t/tmp/tmpnkwkJQ.fasta\n",
      "\t/tmp/tmptg5AA9.fasta\n",
      "\t/tmp/tmpBqNUo7.fasta\n",
      "\t/tmp/tmp6rm5U7.fasta\n",
      "\t/tmp/tmpgsYtt8.fasta\n",
      "\t/tmp/tmpml2Azw.fasta\n",
      "\t/tmp/tmpzeq3m8.fasta\n",
      "\t/tmp/tmpTBwCok.fasta\n",
      "\t/tmp/tmpzRR2cD.fasta\n",
      "\t/tmp/tmpH90TBK.fasta\n",
      "\t/tmp/tmp_nPcox.fasta\n",
      "\t/tmp/tmp_n9W1I.fasta\n",
      "\t/tmp/tmpurIj7B.fasta\n",
      "\t/tmp/tmpIzFvu2.fasta\n",
      "\t/tmp/tmpEqoFFW.fasta\n",
      "\t/tmp/tmp6xf4Ry.fasta\n",
      "\t/tmp/tmpoeGYOo.fasta\n",
      "\t/tmp/tmpZqN_W5.fasta\n",
      "\t/tmp/tmp6cgpjL.fasta\n",
      "\t/tmp/tmpfvKVSt.fasta\n",
      "\t/tmp/tmpIKC8MP.fasta\n",
      "\t/tmp/tmp2E3NFf.fasta\n",
      "\t/tmp/tmpyZFYc3.fasta\n",
      "\t/tmp/tmpNceRTu.fasta\n",
      "\t/tmp/tmp2xyEeH.fasta\n",
      "\t/tmp/tmp4rqnnw.fasta\n",
      "\t/tmp/tmpe0LNMD.fasta\n",
      "\t/tmp/tmpNiCU0I.fasta\n",
      "\t/tmp/tmp72rm3N.fasta\n",
      "\t/tmp/tmpxzTnBz.fasta\n",
      "\t/tmp/tmpPhSEBJ.fasta\n",
      "\t/tmp/tmpyOM13t.fasta\n",
      "\t/tmp/tmpo7xOVw.fasta\n",
      "\t/tmp/tmp0TfzdL.fasta\n",
      "\t/tmp/tmpH0DYuw.fasta\n",
      "\t/tmp/tmpKbH9g9.fasta\n",
      "\t/tmp/tmpRVm9Hm.fasta\n",
      "\t/tmp/tmp1NhHKX.fasta\n",
      "\t/tmp/tmpHGvNmy.fasta\n",
      "\t/tmp/tmptEF3WP.fasta\n",
      "\t/tmp/tmpWPSoAm.fasta\n",
      "\t/tmp/tmpWMkHqh.fasta\n",
      "\t/tmp/tmpbtVCMi.fasta\n",
      "\t/tmp/tmpjlBgzQ.fasta\n",
      "\t/tmp/tmpartAWV.fasta\n",
      "\t/tmp/tmpBYc0wv.fasta\n",
      "\t/tmp/tmp_2o99v.fasta\n",
      "\t/tmp/tmpj7C9az.fasta\n",
      "\t/tmp/tmpsEzkhw.fasta\n",
      "\t/tmp/tmpJGT1Gj.fasta\n",
      "\t/tmp/tmpcXgpe8.fasta\n",
      "\t/tmp/tmpF9J0J1.fasta\n",
      "\t/tmp/tmpPQwmhT.fasta\n",
      "\t/tmp/tmpt2BAoA.fasta\n",
      "\t/tmp/tmpwhqURL.fasta\n",
      "\t/tmp/tmplz_0mr.fasta\n",
      "\t/tmp/tmpOLWdRJ.fasta\n",
      "\t/tmp/tmpYaovJv.fasta\n",
      "\t/tmp/tmpu2L4JW.fasta\n",
      "\t/tmp/tmpqKAmHS.fasta\n",
      "\t/tmp/tmpHuyWFy.fasta\n",
      "\t/tmp/tmpyzOAKr.fasta\n",
      "\t/tmp/tmpeo9d2y.fasta\n",
      "\t/tmp/tmpSzlzUv.fasta\n",
      "\t/tmp/tmpQ_mkBy.fasta\n",
      "\t/tmp/tmpFQhOUI.fasta\n",
      "\t/tmp/tmpscS33c.fasta\n",
      "\t/tmp/tmpW9EtDP.fasta\n",
      "\t/tmp/tmpARYhsw.fasta\n",
      "\t/tmp/tmpsvE8mq.fasta\n",
      "\t/tmp/tmpfD7c0Q.fasta\n",
      "\t/tmp/tmpxNoZfY.fasta\n",
      "\t/tmp/tmp92cxjs.fasta\n",
      "\t/tmp/tmpsb2R9P.fasta\n",
      "\t/tmp/tmpGvk6F5.fasta\n",
      "\t/tmp/tmpIeeEeo.fasta\n",
      "\t/tmp/tmpN7Nm2A.fasta\n",
      "\t/tmp/tmpQ_thhg.fasta\n",
      "\t/tmp/tmpBu4KtT.fasta\n",
      "\t/tmp/tmpEj2c4O.fasta\n",
      "\t/tmp/tmpjZAQ6Z.fasta\n",
      "\t/tmp/tmp1IMuZc.fasta\n",
      "\t/tmp/tmpRXPuRO.fasta\n",
      "\t/tmp/tmpK4ukTf.fasta\n",
      "\t/tmp/tmpSEbFLP.fasta\n",
      "\t/tmp/tmpREquzG.fasta\n",
      "\t/tmp/tmpwnS7Ov.fasta\n",
      "\t/tmp/tmp9DFmDF.fasta\n",
      "\t/tmp/tmpYJHTyR.fasta\n",
      "\t/tmp/tmp27rNeu.fasta\n",
      "\t/tmp/tmpXNrhJR.fasta\n",
      "\t/tmp/tmpBCVfth.fasta\n",
      "\t/tmp/tmpTdrgH0.fasta\n",
      "\t/tmp/tmp9Fu8Pc.fasta\n",
      "\t/tmp/tmpWFFNf7.fasta\n",
      "\t/tmp/tmpqmfw6n.fasta\n",
      "\t/tmp/tmpKfi0hz.fasta\n",
      "\t/tmp/tmpQySheT.fasta\n",
      "\t/tmp/tmpZWHgs1.fasta\n",
      "\t/tmp/tmpkZCK_V.fasta\n",
      "\t/tmp/tmpAjwRZp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd3r10N.lastz\n",
      "\t/tmp/tmpdim516.lastz\n",
      "\t/tmp/tmp16RPCz.lastz\n",
      "\t/tmp/tmpUlkPBe.lastz\n",
      "\t/tmp/tmp7M6Ing.lastz\n",
      "\t/tmp/tmpC9UGcb.lastz\n",
      "\t/tmp/tmpYF05QP.lastz\n",
      "\t/tmp/tmpsBFYLI.lastz\n",
      "\t/tmp/tmpQOO7zo.lastz\n",
      "\t/tmp/tmpcBld_7.lastz\n",
      "\t/tmp/tmp7k7UcM.lastz\n",
      "\t/tmp/tmpx3pbWb.lastz\n",
      "\t/tmp/tmp3lZpz9.lastz\n",
      "\t/tmp/tmpjowweO.lastz\n",
      "\t/tmp/tmpDexXcL.lastz\n",
      "\t/tmp/tmpKfW9Nq.lastz\n",
      "\t/tmp/tmpL4IpaV.lastz\n",
      "\t/tmp/tmpOw9jrz.lastz\n",
      "\t/tmp/tmpkRN6gm.lastz\n",
      "\t/tmp/tmpvCyeIw.lastz\n",
      "\t/tmp/tmpZSZGNB.lastz\n",
      "\t/tmp/tmpfISTcS.lastz\n",
      "\t/tmp/tmpYytYfX.lastz\n",
      "\t/tmp/tmpu8PU9w.lastz\n",
      "\t/tmp/tmpm6yoks.lastz\n",
      "\t/tmp/tmpMvYYSx.lastz\n",
      "\t/tmp/tmp0KHFLZ.lastz\n",
      "\t/tmp/tmpcUFSPc.lastz\n",
      "\t/tmp/tmpZV_7NT.lastz\n",
      "\t/tmp/tmpv8xZFP.lastz\n",
      "\t/tmp/tmpuR9hwv.lastz\n",
      "\t/tmp/tmpzPyoU9.lastz\n",
      "\t/tmp/tmpgihP62.lastz\n",
      "\t/tmp/tmphCJhF7.lastz\n",
      "\t/tmp/tmppLluAG.lastz\n",
      "\t/tmp/tmp8LbVBF.lastz\n",
      "\t/tmp/tmpIQ5ZUT.lastz\n",
      "\t/tmp/tmpXgTfvz.lastz\n",
      "\t/tmp/tmpmBW7Wh.lastz\n",
      "\t/tmp/tmpGQrrtB.lastz\n",
      "\t/tmp/tmpBfcPoM.lastz\n",
      "\t/tmp/tmpbnLNtj.lastz\n",
      "\t/tmp/tmpG7oHQF.lastz\n",
      "\t/tmp/tmpygqc6O.lastz\n",
      "\t/tmp/tmp5Edz5C.lastz\n",
      "\t/tmp/tmpLCRgm5.lastz\n",
      "\t/tmp/tmpUx87CE.lastz\n",
      "\t/tmp/tmplqATqm.lastz\n",
      "\t/tmp/tmplYMkC5.lastz\n",
      "\t/tmp/tmp30POwg.lastz\n",
      "\t/tmp/tmp5j7YJO.lastz\n",
      "\t/tmp/tmpwzR34i.lastz\n",
      "\t/tmp/tmpgUdhGb.lastz\n",
      "\t/tmp/tmp9gNQYv.lastz\n",
      "\t/tmp/tmprWEvFm.lastz\n",
      "\t/tmp/tmplDObek.lastz\n",
      "\t/tmp/tmpXY4zAN.lastz\n",
      "\t/tmp/tmphLzrvz.lastz\n",
      "\t/tmp/tmpVbxfZ6.lastz\n",
      "\t/tmp/tmpUUZTrE.lastz\n",
      "\t/tmp/tmpC7GzX8.lastz\n",
      "\t/tmp/tmpKDbyUF.lastz\n",
      "\t/tmp/tmpKWLFTe.lastz\n",
      "\t/tmp/tmpZGpIdX.lastz\n",
      "\t/tmp/tmp7JxY_Z.lastz\n",
      "\t/tmp/tmpOd_niX.lastz\n",
      "\t/tmp/tmpoFbelB.lastz\n",
      "\t/tmp/tmpaSyFpj.lastz\n",
      "\t/tmp/tmp8oBjvS.lastz\n",
      "\t/tmp/tmprNWVZf.lastz\n",
      "\t/tmp/tmpIgTsoF.lastz\n",
      "\t/tmp/tmpdvQk8g.lastz\n",
      "\t/tmp/tmp4fMhYi.lastz\n",
      "\t/tmp/tmpOk8c9u.lastz\n",
      "\t/tmp/tmpn9Aezt.lastz\n",
      "\t/tmp/tmpvNQJjo.lastz\n",
      "\t/tmp/tmpP0ugdu.lastz\n",
      "\t/tmp/tmpaByFsP.lastz\n",
      "\t/tmp/tmpNrkqkl.lastz\n",
      "\t/tmp/tmpCAxkxm.lastz\n",
      "\t/tmp/tmpIE_ii3.lastz\n",
      "\t/tmp/tmpi7KMiM.lastz\n",
      "\t/tmp/tmpzc1MG8.lastz\n",
      "\t/tmp/tmpYrlMLi.lastz\n",
      "\t/tmp/tmpIpdYL5.lastz\n",
      "\t/tmp/tmpmbkZYS.lastz\n",
      "\t/tmp/tmpkYwQgb.lastz\n",
      "\t/tmp/tmpbq09Dv.lastz\n",
      "\t/tmp/tmpBD9CSY.lastz\n",
      "\t/tmp/tmpQmpxsO.lastz\n",
      "\t/tmp/tmpV6clrj.lastz\n",
      "\t/tmp/tmpN0HB61.lastz\n",
      "\t/tmp/tmp3cc9AE.lastz\n",
      "\t/tmp/tmpGWxJnk.lastz\n",
      "\t/tmp/tmp2k2Zb3.lastz\n",
      "\t/tmp/tmpdSQG73.lastz\n",
      "\t/tmp/tmpgdvy3b.lastz\n",
      "\t/tmp/tmpSgnDqk.lastz\n",
      "\t/tmp/tmpxsq0Vr.lastz\n",
      "\t/tmp/tmpJ0jiO1.lastz\n",
      "\t/tmp/tmp7nXx26.lastz\n",
      "\t/tmp/tmpXJ76BJ.lastz\n",
      "\t/tmp/tmplmXXVP.lastz\n",
      "\t/tmp/tmpMkzM1B.lastz\n",
      "\t/tmp/tmpHSSsQS.lastz\n",
      "\t/tmp/tmp4s5uBI.lastz\n",
      "\t/tmp/tmpGwc4uA.lastz\n",
      "\t/tmp/tmpL7mvO8.lastz\n",
      "\t/tmp/tmp4qH53s.lastz\n",
      "\t/tmp/tmpyJjh18.lastz\n",
      "\t/tmp/tmpZ3qYAe.lastz\n",
      "\t/tmp/tmpIac7Yp.lastz\n",
      "\t/tmp/tmpO57gbn.lastz\n",
      "\t/tmp/tmpy6gXYm.lastz\n",
      "\t/tmp/tmpor9R3y.lastz\n",
      "\t/tmp/tmpjNneG5.lastz\n",
      "\t/tmp/tmp2gzFZ9.lastz\n",
      "\t/tmp/tmpDdCGvE.lastz\n",
      "\t/tmp/tmpW9fw1h.lastz\n",
      "\t/tmp/tmpYGsilS.lastz\n",
      "\t/tmp/tmpQeeo6h.lastz\n",
      "\t/tmp/tmprThz0h.lastz\n",
      "\t/tmp/tmp_E9ZNt.lastz\n",
      "\t/tmp/tmpBQM5J_.lastz\n",
      "\t/tmp/tmpMYJsfc.lastz\n",
      "\t/tmp/tmpHqZTEx.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Spirometra_erinaceieuropaei_GCA_000951995.lastz\n",
      "Creating Spirometra_erinaceieuropaei_GCA_000951995 table\n",
      "Inserting data to Spirometra_erinaceieuropaei_GCA_000951995 table\n",
      "\n",
      "Running against Taenia_asiatica_GCA_001693035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 17 queries...\n",
      "\t/tmp/tmpzO_lrh.fasta\n",
      "\t/tmp/tmpl60uqi.fasta\n",
      "\t/tmp/tmphtlpvg.fasta\n",
      "\t/tmp/tmpikXAEX.fasta\n",
      "\t/tmp/tmpu0u55i.fasta\n",
      "\t/tmp/tmpLR5e4n.fasta\n",
      "\t/tmp/tmpqcjaHs.fasta\n",
      "\t/tmp/tmpUT7q5Q.fasta\n",
      "\t/tmp/tmpfQTcDX.fasta\n",
      "\t/tmp/tmp6ziVjn.fasta\n",
      "\t/tmp/tmp3zI_TI.fasta\n",
      "\t/tmp/tmpjf8564.fasta\n",
      "\t/tmp/tmpk8ls5K.fasta\n",
      "\t/tmp/tmp0l8XC_.fasta\n",
      "\t/tmp/tmp3zY9hZ.fasta\n",
      "\t/tmp/tmpJt0pno.fasta\n",
      "\t/tmp/tmp2DkN2D.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp79n__V.lastz\n",
      "\t/tmp/tmp35Mliw.lastz\n",
      "\t/tmp/tmpcs1Vll.lastz\n",
      "\t/tmp/tmp8Jcwiv.lastz\n",
      "\t/tmp/tmpU9eZ9D.lastz\n",
      "\t/tmp/tmpd137Dd.lastz\n",
      "\t/tmp/tmpOKLyjz.lastz\n",
      "\t/tmp/tmp63yrTw.lastz\n",
      "\t/tmp/tmpBFUcvA.lastz\n",
      "\t/tmp/tmpD3uP6j.lastz\n",
      "\t/tmp/tmprrje8A.lastz\n",
      "\t/tmp/tmpYR4kRH.lastz\n",
      "\t/tmp/tmpWNwfIa.lastz\n",
      "\t/tmp/tmpV9YVRt.lastz\n",
      "\t/tmp/tmpWdX5pN.lastz\n",
      "\t/tmp/tmpz_5r1q.lastz\n",
      "\t/tmp/tmpbOASZR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Taenia_asiatica_GCA_001693035.lastz\n",
      "Creating Taenia_asiatica_GCA_001693035 table\n",
      "Inserting data to Taenia_asiatica_GCA_001693035 table\n",
      "\n",
      "Running against Taenia_saginata_GCA_001693075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 16 queries...\n",
      "\t/tmp/tmp5ZMxa2.fasta\n",
      "\t/tmp/tmpPTwqBg.fasta\n",
      "\t/tmp/tmpF4LdyW.fasta\n",
      "\t/tmp/tmpnC5Up9.fasta\n",
      "\t/tmp/tmpO_RB9i.fasta\n",
      "\t/tmp/tmpZ56Jbo.fasta\n",
      "\t/tmp/tmpR55YFL.fasta\n",
      "\t/tmp/tmp89py3h.fasta\n",
      "\t/tmp/tmp1KSsnC.fasta\n",
      "\t/tmp/tmphNJ5RP.fasta\n",
      "\t/tmp/tmp8tY9hJ.fasta\n",
      "\t/tmp/tmpDHbCSQ.fasta\n",
      "\t/tmp/tmpzSJZqA.fasta\n",
      "\t/tmp/tmpFu4Njf.fasta\n",
      "\t/tmp/tmpdpabCV.fasta\n",
      "\t/tmp/tmpDwc64a.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdyA0ba.lastz\n",
      "\t/tmp/tmpAO2JAh.lastz\n",
      "\t/tmp/tmpdr9zXD.lastz\n",
      "\t/tmp/tmpRRgI4C.lastz\n",
      "\t/tmp/tmpiyLLbN.lastz\n",
      "\t/tmp/tmpG_ILGw.lastz\n",
      "\t/tmp/tmppzusU_.lastz\n",
      "\t/tmp/tmprlWAdj.lastz\n",
      "\t/tmp/tmpVj1sAz.lastz\n",
      "\t/tmp/tmpckAbbL.lastz\n",
      "\t/tmp/tmpqxw9lt.lastz\n",
      "\t/tmp/tmpJGyxpn.lastz\n",
      "\t/tmp/tmpG8jmob.lastz\n",
      "\t/tmp/tmpObplsX.lastz\n",
      "\t/tmp/tmpCvy8K4.lastz\n",
      "\t/tmp/tmp5QVgXd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Taenia_saginata_GCA_001693075.lastz\n",
      "Creating Taenia_saginata_GCA_001693075 table\n",
      "Inserting data to Taenia_saginata_GCA_001693075 table\n",
      "\n",
      "Running against Taenia_solium_GCA_001870725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 13 queries...\n",
      "\t/tmp/tmpfCUfiu.fasta\n",
      "\t/tmp/tmpxBrNTG.fasta\n",
      "\t/tmp/tmpFfAukZ.fasta\n",
      "\t/tmp/tmpGgO_x2.fasta\n",
      "\t/tmp/tmpkDnjSI.fasta\n",
      "\t/tmp/tmpexlNm8.fasta\n",
      "\t/tmp/tmpUf1Mqf.fasta\n",
      "\t/tmp/tmpenbxup.fasta\n",
      "\t/tmp/tmpqHEzLr.fasta\n",
      "\t/tmp/tmpURnx_v.fasta\n",
      "\t/tmp/tmpLCqmrd.fasta\n",
      "\t/tmp/tmpVgeOQv.fasta\n",
      "\t/tmp/tmp2hvWID.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4IJ2Cg.lastz\n",
      "\t/tmp/tmpiHZVJM.lastz\n",
      "\t/tmp/tmpkE9QVu.lastz\n",
      "\t/tmp/tmpKE3ere.lastz\n",
      "\t/tmp/tmpia9z0R.lastz\n",
      "\t/tmp/tmpXQzf7E.lastz\n",
      "\t/tmp/tmpDBf8xv.lastz\n",
      "\t/tmp/tmp3mDzny.lastz\n",
      "\t/tmp/tmpaPSFX7.lastz\n",
      "\t/tmp/tmpRsEU1U.lastz\n",
      "\t/tmp/tmpl3GYrO.lastz\n",
      "\t/tmp/tmpgrve3t.lastz\n",
      "\t/tmp/tmpbslF0O.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Taenia_solium_GCA_001870725.lastz\n",
      "Creating Taenia_solium_GCA_001870725 table\n",
      "Inserting data to Taenia_solium_GCA_001870725 table\n",
      "\n",
      "Running against Taenia_multiceps_GCA_001923025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmp9zgGHz.fasta\n",
      "\t/tmp/tmpQFIfnD.fasta\n",
      "\t/tmp/tmp6RLylG.fasta\n",
      "\t/tmp/tmpP1W03y.fasta\n",
      "\t/tmp/tmpx2lCcK.fasta\n",
      "\t/tmp/tmpw1pgtj.fasta\n",
      "\t/tmp/tmplz6lU8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd5O2Df.lastz\n",
      "\t/tmp/tmpKTOlfG.lastz\n",
      "\t/tmp/tmpToXVgQ.lastz\n",
      "\t/tmp/tmpxwEYb6.lastz\n",
      "\t/tmp/tmpF65bu2.lastz\n",
      "\t/tmp/tmpJrizui.lastz\n",
      "\t/tmp/tmp_IOxPZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/cestoda/validate_intervals/lastz/Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_Taenia_multiceps_GCA_001923025.lastz\n",
      "Creating Taenia_multiceps_GCA_001923025 table\n",
      "Inserting data to Taenia_multiceps_GCA_001923025 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/cestoda/extract_probes_from_group/cestoda_genome.conf --lastz results/phyluce/cestoda/validate_intervals/lastz --probes 120 --probe-prefix uce_cestoda_ --name-pattern \"Taenia_multiceps_GCA_001923025_+8_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/cestoda/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 14:50:49,338 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 14:50:49,343 - Phyluce - INFO - ---- Working on Dibothriocephalus_latus_GCA_900617775 genome ----\n",
      "2020-02-11 14:50:49,344 - Phyluce - INFO - Reading Dibothriocephalus_latus_GCA_900617775 genome\n",
      "2020-02-11 14:51:02,766 - Phyluce - INFO - Dibothriocephalus_latus_GCA_900617775: 1907 uces, 390 dupes, 1517 non-dupes, 0 orient drop, 1 length drop, 1513 written\n",
      "2020-02-11 14:51:02,766 - Phyluce - INFO - ---- Working on Echinococcus_canadensis_GCA_900004735 genome ----\n",
      "2020-02-11 14:51:02,803 - Phyluce - INFO - Reading Echinococcus_canadensis_GCA_900004735 genome\n",
      "2020-02-11 14:51:17,651 - Phyluce - INFO - Echinococcus_canadensis_GCA_900004735: 3349 uces, 390 dupes, 2959 non-dupes, 8 orient drop, 30 length drop, 2921 written\n",
      "2020-02-11 14:51:17,652 - Phyluce - INFO - ---- Working on Echinococcus_granulosus_GCF_000524195 genome ----\n",
      "2020-02-11 14:51:17,668 - Phyluce - INFO - Reading Echinococcus_granulosus_GCF_000524195 genome\n",
      "2020-02-11 14:51:31,812 - Phyluce - INFO - Echinococcus_granulosus_GCF_000524195: 3343 uces, 370 dupes, 2973 non-dupes, 9 orient drop, 87 length drop, 2877 written\n",
      "2020-02-11 14:51:31,813 - Phyluce - INFO - -- Working on Echinococcus_multilocularis_GCA_000469725 genome --\n",
      "2020-02-11 14:51:31,825 - Phyluce - INFO - Reading Echinococcus_multilocularis_GCA_000469725 genome\n",
      "2020-02-11 14:51:46,398 - Phyluce - INFO - Echinococcus_multilocularis_GCA_000469725: 3348 uces, 353 dupes, 2995 non-dupes, 62 orient drop, 58 length drop, 2875 written\n",
      "2020-02-11 14:51:46,399 - Phyluce - INFO - ---- Working on Echinococcus_oligarthrus_GCA_900683695 genome ---\n",
      "2020-02-11 14:51:46,404 - Phyluce - INFO - Reading Echinococcus_oligarthrus_GCA_900683695 genome\n",
      "2020-02-11 14:51:58,848 - Phyluce - INFO - Echinococcus_oligarthrus_GCA_900683695: 3070 uces, 294 dupes, 2776 non-dupes, 1 orient drop, 5 length drop, 2724 written\n",
      "2020-02-11 14:51:58,848 - Phyluce - INFO - ---- Working on Hydatigera_taeniaeformis_GCA_900622495 genome ---\n",
      "2020-02-11 14:51:58,854 - Phyluce - INFO - Reading Hydatigera_taeniaeformis_GCA_900622495 genome\n",
      "2020-02-11 14:52:13,805 - Phyluce - INFO - Hydatigera_taeniaeformis_GCA_900622495: 3345 uces, 400 dupes, 2945 non-dupes, 3 orient drop, 11 length drop, 2931 written\n",
      "2020-02-11 14:52:13,806 - Phyluce - INFO - ------ Working on Hymenolepis_diminuta_GCA_900708905 genome -----\n",
      "2020-02-11 14:52:13,849 - Phyluce - INFO - Reading Hymenolepis_diminuta_GCA_900708905 genome\n",
      "2020-02-11 14:52:24,657 - Phyluce - INFO - Hymenolepis_diminuta_GCA_900708905: 2603 uces, 377 dupes, 2226 non-dupes, 22 orient drop, 36 length drop, 2168 written\n",
      "2020-02-11 14:52:24,657 - Phyluce - INFO - ----- Working on Hymenolepis_microstoma_GCA_000469805 genome ----\n",
      "2020-02-11 14:52:24,674 - Phyluce - INFO - Reading Hymenolepis_microstoma_GCA_000469805 genome\n",
      "2020-02-11 14:52:37,817 - Phyluce - INFO - Hymenolepis_microstoma_GCA_000469805: 2556 uces, 305 dupes, 2251 non-dupes, 63 orient drop, 75 length drop, 2113 written\n",
      "2020-02-11 14:52:37,817 - Phyluce - INFO - ------ Working on Mesocestoides_corti_GCA_900604375 genome ------\n",
      "2020-02-11 14:52:38,405 - Phyluce - INFO - Reading Mesocestoides_corti_GCA_900604375 genome\n",
      "2020-02-11 14:52:49,810 - Phyluce - INFO - Mesocestoides_corti_GCA_900604375: 2823 uces, 384 dupes, 2439 non-dupes, 3 orient drop, 17 length drop, 2418 written\n",
      "2020-02-11 14:52:49,810 - Phyluce - INFO - ------- Working on Rodentolepis_nana_GCA_900617975 genome -------\n",
      "2020-02-11 14:52:49,817 - Phyluce - INFO - Reading Rodentolepis_nana_GCA_900617975 genome\n",
      "2020-02-11 14:53:00,424 - Phyluce - INFO - Rodentolepis_nana_GCA_900617975: 2565 uces, 419 dupes, 2146 non-dupes, 1 orient drop, 11 length drop, 2133 written\n",
      "2020-02-11 14:53:00,424 - Phyluce - INFO - ---- Working on Schistocephalus_solidus_GCA_900618435 genome ----\n",
      "2020-02-11 14:53:00,441 - Phyluce - INFO - Reading Schistocephalus_solidus_GCA_900618435 genome\n",
      "2020-02-11 14:53:10,363 - Phyluce - INFO - Schistocephalus_solidus_GCA_900618435: 2064 uces, 430 dupes, 1634 non-dupes, 1 orient drop, 12 length drop, 1621 written\n",
      "2020-02-11 14:53:10,364 - Phyluce - INFO - ------ Working on Sparganum_proliferum_GCA_902702955 genome -----\n",
      "2020-02-11 14:53:10,371 - Phyluce - INFO - Reading Sparganum_proliferum_GCA_902702955 genome\n",
      "2020-02-11 14:53:20,583 - Phyluce - INFO - Sparganum_proliferum_GCA_902702955: 2064 uces, 456 dupes, 1608 non-dupes, 10 orient drop, 37 length drop, 1561 written\n",
      "2020-02-11 14:53:20,583 - Phyluce - INFO - -- Working on Spirometra_erinaceieuropaei_GCA_000951995 genome --\n",
      "2020-02-11 14:53:20,593 - Phyluce - INFO - Reading Spirometra_erinaceieuropaei_GCA_000951995 genome\n",
      "2020-02-11 14:53:44,794 - Phyluce - INFO - Spirometra_erinaceieuropaei_GCA_000951995: 1985 uces, 1154 dupes, 831 non-dupes, 1 orient drop, 1 length drop, 823 written\n",
      "2020-02-11 14:53:44,795 - Phyluce - INFO - -------- Working on Taenia_asiatica_GCA_001693035 genome --------\n",
      "2020-02-11 14:53:44,802 - Phyluce - INFO - Reading Taenia_asiatica_GCA_001693035 genome\n",
      "2020-02-11 14:54:00,748 - Phyluce - INFO - Taenia_asiatica_GCA_001693035: 3349 uces, 434 dupes, 2915 non-dupes, 25 orient drop, 42 length drop, 2848 written\n",
      "2020-02-11 14:54:00,748 - Phyluce - INFO - -------- Working on Taenia_saginata_GCA_001693075 genome --------\n",
      "2020-02-11 14:54:00,753 - Phyluce - INFO - Reading Taenia_saginata_GCA_001693075 genome\n",
      "2020-02-11 14:54:16,231 - Phyluce - INFO - Taenia_saginata_GCA_001693075: 3349 uces, 386 dupes, 2963 non-dupes, 14 orient drop, 39 length drop, 2910 written\n",
      "2020-02-11 14:54:16,231 - Phyluce - INFO - --------- Working on Taenia_solium_GCA_001870725 genome ---------\n",
      "2020-02-11 14:54:16,237 - Phyluce - INFO - Reading Taenia_solium_GCA_001870725 genome\n",
      "2020-02-11 14:54:31,106 - Phyluce - INFO - Taenia_solium_GCA_001870725: 3349 uces, 394 dupes, 2955 non-dupes, 14 orient drop, 27 length drop, 2914 written\n",
      "2020-02-11 14:54:31,106 - Phyluce - INFO - -------- Working on Taenia_multiceps_GCA_001923025 genome -------\n",
      "2020-02-11 14:54:31,111 - Phyluce - INFO - Reading Taenia_multiceps_GCA_001923025 genome\n",
      "2020-02-11 14:54:48,138 - Phyluce - INFO - Taenia_multiceps_GCA_001923025: 3349 uces, 210 dupes, 3139 non-dupes, 155 orient drop, 112 length drop, 2872 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/cestoda/extract_probes_from_group/probe_fasta --output results/phyluce/cestoda/extract_probes_from_group/multifastas.sqlite --base-taxon Taenia_multiceps_GCA_001923025;\n",
      "dibothriocephalus_latus_gca_900617775..\n",
      "echinococcus_canadensis_gca_900004735...\n",
      "echinococcus_granulosus_gcf_000524195...\n",
      "echinococcus_multilocularis_gca_000469725...\n",
      "echinococcus_oligarthrus_gca_900683695...\n",
      "hydatigera_taeniaeformis_gca_900622495...\n",
      "hymenolepis_diminuta_gca_900708905...\n",
      "hymenolepis_microstoma_gca_000469805...\n",
      "mesocestoides_corti_gca_900604375...\n",
      "rodentolepis_nana_gca_900617975...\n",
      "schistocephalus_solidus_gca_900618435..\n",
      "sparganum_proliferum_gca_902702955..\n",
      "spirometra_erinaceieuropaei_gca_000951995.\n",
      "taenia_asiatica_gca_001693035...\n",
      "taenia_saginata_gca_001693075...\n",
      "taenia_solium_gca_001870725...\n",
      "taenia_multiceps_gca_001923025...\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/cestoda/extract_probes_from_group/multifastas.sqlite --base-taxon Taenia_multiceps_GCA_001923025;\n",
      "Loci shared by 0 taxa:\t3,193.0\n",
      "Loci shared by 1 taxa:\t3,193.0\n",
      "Loci shared by 2 taxa:\t3,122.0\n",
      "Loci shared by 3 taxa:\t3,093.0\n",
      "Loci shared by 4 taxa:\t3,064.0\n",
      "Loci shared by 5 taxa:\t3,033.0\n",
      "Loci shared by 6 taxa:\t3,010.0\n",
      "Loci shared by 7 taxa:\t2,976.0\n",
      "Loci shared by 8 taxa:\t2,942.0\n",
      "Loci shared by 9 taxa:\t2,873.0\n",
      "Loci shared by 10 taxa:\t2,467.0\n",
      "Loci shared by 11 taxa:\t2,253.0\n",
      "Loci shared by 12 taxa:\t2,135.0\n",
      "Loci shared by 13 taxa:\t1,938.0\n",
      "Loci shared by 14 taxa:\t1,566.0\n",
      "Loci shared by 15 taxa:\t1,311.0\n",
      "Loci shared by 16 taxa:\t912.0\n",
      "Loci shared by 17 taxa:\t334.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/cestoda/extract_probes_from_group/multifastas.sqlite  --base-taxon Taenia_multiceps_GCA_001923025 --output results/phyluce/cestoda/extract_probes_from_group/Taenia_multiceps_GCA_001923025+8-back-to-17.conf --specific-counts 17;\n",
      "Counter({'dibothriocephalus_latus_gca_900617775': 334, 'echinococcus_canadensis_gca_900004735': 334, 'taenia_saginata_gca_001693075': 334, 'echinococcus_multilocularis_gca_000469725': 334, 'echinococcus_oligarthrus_gca_900683695': 334, 'hymenolepis_diminuta_gca_900708905': 334, 'hymenolepis_microstoma_gca_000469805': 334, 'rodentolepis_nana_gca_900617975': 334, 'taenia_multiceps_gca_001923025': 334, 'mesocestoides_corti_gca_900604375': 334, 'schistocephalus_solidus_gca_900618435': 334, 'echinococcus_granulosus_gcf_000524195': 334, 'sparganum_proliferum_gca_902702955': 334, 'taenia_solium_gca_001870725': 334, 'taenia_asiatica_gca_001693035': 334, 'spirometra_erinaceieuropaei_gca_000951995': 334, 'hydatigera_taeniaeformis_gca_900622495': 334})\n",
      "Total loci = 334\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(17)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/cestoda/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/cestoda/extract_probes_from_group/Taenia_multiceps_GCA_001923025+8-back-to-17.conf --probe-prefix uce_cestoda_ --designer rnplattii --design cestoda_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list.fasta;\n",
      "NNNNNNGGGGGGGNGGGGNGG\n",
      "\n",
      "\n",
      "Conserved locus count = 334\n",
      "Probe Count = 11335\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list.fasta --query results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  14:55:22\n",
      "Ended:  Tue Feb 11, 2020  14:55:40\n",
      "Time for execution:  0.292155579726 minutes\n"
     ]
    }
   ],
   "source": [
    "i_design_txt   = group + \"_v1\"\n",
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "i_probes_fas   = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list.fasta --lastz results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_cestoda_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 11334 fasta sequences.  Filtered 0 duplicates. Kept 11335.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "\n",
    "#i_lastz_tbl = o_lastz_tbl\n",
    "#i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/cestoda/final_probe_design/cestoda_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 15:01:48 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 11335\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 906800\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 2M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 89M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3945719\n",
      "Max number of word counting entries: 88818154\n",
      "\n",
      "# comparing sequences from          0  to       1889\n",
      ".---------- new table with     1201 representatives\n",
      "# comparing sequences from       1889  to       3463\n",
      "----------    360 remaining sequences to the next cycle\n",
      "---------- new table with      776 representatives\n",
      "# comparing sequences from       3103  to       4475\n",
      "----------    239 remaining sequences to the next cycle\n",
      "---------- new table with      746 representatives\n",
      "# comparing sequences from       4236  to       5419\n",
      "----------    175 remaining sequences to the next cycle\n",
      "---------- new table with      665 representatives\n",
      "# comparing sequences from       5244  to       6259\n",
      "----------    178 remaining sequences to the next cycle\n",
      "---------- new table with      548 representatives\n",
      "# comparing sequences from       6081  to       6956\n",
      "----------    175 remaining sequences to the next cycle\n",
      "---------- new table with      441 representatives\n",
      "# comparing sequences from       6781  to       7540\n",
      "----------    215 remaining sequences to the next cycle\n",
      "---------- new table with      347 representatives\n",
      "# comparing sequences from       7325  to       7993\n",
      "----------    198 remaining sequences to the next cycle\n",
      "---------- new table with      316 representatives\n",
      "# comparing sequences from       7795  to       8385\n",
      "----------    134 remaining sequences to the next cycle\n",
      "---------- new table with      284 representatives\n",
      "# comparing sequences from       8251  to       8765\n",
      "----------    179 remaining sequences to the next cycle\n",
      "---------- new table with      229 representatives\n",
      "# comparing sequences from       8586  to       9044\n",
      "----------     56 remaining sequences to the next cycle\n",
      "---------- new table with      261 representatives\n",
      "# comparing sequences from       8988  to       9379\n",
      "----------    145 remaining sequences to the next cycle\n",
      "---------- new table with      161 representatives\n",
      "# comparing sequences from       9234  to       9584\n",
      "----------    123 remaining sequences to the next cycle\n",
      "---------- new table with      134 representatives\n",
      "# comparing sequences from       9461  to       9773\n",
      "----------    105 remaining sequences to the next cycle\n",
      "---------- new table with      131 representatives\n",
      "# comparing sequences from       9668  to       9945\n",
      "99.9%---------- new table with      179 representatives\n",
      "# comparing sequences from       9945  to      10176\n",
      "..........    10000  finished       6454  clusters\n",
      "----------     66 remaining sequences to the next cycle\n",
      "---------- new table with      107 representatives\n",
      "# comparing sequences from      10110  to      10314\n",
      "100.0%---------- new table with      130 representatives\n",
      "# comparing sequences from      10314  to      10484\n",
      "99.9%---------- new table with      115 representatives\n",
      "# comparing sequences from      10484  to      11335\n",
      ".....................---------- new table with      545 representatives\n",
      "\n",
      "    11335  finished       7316  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 90M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.83\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Eurotiales"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'eurotiales'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Asprgillus_aculeatinus_GCF_003184765\",\n",
    "               \"Asprgillus_aculeatus_GCF_001890905\",\n",
    "               \"Asprgillus_aff_GCA_001931935\",\n",
    "               \"Asprgillus_albertensis_GCA_009193605\",\n",
    "               \"Asprgillus_alliaceus_GCF_009176365\",\n",
    "               \"Asprgillus_amoenus_GCA_009812435\",\n",
    "               \"Asprgillus_arachidicola_GCA_009193545\",\n",
    "               \"Asprgillus_avenaceus_GCA_009193465\",\n",
    "               \"Asprgillus_awamori_GCA_003850985\",\n",
    "               \"Asprgillus_bertholletius_GCA_009193595\",\n",
    "               \"Asprgillus_bombycis_GCF_001792695\",\n",
    "               \"Asprgillus_brasiliensis_GCA_001889945\",\n",
    "               \"Asprgillus_brunneoviolaceus_GCF_003184695\",\n",
    "               \"Asprgillus_caelatus_GCF_009193585\",\n",
    "               \"Asprgillus_calidoustus_GCA_001511075\",\n",
    "               \"Asprgillus_campestris_GCF_002847485\",\n",
    "               \"Asprgillus_candidus_GCF_002847045\",\n",
    "               \"Asprgillus_carbonarius_GCA_001990825\",\n",
    "               \"Asprgillus_cejpii_GCA_004769165\",\n",
    "               \"Asprgillus_chevalieri_GCA_001599875\",\n",
    "               \"Asprgillus_clavatus_GCF_000002715\",\n",
    "               \"Asprgillus_coremiiformis_GCA_009193565\",\n",
    "               \"Asprgillus_costaricaensis_GCF_003184835\",\n",
    "               \"Asprgillus_cristatus_GCA_001693355\",\n",
    "               \"Asprgillus_ellipticus_GCA_003184645\",\n",
    "               \"Asprgillus_eucalypticola_GCF_003184535\",\n",
    "               \"Asprgillus_fijiensis_GCA_003184825\",\n",
    "               \"Asprgillus_fischeri_GCF_000149645\",\n",
    "               \"Asprgillus_flavus_GCF_000006275\",\n",
    "               \"Asprgillus_fumigatus_GCF_000002655\",\n",
    "               \"Asprgillus_glaucus_GCF_001890805\",\n",
    "               \"Asprgillus_hancockii_GCA_001696595\",\n",
    "               \"Asprgillus_heteromorphus_GCF_003184545\",\n",
    "               \"Asprgillus_homomorphus_GCF_003184865\",\n",
    "               \"Asprgillus_ibericus_GCF_003184845\",\n",
    "               \"Asprgillus_indologenus_GCA_003184685\",\n",
    "               \"Asprgillus_japonicus_GCF_003184785\",\n",
    "               \"Asprgillus_kawachii_GCA_000239835\",\n",
    "               \"Asprgillus_lentulus_GCA_001445615\",\n",
    "               \"Asprgillus_leporis_GCA_009176345\",\n",
    "               \"Asprgillus_luchuensis_GCA_001890685\",\n",
    "               \"Asprgillus_minisclerotigenes_GCA_009176455\",\n",
    "               \"Asprgillus_mulundensis_GCF_003369625\",\n",
    "               \"Asprgillus_neoellipticus_GCA_003116565\",\n",
    "               \"Asprgillus_neoniger_GCF_003184625\",\n",
    "               \"Asprgillus_nidulans_GCF_000149205\",\n",
    "               \"Asprgillus_niger_GCF_000002855\",\n",
    "               \"Asprgillus_nomius_GCF_001204775\",\n",
    "               \"Asprgillus_novofumigatus_GCF_002847465\",\n",
    "               \"Asprgillus_novoparasiticus_GCA_009176405\",\n",
    "               \"Asprgillus_ochraceoroseus_GCA_002846915\",\n",
    "               \"Asprgillus_ochraceus_GCA_004849945\",\n",
    "               \"Asprgillus_olivimuriae_GCA_003719415\",\n",
    "               \"Asprgillus_oryzae_GCF_000184455\",\n",
    "               \"Asprgillus_parasiticus_GCA_009176385\",\n",
    "               \"Asprgillus_persii_GCA_002215965\",\n",
    "               \"Asprgillus_phoenicis_GCA_003344505\",\n",
    "               \"Asprgillus_piperis_GCF_003184755\",\n",
    "               \"Asprgillus_pseudocaelatus_GCA_009193665\",\n",
    "               \"Asprgillus_pseudotamarii_GCF_009193445\",\n",
    "               \"Asprgillus_pseudoterreus_GCA_002927005\",\n",
    "               \"Asprgillus_rambellii_GCA_000986645\",\n",
    "               \"Asprgillus_ruber_GCA_000600275\",\n",
    "               \"Asprgillus_saccharolyticus_GCF_003184585\",\n",
    "               \"Asprgillus_sclerotialis_GCA_003589665\",\n",
    "               \"Asprgillus_sclerotiicarbonarius_GCA_003184635\",\n",
    "               \"Asprgillus_sclerotioniger_GCF_003184525\",\n",
    "               \"Asprgillus_sclerotiorum_GCA_000530345\",\n",
    "               \"Asprgillus_sergii_GCA_009193525\",\n",
    "               \"Asprgillus_sojae_GCA_008274985\",\n",
    "               \"Asprgillus_sp_GCA_001044295\",\n",
    "               \"Asprgillus_spnulosprus_GCA_003574815\",\n",
    "               \"Asprgillus_steynii_GCF_002849105\",\n",
    "               \"Asprgillus_sydowii_GCA_001890705\",\n",
    "               \"Asprgillus_taichungensis_GCA_002850765\",\n",
    "               \"Asprgillus_tamarii_GCA_009193485\",\n",
    "               \"Asprgillus_tanneri_GCA_003426965\",\n",
    "               \"Asprgillus_terreus_GCF_000149615\",\n",
    "               \"Asprgillus_thermomutatus_GCF_002237265\",\n",
    "               \"Asprgillus_transmontanensis_GCA_009193505\",\n",
    "               \"Asprgillus_tritici_GCA_009812425\",\n",
    "               \"Asprgillus_tubingensis_GCA_001890745\",\n",
    "               \"Asprgillus_turcosus_GCA_002234965\",\n",
    "               \"Asprgillus_udagawae_GCA_001078395\",\n",
    "               \"Asprgillus_unguis_GCA_003324175\",\n",
    "               \"Asprgillus_ustus_GCA_000812125\",\n",
    "               \"Asprgillus_uvarum_GCF_003184745\",\n",
    "               \"Asprgillus_vadensis_GCF_003184925\",\n",
    "               \"Asprgillus_versicolor_GCA_001890125\",\n",
    "               \"Asprgillus_violaceofuscus_GCA_003184705\",\n",
    "               \"Asprgillus_viridinutans_GCA_004368095\",\n",
    "               \"Asprgillus_welwitschiae_GCF_003344945\",\n",
    "               \"Asprgillus_wentii_GCA_001890725\",\n",
    "               \"Asprgillus_westerdijkiae_GCA_001307345\",\n",
    "               \"Basipetospra_chlamydospra_GCA_001599675\",\n",
    "               \"Byssochlamys_nivea_GCA_003116535\",\n",
    "               \"Byssochlamys_sp_GCA_002914405\",\n",
    "               \"Byssochlamys_spctabilis_GCF_004022145\",\n",
    "               \"Elaphomyces_granulatus_GCA_002240705\",\n",
    "               \"Monascus_purpureus_GCA_003184285\",\n",
    "               \"Monascus_ruber_GCA_002976275\",\n",
    "               \"Penicilliopsis_zonata_GCF_001890105\",\n",
    "               \"Penicillium_antarcticum_GCA_002072345\",\n",
    "               \"Penicillium_arizonense_GCF_001773325\",\n",
    "               \"Penicillium_biforme_GCA_000577785\",\n",
    "               \"Penicillium_brasilianum_GCA_001048715\",\n",
    "               \"Penicillium_camemberti_GCA_000513335\",\n",
    "               \"Penicillium_capsulatum_GCA_000943775\",\n",
    "               \"Penicillium_carneum_GCA_000577495\",\n",
    "               \"Penicillium_chrysogenum_GCA_000710275\",\n",
    "               \"Penicillium_citrinum_GCA_001950535\",\n",
    "               \"Penicillium_coprophilum_GCA_002072405\",\n",
    "               \"Penicillium_decumbens_GCA_002072245\",\n",
    "               \"Penicillium_digitatum_GCF_000315645\",\n",
    "               \"Penicillium_expansum_GCF_000769745\",\n",
    "               \"Penicillium_flavigenum_GCA_002072365\",\n",
    "               \"Penicillium_freii_GCA_001513925\",\n",
    "               \"Penicillium_fuscoglaucum_GCA_000576735\",\n",
    "               \"Penicillium_griseofulvum_GCA_001561935\",\n",
    "               \"Penicillium_italicum_GCA_002116305\",\n",
    "               \"Penicillium_janthinellum_GCA_002369805\",\n",
    "               \"Penicillium_nalgiovense_GCA_002072425\",\n",
    "               \"Penicillium_nordicum_GCA_000733025\",\n",
    "               \"Penicillium_oxalicum_GCA_000346795\",\n",
    "               \"Penicillium_paneum_GCA_000577715\",\n",
    "               \"Penicillium_paxilli_GCA_000347475\",\n",
    "               \"Penicillium_polonicum_GCA_003344595\",\n",
    "               \"Penicillium_roqueforti_GCA_000513255\",\n",
    "               \"Penicillium_rubens_GCF_000226395\",\n",
    "               \"Penicillium_sclerotiorum_GCA_001750025\",\n",
    "               \"Penicillium_solitum_GCA_000952775\",\n",
    "               \"Penicillium_sp_GCA_003800485\",\n",
    "               \"Penicillium_steckii_GCA_002072375\",\n",
    "               \"Penicillium_subrubescens_GCA_001908125\",\n",
    "               \"Penicillium_verrucosum_GCA_000970515\",\n",
    "               \"Penicillium_vulpinum_GCA_002072255\",\n",
    "               \"Phialosimplex_sp_GCA_003698115\",\n",
    "               \"Rasamsonia_emersonii_GCF_000968595\",\n",
    "               \"Talaromyces_adpressus_GCA_002775195\",\n",
    "               \"Talaromyces_amestolkiae_GCA_001896365\",\n",
    "               \"Talaromyces_atroroseus_GCF_001907595\",\n",
    "               \"Talaromyces_borbonicus_GCA_002916415\",\n",
    "               \"Talaromyces_cellulolyticus_GCA_000829775\",\n",
    "               \"Talaromyces_funiculosus_GCA_004299765\",\n",
    "               \"Talaromyces_islandicus_GCA_000985935\",          \n",
    "               \"Talaromyces_piceae_GCA_001657655\",\n",
    "               \"Talaromyces_pinophilus_GCA_001571465\",\n",
    "               \"Talaromyces_purpureogenus_GCA_001270325\",\n",
    "               \"Talaromyces_stipitatus_GCF_000003125\",\n",
    "               \"Talaromyces_verruculosus_GCA_001305275\",\n",
    "               \"Talaromyces_wortmannii_GCA_001939245\",\n",
    "               \"Thermoascaceae_sp_GCA_003123655\",\n",
    "               \"Thermoascus_crustaceus_GCA_001599835\",\n",
    "               \"Thermomyces_lanuginosus_GCA_000315935\",\n",
    "               \"Xeromyces_bisprus_GCA_900006255\" ]\n",
    "                    \n",
    "reference_taxon = \"Talaromyces_marneffei_GCF_000001985\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184765.1_Aspacu1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11418006 bytes  4567219.20 bytes/sec\n",
      "total size is 11415111  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890905.1_Aspac1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11112183 bytes  4444890.00 bytes/sec\n",
      "total size is 11109361  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001931935.1_ASM193193v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9969599 bytes  3987856.40 bytes/sec\n",
      "total size is 9967052  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193605.1_Aspalbe1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12643049 bytes  3612311.71 bytes/sec\n",
      "total size is 12639857  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009176365.1_Aspalli1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12652992 bytes  5061213.60 bytes/sec\n",
      "total size is 12649792  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009812435.1_ASM981243v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11134786 bytes  3181379.43 bytes/sec\n",
      "total size is 11131959  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193545.1_Aspara19utr_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12514881 bytes  5005969.20 bytes/sec\n",
      "total size is 12511718  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193465.1_Aspave1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10614000 bytes  4245616.80 bytes/sec\n",
      "total size is 10611305  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003850985.1_Aawm_assembly01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12074280 bytes  3449806.29 bytes/sec\n",
      "total size is 12071217  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193595.1_Aspber1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11676172 bytes  3336061.14 bytes/sec\n",
      "total size is 11673213  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001792695.1_ASM179269v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11814876 bytes  3375690.86 bytes/sec\n",
      "total size is 11811881  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001889945.1_Aspbr1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11291100 bytes  4516456.80 bytes/sec\n",
      "total size is 11288238  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184695.1_Aspbru1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11696911 bytes  3341986.57 bytes/sec\n",
      "total size is 11693952  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193585.1_Aspcae1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12643095 bytes  5057254.80 bytes/sec\n",
      "total size is 12639904  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001511075.1_Acal_Allpaths_LG_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12705590 bytes  5082252.80 bytes/sec\n",
      "total size is 12702374  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002847485.1_Aspcam1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8832279 bytes  3532928.40 bytes/sec\n",
      "total size is 8830016  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002847045.1_Aspcand1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8673347 bytes  3469355.60 bytes/sec\n",
      "total size is 8671123  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001990825.1_Aspca3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10787004 bytes  4314818.40 bytes/sec\n",
      "total size is 10784262  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769165.1_ASM476916v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7416415 bytes  2966582.80 bytes/sec\n",
      "total size is 7414493  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001599875.1_JCM_23047_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8662433 bytes  2474992.86 bytes/sec\n",
      "total size is 8660194  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002715.1_ASM271v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8736242 bytes  1941396.44 bytes/sec\n",
      "total size is 8734002  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_009193565.1_Aspcor1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9304271 bytes  3721725.20 bytes/sec\n",
      "total size is 9301896  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184835.1_Aspcos1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11631406 bytes  2114808.73 bytes/sec\n",
      "total size is 11628463  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001693355.1_ASM169335v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8752378 bytes  3500968.00 bytes/sec\n",
      "total size is 8750127  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184645.1_Aspell1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12693655 bytes  3626770.57 bytes/sec\n",
      "total size is 12690448  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184535.1_Aspeuc1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10955304 bytes  4382138.40 bytes/sec\n",
      "total size is 10952521  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184825.1_Aspfij1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11441192 bytes  3268924.00 bytes/sec\n",
      "total size is 11438289  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000149645.2_ASM14964v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9901947 bytes  3960795.60 bytes/sec\n",
      "total size is 9899417  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006275.2_JCVI-afl1-v2.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11565143 bytes  3304338.57 bytes/sec\n",
      "total size is 11562209  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002655.1_ASM265v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9005458 bytes  3602200.00 bytes/sec\n",
      "total size is 9003154  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890805.1_Aspgl1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8777547 bytes  3511035.60 bytes/sec\n",
      "total size is 8775301  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001696595.1_ASM169659v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12686018 bytes  3624588.57 bytes/sec\n",
      "total size is 12682807  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003184545.1_Asphet1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10892839 bytes  3112251.71 bytes/sec\n",
      "total size is 10890072  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184865.1_Asphom1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10685415 bytes  4274182.80 bytes/sec\n",
      "total size is 10682696  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184845.1_Aspibe1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10519044 bytes  4207634.40 bytes/sec\n",
      "total size is 10516373  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184685.1_Aspind1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11985053 bytes  3424312.86 bytes/sec\n",
      "total size is 11982022  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184785.1_Aspjap1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11330289 bytes  4532132.40 bytes/sec\n",
      "total size is 11327418  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000239835.2_Akaw_assembly01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11667116 bytes  4666863.20 bytes/sec\n",
      "total size is 11664157  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001445615.1_Alt_assembly01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9636685 bytes  3854690.80 bytes/sec\n",
      "total size is 9634215  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009176345.1_Asplep1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12463377 bytes  3560976.86 bytes/sec\n",
      "total size is 12460226  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890685.1_Aspfo1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11787746 bytes  3367939.43 bytes/sec\n",
      "total size is 11784764  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009176455.1_Aspmin1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11669813 bytes  4667942.00 bytes/sec\n",
      "total size is 11666854  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003369625.1_ASM336962v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 13416139 bytes  2981373.56 bytes/sec\n",
      "total size is 13412752  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003116565.1_ASM311656v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8640785 bytes  3456330.80 bytes/sec\n",
      "total size is 8638566  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184625.1_Aspneo1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11166112 bytes  3190329.71 bytes/sec\n",
      "total size is 11163281  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000149205.2_ASM14920v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9368619 bytes  3747464.40 bytes/sec\n",
      "total size is 9366225  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002855.2_ASM285v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10729506 bytes  4291819.20 bytes/sec\n",
      "total size is 10726778  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001204775.1_ASM120477v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11433431 bytes  4573389.20 bytes/sec\n",
      "total size is 11430532  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002847465.1_Aspnov1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10054989 bytes  2872866.00 bytes/sec\n",
      "total size is 10052430  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009176405.1_Aspnovo1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12897997 bytes  5159215.60 bytes/sec\n",
      "total size is 12894741  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002846915.2_Aspergillus_ochraceoroseus_IBT_24754_v1.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8499776 bytes  2428519.43 bytes/sec\n",
      "total size is 8497559  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004849945.1_AoFc_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11560680 bytes  4624288.80 bytes/sec\n",
      "total size is 11557756  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003719415.1_ASM371941v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9541360 bytes  2726114.86 bytes/sec\n",
      "total size is 9538917  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000184455.3_ASM18445v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11694008 bytes  4677620.00 bytes/sec\n",
      "total size is 11691046  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_009176385.1_Asppar1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12042054 bytes  3440598.86 bytes/sec\n",
      "total size is 12039007  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002215965.1_ASM221596v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11949324 bytes  4779746.40 bytes/sec\n",
      "total size is 11946297  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003344505.1_Aspph1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11261614 bytes  4504662.40 bytes/sec\n",
      "total size is 11258760  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184755.1_Asppip1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11120111 bytes  3177186.57 bytes/sec\n",
      "total size is 11117288  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193665.1_Asppsec1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12469101 bytes  4987657.20 bytes/sec\n",
      "total size is 12465949  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193645.1_Asppsen1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11912891 bytes  4765173.20 bytes/sec\n",
      "total size is 11909875  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193445.1_Asppset1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11994510 bytes  4797820.80 bytes/sec\n",
      "total size is 11991478  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002927005.1_ASM292700v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9297835 bytes  3719150.80 bytes/sec\n",
      "total size is 9295456  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000986645.1_ASM98664v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8256928 bytes  3302788.00 bytes/sec\n",
      "total size is 8254807  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000600275.1_Eurhe1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8233894 bytes  3293574.40 bytes/sec\n",
      "total size is 8231777  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184585.1_Aspsac1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9758673 bytes  3903486.00 bytes/sec\n",
      "total size is 9756186  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003589665.1_phiScl_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8936634 bytes  2553336.00 bytes/sec\n",
      "total size is 8934344  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184635.1_Aspscle1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11760199 bytes  2613386.89 bytes/sec\n",
      "total size is 11757223  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184525.1_Aspscl1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11532577 bytes  4613047.60 bytes/sec\n",
      "total size is 11529658  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000530345.1_HBR18v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10451505 bytes  4180618.80 bytes/sec\n",
      "total size is 10448850  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193525.1_Aspser1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12094662 bytes  4837881.60 bytes/sec\n",
      "total size is 12091599  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008274985.1_ASM827498v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12568283 bytes  3590950.00 bytes/sec\n",
      "total size is 12565104  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001044295.1_ASM104429v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10666760 bytes  4266720.80 bytes/sec\n",
      "total size is 10664045  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003574815.1_ASM357481v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10454057 bytes  2986885.43 bytes/sec\n",
      "total size is 10451398  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002849105.1_Aspste1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11708254 bytes  4683318.40 bytes/sec\n",
      "total size is 11705287  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890705.1_Aspsy1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10727349 bytes  3064968.86 bytes/sec\n",
      "total size is 10724623  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002850765.1_Asptaic1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8620561 bytes  3448241.20 bytes/sec\n",
      "total size is 8618345  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_009193485.1_Asptam1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12136008 bytes  4854420.00 bytes/sec\n",
      "total size is 12132937  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003426965.1_ASM342696v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12167707 bytes  3476499.71 bytes/sec\n",
      "total size is 12164624  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000149615.1_ASM14961v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9202153 bytes  2044932.22 bytes/sec\n",
      "total size is 9199799  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002237265.2_ASM223726v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9713684 bytes  2158605.78 bytes/sec\n",
      "total size is 9711201  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193505.1_Asptra1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12382456 bytes  2751666.22 bytes/sec\n",
      "total size is 12379329  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009812425.1_ASM981242v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9148490 bytes  3659412.80 bytes/sec\n",
      "total size is 9146143  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890745.1_Asptu1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11111737 bytes  4444711.60 bytes/sec\n",
      "total size is 11108915  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002234965.2_ASM223496v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10119675 bytes  2891347.71 bytes/sec\n",
      "total size is 10117096  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001078395.1_Aud_assembly01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10107918 bytes  4043184.00 bytes/sec\n",
      "total size is 10105336  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003324175.1_ASM332417v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11742278 bytes  3354948.57 bytes/sec\n",
      "total size is 11739299  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000812125.1_Austus1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12075747 bytes  3450225.43 bytes/sec\n",
      "total size is 12072690  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003184745.1_Aspuva1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11312596 bytes  3232182.29 bytes/sec\n",
      "total size is 11309725  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184925.1_Aspvad1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11277655 bytes  3222199.14 bytes/sec\n",
      "total size is 11274792  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890125.1_Aspve1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10352532 bytes  2957878.29 bytes/sec\n",
      "total size is 10349902  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184705.1_Aspvio1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11358598 bytes  2524142.22 bytes/sec\n",
      "total size is 11355719  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004368095.1_ASM436809v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9327253 bytes  2664941.43 bytes/sec\n",
      "total size is 9324866  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003344945.1_Aspwel1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11847714 bytes  4739102.40 bytes/sec\n",
      "total size is 11844715  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890725.1_Aspwe1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9864276 bytes  3945727.20 bytes/sec\n",
      "total size is 9861766  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001307345.1_ASM130734v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10958003 bytes  4383218.00 bytes/sec\n",
      "total size is 10955216  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001599675.1_JCM_23157_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10687051 bytes  4274837.20 bytes/sec\n",
      "total size is 10684316  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003116535.1_ASM311653v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11372404 bytes  3249270.29 bytes/sec\n",
      "total size is 11369521  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002914405.1_ByssAF2.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11182736 bytes  4473111.20 bytes/sec\n",
      "total size is 11179895  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_004022145.1_Paevar1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9374786 bytes  2678522.29 bytes/sec\n",
      "total size is 9372387  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002240705.1_PX439_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 17344292 bytes  4955524.00 bytes/sec\n",
      "total size is 17339951  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003184285.1_ASM318428v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7067515 bytes  2827022.80 bytes/sec\n",
      "total size is 7065681  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002976275.1_ASM297627v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8258360 bytes  3303360.80 bytes/sec\n",
      "total size is 8256238  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001890105.1_Aspzo1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8189389 bytes  3275772.40 bytes/sec\n",
      "total size is 8187288  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072345.1_ASM207234v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9849871 bytes  3939965.20 bytes/sec\n",
      "total size is 9847356  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001773325.1_ASM177332v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10520528 bytes  4208228.00 bytes/sec\n",
      "total size is 10517853  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000577785.1_PBIFFM169_20131217_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11010617 bytes  4404263.60 bytes/sec\n",
      "total size is 11007815  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001048715.1_Pbras_Allpaths-LG_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11093978 bytes  4437608.00 bytes/sec\n",
      "total size is 11091153  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000513335.1_PCAMFM013_20131106_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10726545 bytes  4290634.80 bytes/sec\n",
      "total size is 10723807  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000943775.1_ASM94377v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10653208 bytes  4261300.00 bytes/sec\n",
      "total size is 10650494  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000577495.1_PCARLCP05634_20131217_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8301986 bytes  3320811.20 bytes/sec\n",
      "total size is 8299838  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000710275.1_ASM71027v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10177663 bytes  4071082.00 bytes/sec\n",
      "total size is 10175069  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001950535.1_JCM_22607_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10596195 bytes  3027496.29 bytes/sec\n",
      "total size is 10593484  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072405.1_ASM207240v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8891208 bytes  2540357.14 bytes/sec\n",
      "total size is 8888925  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072245.1_ASM207224v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7481263 bytes  2992522.00 bytes/sec\n",
      "total size is 7479325  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000315645.2_PdigPd1_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7886777 bytes  2253376.86 bytes/sec\n",
      "total size is 7884744  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000769745.1_ASM76974v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10207506 bytes  4083019.20 bytes/sec\n",
      "total size is 10204904  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072365.1_ASM207236v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10340029 bytes  4136028.40 bytes/sec\n",
      "total size is 10337394  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001513925.1_ASM151392v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10662769 bytes  3046517.43 bytes/sec\n",
      "total size is 10660054  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000576735.1_PFUSFM041_20131217_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11363935 bytes  3246850.57 bytes/sec\n",
      "total size is 11361045  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001561935.1_ASM156193v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9128410 bytes  3651380.80 bytes/sec\n",
      "total size is 9126071  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002116305.1_ASM211630v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9647389 bytes  3858972.40 bytes/sec\n",
      "total size is 9644922  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002369805.1_ASM236980v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11797809 bytes  4719140.40 bytes/sec\n",
      "total size is 11794822  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072425.1_ASM207242v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10350482 bytes  4140209.60 bytes/sec\n",
      "total size is 10347847  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000733025.2_PnBFE487-1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9635914 bytes  3854382.40 bytes/sec\n",
      "total size is 9633454  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000346795.1_pde_v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9425936 bytes  3770391.20 bytes/sec\n",
      "total size is 9423528  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000577715.1_PPANFM227_20131217_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8265624 bytes  1836814.67 bytes/sec\n",
      "total size is 8263487  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000347475.1_Ppaxilli_v.1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11087248 bytes  2463842.22 bytes/sec\n",
      "total size is 11084426  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003344595.1_ASM334459v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10545383 bytes  4218170.00 bytes/sec\n",
      "total size is 10542700  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000513255.1_PROQFM164_20130607_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8774010 bytes  3509620.80 bytes/sec\n",
      "total size is 8771752  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000226395.1_PenChr_Nov2007_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10057833 bytes  4023150.00 bytes/sec\n",
      "total size is 10055267  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001750025.1_ASM175002v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10605702 bytes  4242297.60 bytes/sec\n",
      "total size is 10603003  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000952775.2_ASM95277v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10974357 bytes  4389759.60 bytes/sec\n",
      "total size is 10971571  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003800485.1_ASM380048v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11340985 bytes  3240293.43 bytes/sec\n",
      "total size is 11338102  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072375.1_ASM207237v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10116564 bytes  2890458.86 bytes/sec\n",
      "total size is 10113985  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001908125.1_ASM190812v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12483370 bytes  2774091.56 bytes/sec\n",
      "total size is 12480215  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000970515.2_PV808_April18ReSeqReAss_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9585293 bytes  3834134.00 bytes/sec\n",
      "total size is 9582830  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002072255.1_ASM207225v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9783151 bytes  2174042.89 bytes/sec\n",
      "total size is 9780652  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003698115.1_ASM369811v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6875818 bytes  2750344.00 bytes/sec\n",
      "total size is 6874032  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000968595.1_ASM96859v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9070219 bytes  2591503.14 bytes/sec\n",
      "total size is 9067897  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002775195.1_ASM277519v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11405924 bytes  4562386.40 bytes/sec\n",
      "total size is 11403033  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001896365.1_ASM189636v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10527743 bytes  4211114.00 bytes/sec\n",
      "total size is 10525060  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001907595.1_ASM190759v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9607670 bytes  3843084.80 bytes/sec\n",
      "total size is 9605211  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002916415.1_ASM291641v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8552377 bytes  3420967.60 bytes/sec\n",
      "total size is 8550182  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000829775.1_TcelY94_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11273918 bytes  4509584.00 bytes/sec\n",
      "total size is 11271059  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004299765.1_ASM429976v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8927618 bytes  3571064.00 bytes/sec\n",
      "total size is 8925327  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000985935.1_PIS_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10516854 bytes  3004827.43 bytes/sec\n",
      "total size is 10514187  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000001985.1_JCVI-PMFA1-2.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8913625 bytes  3565466.80 bytes/sec\n",
      "total size is 8911339  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001657655.1_ASM165765v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8333240 bytes  1111104.27 bytes/sec\n",
      "total size is 8331094  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001571465.2_ASM157146v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11422491 bytes  2538340.67 bytes/sec\n",
      "total size is 11419592  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001270325.1_PPv01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11937316 bytes  3410673.71 bytes/sec\n",
      "total size is 11934295  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000003125.1_JCVI-TSTA1-3.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11149456 bytes  3185570.86 bytes/sec\n",
      "total size is 11146618  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001305275.1_ASM130527v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11861807 bytes  3389099.71 bytes/sec\n",
      "total size is 11858804  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001939245.1_ASM193924v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12074548 bytes  4829836.00 bytes/sec\n",
      "total size is 12071489  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003123655.1_ASM312365v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10000311 bytes  4000141.20 bytes/sec\n",
      "total size is 9997756  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001599835.1_JCM_12817_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9870422 bytes  3948185.60 bytes/sec\n",
      "total size is 9867887  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000315935.1_NOKS_genome_assembly_20110720_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6018353 bytes  1719541.43 bytes/sec\n",
      "total size is 6016757  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900006255.1_Xbisp_v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6855670 bytes  1523491.56 bytes/sec\n",
      "total size is 6853885  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484540 (\"sim_Asprgillus_aculeatinus_GCF_003184765\") has been submitted\n",
      "Your job 5484541 (\"sim_Asprgillus_aculeatus_GCF_001890905\") has been submitted\n",
      "Your job 5484542 (\"sim_Asprgillus_aff_GCA_001931935\") has been submitted\n",
      "Your job 5484543 (\"sim_Asprgillus_albertensis_GCA_009193605\") has been submitted\n",
      "Your job 5484544 (\"sim_Asprgillus_alliaceus_GCF_009176365\") has been submitted\n",
      "Your job 5484545 (\"sim_Asprgillus_amoenus_GCA_009812435\") has been submitted\n",
      "Your job 5484546 (\"sim_Asprgillus_arachidicola_GCA_009193545\") has been submitted\n",
      "Your job 5484547 (\"sim_Asprgillus_avenaceus_GCA_009193465\") has been submitted\n",
      "Your job 5484548 (\"sim_Asprgillus_awamori_GCA_003850985\") has been submitted\n",
      "Your job 5484549 (\"sim_Asprgillus_bertholletius_GCA_009193595\") has been submitted\n",
      "Your job 5484550 (\"sim_Asprgillus_bombycis_GCF_001792695\") has been submitted\n",
      "Your job 5484551 (\"sim_Asprgillus_brasiliensis_GCA_001889945\") has been submitted\n",
      "Your job 5484552 (\"sim_Asprgillus_brunneoviolaceus_GCF_003184695\") has been submitted\n",
      "Your job 5484553 (\"sim_Asprgillus_caelatus_GCF_009193585\") has been submitted\n",
      "Your job 5484554 (\"sim_Asprgillus_calidoustus_GCA_001511075\") has been submitted\n",
      "Your job 5484555 (\"sim_Asprgillus_campestris_GCF_002847485\") has been submitted\n",
      "Your job 5484556 (\"sim_Asprgillus_candidus_GCF_002847045\") has been submitted\n",
      "Your job 5484557 (\"sim_Asprgillus_carbonarius_GCA_001990825\") has been submitted\n",
      "Your job 5484558 (\"sim_Asprgillus_cejpii_GCA_004769165\") has been submitted\n",
      "Your job 5484559 (\"sim_Asprgillus_chevalieri_GCA_001599875\") has been submitted\n",
      "Your job 5484560 (\"sim_Asprgillus_clavatus_GCF_000002715\") has been submitted\n",
      "Your job 5484561 (\"sim_Asprgillus_coremiiformis_GCA_009193565\") has been submitted\n",
      "Your job 5484562 (\"sim_Asprgillus_costaricaensis_GCF_003184835\") has been submitted\n",
      "Your job 5484563 (\"sim_Asprgillus_cristatus_GCA_001693355\") has been submitted\n",
      "Your job 5484564 (\"sim_Asprgillus_ellipticus_GCA_003184645\") has been submitted\n",
      "Your job 5484565 (\"sim_Asprgillus_eucalypticola_GCF_003184535\") has been submitted\n",
      "Your job 5484566 (\"sim_Asprgillus_fijiensis_GCA_003184825\") has been submitted\n",
      "Your job 5484567 (\"sim_Asprgillus_fischeri_GCF_000149645\") has been submitted\n",
      "Your job 5484568 (\"sim_Asprgillus_flavus_GCF_000006275\") has been submitted\n",
      "Your job 5484569 (\"sim_Asprgillus_fumigatus_GCF_000002655\") has been submitted\n",
      "Your job 5484570 (\"sim_Asprgillus_glaucus_GCF_001890805\") has been submitted\n",
      "Your job 5484571 (\"sim_Asprgillus_hancockii_GCA_001696595\") has been submitted\n",
      "Your job 5484572 (\"sim_Asprgillus_heteromorphus_GCF_003184545\") has been submitted\n",
      "Your job 5484573 (\"sim_Asprgillus_homomorphus_GCF_003184865\") has been submitted\n",
      "Your job 5484574 (\"sim_Asprgillus_ibericus_GCF_003184845\") has been submitted\n",
      "Your job 5484575 (\"sim_Asprgillus_indologenus_GCA_003184685\") has been submitted\n",
      "Your job 5484576 (\"sim_Asprgillus_japonicus_GCF_003184785\") has been submitted\n",
      "Your job 5484577 (\"sim_Asprgillus_kawachii_GCA_000239835\") has been submitted\n",
      "Your job 5484578 (\"sim_Asprgillus_lentulus_GCA_001445615\") has been submitted\n",
      "Your job 5484579 (\"sim_Asprgillus_leporis_GCA_009176345\") has been submitted\n",
      "Your job 5484580 (\"sim_Asprgillus_luchuensis_GCA_001890685\") has been submitted\n",
      "Your job 5484581 (\"sim_Asprgillus_minisclerotigenes_GCA_009176455\") has been submitted\n",
      "Your job 5484582 (\"sim_Asprgillus_mulundensis_GCF_003369625\") has been submitted\n",
      "Your job 5484583 (\"sim_Asprgillus_neoellipticus_GCA_003116565\") has been submitted\n",
      "Your job 5484584 (\"sim_Asprgillus_neoniger_GCF_003184625\") has been submitted\n",
      "Your job 5484585 (\"sim_Asprgillus_nidulans_GCF_000149205\") has been submitted\n",
      "Your job 5484586 (\"sim_Asprgillus_niger_GCF_000002855\") has been submitted\n",
      "Your job 5484587 (\"sim_Asprgillus_nomius_GCF_001204775\") has been submitted\n",
      "Your job 5484588 (\"sim_Asprgillus_novofumigatus_GCF_002847465\") has been submitted\n",
      "Your job 5484589 (\"sim_Asprgillus_novoparasiticus_GCA_009176405\") has been submitted\n",
      "Your job 5484590 (\"sim_Asprgillus_ochraceoroseus_GCA_002846915\") has been submitted\n",
      "Your job 5484591 (\"sim_Asprgillus_ochraceus_GCA_004849945\") has been submitted\n",
      "Your job 5484592 (\"sim_Asprgillus_olivimuriae_GCA_003719415\") has been submitted\n",
      "Your job 5484593 (\"sim_Asprgillus_oryzae_GCF_000184455\") has been submitted\n",
      "Your job 5484594 (\"sim_Asprgillus_parasiticus_GCA_009176385\") has been submitted\n",
      "Your job 5484595 (\"sim_Asprgillus_persii_GCA_002215965\") has been submitted\n",
      "Your job 5484596 (\"sim_Asprgillus_phoenicis_GCA_003344505\") has been submitted\n",
      "Your job 5484597 (\"sim_Asprgillus_piperis_GCF_003184755\") has been submitted\n",
      "Your job 5484598 (\"sim_Asprgillus_pseudocaelatus_GCA_009193665\") has been submitted\n",
      "Your job 5484599 (\"sim_Asprgillus_pseudotamarii_GCF_009193445\") has been submitted\n",
      "Your job 5484600 (\"sim_Asprgillus_pseudoterreus_GCA_002927005\") has been submitted\n",
      "Your job 5484601 (\"sim_Asprgillus_rambellii_GCA_000986645\") has been submitted\n",
      "Your job 5484602 (\"sim_Asprgillus_ruber_GCA_000600275\") has been submitted\n",
      "Your job 5484603 (\"sim_Asprgillus_saccharolyticus_GCF_003184585\") has been submitted\n",
      "Your job 5484604 (\"sim_Asprgillus_sclerotialis_GCA_003589665\") has been submitted\n",
      "Your job 5484605 (\"sim_Asprgillus_sclerotiicarbonarius_GCA_003184635\") has been submitted\n",
      "Your job 5484606 (\"sim_Asprgillus_sclerotioniger_GCF_003184525\") has been submitted\n",
      "Your job 5484607 (\"sim_Asprgillus_sclerotiorum_GCA_000530345\") has been submitted\n",
      "Your job 5484608 (\"sim_Asprgillus_sergii_GCA_009193525\") has been submitted\n",
      "Your job 5484609 (\"sim_Asprgillus_sojae_GCA_008274985\") has been submitted\n",
      "Your job 5484610 (\"sim_Asprgillus_sp_GCA_001044295\") has been submitted\n",
      "Your job 5484611 (\"sim_Asprgillus_spnulosprus_GCA_003574815\") has been submitted\n",
      "Your job 5484612 (\"sim_Asprgillus_steynii_GCF_002849105\") has been submitted\n",
      "Your job 5484613 (\"sim_Asprgillus_sydowii_GCA_001890705\") has been submitted\n",
      "Your job 5484614 (\"sim_Asprgillus_taichungensis_GCA_002850765\") has been submitted\n",
      "Your job 5484615 (\"sim_Asprgillus_tamarii_GCA_009193485\") has been submitted\n",
      "Your job 5484616 (\"sim_Asprgillus_tanneri_GCA_003426965\") has been submitted\n",
      "Your job 5484617 (\"sim_Asprgillus_terreus_GCF_000149615\") has been submitted\n",
      "Your job 5484618 (\"sim_Asprgillus_thermomutatus_GCF_002237265\") has been submitted\n",
      "Your job 5484619 (\"sim_Asprgillus_transmontanensis_GCA_009193505\") has been submitted\n",
      "Your job 5484620 (\"sim_Asprgillus_tritici_GCA_009812425\") has been submitted\n",
      "Your job 5484621 (\"sim_Asprgillus_tubingensis_GCA_001890745\") has been submitted\n",
      "Your job 5484622 (\"sim_Asprgillus_turcosus_GCA_002234965\") has been submitted\n",
      "Your job 5484623 (\"sim_Asprgillus_udagawae_GCA_001078395\") has been submitted\n",
      "Your job 5484624 (\"sim_Asprgillus_unguis_GCA_003324175\") has been submitted\n",
      "Your job 5484625 (\"sim_Asprgillus_ustus_GCA_000812125\") has been submitted\n",
      "Your job 5484626 (\"sim_Asprgillus_uvarum_GCF_003184745\") has been submitted\n",
      "Your job 5484627 (\"sim_Asprgillus_vadensis_GCF_003184925\") has been submitted\n",
      "Your job 5484628 (\"sim_Asprgillus_versicolor_GCA_001890125\") has been submitted\n",
      "Your job 5484629 (\"sim_Asprgillus_violaceofuscus_GCA_003184705\") has been submitted\n",
      "Your job 5484630 (\"sim_Asprgillus_viridinutans_GCA_004368095\") has been submitted\n",
      "Your job 5484631 (\"sim_Asprgillus_welwitschiae_GCF_003344945\") has been submitted\n",
      "Your job 5484632 (\"sim_Asprgillus_wentii_GCA_001890725\") has been submitted\n",
      "Your job 5484633 (\"sim_Asprgillus_westerdijkiae_GCA_001307345\") has been submitted\n",
      "Your job 5484634 (\"sim_Basipetospra_chlamydospra_GCA_001599675\") has been submitted\n",
      "Your job 5484635 (\"sim_Byssochlamys_nivea_GCA_003116535\") has been submitted\n",
      "Your job 5484636 (\"sim_Byssochlamys_sp_GCA_002914405\") has been submitted\n",
      "Your job 5484637 (\"sim_Byssochlamys_spctabilis_GCF_004022145\") has been submitted\n",
      "Your job 5484638 (\"sim_Elaphomyces_granulatus_GCA_002240705\") has been submitted\n",
      "Your job 5484639 (\"sim_Monascus_purpureus_GCA_003184285\") has been submitted\n",
      "Your job 5484640 (\"sim_Monascus_ruber_GCA_002976275\") has been submitted\n",
      "Your job 5484641 (\"sim_Penicilliopsis_zonata_GCF_001890105\") has been submitted\n",
      "Your job 5484642 (\"sim_Penicillium_antarcticum_GCA_002072345\") has been submitted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484643 (\"sim_Penicillium_arizonense_GCF_001773325\") has been submitted\n",
      "Your job 5484644 (\"sim_Penicillium_biforme_GCA_000577785\") has been submitted\n",
      "Your job 5484645 (\"sim_Penicillium_brasilianum_GCA_001048715\") has been submitted\n",
      "Your job 5484646 (\"sim_Penicillium_camemberti_GCA_000513335\") has been submitted\n",
      "Your job 5484647 (\"sim_Penicillium_capsulatum_GCA_000943775\") has been submitted\n",
      "Your job 5484648 (\"sim_Penicillium_carneum_GCA_000577495\") has been submitted\n",
      "Your job 5484649 (\"sim_Penicillium_chrysogenum_GCA_000710275\") has been submitted\n",
      "Your job 5484650 (\"sim_Penicillium_citrinum_GCA_001950535\") has been submitted\n",
      "Your job 5484651 (\"sim_Penicillium_coprophilum_GCA_002072405\") has been submitted\n",
      "Your job 5484652 (\"sim_Penicillium_decumbens_GCA_002072245\") has been submitted\n",
      "Your job 5484653 (\"sim_Penicillium_digitatum_GCF_000315645\") has been submitted\n",
      "Your job 5484654 (\"sim_Penicillium_expansum_GCF_000769745\") has been submitted\n",
      "Your job 5484655 (\"sim_Penicillium_flavigenum_GCA_002072365\") has been submitted\n",
      "Your job 5484656 (\"sim_Penicillium_freii_GCA_001513925\") has been submitted\n",
      "Your job 5484657 (\"sim_Penicillium_fuscoglaucum_GCA_000576735\") has been submitted\n",
      "Your job 5484658 (\"sim_Penicillium_griseofulvum_GCA_001561935\") has been submitted\n",
      "Your job 5484659 (\"sim_Penicillium_italicum_GCA_002116305\") has been submitted\n",
      "Your job 5484660 (\"sim_Penicillium_janthinellum_GCA_002369805\") has been submitted\n",
      "Your job 5484661 (\"sim_Penicillium_nalgiovense_GCA_002072425\") has been submitted\n",
      "Your job 5484662 (\"sim_Penicillium_nordicum_GCA_000733025\") has been submitted\n",
      "Your job 5484663 (\"sim_Penicillium_oxalicum_GCA_000346795\") has been submitted\n",
      "Your job 5484664 (\"sim_Penicillium_paneum_GCA_000577715\") has been submitted\n",
      "Your job 5484665 (\"sim_Penicillium_paxilli_GCA_000347475\") has been submitted\n",
      "Your job 5484666 (\"sim_Penicillium_polonicum_GCA_003344595\") has been submitted\n",
      "Your job 5484667 (\"sim_Penicillium_roqueforti_GCA_000513255\") has been submitted\n",
      "Your job 5484668 (\"sim_Penicillium_rubens_GCF_000226395\") has been submitted\n",
      "Your job 5484669 (\"sim_Penicillium_sclerotiorum_GCA_001750025\") has been submitted\n",
      "Your job 5484670 (\"sim_Penicillium_solitum_GCA_000952775\") has been submitted\n",
      "Your job 5484671 (\"sim_Penicillium_sp_GCA_003800485\") has been submitted\n",
      "Your job 5484672 (\"sim_Penicillium_steckii_GCA_002072375\") has been submitted\n",
      "Your job 5484673 (\"sim_Penicillium_subrubescens_GCA_001908125\") has been submitted\n",
      "Your job 5484674 (\"sim_Penicillium_verrucosum_GCA_000970515\") has been submitted\n",
      "Your job 5484675 (\"sim_Penicillium_vulpinum_GCA_002072255\") has been submitted\n",
      "Your job 5484676 (\"sim_Phialosimplex_sp_GCA_003698115\") has been submitted\n",
      "Your job 5484677 (\"sim_Rasamsonia_emersonii_GCF_000968595\") has been submitted\n",
      "Your job 5484678 (\"sim_Talaromyces_adpressus_GCA_002775195\") has been submitted\n",
      "Your job 5484679 (\"sim_Talaromyces_amestolkiae_GCA_001896365\") has been submitted\n",
      "Your job 5484680 (\"sim_Talaromyces_atroroseus_GCF_001907595\") has been submitted\n",
      "Your job 5484681 (\"sim_Talaromyces_borbonicus_GCA_002916415\") has been submitted\n",
      "Your job 5484682 (\"sim_Talaromyces_cellulolyticus_GCA_000829775\") has been submitted\n",
      "Your job 5484683 (\"sim_Talaromyces_funiculosus_GCA_004299765\") has been submitted\n",
      "Your job 5484684 (\"sim_Talaromyces_islandicus_GCA_000985935\") has been submitted\n",
      "Your job 5484685 (\"sim_Talaromyces_piceae_GCA_001657655\") has been submitted\n",
      "Your job 5484686 (\"sim_Talaromyces_pinophilus_GCA_001571465\") has been submitted\n",
      "Your job 5484687 (\"sim_Talaromyces_purpureogenus_GCA_001270325\") has been submitted\n",
      "Your job 5484688 (\"sim_Talaromyces_stipitatus_GCF_000003125\") has been submitted\n",
      "Your job 5484689 (\"sim_Talaromyces_verruculosus_GCA_001305275\") has been submitted\n",
      "Your job 5484690 (\"sim_Talaromyces_wortmannii_GCA_001939245\") has been submitted\n",
      "Your job 5484691 (\"sim_Thermoascaceae_sp_GCA_003123655\") has been submitted\n",
      "Your job 5484692 (\"sim_Thermoascus_crustaceus_GCA_001599835\") has been submitted\n",
      "Your job 5484693 (\"sim_Thermomyces_lanuginosus_GCA_000315935\") has been submitted\n",
      "Your job 5484694 (\"sim_Xeromyces_bisprus_GCA_900006255\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/eurotiales/cleaned_genomes/Talaromyces_marneffei_GCF_000001985_formatted.fas path=results/phyluce/eurotiales/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484695 (\"map_Asprgillus_aculeatinus_GCF_003184765\") has been submitted\n",
      "Your job 5484696 (\"map_Asprgillus_aculeatus_GCF_001890905\") has been submitted\n",
      "Your job 5484697 (\"map_Asprgillus_aff_GCA_001931935\") has been submitted\n",
      "Your job 5484698 (\"map_Asprgillus_albertensis_GCA_009193605\") has been submitted\n",
      "Your job 5484699 (\"map_Asprgillus_alliaceus_GCF_009176365\") has been submitted\n",
      "Your job 5484700 (\"map_Asprgillus_amoenus_GCA_009812435\") has been submitted\n",
      "Your job 5484701 (\"map_Asprgillus_arachidicola_GCA_009193545\") has been submitted\n",
      "Your job 5484702 (\"map_Asprgillus_avenaceus_GCA_009193465\") has been submitted\n",
      "Your job 5484703 (\"map_Asprgillus_awamori_GCA_003850985\") has been submitted\n",
      "Your job 5484704 (\"map_Asprgillus_bertholletius_GCA_009193595\") has been submitted\n",
      "Your job 5484705 (\"map_Asprgillus_bombycis_GCF_001792695\") has been submitted\n",
      "Your job 5484706 (\"map_Asprgillus_brasiliensis_GCA_001889945\") has been submitted\n",
      "Your job 5484707 (\"map_Asprgillus_brunneoviolaceus_GCF_003184695\") has been submitted\n",
      "Your job 5484708 (\"map_Asprgillus_caelatus_GCF_009193585\") has been submitted\n",
      "Your job 5484709 (\"map_Asprgillus_calidoustus_GCA_001511075\") has been submitted\n",
      "Your job 5484710 (\"map_Asprgillus_campestris_GCF_002847485\") has been submitted\n",
      "Your job 5484711 (\"map_Asprgillus_candidus_GCF_002847045\") has been submitted\n",
      "Your job 5484712 (\"map_Asprgillus_carbonarius_GCA_001990825\") has been submitted\n",
      "Your job 5484713 (\"map_Asprgillus_cejpii_GCA_004769165\") has been submitted\n",
      "Your job 5484714 (\"map_Asprgillus_chevalieri_GCA_001599875\") has been submitted\n",
      "Your job 5484715 (\"map_Asprgillus_clavatus_GCF_000002715\") has been submitted\n",
      "Your job 5484716 (\"map_Asprgillus_coremiiformis_GCA_009193565\") has been submitted\n",
      "Your job 5484717 (\"map_Asprgillus_costaricaensis_GCF_003184835\") has been submitted\n",
      "Your job 5484718 (\"map_Asprgillus_cristatus_GCA_001693355\") has been submitted\n",
      "Your job 5484719 (\"map_Asprgillus_ellipticus_GCA_003184645\") has been submitted\n",
      "Your job 5484720 (\"map_Asprgillus_eucalypticola_GCF_003184535\") has been submitted\n",
      "Your job 5484721 (\"map_Asprgillus_fijiensis_GCA_003184825\") has been submitted\n",
      "Your job 5484722 (\"map_Asprgillus_fischeri_GCF_000149645\") has been submitted\n",
      "Your job 5484723 (\"map_Asprgillus_flavus_GCF_000006275\") has been submitted\n",
      "Your job 5484724 (\"map_Asprgillus_fumigatus_GCF_000002655\") has been submitted\n",
      "Your job 5484725 (\"map_Asprgillus_glaucus_GCF_001890805\") has been submitted\n",
      "Your job 5484726 (\"map_Asprgillus_hancockii_GCA_001696595\") has been submitted\n",
      "Your job 5484727 (\"map_Asprgillus_heteromorphus_GCF_003184545\") has been submitted\n",
      "Your job 5484728 (\"map_Asprgillus_homomorphus_GCF_003184865\") has been submitted\n",
      "Your job 5484729 (\"map_Asprgillus_ibericus_GCF_003184845\") has been submitted\n",
      "Your job 5484730 (\"map_Asprgillus_indologenus_GCA_003184685\") has been submitted\n",
      "Your job 5484731 (\"map_Asprgillus_japonicus_GCF_003184785\") has been submitted\n",
      "Your job 5484732 (\"map_Asprgillus_kawachii_GCA_000239835\") has been submitted\n",
      "Your job 5484733 (\"map_Asprgillus_lentulus_GCA_001445615\") has been submitted\n",
      "Your job 5484734 (\"map_Asprgillus_leporis_GCA_009176345\") has been submitted\n",
      "Your job 5484735 (\"map_Asprgillus_luchuensis_GCA_001890685\") has been submitted\n",
      "Your job 5484736 (\"map_Asprgillus_minisclerotigenes_GCA_009176455\") has been submitted\n",
      "Your job 5484737 (\"map_Asprgillus_mulundensis_GCF_003369625\") has been submitted\n",
      "Your job 5484738 (\"map_Asprgillus_neoellipticus_GCA_003116565\") has been submitted\n",
      "Your job 5484739 (\"map_Asprgillus_neoniger_GCF_003184625\") has been submitted\n",
      "Your job 5484740 (\"map_Asprgillus_nidulans_GCF_000149205\") has been submitted\n",
      "Your job 5484741 (\"map_Asprgillus_niger_GCF_000002855\") has been submitted\n",
      "Your job 5484742 (\"map_Asprgillus_nomius_GCF_001204775\") has been submitted\n",
      "Your job 5484743 (\"map_Asprgillus_novofumigatus_GCF_002847465\") has been submitted\n",
      "Your job 5484744 (\"map_Asprgillus_novoparasiticus_GCA_009176405\") has been submitted\n",
      "Your job 5484745 (\"map_Asprgillus_ochraceoroseus_GCA_002846915\") has been submitted\n",
      "Your job 5484746 (\"map_Asprgillus_ochraceus_GCA_004849945\") has been submitted\n",
      "Your job 5484747 (\"map_Asprgillus_olivimuriae_GCA_003719415\") has been submitted\n",
      "Your job 5484748 (\"map_Asprgillus_oryzae_GCF_000184455\") has been submitted\n",
      "Your job 5484749 (\"map_Asprgillus_parasiticus_GCA_009176385\") has been submitted\n",
      "Your job 5484750 (\"map_Asprgillus_persii_GCA_002215965\") has been submitted\n",
      "Your job 5484751 (\"map_Asprgillus_phoenicis_GCA_003344505\") has been submitted\n",
      "Your job 5484752 (\"map_Asprgillus_piperis_GCF_003184755\") has been submitted\n",
      "Your job 5484753 (\"map_Asprgillus_pseudocaelatus_GCA_009193665\") has been submitted\n",
      "Your job 5484754 (\"map_Asprgillus_pseudotamarii_GCF_009193445\") has been submitted\n",
      "Your job 5484755 (\"map_Asprgillus_pseudoterreus_GCA_002927005\") has been submitted\n",
      "Your job 5484756 (\"map_Asprgillus_rambellii_GCA_000986645\") has been submitted\n",
      "Your job 5484757 (\"map_Asprgillus_ruber_GCA_000600275\") has been submitted\n",
      "Your job 5484758 (\"map_Asprgillus_saccharolyticus_GCF_003184585\") has been submitted\n",
      "Your job 5484759 (\"map_Asprgillus_sclerotialis_GCA_003589665\") has been submitted\n",
      "Your job 5484760 (\"map_Asprgillus_sclerotiicarbonarius_GCA_003184635\") has been submitted\n",
      "Your job 5484761 (\"map_Asprgillus_sclerotioniger_GCF_003184525\") has been submitted\n",
      "Your job 5484762 (\"map_Asprgillus_sclerotiorum_GCA_000530345\") has been submitted\n",
      "Your job 5484763 (\"map_Asprgillus_sergii_GCA_009193525\") has been submitted\n",
      "Your job 5484764 (\"map_Asprgillus_sojae_GCA_008274985\") has been submitted\n",
      "Your job 5484765 (\"map_Asprgillus_sp_GCA_001044295\") has been submitted\n",
      "Your job 5484766 (\"map_Asprgillus_spnulosprus_GCA_003574815\") has been submitted\n",
      "Your job 5484767 (\"map_Asprgillus_steynii_GCF_002849105\") has been submitted\n",
      "Your job 5484768 (\"map_Asprgillus_sydowii_GCA_001890705\") has been submitted\n",
      "Your job 5484769 (\"map_Asprgillus_taichungensis_GCA_002850765\") has been submitted\n",
      "Your job 5484770 (\"map_Asprgillus_tamarii_GCA_009193485\") has been submitted\n",
      "Your job 5484771 (\"map_Asprgillus_tanneri_GCA_003426965\") has been submitted\n",
      "Your job 5484772 (\"map_Asprgillus_terreus_GCF_000149615\") has been submitted\n",
      "Your job 5484773 (\"map_Asprgillus_thermomutatus_GCF_002237265\") has been submitted\n",
      "Your job 5484774 (\"map_Asprgillus_transmontanensis_GCA_009193505\") has been submitted\n",
      "Your job 5484775 (\"map_Asprgillus_tritici_GCA_009812425\") has been submitted\n",
      "Your job 5484776 (\"map_Asprgillus_tubingensis_GCA_001890745\") has been submitted\n",
      "Your job 5484777 (\"map_Asprgillus_turcosus_GCA_002234965\") has been submitted\n",
      "Your job 5484778 (\"map_Asprgillus_udagawae_GCA_001078395\") has been submitted\n",
      "Your job 5484779 (\"map_Asprgillus_unguis_GCA_003324175\") has been submitted\n",
      "Your job 5484780 (\"map_Asprgillus_ustus_GCA_000812125\") has been submitted\n",
      "Your job 5484781 (\"map_Asprgillus_uvarum_GCF_003184745\") has been submitted\n",
      "Your job 5484782 (\"map_Asprgillus_vadensis_GCF_003184925\") has been submitted\n",
      "Your job 5484783 (\"map_Asprgillus_versicolor_GCA_001890125\") has been submitted\n",
      "Your job 5484784 (\"map_Asprgillus_violaceofuscus_GCA_003184705\") has been submitted\n",
      "Your job 5484785 (\"map_Asprgillus_viridinutans_GCA_004368095\") has been submitted\n",
      "Your job 5484786 (\"map_Asprgillus_welwitschiae_GCF_003344945\") has been submitted\n",
      "Your job 5484787 (\"map_Asprgillus_wentii_GCA_001890725\") has been submitted\n",
      "Your job 5484788 (\"map_Asprgillus_westerdijkiae_GCA_001307345\") has been submitted\n",
      "Your job 5484789 (\"map_Basipetospra_chlamydospra_GCA_001599675\") has been submitted\n",
      "Your job 5484790 (\"map_Byssochlamys_nivea_GCA_003116535\") has been submitted\n",
      "Your job 5484791 (\"map_Byssochlamys_sp_GCA_002914405\") has been submitted\n",
      "Your job 5484792 (\"map_Byssochlamys_spctabilis_GCF_004022145\") has been submitted\n",
      "Your job 5484793 (\"map_Elaphomyces_granulatus_GCA_002240705\") has been submitted\n",
      "Your job 5484794 (\"map_Monascus_purpureus_GCA_003184285\") has been submitted\n",
      "Your job 5484795 (\"map_Monascus_ruber_GCA_002976275\") has been submitted\n",
      "Your job 5484796 (\"map_Penicilliopsis_zonata_GCF_001890105\") has been submitted\n",
      "Your job 5484797 (\"map_Penicillium_antarcticum_GCA_002072345\") has been submitted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484798 (\"map_Penicillium_arizonense_GCF_001773325\") has been submitted\n",
      "Your job 5484799 (\"map_Penicillium_biforme_GCA_000577785\") has been submitted\n",
      "Your job 5484800 (\"map_Penicillium_brasilianum_GCA_001048715\") has been submitted\n",
      "Your job 5484801 (\"map_Penicillium_camemberti_GCA_000513335\") has been submitted\n",
      "Your job 5484802 (\"map_Penicillium_capsulatum_GCA_000943775\") has been submitted\n",
      "Your job 5484803 (\"map_Penicillium_carneum_GCA_000577495\") has been submitted\n",
      "Your job 5484804 (\"map_Penicillium_chrysogenum_GCA_000710275\") has been submitted\n",
      "Your job 5484805 (\"map_Penicillium_citrinum_GCA_001950535\") has been submitted\n",
      "Your job 5484806 (\"map_Penicillium_coprophilum_GCA_002072405\") has been submitted\n",
      "Your job 5484807 (\"map_Penicillium_decumbens_GCA_002072245\") has been submitted\n",
      "Your job 5484808 (\"map_Penicillium_digitatum_GCF_000315645\") has been submitted\n",
      "Your job 5484809 (\"map_Penicillium_expansum_GCF_000769745\") has been submitted\n",
      "Your job 5484810 (\"map_Penicillium_flavigenum_GCA_002072365\") has been submitted\n",
      "Your job 5484811 (\"map_Penicillium_freii_GCA_001513925\") has been submitted\n",
      "Your job 5484812 (\"map_Penicillium_fuscoglaucum_GCA_000576735\") has been submitted\n",
      "Your job 5484813 (\"map_Penicillium_griseofulvum_GCA_001561935\") has been submitted\n",
      "Your job 5484814 (\"map_Penicillium_italicum_GCA_002116305\") has been submitted\n",
      "Your job 5484815 (\"map_Penicillium_janthinellum_GCA_002369805\") has been submitted\n",
      "Your job 5484816 (\"map_Penicillium_nalgiovense_GCA_002072425\") has been submitted\n",
      "Your job 5484817 (\"map_Penicillium_nordicum_GCA_000733025\") has been submitted\n",
      "Your job 5484818 (\"map_Penicillium_oxalicum_GCA_000346795\") has been submitted\n",
      "Your job 5484819 (\"map_Penicillium_paneum_GCA_000577715\") has been submitted\n",
      "Your job 5484820 (\"map_Penicillium_paxilli_GCA_000347475\") has been submitted\n",
      "Your job 5484821 (\"map_Penicillium_polonicum_GCA_003344595\") has been submitted\n",
      "Your job 5484822 (\"map_Penicillium_roqueforti_GCA_000513255\") has been submitted\n",
      "Your job 5484823 (\"map_Penicillium_rubens_GCF_000226395\") has been submitted\n",
      "Your job 5484824 (\"map_Penicillium_sclerotiorum_GCA_001750025\") has been submitted\n",
      "Your job 5484825 (\"map_Penicillium_solitum_GCA_000952775\") has been submitted\n",
      "Your job 5484826 (\"map_Penicillium_sp_GCA_003800485\") has been submitted\n",
      "Your job 5484827 (\"map_Penicillium_steckii_GCA_002072375\") has been submitted\n",
      "Your job 5484828 (\"map_Penicillium_subrubescens_GCA_001908125\") has been submitted\n",
      "Your job 5484829 (\"map_Penicillium_verrucosum_GCA_000970515\") has been submitted\n",
      "Your job 5484830 (\"map_Penicillium_vulpinum_GCA_002072255\") has been submitted\n",
      "Your job 5484831 (\"map_Phialosimplex_sp_GCA_003698115\") has been submitted\n",
      "Your job 5484832 (\"map_Rasamsonia_emersonii_GCF_000968595\") has been submitted\n",
      "Your job 5484833 (\"map_Talaromyces_adpressus_GCA_002775195\") has been submitted\n",
      "Your job 5484834 (\"map_Talaromyces_amestolkiae_GCA_001896365\") has been submitted\n",
      "Your job 5484835 (\"map_Talaromyces_atroroseus_GCF_001907595\") has been submitted\n",
      "Your job 5484836 (\"map_Talaromyces_borbonicus_GCA_002916415\") has been submitted\n",
      "Your job 5484837 (\"map_Talaromyces_cellulolyticus_GCA_000829775\") has been submitted\n",
      "Your job 5484838 (\"map_Talaromyces_funiculosus_GCA_004299765\") has been submitted\n",
      "Your job 5484839 (\"map_Talaromyces_islandicus_GCA_000985935\") has been submitted\n",
      "Your job 5484840 (\"map_Talaromyces_piceae_GCA_001657655\") has been submitted\n",
      "Your job 5484841 (\"map_Talaromyces_pinophilus_GCA_001571465\") has been submitted\n",
      "Your job 5484842 (\"map_Talaromyces_purpureogenus_GCA_001270325\") has been submitted\n",
      "Your job 5484843 (\"map_Talaromyces_stipitatus_GCF_000003125\") has been submitted\n",
      "Your job 5484844 (\"map_Talaromyces_verruculosus_GCA_001305275\") has been submitted\n",
      "Your job 5484845 (\"map_Talaromyces_wortmannii_GCA_001939245\") has been submitted\n",
      "Your job 5484846 (\"map_Thermoascaceae_sp_GCA_003123655\") has been submitted\n",
      "Your job 5484847 (\"map_Thermoascus_crustaceus_GCA_001599835\") has been submitted\n",
      "Your job 5484848 (\"map_Thermomyces_lanuginosus_GCA_000315935\") has been submitted\n",
      "Your job 5484849 (\"map_Xeromyces_bisprus_GCA_900006255\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484850 (\"merge_Asprgillus_aculeatinus_GCF_003184765\") has been submitted\n",
      "Your job 5484851 (\"merge_Asprgillus_aculeatus_GCF_001890905\") has been submitted\n",
      "Your job 5484852 (\"merge_Asprgillus_aff_GCA_001931935\") has been submitted\n",
      "Your job 5484853 (\"merge_Asprgillus_albertensis_GCA_009193605\") has been submitted\n",
      "Your job 5484854 (\"merge_Asprgillus_alliaceus_GCF_009176365\") has been submitted\n",
      "Your job 5484855 (\"merge_Asprgillus_amoenus_GCA_009812435\") has been submitted\n",
      "Your job 5484856 (\"merge_Asprgillus_arachidicola_GCA_009193545\") has been submitted\n",
      "Your job 5484857 (\"merge_Asprgillus_avenaceus_GCA_009193465\") has been submitted\n",
      "Your job 5484858 (\"merge_Asprgillus_awamori_GCA_003850985\") has been submitted\n",
      "Your job 5484859 (\"merge_Asprgillus_bertholletius_GCA_009193595\") has been submitted\n",
      "Your job 5484860 (\"merge_Asprgillus_bombycis_GCF_001792695\") has been submitted\n",
      "Your job 5484861 (\"merge_Asprgillus_brasiliensis_GCA_001889945\") has been submitted\n",
      "Your job 5484862 (\"merge_Asprgillus_brunneoviolaceus_GCF_003184695\") has been submitted\n",
      "Your job 5484863 (\"merge_Asprgillus_caelatus_GCF_009193585\") has been submitted\n",
      "Your job 5484864 (\"merge_Asprgillus_calidoustus_GCA_001511075\") has been submitted\n",
      "Your job 5484865 (\"merge_Asprgillus_campestris_GCF_002847485\") has been submitted\n",
      "Your job 5484866 (\"merge_Asprgillus_candidus_GCF_002847045\") has been submitted\n",
      "Your job 5484867 (\"merge_Asprgillus_carbonarius_GCA_001990825\") has been submitted\n",
      "Your job 5484868 (\"merge_Asprgillus_cejpii_GCA_004769165\") has been submitted\n",
      "Your job 5484869 (\"merge_Asprgillus_chevalieri_GCA_001599875\") has been submitted\n",
      "Your job 5484870 (\"merge_Asprgillus_clavatus_GCF_000002715\") has been submitted\n",
      "Your job 5484871 (\"merge_Asprgillus_coremiiformis_GCA_009193565\") has been submitted\n",
      "Your job 5484872 (\"merge_Asprgillus_costaricaensis_GCF_003184835\") has been submitted\n",
      "Your job 5484873 (\"merge_Asprgillus_cristatus_GCA_001693355\") has been submitted\n",
      "Your job 5484874 (\"merge_Asprgillus_ellipticus_GCA_003184645\") has been submitted\n",
      "Your job 5484875 (\"merge_Asprgillus_eucalypticola_GCF_003184535\") has been submitted\n",
      "Your job 5484876 (\"merge_Asprgillus_fijiensis_GCA_003184825\") has been submitted\n",
      "Your job 5484877 (\"merge_Asprgillus_fischeri_GCF_000149645\") has been submitted\n",
      "Your job 5484878 (\"merge_Asprgillus_flavus_GCF_000006275\") has been submitted\n",
      "Your job 5484879 (\"merge_Asprgillus_fumigatus_GCF_000002655\") has been submitted\n",
      "Your job 5484880 (\"merge_Asprgillus_glaucus_GCF_001890805\") has been submitted\n",
      "Your job 5484881 (\"merge_Asprgillus_hancockii_GCA_001696595\") has been submitted\n",
      "Your job 5484882 (\"merge_Asprgillus_heteromorphus_GCF_003184545\") has been submitted\n",
      "Your job 5484883 (\"merge_Asprgillus_homomorphus_GCF_003184865\") has been submitted\n",
      "Your job 5484884 (\"merge_Asprgillus_ibericus_GCF_003184845\") has been submitted\n",
      "Your job 5484885 (\"merge_Asprgillus_indologenus_GCA_003184685\") has been submitted\n",
      "Your job 5484886 (\"merge_Asprgillus_japonicus_GCF_003184785\") has been submitted\n",
      "Your job 5484887 (\"merge_Asprgillus_kawachii_GCA_000239835\") has been submitted\n",
      "Your job 5484888 (\"merge_Asprgillus_lentulus_GCA_001445615\") has been submitted\n",
      "Your job 5484889 (\"merge_Asprgillus_leporis_GCA_009176345\") has been submitted\n",
      "Your job 5484890 (\"merge_Asprgillus_luchuensis_GCA_001890685\") has been submitted\n",
      "Your job 5484891 (\"merge_Asprgillus_minisclerotigenes_GCA_009176455\") has been submitted\n",
      "Your job 5484892 (\"merge_Asprgillus_mulundensis_GCF_003369625\") has been submitted\n",
      "Your job 5484893 (\"merge_Asprgillus_neoellipticus_GCA_003116565\") has been submitted\n",
      "Your job 5484894 (\"merge_Asprgillus_neoniger_GCF_003184625\") has been submitted\n",
      "Your job 5484895 (\"merge_Asprgillus_nidulans_GCF_000149205\") has been submitted\n",
      "Your job 5484896 (\"merge_Asprgillus_niger_GCF_000002855\") has been submitted\n",
      "Your job 5484897 (\"merge_Asprgillus_nomius_GCF_001204775\") has been submitted\n",
      "Your job 5484898 (\"merge_Asprgillus_novofumigatus_GCF_002847465\") has been submitted\n",
      "Your job 5484899 (\"merge_Asprgillus_novoparasiticus_GCA_009176405\") has been submitted\n",
      "Your job 5484900 (\"merge_Asprgillus_ochraceoroseus_GCA_002846915\") has been submitted\n",
      "Your job 5484901 (\"merge_Asprgillus_ochraceus_GCA_004849945\") has been submitted\n",
      "Your job 5484902 (\"merge_Asprgillus_olivimuriae_GCA_003719415\") has been submitted\n",
      "Your job 5484903 (\"merge_Asprgillus_oryzae_GCF_000184455\") has been submitted\n",
      "Your job 5484904 (\"merge_Asprgillus_parasiticus_GCA_009176385\") has been submitted\n",
      "Your job 5484905 (\"merge_Asprgillus_persii_GCA_002215965\") has been submitted\n",
      "Your job 5484906 (\"merge_Asprgillus_phoenicis_GCA_003344505\") has been submitted\n",
      "Your job 5484907 (\"merge_Asprgillus_piperis_GCF_003184755\") has been submitted\n",
      "Your job 5484908 (\"merge_Asprgillus_pseudocaelatus_GCA_009193665\") has been submitted\n",
      "Your job 5484909 (\"merge_Asprgillus_pseudotamarii_GCF_009193445\") has been submitted\n",
      "Your job 5484910 (\"merge_Asprgillus_pseudoterreus_GCA_002927005\") has been submitted\n",
      "Your job 5484911 (\"merge_Asprgillus_rambellii_GCA_000986645\") has been submitted\n",
      "Your job 5484912 (\"merge_Asprgillus_ruber_GCA_000600275\") has been submitted\n",
      "Your job 5484913 (\"merge_Asprgillus_saccharolyticus_GCF_003184585\") has been submitted\n",
      "Your job 5484914 (\"merge_Asprgillus_sclerotialis_GCA_003589665\") has been submitted\n",
      "Your job 5484915 (\"merge_Asprgillus_sclerotiicarbonarius_GCA_003184635\") has been submitted\n",
      "Your job 5484916 (\"merge_Asprgillus_sclerotioniger_GCF_003184525\") has been submitted\n",
      "Your job 5484917 (\"merge_Asprgillus_sclerotiorum_GCA_000530345\") has been submitted\n",
      "Your job 5484918 (\"merge_Asprgillus_sergii_GCA_009193525\") has been submitted\n",
      "Your job 5484919 (\"merge_Asprgillus_sojae_GCA_008274985\") has been submitted\n",
      "Your job 5484920 (\"merge_Asprgillus_sp_GCA_001044295\") has been submitted\n",
      "Your job 5484921 (\"merge_Asprgillus_spnulosprus_GCA_003574815\") has been submitted\n",
      "Your job 5484922 (\"merge_Asprgillus_steynii_GCF_002849105\") has been submitted\n",
      "Your job 5484923 (\"merge_Asprgillus_sydowii_GCA_001890705\") has been submitted\n",
      "Your job 5484924 (\"merge_Asprgillus_taichungensis_GCA_002850765\") has been submitted\n",
      "Your job 5484925 (\"merge_Asprgillus_tamarii_GCA_009193485\") has been submitted\n",
      "Your job 5484926 (\"merge_Asprgillus_tanneri_GCA_003426965\") has been submitted\n",
      "Your job 5484927 (\"merge_Asprgillus_terreus_GCF_000149615\") has been submitted\n",
      "Your job 5484928 (\"merge_Asprgillus_thermomutatus_GCF_002237265\") has been submitted\n",
      "Your job 5484929 (\"merge_Asprgillus_transmontanensis_GCA_009193505\") has been submitted\n",
      "Your job 5484930 (\"merge_Asprgillus_tritici_GCA_009812425\") has been submitted\n",
      "Your job 5484931 (\"merge_Asprgillus_tubingensis_GCA_001890745\") has been submitted\n",
      "Your job 5484932 (\"merge_Asprgillus_turcosus_GCA_002234965\") has been submitted\n",
      "Your job 5484933 (\"merge_Asprgillus_udagawae_GCA_001078395\") has been submitted\n",
      "Your job 5484934 (\"merge_Asprgillus_unguis_GCA_003324175\") has been submitted\n",
      "Your job 5484935 (\"merge_Asprgillus_ustus_GCA_000812125\") has been submitted\n",
      "Your job 5484936 (\"merge_Asprgillus_uvarum_GCF_003184745\") has been submitted\n",
      "Your job 5484937 (\"merge_Asprgillus_vadensis_GCF_003184925\") has been submitted\n",
      "Your job 5484938 (\"merge_Asprgillus_versicolor_GCA_001890125\") has been submitted\n",
      "Your job 5484939 (\"merge_Asprgillus_violaceofuscus_GCA_003184705\") has been submitted\n",
      "Your job 5484940 (\"merge_Asprgillus_viridinutans_GCA_004368095\") has been submitted\n",
      "Your job 5484941 (\"merge_Asprgillus_welwitschiae_GCF_003344945\") has been submitted\n",
      "Your job 5484942 (\"merge_Asprgillus_wentii_GCA_001890725\") has been submitted\n",
      "Your job 5484943 (\"merge_Asprgillus_westerdijkiae_GCA_001307345\") has been submitted\n",
      "Your job 5484944 (\"merge_Basipetospra_chlamydospra_GCA_001599675\") has been submitted\n",
      "Your job 5484945 (\"merge_Byssochlamys_nivea_GCA_003116535\") has been submitted\n",
      "Your job 5484946 (\"merge_Byssochlamys_sp_GCA_002914405\") has been submitted\n",
      "Your job 5484947 (\"merge_Byssochlamys_spctabilis_GCF_004022145\") has been submitted\n",
      "Your job 5484948 (\"merge_Elaphomyces_granulatus_GCA_002240705\") has been submitted\n",
      "Your job 5484949 (\"merge_Monascus_purpureus_GCA_003184285\") has been submitted\n",
      "Your job 5484950 (\"merge_Monascus_ruber_GCA_002976275\") has been submitted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484951 (\"merge_Penicilliopsis_zonata_GCF_001890105\") has been submitted\n",
      "Your job 5484952 (\"merge_Penicillium_antarcticum_GCA_002072345\") has been submitted\n",
      "Your job 5484953 (\"merge_Penicillium_arizonense_GCF_001773325\") has been submitted\n",
      "Your job 5484954 (\"merge_Penicillium_biforme_GCA_000577785\") has been submitted\n",
      "Your job 5484955 (\"merge_Penicillium_brasilianum_GCA_001048715\") has been submitted\n",
      "Your job 5484956 (\"merge_Penicillium_camemberti_GCA_000513335\") has been submitted\n",
      "Your job 5484957 (\"merge_Penicillium_capsulatum_GCA_000943775\") has been submitted\n",
      "Your job 5484958 (\"merge_Penicillium_carneum_GCA_000577495\") has been submitted\n",
      "Your job 5484959 (\"merge_Penicillium_chrysogenum_GCA_000710275\") has been submitted\n",
      "Your job 5484960 (\"merge_Penicillium_citrinum_GCA_001950535\") has been submitted\n",
      "Your job 5484961 (\"merge_Penicillium_coprophilum_GCA_002072405\") has been submitted\n",
      "Your job 5484962 (\"merge_Penicillium_decumbens_GCA_002072245\") has been submitted\n",
      "Your job 5484963 (\"merge_Penicillium_digitatum_GCF_000315645\") has been submitted\n",
      "Your job 5484964 (\"merge_Penicillium_expansum_GCF_000769745\") has been submitted\n",
      "Your job 5484965 (\"merge_Penicillium_flavigenum_GCA_002072365\") has been submitted\n",
      "Your job 5484966 (\"merge_Penicillium_freii_GCA_001513925\") has been submitted\n",
      "Your job 5484967 (\"merge_Penicillium_fuscoglaucum_GCA_000576735\") has been submitted\n",
      "Your job 5484968 (\"merge_Penicillium_griseofulvum_GCA_001561935\") has been submitted\n",
      "Your job 5484969 (\"merge_Penicillium_italicum_GCA_002116305\") has been submitted\n",
      "Your job 5484970 (\"merge_Penicillium_janthinellum_GCA_002369805\") has been submitted\n",
      "Your job 5484971 (\"merge_Penicillium_nalgiovense_GCA_002072425\") has been submitted\n",
      "Your job 5484972 (\"merge_Penicillium_nordicum_GCA_000733025\") has been submitted\n",
      "Your job 5484973 (\"merge_Penicillium_oxalicum_GCA_000346795\") has been submitted\n",
      "Your job 5484974 (\"merge_Penicillium_paneum_GCA_000577715\") has been submitted\n",
      "Your job 5484975 (\"merge_Penicillium_paxilli_GCA_000347475\") has been submitted\n",
      "Your job 5484976 (\"merge_Penicillium_polonicum_GCA_003344595\") has been submitted\n",
      "Your job 5484977 (\"merge_Penicillium_roqueforti_GCA_000513255\") has been submitted\n",
      "Your job 5484978 (\"merge_Penicillium_rubens_GCF_000226395\") has been submitted\n",
      "Your job 5484979 (\"merge_Penicillium_sclerotiorum_GCA_001750025\") has been submitted\n",
      "Your job 5484980 (\"merge_Penicillium_solitum_GCA_000952775\") has been submitted\n",
      "Your job 5484981 (\"merge_Penicillium_sp_GCA_003800485\") has been submitted\n",
      "Your job 5484982 (\"merge_Penicillium_steckii_GCA_002072375\") has been submitted\n",
      "Your job 5484983 (\"merge_Penicillium_subrubescens_GCA_001908125\") has been submitted\n",
      "Your job 5484984 (\"merge_Penicillium_verrucosum_GCA_000970515\") has been submitted\n",
      "Your job 5484985 (\"merge_Penicillium_vulpinum_GCA_002072255\") has been submitted\n",
      "Your job 5484986 (\"merge_Phialosimplex_sp_GCA_003698115\") has been submitted\n",
      "Your job 5484987 (\"merge_Rasamsonia_emersonii_GCF_000968595\") has been submitted\n",
      "Your job 5484988 (\"merge_Talaromyces_adpressus_GCA_002775195\") has been submitted\n",
      "Your job 5484989 (\"merge_Talaromyces_amestolkiae_GCA_001896365\") has been submitted\n",
      "Your job 5484990 (\"merge_Talaromyces_atroroseus_GCF_001907595\") has been submitted\n",
      "Your job 5484991 (\"merge_Talaromyces_borbonicus_GCA_002916415\") has been submitted\n",
      "Your job 5484992 (\"merge_Talaromyces_cellulolyticus_GCA_000829775\") has been submitted\n",
      "Your job 5484993 (\"merge_Talaromyces_funiculosus_GCA_004299765\") has been submitted\n",
      "Your job 5484994 (\"merge_Talaromyces_islandicus_GCA_000985935\") has been submitted\n",
      "Your job 5484995 (\"merge_Talaromyces_piceae_GCA_001657655\") has been submitted\n",
      "Your job 5484996 (\"merge_Talaromyces_pinophilus_GCA_001571465\") has been submitted\n",
      "Your job 5484997 (\"merge_Talaromyces_purpureogenus_GCA_001270325\") has been submitted\n",
      "Your job 5484998 (\"merge_Talaromyces_stipitatus_GCF_000003125\") has been submitted\n",
      "Your job 5484999 (\"merge_Talaromyces_verruculosus_GCA_001305275\") has been submitted\n",
      "Your job 5485000 (\"merge_Talaromyces_wortmannii_GCA_001939245\") has been submitted\n",
      "Your job 5485001 (\"merge_Thermoascaceae_sp_GCA_003123655\") has been submitted\n",
      "Your job 5485002 (\"merge_Thermoascus_crustaceus_GCA_001599835\") has been submitted\n",
      "Your job 5485003 (\"merge_Thermomyces_lanuginosus_GCA_000315935\") has been submitted\n",
      "Your job 5485004 (\"merge_Xeromyces_bisprus_GCA_900006255\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 419 sequences from Asprgillus_aculeatinus_GCF_003184765_merged.bed.  Filtered 310 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 467 sequences from Asprgillus_aculeatus_GCF_001890905_merged.bed.  Filtered 335 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 132.\n",
      "Screened 461 sequences from Asprgillus_aff_GCA_001931935_merged.bed.  Filtered 333 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 128.\n",
      "Screened 341 sequences from Asprgillus_albertensis_GCA_009193605_merged.bed.  Filtered 256 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 355 sequences from Asprgillus_alliaceus_GCF_009176365_merged.bed.  Filtered 268 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 87.\n",
      "Screened 394 sequences from Asprgillus_amoenus_GCA_009812435_merged.bed.  Filtered 284 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 394 sequences from Asprgillus_arachidicola_GCA_009193545_merged.bed.  Filtered 283 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 362 sequences from Asprgillus_avenaceus_GCA_009193465_merged.bed.  Filtered 254 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 108.\n",
      "Screened 603 sequences from Asprgillus_awamori_GCA_003850985_merged.bed.  Filtered 414 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 189.\n",
      "Screened 355 sequences from Asprgillus_bertholletius_GCA_009193595_merged.bed.  Filtered 258 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 97.\n",
      "Screened 446 sequences from Asprgillus_bombycis_GCF_001792695_merged.bed.  Filtered 310 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 136.\n",
      "Screened 484 sequences from Asprgillus_brasiliensis_GCA_001889945_merged.bed.  Filtered 350 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 134.\n",
      "Screened 403 sequences from Asprgillus_brunneoviolaceus_GCF_003184695_merged.bed.  Filtered 291 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 112.\n",
      "Screened 372 sequences from Asprgillus_caelatus_GCF_009193585_merged.bed.  Filtered 281 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 91.\n",
      "Screened 503 sequences from Asprgillus_calidoustus_GCA_001511075_merged.bed.  Filtered 357 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 146.\n",
      "Screened 363 sequences from Asprgillus_campestris_GCF_002847485_merged.bed.  Filtered 267 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96.\n",
      "Screened 322 sequences from Asprgillus_candidus_GCF_002847045_merged.bed.  Filtered 252 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 70.\n",
      "Screened 422 sequences from Asprgillus_carbonarius_GCA_001990825_merged.bed.  Filtered 286 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 136.\n",
      "Screened 571 sequences from Asprgillus_cejpii_GCA_004769165_merged.bed.  Filtered 401 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 170.\n",
      "Screened 404 sequences from Asprgillus_chevalieri_GCA_001599875_merged.bed.  Filtered 309 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 95.\n",
      "Screened 483 sequences from Asprgillus_clavatus_GCF_000002715_merged.bed.  Filtered 311 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 172.\n",
      "Screened 312 sequences from Asprgillus_coremiiformis_GCA_009193565_merged.bed.  Filtered 243 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 447 sequences from Asprgillus_costaricaensis_GCF_003184835_merged.bed.  Filtered 324 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 123.\n",
      "Screened 494 sequences from Asprgillus_cristatus_GCA_001693355_merged.bed.  Filtered 347 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 147.\n",
      "Screened 426 sequences from Asprgillus_ellipticus_GCA_003184645_merged.bed.  Filtered 306 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 120.\n",
      "Screened 485 sequences from Asprgillus_eucalypticola_GCF_003184535_merged.bed.  Filtered 341 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 144.\n",
      "Screened 416 sequences from Asprgillus_fijiensis_GCA_003184825_merged.bed.  Filtered 307 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 574 sequences from Asprgillus_fischeri_GCF_000149645_merged.bed.  Filtered 387 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 187.\n",
      "Screened 390 sequences from Asprgillus_flavus_GCF_000006275_merged.bed.  Filtered 276 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 114.\n",
      "Screened 395 sequences from Asprgillus_fumigatus_GCF_000002655_merged.bed.  Filtered 278 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 387 sequences from Asprgillus_glaucus_GCF_001890805_merged.bed.  Filtered 283 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 104.\n",
      "Screened 406 sequences from Asprgillus_hancockii_GCA_001696595_merged.bed.  Filtered 280 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 126.\n",
      "Screened 428 sequences from Asprgillus_heteromorphus_GCF_003184545_merged.bed.  Filtered 304 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 124.\n",
      "Screened 411 sequences from Asprgillus_homomorphus_GCF_003184865_merged.bed.  Filtered 304 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 418 sequences from Asprgillus_ibericus_GCF_003184845_merged.bed.  Filtered 295 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 123.\n",
      "Screened 399 sequences from Asprgillus_indologenus_GCA_003184685_merged.bed.  Filtered 290 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 398 sequences from Asprgillus_japonicus_GCF_003184785_merged.bed.  Filtered 291 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 571 sequences from Asprgillus_kawachii_GCA_000239835_merged.bed.  Filtered 419 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 152.\n",
      "Screened 536 sequences from Asprgillus_lentulus_GCA_001445615_merged.bed.  Filtered 345 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 191.\n",
      "Screened 381 sequences from Asprgillus_leporis_GCA_009176345_merged.bed.  Filtered 268 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 527 sequences from Asprgillus_luchuensis_GCA_001890685_merged.bed.  Filtered 390 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 137.\n",
      "Screened 363 sequences from Asprgillus_minisclerotigenes_GCA_009176455_merged.bed.  Filtered 259 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 104.\n",
      "Screened 392 sequences from Asprgillus_mulundensis_GCF_003369625_merged.bed.  Filtered 290 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102.\n",
      "Screened 477 sequences from Asprgillus_neoellipticus_GCA_003116565_merged.bed.  Filtered 327 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 150.\n",
      "Screened 446 sequences from Asprgillus_neoniger_GCF_003184625_merged.bed.  Filtered 324 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 122.\n",
      "Screened 421 sequences from Asprgillus_nidulans_GCF_000149205_merged.bed.  Filtered 316 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 105.\n",
      "Screened 489 sequences from Asprgillus_niger_GCF_000002855_merged.bed.  Filtered 336 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 153.\n",
      "Screened 439 sequences from Asprgillus_nomius_GCF_001204775_merged.bed.  Filtered 296 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 143.\n",
      "Screened 469 sequences from Asprgillus_novofumigatus_GCF_002847465_merged.bed.  Filtered 321 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 148.\n",
      "Screened 391 sequences from Asprgillus_novoparasiticus_GCA_009176405_merged.bed.  Filtered 284 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 397 sequences from Asprgillus_ochraceoroseus_GCA_002846915_merged.bed.  Filtered 266 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 485 sequences from Asprgillus_ochraceus_GCA_004849945_merged.bed.  Filtered 343 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 142.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 421 sequences from Asprgillus_olivimuriae_GCA_003719415_merged.bed.  Filtered 300 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 121.\n",
      "Screened 464 sequences from Asprgillus_oryzae_GCF_000184455_merged.bed.  Filtered 320 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 144.\n",
      "Screened 378 sequences from Asprgillus_parasiticus_GCA_009176385_merged.bed.  Filtered 276 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102.\n",
      "Screened 398 sequences from Asprgillus_persii_GCA_002215965_merged.bed.  Filtered 283 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 115.\n",
      "Screened 453 sequences from Asprgillus_phoenicis_GCA_003344505_merged.bed.  Filtered 330 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 123.\n",
      "Screened 444 sequences from Asprgillus_piperis_GCF_003184755_merged.bed.  Filtered 327 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 375 sequences from Asprgillus_pseudocaelatus_GCA_009193665_merged.bed.  Filtered 282 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 93.\n",
      "Screened 353 sequences from Asprgillus_pseudotamarii_GCF_009193445_merged.bed.  Filtered 264 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 455 sequences from Asprgillus_pseudoterreus_GCA_002927005_merged.bed.  Filtered 321 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 134.\n",
      "Screened 479 sequences from Asprgillus_rambellii_GCA_000986645_merged.bed.  Filtered 330 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 149.\n",
      "Screened 415 sequences from Asprgillus_ruber_GCA_000600275_merged.bed.  Filtered 306 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 379 sequences from Asprgillus_saccharolyticus_GCF_003184585_merged.bed.  Filtered 283 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96.\n",
      "Screened 362 sequences from Asprgillus_sclerotialis_GCA_003589665_merged.bed.  Filtered 273 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 421 sequences from Asprgillus_sclerotiicarbonarius_GCA_003184635_merged.bed.  Filtered 301 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 120.\n",
      "Screened 442 sequences from Asprgillus_sclerotioniger_GCF_003184525_merged.bed.  Filtered 311 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 430 sequences from Asprgillus_sclerotiorum_GCA_000530345_merged.bed.  Filtered 305 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 125.\n",
      "Screened 375 sequences from Asprgillus_sergii_GCA_009193525_merged.bed.  Filtered 277 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 98.\n",
      "Screened 385 sequences from Asprgillus_sojae_GCA_008274985_merged.bed.  Filtered 289 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96.\n",
      "Screened 470 sequences from Asprgillus_sp_GCA_001044295_merged.bed.  Filtered 335 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 135.\n",
      "Screened 428 sequences from Asprgillus_spnulosprus_GCA_003574815_merged.bed.  Filtered 318 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 407 sequences from Asprgillus_steynii_GCF_002849105_merged.bed.  Filtered 282 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 125.\n",
      "Screened 357 sequences from Asprgillus_sydowii_GCA_001890705_merged.bed.  Filtered 263 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 305 sequences from Asprgillus_taichungensis_GCA_002850765_merged.bed.  Filtered 233 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 72.\n",
      "Screened 366 sequences from Asprgillus_tamarii_GCA_009193485_merged.bed.  Filtered 273 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 93.\n",
      "Screened 335 sequences from Asprgillus_tanneri_GCA_003426965_merged.bed.  Filtered 236 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 99.\n",
      "Screened 500 sequences from Asprgillus_terreus_GCF_000149615_merged.bed.  Filtered 338 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 162.\n",
      "Screened 486 sequences from Asprgillus_thermomutatus_GCF_002237265_merged.bed.  Filtered 332 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 154.\n",
      "Screened 376 sequences from Asprgillus_transmontanensis_GCA_009193505_merged.bed.  Filtered 269 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 327 sequences from Asprgillus_tritici_GCA_009812425_merged.bed.  Filtered 230 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 97.\n",
      "Screened 442 sequences from Asprgillus_tubingensis_GCA_001890745_merged.bed.  Filtered 313 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 129.\n",
      "Screened 480 sequences from Asprgillus_turcosus_GCA_002234965_merged.bed.  Filtered 333 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 147.\n",
      "Screened 531 sequences from Asprgillus_udagawae_GCA_001078395_merged.bed.  Filtered 376 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 155.\n",
      "Screened 521 sequences from Asprgillus_unguis_GCA_003324175_merged.bed.  Filtered 365 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 156.\n",
      "Screened 454 sequences from Asprgillus_ustus_GCA_000812125_merged.bed.  Filtered 329 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 125.\n",
      "Screened 401 sequences from Asprgillus_uvarum_GCF_003184745_merged.bed.  Filtered 295 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 106.\n",
      "Screened 447 sequences from Asprgillus_vadensis_GCF_003184925_merged.bed.  Filtered 334 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 338 sequences from Asprgillus_versicolor_GCA_001890125_merged.bed.  Filtered 249 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 402 sequences from Asprgillus_violaceofuscus_GCA_003184705_merged.bed.  Filtered 293 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 533 sequences from Asprgillus_viridinutans_GCA_004368095_merged.bed.  Filtered 354 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 179.\n",
      "Screened 473 sequences from Asprgillus_welwitschiae_GCF_003344945_merged.bed.  Filtered 344 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 129.\n",
      "Screened 421 sequences from Asprgillus_wentii_GCA_001890725_merged.bed.  Filtered 305 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 116.\n",
      "Screened 434 sequences from Asprgillus_westerdijkiae_GCA_001307345_merged.bed.  Filtered 313 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 121.\n",
      "Screened 109 sequences from Basipetospra_chlamydospra_GCA_001599675_merged.bed.  Filtered 87 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 22.\n",
      "Screened 535 sequences from Byssochlamys_nivea_GCA_003116535_merged.bed.  Filtered 369 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 166.\n",
      "Screened 467 sequences from Byssochlamys_sp_GCA_002914405_merged.bed.  Filtered 306 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 161.\n",
      "Screened 483 sequences from Byssochlamys_spctabilis_GCF_004022145_merged.bed.  Filtered 334 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 149.\n",
      "Screened 110 sequences from Elaphomyces_granulatus_GCA_002240705_merged.bed.  Filtered 85 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 25.\n",
      "Screened 366 sequences from Monascus_purpureus_GCA_003184285_merged.bed.  Filtered 242 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 124.\n",
      "Screened 394 sequences from Monascus_ruber_GCA_002976275_merged.bed.  Filtered 274 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 120.\n",
      "Screened 394 sequences from Penicilliopsis_zonata_GCF_001890105_merged.bed.  Filtered 284 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 511 sequences from Penicillium_antarcticum_GCA_002072345_merged.bed.  Filtered 370 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 141.\n",
      "Screened 507 sequences from Penicillium_arizonense_GCF_001773325_merged.bed.  Filtered 348 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 159.\n",
      "Screened 409 sequences from Penicillium_biforme_GCA_000577785_merged.bed.  Filtered 279 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 130.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 506 sequences from Penicillium_brasilianum_GCA_001048715_merged.bed.  Filtered 349 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 157.\n",
      "Screened 420 sequences from Penicillium_camemberti_GCA_000513335_merged.bed.  Filtered 281 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 139.\n",
      "Screened 514 sequences from Penicillium_capsulatum_GCA_000943775_merged.bed.  Filtered 347 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 167.\n",
      "Screened 355 sequences from Penicillium_carneum_GCA_000577495_merged.bed.  Filtered 251 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 104.\n",
      "Screened 452 sequences from Penicillium_chrysogenum_GCA_000710275_merged.bed.  Filtered 312 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 140.\n",
      "Screened 369 sequences from Penicillium_citrinum_GCA_001950535_merged.bed.  Filtered 265 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 104.\n",
      "Screened 432 sequences from Penicillium_coprophilum_GCA_002072405_merged.bed.  Filtered 311 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 121.\n",
      "Screened 446 sequences from Penicillium_decumbens_GCA_002072245_merged.bed.  Filtered 319 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 127.\n",
      "Screened 402 sequences from Penicillium_digitatum_GCF_000315645_merged.bed.  Filtered 285 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 363 sequences from Penicillium_expansum_GCF_000769745_merged.bed.  Filtered 267 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96.\n",
      "Screened 474 sequences from Penicillium_flavigenum_GCA_002072365_merged.bed.  Filtered 329 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 145.\n",
      "Screened 447 sequences from Penicillium_freii_GCA_001513925_merged.bed.  Filtered 312 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 135.\n",
      "Screened 436 sequences from Penicillium_fuscoglaucum_GCA_000576735_merged.bed.  Filtered 303 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 133.\n",
      "Screened 477 sequences from Penicillium_griseofulvum_GCA_001561935_merged.bed.  Filtered 325 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 152.\n",
      "Screened 436 sequences from Penicillium_italicum_GCA_002116305_merged.bed.  Filtered 306 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 130.\n",
      "Screened 526 sequences from Penicillium_janthinellum_GCA_002369805_merged.bed.  Filtered 365 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 161.\n",
      "Screened 458 sequences from Penicillium_nalgiovense_GCA_002072425_merged.bed.  Filtered 326 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 132.\n",
      "Screened 438 sequences from Penicillium_nordicum_GCA_000733025_merged.bed.  Filtered 303 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 135.\n",
      "Screened 449 sequences from Penicillium_oxalicum_GCA_000346795_merged.bed.  Filtered 328 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 121.\n",
      "Screened 344 sequences from Penicillium_paneum_GCA_000577715_merged.bed.  Filtered 254 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 90.\n",
      "Screened 485 sequences from Penicillium_paxilli_GCA_000347475_merged.bed.  Filtered 340 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 145.\n",
      "Screened 356 sequences from Penicillium_polonicum_GCA_003344595_merged.bed.  Filtered 252 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 104.\n",
      "Screened 370 sequences from Penicillium_roqueforti_GCA_000513255_merged.bed.  Filtered 277 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 93.\n",
      "Screened 450 sequences from Penicillium_rubens_GCF_000226395_merged.bed.  Filtered 320 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 130.\n",
      "Screened 478 sequences from Penicillium_sclerotiorum_GCA_001750025_merged.bed.  Filtered 322 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 156.\n",
      "Screened 463 sequences from Penicillium_solitum_GCA_000952775_merged.bed.  Filtered 313 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 150.\n",
      "Screened 455 sequences from Penicillium_sp_GCA_003800485_merged.bed.  Filtered 310 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 145.\n",
      "Screened 473 sequences from Penicillium_steckii_GCA_002072375_merged.bed.  Filtered 343 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 130.\n",
      "Screened 570 sequences from Penicillium_subrubescens_GCA_001908125_merged.bed.  Filtered 387 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 183.\n",
      "Screened 396 sequences from Penicillium_verrucosum_GCA_000970515_merged.bed.  Filtered 279 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 443 sequences from Penicillium_vulpinum_GCA_002072255_merged.bed.  Filtered 313 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 130.\n",
      "Screened 275 sequences from Phialosimplex_sp_GCA_003698115_merged.bed.  Filtered 189 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 86.\n",
      "Screened 399 sequences from Rasamsonia_emersonii_GCF_000968595_merged.bed.  Filtered 319 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 37359 sequences from Talaromyces_adpressus_GCA_002775195_merged.bed.  Filtered 18435 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 18924.\n",
      "Screened 31770 sequences from Talaromyces_amestolkiae_GCA_001896365_merged.bed.  Filtered 17431 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 14339.\n",
      "Screened 3075 sequences from Talaromyces_atroroseus_GCF_001907595_merged.bed.  Filtered 2327 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 748.\n",
      "Screened 2330 sequences from Talaromyces_borbonicus_GCA_002916415_merged.bed.  Filtered 1683 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 647.\n",
      "Screened 37528 sequences from Talaromyces_cellulolyticus_GCA_000829775_merged.bed.  Filtered 18452 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 19076.\n",
      "Screened 26540 sequences from Talaromyces_funiculosus_GCA_004299765_merged.bed.  Filtered 15301 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11239.\n",
      "Screened 1329 sequences from Talaromyces_islandicus_GCA_000985935_merged.bed.  Filtered 925 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 404.\n",
      "Screened 928 sequences from Talaromyces_piceae_GCA_001657655_merged.bed.  Filtered 631 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 297.\n",
      "Screened 37757 sequences from Talaromyces_pinophilus_GCA_001571465_merged.bed.  Filtered 18596 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 19161.\n",
      "Screened 482 sequences from Talaromyces_purpureogenus_GCA_001270325_merged.bed.  Filtered 347 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 135.\n",
      "Screened 18478 sequences from Talaromyces_stipitatus_GCF_000003125_merged.bed.  Filtered 11977 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6501.\n",
      "Screened 37687 sequences from Talaromyces_verruculosus_GCA_001305275_merged.bed.  Filtered 18437 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 19250.\n",
      "Screened 1377 sequences from Talaromyces_wortmannii_GCA_001939245_merged.bed.  Filtered 962 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 415.\n",
      "Screened 579 sequences from Thermoascaceae_sp_GCA_003123655_merged.bed.  Filtered 389 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 190.\n",
      "Screened 423 sequences from Thermoascus_crustaceus_GCA_001599835_merged.bed.  Filtered 319 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 104.\n",
      "Screened 295 sequences from Thermomyces_lanuginosus_GCA_000315935_merged.bed.  Filtered 227 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 68.\n",
      "Screened 306 sequences from Xeromyces_bisprus_GCA_900006255_merged.bed.  Filtered 230 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 76.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/eurotiales/initial_intervals/Xeromyces_bisprus_GCA_900006255_merged.bed --twobit results/phyluce/eurotiales/cleaned_genomes/Talaromyces_marneffei_GCF_000001985_formatted.2bit --output results/phyluce/eurotiales/initial_intervals/Xeromyces_bisprus_GCA_900006255_stripped.bed;\n",
      "asprgillus_aculeatinus_gcf_003184765.\n",
      "asprgillus_aculeatus_gcf_001890905.\n",
      "asprgillus_aff_gca_001931935.\n",
      "asprgillus_albertensis_gca_009193605.\n",
      "asprgillus_alliaceus_gcf_009176365.\n",
      "asprgillus_amoenus_gca_009812435.\n",
      "asprgillus_arachidicola_gca_009193545.\n",
      "asprgillus_avenaceus_gca_009193465.\n",
      "asprgillus_awamori_gca_003850985.\n",
      "asprgillus_bertholletius_gca_009193595.\n",
      "asprgillus_bombycis_gcf_001792695.\n",
      "asprgillus_brasiliensis_gca_001889945.\n",
      "asprgillus_brunneoviolaceus_gcf_003184695.\n",
      "asprgillus_caelatus_gcf_009193585.\n",
      "asprgillus_calidoustus_gca_001511075.\n",
      "asprgillus_campestris_gcf_002847485.\n",
      "asprgillus_candidus_gcf_002847045.\n",
      "asprgillus_carbonarius_gca_001990825.\n",
      "asprgillus_cejpii_gca_004769165.\n",
      "asprgillus_chevalieri_gca_001599875.\n",
      "asprgillus_clavatus_gcf_000002715.\n",
      "asprgillus_coremiiformis_gca_009193565.\n",
      "asprgillus_costaricaensis_gcf_003184835.\n",
      "asprgillus_cristatus_gca_001693355.\n",
      "asprgillus_ellipticus_gca_003184645.\n",
      "asprgillus_eucalypticola_gcf_003184535.\n",
      "asprgillus_fijiensis_gca_003184825.\n",
      "asprgillus_fischeri_gcf_000149645.\n",
      "asprgillus_flavus_gcf_000006275.\n",
      "asprgillus_fumigatus_gcf_000002655.\n",
      "asprgillus_glaucus_gcf_001890805.\n",
      "asprgillus_hancockii_gca_001696595.\n",
      "asprgillus_heteromorphus_gcf_003184545.\n",
      "asprgillus_homomorphus_gcf_003184865.\n",
      "asprgillus_ibericus_gcf_003184845.\n",
      "asprgillus_indologenus_gca_003184685.\n",
      "asprgillus_japonicus_gcf_003184785.\n",
      "asprgillus_kawachii_gca_000239835.\n",
      "asprgillus_lentulus_gca_001445615.\n",
      "asprgillus_leporis_gca_009176345.\n",
      "asprgillus_luchuensis_gca_001890685.\n",
      "asprgillus_minisclerotigenes_gca_009176455.\n",
      "asprgillus_mulundensis_gcf_003369625.\n",
      "asprgillus_neoellipticus_gca_003116565.\n",
      "asprgillus_neoniger_gcf_003184625.\n",
      "asprgillus_nidulans_gcf_000149205.\n",
      "asprgillus_niger_gcf_000002855.\n",
      "asprgillus_nomius_gcf_001204775.\n",
      "asprgillus_novofumigatus_gcf_002847465.\n",
      "asprgillus_novoparasiticus_gca_009176405.\n",
      "asprgillus_ochraceoroseus_gca_002846915.\n",
      "asprgillus_ochraceus_gca_004849945.\n",
      "asprgillus_olivimuriae_gca_003719415.\n",
      "asprgillus_oryzae_gcf_000184455.\n",
      "asprgillus_parasiticus_gca_009176385.\n",
      "asprgillus_persii_gca_002215965.\n",
      "asprgillus_phoenicis_gca_003344505.\n",
      "asprgillus_piperis_gcf_003184755.\n",
      "asprgillus_pseudocaelatus_gca_009193665.\n",
      "asprgillus_pseudotamarii_gcf_009193445.\n",
      "asprgillus_pseudoterreus_gca_002927005.\n",
      "asprgillus_rambellii_gca_000986645.\n",
      "asprgillus_ruber_gca_000600275.\n",
      "asprgillus_saccharolyticus_gcf_003184585.\n",
      "asprgillus_sclerotialis_gca_003589665.\n",
      "asprgillus_sclerotiicarbonarius_gca_003184635.\n",
      "asprgillus_sclerotioniger_gcf_003184525.\n",
      "asprgillus_sclerotiorum_gca_000530345.\n",
      "asprgillus_sergii_gca_009193525.\n",
      "asprgillus_sojae_gca_008274985.\n",
      "asprgillus_sp_gca_001044295.\n",
      "asprgillus_spnulosprus_gca_003574815.\n",
      "asprgillus_steynii_gcf_002849105.\n",
      "asprgillus_sydowii_gca_001890705.\n",
      "asprgillus_taichungensis_gca_002850765.\n",
      "asprgillus_tamarii_gca_009193485.\n",
      "asprgillus_tanneri_gca_003426965.\n",
      "asprgillus_terreus_gcf_000149615.\n",
      "asprgillus_thermomutatus_gcf_002237265.\n",
      "asprgillus_transmontanensis_gca_009193505.\n",
      "asprgillus_tritici_gca_009812425.\n",
      "asprgillus_tubingensis_gca_001890745.\n",
      "asprgillus_turcosus_gca_002234965.\n",
      "asprgillus_udagawae_gca_001078395.\n",
      "asprgillus_unguis_gca_003324175.\n",
      "asprgillus_ustus_gca_000812125.\n",
      "asprgillus_uvarum_gcf_003184745.\n",
      "asprgillus_vadensis_gcf_003184925.\n",
      "asprgillus_versicolor_gca_001890125.\n",
      "asprgillus_violaceofuscus_gca_003184705.\n",
      "asprgillus_viridinutans_gca_004368095.\n",
      "asprgillus_welwitschiae_gcf_003344945.\n",
      "asprgillus_wentii_gca_001890725.\n",
      "asprgillus_westerdijkiae_gca_001307345.\n",
      "basipetospra_chlamydospra_gca_001599675.\n",
      "byssochlamys_nivea_gca_003116535.\n",
      "byssochlamys_sp_gca_002914405.\n",
      "byssochlamys_spctabilis_gcf_004022145.\n",
      "elaphomyces_granulatus_gca_002240705.\n",
      "monascus_purpureus_gca_003184285.\n",
      "monascus_ruber_gca_002976275.\n",
      "penicilliopsis_zonata_gcf_001890105.\n",
      "penicillium_antarcticum_gca_002072345.\n",
      "penicillium_arizonense_gcf_001773325.\n",
      "penicillium_biforme_gca_000577785.\n",
      "penicillium_brasilianum_gca_001048715.\n",
      "penicillium_camemberti_gca_000513335.\n",
      "penicillium_capsulatum_gca_000943775.\n",
      "penicillium_carneum_gca_000577495.\n",
      "penicillium_chrysogenum_gca_000710275.\n",
      "penicillium_citrinum_gca_001950535.\n",
      "penicillium_coprophilum_gca_002072405.\n",
      "penicillium_decumbens_gca_002072245.\n",
      "penicillium_digitatum_gcf_000315645.\n",
      "penicillium_expansum_gcf_000769745.\n",
      "penicillium_flavigenum_gca_002072365.\n",
      "penicillium_freii_gca_001513925.\n",
      "penicillium_fuscoglaucum_gca_000576735.\n",
      "penicillium_griseofulvum_gca_001561935.\n",
      "penicillium_italicum_gca_002116305.\n",
      "penicillium_janthinellum_gca_002369805.\n",
      "penicillium_nalgiovense_gca_002072425.\n",
      "penicillium_nordicum_gca_000733025.\n",
      "penicillium_oxalicum_gca_000346795.\n",
      "penicillium_paneum_gca_000577715.\n",
      "penicillium_paxilli_gca_000347475.\n",
      "penicillium_polonicum_gca_003344595.\n",
      "penicillium_roqueforti_gca_000513255.\n",
      "penicillium_rubens_gcf_000226395.\n",
      "penicillium_sclerotiorum_gca_001750025.\n",
      "penicillium_solitum_gca_000952775.\n",
      "penicillium_sp_gca_003800485.\n",
      "penicillium_steckii_gca_002072375.\n",
      "penicillium_subrubescens_gca_001908125.\n",
      "penicillium_verrucosum_gca_000970515.\n",
      "penicillium_vulpinum_gca_002072255.\n",
      "phialosimplex_sp_gca_003698115.\n",
      "rasamsonia_emersonii_gcf_000968595.\n",
      "talaromyces_adpressus_gca_002775195...................\n",
      "talaromyces_amestolkiae_gca_001896365...............\n",
      "talaromyces_atroroseus_gcf_001907595.\n",
      "talaromyces_borbonicus_gca_002916415.\n",
      "talaromyces_cellulolyticus_gca_000829775....................\n",
      "talaromyces_funiculosus_gca_004299765............\n",
      "talaromyces_islandicus_gca_000985935.\n",
      "talaromyces_piceae_gca_001657655.\n",
      "talaromyces_pinophilus_gca_001571465....................\n",
      "talaromyces_purpureogenus_gca_001270325.\n",
      "talaromyces_stipitatus_gcf_000003125.......\n",
      "talaromyces_verruculosus_gca_001305275....................\n",
      "talaromyces_wortmannii_gca_001939245.\n",
      "thermoascaceae_sp_gca_003123655.\n",
      "thermoascus_crustaceus_gca_001599835.\n",
      "thermomyces_lanuginosus_gca_000315935.\n",
      "xeromyces_bisprus_gca_900006255.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/eurotiales/initial_intervals/eurotiales-to-Talaromyces_marneffei_GCF_000001985.sqlite --base-taxon Talaromyces_marneffei_GCF_000001985\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 0 taxa:\t26,540.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 1 taxa:\t26,540.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 2 taxa:\t22,589.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 3 taxa:\t19,605.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 4 taxa:\t16,932.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 5 taxa:\t12,812.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 6 taxa:\t8,444.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 7 taxa:\t3,977.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 8 taxa:\t999.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 9 taxa:\t612.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 10 taxa:\t491.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 11 taxa:\t432.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 12 taxa:\t404.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 13 taxa:\t383.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 14 taxa:\t362.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 15 taxa:\t344.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 16 taxa:\t330.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 17 taxa:\t315.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 18 taxa:\t305.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 19 taxa:\t299.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 20 taxa:\t290.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 21 taxa:\t285.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 22 taxa:\t276.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 23 taxa:\t274.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 24 taxa:\t266.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 25 taxa:\t256.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 26 taxa:\t251.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 27 taxa:\t248.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 28 taxa:\t241.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 29 taxa:\t235.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 30 taxa:\t229.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 31 taxa:\t226.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 32 taxa:\t223.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 33 taxa:\t221.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 34 taxa:\t218.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 35 taxa:\t214.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 36 taxa:\t212.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 37 taxa:\t209.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 38 taxa:\t203.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 39 taxa:\t202.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 40 taxa:\t197.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 41 taxa:\t191.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 42 taxa:\t187.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 43 taxa:\t187.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 44 taxa:\t185.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 45 taxa:\t184.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 46 taxa:\t182.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 47 taxa:\t176.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 48 taxa:\t175.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 49 taxa:\t171.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 50 taxa:\t166.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 51 taxa:\t164.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 52 taxa:\t163.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 53 taxa:\t158.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 54 taxa:\t152.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 55 taxa:\t151.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 56 taxa:\t144.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 57 taxa:\t142.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 58 taxa:\t141.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 59 taxa:\t139.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 60 taxa:\t135.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 61 taxa:\t133.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 62 taxa:\t129.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 63 taxa:\t129.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 64 taxa:\t127.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 65 taxa:\t124.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 66 taxa:\t121.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 67 taxa:\t117.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 68 taxa:\t113.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 69 taxa:\t112.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 70 taxa:\t112.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 71 taxa:\t111.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 72 taxa:\t107.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 73 taxa:\t104.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 74 taxa:\t103.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 75 taxa:\t100.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 76 taxa:\t100.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 77 taxa:\t96.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 78 taxa:\t95.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 79 taxa:\t95.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 80 taxa:\t93.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 81 taxa:\t92.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 82 taxa:\t90.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 83 taxa:\t89.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 84 taxa:\t87.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 85 taxa:\t85.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 86 taxa:\t85.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 87 taxa:\t80.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 88 taxa:\t80.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 89 taxa:\t80.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 90 taxa:\t79.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 91 taxa:\t79.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 92 taxa:\t75.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 93 taxa:\t74.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 94 taxa:\t73.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 95 taxa:\t72.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 96 taxa:\t71.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 97 taxa:\t71.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 98 taxa:\t71.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 99 taxa:\t69.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 100 taxa:\t68.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 101 taxa:\t67.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 102 taxa:\t67.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 103 taxa:\t66.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 104 taxa:\t66.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 105 taxa:\t65.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 106 taxa:\t64.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 107 taxa:\t63.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 108 taxa:\t61.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 109 taxa:\t61.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 110 taxa:\t59.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 111 taxa:\t58.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 112 taxa:\t53.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 113 taxa:\t50.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 114 taxa:\t49.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 115 taxa:\t48.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 116 taxa:\t46.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 117 taxa:\t44.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 118 taxa:\t44.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 119 taxa:\t43.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 120 taxa:\t43.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 121 taxa:\t42.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 122 taxa:\t40.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 123 taxa:\t39.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 124 taxa:\t39.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 125 taxa:\t37.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 126 taxa:\t34.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 127 taxa:\t32.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 128 taxa:\t29.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 129 taxa:\t28.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 130 taxa:\t28.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 131 taxa:\t28.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 132 taxa:\t26.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 133 taxa:\t26.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 134 taxa:\t24.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 135 taxa:\t24.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 136 taxa:\t24.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 137 taxa:\t23.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 138 taxa:\t23.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 139 taxa:\t23.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 140 taxa:\t23.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 141 taxa:\t22.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 142 taxa:\t22.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 143 taxa:\t22.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 144 taxa:\t22.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 145 taxa:\t20.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 146 taxa:\t20.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 147 taxa:\t19.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 148 taxa:\t18.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 149 taxa:\t17.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 150 taxa:\t14.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 151 taxa:\t12.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 152 taxa:\t8.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 153 taxa:\t5.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 154 taxa:\t3.0\n",
      "Loci shared by Talaromyces_marneffei_GCF_000001985 + 155 taxa:\t1.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/eurotiales/initial_intervals/eurotiales-to-Talaromyces_marneffei_GCF_000001985.sqlite --base-taxon Talaromyces_marneffei_GCF_000001985 --output results/phyluce/eurotiales/initial_intervals/Talaromyces_marneffei_GCF_000001985_+50.bed --specific-counts 50;\n",
      "Counter({'talaromyces_pinophilus_gca_001571465': 164, 'talaromyces_adpressus_gca_002775195': 163, 'talaromyces_amestolkiae_gca_001896365': 163, 'talaromyces_stipitatus_gcf_000003125': 163, 'talaromyces_verruculosus_gca_001305275': 160, 'talaromyces_funiculosus_gca_004299765': 150, 'talaromyces_wortmannii_gca_001939245': 144, 'talaromyces_islandicus_gca_000985935': 139, 'talaromyces_cellulolyticus_gca_000829775': 137, 'asprgillus_awamori_gca_003850985': 129, 'talaromyces_atroroseus_gcf_001907595': 126, 'penicillium_capsulatum_gca_000943775': 124, 'thermoascaceae_sp_gca_003123655': 124, 'talaromyces_borbonicus_gca_002916415': 123, 'penicillium_brasilianum_gca_001048715': 122, 'penicillium_solitum_gca_000952775': 122, 'asprgillus_eucalypticola_gcf_003184535': 122, 'penicillium_subrubescens_gca_001908125': 122, 'penicillium_sclerotiorum_gca_001750025': 121, 'asprgillus_clavatus_gcf_000002715': 121, 'penicillium_arizonense_gcf_001773325': 121, 'penicillium_sp_gca_003800485': 121, 'penicillium_freii_gca_001513925': 120, 'talaromyces_piceae_gca_001657655': 119, 'asprgillus_cristatus_gca_001693355': 119, 'penicillium_chrysogenum_gca_000710275': 119, 'asprgillus_lentulus_gca_001445615': 119, 'asprgillus_niger_gcf_000002855': 119, 'penicillium_flavigenum_gca_002072365': 119, 'asprgillus_kawachii_gca_000239835': 119, 'penicillium_paxilli_gca_000347475': 117, 'penicillium_janthinellum_gca_002369805': 116, 'asprgillus_terreus_gcf_000149615': 115, 'byssochlamys_spctabilis_gcf_004022145': 115, 'asprgillus_nomius_gcf_001204775': 115, 'penicillium_griseofulvum_gca_001561935': 114, 'asprgillus_fischeri_gcf_000149645': 114, 'penicillium_antarcticum_gca_002072345': 113, 'asprgillus_unguis_gca_003324175': 113, 'penicillium_camemberti_gca_000513335': 113, 'asprgillus_oryzae_gcf_000184455': 112, 'asprgillus_aculeatus_gcf_001890905': 112, 'asprgillus_sclerotioniger_gcf_003184525': 111, 'penicillium_nalgiovense_gca_002072425': 111, 'asprgillus_viridinutans_gca_004368095': 110, 'asprgillus_ochraceus_gca_004849945': 109, 'asprgillus_bombycis_gcf_001792695': 109, 'byssochlamys_sp_gca_002914405': 108, 'penicillium_nordicum_gca_000733025': 108, 'asprgillus_rambellii_gca_000986645': 108, 'asprgillus_tubingensis_gca_001890745': 108, 'asprgillus_carbonarius_gca_001990825': 107, 'penicillium_rubens_gcf_000226395': 107, 'penicillium_fuscoglaucum_gca_000576735': 107, 'penicillium_biforme_gca_000577785': 107, 'asprgillus_neoellipticus_gca_003116565': 106, 'asprgillus_udagawae_gca_001078395': 106, 'asprgillus_ochraceoroseus_gca_002846915': 105, 'asprgillus_brasiliensis_gca_001889945': 105, 'asprgillus_phoenicis_gca_003344505': 104, 'byssochlamys_nivea_gca_003116535': 104, 'asprgillus_novofumigatus_gcf_002847465': 104, 'asprgillus_heteromorphus_gcf_003184545': 104, 'penicillium_vulpinum_gca_002072255': 103, 'asprgillus_luchuensis_gca_001890685': 103, 'penicillium_italicum_gca_002116305': 103, 'penicillium_steckii_gca_002072375': 103, 'asprgillus_cejpii_gca_004769165': 103, 'asprgillus_costaricaensis_gcf_003184835': 102, 'asprgillus_sp_gca_001044295': 102, 'asprgillus_neoniger_gcf_003184625': 102, 'asprgillus_piperis_gcf_003184755': 102, 'asprgillus_aff_gca_001931935': 102, 'asprgillus_pseudoterreus_gca_002927005': 101, 'asprgillus_welwitschiae_gcf_003344945': 101, 'penicillium_oxalicum_gca_000346795': 101, 'penicillium_coprophilum_gca_002072405': 101, 'asprgillus_thermomutatus_gcf_002237265': 101, 'asprgillus_steynii_gcf_002849105': 100, 'asprgillus_ibericus_gcf_003184845': 100, 'asprgillus_flavus_gcf_000006275': 100, 'asprgillus_turcosus_gca_002234965': 99, 'asprgillus_hancockii_gca_001696595': 99, 'asprgillus_fumigatus_gcf_000002655': 99, 'penicillium_decumbens_gca_002072245': 99, 'asprgillus_sclerotiorum_gca_000530345': 99, 'penicillium_verrucosum_gca_000970515': 98, 'asprgillus_vadensis_gcf_003184925': 98, 'penicillium_digitatum_gcf_000315645': 98, 'asprgillus_westerdijkiae_gca_001307345': 97, 'asprgillus_olivimuriae_gca_003719415': 97, 'penicillium_polonicum_gca_003344595': 96, 'asprgillus_calidoustus_gca_001511075': 94, 'asprgillus_transmontanensis_gca_009193505': 93, 'asprgillus_indologenus_gca_003184685': 93, 'asprgillus_ellipticus_gca_003184645': 93, 'asprgillus_ustus_gca_000812125': 93, 'asprgillus_violaceofuscus_gca_003184705': 92, 'asprgillus_sclerotiicarbonarius_gca_003184635': 92, 'penicillium_carneum_gca_000577495': 92, 'asprgillus_fijiensis_gca_003184825': 92, 'asprgillus_brunneoviolaceus_gcf_003184695': 92, 'talaromyces_purpureogenus_gca_001270325': 91, 'asprgillus_japonicus_gcf_003184785': 91, 'asprgillus_ruber_gca_000600275': 91, 'asprgillus_aculeatinus_gcf_003184765': 91, 'asprgillus_minisclerotigenes_gca_009176455': 90, 'asprgillus_spnulosprus_gca_003574815': 90, 'asprgillus_homomorphus_gcf_003184865': 90, 'asprgillus_uvarum_gcf_003184745': 90, 'asprgillus_persii_gca_002215965': 89, 'asprgillus_wentii_gca_001890725': 89, 'penicillium_citrinum_gca_001950535': 88, 'asprgillus_novoparasiticus_gca_009176405': 88, 'asprgillus_chevalieri_gca_001599875': 88, 'asprgillus_avenaceus_gca_009193465': 88, 'asprgillus_sergii_gca_009193525': 87, 'asprgillus_saccharolyticus_gcf_003184585': 86, 'asprgillus_arachidicola_gca_009193545': 86, 'asprgillus_tritici_gca_009812425': 85, 'asprgillus_glaucus_gcf_001890805': 85, 'asprgillus_tamarii_gca_009193485': 85, 'asprgillus_pseudocaelatus_gca_009193665': 85, 'asprgillus_parasiticus_gca_009176385': 85, 'asprgillus_amoenus_gca_009812435': 84, 'penicillium_roqueforti_gca_000513255': 84, 'penicillium_expansum_gcf_000769745': 84, 'asprgillus_leporis_gca_009176345': 83, 'asprgillus_tanneri_gca_003426965': 83, 'penicilliopsis_zonata_gcf_001890105': 83, 'asprgillus_caelatus_gcf_009193585': 83, 'asprgillus_bertholletius_gca_009193595': 83, 'monascus_ruber_gca_002976275': 83, 'thermoascus_crustaceus_gca_001599835': 82, 'asprgillus_sojae_gca_008274985': 81, 'asprgillus_pseudotamarii_gcf_009193445': 80, 'asprgillus_nidulans_gcf_000149205': 78, 'asprgillus_albertensis_gca_009193605': 77, 'asprgillus_mulundensis_gcf_003369625': 77, 'asprgillus_campestris_gcf_002847485': 77, 'asprgillus_versicolor_gca_001890125': 76, 'monascus_purpureus_gca_003184285': 76, 'penicillium_paneum_gca_000577715': 75, 'asprgillus_sydowii_gca_001890705': 74, 'asprgillus_alliaceus_gcf_009176365': 73, 'asprgillus_sclerotialis_gca_003589665': 70, 'asprgillus_coremiiformis_gca_009193565': 64, 'xeromyces_bisprus_gca_900006255': 63, 'phialosimplex_sp_gca_003698115': 62, 'asprgillus_candidus_gcf_002847045': 62, 'rasamsonia_emersonii_gcf_000968595': 61, 'asprgillus_taichungensis_gca_002850765': 59, 'thermomyces_lanuginosus_gca_000315935': 33, 'basipetospra_chlamydospra_gca_001599675': 21, 'elaphomyces_granulatus_gca_002240705': 19})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 50\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/eurotiales/initial_intervals/Talaromyces_marneffei_GCF_000001985_+50.bed --twobit results/phyluce/eurotiales/cleaned_genomes/Talaromyces_marneffei_GCF_000001985_formatted.2bit --buffer-to 160 --output results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50.fasta;\n",
      "Screened 166 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 166.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50.fasta --probe-prefix uce_eurotiales_ --design eurotiales_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 163\n",
      "Probe Count = 323\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas --query results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes_vself.lastz;\n",
      "Started:  Mon Feb 10, 2020  10:31:31\n",
      "Ended:  Mon Feb 10, 2020  10:31:31\n",
      "Time for execution:  0.0032263358434 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas                        --lastz results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes_vself.lastz                       --probe-prefix=uce_eurotiales_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 322 fasta sequences.  Filtered 22 duplicates. Kept 279.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/eurotiales/validate_intervals/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas --scaffoldlist Asprgillus_aculeatinus_GCF_003184765 Asprgillus_aculeatus_GCF_001890905 Asprgillus_aff_GCA_001931935 Asprgillus_albertensis_GCA_009193605 Asprgillus_alliaceus_GCF_009176365 Asprgillus_amoenus_GCA_009812435 Asprgillus_arachidicola_GCA_009193545 Asprgillus_avenaceus_GCA_009193465 Asprgillus_awamori_GCA_003850985 Asprgillus_bertholletius_GCA_009193595 Asprgillus_bombycis_GCF_001792695 Asprgillus_brasiliensis_GCA_001889945 Asprgillus_brunneoviolaceus_GCF_003184695 Asprgillus_caelatus_GCF_009193585 Asprgillus_calidoustus_GCA_001511075 Asprgillus_campestris_GCF_002847485 Asprgillus_candidus_GCF_002847045 Asprgillus_carbonarius_GCA_001990825 Asprgillus_cejpii_GCA_004769165 Asprgillus_chevalieri_GCA_001599875 Asprgillus_clavatus_GCF_000002715 Asprgillus_coremiiformis_GCA_009193565 Asprgillus_costaricaensis_GCF_003184835 Asprgillus_cristatus_GCA_001693355 Asprgillus_ellipticus_GCA_003184645 Asprgillus_eucalypticola_GCF_003184535 Asprgillus_fijiensis_GCA_003184825 Asprgillus_fischeri_GCF_000149645 Asprgillus_flavus_GCF_000006275 Asprgillus_fumigatus_GCF_000002655 Asprgillus_glaucus_GCF_001890805 Asprgillus_hancockii_GCA_001696595 Asprgillus_heteromorphus_GCF_003184545 Asprgillus_homomorphus_GCF_003184865 Asprgillus_ibericus_GCF_003184845 Asprgillus_indologenus_GCA_003184685 Asprgillus_japonicus_GCF_003184785 Asprgillus_kawachii_GCA_000239835 Asprgillus_lentulus_GCA_001445615 Asprgillus_leporis_GCA_009176345 Asprgillus_luchuensis_GCA_001890685 Asprgillus_minisclerotigenes_GCA_009176455 Asprgillus_mulundensis_GCF_003369625 Asprgillus_neoellipticus_GCA_003116565 Asprgillus_neoniger_GCF_003184625 Asprgillus_nidulans_GCF_000149205 Asprgillus_niger_GCF_000002855 Asprgillus_nomius_GCF_001204775 Asprgillus_novofumigatus_GCF_002847465 Asprgillus_novoparasiticus_GCA_009176405 Asprgillus_ochraceoroseus_GCA_002846915 Asprgillus_ochraceus_GCA_004849945 Asprgillus_olivimuriae_GCA_003719415 Asprgillus_oryzae_GCF_000184455 Asprgillus_parasiticus_GCA_009176385 Asprgillus_persii_GCA_002215965 Asprgillus_phoenicis_GCA_003344505 Asprgillus_piperis_GCF_003184755 Asprgillus_pseudocaelatus_GCA_009193665 Asprgillus_pseudotamarii_GCF_009193445 Asprgillus_pseudoterreus_GCA_002927005 Asprgillus_rambellii_GCA_000986645 Asprgillus_ruber_GCA_000600275 Asprgillus_saccharolyticus_GCF_003184585 Asprgillus_sclerotialis_GCA_003589665 Asprgillus_sclerotiicarbonarius_GCA_003184635 Asprgillus_sclerotioniger_GCF_003184525 Asprgillus_sclerotiorum_GCA_000530345 Asprgillus_sergii_GCA_009193525 Asprgillus_sojae_GCA_008274985 Asprgillus_sp_GCA_001044295 Asprgillus_spnulosprus_GCA_003574815 Asprgillus_steynii_GCF_002849105 Asprgillus_sydowii_GCA_001890705 Asprgillus_taichungensis_GCA_002850765 Asprgillus_tamarii_GCA_009193485 Asprgillus_tanneri_GCA_003426965 Asprgillus_terreus_GCF_000149615 Asprgillus_thermomutatus_GCF_002237265 Asprgillus_transmontanensis_GCA_009193505 Asprgillus_tritici_GCA_009812425 Asprgillus_tubingensis_GCA_001890745 Asprgillus_turcosus_GCA_002234965 Asprgillus_udagawae_GCA_001078395 Asprgillus_unguis_GCA_003324175 Asprgillus_ustus_GCA_000812125 Asprgillus_uvarum_GCF_003184745 Asprgillus_vadensis_GCF_003184925 Asprgillus_versicolor_GCA_001890125 Asprgillus_violaceofuscus_GCA_003184705 Asprgillus_viridinutans_GCA_004368095 Asprgillus_welwitschiae_GCF_003344945 Asprgillus_wentii_GCA_001890725 Asprgillus_westerdijkiae_GCA_001307345 Basipetospra_chlamydospra_GCA_001599675 Byssochlamys_nivea_GCA_003116535 Byssochlamys_sp_GCA_002914405 Byssochlamys_spctabilis_GCF_004022145 Elaphomyces_granulatus_GCA_002240705 Monascus_purpureus_GCA_003184285 Monascus_ruber_GCA_002976275 Penicilliopsis_zonata_GCF_001890105 Penicillium_antarcticum_GCA_002072345 Penicillium_arizonense_GCF_001773325 Penicillium_biforme_GCA_000577785 Penicillium_brasilianum_GCA_001048715 Penicillium_camemberti_GCA_000513335 Penicillium_capsulatum_GCA_000943775 Penicillium_carneum_GCA_000577495 Penicillium_chrysogenum_GCA_000710275 Penicillium_citrinum_GCA_001950535 Penicillium_coprophilum_GCA_002072405 Penicillium_decumbens_GCA_002072245 Penicillium_digitatum_GCF_000315645 Penicillium_expansum_GCF_000769745 Penicillium_flavigenum_GCA_002072365 Penicillium_freii_GCA_001513925 Penicillium_fuscoglaucum_GCA_000576735 Penicillium_griseofulvum_GCA_001561935 Penicillium_italicum_GCA_002116305 Penicillium_janthinellum_GCA_002369805 Penicillium_nalgiovense_GCA_002072425 Penicillium_nordicum_GCA_000733025 Penicillium_oxalicum_GCA_000346795 Penicillium_paneum_GCA_000577715 Penicillium_paxilli_GCA_000347475 Penicillium_polonicum_GCA_003344595 Penicillium_roqueforti_GCA_000513255 Penicillium_rubens_GCF_000226395 Penicillium_sclerotiorum_GCA_001750025 Penicillium_solitum_GCA_000952775 Penicillium_sp_GCA_003800485 Penicillium_steckii_GCA_002072375 Penicillium_subrubescens_GCA_001908125 Penicillium_verrucosum_GCA_000970515 Penicillium_vulpinum_GCA_002072255 Phialosimplex_sp_GCA_003698115 Rasamsonia_emersonii_GCF_000968595 Talaromyces_adpressus_GCA_002775195 Talaromyces_amestolkiae_GCA_001896365 Talaromyces_atroroseus_GCF_001907595 Talaromyces_borbonicus_GCA_002916415 Talaromyces_cellulolyticus_GCA_000829775 Talaromyces_funiculosus_GCA_004299765 Talaromyces_islandicus_GCA_000985935 Talaromyces_piceae_GCA_001657655 Talaromyces_pinophilus_GCA_001571465 Talaromyces_purpureogenus_GCA_001270325 Talaromyces_stipitatus_GCF_000003125 Talaromyces_verruculosus_GCA_001305275 Talaromyces_wortmannii_GCA_001939245 Thermoascaceae_sp_GCA_003123655 Thermoascus_crustaceus_GCA_001599835 Thermomyces_lanuginosus_GCA_000315935 Xeromyces_bisprus_GCA_900006255 Talaromyces_marneffei_GCF_000001985 --genome-base-path results/phyluce/eurotiales/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/eurotiales/extract_probes_from_group/eurotiales-to-Talaromyces_marneffei_GCF_000001985.sqlite --output results/phyluce/eurotiales/extract_probes_from_group/lastz/;\n",
      "\n",
      "Running against Asprgillus_aculeatinus_GCF_003184765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpOzggHC.fasta\n",
      "\t/tmp/tmp82wkJP.fasta\n",
      "\t/tmp/tmpiqEXq2.fasta\n",
      "\t/tmp/tmp44sxL9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB63QHg.lastz\n",
      "\t/tmp/tmpUT3CI6.lastz\n",
      "\t/tmp/tmpQzLUH3.lastz\n",
      "\t/tmp/tmpSm3LYC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_aculeatinus_GCF_003184765.lastz\n",
      "Creating Asprgillus_aculeatinus_GCF_003184765 table\n",
      "Inserting data to Asprgillus_aculeatinus_GCF_003184765 table\n",
      "\n",
      "Running against Asprgillus_aculeatus_GCF_001890905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpyexS_P.fasta\n",
      "\t/tmp/tmpkvG3OS.fasta\n",
      "\t/tmp/tmp4TIoqG.fasta\n",
      "\t/tmp/tmp01KOnr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpT5RHYM.lastz\n",
      "\t/tmp/tmp6uu9zj.lastz\n",
      "\t/tmp/tmp7lRYqh.lastz\n",
      "\t/tmp/tmpkwpaLI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_aculeatus_GCF_001890905.lastz\n",
      "Creating Asprgillus_aculeatus_GCF_001890905 table\n",
      "Inserting data to Asprgillus_aculeatus_GCF_001890905 table\n",
      "\n",
      "Running against Asprgillus_aff_GCA_001931935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp2ey3rx.fasta\n",
      "\t/tmp/tmpGJnj5L.fasta\n",
      "\t/tmp/tmp9M1AmA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLMkF8J.lastz\n",
      "\t/tmp/tmpJ09tkM.lastz\n",
      "\t/tmp/tmpBwX1Dv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_aff_GCA_001931935.lastz\n",
      "Creating Asprgillus_aff_GCA_001931935 table\n",
      "Inserting data to Asprgillus_aff_GCA_001931935 table\n",
      "\n",
      "Running against Asprgillus_albertensis_GCA_009193605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpYuzqUB.fasta\n",
      "\t/tmp/tmpi7Q2Wl.fasta\n",
      "\t/tmp/tmpRqs2Xk.fasta\n",
      "\t/tmp/tmp364dyt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaR8Nfs.lastz\n",
      "\t/tmp/tmp8TOxwV.lastz\n",
      "\t/tmp/tmpWt3ObF.lastz\n",
      "\t/tmp/tmpF_BN8e.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_albertensis_GCA_009193605.lastz\n",
      "Creating Asprgillus_albertensis_GCA_009193605 table\n",
      "Inserting data to Asprgillus_albertensis_GCA_009193605 table\n",
      "\n",
      "Running against Asprgillus_alliaceus_GCF_009176365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpZOF58S.fasta\n",
      "\t/tmp/tmpVgvuSB.fasta\n",
      "\t/tmp/tmpjC45cx.fasta\n",
      "\t/tmp/tmpZRVAbf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1jwhNi.lastz\n",
      "\t/tmp/tmp7uZTQI.lastz\n",
      "\t/tmp/tmpugOssw.lastz\n",
      "\t/tmp/tmpKoqEVn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_alliaceus_GCF_009176365.lastz\n",
      "Creating Asprgillus_alliaceus_GCF_009176365 table\n",
      "Inserting data to Asprgillus_alliaceus_GCF_009176365 table\n",
      "\n",
      "Running against Asprgillus_amoenus_GCA_009812435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpoTqJQ1.fasta\n",
      "\t/tmp/tmpFEoq6k.fasta\n",
      "\t/tmp/tmpngoif1.fasta\n",
      "\t/tmp/tmpgNoy8X.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpb0RnRB.lastz\n",
      "\t/tmp/tmp6_naK2.lastz\n",
      "\t/tmp/tmp1z41a6.lastz\n",
      "\t/tmp/tmp4M2B4E.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_amoenus_GCA_009812435.lastz\n",
      "Creating Asprgillus_amoenus_GCA_009812435 table\n",
      "Inserting data to Asprgillus_amoenus_GCA_009812435 table\n",
      "\n",
      "Running against Asprgillus_arachidicola_GCA_009193545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpQ2Gzdw.fasta\n",
      "\t/tmp/tmpPpzvui.fasta\n",
      "\t/tmp/tmpYzJyCy.fasta\n",
      "\t/tmp/tmpQ542uQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7RPM02.lastz\n",
      "\t/tmp/tmp06fgsY.lastz\n",
      "\t/tmp/tmpnBTrpl.lastz\n",
      "\t/tmp/tmp1F7Yae.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_arachidicola_GCA_009193545.lastz\n",
      "Creating Asprgillus_arachidicola_GCA_009193545 table\n",
      "Inserting data to Asprgillus_arachidicola_GCA_009193545 table\n",
      "\n",
      "Running against Asprgillus_avenaceus_GCA_009193465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp8e6zc1.fasta\n",
      "\t/tmp/tmpE0R3mP.fasta\n",
      "\t/tmp/tmp8XnP8j.fasta\n",
      "\t/tmp/tmprGuLw_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZnTd0p.lastz\n",
      "\t/tmp/tmpYBeqzD.lastz\n",
      "\t/tmp/tmpYwNJsS.lastz\n",
      "\t/tmp/tmpuJkZzt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_avenaceus_GCA_009193465.lastz\n",
      "Creating Asprgillus_avenaceus_GCA_009193465 table\n",
      "Inserting data to Asprgillus_avenaceus_GCA_009193465 table\n",
      "\n",
      "Running against Asprgillus_awamori_GCA_003850985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpvAayr2.fasta\n",
      "\t/tmp/tmpwdeYZq.fasta\n",
      "\t/tmp/tmp58Hmph.fasta\n",
      "\t/tmp/tmp0mxCs7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpn1y1Yn.lastz\n",
      "\t/tmp/tmp15xZrw.lastz\n",
      "\t/tmp/tmp1vFt3L.lastz\n",
      "\t/tmp/tmpGnKSDg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_awamori_GCA_003850985.lastz\n",
      "Creating Asprgillus_awamori_GCA_003850985 table\n",
      "Inserting data to Asprgillus_awamori_GCA_003850985 table\n",
      "\n",
      "Running against Asprgillus_bertholletius_GCA_009193595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpGsJn3_.fasta\n",
      "\t/tmp/tmp12e4tK.fasta\n",
      "\t/tmp/tmpFgMQFL.fasta\n",
      "\t/tmp/tmpPRB_TU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwd3ewZ.lastz\n",
      "\t/tmp/tmpChAOG1.lastz\n",
      "\t/tmp/tmp2tRBuJ.lastz\n",
      "\t/tmp/tmpDuU8Cn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_bertholletius_GCA_009193595.lastz\n",
      "Creating Asprgillus_bertholletius_GCA_009193595 table\n",
      "Inserting data to Asprgillus_bertholletius_GCA_009193595 table\n",
      "\n",
      "Running against Asprgillus_bombycis_GCF_001792695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp5VYLRw.fasta\n",
      "\t/tmp/tmpOWLRE3.fasta\n",
      "\t/tmp/tmpip2tbh.fasta\n",
      "\t/tmp/tmpqbhRwV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPeLcQS.lastz\n",
      "\t/tmp/tmpFoelvW.lastz\n",
      "\t/tmp/tmpAVN9DV.lastz\n",
      "\t/tmp/tmp2et66h.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_bombycis_GCF_001792695.lastz\n",
      "Creating Asprgillus_bombycis_GCF_001792695 table\n",
      "Inserting data to Asprgillus_bombycis_GCF_001792695 table\n",
      "\n",
      "Running against Asprgillus_brasiliensis_GCA_001889945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpn1_37p.fasta\n",
      "\t/tmp/tmpRpzQP4.fasta\n",
      "\t/tmp/tmpmwXe0w.fasta\n",
      "\t/tmp/tmpY2Lit2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0rP7ms.lastz\n",
      "\t/tmp/tmpKH7LAm.lastz\n",
      "\t/tmp/tmptfLiLc.lastz\n",
      "\t/tmp/tmppKfQFS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_brasiliensis_GCA_001889945.lastz\n",
      "Creating Asprgillus_brasiliensis_GCA_001889945 table\n",
      "Inserting data to Asprgillus_brasiliensis_GCA_001889945 table\n",
      "\n",
      "Running against Asprgillus_brunneoviolaceus_GCF_003184695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpWXQgmh.fasta\n",
      "\t/tmp/tmp53EWHE.fasta\n",
      "\t/tmp/tmpLzvivL.fasta\n",
      "\t/tmp/tmpKOQV5B.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpq_2h0X.lastz\n",
      "\t/tmp/tmpCJgtK1.lastz\n",
      "\t/tmp/tmpt0AvcF.lastz\n",
      "\t/tmp/tmpy8SyfN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_brunneoviolaceus_GCF_003184695.lastz\n",
      "Creating Asprgillus_brunneoviolaceus_GCF_003184695 table\n",
      "Inserting data to Asprgillus_brunneoviolaceus_GCF_003184695 table\n",
      "\n",
      "Running against Asprgillus_caelatus_GCF_009193585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpfg5gQV.fasta\n",
      "\t/tmp/tmpaaibBr.fasta\n",
      "\t/tmp/tmpbY2DYp.fasta\n",
      "\t/tmp/tmp_FeuXy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDADC6b.lastz\n",
      "\t/tmp/tmpXUhEAL.lastz\n",
      "\t/tmp/tmpug2QcG.lastz\n",
      "\t/tmp/tmp2W8pVJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_caelatus_GCF_009193585.lastz\n",
      "Creating Asprgillus_caelatus_GCF_009193585 table\n",
      "Inserting data to Asprgillus_caelatus_GCF_009193585 table\n",
      "\n",
      "Running against Asprgillus_calidoustus_GCA_001511075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpEgnBX2.fasta\n",
      "\t/tmp/tmprKtISm.fasta\n",
      "\t/tmp/tmpsBqNX0.fasta\n",
      "\t/tmp/tmpNj9QmO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpH1J0g1.lastz\n",
      "\t/tmp/tmpZj6ezs.lastz\n",
      "\t/tmp/tmpaYp4dL.lastz\n",
      "\t/tmp/tmphPq932.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_calidoustus_GCA_001511075.lastz\n",
      "Creating Asprgillus_calidoustus_GCA_001511075 table\n",
      "Inserting data to Asprgillus_calidoustus_GCA_001511075 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Asprgillus_campestris_GCF_002847485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpmCOyT3.fasta\n",
      "\t/tmp/tmp25ICUY.fasta\n",
      "\t/tmp/tmpCkArp6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7xQNZv.lastz\n",
      "\t/tmp/tmpFKnbep.lastz\n",
      "\t/tmp/tmpX2czi3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_campestris_GCF_002847485.lastz\n",
      "Creating Asprgillus_campestris_GCF_002847485 table\n",
      "Inserting data to Asprgillus_campestris_GCF_002847485 table\n",
      "\n",
      "Running against Asprgillus_candidus_GCF_002847045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmprwp_8N.fasta\n",
      "\t/tmp/tmpBFylNw.fasta\n",
      "\t/tmp/tmpEz69sj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp41eh0G.lastz\n",
      "\t/tmp/tmpx5SIug.lastz\n",
      "\t/tmp/tmpUMBobR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_candidus_GCF_002847045.lastz\n",
      "Creating Asprgillus_candidus_GCF_002847045 table\n",
      "Inserting data to Asprgillus_candidus_GCF_002847045 table\n",
      "\n",
      "Running against Asprgillus_carbonarius_GCA_001990825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmptr_SUt.fasta\n",
      "\t/tmp/tmpJTvhU6.fasta\n",
      "\t/tmp/tmpz4jfcg.fasta\n",
      "\t/tmp/tmpPRfHzB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfVKvJU.lastz\n",
      "\t/tmp/tmpViI5S5.lastz\n",
      "\t/tmp/tmpb3RXLR.lastz\n",
      "\t/tmp/tmpMQGKY2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_carbonarius_GCA_001990825.lastz\n",
      "Creating Asprgillus_carbonarius_GCA_001990825 table\n",
      "Inserting data to Asprgillus_carbonarius_GCA_001990825 table\n",
      "\n",
      "Running against Asprgillus_cejpii_GCA_004769165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpDgZTRT.fasta\n",
      "\t/tmp/tmpQzxX23.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBKp6Re.lastz\n",
      "\t/tmp/tmpdxA3U0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_cejpii_GCA_004769165.lastz\n",
      "Creating Asprgillus_cejpii_GCA_004769165 table\n",
      "Inserting data to Asprgillus_cejpii_GCA_004769165 table\n",
      "\n",
      "Running against Asprgillus_chevalieri_GCA_001599875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmptJafxX.fasta\n",
      "\t/tmp/tmpJUv0V6.fasta\n",
      "\t/tmp/tmpG4M6Bm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptz8p7o.lastz\n",
      "\t/tmp/tmpUd5PdM.lastz\n",
      "\t/tmp/tmpwf_kBh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_chevalieri_GCA_001599875.lastz\n",
      "Creating Asprgillus_chevalieri_GCA_001599875 table\n",
      "Inserting data to Asprgillus_chevalieri_GCA_001599875 table\n",
      "\n",
      "Running against Asprgillus_clavatus_GCF_000002715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpKfO04q.fasta\n",
      "\t/tmp/tmpWkvTU3.fasta\n",
      "\t/tmp/tmp2ZxacJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnXd9A7.lastz\n",
      "\t/tmp/tmp1lMmvq.lastz\n",
      "\t/tmp/tmpTamBB3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_clavatus_GCF_000002715.lastz\n",
      "Creating Asprgillus_clavatus_GCF_000002715 table\n",
      "Inserting data to Asprgillus_clavatus_GCF_000002715 table\n",
      "\n",
      "Running against Asprgillus_coremiiformis_GCA_009193565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpUtSDVj.fasta\n",
      "\t/tmp/tmpQAI3ko.fasta\n",
      "\t/tmp/tmp20oh1z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPmutRk.lastz\n",
      "\t/tmp/tmp3SlTdN.lastz\n",
      "\t/tmp/tmpEIzIUc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_coremiiformis_GCA_009193565.lastz\n",
      "Creating Asprgillus_coremiiformis_GCA_009193565 table\n",
      "Inserting data to Asprgillus_coremiiformis_GCA_009193565 table\n",
      "\n",
      "Running against Asprgillus_costaricaensis_GCF_003184835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpJhnFhJ.fasta\n",
      "\t/tmp/tmpICKOT3.fasta\n",
      "\t/tmp/tmp3JKeoa.fasta\n",
      "\t/tmp/tmpFYJ7Bt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNIeJds.lastz\n",
      "\t/tmp/tmpU10JV7.lastz\n",
      "\t/tmp/tmpYGzO6B.lastz\n",
      "\t/tmp/tmpIjA9T4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_costaricaensis_GCF_003184835.lastz\n",
      "Creating Asprgillus_costaricaensis_GCF_003184835 table\n",
      "Inserting data to Asprgillus_costaricaensis_GCF_003184835 table\n",
      "\n",
      "Running against Asprgillus_cristatus_GCA_001693355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpn7iqhp.fasta\n",
      "\t/tmp/tmp9851Rk.fasta\n",
      "\t/tmp/tmpT3S29G.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8o1RCn.lastz\n",
      "\t/tmp/tmpYmroQ5.lastz\n",
      "\t/tmp/tmpHJ9gZj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_cristatus_GCA_001693355.lastz\n",
      "Creating Asprgillus_cristatus_GCA_001693355 table\n",
      "Inserting data to Asprgillus_cristatus_GCA_001693355 table\n",
      "\n",
      "Running against Asprgillus_ellipticus_GCA_003184645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpUXQqRh.fasta\n",
      "\t/tmp/tmpWSyVoe.fasta\n",
      "\t/tmp/tmppgnXjv.fasta\n",
      "\t/tmp/tmpAY3mnM.fasta\n",
      "\t/tmp/tmpMAukoD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnN05QM.lastz\n",
      "\t/tmp/tmpvTuEm_.lastz\n",
      "\t/tmp/tmpW5Nrg6.lastz\n",
      "\t/tmp/tmp2ga7bO.lastz\n",
      "\t/tmp/tmpTpT0or.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_ellipticus_GCA_003184645.lastz\n",
      "Creating Asprgillus_ellipticus_GCA_003184645 table\n",
      "Inserting data to Asprgillus_ellipticus_GCA_003184645 table\n",
      "\n",
      "Running against Asprgillus_eucalypticola_GCF_003184535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpJHPbie.fasta\n",
      "\t/tmp/tmpxsKSI5.fasta\n",
      "\t/tmp/tmpzKOkvE.fasta\n",
      "\t/tmp/tmp8QLuRr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGNspRu.lastz\n",
      "\t/tmp/tmpnlJ8PG.lastz\n",
      "\t/tmp/tmprVdLkK.lastz\n",
      "\t/tmp/tmpq8y49f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_eucalypticola_GCF_003184535.lastz\n",
      "Creating Asprgillus_eucalypticola_GCF_003184535 table\n",
      "Inserting data to Asprgillus_eucalypticola_GCF_003184535 table\n",
      "\n",
      "Running against Asprgillus_fijiensis_GCA_003184825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpZ69UVv.fasta\n",
      "\t/tmp/tmpUMPZwC.fasta\n",
      "\t/tmp/tmpxy1CUn.fasta\n",
      "\t/tmp/tmpUTIZDJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZk3De_.lastz\n",
      "\t/tmp/tmpfi9VaF.lastz\n",
      "\t/tmp/tmpawEnWC.lastz\n",
      "\t/tmp/tmp8rgrNO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_fijiensis_GCA_003184825.lastz\n",
      "Creating Asprgillus_fijiensis_GCA_003184825 table\n",
      "Inserting data to Asprgillus_fijiensis_GCA_003184825 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Asprgillus_fischeri_GCF_000149645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpP43o_Z.fasta\n",
      "\t/tmp/tmpDpVbMA.fasta\n",
      "\t/tmp/tmpRuuwnL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvbXQtO.lastz\n",
      "\t/tmp/tmpBXfoY8.lastz\n",
      "\t/tmp/tmpKkh6t0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_fischeri_GCF_000149645.lastz\n",
      "Creating Asprgillus_fischeri_GCF_000149645 table\n",
      "Inserting data to Asprgillus_fischeri_GCF_000149645 table\n",
      "\n",
      "Running against Asprgillus_flavus_GCF_000006275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp3sDgZm.fasta\n",
      "\t/tmp/tmpzabSUS.fasta\n",
      "\t/tmp/tmpKOQZ6T.fasta\n",
      "\t/tmp/tmpHO2BGO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp980ZJq.lastz\n",
      "\t/tmp/tmpzAQhor.lastz\n",
      "\t/tmp/tmpuEaaM1.lastz\n",
      "\t/tmp/tmp9pMZxB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_flavus_GCF_000006275.lastz\n",
      "Creating Asprgillus_flavus_GCF_000006275 table\n",
      "Inserting data to Asprgillus_flavus_GCF_000006275 table\n",
      "\n",
      "Running against Asprgillus_fumigatus_GCF_000002655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmppKjbOm.fasta\n",
      "\t/tmp/tmpAsAYIK.fasta\n",
      "\t/tmp/tmppSuhlT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB4AL55.lastz\n",
      "\t/tmp/tmpODUG1t.lastz\n",
      "\t/tmp/tmpyHlum8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_fumigatus_GCF_000002655.lastz\n",
      "Creating Asprgillus_fumigatus_GCF_000002655 table\n",
      "Inserting data to Asprgillus_fumigatus_GCF_000002655 table\n",
      "\n",
      "Running against Asprgillus_glaucus_GCF_001890805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpM_KgRm.fasta\n",
      "\t/tmp/tmph8gnIQ.fasta\n",
      "\t/tmp/tmpAW3b5g.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpszZVxq.lastz\n",
      "\t/tmp/tmp3JQka_.lastz\n",
      "\t/tmp/tmpBUxrHp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_glaucus_GCF_001890805.lastz\n",
      "Creating Asprgillus_glaucus_GCF_001890805 table\n",
      "Inserting data to Asprgillus_glaucus_GCF_001890805 table\n",
      "\n",
      "Running against Asprgillus_hancockii_GCA_001696595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpY7vqWd.fasta\n",
      "\t/tmp/tmpl1ua4R.fasta\n",
      "\t/tmp/tmp5gJutg.fasta\n",
      "\t/tmp/tmpuh8Djk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcmCwtu.lastz\n",
      "\t/tmp/tmpyDIohZ.lastz\n",
      "\t/tmp/tmpWKsotY.lastz\n",
      "\t/tmp/tmph5YekN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_hancockii_GCA_001696595.lastz\n",
      "Creating Asprgillus_hancockii_GCA_001696595 table\n",
      "Inserting data to Asprgillus_hancockii_GCA_001696595 table\n",
      "\n",
      "Running against Asprgillus_heteromorphus_GCF_003184545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpum0hpY.fasta\n",
      "\t/tmp/tmppe3EPj.fasta\n",
      "\t/tmp/tmpJke70z.fasta\n",
      "\t/tmp/tmpo2yd_E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkA_uYV.lastz\n",
      "\t/tmp/tmpo83Hsk.lastz\n",
      "\t/tmp/tmp_31VyG.lastz\n",
      "\t/tmp/tmp3fzub_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_heteromorphus_GCF_003184545.lastz\n",
      "Creating Asprgillus_heteromorphus_GCF_003184545 table\n",
      "Inserting data to Asprgillus_heteromorphus_GCF_003184545 table\n",
      "\n",
      "Running against Asprgillus_homomorphus_GCF_003184865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpA8YmeP.fasta\n",
      "\t/tmp/tmp4CUNV1.fasta\n",
      "\t/tmp/tmpx1uIUc.fasta\n",
      "\t/tmp/tmpxaBQws.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTZAVm6.lastz\n",
      "\t/tmp/tmpO81PmV.lastz\n",
      "\t/tmp/tmpDiZUuH.lastz\n",
      "\t/tmp/tmpCy6e_A.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_homomorphus_GCF_003184865.lastz\n",
      "Creating Asprgillus_homomorphus_GCF_003184865 table\n",
      "Inserting data to Asprgillus_homomorphus_GCF_003184865 table\n",
      "\n",
      "Running against Asprgillus_ibericus_GCF_003184845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpsGwxno.fasta\n",
      "\t/tmp/tmp9i3xJf.fasta\n",
      "\t/tmp/tmpKyKxRX.fasta\n",
      "\t/tmp/tmpFXlFuJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpC6MhcL.lastz\n",
      "\t/tmp/tmpcwTCzP.lastz\n",
      "\t/tmp/tmp5wH7cB.lastz\n",
      "\t/tmp/tmpCGPXg6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_ibericus_GCF_003184845.lastz\n",
      "Creating Asprgillus_ibericus_GCF_003184845 table\n",
      "Inserting data to Asprgillus_ibericus_GCF_003184845 table\n",
      "\n",
      "Running against Asprgillus_indologenus_GCA_003184685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpH0zaKP.fasta\n",
      "\t/tmp/tmpmCjdvQ.fasta\n",
      "\t/tmp/tmp2dhVzh.fasta\n",
      "\t/tmp/tmpXrUMmK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOgqavx.lastz\n",
      "\t/tmp/tmpPblXV7.lastz\n",
      "\t/tmp/tmpBP1kUF.lastz\n",
      "\t/tmp/tmpy7xfWE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_indologenus_GCA_003184685.lastz\n",
      "Creating Asprgillus_indologenus_GCA_003184685 table\n",
      "Inserting data to Asprgillus_indologenus_GCA_003184685 table\n",
      "\n",
      "Running against Asprgillus_japonicus_GCF_003184785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp5O3iA5.fasta\n",
      "\t/tmp/tmpPoylGw.fasta\n",
      "\t/tmp/tmp0Sayjg.fasta\n",
      "\t/tmp/tmp0CfhFU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYzTTZi.lastz\n",
      "\t/tmp/tmpEkAhDm.lastz\n",
      "\t/tmp/tmpGAfSv1.lastz\n",
      "\t/tmp/tmpEOcXbg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_japonicus_GCF_003184785.lastz\n",
      "Creating Asprgillus_japonicus_GCF_003184785 table\n",
      "Inserting data to Asprgillus_japonicus_GCF_003184785 table\n",
      "\n",
      "Running against Asprgillus_kawachii_GCA_000239835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpFaoITY.fasta\n",
      "\t/tmp/tmptoYvj7.fasta\n",
      "\t/tmp/tmpNlRIz2.fasta\n",
      "\t/tmp/tmpj5NDat.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCZXFGA.lastz\n",
      "\t/tmp/tmpDHHBod.lastz\n",
      "\t/tmp/tmpmZv6F1.lastz\n",
      "\t/tmp/tmpqZDdyB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_kawachii_GCA_000239835.lastz\n",
      "Creating Asprgillus_kawachii_GCA_000239835 table\n",
      "Inserting data to Asprgillus_kawachii_GCA_000239835 table\n",
      "\n",
      "Running against Asprgillus_lentulus_GCA_001445615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp1PjGDL.fasta\n",
      "\t/tmp/tmpzven1o.fasta\n",
      "\t/tmp/tmpdrqBLE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp88hvM6.lastz\n",
      "\t/tmp/tmpmXivjP.lastz\n",
      "\t/tmp/tmpCUAFAy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_lentulus_GCA_001445615.lastz\n",
      "Creating Asprgillus_lentulus_GCA_001445615 table\n",
      "Inserting data to Asprgillus_lentulus_GCA_001445615 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Asprgillus_leporis_GCA_009176345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpR8zoX9.fasta\n",
      "\t/tmp/tmp5W1cHq.fasta\n",
      "\t/tmp/tmpnlD0mn.fasta\n",
      "\t/tmp/tmpxnujJS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfON7GH.lastz\n",
      "\t/tmp/tmp61muzU.lastz\n",
      "\t/tmp/tmpdNzDTL.lastz\n",
      "\t/tmp/tmpHpCjFf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_leporis_GCA_009176345.lastz\n",
      "Creating Asprgillus_leporis_GCA_009176345 table\n",
      "Inserting data to Asprgillus_leporis_GCA_009176345 table\n",
      "\n",
      "Running against Asprgillus_luchuensis_GCA_001890685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpaL2fzB.fasta\n",
      "\t/tmp/tmpNLvVQw.fasta\n",
      "\t/tmp/tmpAJ5Vj6.fasta\n",
      "\t/tmp/tmpY0GxJH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_v1S_U.lastz\n",
      "\t/tmp/tmp_wPK2x.lastz\n",
      "\t/tmp/tmppuODPS.lastz\n",
      "\t/tmp/tmpvevkVO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_luchuensis_GCA_001890685.lastz\n",
      "Creating Asprgillus_luchuensis_GCA_001890685 table\n",
      "Inserting data to Asprgillus_luchuensis_GCA_001890685 table\n",
      "\n",
      "Running against Asprgillus_minisclerotigenes_GCA_009176455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpNmz080.fasta\n",
      "\t/tmp/tmp5kW0EZ.fasta\n",
      "\t/tmp/tmpggqlzC.fasta\n",
      "\t/tmp/tmp52JfVL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCtuZSs.lastz\n",
      "\t/tmp/tmp8LU0DA.lastz\n",
      "\t/tmp/tmpzbwxtD.lastz\n",
      "\t/tmp/tmpCRElH4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_minisclerotigenes_GCA_009176455.lastz\n",
      "Creating Asprgillus_minisclerotigenes_GCA_009176455 table\n",
      "Inserting data to Asprgillus_minisclerotigenes_GCA_009176455 table\n",
      "\n",
      "Running against Asprgillus_mulundensis_GCF_003369625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpBrgbbr.fasta\n",
      "\t/tmp/tmpjlJf5V.fasta\n",
      "\t/tmp/tmp16BHT6.fasta\n",
      "\t/tmp/tmpRJ_oki.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwXujlK.lastz\n",
      "\t/tmp/tmp157wWS.lastz\n",
      "\t/tmp/tmp_w6PNJ.lastz\n",
      "\t/tmp/tmpyr4ci9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_mulundensis_GCF_003369625.lastz\n",
      "Creating Asprgillus_mulundensis_GCF_003369625 table\n",
      "Inserting data to Asprgillus_mulundensis_GCF_003369625 table\n",
      "\n",
      "Running against Asprgillus_neoellipticus_GCA_003116565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpOI4w2N.fasta\n",
      "\t/tmp/tmp8XapyC.fasta\n",
      "\t/tmp/tmpeYyoMb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbjqbV5.lastz\n",
      "\t/tmp/tmpwucw8M.lastz\n",
      "\t/tmp/tmp1wbJ7h.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_neoellipticus_GCA_003116565.lastz\n",
      "Creating Asprgillus_neoellipticus_GCA_003116565 table\n",
      "Inserting data to Asprgillus_neoellipticus_GCA_003116565 table\n",
      "\n",
      "Running against Asprgillus_neoniger_GCF_003184625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp2YTQXh.fasta\n",
      "\t/tmp/tmpOb8_bb.fasta\n",
      "\t/tmp/tmpyocBqo.fasta\n",
      "\t/tmp/tmpzHOl7J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2eGcbP.lastz\n",
      "\t/tmp/tmpUgG8RO.lastz\n",
      "\t/tmp/tmpaPULH5.lastz\n",
      "\t/tmp/tmpsneRtl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_neoniger_GCF_003184625.lastz\n",
      "Creating Asprgillus_neoniger_GCF_003184625 table\n",
      "Inserting data to Asprgillus_neoniger_GCF_003184625 table\n",
      "\n",
      "Running against Asprgillus_nidulans_GCF_000149205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpkA8RTM.fasta\n",
      "\t/tmp/tmpeNBjSo.fasta\n",
      "\t/tmp/tmp7dG75k.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5B4kAJ.lastz\n",
      "\t/tmp/tmpqs00hi.lastz\n",
      "\t/tmp/tmpZO_kjK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_nidulans_GCF_000149205.lastz\n",
      "Creating Asprgillus_nidulans_GCF_000149205 table\n",
      "Inserting data to Asprgillus_nidulans_GCF_000149205 table\n",
      "\n",
      "Running against Asprgillus_niger_GCF_000002855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpoR57yi.fasta\n",
      "\t/tmp/tmpUCZyOw.fasta\n",
      "\t/tmp/tmpy3Yrui.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPnhq0x.lastz\n",
      "\t/tmp/tmp2MYzYT.lastz\n",
      "\t/tmp/tmpiIolGz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_niger_GCF_000002855.lastz\n",
      "Creating Asprgillus_niger_GCF_000002855 table\n",
      "Inserting data to Asprgillus_niger_GCF_000002855 table\n",
      "\n",
      "Running against Asprgillus_nomius_GCF_001204775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpCD6LXY.fasta\n",
      "\t/tmp/tmp_uGxFl.fasta\n",
      "\t/tmp/tmppvI_GV.fasta\n",
      "\t/tmp/tmpbA9xtt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGXOOOr.lastz\n",
      "\t/tmp/tmpxwDUjV.lastz\n",
      "\t/tmp/tmpZme1Cj.lastz\n",
      "\t/tmp/tmp5NWoVJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_nomius_GCF_001204775.lastz\n",
      "Creating Asprgillus_nomius_GCF_001204775 table\n",
      "Inserting data to Asprgillus_nomius_GCF_001204775 table\n",
      "\n",
      "Running against Asprgillus_novofumigatus_GCF_002847465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpx6uqYx.fasta\n",
      "\t/tmp/tmp9mjSXW.fasta\n",
      "\t/tmp/tmp9bcQJI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWwh9D4.lastz\n",
      "\t/tmp/tmpGeaBb6.lastz\n",
      "\t/tmp/tmpG5ptvv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_novofumigatus_GCF_002847465.lastz\n",
      "Creating Asprgillus_novofumigatus_GCF_002847465 table\n",
      "Inserting data to Asprgillus_novofumigatus_GCF_002847465 table\n",
      "\n",
      "Running against Asprgillus_novoparasiticus_GCA_009176405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpzpIMaf.fasta\n",
      "\t/tmp/tmpnFdakG.fasta\n",
      "\t/tmp/tmpvPwRQ4.fasta\n",
      "\t/tmp/tmpURpnJz.fasta\n",
      "\t/tmp/tmpcC87gG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXgmcRI.lastz\n",
      "\t/tmp/tmp6MwOTK.lastz\n",
      "\t/tmp/tmpjFRN5M.lastz\n",
      "\t/tmp/tmpcV3W13.lastz\n",
      "\t/tmp/tmpul1xcL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_novoparasiticus_GCA_009176405.lastz\n",
      "Creating Asprgillus_novoparasiticus_GCA_009176405 table\n",
      "Inserting data to Asprgillus_novoparasiticus_GCA_009176405 table\n",
      "\n",
      "Running against Asprgillus_ochraceoroseus_GCA_002846915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpXo0oyn.fasta\n",
      "\t/tmp/tmpAFvpmg.fasta\n",
      "\t/tmp/tmp44Hpkb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQu0J1x.lastz\n",
      "\t/tmp/tmpoyx_hT.lastz\n",
      "\t/tmp/tmpdXBKzz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_ochraceoroseus_GCA_002846915.lastz\n",
      "Creating Asprgillus_ochraceoroseus_GCA_002846915 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Asprgillus_ochraceoroseus_GCA_002846915 table\n",
      "\n",
      "Running against Asprgillus_ochraceus_GCA_004849945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpaBSzHm.fasta\n",
      "\t/tmp/tmpWWJEnx.fasta\n",
      "\t/tmp/tmp5yiSyE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2fD82K.lastz\n",
      "\t/tmp/tmpA5qUBH.lastz\n",
      "\t/tmp/tmpw6biA6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_ochraceus_GCA_004849945.lastz\n",
      "Creating Asprgillus_ochraceus_GCA_004849945 table\n",
      "Inserting data to Asprgillus_ochraceus_GCA_004849945 table\n",
      "\n",
      "Running against Asprgillus_olivimuriae_GCA_003719415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp6Gp5Tt.fasta\n",
      "\t/tmp/tmpy6AVYS.fasta\n",
      "\t/tmp/tmpGWOcnj.fasta\n",
      "\t/tmp/tmpAEeUPi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCb8g3f.lastz\n",
      "\t/tmp/tmpkVkzTH.lastz\n",
      "\t/tmp/tmpxvRYbW.lastz\n",
      "\t/tmp/tmpN5WzEK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_olivimuriae_GCA_003719415.lastz\n",
      "Creating Asprgillus_olivimuriae_GCA_003719415 table\n",
      "Inserting data to Asprgillus_olivimuriae_GCA_003719415 table\n",
      "\n",
      "Running against Asprgillus_oryzae_GCF_000184455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpDgF6w3.fasta\n",
      "\t/tmp/tmpd_xTkI.fasta\n",
      "\t/tmp/tmpuW51Lw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKetdw5.lastz\n",
      "\t/tmp/tmpVyHo4p.lastz\n",
      "\t/tmp/tmpGEsxAu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_oryzae_GCF_000184455.lastz\n",
      "Creating Asprgillus_oryzae_GCF_000184455 table\n",
      "Inserting data to Asprgillus_oryzae_GCF_000184455 table\n",
      "\n",
      "Running against Asprgillus_parasiticus_GCA_009176385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpkBrFQw.fasta\n",
      "\t/tmp/tmpGYx1Pm.fasta\n",
      "\t/tmp/tmpWaZ0Yp.fasta\n",
      "\t/tmp/tmpNBqaoS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7FRL2W.lastz\n",
      "\t/tmp/tmpAPJq55.lastz\n",
      "\t/tmp/tmp4RJLxr.lastz\n",
      "\t/tmp/tmpeqS0CS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_parasiticus_GCA_009176385.lastz\n",
      "Creating Asprgillus_parasiticus_GCA_009176385 table\n",
      "Inserting data to Asprgillus_parasiticus_GCA_009176385 table\n",
      "\n",
      "Running against Asprgillus_persii_GCA_002215965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpLdKaTQ.fasta\n",
      "\t/tmp/tmp1KlHIj.fasta\n",
      "\t/tmp/tmpsPjjW9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOWTupF.lastz\n",
      "\t/tmp/tmpz2YWct.lastz\n",
      "\t/tmp/tmpGgtZPj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_persii_GCA_002215965.lastz\n",
      "Creating Asprgillus_persii_GCA_002215965 table\n",
      "Inserting data to Asprgillus_persii_GCA_002215965 table\n",
      "\n",
      "Running against Asprgillus_phoenicis_GCA_003344505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpCBy_UV.fasta\n",
      "\t/tmp/tmpQSoJ93.fasta\n",
      "\t/tmp/tmpQADo4B.fasta\n",
      "\t/tmp/tmpKBUCCe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZIOmdR.lastz\n",
      "\t/tmp/tmp2INS7r.lastz\n",
      "\t/tmp/tmp8OmWLI.lastz\n",
      "\t/tmp/tmpRRaXbE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_phoenicis_GCA_003344505.lastz\n",
      "Creating Asprgillus_phoenicis_GCA_003344505 table\n",
      "Inserting data to Asprgillus_phoenicis_GCA_003344505 table\n",
      "\n",
      "Running against Asprgillus_piperis_GCF_003184755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp7TAmxl.fasta\n",
      "\t/tmp/tmpietlWW.fasta\n",
      "\t/tmp/tmpT1EeYr.fasta\n",
      "\t/tmp/tmpgbDNG7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLhq1Bh.lastz\n",
      "\t/tmp/tmpQ5unFs.lastz\n",
      "\t/tmp/tmpegiYHU.lastz\n",
      "\t/tmp/tmpgcQFQ0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_piperis_GCF_003184755.lastz\n",
      "Creating Asprgillus_piperis_GCF_003184755 table\n",
      "Inserting data to Asprgillus_piperis_GCF_003184755 table\n",
      "\n",
      "Running against Asprgillus_pseudocaelatus_GCA_009193665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpWJi8Y6.fasta\n",
      "\t/tmp/tmpIraKFs.fasta\n",
      "\t/tmp/tmptQdw3R.fasta\n",
      "\t/tmp/tmpSGDtz0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEM2Uhv.lastz\n",
      "\t/tmp/tmpDA6P3S.lastz\n",
      "\t/tmp/tmpJgvgdz.lastz\n",
      "\t/tmp/tmpD2e4hf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_pseudocaelatus_GCA_009193665.lastz\n",
      "Creating Asprgillus_pseudocaelatus_GCA_009193665 table\n",
      "Inserting data to Asprgillus_pseudocaelatus_GCA_009193665 table\n",
      "\n",
      "Running against Asprgillus_pseudotamarii_GCF_009193445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpIps1wj.fasta\n",
      "\t/tmp/tmp3G8IGV.fasta\n",
      "\t/tmp/tmph1lqxJ.fasta\n",
      "\t/tmp/tmp8LXx_m.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphX8TBj.lastz\n",
      "\t/tmp/tmpyGpk5m.lastz\n",
      "\t/tmp/tmpRZO7gs.lastz\n",
      "\t/tmp/tmpd8uGX6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_pseudotamarii_GCF_009193445.lastz\n",
      "Creating Asprgillus_pseudotamarii_GCF_009193445 table\n",
      "Inserting data to Asprgillus_pseudotamarii_GCF_009193445 table\n",
      "\n",
      "Running against Asprgillus_pseudoterreus_GCA_002927005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpU49w_G.fasta\n",
      "\t/tmp/tmp2WkE2V.fasta\n",
      "\t/tmp/tmpZizv9A.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYoK8do.lastz\n",
      "\t/tmp/tmpUj8x0k.lastz\n",
      "\t/tmp/tmp63C7nV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_pseudoterreus_GCA_002927005.lastz\n",
      "Creating Asprgillus_pseudoterreus_GCA_002927005 table\n",
      "Inserting data to Asprgillus_pseudoterreus_GCA_002927005 table\n",
      "\n",
      "Running against Asprgillus_rambellii_GCA_000986645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpoHBDLm.fasta\n",
      "\t/tmp/tmp7y08_W.fasta\n",
      "\t/tmp/tmpO7MTr4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiPxA6N.lastz\n",
      "\t/tmp/tmpPYo9zL.lastz\n",
      "\t/tmp/tmpHWzs1W.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_rambellii_GCA_000986645.lastz\n",
      "Creating Asprgillus_rambellii_GCA_000986645 table\n",
      "Inserting data to Asprgillus_rambellii_GCA_000986645 table\n",
      "\n",
      "Running against Asprgillus_ruber_GCA_000600275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp6aIijD.fasta\n",
      "\t/tmp/tmpNL55jS.fasta\n",
      "\t/tmp/tmpq6hi7N.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpf_4pLN.lastz\n",
      "\t/tmp/tmp3XgGi8.lastz\n",
      "\t/tmp/tmp3uXxKu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_ruber_GCA_000600275.lastz\n",
      "Creating Asprgillus_ruber_GCA_000600275 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Asprgillus_ruber_GCA_000600275 table\n",
      "\n",
      "Running against Asprgillus_saccharolyticus_GCF_003184585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpsZeQmG.fasta\n",
      "\t/tmp/tmpmvEGoh.fasta\n",
      "\t/tmp/tmpbhwBbW.fasta\n",
      "\t/tmp/tmpkDFCxh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjT7zyj.lastz\n",
      "\t/tmp/tmpFGdvoA.lastz\n",
      "\t/tmp/tmpHZJw9O.lastz\n",
      "\t/tmp/tmpbeuh2f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_saccharolyticus_GCF_003184585.lastz\n",
      "Creating Asprgillus_saccharolyticus_GCF_003184585 table\n",
      "Inserting data to Asprgillus_saccharolyticus_GCF_003184585 table\n",
      "\n",
      "Running against Asprgillus_sclerotialis_GCA_003589665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpMi3IJg.fasta\n",
      "\t/tmp/tmpSwtrmn.fasta\n",
      "\t/tmp/tmp6NzWW3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLquQjM.lastz\n",
      "\t/tmp/tmpd0DvB5.lastz\n",
      "\t/tmp/tmpSrXPcl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sclerotialis_GCA_003589665.lastz\n",
      "Creating Asprgillus_sclerotialis_GCA_003589665 table\n",
      "Inserting data to Asprgillus_sclerotialis_GCA_003589665 table\n",
      "\n",
      "Running against Asprgillus_sclerotiicarbonarius_GCA_003184635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpYUKKVo.fasta\n",
      "\t/tmp/tmpZx_iIz.fasta\n",
      "\t/tmp/tmpXTavRw.fasta\n",
      "\t/tmp/tmpwz7w7J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpC22AXH.lastz\n",
      "\t/tmp/tmpMBK2Lo.lastz\n",
      "\t/tmp/tmp7TXCme.lastz\n",
      "\t/tmp/tmp0qZS_C.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sclerotiicarbonarius_GCA_003184635.lastz\n",
      "Creating Asprgillus_sclerotiicarbonarius_GCA_003184635 table\n",
      "Inserting data to Asprgillus_sclerotiicarbonarius_GCA_003184635 table\n",
      "\n",
      "Running against Asprgillus_sclerotioniger_GCF_003184525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpXhCcsn.fasta\n",
      "\t/tmp/tmpriliCF.fasta\n",
      "\t/tmp/tmpVndinG.fasta\n",
      "\t/tmp/tmpXmnRxC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWXhE2A.lastz\n",
      "\t/tmp/tmp0hyOUo.lastz\n",
      "\t/tmp/tmpIDW0_S.lastz\n",
      "\t/tmp/tmpWBiL3g.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sclerotioniger_GCF_003184525.lastz\n",
      "Creating Asprgillus_sclerotioniger_GCF_003184525 table\n",
      "Inserting data to Asprgillus_sclerotioniger_GCF_003184525 table\n",
      "\n",
      "Running against Asprgillus_sclerotiorum_GCA_000530345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpJfM02d.fasta\n",
      "\t/tmp/tmpCquhi2.fasta\n",
      "\t/tmp/tmpTrTlhN.fasta\n",
      "\t/tmp/tmpgxVJP0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB4smuc.lastz\n",
      "\t/tmp/tmpreg_NK.lastz\n",
      "\t/tmp/tmp9IctiF.lastz\n",
      "\t/tmp/tmpp7sGtf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sclerotiorum_GCA_000530345.lastz\n",
      "Creating Asprgillus_sclerotiorum_GCA_000530345 table\n",
      "Inserting data to Asprgillus_sclerotiorum_GCA_000530345 table\n",
      "\n",
      "Running against Asprgillus_sergii_GCA_009193525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp7tofRz.fasta\n",
      "\t/tmp/tmp5vHUMI.fasta\n",
      "\t/tmp/tmpPw6fus.fasta\n",
      "\t/tmp/tmpco_UTU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMTn2tf.lastz\n",
      "\t/tmp/tmpNp7kBt.lastz\n",
      "\t/tmp/tmpfOe_b3.lastz\n",
      "\t/tmp/tmpgD1Itm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sergii_GCA_009193525.lastz\n",
      "Creating Asprgillus_sergii_GCA_009193525 table\n",
      "Inserting data to Asprgillus_sergii_GCA_009193525 table\n",
      "\n",
      "Running against Asprgillus_sojae_GCA_008274985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpXQLXFy.fasta\n",
      "\t/tmp/tmpAu5r5a.fasta\n",
      "\t/tmp/tmplN16ka.fasta\n",
      "\t/tmp/tmpDXs0m4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0AzdA6.lastz\n",
      "\t/tmp/tmpGrCb05.lastz\n",
      "\t/tmp/tmpk0v5RA.lastz\n",
      "\t/tmp/tmp_AUrPG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sojae_GCA_008274985.lastz\n",
      "Creating Asprgillus_sojae_GCA_008274985 table\n",
      "Inserting data to Asprgillus_sojae_GCA_008274985 table\n",
      "\n",
      "Running against Asprgillus_sp_GCA_001044295.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpSWQXp0.fasta\n",
      "\t/tmp/tmpxvArA8.fasta\n",
      "\t/tmp/tmpqGV7kQ.fasta\n",
      "\t/tmp/tmpCT3931.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphnCJp3.lastz\n",
      "\t/tmp/tmp2RsRs1.lastz\n",
      "\t/tmp/tmpCB2aIM.lastz\n",
      "\t/tmp/tmpDPZ7wb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sp_GCA_001044295.lastz\n",
      "Creating Asprgillus_sp_GCA_001044295 table\n",
      "Inserting data to Asprgillus_sp_GCA_001044295 table\n",
      "\n",
      "Running against Asprgillus_spnulosprus_GCA_003574815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp5iFQB5.fasta\n",
      "\t/tmp/tmppXzIP7.fasta\n",
      "\t/tmp/tmpfmgR7t.fasta\n",
      "\t/tmp/tmpzbveIW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvc3mtd.lastz\n",
      "\t/tmp/tmpm1qEzd.lastz\n",
      "\t/tmp/tmpE39WN6.lastz\n",
      "\t/tmp/tmp2MfmNn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_spnulosprus_GCA_003574815.lastz\n",
      "Creating Asprgillus_spnulosprus_GCA_003574815 table\n",
      "Inserting data to Asprgillus_spnulosprus_GCA_003574815 table\n",
      "\n",
      "Running against Asprgillus_steynii_GCF_002849105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpvTePD3.fasta\n",
      "\t/tmp/tmpb1B8Fa.fasta\n",
      "\t/tmp/tmpJXYmdT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcMciYT.lastz\n",
      "\t/tmp/tmpZIbN2Y.lastz\n",
      "\t/tmp/tmp8JnqVK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_steynii_GCF_002849105.lastz\n",
      "Creating Asprgillus_steynii_GCF_002849105 table\n",
      "Inserting data to Asprgillus_steynii_GCF_002849105 table\n",
      "\n",
      "Running against Asprgillus_sydowii_GCA_001890705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmppm16K2.fasta\n",
      "\t/tmp/tmpfsdT3a.fasta\n",
      "\t/tmp/tmpa0PIPy.fasta\n",
      "\t/tmp/tmpa9FS4m.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgMS24r.lastz\n",
      "\t/tmp/tmp9iYL52.lastz\n",
      "\t/tmp/tmpvQykmN.lastz\n",
      "\t/tmp/tmp6gnhwa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_sydowii_GCA_001890705.lastz\n",
      "Creating Asprgillus_sydowii_GCA_001890705 table\n",
      "Inserting data to Asprgillus_sydowii_GCA_001890705 table\n",
      "\n",
      "Running against Asprgillus_taichungensis_GCA_002850765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpJTTvtx.fasta\n",
      "\t/tmp/tmp0pqzto.fasta\n",
      "\t/tmp/tmpM0tFfe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp54UVSG.lastz\n",
      "\t/tmp/tmpBG4X4z.lastz\n",
      "\t/tmp/tmpuVNmFL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_taichungensis_GCA_002850765.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Asprgillus_taichungensis_GCA_002850765 table\n",
      "Inserting data to Asprgillus_taichungensis_GCA_002850765 table\n",
      "\n",
      "Running against Asprgillus_tamarii_GCA_009193485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpjg_XE_.fasta\n",
      "\t/tmp/tmpUhYGSb.fasta\n",
      "\t/tmp/tmpT7FjZi.fasta\n",
      "\t/tmp/tmpGf4Ayi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLnksZr.lastz\n",
      "\t/tmp/tmpXgW34a.lastz\n",
      "\t/tmp/tmphG_QRH.lastz\n",
      "\t/tmp/tmp54nViJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_tamarii_GCA_009193485.lastz\n",
      "Creating Asprgillus_tamarii_GCA_009193485 table\n",
      "Inserting data to Asprgillus_tamarii_GCA_009193485 table\n",
      "\n",
      "Running against Asprgillus_tanneri_GCA_003426965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpuACHUC.fasta\n",
      "\t/tmp/tmpTg5JVT.fasta\n",
      "\t/tmp/tmpzS1Kkq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoiDfHo.lastz\n",
      "\t/tmp/tmpyykn8s.lastz\n",
      "\t/tmp/tmp1EAV6G.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_tanneri_GCA_003426965.lastz\n",
      "Creating Asprgillus_tanneri_GCA_003426965 table\n",
      "Inserting data to Asprgillus_tanneri_GCA_003426965 table\n",
      "\n",
      "Running against Asprgillus_terreus_GCF_000149615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpYufm38.fasta\n",
      "\t/tmp/tmps_6LQf.fasta\n",
      "\t/tmp/tmpyfJ7o2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjt_9yr.lastz\n",
      "\t/tmp/tmpqLVEt5.lastz\n",
      "\t/tmp/tmpy5qnBu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_terreus_GCF_000149615.lastz\n",
      "Creating Asprgillus_terreus_GCF_000149615 table\n",
      "Inserting data to Asprgillus_terreus_GCF_000149615 table\n",
      "\n",
      "Running against Asprgillus_thermomutatus_GCF_002237265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpbp79qq.fasta\n",
      "\t/tmp/tmpUMDInr.fasta\n",
      "\t/tmp/tmpfWeBYY.fasta\n",
      "\t/tmp/tmpcJs8VC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnLJdbS.lastz\n",
      "\t/tmp/tmptRj7IP.lastz\n",
      "\t/tmp/tmp_IpHH8.lastz\n",
      "\t/tmp/tmpC0aPP8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_thermomutatus_GCF_002237265.lastz\n",
      "Creating Asprgillus_thermomutatus_GCF_002237265 table\n",
      "Inserting data to Asprgillus_thermomutatus_GCF_002237265 table\n",
      "\n",
      "Running against Asprgillus_transmontanensis_GCA_009193505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpDCKDaW.fasta\n",
      "\t/tmp/tmp6Sq13E.fasta\n",
      "\t/tmp/tmpu5G5PJ.fasta\n",
      "\t/tmp/tmpd7Ubhu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxhFP4k.lastz\n",
      "\t/tmp/tmp8_iC7I.lastz\n",
      "\t/tmp/tmpqAb0tF.lastz\n",
      "\t/tmp/tmpn8HmNo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_transmontanensis_GCA_009193505.lastz\n",
      "Creating Asprgillus_transmontanensis_GCA_009193505 table\n",
      "Inserting data to Asprgillus_transmontanensis_GCA_009193505 table\n",
      "\n",
      "Running against Asprgillus_tritici_GCA_009812425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpoAlphG.fasta\n",
      "\t/tmp/tmpT6UHRk.fasta\n",
      "\t/tmp/tmpMVs0ly.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyyP8e0.lastz\n",
      "\t/tmp/tmpuMXk4h.lastz\n",
      "\t/tmp/tmpw7lILO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_tritici_GCA_009812425.lastz\n",
      "Creating Asprgillus_tritici_GCA_009812425 table\n",
      "Inserting data to Asprgillus_tritici_GCA_009812425 table\n",
      "\n",
      "Running against Asprgillus_tubingensis_GCA_001890745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpBsx65x.fasta\n",
      "\t/tmp/tmpChOZhe.fasta\n",
      "\t/tmp/tmp49SmxD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE3_du4.lastz\n",
      "\t/tmp/tmpYJywKG.lastz\n",
      "\t/tmp/tmpp0heAB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_tubingensis_GCA_001890745.lastz\n",
      "Creating Asprgillus_tubingensis_GCA_001890745 table\n",
      "Inserting data to Asprgillus_tubingensis_GCA_001890745 table\n",
      "\n",
      "Running against Asprgillus_turcosus_GCA_002234965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp8Oh9Rg.fasta\n",
      "\t/tmp/tmpBlDGCs.fasta\n",
      "\t/tmp/tmpldrJV1.fasta\n",
      "\t/tmp/tmpzYWf52.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpY4drd3.lastz\n",
      "\t/tmp/tmpamzDwQ.lastz\n",
      "\t/tmp/tmpUMqUMQ.lastz\n",
      "\t/tmp/tmppL6Y37.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_turcosus_GCA_002234965.lastz\n",
      "Creating Asprgillus_turcosus_GCA_002234965 table\n",
      "Inserting data to Asprgillus_turcosus_GCA_002234965 table\n",
      "\n",
      "Running against Asprgillus_udagawae_GCA_001078395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp9epgzr.fasta\n",
      "\t/tmp/tmphHvESU.fasta\n",
      "\t/tmp/tmpDGnWhn.fasta\n",
      "\t/tmp/tmppCtnaD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6TbJ0r.lastz\n",
      "\t/tmp/tmpNAc7Jx.lastz\n",
      "\t/tmp/tmpesre7X.lastz\n",
      "\t/tmp/tmpSJmz4V.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_udagawae_GCA_001078395.lastz\n",
      "Creating Asprgillus_udagawae_GCA_001078395 table\n",
      "Inserting data to Asprgillus_udagawae_GCA_001078395 table\n",
      "\n",
      "Running against Asprgillus_unguis_GCA_003324175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpkygFHx.fasta\n",
      "\t/tmp/tmpoxxdEY.fasta\n",
      "\t/tmp/tmp7eSeYD.fasta\n",
      "\t/tmp/tmpHumNHl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpK0G41a.lastz\n",
      "\t/tmp/tmp2FqG3C.lastz\n",
      "\t/tmp/tmpYLTnwo.lastz\n",
      "\t/tmp/tmpI9tyw6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_unguis_GCA_003324175.lastz\n",
      "Creating Asprgillus_unguis_GCA_003324175 table\n",
      "Inserting data to Asprgillus_unguis_GCA_003324175 table\n",
      "\n",
      "Running against Asprgillus_ustus_GCA_000812125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpErUvls.fasta\n",
      "\t/tmp/tmp3zEN9B.fasta\n",
      "\t/tmp/tmpZXBd94.fasta\n",
      "\t/tmp/tmpV9nTYP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl1Bx11.lastz\n",
      "\t/tmp/tmpl2rqKX.lastz\n",
      "\t/tmp/tmpAbrS5M.lastz\n",
      "\t/tmp/tmpkAHI5H.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_ustus_GCA_000812125.lastz\n",
      "Creating Asprgillus_ustus_GCA_000812125 table\n",
      "Inserting data to Asprgillus_ustus_GCA_000812125 table\n",
      "\n",
      "Running against Asprgillus_uvarum_GCF_003184745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpRjB0Zj.fasta\n",
      "\t/tmp/tmpQiyzO_.fasta\n",
      "\t/tmp/tmpJlWUFE.fasta\n",
      "\t/tmp/tmplHNdMr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBqOYjW.lastz\n",
      "\t/tmp/tmpYUaiTh.lastz\n",
      "\t/tmp/tmpCBAATX.lastz\n",
      "\t/tmp/tmp680UIc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_uvarum_GCF_003184745.lastz\n",
      "Creating Asprgillus_uvarum_GCF_003184745 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Asprgillus_uvarum_GCF_003184745 table\n",
      "\n",
      "Running against Asprgillus_vadensis_GCF_003184925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpjdCW5j.fasta\n",
      "\t/tmp/tmpBu2YgZ.fasta\n",
      "\t/tmp/tmpIKOczp.fasta\n",
      "\t/tmp/tmpOmpFVA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTnmIaz.lastz\n",
      "\t/tmp/tmpNFVr91.lastz\n",
      "\t/tmp/tmpyE4BwN.lastz\n",
      "\t/tmp/tmp_4aMDE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_vadensis_GCF_003184925.lastz\n",
      "Creating Asprgillus_vadensis_GCF_003184925 table\n",
      "Inserting data to Asprgillus_vadensis_GCF_003184925 table\n",
      "\n",
      "Running against Asprgillus_versicolor_GCA_001890125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpygtUxf.fasta\n",
      "\t/tmp/tmpgngklV.fasta\n",
      "\t/tmp/tmpXMaRQu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuTUEvO.lastz\n",
      "\t/tmp/tmpHZwFwd.lastz\n",
      "\t/tmp/tmpuy5MU6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_versicolor_GCA_001890125.lastz\n",
      "Creating Asprgillus_versicolor_GCA_001890125 table\n",
      "Inserting data to Asprgillus_versicolor_GCA_001890125 table\n",
      "\n",
      "Running against Asprgillus_violaceofuscus_GCA_003184705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpqxYF0i.fasta\n",
      "\t/tmp/tmppWNcLY.fasta\n",
      "\t/tmp/tmpbN5MZK.fasta\n",
      "\t/tmp/tmp0S272z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOJ2F_1.lastz\n",
      "\t/tmp/tmpwpQvEq.lastz\n",
      "\t/tmp/tmpy3Rhcc.lastz\n",
      "\t/tmp/tmpHp_25u.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_violaceofuscus_GCA_003184705.lastz\n",
      "Creating Asprgillus_violaceofuscus_GCA_003184705 table\n",
      "Inserting data to Asprgillus_violaceofuscus_GCA_003184705 table\n",
      "\n",
      "Running against Asprgillus_viridinutans_GCA_004368095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpycuDR8.fasta\n",
      "\t/tmp/tmpByGH2X.fasta\n",
      "\t/tmp/tmpidbPDY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3Udt80.lastz\n",
      "\t/tmp/tmp9JkdCt.lastz\n",
      "\t/tmp/tmpR3qvzB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_viridinutans_GCA_004368095.lastz\n",
      "Creating Asprgillus_viridinutans_GCA_004368095 table\n",
      "Inserting data to Asprgillus_viridinutans_GCA_004368095 table\n",
      "\n",
      "Running against Asprgillus_welwitschiae_GCF_003344945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpXTBsu9.fasta\n",
      "\t/tmp/tmpC1Frpp.fasta\n",
      "\t/tmp/tmpN6u6OP.fasta\n",
      "\t/tmp/tmpamvCfS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmh0A6S.lastz\n",
      "\t/tmp/tmpqlIqkK.lastz\n",
      "\t/tmp/tmppHxs4w.lastz\n",
      "\t/tmp/tmpiAgoN2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_welwitschiae_GCF_003344945.lastz\n",
      "Creating Asprgillus_welwitschiae_GCF_003344945 table\n",
      "Inserting data to Asprgillus_welwitschiae_GCF_003344945 table\n",
      "\n",
      "Running against Asprgillus_wentii_GCA_001890725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpv2Yg0W.fasta\n",
      "\t/tmp/tmpKOPoxT.fasta\n",
      "\t/tmp/tmps27nlP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpipYouS.lastz\n",
      "\t/tmp/tmp63pH5X.lastz\n",
      "\t/tmp/tmpBHkUmp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_wentii_GCA_001890725.lastz\n",
      "Creating Asprgillus_wentii_GCA_001890725 table\n",
      "Inserting data to Asprgillus_wentii_GCA_001890725 table\n",
      "\n",
      "Running against Asprgillus_westerdijkiae_GCA_001307345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpOL52zn.fasta\n",
      "\t/tmp/tmpniguDq.fasta\n",
      "\t/tmp/tmp2UZgau.fasta\n",
      "\t/tmp/tmpXc73iW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbEZTke.lastz\n",
      "\t/tmp/tmp41ouef.lastz\n",
      "\t/tmp/tmpPo0EKV.lastz\n",
      "\t/tmp/tmpxvhhki.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Asprgillus_westerdijkiae_GCA_001307345.lastz\n",
      "Creating Asprgillus_westerdijkiae_GCA_001307345 table\n",
      "Inserting data to Asprgillus_westerdijkiae_GCA_001307345 table\n",
      "\n",
      "Running against Basipetospra_chlamydospra_GCA_001599675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpn0EsbA.fasta\n",
      "\t/tmp/tmpCRUDkQ.fasta\n",
      "\t/tmp/tmpzTmpku.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4FsnZK.lastz\n",
      "\t/tmp/tmpbv2fXv.lastz\n",
      "\t/tmp/tmpJ8dw7w.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Basipetospra_chlamydospra_GCA_001599675.lastz\n",
      "Creating Basipetospra_chlamydospra_GCA_001599675 table\n",
      "Inserting data to Basipetospra_chlamydospra_GCA_001599675 table\n",
      "\n",
      "Running against Byssochlamys_nivea_GCA_003116535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpmjz1mK.fasta\n",
      "\t/tmp/tmp2tmRk0.fasta\n",
      "\t/tmp/tmph_dK_T.fasta\n",
      "\t/tmp/tmpw0y3oJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfzqgy6.lastz\n",
      "\t/tmp/tmpgoImlY.lastz\n",
      "\t/tmp/tmpTdnC0A.lastz\n",
      "\t/tmp/tmppZ7y6N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Byssochlamys_nivea_GCA_003116535.lastz\n",
      "Creating Byssochlamys_nivea_GCA_003116535 table\n",
      "Inserting data to Byssochlamys_nivea_GCA_003116535 table\n",
      "\n",
      "Running against Byssochlamys_sp_GCA_002914405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpHOWtIV.fasta\n",
      "\t/tmp/tmpA_FDTq.fasta\n",
      "\t/tmp/tmpBHhOps.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDPrr4u.lastz\n",
      "\t/tmp/tmpWdDVvB.lastz\n",
      "\t/tmp/tmppK4H_h.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Byssochlamys_sp_GCA_002914405.lastz\n",
      "Creating Byssochlamys_sp_GCA_002914405 table\n",
      "Inserting data to Byssochlamys_sp_GCA_002914405 table\n",
      "\n",
      "Running against Byssochlamys_spctabilis_GCF_004022145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpGAzxzD.fasta\n",
      "\t/tmp/tmpuFB1bn.fasta\n",
      "\t/tmp/tmpJDK8hW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRQ4MK_.lastz\n",
      "\t/tmp/tmpmRbgv6.lastz\n",
      "\t/tmp/tmp0ahwRT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Byssochlamys_spctabilis_GCF_004022145.lastz\n",
      "Creating Byssochlamys_spctabilis_GCF_004022145 table\n",
      "Inserting data to Byssochlamys_spctabilis_GCF_004022145 table\n",
      "\n",
      "Running against Elaphomyces_granulatus_GCA_002240705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpCIwYyZ.fasta\n",
      "\t/tmp/tmp1AZUkY.fasta\n",
      "\t/tmp/tmp5PHUOL.fasta\n",
      "\t/tmp/tmpHKe9wy.fasta\n",
      "\t/tmp/tmp4QTFqP.fasta\n",
      "\t/tmp/tmpfKv6jx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZwYLyr.lastz\n",
      "\t/tmp/tmpfUoVEb.lastz\n",
      "\t/tmp/tmpvcEcK6.lastz\n",
      "\t/tmp/tmpHdePMp.lastz\n",
      "\t/tmp/tmpHPXJRZ.lastz\n",
      "\t/tmp/tmpmrDXJ7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Elaphomyces_granulatus_GCA_002240705.lastz\n",
      "Creating Elaphomyces_granulatus_GCA_002240705 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Elaphomyces_granulatus_GCA_002240705 table\n",
      "\n",
      "Running against Monascus_purpureus_GCA_003184285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpuGMOqU.fasta\n",
      "\t/tmp/tmpEoqSuB.fasta\n",
      "\t/tmp/tmp8v3Of4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4txp5l.lastz\n",
      "\t/tmp/tmpO55uSC.lastz\n",
      "\t/tmp/tmpuxfdC2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Monascus_purpureus_GCA_003184285.lastz\n",
      "Creating Monascus_purpureus_GCA_003184285 table\n",
      "Inserting data to Monascus_purpureus_GCA_003184285 table\n",
      "\n",
      "Running against Monascus_ruber_GCA_002976275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpAezBtt.fasta\n",
      "\t/tmp/tmpx6V6qE.fasta\n",
      "\t/tmp/tmpjOk8Zu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp14Iq5M.lastz\n",
      "\t/tmp/tmpPSlAr1.lastz\n",
      "\t/tmp/tmpnVIiaX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Monascus_ruber_GCA_002976275.lastz\n",
      "Creating Monascus_ruber_GCA_002976275 table\n",
      "Inserting data to Monascus_ruber_GCA_002976275 table\n",
      "\n",
      "Running against Penicilliopsis_zonata_GCF_001890105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpzr6VJl.fasta\n",
      "\t/tmp/tmp0u3z4O.fasta\n",
      "\t/tmp/tmp0iB3pI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpm8F3QF.lastz\n",
      "\t/tmp/tmp20eyCT.lastz\n",
      "\t/tmp/tmpytgpn3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicilliopsis_zonata_GCF_001890105.lastz\n",
      "Creating Penicilliopsis_zonata_GCF_001890105 table\n",
      "Inserting data to Penicilliopsis_zonata_GCF_001890105 table\n",
      "\n",
      "Running against Penicillium_antarcticum_GCA_002072345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpP4MKsS.fasta\n",
      "\t/tmp/tmpf4xRvp.fasta\n",
      "\t/tmp/tmpxFzij8.fasta\n",
      "\t/tmp/tmpsjmeR3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvRWZsM.lastz\n",
      "\t/tmp/tmpn0_CRc.lastz\n",
      "\t/tmp/tmpxPBP8q.lastz\n",
      "\t/tmp/tmped5Qvt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_antarcticum_GCA_002072345.lastz\n",
      "Creating Penicillium_antarcticum_GCA_002072345 table\n",
      "Inserting data to Penicillium_antarcticum_GCA_002072345 table\n",
      "\n",
      "Running against Penicillium_arizonense_GCF_001773325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpDdCDgM.fasta\n",
      "\t/tmp/tmppy_bqz.fasta\n",
      "\t/tmp/tmpfwAxTa.fasta\n",
      "\t/tmp/tmpJICprG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeTrhOZ.lastz\n",
      "\t/tmp/tmpW35FdZ.lastz\n",
      "\t/tmp/tmpkTiJPM.lastz\n",
      "\t/tmp/tmpGtEWaB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_arizonense_GCF_001773325.lastz\n",
      "Creating Penicillium_arizonense_GCF_001773325 table\n",
      "Inserting data to Penicillium_arizonense_GCF_001773325 table\n",
      "\n",
      "Running against Penicillium_biforme_GCA_000577785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpm4N41z.fasta\n",
      "\t/tmp/tmp6lHxGY.fasta\n",
      "\t/tmp/tmpPJWRby.fasta\n",
      "\t/tmp/tmp1iuLoz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZbJidy.lastz\n",
      "\t/tmp/tmp8kY8Ek.lastz\n",
      "\t/tmp/tmpBnGDre.lastz\n",
      "\t/tmp/tmpLXDv0T.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_biforme_GCA_000577785.lastz\n",
      "Creating Penicillium_biforme_GCA_000577785 table\n",
      "Inserting data to Penicillium_biforme_GCA_000577785 table\n",
      "\n",
      "Running against Penicillium_brasilianum_GCA_001048715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpubFTje.fasta\n",
      "\t/tmp/tmpmYBhiv.fasta\n",
      "\t/tmp/tmpFWarV0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGqrWzN.lastz\n",
      "\t/tmp/tmpPyPXVR.lastz\n",
      "\t/tmp/tmp39ow8j.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_brasilianum_GCA_001048715.lastz\n",
      "Creating Penicillium_brasilianum_GCA_001048715 table\n",
      "Inserting data to Penicillium_brasilianum_GCA_001048715 table\n",
      "\n",
      "Running against Penicillium_camemberti_GCA_000513335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmptgEgcA.fasta\n",
      "\t/tmp/tmpHsEQI4.fasta\n",
      "\t/tmp/tmp7BuCM8.fasta\n",
      "\t/tmp/tmpGLjOwQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpz3TtpP.lastz\n",
      "\t/tmp/tmpn54EJ8.lastz\n",
      "\t/tmp/tmpzX5tuI.lastz\n",
      "\t/tmp/tmp4BhOfr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_camemberti_GCA_000513335.lastz\n",
      "Creating Penicillium_camemberti_GCA_000513335 table\n",
      "Inserting data to Penicillium_camemberti_GCA_000513335 table\n",
      "\n",
      "Running against Penicillium_capsulatum_GCA_000943775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp4Rbl3G.fasta\n",
      "\t/tmp/tmp4SrWSU.fasta\n",
      "\t/tmp/tmp431AFs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpM__xzx.lastz\n",
      "\t/tmp/tmpiw2GEh.lastz\n",
      "\t/tmp/tmpcrhrmd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_capsulatum_GCA_000943775.lastz\n",
      "Creating Penicillium_capsulatum_GCA_000943775 table\n",
      "Inserting data to Penicillium_capsulatum_GCA_000943775 table\n",
      "\n",
      "Running against Penicillium_carneum_GCA_000577495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpgo_CDD.fasta\n",
      "\t/tmp/tmpYWY4zM.fasta\n",
      "\t/tmp/tmpY4pZAt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpK737Ug.lastz\n",
      "\t/tmp/tmplpwELm.lastz\n",
      "\t/tmp/tmpojboaS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_carneum_GCA_000577495.lastz\n",
      "Creating Penicillium_carneum_GCA_000577495 table\n",
      "Inserting data to Penicillium_carneum_GCA_000577495 table\n",
      "\n",
      "Running against Penicillium_chrysogenum_GCA_000710275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpkNTzOH.fasta\n",
      "\t/tmp/tmp5nv26n.fasta\n",
      "\t/tmp/tmpDb9E3o.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpal2aZB.lastz\n",
      "\t/tmp/tmpCWmF16.lastz\n",
      "\t/tmp/tmpCxVaA9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_chrysogenum_GCA_000710275.lastz\n",
      "Creating Penicillium_chrysogenum_GCA_000710275 table\n",
      "Inserting data to Penicillium_chrysogenum_GCA_000710275 table\n",
      "\n",
      "Running against Penicillium_citrinum_GCA_001950535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp7o7cWN.fasta\n",
      "\t/tmp/tmpL2oEH7.fasta\n",
      "\t/tmp/tmpgl8eNb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprRb0RS.lastz\n",
      "\t/tmp/tmp0hgPzG.lastz\n",
      "\t/tmp/tmpR9RTTn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_citrinum_GCA_001950535.lastz\n",
      "Creating Penicillium_citrinum_GCA_001950535 table\n",
      "Inserting data to Penicillium_citrinum_GCA_001950535 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Penicillium_coprophilum_GCA_002072405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpScZ4Ho.fasta\n",
      "\t/tmp/tmpXulggh.fasta\n",
      "\t/tmp/tmpu4_nNg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMFq62Y.lastz\n",
      "\t/tmp/tmp5iJuv8.lastz\n",
      "\t/tmp/tmpk0JLFO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_coprophilum_GCA_002072405.lastz\n",
      "Creating Penicillium_coprophilum_GCA_002072405 table\n",
      "Inserting data to Penicillium_coprophilum_GCA_002072405 table\n",
      "\n",
      "Running against Penicillium_decumbens_GCA_002072245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp1UxKMU.fasta\n",
      "\t/tmp/tmpnzEnM4.fasta\n",
      "\t/tmp/tmpBEqQx_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBmeIWx.lastz\n",
      "\t/tmp/tmpCWgkL9.lastz\n",
      "\t/tmp/tmpDhIeII.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_decumbens_GCA_002072245.lastz\n",
      "Creating Penicillium_decumbens_GCA_002072245 table\n",
      "Inserting data to Penicillium_decumbens_GCA_002072245 table\n",
      "\n",
      "Running against Penicillium_digitatum_GCF_000315645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpoMa48I.fasta\n",
      "\t/tmp/tmpLlQF2z.fasta\n",
      "\t/tmp/tmpbF55vN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwjsHrO.lastz\n",
      "\t/tmp/tmplwZfO_.lastz\n",
      "\t/tmp/tmpvE9kmD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_digitatum_GCF_000315645.lastz\n",
      "Creating Penicillium_digitatum_GCF_000315645 table\n",
      "Inserting data to Penicillium_digitatum_GCF_000315645 table\n",
      "\n",
      "Running against Penicillium_expansum_GCF_000769745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmprOoCU7.fasta\n",
      "\t/tmp/tmpz6qAdP.fasta\n",
      "\t/tmp/tmp5SWD_4.fasta\n",
      "\t/tmp/tmpyqGPcO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpexJi3M.lastz\n",
      "\t/tmp/tmpjdsJCK.lastz\n",
      "\t/tmp/tmphF3aE7.lastz\n",
      "\t/tmp/tmpyQBxaH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_expansum_GCF_000769745.lastz\n",
      "Creating Penicillium_expansum_GCF_000769745 table\n",
      "Inserting data to Penicillium_expansum_GCF_000769745 table\n",
      "\n",
      "Running against Penicillium_flavigenum_GCA_002072365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpxm9kQL.fasta\n",
      "\t/tmp/tmp04nDi1.fasta\n",
      "\t/tmp/tmpkV1G0m.fasta\n",
      "\t/tmp/tmpnfvXek.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHx_GUf.lastz\n",
      "\t/tmp/tmpJkPxCH.lastz\n",
      "\t/tmp/tmpAltse8.lastz\n",
      "\t/tmp/tmpNCw0zx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_flavigenum_GCA_002072365.lastz\n",
      "Creating Penicillium_flavigenum_GCA_002072365 table\n",
      "Inserting data to Penicillium_flavigenum_GCA_002072365 table\n",
      "\n",
      "Running against Penicillium_freii_GCA_001513925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpu9k6KX.fasta\n",
      "\t/tmp/tmpnckQ9H.fasta\n",
      "\t/tmp/tmpLYlY9N.fasta\n",
      "\t/tmp/tmpehGNuR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNn_Mvg.lastz\n",
      "\t/tmp/tmp_U99GO.lastz\n",
      "\t/tmp/tmp43fDwG.lastz\n",
      "\t/tmp/tmpTsy4cL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_freii_GCA_001513925.lastz\n",
      "Creating Penicillium_freii_GCA_001513925 table\n",
      "Inserting data to Penicillium_freii_GCA_001513925 table\n",
      "\n",
      "Running against Penicillium_fuscoglaucum_GCA_000576735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpMVYzIB.fasta\n",
      "\t/tmp/tmpWTRJL7.fasta\n",
      "\t/tmp/tmpTEoEhq.fasta\n",
      "\t/tmp/tmpriwcME.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpX9IQxJ.lastz\n",
      "\t/tmp/tmpJNvPAP.lastz\n",
      "\t/tmp/tmp9TgQVt.lastz\n",
      "\t/tmp/tmp5yJmcK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_fuscoglaucum_GCA_000576735.lastz\n",
      "Creating Penicillium_fuscoglaucum_GCA_000576735 table\n",
      "Inserting data to Penicillium_fuscoglaucum_GCA_000576735 table\n",
      "\n",
      "Running against Penicillium_griseofulvum_GCA_001561935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpa9V5dG.fasta\n",
      "\t/tmp/tmpK5TX9W.fasta\n",
      "\t/tmp/tmpj3tXpE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUCVnIJ.lastz\n",
      "\t/tmp/tmpNmz_Zo.lastz\n",
      "\t/tmp/tmpZf7fux.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_griseofulvum_GCA_001561935.lastz\n",
      "Creating Penicillium_griseofulvum_GCA_001561935 table\n",
      "Inserting data to Penicillium_griseofulvum_GCA_001561935 table\n",
      "\n",
      "Running against Penicillium_italicum_GCA_002116305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpoYyf2Y.fasta\n",
      "\t/tmp/tmp1tcbGp.fasta\n",
      "\t/tmp/tmpxDn1ES.fasta\n",
      "\t/tmp/tmpjVuJ4j.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptB3fdi.lastz\n",
      "\t/tmp/tmpwXZTJw.lastz\n",
      "\t/tmp/tmpjIityN.lastz\n",
      "\t/tmp/tmpP1iXUh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_italicum_GCA_002116305.lastz\n",
      "Creating Penicillium_italicum_GCA_002116305 table\n",
      "Inserting data to Penicillium_italicum_GCA_002116305 table\n",
      "\n",
      "Running against Penicillium_janthinellum_GCA_002369805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpJVNB7q.fasta\n",
      "\t/tmp/tmpqJufG9.fasta\n",
      "\t/tmp/tmp7GdAwz.fasta\n",
      "\t/tmp/tmp_fyssv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvN_763.lastz\n",
      "\t/tmp/tmpjZUTWd.lastz\n",
      "\t/tmp/tmp9bHcyt.lastz\n",
      "\t/tmp/tmpxA5TlY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_janthinellum_GCA_002369805.lastz\n",
      "Creating Penicillium_janthinellum_GCA_002369805 table\n",
      "Inserting data to Penicillium_janthinellum_GCA_002369805 table\n",
      "\n",
      "Running against Penicillium_nalgiovense_GCA_002072425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpy55w94.fasta\n",
      "\t/tmp/tmp3m9mIE.fasta\n",
      "\t/tmp/tmpo8XPB6.fasta\n",
      "\t/tmp/tmpjGol8b.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvZDOS7.lastz\n",
      "\t/tmp/tmpnBmMd4.lastz\n",
      "\t/tmp/tmpMB7lOo.lastz\n",
      "\t/tmp/tmp_k_mIw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_nalgiovense_GCA_002072425.lastz\n",
      "Creating Penicillium_nalgiovense_GCA_002072425 table\n",
      "Inserting data to Penicillium_nalgiovense_GCA_002072425 table\n",
      "\n",
      "Running against Penicillium_nordicum_GCA_000733025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp_QP60Q.fasta\n",
      "\t/tmp/tmpjfhdG3.fasta\n",
      "\t/tmp/tmpFtRIpi.fasta\n",
      "\t/tmp/tmp3tCihr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7FZ4qz.lastz\n",
      "\t/tmp/tmpItNK9c.lastz\n",
      "\t/tmp/tmpkORqL1.lastz\n",
      "\t/tmp/tmpA5qSL1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_nordicum_GCA_000733025.lastz\n",
      "Creating Penicillium_nordicum_GCA_000733025 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Penicillium_nordicum_GCA_000733025 table\n",
      "\n",
      "Running against Penicillium_oxalicum_GCA_000346795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpY9cUFM.fasta\n",
      "\t/tmp/tmp3HjynV.fasta\n",
      "\t/tmp/tmp9BCFzv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpC3aNUX.lastz\n",
      "\t/tmp/tmpzfwSgZ.lastz\n",
      "\t/tmp/tmpD8E7wQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_oxalicum_GCA_000346795.lastz\n",
      "Creating Penicillium_oxalicum_GCA_000346795 table\n",
      "Inserting data to Penicillium_oxalicum_GCA_000346795 table\n",
      "\n",
      "Running against Penicillium_paneum_GCA_000577715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpDaZfka.fasta\n",
      "\t/tmp/tmpKTCEh7.fasta\n",
      "\t/tmp/tmpidqnFw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpT25hF1.lastz\n",
      "\t/tmp/tmp_zgZF6.lastz\n",
      "\t/tmp/tmpotZgn0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_paneum_GCA_000577715.lastz\n",
      "Creating Penicillium_paneum_GCA_000577715 table\n",
      "Inserting data to Penicillium_paneum_GCA_000577715 table\n",
      "\n",
      "Running against Penicillium_paxilli_GCA_000347475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpbor1Rx.fasta\n",
      "\t/tmp/tmpWJFswx.fasta\n",
      "\t/tmp/tmp06inaG.fasta\n",
      "\t/tmp/tmptwIWLX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpw21J3O.lastz\n",
      "\t/tmp/tmpnDcWHj.lastz\n",
      "\t/tmp/tmpt02JPC.lastz\n",
      "\t/tmp/tmp6fK_Zu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_paxilli_GCA_000347475.lastz\n",
      "Creating Penicillium_paxilli_GCA_000347475 table\n",
      "Inserting data to Penicillium_paxilli_GCA_000347475 table\n",
      "\n",
      "Running against Penicillium_polonicum_GCA_003344595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpyJDHNi.fasta\n",
      "\t/tmp/tmpfC7K9V.fasta\n",
      "\t/tmp/tmp2RJRUX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4PeuHS.lastz\n",
      "\t/tmp/tmpsLCxke.lastz\n",
      "\t/tmp/tmp08O4qr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_polonicum_GCA_003344595.lastz\n",
      "Creating Penicillium_polonicum_GCA_003344595 table\n",
      "Inserting data to Penicillium_polonicum_GCA_003344595 table\n",
      "\n",
      "Running against Penicillium_roqueforti_GCA_000513255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpbNHLx2.fasta\n",
      "\t/tmp/tmph6NgCP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoGsz29.lastz\n",
      "\t/tmp/tmp8rs2cQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_roqueforti_GCA_000513255.lastz\n",
      "Creating Penicillium_roqueforti_GCA_000513255 table\n",
      "Inserting data to Penicillium_roqueforti_GCA_000513255 table\n",
      "\n",
      "Running against Penicillium_rubens_GCF_000226395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpjo_iF9.fasta\n",
      "\t/tmp/tmpa9981K.fasta\n",
      "\t/tmp/tmpyZz8RK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVy0mqq.lastz\n",
      "\t/tmp/tmpUDNwQU.lastz\n",
      "\t/tmp/tmpD6t2Iy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_rubens_GCF_000226395.lastz\n",
      "Creating Penicillium_rubens_GCF_000226395 table\n",
      "Inserting data to Penicillium_rubens_GCF_000226395 table\n",
      "\n",
      "Running against Penicillium_sclerotiorum_GCA_001750025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpP4T_5H.fasta\n",
      "\t/tmp/tmpL1GtpX.fasta\n",
      "\t/tmp/tmpruupAi.fasta\n",
      "\t/tmp/tmp4g8zzi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOgQB1M.lastz\n",
      "\t/tmp/tmp5G_467.lastz\n",
      "\t/tmp/tmp_YfMoI.lastz\n",
      "\t/tmp/tmpSP3DUi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_sclerotiorum_GCA_001750025.lastz\n",
      "Creating Penicillium_sclerotiorum_GCA_001750025 table\n",
      "Inserting data to Penicillium_sclerotiorum_GCA_001750025 table\n",
      "\n",
      "Running against Penicillium_solitum_GCA_000952775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpL8L1sP.fasta\n",
      "\t/tmp/tmpIpwivT.fasta\n",
      "\t/tmp/tmpBcD8uz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3G7ynE.lastz\n",
      "\t/tmp/tmptE40PN.lastz\n",
      "\t/tmp/tmpSCjAII.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_solitum_GCA_000952775.lastz\n",
      "Creating Penicillium_solitum_GCA_000952775 table\n",
      "Inserting data to Penicillium_solitum_GCA_000952775 table\n",
      "\n",
      "Running against Penicillium_sp_GCA_003800485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpOyM4qJ.fasta\n",
      "\t/tmp/tmpowKbdL.fasta\n",
      "\t/tmp/tmpxx5Qof.fasta\n",
      "\t/tmp/tmpyJMPbo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1orvVo.lastz\n",
      "\t/tmp/tmpb7B6Qq.lastz\n",
      "\t/tmp/tmp20Di7Q.lastz\n",
      "\t/tmp/tmpRKqvbp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_sp_GCA_003800485.lastz\n",
      "Creating Penicillium_sp_GCA_003800485 table\n",
      "Inserting data to Penicillium_sp_GCA_003800485 table\n",
      "\n",
      "Running against Penicillium_steckii_GCA_002072375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp_idjdN.fasta\n",
      "\t/tmp/tmp32ifQC.fasta\n",
      "\t/tmp/tmpw38Rtj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptjoWi1.lastz\n",
      "\t/tmp/tmpMyD8kB.lastz\n",
      "\t/tmp/tmpH2DYaB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_steckii_GCA_002072375.lastz\n",
      "Creating Penicillium_steckii_GCA_002072375 table\n",
      "Inserting data to Penicillium_steckii_GCA_002072375 table\n",
      "\n",
      "Running against Penicillium_subrubescens_GCA_001908125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpEJGauY.fasta\n",
      "\t/tmp/tmpEIEcUL.fasta\n",
      "\t/tmp/tmpqg3WuO.fasta\n",
      "\t/tmp/tmpwWqpSE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9B1dr2.lastz\n",
      "\t/tmp/tmp1YRaUB.lastz\n",
      "\t/tmp/tmpSFkj9L.lastz\n",
      "\t/tmp/tmpKTzmBl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_subrubescens_GCA_001908125.lastz\n",
      "Creating Penicillium_subrubescens_GCA_001908125 table\n",
      "Inserting data to Penicillium_subrubescens_GCA_001908125 table\n",
      "\n",
      "Running against Penicillium_verrucosum_GCA_000970515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpwvbkHJ.fasta\n",
      "\t/tmp/tmpoC6KHG.fasta\n",
      "\t/tmp/tmpqTit0u.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiFh1CB.lastz\n",
      "\t/tmp/tmpxvdohD.lastz\n",
      "\t/tmp/tmpPn13Bf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_verrucosum_GCA_000970515.lastz\n",
      "Creating Penicillium_verrucosum_GCA_000970515 table\n",
      "Inserting data to Penicillium_verrucosum_GCA_000970515 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Penicillium_vulpinum_GCA_002072255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpQOJoke.fasta\n",
      "\t/tmp/tmpqu_qF5.fasta\n",
      "\t/tmp/tmpqarVJj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGqXuaN.lastz\n",
      "\t/tmp/tmpftUkzh.lastz\n",
      "\t/tmp/tmp7eAo7v.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Penicillium_vulpinum_GCA_002072255.lastz\n",
      "Creating Penicillium_vulpinum_GCA_002072255 table\n",
      "Inserting data to Penicillium_vulpinum_GCA_002072255 table\n",
      "\n",
      "Running against Phialosimplex_sp_GCA_003698115.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp4paltM.fasta\n",
      "\t/tmp/tmpcPg78V.fasta\n",
      "\t/tmp/tmpneXiCj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYdgR32.lastz\n",
      "\t/tmp/tmpqIhb1q.lastz\n",
      "\t/tmp/tmpgUGtJd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Phialosimplex_sp_GCA_003698115.lastz\n",
      "Creating Phialosimplex_sp_GCA_003698115 table\n",
      "Inserting data to Phialosimplex_sp_GCA_003698115 table\n",
      "\n",
      "Running against Rasamsonia_emersonii_GCF_000968595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpQVWCOd.fasta\n",
      "\t/tmp/tmplnbhSb.fasta\n",
      "\t/tmp/tmp_vw_7o.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpP4KAlB.lastz\n",
      "\t/tmp/tmpBJGmcR.lastz\n",
      "\t/tmp/tmpEVrkE4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Rasamsonia_emersonii_GCF_000968595.lastz\n",
      "Creating Rasamsonia_emersonii_GCF_000968595 table\n",
      "Inserting data to Rasamsonia_emersonii_GCF_000968595 table\n",
      "\n",
      "Running against Talaromyces_adpressus_GCA_002775195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpPFeRqs.fasta\n",
      "\t/tmp/tmpJQAbRe.fasta\n",
      "\t/tmp/tmpYmXt6O.fasta\n",
      "\t/tmp/tmpPHb3S_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptCs5eZ.lastz\n",
      "\t/tmp/tmpfkJMeR.lastz\n",
      "\t/tmp/tmp9wEAu2.lastz\n",
      "\t/tmp/tmpMZumk4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Talaromyces_adpressus_GCA_002775195.lastz\n",
      "Creating Talaromyces_adpressus_GCA_002775195 table\n",
      "Inserting data to Talaromyces_adpressus_GCA_002775195 table\n",
      "\n",
      "Running against Talaromyces_amestolkiae_GCA_001896365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpfp5rND.fasta\n",
      "\t/tmp/tmp3paPuB.fasta\n",
      "\t/tmp/tmpxXtE8b.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOGSC9l.lastz\n",
      "\t/tmp/tmpUVkzyi.lastz\n",
      "\t/tmp/tmpwolbNN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/eurotiales/extract_probes_from_group/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Talaromyces_amestolkiae_GCA_001896365.lastz\n",
      "Creating Talaromyces_amestolkiae_GCA_001896365 table\n",
      "Inserting data to Talaromyces_amestolkiae_GCA_001896365 table\n",
      "\n",
      "Running against Talaromyces_atroroseus_GCF_001907595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_run_multiple_lastzs_sqlite\", line 257, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_run_multiple_lastzs_sqlite\", line 250, in main\n",
      "    align_against_scaffolds(cur, args, path)\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_run_multiple_lastzs_sqlite\", line 200, in align_against_scaffolds\n",
      "    multi_lastz_runner(output, args.cores, target, args.probefile, True, args.coverage, args.identity)\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/lib/python2.7/site-packages/phyluce/many_lastz.py\", line 122, in multi_lastz_runner\n",
      "    pool = multiprocessing.Pool(cores)\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/lib/python2.7/multiprocessing/__init__.py\", line 232, in Pool\n",
      "    return Pool(processes, initializer, initargs, maxtasksperchild)\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/lib/python2.7/multiprocessing/pool.py\", line 179, in __init__\n",
      "    self._task_handler.start()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/lib/python2.7/threading.py\", line 736, in start\n",
      "    _start_new_thread(self.__bootstrap, ())\n",
      "thread.error: can't start new thread\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/eurotiales/extract_probes_from_group/eurotiales_genome.conf --lastz results/phyluce/eurotiales/validate_intervals/lastz --probes 120 --probe-prefix uce_eurotiales_ --name-pattern \"Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/eurotiales/extract_probes_from_group/probe_fasta;\n",
      "2020-02-10 10:59:01,543 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-10 10:59:01,555 - Phyluce - INFO - ----- Working on Asprgillus_aculeatinus_GCF_003184765 genome ----\n",
      "2020-02-10 10:59:01,556 - Phyluce - INFO - Reading Asprgillus_aculeatinus_GCF_003184765 genome\n",
      "2020-02-10 10:59:03,181 - Phyluce - INFO - Asprgillus_aculeatinus_GCF_003184765: 142 uces, 41 dupes, 101 non-dupes, 1 orient drop, 0 length drop, 100 written\n",
      "2020-02-10 10:59:03,181 - Phyluce - INFO - ------ Working on Asprgillus_aculeatus_GCF_001890905 genome -----\n",
      "2020-02-10 10:59:03,182 - Phyluce - INFO - Reading Asprgillus_aculeatus_GCF_001890905 genome\n",
      "2020-02-10 10:59:04,276 - Phyluce - INFO - Asprgillus_aculeatus_GCF_001890905: 154 uces, 57 dupes, 97 non-dupes, 0 orient drop, 1 length drop, 96 written\n",
      "2020-02-10 10:59:04,277 - Phyluce - INFO - --------- Working on Asprgillus_aff_GCA_001931935 genome --------\n",
      "2020-02-10 10:59:04,278 - Phyluce - INFO - Reading Asprgillus_aff_GCA_001931935 genome\n",
      "2020-02-10 10:59:05,184 - Phyluce - INFO - Asprgillus_aff_GCA_001931935: 142 uces, 49 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-10 10:59:05,184 - Phyluce - INFO - ----- Working on Asprgillus_albertensis_GCA_009193605 genome ----\n",
      "2020-02-10 10:59:05,185 - Phyluce - INFO - Reading Asprgillus_albertensis_GCA_009193605 genome\n",
      "2020-02-10 10:59:06,236 - Phyluce - INFO - Asprgillus_albertensis_GCA_009193605: 136 uces, 49 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-10 10:59:06,236 - Phyluce - INFO - ------ Working on Asprgillus_alliaceus_GCF_009176365 genome -----\n",
      "2020-02-10 10:59:06,237 - Phyluce - INFO - Reading Asprgillus_alliaceus_GCF_009176365 genome\n",
      "2020-02-10 10:59:07,234 - Phyluce - INFO - Asprgillus_alliaceus_GCF_009176365: 136 uces, 48 dupes, 88 non-dupes, 0 orient drop, 0 length drop, 88 written\n",
      "2020-02-10 10:59:07,234 - Phyluce - INFO - ------- Working on Asprgillus_amoenus_GCA_009812435 genome ------\n",
      "2020-02-10 10:59:07,235 - Phyluce - INFO - Reading Asprgillus_amoenus_GCA_009812435 genome\n",
      "2020-02-10 10:59:08,218 - Phyluce - INFO - Asprgillus_amoenus_GCA_009812435: 150 uces, 45 dupes, 105 non-dupes, 0 orient drop, 0 length drop, 105 written\n",
      "2020-02-10 10:59:08,218 - Phyluce - INFO - ---- Working on Asprgillus_arachidicola_GCA_009193545 genome ----\n",
      "2020-02-10 10:59:08,219 - Phyluce - INFO - Reading Asprgillus_arachidicola_GCA_009193545 genome\n",
      "2020-02-10 10:59:09,242 - Phyluce - INFO - Asprgillus_arachidicola_GCA_009193545: 140 uces, 47 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-10 10:59:09,242 - Phyluce - INFO - ------ Working on Asprgillus_avenaceus_GCA_009193465 genome -----\n",
      "2020-02-10 10:59:09,243 - Phyluce - INFO - Reading Asprgillus_avenaceus_GCA_009193465 genome\n",
      "2020-02-10 10:59:10,310 - Phyluce - INFO - Asprgillus_avenaceus_GCA_009193465: 134 uces, 47 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-10 10:59:10,310 - Phyluce - INFO - ------- Working on Asprgillus_awamori_GCA_003850985 genome ------\n",
      "2020-02-10 10:59:10,311 - Phyluce - INFO - Reading Asprgillus_awamori_GCA_003850985 genome\n",
      "2020-02-10 10:59:11,295 - Phyluce - INFO - Asprgillus_awamori_GCA_003850985: 145 uces, 46 dupes, 99 non-dupes, 1 orient drop, 10 length drop, 88 written\n",
      "2020-02-10 10:59:11,296 - Phyluce - INFO - ---- Working on Asprgillus_bertholletius_GCA_009193595 genome ---\n",
      "2020-02-10 10:59:11,297 - Phyluce - INFO - Reading Asprgillus_bertholletius_GCA_009193595 genome\n",
      "2020-02-10 10:59:12,257 - Phyluce - INFO - Asprgillus_bertholletius_GCA_009193595: 135 uces, 46 dupes, 89 non-dupes, 0 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 10:59:12,258 - Phyluce - INFO - ------ Working on Asprgillus_bombycis_GCF_001792695 genome ------\n",
      "2020-02-10 10:59:12,271 - Phyluce - INFO - Reading Asprgillus_bombycis_GCF_001792695 genome\n",
      "2020-02-10 10:59:13,253 - Phyluce - INFO - Asprgillus_bombycis_GCF_001792695: 162 uces, 48 dupes, 114 non-dupes, 0 orient drop, 0 length drop, 114 written\n",
      "2020-02-10 10:59:13,254 - Phyluce - INFO - ---- Working on Asprgillus_brasiliensis_GCA_001889945 genome ----\n",
      "2020-02-10 10:59:13,255 - Phyluce - INFO - Reading Asprgillus_brasiliensis_GCA_001889945 genome\n",
      "2020-02-10 10:59:14,146 - Phyluce - INFO - Asprgillus_brasiliensis_GCA_001889945: 137 uces, 40 dupes, 97 non-dupes, 1 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 10:59:14,146 - Phyluce - INFO - -- Working on Asprgillus_brunneoviolaceus_GCF_003184695 genome --\n",
      "2020-02-10 10:59:14,147 - Phyluce - INFO - Reading Asprgillus_brunneoviolaceus_GCF_003184695 genome\n",
      "2020-02-10 10:59:15,033 - Phyluce - INFO - Asprgillus_brunneoviolaceus_GCF_003184695: 139 uces, 43 dupes, 96 non-dupes, 1 orient drop, 0 length drop, 95 written\n",
      "2020-02-10 10:59:15,034 - Phyluce - INFO - ------ Working on Asprgillus_caelatus_GCF_009193585 genome ------\n",
      "2020-02-10 10:59:15,035 - Phyluce - INFO - Reading Asprgillus_caelatus_GCF_009193585 genome\n",
      "2020-02-10 10:59:15,947 - Phyluce - INFO - Asprgillus_caelatus_GCF_009193585: 141 uces, 47 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-10 10:59:15,947 - Phyluce - INFO - ----- Working on Asprgillus_calidoustus_GCA_001511075 genome ----\n",
      "2020-02-10 10:59:15,948 - Phyluce - INFO - Reading Asprgillus_calidoustus_GCA_001511075 genome\n",
      "2020-02-10 10:59:16,840 - Phyluce - INFO - Asprgillus_calidoustus_GCA_001511075: 154 uces, 44 dupes, 110 non-dupes, 1 orient drop, 1 length drop, 108 written\n",
      "2020-02-10 10:59:16,840 - Phyluce - INFO - ----- Working on Asprgillus_campestris_GCF_002847485 genome -----\n",
      "2020-02-10 10:59:16,841 - Phyluce - INFO - Reading Asprgillus_campestris_GCF_002847485 genome\n",
      "2020-02-10 10:59:18,085 - Phyluce - INFO - Asprgillus_campestris_GCF_002847485: 149 uces, 49 dupes, 100 non-dupes, 2 orient drop, 1 length drop, 97 written\n",
      "2020-02-10 10:59:18,086 - Phyluce - INFO - ------ Working on Asprgillus_candidus_GCF_002847045 genome ------\n",
      "2020-02-10 10:59:18,087 - Phyluce - INFO - Reading Asprgillus_candidus_GCF_002847045 genome\n",
      "2020-02-10 10:59:18,938 - Phyluce - INFO - Asprgillus_candidus_GCF_002847045: 136 uces, 41 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-10 10:59:18,938 - Phyluce - INFO - ----- Working on Asprgillus_carbonarius_GCA_001990825 genome ----\n",
      "2020-02-10 10:59:18,939 - Phyluce - INFO - Reading Asprgillus_carbonarius_GCA_001990825 genome\n",
      "2020-02-10 10:59:19,944 - Phyluce - INFO - Asprgillus_carbonarius_GCA_001990825: 143 uces, 47 dupes, 96 non-dupes, 2 orient drop, 4 length drop, 90 written\n",
      "2020-02-10 10:59:19,945 - Phyluce - INFO - ------- Working on Asprgillus_cejpii_GCA_004769165 genome -------\n",
      "2020-02-10 10:59:19,946 - Phyluce - INFO - Reading Asprgillus_cejpii_GCA_004769165 genome\n",
      "2020-02-10 10:59:20,816 - Phyluce - INFO - Asprgillus_cejpii_GCA_004769165: 143 uces, 46 dupes, 97 non-dupes, 0 orient drop, 10 length drop, 87 written\n",
      "2020-02-10 10:59:20,816 - Phyluce - INFO - ----- Working on Asprgillus_chevalieri_GCA_001599875 genome -----\n",
      "2020-02-10 10:59:20,817 - Phyluce - INFO - Reading Asprgillus_chevalieri_GCA_001599875 genome\n",
      "2020-02-10 10:59:21,653 - Phyluce - INFO - Asprgillus_chevalieri_GCA_001599875: 133 uces, 50 dupes, 83 non-dupes, 0 orient drop, 0 length drop, 83 written\n",
      "2020-02-10 10:59:21,653 - Phyluce - INFO - ------ Working on Asprgillus_clavatus_GCF_000002715 genome ------\n",
      "2020-02-10 10:59:21,663 - Phyluce - INFO - Reading Asprgillus_clavatus_GCF_000002715 genome\n",
      "2020-02-10 10:59:22,775 - Phyluce - INFO - Asprgillus_clavatus_GCF_000002715: 152 uces, 55 dupes, 97 non-dupes, 0 orient drop, 1 length drop, 96 written\n",
      "2020-02-10 10:59:22,775 - Phyluce - INFO - ---- Working on Asprgillus_coremiiformis_GCA_009193565 genome ---\n",
      "2020-02-10 10:59:22,787 - Phyluce - INFO - Reading Asprgillus_coremiiformis_GCA_009193565 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 10:59:23,672 - Phyluce - INFO - Asprgillus_coremiiformis_GCA_009193565: 133 uces, 44 dupes, 89 non-dupes, 0 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 10:59:23,672 - Phyluce - INFO - --- Working on Asprgillus_costaricaensis_GCF_003184835 genome ---\n",
      "2020-02-10 10:59:23,674 - Phyluce - INFO - Reading Asprgillus_costaricaensis_GCF_003184835 genome\n",
      "2020-02-10 10:59:24,526 - Phyluce - INFO - Asprgillus_costaricaensis_GCF_003184835: 133 uces, 42 dupes, 91 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-10 10:59:24,526 - Phyluce - INFO - ------ Working on Asprgillus_cristatus_GCA_001693355 genome -----\n",
      "2020-02-10 10:59:24,527 - Phyluce - INFO - Reading Asprgillus_cristatus_GCA_001693355 genome\n",
      "2020-02-10 10:59:25,903 - Phyluce - INFO - Asprgillus_cristatus_GCA_001693355: 158 uces, 46 dupes, 112 non-dupes, 18 orient drop, 0 length drop, 94 written\n",
      "2020-02-10 10:59:25,903 - Phyluce - INFO - ----- Working on Asprgillus_ellipticus_GCA_003184645 genome -----\n",
      "2020-02-10 10:59:25,904 - Phyluce - INFO - Reading Asprgillus_ellipticus_GCA_003184645 genome\n",
      "2020-02-10 10:59:26,952 - Phyluce - INFO - Asprgillus_ellipticus_GCA_003184645: 135 uces, 45 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 10:59:26,953 - Phyluce - INFO - ---- Working on Asprgillus_eucalypticola_GCF_003184535 genome ---\n",
      "2020-02-10 10:59:26,968 - Phyluce - INFO - Reading Asprgillus_eucalypticola_GCF_003184535 genome\n",
      "2020-02-10 10:59:27,991 - Phyluce - INFO - Asprgillus_eucalypticola_GCF_003184535: 146 uces, 56 dupes, 90 non-dupes, 1 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 10:59:27,991 - Phyluce - INFO - ------ Working on Asprgillus_fijiensis_GCA_003184825 genome -----\n",
      "2020-02-10 10:59:27,992 - Phyluce - INFO - Reading Asprgillus_fijiensis_GCA_003184825 genome\n",
      "2020-02-10 10:59:28,882 - Phyluce - INFO - Asprgillus_fijiensis_GCA_003184825: 139 uces, 41 dupes, 98 non-dupes, 1 orient drop, 0 length drop, 97 written\n",
      "2020-02-10 10:59:28,883 - Phyluce - INFO - ------ Working on Asprgillus_fischeri_GCF_000149645 genome ------\n",
      "2020-02-10 10:59:28,884 - Phyluce - INFO - Reading Asprgillus_fischeri_GCF_000149645 genome\n",
      "2020-02-10 10:59:29,991 - Phyluce - INFO - Asprgillus_fischeri_GCF_000149645: 159 uces, 51 dupes, 108 non-dupes, 2 orient drop, 0 length drop, 106 written\n",
      "2020-02-10 10:59:29,992 - Phyluce - INFO - ------- Working on Asprgillus_flavus_GCF_000006275 genome -------\n",
      "2020-02-10 10:59:29,999 - Phyluce - INFO - Reading Asprgillus_flavus_GCF_000006275 genome\n",
      "2020-02-10 10:59:31,045 - Phyluce - INFO - Asprgillus_flavus_GCF_000006275: 148 uces, 47 dupes, 101 non-dupes, 3 orient drop, 1 length drop, 97 written\n",
      "2020-02-10 10:59:31,045 - Phyluce - INFO - ------ Working on Asprgillus_fumigatus_GCF_000002655 genome -----\n",
      "2020-02-10 10:59:31,054 - Phyluce - INFO - Reading Asprgillus_fumigatus_GCF_000002655 genome\n",
      "2020-02-10 10:59:31,938 - Phyluce - INFO - Asprgillus_fumigatus_GCF_000002655: 154 uces, 41 dupes, 113 non-dupes, 1 orient drop, 11 length drop, 101 written\n",
      "2020-02-10 10:59:31,938 - Phyluce - INFO - ------- Working on Asprgillus_glaucus_GCF_001890805 genome ------\n",
      "2020-02-10 10:59:31,940 - Phyluce - INFO - Reading Asprgillus_glaucus_GCF_001890805 genome\n",
      "2020-02-10 10:59:32,945 - Phyluce - INFO - Asprgillus_glaucus_GCF_001890805: 132 uces, 47 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-10 10:59:32,945 - Phyluce - INFO - ------ Working on Asprgillus_hancockii_GCA_001696595 genome -----\n",
      "2020-02-10 10:59:32,946 - Phyluce - INFO - Reading Asprgillus_hancockii_GCA_001696595 genome\n",
      "2020-02-10 10:59:33,971 - Phyluce - INFO - Asprgillus_hancockii_GCA_001696595: 150 uces, 55 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-10 10:59:33,971 - Phyluce - INFO - ---- Working on Asprgillus_heteromorphus_GCF_003184545 genome ---\n",
      "2020-02-10 10:59:33,973 - Phyluce - INFO - Reading Asprgillus_heteromorphus_GCF_003184545 genome\n",
      "2020-02-10 10:59:34,823 - Phyluce - INFO - Asprgillus_heteromorphus_GCF_003184545: 133 uces, 43 dupes, 90 non-dupes, 1 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 10:59:34,824 - Phyluce - INFO - ----- Working on Asprgillus_homomorphus_GCF_003184865 genome ----\n",
      "2020-02-10 10:59:34,849 - Phyluce - INFO - Reading Asprgillus_homomorphus_GCF_003184865 genome\n",
      "2020-02-10 10:59:35,718 - Phyluce - INFO - Asprgillus_homomorphus_GCF_003184865: 137 uces, 41 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 10:59:35,718 - Phyluce - INFO - ------ Working on Asprgillus_ibericus_GCF_003184845 genome ------\n",
      "2020-02-10 10:59:35,728 - Phyluce - INFO - Reading Asprgillus_ibericus_GCF_003184845 genome\n",
      "2020-02-10 10:59:36,599 - Phyluce - INFO - Asprgillus_ibericus_GCF_003184845: 136 uces, 38 dupes, 98 non-dupes, 1 orient drop, 0 length drop, 97 written\n",
      "2020-02-10 10:59:36,600 - Phyluce - INFO - ----- Working on Asprgillus_indologenus_GCA_003184685 genome ----\n",
      "2020-02-10 10:59:36,601 - Phyluce - INFO - Reading Asprgillus_indologenus_GCA_003184685 genome\n",
      "2020-02-10 10:59:37,534 - Phyluce - INFO - Asprgillus_indologenus_GCA_003184685: 137 uces, 44 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-10 10:59:37,535 - Phyluce - INFO - ------ Working on Asprgillus_japonicus_GCF_003184785 genome -----\n",
      "2020-02-10 10:59:37,536 - Phyluce - INFO - Reading Asprgillus_japonicus_GCF_003184785 genome\n",
      "2020-02-10 10:59:38,500 - Phyluce - INFO - Asprgillus_japonicus_GCF_003184785: 137 uces, 44 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-10 10:59:38,501 - Phyluce - INFO - ------ Working on Asprgillus_kawachii_GCA_000239835 genome ------\n",
      "2020-02-10 10:59:38,504 - Phyluce - INFO - Reading Asprgillus_kawachii_GCA_000239835 genome\n",
      "2020-02-10 10:59:39,468 - Phyluce - INFO - Asprgillus_kawachii_GCA_000239835: 158 uces, 51 dupes, 107 non-dupes, 1 orient drop, 0 length drop, 105 written\n",
      "2020-02-10 10:59:39,468 - Phyluce - INFO - ------ Working on Asprgillus_lentulus_GCA_001445615 genome ------\n",
      "2020-02-10 10:59:39,486 - Phyluce - INFO - Reading Asprgillus_lentulus_GCA_001445615 genome\n",
      "2020-02-10 10:59:40,459 - Phyluce - INFO - Asprgillus_lentulus_GCA_001445615: 156 uces, 46 dupes, 110 non-dupes, 1 orient drop, 10 length drop, 99 written\n",
      "2020-02-10 10:59:40,460 - Phyluce - INFO - ------- Working on Asprgillus_leporis_GCA_009176345 genome ------\n",
      "2020-02-10 10:59:40,461 - Phyluce - INFO - Reading Asprgillus_leporis_GCA_009176345 genome\n",
      "2020-02-10 10:59:41,303 - Phyluce - INFO - Asprgillus_leporis_GCA_009176345: 140 uces, 46 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-10 10:59:41,304 - Phyluce - INFO - ----- Working on Asprgillus_luchuensis_GCA_001890685 genome -----\n",
      "2020-02-10 10:59:41,316 - Phyluce - INFO - Reading Asprgillus_luchuensis_GCA_001890685 genome\n",
      "2020-02-10 10:59:42,177 - Phyluce - INFO - Asprgillus_luchuensis_GCA_001890685: 139 uces, 43 dupes, 96 non-dupes, 1 orient drop, 0 length drop, 95 written\n",
      "2020-02-10 10:59:42,178 - Phyluce - INFO - -- Working on Asprgillus_minisclerotigenes_GCA_009176455 genome -\n",
      "2020-02-10 10:59:42,179 - Phyluce - INFO - Reading Asprgillus_minisclerotigenes_GCA_009176455 genome\n",
      "2020-02-10 10:59:43,045 - Phyluce - INFO - Asprgillus_minisclerotigenes_GCA_009176455: 138 uces, 48 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 10:59:43,045 - Phyluce - INFO - ----- Working on Asprgillus_mulundensis_GCF_003369625 genome ----\n",
      "2020-02-10 10:59:43,055 - Phyluce - INFO - Reading Asprgillus_mulundensis_GCF_003369625 genome\n",
      "2020-02-10 10:59:44,006 - Phyluce - INFO - Asprgillus_mulundensis_GCF_003369625: 137 uces, 47 dupes, 90 non-dupes, 1 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 10:59:44,007 - Phyluce - INFO - ---- Working on Asprgillus_neoellipticus_GCA_003116565 genome ---\n",
      "2020-02-10 10:59:44,008 - Phyluce - INFO - Reading Asprgillus_neoellipticus_GCA_003116565 genome\n",
      "2020-02-10 10:59:44,964 - Phyluce - INFO - Asprgillus_neoellipticus_GCA_003116565: 163 uces, 50 dupes, 113 non-dupes, 1 orient drop, 0 length drop, 112 written\n",
      "2020-02-10 10:59:44,964 - Phyluce - INFO - ------ Working on Asprgillus_neoniger_GCF_003184625 genome ------\n",
      "2020-02-10 10:59:44,972 - Phyluce - INFO - Reading Asprgillus_neoniger_GCF_003184625 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 10:59:45,836 - Phyluce - INFO - Asprgillus_neoniger_GCF_003184625: 136 uces, 40 dupes, 96 non-dupes, 1 orient drop, 0 length drop, 95 written\n",
      "2020-02-10 10:59:45,836 - Phyluce - INFO - ------ Working on Asprgillus_nidulans_GCF_000149205 genome ------\n",
      "2020-02-10 10:59:45,847 - Phyluce - INFO - Reading Asprgillus_nidulans_GCF_000149205 genome\n",
      "2020-02-10 10:59:46,819 - Phyluce - INFO - Asprgillus_nidulans_GCF_000149205: 149 uces, 52 dupes, 97 non-dupes, 1 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 10:59:46,820 - Phyluce - INFO - -------- Working on Asprgillus_niger_GCF_000002855 genome -------\n",
      "2020-02-10 10:59:46,828 - Phyluce - INFO - Reading Asprgillus_niger_GCF_000002855 genome\n",
      "2020-02-10 10:59:47,904 - Phyluce - INFO - Asprgillus_niger_GCF_000002855: 149 uces, 39 dupes, 110 non-dupes, 1 orient drop, 17 length drop, 92 written\n",
      "2020-02-10 10:59:47,904 - Phyluce - INFO - ------- Working on Asprgillus_nomius_GCF_001204775 genome -------\n",
      "2020-02-10 10:59:47,905 - Phyluce - INFO - Reading Asprgillus_nomius_GCF_001204775 genome\n",
      "2020-02-10 10:59:48,900 - Phyluce - INFO - Asprgillus_nomius_GCF_001204775: 162 uces, 52 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-10 10:59:48,900 - Phyluce - INFO - ---- Working on Asprgillus_novofumigatus_GCF_002847465 genome ---\n",
      "2020-02-10 10:59:48,909 - Phyluce - INFO - Reading Asprgillus_novofumigatus_GCF_002847465 genome\n",
      "2020-02-10 10:59:50,097 - Phyluce - INFO - Asprgillus_novofumigatus_GCF_002847465: 149 uces, 59 dupes, 90 non-dupes, 1 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 10:59:50,097 - Phyluce - INFO - --- Working on Asprgillus_novoparasiticus_GCA_009176405 genome --\n",
      "2020-02-10 10:59:50,099 - Phyluce - INFO - Reading Asprgillus_novoparasiticus_GCA_009176405 genome\n",
      "2020-02-10 10:59:50,979 - Phyluce - INFO - Asprgillus_novoparasiticus_GCA_009176405: 144 uces, 48 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 10:59:50,979 - Phyluce - INFO - --- Working on Asprgillus_ochraceoroseus_GCA_002846915 genome ---\n",
      "2020-02-10 10:59:50,992 - Phyluce - INFO - Reading Asprgillus_ochraceoroseus_GCA_002846915 genome\n",
      "2020-02-10 10:59:52,480 - Phyluce - INFO - Asprgillus_ochraceoroseus_GCA_002846915: 151 uces, 55 dupes, 96 non-dupes, 0 orient drop, 1 length drop, 95 written\n",
      "2020-02-10 10:59:52,480 - Phyluce - INFO - ------ Working on Asprgillus_ochraceus_GCA_004849945 genome -----\n",
      "2020-02-10 10:59:52,481 - Phyluce - INFO - Reading Asprgillus_ochraceus_GCA_004849945 genome\n",
      "2020-02-10 10:59:53,397 - Phyluce - INFO - Asprgillus_ochraceus_GCA_004849945: 159 uces, 41 dupes, 118 non-dupes, 2 orient drop, 0 length drop, 116 written\n",
      "2020-02-10 10:59:53,398 - Phyluce - INFO - ----- Working on Asprgillus_olivimuriae_GCA_003719415 genome ----\n",
      "2020-02-10 10:59:53,399 - Phyluce - INFO - Reading Asprgillus_olivimuriae_GCA_003719415 genome\n",
      "2020-02-10 10:59:54,291 - Phyluce - INFO - Asprgillus_olivimuriae_GCA_003719415: 150 uces, 43 dupes, 107 non-dupes, 0 orient drop, 0 length drop, 105 written\n",
      "2020-02-10 10:59:54,291 - Phyluce - INFO - ------- Working on Asprgillus_oryzae_GCF_000184455 genome -------\n",
      "2020-02-10 10:59:54,311 - Phyluce - INFO - Reading Asprgillus_oryzae_GCF_000184455 genome\n",
      "2020-02-10 10:59:55,293 - Phyluce - INFO - Asprgillus_oryzae_GCF_000184455: 162 uces, 41 dupes, 121 non-dupes, 7 orient drop, 8 length drop, 106 written\n",
      "2020-02-10 10:59:55,293 - Phyluce - INFO - ----- Working on Asprgillus_parasiticus_GCA_009176385 genome ----\n",
      "2020-02-10 10:59:55,294 - Phyluce - INFO - Reading Asprgillus_parasiticus_GCA_009176385 genome\n",
      "2020-02-10 10:59:56,158 - Phyluce - INFO - Asprgillus_parasiticus_GCA_009176385: 139 uces, 49 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 10:59:56,158 - Phyluce - INFO - ------- Working on Asprgillus_persii_GCA_002215965 genome -------\n",
      "2020-02-10 10:59:56,159 - Phyluce - INFO - Reading Asprgillus_persii_GCA_002215965 genome\n",
      "2020-02-10 10:59:57,370 - Phyluce - INFO - Asprgillus_persii_GCA_002215965: 154 uces, 43 dupes, 111 non-dupes, 1 orient drop, 18 length drop, 92 written\n",
      "2020-02-10 10:59:57,370 - Phyluce - INFO - ------ Working on Asprgillus_phoenicis_GCA_003344505 genome -----\n",
      "2020-02-10 10:59:57,384 - Phyluce - INFO - Reading Asprgillus_phoenicis_GCA_003344505 genome\n",
      "2020-02-10 10:59:58,249 - Phyluce - INFO - Asprgillus_phoenicis_GCA_003344505: 133 uces, 45 dupes, 88 non-dupes, 0 orient drop, 0 length drop, 88 written\n",
      "2020-02-10 10:59:58,249 - Phyluce - INFO - ------- Working on Asprgillus_piperis_GCF_003184755 genome ------\n",
      "2020-02-10 10:59:58,258 - Phyluce - INFO - Reading Asprgillus_piperis_GCF_003184755 genome\n",
      "2020-02-10 10:59:59,099 - Phyluce - INFO - Asprgillus_piperis_GCF_003184755: 134 uces, 40 dupes, 94 non-dupes, 2 orient drop, 0 length drop, 92 written\n",
      "2020-02-10 10:59:59,099 - Phyluce - INFO - --- Working on Asprgillus_pseudocaelatus_GCA_009193665 genome ---\n",
      "2020-02-10 10:59:59,101 - Phyluce - INFO - Reading Asprgillus_pseudocaelatus_GCA_009193665 genome\n",
      "2020-02-10 11:00:00,106 - Phyluce - INFO - Asprgillus_pseudocaelatus_GCA_009193665: 138 uces, 48 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 11:00:00,107 - Phyluce - INFO - ---- Working on Asprgillus_pseudotamarii_GCF_009193445 genome ---\n",
      "2020-02-10 11:00:00,108 - Phyluce - INFO - Reading Asprgillus_pseudotamarii_GCF_009193445 genome\n",
      "2020-02-10 11:00:01,076 - Phyluce - INFO - Asprgillus_pseudotamarii_GCF_009193445: 141 uces, 51 dupes, 90 non-dupes, 0 orient drop, 1 length drop, 89 written\n",
      "2020-02-10 11:00:01,076 - Phyluce - INFO - ---- Working on Asprgillus_pseudoterreus_GCA_002927005 genome ---\n",
      "2020-02-10 11:00:01,077 - Phyluce - INFO - Reading Asprgillus_pseudoterreus_GCA_002927005 genome\n",
      "2020-02-10 11:00:01,955 - Phyluce - INFO - Asprgillus_pseudoterreus_GCA_002927005: 147 uces, 44 dupes, 103 non-dupes, 0 orient drop, 0 length drop, 103 written\n",
      "2020-02-10 11:00:01,955 - Phyluce - INFO - ------ Working on Asprgillus_rambellii_GCA_000986645 genome -----\n",
      "2020-02-10 11:00:01,964 - Phyluce - INFO - Reading Asprgillus_rambellii_GCA_000986645 genome\n",
      "2020-02-10 11:00:02,873 - Phyluce - INFO - Asprgillus_rambellii_GCA_000986645: 160 uces, 42 dupes, 118 non-dupes, 0 orient drop, 0 length drop, 118 written\n",
      "2020-02-10 11:00:02,874 - Phyluce - INFO - -------- Working on Asprgillus_ruber_GCA_000600275 genome -------\n",
      "2020-02-10 11:00:02,875 - Phyluce - INFO - Reading Asprgillus_ruber_GCA_000600275 genome\n",
      "2020-02-10 11:00:03,780 - Phyluce - INFO - Asprgillus_ruber_GCA_000600275: 144 uces, 45 dupes, 99 non-dupes, 0 orient drop, 0 length drop, 99 written\n",
      "2020-02-10 11:00:03,780 - Phyluce - INFO - --- Working on Asprgillus_saccharolyticus_GCF_003184585 genome --\n",
      "2020-02-10 11:00:03,781 - Phyluce - INFO - Reading Asprgillus_saccharolyticus_GCF_003184585 genome\n",
      "2020-02-10 11:00:04,630 - Phyluce - INFO - Asprgillus_saccharolyticus_GCF_003184585: 139 uces, 40 dupes, 99 non-dupes, 0 orient drop, 1 length drop, 98 written\n",
      "2020-02-10 11:00:04,631 - Phyluce - INFO - ---- Working on Asprgillus_sclerotialis_GCA_003589665 genome ----\n",
      "2020-02-10 11:00:04,632 - Phyluce - INFO - Reading Asprgillus_sclerotialis_GCA_003589665 genome\n",
      "2020-02-10 11:00:05,785 - Phyluce - INFO - Asprgillus_sclerotialis_GCA_003589665: 150 uces, 74 dupes, 76 non-dupes, 0 orient drop, 0 length drop, 76 written\n",
      "2020-02-10 11:00:05,785 - Phyluce - INFO -  Working on Asprgillus_sclerotiicarbonarius_GCA_003184635 genome \n",
      "2020-02-10 11:00:05,786 - Phyluce - INFO - Reading Asprgillus_sclerotiicarbonarius_GCA_003184635 genome\n",
      "2020-02-10 11:00:06,780 - Phyluce - INFO - Asprgillus_sclerotiicarbonarius_GCA_003184635: 134 uces, 43 dupes, 91 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-10 11:00:06,780 - Phyluce - INFO - --- Working on Asprgillus_sclerotioniger_GCF_003184525 genome ---\n",
      "2020-02-10 11:00:06,781 - Phyluce - INFO - Reading Asprgillus_sclerotioniger_GCF_003184525 genome\n",
      "2020-02-10 11:00:07,891 - Phyluce - INFO - Asprgillus_sclerotioniger_GCF_003184525: 150 uces, 52 dupes, 98 non-dupes, 9 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 11:00:07,891 - Phyluce - INFO - ---- Working on Asprgillus_sclerotiorum_GCA_000530345 genome ----\n",
      "2020-02-10 11:00:07,902 - Phyluce - INFO - Reading Asprgillus_sclerotiorum_GCA_000530345 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 11:00:08,870 - Phyluce - INFO - Asprgillus_sclerotiorum_GCA_000530345: 160 uces, 45 dupes, 115 non-dupes, 0 orient drop, 0 length drop, 115 written\n",
      "2020-02-10 11:00:08,870 - Phyluce - INFO - ------- Working on Asprgillus_sergii_GCA_009193525 genome -------\n",
      "2020-02-10 11:00:08,877 - Phyluce - INFO - Reading Asprgillus_sergii_GCA_009193525 genome\n",
      "2020-02-10 11:00:09,930 - Phyluce - INFO - Asprgillus_sergii_GCA_009193525: 137 uces, 49 dupes, 88 non-dupes, 0 orient drop, 0 length drop, 88 written\n",
      "2020-02-10 11:00:09,930 - Phyluce - INFO - -------- Working on Asprgillus_sojae_GCA_008274985 genome -------\n",
      "2020-02-10 11:00:09,944 - Phyluce - INFO - Reading Asprgillus_sojae_GCA_008274985 genome\n",
      "2020-02-10 11:00:10,839 - Phyluce - INFO - Asprgillus_sojae_GCA_008274985: 137 uces, 45 dupes, 92 non-dupes, 5 orient drop, 1 length drop, 86 written\n",
      "2020-02-10 11:00:10,839 - Phyluce - INFO - --------- Working on Asprgillus_sp_GCA_001044295 genome ---------\n",
      "2020-02-10 11:00:10,840 - Phyluce - INFO - Reading Asprgillus_sp_GCA_001044295 genome\n",
      "2020-02-10 11:00:11,830 - Phyluce - INFO - Asprgillus_sp_GCA_001044295: 157 uces, 43 dupes, 114 non-dupes, 0 orient drop, 0 length drop, 114 written\n",
      "2020-02-10 11:00:11,830 - Phyluce - INFO - ----- Working on Asprgillus_spnulosprus_GCA_003574815 genome ----\n",
      "2020-02-10 11:00:11,831 - Phyluce - INFO - Reading Asprgillus_spnulosprus_GCA_003574815 genome\n",
      "2020-02-10 11:00:12,963 - Phyluce - INFO - Asprgillus_spnulosprus_GCA_003574815: 154 uces, 57 dupes, 97 non-dupes, 1 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 11:00:12,963 - Phyluce - INFO - ------- Working on Asprgillus_steynii_GCF_002849105 genome ------\n",
      "2020-02-10 11:00:12,975 - Phyluce - INFO - Reading Asprgillus_steynii_GCF_002849105 genome\n",
      "2020-02-10 11:00:13,835 - Phyluce - INFO - Asprgillus_steynii_GCF_002849105: 137 uces, 35 dupes, 102 non-dupes, 2 orient drop, 2 length drop, 98 written\n",
      "2020-02-10 11:00:13,835 - Phyluce - INFO - ------- Working on Asprgillus_sydowii_GCA_001890705 genome ------\n",
      "2020-02-10 11:00:13,845 - Phyluce - INFO - Reading Asprgillus_sydowii_GCA_001890705 genome\n",
      "2020-02-10 11:00:14,763 - Phyluce - INFO - Asprgillus_sydowii_GCA_001890705: 134 uces, 43 dupes, 91 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-10 11:00:14,763 - Phyluce - INFO - ---- Working on Asprgillus_taichungensis_GCA_002850765 genome ---\n",
      "2020-02-10 11:00:14,770 - Phyluce - INFO - Reading Asprgillus_taichungensis_GCA_002850765 genome\n",
      "2020-02-10 11:00:15,625 - Phyluce - INFO - Asprgillus_taichungensis_GCA_002850765: 134 uces, 40 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-10 11:00:15,626 - Phyluce - INFO - ------- Working on Asprgillus_tamarii_GCA_009193485 genome ------\n",
      "2020-02-10 11:00:15,650 - Phyluce - INFO - Reading Asprgillus_tamarii_GCA_009193485 genome\n",
      "2020-02-10 11:00:16,643 - Phyluce - INFO - Asprgillus_tamarii_GCA_009193485: 137 uces, 46 dupes, 91 non-dupes, 0 orient drop, 1 length drop, 90 written\n",
      "2020-02-10 11:00:16,643 - Phyluce - INFO - ------- Working on Asprgillus_tanneri_GCA_003426965 genome ------\n",
      "2020-02-10 11:00:16,644 - Phyluce - INFO - Reading Asprgillus_tanneri_GCA_003426965 genome\n",
      "2020-02-10 11:00:17,622 - Phyluce - INFO - Asprgillus_tanneri_GCA_003426965: 150 uces, 45 dupes, 105 non-dupes, 0 orient drop, 7 length drop, 98 written\n",
      "2020-02-10 11:00:17,622 - Phyluce - INFO - ------- Working on Asprgillus_terreus_GCF_000149615 genome ------\n",
      "2020-02-10 11:00:17,629 - Phyluce - INFO - Reading Asprgillus_terreus_GCF_000149615 genome\n",
      "2020-02-10 11:00:18,557 - Phyluce - INFO - Asprgillus_terreus_GCF_000149615: 163 uces, 44 dupes, 119 non-dupes, 0 orient drop, 8 length drop, 111 written\n",
      "2020-02-10 11:00:18,557 - Phyluce - INFO - ---- Working on Asprgillus_thermomutatus_GCF_002237265 genome ---\n",
      "2020-02-10 11:00:18,585 - Phyluce - INFO - Reading Asprgillus_thermomutatus_GCF_002237265 genome\n",
      "2020-02-10 11:00:19,612 - Phyluce - INFO - Asprgillus_thermomutatus_GCF_002237265: 152 uces, 49 dupes, 103 non-dupes, 12 orient drop, 0 length drop, 91 written\n",
      "2020-02-10 11:00:19,613 - Phyluce - INFO - -- Working on Asprgillus_transmontanensis_GCA_009193505 genome --\n",
      "2020-02-10 11:00:19,621 - Phyluce - INFO - Reading Asprgillus_transmontanensis_GCA_009193505 genome\n",
      "2020-02-10 11:00:20,614 - Phyluce - INFO - Asprgillus_transmontanensis_GCA_009193505: 141 uces, 51 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 11:00:20,614 - Phyluce - INFO - ------- Working on Asprgillus_tritici_GCA_009812425 genome ------\n",
      "2020-02-10 11:00:20,615 - Phyluce - INFO - Reading Asprgillus_tritici_GCA_009812425 genome\n",
      "2020-02-10 11:00:21,513 - Phyluce - INFO - Asprgillus_tritici_GCA_009812425: 148 uces, 44 dupes, 104 non-dupes, 0 orient drop, 0 length drop, 104 written\n",
      "2020-02-10 11:00:21,513 - Phyluce - INFO - ----- Working on Asprgillus_tubingensis_GCA_001890745 genome ----\n",
      "2020-02-10 11:00:21,515 - Phyluce - INFO - Reading Asprgillus_tubingensis_GCA_001890745 genome\n",
      "2020-02-10 11:00:22,371 - Phyluce - INFO - Asprgillus_tubingensis_GCA_001890745: 134 uces, 42 dupes, 92 non-dupes, 1 orient drop, 0 length drop, 91 written\n",
      "2020-02-10 11:00:22,371 - Phyluce - INFO - ------ Working on Asprgillus_turcosus_GCA_002234965 genome ------\n",
      "2020-02-10 11:00:22,380 - Phyluce - INFO - Reading Asprgillus_turcosus_GCA_002234965 genome\n",
      "2020-02-10 11:00:23,341 - Phyluce - INFO - Asprgillus_turcosus_GCA_002234965: 156 uces, 49 dupes, 107 non-dupes, 0 orient drop, 0 length drop, 107 written\n",
      "2020-02-10 11:00:23,342 - Phyluce - INFO - ------ Working on Asprgillus_udagawae_GCA_001078395 genome ------\n",
      "2020-02-10 11:00:23,343 - Phyluce - INFO - Reading Asprgillus_udagawae_GCA_001078395 genome\n",
      "2020-02-10 11:00:24,369 - Phyluce - INFO - Asprgillus_udagawae_GCA_001078395: 158 uces, 46 dupes, 112 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-10 11:00:24,370 - Phyluce - INFO - ------- Working on Asprgillus_unguis_GCA_003324175 genome -------\n",
      "2020-02-10 11:00:24,372 - Phyluce - INFO - Reading Asprgillus_unguis_GCA_003324175 genome\n",
      "2020-02-10 11:00:25,801 - Phyluce - INFO - Asprgillus_unguis_GCA_003324175: 163 uces, 130 dupes, 33 non-dupes, 0 orient drop, 0 length drop, 33 written\n",
      "2020-02-10 11:00:25,801 - Phyluce - INFO - -------- Working on Asprgillus_ustus_GCA_000812125 genome -------\n",
      "2020-02-10 11:00:25,802 - Phyluce - INFO - Reading Asprgillus_ustus_GCA_000812125 genome\n",
      "2020-02-10 11:00:26,671 - Phyluce - INFO - Asprgillus_ustus_GCA_000812125: 150 uces, 45 dupes, 105 non-dupes, 0 orient drop, 0 length drop, 105 written\n",
      "2020-02-10 11:00:26,671 - Phyluce - INFO - ------- Working on Asprgillus_uvarum_GCF_003184745 genome -------\n",
      "2020-02-10 11:00:26,681 - Phyluce - INFO - Reading Asprgillus_uvarum_GCF_003184745 genome\n",
      "2020-02-10 11:00:27,550 - Phyluce - INFO - Asprgillus_uvarum_GCF_003184745: 138 uces, 45 dupes, 93 non-dupes, 1 orient drop, 0 length drop, 92 written\n",
      "2020-02-10 11:00:27,550 - Phyluce - INFO - ------ Working on Asprgillus_vadensis_GCF_003184925 genome ------\n",
      "2020-02-10 11:00:27,560 - Phyluce - INFO - Reading Asprgillus_vadensis_GCF_003184925 genome\n",
      "2020-02-10 11:00:28,476 - Phyluce - INFO - Asprgillus_vadensis_GCF_003184925: 133 uces, 43 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 11:00:28,476 - Phyluce - INFO - ----- Working on Asprgillus_versicolor_GCA_001890125 genome -----\n",
      "2020-02-10 11:00:28,483 - Phyluce - INFO - Reading Asprgillus_versicolor_GCA_001890125 genome\n",
      "2020-02-10 11:00:29,328 - Phyluce - INFO - Asprgillus_versicolor_GCA_001890125: 133 uces, 43 dupes, 90 non-dupes, 1 orient drop, 0 length drop, 89 written\n",
      "2020-02-10 11:00:29,329 - Phyluce - INFO - --- Working on Asprgillus_violaceofuscus_GCA_003184705 genome ---\n",
      "2020-02-10 11:00:29,330 - Phyluce - INFO - Reading Asprgillus_violaceofuscus_GCA_003184705 genome\n",
      "2020-02-10 11:00:30,304 - Phyluce - INFO - Asprgillus_violaceofuscus_GCA_003184705: 138 uces, 45 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-10 11:00:30,305 - Phyluce - INFO - ---- Working on Asprgillus_viridinutans_GCA_004368095 genome ----\n",
      "2020-02-10 11:00:30,306 - Phyluce - INFO - Reading Asprgillus_viridinutans_GCA_004368095 genome\n",
      "2020-02-10 11:00:31,310 - Phyluce - INFO - Asprgillus_viridinutans_GCA_004368095: 163 uces, 49 dupes, 114 non-dupes, 1 orient drop, 0 length drop, 113 written\n",
      "2020-02-10 11:00:31,310 - Phyluce - INFO - ---- Working on Asprgillus_welwitschiae_GCF_003344945 genome ----\n",
      "2020-02-10 11:00:31,316 - Phyluce - INFO - Reading Asprgillus_welwitschiae_GCF_003344945 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 11:00:32,168 - Phyluce - INFO - Asprgillus_welwitschiae_GCF_003344945: 135 uces, 43 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-10 11:00:32,169 - Phyluce - INFO - ------- Working on Asprgillus_wentii_GCA_001890725 genome -------\n",
      "2020-02-10 11:00:32,177 - Phyluce - INFO - Reading Asprgillus_wentii_GCA_001890725 genome\n",
      "2020-02-10 11:00:33,049 - Phyluce - INFO - Asprgillus_wentii_GCA_001890725: 134 uces, 49 dupes, 85 non-dupes, 2 orient drop, 0 length drop, 83 written\n",
      "2020-02-10 11:00:33,050 - Phyluce - INFO - ---- Working on Asprgillus_westerdijkiae_GCA_001307345 genome ---\n",
      "2020-02-10 11:00:33,059 - Phyluce - INFO - Reading Asprgillus_westerdijkiae_GCA_001307345 genome\n",
      "2020-02-10 11:00:33,989 - Phyluce - INFO - Asprgillus_westerdijkiae_GCA_001307345: 145 uces, 42 dupes, 103 non-dupes, 0 orient drop, 0 length drop, 103 written\n",
      "2020-02-10 11:00:33,990 - Phyluce - INFO - --- Working on Basipetospra_chlamydospra_GCA_001599675 genome ---\n",
      "2020-02-10 11:00:33,991 - Phyluce - INFO - Reading Basipetospra_chlamydospra_GCA_001599675 genome\n",
      "2020-02-10 11:00:34,694 - Phyluce - INFO - Basipetospra_chlamydospra_GCA_001599675: 127 uces, 28 dupes, 99 non-dupes, 1 orient drop, 0 length drop, 98 written\n",
      "2020-02-10 11:00:34,695 - Phyluce - INFO - ------- Working on Byssochlamys_nivea_GCA_003116535 genome ------\n",
      "2020-02-10 11:00:34,706 - Phyluce - INFO - Reading Byssochlamys_nivea_GCA_003116535 genome\n",
      "2020-02-10 11:00:35,698 - Phyluce - INFO - Byssochlamys_nivea_GCA_003116535: 158 uces, 45 dupes, 113 non-dupes, 0 orient drop, 0 length drop, 113 written\n",
      "2020-02-10 11:00:35,698 - Phyluce - INFO - -------- Working on Byssochlamys_sp_GCA_002914405 genome --------\n",
      "2020-02-10 11:00:35,706 - Phyluce - INFO - Reading Byssochlamys_sp_GCA_002914405 genome\n",
      "2020-02-10 11:00:36,557 - Phyluce - INFO - Byssochlamys_sp_GCA_002914405: 150 uces, 41 dupes, 109 non-dupes, 1 orient drop, 6 length drop, 102 written\n",
      "2020-02-10 11:00:36,557 - Phyluce - INFO - ---- Working on Byssochlamys_spctabilis_GCF_004022145 genome ----\n",
      "2020-02-10 11:00:36,566 - Phyluce - INFO - Reading Byssochlamys_spctabilis_GCF_004022145 genome\n",
      "2020-02-10 11:00:37,568 - Phyluce - INFO - Byssochlamys_spctabilis_GCF_004022145: 150 uces, 62 dupes, 88 non-dupes, 0 orient drop, 2 length drop, 86 written\n",
      "2020-02-10 11:00:37,568 - Phyluce - INFO - ----- Working on Elaphomyces_granulatus_GCA_002240705 genome ----\n",
      "2020-02-10 11:00:37,569 - Phyluce - INFO - Reading Elaphomyces_granulatus_GCA_002240705 genome\n",
      "2020-02-10 11:00:38,490 - Phyluce - INFO - Elaphomyces_granulatus_GCA_002240705: 139 uces, 33 dupes, 106 non-dupes, 0 orient drop, 0 length drop, 106 written\n",
      "2020-02-10 11:00:38,490 - Phyluce - INFO - ------- Working on Monascus_purpureus_GCA_003184285 genome ------\n",
      "2020-02-10 11:00:38,492 - Phyluce - INFO - Reading Monascus_purpureus_GCA_003184285 genome\n",
      "2020-02-10 11:00:39,293 - Phyluce - INFO - Monascus_purpureus_GCA_003184285: 151 uces, 35 dupes, 116 non-dupes, 0 orient drop, 2 length drop, 114 written\n",
      "2020-02-10 11:00:39,294 - Phyluce - INFO - --------- Working on Monascus_ruber_GCA_002976275 genome --------\n",
      "2020-02-10 11:00:39,295 - Phyluce - INFO - Reading Monascus_ruber_GCA_002976275 genome\n",
      "2020-02-10 11:00:40,407 - Phyluce - INFO - Monascus_ruber_GCA_002976275: 150 uces, 41 dupes, 109 non-dupes, 1 orient drop, 9 length drop, 99 written\n",
      "2020-02-10 11:00:40,407 - Phyluce - INFO - ----- Working on Penicilliopsis_zonata_GCF_001890105 genome -----\n",
      "2020-02-10 11:00:40,409 - Phyluce - INFO - Reading Penicilliopsis_zonata_GCF_001890105 genome\n",
      "2020-02-10 11:00:41,163 - Phyluce - INFO - Penicilliopsis_zonata_GCF_001890105: 146 uces, 38 dupes, 108 non-dupes, 1 orient drop, 0 length drop, 107 written\n",
      "2020-02-10 11:00:41,164 - Phyluce - INFO - ---- Working on Penicillium_antarcticum_GCA_002072345 genome ----\n",
      "2020-02-10 11:00:41,170 - Phyluce - INFO - Reading Penicillium_antarcticum_GCA_002072345 genome\n",
      "2020-02-10 11:00:42,247 - Phyluce - INFO - Penicillium_antarcticum_GCA_002072345: 161 uces, 45 dupes, 116 non-dupes, 0 orient drop, 0 length drop, 116 written\n",
      "2020-02-10 11:00:42,248 - Phyluce - INFO - ----- Working on Penicillium_arizonense_GCF_001773325 genome ----\n",
      "2020-02-10 11:00:42,249 - Phyluce - INFO - Reading Penicillium_arizonense_GCF_001773325 genome\n",
      "2020-02-10 11:00:43,229 - Phyluce - INFO - Penicillium_arizonense_GCF_001773325: 162 uces, 53 dupes, 109 non-dupes, 0 orient drop, 2 length drop, 107 written\n",
      "2020-02-10 11:00:43,230 - Phyluce - INFO - ------ Working on Penicillium_biforme_GCA_000577785 genome ------\n",
      "2020-02-10 11:00:43,231 - Phyluce - INFO - Reading Penicillium_biforme_GCA_000577785 genome\n",
      "2020-02-10 11:00:44,066 - Phyluce - INFO - Penicillium_biforme_GCA_000577785: 147 uces, 47 dupes, 100 non-dupes, 0 orient drop, 0 length drop, 100 written\n",
      "2020-02-10 11:00:44,066 - Phyluce - INFO - ---- Working on Penicillium_brasilianum_GCA_001048715 genome ----\n",
      "2020-02-10 11:00:44,067 - Phyluce - INFO - Reading Penicillium_brasilianum_GCA_001048715 genome\n",
      "2020-02-10 11:00:45,031 - Phyluce - INFO - Penicillium_brasilianum_GCA_001048715: 161 uces, 53 dupes, 108 non-dupes, 0 orient drop, 2 length drop, 106 written\n",
      "2020-02-10 11:00:45,031 - Phyluce - INFO - ----- Working on Penicillium_camemberti_GCA_000513335 genome ----\n",
      "2020-02-10 11:00:45,032 - Phyluce - INFO - Reading Penicillium_camemberti_GCA_000513335 genome\n",
      "2020-02-10 11:00:45,979 - Phyluce - INFO - Penicillium_camemberti_GCA_000513335: 153 uces, 48 dupes, 105 non-dupes, 0 orient drop, 0 length drop, 105 written\n",
      "2020-02-10 11:00:45,979 - Phyluce - INFO - ----- Working on Penicillium_capsulatum_GCA_000943775 genome ----\n",
      "2020-02-10 11:00:45,981 - Phyluce - INFO - Reading Penicillium_capsulatum_GCA_000943775 genome\n",
      "2020-02-10 11:00:47,116 - Phyluce - INFO - Penicillium_capsulatum_GCA_000943775: 163 uces, 56 dupes, 107 non-dupes, 2 orient drop, 8 length drop, 97 written\n",
      "2020-02-10 11:00:47,117 - Phyluce - INFO - ------ Working on Penicillium_carneum_GCA_000577495 genome ------\n",
      "2020-02-10 11:00:47,118 - Phyluce - INFO - Reading Penicillium_carneum_GCA_000577495 genome\n",
      "2020-02-10 11:00:48,088 - Phyluce - INFO - Penicillium_carneum_GCA_000577495: 156 uces, 48 dupes, 108 non-dupes, 0 orient drop, 0 length drop, 108 written\n",
      "2020-02-10 11:00:48,088 - Phyluce - INFO - ---- Working on Penicillium_chrysogenum_GCA_000710275 genome ----\n",
      "2020-02-10 11:00:48,089 - Phyluce - INFO - Reading Penicillium_chrysogenum_GCA_000710275 genome\n",
      "2020-02-10 11:00:49,073 - Phyluce - INFO - Penicillium_chrysogenum_GCA_000710275: 163 uces, 38 dupes, 125 non-dupes, 5 orient drop, 19 length drop, 101 written\n",
      "2020-02-10 11:00:49,074 - Phyluce - INFO - ------ Working on Penicillium_citrinum_GCA_001950535 genome -----\n",
      "2020-02-10 11:00:49,087 - Phyluce - INFO - Reading Penicillium_citrinum_GCA_001950535 genome\n",
      "2020-02-10 11:00:49,990 - Phyluce - INFO - Penicillium_citrinum_GCA_001950535: 140 uces, 44 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 11:00:49,990 - Phyluce - INFO - ---- Working on Penicillium_coprophilum_GCA_002072405 genome ----\n",
      "2020-02-10 11:00:50,000 - Phyluce - INFO - Reading Penicillium_coprophilum_GCA_002072405 genome\n",
      "2020-02-10 11:00:50,926 - Phyluce - INFO - Penicillium_coprophilum_GCA_002072405: 160 uces, 46 dupes, 114 non-dupes, 2 orient drop, 2 length drop, 110 written\n",
      "2020-02-10 11:00:50,926 - Phyluce - INFO - ----- Working on Penicillium_decumbens_GCA_002072245 genome -----\n",
      "2020-02-10 11:00:50,927 - Phyluce - INFO - Reading Penicillium_decumbens_GCA_002072245 genome\n",
      "2020-02-10 11:00:51,794 - Phyluce - INFO - Penicillium_decumbens_GCA_002072245: 159 uces, 42 dupes, 117 non-dupes, 2 orient drop, 0 length drop, 115 written\n",
      "2020-02-10 11:00:51,794 - Phyluce - INFO - ----- Working on Penicillium_digitatum_GCF_000315645 genome -----\n",
      "2020-02-10 11:00:51,795 - Phyluce - INFO - Reading Penicillium_digitatum_GCF_000315645 genome\n",
      "2020-02-10 11:00:52,754 - Phyluce - INFO - Penicillium_digitatum_GCF_000315645: 160 uces, 38 dupes, 122 non-dupes, 2 orient drop, 2 length drop, 118 written\n",
      "2020-02-10 11:00:52,754 - Phyluce - INFO - ------ Working on Penicillium_expansum_GCF_000769745 genome -----\n",
      "2020-02-10 11:00:52,756 - Phyluce - INFO - Reading Penicillium_expansum_GCF_000769745 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 11:00:53,611 - Phyluce - INFO - Penicillium_expansum_GCF_000769745: 135 uces, 42 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-10 11:00:53,612 - Phyluce - INFO - ----- Working on Penicillium_flavigenum_GCA_002072365 genome ----\n",
      "2020-02-10 11:00:53,619 - Phyluce - INFO - Reading Penicillium_flavigenum_GCA_002072365 genome\n",
      "2020-02-10 11:00:54,662 - Phyluce - INFO - Penicillium_flavigenum_GCA_002072365: 163 uces, 48 dupes, 115 non-dupes, 0 orient drop, 0 length drop, 115 written\n",
      "2020-02-10 11:00:54,663 - Phyluce - INFO - ------- Working on Penicillium_freii_GCA_001513925 genome -------\n",
      "2020-02-10 11:00:54,671 - Phyluce - INFO - Reading Penicillium_freii_GCA_001513925 genome\n",
      "2020-02-10 11:00:55,582 - Phyluce - INFO - Penicillium_freii_GCA_001513925: 161 uces, 48 dupes, 113 non-dupes, 0 orient drop, 0 length drop, 113 written\n",
      "2020-02-10 11:00:55,582 - Phyluce - INFO - ---- Working on Penicillium_fuscoglaucum_GCA_000576735 genome ---\n",
      "2020-02-10 11:00:55,589 - Phyluce - INFO - Reading Penicillium_fuscoglaucum_GCA_000576735 genome\n",
      "2020-02-10 11:00:56,532 - Phyluce - INFO - Penicillium_fuscoglaucum_GCA_000576735: 153 uces, 47 dupes, 106 non-dupes, 0 orient drop, 0 length drop, 106 written\n",
      "2020-02-10 11:00:56,532 - Phyluce - INFO - ---- Working on Penicillium_griseofulvum_GCA_001561935 genome ---\n",
      "2020-02-10 11:00:56,539 - Phyluce - INFO - Reading Penicillium_griseofulvum_GCA_001561935 genome\n",
      "2020-02-10 11:00:57,519 - Phyluce - INFO - Penicillium_griseofulvum_GCA_001561935: 159 uces, 55 dupes, 104 non-dupes, 1 orient drop, 2 length drop, 101 written\n",
      "2020-02-10 11:00:57,520 - Phyluce - INFO - ------ Working on Penicillium_italicum_GCA_002116305 genome -----\n",
      "2020-02-10 11:00:57,532 - Phyluce - INFO - Reading Penicillium_italicum_GCA_002116305 genome\n",
      "2020-02-10 11:00:58,407 - Phyluce - INFO - Penicillium_italicum_GCA_002116305: 157 uces, 43 dupes, 114 non-dupes, 0 orient drop, 0 length drop, 114 written\n",
      "2020-02-10 11:00:58,407 - Phyluce - INFO - ---- Working on Penicillium_janthinellum_GCA_002369805 genome ---\n",
      "2020-02-10 11:00:58,408 - Phyluce - INFO - Reading Penicillium_janthinellum_GCA_002369805 genome\n",
      "2020-02-10 11:00:59,461 - Phyluce - INFO - Penicillium_janthinellum_GCA_002369805: 160 uces, 64 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-10 11:00:59,462 - Phyluce - INFO - ---- Working on Penicillium_nalgiovense_GCA_002072425 genome ----\n",
      "2020-02-10 11:00:59,476 - Phyluce - INFO - Reading Penicillium_nalgiovense_GCA_002072425 genome\n",
      "2020-02-10 11:01:00,434 - Phyluce - INFO - Penicillium_nalgiovense_GCA_002072425: 163 uces, 47 dupes, 116 non-dupes, 0 orient drop, 0 length drop, 116 written\n",
      "2020-02-10 11:01:00,434 - Phyluce - INFO - ------ Working on Penicillium_nordicum_GCA_000733025 genome -----\n",
      "2020-02-10 11:01:00,441 - Phyluce - INFO - Reading Penicillium_nordicum_GCA_000733025 genome\n",
      "2020-02-10 11:01:01,339 - Phyluce - INFO - Penicillium_nordicum_GCA_000733025: 163 uces, 47 dupes, 116 non-dupes, 0 orient drop, 0 length drop, 116 written\n",
      "2020-02-10 11:01:01,339 - Phyluce - INFO - ------ Working on Penicillium_oxalicum_GCA_000346795 genome -----\n",
      "2020-02-10 11:01:01,340 - Phyluce - INFO - Reading Penicillium_oxalicum_GCA_000346795 genome\n",
      "2020-02-10 11:01:02,214 - Phyluce - INFO - Penicillium_oxalicum_GCA_000346795: 157 uces, 42 dupes, 115 non-dupes, 0 orient drop, 3 length drop, 112 written\n",
      "2020-02-10 11:01:02,214 - Phyluce - INFO - ------- Working on Penicillium_paneum_GCA_000577715 genome ------\n",
      "2020-02-10 11:01:02,221 - Phyluce - INFO - Reading Penicillium_paneum_GCA_000577715 genome\n",
      "2020-02-10 11:01:03,095 - Phyluce - INFO - Penicillium_paneum_GCA_000577715: 149 uces, 40 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-10 11:01:03,096 - Phyluce - INFO - ------ Working on Penicillium_paxilli_GCA_000347475 genome ------\n",
      "2020-02-10 11:01:03,107 - Phyluce - INFO - Reading Penicillium_paxilli_GCA_000347475 genome\n",
      "2020-02-10 11:01:04,136 - Phyluce - INFO - Penicillium_paxilli_GCA_000347475: 163 uces, 47 dupes, 116 non-dupes, 0 orient drop, 0 length drop, 116 written\n",
      "2020-02-10 11:01:04,137 - Phyluce - INFO - ----- Working on Penicillium_polonicum_GCA_003344595 genome -----\n",
      "2020-02-10 11:01:04,147 - Phyluce - INFO - Reading Penicillium_polonicum_GCA_003344595 genome\n",
      "2020-02-10 11:01:04,978 - Phyluce - INFO - Penicillium_polonicum_GCA_003344595: 138 uces, 43 dupes, 95 non-dupes, 0 orient drop, 1 length drop, 94 written\n",
      "2020-02-10 11:01:04,978 - Phyluce - INFO - ----- Working on Penicillium_roqueforti_GCA_000513255 genome ----\n",
      "2020-02-10 11:01:04,994 - Phyluce - INFO - Reading Penicillium_roqueforti_GCA_000513255 genome\n",
      "2020-02-10 11:01:05,848 - Phyluce - INFO - Penicillium_roqueforti_GCA_000513255: 146 uces, 34 dupes, 112 non-dupes, 4 orient drop, 4 length drop, 104 written\n",
      "2020-02-10 11:01:05,849 - Phyluce - INFO - ------- Working on Penicillium_rubens_GCF_000226395 genome ------\n",
      "2020-02-10 11:01:05,861 - Phyluce - INFO - Reading Penicillium_rubens_GCF_000226395 genome\n",
      "2020-02-10 11:01:06,771 - Phyluce - INFO - Penicillium_rubens_GCF_000226395: 152 uces, 45 dupes, 107 non-dupes, 1 orient drop, 3 length drop, 103 written\n",
      "2020-02-10 11:01:06,771 - Phyluce - INFO - ---- Working on Penicillium_sclerotiorum_GCA_001750025 genome ---\n",
      "2020-02-10 11:01:06,773 - Phyluce - INFO - Reading Penicillium_sclerotiorum_GCA_001750025 genome\n",
      "2020-02-10 11:01:07,725 - Phyluce - INFO - Penicillium_sclerotiorum_GCA_001750025: 163 uces, 47 dupes, 116 non-dupes, 4 orient drop, 0 length drop, 112 written\n",
      "2020-02-10 11:01:07,725 - Phyluce - INFO - ------ Working on Penicillium_solitum_GCA_000952775 genome ------\n",
      "2020-02-10 11:01:07,726 - Phyluce - INFO - Reading Penicillium_solitum_GCA_000952775 genome\n",
      "2020-02-10 11:01:08,889 - Phyluce - INFO - Penicillium_solitum_GCA_000952775: 163 uces, 45 dupes, 118 non-dupes, 3 orient drop, 23 length drop, 92 written\n",
      "2020-02-10 11:01:08,890 - Phyluce - INFO - --------- Working on Penicillium_sp_GCA_003800485 genome --------\n",
      "2020-02-10 11:01:08,890 - Phyluce - INFO - Reading Penicillium_sp_GCA_003800485 genome\n",
      "2020-02-10 11:01:09,968 - Phyluce - INFO - Penicillium_sp_GCA_003800485: 163 uces, 57 dupes, 106 non-dupes, 0 orient drop, 0 length drop, 106 written\n",
      "2020-02-10 11:01:09,969 - Phyluce - INFO - ------ Working on Penicillium_steckii_GCA_002072375 genome ------\n",
      "2020-02-10 11:01:09,970 - Phyluce - INFO - Reading Penicillium_steckii_GCA_002072375 genome\n",
      "2020-02-10 11:01:10,826 - Phyluce - INFO - Penicillium_steckii_GCA_002072375: 159 uces, 47 dupes, 112 non-dupes, 0 orient drop, 0 length drop, 112 written\n",
      "2020-02-10 11:01:10,827 - Phyluce - INFO - ---- Working on Penicillium_subrubescens_GCA_001908125 genome ---\n",
      "2020-02-10 11:01:10,828 - Phyluce - INFO - Reading Penicillium_subrubescens_GCA_001908125 genome\n",
      "2020-02-10 11:01:11,781 - Phyluce - INFO - Penicillium_subrubescens_GCA_001908125: 163 uces, 48 dupes, 115 non-dupes, 0 orient drop, 0 length drop, 115 written\n",
      "2020-02-10 11:01:11,781 - Phyluce - INFO - ----- Working on Penicillium_verrucosum_GCA_000970515 genome ----\n",
      "2020-02-10 11:01:11,782 - Phyluce - INFO - Reading Penicillium_verrucosum_GCA_000970515 genome\n",
      "2020-02-10 11:01:12,736 - Phyluce - INFO - Penicillium_verrucosum_GCA_000970515: 156 uces, 54 dupes, 102 non-dupes, 0 orient drop, 0 length drop, 102 written\n",
      "2020-02-10 11:01:12,736 - Phyluce - INFO - ------ Working on Penicillium_vulpinum_GCA_002072255 genome -----\n",
      "2020-02-10 11:01:12,737 - Phyluce - INFO - Reading Penicillium_vulpinum_GCA_002072255 genome\n",
      "2020-02-10 11:01:13,714 - Phyluce - INFO - Penicillium_vulpinum_GCA_002072255: 159 uces, 48 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-10 11:01:13,714 - Phyluce - INFO - -------- Working on Phialosimplex_sp_GCA_003698115 genome -------\n",
      "2020-02-10 11:01:13,728 - Phyluce - INFO - Reading Phialosimplex_sp_GCA_003698115 genome\n",
      "2020-02-10 11:01:14,695 - Phyluce - INFO - Phialosimplex_sp_GCA_003698115: 140 uces, 50 dupes, 90 non-dupes, 0 orient drop, 0 length drop, 90 written\n",
      "2020-02-10 11:01:14,695 - Phyluce - INFO - ------ Working on Rasamsonia_emersonii_GCF_000968595 genome -----\n",
      "2020-02-10 11:01:14,703 - Phyluce - INFO - Reading Rasamsonia_emersonii_GCF_000968595 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 11:01:15,353 - Phyluce - INFO - Rasamsonia_emersonii_GCF_000968595: 137 uces, 42 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-10 11:01:15,353 - Phyluce - INFO - ----- Working on Talaromyces_adpressus_GCA_002775195 genome -----\n",
      "2020-02-10 11:01:15,354 - Phyluce - INFO - Reading Talaromyces_adpressus_GCA_002775195 genome\n",
      "2020-02-10 11:01:16,298 - Phyluce - INFO - Talaromyces_adpressus_GCA_002775195: 162 uces, 47 dupes, 115 non-dupes, 0 orient drop, 0 length drop, 115 written\n",
      "2020-02-10 11:01:16,298 - Phyluce - INFO - ---- Working on Talaromyces_amestolkiae_GCA_001896365 genome ----\n",
      "2020-02-10 11:01:16,300 - Phyluce - INFO - Reading Talaromyces_amestolkiae_GCA_001896365 genome\n",
      "2020-02-10 11:01:17,103 - Phyluce - INFO - Talaromyces_amestolkiae_GCA_001896365: 163 uces, 55 dupes, 108 non-dupes, 0 orient drop, 1 length drop, 107 written\n",
      "2020-02-10 11:01:17,103 - Phyluce - INFO - ----- Working on Talaromyces_atroroseus_GCF_001907595 genome ----\n",
      "2020-02-10 11:01:17,104 - Phyluce - INFO - Reading Talaromyces_atroroseus_GCF_001907595 genome\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_slice_sequence_from_genomes\", line 348, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_slice_sequence_from_genomes\", line 287, in main\n",
      "    all_uce_names, uce_matches, orientation = parse_lastz_file(lz, args.contig_orient, regex)\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_slice_sequence_from_genomes\", line 228, in parse_lastz_file\n",
      "    for lz in lastz.Reader(lz, long_format=True):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/lib/python2.7/site-packages/phyluce/lastz.py\", line 109, in __init__\n",
      "    self.file = open(lastz_file, 'rU')\n",
      "IOError: [Errno 2] No such file or directory: '/master/nplatt/pathogen_probes/results/phyluce/eurotiales/validate_intervals/lastz/Talaromyces_marneffei_GCF_000001985_+50_temp_probes.fas_v_Talaromyces_atroroseus_GCF_001907595.lastz.clean'\n",
      "Exception AttributeError: \"Reader instance has no attribute 'file'\" in <bound method Reader.__del__ of <phyluce.lastz.Reader instance at 0x7f12d832d290>> ignored\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/eurotiales/extract_probes_from_group/probe_fasta --output results/phyluce/eurotiales/extract_probes_from_group/multifastas.sqlite --base-taxon Talaromyces_marneffei_GCF_000001985;\n",
      "asprgillus_aculeatinus_gcf_003184765.\n",
      "asprgillus_aculeatus_gcf_001890905.\n",
      "asprgillus_aff_gca_001931935.\n",
      "asprgillus_albertensis_gca_009193605.\n",
      "asprgillus_alliaceus_gcf_009176365.\n",
      "asprgillus_amoenus_gca_009812435.\n",
      "asprgillus_arachidicola_gca_009193545.\n",
      "asprgillus_avenaceus_gca_009193465.\n",
      "asprgillus_awamori_gca_003850985.\n",
      "asprgillus_bertholletius_gca_009193595.\n",
      "asprgillus_bombycis_gcf_001792695.\n",
      "asprgillus_brasiliensis_gca_001889945.\n",
      "asprgillus_brunneoviolaceus_gcf_003184695.\n",
      "asprgillus_caelatus_gcf_009193585.\n",
      "asprgillus_calidoustus_gca_001511075.\n",
      "asprgillus_campestris_gcf_002847485.\n",
      "asprgillus_candidus_gcf_002847045.\n",
      "asprgillus_carbonarius_gca_001990825.\n",
      "asprgillus_cejpii_gca_004769165.\n",
      "asprgillus_chevalieri_gca_001599875.\n",
      "asprgillus_clavatus_gcf_000002715.\n",
      "asprgillus_coremiiformis_gca_009193565.\n",
      "asprgillus_costaricaensis_gcf_003184835.\n",
      "asprgillus_cristatus_gca_001693355.\n",
      "asprgillus_ellipticus_gca_003184645.\n",
      "asprgillus_eucalypticola_gcf_003184535.\n",
      "asprgillus_fijiensis_gca_003184825.\n",
      "asprgillus_fischeri_gcf_000149645.\n",
      "asprgillus_flavus_gcf_000006275.\n",
      "asprgillus_fumigatus_gcf_000002655.\n",
      "asprgillus_glaucus_gcf_001890805.\n",
      "asprgillus_hancockii_gca_001696595.\n",
      "asprgillus_heteromorphus_gcf_003184545.\n",
      "asprgillus_homomorphus_gcf_003184865.\n",
      "asprgillus_ibericus_gcf_003184845.\n",
      "asprgillus_indologenus_gca_003184685.\n",
      "asprgillus_japonicus_gcf_003184785.\n",
      "asprgillus_kawachii_gca_000239835.\n",
      "asprgillus_lentulus_gca_001445615.\n",
      "asprgillus_leporis_gca_009176345.\n",
      "asprgillus_luchuensis_gca_001890685.\n",
      "asprgillus_minisclerotigenes_gca_009176455.\n",
      "asprgillus_mulundensis_gcf_003369625.\n",
      "asprgillus_neoellipticus_gca_003116565.\n",
      "asprgillus_neoniger_gcf_003184625.\n",
      "asprgillus_nidulans_gcf_000149205.\n",
      "asprgillus_niger_gcf_000002855.\n",
      "asprgillus_nomius_gcf_001204775.\n",
      "asprgillus_novofumigatus_gcf_002847465.\n",
      "asprgillus_novoparasiticus_gca_009176405.\n",
      "asprgillus_ochraceoroseus_gca_002846915.\n",
      "asprgillus_ochraceus_gca_004849945.\n",
      "asprgillus_olivimuriae_gca_003719415.\n",
      "asprgillus_oryzae_gcf_000184455.\n",
      "asprgillus_parasiticus_gca_009176385.\n",
      "asprgillus_persii_gca_002215965.\n",
      "asprgillus_phoenicis_gca_003344505.\n",
      "asprgillus_piperis_gcf_003184755.\n",
      "asprgillus_pseudocaelatus_gca_009193665.\n",
      "asprgillus_pseudotamarii_gcf_009193445.\n",
      "asprgillus_pseudoterreus_gca_002927005.\n",
      "asprgillus_rambellii_gca_000986645.\n",
      "asprgillus_ruber_gca_000600275.\n",
      "asprgillus_saccharolyticus_gcf_003184585.\n",
      "asprgillus_sclerotialis_gca_003589665.\n",
      "asprgillus_sclerotiicarbonarius_gca_003184635.\n",
      "asprgillus_sclerotioniger_gcf_003184525.\n",
      "asprgillus_sclerotiorum_gca_000530345.\n",
      "asprgillus_sergii_gca_009193525.\n",
      "asprgillus_sojae_gca_008274985.\n",
      "asprgillus_sp_gca_001044295.\n",
      "asprgillus_spnulosprus_gca_003574815.\n",
      "asprgillus_steynii_gcf_002849105.\n",
      "asprgillus_sydowii_gca_001890705.\n",
      "asprgillus_taichungensis_gca_002850765.\n",
      "asprgillus_tamarii_gca_009193485.\n",
      "asprgillus_tanneri_gca_003426965.\n",
      "asprgillus_terreus_gcf_000149615.\n",
      "asprgillus_thermomutatus_gcf_002237265.\n",
      "asprgillus_transmontanensis_gca_009193505.\n",
      "asprgillus_tritici_gca_009812425.\n",
      "asprgillus_tubingensis_gca_001890745.\n",
      "asprgillus_turcosus_gca_002234965.\n",
      "asprgillus_udagawae_gca_001078395.\n",
      "asprgillus_unguis_gca_003324175.\n",
      "asprgillus_ustus_gca_000812125.\n",
      "asprgillus_uvarum_gcf_003184745.\n",
      "asprgillus_vadensis_gcf_003184925.\n",
      "asprgillus_versicolor_gca_001890125.\n",
      "asprgillus_violaceofuscus_gca_003184705.\n",
      "asprgillus_viridinutans_gca_004368095.\n",
      "asprgillus_welwitschiae_gcf_003344945.\n",
      "asprgillus_wentii_gca_001890725.\n",
      "asprgillus_westerdijkiae_gca_001307345.\n",
      "basipetospra_chlamydospra_gca_001599675.\n",
      "byssochlamys_nivea_gca_003116535.\n",
      "byssochlamys_sp_gca_002914405.\n",
      "byssochlamys_spctabilis_gcf_004022145.\n",
      "elaphomyces_granulatus_gca_002240705.\n",
      "monascus_purpureus_gca_003184285.\n",
      "monascus_ruber_gca_002976275.\n",
      "penicilliopsis_zonata_gcf_001890105.\n",
      "penicillium_antarcticum_gca_002072345.\n",
      "penicillium_arizonense_gcf_001773325.\n",
      "penicillium_biforme_gca_000577785.\n",
      "penicillium_brasilianum_gca_001048715.\n",
      "penicillium_camemberti_gca_000513335.\n",
      "penicillium_capsulatum_gca_000943775.\n",
      "penicillium_carneum_gca_000577495.\n",
      "penicillium_chrysogenum_gca_000710275.\n",
      "penicillium_citrinum_gca_001950535.\n",
      "penicillium_coprophilum_gca_002072405.\n",
      "penicillium_decumbens_gca_002072245.\n",
      "penicillium_digitatum_gcf_000315645.\n",
      "penicillium_expansum_gcf_000769745.\n",
      "penicillium_flavigenum_gca_002072365.\n",
      "penicillium_freii_gca_001513925.\n",
      "penicillium_fuscoglaucum_gca_000576735.\n",
      "penicillium_griseofulvum_gca_001561935.\n",
      "penicillium_italicum_gca_002116305.\n",
      "penicillium_janthinellum_gca_002369805.\n",
      "penicillium_nalgiovense_gca_002072425.\n",
      "penicillium_nordicum_gca_000733025.\n",
      "penicillium_oxalicum_gca_000346795.\n",
      "penicillium_paneum_gca_000577715.\n",
      "penicillium_paxilli_gca_000347475.\n",
      "penicillium_polonicum_gca_003344595.\n",
      "penicillium_roqueforti_gca_000513255.\n",
      "penicillium_rubens_gcf_000226395.\n",
      "penicillium_sclerotiorum_gca_001750025.\n",
      "penicillium_solitum_gca_000952775.\n",
      "penicillium_sp_gca_003800485.\n",
      "penicillium_steckii_gca_002072375.\n",
      "penicillium_subrubescens_gca_001908125.\n",
      "penicillium_verrucosum_gca_000970515.\n",
      "penicillium_vulpinum_gca_002072255.\n",
      "phialosimplex_sp_gca_003698115.\n",
      "rasamsonia_emersonii_gcf_000968595.\n",
      "talaromyces_adpressus_gca_002775195.\n",
      "talaromyces_amestolkiae_gca_001896365.\n",
      "talaromyces_atroroseus_gcf_001907595\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/eurotiales/extract_probes_from_group/multifastas.sqlite --base-taxon Talaromyces_marneffei_GCF_000001985;\n",
      "Loci shared by 0 taxa:\t152.0\n",
      "Loci shared by 1 taxa:\t152.0\n",
      "Loci shared by 2 taxa:\t145.0\n",
      "Loci shared by 3 taxa:\t143.0\n",
      "Loci shared by 4 taxa:\t139.0\n",
      "Loci shared by 5 taxa:\t138.0\n",
      "Loci shared by 6 taxa:\t137.0\n",
      "Loci shared by 7 taxa:\t137.0\n",
      "Loci shared by 8 taxa:\t137.0\n",
      "Loci shared by 9 taxa:\t136.0\n",
      "Loci shared by 10 taxa:\t135.0\n",
      "Loci shared by 11 taxa:\t134.0\n",
      "Loci shared by 12 taxa:\t133.0\n",
      "Loci shared by 13 taxa:\t132.0\n",
      "Loci shared by 14 taxa:\t130.0\n",
      "Loci shared by 15 taxa:\t130.0\n",
      "Loci shared by 16 taxa:\t130.0\n",
      "Loci shared by 17 taxa:\t130.0\n",
      "Loci shared by 18 taxa:\t130.0\n",
      "Loci shared by 19 taxa:\t129.0\n",
      "Loci shared by 20 taxa:\t129.0\n",
      "Loci shared by 21 taxa:\t127.0\n",
      "Loci shared by 22 taxa:\t127.0\n",
      "Loci shared by 23 taxa:\t126.0\n",
      "Loci shared by 24 taxa:\t126.0\n",
      "Loci shared by 25 taxa:\t126.0\n",
      "Loci shared by 26 taxa:\t126.0\n",
      "Loci shared by 27 taxa:\t126.0\n",
      "Loci shared by 28 taxa:\t125.0\n",
      "Loci shared by 29 taxa:\t125.0\n",
      "Loci shared by 30 taxa:\t124.0\n",
      "Loci shared by 31 taxa:\t123.0\n",
      "Loci shared by 32 taxa:\t123.0\n",
      "Loci shared by 33 taxa:\t123.0\n",
      "Loci shared by 34 taxa:\t123.0\n",
      "Loci shared by 35 taxa:\t123.0\n",
      "Loci shared by 36 taxa:\t123.0\n",
      "Loci shared by 37 taxa:\t123.0\n",
      "Loci shared by 38 taxa:\t122.0\n",
      "Loci shared by 39 taxa:\t121.0\n",
      "Loci shared by 40 taxa:\t119.0\n",
      "Loci shared by 41 taxa:\t118.0\n",
      "Loci shared by 42 taxa:\t116.0\n",
      "Loci shared by 43 taxa:\t115.0\n",
      "Loci shared by 44 taxa:\t113.0\n",
      "Loci shared by 45 taxa:\t112.0\n",
      "Loci shared by 46 taxa:\t109.0\n",
      "Loci shared by 47 taxa:\t108.0\n",
      "Loci shared by 48 taxa:\t108.0\n",
      "Loci shared by 49 taxa:\t106.0\n",
      "Loci shared by 50 taxa:\t104.0\n",
      "Loci shared by 51 taxa:\t102.0\n",
      "Loci shared by 52 taxa:\t102.0\n",
      "Loci shared by 53 taxa:\t100.0\n",
      "Loci shared by 54 taxa:\t97.0\n",
      "Loci shared by 55 taxa:\t97.0\n",
      "Loci shared by 56 taxa:\t97.0\n",
      "Loci shared by 57 taxa:\t95.0\n",
      "Loci shared by 58 taxa:\t95.0\n",
      "Loci shared by 59 taxa:\t95.0\n",
      "Loci shared by 60 taxa:\t95.0\n",
      "Loci shared by 61 taxa:\t95.0\n",
      "Loci shared by 62 taxa:\t94.0\n",
      "Loci shared by 63 taxa:\t94.0\n",
      "Loci shared by 64 taxa:\t94.0\n",
      "Loci shared by 65 taxa:\t93.0\n",
      "Loci shared by 66 taxa:\t93.0\n",
      "Loci shared by 67 taxa:\t93.0\n",
      "Loci shared by 68 taxa:\t93.0\n",
      "Loci shared by 69 taxa:\t93.0\n",
      "Loci shared by 70 taxa:\t93.0\n",
      "Loci shared by 71 taxa:\t93.0\n",
      "Loci shared by 72 taxa:\t93.0\n",
      "Loci shared by 73 taxa:\t92.0\n",
      "Loci shared by 74 taxa:\t92.0\n",
      "Loci shared by 75 taxa:\t92.0\n",
      "Loci shared by 76 taxa:\t91.0\n",
      "Loci shared by 77 taxa:\t91.0\n",
      "Loci shared by 78 taxa:\t90.0\n",
      "Loci shared by 79 taxa:\t89.0\n",
      "Loci shared by 80 taxa:\t88.0\n",
      "Loci shared by 81 taxa:\t88.0\n",
      "Loci shared by 82 taxa:\t87.0\n",
      "Loci shared by 83 taxa:\t87.0\n",
      "Loci shared by 84 taxa:\t87.0\n",
      "Loci shared by 85 taxa:\t87.0\n",
      "Loci shared by 86 taxa:\t87.0\n",
      "Loci shared by 87 taxa:\t87.0\n",
      "Loci shared by 88 taxa:\t87.0\n",
      "Loci shared by 89 taxa:\t87.0\n",
      "Loci shared by 90 taxa:\t87.0\n",
      "Loci shared by 91 taxa:\t87.0\n",
      "Loci shared by 92 taxa:\t87.0\n",
      "Loci shared by 93 taxa:\t87.0\n",
      "Loci shared by 94 taxa:\t87.0\n",
      "Loci shared by 95 taxa:\t87.0\n",
      "Loci shared by 96 taxa:\t87.0\n",
      "Loci shared by 97 taxa:\t87.0\n",
      "Loci shared by 98 taxa:\t87.0\n",
      "Loci shared by 99 taxa:\t87.0\n",
      "Loci shared by 100 taxa:\t87.0\n",
      "Loci shared by 101 taxa:\t87.0\n",
      "Loci shared by 102 taxa:\t87.0\n",
      "Loci shared by 103 taxa:\t86.0\n",
      "Loci shared by 104 taxa:\t86.0\n",
      "Loci shared by 105 taxa:\t86.0\n",
      "Loci shared by 106 taxa:\t86.0\n",
      "Loci shared by 107 taxa:\t86.0\n",
      "Loci shared by 108 taxa:\t86.0\n",
      "Loci shared by 109 taxa:\t86.0\n",
      "Loci shared by 110 taxa:\t86.0\n",
      "Loci shared by 111 taxa:\t85.0\n",
      "Loci shared by 112 taxa:\t85.0\n",
      "Loci shared by 113 taxa:\t85.0\n",
      "Loci shared by 114 taxa:\t85.0\n",
      "Loci shared by 115 taxa:\t85.0\n",
      "Loci shared by 116 taxa:\t84.0\n",
      "Loci shared by 117 taxa:\t82.0\n",
      "Loci shared by 118 taxa:\t82.0\n",
      "Loci shared by 119 taxa:\t81.0\n",
      "Loci shared by 120 taxa:\t81.0\n",
      "Loci shared by 121 taxa:\t79.0\n",
      "Loci shared by 122 taxa:\t76.0\n",
      "Loci shared by 123 taxa:\t76.0\n",
      "Loci shared by 124 taxa:\t73.0\n",
      "Loci shared by 125 taxa:\t73.0\n",
      "Loci shared by 126 taxa:\t72.0\n",
      "Loci shared by 127 taxa:\t71.0\n",
      "Loci shared by 128 taxa:\t71.0\n",
      "Loci shared by 129 taxa:\t70.0\n",
      "Loci shared by 130 taxa:\t70.0\n",
      "Loci shared by 131 taxa:\t70.0\n",
      "Loci shared by 132 taxa:\t70.0\n",
      "Loci shared by 133 taxa:\t69.0\n",
      "Loci shared by 134 taxa:\t67.0\n",
      "Loci shared by 135 taxa:\t64.0\n",
      "Loci shared by 136 taxa:\t63.0\n",
      "Loci shared by 137 taxa:\t58.0\n",
      "Loci shared by 138 taxa:\t41.0\n",
      "Loci shared by 139 taxa:\t25.0\n",
      "Loci shared by 140 taxa:\t5.0\n",
      "Loci shared by 141 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/eurotiales/extract_probes_from_group/multifastas.sqlite  --base-taxon Talaromyces_marneffei_GCF_000001985 --output results/phyluce/eurotiales/extract_probes_from_group/Talaromyces_marneffei_GCF_000001985+50-back-to-137.conf --specific-counts 137;\n",
      "Counter({'asprgillus_steynii_gcf_002849105': 58, 'penicillium_vulpinum_gca_002072255': 58, 'penicillium_griseofulvum_gca_001561935': 58, 'asprgillus_fischeri_gcf_000149645': 58, 'asprgillus_pseudoterreus_gca_002927005': 58, 'asprgillus_versicolor_gca_001890125': 58, 'byssochlamys_sp_gca_002914405': 58, 'asprgillus_welwitschiae_gcf_003344945': 58, 'asprgillus_terreus_gcf_000149615': 58, 'penicillium_antarcticum_gca_002072345': 58, 'penicillium_oxalicum_gca_000346795': 58, 'asprgillus_turcosus_gca_002234965': 58, 'asprgillus_sojae_gca_008274985': 58, 'asprgillus_saccharolyticus_gcf_003184585': 58, 'penicillium_capsulatum_gca_000943775': 58, 'asprgillus_sydowii_gca_001890705': 58, 'penicillium_nordicum_gca_000733025': 58, 'asprgillus_amoenus_gca_009812435': 58, 'penicillium_brasilianum_gca_001048715': 58, 'asprgillus_transmontanensis_gca_009193505': 58, 'asprgillus_arachidicola_gca_009193545': 58, 'asprgillus_japonicus_gcf_003184785': 58, 'penicillium_subrubescens_gca_001908125': 58, 'monascus_purpureus_gca_003184285': 58, 'asprgillus_ochraceoroseus_gca_002846915': 58, 'penicillium_roqueforti_gca_000513255': 58, 'asprgillus_tanneri_gca_003426965': 58, 'asprgillus_fumigatus_gcf_000002655': 58, 'asprgillus_phoenicis_gca_003344505': 58, 'monascus_ruber_gca_002976275': 58, 'asprgillus_clavatus_gcf_000002715': 58, 'asprgillus_tritici_gca_009812425': 58, 'penicillium_citrinum_gca_001950535': 58, 'asprgillus_ibericus_gcf_003184845': 58, 'penicillium_arizonense_gcf_001773325': 58, 'asprgillus_ruber_gca_000600275': 58, 'asprgillus_indologenus_gca_003184685': 58, 'penicillium_rubens_gcf_000226395': 58, 'asprgillus_fijiensis_gca_003184825': 58, 'asprgillus_calidoustus_gca_001511075': 58, 'asprgillus_costaricaensis_gcf_003184835': 58, 'penicillium_fuscoglaucum_gca_000576735': 58, 'asprgillus_neoellipticus_gca_003116565': 58, 'asprgillus_brunneoviolaceus_gcf_003184695': 58, 'asprgillus_sclerotioniger_gcf_003184525': 58, 'penicilliopsis_zonata_gcf_001890105': 58, 'asprgillus_minisclerotigenes_gca_009176455': 58, 'penicillium_biforme_gca_000577785': 58, 'byssochlamys_nivea_gca_003116535': 58, 'asprgillus_coremiiformis_gca_009193565': 58, 'asprgillus_cristatus_gca_001693355': 58, 'penicillium_verrucosum_gca_000970515': 58, 'asprgillus_novoparasiticus_gca_009176405': 58, 'penicillium_freii_gca_001513925': 58, 'asprgillus_awamori_gca_003850985': 58, 'penicillium_solitum_gca_000952775': 58, 'asprgillus_bombycis_gcf_001792695': 58, 'penicillium_steckii_gca_002072375': 58, 'asprgillus_cejpii_gca_004769165': 58, 'asprgillus_brasiliensis_gca_001889945': 58, 'asprgillus_tamarii_gca_009193485': 58, 'penicillium_chrysogenum_gca_000710275': 58, 'penicillium_coprophilum_gca_002072405': 58, 'asprgillus_sp_gca_001044295': 58, 'asprgillus_flavus_gcf_000006275': 58, 'asprgillus_pseudotamarii_gcf_009193445': 58, 'penicillium_polonicum_gca_003344595': 58, 'asprgillus_pseudocaelatus_gca_009193665': 58, 'asprgillus_wentii_gca_001890725': 58, 'byssochlamys_spctabilis_gcf_004022145': 58, 'asprgillus_thermomutatus_gcf_002237265': 58, 'asprgillus_oryzae_gcf_000184455': 58, 'asprgillus_taichungensis_gca_002850765': 58, 'asprgillus_neoniger_gcf_003184625': 58, 'penicillium_paxilli_gca_000347475': 58, 'penicillium_sp_gca_003800485': 58, 'asprgillus_lentulus_gca_001445615': 58, 'asprgillus_parasiticus_gca_009176385': 58, 'asprgillus_vadensis_gcf_003184925': 58, 'talaromyces_adpressus_gca_002775195': 58, 'asprgillus_viridinutans_gca_004368095': 58, 'asprgillus_luchuensis_gca_001890685': 58, 'penicillium_janthinellum_gca_002369805': 58, 'asprgillus_caelatus_gcf_009193585': 58, 'asprgillus_aculeatinus_gcf_003184765': 58, 'penicillium_italicum_gca_002116305': 58, 'asprgillus_udagawae_gca_001078395': 58, 'asprgillus_tubingensis_gca_001890745': 58, 'penicillium_flavigenum_gca_002072365': 58, 'asprgillus_alliaceus_gcf_009176365': 58, 'asprgillus_piperis_gcf_003184755': 58, 'asprgillus_bertholletius_gca_009193595': 58, 'penicillium_digitatum_gcf_000315645': 58, 'asprgillus_hancockii_gca_001696595': 58, 'penicillium_nalgiovense_gca_002072425': 58, 'penicillium_expansum_gcf_000769745': 58, 'asprgillus_albertensis_gca_009193605': 58, 'asprgillus_campestris_gcf_002847485': 58, 'asprgillus_ellipticus_gca_003184645': 58, 'asprgillus_sergii_gca_009193525': 58, 'asprgillus_nomius_gcf_001204775': 58, 'penicillium_camemberti_gca_000513335': 58, 'asprgillus_aculeatus_gcf_001890905': 58, 'asprgillus_persii_gca_002215965': 58, 'asprgillus_uvarum_gcf_003184745': 58, 'asprgillus_avenaceus_gca_009193465': 58, 'talaromyces_amestolkiae_gca_001896365': 58, 'asprgillus_leporis_gca_009176345': 57, 'asprgillus_violaceofuscus_gca_003184705': 57, 'asprgillus_westerdijkiae_gca_001307345': 57, 'asprgillus_rambellii_gca_000986645': 57, 'asprgillus_ochraceus_gca_004849945': 57, 'asprgillus_sclerotiicarbonarius_gca_003184635': 57, 'asprgillus_nidulans_gcf_000149205': 57, 'asprgillus_glaucus_gcf_001890805': 57, 'penicillium_decumbens_gca_002072245': 57, 'asprgillus_novofumigatus_gcf_002847465': 57, 'asprgillus_candidus_gcf_002847045': 57, 'asprgillus_heteromorphus_gcf_003184545': 57, 'asprgillus_chevalieri_gca_001599875': 57, 'asprgillus_eucalypticola_gcf_003184535': 57, 'asprgillus_niger_gcf_000002855': 57, 'asprgillus_homomorphus_gcf_003184865': 57, 'asprgillus_spnulosprus_gca_003574815': 57, 'asprgillus_aff_gca_001931935': 57, 'asprgillus_mulundensis_gcf_003369625': 57, 'rasamsonia_emersonii_gcf_000968595': 57, 'asprgillus_kawachii_gca_000239835': 57, 'asprgillus_olivimuriae_gca_003719415': 57, 'asprgillus_ustus_gca_000812125': 57, 'phialosimplex_sp_gca_003698115': 56, 'penicillium_sclerotiorum_gca_001750025': 56, 'asprgillus_carbonarius_gca_001990825': 56, 'penicillium_paneum_gca_000577715': 56, 'asprgillus_sclerotiorum_gca_000530345': 56, 'penicillium_carneum_gca_000577495': 55, 'basipetospra_chlamydospra_gca_001599675': 54, 'elaphomyces_granulatus_gca_002240705': 52, 'asprgillus_sclerotialis_gca_003589665': 48, 'asprgillus_unguis_gca_003324175': 11})\n",
      "Total loci = 58\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(137)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/eurotiales/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/eurotiales/extract_probes_from_group/Talaromyces_marneffei_GCF_000001985+50-back-to-137.conf --probe-prefix uce_eurotiales_ --designer rnplattii --design eurotiales_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list.fasta;\n",
      "GGGGGGGNNGGGNNNNGGGGGGGGNGGGGGGGGGGGGNGGGGGGNNNN\n",
      "\n",
      "\n",
      "Conserved locus count = 58\n",
      "Probe Count = 15986\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list.fasta --query results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Mon Feb 10, 2020  11:20:53\n",
      "Ended:  Mon Feb 10, 2020  11:24:35\n",
      "Time for execution:  3.7069364508 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list.fasta --lastz results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_eurotiales_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 15985 fasta sequences.  Filtered 0 duplicates. Kept 15986.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/eurotiales/final_probe_design/eurotiales_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Mon Feb 10 13:00:26 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 15986\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1278880\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 49M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 90M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3937783\n",
      "Max number of word counting entries: 88639506\n",
      "\n",
      "# comparing sequences from          0  to       2664\n",
      "..---------- new table with      991 representatives\n",
      "# comparing sequences from       2664  to       4884\n",
      "----------   1506 remaining sequences to the next cycle\n",
      "---------- new table with      287 representatives\n",
      "# comparing sequences from       3378  to       5479\n",
      "----------   1407 remaining sequences to the next cycle\n",
      "---------- new table with      256 representatives\n",
      "# comparing sequences from       4072  to       6057\n",
      "----------   1272 remaining sequences to the next cycle\n",
      "---------- new table with      196 representatives\n",
      "# comparing sequences from       4785  to       6651\n",
      "----------   1170 remaining sequences to the next cycle\n",
      "---------- new table with      187 representatives\n",
      "# comparing sequences from       5481  to       7231\n",
      "----------   1082 remaining sequences to the next cycle\n",
      "---------- new table with      186 representatives\n",
      "# comparing sequences from       6149  to       7788\n",
      "----------    972 remaining sequences to the next cycle\n",
      "---------- new table with      155 representatives\n",
      "# comparing sequences from       6816  to       8344\n",
      "----------    980 remaining sequences to the next cycle\n",
      "---------- new table with      148 representatives\n",
      "# comparing sequences from       7364  to       8801\n",
      "----------    892 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       7909  to       9255\n",
      "----------    852 remaining sequences to the next cycle\n",
      "---------- new table with      114 representatives\n",
      "# comparing sequences from       8403  to       9666\n",
      "----------    846 remaining sequences to the next cycle\n",
      "---------- new table with      173 representatives\n",
      "# comparing sequences from       8820  to      10014\n",
      "----------    708 remaining sequences to the next cycle\n",
      "---------- new table with      145 representatives\n",
      "# comparing sequences from       9306  to      10419\n",
      "----------    580 remaining sequences to the next cycle\n",
      "---------- new table with      116 representatives\n",
      "# comparing sequences from       9839  to      10863\n",
      "..........    10000  finished       3110  clusters\n",
      "----------    711 remaining sequences to the next cycle\n",
      "---------- new table with      109 representatives\n",
      "# comparing sequences from      10152  to      11124\n",
      "----------    710 remaining sequences to the next cycle\n",
      "---------- new table with      112 representatives\n",
      "# comparing sequences from      10414  to      11342\n",
      "----------    555 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      10787  to      11653\n",
      "----------    527 remaining sequences to the next cycle\n",
      "---------- new table with      104 representatives\n",
      "# comparing sequences from      11126  to      11936\n",
      "----------    294 remaining sequences to the next cycle\n",
      "---------- new table with      168 representatives\n",
      "# comparing sequences from      11642  to      12366\n",
      "----------    299 remaining sequences to the next cycle\n",
      "---------- new table with      145 representatives\n",
      "# comparing sequences from      12067  to      12720\n",
      "----------    182 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      12538  to      13112\n",
      "----------    101 remaining sequences to the next cycle\n",
      "---------- new table with      115 representatives\n",
      "# comparing sequences from      13011  to      13506\n",
      "----------    141 remaining sequences to the next cycle\n",
      "---------- new table with      106 representatives\n",
      "# comparing sequences from      13365  to      13801\n",
      "----------      3 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      13798  to      14162\n",
      "99.9%---------- new table with       54 representatives\n",
      "# comparing sequences from      14162  to      14466\n",
      "...................---------- new table with       92 representatives\n",
      "# comparing sequences from      14466  to      14719\n",
      "99.9%---------- new table with       90 representatives\n",
      "# comparing sequences from      14719  to      14930\n",
      "...................---------- new table with       86 representatives\n",
      "# comparing sequences from      14930  to      15106\n",
      "100.0%---------- new table with       21 representatives\n",
      "# comparing sequences from      15106  to      15986\n",
      "...................---------- new table with      308 representatives\n",
      "\n",
      "    15986  finished       4864  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 91M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 3.13\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Hexamitidae"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'hexamitidae'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Giardia_intestinalis_GCA_000182405\",\n",
    "               \"Giardia_intestinalis_GCA_000182665\",\n",
    "               \"Giardia_intestinalis_GCA_000498715\",\n",
    "               \"Giardia_intestinalis_GCA_000498735\",\n",
    "               \"Giardia_intestinalis_GCA_001493575\",\n",
    "               \"Giardia_intestinalis_GCA_001543975\",\n",
    "               \"Giardia_intestinalis_GCA_009192805\",\n",
    "               \"Giardia_intestinalis_GCA_009192825\",\n",
    "               \"Giardia_intestinalis_GCA_902209425\",\n",
    "               \"Giardia_intestinalis_GCA_902221465\",\n",
    "               \"Giardia_intestinalis_GCA_902221485\",\n",
    "               \"Giardia_intestinalis_GCA_902221515\",\n",
    "               \"Giardia_intestinalis_GCA_902221535\",\n",
    "               \"Giardia_intestinalis_GCA_902221545\",\n",
    "               \"Giardia_lamblia_GCA_000182665\",\n",
    "               \"Giardia_muris_GCA_006247105\",\n",
    "               \"Spironucleus_salmonicida_GCA_000497125\" ]\n",
    "                    \n",
    "reference_taxon = \"Giardia_intestinalis_GCA_000002435\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002435.2_UU_WB_2.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3518754 bytes  2345864.00 bytes/sec\n",
      "total size is 3517786  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000182405.1_ASM18240v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3438178 bytes  1375288.00 bytes/sec\n",
      "total size is 3437233  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000182665.1_ASM18266v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3566636 bytes  2377785.33 bytes/sec\n",
      "total size is 3565659  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000498715.1_ASM49871v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3328220 bytes  739613.78 bytes/sec\n",
      "total size is 3327299  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000498735.1_ASM49873v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3714855 bytes  1485958.80 bytes/sec\n",
      "total size is 3713838  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001493575.1_GIAS175_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3300060 bytes  1320040.80 bytes/sec\n",
      "total size is 3299150  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001543975.1_G_duodenalis_AssB_BAH15c1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3223209 bytes  1289300.40 bytes/sec\n",
      "total size is 3222297  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009192805.1_ASM919280v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3177608 bytes  2118433.33 bytes/sec\n",
      "total size is 3176726  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009192825.1_ASM919282v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3196323 bytes  2130910.00 bytes/sec\n",
      "total size is 3195433  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902209425.1_cyste1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3571718 bytes  1428704.00 bytes/sec\n",
      "total size is 3570745  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_902221465.1_cyste2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3494782 bytes  258875.85 bytes/sec\n",
      "total size is 3493825  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902221485.1_cyste4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3458534 bytes  628832.00 bytes/sec\n",
      "total size is 3457585  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902221515.1_cyste3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3575635 bytes  2383784.67 bytes/sec\n",
      "total size is 3574654  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902221535.1_pool5_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3533124 bytes  1413266.40 bytes/sec\n",
      "total size is 3532160  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902221545.1_pool8_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3716549 bytes  1486636.40 bytes/sec\n",
      "total size is 3715537  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000182665.1_ASM18266v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 3566636 bytes  2377785.33 bytes/sec\n",
      "total size is 3565659  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006247105.1_UU_GM_1.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2821076 bytes  1128447.20 bytes/sec\n",
      "total size is 2820276  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000497125.1_SSK3.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 4055372 bytes  2703609.33 bytes/sec\n",
      "total size is 4054279  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485005 (\"sim_Giardia_intestinalis_GCA_000182405\") has been submitted\n",
      "Your job 5485006 (\"sim_Giardia_intestinalis_GCA_000182665\") has been submitted\n",
      "Your job 5485007 (\"sim_Giardia_intestinalis_GCA_000498715\") has been submitted\n",
      "Your job 5485008 (\"sim_Giardia_intestinalis_GCA_000498735\") has been submitted\n",
      "Your job 5485009 (\"sim_Giardia_intestinalis_GCA_001493575\") has been submitted\n",
      "Your job 5485010 (\"sim_Giardia_intestinalis_GCA_001543975\") has been submitted\n",
      "Your job 5485011 (\"sim_Giardia_intestinalis_GCA_009192805\") has been submitted\n",
      "Your job 5485012 (\"sim_Giardia_intestinalis_GCA_009192825\") has been submitted\n",
      "Your job 5485013 (\"sim_Giardia_intestinalis_GCA_902209425\") has been submitted\n",
      "Your job 5485014 (\"sim_Giardia_intestinalis_GCA_902221465\") has been submitted\n",
      "Your job 5485015 (\"sim_Giardia_intestinalis_GCA_902221485\") has been submitted\n",
      "Your job 5485016 (\"sim_Giardia_intestinalis_GCA_902221515\") has been submitted\n",
      "Your job 5485017 (\"sim_Giardia_intestinalis_GCA_902221535\") has been submitted\n",
      "Your job 5485018 (\"sim_Giardia_intestinalis_GCA_902221545\") has been submitted\n",
      "Your job 5485019 (\"sim_Giardia_lamblia_GCA_000182665\") has been submitted\n",
      "Your job 5485020 (\"sim_Giardia_muris_GCA_006247105\") has been submitted\n",
      "Your job 5485021 (\"sim_Spironucleus_salmonicida_GCA_000497125\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/hexamitidae/cleaned_genomes/Giardia_intestinalis_GCA_000002435_formatted.fas path=results/phyluce/hexamitidae/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485022 (\"map_Giardia_intestinalis_GCA_000182405\") has been submitted\n",
      "Your job 5485023 (\"map_Giardia_intestinalis_GCA_000182665\") has been submitted\n",
      "Your job 5485024 (\"map_Giardia_intestinalis_GCA_000498715\") has been submitted\n",
      "Your job 5485025 (\"map_Giardia_intestinalis_GCA_000498735\") has been submitted\n",
      "Your job 5485026 (\"map_Giardia_intestinalis_GCA_001493575\") has been submitted\n",
      "Your job 5485027 (\"map_Giardia_intestinalis_GCA_001543975\") has been submitted\n",
      "Your job 5485028 (\"map_Giardia_intestinalis_GCA_009192805\") has been submitted\n",
      "Your job 5485029 (\"map_Giardia_intestinalis_GCA_009192825\") has been submitted\n",
      "Your job 5485030 (\"map_Giardia_intestinalis_GCA_902209425\") has been submitted\n",
      "Your job 5485031 (\"map_Giardia_intestinalis_GCA_902221465\") has been submitted\n",
      "Your job 5485032 (\"map_Giardia_intestinalis_GCA_902221485\") has been submitted\n",
      "Your job 5485033 (\"map_Giardia_intestinalis_GCA_902221515\") has been submitted\n",
      "Your job 5485034 (\"map_Giardia_intestinalis_GCA_902221535\") has been submitted\n",
      "Your job 5485035 (\"map_Giardia_intestinalis_GCA_902221545\") has been submitted\n",
      "Your job 5485036 (\"map_Giardia_lamblia_GCA_000182665\") has been submitted\n",
      "Your job 5485037 (\"map_Giardia_muris_GCA_006247105\") has been submitted\n",
      "Your job 5485038 (\"map_Spironucleus_salmonicida_GCA_000497125\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485039 (\"merge_Giardia_intestinalis_GCA_000182405\") has been submitted\n",
      "Your job 5485040 (\"merge_Giardia_intestinalis_GCA_000182665\") has been submitted\n",
      "Your job 5485041 (\"merge_Giardia_intestinalis_GCA_000498715\") has been submitted\n",
      "Your job 5485042 (\"merge_Giardia_intestinalis_GCA_000498735\") has been submitted\n",
      "Your job 5485043 (\"merge_Giardia_intestinalis_GCA_001493575\") has been submitted\n",
      "Your job 5485044 (\"merge_Giardia_intestinalis_GCA_001543975\") has been submitted\n",
      "Your job 5485045 (\"merge_Giardia_intestinalis_GCA_009192805\") has been submitted\n",
      "Your job 5485046 (\"merge_Giardia_intestinalis_GCA_009192825\") has been submitted\n",
      "Your job 5485047 (\"merge_Giardia_intestinalis_GCA_902209425\") has been submitted\n",
      "Your job 5485048 (\"merge_Giardia_intestinalis_GCA_902221465\") has been submitted\n",
      "Your job 5485049 (\"merge_Giardia_intestinalis_GCA_902221485\") has been submitted\n",
      "Your job 5485050 (\"merge_Giardia_intestinalis_GCA_902221515\") has been submitted\n",
      "Your job 5485051 (\"merge_Giardia_intestinalis_GCA_902221535\") has been submitted\n",
      "Your job 5485052 (\"merge_Giardia_intestinalis_GCA_902221545\") has been submitted\n",
      "Your job 5485053 (\"merge_Giardia_lamblia_GCA_000182665\") has been submitted\n",
      "Your job 5485054 (\"merge_Giardia_muris_GCA_006247105\") has been submitted\n",
      "Your job 5485055 (\"merge_Spironucleus_salmonicida_GCA_000497125\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 2565 sequences from Giardia_intestinalis_GCA_000182405_merged.bed.  Filtered 1939 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 626.\n",
      "Screened 21421 sequences from Giardia_intestinalis_GCA_000182665_merged.bed.  Filtered 8700 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12721.\n",
      "Screened 3667 sequences from Giardia_intestinalis_GCA_000498715_merged.bed.  Filtered 1763 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1904.\n",
      "Screened 2632 sequences from Giardia_intestinalis_GCA_000498735_merged.bed.  Filtered 1987 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 645.\n",
      "Screened 3610 sequences from Giardia_intestinalis_GCA_001493575_merged.bed.  Filtered 1662 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1948.\n",
      "Screened 2555 sequences from Giardia_intestinalis_GCA_001543975_merged.bed.  Filtered 1967 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 588.\n",
      "Screened 1631 sequences from Giardia_intestinalis_GCA_009192805_merged.bed.  Filtered 750 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 881.\n",
      "Screened 1572 sequences from Giardia_intestinalis_GCA_009192825_merged.bed.  Filtered 697 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 875.\n",
      "Screened 1522 sequences from Giardia_intestinalis_GCA_902209425_merged.bed.  Filtered 1162 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 360.\n",
      "Screened 699 sequences from Giardia_intestinalis_GCA_902221465_merged.bed.  Filtered 554 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 145.\n",
      "Screened 704 sequences from Giardia_intestinalis_GCA_902221485_merged.bed.  Filtered 571 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 133.\n",
      "Screened 1538 sequences from Giardia_intestinalis_GCA_902221515_merged.bed.  Filtered 1173 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 365.\n",
      "Screened 682 sequences from Giardia_intestinalis_GCA_902221535_merged.bed.  Filtered 539 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 143.\n",
      "Screened 1473 sequences from Giardia_intestinalis_GCA_902221545_merged.bed.  Filtered 1117 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 356.\n",
      "Screened 21446 sequences from Giardia_lamblia_GCA_000182665_merged.bed.  Filtered 8666 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12780.\n",
      "Screened 68 sequences from Giardia_muris_GCA_006247105_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 22.\n",
      "Screened 4 sequences from Spironucleus_salmonicida_GCA_000497125_merged.bed.  Filtered 3 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/hexamitidae/initial_intervals/Spironucleus_salmonicida_GCA_000497125_merged.bed --twobit results/phyluce/hexamitidae/cleaned_genomes/Giardia_intestinalis_GCA_000002435_formatted.2bit --output results/phyluce/hexamitidae/initial_intervals/Spironucleus_salmonicida_GCA_000497125_stripped.bed;\n",
      "giardia_intestinalis_gca_000182405.\n",
      "giardia_intestinalis_gca_000182665.............\n",
      "giardia_intestinalis_gca_000498715..\n",
      "giardia_intestinalis_gca_000498735.\n",
      "giardia_intestinalis_gca_001493575..\n",
      "giardia_intestinalis_gca_001543975.\n",
      "giardia_intestinalis_gca_009192805.\n",
      "giardia_intestinalis_gca_009192825.\n",
      "giardia_intestinalis_gca_902209425.\n",
      "giardia_intestinalis_gca_902221465.\n",
      "giardia_intestinalis_gca_902221485.\n",
      "giardia_intestinalis_gca_902221515.\n",
      "giardia_intestinalis_gca_902221535.\n",
      "giardia_intestinalis_gca_902221545.\n",
      "giardia_lamblia_gca_000182665.............\n",
      "giardia_muris_gca_006247105.\n",
      "spironucleus_salmonicida_gca_000497125.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/hexamitidae/initial_intervals/hexamitidae-to-Giardia_intestinalis_GCA_000002435.sqlite --base-taxon Giardia_intestinalis_GCA_000002435\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 0 taxa:\t15,148.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 1 taxa:\t15,148.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 2 taxa:\t14,275.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 3 taxa:\t13,949.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 4 taxa:\t13,569.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 5 taxa:\t12,664.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 6 taxa:\t12,069.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 7 taxa:\t856.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 8 taxa:\t698.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 9 taxa:\t552.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 10 taxa:\t256.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 11 taxa:\t222.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 12 taxa:\t170.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 13 taxa:\t90.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 14 taxa:\t74.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 15 taxa:\t62.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 16 taxa:\t9.0\n",
      "Loci shared by Giardia_intestinalis_GCA_000002435 + 17 taxa:\t1.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/hexamitidae/initial_intervals/hexamitidae-to-Giardia_intestinalis_GCA_000002435.sqlite --base-taxon Giardia_intestinalis_GCA_000002435 --output results/phyluce/hexamitidae/initial_intervals/Giardia_intestinalis_GCA_000002435_+7.bed --specific-counts 7;\n",
      "Counter({'giardia_lamblia_gca_000182665': 848, 'giardia_intestinalis_gca_001493575': 846, 'giardia_intestinalis_gca_000182665': 835, 'giardia_intestinalis_gca_009192825': 832, 'giardia_intestinalis_gca_000498715': 831, 'giardia_intestinalis_gca_009192805': 829, 'giardia_intestinalis_gca_000498735': 600, 'giardia_intestinalis_gca_000182405': 598, 'giardia_intestinalis_gca_001543975': 549, 'giardia_intestinalis_gca_902221545': 324, 'giardia_intestinalis_gca_902221515': 324, 'giardia_intestinalis_gca_902209425': 316, 'giardia_intestinalis_gca_902221535': 127, 'giardia_intestinalis_gca_902221465': 126, 'giardia_intestinalis_gca_902221485': 122, 'giardia_muris_gca_006247105': 18, 'spironucleus_salmonicida_gca_000497125': 1})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 7\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/hexamitidae/initial_intervals/Giardia_intestinalis_GCA_000002435_+7.bed --twobit results/phyluce/hexamitidae/cleaned_genomes/Giardia_intestinalis_GCA_000002435_formatted.2bit --buffer-to 160 --output results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7.fasta;\n",
      "Screened 856 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 856.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7.fasta --probe-prefix uce_hexamitidae_ --design hexamitidae_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 832\n",
      "Probe Count = 1652\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas --query results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes_vself.lastz;\n",
      "Started:  Mon Feb 10, 2020  13:10:04\n",
      "Ended:  Mon Feb 10, 2020  13:10:05\n",
      "Time for execution:  0.0107145984968 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas                        --lastz results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes_vself.lastz                       --probe-prefix=uce_hexamitidae_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 1651 fasta sequences.  Filtered 43 duplicates. Kept 1570.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/hexamitidae/validate_intervals/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas --scaffoldlist Giardia_intestinalis_GCA_000182405 Giardia_intestinalis_GCA_000182665 Giardia_intestinalis_GCA_000498715 Giardia_intestinalis_GCA_000498735 Giardia_intestinalis_GCA_001493575 Giardia_intestinalis_GCA_001543975 Giardia_intestinalis_GCA_009192805 Giardia_intestinalis_GCA_009192825 Giardia_intestinalis_GCA_902209425 Giardia_intestinalis_GCA_902221465 Giardia_intestinalis_GCA_902221485 Giardia_intestinalis_GCA_902221515 Giardia_intestinalis_GCA_902221535 Giardia_intestinalis_GCA_902221545 Giardia_lamblia_GCA_000182665 Giardia_muris_GCA_006247105 Spironucleus_salmonicida_GCA_000497125 Giardia_intestinalis_GCA_000002435 --genome-base-path results/phyluce/hexamitidae/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/hexamitidae/validate_intervals/hexamitidae-to-Giardia_intestinalis_GCA_000002435.sqlite --output results/phyluce/hexamitidae/validate_intervals/lastz/;\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_000182405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpL8LEhz.fasta\n",
      "\t/tmp/tmpjJfbC7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6jX7cJ.lastz\n",
      "\t/tmp/tmp4_mUdD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_000182405.lastz\n",
      "Creating Giardia_intestinalis_GCA_000182405 table\n",
      "Inserting data to Giardia_intestinalis_GCA_000182405 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_000182665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpENxEbj.fasta\n",
      "\t/tmp/tmphBixql.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiZZgJA.lastz\n",
      "\t/tmp/tmpDTJ6wa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_000182665.lastz\n",
      "Creating Giardia_intestinalis_GCA_000182665 table\n",
      "Inserting data to Giardia_intestinalis_GCA_000182665 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_000498715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpn0BrF1.fasta\n",
      "\t/tmp/tmpG7AAR3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmi6Tm6.lastz\n",
      "\t/tmp/tmpaOJU5z.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_000498715.lastz\n",
      "Creating Giardia_intestinalis_GCA_000498715 table\n",
      "Inserting data to Giardia_intestinalis_GCA_000498715 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_000498735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpowrQLx.fasta\n",
      "\t/tmp/tmpj68yx3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSQzmmc.lastz\n",
      "\t/tmp/tmpQX4WU2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_000498735.lastz\n",
      "Creating Giardia_intestinalis_GCA_000498735 table\n",
      "Inserting data to Giardia_intestinalis_GCA_000498735 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_001493575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpAeMb4O.fasta\n",
      "\t/tmp/tmpN_i0dm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXnAvwU.lastz\n",
      "\t/tmp/tmpsiO6K6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_001493575.lastz\n",
      "Creating Giardia_intestinalis_GCA_001493575 table\n",
      "Inserting data to Giardia_intestinalis_GCA_001493575 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_001543975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpaPQOxb.fasta\n",
      "\t/tmp/tmpG3I6K5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXu8l9p.lastz\n",
      "\t/tmp/tmpLiC7hM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_001543975.lastz\n",
      "Creating Giardia_intestinalis_GCA_001543975 table\n",
      "Inserting data to Giardia_intestinalis_GCA_001543975 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_009192805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp4iCua8.fasta\n",
      "\t/tmp/tmpD0LVhn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEwRGlo.lastz\n",
      "\t/tmp/tmpBYsxeB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_009192805.lastz\n",
      "Creating Giardia_intestinalis_GCA_009192805 table\n",
      "Inserting data to Giardia_intestinalis_GCA_009192805 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_009192825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpmKZwHH.fasta\n",
      "\t/tmp/tmpsrzvZF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyRwEIK.lastz\n",
      "\t/tmp/tmpI7mErS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_009192825.lastz\n",
      "Creating Giardia_intestinalis_GCA_009192825 table\n",
      "Inserting data to Giardia_intestinalis_GCA_009192825 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_902209425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpACFEwJ.fasta\n",
      "\t/tmp/tmpAybQPE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaaiGRm.lastz\n",
      "\t/tmp/tmpQlP83M.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_902209425.lastz\n",
      "Creating Giardia_intestinalis_GCA_902209425 table\n",
      "Inserting data to Giardia_intestinalis_GCA_902209425 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_902221465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpjuOK8f.fasta\n",
      "\t/tmp/tmphoplDt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgdLxLz.lastz\n",
      "\t/tmp/tmpXvwMwI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_902221465.lastz\n",
      "Creating Giardia_intestinalis_GCA_902221465 table\n",
      "Inserting data to Giardia_intestinalis_GCA_902221465 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_902221485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpG5yvO0.fasta\n",
      "\t/tmp/tmpvBeT5s.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuxW2_A.lastz\n",
      "\t/tmp/tmpIREhHb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_902221485.lastz\n",
      "Creating Giardia_intestinalis_GCA_902221485 table\n",
      "Inserting data to Giardia_intestinalis_GCA_902221485 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_902221515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpNBVcDC.fasta\n",
      "\t/tmp/tmp3mwzlN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl5J86w.lastz\n",
      "\t/tmp/tmpn_qcP3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_902221515.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Giardia_intestinalis_GCA_902221515 table\n",
      "Inserting data to Giardia_intestinalis_GCA_902221515 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_902221535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpq8npOX.fasta\n",
      "\t/tmp/tmpRQIbRX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIwMAu4.lastz\n",
      "\t/tmp/tmpaU5v65.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_902221535.lastz\n",
      "Creating Giardia_intestinalis_GCA_902221535 table\n",
      "Inserting data to Giardia_intestinalis_GCA_902221535 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_902221545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp8vQ9lk.fasta\n",
      "\t/tmp/tmpF9rkSC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE2NG4F.lastz\n",
      "\t/tmp/tmpVf0y7K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_902221545.lastz\n",
      "Creating Giardia_intestinalis_GCA_902221545 table\n",
      "Inserting data to Giardia_intestinalis_GCA_902221545 table\n",
      "\n",
      "Running against Giardia_lamblia_GCA_000182665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpfXTmjn.fasta\n",
      "\t/tmp/tmpynF8tn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4vOxjf.lastz\n",
      "\t/tmp/tmpvVbHAz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_lamblia_GCA_000182665.lastz\n",
      "Creating Giardia_lamblia_GCA_000182665 table\n",
      "Inserting data to Giardia_lamblia_GCA_000182665 table\n",
      "\n",
      "Running against Giardia_muris_GCA_006247105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpXRuv6g.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPWfcoi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_muris_GCA_006247105.lastz\n",
      "Creating Giardia_muris_GCA_006247105 table\n",
      "Inserting data to Giardia_muris_GCA_006247105 table\n",
      "\n",
      "Running against Spironucleus_salmonicida_GCA_000497125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpOiGZs2.fasta\n",
      "\t/tmp/tmp6ypAZb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOJQELk.lastz\n",
      "\t/tmp/tmpXvx80X.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Spironucleus_salmonicida_GCA_000497125.lastz\n",
      "Creating Spironucleus_salmonicida_GCA_000497125 table\n",
      "Inserting data to Spironucleus_salmonicida_GCA_000497125 table\n",
      "\n",
      "Running against Giardia_intestinalis_GCA_000002435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpsf3uAE.fasta\n",
      "\t/tmp/tmpI2MsUD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpj5VX7s.lastz\n",
      "\t/tmp/tmpWH7R2T.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/hexamitidae/validate_intervals/lastz/Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_Giardia_intestinalis_GCA_000002435.lastz\n",
      "Creating Giardia_intestinalis_GCA_000002435 table\n",
      "Inserting data to Giardia_intestinalis_GCA_000002435 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/hexamitidae/extract_probes_from_group/hexamitidae_genome.conf --lastz results/phyluce/hexamitidae/validate_intervals/lastz --probes 120 --probe-prefix uce_hexamitidae_ --name-pattern \"Giardia_intestinalis_GCA_000002435_+7_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/hexamitidae/extract_probes_from_group/probe_fasta;\n",
      "2020-02-10 13:11:43,097 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-10 13:11:43,100 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_000182405 genome -----\n",
      "2020-02-10 13:11:43,101 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_000182405 genome\n",
      "2020-02-10 13:11:46,703 - Phyluce - INFO - Giardia_intestinalis_GCA_000182405: 831 uces, 108 dupes, 723 non-dupes, 1 orient drop, 2 length drop, 720 written\n",
      "2020-02-10 13:11:46,703 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_000182665 genome -----\n",
      "2020-02-10 13:11:46,704 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_000182665 genome\n",
      "2020-02-10 13:11:50,054 - Phyluce - INFO - Giardia_intestinalis_GCA_000182665: 832 uces, 126 dupes, 706 non-dupes, 3 orient drop, 2 length drop, 701 written\n",
      "2020-02-10 13:11:50,054 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_000498715 genome -----\n",
      "2020-02-10 13:11:50,055 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_000498715 genome\n",
      "2020-02-10 13:11:53,288 - Phyluce - INFO - Giardia_intestinalis_GCA_000498715: 823 uces, 99 dupes, 724 non-dupes, 17 orient drop, 5 length drop, 702 written\n",
      "2020-02-10 13:11:53,288 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_000498735 genome -----\n",
      "2020-02-10 13:11:53,289 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_000498735 genome\n",
      "2020-02-10 13:11:56,654 - Phyluce - INFO - Giardia_intestinalis_GCA_000498735: 825 uces, 123 dupes, 702 non-dupes, 3 orient drop, 0 length drop, 699 written\n",
      "2020-02-10 13:11:56,654 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_001493575 genome -----\n",
      "2020-02-10 13:11:56,654 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_001493575 genome\n",
      "2020-02-10 13:11:59,847 - Phyluce - INFO - Giardia_intestinalis_GCA_001493575: 827 uces, 109 dupes, 718 non-dupes, 5 orient drop, 2 length drop, 711 written\n",
      "2020-02-10 13:11:59,848 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_001543975 genome -----\n",
      "2020-02-10 13:11:59,848 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_001543975 genome\n",
      "2020-02-10 13:12:02,920 - Phyluce - INFO - Giardia_intestinalis_GCA_001543975: 827 uces, 83 dupes, 744 non-dupes, 2 orient drop, 2 length drop, 740 written\n",
      "2020-02-10 13:12:02,920 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_009192805 genome -----\n",
      "2020-02-10 13:12:02,921 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_009192805 genome\n",
      "2020-02-10 13:12:05,876 - Phyluce - INFO - Giardia_intestinalis_GCA_009192805: 813 uces, 79 dupes, 734 non-dupes, 3 orient drop, 3 length drop, 728 written\n",
      "2020-02-10 13:12:05,876 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_009192825 genome -----\n",
      "2020-02-10 13:12:05,877 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_009192825 genome\n",
      "2020-02-10 13:12:08,852 - Phyluce - INFO - Giardia_intestinalis_GCA_009192825: 816 uces, 80 dupes, 736 non-dupes, 3 orient drop, 1 length drop, 732 written\n",
      "2020-02-10 13:12:08,853 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_902209425 genome -----\n",
      "2020-02-10 13:12:08,854 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_902209425 genome\n",
      "2020-02-10 13:12:12,056 - Phyluce - INFO - Giardia_intestinalis_GCA_902209425: 826 uces, 101 dupes, 725 non-dupes, 2 orient drop, 1 length drop, 722 written\n",
      "2020-02-10 13:12:12,056 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_902221465 genome -----\n",
      "2020-02-10 13:12:12,057 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_902221465 genome\n",
      "2020-02-10 13:12:15,259 - Phyluce - INFO - Giardia_intestinalis_GCA_902221465: 830 uces, 81 dupes, 749 non-dupes, 0 orient drop, 2 length drop, 747 written\n",
      "2020-02-10 13:12:15,260 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_902221485 genome -----\n",
      "2020-02-10 13:12:15,265 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_902221485 genome\n",
      "2020-02-10 13:12:18,368 - Phyluce - INFO - Giardia_intestinalis_GCA_902221485: 829 uces, 78 dupes, 751 non-dupes, 0 orient drop, 2 length drop, 749 written\n",
      "2020-02-10 13:12:18,369 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_902221515 genome -----\n",
      "2020-02-10 13:12:18,378 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_902221515 genome\n",
      "2020-02-10 13:12:21,587 - Phyluce - INFO - Giardia_intestinalis_GCA_902221515: 829 uces, 96 dupes, 733 non-dupes, 1 orient drop, 1 length drop, 731 written\n",
      "2020-02-10 13:12:21,587 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_902221535 genome -----\n",
      "2020-02-10 13:12:21,588 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_902221535 genome\n",
      "2020-02-10 13:12:24,954 - Phyluce - INFO - Giardia_intestinalis_GCA_902221535: 831 uces, 87 dupes, 744 non-dupes, 0 orient drop, 1 length drop, 743 written\n",
      "2020-02-10 13:12:24,955 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_902221545 genome -----\n",
      "2020-02-10 13:12:24,955 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_902221545 genome\n",
      "2020-02-10 13:12:28,144 - Phyluce - INFO - Giardia_intestinalis_GCA_902221545: 818 uces, 102 dupes, 716 non-dupes, 1 orient drop, 1 length drop, 714 written\n",
      "2020-02-10 13:12:28,144 - Phyluce - INFO - -------- Working on Giardia_lamblia_GCA_000182665 genome --------\n",
      "2020-02-10 13:12:28,154 - Phyluce - INFO - Reading Giardia_lamblia_GCA_000182665 genome\n",
      "2020-02-10 13:12:31,536 - Phyluce - INFO - Giardia_lamblia_GCA_000182665: 832 uces, 126 dupes, 706 non-dupes, 3 orient drop, 2 length drop, 701 written\n",
      "2020-02-10 13:12:31,536 - Phyluce - INFO - --------- Working on Giardia_muris_GCA_006247105 genome ---------\n",
      "2020-02-10 13:12:31,537 - Phyluce - INFO - Reading Giardia_muris_GCA_006247105 genome\n",
      "2020-02-10 13:12:33,387 - Phyluce - INFO - Giardia_muris_GCA_006247105: 501 uces, 59 dupes, 442 non-dupes, 12 orient drop, 5 length drop, 425 written\n",
      "2020-02-10 13:12:33,387 - Phyluce - INFO - ---- Working on Spironucleus_salmonicida_GCA_000497125 genome ---\n",
      "2020-02-10 13:12:33,388 - Phyluce - INFO - Reading Spironucleus_salmonicida_GCA_000497125 genome\n",
      "2020-02-10 13:12:33,970 - Phyluce - INFO - Spironucleus_salmonicida_GCA_000497125: 152 uces, 39 dupes, 113 non-dupes, 7 orient drop, 1 length drop, 105 written\n",
      "2020-02-10 13:12:33,971 - Phyluce - INFO - ------ Working on Giardia_intestinalis_GCA_000002435 genome -----\n",
      "2020-02-10 13:12:33,971 - Phyluce - INFO - Reading Giardia_intestinalis_GCA_000002435 genome\n",
      "2020-02-10 13:12:37,369 - Phyluce - INFO - Giardia_intestinalis_GCA_000002435: 832 uces, 82 dupes, 750 non-dupes, 40 orient drop, 17 length drop, 693 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/hexamitidae/extract_probes_from_group/probe_fasta --output results/phyluce/hexamitidae/extract_probes_from_group/multifastas.sqlite --base-taxon Giardia_intestinalis_GCA_000002435;\n",
      "giardia_intestinalis_gca_000182405.\n",
      "giardia_intestinalis_gca_000182665.\n",
      "giardia_intestinalis_gca_000498715.\n",
      "giardia_intestinalis_gca_000498735.\n",
      "giardia_intestinalis_gca_001493575.\n",
      "giardia_intestinalis_gca_001543975.\n",
      "giardia_intestinalis_gca_009192805.\n",
      "giardia_intestinalis_gca_009192825.\n",
      "giardia_intestinalis_gca_902209425.\n",
      "giardia_intestinalis_gca_902221465.\n",
      "giardia_intestinalis_gca_902221485.\n",
      "giardia_intestinalis_gca_902221515.\n",
      "giardia_intestinalis_gca_902221535.\n",
      "giardia_intestinalis_gca_902221545.\n",
      "giardia_lamblia_gca_000182665.\n",
      "giardia_muris_gca_006247105.\n",
      "spironucleus_salmonicida_gca_000497125.\n",
      "giardia_intestinalis_gca_000002435.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/hexamitidae/extract_probes_from_group/multifastas.sqlite --base-taxon Giardia_intestinalis_GCA_000002435;\n",
      "Loci shared by 0 taxa:\t808.0\n",
      "Loci shared by 1 taxa:\t808.0\n",
      "Loci shared by 2 taxa:\t786.0\n",
      "Loci shared by 3 taxa:\t778.0\n",
      "Loci shared by 4 taxa:\t772.0\n",
      "Loci shared by 5 taxa:\t762.0\n",
      "Loci shared by 6 taxa:\t753.0\n",
      "Loci shared by 7 taxa:\t744.0\n",
      "Loci shared by 8 taxa:\t735.0\n",
      "Loci shared by 9 taxa:\t727.0\n",
      "Loci shared by 10 taxa:\t714.0\n",
      "Loci shared by 11 taxa:\t704.0\n",
      "Loci shared by 12 taxa:\t691.0\n",
      "Loci shared by 13 taxa:\t685.0\n",
      "Loci shared by 14 taxa:\t673.0\n",
      "Loci shared by 15 taxa:\t655.0\n",
      "Loci shared by 16 taxa:\t639.0\n",
      "Loci shared by 17 taxa:\t363.0\n",
      "Loci shared by 18 taxa:\t74.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/hexamitidae/extract_probes_from_group/multifastas.sqlite  --base-taxon Giardia_intestinalis_GCA_000002435 --output results/phyluce/hexamitidae/extract_probes_from_group/Giardia_intestinalis_GCA_000002435+7-back-to-18.conf --specific-counts 18;\n",
      "Counter({'giardia_intestinalis_gca_902221465': 74, 'giardia_intestinalis_gca_000498715': 74, 'giardia_intestinalis_gca_902221485': 74, 'giardia_intestinalis_gca_000182665': 74, 'giardia_intestinalis_gca_009192825': 74, 'giardia_muris_gca_006247105': 74, 'giardia_intestinalis_gca_902221535': 74, 'giardia_intestinalis_gca_001543975': 74, 'giardia_intestinalis_gca_001493575': 74, 'giardia_intestinalis_gca_902221545': 74, 'giardia_intestinalis_gca_000002435': 74, 'giardia_intestinalis_gca_000182405': 74, 'giardia_intestinalis_gca_902209425': 74, 'giardia_intestinalis_gca_902221515': 74, 'giardia_lamblia_gca_000182665': 74, 'giardia_intestinalis_gca_000498735': 74, 'giardia_intestinalis_gca_009192805': 74, 'spironucleus_salmonicida_gca_000497125': 74})\n",
      "Total loci = 74\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(18)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/hexamitidae/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/hexamitidae/extract_probes_from_group/Giardia_intestinalis_GCA_000002435+7-back-to-18.conf --probe-prefix uce_hexamitidae_ --designer rnplattii --design hexamitidae_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list.fasta;\n",
      "GGNNGNNGGGNNG\n",
      "\n",
      "\n",
      "Conserved locus count = 74\n",
      "Probe Count = 2651\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list.fasta --query results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  12:18:12\n",
      "Ended:  Tue Feb 11, 2020  12:18:17\n",
      "Time for execution:  0.0834663192431 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list.fasta --lastz results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_hexamitidae_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 2650 fasta sequences.  Filtered 0 duplicates. Kept 2651.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/hexamitidae/final_probe_design/hexamitidae_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 12:20:08 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 2651\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 212080\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3960486\n",
      "Max number of word counting entries: 89150560\n",
      "\n",
      "# comparing sequences from          0  to        441\n",
      "---------- new table with      204 representatives\n",
      "# comparing sequences from        441  to        809\n",
      "----------    151 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from        658  to        990\n",
      "----------     30 remaining sequences to the next cycle\n",
      "---------- new table with      138 representatives\n",
      "# comparing sequences from        960  to       1241\n",
      "100.0%---------- new table with      132 representatives\n",
      "# comparing sequences from       1241  to       1476\n",
      "----------     22 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       1454  to       1653\n",
      "99.9%---------- new table with       90 representatives\n",
      "# comparing sequences from       1653  to       2651\n",
      ".....................---------- new table with      477 representatives\n",
      "\n",
      "     2651  finished       1241  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.67\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Kinetoplastea"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'kinetoplastea'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Angomonas_deanei_GCA_001659865\",\n",
    "               \"Angomonas_desouzai_GCA_000482185\",\n",
    "               \"Bodo_saltans_GCA_001460835\",\n",
    "               \"Crithidia_acanthocephali_GCA_000482105\",\n",
    "               \"Crithidia_bombi_GCA_900240985\",\n",
    "               \"Crithidia_expoeki_GCA_900240875\",\n",
    "               \"Crithidia_fasciculata_GCA_000331325\",\n",
    "               \"Crithidia_mellificae_GCA_002216565\",\n",
    "               \"Endotrypanum_monterogeii_GCA_000333855\",\n",
    "               \"Herpetomonas_muscarum_GCA_000482205\",\n",
    "               \"Leishmania_aethiopica_GCA_000444285\",\n",
    "               \"Leishmania_amazonensis_GCA_005317125\",\n",
    "               \"Leishmania_arabica_GCA_000410695\",\n",
    "               \"Leishmania_braziliensis_GCF_000002845\",\n",
    "               \"Leishmania_donovani_GCF_000227135\",\n",
    "               \"Leishmania_enriettii_GCA_000410755\",\n",
    "               \"Leishmania_gerbilli_GCA_000443025\",\n",
    "               \"Leishmania_infantum_GCF_000002875\",\n",
    "               \"Leishmania_lainsoni_GCA_003664395\",\n",
    "               \"Leishmania_mexicana_GCF_000234665\",\n",
    "               \"Leishmania_panamensis_GCF_000755165\",\n",
    "               \"Leishmania_peruviana_GCA_001403675\",\n",
    "               \"Leishmania_sp_GCA_000409445\",\n",
    "               \"Leishmania_tarentolae_GCA_009731335\",\n",
    "               \"Leishmania_tropica_GCA_000410715\",\n",
    "               \"Leishmania_turanica_GCA_000441995\",\n",
    "               \"Leptomonas_pyrrhocoris_GCF_001293395\",\n",
    "               \"Leptomonas_seymouri_GCA_001299535\",\n",
    "               \"Lotmaria_passim_GCA_000635995\",\n",
    "               \"Paratrypanosoma_confusum_GCA_002921335\",\n",
    "               \"Perkinsela_sp_GCA_001235845\",\n",
    "               \"Phytomonas_francai_GCA_001766655\",\n",
    "               \"Phytomonas_serpens_GCA_000331125\",\n",
    "               \"Phytomonas_sp_GCA_000582765\",\n",
    "               \"Strigomonas_culicis_GCA_000482145\",\n",
    "               \"Strigomonas_galati_GCA_000482125\",\n",
    "               \"Strigomonas_oncopelti_GCA_000482165\",\n",
    "               \"Trypanosoma_brucei_GCF_000210295\",\n",
    "               \"Trypanosoma_congolense_GCA_002287245\",\n",
    "               \"Trypanosoma_conorhini_GCF_003719485\",\n",
    "               \"Trypanosoma_cruzi_GCF_000209065\",\n",
    "               \"Trypanosoma_equiperdum_GCA_001457755\",\n",
    "               \"Trypanosoma_grayi_GCF_000691245\",\n",
    "               \"Trypanosoma_rangeli_GCF_003719475\",\n",
    "               \"Trypanosoma_theileri_GCF_002087225\",\n",
    "               \"Trypanosoma_vivax_GCA_000227375\",\n",
    "               \"Trypanosomatidae_sp_GCA_003671325\" ]\n",
    "                    \n",
    "reference_taxon = \"Leishmania_major_GCF_000002725\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001659865.1_Angomonas_deanei_v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5704401 bytes  2281777.20 bytes/sec\n",
      "total size is 5702885  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000482185.1_Ades_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7890725 bytes  3156306.80 bytes/sec\n",
      "total size is 7888694  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001460835.1_BSAL_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12198374 bytes  3485261.71 bytes/sec\n",
      "total size is 12195290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000482105.1_Caca_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11043288 bytes  7362220.00 bytes/sec\n",
      "total size is 11040488  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900240985.1_crithidia-bombi.GDC.2013.v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10268745 bytes  4107514.80 bytes/sec\n",
      "total size is 10266110  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900240875.1_crithidia-expoeki.GDC.2015.v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9965260 bytes  3986120.80 bytes/sec\n",
      "total size is 9962695  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000331325.2_Crithidia_fasciculata-14.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12875302 bytes  3678669.71 bytes/sec\n",
      "total size is 12872036  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002216565.1_ASM221656v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15385879 bytes  4395977.43 bytes/sec\n",
      "total size is 15382012  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000333855.2_Endotrypanum_monterogeii-LV88-1.0.3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10201035 bytes  4080430.80 bytes/sec\n",
      "total size is 10198408  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000482205.1_Hmus_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10174654 bytes  4069878.40 bytes/sec\n",
      "total size is 10172062  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_902369305.1_Lad_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6014 bytes  1101.09 bytes/sec\n",
      "total size is 5900  speedup is 0.97\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000444285.2_Leishmania_aethiopica-L147-2.0.3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9904650 bytes  2829912.00 bytes/sec\n",
      "total size is 9902098  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005317125.1_ASM531712v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10144638 bytes  4057872.00 bytes/sec\n",
      "total size is 10142051  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000410695.2_Leishmania_arabica_LEM1108-1.0.3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9834178 bytes  3933688.00 bytes/sec\n",
      "total size is 9831642  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002845.2_ASM284v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9996336 bytes  3998551.20 bytes/sec\n",
      "total size is 9993792  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000227135.2_ASM22713v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9965348 bytes  3986156.00 bytes/sec\n",
      "total size is 9962802  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000410755.2_Leishmania_enrietti_LEM3045-1.0.2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9696890 bytes  3878772.80 bytes/sec\n",
      "total size is 9694393  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000443025.1_Leishmania_gerbilii_LEM452-1.0.2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9875164 bytes  2821487.43 bytes/sec\n",
      "total size is 9872620  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902369315.1_Lgu_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6133 bytes  1372.22 bytes/sec\n",
      "total size is 6019  speedup is 0.97\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002875.2_ASM287v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10058500 bytes  2873869.14 bytes/sec\n",
      "total size is 10055940  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003664395.1_CDC_Llain_216-34_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10259758 bytes  2931371.43 bytes/sec\n",
      "total size is 10257131  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000002725.2_ASM272v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10077710 bytes  4031100.80 bytes/sec\n",
      "total size is 10075142  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000234665.4_ASM23466v4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10052517 bytes  4021023.60 bytes/sec\n",
      "total size is 10049955  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000755165.1_ASM75516v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9640717 bytes  3856303.60 bytes/sec\n",
      "total size is 9638251  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001403675.1_Leishmania_peruviana_PAB-4377_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9647014 bytes  2756301.71 bytes/sec\n",
      "total size is 9644526  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000409445.2_Leishmania_MAR_LEM2494-1.0.3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9740215 bytes  2782930.57 bytes/sec\n",
      "total size is 9737707  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009731335.1_Lta_assembly01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10444805 bytes  2984242.00 bytes/sec\n",
      "total size is 10442143  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000410715.1_Leishmania_tropica_L590-2.0.2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10067908 bytes  2876557.14 bytes/sec\n",
      "total size is 10065319  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000441995.1_Leishmania_turanica_LEM423-1.0.2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9923740 bytes  2835366.29 bytes/sec\n",
      "total size is 9921188  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001293395.1_ASM129339v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9460684 bytes  3784290.40 bytes/sec\n",
      "total size is 9458265  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001299535.1_ASM129953v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8974129 bytes  3589668.40 bytes/sec\n",
      "total size is 8971830  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000635995.1_ASM63599v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10591871 bytes  3026260.86 bytes/sec\n",
      "total size is 10589173  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002921335.1_ASM292133v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8214696 bytes  3285895.20 bytes/sec\n",
      "total size is 8212582  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001235845.1_ASM123584v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2861076 bytes  1907412.00 bytes/sec\n",
      "total size is 2860266  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001766655.1_ASM176665v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5537173 bytes  2214886.00 bytes/sec\n",
      "total size is 5535715  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000331125.1_PhytSerpensv01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8381554 bytes  3352638.40 bytes/sec\n",
      "total size is 8379397  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000582765.1_AKH_PRJEB1535_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5207235 bytes  2082910.80 bytes/sec\n",
      "total size is 5205852  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000482145.1_Scul_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7654265 bytes  3061722.80 bytes/sec\n",
      "total size is 7652290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000482125.1_Sgal_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8899476 bytes  2542719.43 bytes/sec\n",
      "total size is 8897196  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000482165.1_Sonc_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8155752 bytes  2330226.86 bytes/sec\n",
      "total size is 8153657  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000210295.1_ASM21029v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6763209 bytes  1932357.43 bytes/sec\n",
      "total size is 6761448  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002287245.1_ASM228724v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11189331 bytes  4475749.20 bytes/sec\n",
      "total size is 11186488  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003719485.1_ASM371948v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6796481 bytes  2718609.20 bytes/sec\n",
      "total size is 6794711  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000209065.1_ASM20906v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 26710398 bytes  5935653.33 bytes/sec\n",
      "total size is 26703772  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001457755.2_Trypanosoma_equiperdum_OVI_V2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8486624 bytes  3394666.40 bytes/sec\n",
      "total size is 8484427  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000691245.1_Tgr_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6692933 bytes  4461983.33 bytes/sec\n",
      "total size is 6691192  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003719475.1_ASM371947v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6675731 bytes  2670309.20 bytes/sec\n",
      "total size is 6673993  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002087225.1_Tth_v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8149087 bytes  3259651.60 bytes/sec\n",
      "total size is 8146994  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000227375.1_ASM22737v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7855181 bytes  3142089.20 bytes/sec\n",
      "total size is 7853156  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003671325.1_LVH60_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16752753 bytes  4786512.86 bytes/sec\n",
      "total size is 16748556  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487555 (\"sim_Angomonas_deanei_GCA_001659865\") has been submitted\n",
      "Your job 5487556 (\"sim_Angomonas_desouzai_GCA_000482185\") has been submitted\n",
      "Your job 5487557 (\"sim_Bodo_saltans_GCA_001460835\") has been submitted\n",
      "Your job 5487558 (\"sim_Crithidia_acanthocephali_GCA_000482105\") has been submitted\n",
      "Your job 5487559 (\"sim_Crithidia_bombi_GCA_900240985\") has been submitted\n",
      "Your job 5487560 (\"sim_Crithidia_expoeki_GCA_900240875\") has been submitted\n",
      "Your job 5487561 (\"sim_Crithidia_fasciculata_GCA_000331325\") has been submitted\n",
      "Your job 5487562 (\"sim_Crithidia_mellificae_GCA_002216565\") has been submitted\n",
      "Your job 5487563 (\"sim_Endotrypanum_monterogeii_GCA_000333855\") has been submitted\n",
      "Your job 5487564 (\"sim_Herpetomonas_muscarum_GCA_000482205\") has been submitted\n",
      "Your job 5487565 (\"sim_Leishmania_adleri_GCA_902369305\") has been submitted\n",
      "Your job 5487566 (\"sim_Leishmania_aethiopica_GCA_000444285\") has been submitted\n",
      "Your job 5487567 (\"sim_Leishmania_amazonensis_GCA_005317125\") has been submitted\n",
      "Your job 5487568 (\"sim_Leishmania_arabica_GCA_000410695\") has been submitted\n",
      "Your job 5487569 (\"sim_Leishmania_braziliensis_GCF_000002845\") has been submitted\n",
      "Your job 5487570 (\"sim_Leishmania_donovani_GCF_000227135\") has been submitted\n",
      "Your job 5487571 (\"sim_Leishmania_enriettii_GCA_000410755\") has been submitted\n",
      "Your job 5487572 (\"sim_Leishmania_gerbilli_GCA_000443025\") has been submitted\n",
      "Your job 5487573 (\"sim_Leishmania_guyanensis_GCA_902369315\") has been submitted\n",
      "Your job 5487574 (\"sim_Leishmania_infantum_GCF_000002875\") has been submitted\n",
      "Your job 5487575 (\"sim_Leishmania_lainsoni_GCA_003664395\") has been submitted\n",
      "Your job 5487576 (\"sim_Leishmania_major_GCF_000002725\") has been submitted\n",
      "Your job 5487577 (\"sim_Leishmania_mexicana_GCF_000234665\") has been submitted\n",
      "Your job 5487578 (\"sim_Leishmania_panamensis_GCF_000755165\") has been submitted\n",
      "Your job 5487579 (\"sim_Leishmania_peruviana_GCA_001403675\") has been submitted\n",
      "Your job 5487580 (\"sim_Leishmania_sp_GCA_000409445\") has been submitted\n",
      "Your job 5487581 (\"sim_Leishmania_tarentolae_GCA_009731335\") has been submitted\n",
      "Your job 5487582 (\"sim_Leishmania_tropica_GCA_000410715\") has been submitted\n",
      "Your job 5487583 (\"sim_Leishmania_turanica_GCA_000441995\") has been submitted\n",
      "Your job 5487584 (\"sim_Leptomonas_pyrrhocoris_GCF_001293395\") has been submitted\n",
      "Your job 5487585 (\"sim_Leptomonas_seymouri_GCA_001299535\") has been submitted\n",
      "Your job 5487586 (\"sim_Lotmaria_passim_GCA_000635995\") has been submitted\n",
      "Your job 5487587 (\"sim_Paratrypanosoma_confusum_GCA_002921335\") has been submitted\n",
      "Your job 5487588 (\"sim_Perkinsela_sp_GCA_001235845\") has been submitted\n",
      "Your job 5487589 (\"sim_Phytomonas_francai_GCA_001766655\") has been submitted\n",
      "Your job 5487590 (\"sim_Phytomonas_serpens_GCA_000331125\") has been submitted\n",
      "Your job 5487591 (\"sim_Phytomonas_sp_GCA_000582765\") has been submitted\n",
      "Your job 5487592 (\"sim_Strigomonas_culicis_GCA_000482145\") has been submitted\n",
      "Your job 5487593 (\"sim_Strigomonas_galati_GCA_000482125\") has been submitted\n",
      "Your job 5487594 (\"sim_Strigomonas_oncopelti_GCA_000482165\") has been submitted\n",
      "Your job 5487595 (\"sim_Trypanosoma_brucei_GCF_000210295\") has been submitted\n",
      "Your job 5487596 (\"sim_Trypanosoma_congolense_GCA_002287245\") has been submitted\n",
      "Your job 5487597 (\"sim_Trypanosoma_conorhini_GCF_003719485\") has been submitted\n",
      "Your job 5487598 (\"sim_Trypanosoma_cruzi_GCF_000209065\") has been submitted\n",
      "Your job 5487599 (\"sim_Trypanosoma_equiperdum_GCA_001457755\") has been submitted\n",
      "Your job 5487600 (\"sim_Trypanosoma_grayi_GCF_000691245\") has been submitted\n",
      "Your job 5487601 (\"sim_Trypanosoma_rangeli_GCF_003719475\") has been submitted\n",
      "Your job 5487602 (\"sim_Trypanosoma_theileri_GCF_002087225\") has been submitted\n",
      "Your job 5487603 (\"sim_Trypanosoma_vivax_GCA_000227375\") has been submitted\n",
      "Your job 5487604 (\"sim_Trypanosomatidae_sp_GCA_003671325\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/kinetoplastea/cleaned_genomes/Leishmania_major_GCF_000002725_formatted.fas path=results/phyluce/kinetoplastea/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 279,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487605 (\"map_Angomonas_deanei_GCA_001659865\") has been submitted\n",
      "Your job 5487606 (\"map_Angomonas_desouzai_GCA_000482185\") has been submitted\n",
      "Your job 5487607 (\"map_Bodo_saltans_GCA_001460835\") has been submitted\n",
      "Your job 5487608 (\"map_Crithidia_acanthocephali_GCA_000482105\") has been submitted\n",
      "Your job 5487609 (\"map_Crithidia_bombi_GCA_900240985\") has been submitted\n",
      "Your job 5487610 (\"map_Crithidia_expoeki_GCA_900240875\") has been submitted\n",
      "Your job 5487611 (\"map_Crithidia_fasciculata_GCA_000331325\") has been submitted\n",
      "Your job 5487612 (\"map_Crithidia_mellificae_GCA_002216565\") has been submitted\n",
      "Your job 5487613 (\"map_Endotrypanum_monterogeii_GCA_000333855\") has been submitted\n",
      "Your job 5487614 (\"map_Herpetomonas_muscarum_GCA_000482205\") has been submitted\n",
      "Your job 5487615 (\"map_Leishmania_adleri_GCA_902369305\") has been submitted\n",
      "Your job 5487616 (\"map_Leishmania_aethiopica_GCA_000444285\") has been submitted\n",
      "Your job 5487617 (\"map_Leishmania_amazonensis_GCA_005317125\") has been submitted\n",
      "Your job 5487618 (\"map_Leishmania_arabica_GCA_000410695\") has been submitted\n",
      "Your job 5487619 (\"map_Leishmania_braziliensis_GCF_000002845\") has been submitted\n",
      "Your job 5487620 (\"map_Leishmania_donovani_GCF_000227135\") has been submitted\n",
      "Your job 5487621 (\"map_Leishmania_enriettii_GCA_000410755\") has been submitted\n",
      "Your job 5487622 (\"map_Leishmania_gerbilli_GCA_000443025\") has been submitted\n",
      "Your job 5487623 (\"map_Leishmania_guyanensis_GCA_902369315\") has been submitted\n",
      "Your job 5487624 (\"map_Leishmania_infantum_GCF_000002875\") has been submitted\n",
      "Your job 5487625 (\"map_Leishmania_lainsoni_GCA_003664395\") has been submitted\n",
      "Your job 5487626 (\"map_Leishmania_major_GCF_000002725\") has been submitted\n",
      "Your job 5487627 (\"map_Leishmania_mexicana_GCF_000234665\") has been submitted\n",
      "Your job 5487628 (\"map_Leishmania_panamensis_GCF_000755165\") has been submitted\n",
      "Your job 5487629 (\"map_Leishmania_peruviana_GCA_001403675\") has been submitted\n",
      "Your job 5487630 (\"map_Leishmania_sp_GCA_000409445\") has been submitted\n",
      "Your job 5487631 (\"map_Leishmania_tarentolae_GCA_009731335\") has been submitted\n",
      "Your job 5487632 (\"map_Leishmania_tropica_GCA_000410715\") has been submitted\n",
      "Your job 5487633 (\"map_Leishmania_turanica_GCA_000441995\") has been submitted\n",
      "Your job 5487634 (\"map_Leptomonas_pyrrhocoris_GCF_001293395\") has been submitted\n",
      "Your job 5487635 (\"map_Leptomonas_seymouri_GCA_001299535\") has been submitted\n",
      "Your job 5487636 (\"map_Lotmaria_passim_GCA_000635995\") has been submitted\n",
      "Your job 5487637 (\"map_Paratrypanosoma_confusum_GCA_002921335\") has been submitted\n",
      "Your job 5487638 (\"map_Perkinsela_sp_GCA_001235845\") has been submitted\n",
      "Your job 5487639 (\"map_Phytomonas_francai_GCA_001766655\") has been submitted\n",
      "Your job 5487640 (\"map_Phytomonas_serpens_GCA_000331125\") has been submitted\n",
      "Your job 5487641 (\"map_Phytomonas_sp_GCA_000582765\") has been submitted\n",
      "Your job 5487642 (\"map_Strigomonas_culicis_GCA_000482145\") has been submitted\n",
      "Your job 5487643 (\"map_Strigomonas_galati_GCA_000482125\") has been submitted\n",
      "Your job 5487644 (\"map_Strigomonas_oncopelti_GCA_000482165\") has been submitted\n",
      "Your job 5487645 (\"map_Trypanosoma_brucei_GCF_000210295\") has been submitted\n",
      "Your job 5487646 (\"map_Trypanosoma_congolense_GCA_002287245\") has been submitted\n",
      "Your job 5487647 (\"map_Trypanosoma_conorhini_GCF_003719485\") has been submitted\n",
      "Your job 5487648 (\"map_Trypanosoma_cruzi_GCF_000209065\") has been submitted\n",
      "Your job 5487649 (\"map_Trypanosoma_equiperdum_GCA_001457755\") has been submitted\n",
      "Your job 5487650 (\"map_Trypanosoma_grayi_GCF_000691245\") has been submitted\n",
      "Your job 5487651 (\"map_Trypanosoma_rangeli_GCF_003719475\") has been submitted\n",
      "Your job 5487652 (\"map_Trypanosoma_theileri_GCF_002087225\") has been submitted\n",
      "Your job 5487653 (\"map_Trypanosoma_vivax_GCA_000227375\") has been submitted\n",
      "Your job 5487654 (\"map_Trypanosomatidae_sp_GCA_003671325\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487655 (\"merge_Angomonas_deanei_GCA_001659865\") has been submitted\n",
      "Your job 5487656 (\"merge_Angomonas_desouzai_GCA_000482185\") has been submitted\n",
      "Your job 5487657 (\"merge_Bodo_saltans_GCA_001460835\") has been submitted\n",
      "Your job 5487658 (\"merge_Crithidia_acanthocephali_GCA_000482105\") has been submitted\n",
      "Your job 5487659 (\"merge_Crithidia_bombi_GCA_900240985\") has been submitted\n",
      "Your job 5487660 (\"merge_Crithidia_expoeki_GCA_900240875\") has been submitted\n",
      "Your job 5487661 (\"merge_Crithidia_fasciculata_GCA_000331325\") has been submitted\n",
      "Your job 5487662 (\"merge_Crithidia_mellificae_GCA_002216565\") has been submitted\n",
      "Your job 5487663 (\"merge_Endotrypanum_monterogeii_GCA_000333855\") has been submitted\n",
      "Your job 5487664 (\"merge_Herpetomonas_muscarum_GCA_000482205\") has been submitted\n",
      "Your job 5487665 (\"merge_Leishmania_adleri_GCA_902369305\") has been submitted\n",
      "Your job 5487666 (\"merge_Leishmania_aethiopica_GCA_000444285\") has been submitted\n",
      "Your job 5487667 (\"merge_Leishmania_amazonensis_GCA_005317125\") has been submitted\n",
      "Your job 5487668 (\"merge_Leishmania_arabica_GCA_000410695\") has been submitted\n",
      "Your job 5487669 (\"merge_Leishmania_braziliensis_GCF_000002845\") has been submitted\n",
      "Your job 5487670 (\"merge_Leishmania_donovani_GCF_000227135\") has been submitted\n",
      "Your job 5487671 (\"merge_Leishmania_enriettii_GCA_000410755\") has been submitted\n",
      "Your job 5487672 (\"merge_Leishmania_gerbilli_GCA_000443025\") has been submitted\n",
      "Your job 5487673 (\"merge_Leishmania_guyanensis_GCA_902369315\") has been submitted\n",
      "Your job 5487674 (\"merge_Leishmania_infantum_GCF_000002875\") has been submitted\n",
      "Your job 5487675 (\"merge_Leishmania_lainsoni_GCA_003664395\") has been submitted\n",
      "Your job 5487676 (\"merge_Leishmania_major_GCF_000002725\") has been submitted\n",
      "Your job 5487677 (\"merge_Leishmania_mexicana_GCF_000234665\") has been submitted\n",
      "Your job 5487678 (\"merge_Leishmania_panamensis_GCF_000755165\") has been submitted\n",
      "Your job 5487679 (\"merge_Leishmania_peruviana_GCA_001403675\") has been submitted\n",
      "Your job 5487680 (\"merge_Leishmania_sp_GCA_000409445\") has been submitted\n",
      "Your job 5487681 (\"merge_Leishmania_tarentolae_GCA_009731335\") has been submitted\n",
      "Your job 5487682 (\"merge_Leishmania_tropica_GCA_000410715\") has been submitted\n",
      "Your job 5487683 (\"merge_Leishmania_turanica_GCA_000441995\") has been submitted\n",
      "Your job 5487684 (\"merge_Leptomonas_pyrrhocoris_GCF_001293395\") has been submitted\n",
      "Your job 5487685 (\"merge_Leptomonas_seymouri_GCA_001299535\") has been submitted\n",
      "Your job 5487686 (\"merge_Lotmaria_passim_GCA_000635995\") has been submitted\n",
      "Your job 5487687 (\"merge_Paratrypanosoma_confusum_GCA_002921335\") has been submitted\n",
      "Your job 5487688 (\"merge_Perkinsela_sp_GCA_001235845\") has been submitted\n",
      "Your job 5487689 (\"merge_Phytomonas_francai_GCA_001766655\") has been submitted\n",
      "Your job 5487690 (\"merge_Phytomonas_serpens_GCA_000331125\") has been submitted\n",
      "Your job 5487691 (\"merge_Phytomonas_sp_GCA_000582765\") has been submitted\n",
      "Your job 5487692 (\"merge_Strigomonas_culicis_GCA_000482145\") has been submitted\n",
      "Your job 5487693 (\"merge_Strigomonas_galati_GCA_000482125\") has been submitted\n",
      "Your job 5487694 (\"merge_Strigomonas_oncopelti_GCA_000482165\") has been submitted\n",
      "Your job 5487695 (\"merge_Trypanosoma_brucei_GCF_000210295\") has been submitted\n",
      "Your job 5487696 (\"merge_Trypanosoma_congolense_GCA_002287245\") has been submitted\n",
      "Your job 5487697 (\"merge_Trypanosoma_conorhini_GCF_003719485\") has been submitted\n",
      "Your job 5487698 (\"merge_Trypanosoma_cruzi_GCF_000209065\") has been submitted\n",
      "Your job 5487699 (\"merge_Trypanosoma_equiperdum_GCA_001457755\") has been submitted\n",
      "Your job 5487700 (\"merge_Trypanosoma_grayi_GCF_000691245\") has been submitted\n",
      "Your job 5487701 (\"merge_Trypanosoma_rangeli_GCF_003719475\") has been submitted\n",
      "Your job 5487702 (\"merge_Trypanosoma_theileri_GCF_002087225\") has been submitted\n",
      "Your job 5487703 (\"merge_Trypanosoma_vivax_GCA_000227375\") has been submitted\n",
      "Your job 5487704 (\"merge_Trypanosomatidae_sp_GCA_003671325\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 210 sequences from Angomonas_deanei_GCA_001659865_merged.bed.  Filtered 170 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 40.\n",
      "Screened 258 sequences from Angomonas_desouzai_GCA_000482185_merged.bed.  Filtered 206 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 163 sequences from Bodo_saltans_GCA_001460835_merged.bed.  Filtered 121 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 42.\n",
      "Screened 3876 sequences from Crithidia_acanthocephali_GCA_000482105_merged.bed.  Filtered 2691 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1185.\n",
      "Screened 2409 sequences from Crithidia_bombi_GCA_900240985_merged.bed.  Filtered 1791 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 618.\n",
      "Screened 2022 sequences from Crithidia_expoeki_GCA_900240875_merged.bed.  Filtered 1535 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 487.\n",
      "Screened 3696 sequences from Crithidia_fasciculata_GCA_000331325_merged.bed.  Filtered 2652 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1044.\n",
      "Screened 3373 sequences from Crithidia_mellificae_GCA_002216565_merged.bed.  Filtered 2345 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1028.\n",
      "Screened 4345 sequences from Endotrypanum_monterogeii_GCA_000333855_merged.bed.  Filtered 3269 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1076.\n",
      "Screened 1030 sequences from Herpetomonas_muscarum_GCA_000482205_merged.bed.  Filtered 789 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 241.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 27010 sequences from Leishmania_aethiopica_GCA_000444285_merged.bed.  Filtered 11724 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15286.\n",
      "Screened 37361 sequences from Leishmania_amazonensis_GCA_005317125_merged.bed.  Filtered 17288 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 20073.\n",
      "Screened 21554 sequences from Leishmania_arabica_GCA_000410695_merged.bed.  Filtered 9117 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12437.\n",
      "Screened 23032 sequences from Leishmania_braziliensis_GCF_000002845_merged.bed.  Filtered 12458 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10574.\n",
      "Screened 44670 sequences from Leishmania_donovani_GCF_000227135_merged.bed.  Filtered 21846 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 22824.\n",
      "Screened 18754 sequences from Leishmania_enriettii_GCA_000410755_merged.bed.  Filtered 10895 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7859.\n",
      "Screened 17591 sequences from Leishmania_gerbilli_GCA_000443025_merged.bed.  Filtered 7445 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10146.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 45143 sequences from Leishmania_infantum_GCF_000002875_merged.bed.  Filtered 22037 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 23106.\n",
      "Screened 21637 sequences from Leishmania_lainsoni_GCA_003664395_merged.bed.  Filtered 11968 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9669.\n",
      "Screened 2792 sequences from Leishmania_major_GCF_000002725_merged.bed.  Filtered 1219 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1573.\n",
      "Screened 37704 sequences from Leishmania_mexicana_GCF_000234665_merged.bed.  Filtered 17612 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 20092.\n",
      "Screened 22754 sequences from Leishmania_panamensis_GCF_000755165_merged.bed.  Filtered 12261 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10493.\n",
      "Screened 21831 sequences from Leishmania_peruviana_GCA_001403675_merged.bed.  Filtered 11867 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9964.\n",
      "Screened 16348 sequences from Leishmania_sp_GCA_000409445_merged.bed.  Filtered 9822 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6526.\n",
      "Screened 27556 sequences from Leishmania_tarentolae_GCA_009731335_merged.bed.  Filtered 13696 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13860.\n",
      "Screened 26170 sequences from Leishmania_tropica_GCA_000410715_merged.bed.  Filtered 11256 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 14914.\n",
      "Screened 20082 sequences from Leishmania_turanica_GCA_000441995_merged.bed.  Filtered 8568 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11514.\n",
      "Screened 2830 sequences from Leptomonas_pyrrhocoris_GCF_001293395_merged.bed.  Filtered 2028 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 802.\n",
      "Screened 1935 sequences from Leptomonas_seymouri_GCA_001299535_merged.bed.  Filtered 1429 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 506.\n",
      "Screened 2898 sequences from Lotmaria_passim_GCA_000635995_merged.bed.  Filtered 2102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 796.\n",
      "Screened 286 sequences from Paratrypanosoma_confusum_GCA_002921335_merged.bed.  Filtered 215 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 71.\n",
      "Screened 21 sequences from Perkinsela_sp_GCA_001235845_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 272 sequences from Phytomonas_francai_GCA_001766655_merged.bed.  Filtered 198 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 74.\n",
      "Screened 236 sequences from Phytomonas_serpens_GCA_000331125_merged.bed.  Filtered 178 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 58.\n",
      "Screened 188 sequences from Phytomonas_sp_GCA_000582765_merged.bed.  Filtered 145 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 957 sequences from Strigomonas_culicis_GCA_000482145_merged.bed.  Filtered 712 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 245.\n",
      "Screened 824 sequences from Strigomonas_galati_GCA_000482125_merged.bed.  Filtered 633 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 191.\n",
      "Screened 901 sequences from Strigomonas_oncopelti_GCA_000482165_merged.bed.  Filtered 694 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 207.\n",
      "Screened 67 sequences from Trypanosoma_brucei_GCF_000210295_merged.bed.  Filtered 51 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 16.\n",
      "Screened 134 sequences from Trypanosoma_congolense_GCA_002287245_merged.bed.  Filtered 110 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 24.\n",
      "Screened 341 sequences from Trypanosoma_conorhini_GCF_003719485_merged.bed.  Filtered 244 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 97.\n",
      "Screened 210 sequences from Trypanosoma_cruzi_GCF_000209065_merged.bed.  Filtered 154 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 56.\n",
      "Screened 56 sequences from Trypanosoma_equiperdum_GCA_001457755_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 342 sequences from Trypanosoma_grayi_GCF_000691245_merged.bed.  Filtered 264 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 78.\n",
      "Screened 227 sequences from Trypanosoma_rangeli_GCF_003719475_merged.bed.  Filtered 178 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 49.\n",
      "Screened 136 sequences from Trypanosoma_theileri_GCF_002087225_merged.bed.  Filtered 99 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 37.\n",
      "Screened 63 sequences from Trypanosoma_vivax_GCA_000227375_merged.bed.  Filtered 50 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 3786 sequences from Trypanosomatidae_sp_GCA_003671325_merged.bed.  Filtered 2684 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1102.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/kinetoplastea/initial_intervals/Trypanosomatidae_sp_GCA_003671325_merged.bed --twobit results/phyluce/kinetoplastea/cleaned_genomes/Leishmania_major_GCF_000002725_formatted.2bit --output results/phyluce/kinetoplastea/initial_intervals/Trypanosomatidae_sp_GCA_003671325_stripped.bed;\n",
      "angomonas_deanei_gca_001659865.\n",
      "angomonas_desouzai_gca_000482185.\n",
      "bodo_saltans_gca_001460835.\n",
      "crithidia_acanthocephali_gca_000482105..\n",
      "crithidia_bombi_gca_900240985.\n",
      "crithidia_expoeki_gca_900240875.\n",
      "crithidia_fasciculata_gca_000331325..\n",
      "crithidia_mellificae_gca_002216565..\n",
      "endotrypanum_monterogeii_gca_000333855..\n",
      "herpetomonas_muscarum_gca_000482205.\n",
      "leishmania_adleri_gca_902369305\n",
      "leishmania_aethiopica_gca_000444285................\n",
      "leishmania_amazonensis_gca_005317125.....................\n",
      "leishmania_arabica_gca_000410695.............\n",
      "leishmania_braziliensis_gcf_000002845...........\n",
      "leishmania_donovani_gcf_000227135.......................\n",
      "leishmania_enriettii_gca_000410755........\n",
      "leishmania_gerbilli_gca_000443025...........\n",
      "leishmania_guyanensis_gca_902369315\n",
      "leishmania_infantum_gcf_000002875........................\n",
      "leishmania_lainsoni_gca_003664395..........\n",
      "leishmania_major_gcf_000002725..\n",
      "leishmania_mexicana_gcf_000234665.....................\n",
      "leishmania_panamensis_gcf_000755165...........\n",
      "leishmania_peruviana_gca_001403675..........\n",
      "leishmania_sp_gca_000409445.......\n",
      "leishmania_tarentolae_gca_009731335..............\n",
      "leishmania_tropica_gca_000410715...............\n",
      "leishmania_turanica_gca_000441995............\n",
      "leptomonas_pyrrhocoris_gcf_001293395.\n",
      "leptomonas_seymouri_gca_001299535.\n",
      "lotmaria_passim_gca_000635995.\n",
      "paratrypanosoma_confusum_gca_002921335.\n",
      "perkinsela_sp_gca_001235845.\n",
      "phytomonas_francai_gca_001766655.\n",
      "phytomonas_serpens_gca_000331125.\n",
      "phytomonas_sp_gca_000582765.\n",
      "strigomonas_culicis_gca_000482145.\n",
      "strigomonas_galati_gca_000482125.\n",
      "strigomonas_oncopelti_gca_000482165.\n",
      "trypanosoma_brucei_gcf_000210295.\n",
      "trypanosoma_congolense_gca_002287245.\n",
      "trypanosoma_conorhini_gcf_003719485.\n",
      "trypanosoma_cruzi_gcf_000209065.\n",
      "trypanosoma_equiperdum_gca_001457755.\n",
      "trypanosoma_grayi_gcf_000691245.\n",
      "trypanosoma_rangeli_gcf_003719475.\n",
      "trypanosoma_theileri_gcf_002087225.\n",
      "trypanosoma_vivax_gca_000227375.\n",
      "trypanosomatidae_sp_gca_003671325..\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/kinetoplastea/initial_intervals/kinetoplastea-to-Leishmania_major_GCF_000002725.sqlite --base-taxon Leishmania_major_GCF_000002725\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 0 taxa:\t26,983.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 1 taxa:\t26,983.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 2 taxa:\t25,971.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 3 taxa:\t24,871.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 4 taxa:\t23,301.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 5 taxa:\t21,705.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 6 taxa:\t19,962.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 7 taxa:\t17,058.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 8 taxa:\t15,509.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 9 taxa:\t12,350.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 10 taxa:\t10,801.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 11 taxa:\t8,754.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 12 taxa:\t7,691.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 13 taxa:\t6,876.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 14 taxa:\t6,132.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 15 taxa:\t5,191.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 16 taxa:\t4,390.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 17 taxa:\t3,352.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 18 taxa:\t1,922.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 19 taxa:\t1,361.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 20 taxa:\t1,087.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 21 taxa:\t881.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 22 taxa:\t695.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 23 taxa:\t568.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 24 taxa:\t460.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 25 taxa:\t375.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 26 taxa:\t295.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 27 taxa:\t238.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 28 taxa:\t178.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 29 taxa:\t129.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 30 taxa:\t107.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 31 taxa:\t82.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 32 taxa:\t63.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 33 taxa:\t56.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 34 taxa:\t47.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 35 taxa:\t36.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 36 taxa:\t27.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 37 taxa:\t22.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 38 taxa:\t17.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 39 taxa:\t14.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 40 taxa:\t12.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 41 taxa:\t11.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 42 taxa:\t10.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 43 taxa:\t3.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 44 taxa:\t3.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 45 taxa:\t2.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 46 taxa:\t0.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 47 taxa:\t0.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 48 taxa:\t0.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 49 taxa:\t0.0\n",
      "Loci shared by Leishmania_major_GCF_000002725 + 50 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/kinetoplastea/initial_intervals/kinetoplastea-to-Leishmania_major_GCF_000002725.sqlite --base-taxon Leishmania_major_GCF_000002725 --output results/phyluce/kinetoplastea/initial_intervals/Leishmania_major_GCF_000002725_+17.bed --specific-counts 17;\n",
      "Counter({'leishmania_donovani_gcf_000227135': 3332, 'leishmania_infantum_gcf_000002875': 3331, 'leishmania_mexicana_gcf_000234665': 3330, 'leishmania_amazonensis_gca_005317125': 3320, 'leishmania_tarentolae_gca_009731335': 3313, 'leishmania_panamensis_gcf_000755165': 3308, 'leishmania_arabica_gca_000410695': 3304, 'leishmania_turanica_gca_000441995': 3301, 'leishmania_gerbilli_gca_000443025': 3294, 'leishmania_aethiopica_gca_000444285': 3293, 'leishmania_braziliensis_gcf_000002845': 3292, 'leishmania_tropica_gca_000410715': 3286, 'leishmania_lainsoni_gca_003664395': 3278, 'leishmania_peruviana_gca_001403675': 3266, 'leishmania_enriettii_gca_000410755': 3232, 'leishmania_major_gcf_000002725': 3222, 'leishmania_sp_gca_000409445': 3145, 'crithidia_acanthocephali_gca_000482105': 1117, 'trypanosomatidae_sp_gca_003671325': 1034, 'crithidia_mellificae_gca_002216565': 1012, 'crithidia_fasciculata_gca_000331325': 1006, 'endotrypanum_monterogeii_gca_000333855': 994, 'lotmaria_passim_gca_000635995': 758, 'leptomonas_pyrrhocoris_gcf_001293395': 756, 'crithidia_bombi_gca_900240985': 587, 'leptomonas_seymouri_gca_001299535': 506, 'crithidia_expoeki_gca_900240875': 462, 'strigomonas_culicis_gca_000482145': 241, 'herpetomonas_muscarum_gca_000482205': 241, 'strigomonas_oncopelti_gca_000482165': 215, 'strigomonas_galati_gca_000482125': 194, 'trypanosoma_conorhini_gcf_003719485': 88, 'trypanosoma_grayi_gcf_000691245': 74, 'paratrypanosoma_confusum_gca_002921335': 69, 'phytomonas_francai_gca_001766655': 68, 'phytomonas_serpens_gca_000331125': 56, 'angomonas_desouzai_gca_000482185': 52, 'trypanosoma_cruzi_gcf_000209065': 48, 'trypanosoma_rangeli_gcf_003719475': 47, 'phytomonas_sp_gca_000582765': 40, 'angomonas_deanei_gca_001659865': 39, 'trypanosoma_theileri_gcf_002087225': 39, 'bodo_saltans_gca_001460835': 37, 'trypanosoma_congolense_gca_002287245': 19, 'trypanosoma_brucei_gcf_000210295': 15, 'trypanosoma_equiperdum_gca_001457755': 11, 'trypanosoma_vivax_gca_000227375': 9, 'perkinsela_sp_gca_001235845': 4})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 17\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/kinetoplastea/initial_intervals/Leishmania_major_GCF_000002725_+17.bed --twobit results/phyluce/kinetoplastea/cleaned_genomes/Leishmania_major_GCF_000002725_formatted.2bit --buffer-to 160 --output results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17.fasta;\n",
      "Screened 3352 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 3352.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17.fasta --probe-prefix uce_kinetoplastea_ --design kinetoplastea_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 3326\n",
      "Probe Count = 6579\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes.fas --query results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  10:45:14\n",
      "Ended:  Tue Feb 11, 2020  10:45:18\n",
      "Time for execution:  0.0593067526817 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes.fas                        --lastz results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes_vself.lastz                       --probe-prefix=uce_kinetoplastea_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6578 fasta sequences.  Filtered 298 duplicates. Kept 5989.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/kinetoplastea/validate_intervals/Leishmania_major_GCF_000002725_+17_temp_probes.fas --scaffoldlist Angomonas_deanei_GCA_001659865 Angomonas_desouzai_GCA_000482185 Bodo_saltans_GCA_001460835 Crithidia_acanthocephali_GCA_000482105 Crithidia_bombi_GCA_900240985 Crithidia_expoeki_GCA_900240875 Crithidia_fasciculata_GCA_000331325 Crithidia_mellificae_GCA_002216565 Endotrypanum_monterogeii_GCA_000333855 Herpetomonas_muscarum_GCA_000482205 Leishmania_aethiopica_GCA_000444285 Leishmania_amazonensis_GCA_005317125 Leishmania_arabica_GCA_000410695 Leishmania_braziliensis_GCF_000002845 Leishmania_donovani_GCF_000227135 Leishmania_enriettii_GCA_000410755 Leishmania_gerbilli_GCA_000443025 Leishmania_infantum_GCF_000002875 Leishmania_lainsoni_GCA_003664395 Leishmania_mexicana_GCF_000234665 Leishmania_panamensis_GCF_000755165 Leishmania_peruviana_GCA_001403675 Leishmania_sp_GCA_000409445 Leishmania_tarentolae_GCA_009731335 Leishmania_tropica_GCA_000410715 Leishmania_turanica_GCA_000441995 Leptomonas_pyrrhocoris_GCF_001293395 Leptomonas_seymouri_GCA_001299535 Lotmaria_passim_GCA_000635995 Paratrypanosoma_confusum_GCA_002921335 Perkinsela_sp_GCA_001235845 Phytomonas_francai_GCA_001766655 Phytomonas_serpens_GCA_000331125 Phytomonas_sp_GCA_000582765 Strigomonas_culicis_GCA_000482145 Strigomonas_galati_GCA_000482125 Strigomonas_oncopelti_GCA_000482165 Trypanosoma_brucei_GCF_000210295 Trypanosoma_congolense_GCA_002287245 Trypanosoma_conorhini_GCF_003719485 Trypanosoma_cruzi_GCF_000209065 Trypanosoma_equiperdum_GCA_001457755 Trypanosoma_grayi_GCF_000691245 Trypanosoma_rangeli_GCF_003719475 Trypanosoma_theileri_GCF_002087225 Trypanosoma_vivax_GCA_000227375 Trypanosomatidae_sp_GCA_003671325 Leishmania_major_GCF_000002725 --genome-base-path results/phyluce/kinetoplastea/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/kinetoplastea/validate_intervals/kinetoplastea-to-Leishmania_major_GCF_000002725.sqlite --output results/phyluce/kinetoplastea/validate_intervals/lastz/;\n",
      "\n",
      "Running against Angomonas_deanei_GCA_001659865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpYcQzzw.fasta\n",
      "\t/tmp/tmpeNafXY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUF5CQm.lastz\n",
      "\t/tmp/tmp2lw89F.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Angomonas_deanei_GCA_001659865.lastz\n",
      "Creating Angomonas_deanei_GCA_001659865 table\n",
      "Inserting data to Angomonas_deanei_GCA_001659865 table\n",
      "\n",
      "Running against Angomonas_desouzai_GCA_000482185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpERAVI0.fasta\n",
      "\t/tmp/tmpNfLUkO.fasta\n",
      "\t/tmp/tmp6lOHzk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp43s3ZR.lastz\n",
      "\t/tmp/tmprRc9qu.lastz\n",
      "\t/tmp/tmp6J2_fZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Angomonas_desouzai_GCA_000482185.lastz\n",
      "Creating Angomonas_desouzai_GCA_000482185 table\n",
      "Inserting data to Angomonas_desouzai_GCA_000482185 table\n",
      "\n",
      "Running against Bodo_saltans_GCA_001460835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpqaMoQH.fasta\n",
      "\t/tmp/tmpVuAufU.fasta\n",
      "\t/tmp/tmpDijbqz.fasta\n",
      "\t/tmp/tmpoiyq5J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0pmQJW.lastz\n",
      "\t/tmp/tmpyX9H2Z.lastz\n",
      "\t/tmp/tmpIA0TZ3.lastz\n",
      "\t/tmp/tmpG1aEFP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Bodo_saltans_GCA_001460835.lastz\n",
      "Creating Bodo_saltans_GCA_001460835 table\n",
      "Inserting data to Bodo_saltans_GCA_001460835 table\n",
      "\n",
      "Running against Crithidia_acanthocephali_GCA_000482105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpHT8CJQ.fasta\n",
      "\t/tmp/tmpD6hR8I.fasta\n",
      "\t/tmp/tmpMjc2VX.fasta\n",
      "\t/tmp/tmpIjOVVS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE1zKky.lastz\n",
      "\t/tmp/tmpbbuH9n.lastz\n",
      "\t/tmp/tmp7UBBQU.lastz\n",
      "\t/tmp/tmp0BucGt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Crithidia_acanthocephali_GCA_000482105.lastz\n",
      "Creating Crithidia_acanthocephali_GCA_000482105 table\n",
      "Inserting data to Crithidia_acanthocephali_GCA_000482105 table\n",
      "\n",
      "Running against Crithidia_bombi_GCA_900240985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpMKGNHt.fasta\n",
      "\t/tmp/tmpH6Z_F5.fasta\n",
      "\t/tmp/tmps4PoqR.fasta\n",
      "\t/tmp/tmptraliy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiqzSUX.lastz\n",
      "\t/tmp/tmpaY5z9U.lastz\n",
      "\t/tmp/tmpTVEPzB.lastz\n",
      "\t/tmp/tmplNwHdH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Crithidia_bombi_GCA_900240985.lastz\n",
      "Creating Crithidia_bombi_GCA_900240985 table\n",
      "Inserting data to Crithidia_bombi_GCA_900240985 table\n",
      "\n",
      "Running against Crithidia_expoeki_GCA_900240875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpJM8goI.fasta\n",
      "\t/tmp/tmphQw4F1.fasta\n",
      "\t/tmp/tmpqHaVnY.fasta\n",
      "\t/tmp/tmpk5WFOT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdTH1a7.lastz\n",
      "\t/tmp/tmp9blcTv.lastz\n",
      "\t/tmp/tmplDCyFz.lastz\n",
      "\t/tmp/tmpZJtVcm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Crithidia_expoeki_GCA_900240875.lastz\n",
      "Creating Crithidia_expoeki_GCA_900240875 table\n",
      "Inserting data to Crithidia_expoeki_GCA_900240875 table\n",
      "\n",
      "Running against Crithidia_fasciculata_GCA_000331325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpygnJxu.fasta\n",
      "\t/tmp/tmpqgRq_6.fasta\n",
      "\t/tmp/tmpiJHP0p.fasta\n",
      "\t/tmp/tmp7x7nfN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjjj1WU.lastz\n",
      "\t/tmp/tmpswwsm8.lastz\n",
      "\t/tmp/tmp7dzRmd.lastz\n",
      "\t/tmp/tmp1umrNe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Crithidia_fasciculata_GCA_000331325.lastz\n",
      "Creating Crithidia_fasciculata_GCA_000331325 table\n",
      "Inserting data to Crithidia_fasciculata_GCA_000331325 table\n",
      "\n",
      "Running against Crithidia_mellificae_GCA_002216565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpMf8IhO.fasta\n",
      "\t/tmp/tmpI7YdE7.fasta\n",
      "\t/tmp/tmpb9z4lp.fasta\n",
      "\t/tmp/tmplq9NER.fasta\n",
      "\t/tmp/tmpWY1Yat.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzm3bjz.lastz\n",
      "\t/tmp/tmpbUgoIK.lastz\n",
      "\t/tmp/tmpPO7gNd.lastz\n",
      "\t/tmp/tmpdNcvjt.lastz\n",
      "\t/tmp/tmp_vORem.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Crithidia_mellificae_GCA_002216565.lastz\n",
      "Creating Crithidia_mellificae_GCA_002216565 table\n",
      "Inserting data to Crithidia_mellificae_GCA_002216565 table\n",
      "\n",
      "Running against Endotrypanum_monterogeii_GCA_000333855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmptk5SpU.fasta\n",
      "\t/tmp/tmpiHAVRS.fasta\n",
      "\t/tmp/tmp3O31Et.fasta\n",
      "\t/tmp/tmpcUvlDV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCJnD4u.lastz\n",
      "\t/tmp/tmp65cPDi.lastz\n",
      "\t/tmp/tmpEPFWiZ.lastz\n",
      "\t/tmp/tmpvbGB57.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Endotrypanum_monterogeii_GCA_000333855.lastz\n",
      "Creating Endotrypanum_monterogeii_GCA_000333855 table\n",
      "Inserting data to Endotrypanum_monterogeii_GCA_000333855 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Herpetomonas_muscarum_GCA_000482205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp3PqTwS.fasta\n",
      "\t/tmp/tmp0vm7qw.fasta\n",
      "\t/tmp/tmp9FEl1w.fasta\n",
      "\t/tmp/tmpgROPbK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKRkokO.lastz\n",
      "\t/tmp/tmpgqIFjI.lastz\n",
      "\t/tmp/tmp_1wwYA.lastz\n",
      "\t/tmp/tmpyQ1jjM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Herpetomonas_muscarum_GCA_000482205.lastz\n",
      "Creating Herpetomonas_muscarum_GCA_000482205 table\n",
      "Inserting data to Herpetomonas_muscarum_GCA_000482205 table\n",
      "\n",
      "Running against Leishmania_aethiopica_GCA_000444285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpPICDD6.fasta\n",
      "\t/tmp/tmpOu0FaX.fasta\n",
      "\t/tmp/tmpwt5k07.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHn9tgc.lastz\n",
      "\t/tmp/tmpzrQ64H.lastz\n",
      "\t/tmp/tmph0PoOV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_aethiopica_GCA_000444285.lastz\n",
      "Creating Leishmania_aethiopica_GCA_000444285 table\n",
      "Inserting data to Leishmania_aethiopica_GCA_000444285 table\n",
      "\n",
      "Running against Leishmania_amazonensis_GCA_005317125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpSRHkhF.fasta\n",
      "\t/tmp/tmpDFEhvb.fasta\n",
      "\t/tmp/tmp4ZqslX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnwJVBf.lastz\n",
      "\t/tmp/tmpE_HBvM.lastz\n",
      "\t/tmp/tmp0xf8tV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_amazonensis_GCA_005317125.lastz\n",
      "Creating Leishmania_amazonensis_GCA_005317125 table\n",
      "Inserting data to Leishmania_amazonensis_GCA_005317125 table\n",
      "\n",
      "Running against Leishmania_arabica_GCA_000410695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmplZqOtb.fasta\n",
      "\t/tmp/tmpVFgCbf.fasta\n",
      "\t/tmp/tmpf9k0Bo.fasta\n",
      "\t/tmp/tmpXqlPg9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1ZkYsC.lastz\n",
      "\t/tmp/tmppCfKJy.lastz\n",
      "\t/tmp/tmps4fEF3.lastz\n",
      "\t/tmp/tmpSQoDUw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_arabica_GCA_000410695.lastz\n",
      "Creating Leishmania_arabica_GCA_000410695 table\n",
      "Inserting data to Leishmania_arabica_GCA_000410695 table\n",
      "\n",
      "Running against Leishmania_braziliensis_GCF_000002845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp7ddwtn.fasta\n",
      "\t/tmp/tmprV3Em8.fasta\n",
      "\t/tmp/tmp70O9Xm.fasta\n",
      "\t/tmp/tmpW_yB7b.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZDze_3.lastz\n",
      "\t/tmp/tmph16X3u.lastz\n",
      "\t/tmp/tmpD41jZP.lastz\n",
      "\t/tmp/tmpQRG3Pe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_braziliensis_GCF_000002845.lastz\n",
      "Creating Leishmania_braziliensis_GCF_000002845 table\n",
      "Inserting data to Leishmania_braziliensis_GCF_000002845 table\n",
      "\n",
      "Running against Leishmania_donovani_GCF_000227135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpYoxXlW.fasta\n",
      "\t/tmp/tmpHwLu8_.fasta\n",
      "\t/tmp/tmpDtkmWm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiB8gAZ.lastz\n",
      "\t/tmp/tmpWvJSGK.lastz\n",
      "\t/tmp/tmpJYrkdG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_donovani_GCF_000227135.lastz\n",
      "Creating Leishmania_donovani_GCF_000227135 table\n",
      "Inserting data to Leishmania_donovani_GCF_000227135 table\n",
      "\n",
      "Running against Leishmania_enriettii_GCA_000410755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpkBMQLw.fasta\n",
      "\t/tmp/tmplHTlEX.fasta\n",
      "\t/tmp/tmptgfld8.fasta\n",
      "\t/tmp/tmp83R64c.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4ilG24.lastz\n",
      "\t/tmp/tmpsW3gtm.lastz\n",
      "\t/tmp/tmpeXOXVR.lastz\n",
      "\t/tmp/tmpRmWVF3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_enriettii_GCA_000410755.lastz\n",
      "Creating Leishmania_enriettii_GCA_000410755 table\n",
      "Inserting data to Leishmania_enriettii_GCA_000410755 table\n",
      "\n",
      "Running against Leishmania_gerbilli_GCA_000443025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpBG4ZHy.fasta\n",
      "\t/tmp/tmpQkwNCy.fasta\n",
      "\t/tmp/tmpQKhtUy.fasta\n",
      "\t/tmp/tmppDfY88.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOMH6RZ.lastz\n",
      "\t/tmp/tmpd_92HA.lastz\n",
      "\t/tmp/tmpdHvIsH.lastz\n",
      "\t/tmp/tmp7GT6s1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_gerbilli_GCA_000443025.lastz\n",
      "Creating Leishmania_gerbilli_GCA_000443025 table\n",
      "Inserting data to Leishmania_gerbilli_GCA_000443025 table\n",
      "\n",
      "Running against Leishmania_infantum_GCF_000002875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpHZKNX1.fasta\n",
      "\t/tmp/tmpcFxU0s.fasta\n",
      "\t/tmp/tmpYyEfOK.fasta\n",
      "\t/tmp/tmpBKVMJt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8XEhWD.lastz\n",
      "\t/tmp/tmp08dyWS.lastz\n",
      "\t/tmp/tmpn4MC6r.lastz\n",
      "\t/tmp/tmpDBznRZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_infantum_GCF_000002875.lastz\n",
      "Creating Leishmania_infantum_GCF_000002875 table\n",
      "Inserting data to Leishmania_infantum_GCF_000002875 table\n",
      "\n",
      "Running against Leishmania_lainsoni_GCA_003664395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpVkWXge.fasta\n",
      "\t/tmp/tmpE2QtJj.fasta\n",
      "\t/tmp/tmpohWmCF.fasta\n",
      "\t/tmp/tmpHo5VSD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0h_7UI.lastz\n",
      "\t/tmp/tmp6y2qNu.lastz\n",
      "\t/tmp/tmpCDuz90.lastz\n",
      "\t/tmp/tmp3GCmQ7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_lainsoni_GCA_003664395.lastz\n",
      "Creating Leishmania_lainsoni_GCA_003664395 table\n",
      "Inserting data to Leishmania_lainsoni_GCA_003664395 table\n",
      "\n",
      "Running against Leishmania_mexicana_GCF_000234665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmphowsS_.fasta\n",
      "\t/tmp/tmpKsTgdR.fasta\n",
      "\t/tmp/tmpwI1vwr.fasta\n",
      "\t/tmp/tmphZZ_1f.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnjwzIl.lastz\n",
      "\t/tmp/tmp1ZuO8O.lastz\n",
      "\t/tmp/tmpz1Wgne.lastz\n",
      "\t/tmp/tmpGB2OGe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_mexicana_GCF_000234665.lastz\n",
      "Creating Leishmania_mexicana_GCF_000234665 table\n",
      "Inserting data to Leishmania_mexicana_GCF_000234665 table\n",
      "\n",
      "Running against Leishmania_panamensis_GCF_000755165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpaq5g7M.fasta\n",
      "\t/tmp/tmpWUUuwj.fasta\n",
      "\t/tmp/tmpkVmrZh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1xRgz6.lastz\n",
      "\t/tmp/tmprXnn1d.lastz\n",
      "\t/tmp/tmp55X9p8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_panamensis_GCF_000755165.lastz\n",
      "Creating Leishmania_panamensis_GCF_000755165 table\n",
      "Inserting data to Leishmania_panamensis_GCF_000755165 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Leishmania_peruviana_GCA_001403675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpnbhNGW.fasta\n",
      "\t/tmp/tmpZi_KRt.fasta\n",
      "\t/tmp/tmpeLxiQn.fasta\n",
      "\t/tmp/tmptwVbAZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoXCXKe.lastz\n",
      "\t/tmp/tmprkA2Zw.lastz\n",
      "\t/tmp/tmpTo8eSh.lastz\n",
      "\t/tmp/tmpdZchHG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_peruviana_GCA_001403675.lastz\n",
      "Creating Leishmania_peruviana_GCA_001403675 table\n",
      "Inserting data to Leishmania_peruviana_GCA_001403675 table\n",
      "\n",
      "Running against Leishmania_sp_GCA_000409445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp8WUUlm.fasta\n",
      "\t/tmp/tmpYMyUt9.fasta\n",
      "\t/tmp/tmp0hOJXC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjflZzk.lastz\n",
      "\t/tmp/tmpu_FZ9e.lastz\n",
      "\t/tmp/tmpGMTBR2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_sp_GCA_000409445.lastz\n",
      "Creating Leishmania_sp_GCA_000409445 table\n",
      "Inserting data to Leishmania_sp_GCA_000409445 table\n",
      "\n",
      "Running against Leishmania_tarentolae_GCA_009731335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpZkHgNr.fasta\n",
      "\t/tmp/tmpqUSbfO.fasta\n",
      "\t/tmp/tmp1BMuzw.fasta\n",
      "\t/tmp/tmpZK0JrV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbt6z2z.lastz\n",
      "\t/tmp/tmpjDqMJU.lastz\n",
      "\t/tmp/tmp4_T91T.lastz\n",
      "\t/tmp/tmp9PrngN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_tarentolae_GCA_009731335.lastz\n",
      "Creating Leishmania_tarentolae_GCA_009731335 table\n",
      "Inserting data to Leishmania_tarentolae_GCA_009731335 table\n",
      "\n",
      "Running against Leishmania_tropica_GCA_000410715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpCt3X2E.fasta\n",
      "\t/tmp/tmpDXYcsS.fasta\n",
      "\t/tmp/tmpFhr0_N.fasta\n",
      "\t/tmp/tmp__TW7R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_0Q6FP.lastz\n",
      "\t/tmp/tmpbCnmlX.lastz\n",
      "\t/tmp/tmp5vTZa3.lastz\n",
      "\t/tmp/tmpE6qp6g.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_tropica_GCA_000410715.lastz\n",
      "Creating Leishmania_tropica_GCA_000410715 table\n",
      "Inserting data to Leishmania_tropica_GCA_000410715 table\n",
      "\n",
      "Running against Leishmania_turanica_GCA_000441995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpXb276R.fasta\n",
      "\t/tmp/tmpVpf5Sz.fasta\n",
      "\t/tmp/tmpia4Jlc.fasta\n",
      "\t/tmp/tmpdq0WY_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBIAggQ.lastz\n",
      "\t/tmp/tmpJMKjMw.lastz\n",
      "\t/tmp/tmpPrDTLc.lastz\n",
      "\t/tmp/tmpGlBQdJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_turanica_GCA_000441995.lastz\n",
      "Creating Leishmania_turanica_GCA_000441995 table\n",
      "Inserting data to Leishmania_turanica_GCA_000441995 table\n",
      "\n",
      "Running against Leptomonas_pyrrhocoris_GCF_001293395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpAElGNx.fasta\n",
      "\t/tmp/tmpMQW8AC.fasta\n",
      "\t/tmp/tmpOpyGdW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppaR40c.lastz\n",
      "\t/tmp/tmpc51U5b.lastz\n",
      "\t/tmp/tmpeCu30Y.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leptomonas_pyrrhocoris_GCF_001293395.lastz\n",
      "Creating Leptomonas_pyrrhocoris_GCF_001293395 table\n",
      "Inserting data to Leptomonas_pyrrhocoris_GCF_001293395 table\n",
      "\n",
      "Running against Leptomonas_seymouri_GCA_001299535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpvSLMyw.fasta\n",
      "\t/tmp/tmpLpJDkF.fasta\n",
      "\t/tmp/tmptDYh6W.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2klh8Z.lastz\n",
      "\t/tmp/tmpsYU1Qv.lastz\n",
      "\t/tmp/tmpswSyRz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leptomonas_seymouri_GCA_001299535.lastz\n",
      "Creating Leptomonas_seymouri_GCA_001299535 table\n",
      "Inserting data to Leptomonas_seymouri_GCA_001299535 table\n",
      "\n",
      "Running against Lotmaria_passim_GCA_000635995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpNSTZ4A.fasta\n",
      "\t/tmp/tmp1NnSQU.fasta\n",
      "\t/tmp/tmpLUOjKt.fasta\n",
      "\t/tmp/tmp1b29cG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpx26udl.lastz\n",
      "\t/tmp/tmpyHNFVV.lastz\n",
      "\t/tmp/tmpu5CDOt.lastz\n",
      "\t/tmp/tmpld27bJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Lotmaria_passim_GCA_000635995.lastz\n",
      "Creating Lotmaria_passim_GCA_000635995 table\n",
      "Inserting data to Lotmaria_passim_GCA_000635995 table\n",
      "\n",
      "Running against Paratrypanosoma_confusum_GCA_002921335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp0uI0dC.fasta\n",
      "\t/tmp/tmpwKLwzW.fasta\n",
      "\t/tmp/tmpk7ok5k.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzI5ZaG.lastz\n",
      "\t/tmp/tmptVqDIr.lastz\n",
      "\t/tmp/tmpMq8pKz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Paratrypanosoma_confusum_GCA_002921335.lastz\n",
      "Creating Paratrypanosoma_confusum_GCA_002921335 table\n",
      "Inserting data to Paratrypanosoma_confusum_GCA_002921335 table\n",
      "\n",
      "Running against Perkinsela_sp_GCA_001235845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFIozoP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAnV43n.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Perkinsela_sp_GCA_001235845.lastz\n",
      "Creating Perkinsela_sp_GCA_001235845 table\n",
      "Inserting data to Perkinsela_sp_GCA_001235845 table\n",
      "\n",
      "Running against Phytomonas_francai_GCA_001766655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpw4BTvi.fasta\n",
      "\t/tmp/tmpGxXw05.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYk9_cS.lastz\n",
      "\t/tmp/tmp4DLHr2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Phytomonas_francai_GCA_001766655.lastz\n",
      "Creating Phytomonas_francai_GCA_001766655 table\n",
      "Inserting data to Phytomonas_francai_GCA_001766655 table\n",
      "\n",
      "Running against Phytomonas_serpens_GCA_000331125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpYqX7XK.fasta\n",
      "\t/tmp/tmpqRPKGO.fasta\n",
      "\t/tmp/tmpPd8Pcy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8epasA.lastz\n",
      "\t/tmp/tmpv2903n.lastz\n",
      "\t/tmp/tmpRIkD39.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Phytomonas_serpens_GCA_000331125.lastz\n",
      "Creating Phytomonas_serpens_GCA_000331125 table\n",
      "Inserting data to Phytomonas_serpens_GCA_000331125 table\n",
      "\n",
      "Running against Phytomonas_sp_GCA_000582765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpAoLFhI.fasta\n",
      "\t/tmp/tmphiCaXL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWLvt2V.lastz\n",
      "\t/tmp/tmpnzFMh0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Phytomonas_sp_GCA_000582765.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Phytomonas_sp_GCA_000582765 table\n",
      "Inserting data to Phytomonas_sp_GCA_000582765 table\n",
      "\n",
      "Running against Strigomonas_culicis_GCA_000482145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpCAOV9Z.fasta\n",
      "\t/tmp/tmpFVfihh.fasta\n",
      "\t/tmp/tmpj5C4AM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3GGF7g.lastz\n",
      "\t/tmp/tmplNwQU0.lastz\n",
      "\t/tmp/tmpgnlRI2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Strigomonas_culicis_GCA_000482145.lastz\n",
      "Creating Strigomonas_culicis_GCA_000482145 table\n",
      "Inserting data to Strigomonas_culicis_GCA_000482145 table\n",
      "\n",
      "Running against Strigomonas_galati_GCA_000482125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpNkftXd.fasta\n",
      "\t/tmp/tmpRu7Kty.fasta\n",
      "\t/tmp/tmpsTKSWM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppgKZca.lastz\n",
      "\t/tmp/tmpve8mJn.lastz\n",
      "\t/tmp/tmpKSKiG6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Strigomonas_galati_GCA_000482125.lastz\n",
      "Creating Strigomonas_galati_GCA_000482125 table\n",
      "Inserting data to Strigomonas_galati_GCA_000482125 table\n",
      "\n",
      "Running against Strigomonas_oncopelti_GCA_000482165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpRHGQSQ.fasta\n",
      "\t/tmp/tmpxjkcDi.fasta\n",
      "\t/tmp/tmpBbw1FD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCLTdqZ.lastz\n",
      "\t/tmp/tmpV2JfXu.lastz\n",
      "\t/tmp/tmpzqZkF6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Strigomonas_oncopelti_GCA_000482165.lastz\n",
      "Creating Strigomonas_oncopelti_GCA_000482165 table\n",
      "Inserting data to Strigomonas_oncopelti_GCA_000482165 table\n",
      "\n",
      "Running against Trypanosoma_brucei_GCF_000210295.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp9iWW9y.fasta\n",
      "\t/tmp/tmpGRKuRd.fasta\n",
      "\t/tmp/tmpyfbyfz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpevoppx.lastz\n",
      "\t/tmp/tmpBtsAId.lastz\n",
      "\t/tmp/tmpyYtYdL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_brucei_GCF_000210295.lastz\n",
      "Creating Trypanosoma_brucei_GCF_000210295 table\n",
      "Inserting data to Trypanosoma_brucei_GCF_000210295 table\n",
      "\n",
      "Running against Trypanosoma_congolense_GCA_002287245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmp8KjlF5.fasta\n",
      "\t/tmp/tmpkwOLBm.fasta\n",
      "\t/tmp/tmph9HJVi.fasta\n",
      "\t/tmp/tmp0aj2vf.fasta\n",
      "\t/tmp/tmpD9QMkW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpX7EOFe.lastz\n",
      "\t/tmp/tmpW81WJn.lastz\n",
      "\t/tmp/tmpTTlLlB.lastz\n",
      "\t/tmp/tmpcseDR4.lastz\n",
      "\t/tmp/tmpKyplZY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_congolense_GCA_002287245.lastz\n",
      "Creating Trypanosoma_congolense_GCA_002287245 table\n",
      "Inserting data to Trypanosoma_congolense_GCA_002287245 table\n",
      "\n",
      "Running against Trypanosoma_conorhini_GCF_003719485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpnQOp99.fasta\n",
      "\t/tmp/tmp9dNW3w.fasta\n",
      "\t/tmp/tmp_LhKRJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDzy4LH.lastz\n",
      "\t/tmp/tmp7KK3Tb.lastz\n",
      "\t/tmp/tmpr0itNB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_conorhini_GCF_003719485.lastz\n",
      "Creating Trypanosoma_conorhini_GCF_003719485 table\n",
      "Inserting data to Trypanosoma_conorhini_GCF_003719485 table\n",
      "\n",
      "Running against Trypanosoma_cruzi_GCF_000209065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpgpnvkq.fasta\n",
      "\t/tmp/tmpJk4TFx.fasta\n",
      "\t/tmp/tmpdaW6e8.fasta\n",
      "\t/tmp/tmpv01HmO.fasta\n",
      "\t/tmp/tmp4m8sHj.fasta\n",
      "\t/tmp/tmpnp4GsL.fasta\n",
      "\t/tmp/tmpN4xNx8.fasta\n",
      "\t/tmp/tmpPadO_n.fasta\n",
      "\t/tmp/tmpHqQwZB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSeXLBN.lastz\n",
      "\t/tmp/tmpx5RgKb.lastz\n",
      "\t/tmp/tmpvFc088.lastz\n",
      "\t/tmp/tmpI7ps3P.lastz\n",
      "\t/tmp/tmpHf9A_W.lastz\n",
      "\t/tmp/tmpmpmlwR.lastz\n",
      "\t/tmp/tmpBp9FbJ.lastz\n",
      "\t/tmp/tmppTXYJt.lastz\n",
      "\t/tmp/tmp9JLbZY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_cruzi_GCF_000209065.lastz\n",
      "Creating Trypanosoma_cruzi_GCF_000209065 table\n",
      "Inserting data to Trypanosoma_cruzi_GCF_000209065 table\n",
      "\n",
      "Running against Trypanosoma_equiperdum_GCA_001457755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpoZojVe.fasta\n",
      "\t/tmp/tmpVy2U6U.fasta\n",
      "\t/tmp/tmpiiRydW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpI2Dch8.lastz\n",
      "\t/tmp/tmpv3R0NB.lastz\n",
      "\t/tmp/tmpjMjqgy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_equiperdum_GCA_001457755.lastz\n",
      "Creating Trypanosoma_equiperdum_GCA_001457755 table\n",
      "Inserting data to Trypanosoma_equiperdum_GCA_001457755 table\n",
      "\n",
      "Running against Trypanosoma_grayi_GCF_000691245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpUtCvUJ.fasta\n",
      "\t/tmp/tmpw2KvK8.fasta\n",
      "\t/tmp/tmphWSbDp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwUZScI.lastz\n",
      "\t/tmp/tmpjnMwAE.lastz\n",
      "\t/tmp/tmp2qpKqx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_grayi_GCF_000691245.lastz\n",
      "Creating Trypanosoma_grayi_GCF_000691245 table\n",
      "Inserting data to Trypanosoma_grayi_GCF_000691245 table\n",
      "\n",
      "Running against Trypanosoma_rangeli_GCF_003719475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpzScBXv.fasta\n",
      "\t/tmp/tmpolBc2Y.fasta\n",
      "\t/tmp/tmpZLzbhS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIuxPZz.lastz\n",
      "\t/tmp/tmp2gHVGw.lastz\n",
      "\t/tmp/tmpqwJSVh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_rangeli_GCF_003719475.lastz\n",
      "Creating Trypanosoma_rangeli_GCF_003719475 table\n",
      "Inserting data to Trypanosoma_rangeli_GCF_003719475 table\n",
      "\n",
      "Running against Trypanosoma_theileri_GCF_002087225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpPUzBsL.fasta\n",
      "\t/tmp/tmp_N3Swn.fasta\n",
      "\t/tmp/tmpKS0wOZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmuxG7j.lastz\n",
      "\t/tmp/tmpKIc6IF.lastz\n",
      "\t/tmp/tmp4thAFn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_theileri_GCF_002087225.lastz\n",
      "Creating Trypanosoma_theileri_GCF_002087225 table\n",
      "Inserting data to Trypanosoma_theileri_GCF_002087225 table\n",
      "\n",
      "Running against Trypanosoma_vivax_GCA_000227375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpyfhniI.fasta\n",
      "\t/tmp/tmp918pm9.fasta\n",
      "\t/tmp/tmpmGf1Sz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYj24qh.lastz\n",
      "\t/tmp/tmpARAC1a.lastz\n",
      "\t/tmp/tmpsVpCsc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosoma_vivax_GCA_000227375.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Trypanosoma_vivax_GCA_000227375 table\n",
      "Inserting data to Trypanosoma_vivax_GCA_000227375 table\n",
      "\n",
      "Running against Trypanosomatidae_sp_GCA_003671325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpC1Eta_.fasta\n",
      "\t/tmp/tmpuf9ltq.fasta\n",
      "\t/tmp/tmpQXjF5d.fasta\n",
      "\t/tmp/tmpOvXZw8.fasta\n",
      "\t/tmp/tmpghw5ov.fasta\n",
      "\t/tmp/tmphJygnb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9pzt0z.lastz\n",
      "\t/tmp/tmpwo8iIn.lastz\n",
      "\t/tmp/tmpBVdPWO.lastz\n",
      "\t/tmp/tmpfHW7de.lastz\n",
      "\t/tmp/tmpg4eG_f.lastz\n",
      "\t/tmp/tmp5V3Afx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Trypanosomatidae_sp_GCA_003671325.lastz\n",
      "Creating Trypanosomatidae_sp_GCA_003671325 table\n",
      "Inserting data to Trypanosomatidae_sp_GCA_003671325 table\n",
      "\n",
      "Running against Leishmania_major_GCF_000002725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpo9uVQr.fasta\n",
      "\t/tmp/tmpBefS7_.fasta\n",
      "\t/tmp/tmpBHI6js.fasta\n",
      "\t/tmp/tmpOKdOcI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmtbMz9.lastz\n",
      "\t/tmp/tmpyI5UmE.lastz\n",
      "\t/tmp/tmpjRI_PR.lastz\n",
      "\t/tmp/tmpHR9xFV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/kinetoplastea/validate_intervals/lastz/Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_Leishmania_major_GCF_000002725.lastz\n",
      "Creating Leishmania_major_GCF_000002725 table\n",
      "Inserting data to Leishmania_major_GCF_000002725 table\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "num_taxa = 17\n",
    "i_probes_fas    = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/kinetoplastea/extract_probes_from_group/kinetoplastea_genome.conf --lastz results/phyluce/kinetoplastea/validate_intervals/lastz --probes 120 --probe-prefix uce_kinetoplastea_ --name-pattern \"Leishmania_major_GCF_000002725_+17_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/kinetoplastea/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 12:42:30,899 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 12:42:30,907 - Phyluce - INFO - -------- Working on Angomonas_deanei_GCA_001659865 genome -------\n",
      "2020-02-11 12:42:31,025 - Phyluce - INFO - Reading Angomonas_deanei_GCA_001659865 genome\n",
      "2020-02-11 12:42:42,040 - Phyluce - INFO - Angomonas_deanei_GCA_001659865: 2112 uces, 342 dupes, 1770 non-dupes, 0 orient drop, 50 length drop, 1720 written\n",
      "2020-02-11 12:42:42,041 - Phyluce - INFO - ------- Working on Angomonas_desouzai_GCA_000482185 genome ------\n",
      "2020-02-11 12:42:42,056 - Phyluce - INFO - Reading Angomonas_desouzai_GCA_000482185 genome\n",
      "2020-02-11 12:42:51,999 - Phyluce - INFO - Angomonas_desouzai_GCA_000482185: 2164 uces, 515 dupes, 1649 non-dupes, 0 orient drop, 10 length drop, 1639 written\n",
      "2020-02-11 12:42:51,999 - Phyluce - INFO - ---------- Working on Bodo_saltans_GCA_001460835 genome ---------\n",
      "2020-02-11 12:42:52,153 - Phyluce - INFO - Reading Bodo_saltans_GCA_001460835 genome\n",
      "2020-02-11 12:42:59,495 - Phyluce - INFO - Bodo_saltans_GCA_001460835: 1698 uces, 410 dupes, 1288 non-dupes, 0 orient drop, 16 length drop, 1272 written\n",
      "2020-02-11 12:42:59,495 - Phyluce - INFO - ---- Working on Crithidia_acanthocephali_GCA_000482105 genome ---\n",
      "2020-02-11 12:42:59,612 - Phyluce - INFO - Reading Crithidia_acanthocephali_GCA_000482105 genome\n",
      "2020-02-11 12:43:12,787 - Phyluce - INFO - Crithidia_acanthocephali_GCA_000482105: 2736 uces, 608 dupes, 2128 non-dupes, 0 orient drop, 33 length drop, 2095 written\n",
      "2020-02-11 12:43:12,788 - Phyluce - INFO - -------- Working on Crithidia_bombi_GCA_900240985 genome --------\n",
      "2020-02-11 12:43:12,806 - Phyluce - INFO - Reading Crithidia_bombi_GCA_900240985 genome\n",
      "2020-02-11 12:43:24,586 - Phyluce - INFO - Crithidia_bombi_GCA_900240985: 2763 uces, 432 dupes, 2331 non-dupes, 5 orient drop, 99 length drop, 2227 written\n",
      "2020-02-11 12:43:24,586 - Phyluce - INFO - ------- Working on Crithidia_expoeki_GCA_900240875 genome -------\n",
      "2020-02-11 12:43:24,598 - Phyluce - INFO - Reading Crithidia_expoeki_GCA_900240875 genome\n",
      "2020-02-11 12:43:39,829 - Phyluce - INFO - Crithidia_expoeki_GCA_900240875: 2818 uces, 486 dupes, 2332 non-dupes, 7 orient drop, 192 length drop, 2133 written\n",
      "2020-02-11 12:43:39,829 - Phyluce - INFO - ----- Working on Crithidia_fasciculata_GCA_000331325 genome -----\n",
      "2020-02-11 12:43:40,053 - Phyluce - INFO - Reading Crithidia_fasciculata_GCA_000331325 genome\n",
      "2020-02-11 12:43:57,035 - Phyluce - INFO - Crithidia_fasciculata_GCA_000331325: 2751 uces, 820 dupes, 1931 non-dupes, 6 orient drop, 156 length drop, 1769 written\n",
      "2020-02-11 12:43:57,036 - Phyluce - INFO - ------ Working on Crithidia_mellificae_GCA_002216565 genome -----\n",
      "2020-02-11 12:43:57,062 - Phyluce - INFO - Reading Crithidia_mellificae_GCA_002216565 genome\n",
      "2020-02-11 12:44:15,493 - Phyluce - INFO - Crithidia_mellificae_GCA_002216565: 2672 uces, 1284 dupes, 1388 non-dupes, 0 orient drop, 19 length drop, 1369 written\n",
      "2020-02-11 12:44:15,494 - Phyluce - INFO - ---- Working on Endotrypanum_monterogeii_GCA_000333855 genome ---\n",
      "2020-02-11 12:44:15,528 - Phyluce - INFO - Reading Endotrypanum_monterogeii_GCA_000333855 genome\n",
      "2020-02-11 12:44:28,608 - Phyluce - INFO - Endotrypanum_monterogeii_GCA_000333855: 3060 uces, 432 dupes, 2628 non-dupes, 5 orient drop, 123 length drop, 2500 written\n",
      "2020-02-11 12:44:28,608 - Phyluce - INFO - ----- Working on Herpetomonas_muscarum_GCA_000482205 genome -----\n",
      "2020-02-11 12:44:28,621 - Phyluce - INFO - Reading Herpetomonas_muscarum_GCA_000482205 genome\n",
      "2020-02-11 12:44:39,955 - Phyluce - INFO - Herpetomonas_muscarum_GCA_000482205: 2242 uces, 432 dupes, 1810 non-dupes, 0 orient drop, 20 length drop, 1788 written\n",
      "2020-02-11 12:44:39,955 - Phyluce - INFO - ----- Working on Leishmania_aethiopica_GCA_000444285 genome -----\n",
      "2020-02-11 12:44:40,627 - Phyluce - INFO - Reading Leishmania_aethiopica_GCA_000444285 genome\n",
      "2020-02-11 12:44:54,318 - Phyluce - INFO - Leishmania_aethiopica_GCA_000444285: 3286 uces, 442 dupes, 2844 non-dupes, 8 orient drop, 131 length drop, 2705 written\n",
      "2020-02-11 12:44:54,318 - Phyluce - INFO - ----- Working on Leishmania_amazonensis_GCA_005317125 genome ----\n",
      "2020-02-11 12:44:54,321 - Phyluce - INFO - Reading Leishmania_amazonensis_GCA_005317125 genome\n",
      "2020-02-11 12:45:09,071 - Phyluce - INFO - Leishmania_amazonensis_GCA_005317125: 3289 uces, 443 dupes, 2846 non-dupes, 9 orient drop, 313 length drop, 2524 written\n",
      "2020-02-11 12:45:09,072 - Phyluce - INFO - ------- Working on Leishmania_arabica_GCA_000410695 genome ------\n",
      "2020-02-11 12:45:09,093 - Phyluce - INFO - Reading Leishmania_arabica_GCA_000410695 genome\n",
      "2020-02-11 12:45:23,709 - Phyluce - INFO - Leishmania_arabica_GCA_000410695: 3291 uces, 452 dupes, 2839 non-dupes, 6 orient drop, 98 length drop, 2735 written\n",
      "2020-02-11 12:45:23,709 - Phyluce - INFO - ---- Working on Leishmania_braziliensis_GCF_000002845 genome ----\n",
      "2020-02-11 12:45:24,646 - Phyluce - INFO - Reading Leishmania_braziliensis_GCF_000002845 genome\n",
      "2020-02-11 12:45:38,532 - Phyluce - INFO - Leishmania_braziliensis_GCF_000002845: 3179 uces, 463 dupes, 2716 non-dupes, 7 orient drop, 166 length drop, 2543 written\n",
      "2020-02-11 12:45:38,533 - Phyluce - INFO - ------ Working on Leishmania_donovani_GCF_000227135 genome ------\n",
      "2020-02-11 12:45:38,831 - Phyluce - INFO - Reading Leishmania_donovani_GCF_000227135 genome\n",
      "2020-02-11 12:45:52,714 - Phyluce - INFO - Leishmania_donovani_GCF_000227135: 3303 uces, 435 dupes, 2868 non-dupes, 5 orient drop, 94 length drop, 2769 written\n",
      "2020-02-11 12:45:52,714 - Phyluce - INFO - ------ Working on Leishmania_enriettii_GCA_000410755 genome -----\n",
      "2020-02-11 12:45:52,766 - Phyluce - INFO - Reading Leishmania_enriettii_GCA_000410755 genome\n",
      "2020-02-11 12:46:06,205 - Phyluce - INFO - Leishmania_enriettii_GCA_000410755: 3149 uces, 455 dupes, 2694 non-dupes, 9 orient drop, 111 length drop, 2574 written\n",
      "2020-02-11 12:46:06,205 - Phyluce - INFO - ------ Working on Leishmania_gerbilli_GCA_000443025 genome ------\n",
      "2020-02-11 12:46:06,363 - Phyluce - INFO - Reading Leishmania_gerbilli_GCA_000443025 genome\n",
      "2020-02-11 12:46:20,258 - Phyluce - INFO - Leishmania_gerbilli_GCA_000443025: 3289 uces, 476 dupes, 2813 non-dupes, 4 orient drop, 119 length drop, 2690 written\n",
      "2020-02-11 12:46:20,259 - Phyluce - INFO - ------ Working on Leishmania_infantum_GCF_000002875 genome ------\n",
      "2020-02-11 12:46:20,297 - Phyluce - INFO - Reading Leishmania_infantum_GCF_000002875 genome\n",
      "2020-02-11 12:46:35,337 - Phyluce - INFO - Leishmania_infantum_GCF_000002875: 3302 uces, 470 dupes, 2832 non-dupes, 9 orient drop, 170 length drop, 2652 written\n",
      "2020-02-11 12:46:35,338 - Phyluce - INFO - ------ Working on Leishmania_lainsoni_GCA_003664395 genome ------\n",
      "2020-02-11 12:46:35,498 - Phyluce - INFO - Reading Leishmania_lainsoni_GCA_003664395 genome\n",
      "2020-02-11 12:46:51,541 - Phyluce - INFO - Leishmania_lainsoni_GCA_003664395: 3200 uces, 553 dupes, 2647 non-dupes, 6 orient drop, 178 length drop, 2463 written\n",
      "2020-02-11 12:46:51,541 - Phyluce - INFO - ------ Working on Leishmania_mexicana_GCF_000234665 genome ------\n",
      "2020-02-11 12:46:51,760 - Phyluce - INFO - Reading Leishmania_mexicana_GCF_000234665 genome\n",
      "2020-02-11 12:47:06,650 - Phyluce - INFO - Leishmania_mexicana_GCF_000234665: 3300 uces, 492 dupes, 2808 non-dupes, 8 orient drop, 156 length drop, 2644 written\n",
      "2020-02-11 12:47:06,651 - Phyluce - INFO - ----- Working on Leishmania_panamensis_GCF_000755165 genome -----\n",
      "2020-02-11 12:47:06,916 - Phyluce - INFO - Reading Leishmania_panamensis_GCF_000755165 genome\n",
      "2020-02-11 12:47:20,457 - Phyluce - INFO - Leishmania_panamensis_GCF_000755165: 3194 uces, 430 dupes, 2764 non-dupes, 7 orient drop, 91 length drop, 2666 written\n",
      "2020-02-11 12:47:20,458 - Phyluce - INFO - ------ Working on Leishmania_peruviana_GCA_001403675 genome -----\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-11 12:47:21,466 - Phyluce - INFO - Reading Leishmania_peruviana_GCA_001403675 genome\n",
      "2020-02-11 12:47:34,536 - Phyluce - INFO - Leishmania_peruviana_GCA_001403675: 3117 uces, 420 dupes, 2697 non-dupes, 7 orient drop, 124 length drop, 2566 written\n",
      "2020-02-11 12:47:34,536 - Phyluce - INFO - --------- Working on Leishmania_sp_GCA_000409445 genome ---------\n",
      "2020-02-11 12:47:34,618 - Phyluce - INFO - Reading Leishmania_sp_GCA_000409445 genome\n",
      "2020-02-11 12:47:47,689 - Phyluce - INFO - Leishmania_sp_GCA_000409445: 3128 uces, 449 dupes, 2679 non-dupes, 7 orient drop, 140 length drop, 2532 written\n",
      "2020-02-11 12:47:47,690 - Phyluce - INFO - ----- Working on Leishmania_tarentolae_GCA_009731335 genome -----\n",
      "2020-02-11 12:47:47,700 - Phyluce - INFO - Reading Leishmania_tarentolae_GCA_009731335 genome\n",
      "2020-02-11 12:48:03,961 - Phyluce - INFO - Leishmania_tarentolae_GCA_009731335: 3258 uces, 508 dupes, 2750 non-dupes, 12 orient drop, 185 length drop, 2553 written\n",
      "2020-02-11 12:48:03,962 - Phyluce - INFO - ------- Working on Leishmania_tropica_GCA_000410715 genome ------\n",
      "2020-02-11 12:48:03,965 - Phyluce - INFO - Reading Leishmania_tropica_GCA_000410715 genome\n",
      "2020-02-11 12:48:18,005 - Phyluce - INFO - Leishmania_tropica_GCA_000410715: 3265 uces, 474 dupes, 2791 non-dupes, 3 orient drop, 109 length drop, 2679 written\n",
      "2020-02-11 12:48:18,005 - Phyluce - INFO - ------ Working on Leishmania_turanica_GCA_000441995 genome ------\n",
      "2020-02-11 12:48:18,071 - Phyluce - INFO - Reading Leishmania_turanica_GCA_000441995 genome\n",
      "2020-02-11 12:48:31,878 - Phyluce - INFO - Leishmania_turanica_GCA_000441995: 3291 uces, 451 dupes, 2840 non-dupes, 6 orient drop, 106 length drop, 2727 written\n",
      "2020-02-11 12:48:31,878 - Phyluce - INFO - ----- Working on Leptomonas_pyrrhocoris_GCF_001293395 genome ----\n",
      "2020-02-11 12:48:31,889 - Phyluce - INFO - Reading Leptomonas_pyrrhocoris_GCF_001293395 genome\n",
      "2020-02-11 12:48:44,990 - Phyluce - INFO - Leptomonas_pyrrhocoris_GCF_001293395: 2752 uces, 431 dupes, 2321 non-dupes, 7 orient drop, 208 length drop, 2106 written\n",
      "2020-02-11 12:48:44,990 - Phyluce - INFO - ------ Working on Leptomonas_seymouri_GCA_001299535 genome ------\n",
      "2020-02-11 12:48:45,000 - Phyluce - INFO - Reading Leptomonas_seymouri_GCA_001299535 genome\n",
      "2020-02-11 12:48:55,845 - Phyluce - INFO - Leptomonas_seymouri_GCA_001299535: 2761 uces, 385 dupes, 2376 non-dupes, 0 orient drop, 27 length drop, 2347 written\n",
      "2020-02-11 12:48:55,845 - Phyluce - INFO - -------- Working on Lotmaria_passim_GCA_000635995 genome --------\n",
      "2020-02-11 12:48:55,856 - Phyluce - INFO - Reading Lotmaria_passim_GCA_000635995 genome\n",
      "2020-02-11 12:49:10,376 - Phyluce - INFO - Lotmaria_passim_GCA_000635995: 2799 uces, 686 dupes, 2113 non-dupes, 19 orient drop, 51 length drop, 2043 written\n",
      "2020-02-11 12:49:10,376 - Phyluce - INFO - ---- Working on Paratrypanosoma_confusum_GCA_002921335 genome ---\n",
      "2020-02-11 12:49:10,618 - Phyluce - INFO - Reading Paratrypanosoma_confusum_GCA_002921335 genome\n",
      "2020-02-11 12:49:18,663 - Phyluce - INFO - Paratrypanosoma_confusum_GCA_002921335: 2038 uces, 351 dupes, 1687 non-dupes, 0 orient drop, 19 length drop, 1668 written\n",
      "2020-02-11 12:49:18,664 - Phyluce - INFO - --------- Working on Perkinsela_sp_GCA_001235845 genome ---------\n",
      "2020-02-11 12:49:18,724 - Phyluce - INFO - Reading Perkinsela_sp_GCA_001235845 genome\n",
      "2020-02-11 12:49:20,851 - Phyluce - INFO - Perkinsela_sp_GCA_001235845: 511 uces, 158 dupes, 353 non-dupes, 5 orient drop, 2 length drop, 346 written\n",
      "2020-02-11 12:49:20,851 - Phyluce - INFO - ------- Working on Phytomonas_francai_GCA_001766655 genome ------\n",
      "2020-02-11 12:49:20,937 - Phyluce - INFO - Reading Phytomonas_francai_GCA_001766655 genome\n",
      "2020-02-11 12:49:28,897 - Phyluce - INFO - Phytomonas_francai_GCA_001766655: 2068 uces, 325 dupes, 1743 non-dupes, 19 orient drop, 77 length drop, 1647 written\n",
      "2020-02-11 12:49:28,897 - Phyluce - INFO - ------- Working on Phytomonas_serpens_GCA_000331125 genome ------\n",
      "2020-02-11 12:49:28,911 - Phyluce - INFO - Reading Phytomonas_serpens_GCA_000331125 genome\n",
      "2020-02-11 12:49:39,244 - Phyluce - INFO - Phytomonas_serpens_GCA_000331125: 2117 uces, 804 dupes, 1313 non-dupes, 0 orient drop, 2 length drop, 1309 written\n",
      "2020-02-11 12:49:39,244 - Phyluce - INFO - --------- Working on Phytomonas_sp_GCA_000582765 genome ---------\n",
      "2020-02-11 12:49:39,248 - Phyluce - INFO - Reading Phytomonas_sp_GCA_000582765 genome\n",
      "2020-02-11 12:49:46,814 - Phyluce - INFO - Phytomonas_sp_GCA_000582765: 2028 uces, 291 dupes, 1737 non-dupes, 1 orient drop, 25 length drop, 1711 written\n",
      "2020-02-11 12:49:46,815 - Phyluce - INFO - ------ Working on Strigomonas_culicis_GCA_000482145 genome ------\n",
      "2020-02-11 12:49:46,821 - Phyluce - INFO - Reading Strigomonas_culicis_GCA_000482145 genome\n",
      "2020-02-11 12:49:56,201 - Phyluce - INFO - Strigomonas_culicis_GCA_000482145: 2140 uces, 457 dupes, 1683 non-dupes, 0 orient drop, 26 length drop, 1657 written\n",
      "2020-02-11 12:49:56,202 - Phyluce - INFO - ------- Working on Strigomonas_galati_GCA_000482125 genome ------\n",
      "2020-02-11 12:49:56,280 - Phyluce - INFO - Reading Strigomonas_galati_GCA_000482125 genome\n",
      "2020-02-11 12:50:06,742 - Phyluce - INFO - Strigomonas_galati_GCA_000482125: 2271 uces, 542 dupes, 1729 non-dupes, 0 orient drop, 15 length drop, 1713 written\n",
      "2020-02-11 12:50:06,743 - Phyluce - INFO - ----- Working on Strigomonas_oncopelti_GCA_000482165 genome -----\n",
      "2020-02-11 12:50:06,761 - Phyluce - INFO - Reading Strigomonas_oncopelti_GCA_000482165 genome\n",
      "2020-02-11 12:50:16,623 - Phyluce - INFO - Strigomonas_oncopelti_GCA_000482165: 2239 uces, 468 dupes, 1771 non-dupes, 0 orient drop, 10 length drop, 1760 written\n",
      "2020-02-11 12:50:16,623 - Phyluce - INFO - ------- Working on Trypanosoma_brucei_GCF_000210295 genome ------\n",
      "2020-02-11 12:50:16,636 - Phyluce - INFO - Reading Trypanosoma_brucei_GCF_000210295 genome\n",
      "2020-02-11 12:50:24,703 - Phyluce - INFO - Trypanosoma_brucei_GCF_000210295: 1990 uces, 290 dupes, 1700 non-dupes, 10 orient drop, 150 length drop, 1540 written\n",
      "2020-02-11 12:50:24,703 - Phyluce - INFO - ----- Working on Trypanosoma_congolense_GCA_002287245 genome ----\n",
      "2020-02-11 12:50:25,060 - Phyluce - INFO - Reading Trypanosoma_congolense_GCA_002287245 genome\n",
      "2020-02-11 12:50:37,463 - Phyluce - INFO - Trypanosoma_congolense_GCA_002287245: 2014 uces, 429 dupes, 1585 non-dupes, 9 orient drop, 147 length drop, 1429 written\n",
      "2020-02-11 12:50:37,463 - Phyluce - INFO - ----- Working on Trypanosoma_conorhini_GCF_003719485 genome -----\n",
      "2020-02-11 12:50:37,600 - Phyluce - INFO - Reading Trypanosoma_conorhini_GCF_003719485 genome\n",
      "2020-02-11 12:50:46,193 - Phyluce - INFO - Trypanosoma_conorhini_GCF_003719485: 2141 uces, 408 dupes, 1733 non-dupes, 0 orient drop, 45 length drop, 1688 written\n",
      "2020-02-11 12:50:46,193 - Phyluce - INFO - ------- Working on Trypanosoma_cruzi_GCF_000209065 genome -------\n",
      "2020-02-11 12:50:46,206 - Phyluce - INFO - Reading Trypanosoma_cruzi_GCF_000209065 genome\n",
      "2020-02-11 12:51:06,314 - Phyluce - INFO - Trypanosoma_cruzi_GCF_000209065: 2209 uces, 1737 dupes, 472 non-dupes, 0 orient drop, 25 length drop, 447 written\n",
      "2020-02-11 12:51:06,314 - Phyluce - INFO - ----- Working on Trypanosoma_equiperdum_GCA_001457755 genome ----\n",
      "2020-02-11 12:51:06,634 - Phyluce - INFO - Reading Trypanosoma_equiperdum_GCA_001457755 genome\n",
      "2020-02-11 12:51:13,770 - Phyluce - INFO - Trypanosoma_equiperdum_GCA_001457755: 1952 uces, 263 dupes, 1689 non-dupes, 0 orient drop, 18 length drop, 1670 written\n",
      "2020-02-11 12:51:13,771 - Phyluce - INFO - ------- Working on Trypanosoma_grayi_GCF_000691245 genome -------\n",
      "2020-02-11 12:51:13,797 - Phyluce - INFO - Reading Trypanosoma_grayi_GCF_000691245 genome\n",
      "2020-02-11 12:51:22,334 - Phyluce - INFO - Trypanosoma_grayi_GCF_000691245: 2163 uces, 388 dupes, 1775 non-dupes, 0 orient drop, 16 length drop, 1758 written\n",
      "2020-02-11 12:51:22,334 - Phyluce - INFO - ------ Working on Trypanosoma_rangeli_GCF_003719475 genome ------\n",
      "2020-02-11 12:51:22,357 - Phyluce - INFO - Reading Trypanosoma_rangeli_GCF_003719475 genome\n",
      "2020-02-11 12:51:31,229 - Phyluce - INFO - Trypanosoma_rangeli_GCF_003719475: 2133 uces, 385 dupes, 1748 non-dupes, 0 orient drop, 59 length drop, 1689 written\n",
      "2020-02-11 12:51:31,230 - Phyluce - INFO - ------ Working on Trypanosoma_theileri_GCF_002087225 genome -----\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-11 12:51:31,815 - Phyluce - INFO - Reading Trypanosoma_theileri_GCF_002087225 genome\n",
      "2020-02-11 12:51:39,545 - Phyluce - INFO - Trypanosoma_theileri_GCF_002087225: 1831 uces, 270 dupes, 1561 non-dupes, 7 orient drop, 90 length drop, 1464 written\n",
      "2020-02-11 12:51:39,546 - Phyluce - INFO - ------- Working on Trypanosoma_vivax_GCA_000227375 genome -------\n",
      "2020-02-11 12:51:40,035 - Phyluce - INFO - Reading Trypanosoma_vivax_GCA_000227375 genome\n",
      "2020-02-11 12:51:41,035 - Phyluce - INFO - Trypanosoma_vivax_GCA_000227375: 178 uces, 58 dupes, 120 non-dupes, 1 orient drop, 7 length drop, 111 written\n",
      "2020-02-11 12:51:41,035 - Phyluce - INFO - ------ Working on Trypanosomatidae_sp_GCA_003671325 genome ------\n",
      "2020-02-11 12:51:41,044 - Phyluce - INFO - Reading Trypanosomatidae_sp_GCA_003671325 genome\n",
      "2020-02-11 12:52:01,429 - Phyluce - INFO - Trypanosomatidae_sp_GCA_003671325: 2721 uces, 1277 dupes, 1444 non-dupes, 0 orient drop, 38 length drop, 1406 written\n",
      "2020-02-11 12:52:01,429 - Phyluce - INFO - -------- Working on Leishmania_major_GCF_000002725 genome -------\n",
      "2020-02-11 12:52:01,433 - Phyluce - INFO - Reading Leishmania_major_GCF_000002725 genome\n",
      "2020-02-11 12:52:16,733 - Phyluce - INFO - Leishmania_major_GCF_000002725: 3321 uces, 458 dupes, 2863 non-dupes, 11 orient drop, 198 length drop, 2654 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/kinetoplastea/extract_probes_from_group/probe_fasta --output results/phyluce/kinetoplastea/extract_probes_from_group/multifastas.sqlite --base-taxon Leishmania_major_GCF_000002725;\n",
      "angomonas_deanei_gca_001659865..\n",
      "angomonas_desouzai_gca_000482185..\n",
      "bodo_saltans_gca_001460835..\n",
      "crithidia_acanthocephali_gca_000482105...\n",
      "crithidia_bombi_gca_900240985...\n",
      "crithidia_expoeki_gca_900240875...\n",
      "crithidia_fasciculata_gca_000331325..\n",
      "crithidia_mellificae_gca_002216565..\n",
      "endotrypanum_monterogeii_gca_000333855...\n",
      "herpetomonas_muscarum_gca_000482205..\n",
      "leishmania_aethiopica_gca_000444285...\n",
      "leishmania_amazonensis_gca_005317125...\n",
      "leishmania_arabica_gca_000410695...\n",
      "leishmania_braziliensis_gcf_000002845...\n",
      "leishmania_donovani_gcf_000227135...\n",
      "leishmania_enriettii_gca_000410755...\n",
      "leishmania_gerbilli_gca_000443025...\n",
      "leishmania_infantum_gcf_000002875...\n",
      "leishmania_lainsoni_gca_003664395...\n",
      "leishmania_mexicana_gcf_000234665...\n",
      "leishmania_panamensis_gcf_000755165...\n",
      "leishmania_peruviana_gca_001403675...\n",
      "leishmania_sp_gca_000409445...\n",
      "leishmania_tarentolae_gca_009731335...\n",
      "leishmania_tropica_gca_000410715...\n",
      "leishmania_turanica_gca_000441995...\n",
      "leptomonas_pyrrhocoris_gcf_001293395...\n",
      "leptomonas_seymouri_gca_001299535...\n",
      "lotmaria_passim_gca_000635995...\n",
      "paratrypanosoma_confusum_gca_002921335..\n",
      "perkinsela_sp_gca_001235845.\n",
      "phytomonas_francai_gca_001766655..\n",
      "phytomonas_serpens_gca_000331125..\n",
      "phytomonas_sp_gca_000582765..\n",
      "strigomonas_culicis_gca_000482145..\n",
      "strigomonas_galati_gca_000482125..\n",
      "strigomonas_oncopelti_gca_000482165..\n",
      "trypanosoma_brucei_gcf_000210295..\n",
      "trypanosoma_congolense_gca_002287245..\n",
      "trypanosoma_conorhini_gcf_003719485..\n",
      "trypanosoma_cruzi_gcf_000209065.\n",
      "trypanosoma_equiperdum_gca_001457755..\n",
      "trypanosoma_grayi_gcf_000691245..\n",
      "trypanosoma_rangeli_gcf_003719475..\n",
      "trypanosoma_theileri_gcf_002087225..\n",
      "trypanosoma_vivax_gca_000227375.\n",
      "trypanosomatidae_sp_gca_003671325..\n",
      "leishmania_major_gcf_000002725...\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/kinetoplastea/extract_probes_from_group/multifastas.sqlite --base-taxon Leishmania_major_GCF_000002725;\n",
      "Loci shared by 0 taxa:\t3,274.0\n",
      "Loci shared by 1 taxa:\t3,274.0\n",
      "Loci shared by 2 taxa:\t3,225.0\n",
      "Loci shared by 3 taxa:\t3,193.0\n",
      "Loci shared by 4 taxa:\t3,161.0\n",
      "Loci shared by 5 taxa:\t3,126.0\n",
      "Loci shared by 6 taxa:\t3,097.0\n",
      "Loci shared by 7 taxa:\t3,051.0\n",
      "Loci shared by 8 taxa:\t3,000.0\n",
      "Loci shared by 9 taxa:\t2,968.0\n",
      "Loci shared by 10 taxa:\t2,934.0\n",
      "Loci shared by 11 taxa:\t2,871.0\n",
      "Loci shared by 12 taxa:\t2,810.0\n",
      "Loci shared by 13 taxa:\t2,767.0\n",
      "Loci shared by 14 taxa:\t2,731.0\n",
      "Loci shared by 15 taxa:\t2,694.0\n",
      "Loci shared by 16 taxa:\t2,648.0\n",
      "Loci shared by 17 taxa:\t2,588.0\n",
      "Loci shared by 18 taxa:\t2,512.0\n",
      "Loci shared by 19 taxa:\t2,395.0\n",
      "Loci shared by 20 taxa:\t2,341.0\n",
      "Loci shared by 21 taxa:\t2,294.0\n",
      "Loci shared by 22 taxa:\t2,253.0\n",
      "Loci shared by 23 taxa:\t2,203.0\n",
      "Loci shared by 24 taxa:\t2,150.0\n",
      "Loci shared by 25 taxa:\t2,104.0\n",
      "Loci shared by 26 taxa:\t2,035.0\n",
      "Loci shared by 27 taxa:\t1,952.0\n",
      "Loci shared by 28 taxa:\t1,865.0\n",
      "Loci shared by 29 taxa:\t1,810.0\n",
      "Loci shared by 30 taxa:\t1,751.0\n",
      "Loci shared by 31 taxa:\t1,689.0\n",
      "Loci shared by 32 taxa:\t1,629.0\n",
      "Loci shared by 33 taxa:\t1,575.0\n",
      "Loci shared by 34 taxa:\t1,512.0\n",
      "Loci shared by 35 taxa:\t1,434.0\n",
      "Loci shared by 36 taxa:\t1,378.0\n",
      "Loci shared by 37 taxa:\t1,307.0\n",
      "Loci shared by 38 taxa:\t1,228.0\n",
      "Loci shared by 39 taxa:\t1,140.0\n",
      "Loci shared by 40 taxa:\t1,053.0\n",
      "Loci shared by 41 taxa:\t942.0\n",
      "Loci shared by 42 taxa:\t796.0\n",
      "Loci shared by 43 taxa:\t609.0\n",
      "Loci shared by 44 taxa:\t383.0\n",
      "Loci shared by 45 taxa:\t169.0\n",
      "Loci shared by 46 taxa:\t42.0\n",
      "Loci shared by 47 taxa:\t8.0\n",
      "Loci shared by 48 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/kinetoplastea/extract_probes_from_group/multifastas.sqlite  --base-taxon Leishmania_major_GCF_000002725 --output results/phyluce/kinetoplastea/extract_probes_from_group/Leishmania_major_GCF_000002725+17-back-to-45.conf --specific-counts 45;\n",
      "Counter({'trypanosoma_rangeli_gcf_003719475': 169, 'trypanosoma_brucei_gcf_000210295': 169, 'leishmania_mexicana_gcf_000234665': 169, 'endotrypanum_monterogeii_gca_000333855': 169, 'leptomonas_seymouri_gca_001299535': 169, 'leptomonas_pyrrhocoris_gcf_001293395': 169, 'leishmania_peruviana_gca_001403675': 169, 'leishmania_turanica_gca_000441995': 169, 'leishmania_gerbilli_gca_000443025': 169, 'leishmania_sp_gca_000409445': 169, 'leishmania_aethiopica_gca_000444285': 169, 'strigomonas_oncopelti_gca_000482165': 169, 'angomonas_desouzai_gca_000482185': 169, 'trypanosoma_grayi_gcf_000691245': 169, 'leishmania_enriettii_gca_000410755': 169, 'crithidia_acanthocephali_gca_000482105': 169, 'leishmania_tropica_gca_000410715': 169, 'leishmania_infantum_gcf_000002875': 169, 'leishmania_panamensis_gcf_000755165': 169, 'leishmania_lainsoni_gca_003664395': 169, 'leishmania_tarentolae_gca_009731335': 169, 'leishmania_major_gcf_000002725': 169, 'leishmania_donovani_gcf_000227135': 169, 'crithidia_expoeki_gca_900240875': 168, 'strigomonas_galati_gca_000482125': 168, 'leishmania_braziliensis_gcf_000002845': 168, 'leishmania_arabica_gca_000410695': 168, 'crithidia_bombi_gca_900240985': 168, 'angomonas_deanei_gca_001659865': 168, 'phytomonas_sp_gca_000582765': 168, 'trypanosoma_equiperdum_gca_001457755': 168, 'trypanosoma_conorhini_gcf_003719485': 168, 'phytomonas_francai_gca_001766655': 168, 'herpetomonas_muscarum_gca_000482205': 167, 'lotmaria_passim_gca_000635995': 167, 'paratrypanosoma_confusum_gca_002921335': 167, 'trypanosoma_congolense_gca_002287245': 167, 'strigomonas_culicis_gca_000482145': 166, 'trypanosoma_theileri_gcf_002087225': 166, 'bodo_saltans_gca_001460835': 165, 'leishmania_amazonensis_gca_005317125': 163, 'crithidia_fasciculata_gca_000331325': 161, 'phytomonas_serpens_gca_000331125': 158, 'trypanosomatidae_sp_gca_003671325': 147, 'crithidia_mellificae_gca_002216565': 140, 'trypanosoma_cruzi_gcf_000209065': 74, 'perkinsela_sp_gca_001235845': 72, 'trypanosoma_vivax_gca_000227375': 8})\n",
      "Total loci = 169\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(45)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/kinetoplastea/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/kinetoplastea/extract_probes_from_group/Leishmania_major_GCF_000002725+17-back-to-45.conf --probe-prefix uce_kinetoplastea_ --designer rnplattii --design kinetoplastea_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGNNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 169\n",
      "Probe Count = 15149\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list.fasta --query results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  12:52:56\n",
      "Ended:  Tue Feb 11, 2020  12:53:57\n",
      "Time for execution:  1.01459121704 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list.fasta --lastz results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_kinetoplastea_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 15148 fasta sequences.  Filtered 0 duplicates. Kept 15149.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/kinetoplastea/final_probe_design/kinetoplastea_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 13:17:09 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 15149\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1211920\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 90M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3939168\n",
      "Max number of word counting entries: 88670684\n",
      "\n",
      "# comparing sequences from          0  to       2524\n",
      "..---------- new table with     1780 representatives\n",
      "# comparing sequences from       2524  to       4628\n",
      "----------    281 remaining sequences to the next cycle\n",
      "---------- new table with     1309 representatives\n",
      "# comparing sequences from       4347  to       6147\n",
      "----------    366 remaining sequences to the next cycle\n",
      "---------- new table with     1012 representatives\n",
      "# comparing sequences from       5781  to       7342\n",
      "----------    305 remaining sequences to the next cycle\n",
      "---------- new table with      889 representatives\n",
      "# comparing sequences from       7037  to       8389\n",
      "----------    262 remaining sequences to the next cycle\n",
      "---------- new table with      765 representatives\n",
      "# comparing sequences from       8127  to       9297\n",
      "----------    228 remaining sequences to the next cycle\n",
      "---------- new table with      680 representatives\n",
      "# comparing sequences from       9069  to      10082\n",
      "----------    236 remaining sequences to the next cycle\n",
      "---------- new table with      546 representatives\n",
      "# comparing sequences from       9846  to      10729\n",
      "..........    10000  finished       7082  clusters\n",
      "----------    299 remaining sequences to the next cycle\n",
      "---------- new table with      411 representatives\n",
      "# comparing sequences from      10430  to      11216\n",
      "----------    292 remaining sequences to the next cycle\n",
      "---------- new table with      351 representatives\n",
      "# comparing sequences from      10924  to      11628\n",
      "----------    121 remaining sequences to the next cycle\n",
      "---------- new table with      424 representatives\n",
      "# comparing sequences from      11507  to      12114\n",
      "----------    118 remaining sequences to the next cycle\n",
      "---------- new table with      354 representatives\n",
      "# comparing sequences from      11996  to      12521\n",
      "----------    123 remaining sequences to the next cycle\n",
      "---------- new table with      283 representatives\n",
      "# comparing sequences from      12398  to      12856\n",
      "----------     41 remaining sequences to the next cycle\n",
      "---------- new table with      292 representatives\n",
      "# comparing sequences from      12815  to      13204\n",
      "----------     60 remaining sequences to the next cycle\n",
      "---------- new table with      230 representatives\n",
      "# comparing sequences from      13144  to      13478\n",
      "99.9%---------- new table with      244 representatives\n",
      "# comparing sequences from      13478  to      13756\n",
      "----------     34 remaining sequences to the next cycle\n",
      "---------- new table with      183 representatives\n",
      "# comparing sequences from      13722  to      13959\n",
      "----------     22 remaining sequences to the next cycle\n",
      "---------- new table with      145 representatives\n",
      "# comparing sequences from      13937  to      14139\n",
      "100.0%---------- new table with      151 representatives\n",
      "# comparing sequences from      14139  to      14307\n",
      "99.9%---------- new table with      120 representatives\n",
      "# comparing sequences from      14307  to      15149\n",
      ".....................---------- new table with      611 representatives\n",
      "\n",
      "    15149  finished      10780  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 92M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 2.31\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Onygenales"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'onygenales'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Amauroascus_mutatus_GCA_001430935\",\n",
    "               \"Amauroascus_niger_GCA_001430945\",\n",
    "               \"Ascospaera_apis_GCA_001636715\",\n",
    "               \"Blastomyces_dermatitidis_GCA_000003525\",\n",
    "               \"Blastomyces_gilchristii_GCF_000003855\",\n",
    "               \"Blastomyces_parvus_GCA_002572885\",\n",
    "               \"Blastomyces_percursus_GCA_003206225\",\n",
    "               \"Blastomyces_silverae_GCA_001014755\",\n",
    "               \"Blastomyces_sp_GCA_003206725\",\n",
    "               \"Byssoonygena_ceratinophila_GCA_001430925\",\n",
    "               \"Chrysosprium_queenslandicum_GCA_001430955\",\n",
    "               \"Coccidioides_immitis_GCF_000149335\",\n",
    "               \"Coccidioides_posadasii_GCF_000151335\",\n",
    "               \"Emergomyces_orientalis_GCA_002110485\",\n",
    "               \"Emergomyces_pasteurianus_GCA_001883825\",\n",
    "               \"Emmonsia_crescens_GCA_002572855\",\n",
    "               \"Emmonsia_sp_GCA_001660665\",\n",
    "               \"Helicocarpus_griseus_GCA_002573585\",\n",
    "               \"Malbranchea_cinnamomea_GCA_900128795\",\n",
    "               \"Microsprum_canis_GCF_000151145\",\n",
    "               \"Nannizzia_gypsea_GCF_000150975\",\n",
    "               \"Onygena_corvina_GCA_000812245\",\n",
    "               \"Ophidiomyces_ophiodiicola_GCA_002167195\",\n",
    "               \"Paracoccidioides_brasiliensis_GCF_000150735\",\n",
    "               \"Paracoccidioides_lutzii_GCF_000150705\",\n",
    "               \"Polytolypa_hystricis_GCA_002573605\",\n",
    "               \"Trichophyton_benhamiae_GCF_000151125\",\n",
    "               \"Trichophyton_equinum_GCA_000151175\",\n",
    "               \"Trichophyton_interdigitale_GCA_000622975\",\n",
    "               \"Trichophyton_mentagrophytes_GCA_003664465\",\n",
    "               \"Trichophyton_rubrum_GCF_000151425\",\n",
    "               \"Trichophyton_soudanense_GCA_000616865\",\n",
    "               \"Trichophyton_tonsurans_GCA_000151455\",\n",
    "               \"Trichophyton_verrucosum_GCF_000151505\",\n",
    "               \"Trichophyton_violaceum_GCA_001651435\",\n",
    "               \"Uncinocarpus_reesii_GCF_000003515\" ]\n",
    "reference_taxon = \"Histoplasma_capsulatum_GCF_000149585\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001430935.1_ASM143093v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8869677 bytes  3547887.60 bytes/sec\n",
      "total size is 8867402  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001430945.1_ASM143094v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11112123 bytes  4444866.00 bytes/sec\n",
      "total size is 11109296  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001636715.1_AAP_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6406318 bytes  753689.41 bytes/sec\n",
      "total size is 6404648  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000003525.2_BD_ER3_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20167502 bytes  5762155.43 bytes/sec\n",
      "total size is 20162469  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000003855.2_BD_SLH14081_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 22662879 bytes  1295024.06 bytes/sec\n",
      "total size is 22657233  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002572885.1_Emmon_parv_130_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8864238 bytes  3545712.00 bytes/sec\n",
      "total size is 8861960  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003206225.1_ASM320622v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10439642 bytes  6959789.33 bytes/sec\n",
      "total size is 10436983  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001014755.1_ASM101475v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9869194 bytes  6579490.67 bytes/sec\n",
      "total size is 9866671  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003206725.1_ASM320672v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10903707 bytes  4361499.60 bytes/sec\n",
      "total size is 10900936  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001430925.1_ASM143092v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8205183 bytes  5470150.00 bytes/sec\n",
      "total size is 8203069  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001430955.1_ASM143095v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9716061 bytes  6477402.00 bytes/sec\n",
      "total size is 9713578  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000149335.2_ASM14933v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9036510 bytes  6024368.00 bytes/sec\n",
      "total size is 9034196  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151335.1_JCVI-cpa1-1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8421143 bytes  3368474.00 bytes/sec\n",
      "total size is 8418978  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002110485.1_ASM211048v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11144993 bytes  891602.80 bytes/sec\n",
      "total size is 11142158  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001883825.1_Emmo_past_UAMH9510_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10380655 bytes  4152278.80 bytes/sec\n",
      "total size is 10378002  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002572855.1_Emmon_cres_4076_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10722861 bytes  7148602.00 bytes/sec\n",
      "total size is 10720123  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001660665.1_Emmo_afri_EA111_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9553263 bytes  6368870.00 bytes/sec\n",
      "total size is 9550816  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002573585.1_Heli_gris_5409_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10240156 bytes  4096079.20 bytes/sec\n",
      "total size is 10237539  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000149585.1_ASM14958v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9737714 bytes  3895102.40 bytes/sec\n",
      "total size is 9735224  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900128795.2_FCH_10_5_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7955435 bytes  3182190.80 bytes/sec\n",
      "total size is 7953388  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151145.1_ASM15114v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7339135 bytes  4892784.67 bytes/sec\n",
      "total size is 7337238  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000150975.2_MS_CBS118893_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7334051 bytes  4889395.33 bytes/sec\n",
      "total size is 7332152  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000812245.1_ASM81224v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6790128 bytes  2716068.00 bytes/sec\n",
      "total size is 6788359  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002167195.1_ASM216719v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6777112 bytes  366332.65 bytes/sec\n",
      "total size is 6775350  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000150735.2_Paracocci_br_Pb18_V2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9317065 bytes  3726842.80 bytes/sec\n",
      "total size is 9314669  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000150705.2_Paracocci_br_Pb01_V2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10291032 bytes  4116429.60 bytes/sec\n",
      "total size is 10288404  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002573605.1_Poly_hyst_7299_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 10943340 bytes  4377352.80 bytes/sec\n",
      "total size is 10940555  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151125.2_ASM15112v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7089848 bytes  2835956.00 bytes/sec\n",
      "total size is 7088007  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151175.1_ASM15117v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7432053 bytes  2972838.00 bytes/sec\n",
      "total size is 7430132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000622975.1_Tric_inte_MR816_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7172896 bytes  4781958.67 bytes/sec\n",
      "total size is 7171031  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003664465.1_ASM366446v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7591794 bytes  3036734.40 bytes/sec\n",
      "total size is 7589832  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151425.1_ASM15142v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7036201 bytes  2814497.20 bytes/sec\n",
      "total size is 7034376  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000616865.1_Tric_soud_CBS_452_61_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7241144 bytes  499392.14 bytes/sec\n",
      "total size is 7239258  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151455.1_ASM15145v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7165270 bytes  4776874.67 bytes/sec\n",
      "total size is 7163413  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000151505.1_ASM15150v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7208128 bytes  2883268.00 bytes/sec\n",
      "total size is 7206263  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001651435.1_CMCC_F_T3l_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7303585 bytes  2921450.80 bytes/sec\n",
      "total size is 7301696  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000003515.2_ASM351v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6887227 bytes  2754907.60 bytes/sec\n",
      "total size is 6885436  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "30204if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485968 (\"sim_Amauroascus_mutatus_GCA_001430935\") has been submitted\n",
      "Your job 5485969 (\"sim_Amauroascus_niger_GCA_001430945\") has been submitted\n",
      "Your job 5485970 (\"sim_Ascospaera_apis_GCA_001636715\") has been submitted\n",
      "Your job 5485971 (\"sim_Blastomyces_dermatitidis_GCA_000003525\") has been submitted\n",
      "Your job 5485972 (\"sim_Blastomyces_gilchristii_GCF_000003855\") has been submitted\n",
      "Your job 5485973 (\"sim_Blastomyces_parvus_GCA_002572885\") has been submitted\n",
      "Your job 5485974 (\"sim_Blastomyces_percursus_GCA_003206225\") has been submitted\n",
      "Your job 5485975 (\"sim_Blastomyces_silverae_GCA_001014755\") has been submitted\n",
      "Your job 5485976 (\"sim_Blastomyces_sp_GCA_003206725\") has been submitted\n",
      "Your job 5485977 (\"sim_Byssoonygena_ceratinophila_GCA_001430925\") has been submitted\n",
      "Your job 5485978 (\"sim_Chrysosprium_queenslandicum_GCA_001430955\") has been submitted\n",
      "Your job 5485979 (\"sim_Coccidioides_immitis_GCF_000149335\") has been submitted\n",
      "Your job 5485980 (\"sim_Coccidioides_posadasii_GCF_000151335\") has been submitted\n",
      "Your job 5485981 (\"sim_Emergomyces_orientalis_GCA_002110485\") has been submitted\n",
      "Your job 5485982 (\"sim_Emergomyces_pasteurianus_GCA_001883825\") has been submitted\n",
      "Your job 5485983 (\"sim_Emmonsia_crescens_GCA_002572855\") has been submitted\n",
      "Your job 5485984 (\"sim_Emmonsia_sp_GCA_001660665\") has been submitted\n",
      "Your job 5485985 (\"sim_Helicocarpus_griseus_GCA_002573585\") has been submitted\n",
      "Your job 5485986 (\"sim_Malbranchea_cinnamomea_GCA_900128795\") has been submitted\n",
      "Your job 5485987 (\"sim_Microsprum_canis_GCF_000151145\") has been submitted\n",
      "Your job 5485988 (\"sim_Nannizzia_gypsea_GCF_000150975\") has been submitted\n",
      "Your job 5485989 (\"sim_Onygena_corvina_GCA_000812245\") has been submitted\n",
      "Your job 5485990 (\"sim_Ophidiomyces_ophiodiicola_GCA_002167195\") has been submitted\n",
      "Your job 5485991 (\"sim_Paracoccidioides_brasiliensis_GCF_000150735\") has been submitted\n",
      "Your job 5485992 (\"sim_Paracoccidioides_lutzii_GCF_000150705\") has been submitted\n",
      "Your job 5485993 (\"sim_Polytolypa_hystricis_GCA_002573605\") has been submitted\n",
      "Your job 5485994 (\"sim_Trichophyton_benhamiae_GCF_000151125\") has been submitted\n",
      "Your job 5485995 (\"sim_Trichophyton_equinum_GCA_000151175\") has been submitted\n",
      "Your job 5485996 (\"sim_Trichophyton_interdigitale_GCA_000622975\") has been submitted\n",
      "Your job 5485997 (\"sim_Trichophyton_mentagrophytes_GCA_003664465\") has been submitted\n",
      "Your job 5485998 (\"sim_Trichophyton_rubrum_GCF_000151425\") has been submitted\n",
      "Your job 5485999 (\"sim_Trichophyton_soudanense_GCA_000616865\") has been submitted\n",
      "Your job 5486000 (\"sim_Trichophyton_tonsurans_GCA_000151455\") has been submitted\n",
      "Your job 5486001 (\"sim_Trichophyton_verrucosum_GCF_000151505\") has been submitted\n",
      "Your job 5486002 (\"sim_Trichophyton_violaceum_GCA_001651435\") has been submitted\n",
      "Your job 5486003 (\"sim_Uncinocarpus_reesii_GCF_000003515\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/onygenales/cleaned_genomes/Histoplasma_capsulatum_GCF_000149585_formatted.fas path=results/phyluce/onygenales/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486004 (\"map_Amauroascus_mutatus_GCA_001430935\") has been submitted\n",
      "Your job 5486005 (\"map_Amauroascus_niger_GCA_001430945\") has been submitted\n",
      "Your job 5486006 (\"map_Ascospaera_apis_GCA_001636715\") has been submitted\n",
      "Your job 5486007 (\"map_Blastomyces_dermatitidis_GCA_000003525\") has been submitted\n",
      "Your job 5486008 (\"map_Blastomyces_gilchristii_GCF_000003855\") has been submitted\n",
      "Your job 5486009 (\"map_Blastomyces_parvus_GCA_002572885\") has been submitted\n",
      "Your job 5486010 (\"map_Blastomyces_percursus_GCA_003206225\") has been submitted\n",
      "Your job 5486011 (\"map_Blastomyces_silverae_GCA_001014755\") has been submitted\n",
      "Your job 5486012 (\"map_Blastomyces_sp_GCA_003206725\") has been submitted\n",
      "Your job 5486013 (\"map_Byssoonygena_ceratinophila_GCA_001430925\") has been submitted\n",
      "Your job 5486014 (\"map_Chrysosprium_queenslandicum_GCA_001430955\") has been submitted\n",
      "Your job 5486015 (\"map_Coccidioides_immitis_GCF_000149335\") has been submitted\n",
      "Your job 5486016 (\"map_Coccidioides_posadasii_GCF_000151335\") has been submitted\n",
      "Your job 5486017 (\"map_Emergomyces_orientalis_GCA_002110485\") has been submitted\n",
      "Your job 5486018 (\"map_Emergomyces_pasteurianus_GCA_001883825\") has been submitted\n",
      "Your job 5486019 (\"map_Emmonsia_crescens_GCA_002572855\") has been submitted\n",
      "Your job 5486020 (\"map_Emmonsia_sp_GCA_001660665\") has been submitted\n",
      "Your job 5486021 (\"map_Helicocarpus_griseus_GCA_002573585\") has been submitted\n",
      "Your job 5486022 (\"map_Malbranchea_cinnamomea_GCA_900128795\") has been submitted\n",
      "Your job 5486023 (\"map_Microsprum_canis_GCF_000151145\") has been submitted\n",
      "Your job 5486024 (\"map_Nannizzia_gypsea_GCF_000150975\") has been submitted\n",
      "Your job 5486025 (\"map_Onygena_corvina_GCA_000812245\") has been submitted\n",
      "Your job 5486026 (\"map_Ophidiomyces_ophiodiicola_GCA_002167195\") has been submitted\n",
      "Your job 5486027 (\"map_Paracoccidioides_brasiliensis_GCF_000150735\") has been submitted\n",
      "Your job 5486028 (\"map_Paracoccidioides_lutzii_GCF_000150705\") has been submitted\n",
      "Your job 5486029 (\"map_Polytolypa_hystricis_GCA_002573605\") has been submitted\n",
      "Your job 5486030 (\"map_Trichophyton_benhamiae_GCF_000151125\") has been submitted\n",
      "Your job 5486031 (\"map_Trichophyton_equinum_GCA_000151175\") has been submitted\n",
      "Your job 5486032 (\"map_Trichophyton_interdigitale_GCA_000622975\") has been submitted\n",
      "Your job 5486033 (\"map_Trichophyton_mentagrophytes_GCA_003664465\") has been submitted\n",
      "Your job 5486034 (\"map_Trichophyton_rubrum_GCF_000151425\") has been submitted\n",
      "Your job 5486035 (\"map_Trichophyton_soudanense_GCA_000616865\") has been submitted\n",
      "Your job 5486036 (\"map_Trichophyton_tonsurans_GCA_000151455\") has been submitted\n",
      "Your job 5486037 (\"map_Trichophyton_verrucosum_GCF_000151505\") has been submitted\n",
      "Your job 5486038 (\"map_Trichophyton_violaceum_GCA_001651435\") has been submitted\n",
      "Your job 5486039 (\"map_Uncinocarpus_reesii_GCF_000003515\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486040 (\"merge_Amauroascus_mutatus_GCA_001430935\") has been submitted\n",
      "Your job 5486041 (\"merge_Amauroascus_niger_GCA_001430945\") has been submitted\n",
      "Your job 5486042 (\"merge_Ascospaera_apis_GCA_001636715\") has been submitted\n",
      "Your job 5486043 (\"merge_Blastomyces_dermatitidis_GCA_000003525\") has been submitted\n",
      "Your job 5486044 (\"merge_Blastomyces_gilchristii_GCF_000003855\") has been submitted\n",
      "Your job 5486045 (\"merge_Blastomyces_parvus_GCA_002572885\") has been submitted\n",
      "Your job 5486046 (\"merge_Blastomyces_percursus_GCA_003206225\") has been submitted\n",
      "Your job 5486047 (\"merge_Blastomyces_silverae_GCA_001014755\") has been submitted\n",
      "Your job 5486048 (\"merge_Blastomyces_sp_GCA_003206725\") has been submitted\n",
      "Your job 5486049 (\"merge_Byssoonygena_ceratinophila_GCA_001430925\") has been submitted\n",
      "Your job 5486050 (\"merge_Chrysosprium_queenslandicum_GCA_001430955\") has been submitted\n",
      "Your job 5486051 (\"merge_Coccidioides_immitis_GCF_000149335\") has been submitted\n",
      "Your job 5486052 (\"merge_Coccidioides_posadasii_GCF_000151335\") has been submitted\n",
      "Your job 5486053 (\"merge_Emergomyces_orientalis_GCA_002110485\") has been submitted\n",
      "Your job 5486054 (\"merge_Emergomyces_pasteurianus_GCA_001883825\") has been submitted\n",
      "Your job 5486055 (\"merge_Emmonsia_crescens_GCA_002572855\") has been submitted\n",
      "Your job 5486056 (\"merge_Emmonsia_sp_GCA_001660665\") has been submitted\n",
      "Your job 5486057 (\"merge_Helicocarpus_griseus_GCA_002573585\") has been submitted\n",
      "Your job 5486058 (\"merge_Malbranchea_cinnamomea_GCA_900128795\") has been submitted\n",
      "Your job 5486059 (\"merge_Microsprum_canis_GCF_000151145\") has been submitted\n",
      "Your job 5486060 (\"merge_Nannizzia_gypsea_GCF_000150975\") has been submitted\n",
      "Your job 5486061 (\"merge_Onygena_corvina_GCA_000812245\") has been submitted\n",
      "Your job 5486062 (\"merge_Ophidiomyces_ophiodiicola_GCA_002167195\") has been submitted\n",
      "Your job 5486063 (\"merge_Paracoccidioides_brasiliensis_GCF_000150735\") has been submitted\n",
      "Your job 5486064 (\"merge_Paracoccidioides_lutzii_GCF_000150705\") has been submitted\n",
      "Your job 5486065 (\"merge_Polytolypa_hystricis_GCA_002573605\") has been submitted\n",
      "Your job 5486066 (\"merge_Trichophyton_benhamiae_GCF_000151125\") has been submitted\n",
      "Your job 5486067 (\"merge_Trichophyton_equinum_GCA_000151175\") has been submitted\n",
      "Your job 5486068 (\"merge_Trichophyton_interdigitale_GCA_000622975\") has been submitted\n",
      "Your job 5486069 (\"merge_Trichophyton_mentagrophytes_GCA_003664465\") has been submitted\n",
      "Your job 5486070 (\"merge_Trichophyton_rubrum_GCF_000151425\") has been submitted\n",
      "Your job 5486071 (\"merge_Trichophyton_soudanense_GCA_000616865\") has been submitted\n",
      "Your job 5486072 (\"merge_Trichophyton_tonsurans_GCA_000151455\") has been submitted\n",
      "Your job 5486073 (\"merge_Trichophyton_verrucosum_GCF_000151505\") has been submitted\n",
      "Your job 5486074 (\"merge_Trichophyton_violaceum_GCA_001651435\") has been submitted\n",
      "Your job 5486075 (\"merge_Uncinocarpus_reesii_GCF_000003515\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 350 sequences from Amauroascus_mutatus_GCA_001430935_merged.bed.  Filtered 258 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 92.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 20819 sequences from Blastomyces_dermatitidis_GCA_000003525_merged.bed.  Filtered 12222 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8597.\n",
      "Screened 21557 sequences from Blastomyces_gilchristii_GCF_000003855_merged.bed.  Filtered 12671 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8886.\n",
      "Screened 17946 sequences from Blastomyces_parvus_GCA_002572885_merged.bed.  Filtered 10734 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7212.\n",
      "Screened 20335 sequences from Blastomyces_percursus_GCA_003206225_merged.bed.  Filtered 12072 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8263.\n",
      "Screened 19925 sequences from Blastomyces_silverae_GCA_001014755_merged.bed.  Filtered 11438 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8487.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 270 sequences from Chrysosprium_queenslandicum_GCA_001430955_merged.bed.  Filtered 199 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 71.\n",
      "Screened 713 sequences from Coccidioides_immitis_GCF_000149335_merged.bed.  Filtered 440 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 273.\n",
      "Screened 444 sequences from Coccidioides_posadasii_GCF_000151335_merged.bed.  Filtered 277 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 167.\n",
      "Screened 20949 sequences from Emergomyces_orientalis_GCA_002110485_merged.bed.  Filtered 12425 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8524.\n",
      "Screened 14944 sequences from Emergomyces_pasteurianus_GCA_001883825_merged.bed.  Filtered 9775 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5169.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 12660 sequences from Emmonsia_sp_GCA_001660665_merged.bed.  Filtered 8343 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4317.\n",
      "Screened 2570 sequences from Helicocarpus_griseus_GCA_002573585_merged.bed.  Filtered 2024 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 546.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 4046 sequences from Paracoccidioides_brasiliensis_GCF_000150735_merged.bed.  Filtered 3048 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 998.\n",
      "Screened 4520 sequences from Paracoccidioides_lutzii_GCF_000150705_merged.bed.  Filtered 3325 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1195.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 212 sequences from Trichophyton_mentagrophytes_GCA_003664465_merged.bed.  Filtered 157 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 55.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/onygenales/initial_intervals/Uncinocarpus_reesii_GCF_000003515_merged.bed --twobit results/phyluce/onygenales/cleaned_genomes/Histoplasma_capsulatum_GCF_000149585_formatted.2bit --output results/phyluce/onygenales/initial_intervals/Uncinocarpus_reesii_GCF_000003515_stripped.bed;\n",
      "amauroascus_mutatus_gca_001430935.\n",
      "amauroascus_niger_gca_001430945\n",
      "ascospaera_apis_gca_001636715\n",
      "blastomyces_dermatitidis_gca_000003525.........\n",
      "blastomyces_gilchristii_gcf_000003855.........\n",
      "blastomyces_parvus_gca_002572885........\n",
      "blastomyces_percursus_gca_003206225.........\n",
      "blastomyces_silverae_gca_001014755.........\n",
      "blastomyces_sp_gca_003206725\n",
      "byssoonygena_ceratinophila_gca_001430925\n",
      "chrysosprium_queenslandicum_gca_001430955.\n",
      "coccidioides_immitis_gcf_000149335.\n",
      "coccidioides_posadasii_gcf_000151335.\n",
      "emergomyces_orientalis_gca_002110485.........\n",
      "emergomyces_pasteurianus_gca_001883825......\n",
      "emmonsia_crescens_gca_002572855\n",
      "emmonsia_sp_gca_001660665.....\n",
      "helicocarpus_griseus_gca_002573585.\n",
      "malbranchea_cinnamomea_gca_900128795\n",
      "microsprum_canis_gcf_000151145\n",
      "nannizzia_gypsea_gcf_000150975\n",
      "onygena_corvina_gca_000812245\n",
      "ophidiomyces_ophiodiicola_gca_002167195\n",
      "paracoccidioides_brasiliensis_gcf_000150735.\n",
      "paracoccidioides_lutzii_gcf_000150705..\n",
      "polytolypa_hystricis_gca_002573605\n",
      "trichophyton_benhamiae_gcf_000151125\n",
      "trichophyton_equinum_gca_000151175\n",
      "trichophyton_interdigitale_gca_000622975\n",
      "trichophyton_mentagrophytes_gca_003664465.\n",
      "trichophyton_rubrum_gcf_000151425\n",
      "trichophyton_soudanense_gca_000616865\n",
      "trichophyton_tonsurans_gca_000151455\n",
      "trichophyton_verrucosum_gcf_000151505\n",
      "trichophyton_violaceum_gca_001651435\n",
      "uncinocarpus_reesii_gcf_000003515\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/onygenales/initial_intervals/onygenales-to-Histoplasma_capsulatum_GCF_000149585.sqlite --base-taxon Histoplasma_capsulatum_GCF_000149585\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 0 taxa:\t16,588.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 1 taxa:\t16,588.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 2 taxa:\t12,521.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 3 taxa:\t9,887.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 4 taxa:\t7,898.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 5 taxa:\t6,143.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 6 taxa:\t4,434.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 7 taxa:\t3,004.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 8 taxa:\t1,812.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 9 taxa:\t618.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 10 taxa:\t329.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 11 taxa:\t146.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 12 taxa:\t29.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 13 taxa:\t13.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 14 taxa:\t10.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 15 taxa:\t5.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 16 taxa:\t2.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 17 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 18 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 19 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 20 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 21 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 22 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 23 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 24 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 25 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 26 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 27 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 28 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 29 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 30 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 31 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 32 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 33 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 34 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 35 taxa:\t0.0\n",
      "Loci shared by Histoplasma_capsulatum_GCF_000149585 + 36 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/onygenales/initial_intervals/onygenales-to-Histoplasma_capsulatum_GCF_000149585.sqlite --base-taxon Histoplasma_capsulatum_GCF_000149585 --output results/phyluce/onygenales/initial_intervals/Histoplasma_capsulatum_GCF_000149585_+9.bed --specific-counts 9;\n",
      "Counter({'blastomyces_gilchristii_gcf_000003855': 614, 'blastomyces_dermatitidis_gca_000003525': 612, 'blastomyces_percursus_gca_003206225': 611, 'emergomyces_orientalis_gca_002110485': 605, 'blastomyces_parvus_gca_002572885': 591, 'emergomyces_pasteurianus_gca_001883825': 581, 'emmonsia_sp_gca_001660665': 572, 'blastomyces_silverae_gca_001014755': 572, 'paracoccidioides_brasiliensis_gcf_000150735': 438, 'paracoccidioides_lutzii_gcf_000150705': 409, 'helicocarpus_griseus_gca_002573585': 326, 'coccidioides_posadasii_gcf_000151335': 45, 'coccidioides_immitis_gcf_000149335': 42, 'trichophyton_mentagrophytes_gca_003664465': 32, 'chrysosprium_queenslandicum_gca_001430955': 31, 'amauroascus_mutatus_gca_001430935': 15})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 9\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/onygenales/initial_intervals/Histoplasma_capsulatum_GCF_000149585_+9.bed --twobit results/phyluce/onygenales/cleaned_genomes/Histoplasma_capsulatum_GCF_000149585_formatted.2bit --buffer-to 160 --output results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9.fasta;\n",
      "Screened 618 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 618.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9.fasta --probe-prefix uce_onygenales_ --design onygenales_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "G\n",
      "\n",
      "\n",
      "Conserved locus count = 618\n",
      "Probe Count = 1235\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas --query results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes_vself.lastz;\n",
      "Started:  Mon Feb 10, 2020  13:46:56\n",
      "Ended:  Mon Feb 10, 2020  13:46:56\n",
      "Time for execution:  0.00786991516749 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas                        --lastz results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes_vself.lastz                       --probe-prefix=uce_onygenales_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 1234 fasta sequences.  Filtered 8 duplicates. Kept 1219.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/onygenales/validate_intervals/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas --scaffoldlist Amauroascus_mutatus_GCA_001430935 Amauroascus_niger_GCA_001430945 Ascospaera_apis_GCA_001636715 Blastomyces_dermatitidis_GCA_000003525 Blastomyces_gilchristii_GCF_000003855 Blastomyces_parvus_GCA_002572885 Blastomyces_percursus_GCA_003206225 Blastomyces_silverae_GCA_001014755 Blastomyces_sp_GCA_003206725 Byssoonygena_ceratinophila_GCA_001430925 Chrysosprium_queenslandicum_GCA_001430955 Coccidioides_immitis_GCF_000149335 Coccidioides_posadasii_GCF_000151335 Emergomyces_orientalis_GCA_002110485 Emergomyces_pasteurianus_GCA_001883825 Emmonsia_crescens_GCA_002572855 Emmonsia_sp_GCA_001660665 Helicocarpus_griseus_GCA_002573585 Malbranchea_cinnamomea_GCA_900128795 Microsprum_canis_GCF_000151145 Nannizzia_gypsea_GCF_000150975 Onygena_corvina_GCA_000812245 Ophidiomyces_ophiodiicola_GCA_002167195 Paracoccidioides_brasiliensis_GCF_000150735 Paracoccidioides_lutzii_GCF_000150705 Polytolypa_hystricis_GCA_002573605 Trichophyton_benhamiae_GCF_000151125 Trichophyton_equinum_GCA_000151175 Trichophyton_interdigitale_GCA_000622975 Trichophyton_mentagrophytes_GCA_003664465 Trichophyton_rubrum_GCF_000151425 Trichophyton_soudanense_GCA_000616865 Trichophyton_tonsurans_GCA_000151455 Trichophyton_verrucosum_GCF_000151505 Trichophyton_violaceum_GCA_001651435 Uncinocarpus_reesii_GCF_000003515 Histoplasma_capsulatum_GCF_000149585 --genome-base-path results/phyluce/onygenales/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/onygenales/validate_intervals/onygenales-to-Histoplasma_capsulatum_GCF_000149585.sqlite --output results/phyluce/onygenales/validate_intervals/lastz/;\n",
      "\n",
      "Running against Amauroascus_mutatus_GCA_001430935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpo7j2g1.fasta\n",
      "\t/tmp/tmp932PKZ.fasta\n",
      "\t/tmp/tmp66wR1I.fasta\n",
      "\t/tmp/tmp6qjBNd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSXyP1y.lastz\n",
      "\t/tmp/tmpLwJ2mt.lastz\n",
      "\t/tmp/tmpeotdUm.lastz\n",
      "\t/tmp/tmpU0k3ke.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Amauroascus_mutatus_GCA_001430935.lastz\n",
      "Creating Amauroascus_mutatus_GCA_001430935 table\n",
      "Inserting data to Amauroascus_mutatus_GCA_001430935 table\n",
      "\n",
      "Running against Amauroascus_niger_GCA_001430945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpff0TpN.fasta\n",
      "\t/tmp/tmpyxNBRK.fasta\n",
      "\t/tmp/tmpWUANBD.fasta\n",
      "\t/tmp/tmpcPpWqS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGmOt6_.lastz\n",
      "\t/tmp/tmpS05qgY.lastz\n",
      "\t/tmp/tmpa2tUEO.lastz\n",
      "\t/tmp/tmpyF6t5q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Amauroascus_niger_GCA_001430945.lastz\n",
      "Creating Amauroascus_niger_GCA_001430945 table\n",
      "Inserting data to Amauroascus_niger_GCA_001430945 table\n",
      "\n",
      "Running against Ascospaera_apis_GCA_001636715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpPaK_Ds.fasta\n",
      "\t/tmp/tmpFPNNsJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpm0r8mK.lastz\n",
      "\t/tmp/tmpB1se4b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Ascospaera_apis_GCA_001636715.lastz\n",
      "Creating Ascospaera_apis_GCA_001636715 table\n",
      "Inserting data to Ascospaera_apis_GCA_001636715 table\n",
      "\n",
      "Running against Blastomyces_dermatitidis_GCA_000003525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpT9kNxa.fasta\n",
      "\t/tmp/tmpgKZ20v.fasta\n",
      "\t/tmp/tmptlEvXx.fasta\n",
      "\t/tmp/tmpUUQ4eG.fasta\n",
      "\t/tmp/tmp5fQtYh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpU_k_TN.lastz\n",
      "\t/tmp/tmpYsCREF.lastz\n",
      "\t/tmp/tmpQ30NhG.lastz\n",
      "\t/tmp/tmp9uuDrf.lastz\n",
      "\t/tmp/tmpcuqfb5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Blastomyces_dermatitidis_GCA_000003525.lastz\n",
      "Creating Blastomyces_dermatitidis_GCA_000003525 table\n",
      "Inserting data to Blastomyces_dermatitidis_GCA_000003525 table\n",
      "\n",
      "Running against Blastomyces_gilchristii_GCF_000003855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmp_hBKeC.fasta\n",
      "\t/tmp/tmpvbh8K5.fasta\n",
      "\t/tmp/tmpEs7QN2.fasta\n",
      "\t/tmp/tmpSG_e5D.fasta\n",
      "\t/tmp/tmpOW4O3A.fasta\n",
      "\t/tmp/tmpTX_uA9.fasta\n",
      "\t/tmp/tmpYyuBHx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpu9tYQf.lastz\n",
      "\t/tmp/tmpqNKfwx.lastz\n",
      "\t/tmp/tmpPGRFqh.lastz\n",
      "\t/tmp/tmpDhzz3F.lastz\n",
      "\t/tmp/tmpq_c6v_.lastz\n",
      "\t/tmp/tmpQrini0.lastz\n",
      "\t/tmp/tmpc14xP0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Blastomyces_gilchristii_GCF_000003855.lastz\n",
      "Creating Blastomyces_gilchristii_GCF_000003855 table\n",
      "Inserting data to Blastomyces_gilchristii_GCF_000003855 table\n",
      "\n",
      "Running against Blastomyces_parvus_GCA_002572885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpTB3drb.fasta\n",
      "\t/tmp/tmpzatnks.fasta\n",
      "\t/tmp/tmpByWAeF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1DcVLs.lastz\n",
      "\t/tmp/tmpAC5chZ.lastz\n",
      "\t/tmp/tmpCkY0Yg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Blastomyces_parvus_GCA_002572885.lastz\n",
      "Creating Blastomyces_parvus_GCA_002572885 table\n",
      "Inserting data to Blastomyces_parvus_GCA_002572885 table\n",
      "\n",
      "Running against Blastomyces_percursus_GCA_003206225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpxkGr6i.fasta\n",
      "\t/tmp/tmpNvdLhp.fasta\n",
      "\t/tmp/tmpTbRnVw.fasta\n",
      "\t/tmp/tmpcm0V4G.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpc9OxFE.lastz\n",
      "\t/tmp/tmpeOzHlr.lastz\n",
      "\t/tmp/tmpvtMD7k.lastz\n",
      "\t/tmp/tmpIf00AA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Blastomyces_percursus_GCA_003206225.lastz\n",
      "Creating Blastomyces_percursus_GCA_003206225 table\n",
      "Inserting data to Blastomyces_percursus_GCA_003206225 table\n",
      "\n",
      "Running against Blastomyces_silverae_GCA_001014755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpBb6Hu4.fasta\n",
      "\t/tmp/tmpi_I0z3.fasta\n",
      "\t/tmp/tmpQebLVo.fasta\n",
      "\t/tmp/tmpZSXGHg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpT8hT3f.lastz\n",
      "\t/tmp/tmpyxFQoA.lastz\n",
      "\t/tmp/tmpqdNAiN.lastz\n",
      "\t/tmp/tmpd3zkUS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Blastomyces_silverae_GCA_001014755.lastz\n",
      "Creating Blastomyces_silverae_GCA_001014755 table\n",
      "Inserting data to Blastomyces_silverae_GCA_001014755 table\n",
      "\n",
      "Running against Blastomyces_sp_GCA_003206725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpbGcrug.fasta\n",
      "\t/tmp/tmp_7ZkCj.fasta\n",
      "\t/tmp/tmpsv97b6.fasta\n",
      "\t/tmp/tmp5FcGTf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdaOFwK.lastz\n",
      "\t/tmp/tmpFfLfpP.lastz\n",
      "\t/tmp/tmpkXddU3.lastz\n",
      "\t/tmp/tmpaMwTbf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Blastomyces_sp_GCA_003206725.lastz\n",
      "Creating Blastomyces_sp_GCA_003206725 table\n",
      "Inserting data to Blastomyces_sp_GCA_003206725 table\n",
      "\n",
      "Running against Byssoonygena_ceratinophila_GCA_001430925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpUNunqp.fasta\n",
      "\t/tmp/tmpwBJQIA.fasta\n",
      "\t/tmp/tmpYWYdgP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpb659Bz.lastz\n",
      "\t/tmp/tmpbO2SI2.lastz\n",
      "\t/tmp/tmpZUzx6p.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Byssoonygena_ceratinophila_GCA_001430925.lastz\n",
      "Creating Byssoonygena_ceratinophila_GCA_001430925 table\n",
      "Inserting data to Byssoonygena_ceratinophila_GCA_001430925 table\n",
      "\n",
      "Running against Chrysosprium_queenslandicum_GCA_001430955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp_jKUd0.fasta\n",
      "\t/tmp/tmpCzN5Of.fasta\n",
      "\t/tmp/tmpUdWzd9.fasta\n",
      "\t/tmp/tmpQdaeZU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9K0lUI.lastz\n",
      "\t/tmp/tmptmJW5D.lastz\n",
      "\t/tmp/tmptrqzUT.lastz\n",
      "\t/tmp/tmpJeJpab.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Chrysosprium_queenslandicum_GCA_001430955.lastz\n",
      "Creating Chrysosprium_queenslandicum_GCA_001430955 table\n",
      "Inserting data to Chrysosprium_queenslandicum_GCA_001430955 table\n",
      "\n",
      "Running against Coccidioides_immitis_GCF_000149335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpC9OteC.fasta\n",
      "\t/tmp/tmpOLqBTf.fasta\n",
      "\t/tmp/tmpPITmSQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGp4hmP.lastz\n",
      "\t/tmp/tmpGUM28n.lastz\n",
      "\t/tmp/tmpcf5oQx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Coccidioides_immitis_GCF_000149335.lastz\n",
      "Creating Coccidioides_immitis_GCF_000149335 table\n",
      "Inserting data to Coccidioides_immitis_GCF_000149335 table\n",
      "\n",
      "Running against Coccidioides_posadasii_GCF_000151335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpIZBDyE.fasta\n",
      "\t/tmp/tmpDh_bYa.fasta\n",
      "\t/tmp/tmpVsu0Bv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJJArmN.lastz\n",
      "\t/tmp/tmp53OXGi.lastz\n",
      "\t/tmp/tmp_BwPXt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Coccidioides_posadasii_GCF_000151335.lastz\n",
      "Creating Coccidioides_posadasii_GCF_000151335 table\n",
      "Inserting data to Coccidioides_posadasii_GCF_000151335 table\n",
      "\n",
      "Running against Emergomyces_orientalis_GCA_002110485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpPrgDkJ.fasta\n",
      "\t/tmp/tmpSJQDCL.fasta\n",
      "\t/tmp/tmpg5uwwe.fasta\n",
      "\t/tmp/tmpz1Z7O9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpF3QXSz.lastz\n",
      "\t/tmp/tmpmwvGwY.lastz\n",
      "\t/tmp/tmpVJzAc6.lastz\n",
      "\t/tmp/tmp15wfp6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Emergomyces_orientalis_GCA_002110485.lastz\n",
      "Creating Emergomyces_orientalis_GCA_002110485 table\n",
      "Inserting data to Emergomyces_orientalis_GCA_002110485 table\n",
      "\n",
      "Running against Emergomyces_pasteurianus_GCA_001883825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpqMIdz7.fasta\n",
      "\t/tmp/tmpHl_kJl.fasta\n",
      "\t/tmp/tmpcw1i22.fasta\n",
      "\t/tmp/tmpjJh9ci.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptdlH19.lastz\n",
      "\t/tmp/tmpN9HpN_.lastz\n",
      "\t/tmp/tmpKoBrDj.lastz\n",
      "\t/tmp/tmp1SQYp5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Emergomyces_pasteurianus_GCA_001883825.lastz\n",
      "Creating Emergomyces_pasteurianus_GCA_001883825 table\n",
      "Inserting data to Emergomyces_pasteurianus_GCA_001883825 table\n",
      "\n",
      "Running against Emmonsia_crescens_GCA_002572855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmp9JFkNG.fasta\n",
      "\t/tmp/tmpURTAWd.fasta\n",
      "\t/tmp/tmpo0RPGz.fasta\n",
      "\t/tmp/tmp471d7U.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplEDzFC.lastz\n",
      "\t/tmp/tmpSeExzO.lastz\n",
      "\t/tmp/tmpotPCYS.lastz\n",
      "\t/tmp/tmpVjBrMx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Emmonsia_crescens_GCA_002572855.lastz\n",
      "Creating Emmonsia_crescens_GCA_002572855 table\n",
      "Inserting data to Emmonsia_crescens_GCA_002572855 table\n",
      "\n",
      "Running against Emmonsia_sp_GCA_001660665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpW8ZTeZ.fasta\n",
      "\t/tmp/tmp6Je_4u.fasta\n",
      "\t/tmp/tmpffos3_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUrJYsu.lastz\n",
      "\t/tmp/tmpkqFiJw.lastz\n",
      "\t/tmp/tmpriqH3e.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Emmonsia_sp_GCA_001660665.lastz\n",
      "Creating Emmonsia_sp_GCA_001660665 table\n",
      "Inserting data to Emmonsia_sp_GCA_001660665 table\n",
      "\n",
      "Running against Helicocarpus_griseus_GCA_002573585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmppYfqNN.fasta\n",
      "\t/tmp/tmp626Phe.fasta\n",
      "\t/tmp/tmpauLnQc.fasta\n",
      "\t/tmp/tmp_hrg2f.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpydp9bs.lastz\n",
      "\t/tmp/tmp8uFED8.lastz\n",
      "\t/tmp/tmpbWSVs5.lastz\n",
      "\t/tmp/tmptDcR5m.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Helicocarpus_griseus_GCA_002573585.lastz\n",
      "Creating Helicocarpus_griseus_GCA_002573585 table\n",
      "Inserting data to Helicocarpus_griseus_GCA_002573585 table\n",
      "\n",
      "Running against Malbranchea_cinnamomea_GCA_900128795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpNN0kNn.fasta\n",
      "\t/tmp/tmp_RryYl.fasta\n",
      "\t/tmp/tmp1yD8Ff.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_BqbDn.lastz\n",
      "\t/tmp/tmpGAKGME.lastz\n",
      "\t/tmp/tmp7eLeTT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Malbranchea_cinnamomea_GCA_900128795.lastz\n",
      "Creating Malbranchea_cinnamomea_GCA_900128795 table\n",
      "Inserting data to Malbranchea_cinnamomea_GCA_900128795 table\n",
      "\n",
      "Running against Microsprum_canis_GCF_000151145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp7lg0DH.fasta\n",
      "\t/tmp/tmp7GqYNU.fasta\n",
      "\t/tmp/tmpq6ILhF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6t2V2T.lastz\n",
      "\t/tmp/tmpNTYgk3.lastz\n",
      "\t/tmp/tmpDeSCkG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Microsprum_canis_GCF_000151145.lastz\n",
      "Creating Microsprum_canis_GCF_000151145 table\n",
      "Inserting data to Microsprum_canis_GCF_000151145 table\n",
      "\n",
      "Running against Nannizzia_gypsea_GCF_000150975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp5xWzHz.fasta\n",
      "\t/tmp/tmpweJmy1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVBg2BH.lastz\n",
      "\t/tmp/tmppkzY8m.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Nannizzia_gypsea_GCF_000150975.lastz\n",
      "Creating Nannizzia_gypsea_GCF_000150975 table\n",
      "Inserting data to Nannizzia_gypsea_GCF_000150975 table\n",
      "\n",
      "Running against Onygena_corvina_GCA_000812245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpwSlf_I.fasta\n",
      "\t/tmp/tmp_vbzIl.fasta\n",
      "\t/tmp/tmpIkvu6f.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6Zs0Fe.lastz\n",
      "\t/tmp/tmpWcb1ZJ.lastz\n",
      "\t/tmp/tmpw3QHqC.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Onygena_corvina_GCA_000812245.lastz\n",
      "Creating Onygena_corvina_GCA_000812245 table\n",
      "Inserting data to Onygena_corvina_GCA_000812245 table\n",
      "\n",
      "Running against Ophidiomyces_ophiodiicola_GCA_002167195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpp6yWQ5.fasta\n",
      "\t/tmp/tmp1P9uVk.fasta\n",
      "\t/tmp/tmpwfzTwI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpksKEFm.lastz\n",
      "\t/tmp/tmpo9KwCy.lastz\n",
      "\t/tmp/tmpnxur2A.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Ophidiomyces_ophiodiicola_GCA_002167195.lastz\n",
      "Creating Ophidiomyces_ophiodiicola_GCA_002167195 table\n",
      "Inserting data to Ophidiomyces_ophiodiicola_GCA_002167195 table\n",
      "\n",
      "Running against Paracoccidioides_brasiliensis_GCF_000150735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpEsJK2P.fasta\n",
      "\t/tmp/tmpBKNpnv.fasta\n",
      "\t/tmp/tmpOojSiI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpW1JjgX.lastz\n",
      "\t/tmp/tmpNQQKNo.lastz\n",
      "\t/tmp/tmp9Mw2UZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Paracoccidioides_brasiliensis_GCF_000150735.lastz\n",
      "Creating Paracoccidioides_brasiliensis_GCF_000150735 table\n",
      "Inserting data to Paracoccidioides_brasiliensis_GCF_000150735 table\n",
      "\n",
      "Running against Paracoccidioides_lutzii_GCF_000150705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpYWBP_z.fasta\n",
      "\t/tmp/tmpIj9Bx5.fasta\n",
      "\t/tmp/tmpYmIr1d.fasta\n",
      "\t/tmp/tmpU3tUD6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAg49i3.lastz\n",
      "\t/tmp/tmpm1cNFU.lastz\n",
      "\t/tmp/tmpyQ42L2.lastz\n",
      "\t/tmp/tmplroprh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Paracoccidioides_lutzii_GCF_000150705.lastz\n",
      "Creating Paracoccidioides_lutzii_GCF_000150705 table\n",
      "Inserting data to Paracoccidioides_lutzii_GCF_000150705 table\n",
      "\n",
      "Running against Polytolypa_hystricis_GCA_002573605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpTng1d4.fasta\n",
      "\t/tmp/tmpVMBQJf.fasta\n",
      "\t/tmp/tmptyvWHZ.fasta\n",
      "\t/tmp/tmppuhYV2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpknVvE5.lastz\n",
      "\t/tmp/tmpsFpdbI.lastz\n",
      "\t/tmp/tmpQj5Zuz.lastz\n",
      "\t/tmp/tmpBpSkxI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Polytolypa_hystricis_GCA_002573605.lastz\n",
      "Creating Polytolypa_hystricis_GCA_002573605 table\n",
      "Inserting data to Polytolypa_hystricis_GCA_002573605 table\n",
      "\n",
      "Running against Trichophyton_benhamiae_GCF_000151125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmptcqt2v.fasta\n",
      "\t/tmp/tmpRPsEVc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMukN04.lastz\n",
      "\t/tmp/tmpJTDvaQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_benhamiae_GCF_000151125.lastz\n",
      "Creating Trichophyton_benhamiae_GCF_000151125 table\n",
      "Inserting data to Trichophyton_benhamiae_GCF_000151125 table\n",
      "\n",
      "Running against Trichophyton_equinum_GCA_000151175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmprBXF04.fasta\n",
      "\t/tmp/tmp4agGIs.fasta\n",
      "\t/tmp/tmpmwzEMN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6NAUjV.lastz\n",
      "\t/tmp/tmpDIFZbj.lastz\n",
      "\t/tmp/tmp55H97_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_equinum_GCA_000151175.lastz\n",
      "Creating Trichophyton_equinum_GCA_000151175 table\n",
      "Inserting data to Trichophyton_equinum_GCA_000151175 table\n",
      "\n",
      "Running against Trichophyton_interdigitale_GCA_000622975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpnBjNOn.fasta\n",
      "\t/tmp/tmpYMkWDM.fasta\n",
      "\t/tmp/tmpu3kYbM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_R2Xq0.lastz\n",
      "\t/tmp/tmpfo9G42.lastz\n",
      "\t/tmp/tmpMYlqch.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_interdigitale_GCA_000622975.lastz\n",
      "Creating Trichophyton_interdigitale_GCA_000622975 table\n",
      "Inserting data to Trichophyton_interdigitale_GCA_000622975 table\n",
      "\n",
      "Running against Trichophyton_mentagrophytes_GCA_003664465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpexSF2U.fasta\n",
      "\t/tmp/tmpNCGJCi.fasta\n",
      "\t/tmp/tmp9Zy4cR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPbJp5B.lastz\n",
      "\t/tmp/tmpgeRZTi.lastz\n",
      "\t/tmp/tmpvRmfzN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_mentagrophytes_GCA_003664465.lastz\n",
      "Creating Trichophyton_mentagrophytes_GCA_003664465 table\n",
      "Inserting data to Trichophyton_mentagrophytes_GCA_003664465 table\n",
      "\n",
      "Running against Trichophyton_rubrum_GCF_000151425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpJzSuxm.fasta\n",
      "\t/tmp/tmpuFI95g.fasta\n",
      "\t/tmp/tmpLkK1qk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFZsE6D.lastz\n",
      "\t/tmp/tmps_anWh.lastz\n",
      "\t/tmp/tmp8ethyN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_rubrum_GCF_000151425.lastz\n",
      "Creating Trichophyton_rubrum_GCF_000151425 table\n",
      "Inserting data to Trichophyton_rubrum_GCF_000151425 table\n",
      "\n",
      "Running against Trichophyton_soudanense_GCA_000616865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpEkl7jx.fasta\n",
      "\t/tmp/tmppDhWwc.fasta\n",
      "\t/tmp/tmpLkeHds.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxeQJ_R.lastz\n",
      "\t/tmp/tmp3TP3Yb.lastz\n",
      "\t/tmp/tmprzkZ64.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_soudanense_GCA_000616865.lastz\n",
      "Creating Trichophyton_soudanense_GCA_000616865 table\n",
      "Inserting data to Trichophyton_soudanense_GCA_000616865 table\n",
      "\n",
      "Running against Trichophyton_tonsurans_GCA_000151455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpwqrM9e.fasta\n",
      "\t/tmp/tmp0zEGI1.fasta\n",
      "\t/tmp/tmpmaOuCl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDaOEX8.lastz\n",
      "\t/tmp/tmpstfKcn.lastz\n",
      "\t/tmp/tmpDqa4JZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_tonsurans_GCA_000151455.lastz\n",
      "Creating Trichophyton_tonsurans_GCA_000151455 table\n",
      "Inserting data to Trichophyton_tonsurans_GCA_000151455 table\n",
      "\n",
      "Running against Trichophyton_verrucosum_GCF_000151505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpVcSx6l.fasta\n",
      "\t/tmp/tmpgAeiH7.fasta\n",
      "\t/tmp/tmp79MZX8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTkBanh.lastz\n",
      "\t/tmp/tmpHGBgHu.lastz\n",
      "\t/tmp/tmpncnBmf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_verrucosum_GCF_000151505.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Trichophyton_verrucosum_GCF_000151505 table\n",
      "Inserting data to Trichophyton_verrucosum_GCF_000151505 table\n",
      "\n",
      "Running against Trichophyton_violaceum_GCA_001651435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpGj6VkP.fasta\n",
      "\t/tmp/tmp6TShDG.fasta\n",
      "\t/tmp/tmpKaQb0r.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCYpw03.lastz\n",
      "\t/tmp/tmpKavwzi.lastz\n",
      "\t/tmp/tmpQ_dWMn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Trichophyton_violaceum_GCA_001651435.lastz\n",
      "Creating Trichophyton_violaceum_GCA_001651435 table\n",
      "Inserting data to Trichophyton_violaceum_GCA_001651435 table\n",
      "\n",
      "Running against Uncinocarpus_reesii_GCF_000003515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp9kHFj5.fasta\n",
      "\t/tmp/tmpklxJO1.fasta\n",
      "\t/tmp/tmpFZQro8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGi_HHE.lastz\n",
      "\t/tmp/tmpLmg8_x.lastz\n",
      "\t/tmp/tmpeCe_X7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Uncinocarpus_reesii_GCF_000003515.lastz\n",
      "Creating Uncinocarpus_reesii_GCF_000003515 table\n",
      "Inserting data to Uncinocarpus_reesii_GCF_000003515 table\n",
      "\n",
      "Running against Histoplasma_capsulatum_GCF_000149585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpVvOjuX.fasta\n",
      "\t/tmp/tmpplkuqO.fasta\n",
      "\t/tmp/tmpNjkU5w.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJU1GKA.lastz\n",
      "\t/tmp/tmp90CwKc.lastz\n",
      "\t/tmp/tmpKObMIg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/onygenales/validate_intervals/lastz/Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_Histoplasma_capsulatum_GCF_000149585.lastz\n",
      "Creating Histoplasma_capsulatum_GCF_000149585 table\n",
      "Inserting data to Histoplasma_capsulatum_GCF_000149585 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/onygenales/extract_probes_from_group/onygenales_genome.conf --lastz results/phyluce/onygenales/validate_intervals/lastz --probes 120 --probe-prefix uce_onygenales_ --name-pattern \"Histoplasma_capsulatum_GCF_000149585_+9_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/onygenales/extract_probes_from_group/probe_fasta;\n",
      "2020-02-10 13:49:49,232 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-10 13:49:49,236 - Phyluce - INFO - ------ Working on Amauroascus_mutatus_GCA_001430935 genome ------\n",
      "2020-02-10 13:49:49,247 - Phyluce - INFO - Reading Amauroascus_mutatus_GCA_001430935 genome\n",
      "2020-02-10 13:49:51,440 - Phyluce - INFO - Amauroascus_mutatus_GCA_001430935: 584 uces, 77 dupes, 507 non-dupes, 0 orient drop, 1 length drop, 506 written\n",
      "2020-02-10 13:49:51,441 - Phyluce - INFO - ------- Working on Amauroascus_niger_GCA_001430945 genome -------\n",
      "2020-02-10 13:49:51,441 - Phyluce - INFO - Reading Amauroascus_niger_GCA_001430945 genome\n",
      "2020-02-10 13:49:53,645 - Phyluce - INFO - Amauroascus_niger_GCA_001430945: 590 uces, 83 dupes, 507 non-dupes, 0 orient drop, 1 length drop, 506 written\n",
      "2020-02-10 13:49:53,645 - Phyluce - INFO - -------- Working on Ascospaera_apis_GCA_001636715 genome --------\n",
      "2020-02-10 13:49:53,646 - Phyluce - INFO - Reading Ascospaera_apis_GCA_001636715 genome\n",
      "2020-02-10 13:49:55,355 - Phyluce - INFO - Ascospaera_apis_GCA_001636715: 501 uces, 54 dupes, 447 non-dupes, 0 orient drop, 1 length drop, 446 written\n",
      "2020-02-10 13:49:55,355 - Phyluce - INFO - ---- Working on Blastomyces_dermatitidis_GCA_000003525 genome ---\n",
      "2020-02-10 13:49:55,356 - Phyluce - INFO - Reading Blastomyces_dermatitidis_GCA_000003525 genome\n",
      "2020-02-10 13:49:58,032 - Phyluce - INFO - Blastomyces_dermatitidis_GCA_000003525: 615 uces, 85 dupes, 530 non-dupes, 3 orient drop, 1 length drop, 526 written\n",
      "2020-02-10 13:49:58,032 - Phyluce - INFO - ---- Working on Blastomyces_gilchristii_GCF_000003855 genome ----\n",
      "2020-02-10 13:49:58,033 - Phyluce - INFO - Reading Blastomyces_gilchristii_GCF_000003855 genome\n",
      "2020-02-10 13:50:00,657 - Phyluce - INFO - Blastomyces_gilchristii_GCF_000003855: 617 uces, 77 dupes, 540 non-dupes, 12 orient drop, 1 length drop, 527 written\n",
      "2020-02-10 13:50:00,657 - Phyluce - INFO - ------- Working on Blastomyces_parvus_GCA_002572885 genome ------\n",
      "2020-02-10 13:50:00,671 - Phyluce - INFO - Reading Blastomyces_parvus_GCA_002572885 genome\n",
      "2020-02-10 13:50:02,931 - Phyluce - INFO - Blastomyces_parvus_GCA_002572885: 602 uces, 66 dupes, 536 non-dupes, 1 orient drop, 0 length drop, 535 written\n",
      "2020-02-10 13:50:02,932 - Phyluce - INFO - ----- Working on Blastomyces_percursus_GCA_003206225 genome -----\n",
      "2020-02-10 13:50:02,932 - Phyluce - INFO - Reading Blastomyces_percursus_GCA_003206225 genome\n",
      "2020-02-10 13:50:05,476 - Phyluce - INFO - Blastomyces_percursus_GCA_003206225: 616 uces, 83 dupes, 533 non-dupes, 1 orient drop, 0 length drop, 532 written\n",
      "2020-02-10 13:50:05,476 - Phyluce - INFO - ------ Working on Blastomyces_silverae_GCA_001014755 genome -----\n",
      "2020-02-10 13:50:05,477 - Phyluce - INFO - Reading Blastomyces_silverae_GCA_001014755 genome\n",
      "2020-02-10 13:50:07,625 - Phyluce - INFO - Blastomyces_silverae_GCA_001014755: 569 uces, 57 dupes, 512 non-dupes, 1 orient drop, 0 length drop, 511 written\n",
      "2020-02-10 13:50:07,625 - Phyluce - INFO - --------- Working on Blastomyces_sp_GCA_003206725 genome --------\n",
      "2020-02-10 13:50:07,633 - Phyluce - INFO - Reading Blastomyces_sp_GCA_003206725 genome\n",
      "2020-02-10 13:50:09,946 - Phyluce - INFO - Blastomyces_sp_GCA_003206725: 614 uces, 71 dupes, 543 non-dupes, 1 orient drop, 0 length drop, 542 written\n",
      "2020-02-10 13:50:09,946 - Phyluce - INFO - --- Working on Byssoonygena_ceratinophila_GCA_001430925 genome --\n",
      "2020-02-10 13:50:09,947 - Phyluce - INFO - Reading Byssoonygena_ceratinophila_GCA_001430925 genome\n",
      "2020-02-10 13:50:12,150 - Phyluce - INFO - Byssoonygena_ceratinophila_GCA_001430925: 588 uces, 81 dupes, 507 non-dupes, 0 orient drop, 0 length drop, 507 written\n",
      "2020-02-10 13:50:12,150 - Phyluce - INFO - -- Working on Chrysosprium_queenslandicum_GCA_001430955 genome --\n",
      "2020-02-10 13:50:12,151 - Phyluce - INFO - Reading Chrysosprium_queenslandicum_GCA_001430955 genome\n",
      "2020-02-10 13:50:14,228 - Phyluce - INFO - Chrysosprium_queenslandicum_GCA_001430955: 582 uces, 73 dupes, 509 non-dupes, 2 orient drop, 0 length drop, 507 written\n",
      "2020-02-10 13:50:14,228 - Phyluce - INFO - ------ Working on Coccidioides_immitis_GCF_000149335 genome -----\n",
      "2020-02-10 13:50:14,229 - Phyluce - INFO - Reading Coccidioides_immitis_GCF_000149335 genome\n",
      "2020-02-10 13:50:16,638 - Phyluce - INFO - Coccidioides_immitis_GCF_000149335: 597 uces, 61 dupes, 536 non-dupes, 17 orient drop, 8 length drop, 511 written\n",
      "2020-02-10 13:50:16,638 - Phyluce - INFO - ----- Working on Coccidioides_posadasii_GCF_000151335 genome ----\n",
      "2020-02-10 13:50:16,654 - Phyluce - INFO - Reading Coccidioides_posadasii_GCF_000151335 genome\n",
      "2020-02-10 13:50:18,902 - Phyluce - INFO - Coccidioides_posadasii_GCF_000151335: 596 uces, 79 dupes, 517 non-dupes, 5 orient drop, 3 length drop, 509 written\n",
      "2020-02-10 13:50:18,902 - Phyluce - INFO - ----- Working on Emergomyces_orientalis_GCA_002110485 genome ----\n",
      "2020-02-10 13:50:18,903 - Phyluce - INFO - Reading Emergomyces_orientalis_GCA_002110485 genome\n",
      "2020-02-10 13:50:21,593 - Phyluce - INFO - Emergomyces_orientalis_GCA_002110485: 617 uces, 85 dupes, 532 non-dupes, 1 orient drop, 1 length drop, 530 written\n",
      "2020-02-10 13:50:21,593 - Phyluce - INFO - ---- Working on Emergomyces_pasteurianus_GCA_001883825 genome ---\n",
      "2020-02-10 13:50:21,594 - Phyluce - INFO - Reading Emergomyces_pasteurianus_GCA_001883825 genome\n",
      "2020-02-10 13:50:23,941 - Phyluce - INFO - Emergomyces_pasteurianus_GCA_001883825: 608 uces, 75 dupes, 533 non-dupes, 1 orient drop, 0 length drop, 532 written\n",
      "2020-02-10 13:50:23,941 - Phyluce - INFO - ------- Working on Emmonsia_crescens_GCA_002572855 genome -------\n",
      "2020-02-10 13:50:23,941 - Phyluce - INFO - Reading Emmonsia_crescens_GCA_002572855 genome\n",
      "2020-02-10 13:50:26,252 - Phyluce - INFO - Emmonsia_crescens_GCA_002572855: 605 uces, 71 dupes, 534 non-dupes, 1 orient drop, 0 length drop, 533 written\n",
      "2020-02-10 13:50:26,252 - Phyluce - INFO - ---------- Working on Emmonsia_sp_GCA_001660665 genome ----------\n",
      "2020-02-10 13:50:26,253 - Phyluce - INFO - Reading Emmonsia_sp_GCA_001660665 genome\n",
      "2020-02-10 13:50:28,565 - Phyluce - INFO - Emmonsia_sp_GCA_001660665: 597 uces, 72 dupes, 525 non-dupes, 4 orient drop, 0 length drop, 521 written\n",
      "2020-02-10 13:50:28,565 - Phyluce - INFO - ------ Working on Helicocarpus_griseus_GCA_002573585 genome -----\n",
      "2020-02-10 13:50:28,573 - Phyluce - INFO - Reading Helicocarpus_griseus_GCA_002573585 genome\n",
      "2020-02-10 13:50:30,767 - Phyluce - INFO - Helicocarpus_griseus_GCA_002573585: 592 uces, 68 dupes, 524 non-dupes, 0 orient drop, 0 length drop, 524 written\n",
      "2020-02-10 13:50:30,767 - Phyluce - INFO - ----- Working on Malbranchea_cinnamomea_GCA_900128795 genome ----\n",
      "2020-02-10 13:50:30,783 - Phyluce - INFO - Reading Malbranchea_cinnamomea_GCA_900128795 genome\n",
      "2020-02-10 13:50:32,905 - Phyluce - INFO - Malbranchea_cinnamomea_GCA_900128795: 588 uces, 69 dupes, 519 non-dupes, 1 orient drop, 0 length drop, 518 written\n",
      "2020-02-10 13:50:32,906 - Phyluce - INFO - -------- Working on Microsprum_canis_GCF_000151145 genome -------\n",
      "2020-02-10 13:50:32,910 - Phyluce - INFO - Reading Microsprum_canis_GCF_000151145 genome\n",
      "2020-02-10 13:50:35,053 - Phyluce - INFO - Microsprum_canis_GCF_000151145: 577 uces, 50 dupes, 527 non-dupes, 1 orient drop, 14 length drop, 512 written\n",
      "2020-02-10 13:50:35,053 - Phyluce - INFO - -------- Working on Nannizzia_gypsea_GCF_000150975 genome -------\n",
      "2020-02-10 13:50:35,063 - Phyluce - INFO - Reading Nannizzia_gypsea_GCF_000150975 genome\n",
      "2020-02-10 13:50:37,163 - Phyluce - INFO - Nannizzia_gypsea_GCF_000150975: 573 uces, 44 dupes, 529 non-dupes, 4 orient drop, 15 length drop, 510 written\n",
      "2020-02-10 13:50:37,163 - Phyluce - INFO - -------- Working on Onygena_corvina_GCA_000812245 genome --------\n",
      "2020-02-10 13:50:37,185 - Phyluce - INFO - Reading Onygena_corvina_GCA_000812245 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 13:50:39,196 - Phyluce - INFO - Onygena_corvina_GCA_000812245: 567 uces, 66 dupes, 501 non-dupes, 0 orient drop, 0 length drop, 501 written\n",
      "2020-02-10 13:50:39,196 - Phyluce - INFO - --- Working on Ophidiomyces_ophiodiicola_GCA_002167195 genome ---\n",
      "2020-02-10 13:50:39,197 - Phyluce - INFO - Reading Ophidiomyces_ophiodiicola_GCA_002167195 genome\n",
      "2020-02-10 13:50:41,266 - Phyluce - INFO - Ophidiomyces_ophiodiicola_GCA_002167195: 584 uces, 61 dupes, 523 non-dupes, 2 orient drop, 4 length drop, 517 written\n",
      "2020-02-10 13:50:41,267 - Phyluce - INFO - - Working on Paracoccidioides_brasiliensis_GCF_000150735 genome -\n",
      "2020-02-10 13:50:41,280 - Phyluce - INFO - Reading Paracoccidioides_brasiliensis_GCF_000150735 genome\n",
      "2020-02-10 13:50:43,924 - Phyluce - INFO - Paracoccidioides_brasiliensis_GCF_000150735: 610 uces, 72 dupes, 538 non-dupes, 5 orient drop, 5 length drop, 528 written\n",
      "2020-02-10 13:50:43,924 - Phyluce - INFO - ---- Working on Paracoccidioides_lutzii_GCF_000150705 genome ----\n",
      "2020-02-10 13:50:43,925 - Phyluce - INFO - Reading Paracoccidioides_lutzii_GCF_000150705 genome\n",
      "2020-02-10 13:50:46,493 - Phyluce - INFO - Paracoccidioides_lutzii_GCF_000150705: 612 uces, 79 dupes, 533 non-dupes, 4 orient drop, 3 length drop, 526 written\n",
      "2020-02-10 13:50:46,493 - Phyluce - INFO - ------ Working on Polytolypa_hystricis_GCA_002573605 genome -----\n",
      "2020-02-10 13:50:46,494 - Phyluce - INFO - Reading Polytolypa_hystricis_GCA_002573605 genome\n",
      "2020-02-10 13:50:48,677 - Phyluce - INFO - Polytolypa_hystricis_GCA_002573605: 596 uces, 76 dupes, 520 non-dupes, 1 orient drop, 2 length drop, 517 written\n",
      "2020-02-10 13:50:48,677 - Phyluce - INFO - ----- Working on Trichophyton_benhamiae_GCF_000151125 genome ----\n",
      "2020-02-10 13:50:48,678 - Phyluce - INFO - Reading Trichophyton_benhamiae_GCF_000151125 genome\n",
      "2020-02-10 13:50:50,689 - Phyluce - INFO - Trichophyton_benhamiae_GCF_000151125: 569 uces, 63 dupes, 506 non-dupes, 0 orient drop, 3 length drop, 503 written\n",
      "2020-02-10 13:50:50,690 - Phyluce - INFO - ------ Working on Trichophyton_equinum_GCA_000151175 genome -----\n",
      "2020-02-10 13:50:50,699 - Phyluce - INFO - Reading Trichophyton_equinum_GCA_000151175 genome\n",
      "2020-02-10 13:50:52,803 - Phyluce - INFO - Trichophyton_equinum_GCA_000151175: 555 uces, 75 dupes, 480 non-dupes, 0 orient drop, 1 length drop, 479 written\n",
      "2020-02-10 13:50:52,803 - Phyluce - INFO - --- Working on Trichophyton_interdigitale_GCA_000622975 genome --\n",
      "2020-02-10 13:50:52,812 - Phyluce - INFO - Reading Trichophyton_interdigitale_GCA_000622975 genome\n",
      "2020-02-10 13:50:54,750 - Phyluce - INFO - Trichophyton_interdigitale_GCA_000622975: 551 uces, 57 dupes, 494 non-dupes, 0 orient drop, 0 length drop, 494 written\n",
      "2020-02-10 13:50:54,750 - Phyluce - INFO - -- Working on Trichophyton_mentagrophytes_GCA_003664465 genome --\n",
      "2020-02-10 13:50:54,760 - Phyluce - INFO - Reading Trichophyton_mentagrophytes_GCA_003664465 genome\n",
      "2020-02-10 13:50:56,756 - Phyluce - INFO - Trichophyton_mentagrophytes_GCA_003664465: 565 uces, 61 dupes, 504 non-dupes, 0 orient drop, 0 length drop, 504 written\n",
      "2020-02-10 13:50:56,756 - Phyluce - INFO - ------ Working on Trichophyton_rubrum_GCF_000151425 genome ------\n",
      "2020-02-10 13:50:56,766 - Phyluce - INFO - Reading Trichophyton_rubrum_GCF_000151425 genome\n",
      "2020-02-10 13:50:58,705 - Phyluce - INFO - Trichophyton_rubrum_GCF_000151425: 553 uces, 53 dupes, 500 non-dupes, 0 orient drop, 3 length drop, 497 written\n",
      "2020-02-10 13:50:58,705 - Phyluce - INFO - ---- Working on Trichophyton_soudanense_GCA_000616865 genome ----\n",
      "2020-02-10 13:50:58,706 - Phyluce - INFO - Reading Trichophyton_soudanense_GCA_000616865 genome\n",
      "2020-02-10 13:51:00,676 - Phyluce - INFO - Trichophyton_soudanense_GCA_000616865: 562 uces, 57 dupes, 505 non-dupes, 1 orient drop, 0 length drop, 504 written\n",
      "2020-02-10 13:51:00,677 - Phyluce - INFO - ----- Working on Trichophyton_tonsurans_GCA_000151455 genome ----\n",
      "2020-02-10 13:51:00,677 - Phyluce - INFO - Reading Trichophyton_tonsurans_GCA_000151455 genome\n",
      "2020-02-10 13:51:02,675 - Phyluce - INFO - Trichophyton_tonsurans_GCA_000151455: 563 uces, 59 dupes, 504 non-dupes, 0 orient drop, 13 length drop, 491 written\n",
      "2020-02-10 13:51:02,676 - Phyluce - INFO - ---- Working on Trichophyton_verrucosum_GCF_000151505 genome ----\n",
      "2020-02-10 13:51:02,676 - Phyluce - INFO - Reading Trichophyton_verrucosum_GCF_000151505 genome\n",
      "2020-02-10 13:51:04,646 - Phyluce - INFO - Trichophyton_verrucosum_GCF_000151505: 570 uces, 63 dupes, 507 non-dupes, 0 orient drop, 0 length drop, 507 written\n",
      "2020-02-10 13:51:04,646 - Phyluce - INFO - ----- Working on Trichophyton_violaceum_GCA_001651435 genome ----\n",
      "2020-02-10 13:51:04,647 - Phyluce - INFO - Reading Trichophyton_violaceum_GCA_001651435 genome\n",
      "2020-02-10 13:51:06,671 - Phyluce - INFO - Trichophyton_violaceum_GCA_001651435: 569 uces, 62 dupes, 507 non-dupes, 1 orient drop, 1 length drop, 504 written\n",
      "2020-02-10 13:51:06,671 - Phyluce - INFO - ------ Working on Uncinocarpus_reesii_GCF_000003515 genome ------\n",
      "2020-02-10 13:51:06,672 - Phyluce - INFO - Reading Uncinocarpus_reesii_GCF_000003515 genome\n",
      "2020-02-10 13:51:08,737 - Phyluce - INFO - Uncinocarpus_reesii_GCF_000003515: 571 uces, 58 dupes, 513 non-dupes, 4 orient drop, 6 length drop, 503 written\n",
      "2020-02-10 13:51:08,738 - Phyluce - INFO - ----- Working on Histoplasma_capsulatum_GCF_000149585 genome ----\n",
      "2020-02-10 13:51:08,738 - Phyluce - INFO - Reading Histoplasma_capsulatum_GCF_000149585 genome\n",
      "2020-02-10 13:51:11,379 - Phyluce - INFO - Histoplasma_capsulatum_GCF_000149585: 618 uces, 79 dupes, 539 non-dupes, 5 orient drop, 5 length drop, 529 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/onygenales/extract_probes_from_group/probe_fasta --output results/phyluce/onygenales/extract_probes_from_group/multifastas.sqlite --base-taxon Histoplasma_capsulatum_GCF_000149585;\n",
      "amauroascus_mutatus_gca_001430935.\n",
      "amauroascus_niger_gca_001430945.\n",
      "ascospaera_apis_gca_001636715.\n",
      "blastomyces_dermatitidis_gca_000003525.\n",
      "blastomyces_gilchristii_gcf_000003855.\n",
      "blastomyces_parvus_gca_002572885.\n",
      "blastomyces_percursus_gca_003206225.\n",
      "blastomyces_silverae_gca_001014755.\n",
      "blastomyces_sp_gca_003206725.\n",
      "byssoonygena_ceratinophila_gca_001430925.\n",
      "chrysosprium_queenslandicum_gca_001430955.\n",
      "coccidioides_immitis_gcf_000149335.\n",
      "coccidioides_posadasii_gcf_000151335.\n",
      "emergomyces_orientalis_gca_002110485.\n",
      "emergomyces_pasteurianus_gca_001883825.\n",
      "emmonsia_crescens_gca_002572855.\n",
      "emmonsia_sp_gca_001660665.\n",
      "helicocarpus_griseus_gca_002573585.\n",
      "malbranchea_cinnamomea_gca_900128795.\n",
      "microsprum_canis_gcf_000151145.\n",
      "nannizzia_gypsea_gcf_000150975.\n",
      "onygena_corvina_gca_000812245.\n",
      "ophidiomyces_ophiodiicola_gca_002167195.\n",
      "paracoccidioides_brasiliensis_gcf_000150735.\n",
      "paracoccidioides_lutzii_gcf_000150705.\n",
      "polytolypa_hystricis_gca_002573605.\n",
      "trichophyton_benhamiae_gcf_000151125.\n",
      "trichophyton_equinum_gca_000151175.\n",
      "trichophyton_interdigitale_gca_000622975.\n",
      "trichophyton_mentagrophytes_gca_003664465.\n",
      "trichophyton_rubrum_gcf_000151425.\n",
      "trichophyton_soudanense_gca_000616865.\n",
      "trichophyton_tonsurans_gca_000151455.\n",
      "trichophyton_verrucosum_gcf_000151505.\n",
      "trichophyton_violaceum_gca_001651435.\n",
      "uncinocarpus_reesii_gcf_000003515.\n",
      "histoplasma_capsulatum_gcf_000149585.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/onygenales/extract_probes_from_group/multifastas.sqlite --base-taxon Histoplasma_capsulatum_GCF_000149585;\n",
      "Loci shared by 0 taxa:\t600.0\n",
      "Loci shared by 1 taxa:\t600.0\n",
      "Loci shared by 2 taxa:\t589.0\n",
      "Loci shared by 3 taxa:\t588.0\n",
      "Loci shared by 4 taxa:\t586.0\n",
      "Loci shared by 5 taxa:\t582.0\n",
      "Loci shared by 6 taxa:\t574.0\n",
      "Loci shared by 7 taxa:\t570.0\n",
      "Loci shared by 8 taxa:\t567.0\n",
      "Loci shared by 9 taxa:\t562.0\n",
      "Loci shared by 10 taxa:\t557.0\n",
      "Loci shared by 11 taxa:\t557.0\n",
      "Loci shared by 12 taxa:\t553.0\n",
      "Loci shared by 13 taxa:\t545.0\n",
      "Loci shared by 14 taxa:\t543.0\n",
      "Loci shared by 15 taxa:\t540.0\n",
      "Loci shared by 16 taxa:\t536.0\n",
      "Loci shared by 17 taxa:\t530.0\n",
      "Loci shared by 18 taxa:\t527.0\n",
      "Loci shared by 19 taxa:\t526.0\n",
      "Loci shared by 20 taxa:\t521.0\n",
      "Loci shared by 21 taxa:\t515.0\n",
      "Loci shared by 22 taxa:\t513.0\n",
      "Loci shared by 23 taxa:\t507.0\n",
      "Loci shared by 24 taxa:\t506.0\n",
      "Loci shared by 25 taxa:\t501.0\n",
      "Loci shared by 26 taxa:\t494.0\n",
      "Loci shared by 27 taxa:\t490.0\n",
      "Loci shared by 28 taxa:\t482.0\n",
      "Loci shared by 29 taxa:\t480.0\n",
      "Loci shared by 30 taxa:\t476.0\n",
      "Loci shared by 31 taxa:\t471.0\n",
      "Loci shared by 32 taxa:\t458.0\n",
      "Loci shared by 33 taxa:\t447.0\n",
      "Loci shared by 34 taxa:\t429.0\n",
      "Loci shared by 35 taxa:\t413.0\n",
      "Loci shared by 36 taxa:\t364.0\n",
      "Loci shared by 37 taxa:\t250.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/onygenales/extract_probes_from_group/multifastas.sqlite  --base-taxon Histoplasma_capsulatum_GCF_000149585 --output results/phyluce/onygenales/extract_probes_from_group/Histoplasma_capsulatum_GCF_000149585+9-back-to-37.conf --specific-counts 37;\n",
      "Counter({'trichophyton_verrucosum_gcf_000151505': 250, 'coccidioides_immitis_gcf_000149335': 250, 'amauroascus_niger_gca_001430945': 250, 'blastomyces_percursus_gca_003206225': 250, 'microsprum_canis_gcf_000151145': 250, 'blastomyces_parvus_gca_002572885': 250, 'blastomyces_sp_gca_003206725': 250, 'histoplasma_capsulatum_gcf_000149585': 250, 'blastomyces_gilchristii_gcf_000003855': 250, 'emergomyces_orientalis_gca_002110485': 250, 'trichophyton_soudanense_gca_000616865': 250, 'malbranchea_cinnamomea_gca_900128795': 250, 'trichophyton_violaceum_gca_001651435': 250, 'onygena_corvina_gca_000812245': 250, 'amauroascus_mutatus_gca_001430935': 250, 'polytolypa_hystricis_gca_002573605': 250, 'byssoonygena_ceratinophila_gca_001430925': 250, 'emergomyces_pasteurianus_gca_001883825': 250, 'chrysosprium_queenslandicum_gca_001430955': 250, 'coccidioides_posadasii_gcf_000151335': 250, 'blastomyces_dermatitidis_gca_000003525': 250, 'paracoccidioides_brasiliensis_gcf_000150735': 250, 'emmonsia_crescens_gca_002572855': 250, 'emmonsia_sp_gca_001660665': 250, 'blastomyces_silverae_gca_001014755': 250, 'paracoccidioides_lutzii_gcf_000150705': 250, 'ascospaera_apis_gca_001636715': 250, 'uncinocarpus_reesii_gcf_000003515': 250, 'helicocarpus_griseus_gca_002573585': 250, 'trichophyton_mentagrophytes_gca_003664465': 250, 'trichophyton_equinum_gca_000151175': 250, 'nannizzia_gypsea_gcf_000150975': 250, 'trichophyton_interdigitale_gca_000622975': 250, 'ophidiomyces_ophiodiicola_gca_002167195': 250, 'trichophyton_benhamiae_gcf_000151125': 250, 'trichophyton_tonsurans_gca_000151455': 250, 'trichophyton_rubrum_gcf_000151425': 250})\n",
      "Total loci = 250\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(37)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/onygenales/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/onygenales/extract_probes_from_group/Histoplasma_capsulatum_GCF_000149585+9-back-to-37.conf --probe-prefix uce_onygenales_ --designer rnplattii --design onygenales_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGNNGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 250\n",
      "Probe Count = 18482\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list.fasta --query results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Mon Feb 10, 2020  13:54:04\n",
      "Ended:  Mon Feb 10, 2020  13:55:09\n",
      "Time for execution:  1.08550671736 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list.fasta --lastz results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_onygenales_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 18481 fasta sequences.  Filtered 0 duplicates. Kept 18482.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/onygenales/final_probe_design/onygenales_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Mon Feb 10 14:20:39 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 18482\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1478560\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 49M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 91M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3933501\n",
      "Max number of word counting entries: 88543113\n",
      "\n",
      "# comparing sequences from          0  to       3080\n",
      "...---------- new table with     1970 representatives\n",
      "# comparing sequences from       3080  to       5647\n",
      "----------    676 remaining sequences to the next cycle\n",
      "---------- new table with     1167 representatives\n",
      "# comparing sequences from       4971  to       7222\n",
      "----------    761 remaining sequences to the next cycle\n",
      "---------- new table with      872 representatives\n",
      "# comparing sequences from       6461  to       8464\n",
      "----------    751 remaining sequences to the next cycle\n",
      "---------- new table with      831 representatives\n",
      "# comparing sequences from       7713  to       9507\n",
      "----------    456 remaining sequences to the next cycle\n",
      "---------- new table with      838 representatives\n",
      "# comparing sequences from       9051  to      10622\n",
      "----------    640 remaining sequences to the next cycle\n",
      "---------- new table with      588 representatives\n",
      "# comparing sequences from       9982  to      11398\n",
      "..........    10000  finished       6279  clusters\n",
      "----------    354 remaining sequences to the next cycle\n",
      "---------- new table with      722 representatives\n",
      "# comparing sequences from      11044  to      12283\n",
      "----------    491 remaining sequences to the next cycle\n",
      "---------- new table with      484 representatives\n",
      "# comparing sequences from      11792  to      12907\n",
      "----------    246 remaining sequences to the next cycle\n",
      "---------- new table with      557 representatives\n",
      "# comparing sequences from      12661  to      13631\n",
      "----------    219 remaining sequences to the next cycle\n",
      "---------- new table with      505 representatives\n",
      "# comparing sequences from      13412  to      14257\n",
      "----------    181 remaining sequences to the next cycle\n",
      "---------- new table with      410 representatives\n",
      "# comparing sequences from      14076  to      14810\n",
      "----------    300 remaining sequences to the next cycle\n",
      "---------- new table with      268 representatives\n",
      "# comparing sequences from      14510  to      15172\n",
      "----------    289 remaining sequences to the next cycle\n",
      "---------- new table with      240 representatives\n",
      "# comparing sequences from      14883  to      15482\n",
      "----------    240 remaining sequences to the next cycle\n",
      "---------- new table with      223 representatives\n",
      "# comparing sequences from      15242  to      15782\n",
      "----------    182 remaining sequences to the next cycle\n",
      "---------- new table with      235 representatives\n",
      "# comparing sequences from      15600  to      16080\n",
      "----------     73 remaining sequences to the next cycle\n",
      "---------- new table with      256 representatives\n",
      "# comparing sequences from      16007  to      16419\n",
      "----------     58 remaining sequences to the next cycle\n",
      "---------- new table with      234 representatives\n",
      "# comparing sequences from      16361  to      16714\n",
      "----------    145 remaining sequences to the next cycle\n",
      "---------- new table with      139 representatives\n",
      "# comparing sequences from      16569  to      16887\n",
      "----------    131 remaining sequences to the next cycle\n",
      "---------- new table with      125 representatives\n",
      "# comparing sequences from      16756  to      17043\n",
      "----------     56 remaining sequences to the next cycle\n",
      "---------- new table with      151 representatives\n",
      "# comparing sequences from      16987  to      17236\n",
      "99.9%---------- new table with      167 representatives\n",
      "# comparing sequences from      17236  to      17443\n",
      "99.9%---------- new table with      132 representatives\n",
      "# comparing sequences from      17443  to      17616\n",
      "----------     18 remaining sequences to the next cycle\n",
      "---------- new table with      105 representatives\n",
      "# comparing sequences from      17598  to      18482\n",
      "....................---------- new table with      534 representatives\n",
      "\n",
      "    18482  finished      11753  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 93M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 2.83\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Trematodes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'trematoda'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Clonorchis_sinensis_GCA_003604175\",\n",
    "               \"Dicrocoelium_dendriticum_GCA_000950715\",\n",
    "               \"Echinostoma_caproni_GCA_900618425\",\n",
    "               \"Fasciola_gigantica_GCA_006461475\",\n",
    "               \"Fasciola_hepatica_GCA_900302435\",\n",
    "               \"Fasciolopsis_buski_GCA_008360955\",\n",
    "               \"Opisthorchis_felineus_GCA_004794785\",\n",
    "               \"Opisthorchis_viverrini_GCF_000715545\",\n",
    "               \"Paragonimus_westermani_GCA_008508345\",\n",
    "               \"Schistosoma_bovis_GCA_003958945\",\n",
    "               \"Schistosoma_curassoni_GCA_900618015\",\n",
    "               \"Schistosoma_haematobium_GCF_000699445\",\n",
    "               \"Schistosoma_japonicum_GCA_006368765\",\n",
    "               \"Schistosoma_margrebowiei_GCA_900618395\",\n",
    "               \"Schistosoma_mattheei_GCA_900617995\",\n",
    "               \"Trichobilharzia_regenti_GCA_900618515\" ]\n",
    "\n",
    "\n",
    "reference_taxon = \"Schistosoma_mansoni_GCA_000237925\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003604175.1_ASM360417v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 175932734 bytes  7180929.63 bytes/sec\n",
      "total size is 175889683  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000950715.1_D_dendriticum_Leon_v1_0_4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 150139750 bytes  7699476.51 bytes/sec\n",
      "total size is 150102981  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900618425.1_E_caproni_Egypt_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 257795544 bytes  7695390.63 bytes/sec\n",
      "total size is 257732496  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006461475.1_F_gigantica_1.0.allpaths_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 349725594 bytes  8853813.57 bytes/sec\n",
      "total size is 349640106  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900302435.1_Fasciola_10x_pilon_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 377479810 bytes  8296260.48 bytes/sec\n",
      "total size is 377387560  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008360955.1_F_buski_1.0.allpaths-lg_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 234982973 bytes  7459778.25 bytes/sec\n",
      "total size is 234925494  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004794785.1_ICG_Ofel_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 198594737 bytes  6732026.41 bytes/sec\n",
      "total size is 198546149  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000715545.1_OpiViv1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 184004534 bytes  7215865.73 bytes/sec\n",
      "total size is 183959517  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008508345.1_ASM850834v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 285449871 bytes  7611997.68 bytes/sec\n",
      "total size is 285380084  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003958945.1_ASM395894v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 114689269 bytes  7399310.39 bytes/sec\n",
      "total size is 114661162  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GCA_900618015.1_S_curassoni_Dakar_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 110903031 bytes  7155036.97 bytes/sec\n",
      "total size is 110875837  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000699445.1_SchHae_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 115390522 bytes  6993367.52 bytes/sec\n",
      "total size is 115362248  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006368765.1_ASM636876v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 118538283 bytes  8780616.67 bytes/sec\n",
      "total size is 118509240  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000237925.3_Smansoni_v7_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 124428405 bytes  9216922.00 bytes/sec\n",
      "total size is 124397922  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900618395.1_S_margrebowiei_Zambia_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 119599094 bytes  7716073.29 bytes/sec\n",
      "total size is 119569776  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617995.1_S_mattheei_Denwood_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 108344905 bytes  8667595.76 bytes/sec\n",
      "total size is 108318334  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900618515.1_T_regenti_v1_0_4_001_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 212546486 bytes  9044533.11 bytes/sec\n",
      "total size is 212494486  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486076 (\"sim_Clonorchis_sinensis_GCA_003604175\") has been submitted\n",
      "Your job 5486077 (\"sim_Dicrocoelium_dendriticum_GCA_000950715\") has been submitted\n",
      "Your job 5486078 (\"sim_Echinostoma_caproni_GCA_900618425\") has been submitted\n",
      "Your job 5486079 (\"sim_Fasciola_gigantica_GCA_006461475\") has been submitted\n",
      "Your job 5486080 (\"sim_Fasciola_hepatica_GCA_900302435\") has been submitted\n",
      "Your job 5486081 (\"sim_Fasciolopsis_buski_GCA_008360955\") has been submitted\n",
      "Your job 5486082 (\"sim_Opisthorchis_felineus_GCA_004794785\") has been submitted\n",
      "Your job 5486083 (\"sim_Opisthorchis_viverrini_GCF_000715545\") has been submitted\n",
      "Your job 5486084 (\"sim_Paragonimus_westermani_GCA_008508345\") has been submitted\n",
      "Your job 5486085 (\"sim_Schistosoma_bovis_GCA_003958945\") has been submitted\n",
      "Your job 5486086 (\"sim_Schistosoma_curassoni_GCA_900618015\") has been submitted\n",
      "Your job 5486087 (\"sim_Schistosoma_haematobium_GCF_000699445\") has been submitted\n",
      "Your job 5486088 (\"sim_Schistosoma_japonicum_GCA_006368765\") has been submitted\n",
      "Your job 5486089 (\"sim_Schistosoma_margrebowiei_GCA_900618395\") has been submitted\n",
      "Your job 5486090 (\"sim_Schistosoma_mattheei_GCA_900617995\") has been submitted\n",
      "Your job 5486091 (\"sim_Trichobilharzia_regenti_GCA_900618515\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/trematoda/cleaned_genomes/Schistosoma_mansoni_GCA_000237925_formatted.fas path=results/phyluce/trematoda/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486092 (\"map_Clonorchis_sinensis_GCA_003604175\") has been submitted\n",
      "Your job 5486093 (\"map_Dicrocoelium_dendriticum_GCA_000950715\") has been submitted\n",
      "Your job 5486094 (\"map_Echinostoma_caproni_GCA_900618425\") has been submitted\n",
      "Your job 5486095 (\"map_Fasciola_gigantica_GCA_006461475\") has been submitted\n",
      "Your job 5486096 (\"map_Fasciola_hepatica_GCA_900302435\") has been submitted\n",
      "Your job 5486097 (\"map_Fasciolopsis_buski_GCA_008360955\") has been submitted\n",
      "Your job 5486098 (\"map_Opisthorchis_felineus_GCA_004794785\") has been submitted\n",
      "Your job 5486099 (\"map_Opisthorchis_viverrini_GCF_000715545\") has been submitted\n",
      "Your job 5486100 (\"map_Paragonimus_westermani_GCA_008508345\") has been submitted\n",
      "Your job 5486101 (\"map_Schistosoma_bovis_GCA_003958945\") has been submitted\n",
      "Your job 5486102 (\"map_Schistosoma_curassoni_GCA_900618015\") has been submitted\n",
      "Your job 5486103 (\"map_Schistosoma_haematobium_GCF_000699445\") has been submitted\n",
      "Your job 5486104 (\"map_Schistosoma_japonicum_GCA_006368765\") has been submitted\n",
      "Your job 5486105 (\"map_Schistosoma_margrebowiei_GCA_900618395\") has been submitted\n",
      "Your job 5486106 (\"map_Schistosoma_mattheei_GCA_900617995\") has been submitted\n",
      "Your job 5486107 (\"map_Trichobilharzia_regenti_GCA_900618515\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486832 (\"merge_Clonorchis_sinensis_GCA_003604175\") has been submitted\n",
      "Your job 5486833 (\"merge_Dicrocoelium_dendriticum_GCA_000950715\") has been submitted\n",
      "Your job 5486834 (\"merge_Echinostoma_caproni_GCA_900618425\") has been submitted\n",
      "Your job 5486835 (\"merge_Fasciola_gigantica_GCA_006461475\") has been submitted\n",
      "Your job 5486836 (\"merge_Fasciola_hepatica_GCA_900302435\") has been submitted\n",
      "Your job 5486837 (\"merge_Fasciolopsis_buski_GCA_008360955\") has been submitted\n",
      "Your job 5486838 (\"merge_Opisthorchis_felineus_GCA_004794785\") has been submitted\n",
      "Your job 5486839 (\"merge_Opisthorchis_viverrini_GCF_000715545\") has been submitted\n",
      "Your job 5486840 (\"merge_Paragonimus_westermani_GCA_008508345\") has been submitted\n",
      "Your job 5486841 (\"merge_Schistosoma_bovis_GCA_003958945\") has been submitted\n",
      "Your job 5486842 (\"merge_Schistosoma_curassoni_GCA_900618015\") has been submitted\n",
      "Your job 5486843 (\"merge_Schistosoma_haematobium_GCF_000699445\") has been submitted\n",
      "Your job 5486844 (\"merge_Schistosoma_japonicum_GCA_006368765\") has been submitted\n",
      "Your job 5486845 (\"merge_Schistosoma_margrebowiei_GCA_900618395\") has been submitted\n",
      "Your job 5486846 (\"merge_Schistosoma_mattheei_GCA_900617995\") has been submitted\n",
      "Your job 5486847 (\"merge_Trichobilharzia_regenti_GCA_900618515\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 831 sequences from Clonorchis_sinensis_GCA_003604175_merged.bed.  Filtered 805 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 26.\n",
      "Screened 217 sequences from Dicrocoelium_dendriticum_GCA_000950715_merged.bed.  Filtered 213 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 926 sequences from Echinostoma_caproni_GCA_900618425_merged.bed.  Filtered 909 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 17.\n",
      "Screened 1179 sequences from Fasciola_gigantica_GCA_006461475_merged.bed.  Filtered 1138 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 41.\n",
      "Screened 1337 sequences from Fasciola_hepatica_GCA_900302435_merged.bed.  Filtered 1297 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 40.\n",
      "Screened 1077 sequences from Fasciolopsis_buski_GCA_008360955_merged.bed.  Filtered 1042 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 35.\n",
      "Screened 620 sequences from Opisthorchis_felineus_GCA_004794785_merged.bed.  Filtered 597 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 23.\n",
      "Screened 819 sequences from Opisthorchis_viverrini_GCF_000715545_merged.bed.  Filtered 799 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 20.\n",
      "Screened 568 sequences from Paragonimus_westermani_GCA_008508345_merged.bed.  Filtered 537 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 31.\n",
      "Screened 442569 sequences from Schistosoma_bovis_GCA_003958945_merged.bed.  Filtered 341104 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 101465.\n",
      "Screened 455856 sequences from Schistosoma_curassoni_GCA_900618015_merged.bed.  Filtered 353468 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102388.\n",
      "Screened 469562 sequences from Schistosoma_haematobium_GCF_000699445_merged.bed.  Filtered 360386 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109176.\n",
      "Screened 36000 sequences from Schistosoma_japonicum_GCA_006368765_merged.bed.  Filtered 28667 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7333.\n",
      "Screened 442733 sequences from Schistosoma_margrebowiei_GCA_900618395_merged.bed.  Filtered 346676 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96057.\n",
      "Screened 449483 sequences from Schistosoma_mattheei_GCA_900617995_merged.bed.  Filtered 346486 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102997.\n",
      "Screened 5422 sequences from Trichobilharzia_regenti_GCA_900618515_merged.bed.  Filtered 4444 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 978.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/trematoda/initial_intervals/Trichobilharzia_regenti_GCA_900618515_merged.bed --twobit results/phyluce/trematoda/cleaned_genomes/Schistosoma_mansoni_GCA_000237925_formatted.2bit --output results/phyluce/trematoda/initial_intervals/Trichobilharzia_regenti_GCA_900618515_stripped.bed;\n",
      "clonorchis_sinensis_gca_003604175.\n",
      "dicrocoelium_dendriticum_gca_000950715.\n",
      "echinostoma_caproni_gca_900618425.\n",
      "fasciola_gigantica_gca_006461475.\n",
      "fasciola_hepatica_gca_900302435.\n",
      "fasciolopsis_buski_gca_008360955.\n",
      "opisthorchis_felineus_gca_004794785.\n",
      "opisthorchis_viverrini_gcf_000715545.\n",
      "paragonimus_westermani_gca_008508345.\n",
      "schistosoma_bovis_gca_003958945......................................................................................................\n",
      "schistosoma_curassoni_gca_900618015.......................................................................................................\n",
      "schistosoma_haematobium_gcf_000699445..............................................................................................................\n",
      "schistosoma_japonicum_gca_006368765........\n",
      "schistosoma_margrebowiei_gca_900618395.................................................................................................\n",
      "schistosoma_mattheei_gca_900617995.......................................................................................................\n",
      "trichobilharzia_regenti_gca_900618515.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/trematoda/initial_intervals/trematoda-to-Schistosoma_mansoni_GCA_000237925.sqlite --base-taxon Schistosoma_mansoni_GCA_000237925\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 0 taxa:\t150,106.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 1 taxa:\t150,106.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 2 taxa:\t120,696.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 3 taxa:\t101,035.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 4 taxa:\t82,688.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 5 taxa:\t59,269.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 6 taxa:\t6,107.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 7 taxa:\t695.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 8 taxa:\t38.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 9 taxa:\t28.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 10 taxa:\t23.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 11 taxa:\t20.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 12 taxa:\t18.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 13 taxa:\t14.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 14 taxa:\t12.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 15 taxa:\t7.0\n",
      "Loci shared by Schistosoma_mansoni_GCA_000237925 + 16 taxa:\t2.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/trematoda/initial_intervals/trematoda-to-Schistosoma_mansoni_GCA_000237925.sqlite --base-taxon Schistosoma_mansoni_GCA_000237925 --output results/phyluce/trematoda/initial_intervals/Schistosoma_mansoni_GCA_000237925_+6.bed --specific-counts 6;\n",
      "Counter({'schistosoma_curassoni_gca_900618015': 6098, 'schistosoma_bovis_gca_003958945': 6094, 'schistosoma_margrebowiei_gca_900618395': 6093, 'schistosoma_haematobium_gcf_000699445': 6089, 'schistosoma_mattheei_gca_900617995': 6082, 'schistosoma_japonicum_gca_006368765': 5974, 'trichobilharzia_regenti_gca_900618515': 860, 'fasciola_gigantica_gca_006461475': 33, 'fasciolopsis_buski_gca_008360955': 33, 'fasciola_hepatica_gca_900302435': 29, 'paragonimus_westermani_gca_008508345': 27, 'clonorchis_sinensis_gca_003604175': 25, 'opisthorchis_felineus_gca_004794785': 22, 'opisthorchis_viverrini_gcf_000715545': 19, 'echinostoma_caproni_gca_900618425': 17, 'dicrocoelium_dendriticum_gca_000950715': 4})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 6\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/trematoda/initial_intervals/Schistosoma_mansoni_GCA_000237925_+6.bed --twobit results/phyluce/trematoda/cleaned_genomes/Schistosoma_mansoni_GCA_000237925_formatted.2bit --buffer-to 160 --output results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6.fasta;\n",
      "Screened 6107 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 6107.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6.fasta --probe-prefix uce_trematoda_ --design trematoda_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 5435\n",
      "Probe Count = 10163\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas --query results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes_vself.lastz;\n",
      "Started:  Mon Feb 10, 2020  14:41:58\n",
      "Ended:  Mon Feb 10, 2020  14:42:05\n",
      "Time for execution:  0.104172933102 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas                        --lastz results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes_vself.lastz                       --probe-prefix=uce_trematoda_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 10162 fasta sequences.  Filtered 372 duplicates. Kept 9461.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/trematoda/validate_intervals/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas --scaffoldlist Clonorchis_sinensis_GCA_003604175 Dicrocoelium_dendriticum_GCA_000950715 Echinostoma_caproni_GCA_900618425 Fasciola_gigantica_GCA_006461475 Fasciola_hepatica_GCA_900302435 Fasciolopsis_buski_GCA_008360955 Opisthorchis_felineus_GCA_004794785 Opisthorchis_viverrini_GCF_000715545 Paragonimus_westermani_GCA_008508345 Schistosoma_bovis_GCA_003958945 Schistosoma_curassoni_GCA_900618015 Schistosoma_haematobium_GCF_000699445 Schistosoma_japonicum_GCA_006368765 Schistosoma_margrebowiei_GCA_900618395 Schistosoma_mattheei_GCA_900617995 Trichobilharzia_regenti_GCA_900618515 Schistosoma_mansoni_GCA_000237925 --genome-base-path results/phyluce/trematoda/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/trematoda/validate_intervals/trematoda-to-Schistosoma_mansoni_GCA_000237925.sqlite --output results/phyluce/trematoda/validate_intervals/lastz/;\n",
      "\n",
      "Running against Clonorchis_sinensis_GCA_003604175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 52 queries...\n",
      "\t/tmp/tmp4WeOV_.fasta\n",
      "\t/tmp/tmp2qsdrU.fasta\n",
      "\t/tmp/tmpm_apsD.fasta\n",
      "\t/tmp/tmpYjn0q1.fasta\n",
      "\t/tmp/tmpSYGREd.fasta\n",
      "\t/tmp/tmpyn7MOD.fasta\n",
      "\t/tmp/tmpx5CIJ4.fasta\n",
      "\t/tmp/tmp9WexI8.fasta\n",
      "\t/tmp/tmplWWhhj.fasta\n",
      "\t/tmp/tmpudXrNA.fasta\n",
      "\t/tmp/tmpfkJq8E.fasta\n",
      "\t/tmp/tmpOS3Sao.fasta\n",
      "\t/tmp/tmp7UKzWF.fasta\n",
      "\t/tmp/tmpyHoTOT.fasta\n",
      "\t/tmp/tmpSzWQH6.fasta\n",
      "\t/tmp/tmp06a9uy.fasta\n",
      "\t/tmp/tmpf5A4Xt.fasta\n",
      "\t/tmp/tmppV8xf_.fasta\n",
      "\t/tmp/tmp1UHBTW.fasta\n",
      "\t/tmp/tmpZl9OIX.fasta\n",
      "\t/tmp/tmpdU8Jtq.fasta\n",
      "\t/tmp/tmpFyrJiL.fasta\n",
      "\t/tmp/tmpL5_0rI.fasta\n",
      "\t/tmp/tmpf6YafE.fasta\n",
      "\t/tmp/tmplppcKB.fasta\n",
      "\t/tmp/tmpk1cBSt.fasta\n",
      "\t/tmp/tmplMdmjY.fasta\n",
      "\t/tmp/tmpUu6YQO.fasta\n",
      "\t/tmp/tmpAxRT_l.fasta\n",
      "\t/tmp/tmpfjAYu4.fasta\n",
      "\t/tmp/tmpNDBS2k.fasta\n",
      "\t/tmp/tmpOqu_SR.fasta\n",
      "\t/tmp/tmpj5KRXT.fasta\n",
      "\t/tmp/tmpgL9_8y.fasta\n",
      "\t/tmp/tmpWQ3O0U.fasta\n",
      "\t/tmp/tmpBeUEI1.fasta\n",
      "\t/tmp/tmpPSOELn.fasta\n",
      "\t/tmp/tmpQgbBL0.fasta\n",
      "\t/tmp/tmpYdLua_.fasta\n",
      "\t/tmp/tmpkV4GNw.fasta\n",
      "\t/tmp/tmpohTa0I.fasta\n",
      "\t/tmp/tmpRd2OPE.fasta\n",
      "\t/tmp/tmpkie9eK.fasta\n",
      "\t/tmp/tmpTgRFc9.fasta\n",
      "\t/tmp/tmpc4TiVf.fasta\n",
      "\t/tmp/tmpyseAYd.fasta\n",
      "\t/tmp/tmptS5Rfl.fasta\n",
      "\t/tmp/tmpqUYwO5.fasta\n",
      "\t/tmp/tmpeXtI1o.fasta\n",
      "\t/tmp/tmpnxmXmK.fasta\n",
      "\t/tmp/tmpKmmCRC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpw4XYRh.lastz\n",
      "\t/tmp/tmpFFwX6V.lastz\n",
      "\t/tmp/tmphuPi2X.lastz\n",
      "\t/tmp/tmpKVcF1Q.lastz\n",
      "\t/tmp/tmpT6wt2G.lastz\n",
      "\t/tmp/tmpGDUilF.lastz\n",
      "\t/tmp/tmpGsxyuQ.lastz\n",
      "\t/tmp/tmpXLGXjo.lastz\n",
      "\t/tmp/tmpbpJx6S.lastz\n",
      "\t/tmp/tmp0tFKas.lastz\n",
      "\t/tmp/tmpEeco63.lastz\n",
      "\t/tmp/tmpivmde0.lastz\n",
      "\t/tmp/tmpCH965H.lastz\n",
      "\t/tmp/tmpIkQc0m.lastz\n",
      "\t/tmp/tmpiuarV5.lastz\n",
      "\t/tmp/tmpx4mO18.lastz\n",
      "\t/tmp/tmpEVAxyK.lastz\n",
      "\t/tmp/tmpFjgqsq.lastz\n",
      "\t/tmp/tmpwRhiKs.lastz\n",
      "\t/tmp/tmpdABuBu.lastz\n",
      "\t/tmp/tmplKvBRS.lastz\n",
      "\t/tmp/tmpbNkZ7A.lastz\n",
      "\t/tmp/tmpNacSa8.lastz\n",
      "\t/tmp/tmp5QAaey.lastz\n",
      "\t/tmp/tmpb3xK4r.lastz\n",
      "\t/tmp/tmpEoeZoj.lastz\n",
      "\t/tmp/tmpOoC9KE.lastz\n",
      "\t/tmp/tmpFclbiv.lastz\n",
      "\t/tmp/tmpW_oKvp.lastz\n",
      "\t/tmp/tmpad3Ncm.lastz\n",
      "\t/tmp/tmp11xSA6.lastz\n",
      "\t/tmp/tmpV6BIrt.lastz\n",
      "\t/tmp/tmpd7bD08.lastz\n",
      "\t/tmp/tmpu9hHUM.lastz\n",
      "\t/tmp/tmpeUEdAb.lastz\n",
      "\t/tmp/tmpwNTI6S.lastz\n",
      "\t/tmp/tmpSc2P7a.lastz\n",
      "\t/tmp/tmpheOlYs.lastz\n",
      "\t/tmp/tmp2ZylIi.lastz\n",
      "\t/tmp/tmpehznl2.lastz\n",
      "\t/tmp/tmpt4umF6.lastz\n",
      "\t/tmp/tmp66gJnf.lastz\n",
      "\t/tmp/tmpdC61ex.lastz\n",
      "\t/tmp/tmpqmUfgO.lastz\n",
      "\t/tmp/tmpQOcb6j.lastz\n",
      "\t/tmp/tmpXL1_Xo.lastz\n",
      "\t/tmp/tmpmojKV_.lastz\n",
      "\t/tmp/tmp7Z4Ip8.lastz\n",
      "\t/tmp/tmpHZk5f5.lastz\n",
      "\t/tmp/tmpKk5ubF.lastz\n",
      "\t/tmp/tmphDsBmD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Clonorchis_sinensis_GCA_003604175.lastz\n",
      "Creating Clonorchis_sinensis_GCA_003604175 table\n",
      "Inserting data to Clonorchis_sinensis_GCA_003604175 table\n",
      "\n",
      "Running against Dicrocoelium_dendriticum_GCA_000950715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 55 queries...\n",
      "\t/tmp/tmpf6_cUD.fasta\n",
      "\t/tmp/tmp8KIeLV.fasta\n",
      "\t/tmp/tmpHIwGN8.fasta\n",
      "\t/tmp/tmpxYpW9H.fasta\n",
      "\t/tmp/tmpmmVN70.fasta\n",
      "\t/tmp/tmpHLYQbV.fasta\n",
      "\t/tmp/tmpZZArdP.fasta\n",
      "\t/tmp/tmpgFMeqw.fasta\n",
      "\t/tmp/tmpccb2B6.fasta\n",
      "\t/tmp/tmp6P3JWs.fasta\n",
      "\t/tmp/tmp541Oa3.fasta\n",
      "\t/tmp/tmp_BKTFR.fasta\n",
      "\t/tmp/tmpFcj4rg.fasta\n",
      "\t/tmp/tmp6Td11S.fasta\n",
      "\t/tmp/tmpS5gbt2.fasta\n",
      "\t/tmp/tmp2PS_D6.fasta\n",
      "\t/tmp/tmpQ1mmZ3.fasta\n",
      "\t/tmp/tmpzH4s7y.fasta\n",
      "\t/tmp/tmp5SoOZa.fasta\n",
      "\t/tmp/tmpP3HPui.fasta\n",
      "\t/tmp/tmpVV3EkG.fasta\n",
      "\t/tmp/tmp6GH5Wf.fasta\n",
      "\t/tmp/tmp7RLn1d.fasta\n",
      "\t/tmp/tmpqzrAmn.fasta\n",
      "\t/tmp/tmp0txhAF.fasta\n",
      "\t/tmp/tmpYnDIM3.fasta\n",
      "\t/tmp/tmpPTfkOc.fasta\n",
      "\t/tmp/tmpgS5lCj.fasta\n",
      "\t/tmp/tmpkaAaNl.fasta\n",
      "\t/tmp/tmpe1RdlO.fasta\n",
      "\t/tmp/tmpV1NhaA.fasta\n",
      "\t/tmp/tmpetcqwH.fasta\n",
      "\t/tmp/tmp4kp8UV.fasta\n",
      "\t/tmp/tmpQzwWEF.fasta\n",
      "\t/tmp/tmpGkeGgU.fasta\n",
      "\t/tmp/tmp9yW83M.fasta\n",
      "\t/tmp/tmpa5x3Kl.fasta\n",
      "\t/tmp/tmpdkDqp9.fasta\n",
      "\t/tmp/tmpf1eisv.fasta\n",
      "\t/tmp/tmpQdD6U6.fasta\n",
      "\t/tmp/tmpQztqnG.fasta\n",
      "\t/tmp/tmpL6asMq.fasta\n",
      "\t/tmp/tmp_EGbiw.fasta\n",
      "\t/tmp/tmp_l6vmn.fasta\n",
      "\t/tmp/tmpykua4M.fasta\n",
      "\t/tmp/tmpbYLvQU.fasta\n",
      "\t/tmp/tmpdjb5Fe.fasta\n",
      "\t/tmp/tmpkYvKlI.fasta\n",
      "\t/tmp/tmpmQWbpz.fasta\n",
      "\t/tmp/tmpUWs81W.fasta\n",
      "\t/tmp/tmpcmJdFp.fasta\n",
      "\t/tmp/tmplWwoGb.fasta\n",
      "\t/tmp/tmp4TdG37.fasta\n",
      "\t/tmp/tmpPJIIR0.fasta\n",
      "\t/tmp/tmpyhMkzU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQLCUjx.lastz\n",
      "\t/tmp/tmpa2H8aF.lastz\n",
      "\t/tmp/tmpRfoNc0.lastz\n",
      "\t/tmp/tmp8Wtypo.lastz\n",
      "\t/tmp/tmp5vJ2Xp.lastz\n",
      "\t/tmp/tmpLYFNst.lastz\n",
      "\t/tmp/tmpc6Cntg.lastz\n",
      "\t/tmp/tmp96i3Fh.lastz\n",
      "\t/tmp/tmpVehHcn.lastz\n",
      "\t/tmp/tmpIwy3OB.lastz\n",
      "\t/tmp/tmpl996rU.lastz\n",
      "\t/tmp/tmpOIyOQG.lastz\n",
      "\t/tmp/tmpgpyz5B.lastz\n",
      "\t/tmp/tmphrp2n5.lastz\n",
      "\t/tmp/tmpB7o0F6.lastz\n",
      "\t/tmp/tmphWJTDI.lastz\n",
      "\t/tmp/tmpZzUzoK.lastz\n",
      "\t/tmp/tmpA_PBLm.lastz\n",
      "\t/tmp/tmpcMpU1S.lastz\n",
      "\t/tmp/tmp4R0kb0.lastz\n",
      "\t/tmp/tmpBGrgst.lastz\n",
      "\t/tmp/tmpqWtDoh.lastz\n",
      "\t/tmp/tmpReDKTL.lastz\n",
      "\t/tmp/tmpHvngSp.lastz\n",
      "\t/tmp/tmp6pY8gs.lastz\n",
      "\t/tmp/tmpj5Sy_k.lastz\n",
      "\t/tmp/tmpqo2hcz.lastz\n",
      "\t/tmp/tmpoohZW1.lastz\n",
      "\t/tmp/tmpGcfTD9.lastz\n",
      "\t/tmp/tmpzu_GqB.lastz\n",
      "\t/tmp/tmp8zxVsk.lastz\n",
      "\t/tmp/tmp_utpaR.lastz\n",
      "\t/tmp/tmpwX_4kf.lastz\n",
      "\t/tmp/tmp1cIQUk.lastz\n",
      "\t/tmp/tmpOAqkHS.lastz\n",
      "\t/tmp/tmpWNW9eu.lastz\n",
      "\t/tmp/tmpZYT2Sk.lastz\n",
      "\t/tmp/tmpclnGnz.lastz\n",
      "\t/tmp/tmpRvo2lC.lastz\n",
      "\t/tmp/tmph6EQCT.lastz\n",
      "\t/tmp/tmpuUFWTW.lastz\n",
      "\t/tmp/tmpFgig7J.lastz\n",
      "\t/tmp/tmpa7O3j4.lastz\n",
      "\t/tmp/tmpJ0gTBB.lastz\n",
      "\t/tmp/tmpO8gMPy.lastz\n",
      "\t/tmp/tmp27UHHK.lastz\n",
      "\t/tmp/tmpfulbGq.lastz\n",
      "\t/tmp/tmp3Co572.lastz\n",
      "\t/tmp/tmpyeIxRn.lastz\n",
      "\t/tmp/tmpPtMtKf.lastz\n",
      "\t/tmp/tmphZYbf6.lastz\n",
      "\t/tmp/tmpPuRcZK.lastz\n",
      "\t/tmp/tmpvTod7B.lastz\n",
      "\t/tmp/tmpLyWpxD.lastz\n",
      "\t/tmp/tmpCf04wK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Dicrocoelium_dendriticum_GCA_000950715.lastz\n",
      "Creating Dicrocoelium_dendriticum_GCA_000950715 table\n",
      "Inserting data to Dicrocoelium_dendriticum_GCA_000950715 table\n",
      "\n",
      "Running against Echinostoma_caproni_GCA_900618425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 84 queries...\n",
      "\t/tmp/tmpVpZ7L7.fasta\n",
      "\t/tmp/tmp3XQQQK.fasta\n",
      "\t/tmp/tmpICwvVa.fasta\n",
      "\t/tmp/tmp5CJIV2.fasta\n",
      "\t/tmp/tmp7ht6A7.fasta\n",
      "\t/tmp/tmpsAa18j.fasta\n",
      "\t/tmp/tmp5kEM8I.fasta\n",
      "\t/tmp/tmptMzHRR.fasta\n",
      "\t/tmp/tmpnfZjkG.fasta\n",
      "\t/tmp/tmpcUVmBS.fasta\n",
      "\t/tmp/tmpGgtQGQ.fasta\n",
      "\t/tmp/tmppT8JVS.fasta\n",
      "\t/tmp/tmps0Uac4.fasta\n",
      "\t/tmp/tmpb5Ch14.fasta\n",
      "\t/tmp/tmp_rHs0O.fasta\n",
      "\t/tmp/tmpO_qZlO.fasta\n",
      "\t/tmp/tmpPAXoz_.fasta\n",
      "\t/tmp/tmpgSIQVW.fasta\n",
      "\t/tmp/tmpds5IZ_.fasta\n",
      "\t/tmp/tmpa13mJN.fasta\n",
      "\t/tmp/tmpePXISR.fasta\n",
      "\t/tmp/tmp82Uh6x.fasta\n",
      "\t/tmp/tmpOryabD.fasta\n",
      "\t/tmp/tmplQjqTN.fasta\n",
      "\t/tmp/tmpTSLFn0.fasta\n",
      "\t/tmp/tmpRmTZA0.fasta\n",
      "\t/tmp/tmpaPfY1r.fasta\n",
      "\t/tmp/tmpsnztN_.fasta\n",
      "\t/tmp/tmpgLSDHz.fasta\n",
      "\t/tmp/tmp5SQBJv.fasta\n",
      "\t/tmp/tmptpIvT7.fasta\n",
      "\t/tmp/tmppN27Lx.fasta\n",
      "\t/tmp/tmpq5uJy7.fasta\n",
      "\t/tmp/tmpS9j5Vc.fasta\n",
      "\t/tmp/tmpIbk5nF.fasta\n",
      "\t/tmp/tmpLW7e4E.fasta\n",
      "\t/tmp/tmpP9v6pl.fasta\n",
      "\t/tmp/tmp6M2grI.fasta\n",
      "\t/tmp/tmpVNGjK8.fasta\n",
      "\t/tmp/tmpJu35EV.fasta\n",
      "\t/tmp/tmprfyrHE.fasta\n",
      "\t/tmp/tmpjHzJdy.fasta\n",
      "\t/tmp/tmpkUTHTT.fasta\n",
      "\t/tmp/tmpfn4LRA.fasta\n",
      "\t/tmp/tmpQprVOp.fasta\n",
      "\t/tmp/tmpdOoNAh.fasta\n",
      "\t/tmp/tmpMO_x2g.fasta\n",
      "\t/tmp/tmp39Zzd7.fasta\n",
      "\t/tmp/tmp1NJfqO.fasta\n",
      "\t/tmp/tmpSgQtmN.fasta\n",
      "\t/tmp/tmpyQcUPP.fasta\n",
      "\t/tmp/tmp3ZLkVU.fasta\n",
      "\t/tmp/tmpH33Jec.fasta\n",
      "\t/tmp/tmpZO5PH9.fasta\n",
      "\t/tmp/tmp2RLAAb.fasta\n",
      "\t/tmp/tmp3lPyrQ.fasta\n",
      "\t/tmp/tmpiluuGt.fasta\n",
      "\t/tmp/tmps943wP.fasta\n",
      "\t/tmp/tmpik1Eif.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmphtGoHb.fasta\n",
      "\t/tmp/tmpaNQPtm.fasta\n",
      "\t/tmp/tmprdqv48.fasta\n",
      "\t/tmp/tmpLSPKuW.fasta\n",
      "\t/tmp/tmpEevwvq.fasta\n",
      "\t/tmp/tmp3bOcBU.fasta\n",
      "\t/tmp/tmps7028B.fasta\n",
      "\t/tmp/tmpoRdEiP.fasta\n",
      "\t/tmp/tmpMRuoWu.fasta\n",
      "\t/tmp/tmpsJAnDi.fasta\n",
      "\t/tmp/tmpxwzcsT.fasta\n",
      "\t/tmp/tmpnYZh_j.fasta\n",
      "\t/tmp/tmpuBRM4o.fasta\n",
      "\t/tmp/tmp0HeQIx.fasta\n",
      "\t/tmp/tmpbklzqZ.fasta\n",
      "\t/tmp/tmp5_26ZD.fasta\n",
      "\t/tmp/tmp5p7i6F.fasta\n",
      "\t/tmp/tmpaPMPZO.fasta\n",
      "\t/tmp/tmpF0apuw.fasta\n",
      "\t/tmp/tmpvRA510.fasta\n",
      "\t/tmp/tmpoYy_F2.fasta\n",
      "\t/tmp/tmpY2XeKQ.fasta\n",
      "\t/tmp/tmphflLNO.fasta\n",
      "\t/tmp/tmpLXKKXE.fasta\n",
      "\t/tmp/tmp1Sdykm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpC_5D8M.lastz\n",
      "\t/tmp/tmp0D0Vvt.lastz\n",
      "\t/tmp/tmpwY9Crd.lastz\n",
      "\t/tmp/tmpi6l40o.lastz\n",
      "\t/tmp/tmpsPWKEn.lastz\n",
      "\t/tmp/tmpAhJb1O.lastz\n",
      "\t/tmp/tmpOHocZk.lastz\n",
      "\t/tmp/tmp6svkQu.lastz\n",
      "\t/tmp/tmpG4aFE3.lastz\n",
      "\t/tmp/tmpy6oLzY.lastz\n",
      "\t/tmp/tmpjzA41W.lastz\n",
      "\t/tmp/tmpq2U6HU.lastz\n",
      "\t/tmp/tmpX1oJLf.lastz\n",
      "\t/tmp/tmptRnhaW.lastz\n",
      "\t/tmp/tmp91_j2c.lastz\n",
      "\t/tmp/tmpCHzMDi.lastz\n",
      "\t/tmp/tmp4jghab.lastz\n",
      "\t/tmp/tmpikm2hK.lastz\n",
      "\t/tmp/tmpf3UWDQ.lastz\n",
      "\t/tmp/tmpjpFWz6.lastz\n",
      "\t/tmp/tmpJ4RKFj.lastz\n",
      "\t/tmp/tmpacNPhK.lastz\n",
      "\t/tmp/tmp7MEdq5.lastz\n",
      "\t/tmp/tmptvQzeq.lastz\n",
      "\t/tmp/tmpwxoGPt.lastz\n",
      "\t/tmp/tmpip7G7K.lastz\n",
      "\t/tmp/tmp2bCmQz.lastz\n",
      "\t/tmp/tmpWljbKd.lastz\n",
      "\t/tmp/tmpIK4X9G.lastz\n",
      "\t/tmp/tmpDLXCwJ.lastz\n",
      "\t/tmp/tmpvucWos.lastz\n",
      "\t/tmp/tmp1IWds7.lastz\n",
      "\t/tmp/tmpXsXOuj.lastz\n",
      "\t/tmp/tmp2rrbvl.lastz\n",
      "\t/tmp/tmpc0tUf2.lastz\n",
      "\t/tmp/tmpjdqLEn.lastz\n",
      "\t/tmp/tmps4asDH.lastz\n",
      "\t/tmp/tmpxORbF5.lastz\n",
      "\t/tmp/tmpcAsbhN.lastz\n",
      "\t/tmp/tmpPp1rNf.lastz\n",
      "\t/tmp/tmpL9vf5I.lastz\n",
      "\t/tmp/tmpzaa5ex.lastz\n",
      "\t/tmp/tmp4AQJ_C.lastz\n",
      "\t/tmp/tmpUnoMU5.lastz\n",
      "\t/tmp/tmpJpe69Y.lastz\n",
      "\t/tmp/tmpugFtSB.lastz\n",
      "\t/tmp/tmpeiwQQH.lastz\n",
      "\t/tmp/tmpqyTfcU.lastz\n",
      "\t/tmp/tmphNqIy5.lastz\n",
      "\t/tmp/tmpxAGLtf.lastz\n",
      "\t/tmp/tmpnob8mm.lastz\n",
      "\t/tmp/tmpWiF_ap.lastz\n",
      "\t/tmp/tmpZhppzJ.lastz\n",
      "\t/tmp/tmpKn5Bz3.lastz\n",
      "\t/tmp/tmpoAfnoq.lastz\n",
      "\t/tmp/tmpGXYpps.lastz\n",
      "\t/tmp/tmpsnlyZT.lastz\n",
      "\t/tmp/tmpL0ZNhK.lastz\n",
      "\t/tmp/tmpQVhnOa.lastz\n",
      "\t/tmp/tmp1vhq94.lastz\n",
      "\t/tmp/tmpiWDLNE.lastz\n",
      "\t/tmp/tmpF5cwLB.lastz\n",
      "\t/tmp/tmpdDafu_.lastz\n",
      "\t/tmp/tmp8OqRqp.lastz\n",
      "\t/tmp/tmpcQFjRF.lastz\n",
      "\t/tmp/tmpAJfc2g.lastz\n",
      "\t/tmp/tmpBk1e4j.lastz\n",
      "\t/tmp/tmpQ8mUI8.lastz\n",
      "\t/tmp/tmpLAZsc0.lastz\n",
      "\t/tmp/tmpKknakG.lastz\n",
      "\t/tmp/tmpTUwXAK.lastz\n",
      "\t/tmp/tmp_rD50j.lastz\n",
      "\t/tmp/tmppNqJp9.lastz\n",
      "\t/tmp/tmpDzgeZK.lastz\n",
      "\t/tmp/tmpksbEgJ.lastz\n",
      "\t/tmp/tmpvzUhjK.lastz\n",
      "\t/tmp/tmptU5nng.lastz\n",
      "\t/tmp/tmp94A8Hb.lastz\n",
      "\t/tmp/tmpOyNUeC.lastz\n",
      "\t/tmp/tmp9Q2v7V.lastz\n",
      "\t/tmp/tmpO2hxKU.lastz\n",
      "\t/tmp/tmpzVwal_.lastz\n",
      "\t/tmp/tmp_YynX5.lastz\n",
      "\t/tmp/tmpbd2HPK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Echinostoma_caproni_GCA_900618425.lastz\n",
      "Creating Echinostoma_caproni_GCA_900618425 table\n",
      "Inserting data to Echinostoma_caproni_GCA_900618425 table\n",
      "\n",
      "Running against Fasciola_gigantica_GCA_006461475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 112 queries...\n",
      "\t/tmp/tmpnXXZo3.fasta\n",
      "\t/tmp/tmpRZJPyc.fasta\n",
      "\t/tmp/tmpgCzdgQ.fasta\n",
      "\t/tmp/tmpVwZDYl.fasta\n",
      "\t/tmp/tmpIlfWfs.fasta\n",
      "\t/tmp/tmp2O6Y0K.fasta\n",
      "\t/tmp/tmp3i5l8I.fasta\n",
      "\t/tmp/tmphr56kZ.fasta\n",
      "\t/tmp/tmpQAapRH.fasta\n",
      "\t/tmp/tmpdRLNhy.fasta\n",
      "\t/tmp/tmpHG5mPv.fasta\n",
      "\t/tmp/tmpwhMARl.fasta\n",
      "\t/tmp/tmpnhP9KI.fasta\n",
      "\t/tmp/tmpOv5dLd.fasta\n",
      "\t/tmp/tmpMOOvzs.fasta\n",
      "\t/tmp/tmpdlcDTa.fasta\n",
      "\t/tmp/tmpAQG4nf.fasta\n",
      "\t/tmp/tmpaus4I4.fasta\n",
      "\t/tmp/tmptE8EQ_.fasta\n",
      "\t/tmp/tmpmkC9CT.fasta\n",
      "\t/tmp/tmpb6Pk4q.fasta\n",
      "\t/tmp/tmp252T3i.fasta\n",
      "\t/tmp/tmpFUnazC.fasta\n",
      "\t/tmp/tmpaitE9u.fasta\n",
      "\t/tmp/tmpZuGz84.fasta\n",
      "\t/tmp/tmphA9vr9.fasta\n",
      "\t/tmp/tmpWC_nKP.fasta\n",
      "\t/tmp/tmpnZkoHR.fasta\n",
      "\t/tmp/tmp1Os_Z0.fasta\n",
      "\t/tmp/tmp4ErvjF.fasta\n",
      "\t/tmp/tmp8uz1RX.fasta\n",
      "\t/tmp/tmpMyh2nm.fasta\n",
      "\t/tmp/tmp20x8ei.fasta\n",
      "\t/tmp/tmpfUMYSL.fasta\n",
      "\t/tmp/tmpUsKGub.fasta\n",
      "\t/tmp/tmpzIVeWc.fasta\n",
      "\t/tmp/tmpZ3xR8k.fasta\n",
      "\t/tmp/tmpyAfObn.fasta\n",
      "\t/tmp/tmpp6iY9P.fasta\n",
      "\t/tmp/tmpwcV4Ll.fasta\n",
      "\t/tmp/tmpJwhXOf.fasta\n",
      "\t/tmp/tmpVyNrYX.fasta\n",
      "\t/tmp/tmpp_HBqd.fasta\n",
      "\t/tmp/tmpviMd38.fasta\n",
      "\t/tmp/tmpDD5akj.fasta\n",
      "\t/tmp/tmpeXcd_7.fasta\n",
      "\t/tmp/tmpgrHkqG.fasta\n",
      "\t/tmp/tmpWLCjKK.fasta\n",
      "\t/tmp/tmpNmts2g.fasta\n",
      "\t/tmp/tmpWjJJXX.fasta\n",
      "\t/tmp/tmp2GGcRq.fasta\n",
      "\t/tmp/tmp2S6FWm.fasta\n",
      "\t/tmp/tmp5B_SSJ.fasta\n",
      "\t/tmp/tmpDwPRJb.fasta\n",
      "\t/tmp/tmp4druxA.fasta\n",
      "\t/tmp/tmpBzMtwu.fasta\n",
      "\t/tmp/tmp6KqQiu.fasta\n",
      "\t/tmp/tmpcCPvOy.fasta\n",
      "\t/tmp/tmpJf_Mwb.fasta\n",
      "\t/tmp/tmpLy2aVp.fasta\n",
      "\t/tmp/tmpFCPrCy.fasta\n",
      "\t/tmp/tmp_OHtLa.fasta\n",
      "\t/tmp/tmp95IKGq.fasta\n",
      "\t/tmp/tmp4o2my1.fasta\n",
      "\t/tmp/tmpXWFZtE.fasta\n",
      "\t/tmp/tmp3j3wxU.fasta\n",
      "\t/tmp/tmp9D5crI.fasta\n",
      "\t/tmp/tmps7ZeaU.fasta\n",
      "\t/tmp/tmpdEFFR3.fasta\n",
      "\t/tmp/tmpZjiyfY.fasta\n",
      "\t/tmp/tmplq6k7e.fasta\n",
      "\t/tmp/tmpnTg1uu.fasta\n",
      "\t/tmp/tmpTQVrUb.fasta\n",
      "\t/tmp/tmp6DQq3C.fasta\n",
      "\t/tmp/tmpglghCt.fasta\n",
      "\t/tmp/tmpGb28sC.fasta\n",
      "\t/tmp/tmp3zlvjj.fasta\n",
      "\t/tmp/tmp963Q8y.fasta\n",
      "\t/tmp/tmpuEWeCq.fasta\n",
      "\t/tmp/tmpmBww_I.fasta\n",
      "\t/tmp/tmpzyur9Z.fasta\n",
      "\t/tmp/tmpO4NTii.fasta\n",
      "\t/tmp/tmpPd82Dx.fasta\n",
      "\t/tmp/tmpddF0jk.fasta\n",
      "\t/tmp/tmp8RzzOY.fasta\n",
      "\t/tmp/tmpBspuSC.fasta\n",
      "\t/tmp/tmpSCF9oi.fasta\n",
      "\t/tmp/tmpFbepgo.fasta\n",
      "\t/tmp/tmp9F2v3t.fasta\n",
      "\t/tmp/tmpKnfxNJ.fasta\n",
      "\t/tmp/tmphKDQet.fasta\n",
      "\t/tmp/tmpyNvrWU.fasta\n",
      "\t/tmp/tmptBUilq.fasta\n",
      "\t/tmp/tmpd_OR3a.fasta\n",
      "\t/tmp/tmpLEvyyF.fasta\n",
      "\t/tmp/tmpAveY2t.fasta\n",
      "\t/tmp/tmpSPyhjb.fasta\n",
      "\t/tmp/tmpkhrWdO.fasta\n",
      "\t/tmp/tmpknW_bt.fasta\n",
      "\t/tmp/tmpdjiXPi.fasta\n",
      "\t/tmp/tmpGAs_16.fasta\n",
      "\t/tmp/tmpM7pvjz.fasta\n",
      "\t/tmp/tmp1i8baU.fasta\n",
      "\t/tmp/tmpwjKSmg.fasta\n",
      "\t/tmp/tmp8qnYG0.fasta\n",
      "\t/tmp/tmp5brrp6.fasta\n",
      "\t/tmp/tmporIYp3.fasta\n",
      "\t/tmp/tmpYrk5QA.fasta\n",
      "\t/tmp/tmp741q69.fasta\n",
      "\t/tmp/tmpQlpZAC.fasta\n",
      "\t/tmp/tmpCk4gcV.fasta\n",
      "\t/tmp/tmpLX61TF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNc8aa3.lastz\n",
      "\t/tmp/tmponBGL5.lastz\n",
      "\t/tmp/tmpSlHG9e.lastz\n",
      "\t/tmp/tmp_N2nYA.lastz\n",
      "\t/tmp/tmpIFnfyd.lastz\n",
      "\t/tmp/tmp5k9zP4.lastz\n",
      "\t/tmp/tmpNj2nsm.lastz\n",
      "\t/tmp/tmp28m56X.lastz\n",
      "\t/tmp/tmp0F9Vwt.lastz\n",
      "\t/tmp/tmpB2UDEf.lastz\n",
      "\t/tmp/tmpPLj06f.lastz\n",
      "\t/tmp/tmpgCOpB3.lastz\n",
      "\t/tmp/tmpSn1aZy.lastz\n",
      "\t/tmp/tmpXYLY3v.lastz\n",
      "\t/tmp/tmpRQjb1x.lastz\n",
      "\t/tmp/tmpTXx1cO.lastz\n",
      "\t/tmp/tmpzbqS6i.lastz\n",
      "\t/tmp/tmplQRPaU.lastz\n",
      "\t/tmp/tmpgZc3ji.lastz\n",
      "\t/tmp/tmpbbiDuf.lastz\n",
      "\t/tmp/tmpcfyGlN.lastz\n",
      "\t/tmp/tmpH4fHtR.lastz\n",
      "\t/tmp/tmp89cF7p.lastz\n",
      "\t/tmp/tmpKu2dJ7.lastz\n",
      "\t/tmp/tmpUbq6uT.lastz\n",
      "\t/tmp/tmp68ycEY.lastz\n",
      "\t/tmp/tmpd7y65r.lastz\n",
      "\t/tmp/tmpAj8zPD.lastz\n",
      "\t/tmp/tmprPKmBo.lastz\n",
      "\t/tmp/tmpNLvsEf.lastz\n",
      "\t/tmp/tmpVYzYMT.lastz\n",
      "\t/tmp/tmpgkNhGv.lastz\n",
      "\t/tmp/tmp9UkmSt.lastz\n",
      "\t/tmp/tmp2fS4I4.lastz\n",
      "\t/tmp/tmpPrRmjp.lastz\n",
      "\t/tmp/tmp20anoN.lastz\n",
      "\t/tmp/tmpmISxzU.lastz\n",
      "\t/tmp/tmp3hh70T.lastz\n",
      "\t/tmp/tmp3AR5FL.lastz\n",
      "\t/tmp/tmpFXD2HY.lastz\n",
      "\t/tmp/tmpxKHPVl.lastz\n",
      "\t/tmp/tmpf_VWzu.lastz\n",
      "\t/tmp/tmpmcAGXW.lastz\n",
      "\t/tmp/tmpc33t_F.lastz\n",
      "\t/tmp/tmpmrLdpZ.lastz\n",
      "\t/tmp/tmpxr0oDo.lastz\n",
      "\t/tmp/tmphkM9JV.lastz\n",
      "\t/tmp/tmpcf2xrz.lastz\n",
      "\t/tmp/tmpfllXbK.lastz\n",
      "\t/tmp/tmpEfosS2.lastz\n",
      "\t/tmp/tmpvBbXEb.lastz\n",
      "\t/tmp/tmpmCotKx.lastz\n",
      "\t/tmp/tmpaVFvfa.lastz\n",
      "\t/tmp/tmp4tVnQP.lastz\n",
      "\t/tmp/tmpScxfKx.lastz\n",
      "\t/tmp/tmpsvbBJW.lastz\n",
      "\t/tmp/tmpSOWsHc.lastz\n",
      "\t/tmp/tmpJcwJYL.lastz\n",
      "\t/tmp/tmpoEaDUK.lastz\n",
      "\t/tmp/tmp1EQ3Rx.lastz\n",
      "\t/tmp/tmp3EXycr.lastz\n",
      "\t/tmp/tmpOk8YH8.lastz\n",
      "\t/tmp/tmpF8H3bE.lastz\n",
      "\t/tmp/tmpTBM0vl.lastz\n",
      "\t/tmp/tmp6bNC0q.lastz\n",
      "\t/tmp/tmpS0JY0Q.lastz\n",
      "\t/tmp/tmp7sHmJk.lastz\n",
      "\t/tmp/tmpYvluIk.lastz\n",
      "\t/tmp/tmpoysRRA.lastz\n",
      "\t/tmp/tmpVG7BF5.lastz\n",
      "\t/tmp/tmpjFXjqv.lastz\n",
      "\t/tmp/tmpxCUeA7.lastz\n",
      "\t/tmp/tmpzGX1Em.lastz\n",
      "\t/tmp/tmpzNmrZ2.lastz\n",
      "\t/tmp/tmpMMTHHy.lastz\n",
      "\t/tmp/tmpcHx6zY.lastz\n",
      "\t/tmp/tmp7XZ1aC.lastz\n",
      "\t/tmp/tmpr5fbcJ.lastz\n",
      "\t/tmp/tmpfrvQz9.lastz\n",
      "\t/tmp/tmpGHDkq_.lastz\n",
      "\t/tmp/tmpVGExdT.lastz\n",
      "\t/tmp/tmpkJVAYM.lastz\n",
      "\t/tmp/tmpYeBkmb.lastz\n",
      "\t/tmp/tmp0i5NAq.lastz\n",
      "\t/tmp/tmppF7bOE.lastz\n",
      "\t/tmp/tmpjoielF.lastz\n",
      "\t/tmp/tmp41WsqY.lastz\n",
      "\t/tmp/tmpRm7vIA.lastz\n",
      "\t/tmp/tmpLeihEl.lastz\n",
      "\t/tmp/tmpeSLa9h.lastz\n",
      "\t/tmp/tmpOSOaog.lastz\n",
      "\t/tmp/tmpd_xhcU.lastz\n",
      "\t/tmp/tmp8uM3rX.lastz\n",
      "\t/tmp/tmp0Gm41M.lastz\n",
      "\t/tmp/tmplhuE3p.lastz\n",
      "\t/tmp/tmpVlz_hB.lastz\n",
      "\t/tmp/tmpDaegmX.lastz\n",
      "\t/tmp/tmpUKmhz6.lastz\n",
      "\t/tmp/tmperxdgf.lastz\n",
      "\t/tmp/tmpWjMt8E.lastz\n",
      "\t/tmp/tmpQyr3cF.lastz\n",
      "\t/tmp/tmp6Ey94c.lastz\n",
      "\t/tmp/tmpxJ2lJe.lastz\n",
      "\t/tmp/tmpelR1tJ.lastz\n",
      "\t/tmp/tmpRbMaLM.lastz\n",
      "\t/tmp/tmpeDXrpl.lastz\n",
      "\t/tmp/tmpgMifT5.lastz\n",
      "\t/tmp/tmp9w_NpH.lastz\n",
      "\t/tmp/tmpmV1LAm.lastz\n",
      "\t/tmp/tmpO1h3uE.lastz\n",
      "\t/tmp/tmpXE7h_5.lastz\n",
      "\t/tmp/tmpS4zu7H.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Fasciola_gigantica_GCA_006461475.lastz\n",
      "Creating Fasciola_gigantica_GCA_006461475 table\n",
      "Inserting data to Fasciola_gigantica_GCA_006461475 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Fasciola_hepatica_GCA_900302435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 109 queries...\n",
      "\t/tmp/tmpnVQOOh.fasta\n",
      "\t/tmp/tmpdaJXrM.fasta\n",
      "\t/tmp/tmple6b5z.fasta\n",
      "\t/tmp/tmpxNA6GW.fasta\n",
      "\t/tmp/tmpGaZdFs.fasta\n",
      "\t/tmp/tmpzQIQqL.fasta\n",
      "\t/tmp/tmpwEiolZ.fasta\n",
      "\t/tmp/tmpfsC0t1.fasta\n",
      "\t/tmp/tmpgpk1Y1.fasta\n",
      "\t/tmp/tmpSnf4Go.fasta\n",
      "\t/tmp/tmpAe0i_p.fasta\n",
      "\t/tmp/tmp4FbAqs.fasta\n",
      "\t/tmp/tmpVgG21_.fasta\n",
      "\t/tmp/tmpTd2QtC.fasta\n",
      "\t/tmp/tmpIuSUGN.fasta\n",
      "\t/tmp/tmpALiS38.fasta\n",
      "\t/tmp/tmpXFygqv.fasta\n",
      "\t/tmp/tmpAt7qNF.fasta\n",
      "\t/tmp/tmplqkmZM.fasta\n",
      "\t/tmp/tmp01WsNt.fasta\n",
      "\t/tmp/tmpPcPDeB.fasta\n",
      "\t/tmp/tmpREhYZT.fasta\n",
      "\t/tmp/tmpDcM0P1.fasta\n",
      "\t/tmp/tmpByiKte.fasta\n",
      "\t/tmp/tmpUuKiEk.fasta\n",
      "\t/tmp/tmphEP0eY.fasta\n",
      "\t/tmp/tmpYWhEEt.fasta\n",
      "\t/tmp/tmpx9rAE1.fasta\n",
      "\t/tmp/tmp2akLwZ.fasta\n",
      "\t/tmp/tmpA7TgXN.fasta\n",
      "\t/tmp/tmpF029_J.fasta\n",
      "\t/tmp/tmpRgodXx.fasta\n",
      "\t/tmp/tmpQlBGCT.fasta\n",
      "\t/tmp/tmpU1qP0T.fasta\n",
      "\t/tmp/tmpa4eZnQ.fasta\n",
      "\t/tmp/tmpqOznJS.fasta\n",
      "\t/tmp/tmpoGxP8K.fasta\n",
      "\t/tmp/tmpmFPyy7.fasta\n",
      "\t/tmp/tmpWcqg2I.fasta\n",
      "\t/tmp/tmpqglMF6.fasta\n",
      "\t/tmp/tmp0cnhLN.fasta\n",
      "\t/tmp/tmpzeCRsZ.fasta\n",
      "\t/tmp/tmpkfhCWQ.fasta\n",
      "\t/tmp/tmpv6ChwZ.fasta\n",
      "\t/tmp/tmpE8wNr1.fasta\n",
      "\t/tmp/tmprasckp.fasta\n",
      "\t/tmp/tmpkL6LsH.fasta\n",
      "\t/tmp/tmpDEPoLN.fasta\n",
      "\t/tmp/tmpGqBZn1.fasta\n",
      "\t/tmp/tmpEBAJoK.fasta\n",
      "\t/tmp/tmpFSGPzP.fasta\n",
      "\t/tmp/tmpXc_7wl.fasta\n",
      "\t/tmp/tmp6Qyaj1.fasta\n",
      "\t/tmp/tmpjbKwDU.fasta\n",
      "\t/tmp/tmpyrsTlh.fasta\n",
      "\t/tmp/tmpQeEChD.fasta\n",
      "\t/tmp/tmpmiyw56.fasta\n",
      "\t/tmp/tmpqMa8k9.fasta\n",
      "\t/tmp/tmpjXm5i2.fasta\n",
      "\t/tmp/tmpVtbLBL.fasta\n",
      "\t/tmp/tmpnDEWPb.fasta\n",
      "\t/tmp/tmpguIc0s.fasta\n",
      "\t/tmp/tmp35O2Eq.fasta\n",
      "\t/tmp/tmpLlGRhE.fasta\n",
      "\t/tmp/tmp6EhmQ4.fasta\n",
      "\t/tmp/tmpdWJbrb.fasta\n",
      "\t/tmp/tmpE3w1cD.fasta\n",
      "\t/tmp/tmpwqMHAd.fasta\n",
      "\t/tmp/tmpI9vL0R.fasta\n",
      "\t/tmp/tmpP96md_.fasta\n",
      "\t/tmp/tmp5Sl_Zz.fasta\n",
      "\t/tmp/tmpBGotX7.fasta\n",
      "\t/tmp/tmp4vj0rA.fasta\n",
      "\t/tmp/tmpuIIc4q.fasta\n",
      "\t/tmp/tmp2zFqoU.fasta\n",
      "\t/tmp/tmpbYV6lw.fasta\n",
      "\t/tmp/tmpv9DU5h.fasta\n",
      "\t/tmp/tmpURwytF.fasta\n",
      "\t/tmp/tmpDb31dc.fasta\n",
      "\t/tmp/tmpTSPXIe.fasta\n",
      "\t/tmp/tmpeLNW6v.fasta\n",
      "\t/tmp/tmpRCkcgx.fasta\n",
      "\t/tmp/tmpHKcxEL.fasta\n",
      "\t/tmp/tmpTXtI8c.fasta\n",
      "\t/tmp/tmpcJtlLr.fasta\n",
      "\t/tmp/tmpT7q42c.fasta\n",
      "\t/tmp/tmpTocc0G.fasta\n",
      "\t/tmp/tmpXrmEAJ.fasta\n",
      "\t/tmp/tmpO9Xfc7.fasta\n",
      "\t/tmp/tmp7LIqyb.fasta\n",
      "\t/tmp/tmptKoVkc.fasta\n",
      "\t/tmp/tmpcgf7CH.fasta\n",
      "\t/tmp/tmpD08nF2.fasta\n",
      "\t/tmp/tmpfgRP5m.fasta\n",
      "\t/tmp/tmp2OBLGo.fasta\n",
      "\t/tmp/tmpHfto35.fasta\n",
      "\t/tmp/tmpkHnc2q.fasta\n",
      "\t/tmp/tmpgEGDhy.fasta\n",
      "\t/tmp/tmpBXgQ6Y.fasta\n",
      "\t/tmp/tmpNvhUUO.fasta\n",
      "\t/tmp/tmpVDp5mW.fasta\n",
      "\t/tmp/tmpbYbATp.fasta\n",
      "\t/tmp/tmp0gI5_0.fasta\n",
      "\t/tmp/tmp_uJTa6.fasta\n",
      "\t/tmp/tmp5offB2.fasta\n",
      "\t/tmp/tmpGKTPO8.fasta\n",
      "\t/tmp/tmpCLAnSE.fasta\n",
      "\t/tmp/tmpecZWoG.fasta\n",
      "\t/tmp/tmpR25lQ4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwuqjuF.lastz\n",
      "\t/tmp/tmpr_wji7.lastz\n",
      "\t/tmp/tmpA50hoF.lastz\n",
      "\t/tmp/tmpIxgLYT.lastz\n",
      "\t/tmp/tmpT5nOZX.lastz\n",
      "\t/tmp/tmprdMTGR.lastz\n",
      "\t/tmp/tmpsxitlk.lastz\n",
      "\t/tmp/tmp2EO0uT.lastz\n",
      "\t/tmp/tmpEXnhzC.lastz\n",
      "\t/tmp/tmpCCEsym.lastz\n",
      "\t/tmp/tmpKsjle9.lastz\n",
      "\t/tmp/tmpdpuqMq.lastz\n",
      "\t/tmp/tmpc0dJCE.lastz\n",
      "\t/tmp/tmpB77935.lastz\n",
      "\t/tmp/tmpJGlo5e.lastz\n",
      "\t/tmp/tmpTjaloG.lastz\n",
      "\t/tmp/tmpUvuxjK.lastz\n",
      "\t/tmp/tmpNhaTmQ.lastz\n",
      "\t/tmp/tmpT6cwhR.lastz\n",
      "\t/tmp/tmpt9XjYi.lastz\n",
      "\t/tmp/tmpBWQKIF.lastz\n",
      "\t/tmp/tmp5OHKWE.lastz\n",
      "\t/tmp/tmpyyODqY.lastz\n",
      "\t/tmp/tmpt5fFL6.lastz\n",
      "\t/tmp/tmpmUMtZS.lastz\n",
      "\t/tmp/tmpRegVxL.lastz\n",
      "\t/tmp/tmpkWUsAT.lastz\n",
      "\t/tmp/tmp9c4KTD.lastz\n",
      "\t/tmp/tmpQ3x1hr.lastz\n",
      "\t/tmp/tmpsvPgJw.lastz\n",
      "\t/tmp/tmp53tKn4.lastz\n",
      "\t/tmp/tmphd4j3x.lastz\n",
      "\t/tmp/tmp5uPgxo.lastz\n",
      "\t/tmp/tmpOE2J6W.lastz\n",
      "\t/tmp/tmpo1M90X.lastz\n",
      "\t/tmp/tmpQ21_Ui.lastz\n",
      "\t/tmp/tmpywPRYA.lastz\n",
      "\t/tmp/tmp5TA77N.lastz\n",
      "\t/tmp/tmpBgme1M.lastz\n",
      "\t/tmp/tmpTpHv3R.lastz\n",
      "\t/tmp/tmp_Ekigl.lastz\n",
      "\t/tmp/tmpLXh5Am.lastz\n",
      "\t/tmp/tmpxulDFN.lastz\n",
      "\t/tmp/tmpRIl4aN.lastz\n",
      "\t/tmp/tmp7F_bKi.lastz\n",
      "\t/tmp/tmpmYZajI.lastz\n",
      "\t/tmp/tmpW3g0bc.lastz\n",
      "\t/tmp/tmp3U9vWI.lastz\n",
      "\t/tmp/tmpbuIaL2.lastz\n",
      "\t/tmp/tmpusiQ5j.lastz\n",
      "\t/tmp/tmpXbpSCq.lastz\n",
      "\t/tmp/tmppDUEEb.lastz\n",
      "\t/tmp/tmpecsyHs.lastz\n",
      "\t/tmp/tmpUeNsaO.lastz\n",
      "\t/tmp/tmpDpPcSs.lastz\n",
      "\t/tmp/tmpUuyaLq.lastz\n",
      "\t/tmp/tmpODKNbZ.lastz\n",
      "\t/tmp/tmpClKwTL.lastz\n",
      "\t/tmp/tmpSimmpX.lastz\n",
      "\t/tmp/tmpRISlY3.lastz\n",
      "\t/tmp/tmpMGxSv0.lastz\n",
      "\t/tmp/tmpJfVwf2.lastz\n",
      "\t/tmp/tmps1S6z1.lastz\n",
      "\t/tmp/tmpxY94te.lastz\n",
      "\t/tmp/tmpqDKzZP.lastz\n",
      "\t/tmp/tmpWxBvX7.lastz\n",
      "\t/tmp/tmpL8fRQm.lastz\n",
      "\t/tmp/tmpdKbKZF.lastz\n",
      "\t/tmp/tmpvzaBYO.lastz\n",
      "\t/tmp/tmpPmEY7b.lastz\n",
      "\t/tmp/tmpfS6L8D.lastz\n",
      "\t/tmp/tmplFZNcc.lastz\n",
      "\t/tmp/tmpy9_3Iz.lastz\n",
      "\t/tmp/tmpTjRudy.lastz\n",
      "\t/tmp/tmpih8kE9.lastz\n",
      "\t/tmp/tmpJfofaU.lastz\n",
      "\t/tmp/tmp96_Z7c.lastz\n",
      "\t/tmp/tmpsI9cD_.lastz\n",
      "\t/tmp/tmpjNFU4s.lastz\n",
      "\t/tmp/tmpthqkKi.lastz\n",
      "\t/tmp/tmpuh0DlE.lastz\n",
      "\t/tmp/tmpOYOISs.lastz\n",
      "\t/tmp/tmpxug7mc.lastz\n",
      "\t/tmp/tmpTb8pPZ.lastz\n",
      "\t/tmp/tmpupshJ_.lastz\n",
      "\t/tmp/tmpolI4yY.lastz\n",
      "\t/tmp/tmpQeZDBs.lastz\n",
      "\t/tmp/tmpfqS17U.lastz\n",
      "\t/tmp/tmpZHntob.lastz\n",
      "\t/tmp/tmps8BwPL.lastz\n",
      "\t/tmp/tmpg81YNJ.lastz\n",
      "\t/tmp/tmphq23vd.lastz\n",
      "\t/tmp/tmpWtX6dW.lastz\n",
      "\t/tmp/tmpIqwbAB.lastz\n",
      "\t/tmp/tmp15foAl.lastz\n",
      "\t/tmp/tmpMap9X0.lastz\n",
      "\t/tmp/tmpyIUvvI.lastz\n",
      "\t/tmp/tmpsx14kZ.lastz\n",
      "\t/tmp/tmpkgksfy.lastz\n",
      "\t/tmp/tmpnWjz9Q.lastz\n",
      "\t/tmp/tmpqg2B_u.lastz\n",
      "\t/tmp/tmpasKERn.lastz\n",
      "\t/tmp/tmp17cwt8.lastz\n",
      "\t/tmp/tmpATYC5Y.lastz\n",
      "\t/tmp/tmpkeBd8W.lastz\n",
      "\t/tmp/tmplegstX.lastz\n",
      "\t/tmp/tmpB4U2MX.lastz\n",
      "\t/tmp/tmpTu5gwL.lastz\n",
      "\t/tmp/tmpDcMAuY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Fasciola_hepatica_GCA_900302435.lastz\n",
      "Creating Fasciola_hepatica_GCA_900302435 table\n",
      "Inserting data to Fasciola_hepatica_GCA_900302435 table\n",
      "\n",
      "Running against Fasciolopsis_buski_GCA_008360955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 75 queries...\n",
      "\t/tmp/tmp2GHtlO.fasta\n",
      "\t/tmp/tmp1qvian.fasta\n",
      "\t/tmp/tmp6ZR0kU.fasta\n",
      "\t/tmp/tmpInyWD8.fasta\n",
      "\t/tmp/tmpljlqnn.fasta\n",
      "\t/tmp/tmpBRj5Y4.fasta\n",
      "\t/tmp/tmpHXHDRt.fasta\n",
      "\t/tmp/tmpT3WBgU.fasta\n",
      "\t/tmp/tmpYKEsPA.fasta\n",
      "\t/tmp/tmpT_uBFd.fasta\n",
      "\t/tmp/tmpuetnyO.fasta\n",
      "\t/tmp/tmp9oeEre.fasta\n",
      "\t/tmp/tmpsdEj7O.fasta\n",
      "\t/tmp/tmpDM5Zzy.fasta\n",
      "\t/tmp/tmpHuSI2l.fasta\n",
      "\t/tmp/tmp6gO8P3.fasta\n",
      "\t/tmp/tmpqUgHq7.fasta\n",
      "\t/tmp/tmp2E8u9Z.fasta\n",
      "\t/tmp/tmpePKZTP.fasta\n",
      "\t/tmp/tmpOqEDuD.fasta\n",
      "\t/tmp/tmpCXPsPh.fasta\n",
      "\t/tmp/tmpJeg1zK.fasta\n",
      "\t/tmp/tmpMpVB7C.fasta\n",
      "\t/tmp/tmpFOWCCF.fasta\n",
      "\t/tmp/tmp74IkUx.fasta\n",
      "\t/tmp/tmpyFiUxi.fasta\n",
      "\t/tmp/tmps7e4hd.fasta\n",
      "\t/tmp/tmpPVNyGR.fasta\n",
      "\t/tmp/tmpZ72apk.fasta\n",
      "\t/tmp/tmpmnCF01.fasta\n",
      "\t/tmp/tmpUeWnBz.fasta\n",
      "\t/tmp/tmpK8_KJE.fasta\n",
      "\t/tmp/tmp3SrljH.fasta\n",
      "\t/tmp/tmpFKQEA5.fasta\n",
      "\t/tmp/tmpa_eHcr.fasta\n",
      "\t/tmp/tmpqjb7jY.fasta\n",
      "\t/tmp/tmpcFzJyx.fasta\n",
      "\t/tmp/tmp3AsV_s.fasta\n",
      "\t/tmp/tmpwNFf0W.fasta\n",
      "\t/tmp/tmpW8gshQ.fasta\n",
      "\t/tmp/tmpUY0jXP.fasta\n",
      "\t/tmp/tmp2KB5cl.fasta\n",
      "\t/tmp/tmpnnPe62.fasta\n",
      "\t/tmp/tmpDOUrC4.fasta\n",
      "\t/tmp/tmpzwVEIv.fasta\n",
      "\t/tmp/tmpyFxnB7.fasta\n",
      "\t/tmp/tmp0wfy6t.fasta\n",
      "\t/tmp/tmpPnGYkT.fasta\n",
      "\t/tmp/tmpBE2mP4.fasta\n",
      "\t/tmp/tmpB8i96j.fasta\n",
      "\t/tmp/tmpmob2XP.fasta\n",
      "\t/tmp/tmpnQOzZt.fasta\n",
      "\t/tmp/tmpTGOyHl.fasta\n",
      "\t/tmp/tmpZXUysi.fasta\n",
      "\t/tmp/tmpXSxMn8.fasta\n",
      "\t/tmp/tmpXPbUSp.fasta\n",
      "\t/tmp/tmprJ0mpu.fasta\n",
      "\t/tmp/tmpqsBDNG.fasta\n",
      "\t/tmp/tmpTeyyc3.fasta\n",
      "\t/tmp/tmpVUjs_U.fasta\n",
      "\t/tmp/tmpn0bY6k.fasta\n",
      "\t/tmp/tmpKwP6C2.fasta\n",
      "\t/tmp/tmpz84jId.fasta\n",
      "\t/tmp/tmpFm6K8q.fasta\n",
      "\t/tmp/tmpRrs1Ps.fasta\n",
      "\t/tmp/tmp1Oybm7.fasta\n",
      "\t/tmp/tmpQU3U0w.fasta\n",
      "\t/tmp/tmpatS7gG.fasta\n",
      "\t/tmp/tmpchXq7C.fasta\n",
      "\t/tmp/tmpKEUtj6.fasta\n",
      "\t/tmp/tmpvY7tZy.fasta\n",
      "\t/tmp/tmpFwPWpt.fasta\n",
      "\t/tmp/tmpUILFco.fasta\n",
      "\t/tmp/tmpST3fFa.fasta\n",
      "\t/tmp/tmpdo8nah.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8boLws.lastz\n",
      "\t/tmp/tmpCu3VH1.lastz\n",
      "\t/tmp/tmpME4nK5.lastz\n",
      "\t/tmp/tmpfWRoVR.lastz\n",
      "\t/tmp/tmpw7fiFo.lastz\n",
      "\t/tmp/tmpS_VV9Y.lastz\n",
      "\t/tmp/tmp4iI4aw.lastz\n",
      "\t/tmp/tmptlqrd0.lastz\n",
      "\t/tmp/tmpLxaHWi.lastz\n",
      "\t/tmp/tmpzeelC2.lastz\n",
      "\t/tmp/tmpDuxDph.lastz\n",
      "\t/tmp/tmpwMnZMt.lastz\n",
      "\t/tmp/tmpkCoOQt.lastz\n",
      "\t/tmp/tmptYuPLv.lastz\n",
      "\t/tmp/tmp0LL9zq.lastz\n",
      "\t/tmp/tmp4VE6v3.lastz\n",
      "\t/tmp/tmp7yq1n0.lastz\n",
      "\t/tmp/tmpFYezav.lastz\n",
      "\t/tmp/tmpBnhJym.lastz\n",
      "\t/tmp/tmpvidmmg.lastz\n",
      "\t/tmp/tmpddg2XC.lastz\n",
      "\t/tmp/tmpVRcNS2.lastz\n",
      "\t/tmp/tmpbKdoKg.lastz\n",
      "\t/tmp/tmpXSiy_M.lastz\n",
      "\t/tmp/tmpoyN_Xs.lastz\n",
      "\t/tmp/tmpFDx4MO.lastz\n",
      "\t/tmp/tmps4L7uN.lastz\n",
      "\t/tmp/tmpay5vqr.lastz\n",
      "\t/tmp/tmpi47gAI.lastz\n",
      "\t/tmp/tmpcD0ihw.lastz\n",
      "\t/tmp/tmp5S_Q4Z.lastz\n",
      "\t/tmp/tmpI64uag.lastz\n",
      "\t/tmp/tmpXbFOOo.lastz\n",
      "\t/tmp/tmprLRLqT.lastz\n",
      "\t/tmp/tmpTgMVEr.lastz\n",
      "\t/tmp/tmp9kO1RU.lastz\n",
      "\t/tmp/tmpLkYSb6.lastz\n",
      "\t/tmp/tmpo0jDs6.lastz\n",
      "\t/tmp/tmp2kbpkS.lastz\n",
      "\t/tmp/tmp2tfTq5.lastz\n",
      "\t/tmp/tmpLLQAAx.lastz\n",
      "\t/tmp/tmpoC6Qdd.lastz\n",
      "\t/tmp/tmpKVlSiY.lastz\n",
      "\t/tmp/tmp2YY9eQ.lastz\n",
      "\t/tmp/tmp__PYYy.lastz\n",
      "\t/tmp/tmpRrHzLh.lastz\n",
      "\t/tmp/tmpJthzBO.lastz\n",
      "\t/tmp/tmp0gXWJR.lastz\n",
      "\t/tmp/tmpFBuM1w.lastz\n",
      "\t/tmp/tmpSe3ktC.lastz\n",
      "\t/tmp/tmpI0C_ti.lastz\n",
      "\t/tmp/tmpNmaidS.lastz\n",
      "\t/tmp/tmpPEn_nL.lastz\n",
      "\t/tmp/tmpGQfVA7.lastz\n",
      "\t/tmp/tmpzc8u1D.lastz\n",
      "\t/tmp/tmp2GRcn8.lastz\n",
      "\t/tmp/tmpa8al94.lastz\n",
      "\t/tmp/tmpwKbDI1.lastz\n",
      "\t/tmp/tmpcHfQAh.lastz\n",
      "\t/tmp/tmpLynzxY.lastz\n",
      "\t/tmp/tmpW_6LkC.lastz\n",
      "\t/tmp/tmppHun4X.lastz\n",
      "\t/tmp/tmpkA7Pan.lastz\n",
      "\t/tmp/tmp9t30QI.lastz\n",
      "\t/tmp/tmp3n1n0o.lastz\n",
      "\t/tmp/tmpo_sGxf.lastz\n",
      "\t/tmp/tmpWZciVf.lastz\n",
      "\t/tmp/tmp0YNZ9m.lastz\n",
      "\t/tmp/tmpOnA2MQ.lastz\n",
      "\t/tmp/tmpUQVdKc.lastz\n",
      "\t/tmp/tmph2Mj1S.lastz\n",
      "\t/tmp/tmp6hU2fu.lastz\n",
      "\t/tmp/tmpQpDdAs.lastz\n",
      "\t/tmp/tmpNLFZQ1.lastz\n",
      "\t/tmp/tmprc_6Et.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Fasciolopsis_buski_GCA_008360955.lastz\n",
      "Creating Fasciolopsis_buski_GCA_008360955 table\n",
      "Inserting data to Fasciolopsis_buski_GCA_008360955 table\n",
      "\n",
      "Running against Opisthorchis_felineus_GCA_004794785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 66 queries...\n",
      "\t/tmp/tmp9qtCIs.fasta\n",
      "\t/tmp/tmptJ683S.fasta\n",
      "\t/tmp/tmpgrQGc_.fasta\n",
      "\t/tmp/tmpajFItH.fasta\n",
      "\t/tmp/tmpKWMhrL.fasta\n",
      "\t/tmp/tmp5VvR57.fasta\n",
      "\t/tmp/tmpcxsbzj.fasta\n",
      "\t/tmp/tmpSSjxFt.fasta\n",
      "\t/tmp/tmpD8Q3Rs.fasta\n",
      "\t/tmp/tmpevU8bg.fasta\n",
      "\t/tmp/tmpnghmof.fasta\n",
      "\t/tmp/tmpbEqyXV.fasta\n",
      "\t/tmp/tmpQcmkw5.fasta\n",
      "\t/tmp/tmpRITyco.fasta\n",
      "\t/tmp/tmpDJ6how.fasta\n",
      "\t/tmp/tmpiFpa1v.fasta\n",
      "\t/tmp/tmpQgWaKj.fasta\n",
      "\t/tmp/tmpFjDEaS.fasta\n",
      "\t/tmp/tmpJ09B5K.fasta\n",
      "\t/tmp/tmpKG_QLO.fasta\n",
      "\t/tmp/tmps7yh82.fasta\n",
      "\t/tmp/tmpj7Nbge.fasta\n",
      "\t/tmp/tmpdlWnjK.fasta\n",
      "\t/tmp/tmpL4RQ_m.fasta\n",
      "\t/tmp/tmplw2t_p.fasta\n",
      "\t/tmp/tmp0tnvGI.fasta\n",
      "\t/tmp/tmpn_yA3S.fasta\n",
      "\t/tmp/tmp7ZKDnQ.fasta\n",
      "\t/tmp/tmpEoUjL6.fasta\n",
      "\t/tmp/tmpTN79HT.fasta\n",
      "\t/tmp/tmpxjCTgq.fasta\n",
      "\t/tmp/tmp22FRdJ.fasta\n",
      "\t/tmp/tmpeG9Ui8.fasta\n",
      "\t/tmp/tmpe_zzmb.fasta\n",
      "\t/tmp/tmpyFh54_.fasta\n",
      "\t/tmp/tmpO6Zgav.fasta\n",
      "\t/tmp/tmpILoWj1.fasta\n",
      "\t/tmp/tmp3Qdo6U.fasta\n",
      "\t/tmp/tmpT5yVP8.fasta\n",
      "\t/tmp/tmpwMmQDB.fasta\n",
      "\t/tmp/tmpY2NZLD.fasta\n",
      "\t/tmp/tmpZecGGH.fasta\n",
      "\t/tmp/tmpvjNLzo.fasta\n",
      "\t/tmp/tmp_BOEnh.fasta\n",
      "\t/tmp/tmpiKqxgv.fasta\n",
      "\t/tmp/tmphhALl0.fasta\n",
      "\t/tmp/tmp6byuSb.fasta\n",
      "\t/tmp/tmpHzqX6J.fasta\n",
      "\t/tmp/tmpKYdX5Q.fasta\n",
      "\t/tmp/tmpPtrv0c.fasta\n",
      "\t/tmp/tmp4sLwCS.fasta\n",
      "\t/tmp/tmpMC2jZJ.fasta\n",
      "\t/tmp/tmpbURidI.fasta\n",
      "\t/tmp/tmpQ8rhYp.fasta\n",
      "\t/tmp/tmpBVwMAN.fasta\n",
      "\t/tmp/tmpraN3bn.fasta\n",
      "\t/tmp/tmplzI3NJ.fasta\n",
      "\t/tmp/tmph993Uv.fasta\n",
      "\t/tmp/tmpXCjEYO.fasta\n",
      "\t/tmp/tmp33QPA6.fasta\n",
      "\t/tmp/tmp0G7bWV.fasta\n",
      "\t/tmp/tmpfC0TTo.fasta\n",
      "\t/tmp/tmp9lQHqh.fasta\n",
      "\t/tmp/tmpz6hAk6.fasta\n",
      "\t/tmp/tmp3xaH3H.fasta\n",
      "\t/tmp/tmpWTfW70.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCeNovB.lastz\n",
      "\t/tmp/tmpEf1HMQ.lastz\n",
      "\t/tmp/tmph_0CKO.lastz\n",
      "\t/tmp/tmp_iP4Oq.lastz\n",
      "\t/tmp/tmpFUmLUg.lastz\n",
      "\t/tmp/tmpU1WGhu.lastz\n",
      "\t/tmp/tmpZb5GD_.lastz\n",
      "\t/tmp/tmpZFqEe3.lastz\n",
      "\t/tmp/tmpRAuVmY.lastz\n",
      "\t/tmp/tmpDWifUX.lastz\n",
      "\t/tmp/tmpVFYNER.lastz\n",
      "\t/tmp/tmpoRecbt.lastz\n",
      "\t/tmp/tmpajhD4x.lastz\n",
      "\t/tmp/tmpK_pCzp.lastz\n",
      "\t/tmp/tmpxdOTJO.lastz\n",
      "\t/tmp/tmpVwbbJv.lastz\n",
      "\t/tmp/tmpGfcWdn.lastz\n",
      "\t/tmp/tmpg569v6.lastz\n",
      "\t/tmp/tmpkXKUR_.lastz\n",
      "\t/tmp/tmpJj9QkM.lastz\n",
      "\t/tmp/tmpIySGvA.lastz\n",
      "\t/tmp/tmprGOCpA.lastz\n",
      "\t/tmp/tmpRfokhS.lastz\n",
      "\t/tmp/tmpL6Bpuf.lastz\n",
      "\t/tmp/tmp3HNEFA.lastz\n",
      "\t/tmp/tmp7fOtWA.lastz\n",
      "\t/tmp/tmpXUbn_c.lastz\n",
      "\t/tmp/tmp5J0xpx.lastz\n",
      "\t/tmp/tmpTVjd46.lastz\n",
      "\t/tmp/tmpz3v9aa.lastz\n",
      "\t/tmp/tmprs2Giv.lastz\n",
      "\t/tmp/tmpjAnmUd.lastz\n",
      "\t/tmp/tmpnxHr3D.lastz\n",
      "\t/tmp/tmp3s330i.lastz\n",
      "\t/tmp/tmpwwBrzM.lastz\n",
      "\t/tmp/tmp7Z9Wwr.lastz\n",
      "\t/tmp/tmpwJtpG3.lastz\n",
      "\t/tmp/tmpb7HboB.lastz\n",
      "\t/tmp/tmpywMtgr.lastz\n",
      "\t/tmp/tmpSL3ClB.lastz\n",
      "\t/tmp/tmpBfk3a1.lastz\n",
      "\t/tmp/tmpquPrTU.lastz\n",
      "\t/tmp/tmp5Mu9ap.lastz\n",
      "\t/tmp/tmpHWnNxO.lastz\n",
      "\t/tmp/tmpitfy8g.lastz\n",
      "\t/tmp/tmpNIRBsl.lastz\n",
      "\t/tmp/tmpnr53Rl.lastz\n",
      "\t/tmp/tmpbg7Q3V.lastz\n",
      "\t/tmp/tmpry8LPI.lastz\n",
      "\t/tmp/tmpODE0z1.lastz\n",
      "\t/tmp/tmpA_6ef1.lastz\n",
      "\t/tmp/tmpc_9v9Z.lastz\n",
      "\t/tmp/tmpEuhQTb.lastz\n",
      "\t/tmp/tmpCi2go5.lastz\n",
      "\t/tmp/tmppuw1MS.lastz\n",
      "\t/tmp/tmpVcqfFw.lastz\n",
      "\t/tmp/tmplPtb9r.lastz\n",
      "\t/tmp/tmplxoG4w.lastz\n",
      "\t/tmp/tmpSyXJqV.lastz\n",
      "\t/tmp/tmpgfM9OI.lastz\n",
      "\t/tmp/tmpeclTbL.lastz\n",
      "\t/tmp/tmpMUD1zU.lastz\n",
      "\t/tmp/tmp1pvxnW.lastz\n",
      "\t/tmp/tmph49Hd2.lastz\n",
      "\t/tmp/tmppqNsU1.lastz\n",
      "\t/tmp/tmpgALhov.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Opisthorchis_felineus_GCA_004794785.lastz\n",
      "Creating Opisthorchis_felineus_GCA_004794785 table\n",
      "Inserting data to Opisthorchis_felineus_GCA_004794785 table\n",
      "\n",
      "Running against Opisthorchis_viverrini_GCF_000715545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 58 queries...\n",
      "\t/tmp/tmpspaNet.fasta\n",
      "\t/tmp/tmp_LsOj_.fasta\n",
      "\t/tmp/tmp_rvPt3.fasta\n",
      "\t/tmp/tmpg6ZATY.fasta\n",
      "\t/tmp/tmpPWYZ54.fasta\n",
      "\t/tmp/tmp1JVr3q.fasta\n",
      "\t/tmp/tmps1K1_N.fasta\n",
      "\t/tmp/tmpU7bk9U.fasta\n",
      "\t/tmp/tmpE6xx2S.fasta\n",
      "\t/tmp/tmpZ6yw_J.fasta\n",
      "\t/tmp/tmpnarLgh.fasta\n",
      "\t/tmp/tmpkLQUoL.fasta\n",
      "\t/tmp/tmp0ipoiQ.fasta\n",
      "\t/tmp/tmpLe104l.fasta\n",
      "\t/tmp/tmpodnZ7b.fasta\n",
      "\t/tmp/tmp4WD57_.fasta\n",
      "\t/tmp/tmp8I0JXu.fasta\n",
      "\t/tmp/tmpfJQyzq.fasta\n",
      "\t/tmp/tmpRMaBhp.fasta\n",
      "\t/tmp/tmp60rVRl.fasta\n",
      "\t/tmp/tmpLdzf_A.fasta\n",
      "\t/tmp/tmpoHaEzw.fasta\n",
      "\t/tmp/tmpwN8JxL.fasta\n",
      "\t/tmp/tmpsggG7f.fasta\n",
      "\t/tmp/tmprMM8ED.fasta\n",
      "\t/tmp/tmp81UyE1.fasta\n",
      "\t/tmp/tmp5oIkYL.fasta\n",
      "\t/tmp/tmp5p3Gr1.fasta\n",
      "\t/tmp/tmp1GpWMF.fasta\n",
      "\t/tmp/tmpFw_RYD.fasta\n",
      "\t/tmp/tmpstGet6.fasta\n",
      "\t/tmp/tmp_Nh6n8.fasta\n",
      "\t/tmp/tmpNhfgom.fasta\n",
      "\t/tmp/tmp92lejx.fasta\n",
      "\t/tmp/tmp1UCvuA.fasta\n",
      "\t/tmp/tmpF2hWwm.fasta\n",
      "\t/tmp/tmpXfyC5w.fasta\n",
      "\t/tmp/tmpRwc60H.fasta\n",
      "\t/tmp/tmp3cSQ2d.fasta\n",
      "\t/tmp/tmpjl97nc.fasta\n",
      "\t/tmp/tmpJSqjzQ.fasta\n",
      "\t/tmp/tmpRcPkeg.fasta\n",
      "\t/tmp/tmpY2cB9z.fasta\n",
      "\t/tmp/tmpbblfiX.fasta\n",
      "\t/tmp/tmpoXjZ9t.fasta\n",
      "\t/tmp/tmpWUTi0E.fasta\n",
      "\t/tmp/tmpTx46wT.fasta\n",
      "\t/tmp/tmpXxSKEg.fasta\n",
      "\t/tmp/tmp1s_WRi.fasta\n",
      "\t/tmp/tmpI_dMWx.fasta\n",
      "\t/tmp/tmp5JfiZ4.fasta\n",
      "\t/tmp/tmplo9mB7.fasta\n",
      "\t/tmp/tmp4RrmYE.fasta\n",
      "\t/tmp/tmppOhNdm.fasta\n",
      "\t/tmp/tmptNaa98.fasta\n",
      "\t/tmp/tmpEs9hrx.fasta\n",
      "\t/tmp/tmpPTc3Xd.fasta\n",
      "\t/tmp/tmpPgT8Ye.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpR6C9F1.lastz\n",
      "\t/tmp/tmpwNznIB.lastz\n",
      "\t/tmp/tmpWjmt7L.lastz\n",
      "\t/tmp/tmpBUf_7f.lastz\n",
      "\t/tmp/tmpDwbK2T.lastz\n",
      "\t/tmp/tmpZCPtLa.lastz\n",
      "\t/tmp/tmpgH6foF.lastz\n",
      "\t/tmp/tmpv4n53M.lastz\n",
      "\t/tmp/tmp3HhX5d.lastz\n",
      "\t/tmp/tmpemUvMg.lastz\n",
      "\t/tmp/tmpTeRC8r.lastz\n",
      "\t/tmp/tmpW7IYDj.lastz\n",
      "\t/tmp/tmpA3X1Px.lastz\n",
      "\t/tmp/tmpPzpdMq.lastz\n",
      "\t/tmp/tmph4HvFi.lastz\n",
      "\t/tmp/tmpVhsGWb.lastz\n",
      "\t/tmp/tmpBHaI8_.lastz\n",
      "\t/tmp/tmpHnPZpg.lastz\n",
      "\t/tmp/tmp6MrNks.lastz\n",
      "\t/tmp/tmp6o5NTj.lastz\n",
      "\t/tmp/tmpi6iZA_.lastz\n",
      "\t/tmp/tmpqAkQQm.lastz\n",
      "\t/tmp/tmp45bGbQ.lastz\n",
      "\t/tmp/tmpoILWZf.lastz\n",
      "\t/tmp/tmpzZCoSi.lastz\n",
      "\t/tmp/tmpSqfgZr.lastz\n",
      "\t/tmp/tmpN8PAqZ.lastz\n",
      "\t/tmp/tmpr549ly.lastz\n",
      "\t/tmp/tmpYVhQaY.lastz\n",
      "\t/tmp/tmpTIHHeN.lastz\n",
      "\t/tmp/tmpCK81ri.lastz\n",
      "\t/tmp/tmpU9JtN0.lastz\n",
      "\t/tmp/tmpaw_nVf.lastz\n",
      "\t/tmp/tmpZJ81bI.lastz\n",
      "\t/tmp/tmpy84vWp.lastz\n",
      "\t/tmp/tmpviQdHl.lastz\n",
      "\t/tmp/tmpIXWA6Z.lastz\n",
      "\t/tmp/tmpUojrER.lastz\n",
      "\t/tmp/tmpBHVC1I.lastz\n",
      "\t/tmp/tmp6M0dTK.lastz\n",
      "\t/tmp/tmpmR6cFS.lastz\n",
      "\t/tmp/tmpNGfhtd.lastz\n",
      "\t/tmp/tmpbWUKlG.lastz\n",
      "\t/tmp/tmp6u1icz.lastz\n",
      "\t/tmp/tmpiyAt5y.lastz\n",
      "\t/tmp/tmpE6zFIz.lastz\n",
      "\t/tmp/tmpitUDC6.lastz\n",
      "\t/tmp/tmpTRHzZK.lastz\n",
      "\t/tmp/tmpzriEMz.lastz\n",
      "\t/tmp/tmp_7OXew.lastz\n",
      "\t/tmp/tmpUGL747.lastz\n",
      "\t/tmp/tmpuub35L.lastz\n",
      "\t/tmp/tmp8vDIv8.lastz\n",
      "\t/tmp/tmpZlFEDD.lastz\n",
      "\t/tmp/tmpNd8JKX.lastz\n",
      "\t/tmp/tmpk6kLfT.lastz\n",
      "\t/tmp/tmpNoH5xz.lastz\n",
      "\t/tmp/tmpA9FJwz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Opisthorchis_viverrini_GCF_000715545.lastz\n",
      "Creating Opisthorchis_viverrini_GCF_000715545 table\n",
      "Inserting data to Opisthorchis_viverrini_GCF_000715545 table\n",
      "\n",
      "Running against Paragonimus_westermani_GCA_008508345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 92 queries...\n",
      "\t/tmp/tmp9TTls_.fasta\n",
      "\t/tmp/tmpU2Q81B.fasta\n",
      "\t/tmp/tmph75b0q.fasta\n",
      "\t/tmp/tmpD5ZBT6.fasta\n",
      "\t/tmp/tmpWVK429.fasta\n",
      "\t/tmp/tmpNYh1o8.fasta\n",
      "\t/tmp/tmpPFj0CQ.fasta\n",
      "\t/tmp/tmpdhD7jT.fasta\n",
      "\t/tmp/tmp0DdFjv.fasta\n",
      "\t/tmp/tmpJXGe5L.fasta\n",
      "\t/tmp/tmpIJH59A.fasta\n",
      "\t/tmp/tmpZ3j1iI.fasta\n",
      "\t/tmp/tmpZyGMH_.fasta\n",
      "\t/tmp/tmp0JR1cK.fasta\n",
      "\t/tmp/tmpw1xlXs.fasta\n",
      "\t/tmp/tmpN6zjXJ.fasta\n",
      "\t/tmp/tmpYsfPfC.fasta\n",
      "\t/tmp/tmpu6HWyh.fasta\n",
      "\t/tmp/tmpKOGXam.fasta\n",
      "\t/tmp/tmpYStt0u.fasta\n",
      "\t/tmp/tmpoSXXfP.fasta\n",
      "\t/tmp/tmpdWFkM9.fasta\n",
      "\t/tmp/tmp191ng6.fasta\n",
      "\t/tmp/tmp5MOryf.fasta\n",
      "\t/tmp/tmptrVDTf.fasta\n",
      "\t/tmp/tmpeATGhb.fasta\n",
      "\t/tmp/tmpPhHy0Y.fasta\n",
      "\t/tmp/tmpGfnUR3.fasta\n",
      "\t/tmp/tmpJEVQP0.fasta\n",
      "\t/tmp/tmpDYAG2R.fasta\n",
      "\t/tmp/tmpcptyoA.fasta\n",
      "\t/tmp/tmpeYpoHf.fasta\n",
      "\t/tmp/tmp_Ed0VI.fasta\n",
      "\t/tmp/tmpdO1eOd.fasta\n",
      "\t/tmp/tmp1HWHl3.fasta\n",
      "\t/tmp/tmpUE3ABS.fasta\n",
      "\t/tmp/tmp2slVVM.fasta\n",
      "\t/tmp/tmptWAu2E.fasta\n",
      "\t/tmp/tmpCGy_10.fasta\n",
      "\t/tmp/tmpdIqzSH.fasta\n",
      "\t/tmp/tmpSRkwZ2.fasta\n",
      "\t/tmp/tmpgvX3Vs.fasta\n",
      "\t/tmp/tmpg9ittl.fasta\n",
      "\t/tmp/tmpRB5OBe.fasta\n",
      "\t/tmp/tmpgUA5Pe.fasta\n",
      "\t/tmp/tmpt_7Pot.fasta\n",
      "\t/tmp/tmprqdh_f.fasta\n",
      "\t/tmp/tmpKS_4N8.fasta\n",
      "\t/tmp/tmpei0NjA.fasta\n",
      "\t/tmp/tmpOTtIjW.fasta\n",
      "\t/tmp/tmp2WdDOx.fasta\n",
      "\t/tmp/tmpH8ymFS.fasta\n",
      "\t/tmp/tmpU3QJnD.fasta\n",
      "\t/tmp/tmpWoXQDB.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpKWM4EM.fasta\n",
      "\t/tmp/tmp29XTlF.fasta\n",
      "\t/tmp/tmp89qasT.fasta\n",
      "\t/tmp/tmpuUxLzL.fasta\n",
      "\t/tmp/tmp45VrIg.fasta\n",
      "\t/tmp/tmp8zqKVh.fasta\n",
      "\t/tmp/tmpYFfGbR.fasta\n",
      "\t/tmp/tmpVghc5I.fasta\n",
      "\t/tmp/tmp5I72x8.fasta\n",
      "\t/tmp/tmpWIwRqu.fasta\n",
      "\t/tmp/tmpucNX5a.fasta\n",
      "\t/tmp/tmp5eOATR.fasta\n",
      "\t/tmp/tmpJBo2zG.fasta\n",
      "\t/tmp/tmpxTqgZN.fasta\n",
      "\t/tmp/tmpwMqbUT.fasta\n",
      "\t/tmp/tmpk2icEr.fasta\n",
      "\t/tmp/tmpwAW3wW.fasta\n",
      "\t/tmp/tmp2Gi_k6.fasta\n",
      "\t/tmp/tmphu1fx_.fasta\n",
      "\t/tmp/tmpj9aH14.fasta\n",
      "\t/tmp/tmpSy0OF9.fasta\n",
      "\t/tmp/tmp3BNdBi.fasta\n",
      "\t/tmp/tmp9YwPmT.fasta\n",
      "\t/tmp/tmpoG8TqP.fasta\n",
      "\t/tmp/tmpriugKG.fasta\n",
      "\t/tmp/tmplyJqHv.fasta\n",
      "\t/tmp/tmpJdpnVX.fasta\n",
      "\t/tmp/tmpMaEJcQ.fasta\n",
      "\t/tmp/tmp3VpdIY.fasta\n",
      "\t/tmp/tmp8rXwY0.fasta\n",
      "\t/tmp/tmpaq4Bt7.fasta\n",
      "\t/tmp/tmpQgU4WE.fasta\n",
      "\t/tmp/tmpJ1vUaz.fasta\n",
      "\t/tmp/tmpcol2ZG.fasta\n",
      "\t/tmp/tmpkSJGSS.fasta\n",
      "\t/tmp/tmpYSIUHY.fasta\n",
      "\t/tmp/tmpfnO36o.fasta\n",
      "\t/tmp/tmpIGz6MB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNySt7P.lastz\n",
      "\t/tmp/tmp0Fh0b0.lastz\n",
      "\t/tmp/tmpukSfFk.lastz\n",
      "\t/tmp/tmpNrI9Z9.lastz\n",
      "\t/tmp/tmpt3Z7hz.lastz\n",
      "\t/tmp/tmpWfjCgK.lastz\n",
      "\t/tmp/tmpvrIPXM.lastz\n",
      "\t/tmp/tmp77IufB.lastz\n",
      "\t/tmp/tmpnQPnDn.lastz\n",
      "\t/tmp/tmpVOzhUX.lastz\n",
      "\t/tmp/tmpUVi2SA.lastz\n",
      "\t/tmp/tmpEeCYrQ.lastz\n",
      "\t/tmp/tmpWviqtw.lastz\n",
      "\t/tmp/tmpaJaNko.lastz\n",
      "\t/tmp/tmpOkiO5d.lastz\n",
      "\t/tmp/tmpFBj7rq.lastz\n",
      "\t/tmp/tmpfuLIQt.lastz\n",
      "\t/tmp/tmpmX03MF.lastz\n",
      "\t/tmp/tmpp32kwd.lastz\n",
      "\t/tmp/tmp43nv7q.lastz\n",
      "\t/tmp/tmpdUyl6v.lastz\n",
      "\t/tmp/tmpcb_1zA.lastz\n",
      "\t/tmp/tmpTVuda1.lastz\n",
      "\t/tmp/tmptqwAVn.lastz\n",
      "\t/tmp/tmpXVhtOI.lastz\n",
      "\t/tmp/tmpCvtmHO.lastz\n",
      "\t/tmp/tmpcZ0PQS.lastz\n",
      "\t/tmp/tmpv5n1nS.lastz\n",
      "\t/tmp/tmpmskXI2.lastz\n",
      "\t/tmp/tmpMj51eE.lastz\n",
      "\t/tmp/tmpUf_J8D.lastz\n",
      "\t/tmp/tmpa607GE.lastz\n",
      "\t/tmp/tmp5XiRNS.lastz\n",
      "\t/tmp/tmpaXtHwW.lastz\n",
      "\t/tmp/tmp1nqFGa.lastz\n",
      "\t/tmp/tmpIs6Hp0.lastz\n",
      "\t/tmp/tmpNbcd3p.lastz\n",
      "\t/tmp/tmpO7MLGa.lastz\n",
      "\t/tmp/tmpuhc4eT.lastz\n",
      "\t/tmp/tmpstlnO5.lastz\n",
      "\t/tmp/tmp9Wj2gn.lastz\n",
      "\t/tmp/tmpYCcmgu.lastz\n",
      "\t/tmp/tmpf8r2QK.lastz\n",
      "\t/tmp/tmpSCkWSZ.lastz\n",
      "\t/tmp/tmpUfHchF.lastz\n",
      "\t/tmp/tmpySkZ5c.lastz\n",
      "\t/tmp/tmpouYU1q.lastz\n",
      "\t/tmp/tmpQ0u8SK.lastz\n",
      "\t/tmp/tmprlmtfb.lastz\n",
      "\t/tmp/tmpC6zYSy.lastz\n",
      "\t/tmp/tmpsIOLLx.lastz\n",
      "\t/tmp/tmpXh7qMw.lastz\n",
      "\t/tmp/tmpXTKoXX.lastz\n",
      "\t/tmp/tmp89vUul.lastz\n",
      "\t/tmp/tmpa_z6Eb.lastz\n",
      "\t/tmp/tmpn_UB_a.lastz\n",
      "\t/tmp/tmpumescP.lastz\n",
      "\t/tmp/tmpm_6UZ2.lastz\n",
      "\t/tmp/tmp5CMkn8.lastz\n",
      "\t/tmp/tmph9aocV.lastz\n",
      "\t/tmp/tmpJNXCQV.lastz\n",
      "\t/tmp/tmpMkCa3u.lastz\n",
      "\t/tmp/tmpstfDMB.lastz\n",
      "\t/tmp/tmpeX9XH4.lastz\n",
      "\t/tmp/tmpLbkxko.lastz\n",
      "\t/tmp/tmpBX8Z6W.lastz\n",
      "\t/tmp/tmplMjwy4.lastz\n",
      "\t/tmp/tmpOjkwa3.lastz\n",
      "\t/tmp/tmpkPD2K6.lastz\n",
      "\t/tmp/tmpsMhLrb.lastz\n",
      "\t/tmp/tmppjVDXY.lastz\n",
      "\t/tmp/tmpPLEMvm.lastz\n",
      "\t/tmp/tmp_B75E8.lastz\n",
      "\t/tmp/tmp6yHFjO.lastz\n",
      "\t/tmp/tmpDCMePJ.lastz\n",
      "\t/tmp/tmpSg364u.lastz\n",
      "\t/tmp/tmpXgoq0T.lastz\n",
      "\t/tmp/tmpipjnIB.lastz\n",
      "\t/tmp/tmpVGyDG0.lastz\n",
      "\t/tmp/tmp2XcoPu.lastz\n",
      "\t/tmp/tmpK7kGEg.lastz\n",
      "\t/tmp/tmpTFyCQr.lastz\n",
      "\t/tmp/tmpEAMvtb.lastz\n",
      "\t/tmp/tmp7REFQ0.lastz\n",
      "\t/tmp/tmpGpGpuo.lastz\n",
      "\t/tmp/tmpiriLGy.lastz\n",
      "\t/tmp/tmp12oLqJ.lastz\n",
      "\t/tmp/tmpMryYfD.lastz\n",
      "\t/tmp/tmpq3aiiu.lastz\n",
      "\t/tmp/tmplLSBIA.lastz\n",
      "\t/tmp/tmpox5meD.lastz\n",
      "\t/tmp/tmpJEdYmK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Paragonimus_westermani_GCA_008508345.lastz\n",
      "Creating Paragonimus_westermani_GCA_008508345 table\n",
      "Inserting data to Paragonimus_westermani_GCA_008508345 table\n",
      "\n",
      "Running against Schistosoma_bovis_GCA_003958945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 37 queries...\n",
      "\t/tmp/tmpHEtY4c.fasta\n",
      "\t/tmp/tmpM89ask.fasta\n",
      "\t/tmp/tmp1rzfud.fasta\n",
      "\t/tmp/tmpKhFCVR.fasta\n",
      "\t/tmp/tmpQrFjTu.fasta\n",
      "\t/tmp/tmpADaZXR.fasta\n",
      "\t/tmp/tmpT9kb1J.fasta\n",
      "\t/tmp/tmp7XWZOu.fasta\n",
      "\t/tmp/tmpizrhVK.fasta\n",
      "\t/tmp/tmpVBLepF.fasta\n",
      "\t/tmp/tmpG0rTHi.fasta\n",
      "\t/tmp/tmppb5wif.fasta\n",
      "\t/tmp/tmp8NMiGt.fasta\n",
      "\t/tmp/tmp9XczBm.fasta\n",
      "\t/tmp/tmpPZMXNu.fasta\n",
      "\t/tmp/tmpg0v2pR.fasta\n",
      "\t/tmp/tmp7kBaaA.fasta\n",
      "\t/tmp/tmprmcXbk.fasta\n",
      "\t/tmp/tmpZ7S9IQ.fasta\n",
      "\t/tmp/tmpyvEsgZ.fasta\n",
      "\t/tmp/tmpxkqRgk.fasta\n",
      "\t/tmp/tmp42OFBy.fasta\n",
      "\t/tmp/tmpZhzBT8.fasta\n",
      "\t/tmp/tmpNeKjCt.fasta\n",
      "\t/tmp/tmpsCvSKI.fasta\n",
      "\t/tmp/tmpiPp0kC.fasta\n",
      "\t/tmp/tmpDw5x8O.fasta\n",
      "\t/tmp/tmpdRxjIk.fasta\n",
      "\t/tmp/tmprSW91X.fasta\n",
      "\t/tmp/tmpn3vaDI.fasta\n",
      "\t/tmp/tmpc1Lea_.fasta\n",
      "\t/tmp/tmpETW6t4.fasta\n",
      "\t/tmp/tmpxk2jWE.fasta\n",
      "\t/tmp/tmptiS6Xg.fasta\n",
      "\t/tmp/tmpFaooen.fasta\n",
      "\t/tmp/tmpWs7RQL.fasta\n",
      "\t/tmp/tmpORZp4z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeI2fOn.lastz\n",
      "\t/tmp/tmp8AaW3c.lastz\n",
      "\t/tmp/tmpqVHErB.lastz\n",
      "\t/tmp/tmpSUTUMZ.lastz\n",
      "\t/tmp/tmpGtKCyu.lastz\n",
      "\t/tmp/tmpG8iiyx.lastz\n",
      "\t/tmp/tmpSA4vya.lastz\n",
      "\t/tmp/tmpfc9iWF.lastz\n",
      "\t/tmp/tmp1bIcUy.lastz\n",
      "\t/tmp/tmp5VsuWx.lastz\n",
      "\t/tmp/tmp3mu99m.lastz\n",
      "\t/tmp/tmpiEY2Oc.lastz\n",
      "\t/tmp/tmpwea_ZP.lastz\n",
      "\t/tmp/tmp5VJZom.lastz\n",
      "\t/tmp/tmpHk0LlF.lastz\n",
      "\t/tmp/tmpwvHEhY.lastz\n",
      "\t/tmp/tmpV3mCV3.lastz\n",
      "\t/tmp/tmpwJUArX.lastz\n",
      "\t/tmp/tmpVTPldd.lastz\n",
      "\t/tmp/tmp9M5aOY.lastz\n",
      "\t/tmp/tmpaI8UUB.lastz\n",
      "\t/tmp/tmpdrP05s.lastz\n",
      "\t/tmp/tmp7dQhZK.lastz\n",
      "\t/tmp/tmpIa6bZl.lastz\n",
      "\t/tmp/tmpdb_OC1.lastz\n",
      "\t/tmp/tmpXJTs83.lastz\n",
      "\t/tmp/tmpIVpmJG.lastz\n",
      "\t/tmp/tmpeUchQz.lastz\n",
      "\t/tmp/tmpB4WuYx.lastz\n",
      "\t/tmp/tmpq46eHA.lastz\n",
      "\t/tmp/tmptuBMSr.lastz\n",
      "\t/tmp/tmpoC_RiQ.lastz\n",
      "\t/tmp/tmpBSbiEG.lastz\n",
      "\t/tmp/tmpfIzF1G.lastz\n",
      "\t/tmp/tmp1mmyEy.lastz\n",
      "\t/tmp/tmpVSkB2K.lastz\n",
      "\t/tmp/tmpA32fbf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_bovis_GCA_003958945.lastz\n",
      "Creating Schistosoma_bovis_GCA_003958945 table\n",
      "Inserting data to Schistosoma_bovis_GCA_003958945 table\n",
      "\n",
      "Running against Schistosoma_curassoni_GCA_900618015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 35 queries...\n",
      "\t/tmp/tmpOnEqmx.fasta\n",
      "\t/tmp/tmpSKWXOW.fasta\n",
      "\t/tmp/tmp9Trhv1.fasta\n",
      "\t/tmp/tmpWwq80D.fasta\n",
      "\t/tmp/tmp4R7QJV.fasta\n",
      "\t/tmp/tmpdw2pKo.fasta\n",
      "\t/tmp/tmpL7sUVX.fasta\n",
      "\t/tmp/tmpTsZJtI.fasta\n",
      "\t/tmp/tmpZluQpw.fasta\n",
      "\t/tmp/tmpngEtjN.fasta\n",
      "\t/tmp/tmpe4nyII.fasta\n",
      "\t/tmp/tmpCUB08R.fasta\n",
      "\t/tmp/tmpjWitGC.fasta\n",
      "\t/tmp/tmpwt8cLE.fasta\n",
      "\t/tmp/tmpXC9cip.fasta\n",
      "\t/tmp/tmpJhN7p2.fasta\n",
      "\t/tmp/tmpxpEHGN.fasta\n",
      "\t/tmp/tmpKfyptJ.fasta\n",
      "\t/tmp/tmpPBz_LL.fasta\n",
      "\t/tmp/tmpSiDNvP.fasta\n",
      "\t/tmp/tmpA_JgXG.fasta\n",
      "\t/tmp/tmpXasbEW.fasta\n",
      "\t/tmp/tmpk3DxZq.fasta\n",
      "\t/tmp/tmpbhjxji.fasta\n",
      "\t/tmp/tmpEUbJ1S.fasta\n",
      "\t/tmp/tmpXVOC_F.fasta\n",
      "\t/tmp/tmpupMKBf.fasta\n",
      "\t/tmp/tmpnFAD5X.fasta\n",
      "\t/tmp/tmpnYj5mI.fasta\n",
      "\t/tmp/tmpeRtzOa.fasta\n",
      "\t/tmp/tmpqNP5vV.fasta\n",
      "\t/tmp/tmpVEzDHm.fasta\n",
      "\t/tmp/tmpBw9hAn.fasta\n",
      "\t/tmp/tmpm4lMoR.fasta\n",
      "\t/tmp/tmpmYxf6o.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp72EK94.lastz\n",
      "\t/tmp/tmpyqEY9s.lastz\n",
      "\t/tmp/tmpk8IzPe.lastz\n",
      "\t/tmp/tmpuWq3BG.lastz\n",
      "\t/tmp/tmpwzPnZF.lastz\n",
      "\t/tmp/tmpw9_a1L.lastz\n",
      "\t/tmp/tmpDhf9tf.lastz\n",
      "\t/tmp/tmpEFQBma.lastz\n",
      "\t/tmp/tmp3aH50c.lastz\n",
      "\t/tmp/tmpL_CAly.lastz\n",
      "\t/tmp/tmpUsdK6T.lastz\n",
      "\t/tmp/tmpDlKMRW.lastz\n",
      "\t/tmp/tmpo13qio.lastz\n",
      "\t/tmp/tmpOJj33E.lastz\n",
      "\t/tmp/tmpTeGrat.lastz\n",
      "\t/tmp/tmp_xF19p.lastz\n",
      "\t/tmp/tmpaWGyPq.lastz\n",
      "\t/tmp/tmpqhDYIT.lastz\n",
      "\t/tmp/tmpTOb4sF.lastz\n",
      "\t/tmp/tmp7wfTL5.lastz\n",
      "\t/tmp/tmpHoi1du.lastz\n",
      "\t/tmp/tmpadVaZA.lastz\n",
      "\t/tmp/tmp7n9DOz.lastz\n",
      "\t/tmp/tmprCZ37X.lastz\n",
      "\t/tmp/tmpHKumgj.lastz\n",
      "\t/tmp/tmpoHEMXq.lastz\n",
      "\t/tmp/tmpJNsDWI.lastz\n",
      "\t/tmp/tmpItT7KA.lastz\n",
      "\t/tmp/tmpuFpZ7I.lastz\n",
      "\t/tmp/tmpwETZeF.lastz\n",
      "\t/tmp/tmpWdBnwd.lastz\n",
      "\t/tmp/tmpBFn0bg.lastz\n",
      "\t/tmp/tmp8LtwnQ.lastz\n",
      "\t/tmp/tmpCtqRob.lastz\n",
      "\t/tmp/tmp4YWj3C.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_curassoni_GCA_900618015.lastz\n",
      "Creating Schistosoma_curassoni_GCA_900618015 table\n",
      "Inserting data to Schistosoma_curassoni_GCA_900618015 table\n",
      "\n",
      "Running against Schistosoma_haematobium_GCF_000699445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 37 queries...\n",
      "\t/tmp/tmpAcCc8B.fasta\n",
      "\t/tmp/tmpdj9we3.fasta\n",
      "\t/tmp/tmpNRJ5zS.fasta\n",
      "\t/tmp/tmpSsecIk.fasta\n",
      "\t/tmp/tmpzUfF6p.fasta\n",
      "\t/tmp/tmpQxXovO.fasta\n",
      "\t/tmp/tmpbkTOhp.fasta\n",
      "\t/tmp/tmpTVNBPb.fasta\n",
      "\t/tmp/tmpGtEifD.fasta\n",
      "\t/tmp/tmpFtaYzF.fasta\n",
      "\t/tmp/tmppohTnT.fasta\n",
      "\t/tmp/tmppwyg7u.fasta\n",
      "\t/tmp/tmpZkcznP.fasta\n",
      "\t/tmp/tmpkgnnOM.fasta\n",
      "\t/tmp/tmpWFBJkI.fasta\n",
      "\t/tmp/tmpEFg7NK.fasta\n",
      "\t/tmp/tmp7U3syR.fasta\n",
      "\t/tmp/tmpEfBO7q.fasta\n",
      "\t/tmp/tmp33vOZD.fasta\n",
      "\t/tmp/tmpBSawUH.fasta\n",
      "\t/tmp/tmpsK0Y_t.fasta\n",
      "\t/tmp/tmppfduVa.fasta\n",
      "\t/tmp/tmpIqJJuL.fasta\n",
      "\t/tmp/tmpUpWIFE.fasta\n",
      "\t/tmp/tmpG6HrRu.fasta\n",
      "\t/tmp/tmpDFjrro.fasta\n",
      "\t/tmp/tmp3n6rMw.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpv0zHAK.fasta\n",
      "\t/tmp/tmpeJJyec.fasta\n",
      "\t/tmp/tmp0CO5c1.fasta\n",
      "\t/tmp/tmp6yjOQP.fasta\n",
      "\t/tmp/tmpQ1dX1u.fasta\n",
      "\t/tmp/tmp6s_Ei_.fasta\n",
      "\t/tmp/tmpTptiYr.fasta\n",
      "\t/tmp/tmpxX_jMl.fasta\n",
      "\t/tmp/tmpUGwnMI.fasta\n",
      "\t/tmp/tmpVUg5FY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpN7wLx6.lastz\n",
      "\t/tmp/tmpFYE5ID.lastz\n",
      "\t/tmp/tmp2mYVGE.lastz\n",
      "\t/tmp/tmpi7ES4p.lastz\n",
      "\t/tmp/tmpHxaOXS.lastz\n",
      "\t/tmp/tmp2CugYf.lastz\n",
      "\t/tmp/tmpPwgX1F.lastz\n",
      "\t/tmp/tmpOYDUea.lastz\n",
      "\t/tmp/tmpyR5Hf2.lastz\n",
      "\t/tmp/tmp6cq4cB.lastz\n",
      "\t/tmp/tmp8Ir_43.lastz\n",
      "\t/tmp/tmpyQ2FmE.lastz\n",
      "\t/tmp/tmpbGkthy.lastz\n",
      "\t/tmp/tmpWFXS6x.lastz\n",
      "\t/tmp/tmpCG4zCU.lastz\n",
      "\t/tmp/tmpaxQ9GY.lastz\n",
      "\t/tmp/tmp6jDvg5.lastz\n",
      "\t/tmp/tmpw2hi4Z.lastz\n",
      "\t/tmp/tmpzpQEXJ.lastz\n",
      "\t/tmp/tmp1nSo2b.lastz\n",
      "\t/tmp/tmpuip0_f.lastz\n",
      "\t/tmp/tmpGSnFPo.lastz\n",
      "\t/tmp/tmpaYg9Uj.lastz\n",
      "\t/tmp/tmptzUg7I.lastz\n",
      "\t/tmp/tmp4PUn9U.lastz\n",
      "\t/tmp/tmphQxRdO.lastz\n",
      "\t/tmp/tmpLpudh2.lastz\n",
      "\t/tmp/tmp466TzR.lastz\n",
      "\t/tmp/tmpUurls9.lastz\n",
      "\t/tmp/tmpQDu_H7.lastz\n",
      "\t/tmp/tmpOqrtSm.lastz\n",
      "\t/tmp/tmpjHIHD3.lastz\n",
      "\t/tmp/tmpRS3wRZ.lastz\n",
      "\t/tmp/tmpds9pFs.lastz\n",
      "\t/tmp/tmpW3Nbmw.lastz\n",
      "\t/tmp/tmpA33R3M.lastz\n",
      "\t/tmp/tmpCCGsuE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_haematobium_GCF_000699445.lastz\n",
      "Creating Schistosoma_haematobium_GCF_000699445 table\n",
      "Inserting data to Schistosoma_haematobium_GCF_000699445 table\n",
      "\n",
      "Running against Schistosoma_japonicum_GCA_006368765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 35 queries...\n",
      "\t/tmp/tmpR5Zqo7.fasta\n",
      "\t/tmp/tmpZoolfq.fasta\n",
      "\t/tmp/tmpSFwFHv.fasta\n",
      "\t/tmp/tmp60zuBn.fasta\n",
      "\t/tmp/tmpO92bmk.fasta\n",
      "\t/tmp/tmpko0IMo.fasta\n",
      "\t/tmp/tmpye7rQV.fasta\n",
      "\t/tmp/tmpd1W4Bt.fasta\n",
      "\t/tmp/tmpLiMlcg.fasta\n",
      "\t/tmp/tmpUlflDx.fasta\n",
      "\t/tmp/tmpVuxtTA.fasta\n",
      "\t/tmp/tmpjwEjkQ.fasta\n",
      "\t/tmp/tmpYp9YlG.fasta\n",
      "\t/tmp/tmpYgGNhM.fasta\n",
      "\t/tmp/tmpA7Qr2p.fasta\n",
      "\t/tmp/tmpv1o05n.fasta\n",
      "\t/tmp/tmpM_2gNa.fasta\n",
      "\t/tmp/tmp13q1ew.fasta\n",
      "\t/tmp/tmpZ7CoyN.fasta\n",
      "\t/tmp/tmpcEkgBm.fasta\n",
      "\t/tmp/tmp4gUd29.fasta\n",
      "\t/tmp/tmp4B7TRO.fasta\n",
      "\t/tmp/tmpdnK7nu.fasta\n",
      "\t/tmp/tmpTs4N4d.fasta\n",
      "\t/tmp/tmpH_QZPT.fasta\n",
      "\t/tmp/tmpO5fwJT.fasta\n",
      "\t/tmp/tmpNNt0bZ.fasta\n",
      "\t/tmp/tmpxl3jA6.fasta\n",
      "\t/tmp/tmpUK7wXS.fasta\n",
      "\t/tmp/tmpz5Kw7A.fasta\n",
      "\t/tmp/tmp4vf2oL.fasta\n",
      "\t/tmp/tmp6lZqoX.fasta\n",
      "\t/tmp/tmpCoVQsY.fasta\n",
      "\t/tmp/tmpCzCxFn.fasta\n",
      "\t/tmp/tmpIO6UcE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqqgC1l.lastz\n",
      "\t/tmp/tmpLYrLCz.lastz\n",
      "\t/tmp/tmpPDJLr2.lastz\n",
      "\t/tmp/tmp1xU3JE.lastz\n",
      "\t/tmp/tmpAcdbS2.lastz\n",
      "\t/tmp/tmpzv_35q.lastz\n",
      "\t/tmp/tmppCMXEx.lastz\n",
      "\t/tmp/tmpXailUj.lastz\n",
      "\t/tmp/tmpUhdVkh.lastz\n",
      "\t/tmp/tmpHO5Iqe.lastz\n",
      "\t/tmp/tmpzE8LCF.lastz\n",
      "\t/tmp/tmpC_lh0g.lastz\n",
      "\t/tmp/tmpX44Mm1.lastz\n",
      "\t/tmp/tmp08h1M7.lastz\n",
      "\t/tmp/tmpeIchgV.lastz\n",
      "\t/tmp/tmpPU6nRf.lastz\n",
      "\t/tmp/tmpckLrab.lastz\n",
      "\t/tmp/tmpf9VQY6.lastz\n",
      "\t/tmp/tmpQe33I9.lastz\n",
      "\t/tmp/tmpO9eWL5.lastz\n",
      "\t/tmp/tmp1LajqY.lastz\n",
      "\t/tmp/tmp7J_375.lastz\n",
      "\t/tmp/tmpPZMO7Q.lastz\n",
      "\t/tmp/tmp1meE87.lastz\n",
      "\t/tmp/tmpcmOVMz.lastz\n",
      "\t/tmp/tmp2DirHe.lastz\n",
      "\t/tmp/tmpnTu24f.lastz\n",
      "\t/tmp/tmpD2Sxta.lastz\n",
      "\t/tmp/tmpxWEZ9E.lastz\n",
      "\t/tmp/tmpYBb55N.lastz\n",
      "\t/tmp/tmp9CGTT4.lastz\n",
      "\t/tmp/tmppcism_.lastz\n",
      "\t/tmp/tmpf_mlsr.lastz\n",
      "\t/tmp/tmpzZWDCd.lastz\n",
      "\t/tmp/tmpjzFBQE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_japonicum_GCA_006368765.lastz\n",
      "Creating Schistosoma_japonicum_GCA_006368765 table\n",
      "Inserting data to Schistosoma_japonicum_GCA_006368765 table\n",
      "\n",
      "Running against Schistosoma_margrebowiei_GCA_900618395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 37 queries...\n",
      "\t/tmp/tmpzlZbN2.fasta\n",
      "\t/tmp/tmpiHO_Ny.fasta\n",
      "\t/tmp/tmp5zbwV3.fasta\n",
      "\t/tmp/tmp2SzCFc.fasta\n",
      "\t/tmp/tmpzpH6SB.fasta\n",
      "\t/tmp/tmpq5ISPW.fasta\n",
      "\t/tmp/tmpyQVcwO.fasta\n",
      "\t/tmp/tmpDsW06B.fasta\n",
      "\t/tmp/tmpSahW7I.fasta\n",
      "\t/tmp/tmpJaZU9_.fasta\n",
      "\t/tmp/tmpB6AD7y.fasta\n",
      "\t/tmp/tmpZqoW3u.fasta\n",
      "\t/tmp/tmpN4cvex.fasta\n",
      "\t/tmp/tmp7Zl6oH.fasta\n",
      "\t/tmp/tmpog_yrr.fasta\n",
      "\t/tmp/tmpDGV1Bn.fasta\n",
      "\t/tmp/tmp3QVjvC.fasta\n",
      "\t/tmp/tmpYeoM1b.fasta\n",
      "\t/tmp/tmpIM3Lb8.fasta\n",
      "\t/tmp/tmpFP2gan.fasta\n",
      "\t/tmp/tmpA1NfvK.fasta\n",
      "\t/tmp/tmpKX1LhL.fasta\n",
      "\t/tmp/tmpGrJI_M.fasta\n",
      "\t/tmp/tmpl8NZEw.fasta\n",
      "\t/tmp/tmpIIb17g.fasta\n",
      "\t/tmp/tmpAFfOBx.fasta\n",
      "\t/tmp/tmpSLt0eG.fasta\n",
      "\t/tmp/tmpGEKrx6.fasta\n",
      "\t/tmp/tmpkBbpyQ.fasta\n",
      "\t/tmp/tmpUWQhAk.fasta\n",
      "\t/tmp/tmpsPvLQy.fasta\n",
      "\t/tmp/tmpRjXRTe.fasta\n",
      "\t/tmp/tmp5_q0G9.fasta\n",
      "\t/tmp/tmpNo8sH0.fasta\n",
      "\t/tmp/tmpB__aVw.fasta\n",
      "\t/tmp/tmpGeAZ7Q.fasta\n",
      "\t/tmp/tmpvaDVZd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvCXPK5.lastz\n",
      "\t/tmp/tmpYKm2Vo.lastz\n",
      "\t/tmp/tmp3pexWW.lastz\n",
      "\t/tmp/tmpZt3pWe.lastz\n",
      "\t/tmp/tmpjO4bGQ.lastz\n",
      "\t/tmp/tmp9p2MIr.lastz\n",
      "\t/tmp/tmp4wpTat.lastz\n",
      "\t/tmp/tmpJgFkr1.lastz\n",
      "\t/tmp/tmpDLusnx.lastz\n",
      "\t/tmp/tmphhUkD_.lastz\n",
      "\t/tmp/tmpJmK882.lastz\n",
      "\t/tmp/tmpi7xYKQ.lastz\n",
      "\t/tmp/tmpuiZlNF.lastz\n",
      "\t/tmp/tmpDmxlms.lastz\n",
      "\t/tmp/tmpjNdGIC.lastz\n",
      "\t/tmp/tmp4tw1Br.lastz\n",
      "\t/tmp/tmpYkCruv.lastz\n",
      "\t/tmp/tmpmQyoAh.lastz\n",
      "\t/tmp/tmpKVlHPp.lastz\n",
      "\t/tmp/tmpeNxlD_.lastz\n",
      "\t/tmp/tmpQxGjJN.lastz\n",
      "\t/tmp/tmp9AuV0s.lastz\n",
      "\t/tmp/tmpUYHMCY.lastz\n",
      "\t/tmp/tmp0ZJzpZ.lastz\n",
      "\t/tmp/tmpKiV6o9.lastz\n",
      "\t/tmp/tmpE3R3sj.lastz\n",
      "\t/tmp/tmp834uSO.lastz\n",
      "\t/tmp/tmpjte9oB.lastz\n",
      "\t/tmp/tmptITfpM.lastz\n",
      "\t/tmp/tmp9tLIAw.lastz\n",
      "\t/tmp/tmpN8H5Jv.lastz\n",
      "\t/tmp/tmpqH5Fij.lastz\n",
      "\t/tmp/tmp3Qe2Wo.lastz\n",
      "\t/tmp/tmpoxgEwl.lastz\n",
      "\t/tmp/tmpQp3UR4.lastz\n",
      "\t/tmp/tmpJkoLJ2.lastz\n",
      "\t/tmp/tmpqUehrM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_margrebowiei_GCA_900618395.lastz\n",
      "Creating Schistosoma_margrebowiei_GCA_900618395 table\n",
      "Inserting data to Schistosoma_margrebowiei_GCA_900618395 table\n",
      "\n",
      "Running against Schistosoma_mattheei_GCA_900617995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 35 queries...\n",
      "\t/tmp/tmpw0y2Z2.fasta\n",
      "\t/tmp/tmpMN5i78.fasta\n",
      "\t/tmp/tmppY3TLx.fasta\n",
      "\t/tmp/tmph7zF2p.fasta\n",
      "\t/tmp/tmpRAQU5n.fasta\n",
      "\t/tmp/tmpPPBAfg.fasta\n",
      "\t/tmp/tmp3Ufbru.fasta\n",
      "\t/tmp/tmpy2WBF4.fasta\n",
      "\t/tmp/tmpK88lbO.fasta\n",
      "\t/tmp/tmpJiWlPg.fasta\n",
      "\t/tmp/tmptNzUgq.fasta\n",
      "\t/tmp/tmpznhH5P.fasta\n",
      "\t/tmp/tmpXOH86r.fasta\n",
      "\t/tmp/tmpmWmSZM.fasta\n",
      "\t/tmp/tmpmq7tII.fasta\n",
      "\t/tmp/tmpiKysbE.fasta\n",
      "\t/tmp/tmpQMozpL.fasta\n",
      "\t/tmp/tmpSz8oot.fasta\n",
      "\t/tmp/tmpb4rmQr.fasta\n",
      "\t/tmp/tmpTVzgra.fasta\n",
      "\t/tmp/tmp3_4Zlo.fasta\n",
      "\t/tmp/tmplGM8S7.fasta\n",
      "\t/tmp/tmpQEO5is.fasta\n",
      "\t/tmp/tmpxan40A.fasta\n",
      "\t/tmp/tmpFunGr5.fasta\n",
      "\t/tmp/tmp1sbugC.fasta\n",
      "\t/tmp/tmpYMjsGO.fasta\n",
      "\t/tmp/tmpG4Laai.fasta\n",
      "\t/tmp/tmpUmYGky.fasta\n",
      "\t/tmp/tmpRYfUNT.fasta\n",
      "\t/tmp/tmpyfR99M.fasta\n",
      "\t/tmp/tmpeDdGTg.fasta\n",
      "\t/tmp/tmpgnWBa2.fasta\n",
      "\t/tmp/tmp2KpXMh.fasta\n",
      "\t/tmp/tmpxwN4yV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqfAnOB.lastz\n",
      "\t/tmp/tmpm6rwb5.lastz\n",
      "\t/tmp/tmp4eSHzf.lastz\n",
      "\t/tmp/tmpbiwGf7.lastz\n",
      "\t/tmp/tmp4Y9z12.lastz\n",
      "\t/tmp/tmpZqdIl3.lastz\n",
      "\t/tmp/tmpE9r_R7.lastz\n",
      "\t/tmp/tmpl8P2qb.lastz\n",
      "\t/tmp/tmpeV1Z0r.lastz\n",
      "\t/tmp/tmpe4ICm3.lastz\n",
      "\t/tmp/tmpHYMI6i.lastz\n",
      "\t/tmp/tmpVPZsfG.lastz\n",
      "\t/tmp/tmpiDKaPJ.lastz\n",
      "\t/tmp/tmp0aqEaa.lastz\n",
      "\t/tmp/tmpHoex1j.lastz\n",
      "\t/tmp/tmpYB2rO0.lastz\n",
      "\t/tmp/tmpvC82N3.lastz\n",
      "\t/tmp/tmpXOS13V.lastz\n",
      "\t/tmp/tmpYR61ql.lastz\n",
      "\t/tmp/tmp4ihDbl.lastz\n",
      "\t/tmp/tmp00elSw.lastz\n",
      "\t/tmp/tmpszrANj.lastz\n",
      "\t/tmp/tmp9eFmyn.lastz\n",
      "\t/tmp/tmppgXfUy.lastz\n",
      "\t/tmp/tmptcATnV.lastz\n",
      "\t/tmp/tmp7w06pr.lastz\n",
      "\t/tmp/tmpri1JSW.lastz\n",
      "\t/tmp/tmpXWHE9h.lastz\n",
      "\t/tmp/tmp0zUpGt.lastz\n",
      "\t/tmp/tmpYRBKMY.lastz\n",
      "\t/tmp/tmpkzdKMh.lastz\n",
      "\t/tmp/tmpexixnQ.lastz\n",
      "\t/tmp/tmpFbgJqa.lastz\n",
      "\t/tmp/tmpzIpXEh.lastz\n",
      "\t/tmp/tmp0Zr_qi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_mattheei_GCA_900617995.lastz\n",
      "Creating Schistosoma_mattheei_GCA_900617995 table\n",
      "Inserting data to Schistosoma_mattheei_GCA_900617995 table\n",
      "\n",
      "Running against Trichobilharzia_regenti_GCA_900618515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 71 queries...\n",
      "\t/tmp/tmpkrKAIt.fasta\n",
      "\t/tmp/tmprdnyYz.fasta\n",
      "\t/tmp/tmpz5z0fL.fasta\n",
      "\t/tmp/tmpcyC5vS.fasta\n",
      "\t/tmp/tmpOCVx_G.fasta\n",
      "\t/tmp/tmpbvl_mJ.fasta\n",
      "\t/tmp/tmpR6VA0f.fasta\n",
      "\t/tmp/tmpg8cEVq.fasta\n",
      "\t/tmp/tmpWFlI2F.fasta\n",
      "\t/tmp/tmp8MGeyG.fasta\n",
      "\t/tmp/tmpDWKtXB.fasta\n",
      "\t/tmp/tmpqptg6F.fasta\n",
      "\t/tmp/tmpIlfDQO.fasta\n",
      "\t/tmp/tmpH8QcvR.fasta\n",
      "\t/tmp/tmpQyGomH.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpIkrkyX.fasta\n",
      "\t/tmp/tmpdz7mM9.fasta\n",
      "\t/tmp/tmpKxdDe8.fasta\n",
      "\t/tmp/tmpumRl5X.fasta\n",
      "\t/tmp/tmpYCK7cz.fasta\n",
      "\t/tmp/tmpOA_BXF.fasta\n",
      "\t/tmp/tmpUCvkLm.fasta\n",
      "\t/tmp/tmpVHi0Xd.fasta\n",
      "\t/tmp/tmp51K4_B.fasta\n",
      "\t/tmp/tmpQkMY7P.fasta\n",
      "\t/tmp/tmpyPlxxZ.fasta\n",
      "\t/tmp/tmpxfTIJe.fasta\n",
      "\t/tmp/tmplHj17T.fasta\n",
      "\t/tmp/tmp7zNOTJ.fasta\n",
      "\t/tmp/tmp_AwAb2.fasta\n",
      "\t/tmp/tmpevh75M.fasta\n",
      "\t/tmp/tmp7N3TYT.fasta\n",
      "\t/tmp/tmppQh6c6.fasta\n",
      "\t/tmp/tmpsyUI_Z.fasta\n",
      "\t/tmp/tmpwQXQLo.fasta\n",
      "\t/tmp/tmpRmIDeZ.fasta\n",
      "\t/tmp/tmp3Z7lGy.fasta\n",
      "\t/tmp/tmp5r8ACQ.fasta\n",
      "\t/tmp/tmpdOLYd3.fasta\n",
      "\t/tmp/tmpPzXfY9.fasta\n",
      "\t/tmp/tmpQ4A5xP.fasta\n",
      "\t/tmp/tmpUWWU4S.fasta\n",
      "\t/tmp/tmpv7dpZx.fasta\n",
      "\t/tmp/tmpPeC1d1.fasta\n",
      "\t/tmp/tmpXK0ulu.fasta\n",
      "\t/tmp/tmpK1LIMZ.fasta\n",
      "\t/tmp/tmpSuyKjs.fasta\n",
      "\t/tmp/tmpu0uM57.fasta\n",
      "\t/tmp/tmpUy_8Ab.fasta\n",
      "\t/tmp/tmpU38ovy.fasta\n",
      "\t/tmp/tmpL9yqXR.fasta\n",
      "\t/tmp/tmpoOOXQg.fasta\n",
      "\t/tmp/tmpa_urvT.fasta\n",
      "\t/tmp/tmp63wYhK.fasta\n",
      "\t/tmp/tmpmatTGF.fasta\n",
      "\t/tmp/tmpO0k6g7.fasta\n",
      "\t/tmp/tmp3IEPTr.fasta\n",
      "\t/tmp/tmpCivJMB.fasta\n",
      "\t/tmp/tmpiOFp1W.fasta\n",
      "\t/tmp/tmpdSRnWW.fasta\n",
      "\t/tmp/tmp04bEd_.fasta\n",
      "\t/tmp/tmpcNo4gm.fasta\n",
      "\t/tmp/tmpUonQGl.fasta\n",
      "\t/tmp/tmpW6bx0F.fasta\n",
      "\t/tmp/tmpMtgkml.fasta\n",
      "\t/tmp/tmpYyaXNM.fasta\n",
      "\t/tmp/tmpIzyn4Y.fasta\n",
      "\t/tmp/tmpllz8rm.fasta\n",
      "\t/tmp/tmp24L5Yf.fasta\n",
      "\t/tmp/tmpIMpRHd.fasta\n",
      "\t/tmp/tmpi_REax.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXTFbUi.lastz\n",
      "\t/tmp/tmp2bBTk_.lastz\n",
      "\t/tmp/tmpKV4aIU.lastz\n",
      "\t/tmp/tmptmCqTi.lastz\n",
      "\t/tmp/tmpQumBV9.lastz\n",
      "\t/tmp/tmpfRM0gB.lastz\n",
      "\t/tmp/tmpEdzyUg.lastz\n",
      "\t/tmp/tmpU6ZbVe.lastz\n",
      "\t/tmp/tmpX9XgzU.lastz\n",
      "\t/tmp/tmpGTASL3.lastz\n",
      "\t/tmp/tmp7yYpwJ.lastz\n",
      "\t/tmp/tmp8ABwfL.lastz\n",
      "\t/tmp/tmp5N6mnB.lastz\n",
      "\t/tmp/tmpU5sYvh.lastz\n",
      "\t/tmp/tmp13FVli.lastz\n",
      "\t/tmp/tmp8xLfoR.lastz\n",
      "\t/tmp/tmp9XSaxI.lastz\n",
      "\t/tmp/tmpXJ_C90.lastz\n",
      "\t/tmp/tmpCTUC7v.lastz\n",
      "\t/tmp/tmpXdmGJ_.lastz\n",
      "\t/tmp/tmpJapYkA.lastz\n",
      "\t/tmp/tmp7jtibz.lastz\n",
      "\t/tmp/tmpYg3TFO.lastz\n",
      "\t/tmp/tmpGKqjQR.lastz\n",
      "\t/tmp/tmpinxZ1G.lastz\n",
      "\t/tmp/tmpoRZZ_P.lastz\n",
      "\t/tmp/tmpHMhDeX.lastz\n",
      "\t/tmp/tmp8FOgq3.lastz\n",
      "\t/tmp/tmp_1SsSu.lastz\n",
      "\t/tmp/tmphmS2_Y.lastz\n",
      "\t/tmp/tmp2UncG0.lastz\n",
      "\t/tmp/tmprtbBjm.lastz\n",
      "\t/tmp/tmpkjrb4i.lastz\n",
      "\t/tmp/tmpoJBXvd.lastz\n",
      "\t/tmp/tmp4Nelj6.lastz\n",
      "\t/tmp/tmpkhwd9l.lastz\n",
      "\t/tmp/tmpVk6E06.lastz\n",
      "\t/tmp/tmp2XNDWR.lastz\n",
      "\t/tmp/tmp7JmGD8.lastz\n",
      "\t/tmp/tmpl18ml0.lastz\n",
      "\t/tmp/tmpuCfVj1.lastz\n",
      "\t/tmp/tmp5sKSem.lastz\n",
      "\t/tmp/tmpkH_uvy.lastz\n",
      "\t/tmp/tmp5f9KXl.lastz\n",
      "\t/tmp/tmpkPRfmC.lastz\n",
      "\t/tmp/tmphHX0iA.lastz\n",
      "\t/tmp/tmpCOilt9.lastz\n",
      "\t/tmp/tmppdW6iR.lastz\n",
      "\t/tmp/tmpOwHiZj.lastz\n",
      "\t/tmp/tmp7TuesO.lastz\n",
      "\t/tmp/tmpxRoH6v.lastz\n",
      "\t/tmp/tmp5T0tRq.lastz\n",
      "\t/tmp/tmpdsMpZP.lastz\n",
      "\t/tmp/tmpXDLHsk.lastz\n",
      "\t/tmp/tmp9SqqCh.lastz\n",
      "\t/tmp/tmpOTRf2c.lastz\n",
      "\t/tmp/tmp6TEDy9.lastz\n",
      "\t/tmp/tmptKITU5.lastz\n",
      "\t/tmp/tmpQ_BlMs.lastz\n",
      "\t/tmp/tmpzpB6vj.lastz\n",
      "\t/tmp/tmpSqD5v6.lastz\n",
      "\t/tmp/tmp9An9Nf.lastz\n",
      "\t/tmp/tmpISTshh.lastz\n",
      "\t/tmp/tmpDC9Bta.lastz\n",
      "\t/tmp/tmpRfUw6x.lastz\n",
      "\t/tmp/tmpeNDlBk.lastz\n",
      "\t/tmp/tmp69pJJk.lastz\n",
      "\t/tmp/tmpKqnDCP.lastz\n",
      "\t/tmp/tmpeDxCA3.lastz\n",
      "\t/tmp/tmpPRuYmB.lastz\n",
      "\t/tmp/tmpmpg0fr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Trichobilharzia_regenti_GCA_900618515.lastz\n",
      "Creating Trichobilharzia_regenti_GCA_900618515 table\n",
      "Inserting data to Trichobilharzia_regenti_GCA_900618515 table\n",
      "\n",
      "Running against Schistosoma_mansoni_GCA_000237925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpMxF94V.fasta\n",
      "\t/tmp/tmpVtvyqp.fasta\n",
      "\t/tmp/tmpvxi3gp.fasta\n",
      "\t/tmp/tmp2rn8rP.fasta\n",
      "\t/tmp/tmpnC09LA.fasta\n",
      "\t/tmp/tmpalmp8J.fasta\n",
      "\t/tmp/tmpkInA7x.fasta\n",
      "\t/tmp/tmpcikbEv.fasta\n",
      "\t/tmp/tmpgrBRca.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzZQfRX.lastz\n",
      "\t/tmp/tmpekRxZD.lastz\n",
      "\t/tmp/tmpDdlUBW.lastz\n",
      "\t/tmp/tmpdEfyjt.lastz\n",
      "\t/tmp/tmpJdnXP5.lastz\n",
      "\t/tmp/tmpo9cVLn.lastz\n",
      "\t/tmp/tmp6dtLF7.lastz\n",
      "\t/tmp/tmpFEUAtP.lastz\n",
      "\t/tmp/tmpxcCNiI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trematoda/validate_intervals/lastz/Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_Schistosoma_mansoni_GCA_000237925.lastz\n",
      "Creating Schistosoma_mansoni_GCA_000237925 table\n",
      "Inserting data to Schistosoma_mansoni_GCA_000237925 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/trematoda/extract_probes_from_group/trematoda_genome.conf --lastz results/phyluce/trematoda/validate_intervals/lastz --probes 120 --probe-prefix uce_trematoda_ --name-pattern \"Schistosoma_mansoni_GCA_000237925_+6_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/trematoda/extract_probes_from_group/probe_fasta;\n",
      "2020-02-10 15:57:46,446 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-10 15:57:46,449 - Phyluce - INFO - ------ Working on Clonorchis_sinensis_GCA_003604175 genome ------\n",
      "2020-02-10 15:57:46,450 - Phyluce - INFO - Reading Clonorchis_sinensis_GCA_003604175 genome\n",
      "2020-02-10 15:58:01,408 - Phyluce - INFO - Clonorchis_sinensis_GCA_003604175: 3454 uces, 572 dupes, 2882 non-dupes, 15 orient drop, 53 length drop, 2814 written\n",
      "2020-02-10 15:58:01,408 - Phyluce - INFO - ---- Working on Dicrocoelium_dendriticum_GCA_000950715 genome ---\n",
      "2020-02-10 15:58:01,409 - Phyluce - INFO - Reading Dicrocoelium_dendriticum_GCA_000950715 genome\n",
      "2020-02-10 15:58:18,432 - Phyluce - INFO - Dicrocoelium_dendriticum_GCA_000950715: 921 uces, 107 dupes, 814 non-dupes, 0 orient drop, 0 length drop, 798 written\n",
      "2020-02-10 15:58:18,432 - Phyluce - INFO - ------ Working on Echinostoma_caproni_GCA_900618425 genome ------\n",
      "2020-02-10 15:58:18,440 - Phyluce - INFO - Reading Echinostoma_caproni_GCA_900618425 genome\n",
      "2020-02-10 15:58:37,568 - Phyluce - INFO - Echinostoma_caproni_GCA_900618425: 3441 uces, 552 dupes, 2889 non-dupes, 1 orient drop, 11 length drop, 2877 written\n",
      "2020-02-10 15:58:37,569 - Phyluce - INFO - ------- Working on Fasciola_gigantica_GCA_006461475 genome ------\n",
      "2020-02-10 15:58:37,570 - Phyluce - INFO - Reading Fasciola_gigantica_GCA_006461475 genome\n",
      "2020-02-10 15:58:55,771 - Phyluce - INFO - Fasciola_gigantica_GCA_006461475: 3427 uces, 607 dupes, 2820 non-dupes, 18 orient drop, 30 length drop, 2772 written\n",
      "2020-02-10 15:58:55,771 - Phyluce - INFO - ------- Working on Fasciola_hepatica_GCA_900302435 genome -------\n",
      "2020-02-10 15:58:55,780 - Phyluce - INFO - Reading Fasciola_hepatica_GCA_900302435 genome\n",
      "2020-02-10 15:59:16,484 - Phyluce - INFO - Fasciola_hepatica_GCA_900302435: 3556 uces, 637 dupes, 2919 non-dupes, 20 orient drop, 38 length drop, 2861 written\n",
      "2020-02-10 15:59:16,484 - Phyluce - INFO - ------- Working on Fasciolopsis_buski_GCA_008360955 genome ------\n",
      "2020-02-10 15:59:16,494 - Phyluce - INFO - Reading Fasciolopsis_buski_GCA_008360955 genome\n",
      "2020-02-10 15:59:32,165 - Phyluce - INFO - Fasciolopsis_buski_GCA_008360955: 3432 uces, 575 dupes, 2857 non-dupes, 5 orient drop, 23 length drop, 2829 written\n",
      "2020-02-10 15:59:32,165 - Phyluce - INFO - ----- Working on Opisthorchis_felineus_GCA_004794785 genome -----\n",
      "2020-02-10 15:59:32,166 - Phyluce - INFO - Reading Opisthorchis_felineus_GCA_004794785 genome\n",
      "2020-02-10 15:59:47,138 - Phyluce - INFO - Opisthorchis_felineus_GCA_004794785: 3457 uces, 603 dupes, 2854 non-dupes, 6 orient drop, 48 length drop, 2800 written\n",
      "2020-02-10 15:59:47,139 - Phyluce - INFO - ----- Working on Opisthorchis_viverrini_GCF_000715545 genome ----\n",
      "2020-02-10 15:59:47,139 - Phyluce - INFO - Reading Opisthorchis_viverrini_GCF_000715545 genome\n",
      "2020-02-10 16:00:03,291 - Phyluce - INFO - Opisthorchis_viverrini_GCF_000715545: 3445 uces, 586 dupes, 2859 non-dupes, 11 orient drop, 67 length drop, 2780 written\n",
      "2020-02-10 16:00:03,292 - Phyluce - INFO - ----- Working on Paragonimus_westermani_GCA_008508345 genome ----\n",
      "2020-02-10 16:00:03,292 - Phyluce - INFO - Reading Paragonimus_westermani_GCA_008508345 genome\n",
      "2020-02-10 16:00:24,459 - Phyluce - INFO - Paragonimus_westermani_GCA_008508345: 3546 uces, 688 dupes, 2858 non-dupes, 12 orient drop, 34 length drop, 2812 written\n",
      "2020-02-10 16:00:24,459 - Phyluce - INFO - ------- Working on Schistosoma_bovis_GCA_003958945 genome -------\n",
      "2020-02-10 16:00:24,460 - Phyluce - INFO - Reading Schistosoma_bovis_GCA_003958945 genome\n",
      "2020-02-10 16:00:58,211 - Phyluce - INFO - Schistosoma_bovis_GCA_003958945: 5429 uces, 1471 dupes, 3958 non-dupes, 52 orient drop, 135 length drop, 3771 written\n",
      "2020-02-10 16:00:58,211 - Phyluce - INFO - ----- Working on Schistosoma_curassoni_GCA_900618015 genome -----\n",
      "2020-02-10 16:00:58,220 - Phyluce - INFO - Reading Schistosoma_curassoni_GCA_900618015 genome\n",
      "2020-02-10 16:01:25,871 - Phyluce - INFO - Schistosoma_curassoni_GCA_900618015: 5400 uces, 728 dupes, 4672 non-dupes, 3 orient drop, 1 length drop, 4666 written\n",
      "2020-02-10 16:01:25,871 - Phyluce - INFO - ---- Working on Schistosoma_haematobium_GCF_000699445 genome ----\n",
      "2020-02-10 16:01:25,872 - Phyluce - INFO - Reading Schistosoma_haematobium_GCF_000699445 genome\n",
      "2020-02-10 16:01:54,488 - Phyluce - INFO - Schistosoma_haematobium_GCF_000699445: 5417 uces, 853 dupes, 4564 non-dupes, 10 orient drop, 46 length drop, 4508 written\n",
      "2020-02-10 16:01:54,489 - Phyluce - INFO - ----- Working on Schistosoma_japonicum_GCA_006368765 genome -----\n",
      "2020-02-10 16:01:54,490 - Phyluce - INFO - Reading Schistosoma_japonicum_GCA_006368765 genome\n",
      "2020-02-10 16:02:23,694 - Phyluce - INFO - Schistosoma_japonicum_GCA_006368765: 5322 uces, 707 dupes, 4615 non-dupes, 42 orient drop, 66 length drop, 4507 written\n",
      "2020-02-10 16:02:23,694 - Phyluce - INFO - ---- Working on Schistosoma_margrebowiei_GCA_900618395 genome ---\n",
      "2020-02-10 16:02:23,695 - Phyluce - INFO - Reading Schistosoma_margrebowiei_GCA_900618395 genome\n",
      "2020-02-10 16:02:52,660 - Phyluce - INFO - Schistosoma_margrebowiei_GCA_900618395: 5426 uces, 763 dupes, 4663 non-dupes, 7 orient drop, 12 length drop, 4644 written\n",
      "2020-02-10 16:02:52,661 - Phyluce - INFO - ------ Working on Schistosoma_mattheei_GCA_900617995 genome -----\n",
      "2020-02-10 16:02:52,662 - Phyluce - INFO - Reading Schistosoma_mattheei_GCA_900617995 genome\n",
      "2020-02-10 16:03:19,830 - Phyluce - INFO - Schistosoma_mattheei_GCA_900617995: 5405 uces, 733 dupes, 4672 non-dupes, 2 orient drop, 12 length drop, 4658 written\n",
      "2020-02-10 16:03:19,830 - Phyluce - INFO - ---- Working on Trichobilharzia_regenti_GCA_900618515 genome ----\n",
      "2020-02-10 16:03:19,831 - Phyluce - INFO - Reading Trichobilharzia_regenti_GCA_900618515 genome\n",
      "2020-02-10 16:03:47,775 - Phyluce - INFO - Trichobilharzia_regenti_GCA_900618515: 4791 uces, 769 dupes, 4022 non-dupes, 2 orient drop, 3 length drop, 4016 written\n",
      "2020-02-10 16:03:47,775 - Phyluce - INFO - ------ Working on Schistosoma_mansoni_GCA_000237925 genome ------\n",
      "2020-02-10 16:03:47,776 - Phyluce - INFO - Reading Schistosoma_mansoni_GCA_000237925 genome\n",
      "2020-02-10 16:04:25,482 - Phyluce - INFO - Schistosoma_mansoni_GCA_000237925: 5435 uces, 747 dupes, 4688 non-dupes, 93 orient drop, 120 length drop, 4475 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/trematoda/extract_probes_from_group/probe_fasta --output results/phyluce/trematoda/extract_probes_from_group/multifastas.sqlite --base-taxon Schistosoma_mansoni_GCA_000237925;\n",
      "clonorchis_sinensis_gca_003604175...\n",
      "dicrocoelium_dendriticum_gca_000950715.\n",
      "echinostoma_caproni_gca_900618425...\n",
      "fasciola_gigantica_gca_006461475...\n",
      "fasciola_hepatica_gca_900302435...\n",
      "fasciolopsis_buski_gca_008360955...\n",
      "opisthorchis_felineus_gca_004794785...\n",
      "opisthorchis_viverrini_gcf_000715545...\n",
      "paragonimus_westermani_gca_008508345...\n",
      "schistosoma_bovis_gca_003958945....\n",
      "schistosoma_curassoni_gca_900618015.....\n",
      "schistosoma_haematobium_gcf_000699445.....\n",
      "schistosoma_japonicum_gca_006368765.....\n",
      "schistosoma_margrebowiei_gca_900618395.....\n",
      "schistosoma_mattheei_gca_900617995.....\n",
      "trichobilharzia_regenti_gca_900618515.....\n",
      "schistosoma_mansoni_gca_000237925.....\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/trematoda/extract_probes_from_group/multifastas.sqlite --base-taxon Schistosoma_mansoni_GCA_000237925;\n",
      "Loci shared by 0 taxa:\t5,270.0\n",
      "Loci shared by 1 taxa:\t5,270.0\n",
      "Loci shared by 2 taxa:\t5,132.0\n",
      "Loci shared by 3 taxa:\t5,048.0\n",
      "Loci shared by 4 taxa:\t4,991.0\n",
      "Loci shared by 5 taxa:\t4,924.0\n",
      "Loci shared by 6 taxa:\t4,827.0\n",
      "Loci shared by 7 taxa:\t4,625.0\n",
      "Loci shared by 8 taxa:\t4,144.0\n",
      "Loci shared by 9 taxa:\t3,453.0\n",
      "Loci shared by 10 taxa:\t3,213.0\n",
      "Loci shared by 11 taxa:\t3,013.0\n",
      "Loci shared by 12 taxa:\t2,772.0\n",
      "Loci shared by 13 taxa:\t2,465.0\n",
      "Loci shared by 14 taxa:\t2,116.0\n",
      "Loci shared by 15 taxa:\t1,599.0\n",
      "Loci shared by 16 taxa:\t850.0\n",
      "Loci shared by 17 taxa:\t146.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/trematoda/extract_probes_from_group/multifastas.sqlite  --base-taxon Schistosoma_mansoni_GCA_000237925 --output results/phyluce/trematoda/extract_probes_from_group/Schistosoma_mansoni_GCA_000237925+6-back-to-17.conf --specific-counts 17;\n",
      "Counter({'schistosoma_mansoni_gca_000237925': 146, 'trichobilharzia_regenti_gca_900618515': 146, 'schistosoma_mattheei_gca_900617995': 146, 'fasciola_hepatica_gca_900302435': 146, 'schistosoma_japonicum_gca_006368765': 146, 'fasciola_gigantica_gca_006461475': 146, 'schistosoma_margrebowiei_gca_900618395': 146, 'dicrocoelium_dendriticum_gca_000950715': 146, 'opisthorchis_viverrini_gcf_000715545': 146, 'fasciolopsis_buski_gca_008360955': 146, 'paragonimus_westermani_gca_008508345': 146, 'echinostoma_caproni_gca_900618425': 146, 'clonorchis_sinensis_gca_003604175': 146, 'schistosoma_bovis_gca_003958945': 146, 'schistosoma_curassoni_gca_900618015': 146, 'schistosoma_haematobium_gcf_000699445': 146, 'opisthorchis_felineus_gca_004794785': 146})\n",
      "Total loci = 146\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(17)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/trematoda/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/trematoda/extract_probes_from_group/Schistosoma_mansoni_GCA_000237925+6-back-to-17.conf --probe-prefix uce_trematoda_ --designer rnplattii --design trematoda_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGGGGGGGGGGGGGGGNNNNGGGGGGGGGGGNNGGGGGGGGGGGGGGNGNNGGGGGGGGGGGGGGGGNNGGGGGGGGGGGGNNGGGGGGGGGGGGGGGGGNNGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 146\n",
      "Probe Count = 4821\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list.fasta --query results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Mon Feb 10, 2020  16:04:48\n",
      "Ended:  Mon Feb 10, 2020  16:04:56\n",
      "Time for execution:  0.118803016345 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list.fasta --lastz results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_trematoda_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 4820 fasta sequences.  Filtered 0 duplicates. Kept 4821.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/trematoda/final_probe_design/trematoda_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Mon Feb 10 16:07:30 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 4821\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 385680\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3956764\n",
      "Max number of word counting entries: 89066764\n",
      "\n",
      "# comparing sequences from          0  to        803\n",
      "---------- new table with      518 representatives\n",
      "# comparing sequences from        803  to       1472\n",
      "----------     71 remaining sequences to the next cycle\n",
      "---------- new table with      380 representatives\n",
      "# comparing sequences from       1401  to       1971\n",
      "----------     36 remaining sequences to the next cycle\n",
      "---------- new table with      341 representatives\n",
      "# comparing sequences from       1935  to       2416\n",
      "----------    156 remaining sequences to the next cycle\n",
      "---------- new table with      213 representatives\n",
      "# comparing sequences from       2260  to       2686\n",
      "....................---------- new table with      271 representatives\n",
      "# comparing sequences from       2686  to       3041\n",
      "----------    112 remaining sequences to the next cycle\n",
      "---------- new table with      155 representatives\n",
      "# comparing sequences from       2929  to       3244\n",
      "100.0%---------- new table with      192 representatives\n",
      "# comparing sequences from       3244  to       3506\n",
      "----------     26 remaining sequences to the next cycle\n",
      "---------- new table with      148 representatives\n",
      "# comparing sequences from       3480  to       3703\n",
      "99.9%---------- new table with      150 representatives\n",
      "# comparing sequences from       3703  to       3889\n",
      "----------     32 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       3857  to       4821\n",
      ".....................---------- new table with      610 representatives\n",
      "\n",
      "     4821  finished       3078  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.96\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Tremellales"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'tremellales'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Bullera_alba_GCA_001600095\",\n",
    "               \"Cryptococcus_amylolentus_GCF_001720205\",\n",
    "               \"Cryptococcus_depauperatus_GCA_001720245\",\n",
    "               \"Cryptococcus_floricola_GCA_006352305\",\n",
    "               \"Cryptococcus_gattii_GCF_000185945\",\n",
    "               \"Cryptococcus_sp_GCA_001600855\",\n",
    "               \"Cryptococcus_wingfieldii_GCF_001720155\",\n",
    "               \"Dioszegia_aurantiaca_GCA_001600655\",\n",
    "               \"Dioszegia_crocea_GCA_001600615\",\n",
    "               \"Kockovaella_imperatae_GCF_002102565\",\n",
    "               \"Kwoniella_bestiolae_GCF_000512585\",\n",
    "               \"Kwoniella_dejecticola_GCF_000512565\",\n",
    "               \"Kwoniella_heveanensis_GCA_000507405\",\n",
    "               \"Kwoniella_mangrovensis_GCF_000507465\",\n",
    "               \"Kwoniella_pini_GCF_000512605\",\n",
    "               \"Kwoniella_shandongensis_GCF_008629635\",\n",
    "               \"Naematelia_encephala_GCA_002105065\",\n",
    "               \"Papiliotrema_flavescens_GCA_000442785\",\n",
    "               \"Papiliotrema_laurentii_GCA_000738825\",\n",
    "               \"Phaeotremella_fagi_GCA_001599715\",\n",
    "               \"Phaeotremella_skinneri_GCA_001599695\",\n",
    "               \"Saitozyma_podzolica_GCA_003942215\",\n",
    "               \"Tremella_fuciformis_GCA_000987905\",\n",
    "               \"Tremella_mesenterica_GCF_000271645\" ]\n",
    "                    \n",
    "reference_taxon = \"Cryptococcus_neoformans_GCF_000091045\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001600095.1_JCM_2954_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6113863 bytes  4075936.67 bytes/sec\n",
      "total size is 6112250  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001720205.1_Cryp_amyl_CBS6039_V3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6405771 bytes  4270542.00 bytes/sec\n",
      "total size is 6404088  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001720245.1_Filo_depa_CBS7855_V2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 4930190 bytes  3286821.33 bytes/sec\n",
      "total size is 4928867  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006352305.1_ASM635230v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6734119 bytes  4489440.67 bytes/sec\n",
      "total size is 6732365  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000185945.1_ASM18594v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5571422 bytes  3714309.33 bytes/sec\n",
      "total size is 5569957  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000091045.1_ASM9104v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5990061 bytes  2396041.20 bytes/sec\n",
      "total size is 5988493  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001600855.1_JCM_24511_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8854292 bytes  5902889.33 bytes/sec\n",
      "total size is 8852005  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001720155.1_Tsuc_wing_CBS7118_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6313195 bytes  4208824.67 bytes/sec\n",
      "total size is 6311536  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001600655.1_JCM_2956_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6073435 bytes  4048984.67 bytes/sec\n",
      "total size is 6071830  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001600615.1_JCM_2961_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6491075 bytes  4327411.33 bytes/sec\n",
      "total size is 6489366  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002102565.1_Kocim1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 5472348 bytes  2188956.00 bytes/sec\n",
      "total size is 5470911  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000512585.2_Cryp_best_CBS10118_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7754417 bytes  3101783.60 bytes/sec\n",
      "total size is 7752405  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000512565.2_Cryp_deje_CBS10117_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7540289 bytes  3016132.40 bytes/sec\n",
      "total size is 7538325  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000507405.3_Cryp_heve_BCC8398_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8119023 bytes  3247626.00 bytes/sec\n",
      "total size is 8116924  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000507465.3_Kwon_mang_CBS8507_V2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7184782 bytes  2873929.60 bytes/sec\n",
      "total size is 7182907  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000512605.2_Cryp_pinu_CBS10737_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6585761 bytes  2634321.20 bytes/sec\n",
      "total size is 6584037  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008629635.1_Kwon_shan_CBS_12478_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6760899 bytes  2704376.40 bytes/sec\n",
      "total size is 6759126  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002105065.1_Treen1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6242426 bytes  4161645.33 bytes/sec\n",
      "total size is 6240797  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000442785.1_Cf_30_300r_Split10plusN_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7276106 bytes  2910459.20 bytes/sec\n",
      "total size is 7274212  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000738825.1_ASM73882v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6070156 bytes  2428079.20 bytes/sec\n",
      "total size is 6068563  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001599715.1_JCM_13614_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7127917 bytes  2851183.60 bytes/sec\n",
      "total size is 7126055  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001599695.1_JCM_9039_assembly_v001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6495883 bytes  4330616.67 bytes/sec\n",
      "total size is 6494174  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003942215.1_ASM394221v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 9470394 bytes  3788174.40 bytes/sec\n",
      "total size is 9467975  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000987905.1_ASM98790v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7531910 bytes  3012780.80 bytes/sec\n",
      "total size is 7529965  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000271645.1_Treme1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8795164 bytes  5863470.67 bytes/sec\n",
      "total size is 8792910  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486124 (\"sim_Bullera_alba_GCA_001600095\") has been submitted\n",
      "Your job 5486125 (\"sim_Cryptococcus_amylolentus_GCF_001720205\") has been submitted\n",
      "Your job 5486126 (\"sim_Cryptococcus_depauperatus_GCA_001720245\") has been submitted\n",
      "Your job 5486127 (\"sim_Cryptococcus_floricola_GCA_006352305\") has been submitted\n",
      "Your job 5486128 (\"sim_Cryptococcus_gattii_GCF_000185945\") has been submitted\n",
      "Your job 5486129 (\"sim_Cryptococcus_sp_GCA_001600855\") has been submitted\n",
      "Your job 5486130 (\"sim_Cryptococcus_wingfieldii_GCF_001720155\") has been submitted\n",
      "Your job 5486131 (\"sim_Dioszegia_aurantiaca_GCA_001600655\") has been submitted\n",
      "Your job 5486132 (\"sim_Dioszegia_crocea_GCA_001600615\") has been submitted\n",
      "Your job 5486133 (\"sim_Kockovaella_imperatae_GCF_002102565\") has been submitted\n",
      "Your job 5486134 (\"sim_Kwoniella_bestiolae_GCF_000512585\") has been submitted\n",
      "Your job 5486135 (\"sim_Kwoniella_dejecticola_GCF_000512565\") has been submitted\n",
      "Your job 5486136 (\"sim_Kwoniella_heveanensis_GCA_000507405\") has been submitted\n",
      "Your job 5486137 (\"sim_Kwoniella_mangrovensis_GCF_000507465\") has been submitted\n",
      "Your job 5486138 (\"sim_Kwoniella_pini_GCF_000512605\") has been submitted\n",
      "Your job 5486139 (\"sim_Kwoniella_shandongensis_GCF_008629635\") has been submitted\n",
      "Your job 5486140 (\"sim_Naematelia_encephala_GCA_002105065\") has been submitted\n",
      "Your job 5486141 (\"sim_Papiliotrema_flavescens_GCA_000442785\") has been submitted\n",
      "Your job 5486142 (\"sim_Papiliotrema_laurentii_GCA_000738825\") has been submitted\n",
      "Your job 5486143 (\"sim_Phaeotremella_fagi_GCA_001599715\") has been submitted\n",
      "Your job 5486144 (\"sim_Phaeotremella_skinneri_GCA_001599695\") has been submitted\n",
      "Your job 5486145 (\"sim_Saitozyma_podzolica_GCA_003942215\") has been submitted\n",
      "Your job 5486146 (\"sim_Tremella_fuciformis_GCA_000987905\") has been submitted\n",
      "Your job 5486147 (\"sim_Tremella_mesenterica_GCF_000271645\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/tremellales/cleaned_genomes/Cryptococcus_neoformans_GCF_000091045_formatted.fas path=results/phyluce/tremellales/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486148 (\"map_Bullera_alba_GCA_001600095\") has been submitted\n",
      "Your job 5486149 (\"map_Cryptococcus_amylolentus_GCF_001720205\") has been submitted\n",
      "Your job 5486150 (\"map_Cryptococcus_depauperatus_GCA_001720245\") has been submitted\n",
      "Your job 5486151 (\"map_Cryptococcus_floricola_GCA_006352305\") has been submitted\n",
      "Your job 5486152 (\"map_Cryptococcus_gattii_GCF_000185945\") has been submitted\n",
      "Your job 5486153 (\"map_Cryptococcus_sp_GCA_001600855\") has been submitted\n",
      "Your job 5486154 (\"map_Cryptococcus_wingfieldii_GCF_001720155\") has been submitted\n",
      "Your job 5486155 (\"map_Dioszegia_aurantiaca_GCA_001600655\") has been submitted\n",
      "Your job 5486156 (\"map_Dioszegia_crocea_GCA_001600615\") has been submitted\n",
      "Your job 5486157 (\"map_Kockovaella_imperatae_GCF_002102565\") has been submitted\n",
      "Your job 5486158 (\"map_Kwoniella_bestiolae_GCF_000512585\") has been submitted\n",
      "Your job 5486159 (\"map_Kwoniella_dejecticola_GCF_000512565\") has been submitted\n",
      "Your job 5486160 (\"map_Kwoniella_heveanensis_GCA_000507405\") has been submitted\n",
      "Your job 5486161 (\"map_Kwoniella_mangrovensis_GCF_000507465\") has been submitted\n",
      "Your job 5486162 (\"map_Kwoniella_pini_GCF_000512605\") has been submitted\n",
      "Your job 5486163 (\"map_Kwoniella_shandongensis_GCF_008629635\") has been submitted\n",
      "Your job 5486164 (\"map_Naematelia_encephala_GCA_002105065\") has been submitted\n",
      "Your job 5486165 (\"map_Papiliotrema_flavescens_GCA_000442785\") has been submitted\n",
      "Your job 5486166 (\"map_Papiliotrema_laurentii_GCA_000738825\") has been submitted\n",
      "Your job 5486167 (\"map_Phaeotremella_fagi_GCA_001599715\") has been submitted\n",
      "Your job 5486168 (\"map_Phaeotremella_skinneri_GCA_001599695\") has been submitted\n",
      "Your job 5486169 (\"map_Saitozyma_podzolica_GCA_003942215\") has been submitted\n",
      "Your job 5486170 (\"map_Tremella_fuciformis_GCA_000987905\") has been submitted\n",
      "Your job 5486171 (\"map_Tremella_mesenterica_GCF_000271645\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486172 (\"merge_Bullera_alba_GCA_001600095\") has been submitted\n",
      "Your job 5486173 (\"merge_Cryptococcus_amylolentus_GCF_001720205\") has been submitted\n",
      "Your job 5486174 (\"merge_Cryptococcus_depauperatus_GCA_001720245\") has been submitted\n",
      "Your job 5486175 (\"merge_Cryptococcus_floricola_GCA_006352305\") has been submitted\n",
      "Your job 5486176 (\"merge_Cryptococcus_gattii_GCF_000185945\") has been submitted\n",
      "Your job 5486177 (\"merge_Cryptococcus_sp_GCA_001600855\") has been submitted\n",
      "Your job 5486178 (\"merge_Cryptococcus_wingfieldii_GCF_001720155\") has been submitted\n",
      "Your job 5486179 (\"merge_Dioszegia_aurantiaca_GCA_001600655\") has been submitted\n",
      "Your job 5486180 (\"merge_Dioszegia_crocea_GCA_001600615\") has been submitted\n",
      "Your job 5486181 (\"merge_Kockovaella_imperatae_GCF_002102565\") has been submitted\n",
      "Your job 5486182 (\"merge_Kwoniella_bestiolae_GCF_000512585\") has been submitted\n",
      "Your job 5486183 (\"merge_Kwoniella_dejecticola_GCF_000512565\") has been submitted\n",
      "Your job 5486184 (\"merge_Kwoniella_heveanensis_GCA_000507405\") has been submitted\n",
      "Your job 5486185 (\"merge_Kwoniella_mangrovensis_GCF_000507465\") has been submitted\n",
      "Your job 5486186 (\"merge_Kwoniella_pini_GCF_000512605\") has been submitted\n",
      "Your job 5486187 (\"merge_Kwoniella_shandongensis_GCF_008629635\") has been submitted\n",
      "Your job 5486188 (\"merge_Naematelia_encephala_GCA_002105065\") has been submitted\n",
      "Your job 5486189 (\"merge_Papiliotrema_flavescens_GCA_000442785\") has been submitted\n",
      "Your job 5486190 (\"merge_Papiliotrema_laurentii_GCA_000738825\") has been submitted\n",
      "Your job 5486191 (\"merge_Phaeotremella_fagi_GCA_001599715\") has been submitted\n",
      "Your job 5486192 (\"merge_Phaeotremella_skinneri_GCA_001599695\") has been submitted\n",
      "Your job 5486193 (\"merge_Saitozyma_podzolica_GCA_003942215\") has been submitted\n",
      "Your job 5486194 (\"merge_Tremella_fuciformis_GCA_000987905\") has been submitted\n",
      "Your job 5486195 (\"merge_Tremella_mesenterica_GCF_000271645\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 176 sequences from Bullera_alba_GCA_001600095_merged.bed.  Filtered 129 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 47.\n",
      "Screened 940 sequences from Cryptococcus_amylolentus_GCF_001720205_merged.bed.  Filtered 679 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 261.\n",
      "Screened 290 sequences from Cryptococcus_depauperatus_GCA_001720245_merged.bed.  Filtered 236 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 54.\n",
      "Screened 938 sequences from Cryptococcus_floricola_GCA_006352305_merged.bed.  Filtered 666 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 272.\n",
      "Screened 33813 sequences from Cryptococcus_gattii_GCF_000185945_merged.bed.  Filtered 15232 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 18581.\n",
      "Screened 225 sequences from Cryptococcus_sp_GCA_001600855_merged.bed.  Filtered 173 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 953 sequences from Cryptococcus_wingfieldii_GCF_001720155_merged.bed.  Filtered 687 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 266.\n",
      "Screened 151 sequences from Dioszegia_aurantiaca_GCA_001600655_merged.bed.  Filtered 116 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 35.\n",
      "Screened 164 sequences from Dioszegia_crocea_GCA_001600615_merged.bed.  Filtered 132 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 32.\n",
      "Screened 88 sequences from Kockovaella_imperatae_GCF_002102565_merged.bed.  Filtered 70 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 18.\n",
      "Screened 170 sequences from Kwoniella_bestiolae_GCF_000512585_merged.bed.  Filtered 146 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 24.\n",
      "Screened 150 sequences from Kwoniella_dejecticola_GCF_000512565_merged.bed.  Filtered 123 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 27.\n",
      "Screened 474 sequences from Kwoniella_heveanensis_GCA_000507405_merged.bed.  Filtered 357 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 122 sequences from Kwoniella_mangrovensis_GCF_000507465_merged.bed.  Filtered 106 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 16.\n",
      "Screened 76 sequences from Kwoniella_pini_GCF_000512605_merged.bed.  Filtered 63 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 709 sequences from Kwoniella_shandongensis_GCF_008629635_merged.bed.  Filtered 530 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 179.\n",
      "Screened 137 sequences from Naematelia_encephala_GCA_002105065_merged.bed.  Filtered 111 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 26.\n",
      "Screened 214 sequences from Papiliotrema_flavescens_GCA_000442785_merged.bed.  Filtered 162 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 285 sequences from Papiliotrema_laurentii_GCA_000738825_merged.bed.  Filtered 216 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 30 sequences from Phaeotremella_fagi_GCA_001599715_merged.bed.  Filtered 26 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 127 sequences from Phaeotremella_skinneri_GCA_001599695_merged.bed.  Filtered 99 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 28.\n",
      "Screened 251 sequences from Saitozyma_podzolica_GCA_003942215_merged.bed.  Filtered 201 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 50.\n",
      "Screened 151 sequences from Tremella_fuciformis_GCA_000987905_merged.bed.  Filtered 109 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 42.\n",
      "Screened 36 sequences from Tremella_mesenterica_GCF_000271645_merged.bed.  Filtered 23 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/tremellales/initial_intervals/Tremella_mesenterica_GCF_000271645_merged.bed --twobit results/phyluce/tremellales/cleaned_genomes/Cryptococcus_neoformans_GCF_000091045_formatted.2bit --output results/phyluce/tremellales/initial_intervals/Tremella_mesenterica_GCF_000271645_stripped.bed;\n",
      "bullera_alba_gca_001600095.\n",
      "cryptococcus_amylolentus_gcf_001720205.\n",
      "cryptococcus_depauperatus_gca_001720245.\n",
      "cryptococcus_floricola_gca_006352305.\n",
      "cryptococcus_gattii_gcf_000185945...................\n",
      "cryptococcus_sp_gca_001600855.\n",
      "cryptococcus_wingfieldii_gcf_001720155.\n",
      "dioszegia_aurantiaca_gca_001600655.\n",
      "dioszegia_crocea_gca_001600615.\n",
      "kockovaella_imperatae_gcf_002102565.\n",
      "kwoniella_bestiolae_gcf_000512585.\n",
      "kwoniella_dejecticola_gcf_000512565.\n",
      "kwoniella_heveanensis_gca_000507405.\n",
      "kwoniella_mangrovensis_gcf_000507465.\n",
      "kwoniella_pini_gcf_000512605.\n",
      "kwoniella_shandongensis_gcf_008629635.\n",
      "naematelia_encephala_gca_002105065.\n",
      "papiliotrema_flavescens_gca_000442785.\n",
      "papiliotrema_laurentii_gca_000738825.\n",
      "phaeotremella_fagi_gca_001599715.\n",
      "phaeotremella_skinneri_gca_001599695.\n",
      "saitozyma_podzolica_gca_003942215.\n",
      "tremella_fuciformis_gca_000987905.\n",
      "tremella_mesenterica_gcf_000271645.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/tremellales/initial_intervals/tremellales-to-Cryptococcus_neoformans_GCF_000091045.sqlite --base-taxon Cryptococcus_neoformans_GCF_000091045\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 0 taxa:\t18,750.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 1 taxa:\t18,750.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 2 taxa:\t398.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 3 taxa:\t306.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 4 taxa:\t254.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 5 taxa:\t169.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 6 taxa:\t126.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 7 taxa:\t98.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 8 taxa:\t76.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 9 taxa:\t62.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 10 taxa:\t49.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 11 taxa:\t39.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 12 taxa:\t27.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 13 taxa:\t22.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 14 taxa:\t17.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 15 taxa:\t14.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 16 taxa:\t9.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 17 taxa:\t8.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 18 taxa:\t7.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 19 taxa:\t4.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 20 taxa:\t3.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 21 taxa:\t2.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 22 taxa:\t0.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 23 taxa:\t0.0\n",
      "Loci shared by Cryptococcus_neoformans_GCF_000091045 + 24 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/tremellales/initial_intervals/tremellales-to-Cryptococcus_neoformans_GCF_000091045.sqlite --base-taxon Cryptococcus_neoformans_GCF_000091045 --output results/phyluce/tremellales/initial_intervals/Cryptococcus_neoformans_GCF_000091045_+2.bed --specific-counts 2;\n",
      "Counter({'cryptococcus_gattii_gcf_000185945': 374, 'cryptococcus_floricola_gca_006352305': 280, 'cryptococcus_wingfieldii_gcf_001720155': 269, 'cryptococcus_amylolentus_gcf_001720205': 261, 'kwoniella_shandongensis_gcf_008629635': 148, 'kwoniella_heveanensis_gca_000507405': 123, 'papiliotrema_laurentii_gca_000738825': 72, 'cryptococcus_depauperatus_gca_001720245': 55, 'saitozyma_podzolica_gca_003942215': 55, 'cryptococcus_sp_gca_001600855': 54, 'papiliotrema_flavescens_gca_000442785': 53, 'bullera_alba_gca_001600095': 47, 'tremella_fuciformis_gca_000987905': 45, 'dioszegia_aurantiaca_gca_001600655': 37, 'dioszegia_crocea_gca_001600615': 35, 'naematelia_encephala_gca_002105065': 31, 'phaeotremella_skinneri_gca_001599695': 30, 'kwoniella_dejecticola_gcf_000512565': 27, 'kwoniella_bestiolae_gcf_000512585': 24, 'kockovaella_imperatae_gcf_002102565': 21, 'kwoniella_mangrovensis_gcf_000507465': 18, 'kwoniella_pini_gcf_000512605': 13, 'tremella_mesenterica_gcf_000271645': 12, 'phaeotremella_fagi_gca_001599715': 4})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 2\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/tremellales/initial_intervals/Cryptococcus_neoformans_GCF_000091045_+2.bed --twobit results/phyluce/tremellales/cleaned_genomes/Cryptococcus_neoformans_GCF_000091045_formatted.2bit --buffer-to 160 --output results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2.fasta;\n",
      "Screened 398 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 398.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2.fasta --probe-prefix uce_tremellales_ --design tremellales_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "\n",
      "\n",
      "\n",
      "Conserved locus count = 398\n",
      "Probe Count = 796\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas --query results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes_vself.lastz;\n",
      "Started:  Mon Feb 10, 2020  16:10:30\n",
      "Ended:  Mon Feb 10, 2020  16:10:30\n",
      "Time for execution:  0.0063698331515 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas                        --lastz results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes_vself.lastz                       --probe-prefix=uce_tremellales_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 795 fasta sequences.  Filtered 15 duplicates. Kept 766.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/tremellales/validate_intervals/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas --scaffoldlist Bullera_alba_GCA_001600095 Cryptococcus_amylolentus_GCF_001720205 Cryptococcus_depauperatus_GCA_001720245 Cryptococcus_floricola_GCA_006352305 Cryptococcus_gattii_GCF_000185945 Cryptococcus_sp_GCA_001600855 Cryptococcus_wingfieldii_GCF_001720155 Dioszegia_aurantiaca_GCA_001600655 Dioszegia_crocea_GCA_001600615 Kockovaella_imperatae_GCF_002102565 Kwoniella_bestiolae_GCF_000512585 Kwoniella_dejecticola_GCF_000512565 Kwoniella_heveanensis_GCA_000507405 Kwoniella_mangrovensis_GCF_000507465 Kwoniella_pini_GCF_000512605 Kwoniella_shandongensis_GCF_008629635 Naematelia_encephala_GCA_002105065 Papiliotrema_flavescens_GCA_000442785 Papiliotrema_laurentii_GCA_000738825 Phaeotremella_fagi_GCA_001599715 Phaeotremella_skinneri_GCA_001599695 Saitozyma_podzolica_GCA_003942215 Tremella_fuciformis_GCA_000987905 Tremella_mesenterica_GCF_000271645 Cryptococcus_neoformans_GCF_000091045 --genome-base-path results/phyluce/tremellales/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/tremellales/validate_intervals/tremellales-to-Cryptococcus_neoformans_GCF_000091045.sqlite --output results/phyluce/tremellales/validate_intervals/lastz/;\n",
      "\n",
      "Running against Bullera_alba_GCA_001600095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp4OyIy_.fasta\n",
      "\t/tmp/tmpiutE_Q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa9gflF.lastz\n",
      "\t/tmp/tmpdD088p.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Bullera_alba_GCA_001600095.lastz\n",
      "Creating Bullera_alba_GCA_001600095 table\n",
      "Inserting data to Bullera_alba_GCA_001600095 table\n",
      "\n",
      "Running against Cryptococcus_amylolentus_GCF_001720205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpNmV9O5.fasta\n",
      "\t/tmp/tmpc0Pf7A.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdxe5El.lastz\n",
      "\t/tmp/tmpGxokzj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_amylolentus_GCF_001720205.lastz\n",
      "Creating Cryptococcus_amylolentus_GCF_001720205 table\n",
      "Inserting data to Cryptococcus_amylolentus_GCF_001720205 table\n",
      "\n",
      "Running against Cryptococcus_depauperatus_GCA_001720245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp5kFp7W.fasta\n",
      "\t/tmp/tmprhIzF1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKHmZj9.lastz\n",
      "\t/tmp/tmpjN1rdL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_depauperatus_GCA_001720245.lastz\n",
      "Creating Cryptococcus_depauperatus_GCA_001720245 table\n",
      "Inserting data to Cryptococcus_depauperatus_GCA_001720245 table\n",
      "\n",
      "Running against Cryptococcus_floricola_GCA_006352305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpQMYf1q.fasta\n",
      "\t/tmp/tmpK4R8Z_.fasta\n",
      "\t/tmp/tmpeOUhyE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOIWswi.lastz\n",
      "\t/tmp/tmpEq9Kao.lastz\n",
      "\t/tmp/tmpmHM5e7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_floricola_GCA_006352305.lastz\n",
      "Creating Cryptococcus_floricola_GCA_006352305 table\n",
      "Inserting data to Cryptococcus_floricola_GCA_006352305 table\n",
      "\n",
      "Running against Cryptococcus_gattii_GCF_000185945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpLeyBqT.fasta\n",
      "\t/tmp/tmp8XnwWG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpc297aa.lastz\n",
      "\t/tmp/tmpQUT6jl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_gattii_GCF_000185945.lastz\n",
      "Creating Cryptococcus_gattii_GCF_000185945 table\n",
      "Inserting data to Cryptococcus_gattii_GCF_000185945 table\n",
      "\n",
      "Running against Cryptococcus_sp_GCA_001600855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpDs1fQc.fasta\n",
      "\t/tmp/tmps9Kc43.fasta\n",
      "\t/tmp/tmp3jDJZO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFhvaY5.lastz\n",
      "\t/tmp/tmpm_6yLS.lastz\n",
      "\t/tmp/tmpdgettt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_sp_GCA_001600855.lastz\n",
      "Creating Cryptococcus_sp_GCA_001600855 table\n",
      "Inserting data to Cryptococcus_sp_GCA_001600855 table\n",
      "\n",
      "Running against Cryptococcus_wingfieldii_GCF_001720155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpNrr0FK.fasta\n",
      "\t/tmp/tmpANLsT7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBB4Ir_.lastz\n",
      "\t/tmp/tmpSpeoXL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_wingfieldii_GCF_001720155.lastz\n",
      "Creating Cryptococcus_wingfieldii_GCF_001720155 table\n",
      "Inserting data to Cryptococcus_wingfieldii_GCF_001720155 table\n",
      "\n",
      "Running against Dioszegia_aurantiaca_GCA_001600655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpsjYqhD.fasta\n",
      "\t/tmp/tmpQkEXfn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYPI0uJ.lastz\n",
      "\t/tmp/tmpzw66Gd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Dioszegia_aurantiaca_GCA_001600655.lastz\n",
      "Creating Dioszegia_aurantiaca_GCA_001600655 table\n",
      "Inserting data to Dioszegia_aurantiaca_GCA_001600655 table\n",
      "\n",
      "Running against Dioszegia_crocea_GCA_001600615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpTfZlOD.fasta\n",
      "\t/tmp/tmpOkSCtm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEcZQXf.lastz\n",
      "\t/tmp/tmpfocwWr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Dioszegia_crocea_GCA_001600615.lastz\n",
      "Creating Dioszegia_crocea_GCA_001600615 table\n",
      "Inserting data to Dioszegia_crocea_GCA_001600615 table\n",
      "\n",
      "Running against Kockovaella_imperatae_GCF_002102565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpHGvw8A.fasta\n",
      "\t/tmp/tmpfeAzj8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpApRm6z.lastz\n",
      "\t/tmp/tmpR1KAgM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kockovaella_imperatae_GCF_002102565.lastz\n",
      "Creating Kockovaella_imperatae_GCF_002102565 table\n",
      "Inserting data to Kockovaella_imperatae_GCF_002102565 table\n",
      "\n",
      "Running against Kwoniella_bestiolae_GCF_000512585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpWGN1_K.fasta\n",
      "\t/tmp/tmppxuJRZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpH5yRCW.lastz\n",
      "\t/tmp/tmppkLz98.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kwoniella_bestiolae_GCF_000512585.lastz\n",
      "Creating Kwoniella_bestiolae_GCF_000512585 table\n",
      "Inserting data to Kwoniella_bestiolae_GCF_000512585 table\n",
      "\n",
      "Running against Kwoniella_dejecticola_GCF_000512565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp8R_9i9.fasta\n",
      "\t/tmp/tmpASRt7w.fasta\n",
      "\t/tmp/tmpsuo8SU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa9oErW.lastz\n",
      "\t/tmp/tmpIFZ91q.lastz\n",
      "\t/tmp/tmpl5qyDa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kwoniella_dejecticola_GCF_000512565.lastz\n",
      "Creating Kwoniella_dejecticola_GCF_000512565 table\n",
      "Inserting data to Kwoniella_dejecticola_GCF_000512565 table\n",
      "\n",
      "Running against Kwoniella_heveanensis_GCA_000507405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpqbW5kd.fasta\n",
      "\t/tmp/tmpef2tTu.fasta\n",
      "\t/tmp/tmppguC2H.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8I6GpV.lastz\n",
      "\t/tmp/tmp9iyK06.lastz\n",
      "\t/tmp/tmpFPQGDu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kwoniella_heveanensis_GCA_000507405.lastz\n",
      "Creating Kwoniella_heveanensis_GCA_000507405 table\n",
      "Inserting data to Kwoniella_heveanensis_GCA_000507405 table\n",
      "\n",
      "Running against Kwoniella_mangrovensis_GCF_000507465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp7wGr3p.fasta\n",
      "\t/tmp/tmpaCaUWL.fasta\n",
      "\t/tmp/tmpxtIYCX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHqDuxM.lastz\n",
      "\t/tmp/tmpGPGXCE.lastz\n",
      "\t/tmp/tmpc5WoH2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kwoniella_mangrovensis_GCF_000507465.lastz\n",
      "Creating Kwoniella_mangrovensis_GCF_000507465 table\n",
      "Inserting data to Kwoniella_mangrovensis_GCF_000507465 table\n",
      "\n",
      "Running against Kwoniella_pini_GCF_000512605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmp0IiYvE.fasta\n",
      "\t/tmp/tmpW1X_79.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpY69o2u.lastz\n",
      "\t/tmp/tmpJVMIsJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kwoniella_pini_GCF_000512605.lastz\n",
      "Creating Kwoniella_pini_GCF_000512605 table\n",
      "Inserting data to Kwoniella_pini_GCF_000512605 table\n",
      "\n",
      "Running against Kwoniella_shandongensis_GCF_008629635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp58frNZ.fasta\n",
      "\t/tmp/tmpQFkqR9.fasta\n",
      "\t/tmp/tmpxe00zw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRhDOQj.lastz\n",
      "\t/tmp/tmpBGlgV7.lastz\n",
      "\t/tmp/tmpF2K2bB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Kwoniella_shandongensis_GCF_008629635.lastz\n",
      "Creating Kwoniella_shandongensis_GCF_008629635 table\n",
      "Inserting data to Kwoniella_shandongensis_GCF_008629635 table\n",
      "\n",
      "Running against Naematelia_encephala_GCA_002105065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpR1k3Cf.fasta\n",
      "\t/tmp/tmpIL925S.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJi31OU.lastz\n",
      "\t/tmp/tmpk6SoeC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Naematelia_encephala_GCA_002105065.lastz\n",
      "Creating Naematelia_encephala_GCA_002105065 table\n",
      "Inserting data to Naematelia_encephala_GCA_002105065 table\n",
      "\n",
      "Running against Papiliotrema_flavescens_GCA_000442785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpsMsuxN.fasta\n",
      "\t/tmp/tmpSc0dPB.fasta\n",
      "\t/tmp/tmpU5gKfI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_O3YqR.lastz\n",
      "\t/tmp/tmpKGo_cU.lastz\n",
      "\t/tmp/tmpCjlzpV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Papiliotrema_flavescens_GCA_000442785.lastz\n",
      "Creating Papiliotrema_flavescens_GCA_000442785 table\n",
      "Inserting data to Papiliotrema_flavescens_GCA_000442785 table\n",
      "\n",
      "Running against Papiliotrema_laurentii_GCA_000738825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpZoK2B3.fasta\n",
      "\t/tmp/tmpRJC0P9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWLW8aj.lastz\n",
      "\t/tmp/tmpoUBima.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Papiliotrema_laurentii_GCA_000738825.lastz\n",
      "Creating Papiliotrema_laurentii_GCA_000738825 table\n",
      "Inserting data to Papiliotrema_laurentii_GCA_000738825 table\n",
      "\n",
      "Running against Phaeotremella_fagi_GCA_001599715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpbsfr84.fasta\n",
      "\t/tmp/tmpuqVuaD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2ZLDAv.lastz\n",
      "\t/tmp/tmpLsiXfj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Phaeotremella_fagi_GCA_001599715.lastz\n",
      "Creating Phaeotremella_fagi_GCA_001599715 table\n",
      "Inserting data to Phaeotremella_fagi_GCA_001599715 table\n",
      "\n",
      "Running against Phaeotremella_skinneri_GCA_001599695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmptTS7yH.fasta\n",
      "\t/tmp/tmpsR_BXV.fasta\n",
      "\t/tmp/tmppme_Ee.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQh97Wi.lastz\n",
      "\t/tmp/tmpYCbKUs.lastz\n",
      "\t/tmp/tmpy50Hcc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Phaeotremella_skinneri_GCA_001599695.lastz\n",
      "Creating Phaeotremella_skinneri_GCA_001599695 table\n",
      "Inserting data to Phaeotremella_skinneri_GCA_001599695 table\n",
      "\n",
      "Running against Saitozyma_podzolica_GCA_003942215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpq7W0C5.fasta\n",
      "\t/tmp/tmpiaOYvL.fasta\n",
      "\t/tmp/tmpz85AdT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkriw6Q.lastz\n",
      "\t/tmp/tmpkJSFyQ.lastz\n",
      "\t/tmp/tmpe1FL3D.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Saitozyma_podzolica_GCA_003942215.lastz\n",
      "Creating Saitozyma_podzolica_GCA_003942215 table\n",
      "Inserting data to Saitozyma_podzolica_GCA_003942215 table\n",
      "\n",
      "Running against Tremella_fuciformis_GCA_000987905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpMyFDWJ.fasta\n",
      "\t/tmp/tmp8Wu5Nl.fasta\n",
      "\t/tmp/tmpBvEgjH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfLLj7j.lastz\n",
      "\t/tmp/tmpYN7iOM.lastz\n",
      "\t/tmp/tmpifgGDd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Tremella_fuciformis_GCA_000987905.lastz\n",
      "Creating Tremella_fuciformis_GCA_000987905 table\n",
      "Inserting data to Tremella_fuciformis_GCA_000987905 table\n",
      "\n",
      "Running against Tremella_mesenterica_GCF_000271645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmp8wN815.fasta\n",
      "\t/tmp/tmp17rIH4.fasta\n",
      "\t/tmp/tmpDjecE6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprJQScA.lastz\n",
      "\t/tmp/tmpqllADi.lastz\n",
      "\t/tmp/tmp3jI0Dk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Tremella_mesenterica_GCF_000271645.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Tremella_mesenterica_GCF_000271645 table\n",
      "Inserting data to Tremella_mesenterica_GCF_000271645 table\n",
      "\n",
      "Running against Cryptococcus_neoformans_GCF_000091045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 2 queries...\n",
      "\t/tmp/tmpFNEfOI.fasta\n",
      "\t/tmp/tmpuHIYpE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5lrPZE.lastz\n",
      "\t/tmp/tmpGhl1WS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/tremellales/validate_intervals/lastz/Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_Cryptococcus_neoformans_GCF_000091045.lastz\n",
      "Creating Cryptococcus_neoformans_GCF_000091045 table\n",
      "Inserting data to Cryptococcus_neoformans_GCF_000091045 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/tremellales/extract_probes_from_group/tremellales_genome.conf --lastz results/phyluce/tremellales/validate_intervals/lastz --probes 120 --probe-prefix uce_tremellales_ --name-pattern \"Cryptococcus_neoformans_GCF_000091045_+2_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/tremellales/extract_probes_from_group/probe_fasta;\n",
      "2020-02-10 16:12:14,884 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-10 16:12:14,887 - Phyluce - INFO - ---------- Working on Bullera_alba_GCA_001600095 genome ---------\n",
      "2020-02-10 16:12:14,888 - Phyluce - INFO - Reading Bullera_alba_GCA_001600095 genome\n",
      "2020-02-10 16:12:16,228 - Phyluce - INFO - Bullera_alba_GCA_001600095: 362 uces, 38 dupes, 324 non-dupes, 6 orient drop, 2 length drop, 316 written\n",
      "2020-02-10 16:12:16,228 - Phyluce - INFO - ---- Working on Cryptococcus_amylolentus_GCF_001720205 genome ---\n",
      "2020-02-10 16:12:16,235 - Phyluce - INFO - Reading Cryptococcus_amylolentus_GCF_001720205 genome\n",
      "2020-02-10 16:12:17,768 - Phyluce - INFO - Cryptococcus_amylolentus_GCF_001720205: 383 uces, 54 dupes, 329 non-dupes, 4 orient drop, 5 length drop, 320 written\n",
      "2020-02-10 16:12:17,769 - Phyluce - INFO - --- Working on Cryptococcus_depauperatus_GCA_001720245 genome ---\n",
      "2020-02-10 16:12:17,769 - Phyluce - INFO - Reading Cryptococcus_depauperatus_GCA_001720245 genome\n",
      "2020-02-10 16:12:19,250 - Phyluce - INFO - Cryptococcus_depauperatus_GCA_001720245: 383 uces, 53 dupes, 330 non-dupes, 0 orient drop, 1 length drop, 329 written\n",
      "2020-02-10 16:12:19,250 - Phyluce - INFO - ----- Working on Cryptococcus_floricola_GCA_006352305 genome ----\n",
      "2020-02-10 16:12:19,251 - Phyluce - INFO - Reading Cryptococcus_floricola_GCA_006352305 genome\n",
      "2020-02-10 16:12:21,098 - Phyluce - INFO - Cryptococcus_floricola_GCA_006352305: 393 uces, 54 dupes, 339 non-dupes, 5 orient drop, 14 length drop, 320 written\n",
      "2020-02-10 16:12:21,098 - Phyluce - INFO - ------ Working on Cryptococcus_gattii_GCF_000185945 genome ------\n",
      "2020-02-10 16:12:21,099 - Phyluce - INFO - Reading Cryptococcus_gattii_GCF_000185945 genome\n",
      "2020-02-10 16:12:22,671 - Phyluce - INFO - Cryptococcus_gattii_GCF_000185945: 385 uces, 56 dupes, 329 non-dupes, 1 orient drop, 5 length drop, 323 written\n",
      "2020-02-10 16:12:22,671 - Phyluce - INFO - -------- Working on Cryptococcus_sp_GCA_001600855 genome --------\n",
      "2020-02-10 16:12:22,672 - Phyluce - INFO - Reading Cryptococcus_sp_GCA_001600855 genome\n",
      "2020-02-10 16:12:23,923 - Phyluce - INFO - Cryptococcus_sp_GCA_001600855: 357 uces, 43 dupes, 314 non-dupes, 1 orient drop, 1 length drop, 312 written\n",
      "2020-02-10 16:12:23,923 - Phyluce - INFO - ---- Working on Cryptococcus_wingfieldii_GCF_001720155 genome ---\n",
      "2020-02-10 16:12:23,923 - Phyluce - INFO - Reading Cryptococcus_wingfieldii_GCF_001720155 genome\n",
      "2020-02-10 16:12:25,456 - Phyluce - INFO - Cryptococcus_wingfieldii_GCF_001720155: 379 uces, 56 dupes, 323 non-dupes, 4 orient drop, 4 length drop, 315 written\n",
      "2020-02-10 16:12:25,457 - Phyluce - INFO - ------ Working on Dioszegia_aurantiaca_GCA_001600655 genome -----\n",
      "2020-02-10 16:12:25,457 - Phyluce - INFO - Reading Dioszegia_aurantiaca_GCA_001600655 genome\n",
      "2020-02-10 16:12:26,751 - Phyluce - INFO - Dioszegia_aurantiaca_GCA_001600655: 365 uces, 44 dupes, 321 non-dupes, 2 orient drop, 1 length drop, 318 written\n",
      "2020-02-10 16:12:26,751 - Phyluce - INFO - -------- Working on Dioszegia_crocea_GCA_001600615 genome -------\n",
      "2020-02-10 16:12:26,768 - Phyluce - INFO - Reading Dioszegia_crocea_GCA_001600615 genome\n",
      "2020-02-10 16:12:28,148 - Phyluce - INFO - Dioszegia_crocea_GCA_001600615: 369 uces, 50 dupes, 319 non-dupes, 0 orient drop, 1 length drop, 318 written\n",
      "2020-02-10 16:12:28,149 - Phyluce - INFO - ----- Working on Kockovaella_imperatae_GCF_002102565 genome -----\n",
      "2020-02-10 16:12:28,149 - Phyluce - INFO - Reading Kockovaella_imperatae_GCF_002102565 genome\n",
      "2020-02-10 16:12:29,562 - Phyluce - INFO - Kockovaella_imperatae_GCF_002102565: 363 uces, 57 dupes, 306 non-dupes, 2 orient drop, 0 length drop, 304 written\n",
      "2020-02-10 16:12:29,562 - Phyluce - INFO - ------ Working on Kwoniella_bestiolae_GCF_000512585 genome ------\n",
      "2020-02-10 16:12:29,563 - Phyluce - INFO - Reading Kwoniella_bestiolae_GCF_000512585 genome\n",
      "2020-02-10 16:12:30,909 - Phyluce - INFO - Kwoniella_bestiolae_GCF_000512585: 369 uces, 43 dupes, 326 non-dupes, 3 orient drop, 1 length drop, 322 written\n",
      "2020-02-10 16:12:30,910 - Phyluce - INFO - ----- Working on Kwoniella_dejecticola_GCF_000512565 genome -----\n",
      "2020-02-10 16:12:30,918 - Phyluce - INFO - Reading Kwoniella_dejecticola_GCF_000512565 genome\n",
      "2020-02-10 16:12:32,294 - Phyluce - INFO - Kwoniella_dejecticola_GCF_000512565: 377 uces, 42 dupes, 335 non-dupes, 3 orient drop, 1 length drop, 331 written\n",
      "2020-02-10 16:12:32,294 - Phyluce - INFO - ----- Working on Kwoniella_heveanensis_GCA_000507405 genome -----\n",
      "2020-02-10 16:12:32,308 - Phyluce - INFO - Reading Kwoniella_heveanensis_GCA_000507405 genome\n",
      "2020-02-10 16:12:33,688 - Phyluce - INFO - Kwoniella_heveanensis_GCA_000507405: 372 uces, 51 dupes, 321 non-dupes, 1 orient drop, 0 length drop, 320 written\n",
      "2020-02-10 16:12:33,689 - Phyluce - INFO - ----- Working on Kwoniella_mangrovensis_GCF_000507465 genome ----\n",
      "2020-02-10 16:12:33,689 - Phyluce - INFO - Reading Kwoniella_mangrovensis_GCF_000507465 genome\n",
      "2020-02-10 16:12:35,037 - Phyluce - INFO - Kwoniella_mangrovensis_GCF_000507465: 373 uces, 45 dupes, 328 non-dupes, 2 orient drop, 0 length drop, 326 written\n",
      "2020-02-10 16:12:35,037 - Phyluce - INFO - --------- Working on Kwoniella_pini_GCF_000512605 genome --------\n",
      "2020-02-10 16:12:35,038 - Phyluce - INFO - Reading Kwoniella_pini_GCF_000512605 genome\n",
      "2020-02-10 16:12:36,437 - Phyluce - INFO - Kwoniella_pini_GCF_000512605: 374 uces, 41 dupes, 333 non-dupes, 1 orient drop, 2 length drop, 330 written\n",
      "2020-02-10 16:12:36,437 - Phyluce - INFO - ---- Working on Kwoniella_shandongensis_GCF_008629635 genome ----\n",
      "2020-02-10 16:12:36,438 - Phyluce - INFO - Reading Kwoniella_shandongensis_GCF_008629635 genome\n",
      "2020-02-10 16:12:37,876 - Phyluce - INFO - Kwoniella_shandongensis_GCF_008629635: 391 uces, 53 dupes, 338 non-dupes, 0 orient drop, 1 length drop, 337 written\n",
      "2020-02-10 16:12:37,876 - Phyluce - INFO - ------ Working on Naematelia_encephala_GCA_002105065 genome -----\n",
      "2020-02-10 16:12:37,893 - Phyluce - INFO - Reading Naematelia_encephala_GCA_002105065 genome\n",
      "2020-02-10 16:12:39,262 - Phyluce - INFO - Naematelia_encephala_GCA_002105065: 372 uces, 43 dupes, 329 non-dupes, 3 orient drop, 0 length drop, 326 written\n",
      "2020-02-10 16:12:39,263 - Phyluce - INFO - ---- Working on Papiliotrema_flavescens_GCA_000442785 genome ----\n",
      "2020-02-10 16:12:39,274 - Phyluce - INFO - Reading Papiliotrema_flavescens_GCA_000442785 genome\n",
      "2020-02-10 16:12:40,562 - Phyluce - INFO - Papiliotrema_flavescens_GCA_000442785: 373 uces, 38 dupes, 335 non-dupes, 0 orient drop, 0 length drop, 335 written\n",
      "2020-02-10 16:12:40,563 - Phyluce - INFO - ----- Working on Papiliotrema_laurentii_GCA_000738825 genome ----\n",
      "2020-02-10 16:12:40,580 - Phyluce - INFO - Reading Papiliotrema_laurentii_GCA_000738825 genome\n",
      "2020-02-10 16:12:41,976 - Phyluce - INFO - Papiliotrema_laurentii_GCA_000738825: 375 uces, 47 dupes, 328 non-dupes, 0 orient drop, 10 length drop, 318 written\n",
      "2020-02-10 16:12:41,976 - Phyluce - INFO - ------- Working on Phaeotremella_fagi_GCA_001599715 genome ------\n",
      "2020-02-10 16:12:41,981 - Phyluce - INFO - Reading Phaeotremella_fagi_GCA_001599715 genome\n",
      "2020-02-10 16:12:43,235 - Phyluce - INFO - Phaeotremella_fagi_GCA_001599715: 362 uces, 34 dupes, 328 non-dupes, 1 orient drop, 7 length drop, 320 written\n",
      "2020-02-10 16:12:43,235 - Phyluce - INFO - ----- Working on Phaeotremella_skinneri_GCA_001599695 genome ----\n",
      "2020-02-10 16:12:43,236 - Phyluce - INFO - Reading Phaeotremella_skinneri_GCA_001599695 genome\n",
      "2020-02-10 16:12:44,542 - Phyluce - INFO - Phaeotremella_skinneri_GCA_001599695: 363 uces, 43 dupes, 320 non-dupes, 0 orient drop, 3 length drop, 317 written\n",
      "2020-02-10 16:12:44,543 - Phyluce - INFO - ------ Working on Saitozyma_podzolica_GCA_003942215 genome ------\n",
      "2020-02-10 16:12:44,559 - Phyluce - INFO - Reading Saitozyma_podzolica_GCA_003942215 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-10 16:12:46,113 - Phyluce - INFO - Saitozyma_podzolica_GCA_003942215: 366 uces, 53 dupes, 313 non-dupes, 0 orient drop, 2 length drop, 311 written\n",
      "2020-02-10 16:12:46,114 - Phyluce - INFO - ------ Working on Tremella_fuciformis_GCA_000987905 genome ------\n",
      "2020-02-10 16:12:46,115 - Phyluce - INFO - Reading Tremella_fuciformis_GCA_000987905 genome\n",
      "2020-02-10 16:12:47,561 - Phyluce - INFO - Tremella_fuciformis_GCA_000987905: 369 uces, 57 dupes, 312 non-dupes, 0 orient drop, 0 length drop, 312 written\n",
      "2020-02-10 16:12:47,561 - Phyluce - INFO - ------ Working on Tremella_mesenterica_GCF_000271645 genome -----\n",
      "2020-02-10 16:12:47,571 - Phyluce - INFO - Reading Tremella_mesenterica_GCF_000271645 genome\n",
      "2020-02-10 16:12:48,943 - Phyluce - INFO - Tremella_mesenterica_GCF_000271645: 368 uces, 43 dupes, 325 non-dupes, 2 orient drop, 11 length drop, 312 written\n",
      "2020-02-10 16:12:48,943 - Phyluce - INFO - ---- Working on Cryptococcus_neoformans_GCF_000091045 genome ----\n",
      "2020-02-10 16:12:48,944 - Phyluce - INFO - Reading Cryptococcus_neoformans_GCF_000091045 genome\n",
      "2020-02-10 16:12:50,769 - Phyluce - INFO - Cryptococcus_neoformans_GCF_000091045: 398 uces, 52 dupes, 346 non-dupes, 1 orient drop, 21 length drop, 324 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/tremellales/extract_probes_from_group/probe_fasta --output results/phyluce/tremellales/extract_probes_from_group/multifastas.sqlite --base-taxon Cryptococcus_neoformans_GCF_000091045;\n",
      "bullera_alba_gca_001600095.\n",
      "cryptococcus_amylolentus_gcf_001720205.\n",
      "cryptococcus_depauperatus_gca_001720245.\n",
      "cryptococcus_floricola_gca_006352305.\n",
      "cryptococcus_gattii_gcf_000185945.\n",
      "cryptococcus_sp_gca_001600855.\n",
      "cryptococcus_wingfieldii_gcf_001720155.\n",
      "dioszegia_aurantiaca_gca_001600655.\n",
      "dioszegia_crocea_gca_001600615.\n",
      "kockovaella_imperatae_gcf_002102565.\n",
      "kwoniella_bestiolae_gcf_000512585.\n",
      "kwoniella_dejecticola_gcf_000512565.\n",
      "kwoniella_heveanensis_gca_000507405.\n",
      "kwoniella_mangrovensis_gcf_000507465.\n",
      "kwoniella_pini_gcf_000512605.\n",
      "kwoniella_shandongensis_gcf_008629635.\n",
      "naematelia_encephala_gca_002105065.\n",
      "papiliotrema_flavescens_gca_000442785.\n",
      "papiliotrema_laurentii_gca_000738825.\n",
      "phaeotremella_fagi_gca_001599715.\n",
      "phaeotremella_skinneri_gca_001599695.\n",
      "saitozyma_podzolica_gca_003942215.\n",
      "tremella_fuciformis_gca_000987905.\n",
      "tremella_mesenterica_gcf_000271645.\n",
      "cryptococcus_neoformans_gcf_000091045.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/tremellales/extract_probes_from_group/multifastas.sqlite --base-taxon Cryptococcus_neoformans_GCF_000091045;\n",
      "Loci shared by 0 taxa:\t383.0\n",
      "Loci shared by 1 taxa:\t383.0\n",
      "Loci shared by 2 taxa:\t378.0\n",
      "Loci shared by 3 taxa:\t370.0\n",
      "Loci shared by 4 taxa:\t364.0\n",
      "Loci shared by 5 taxa:\t356.0\n",
      "Loci shared by 6 taxa:\t351.0\n",
      "Loci shared by 7 taxa:\t346.0\n",
      "Loci shared by 8 taxa:\t342.0\n",
      "Loci shared by 9 taxa:\t338.0\n",
      "Loci shared by 10 taxa:\t335.0\n",
      "Loci shared by 11 taxa:\t334.0\n",
      "Loci shared by 12 taxa:\t332.0\n",
      "Loci shared by 13 taxa:\t329.0\n",
      "Loci shared by 14 taxa:\t325.0\n",
      "Loci shared by 15 taxa:\t324.0\n",
      "Loci shared by 16 taxa:\t320.0\n",
      "Loci shared by 17 taxa:\t317.0\n",
      "Loci shared by 18 taxa:\t312.0\n",
      "Loci shared by 19 taxa:\t311.0\n",
      "Loci shared by 20 taxa:\t306.0\n",
      "Loci shared by 21 taxa:\t299.0\n",
      "Loci shared by 22 taxa:\t284.0\n",
      "Loci shared by 23 taxa:\t263.0\n",
      "Loci shared by 24 taxa:\t238.0\n",
      "Loci shared by 25 taxa:\t159.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/tremellales/extract_probes_from_group/multifastas.sqlite  --base-taxon Cryptococcus_neoformans_GCF_000091045 --output results/phyluce/tremellales/extract_probes_from_group/Cryptococcus_neoformans_GCF_000091045+2-back-to-25.conf --specific-counts 25;\n",
      "Counter({'kwoniella_dejecticola_gcf_000512565': 159, 'cryptococcus_wingfieldii_gcf_001720155': 159, 'dioszegia_crocea_gca_001600615': 159, 'dioszegia_aurantiaca_gca_001600655': 159, 'cryptococcus_neoformans_gcf_000091045': 159, 'papiliotrema_flavescens_gca_000442785': 159, 'kwoniella_shandongensis_gcf_008629635': 159, 'kockovaella_imperatae_gcf_002102565': 159, 'phaeotremella_skinneri_gca_001599695': 159, 'kwoniella_bestiolae_gcf_000512585': 159, 'naematelia_encephala_gca_002105065': 159, 'cryptococcus_floricola_gca_006352305': 159, 'kwoniella_mangrovensis_gcf_000507465': 159, 'papiliotrema_laurentii_gca_000738825': 159, 'cryptococcus_amylolentus_gcf_001720205': 159, 'phaeotremella_fagi_gca_001599715': 159, 'cryptococcus_sp_gca_001600855': 159, 'cryptococcus_gattii_gcf_000185945': 159, 'kwoniella_pini_gcf_000512605': 159, 'kwoniella_heveanensis_gca_000507405': 159, 'saitozyma_podzolica_gca_003942215': 159, 'bullera_alba_gca_001600095': 159, 'tremella_fuciformis_gca_000987905': 159, 'tremella_mesenterica_gcf_000271645': 159, 'cryptococcus_depauperatus_gca_001720245': 159})\n",
      "Total loci = 159\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(25)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/tremellales/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/tremellales/extract_probes_from_group/Cryptococcus_neoformans_GCF_000091045+2-back-to-25.conf --probe-prefix uce_tremellales_ --designer rnplattii --design tremellales_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 159\n",
      "Probe Count = 7923\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list.fasta --query results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Mon Feb 10, 2020  16:13:35\n",
      "Ended:  Mon Feb 10, 2020  16:13:54\n",
      "Time for execution:  0.322025386492 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list.fasta --lastz results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_tremellales_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 7922 fasta sequences.  Filtered 0 duplicates. Kept 7923.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/tremellales/final_probe_design/tremellales_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Mon Feb 10 16:21:01 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 7923\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 633840\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3951519\n",
      "Max number of word counting entries: 88948698\n",
      "\n",
      "# comparing sequences from          0  to       1320\n",
      ".---------- new table with     1158 representatives\n",
      "# comparing sequences from       1320  to       2420\n",
      "----------    296 remaining sequences to the next cycle\n",
      "---------- new table with      684 representatives\n",
      "# comparing sequences from       2124  to       3090\n",
      "----------    196 remaining sequences to the next cycle\n",
      "---------- new table with      681 representatives\n",
      "# comparing sequences from       2894  to       3732\n",
      "----------    319 remaining sequences to the next cycle\n",
      "---------- new table with      433 representatives\n",
      "# comparing sequences from       3413  to       4164\n",
      "----------    263 remaining sequences to the next cycle\n",
      "---------- new table with      415 representatives\n",
      "# comparing sequences from       3901  to       4571\n",
      "----------    236 remaining sequences to the next cycle\n",
      "---------- new table with      382 representatives\n",
      "# comparing sequences from       4335  to       4933\n",
      "----------    108 remaining sequences to the next cycle\n",
      "---------- new table with      428 representatives\n",
      "# comparing sequences from       4825  to       5341\n",
      "----------    127 remaining sequences to the next cycle\n",
      "---------- new table with      347 representatives\n",
      "# comparing sequences from       5214  to       5665\n",
      "----------     91 remaining sequences to the next cycle\n",
      "---------- new table with      306 representatives\n",
      "# comparing sequences from       5574  to       5965\n",
      "----------    155 remaining sequences to the next cycle\n",
      "---------- new table with      208 representatives\n",
      "# comparing sequences from       5810  to       6162\n",
      "----------    128 remaining sequences to the next cycle\n",
      "---------- new table with      197 representatives\n",
      "# comparing sequences from       6034  to       6348\n",
      "100.0%---------- new table with      286 representatives\n",
      "# comparing sequences from       6348  to       6610\n",
      "----------     40 remaining sequences to the next cycle\n",
      "---------- new table with      191 representatives\n",
      "# comparing sequences from       6570  to       6795\n",
      "99.9%---------- new table with      191 representatives\n",
      "# comparing sequences from       6795  to       6983\n",
      "----------     54 remaining sequences to the next cycle\n",
      "---------- new table with      118 representatives\n",
      "# comparing sequences from       6929  to       7923\n",
      ".....................---------- new table with      872 representatives\n",
      "\n",
      "     7923  finished       6897  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 89M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.53\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Trypanosoma (Supplement)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group = 'trypanosoma'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Trypanosoma_brucei_GCF_000210295\",\n",
    "               \"Trypanosoma_congolense_GCA_002287245\",\n",
    "               \"Trypanosoma_conorhini_GCF_003719485\",\n",
    "               \"Trypanosoma_equiperdum_GCA_001457755\",\n",
    "               \"Trypanosoma_grayi_GCF_000691245\",\n",
    "               \"Trypanosoma_rangeli_GCF_003719475\",\n",
    "               \"Trypanosoma_theileri_GCF_002087225\",\n",
    "               \"Trypanosoma_vivax_GCA_000227375\" ]\n",
    "                    \n",
    "reference_taxon = \"Trypanosoma_cruzi_GCF_000209065\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000210295.1_ASM21029v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6763209 bytes  2705300.40 bytes/sec\n",
      "total size is 6761448  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002287245.1_ASM228724v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11189331 bytes  4475749.20 bytes/sec\n",
      "total size is 11186488  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003719485.1_ASM371948v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6796481 bytes  4531015.33 bytes/sec\n",
      "total size is 6794711  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000209065.1_ASM20906v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 26710398 bytes  7631554.29 bytes/sec\n",
      "total size is 26703772  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001457755.2_Trypanosoma_equiperdum_OVI_V2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8486624 bytes  5657777.33 bytes/sec\n",
      "total size is 8484427  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000691245.1_Tgr_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6692933 bytes  4461983.33 bytes/sec\n",
      "total size is 6691192  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003719475.1_ASM371947v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 6675731 bytes  2670309.20 bytes/sec\n",
      "total size is 6673993  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002087225.1_Tth_v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 8149087 bytes  3259651.60 bytes/sec\n",
      "total size is 8146994  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000227375.1_ASM22737v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 7855181 bytes  3142089.20 bytes/sec\n",
      "total size is 7853156  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487732 (\"sim_Trypanosoma_brucei_GCF_000210295\") has been submitted\n",
      "Your job 5487733 (\"sim_Trypanosoma_congolense_GCA_002287245\") has been submitted\n",
      "Your job 5487734 (\"sim_Trypanosoma_conorhini_GCF_003719485\") has been submitted\n",
      "Your job 5487735 (\"sim_Trypanosoma_equiperdum_GCA_001457755\") has been submitted\n",
      "Your job 5487736 (\"sim_Trypanosoma_grayi_GCF_000691245\") has been submitted\n",
      "Your job 5487737 (\"sim_Trypanosoma_rangeli_GCF_003719475\") has been submitted\n",
      "Your job 5487738 (\"sim_Trypanosoma_theileri_GCF_002087225\") has been submitted\n",
      "Your job 5487739 (\"sim_Trypanosoma_vivax_GCA_000227375\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/trypanosoma/cleaned_genomes/Trypanosoma_cruzi_GCF_000209065_formatted.fas path=results/phyluce/trypanosoma/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 302,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487740 (\"map_Trypanosoma_brucei_GCF_000210295\") has been submitted\n",
      "Your job 5487741 (\"map_Trypanosoma_congolense_GCA_002287245\") has been submitted\n",
      "Your job 5487742 (\"map_Trypanosoma_conorhini_GCF_003719485\") has been submitted\n",
      "Your job 5487743 (\"map_Trypanosoma_equiperdum_GCA_001457755\") has been submitted\n",
      "Your job 5487744 (\"map_Trypanosoma_grayi_GCF_000691245\") has been submitted\n",
      "Your job 5487745 (\"map_Trypanosoma_rangeli_GCF_003719475\") has been submitted\n",
      "Your job 5487746 (\"map_Trypanosoma_theileri_GCF_002087225\") has been submitted\n",
      "Your job 5487747 (\"map_Trypanosoma_vivax_GCA_000227375\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487748 (\"merge_Trypanosoma_brucei_GCF_000210295\") has been submitted\n",
      "Your job 5487749 (\"merge_Trypanosoma_congolense_GCA_002287245\") has been submitted\n",
      "Your job 5487750 (\"merge_Trypanosoma_conorhini_GCF_003719485\") has been submitted\n",
      "Your job 5487751 (\"merge_Trypanosoma_equiperdum_GCA_001457755\") has been submitted\n",
      "Your job 5487752 (\"merge_Trypanosoma_grayi_GCF_000691245\") has been submitted\n",
      "Your job 5487753 (\"merge_Trypanosoma_rangeli_GCF_003719475\") has been submitted\n",
      "Your job 5487754 (\"merge_Trypanosoma_theileri_GCF_002087225\") has been submitted\n",
      "Your job 5487755 (\"merge_Trypanosoma_vivax_GCA_000227375\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 248 sequences from Trypanosoma_brucei_GCF_000210295_merged.bed.  Filtered 205 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 394 sequences from Trypanosoma_congolense_GCA_002287245_merged.bed.  Filtered 312 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 82.\n",
      "Screened 10198 sequences from Trypanosoma_conorhini_GCF_003719485_merged.bed.  Filtered 7875 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2323.\n",
      "Screened 172 sequences from Trypanosoma_equiperdum_GCA_001457755_merged.bed.  Filtered 146 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 26.\n",
      "Screened 1157 sequences from Trypanosoma_grayi_GCF_000691245_merged.bed.  Filtered 904 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 253.\n",
      "Screened 9640 sequences from Trypanosoma_rangeli_GCF_003719475_merged.bed.  Filtered 7490 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2150.\n",
      "Screened 762 sequences from Trypanosoma_theileri_GCF_002087225_merged.bed.  Filtered 603 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 159.\n",
      "Screened 212 sequences from Trypanosoma_vivax_GCA_000227375_merged.bed.  Filtered 178 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 34.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/trypanosoma/initial_intervals/Trypanosoma_vivax_GCA_000227375_merged.bed --twobit results/phyluce/trypanosoma/cleaned_genomes/Trypanosoma_cruzi_GCF_000209065_formatted.2bit --output results/phyluce/trypanosoma/initial_intervals/Trypanosoma_vivax_GCA_000227375_stripped.bed;\n",
      "trypanosoma_brucei_gcf_000210295.\n",
      "trypanosoma_congolense_gca_002287245.\n",
      "trypanosoma_conorhini_gcf_003719485...\n",
      "trypanosoma_equiperdum_gca_001457755.\n",
      "trypanosoma_grayi_gcf_000691245.\n",
      "trypanosoma_rangeli_gcf_003719475...\n",
      "trypanosoma_theileri_gcf_002087225.\n",
      "trypanosoma_vivax_gca_000227375.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/trypanosoma/initial_intervals/trypanosoma-to-Trypanosoma_cruzi_GCF_000209065.sqlite --base-taxon Trypanosoma_cruzi_GCF_000209065\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 0 taxa:\t3,458.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 1 taxa:\t3,458.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 2 taxa:\t1,182.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 3 taxa:\t224.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 4 taxa:\t107.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 5 taxa:\t56.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 6 taxa:\t40.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 7 taxa:\t21.0\n",
      "Loci shared by Trypanosoma_cruzi_GCF_000209065 + 8 taxa:\t5.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/trypanosoma/initial_intervals/trypanosoma-to-Trypanosoma_cruzi_GCF_000209065.sqlite --base-taxon Trypanosoma_cruzi_GCF_000209065 --output results/phyluce/trypanosoma/initial_intervals/Trypanosoma_cruzi_GCF_000209065_+1.bed --specific-counts 1;\n",
      "Counter({'trypanosoma_conorhini_gcf_003719485': 2325, 'trypanosoma_rangeli_gcf_003719475': 2154, 'trypanosoma_grayi_gcf_000691245': 265, 'trypanosoma_theileri_gcf_002087225': 164, 'trypanosoma_congolense_gca_002287245': 83, 'trypanosoma_brucei_gcf_000210295': 42, 'trypanosoma_vivax_gca_000227375': 34, 'trypanosoma_equiperdum_gca_001457755': 26})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 1\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/trypanosoma/initial_intervals/Trypanosoma_cruzi_GCF_000209065_+1.bed --twobit results/phyluce/trypanosoma/cleaned_genomes/Trypanosoma_cruzi_GCF_000209065_formatted.2bit --buffer-to 160 --output results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1.fasta;\n",
      "Screened 3458 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 3458.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1.fasta --probe-prefix uce_trypanosoma_ --design trypanosoma_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 3453\n",
      "Probe Count = 6887\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas --query results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  13:29:45\n",
      "Ended:  Tue Feb 11, 2020  13:29:49\n",
      "Time for execution:  0.0633439660072 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas                        --lastz results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes_vself.lastz                       --probe-prefix=uce_trypanosoma_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6886 fasta sequences.  Filtered 1053 duplicates. Kept 4793.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/trypanosoma/validate_intervals/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas --scaffoldlist Trypanosoma_brucei_GCF_000210295 Trypanosoma_congolense_GCA_002287245 Trypanosoma_conorhini_GCF_003719485 Trypanosoma_equiperdum_GCA_001457755 Trypanosoma_grayi_GCF_000691245 Trypanosoma_rangeli_GCF_003719475 Trypanosoma_theileri_GCF_002087225 Trypanosoma_vivax_GCA_000227375 Trypanosoma_cruzi_GCF_000209065 --genome-base-path results/phyluce/trypanosoma/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/trypanosoma/validate_intervals/trypanosoma-to-Trypanosoma_cruzi_GCF_000209065.sqlite --output results/phyluce/trypanosoma/validate_intervals/lastz/;\n",
      "\n",
      "Running against Trypanosoma_brucei_GCF_000210295.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpzGse5k.fasta\n",
      "\t/tmp/tmpBOm4AV.fasta\n",
      "\t/tmp/tmpg6KY1j.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpR9OIRo.lastz\n",
      "\t/tmp/tmpPWCfq8.lastz\n",
      "\t/tmp/tmpYlw8PR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_brucei_GCF_000210295.lastz\n",
      "Creating Trypanosoma_brucei_GCF_000210295 table\n",
      "Inserting data to Trypanosoma_brucei_GCF_000210295 table\n",
      "\n",
      "Running against Trypanosoma_congolense_GCA_002287245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmp4bvIpL.fasta\n",
      "\t/tmp/tmpQFh2Lc.fasta\n",
      "\t/tmp/tmpyu3iQ4.fasta\n",
      "\t/tmp/tmp8cVenF.fasta\n",
      "\t/tmp/tmpyZGpLj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyRXdi6.lastz\n",
      "\t/tmp/tmpGtgT55.lastz\n",
      "\t/tmp/tmp4qO4nD.lastz\n",
      "\t/tmp/tmpCUcBFm.lastz\n",
      "\t/tmp/tmp_r1DyA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_congolense_GCA_002287245.lastz\n",
      "Creating Trypanosoma_congolense_GCA_002287245 table\n",
      "Inserting data to Trypanosoma_congolense_GCA_002287245 table\n",
      "\n",
      "Running against Trypanosoma_conorhini_GCF_003719485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpfZW3m7.fasta\n",
      "\t/tmp/tmp_T8RjI.fasta\n",
      "\t/tmp/tmpuJKDdH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3_K00z.lastz\n",
      "\t/tmp/tmpQjsmcI.lastz\n",
      "\t/tmp/tmpOn2ATB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_conorhini_GCF_003719485.lastz\n",
      "Creating Trypanosoma_conorhini_GCF_003719485 table\n",
      "Inserting data to Trypanosoma_conorhini_GCF_003719485 table\n",
      "\n",
      "Running against Trypanosoma_equiperdum_GCA_001457755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpuY9Zbz.fasta\n",
      "\t/tmp/tmpTYERkS.fasta\n",
      "\t/tmp/tmpGEE291.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7g9D87.lastz\n",
      "\t/tmp/tmpN2RxcQ.lastz\n",
      "\t/tmp/tmp6WLE3M.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_equiperdum_GCA_001457755.lastz\n",
      "Creating Trypanosoma_equiperdum_GCA_001457755 table\n",
      "Inserting data to Trypanosoma_equiperdum_GCA_001457755 table\n",
      "\n",
      "Running against Trypanosoma_grayi_GCF_000691245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpidUFdo.fasta\n",
      "\t/tmp/tmpKHxtv0.fasta\n",
      "\t/tmp/tmps7iee1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIGGhth.lastz\n",
      "\t/tmp/tmpmpFpIg.lastz\n",
      "\t/tmp/tmp2KSoKz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_grayi_GCF_000691245.lastz\n",
      "Creating Trypanosoma_grayi_GCF_000691245 table\n",
      "Inserting data to Trypanosoma_grayi_GCF_000691245 table\n",
      "\n",
      "Running against Trypanosoma_rangeli_GCF_003719475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmprtHoX6.fasta\n",
      "\t/tmp/tmpNzsneg.fasta\n",
      "\t/tmp/tmplyCVgt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6DiBr1.lastz\n",
      "\t/tmp/tmpQEENQq.lastz\n",
      "\t/tmp/tmphWS59t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_rangeli_GCF_003719475.lastz\n",
      "Creating Trypanosoma_rangeli_GCF_003719475 table\n",
      "Inserting data to Trypanosoma_rangeli_GCF_003719475 table\n",
      "\n",
      "Running against Trypanosoma_theileri_GCF_002087225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpwliLbF.fasta\n",
      "\t/tmp/tmpzU2A5G.fasta\n",
      "\t/tmp/tmpJJR1W8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpP80F4X.lastz\n",
      "\t/tmp/tmpld03Wg.lastz\n",
      "\t/tmp/tmpZxfQHp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_theileri_GCF_002087225.lastz\n",
      "Creating Trypanosoma_theileri_GCF_002087225 table\n",
      "Inserting data to Trypanosoma_theileri_GCF_002087225 table\n",
      "\n",
      "Running against Trypanosoma_vivax_GCA_000227375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 3 queries...\n",
      "\t/tmp/tmpY34Q_U.fasta\n",
      "\t/tmp/tmp29FzBM.fasta\n",
      "\t/tmp/tmpDBfitD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSXUsrh.lastz\n",
      "\t/tmp/tmpNv4bfi.lastz\n",
      "\t/tmp/tmp8PEz1f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_vivax_GCA_000227375.lastz\n",
      "Creating Trypanosoma_vivax_GCA_000227375 table\n",
      "Inserting data to Trypanosoma_vivax_GCA_000227375 table\n",
      "\n",
      "Running against Trypanosoma_cruzi_GCF_000209065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpwRN7RA.fasta\n",
      "\t/tmp/tmpKS2nmJ.fasta\n",
      "\t/tmp/tmpav4ZAZ.fasta\n",
      "\t/tmp/tmpP8bEeH.fasta\n",
      "\t/tmp/tmp6IpT8t.fasta\n",
      "\t/tmp/tmpIRAaRb.fasta\n",
      "\t/tmp/tmpsgc2ad.fasta\n",
      "\t/tmp/tmpIFIEoK.fasta\n",
      "\t/tmp/tmpjAV8Bd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoL1Hyr.lastz\n",
      "\t/tmp/tmp173adz.lastz\n",
      "\t/tmp/tmpTq7Mux.lastz\n",
      "\t/tmp/tmpZWhD0B.lastz\n",
      "\t/tmp/tmpgYQAMD.lastz\n",
      "\t/tmp/tmp2F6sBv.lastz\n",
      "\t/tmp/tmpgRs_Gf.lastz\n",
      "\t/tmp/tmpHRdJYa.lastz\n",
      "\t/tmp/tmpaChFnH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/trypanosoma/validate_intervals/lastz/Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_Trypanosoma_cruzi_GCF_000209065.lastz\n",
      "Creating Trypanosoma_cruzi_GCF_000209065 table\n",
      "Inserting data to Trypanosoma_cruzi_GCF_000209065 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/trypanosoma/extract_probes_from_group/trypanosoma_genome.conf --lastz results/phyluce/trypanosoma/validate_intervals/lastz --probes 120 --probe-prefix uce_trypanosoma_ --name-pattern \"Trypanosoma_cruzi_GCF_000209065_+1_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/trypanosoma/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 13:32:33,653 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 13:32:33,674 - Phyluce - INFO - ------- Working on Trypanosoma_brucei_GCF_000210295 genome ------\n",
      "2020-02-11 13:32:33,693 - Phyluce - INFO - Reading Trypanosoma_brucei_GCF_000210295 genome\n",
      "2020-02-11 13:32:49,814 - Phyluce - INFO - Trypanosoma_brucei_GCF_000210295: 3251 uces, 408 dupes, 2843 non-dupes, 21 orient drop, 309 length drop, 2513 written\n",
      "2020-02-11 13:32:49,815 - Phyluce - INFO - ----- Working on Trypanosoma_congolense_GCA_002287245 genome ----\n",
      "2020-02-11 13:32:49,838 - Phyluce - INFO - Reading Trypanosoma_congolense_GCA_002287245 genome\n",
      "2020-02-11 13:33:16,747 - Phyluce - INFO - Trypanosoma_congolense_GCA_002287245: 3233 uces, 671 dupes, 2562 non-dupes, 17 orient drop, 261 length drop, 2284 written\n",
      "2020-02-11 13:33:16,747 - Phyluce - INFO - ----- Working on Trypanosoma_conorhini_GCF_003719485 genome -----\n",
      "2020-02-11 13:33:16,755 - Phyluce - INFO - Reading Trypanosoma_conorhini_GCF_003719485 genome\n",
      "2020-02-11 13:33:31,211 - Phyluce - INFO - Trypanosoma_conorhini_GCF_003719485: 3416 uces, 527 dupes, 2889 non-dupes, 0 orient drop, 70 length drop, 2819 written\n",
      "2020-02-11 13:33:31,212 - Phyluce - INFO - ----- Working on Trypanosoma_equiperdum_GCA_001457755 genome ----\n",
      "2020-02-11 13:33:31,225 - Phyluce - INFO - Reading Trypanosoma_equiperdum_GCA_001457755 genome\n",
      "2020-02-11 13:33:43,757 - Phyluce - INFO - Trypanosoma_equiperdum_GCA_001457755: 3179 uces, 382 dupes, 2797 non-dupes, 0 orient drop, 17 length drop, 2780 written\n",
      "2020-02-11 13:33:43,757 - Phyluce - INFO - ------- Working on Trypanosoma_grayi_GCF_000691245 genome -------\n",
      "2020-02-11 13:33:43,864 - Phyluce - INFO - Reading Trypanosoma_grayi_GCF_000691245 genome\n",
      "2020-02-11 13:33:56,772 - Phyluce - INFO - Trypanosoma_grayi_GCF_000691245: 3325 uces, 413 dupes, 2912 non-dupes, 0 orient drop, 27 length drop, 2885 written\n",
      "2020-02-11 13:33:56,772 - Phyluce - INFO - ------ Working on Trypanosoma_rangeli_GCF_003719475 genome ------\n",
      "2020-02-11 13:33:56,800 - Phyluce - INFO - Reading Trypanosoma_rangeli_GCF_003719475 genome\n",
      "2020-02-11 13:34:11,793 - Phyluce - INFO - Trypanosoma_rangeli_GCF_003719475: 3427 uces, 545 dupes, 2882 non-dupes, 0 orient drop, 111 length drop, 2771 written\n",
      "2020-02-11 13:34:11,794 - Phyluce - INFO - ------ Working on Trypanosoma_theileri_GCF_002087225 genome -----\n",
      "2020-02-11 13:34:11,804 - Phyluce - INFO - Reading Trypanosoma_theileri_GCF_002087225 genome\n",
      "2020-02-11 13:34:25,623 - Phyluce - INFO - Trypanosoma_theileri_GCF_002087225: 3364 uces, 450 dupes, 2914 non-dupes, 5 orient drop, 175 length drop, 2734 written\n",
      "2020-02-11 13:34:25,623 - Phyluce - INFO - ------- Working on Trypanosoma_vivax_GCA_000227375 genome -------\n",
      "2020-02-11 13:34:25,630 - Phyluce - INFO - Reading Trypanosoma_vivax_GCA_000227375 genome\n",
      "2020-02-11 13:34:28,469 - Phyluce - INFO - Trypanosoma_vivax_GCA_000227375: 347 uces, 144 dupes, 203 non-dupes, 3 orient drop, 5 length drop, 189 written\n",
      "2020-02-11 13:34:28,469 - Phyluce - INFO - ------- Working on Trypanosoma_cruzi_GCF_000209065 genome -------\n",
      "2020-02-11 13:34:28,475 - Phyluce - INFO - Reading Trypanosoma_cruzi_GCF_000209065 genome\n",
      "2020-02-11 13:35:07,987 - Phyluce - INFO - Trypanosoma_cruzi_GCF_000209065: 3453 uces, 2976 dupes, 477 non-dupes, 0 orient drop, 39 length drop, 438 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/trypanosoma/extract_probes_from_group/probe_fasta --output results/phyluce/trypanosoma/extract_probes_from_group/multifastas.sqlite --base-taxon Trypanosoma_cruzi_GCF_000209065;\n",
      "trypanosoma_brucei_gcf_000210295...\n",
      "trypanosoma_congolense_gca_002287245...\n",
      "trypanosoma_conorhini_gcf_003719485...\n",
      "trypanosoma_equiperdum_gca_001457755...\n",
      "trypanosoma_grayi_gcf_000691245...\n",
      "trypanosoma_rangeli_gcf_003719475...\n",
      "trypanosoma_theileri_gcf_002087225...\n",
      "trypanosoma_vivax_gca_000227375.\n",
      "trypanosoma_cruzi_gcf_000209065.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/trypanosoma/extract_probes_from_group/multifastas.sqlite --base-taxon Trypanosoma_cruzi_GCF_000209065;\n",
      "Loci shared by 0 taxa:\t3,251.0\n",
      "Loci shared by 1 taxa:\t3,251.0\n",
      "Loci shared by 2 taxa:\t3,115.0\n",
      "Loci shared by 3 taxa:\t2,970.0\n",
      "Loci shared by 4 taxa:\t2,815.0\n",
      "Loci shared by 5 taxa:\t2,594.0\n",
      "Loci shared by 6 taxa:\t2,370.0\n",
      "Loci shared by 7 taxa:\t1,902.0\n",
      "Loci shared by 8 taxa:\t389.0\n",
      "Loci shared by 9 taxa:\t7.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/trypanosoma/extract_probes_from_group/multifastas.sqlite  --base-taxon Trypanosoma_cruzi_GCF_000209065 --output results/phyluce/trypanosoma/extract_probes_from_group/Trypanosoma_cruzi_GCF_000209065+1-back-to-8.conf --specific-counts 8;\n",
      "Counter({'trypanosoma_grayi_gcf_000691245': 389, 'trypanosoma_congolense_gca_002287245': 389, 'trypanosoma_rangeli_gcf_003719475': 389, 'trypanosoma_equiperdum_gca_001457755': 389, 'trypanosoma_conorhini_gcf_003719485': 389, 'trypanosoma_theileri_gcf_002087225': 388, 'trypanosoma_brucei_gcf_000210295': 388, 'trypanosoma_cruzi_gcf_000209065': 321, 'trypanosoma_vivax_gca_000227375': 77})\n",
      "Total loci = 389\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(8)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/trypanosoma/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/trypanosoma/extract_probes_from_group/Trypanosoma_cruzi_GCF_000209065+1-back-to-8.conf --probe-prefix uce_trypanosoma_ --designer rnplattii --design trypanosoma_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 389\n",
      "Probe Count = 6188\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list.fasta --query results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  13:35:56\n",
      "Ended:  Tue Feb 11, 2020  13:36:01\n",
      "Time for execution:  0.0883343497912 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list.fasta --lastz results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_trypanosoma_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6187 fasta sequences.  Filtered 12 duplicates. Kept 6000.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/trypanosoma/final_probe_design/trypanosoma_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 13:37:46 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 6188\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 495040\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3954451\n",
      "Max number of word counting entries: 89014700\n",
      "\n",
      "# comparing sequences from          0  to       1031\n",
      ".---------- new table with      912 representatives\n",
      "# comparing sequences from       1031  to       1890\n",
      "99.9%---------- new table with      755 representatives\n",
      "# comparing sequences from       1890  to       2606\n",
      "----------     95 remaining sequences to the next cycle\n",
      "---------- new table with      549 representatives\n",
      "# comparing sequences from       2511  to       3123\n",
      "----------    124 remaining sequences to the next cycle\n",
      "---------- new table with      434 representatives\n",
      "# comparing sequences from       2999  to       3530\n",
      "----------     92 remaining sequences to the next cycle\n",
      "---------- new table with      393 representatives\n",
      "# comparing sequences from       3438  to       3896\n",
      "----------    126 remaining sequences to the next cycle\n",
      "---------- new table with      294 representatives\n",
      "# comparing sequences from       3770  to       4173\n",
      "----------    134 remaining sequences to the next cycle\n",
      "---------- new table with      230 representatives\n",
      "# comparing sequences from       4039  to       4397\n",
      "----------      9 remaining sequences to the next cycle\n",
      "---------- new table with      309 representatives\n",
      "# comparing sequences from       4388  to       4688\n",
      "99.9%---------- new table with      264 representatives\n",
      "# comparing sequences from       4688  to       4938\n",
      "----------     71 remaining sequences to the next cycle\n",
      "---------- new table with      158 representatives\n",
      "# comparing sequences from       4867  to       5087\n",
      "100.0%---------- new table with      194 representatives\n",
      "# comparing sequences from       5087  to       5270\n",
      "99.9%---------- new table with      162 representatives\n",
      "# comparing sequences from       5270  to       6188\n",
      ".....................---------- new table with      808 representatives\n",
      "\n",
      "     6188  finished       5462  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.18\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "319px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
