{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import shutil\n",
    "from shutil import copy\n",
    "import time\n",
    "from collections import defaultdict\n",
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "os.chdir(\"/master/nplatt/pathogen_probes/\")\n",
    "\n",
    "\n",
    "def wait_on_running_jobs():\n",
    "   \n",
    "    num_jobs = 1\n",
    "    \n",
    "    while num_jobs > 0:\n",
    "        num_jobs = len(subprocess.check_output('qstat', shell=True).split(\"\\n\")) - 2\n",
    "        time.sleep(60)\n",
    "        print(\".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Chlamydia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'chlamydia'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Chlamydia_abortus_GCF_002895085\",\n",
    "               \"Chlamydia_avium_GCF_000583875\",\n",
    "               \"Chlamydia_buteonis_GCF_900634605\",\n",
    "               \"Chlamydia_caviae_GCF_000007605\",\n",
    "               \"Chlamydia_felis_GCF_000009945\",\n",
    "               \"Chlamydia_gallinacea_GCF_000471025\",\n",
    "               \"Chlamydia_ibidis_GCF_000454725\",\n",
    "               \"Chlamydia_muridarum_GCF_000006685\",\n",
    "               \"Chlamydia_pecorum_GCF_000204135\",\n",
    "               \"Chlamydia_pneumoniae_GCF_000008745\",\n",
    "               \"Chlamydia_psittaci_GCF_000204255\",\n",
    "               \"Chlamydia_sp_GCF_900239945\",\n",
    "               \"Chlamydia_suis_GCF_900169085\" ]\n",
    "\n",
    "reference_taxon = \"Chlamydia_trachomatis_GCF_000008725\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002895085.1_ASM289508v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 341785 bytes  227884.67 bytes/sec\n",
      "total size is 341591  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000583875.1_ASM58387v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 311132 bytes  622348.00 bytes/sec\n",
      "total size is 310947  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900634605.1_RSHA_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 341793 bytes  227890.00 bytes/sec\n",
      "total size is 341606  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007605.1_ASM760v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 351700 bytes  234494.67 bytes/sec\n",
      "total size is 351509  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000009945.1_ASM994v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 349893 bytes  233290.00 bytes/sec\n",
      "total size is 349702  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000471025.2_ASM47102v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 316978 bytes  211346.67 bytes/sec\n",
      "total size is 316793  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000454725.1_ibidis.assembly_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 342508 bytes  228366.67 bytes/sec\n",
      "total size is 342310  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006685.1_ASM668v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 320871 bytes  213942.00 bytes/sec\n",
      "total size is 320688  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000204135.1_ASM20413v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 328845 bytes  219258.00 bytes/sec\n",
      "total size is 328652  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000008745.1_ASM874v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 366606 bytes  244432.00 bytes/sec\n",
      "total size is 366407  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000204255.1_ASM20425v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 351224 bytes  234177.33 bytes/sec\n",
      "total size is 351031  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900239945.1_Chlamydia_sp._nov._H15-1957-10C_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 357880 bytes  238614.67 bytes/sec\n",
      "total size is 357666  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900169085.1_SWA-2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 322351 bytes  644786.00 bytes/sec\n",
      "total size is 322171  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000008725.1_ASM872v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 310297 bytes  206892.67 bytes/sec\n",
      "total size is 310114  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482630 (\"sim_Chlamydia_abortus_GCF_002895085\") has been submitted\n",
      "Your job 5482631 (\"sim_Chlamydia_avium_GCF_000583875\") has been submitted\n",
      "Your job 5482632 (\"sim_Chlamydia_buteonis_GCF_900634605\") has been submitted\n",
      "Your job 5482633 (\"sim_Chlamydia_caviae_GCF_000007605\") has been submitted\n",
      "Your job 5482634 (\"sim_Chlamydia_felis_GCF_000009945\") has been submitted\n",
      "Your job 5482635 (\"sim_Chlamydia_gallinacea_GCF_000471025\") has been submitted\n",
      "Your job 5482636 (\"sim_Chlamydia_ibidis_GCF_000454725\") has been submitted\n",
      "Your job 5482637 (\"sim_Chlamydia_muridarum_GCF_000006685\") has been submitted\n",
      "Your job 5482638 (\"sim_Chlamydia_pecorum_GCF_000204135\") has been submitted\n",
      "Your job 5482639 (\"sim_Chlamydia_pneumoniae_GCF_000008745\") has been submitted\n",
      "Your job 5482640 (\"sim_Chlamydia_psittaci_GCF_000204255\") has been submitted\n",
      "Your job 5482641 (\"sim_Chlamydia_sp_GCF_900239945\") has been submitted\n",
      "Your job 5482642 (\"sim_Chlamydia_suis_GCF_900169085\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/chlamydia/cleaned_genomes/Chlamydia_trachomatis_GCF_000008725_formatted.fas path=results/phyluce/chlamydia/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482643 (\"map_Chlamydia_abortus_GCF_002895085\") has been submitted\n",
      "Your job 5482644 (\"map_Chlamydia_avium_GCF_000583875\") has been submitted\n",
      "Your job 5482645 (\"map_Chlamydia_buteonis_GCF_900634605\") has been submitted\n",
      "Your job 5482646 (\"map_Chlamydia_caviae_GCF_000007605\") has been submitted\n",
      "Your job 5482647 (\"map_Chlamydia_felis_GCF_000009945\") has been submitted\n",
      "Your job 5482648 (\"map_Chlamydia_gallinacea_GCF_000471025\") has been submitted\n",
      "Your job 5482649 (\"map_Chlamydia_ibidis_GCF_000454725\") has been submitted\n",
      "Your job 5482650 (\"map_Chlamydia_muridarum_GCF_000006685\") has been submitted\n",
      "Your job 5482651 (\"map_Chlamydia_pecorum_GCF_000204135\") has been submitted\n",
      "Your job 5482652 (\"map_Chlamydia_pneumoniae_GCF_000008745\") has been submitted\n",
      "Your job 5482653 (\"map_Chlamydia_psittaci_GCF_000204255\") has been submitted\n",
      "Your job 5482654 (\"map_Chlamydia_sp_GCF_900239945\") has been submitted\n",
      "Your job 5482655 (\"map_Chlamydia_suis_GCF_900169085\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482656 (\"merge_Chlamydia_abortus_GCF_002895085\") has been submitted\n",
      "Your job 5482657 (\"merge_Chlamydia_avium_GCF_000583875\") has been submitted\n",
      "Your job 5482658 (\"merge_Chlamydia_buteonis_GCF_900634605\") has been submitted\n",
      "Your job 5482659 (\"merge_Chlamydia_caviae_GCF_000007605\") has been submitted\n",
      "Your job 5482660 (\"merge_Chlamydia_felis_GCF_000009945\") has been submitted\n",
      "Your job 5482661 (\"merge_Chlamydia_gallinacea_GCF_000471025\") has been submitted\n",
      "Your job 5482662 (\"merge_Chlamydia_ibidis_GCF_000454725\") has been submitted\n",
      "Your job 5482663 (\"merge_Chlamydia_muridarum_GCF_000006685\") has been submitted\n",
      "Your job 5482664 (\"merge_Chlamydia_pecorum_GCF_000204135\") has been submitted\n",
      "Your job 5482665 (\"merge_Chlamydia_pneumoniae_GCF_000008745\") has been submitted\n",
      "Your job 5482666 (\"merge_Chlamydia_psittaci_GCF_000204255\") has been submitted\n",
      "Your job 5482667 (\"merge_Chlamydia_sp_GCF_900239945\") has been submitted\n",
      "Your job 5482668 (\"merge_Chlamydia_suis_GCF_900169085\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 49 sequences from Chlamydia_abortus_GCF_002895085_merged.bed.  Filtered 39 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 33 sequences from Chlamydia_avium_GCF_000583875_merged.bed.  Filtered 21 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 47 sequences from Chlamydia_buteonis_GCF_900634605_merged.bed.  Filtered 37 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 45 sequences from Chlamydia_caviae_GCF_000007605_merged.bed.  Filtered 34 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 40 sequences from Chlamydia_felis_GCF_000009945_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 36 sequences from Chlamydia_gallinacea_GCF_000471025_merged.bed.  Filtered 24 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 40 sequences from Chlamydia_ibidis_GCF_000454725_merged.bed.  Filtered 30 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 802 sequences from Chlamydia_muridarum_GCF_000006685_merged.bed.  Filtered 458 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 344.\n",
      "Screened 24 sequences from Chlamydia_pecorum_GCF_000204135_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 35 sequences from Chlamydia_pneumoniae_GCF_000008745_merged.bed.  Filtered 29 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 42 sequences from Chlamydia_psittaci_GCF_000204255_merged.bed.  Filtered 29 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 33 sequences from Chlamydia_sp_GCF_900239945_merged.bed.  Filtered 22 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 653 sequences from Chlamydia_suis_GCF_900169085_merged.bed.  Filtered 375 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 278.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/chlamydia/initial_intervals/Chlamydia_suis_GCF_900169085_merged.bed --twobit results/phyluce/chlamydia/cleaned_genomes/Chlamydia_trachomatis_GCF_000008725_formatted.2bit --output results/phyluce/chlamydia/initial_intervals/Chlamydia_suis_GCF_900169085_stripped.bed;\n",
      "chlamydia_abortus_gcf_002895085.\n",
      "chlamydia_avium_gcf_000583875.\n",
      "chlamydia_buteonis_gcf_900634605.\n",
      "chlamydia_caviae_gcf_000007605.\n",
      "chlamydia_felis_gcf_000009945.\n",
      "chlamydia_gallinacea_gcf_000471025.\n",
      "chlamydia_ibidis_gcf_000454725.\n",
      "chlamydia_muridarum_gcf_000006685.\n",
      "chlamydia_pecorum_gcf_000204135.\n",
      "chlamydia_pneumoniae_gcf_000008745.\n",
      "chlamydia_psittaci_gcf_000204255.\n",
      "chlamydia_sp_gcf_900239945.\n",
      "chlamydia_suis_gcf_900169085.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/chlamydia/initial_intervals/Chlamydia_suis_GCF_900169085_merged.bed --twobit results/phyluce/chlamydia/cleaned_genomes/Chlamydia_trachomatis_GCF_000008725_formatted.2bit --output results/phyluce/chlamydia/initial_intervals/Chlamydia_suis_GCF_900169085_stripped.bed;\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 0 taxa:\t405.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 1 taxa:\t405.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 2 taxa:\t231.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 3 taxa:\t14.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 4 taxa:\t13.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 5 taxa:\t12.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 6 taxa:\t12.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 7 taxa:\t10.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 8 taxa:\t10.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 9 taxa:\t9.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 10 taxa:\t8.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 11 taxa:\t7.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 12 taxa:\t6.0\n",
      "Loci shared by Chlamydia_trachomatis_GCF_000008725 + 13 taxa:\t6.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/chlamydia/initial_intervals/chlamydia-to-Chlamydia_trachomatis_GCF_000008725.sqlite --base-taxon Chlamydia_trachomatis_GCF_000008725 --output results/phyluce/chlamydia/initial_intervals/Chlamydia_trachomatis_GCF_000008725_+2.bed --specific-counts 2;\n",
      "Counter({'chlamydia_muridarum_gcf_000006685': 230, 'chlamydia_suis_gcf_900169085': 229, 'chlamydia_felis_gcf_000009945': 12, 'chlamydia_psittaci_gcf_000204255': 12, 'chlamydia_avium_gcf_000583875': 11, 'chlamydia_gallinacea_gcf_000471025': 10, 'chlamydia_caviae_gcf_000007605': 10, 'chlamydia_sp_gcf_900239945': 10, 'chlamydia_abortus_gcf_002895085': 10, 'chlamydia_pecorum_gcf_000204135': 10, 'chlamydia_ibidis_gcf_000454725': 10, 'chlamydia_buteonis_gcf_900634605': 9, 'chlamydia_pneumoniae_gcf_000008745': 6})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 2\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/chlamydia/initial_intervals/Chlamydia_trachomatis_GCF_000008725_+2.bed --twobit results/phyluce/chlamydia/cleaned_genomes/Chlamydia_trachomatis_GCF_000008725_formatted.2bit --buffer-to 160 --output results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2.fasta;\n",
      "Screened 231 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 231.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2.fasta --probe-prefix uce_chlamydia_ --design chlamydia_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 227\n",
      "Probe Count = 446\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas --query results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  09:29:01\n",
      "Ended:  Fri Feb 07, 2020  09:29:01\n",
      "Time for execution:  0.00403754711151 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas                        --lastz results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes_vself.lastz                       --probe-prefix=uce_chlamydia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 445 fasta sequences.  Filtered 4 duplicates. Kept 438.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/chlamydia/validate_intervals/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas --scaffoldlist Chlamydia_abortus_GCF_002895085 Chlamydia_avium_GCF_000583875 Chlamydia_buteonis_GCF_900634605 Chlamydia_caviae_GCF_000007605 Chlamydia_felis_GCF_000009945 Chlamydia_gallinacea_GCF_000471025 Chlamydia_ibidis_GCF_000454725 Chlamydia_muridarum_GCF_000006685 Chlamydia_pecorum_GCF_000204135 Chlamydia_pneumoniae_GCF_000008745 Chlamydia_psittaci_GCF_000204255 Chlamydia_sp_GCF_900239945 Chlamydia_suis_GCF_900169085 Chlamydia_trachomatis_GCF_000008725 --genome-base-path results/phyluce/chlamydia/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/chlamydia/validate_intervals/chlamydia-to-Chlamydia_trachomatis_GCF_000008725.sqlite --output results/phyluce/chlamydia/validate_intervals/lastz/;\n",
      "\n",
      "Running against Chlamydia_abortus_GCF_002895085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmXuH4u.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSxcq73.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_abortus_GCF_002895085.lastz\n",
      "Creating Chlamydia_abortus_GCF_002895085 table\n",
      "Inserting data to Chlamydia_abortus_GCF_002895085 table\n",
      "\n",
      "Running against Chlamydia_avium_GCF_000583875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxX2hRE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXGCPjK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_avium_GCF_000583875.lastz\n",
      "Creating Chlamydia_avium_GCF_000583875 table\n",
      "Inserting data to Chlamydia_avium_GCF_000583875 table\n",
      "\n",
      "Running against Chlamydia_buteonis_GCF_900634605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTUETwi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqmjk4j.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_buteonis_GCF_900634605.lastz\n",
      "Creating Chlamydia_buteonis_GCF_900634605 table\n",
      "Inserting data to Chlamydia_buteonis_GCF_900634605 table\n",
      "\n",
      "Running against Chlamydia_caviae_GCF_000007605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRDZO4k.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwOBOhE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_caviae_GCF_000007605.lastz\n",
      "Creating Chlamydia_caviae_GCF_000007605 table\n",
      "Inserting data to Chlamydia_caviae_GCF_000007605 table\n",
      "\n",
      "Running against Chlamydia_felis_GCF_000009945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5nXkFY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1oWtNh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_felis_GCF_000009945.lastz\n",
      "Creating Chlamydia_felis_GCF_000009945 table\n",
      "Inserting data to Chlamydia_felis_GCF_000009945 table\n",
      "\n",
      "Running against Chlamydia_gallinacea_GCF_000471025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzgt5ix.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3liO5A.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_gallinacea_GCF_000471025.lastz\n",
      "Creating Chlamydia_gallinacea_GCF_000471025 table\n",
      "Inserting data to Chlamydia_gallinacea_GCF_000471025 table\n",
      "\n",
      "Running against Chlamydia_ibidis_GCF_000454725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0v3oVM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzK_40a.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_ibidis_GCF_000454725.lastz\n",
      "Creating Chlamydia_ibidis_GCF_000454725 table\n",
      "Inserting data to Chlamydia_ibidis_GCF_000454725 table\n",
      "\n",
      "Running against Chlamydia_muridarum_GCF_000006685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKQXVlK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpR_29EQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_muridarum_GCF_000006685.lastz\n",
      "Creating Chlamydia_muridarum_GCF_000006685 table\n",
      "Inserting data to Chlamydia_muridarum_GCF_000006685 table\n",
      "\n",
      "Running against Chlamydia_pecorum_GCF_000204135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpY9Lqov.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXbAoCZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_pecorum_GCF_000204135.lastz\n",
      "Creating Chlamydia_pecorum_GCF_000204135 table\n",
      "Inserting data to Chlamydia_pecorum_GCF_000204135 table\n",
      "\n",
      "Running against Chlamydia_pneumoniae_GCF_000008745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphpnvPF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpflsnEE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_pneumoniae_GCF_000008745.lastz\n",
      "Creating Chlamydia_pneumoniae_GCF_000008745 table\n",
      "Inserting data to Chlamydia_pneumoniae_GCF_000008745 table\n",
      "\n",
      "Running against Chlamydia_psittaci_GCF_000204255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpehayqs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5kY3cs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_psittaci_GCF_000204255.lastz\n",
      "Creating Chlamydia_psittaci_GCF_000204255 table\n",
      "Inserting data to Chlamydia_psittaci_GCF_000204255 table\n",
      "\n",
      "Running against Chlamydia_sp_GCF_900239945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmppGgv6g.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplcJpAg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_sp_GCF_900239945.lastz\n",
      "Creating Chlamydia_sp_GCF_900239945 table\n",
      "Inserting data to Chlamydia_sp_GCF_900239945 table\n",
      "\n",
      "Running against Chlamydia_suis_GCF_900169085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpak0ct5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5g4J2k.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_suis_GCF_900169085.lastz\n",
      "Creating Chlamydia_suis_GCF_900169085 table\n",
      "Inserting data to Chlamydia_suis_GCF_900169085 table\n",
      "\n",
      "Running against Chlamydia_trachomatis_GCF_000008725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpBJesKG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfT00xj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/chlamydia/validate_intervals/lastz/Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_Chlamydia_trachomatis_GCF_000008725.lastz\n",
      "Creating Chlamydia_trachomatis_GCF_000008725 table\n",
      "Inserting data to Chlamydia_trachomatis_GCF_000008725 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/chlamydia/extract_probes_from_group/chlamydia_genome.conf --lastz results/phyluce/chlamydia/validate_intervals/lastz --probes 120 --probe-prefix uce_chlamydia_ --name-pattern \"Chlamydia_trachomatis_GCF_000008725_+2_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/chlamydia/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 09:29:20,337 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 09:29:20,339 - Phyluce - INFO - ------- Working on Chlamydia_abortus_GCF_002895085 genome -------\n",
      "2020-02-07 09:29:20,340 - Phyluce - INFO - Reading Chlamydia_abortus_GCF_002895085 genome\n",
      "2020-02-07 09:29:21,004 - Phyluce - INFO - Chlamydia_abortus_GCF_002895085: 208 uces, 0 dupes, 208 non-dupes, 2 orient drop, 0 length drop, 206 written\n",
      "2020-02-07 09:29:21,004 - Phyluce - INFO - -------- Working on Chlamydia_avium_GCF_000583875 genome --------\n",
      "2020-02-07 09:29:21,005 - Phyluce - INFO - Reading Chlamydia_avium_GCF_000583875 genome\n",
      "2020-02-07 09:29:21,589 - Phyluce - INFO - Chlamydia_avium_GCF_000583875: 200 uces, 0 dupes, 200 non-dupes, 1 orient drop, 0 length drop, 199 written\n",
      "2020-02-07 09:29:21,589 - Phyluce - INFO - ------- Working on Chlamydia_buteonis_GCF_900634605 genome ------\n",
      "2020-02-07 09:29:21,590 - Phyluce - INFO - Reading Chlamydia_buteonis_GCF_900634605 genome\n",
      "2020-02-07 09:29:22,209 - Phyluce - INFO - Chlamydia_buteonis_GCF_900634605: 210 uces, 1 dupes, 209 non-dupes, 2 orient drop, 0 length drop, 207 written\n",
      "2020-02-07 09:29:22,209 - Phyluce - INFO - -------- Working on Chlamydia_caviae_GCF_000007605 genome -------\n",
      "2020-02-07 09:29:22,209 - Phyluce - INFO - Reading Chlamydia_caviae_GCF_000007605 genome\n",
      "2020-02-07 09:29:22,826 - Phyluce - INFO - Chlamydia_caviae_GCF_000007605: 206 uces, 0 dupes, 206 non-dupes, 2 orient drop, 0 length drop, 204 written\n",
      "2020-02-07 09:29:22,827 - Phyluce - INFO - -------- Working on Chlamydia_felis_GCF_000009945 genome --------\n",
      "2020-02-07 09:29:22,827 - Phyluce - INFO - Reading Chlamydia_felis_GCF_000009945 genome\n",
      "2020-02-07 09:29:23,433 - Phyluce - INFO - Chlamydia_felis_GCF_000009945: 208 uces, 0 dupes, 208 non-dupes, 1 orient drop, 0 length drop, 207 written\n",
      "2020-02-07 09:29:23,433 - Phyluce - INFO - ------ Working on Chlamydia_gallinacea_GCF_000471025 genome -----\n",
      "2020-02-07 09:29:23,434 - Phyluce - INFO - Reading Chlamydia_gallinacea_GCF_000471025 genome\n",
      "2020-02-07 09:29:24,023 - Phyluce - INFO - Chlamydia_gallinacea_GCF_000471025: 200 uces, 0 dupes, 200 non-dupes, 1 orient drop, 0 length drop, 199 written\n",
      "2020-02-07 09:29:24,023 - Phyluce - INFO - -------- Working on Chlamydia_ibidis_GCF_000454725 genome -------\n",
      "2020-02-07 09:29:24,024 - Phyluce - INFO - Reading Chlamydia_ibidis_GCF_000454725 genome\n",
      "2020-02-07 09:29:24,613 - Phyluce - INFO - Chlamydia_ibidis_GCF_000454725: 197 uces, 1 dupes, 196 non-dupes, 2 orient drop, 0 length drop, 194 written\n",
      "2020-02-07 09:29:24,613 - Phyluce - INFO - ------ Working on Chlamydia_muridarum_GCF_000006685 genome ------\n",
      "2020-02-07 09:29:24,614 - Phyluce - INFO - Reading Chlamydia_muridarum_GCF_000006685 genome\n",
      "2020-02-07 09:29:25,382 - Phyluce - INFO - Chlamydia_muridarum_GCF_000006685: 227 uces, 0 dupes, 227 non-dupes, 0 orient drop, 11 length drop, 216 written\n",
      "2020-02-07 09:29:25,382 - Phyluce - INFO - ------- Working on Chlamydia_pecorum_GCF_000204135 genome -------\n",
      "2020-02-07 09:29:25,383 - Phyluce - INFO - Reading Chlamydia_pecorum_GCF_000204135 genome\n",
      "2020-02-07 09:29:25,940 - Phyluce - INFO - Chlamydia_pecorum_GCF_000204135: 195 uces, 0 dupes, 195 non-dupes, 2 orient drop, 0 length drop, 193 written\n",
      "2020-02-07 09:29:25,940 - Phyluce - INFO - ------ Working on Chlamydia_pneumoniae_GCF_000008745 genome -----\n",
      "2020-02-07 09:29:25,957 - Phyluce - INFO - Reading Chlamydia_pneumoniae_GCF_000008745 genome\n",
      "2020-02-07 09:29:26,549 - Phyluce - INFO - Chlamydia_pneumoniae_GCF_000008745: 195 uces, 0 dupes, 195 non-dupes, 3 orient drop, 0 length drop, 192 written\n",
      "2020-02-07 09:29:26,550 - Phyluce - INFO - ------- Working on Chlamydia_psittaci_GCF_000204255 genome ------\n",
      "2020-02-07 09:29:26,550 - Phyluce - INFO - Reading Chlamydia_psittaci_GCF_000204255 genome\n",
      "2020-02-07 09:29:27,161 - Phyluce - INFO - Chlamydia_psittaci_GCF_000204255: 208 uces, 0 dupes, 208 non-dupes, 3 orient drop, 0 length drop, 205 written\n",
      "2020-02-07 09:29:27,161 - Phyluce - INFO - ---------- Working on Chlamydia_sp_GCF_900239945 genome ---------\n",
      "2020-02-07 09:29:27,162 - Phyluce - INFO - Reading Chlamydia_sp_GCF_900239945 genome\n",
      "2020-02-07 09:29:27,736 - Phyluce - INFO - Chlamydia_sp_GCF_900239945: 195 uces, 0 dupes, 195 non-dupes, 3 orient drop, 0 length drop, 192 written\n",
      "2020-02-07 09:29:27,737 - Phyluce - INFO - --------- Working on Chlamydia_suis_GCF_900169085 genome --------\n",
      "2020-02-07 09:29:27,737 - Phyluce - INFO - Reading Chlamydia_suis_GCF_900169085 genome\n",
      "2020-02-07 09:29:28,507 - Phyluce - INFO - Chlamydia_suis_GCF_900169085: 227 uces, 0 dupes, 227 non-dupes, 0 orient drop, 12 length drop, 215 written\n",
      "2020-02-07 09:29:28,507 - Phyluce - INFO - ----- Working on Chlamydia_trachomatis_GCF_000008725 genome -----\n",
      "2020-02-07 09:29:28,508 - Phyluce - INFO - Reading Chlamydia_trachomatis_GCF_000008725 genome\n",
      "2020-02-07 09:29:29,270 - Phyluce - INFO - Chlamydia_trachomatis_GCF_000008725: 227 uces, 0 dupes, 227 non-dupes, 1 orient drop, 11 length drop, 215 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/chlamydia/extract_probes_from_group/probe_fasta --output results/phyluce/chlamydia/extract_probes_from_group/multifastas.sqlite --base-taxon Chlamydia_trachomatis_GCF_000008725;\n",
      "chlamydia_abortus_gcf_002895085.\n",
      "chlamydia_avium_gcf_000583875.\n",
      "chlamydia_buteonis_gcf_900634605.\n",
      "chlamydia_caviae_gcf_000007605.\n",
      "chlamydia_felis_gcf_000009945.\n",
      "chlamydia_gallinacea_gcf_000471025.\n",
      "chlamydia_ibidis_gcf_000454725.\n",
      "chlamydia_muridarum_gcf_000006685.\n",
      "chlamydia_pecorum_gcf_000204135.\n",
      "chlamydia_pneumoniae_gcf_000008745.\n",
      "chlamydia_psittaci_gcf_000204255.\n",
      "chlamydia_sp_gcf_900239945.\n",
      "chlamydia_suis_gcf_900169085.\n",
      "chlamydia_trachomatis_gcf_000008725.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/chlamydia/extract_probes_from_group/multifastas.sqlite --base-taxon Chlamydia_trachomatis_GCF_000008725;\n",
      "Loci shared by 0 taxa:\t226.0\n",
      "Loci shared by 1 taxa:\t226.0\n",
      "Loci shared by 2 taxa:\t226.0\n",
      "Loci shared by 3 taxa:\t226.0\n",
      "Loci shared by 4 taxa:\t215.0\n",
      "Loci shared by 5 taxa:\t211.0\n",
      "Loci shared by 6 taxa:\t210.0\n",
      "Loci shared by 7 taxa:\t208.0\n",
      "Loci shared by 8 taxa:\t207.0\n",
      "Loci shared by 9 taxa:\t204.0\n",
      "Loci shared by 10 taxa:\t200.0\n",
      "Loci shared by 11 taxa:\t196.0\n",
      "Loci shared by 12 taxa:\t183.0\n",
      "Loci shared by 13 taxa:\t176.0\n",
      "Loci shared by 14 taxa:\t156.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/chlamydia/extract_probes_from_group/multifastas.sqlite  --base-taxon Chlamydia_trachomatis_GCF_000008725 --output results/phyluce/chlamydia/extract_probes_from_group/Chlamydia_trachomatis_GCF_000008725+2-back-to-14.conf --specific-counts 14;\n",
      "Counter({'chlamydia_buteonis_gcf_900634605': 156, 'chlamydia_avium_gcf_000583875': 156, 'chlamydia_gallinacea_gcf_000471025': 156, 'chlamydia_felis_gcf_000009945': 156, 'chlamydia_caviae_gcf_000007605': 156, 'chlamydia_sp_gcf_900239945': 156, 'chlamydia_pneumoniae_gcf_000008745': 156, 'chlamydia_trachomatis_gcf_000008725': 156, 'chlamydia_psittaci_gcf_000204255': 156, 'chlamydia_suis_gcf_900169085': 156, 'chlamydia_abortus_gcf_002895085': 156, 'chlamydia_pecorum_gcf_000204135': 156, 'chlamydia_ibidis_gcf_000454725': 156, 'chlamydia_muridarum_gcf_000006685': 156})\n",
      "Total loci = 156\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(14)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/chlamydia/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/chlamydia/extract_probes_from_group/Chlamydia_trachomatis_GCF_000008725+2-back-to-14.conf --probe-prefix uce_chlamydia_ --designer rnplattii --design chlamydia_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 156\n",
      "Probe Count = 4238\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list.fasta --query results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  09:29:43\n",
      "Ended:  Fri Feb 07, 2020  09:29:49\n",
      "Time for execution:  0.0977126995722 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list.fasta --lastz results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_chlamydia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 4237 fasta sequences.  Filtered 2 duplicates. Kept 4182.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/chlamydia/final_probe_design/chlamydia_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 09:31:56 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 4238\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 339040\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3957771\n",
      "Max number of word counting entries: 89089442\n",
      "\n",
      "# comparing sequences from          0  to        706\n",
      "---------- new table with      575 representatives\n",
      "# comparing sequences from        706  to       1294\n",
      "----------    129 remaining sequences to the next cycle\n",
      "---------- new table with      380 representatives\n",
      "# comparing sequences from       1165  to       1677\n",
      "----------    221 remaining sequences to the next cycle\n",
      "---------- new table with      253 representatives\n",
      "# comparing sequences from       1456  to       1919\n",
      "----------    173 remaining sequences to the next cycle\n",
      "---------- new table with      243 representatives\n",
      "# comparing sequences from       1746  to       2161\n",
      "----------     47 remaining sequences to the next cycle\n",
      "---------- new table with      299 representatives\n",
      "# comparing sequences from       2114  to       2468\n",
      "----------    146 remaining sequences to the next cycle\n",
      "---------- new table with      164 representatives\n",
      "# comparing sequences from       2322  to       2641\n",
      "----------     96 remaining sequences to the next cycle\n",
      "---------- new table with      188 representatives\n",
      "# comparing sequences from       2545  to       2827\n",
      "----------     37 remaining sequences to the next cycle\n",
      "---------- new table with      201 representatives\n",
      "# comparing sequences from       2790  to       3031\n",
      "----------     49 remaining sequences to the next cycle\n",
      "---------- new table with      150 representatives\n",
      "# comparing sequences from       2982  to       3191\n",
      "99.9%---------- new table with      163 representatives\n",
      "# comparing sequences from       3191  to       3365\n",
      "100.0%---------- new table with      120 representatives\n",
      "# comparing sequences from       3365  to       4238\n",
      ".....................---------- new table with      680 representatives\n",
      "\n",
      "     4238  finished       3416  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.96\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Coxiella burnetti"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'coxiella'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Coxiella_burnetii_GCF_000017105\",\n",
    "               \"Coxiella_burnetii_GCF_001572745\",\n",
    "               \"Coxiella_burnetii_GCF_004006195\",\n",
    "               \"Coxiella_burnetii_GCF_000019885\",\n",
    "               \"Coxiella_burnetii_GCF_000168875\",\n",
    "               \"Coxiella_burnetii_GCF_000018745\",\n",
    "               \"Coxiella_burnetii_GCF_000019865\",\n",
    "               \"Coxiella_burnetii_GCF_000367725\",\n",
    "               \"Coxiella_burnetii_GCF_004087735\",\n",
    "               \"Coxiella_burnetii_GCF_002109345\",\n",
    "               \"Coxiella_burnetii_GCF_002094935\",\n",
    "               \"Coxiella_burnetii_GCF_005280755\",\n",
    "               \"Coxiella_burnetii_GCF_002634005\",\n",
    "               \"Coxiella_burnetii_GCF_002633985\",\n",
    "               \"Coxiella_burnetii_GCF_002633965\",\n",
    "               \"Coxiella_burnetii_GCF_002633945\",\n",
    "               \"Coxiella_burnetii_GCF_002633885\",\n",
    "               \"Coxiella_burnetii_GCF_002634025\",\n",
    "               \"Coxiella_burnetii_GCF_002634045\",\n",
    "               \"Coxiella_burnetii_GCF_000767035\",\n",
    "               \"Coxiella_burnetii_GCF_002634065\",\n",
    "               \"Coxiella_burnetii_GCF_002633925\",\n",
    "               \"Coxiella_burnetii_GCF_002633905\",\n",
    "               \"Coxiella_burnetii_GCF_001572765\",\n",
    "               \"Coxiella_burnetii_GCF_002634085\",\n",
    "               \"Coxiella_burnetii_GCF_000612785\",\n",
    "               \"Coxiella_burnetii_GCF_000826165\",\n",
    "               \"Coxiella_burnetii_GCF_000820465\",\n",
    "               \"Coxiella_burnetii_GCF_000967075\",\n",
    "               \"Coxiella_burnetii_GCF_000470495\",\n",
    "               \"Coxiella_burnetii_GCF_002247545\",\n",
    "               \"Coxiella_burnetii_GCF_002247155\",\n",
    "               \"Coxiella_burnetii_GCF_002896835\",\n",
    "               \"Coxiella_burnetii_GCF_000613025\",\n",
    "               \"Coxiella_burnetii_GCF_000723305\",\n",
    "               \"Coxiella_burnetii_GCF_002924395\",\n",
    "               \"Coxiella_burnetii_GCF_002247185\",\n",
    "               \"Coxiella_burnetii_GCF_002896735\",\n",
    "               \"Coxiella_burnetii_GCF_002896755\",\n",
    "               \"Coxiella_burnetii_GCF_002896775\",\n",
    "               \"Coxiella_burnetii_GCF_002247265\",\n",
    "               \"Coxiella_burnetii_GCF_002247205\",\n",
    "               \"Coxiella_burnetii_GCF_002924305\",\n",
    "               \"Coxiella_burnetii_GCF_002924345\",\n",
    "               \"Coxiella_burnetii_GCF_002924425\",\n",
    "               \"Coxiella_burnetii_GCF_000723245\",\n",
    "               \"Coxiella_burnetii_GCF_002247285\",\n",
    "               \"Coxiella_burnetii_GCF_002247305\",\n",
    "               \"Coxiella_burnetii_GCF_002247335\",\n",
    "               \"Coxiella_burnetii_GCF_002249845\",\n",
    "               \"Coxiella_burnetii_GCF_002924325\",\n",
    "               \"Coxiella_burnetii_GCF_003849965\",\n",
    "               \"Coxiella_burnetii_GCF_002247225\",\n",
    "               \"Coxiella_burnetii_GCF_002896795\",\n",
    "               \"Coxiella_burnetii_GCF_002924385\",\n",
    "               \"Coxiella_burnetii_GCF_000751935\",\n",
    "               \"Coxiella_burnetii_GCF_000820825\",\n",
    "               \"Coxiella_burnetii_GCF_000756325\",\n",
    "               \"Coxiella_burnetii_GCF_002591355\",\n",
    "               \"Coxiella_burnetii_GCF_000169495\",\n",
    "               \"Coxiella_burnetii_GCF_000723285\",\n",
    "               \"Coxiella_burnetii_GCF_002896815\",\n",
    "               \"Coxiella_burnetii_GCF_003849885\",\n",
    "               \"Coxiella_burnetii_GCF_003849875\",\n",
    "               \"Coxiella_burnetii_GCF_003849785\",\n",
    "               \"Coxiella_burnetii_GCF_000723265\",\n",
    "               \"Coxiella_burnetii_GCF_000300315\",\n",
    "               \"Coxiella_burnetii_GCA_000359545\" ]\n",
    "                    \n",
    "reference_taxon = \"Coxiella_burnetii_GCF_000007765\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007765.2_ASM776v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 605169 bytes  242084.40 bytes/sec\n",
      "total size is 604914  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017105.1_ASM1710v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 659451 bytes  439662.00 bytes/sec\n",
      "total size is 659179  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001572745.1_ASM157274v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 619874 bytes  413277.33 bytes/sec\n",
      "total size is 619616  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004006195.1_ASM400619v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 608351 bytes  243357.20 bytes/sec\n",
      "total size is 608093  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000019885.1_ASM1988v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 623461 bytes  415668.67 bytes/sec\n",
      "total size is 623197  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000168875.3_ASM16887v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 623524 bytes  249426.40 bytes/sec\n",
      "total size is 623259  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000018745.1_ASM1874v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 609250 bytes  406194.67 bytes/sec\n",
      "total size is 608994  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000019865.1_ASM1986v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597049 bytes  398060.67 bytes/sec\n",
      "total size is 596793  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000367725.3_Z3055_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 594265 bytes  237722.80 bytes/sec\n",
      "total size is 594013  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004087735.1_ASM408773v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 599598 bytes  399760.00 bytes/sec\n",
      "total size is 599340  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002109345.1_ASM210934v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597412 bytes  238981.60 bytes/sec\n",
      "total size is 597154  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002094935.1_ASM209493v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597419 bytes  398307.33 bytes/sec\n",
      "total size is 597161  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005280755.1_ASM528075v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597371 bytes  238965.20 bytes/sec\n",
      "total size is 597113  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002634005.1_ASM263400v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597633 bytes  398450.00 bytes/sec\n",
      "total size is 597375  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002633985.1_ASM263398v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 598140 bytes  398788.00 bytes/sec\n",
      "total size is 597882  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002633965.1_ASM263396v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 598067 bytes  239243.60 bytes/sec\n",
      "total size is 597809  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002633945.1_ASM263394v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597452 bytes  398329.33 bytes/sec\n",
      "total size is 597194  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002633885.1_ASM263388v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 587882 bytes  391949.33 bytes/sec\n",
      "total size is 587632  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002634025.1_ASM263402v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 595903 bytes  397296.67 bytes/sec\n",
      "total size is 595645  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002634045.1_ASM263404v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 595629 bytes  397114.00 bytes/sec\n",
      "total size is 595371  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000767035.1_ASM76703v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 610232 bytes  406849.33 bytes/sec\n",
      "total size is 609975  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002634065.1_ASM263406v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 608536 bytes  1217156.00 bytes/sec\n",
      "total size is 608278  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002633925.1_ASM263392v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601647 bytes  1203378.00 bytes/sec\n",
      "total size is 601389  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002633905.1_ASM263390v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602387 bytes  401619.33 bytes/sec\n",
      "total size is 602129  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001572765.1_ASM157276v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 595298 bytes  396893.33 bytes/sec\n",
      "total size is 595040  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002634085.1_ASM263408v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 591967 bytes  394672.67 bytes/sec\n",
      "total size is 591709  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000612785.1_Cb-C2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 604561 bytes  93015.85 bytes/sec\n",
      "total size is 604309  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000826165.2_Coxiella_burnetii_Cb171_QLYMPHOMA_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 620084 bytes  413417.33 bytes/sec\n",
      "total size is 619804  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000820465.1_Coxiella_burnetii_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 596982 bytes  398016.00 bytes/sec\n",
      "total size is 596718  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000967075.1_ASM96707v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 636538 bytes  424386.67 bytes/sec\n",
      "total size is 636273  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000470495.1_Cb185_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 593546 bytes  395725.33 bytes/sec\n",
      "total size is 593294  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247545.1_ASM224754v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 656777 bytes  437879.33 bytes/sec\n",
      "total size is 656503  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002247155.1_ASM224715v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657149 bytes  438127.33 bytes/sec\n",
      "total size is 656875  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002896835.1_ASM289683v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602458 bytes  401666.67 bytes/sec\n",
      "total size is 602200  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000613025.1_Cb-B1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602783 bytes  401883.33 bytes/sec\n",
      "total size is 602531  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000723305.1_Cb-B18_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 603153 bytes  402130.00 bytes/sec\n",
      "total size is 602900  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002924395.1_ASM292439v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 591704 bytes  394497.33 bytes/sec\n",
      "total size is 591446  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247185.1_ASM224718v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601877 bytes  401279.33 bytes/sec\n",
      "total size is 601619  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002896735.1_ASM289673v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601963 bytes  401336.67 bytes/sec\n",
      "total size is 601705  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002896755.1_ASM289675v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602191 bytes  401488.67 bytes/sec\n",
      "total size is 601933  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002896775.1_ASM289677v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602724 bytes  401844.00 bytes/sec\n",
      "total size is 602466  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247265.1_ASM224726v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601637 bytes  401119.33 bytes/sec\n",
      "total size is 601379  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247205.1_ASM224720v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601515 bytes  401038.00 bytes/sec\n",
      "total size is 601257  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002924305.1_ASM292430v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601463 bytes  401003.33 bytes/sec\n",
      "total size is 601205  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002924345.1_ASM292434v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601623 bytes  401110.00 bytes/sec\n",
      "total size is 601365  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002924425.1_ASM292442v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601459 bytes  401000.67 bytes/sec\n",
      "total size is 601201  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000723245.1_EV-Cb-BK10_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 600571 bytes  400408.67 bytes/sec\n",
      "total size is 600314  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247285.1_ASM224728v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601652 bytes  70787.53 bytes/sec\n",
      "total size is 601394  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247305.1_ASM224730v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601807 bytes  401232.67 bytes/sec\n",
      "total size is 601549  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247335.1_ASM224733v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601782 bytes  401216.00 bytes/sec\n",
      "total size is 601524  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002249845.1_ASM224984v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601724 bytes  401177.33 bytes/sec\n",
      "total size is 601466  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002924325.1_ASM292432v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602079 bytes  133804.67 bytes/sec\n",
      "total size is 601821  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003849965.1_ASM384996v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602548 bytes  401726.67 bytes/sec\n",
      "total size is 602290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002247225.1_ASM224722v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 592110 bytes  394768.00 bytes/sec\n",
      "total size is 591852  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002896795.1_ASM289679v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597094 bytes  398090.67 bytes/sec\n",
      "total size is 596836  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002924385.1_ASM292438v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601801 bytes  401228.67 bytes/sec\n",
      "total size is 601543  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000751935.2_Coxiella_burnetii_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 586693 bytes  391156.67 bytes/sec\n",
      "total size is 586437  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000820825.1_Coxiella_burnetii_Dog_Utad_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 588901 bytes  1177886.00 bytes/sec\n",
      "total size is 588636  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000756325.1_CoxAra_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 620655 bytes  413798.00 bytes/sec\n",
      "total size is 620402  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002591355.1_ASM259135v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 627194 bytes  418157.33 bytes/sec\n",
      "total size is 626928  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000169495.2_ASM16949v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 598301 bytes  398895.33 bytes/sec\n",
      "total size is 598044  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000723285.1_EV-Cb-C13_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 604197 bytes  402826.00 bytes/sec\n",
      "total size is 603941  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002896815.1_ASM289681v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 603699 bytes  402494.00 bytes/sec\n",
      "total size is 603441  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003849885.1_ASM384988v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 600558 bytes  400400.00 bytes/sec\n",
      "total size is 600300  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003849875.1_ASM384987v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 600910 bytes  400634.67 bytes/sec\n",
      "total size is 600652  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003849785.1_ASM384978v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 599846 bytes  399925.33 bytes/sec\n",
      "total size is 599588  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000723265.1_Cb-O184_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 645790 bytes  430554.67 bytes/sec\n",
      "total size is 645528  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000300315.1_ASM30031v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 588648 bytes  392460.00 bytes/sec\n",
      "total size is 588399  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000359545.5_G255_GS_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 603448 bytes  402326.67 bytes/sec\n",
      "total size is 603194  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482669 (\"sim_Coxiella_burnetii_GCF_000017105\") has been submitted\n",
      "Your job 5482670 (\"sim_Coxiella_burnetii_GCF_001572745\") has been submitted\n",
      "Your job 5482671 (\"sim_Coxiella_burnetii_GCF_004006195\") has been submitted\n",
      "Your job 5482672 (\"sim_Coxiella_burnetii_GCF_000019885\") has been submitted\n",
      "Your job 5482673 (\"sim_Coxiella_burnetii_GCF_000168875\") has been submitted\n",
      "Your job 5482674 (\"sim_Coxiella_burnetii_GCF_000018745\") has been submitted\n",
      "Your job 5482675 (\"sim_Coxiella_burnetii_GCF_000019865\") has been submitted\n",
      "Your job 5482676 (\"sim_Coxiella_burnetii_GCF_000367725\") has been submitted\n",
      "Your job 5482677 (\"sim_Coxiella_burnetii_GCF_004087735\") has been submitted\n",
      "Your job 5482678 (\"sim_Coxiella_burnetii_GCF_002109345\") has been submitted\n",
      "Your job 5482679 (\"sim_Coxiella_burnetii_GCF_002094935\") has been submitted\n",
      "Your job 5482680 (\"sim_Coxiella_burnetii_GCF_005280755\") has been submitted\n",
      "Your job 5482681 (\"sim_Coxiella_burnetii_GCF_002634005\") has been submitted\n",
      "Your job 5482682 (\"sim_Coxiella_burnetii_GCF_002633985\") has been submitted\n",
      "Your job 5482683 (\"sim_Coxiella_burnetii_GCF_002633965\") has been submitted\n",
      "Your job 5482684 (\"sim_Coxiella_burnetii_GCF_002633945\") has been submitted\n",
      "Your job 5482685 (\"sim_Coxiella_burnetii_GCF_002633885\") has been submitted\n",
      "Your job 5482686 (\"sim_Coxiella_burnetii_GCF_002634025\") has been submitted\n",
      "Your job 5482687 (\"sim_Coxiella_burnetii_GCF_002634045\") has been submitted\n",
      "Your job 5482688 (\"sim_Coxiella_burnetii_GCF_000767035\") has been submitted\n",
      "Your job 5482689 (\"sim_Coxiella_burnetii_GCF_002634065\") has been submitted\n",
      "Your job 5482690 (\"sim_Coxiella_burnetii_GCF_002633925\") has been submitted\n",
      "Your job 5482691 (\"sim_Coxiella_burnetii_GCF_002633905\") has been submitted\n",
      "Your job 5482692 (\"sim_Coxiella_burnetii_GCF_001572765\") has been submitted\n",
      "Your job 5482693 (\"sim_Coxiella_burnetii_GCF_002634085\") has been submitted\n",
      "Your job 5482694 (\"sim_Coxiella_burnetii_GCF_000612785\") has been submitted\n",
      "Your job 5482695 (\"sim_Coxiella_burnetii_GCF_000826165\") has been submitted\n",
      "Your job 5482696 (\"sim_Coxiella_burnetii_GCF_000820465\") has been submitted\n",
      "Your job 5482697 (\"sim_Coxiella_burnetii_GCF_000967075\") has been submitted\n",
      "Your job 5482698 (\"sim_Coxiella_burnetii_GCF_000470495\") has been submitted\n",
      "Your job 5482699 (\"sim_Coxiella_burnetii_GCF_002247545\") has been submitted\n",
      "Your job 5482700 (\"sim_Coxiella_burnetii_GCF_002247155\") has been submitted\n",
      "Your job 5482701 (\"sim_Coxiella_burnetii_GCF_002896835\") has been submitted\n",
      "Your job 5482702 (\"sim_Coxiella_burnetii_GCF_000613025\") has been submitted\n",
      "Your job 5482703 (\"sim_Coxiella_burnetii_GCF_000723305\") has been submitted\n",
      "Your job 5482704 (\"sim_Coxiella_burnetii_GCF_002924395\") has been submitted\n",
      "Your job 5482705 (\"sim_Coxiella_burnetii_GCF_002247185\") has been submitted\n",
      "Your job 5482706 (\"sim_Coxiella_burnetii_GCF_002896735\") has been submitted\n",
      "Your job 5482707 (\"sim_Coxiella_burnetii_GCF_002896755\") has been submitted\n",
      "Your job 5482708 (\"sim_Coxiella_burnetii_GCF_002896775\") has been submitted\n",
      "Your job 5482709 (\"sim_Coxiella_burnetii_GCF_002247265\") has been submitted\n",
      "Your job 5482710 (\"sim_Coxiella_burnetii_GCF_002247205\") has been submitted\n",
      "Your job 5482711 (\"sim_Coxiella_burnetii_GCF_002924305\") has been submitted\n",
      "Your job 5482712 (\"sim_Coxiella_burnetii_GCF_002924345\") has been submitted\n",
      "Your job 5482713 (\"sim_Coxiella_burnetii_GCF_002924425\") has been submitted\n",
      "Your job 5482714 (\"sim_Coxiella_burnetii_GCF_000723245\") has been submitted\n",
      "Your job 5482715 (\"sim_Coxiella_burnetii_GCF_002247285\") has been submitted\n",
      "Your job 5482716 (\"sim_Coxiella_burnetii_GCF_002247305\") has been submitted\n",
      "Your job 5482717 (\"sim_Coxiella_burnetii_GCF_002247335\") has been submitted\n",
      "Your job 5482718 (\"sim_Coxiella_burnetii_GCF_002249845\") has been submitted\n",
      "Your job 5482719 (\"sim_Coxiella_burnetii_GCF_002924325\") has been submitted\n",
      "Your job 5482720 (\"sim_Coxiella_burnetii_GCF_003849965\") has been submitted\n",
      "Your job 5482721 (\"sim_Coxiella_burnetii_GCF_002247225\") has been submitted\n",
      "Your job 5482722 (\"sim_Coxiella_burnetii_GCF_002896795\") has been submitted\n",
      "Your job 5482723 (\"sim_Coxiella_burnetii_GCF_002924385\") has been submitted\n",
      "Your job 5482724 (\"sim_Coxiella_burnetii_GCF_000751935\") has been submitted\n",
      "Your job 5482725 (\"sim_Coxiella_burnetii_GCF_000820825\") has been submitted\n",
      "Your job 5482726 (\"sim_Coxiella_burnetii_GCF_000756325\") has been submitted\n",
      "Your job 5482727 (\"sim_Coxiella_burnetii_GCF_002591355\") has been submitted\n",
      "Your job 5482728 (\"sim_Coxiella_burnetii_GCF_000169495\") has been submitted\n",
      "Your job 5482729 (\"sim_Coxiella_burnetii_GCF_000723285\") has been submitted\n",
      "Your job 5482730 (\"sim_Coxiella_burnetii_GCF_002896815\") has been submitted\n",
      "Your job 5482731 (\"sim_Coxiella_burnetii_GCF_003849885\") has been submitted\n",
      "Your job 5482732 (\"sim_Coxiella_burnetii_GCF_003849875\") has been submitted\n",
      "Your job 5482733 (\"sim_Coxiella_burnetii_GCF_003849785\") has been submitted\n",
      "Your job 5482734 (\"sim_Coxiella_burnetii_GCF_000723265\") has been submitted\n",
      "Your job 5482735 (\"sim_Coxiella_burnetii_GCF_000300315\") has been submitted\n",
      "Your job 5482736 (\"sim_Coxiella_burnetii_GCA_000359545\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/coxiella/cleaned_genomes/Coxiella_burnetii_GCF_000007765_formatted.fas path=results/phyluce/coxiella/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482737 (\"map_Coxiella_burnetii_GCF_000017105\") has been submitted\n",
      "Your job 5482738 (\"map_Coxiella_burnetii_GCF_001572745\") has been submitted\n",
      "Your job 5482739 (\"map_Coxiella_burnetii_GCF_004006195\") has been submitted\n",
      "Your job 5482740 (\"map_Coxiella_burnetii_GCF_000019885\") has been submitted\n",
      "Your job 5482741 (\"map_Coxiella_burnetii_GCF_000168875\") has been submitted\n",
      "Your job 5482742 (\"map_Coxiella_burnetii_GCF_000018745\") has been submitted\n",
      "Your job 5482743 (\"map_Coxiella_burnetii_GCF_000019865\") has been submitted\n",
      "Your job 5482744 (\"map_Coxiella_burnetii_GCF_000367725\") has been submitted\n",
      "Your job 5482745 (\"map_Coxiella_burnetii_GCF_004087735\") has been submitted\n",
      "Your job 5482746 (\"map_Coxiella_burnetii_GCF_002109345\") has been submitted\n",
      "Your job 5482747 (\"map_Coxiella_burnetii_GCF_002094935\") has been submitted\n",
      "Your job 5482748 (\"map_Coxiella_burnetii_GCF_005280755\") has been submitted\n",
      "Your job 5482749 (\"map_Coxiella_burnetii_GCF_002634005\") has been submitted\n",
      "Your job 5482750 (\"map_Coxiella_burnetii_GCF_002633985\") has been submitted\n",
      "Your job 5482751 (\"map_Coxiella_burnetii_GCF_002633965\") has been submitted\n",
      "Your job 5482752 (\"map_Coxiella_burnetii_GCF_002633945\") has been submitted\n",
      "Your job 5482753 (\"map_Coxiella_burnetii_GCF_002633885\") has been submitted\n",
      "Your job 5482754 (\"map_Coxiella_burnetii_GCF_002634025\") has been submitted\n",
      "Your job 5482755 (\"map_Coxiella_burnetii_GCF_002634045\") has been submitted\n",
      "Your job 5482756 (\"map_Coxiella_burnetii_GCF_000767035\") has been submitted\n",
      "Your job 5482757 (\"map_Coxiella_burnetii_GCF_002634065\") has been submitted\n",
      "Your job 5482758 (\"map_Coxiella_burnetii_GCF_002633925\") has been submitted\n",
      "Your job 5482759 (\"map_Coxiella_burnetii_GCF_002633905\") has been submitted\n",
      "Your job 5482760 (\"map_Coxiella_burnetii_GCF_001572765\") has been submitted\n",
      "Your job 5482761 (\"map_Coxiella_burnetii_GCF_002634085\") has been submitted\n",
      "Your job 5482762 (\"map_Coxiella_burnetii_GCF_000612785\") has been submitted\n",
      "Your job 5482763 (\"map_Coxiella_burnetii_GCF_000826165\") has been submitted\n",
      "Your job 5482764 (\"map_Coxiella_burnetii_GCF_000820465\") has been submitted\n",
      "Your job 5482765 (\"map_Coxiella_burnetii_GCF_000967075\") has been submitted\n",
      "Your job 5482766 (\"map_Coxiella_burnetii_GCF_000470495\") has been submitted\n",
      "Your job 5482767 (\"map_Coxiella_burnetii_GCF_002247545\") has been submitted\n",
      "Your job 5482768 (\"map_Coxiella_burnetii_GCF_002247155\") has been submitted\n",
      "Your job 5482769 (\"map_Coxiella_burnetii_GCF_002896835\") has been submitted\n",
      "Your job 5482770 (\"map_Coxiella_burnetii_GCF_000613025\") has been submitted\n",
      "Your job 5482771 (\"map_Coxiella_burnetii_GCF_000723305\") has been submitted\n",
      "Your job 5482772 (\"map_Coxiella_burnetii_GCF_002924395\") has been submitted\n",
      "Your job 5482773 (\"map_Coxiella_burnetii_GCF_002247185\") has been submitted\n",
      "Your job 5482774 (\"map_Coxiella_burnetii_GCF_002896735\") has been submitted\n",
      "Your job 5482775 (\"map_Coxiella_burnetii_GCF_002896755\") has been submitted\n",
      "Your job 5482776 (\"map_Coxiella_burnetii_GCF_002896775\") has been submitted\n",
      "Your job 5482777 (\"map_Coxiella_burnetii_GCF_002247265\") has been submitted\n",
      "Your job 5482778 (\"map_Coxiella_burnetii_GCF_002247205\") has been submitted\n",
      "Your job 5482779 (\"map_Coxiella_burnetii_GCF_002924305\") has been submitted\n",
      "Your job 5482780 (\"map_Coxiella_burnetii_GCF_002924345\") has been submitted\n",
      "Your job 5482781 (\"map_Coxiella_burnetii_GCF_002924425\") has been submitted\n",
      "Your job 5482782 (\"map_Coxiella_burnetii_GCF_000723245\") has been submitted\n",
      "Your job 5482783 (\"map_Coxiella_burnetii_GCF_002247285\") has been submitted\n",
      "Your job 5482784 (\"map_Coxiella_burnetii_GCF_002247305\") has been submitted\n",
      "Your job 5482785 (\"map_Coxiella_burnetii_GCF_002247335\") has been submitted\n",
      "Your job 5482786 (\"map_Coxiella_burnetii_GCF_002249845\") has been submitted\n",
      "Your job 5482787 (\"map_Coxiella_burnetii_GCF_002924325\") has been submitted\n",
      "Your job 5482788 (\"map_Coxiella_burnetii_GCF_003849965\") has been submitted\n",
      "Your job 5482789 (\"map_Coxiella_burnetii_GCF_002247225\") has been submitted\n",
      "Your job 5482790 (\"map_Coxiella_burnetii_GCF_002896795\") has been submitted\n",
      "Your job 5482791 (\"map_Coxiella_burnetii_GCF_002924385\") has been submitted\n",
      "Your job 5482792 (\"map_Coxiella_burnetii_GCF_000751935\") has been submitted\n",
      "Your job 5482793 (\"map_Coxiella_burnetii_GCF_000820825\") has been submitted\n",
      "Your job 5482794 (\"map_Coxiella_burnetii_GCF_000756325\") has been submitted\n",
      "Your job 5482795 (\"map_Coxiella_burnetii_GCF_002591355\") has been submitted\n",
      "Your job 5482796 (\"map_Coxiella_burnetii_GCF_000169495\") has been submitted\n",
      "Your job 5482797 (\"map_Coxiella_burnetii_GCF_000723285\") has been submitted\n",
      "Your job 5482798 (\"map_Coxiella_burnetii_GCF_002896815\") has been submitted\n",
      "Your job 5482799 (\"map_Coxiella_burnetii_GCF_003849885\") has been submitted\n",
      "Your job 5482800 (\"map_Coxiella_burnetii_GCF_003849875\") has been submitted\n",
      "Your job 5482801 (\"map_Coxiella_burnetii_GCF_003849785\") has been submitted\n",
      "Your job 5482802 (\"map_Coxiella_burnetii_GCF_000723265\") has been submitted\n",
      "Your job 5482803 (\"map_Coxiella_burnetii_GCF_000300315\") has been submitted\n",
      "Your job 5482804 (\"map_Coxiella_burnetii_GCA_000359545\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482805 (\"merge_Coxiella_burnetii_GCF_000017105\") has been submitted\n",
      "Your job 5482806 (\"merge_Coxiella_burnetii_GCF_001572745\") has been submitted\n",
      "Your job 5482807 (\"merge_Coxiella_burnetii_GCF_004006195\") has been submitted\n",
      "Your job 5482808 (\"merge_Coxiella_burnetii_GCF_000019885\") has been submitted\n",
      "Your job 5482809 (\"merge_Coxiella_burnetii_GCF_000168875\") has been submitted\n",
      "Your job 5482810 (\"merge_Coxiella_burnetii_GCF_000018745\") has been submitted\n",
      "Your job 5482811 (\"merge_Coxiella_burnetii_GCF_000019865\") has been submitted\n",
      "Your job 5482812 (\"merge_Coxiella_burnetii_GCF_000367725\") has been submitted\n",
      "Your job 5482813 (\"merge_Coxiella_burnetii_GCF_004087735\") has been submitted\n",
      "Your job 5482814 (\"merge_Coxiella_burnetii_GCF_002109345\") has been submitted\n",
      "Your job 5482815 (\"merge_Coxiella_burnetii_GCF_002094935\") has been submitted\n",
      "Your job 5482816 (\"merge_Coxiella_burnetii_GCF_005280755\") has been submitted\n",
      "Your job 5482817 (\"merge_Coxiella_burnetii_GCF_002634005\") has been submitted\n",
      "Your job 5482818 (\"merge_Coxiella_burnetii_GCF_002633985\") has been submitted\n",
      "Your job 5482819 (\"merge_Coxiella_burnetii_GCF_002633965\") has been submitted\n",
      "Your job 5482820 (\"merge_Coxiella_burnetii_GCF_002633945\") has been submitted\n",
      "Your job 5482821 (\"merge_Coxiella_burnetii_GCF_002633885\") has been submitted\n",
      "Your job 5482822 (\"merge_Coxiella_burnetii_GCF_002634025\") has been submitted\n",
      "Your job 5482823 (\"merge_Coxiella_burnetii_GCF_002634045\") has been submitted\n",
      "Your job 5482824 (\"merge_Coxiella_burnetii_GCF_000767035\") has been submitted\n",
      "Your job 5482825 (\"merge_Coxiella_burnetii_GCF_002634065\") has been submitted\n",
      "Your job 5482826 (\"merge_Coxiella_burnetii_GCF_002633925\") has been submitted\n",
      "Your job 5482827 (\"merge_Coxiella_burnetii_GCF_002633905\") has been submitted\n",
      "Your job 5482828 (\"merge_Coxiella_burnetii_GCF_001572765\") has been submitted\n",
      "Your job 5482829 (\"merge_Coxiella_burnetii_GCF_002634085\") has been submitted\n",
      "Your job 5482830 (\"merge_Coxiella_burnetii_GCF_000612785\") has been submitted\n",
      "Your job 5482831 (\"merge_Coxiella_burnetii_GCF_000826165\") has been submitted\n",
      "Your job 5482832 (\"merge_Coxiella_burnetii_GCF_000820465\") has been submitted\n",
      "Your job 5482833 (\"merge_Coxiella_burnetii_GCF_000967075\") has been submitted\n",
      "Your job 5482834 (\"merge_Coxiella_burnetii_GCF_000470495\") has been submitted\n",
      "Your job 5482835 (\"merge_Coxiella_burnetii_GCF_002247545\") has been submitted\n",
      "Your job 5482836 (\"merge_Coxiella_burnetii_GCF_002247155\") has been submitted\n",
      "Your job 5482837 (\"merge_Coxiella_burnetii_GCF_002896835\") has been submitted\n",
      "Your job 5482838 (\"merge_Coxiella_burnetii_GCF_000613025\") has been submitted\n",
      "Your job 5482839 (\"merge_Coxiella_burnetii_GCF_000723305\") has been submitted\n",
      "Your job 5482840 (\"merge_Coxiella_burnetii_GCF_002924395\") has been submitted\n",
      "Your job 5482841 (\"merge_Coxiella_burnetii_GCF_002247185\") has been submitted\n",
      "Your job 5482842 (\"merge_Coxiella_burnetii_GCF_002896735\") has been submitted\n",
      "Your job 5482843 (\"merge_Coxiella_burnetii_GCF_002896755\") has been submitted\n",
      "Your job 5482844 (\"merge_Coxiella_burnetii_GCF_002896775\") has been submitted\n",
      "Your job 5482845 (\"merge_Coxiella_burnetii_GCF_002247265\") has been submitted\n",
      "Your job 5482846 (\"merge_Coxiella_burnetii_GCF_002247205\") has been submitted\n",
      "Your job 5482847 (\"merge_Coxiella_burnetii_GCF_002924305\") has been submitted\n",
      "Your job 5482848 (\"merge_Coxiella_burnetii_GCF_002924345\") has been submitted\n",
      "Your job 5482849 (\"merge_Coxiella_burnetii_GCF_002924425\") has been submitted\n",
      "Your job 5482850 (\"merge_Coxiella_burnetii_GCF_000723245\") has been submitted\n",
      "Your job 5482851 (\"merge_Coxiella_burnetii_GCF_002247285\") has been submitted\n",
      "Your job 5482852 (\"merge_Coxiella_burnetii_GCF_002247305\") has been submitted\n",
      "Your job 5482853 (\"merge_Coxiella_burnetii_GCF_002247335\") has been submitted\n",
      "Your job 5482854 (\"merge_Coxiella_burnetii_GCF_002249845\") has been submitted\n",
      "Your job 5482855 (\"merge_Coxiella_burnetii_GCF_002924325\") has been submitted\n",
      "Your job 5482856 (\"merge_Coxiella_burnetii_GCF_003849965\") has been submitted\n",
      "Your job 5482857 (\"merge_Coxiella_burnetii_GCF_002247225\") has been submitted\n",
      "Your job 5482858 (\"merge_Coxiella_burnetii_GCF_002896795\") has been submitted\n",
      "Your job 5482859 (\"merge_Coxiella_burnetii_GCF_002924385\") has been submitted\n",
      "Your job 5482860 (\"merge_Coxiella_burnetii_GCF_000751935\") has been submitted\n",
      "Your job 5482861 (\"merge_Coxiella_burnetii_GCF_000820825\") has been submitted\n",
      "Your job 5482862 (\"merge_Coxiella_burnetii_GCF_000756325\") has been submitted\n",
      "Your job 5482863 (\"merge_Coxiella_burnetii_GCF_002591355\") has been submitted\n",
      "Your job 5482864 (\"merge_Coxiella_burnetii_GCF_000169495\") has been submitted\n",
      "Your job 5482865 (\"merge_Coxiella_burnetii_GCF_000723285\") has been submitted\n",
      "Your job 5482866 (\"merge_Coxiella_burnetii_GCF_002896815\") has been submitted\n",
      "Your job 5482867 (\"merge_Coxiella_burnetii_GCF_003849885\") has been submitted\n",
      "Your job 5482868 (\"merge_Coxiella_burnetii_GCF_003849875\") has been submitted\n",
      "Your job 5482869 (\"merge_Coxiella_burnetii_GCF_003849785\") has been submitted\n",
      "Your job 5482870 (\"merge_Coxiella_burnetii_GCF_000723265\") has been submitted\n",
      "Your job 5482871 (\"merge_Coxiella_burnetii_GCF_000300315\") has been submitted\n",
      "Your job 5482872 (\"merge_Coxiella_burnetii_GCA_000359545\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 109 sequences from Coxiella_burnetii_GCF_000017105_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 130 sequences from Coxiella_burnetii_GCF_001572745_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 126.\n",
      "Screened 140 sequences from Coxiella_burnetii_GCF_004006195_merged.bed.  Filtered 5 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 135.\n",
      "Screened 142 sequences from Coxiella_burnetii_GCF_000019885_merged.bed.  Filtered 8 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 134.\n",
      "Screened 130 sequences from Coxiella_burnetii_GCF_000168875_merged.bed.  Filtered 8 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 122.\n",
      "Screened 102 sequences from Coxiella_burnetii_GCF_000018745_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96.\n",
      "Screened 122 sequences from Coxiella_burnetii_GCF_000019865_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 118.\n",
      "Screened 124 sequences from Coxiella_burnetii_GCF_000367725_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 50 sequences from Coxiella_burnetii_GCF_004087735_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 49.\n",
      "Screened 52 sequences from Coxiella_burnetii_GCF_002109345_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 48 sequences from Coxiella_burnetii_GCF_002094935_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 48.\n",
      "Screened 57 sequences from Coxiella_burnetii_GCF_005280755_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 56.\n",
      "Screened 84 sequences from Coxiella_burnetii_GCF_002634005_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 82.\n",
      "Screened 80 sequences from Coxiella_burnetii_GCF_002633985_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 85 sequences from Coxiella_burnetii_GCF_002633965_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 84.\n",
      "Screened 88 sequences from Coxiella_burnetii_GCF_002633945_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 87.\n",
      "Screened 126 sequences from Coxiella_burnetii_GCF_002633885_merged.bed.  Filtered 5 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 121.\n",
      "Screened 232 sequences from Coxiella_burnetii_GCF_002634025_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 222.\n",
      "Screened 214 sequences from Coxiella_burnetii_GCF_002634045_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 208.\n",
      "Screened 158 sequences from Coxiella_burnetii_GCF_000767035_merged.bed.  Filtered 8 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 150.\n",
      "Screened 145 sequences from Coxiella_burnetii_GCF_002634065_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 139.\n",
      "Screened 248 sequences from Coxiella_burnetii_GCF_002633925_merged.bed.  Filtered 5 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 243.\n",
      "Screened 135 sequences from Coxiella_burnetii_GCF_002633905_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 271 sequences from Coxiella_burnetii_GCF_001572765_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 261.\n",
      "Screened 201 sequences from Coxiella_burnetii_GCF_002634085_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 184.\n",
      "Screened 102 sequences from Coxiella_burnetii_GCF_000612785_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 100.\n",
      "Screened 147 sequences from Coxiella_burnetii_GCF_000826165_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 135.\n",
      "Screened 122 sequences from Coxiella_burnetii_GCF_000820465_merged.bed.  Filtered 8 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 114.\n",
      "Screened 135 sequences from Coxiella_burnetii_GCF_000967075_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 129.\n",
      "Screened 89 sequences from Coxiella_burnetii_GCF_000470495_merged.bed.  Filtered 3 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 86.\n",
      "Screened 110 sequences from Coxiella_burnetii_GCF_002247545_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 100.\n",
      "Screened 117 sequences from Coxiella_burnetii_GCF_002247155_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 84 sequences from Coxiella_burnetii_GCF_002896835_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 89 sequences from Coxiella_burnetii_GCF_000613025_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 87.\n",
      "Screened 95 sequences from Coxiella_burnetii_GCF_000723305_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 54 sequences from Coxiella_burnetii_GCF_002924395_merged.bed.  Filtered 3 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 51.\n",
      "Screened 79 sequences from Coxiella_burnetii_GCF_002247185_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 77.\n",
      "Screened 84 sequences from Coxiella_burnetii_GCF_002896735_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 87 sequences from Coxiella_burnetii_GCF_002896755_merged.bed.  Filtered 3 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 84.\n",
      "Screened 87 sequences from Coxiella_burnetii_GCF_002896775_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 86.\n",
      "Screened 50 sequences from Coxiella_burnetii_GCF_002247265_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 49.\n",
      "Screened 46 sequences from Coxiella_burnetii_GCF_002247205_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 46.\n",
      "Screened 41 sequences from Coxiella_burnetii_GCF_002924305_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 41.\n",
      "Screened 39 sequences from Coxiella_burnetii_GCF_002924345_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 39.\n",
      "Screened 44 sequences from Coxiella_burnetii_GCF_002924425_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 88 sequences from Coxiella_burnetii_GCF_000723245_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 87.\n",
      "Screened 47 sequences from Coxiella_burnetii_GCF_002247285_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 47.\n",
      "Screened 46 sequences from Coxiella_burnetii_GCF_002247305_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 45.\n",
      "Screened 49 sequences from Coxiella_burnetii_GCF_002247335_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 49.\n",
      "Screened 47 sequences from Coxiella_burnetii_GCF_002249845_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 47.\n",
      "Screened 47 sequences from Coxiella_burnetii_GCF_002924325_merged.bed.  Filtered 0 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 47.\n",
      "Screened 91 sequences from Coxiella_burnetii_GCF_003849965_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 125 sequences from Coxiella_burnetii_GCF_002247225_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 119.\n",
      "Screened 112 sequences from Coxiella_burnetii_GCF_002896795_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 108.\n",
      "Screened 98 sequences from Coxiella_burnetii_GCF_002924385_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 153 sequences from Coxiella_burnetii_GCF_000751935_merged.bed.  Filtered 5 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 148.\n",
      "Screened 152 sequences from Coxiella_burnetii_GCF_000820825_merged.bed.  Filtered 5 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 147.\n",
      "Screened 140 sequences from Coxiella_burnetii_GCF_000756325_merged.bed.  Filtered 8 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 132.\n",
      "Screened 137 sequences from Coxiella_burnetii_GCF_002591355_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 133.\n",
      "Screened 214 sequences from Coxiella_burnetii_GCF_000169495_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 198.\n",
      "Screened 130 sequences from Coxiella_burnetii_GCF_000723285_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 124.\n",
      "Screened 131 sequences from Coxiella_burnetii_GCF_002896815_merged.bed.  Filtered 3 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 128.\n",
      "Screened 150 sequences from Coxiella_burnetii_GCF_003849885_merged.bed.  Filtered 7 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 143.\n",
      "Screened 155 sequences from Coxiella_burnetii_GCF_003849875_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 149.\n",
      "Screened 154 sequences from Coxiella_burnetii_GCF_003849785_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 150.\n",
      "Screened 159 sequences from Coxiella_burnetii_GCF_000723265_merged.bed.  Filtered 8 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 151.\n",
      "Screened 311 sequences from Coxiella_burnetii_GCF_000300315_merged.bed.  Filtered 11 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 300.\n",
      "Screened 65 sequences from Coxiella_burnetii_GCA_000359545_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 64.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/coxiella/initial_intervals/Coxiella_burnetii_GCA_000359545_merged.bed --twobit results/phyluce/coxiella/cleaned_genomes/Coxiella_burnetii_GCF_000007765_formatted.2bit --output results/phyluce/coxiella/initial_intervals/Coxiella_burnetii_GCA_000359545_stripped.bed;\n",
      "coxiella_burnetii_gcf_000017105.\n",
      "coxiella_burnetii_gcf_001572745.\n",
      "coxiella_burnetii_gcf_004006195.\n",
      "coxiella_burnetii_gcf_000019885.\n",
      "coxiella_burnetii_gcf_000168875.\n",
      "coxiella_burnetii_gcf_000018745.\n",
      "coxiella_burnetii_gcf_000019865.\n",
      "coxiella_burnetii_gcf_000367725.\n",
      "coxiella_burnetii_gcf_004087735.\n",
      "coxiella_burnetii_gcf_002109345.\n",
      "coxiella_burnetii_gcf_002094935.\n",
      "coxiella_burnetii_gcf_005280755.\n",
      "coxiella_burnetii_gcf_002634005.\n",
      "coxiella_burnetii_gcf_002633985.\n",
      "coxiella_burnetii_gcf_002633965.\n",
      "coxiella_burnetii_gcf_002633945.\n",
      "coxiella_burnetii_gcf_002633885.\n",
      "coxiella_burnetii_gcf_002634025.\n",
      "coxiella_burnetii_gcf_002634045.\n",
      "coxiella_burnetii_gcf_000767035.\n",
      "coxiella_burnetii_gcf_002634065.\n",
      "coxiella_burnetii_gcf_002633925.\n",
      "coxiella_burnetii_gcf_002633905.\n",
      "coxiella_burnetii_gcf_001572765.\n",
      "coxiella_burnetii_gcf_002634085.\n",
      "coxiella_burnetii_gcf_000612785.\n",
      "coxiella_burnetii_gcf_000826165.\n",
      "coxiella_burnetii_gcf_000820465.\n",
      "coxiella_burnetii_gcf_000967075.\n",
      "coxiella_burnetii_gcf_000470495.\n",
      "coxiella_burnetii_gcf_002247545.\n",
      "coxiella_burnetii_gcf_002247155.\n",
      "coxiella_burnetii_gcf_002896835.\n",
      "coxiella_burnetii_gcf_000613025.\n",
      "coxiella_burnetii_gcf_000723305.\n",
      "coxiella_burnetii_gcf_002924395.\n",
      "coxiella_burnetii_gcf_002247185.\n",
      "coxiella_burnetii_gcf_002896735.\n",
      "coxiella_burnetii_gcf_002896755.\n",
      "coxiella_burnetii_gcf_002896775.\n",
      "coxiella_burnetii_gcf_002247265.\n",
      "coxiella_burnetii_gcf_002247205.\n",
      "coxiella_burnetii_gcf_002924305.\n",
      "coxiella_burnetii_gcf_002924345.\n",
      "coxiella_burnetii_gcf_002924425.\n",
      "coxiella_burnetii_gcf_000723245.\n",
      "coxiella_burnetii_gcf_002247285.\n",
      "coxiella_burnetii_gcf_002247305.\n",
      "coxiella_burnetii_gcf_002247335.\n",
      "coxiella_burnetii_gcf_002249845.\n",
      "coxiella_burnetii_gcf_002924325.\n",
      "coxiella_burnetii_gcf_003849965.\n",
      "coxiella_burnetii_gcf_002247225.\n",
      "coxiella_burnetii_gcf_002896795.\n",
      "coxiella_burnetii_gcf_002924385.\n",
      "coxiella_burnetii_gcf_000751935.\n",
      "coxiella_burnetii_gcf_000820825.\n",
      "coxiella_burnetii_gcf_000756325.\n",
      "coxiella_burnetii_gcf_002591355.\n",
      "coxiella_burnetii_gcf_000169495.\n",
      "coxiella_burnetii_gcf_000723285.\n",
      "coxiella_burnetii_gcf_002896815.\n",
      "coxiella_burnetii_gcf_003849885.\n",
      "coxiella_burnetii_gcf_003849875.\n",
      "coxiella_burnetii_gcf_003849785.\n",
      "coxiella_burnetii_gcf_000723265.\n",
      "coxiella_burnetii_gcf_000300315.\n",
      "coxiella_burnetii_gca_000359545.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/coxiella/initial_intervals/Coxiella_burnetii_GCA_000359545_merged.bed --twobit results/phyluce/coxiella/cleaned_genomes/Coxiella_burnetii_GCF_000007765_formatted.2bit --output results/phyluce/coxiella/initial_intervals/Coxiella_burnetii_GCA_000359545_stripped.bed;\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 0 taxa:\t114.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 1 taxa:\t114.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 2 taxa:\t113.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 3 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 4 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 5 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 6 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 7 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 8 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 9 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 10 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 11 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 12 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 13 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 14 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 15 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 16 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 17 taxa:\t111.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 18 taxa:\t110.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 19 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 20 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 21 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 22 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 23 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 24 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 25 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 26 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 27 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 28 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 29 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 30 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 31 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 32 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 33 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 34 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 35 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 36 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 37 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 38 taxa:\t109.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 39 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 40 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 41 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 42 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 43 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 44 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 45 taxa:\t108.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 46 taxa:\t107.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 47 taxa:\t107.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 48 taxa:\t107.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 49 taxa:\t107.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 50 taxa:\t107.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 51 taxa:\t106.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 52 taxa:\t104.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 53 taxa:\t104.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 54 taxa:\t104.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 55 taxa:\t104.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 56 taxa:\t103.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 57 taxa:\t103.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 58 taxa:\t99.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 59 taxa:\t99.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 60 taxa:\t99.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 61 taxa:\t98.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 62 taxa:\t98.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 63 taxa:\t96.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 64 taxa:\t95.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 65 taxa:\t94.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 66 taxa:\t94.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 67 taxa:\t91.0\n",
      "Loci shared by Coxiella_burnetii_GCF_000007765 + 68 taxa:\t91.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/coxiella/initial_intervals/coxiella-to-Coxiella_burnetii_GCF_000007765.sqlite --base-taxon Coxiella_burnetii_GCF_000007765 --output results/phyluce/coxiella/initial_intervals/Coxiella_burnetii_GCF_000007765_+68.bed --specific-counts 68;\n",
      "Counter({'coxiella_burnetii_gcf_005280755': 91, 'coxiella_burnetii_gcf_000169495': 91, 'coxiella_burnetii_gcf_001572765': 91, 'coxiella_burnetii_gcf_002094935': 91, 'coxiella_burnetii_gcf_003849885': 91, 'coxiella_burnetii_gcf_000470495': 91, 'coxiella_burnetii_gcf_002247185': 91, 'coxiella_burnetii_gcf_002924395': 91, 'coxiella_burnetii_gcf_004087735': 91, 'coxiella_burnetii_gcf_002247155': 91, 'coxiella_burnetii_gcf_000723265': 91, 'coxiella_burnetii_gcf_001572745': 91, 'coxiella_burnetii_gcf_002247285': 91, 'coxiella_burnetii_gcf_002924385': 91, 'coxiella_burnetii_gcf_000300315': 91, 'coxiella_burnetii_gcf_002633925': 91, 'coxiella_burnetii_gcf_000767035': 91, 'coxiella_burnetii_gcf_002896835': 91, 'coxiella_burnetii_gca_000359545': 91, 'coxiella_burnetii_gcf_000756325': 91, 'coxiella_burnetii_gcf_002634085': 91, 'coxiella_burnetii_gcf_000019865': 91, 'coxiella_burnetii_gcf_002249845': 91, 'coxiella_burnetii_gcf_002247205': 91, 'coxiella_burnetii_gcf_002633885': 91, 'coxiella_burnetii_gcf_002247265': 91, 'coxiella_burnetii_gcf_002896775': 91, 'coxiella_burnetii_gcf_000168875': 91, 'coxiella_burnetii_gcf_002247225': 91, 'coxiella_burnetii_gcf_002247305': 91, 'coxiella_burnetii_gcf_002634045': 91, 'coxiella_burnetii_gcf_000613025': 91, 'coxiella_burnetii_gcf_002896755': 91, 'coxiella_burnetii_gcf_000826165': 91, 'coxiella_burnetii_gcf_000723305': 91, 'coxiella_burnetii_gcf_000751935': 91, 'coxiella_burnetii_gcf_002634065': 91, 'coxiella_burnetii_gcf_000967075': 91, 'coxiella_burnetii_gcf_002633945': 91, 'coxiella_burnetii_gcf_002896735': 91, 'coxiella_burnetii_gcf_003849875': 91, 'coxiella_burnetii_gcf_004006195': 91, 'coxiella_burnetii_gcf_002924425': 91, 'coxiella_burnetii_gcf_000018745': 91, 'coxiella_burnetii_gcf_002633985': 91, 'coxiella_burnetii_gcf_002896815': 91, 'coxiella_burnetii_gcf_002634025': 91, 'coxiella_burnetii_gcf_002247545': 91, 'coxiella_burnetii_gcf_002896795': 91, 'coxiella_burnetii_gcf_000019885': 91, 'coxiella_burnetii_gcf_000820465': 91, 'coxiella_burnetii_gcf_002109345': 91, 'coxiella_burnetii_gcf_003849965': 91, 'coxiella_burnetii_gcf_000723285': 91, 'coxiella_burnetii_gcf_002924325': 91, 'coxiella_burnetii_gcf_002633905': 91, 'coxiella_burnetii_gcf_002924305': 91, 'coxiella_burnetii_gcf_000723245': 91, 'coxiella_burnetii_gcf_002633965': 91, 'coxiella_burnetii_gcf_002924345': 91, 'coxiella_burnetii_gcf_002247335': 91, 'coxiella_burnetii_gcf_002634005': 91, 'coxiella_burnetii_gcf_000612785': 91, 'coxiella_burnetii_gcf_000367725': 91, 'coxiella_burnetii_gcf_002591355': 91, 'coxiella_burnetii_gcf_000820825': 91, 'coxiella_burnetii_gcf_000017105': 91, 'coxiella_burnetii_gcf_003849785': 91})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 68\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/coxiella/initial_intervals/Coxiella_burnetii_GCF_000007765_+68.bed --twobit results/phyluce/coxiella/cleaned_genomes/Coxiella_burnetii_GCF_000007765_formatted.2bit --buffer-to 160 --output results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68.fasta;\n",
      "Screened 91 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 91.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68.fasta --probe-prefix uce_coxiella_ --design coxiella_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 89\n",
      "Probe Count = 175\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas --query results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  09:44:05\n",
      "Ended:  Fri Feb 07, 2020  09:44:06\n",
      "Time for execution:  0.0123950004578 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas                        --lastz results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes_vself.lastz                       --probe-prefix=uce_coxiella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 174 fasta sequences.  Filtered 0 duplicates. Kept 175.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/coxiella/validate_intervals/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas --scaffoldlist Coxiella_burnetii_GCF_000017105 Coxiella_burnetii_GCF_001572745 Coxiella_burnetii_GCF_004006195 Coxiella_burnetii_GCF_000019885 Coxiella_burnetii_GCF_000168875 Coxiella_burnetii_GCF_000018745 Coxiella_burnetii_GCF_000019865 Coxiella_burnetii_GCF_000367725 Coxiella_burnetii_GCF_004087735 Coxiella_burnetii_GCF_002109345 Coxiella_burnetii_GCF_002094935 Coxiella_burnetii_GCF_005280755 Coxiella_burnetii_GCF_002634005 Coxiella_burnetii_GCF_002633985 Coxiella_burnetii_GCF_002633965 Coxiella_burnetii_GCF_002633945 Coxiella_burnetii_GCF_002633885 Coxiella_burnetii_GCF_002634025 Coxiella_burnetii_GCF_002634045 Coxiella_burnetii_GCF_000767035 Coxiella_burnetii_GCF_002634065 Coxiella_burnetii_GCF_002633925 Coxiella_burnetii_GCF_002633905 Coxiella_burnetii_GCF_001572765 Coxiella_burnetii_GCF_002634085 Coxiella_burnetii_GCF_000612785 Coxiella_burnetii_GCF_000826165 Coxiella_burnetii_GCF_000820465 Coxiella_burnetii_GCF_000967075 Coxiella_burnetii_GCF_000470495 Coxiella_burnetii_GCF_002247545 Coxiella_burnetii_GCF_002247155 Coxiella_burnetii_GCF_002896835 Coxiella_burnetii_GCF_000613025 Coxiella_burnetii_GCF_000723305 Coxiella_burnetii_GCF_002924395 Coxiella_burnetii_GCF_002247185 Coxiella_burnetii_GCF_002896735 Coxiella_burnetii_GCF_002896755 Coxiella_burnetii_GCF_002896775 Coxiella_burnetii_GCF_002247265 Coxiella_burnetii_GCF_002247205 Coxiella_burnetii_GCF_002924305 Coxiella_burnetii_GCF_002924345 Coxiella_burnetii_GCF_002924425 Coxiella_burnetii_GCF_000723245 Coxiella_burnetii_GCF_002247285 Coxiella_burnetii_GCF_002247305 Coxiella_burnetii_GCF_002247335 Coxiella_burnetii_GCF_002249845 Coxiella_burnetii_GCF_002924325 Coxiella_burnetii_GCF_003849965 Coxiella_burnetii_GCF_002247225 Coxiella_burnetii_GCF_002896795 Coxiella_burnetii_GCF_002924385 Coxiella_burnetii_GCF_000751935 Coxiella_burnetii_GCF_000820825 Coxiella_burnetii_GCF_000756325 Coxiella_burnetii_GCF_002591355 Coxiella_burnetii_GCF_000169495 Coxiella_burnetii_GCF_000723285 Coxiella_burnetii_GCF_002896815 Coxiella_burnetii_GCF_003849885 Coxiella_burnetii_GCF_003849875 Coxiella_burnetii_GCF_003849785 Coxiella_burnetii_GCF_000723265 Coxiella_burnetii_GCF_000300315 Coxiella_burnetii_GCA_000359545 Coxiella_burnetii_GCF_000007765 --genome-base-path results/phyluce/coxiella/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/coxiella/validate_intervals/coxiella-to-Coxiella_burnetii_GCF_000007765.sqlite --output results/phyluce/coxiella/validate_intervals/lastz/;\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000017105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmO18Oj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2jTAou.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000017105.lastz\n",
      "Creating Coxiella_burnetii_GCF_000017105 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000017105 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_001572745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCjo6yS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9nubon.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_001572745.lastz\n",
      "Creating Coxiella_burnetii_GCF_001572745 table\n",
      "Inserting data to Coxiella_burnetii_GCF_001572745 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_004006195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpUfsLAn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRDCyPD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_004006195.lastz\n",
      "Creating Coxiella_burnetii_GCF_004006195 table\n",
      "Inserting data to Coxiella_burnetii_GCF_004006195 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000019885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvUgpBa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbKNrfc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000019885.lastz\n",
      "Creating Coxiella_burnetii_GCF_000019885 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000019885 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000168875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxJQVJS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSxoj1m.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000168875.lastz\n",
      "Creating Coxiella_burnetii_GCF_000168875 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000168875 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000018745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfx_Jfu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_qWmRn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000018745.lastz\n",
      "Creating Coxiella_burnetii_GCF_000018745 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000018745 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000019865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprwpJFv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBylozP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000019865.lastz\n",
      "Creating Coxiella_burnetii_GCF_000019865 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000019865 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000367725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiNAVRP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRNu3sY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000367725.lastz\n",
      "Creating Coxiella_burnetii_GCF_000367725 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000367725 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_004087735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGSV2ay.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAVCRT8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_004087735.lastz\n",
      "Creating Coxiella_burnetii_GCF_004087735 table\n",
      "Inserting data to Coxiella_burnetii_GCF_004087735 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002109345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBhpASv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSAj8W9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002109345.lastz\n",
      "Creating Coxiella_burnetii_GCF_002109345 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002109345 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002094935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpvgAQid.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgUZm07.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002094935.lastz\n",
      "Creating Coxiella_burnetii_GCF_002094935 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002094935 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_005280755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvHw9bH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuqRDF4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_005280755.lastz\n",
      "Creating Coxiella_burnetii_GCF_005280755 table\n",
      "Inserting data to Coxiella_burnetii_GCF_005280755 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002634005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpW4HNUJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTJmPOf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002634005.lastz\n",
      "Creating Coxiella_burnetii_GCF_002634005 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002634005 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002633985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTq4yWJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxH9a3o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002633985.lastz\n",
      "Creating Coxiella_burnetii_GCF_002633985 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002633985 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002633965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQum2NT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuTM_6X.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002633965.lastz\n",
      "Creating Coxiella_burnetii_GCF_002633965 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002633965 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002633945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptXAqi_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGuqYFT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002633945.lastz\n",
      "Creating Coxiella_burnetii_GCF_002633945 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002633945 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002633885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHNW693.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBLgA6m.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002633885.lastz\n",
      "Creating Coxiella_burnetii_GCF_002633885 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002633885 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002634025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbFX_6p.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRKL8d8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002634025.lastz\n",
      "Creating Coxiella_burnetii_GCF_002634025 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002634025 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002634045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnym7da.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJ0Y26U.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002634045.lastz\n",
      "Creating Coxiella_burnetii_GCF_002634045 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002634045 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000767035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0_lVSM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwSt7gI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000767035.lastz\n",
      "Creating Coxiella_burnetii_GCF_000767035 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000767035 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002634065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZPbeyH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoDkxCG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002634065.lastz\n",
      "Creating Coxiella_burnetii_GCF_002634065 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002634065 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002633925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGshZTQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmph8MJ_p.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002633925.lastz\n",
      "Creating Coxiella_burnetii_GCF_002633925 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002633925 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002633905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRS3vvT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXqIuhf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002633905.lastz\n",
      "Creating Coxiella_burnetii_GCF_002633905 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002633905 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_001572765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCIU3CJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwTf4wt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_001572765.lastz\n",
      "Creating Coxiella_burnetii_GCF_001572765 table\n",
      "Inserting data to Coxiella_burnetii_GCF_001572765 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002634085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMDFeE7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmph8YB1M.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002634085.lastz\n",
      "Creating Coxiella_burnetii_GCF_002634085 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002634085 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000612785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWAaJv6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCTHE6q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000612785.lastz\n",
      "Creating Coxiella_burnetii_GCF_000612785 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000612785 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000826165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQE1G5J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8kxSfw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000826165.lastz\n",
      "Creating Coxiella_burnetii_GCF_000826165 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000826165 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000820465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4kz1tP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQOGbXx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000820465.lastz\n",
      "Creating Coxiella_burnetii_GCF_000820465 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000820465 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000967075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaWBWng.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWtcPGs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000967075.lastz\n",
      "Creating Coxiella_burnetii_GCF_000967075 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000967075 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000470495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplr8aQj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpc036MK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000470495.lastz\n",
      "Creating Coxiella_burnetii_GCF_000470495 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000470495 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpltchkR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6LZNSe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247545.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247545 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247545 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpccnmu5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQc9MY4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247155.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247155 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247155 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002896835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxUO8c1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDi0Cok.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002896835.lastz\n",
      "Creating Coxiella_burnetii_GCF_002896835 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002896835 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000613025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpSYoEC4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHpxxKD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000613025.lastz\n",
      "Creating Coxiella_burnetii_GCF_000613025 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000613025 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000723305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDfi_SE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRh_ql5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000723305.lastz\n",
      "Creating Coxiella_burnetii_GCF_000723305 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000723305 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002924395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzyia26.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiGRkla.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002924395.lastz\n",
      "Creating Coxiella_burnetii_GCF_002924395 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002924395 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJqnTH4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphDFBYA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247185.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247185 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247185 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002896735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOBjjip.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJJY3KQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002896735.lastz\n",
      "Creating Coxiella_burnetii_GCF_002896735 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002896735 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002896755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnYh5zp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4sWruG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002896755.lastz\n",
      "Creating Coxiella_burnetii_GCF_002896755 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002896755 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002896775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLaNFwU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpllqcTA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002896775.lastz\n",
      "Creating Coxiella_burnetii_GCF_002896775 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002896775 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxk3WK5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOi0BgA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247265.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247265 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247265 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpX_dLCw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPPXQEV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247205.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247205 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247205 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002924305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJyPF_H.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCJeEhf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002924305.lastz\n",
      "Creating Coxiella_burnetii_GCF_002924305 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002924305 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002924345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9lzmt7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYcWTnB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002924345.lastz\n",
      "Creating Coxiella_burnetii_GCF_002924345 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002924345 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002924425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsuI0UR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6gNC0W.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002924425.lastz\n",
      "Creating Coxiella_burnetii_GCF_002924425 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002924425 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000723245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp22sL95.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpo6Px27.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000723245.lastz\n",
      "Creating Coxiella_burnetii_GCF_000723245 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000723245 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0SdyAw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprW3v03.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247285.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247285 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247285 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdE2NXK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfLmY2K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247305.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247305 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247305 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTHZDxP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9RM1_1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247335.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247335 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247335 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002249845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpY_TjCM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPxQmnN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002249845.lastz\n",
      "Creating Coxiella_burnetii_GCF_002249845 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002249845 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002924325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmppL_lWe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEJsI0o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002924325.lastz\n",
      "Creating Coxiella_burnetii_GCF_002924325 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002924325 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_003849965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplIOhmW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpo2J6K0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_003849965.lastz\n",
      "Creating Coxiella_burnetii_GCF_003849965 table\n",
      "Inserting data to Coxiella_burnetii_GCF_003849965 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002247225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8c2Dk5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpShjClA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002247225.lastz\n",
      "Creating Coxiella_burnetii_GCF_002247225 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002247225 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002896795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp513KFW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp221XZU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002896795.lastz\n",
      "Creating Coxiella_burnetii_GCF_002896795 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002896795 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002924385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpN01JEw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2DKW7o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002924385.lastz\n",
      "Creating Coxiella_burnetii_GCF_002924385 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002924385 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000751935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpILbpUy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJhQNPv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000751935.lastz\n",
      "Creating Coxiella_burnetii_GCF_000751935 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000751935 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000820825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPlHg7d.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbd_v7e.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000820825.lastz\n",
      "Creating Coxiella_burnetii_GCF_000820825 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000820825 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000756325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuoFvU9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWWJ_YB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000756325.lastz\n",
      "Creating Coxiella_burnetii_GCF_000756325 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000756325 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002591355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpX0wCfI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7iWgV0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002591355.lastz\n",
      "Creating Coxiella_burnetii_GCF_002591355 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002591355 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000169495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp647QuI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_McHjM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000169495.lastz\n",
      "Creating Coxiella_burnetii_GCF_000169495 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000169495 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000723285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpns8Jpg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMUoPBq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000723285.lastz\n",
      "Creating Coxiella_burnetii_GCF_000723285 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000723285 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_002896815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwwjdUs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6txCav.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_002896815.lastz\n",
      "Creating Coxiella_burnetii_GCF_002896815 table\n",
      "Inserting data to Coxiella_burnetii_GCF_002896815 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_003849885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpClK3JU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBA5ZqI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_003849885.lastz\n",
      "Creating Coxiella_burnetii_GCF_003849885 table\n",
      "Inserting data to Coxiella_burnetii_GCF_003849885 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_003849875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpapkTZC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEWfN8F.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_003849875.lastz\n",
      "Creating Coxiella_burnetii_GCF_003849875 table\n",
      "Inserting data to Coxiella_burnetii_GCF_003849875 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_003849785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpricF11.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2vkK4o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_003849785.lastz\n",
      "Creating Coxiella_burnetii_GCF_003849785 table\n",
      "Inserting data to Coxiella_burnetii_GCF_003849785 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000723265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCPj8S4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEJRn4Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000723265.lastz\n",
      "Creating Coxiella_burnetii_GCF_000723265 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000723265 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000300315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7YczgE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvxrbWM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000300315.lastz\n",
      "Creating Coxiella_burnetii_GCF_000300315 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000300315 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCA_000359545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpeDZL0U.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptlwiJO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCA_000359545.lastz\n",
      "Creating Coxiella_burnetii_GCA_000359545 table\n",
      "Inserting data to Coxiella_burnetii_GCA_000359545 table\n",
      "\n",
      "Running against Coxiella_burnetii_GCF_000007765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJ4heKh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0vHTJt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/coxiella/validate_intervals/lastz/Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_Coxiella_burnetii_GCF_000007765.lastz\n",
      "Creating Coxiella_burnetii_GCF_000007765 table\n",
      "Inserting data to Coxiella_burnetii_GCF_000007765 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/coxiella/extract_probes_from_group/coxiella_genome.conf --lastz results/phyluce/coxiella/validate_intervals/lastz --probes 120 --probe-prefix uce_coxiella_ --name-pattern \"Coxiella_burnetii_GCF_000007765_+68_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/coxiella/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 09:45:58,466 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 09:45:58,473 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000017105 genome -------\n",
      "2020-02-07 09:45:58,474 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000017105 genome\n",
      "2020-02-07 09:45:59,004 - Phyluce - INFO - Coxiella_burnetii_GCF_000017105: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:45:59,005 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_001572745 genome -------\n",
      "2020-02-07 09:45:59,006 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_001572745 genome\n",
      "2020-02-07 09:45:59,303 - Phyluce - INFO - Coxiella_burnetii_GCF_001572745: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:45:59,303 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_004006195 genome -------\n",
      "2020-02-07 09:45:59,304 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_004006195 genome\n",
      "2020-02-07 09:45:59,592 - Phyluce - INFO - Coxiella_burnetii_GCF_004006195: 87 uces, 0 dupes, 87 non-dupes, 1 orient drop, 1 length drop, 85 written\n",
      "2020-02-07 09:45:59,592 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000019885 genome -------\n",
      "2020-02-07 09:45:59,593 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000019885 genome\n",
      "2020-02-07 09:45:59,933 - Phyluce - INFO - Coxiella_burnetii_GCF_000019885: 88 uces, 0 dupes, 88 non-dupes, 1 orient drop, 1 length drop, 86 written\n",
      "2020-02-07 09:45:59,933 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000168875 genome -------\n",
      "2020-02-07 09:45:59,934 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000168875 genome\n",
      "2020-02-07 09:46:00,266 - Phyluce - INFO - Coxiella_burnetii_GCF_000168875: 88 uces, 0 dupes, 88 non-dupes, 1 orient drop, 1 length drop, 86 written\n",
      "2020-02-07 09:46:00,266 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000018745 genome -------\n",
      "2020-02-07 09:46:00,267 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000018745 genome\n",
      "2020-02-07 09:46:00,593 - Phyluce - INFO - Coxiella_burnetii_GCF_000018745: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:00,594 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000019865 genome -------\n",
      "2020-02-07 09:46:00,594 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000019865 genome\n",
      "2020-02-07 09:46:00,927 - Phyluce - INFO - Coxiella_burnetii_GCF_000019865: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:00,927 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000367725 genome -------\n",
      "2020-02-07 09:46:00,928 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000367725 genome\n",
      "2020-02-07 09:46:01,268 - Phyluce - INFO - Coxiella_burnetii_GCF_000367725: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:01,269 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_004087735 genome -------\n",
      "2020-02-07 09:46:01,269 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_004087735 genome\n",
      "2020-02-07 09:46:01,617 - Phyluce - INFO - Coxiella_burnetii_GCF_004087735: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:01,617 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002109345 genome -------\n",
      "2020-02-07 09:46:01,618 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002109345 genome\n",
      "2020-02-07 09:46:01,960 - Phyluce - INFO - Coxiella_burnetii_GCF_002109345: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:01,960 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002094935 genome -------\n",
      "2020-02-07 09:46:01,961 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002094935 genome\n",
      "2020-02-07 09:46:02,306 - Phyluce - INFO - Coxiella_burnetii_GCF_002094935: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:02,306 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_005280755 genome -------\n",
      "2020-02-07 09:46:02,307 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_005280755 genome\n",
      "2020-02-07 09:46:02,650 - Phyluce - INFO - Coxiella_burnetii_GCF_005280755: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:02,650 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002634005 genome -------\n",
      "2020-02-07 09:46:02,651 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002634005 genome\n",
      "2020-02-07 09:46:02,992 - Phyluce - INFO - Coxiella_burnetii_GCF_002634005: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:02,992 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002633985 genome -------\n",
      "2020-02-07 09:46:02,996 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002633985 genome\n",
      "2020-02-07 09:46:03,395 - Phyluce - INFO - Coxiella_burnetii_GCF_002633985: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:03,396 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002633965 genome -------\n",
      "2020-02-07 09:46:03,396 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002633965 genome\n",
      "2020-02-07 09:46:03,758 - Phyluce - INFO - Coxiella_burnetii_GCF_002633965: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:03,759 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002633945 genome -------\n",
      "2020-02-07 09:46:03,759 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002633945 genome\n",
      "2020-02-07 09:46:04,100 - Phyluce - INFO - Coxiella_burnetii_GCF_002633945: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:04,100 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002633885 genome -------\n",
      "2020-02-07 09:46:04,101 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002633885 genome\n",
      "2020-02-07 09:46:04,442 - Phyluce - INFO - Coxiella_burnetii_GCF_002633885: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:04,442 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002634025 genome -------\n",
      "2020-02-07 09:46:04,443 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002634025 genome\n",
      "2020-02-07 09:46:04,755 - Phyluce - INFO - Coxiella_burnetii_GCF_002634025: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:04,756 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002634045 genome -------\n",
      "2020-02-07 09:46:04,756 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002634045 genome\n",
      "2020-02-07 09:46:05,068 - Phyluce - INFO - Coxiella_burnetii_GCF_002634045: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:05,068 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000767035 genome -------\n",
      "2020-02-07 09:46:05,069 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000767035 genome\n",
      "2020-02-07 09:46:05,392 - Phyluce - INFO - Coxiella_burnetii_GCF_000767035: 89 uces, 0 dupes, 89 non-dupes, 1 orient drop, 1 length drop, 87 written\n",
      "2020-02-07 09:46:05,392 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002634065 genome -------\n",
      "2020-02-07 09:46:05,393 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002634065 genome\n",
      "2020-02-07 09:46:05,737 - Phyluce - INFO - Coxiella_burnetii_GCF_002634065: 88 uces, 0 dupes, 88 non-dupes, 1 orient drop, 1 length drop, 86 written\n",
      "2020-02-07 09:46:05,737 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002633925 genome -------\n",
      "2020-02-07 09:46:05,743 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002633925 genome\n",
      "2020-02-07 09:46:06,054 - Phyluce - INFO - Coxiella_burnetii_GCF_002633925: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:06,054 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002633905 genome -------\n",
      "2020-02-07 09:46:06,055 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002633905 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 09:46:06,363 - Phyluce - INFO - Coxiella_burnetii_GCF_002633905: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:06,363 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_001572765 genome -------\n",
      "2020-02-07 09:46:06,364 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_001572765 genome\n",
      "2020-02-07 09:46:06,664 - Phyluce - INFO - Coxiella_burnetii_GCF_001572765: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:06,664 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002634085 genome -------\n",
      "2020-02-07 09:46:06,665 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002634085 genome\n",
      "2020-02-07 09:46:06,962 - Phyluce - INFO - Coxiella_burnetii_GCF_002634085: 87 uces, 0 dupes, 87 non-dupes, 0 orient drop, 1 length drop, 86 written\n",
      "2020-02-07 09:46:06,963 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000612785 genome -------\n",
      "2020-02-07 09:46:06,963 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000612785 genome\n",
      "2020-02-07 09:46:07,312 - Phyluce - INFO - Coxiella_burnetii_GCF_000612785: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:07,312 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000826165 genome -------\n",
      "2020-02-07 09:46:07,313 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000826165 genome\n",
      "2020-02-07 09:46:07,654 - Phyluce - INFO - Coxiella_burnetii_GCF_000826165: 89 uces, 0 dupes, 89 non-dupes, 1 orient drop, 1 length drop, 87 written\n",
      "2020-02-07 09:46:07,654 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000820465 genome -------\n",
      "2020-02-07 09:46:07,667 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000820465 genome\n",
      "2020-02-07 09:46:08,019 - Phyluce - INFO - Coxiella_burnetii_GCF_000820465: 88 uces, 1 dupes, 87 non-dupes, 0 orient drop, 1 length drop, 86 written\n",
      "2020-02-07 09:46:08,019 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000967075 genome -------\n",
      "2020-02-07 09:46:08,020 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000967075 genome\n",
      "2020-02-07 09:46:08,334 - Phyluce - INFO - Coxiella_burnetii_GCF_000967075: 87 uces, 4 dupes, 83 non-dupes, 1 orient drop, 3 length drop, 79 written\n",
      "2020-02-07 09:46:08,334 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000470495 genome -------\n",
      "2020-02-07 09:46:08,340 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000470495 genome\n",
      "2020-02-07 09:46:08,685 - Phyluce - INFO - Coxiella_burnetii_GCF_000470495: 88 uces, 0 dupes, 88 non-dupes, 2 orient drop, 1 length drop, 85 written\n",
      "2020-02-07 09:46:08,686 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247545 genome -------\n",
      "2020-02-07 09:46:08,686 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247545 genome\n",
      "2020-02-07 09:46:09,053 - Phyluce - INFO - Coxiella_burnetii_GCF_002247545: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:09,053 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247155 genome -------\n",
      "2020-02-07 09:46:09,054 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247155 genome\n",
      "2020-02-07 09:46:09,434 - Phyluce - INFO - Coxiella_burnetii_GCF_002247155: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:09,434 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002896835 genome -------\n",
      "2020-02-07 09:46:09,435 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002896835 genome\n",
      "2020-02-07 09:46:09,778 - Phyluce - INFO - Coxiella_burnetii_GCF_002896835: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:09,778 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000613025 genome -------\n",
      "2020-02-07 09:46:09,779 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000613025 genome\n",
      "2020-02-07 09:46:10,127 - Phyluce - INFO - Coxiella_burnetii_GCF_000613025: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:10,127 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000723305 genome -------\n",
      "2020-02-07 09:46:10,128 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000723305 genome\n",
      "2020-02-07 09:46:10,470 - Phyluce - INFO - Coxiella_burnetii_GCF_000723305: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:10,470 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002924395 genome -------\n",
      "2020-02-07 09:46:10,471 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002924395 genome\n",
      "2020-02-07 09:46:10,815 - Phyluce - INFO - Coxiella_burnetii_GCF_002924395: 88 uces, 2 dupes, 86 non-dupes, 0 orient drop, 0 length drop, 86 written\n",
      "2020-02-07 09:46:10,815 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247185 genome -------\n",
      "2020-02-07 09:46:10,816 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247185 genome\n",
      "2020-02-07 09:46:11,158 - Phyluce - INFO - Coxiella_burnetii_GCF_002247185: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:11,158 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002896735 genome -------\n",
      "2020-02-07 09:46:11,158 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002896735 genome\n",
      "2020-02-07 09:46:11,502 - Phyluce - INFO - Coxiella_burnetii_GCF_002896735: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:11,502 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002896755 genome -------\n",
      "2020-02-07 09:46:11,503 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002896755 genome\n",
      "2020-02-07 09:46:11,846 - Phyluce - INFO - Coxiella_burnetii_GCF_002896755: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:11,846 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002896775 genome -------\n",
      "2020-02-07 09:46:11,847 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002896775 genome\n",
      "2020-02-07 09:46:12,206 - Phyluce - INFO - Coxiella_burnetii_GCF_002896775: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:12,206 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247265 genome -------\n",
      "2020-02-07 09:46:12,213 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247265 genome\n",
      "2020-02-07 09:46:12,567 - Phyluce - INFO - Coxiella_burnetii_GCF_002247265: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:12,567 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247205 genome -------\n",
      "2020-02-07 09:46:12,568 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247205 genome\n",
      "2020-02-07 09:46:12,927 - Phyluce - INFO - Coxiella_burnetii_GCF_002247205: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:12,928 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002924305 genome -------\n",
      "2020-02-07 09:46:12,928 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002924305 genome\n",
      "2020-02-07 09:46:13,274 - Phyluce - INFO - Coxiella_burnetii_GCF_002924305: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:13,274 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002924345 genome -------\n",
      "2020-02-07 09:46:13,275 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002924345 genome\n",
      "2020-02-07 09:46:13,621 - Phyluce - INFO - Coxiella_burnetii_GCF_002924345: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:13,621 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002924425 genome -------\n",
      "2020-02-07 09:46:13,627 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002924425 genome\n",
      "2020-02-07 09:46:13,980 - Phyluce - INFO - Coxiella_burnetii_GCF_002924425: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:13,980 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000723245 genome -------\n",
      "2020-02-07 09:46:13,981 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000723245 genome\n",
      "2020-02-07 09:46:14,327 - Phyluce - INFO - Coxiella_burnetii_GCF_000723245: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:14,327 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247285 genome -------\n",
      "2020-02-07 09:46:14,328 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247285 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 09:46:14,683 - Phyluce - INFO - Coxiella_burnetii_GCF_002247285: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:14,683 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247305 genome -------\n",
      "2020-02-07 09:46:14,684 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247305 genome\n",
      "2020-02-07 09:46:15,032 - Phyluce - INFO - Coxiella_burnetii_GCF_002247305: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:15,032 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247335 genome -------\n",
      "2020-02-07 09:46:15,033 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247335 genome\n",
      "2020-02-07 09:46:15,378 - Phyluce - INFO - Coxiella_burnetii_GCF_002247335: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:15,378 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002249845 genome -------\n",
      "2020-02-07 09:46:15,379 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002249845 genome\n",
      "2020-02-07 09:46:15,725 - Phyluce - INFO - Coxiella_burnetii_GCF_002249845: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:15,726 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002924325 genome -------\n",
      "2020-02-07 09:46:15,726 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002924325 genome\n",
      "2020-02-07 09:46:16,074 - Phyluce - INFO - Coxiella_burnetii_GCF_002924325: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:16,074 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_003849965 genome -------\n",
      "2020-02-07 09:46:16,075 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_003849965 genome\n",
      "2020-02-07 09:46:16,426 - Phyluce - INFO - Coxiella_burnetii_GCF_003849965: 89 uces, 2 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:16,426 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002247225 genome -------\n",
      "2020-02-07 09:46:16,426 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002247225 genome\n",
      "2020-02-07 09:46:16,761 - Phyluce - INFO - Coxiella_burnetii_GCF_002247225: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:16,761 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002896795 genome -------\n",
      "2020-02-07 09:46:16,762 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002896795 genome\n",
      "2020-02-07 09:46:17,097 - Phyluce - INFO - Coxiella_burnetii_GCF_002896795: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:17,098 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002924385 genome -------\n",
      "2020-02-07 09:46:17,098 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002924385 genome\n",
      "2020-02-07 09:46:17,419 - Phyluce - INFO - Coxiella_burnetii_GCF_002924385: 88 uces, 2 dupes, 86 non-dupes, 0 orient drop, 0 length drop, 86 written\n",
      "2020-02-07 09:46:17,420 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000751935 genome -------\n",
      "2020-02-07 09:46:17,421 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000751935 genome\n",
      "2020-02-07 09:46:17,761 - Phyluce - INFO - Coxiella_burnetii_GCF_000751935: 87 uces, 0 dupes, 87 non-dupes, 2 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:17,761 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000820825 genome -------\n",
      "2020-02-07 09:46:17,766 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000820825 genome\n",
      "2020-02-07 09:46:18,098 - Phyluce - INFO - Coxiella_burnetii_GCF_000820825: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:18,098 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000756325 genome -------\n",
      "2020-02-07 09:46:18,099 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000756325 genome\n",
      "2020-02-07 09:46:18,419 - Phyluce - INFO - Coxiella_burnetii_GCF_000756325: 88 uces, 2 dupes, 86 non-dupes, 0 orient drop, 0 length drop, 86 written\n",
      "2020-02-07 09:46:18,419 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002591355 genome -------\n",
      "2020-02-07 09:46:18,420 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002591355 genome\n",
      "2020-02-07 09:46:18,750 - Phyluce - INFO - Coxiella_burnetii_GCF_002591355: 88 uces, 2 dupes, 86 non-dupes, 0 orient drop, 0 length drop, 86 written\n",
      "2020-02-07 09:46:18,750 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000169495 genome -------\n",
      "2020-02-07 09:46:18,756 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000169495 genome\n",
      "2020-02-07 09:46:19,058 - Phyluce - INFO - Coxiella_burnetii_GCF_000169495: 84 uces, 3 dupes, 81 non-dupes, 0 orient drop, 0 length drop, 81 written\n",
      "2020-02-07 09:46:19,058 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000723285 genome -------\n",
      "2020-02-07 09:46:19,058 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000723285 genome\n",
      "2020-02-07 09:46:19,347 - Phyluce - INFO - Coxiella_burnetii_GCF_000723285: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:19,347 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_002896815 genome -------\n",
      "2020-02-07 09:46:19,348 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_002896815 genome\n",
      "2020-02-07 09:46:19,638 - Phyluce - INFO - Coxiella_burnetii_GCF_002896815: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:19,638 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_003849885 genome -------\n",
      "2020-02-07 09:46:19,646 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_003849885 genome\n",
      "2020-02-07 09:46:19,945 - Phyluce - INFO - Coxiella_burnetii_GCF_003849885: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:19,945 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_003849875 genome -------\n",
      "2020-02-07 09:46:19,951 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_003849875 genome\n",
      "2020-02-07 09:46:20,264 - Phyluce - INFO - Coxiella_burnetii_GCF_003849875: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:20,264 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_003849785 genome -------\n",
      "2020-02-07 09:46:20,269 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_003849785 genome\n",
      "2020-02-07 09:46:20,563 - Phyluce - INFO - Coxiella_burnetii_GCF_003849785: 87 uces, 2 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:20,563 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000723265 genome -------\n",
      "2020-02-07 09:46:20,564 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000723265 genome\n",
      "2020-02-07 09:46:20,908 - Phyluce - INFO - Coxiella_burnetii_GCF_000723265: 89 uces, 4 dupes, 85 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:20,908 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000300315 genome -------\n",
      "2020-02-07 09:46:20,909 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000300315 genome\n",
      "2020-02-07 09:46:21,190 - Phyluce - INFO - Coxiella_burnetii_GCF_000300315: 87 uces, 1 dupes, 86 non-dupes, 0 orient drop, 0 length drop, 85 written\n",
      "2020-02-07 09:46:21,190 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCA_000359545 genome -------\n",
      "2020-02-07 09:46:21,199 - Phyluce - INFO - Reading Coxiella_burnetii_GCA_000359545 genome\n",
      "2020-02-07 09:46:21,546 - Phyluce - INFO - Coxiella_burnetii_GCA_000359545: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n",
      "2020-02-07 09:46:21,547 - Phyluce - INFO - ------- Working on Coxiella_burnetii_GCF_000007765 genome -------\n",
      "2020-02-07 09:46:21,548 - Phyluce - INFO - Reading Coxiella_burnetii_GCF_000007765 genome\n",
      "2020-02-07 09:46:21,890 - Phyluce - INFO - Coxiella_burnetii_GCF_000007765: 89 uces, 0 dupes, 89 non-dupes, 2 orient drop, 0 length drop, 87 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/coxiella/extract_probes_from_group/probe_fasta --output results/phyluce/coxiella/extract_probes_from_group/multifastas.sqlite --base-taxon Coxiella_burnetii_GCF_000007765;\n",
      "coxiella_burnetii_gcf_000017105.\n",
      "coxiella_burnetii_gcf_001572745.\n",
      "coxiella_burnetii_gcf_004006195.\n",
      "coxiella_burnetii_gcf_000019885.\n",
      "coxiella_burnetii_gcf_000168875.\n",
      "coxiella_burnetii_gcf_000018745.\n",
      "coxiella_burnetii_gcf_000019865.\n",
      "coxiella_burnetii_gcf_000367725.\n",
      "coxiella_burnetii_gcf_004087735.\n",
      "coxiella_burnetii_gcf_002109345.\n",
      "coxiella_burnetii_gcf_002094935.\n",
      "coxiella_burnetii_gcf_005280755.\n",
      "coxiella_burnetii_gcf_002634005.\n",
      "coxiella_burnetii_gcf_002633985.\n",
      "coxiella_burnetii_gcf_002633965.\n",
      "coxiella_burnetii_gcf_002633945.\n",
      "coxiella_burnetii_gcf_002633885.\n",
      "coxiella_burnetii_gcf_002634025.\n",
      "coxiella_burnetii_gcf_002634045.\n",
      "coxiella_burnetii_gcf_000767035.\n",
      "coxiella_burnetii_gcf_002634065.\n",
      "coxiella_burnetii_gcf_002633925.\n",
      "coxiella_burnetii_gcf_002633905.\n",
      "coxiella_burnetii_gcf_001572765.\n",
      "coxiella_burnetii_gcf_002634085.\n",
      "coxiella_burnetii_gcf_000612785.\n",
      "coxiella_burnetii_gcf_000826165.\n",
      "coxiella_burnetii_gcf_000820465.\n",
      "coxiella_burnetii_gcf_000967075.\n",
      "coxiella_burnetii_gcf_000470495.\n",
      "coxiella_burnetii_gcf_002247545.\n",
      "coxiella_burnetii_gcf_002247155.\n",
      "coxiella_burnetii_gcf_002896835.\n",
      "coxiella_burnetii_gcf_000613025.\n",
      "coxiella_burnetii_gcf_000723305.\n",
      "coxiella_burnetii_gcf_002924395.\n",
      "coxiella_burnetii_gcf_002247185.\n",
      "coxiella_burnetii_gcf_002896735.\n",
      "coxiella_burnetii_gcf_002896755.\n",
      "coxiella_burnetii_gcf_002896775.\n",
      "coxiella_burnetii_gcf_002247265.\n",
      "coxiella_burnetii_gcf_002247205.\n",
      "coxiella_burnetii_gcf_002924305.\n",
      "coxiella_burnetii_gcf_002924345.\n",
      "coxiella_burnetii_gcf_002924425.\n",
      "coxiella_burnetii_gcf_000723245.\n",
      "coxiella_burnetii_gcf_002247285.\n",
      "coxiella_burnetii_gcf_002247305.\n",
      "coxiella_burnetii_gcf_002247335.\n",
      "coxiella_burnetii_gcf_002249845.\n",
      "coxiella_burnetii_gcf_002924325.\n",
      "coxiella_burnetii_gcf_003849965.\n",
      "coxiella_burnetii_gcf_002247225.\n",
      "coxiella_burnetii_gcf_002896795.\n",
      "coxiella_burnetii_gcf_002924385.\n",
      "coxiella_burnetii_gcf_000751935.\n",
      "coxiella_burnetii_gcf_000820825.\n",
      "coxiella_burnetii_gcf_000756325.\n",
      "coxiella_burnetii_gcf_002591355.\n",
      "coxiella_burnetii_gcf_000169495.\n",
      "coxiella_burnetii_gcf_000723285.\n",
      "coxiella_burnetii_gcf_002896815.\n",
      "coxiella_burnetii_gcf_003849885.\n",
      "coxiella_burnetii_gcf_003849875.\n",
      "coxiella_burnetii_gcf_003849785.\n",
      "coxiella_burnetii_gcf_000723265.\n",
      "coxiella_burnetii_gcf_000300315.\n",
      "coxiella_burnetii_gca_000359545.\n",
      "coxiella_burnetii_gcf_000007765.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/coxiella/extract_probes_from_group/multifastas.sqlite --base-taxon Coxiella_burnetii_GCF_000007765;\n",
      "Loci shared by 0 taxa:\t88.0\n",
      "Loci shared by 1 taxa:\t88.0\n",
      "Loci shared by 2 taxa:\t88.0\n",
      "Loci shared by 3 taxa:\t88.0\n",
      "Loci shared by 4 taxa:\t87.0\n",
      "Loci shared by 5 taxa:\t87.0\n",
      "Loci shared by 6 taxa:\t87.0\n",
      "Loci shared by 7 taxa:\t87.0\n",
      "Loci shared by 8 taxa:\t87.0\n",
      "Loci shared by 9 taxa:\t87.0\n",
      "Loci shared by 10 taxa:\t87.0\n",
      "Loci shared by 11 taxa:\t87.0\n",
      "Loci shared by 12 taxa:\t87.0\n",
      "Loci shared by 13 taxa:\t87.0\n",
      "Loci shared by 14 taxa:\t87.0\n",
      "Loci shared by 15 taxa:\t87.0\n",
      "Loci shared by 16 taxa:\t87.0\n",
      "Loci shared by 17 taxa:\t87.0\n",
      "Loci shared by 18 taxa:\t87.0\n",
      "Loci shared by 19 taxa:\t87.0\n",
      "Loci shared by 20 taxa:\t87.0\n",
      "Loci shared by 21 taxa:\t87.0\n",
      "Loci shared by 22 taxa:\t87.0\n",
      "Loci shared by 23 taxa:\t87.0\n",
      "Loci shared by 24 taxa:\t87.0\n",
      "Loci shared by 25 taxa:\t87.0\n",
      "Loci shared by 26 taxa:\t87.0\n",
      "Loci shared by 27 taxa:\t87.0\n",
      "Loci shared by 28 taxa:\t87.0\n",
      "Loci shared by 29 taxa:\t87.0\n",
      "Loci shared by 30 taxa:\t87.0\n",
      "Loci shared by 31 taxa:\t87.0\n",
      "Loci shared by 32 taxa:\t87.0\n",
      "Loci shared by 33 taxa:\t87.0\n",
      "Loci shared by 34 taxa:\t87.0\n",
      "Loci shared by 35 taxa:\t87.0\n",
      "Loci shared by 36 taxa:\t87.0\n",
      "Loci shared by 37 taxa:\t87.0\n",
      "Loci shared by 38 taxa:\t87.0\n",
      "Loci shared by 39 taxa:\t87.0\n",
      "Loci shared by 40 taxa:\t87.0\n",
      "Loci shared by 41 taxa:\t87.0\n",
      "Loci shared by 42 taxa:\t87.0\n",
      "Loci shared by 43 taxa:\t87.0\n",
      "Loci shared by 44 taxa:\t87.0\n",
      "Loci shared by 45 taxa:\t87.0\n",
      "Loci shared by 46 taxa:\t87.0\n",
      "Loci shared by 47 taxa:\t87.0\n",
      "Loci shared by 48 taxa:\t87.0\n",
      "Loci shared by 49 taxa:\t86.0\n",
      "Loci shared by 50 taxa:\t86.0\n",
      "Loci shared by 51 taxa:\t86.0\n",
      "Loci shared by 52 taxa:\t86.0\n",
      "Loci shared by 53 taxa:\t86.0\n",
      "Loci shared by 54 taxa:\t86.0\n",
      "Loci shared by 55 taxa:\t86.0\n",
      "Loci shared by 56 taxa:\t86.0\n",
      "Loci shared by 57 taxa:\t85.0\n",
      "Loci shared by 58 taxa:\t85.0\n",
      "Loci shared by 59 taxa:\t85.0\n",
      "Loci shared by 60 taxa:\t85.0\n",
      "Loci shared by 61 taxa:\t85.0\n",
      "Loci shared by 62 taxa:\t84.0\n",
      "Loci shared by 63 taxa:\t84.0\n",
      "Loci shared by 64 taxa:\t84.0\n",
      "Loci shared by 65 taxa:\t82.0\n",
      "Loci shared by 66 taxa:\t82.0\n",
      "Loci shared by 67 taxa:\t82.0\n",
      "Loci shared by 68 taxa:\t81.0\n",
      "Loci shared by 69 taxa:\t65.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/coxiella/extract_probes_from_group/multifastas.sqlite  --base-taxon Coxiella_burnetii_GCF_000007765 --output results/phyluce/coxiella/extract_probes_from_group/Coxiella_burnetii_GCF_000007765+68-back-to-68.conf --specific-counts 68;\n",
      "Counter({'coxiella_burnetii_gcf_005280755': 81, 'coxiella_burnetii_gcf_001572765': 81, 'coxiella_burnetii_gcf_002094935': 81, 'coxiella_burnetii_gcf_003849885': 81, 'coxiella_burnetii_gcf_000470495': 81, 'coxiella_burnetii_gcf_002247185': 81, 'coxiella_burnetii_gcf_004087735': 81, 'coxiella_burnetii_gcf_000007765': 81, 'coxiella_burnetii_gcf_001572745': 81, 'coxiella_burnetii_gcf_002247285': 81, 'coxiella_burnetii_gcf_002924385': 81, 'coxiella_burnetii_gcf_002633925': 81, 'coxiella_burnetii_gcf_002634045': 81, 'coxiella_burnetii_gcf_002896835': 81, 'coxiella_burnetii_gca_000359545': 81, 'coxiella_burnetii_gcf_002634085': 81, 'coxiella_burnetii_gcf_000019865': 81, 'coxiella_burnetii_gcf_002249845': 81, 'coxiella_burnetii_gcf_002247205': 81, 'coxiella_burnetii_gcf_002633885': 81, 'coxiella_burnetii_gcf_002247265': 81, 'coxiella_burnetii_gcf_002634005': 81, 'coxiella_burnetii_gcf_000168875': 81, 'coxiella_burnetii_gcf_002247225': 81, 'coxiella_burnetii_gcf_002247305': 81, 'coxiella_burnetii_gcf_000767035': 81, 'coxiella_burnetii_gcf_000613025': 81, 'coxiella_burnetii_gcf_002896755': 81, 'coxiella_burnetii_gcf_000723305': 81, 'coxiella_burnetii_gcf_000751935': 81, 'coxiella_burnetii_gcf_002633945': 81, 'coxiella_burnetii_gcf_002634065': 81, 'coxiella_burnetii_gcf_002896775': 81, 'coxiella_burnetii_gcf_003849875': 81, 'coxiella_burnetii_gcf_004006195': 81, 'coxiella_burnetii_gcf_002924425': 81, 'coxiella_burnetii_gcf_000018745': 81, 'coxiella_burnetii_gcf_002633985': 81, 'coxiella_burnetii_gcf_002896815': 81, 'coxiella_burnetii_gcf_002634025': 81, 'coxiella_burnetii_gcf_002247545': 81, 'coxiella_burnetii_gcf_000019885': 81, 'coxiella_burnetii_gcf_000820465': 81, 'coxiella_burnetii_gcf_002109345': 81, 'coxiella_burnetii_gcf_003849965': 81, 'coxiella_burnetii_gcf_000723285': 81, 'coxiella_burnetii_gcf_002924325': 81, 'coxiella_burnetii_gcf_002896735': 81, 'coxiella_burnetii_gcf_002924305': 81, 'coxiella_burnetii_gcf_000723245': 81, 'coxiella_burnetii_gcf_002633965': 81, 'coxiella_burnetii_gcf_002924345': 81, 'coxiella_burnetii_gcf_002247335': 81, 'coxiella_burnetii_gcf_002247155': 81, 'coxiella_burnetii_gcf_002633905': 81, 'coxiella_burnetii_gcf_000612785': 81, 'coxiella_burnetii_gcf_000367725': 81, 'coxiella_burnetii_gcf_002591355': 81, 'coxiella_burnetii_gcf_000820825': 81, 'coxiella_burnetii_gcf_000017105': 81, 'coxiella_burnetii_gcf_003849785': 81, 'coxiella_burnetii_gcf_002924395': 80, 'coxiella_burnetii_gcf_000723265': 80, 'coxiella_burnetii_gcf_000300315': 80, 'coxiella_burnetii_gcf_000756325': 80, 'coxiella_burnetii_gcf_000826165': 80, 'coxiella_burnetii_gcf_002896795': 80, 'coxiella_burnetii_gcf_000169495': 76, 'coxiella_burnetii_gcf_000967075': 76})\n",
      "Total loci = 81\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(68)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/coxiella/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/coxiella/extract_probes_from_group/Coxiella_burnetii_GCF_000007765+68-back-to-68.conf --probe-prefix uce_coxiella_ --designer rnplattii --design coxiella_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list.fasta;\n",
      "NNNNGGGGGGGGGGGGGGNNGGGGGGGGGGGGGGGGGGGGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "\n",
      "\n",
      "Conserved locus count = 81\n",
      "Probe Count = 11076\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list.fasta --query results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  09:47:11\n",
      "Ended:  Fri Feb 07, 2020  09:48:29\n",
      "Time for execution:  1.3013803641 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list.fasta --lastz results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_coxiella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 11075 fasta sequences.  Filtered 0 duplicates. Kept 11076.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/coxiella/final_probe_design/coxiella_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 10:21:16 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 11076\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 886080\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 2M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 89M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3946150\n",
      "Max number of word counting entries: 88827838\n",
      "\n",
      "# comparing sequences from          0  to       1846\n",
      ".---------- new table with       42 representatives\n",
      "# comparing sequences from       1846  to       3384\n",
      "......................---------- new table with       38 representatives\n",
      "# comparing sequences from       3384  to       4666\n",
      "....................---------- new table with       32 representatives\n",
      "# comparing sequences from       4666  to       5734\n",
      ".....................---------- new table with       24 representatives\n",
      "# comparing sequences from       5734  to       6624\n",
      ".....................---------- new table with       18 representatives\n",
      "# comparing sequences from       6624  to       7366\n",
      "....................---------- new table with       15 representatives\n",
      "# comparing sequences from       7366  to       7984\n",
      "....................---------- new table with       15 representatives\n",
      "# comparing sequences from       7984  to       8499\n",
      "99.9%---------- new table with       12 representatives\n",
      "# comparing sequences from       8499  to       8928\n",
      "99.8%---------- new table with        9 representatives\n",
      "# comparing sequences from       8928  to       9286\n",
      "....................---------- new table with        5 representatives\n",
      "# comparing sequences from       9286  to       9584\n",
      "...................---------- new table with        9 representatives\n",
      "# comparing sequences from       9584  to       9832\n",
      "100.0%---------- new table with        3 representatives\n",
      "# comparing sequences from       9832  to      10039\n",
      "..........    10000  finished        228  clusters\n",
      "100.0%---------- new table with        6 representatives\n",
      "# comparing sequences from      10039  to      10211\n",
      "100.0%---------- new table with        3 representatives\n",
      "# comparing sequences from      10211  to      11076\n",
      "....................---------- new table with       20 representatives\n",
      "\n",
      "    11076  finished        251  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 89M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.37\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ehrlichia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'ehrlichia'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Ehrlichia_chaffeensis_GCF_000013145\",\n",
    "               \"Ehrlichia_minasensis_GCF_000825765\",\n",
    "               \"Ehrlichia_muris_GCF_000508225\",\n",
    "               \"Ehrlichia_ruminantium_GCF_000026005\",\n",
    "               \"Ehrlichia_sp_GCF_000632845\" ]\n",
    "                    \n",
    "reference_taxon = \"Ehrlichia_canis_GCF_000012565\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000012565.1_ASM1256v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 373614 bytes  149462.40 bytes/sec\n",
      "total size is 373414  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000013145.1_ASM1314v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 334513 bytes  223036.67 bytes/sec\n",
      "total size is 334321  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000825765.1_Assembly_of_Ehrlichia_mineirensis_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 410310 bytes  273568.00 bytes/sec\n",
      "total size is 410078  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000508225.1_ASM50822v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 335163 bytes  134082.00 bytes/sec\n",
      "total size is 334970  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000026005.1_ASM2600v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 425221 bytes  170105.20 bytes/sec\n",
      "total size is 425013  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000632845.1_ASM63284v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 326450 bytes  217661.33 bytes/sec\n",
      "total size is 326265  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482997 (\"sim_Ehrlichia_chaffeensis_GCF_000013145\") has been submitted\n",
      "Your job 5482998 (\"sim_Ehrlichia_minasensis_GCF_000825765\") has been submitted\n",
      "Your job 5482999 (\"sim_Ehrlichia_muris_GCF_000508225\") has been submitted\n",
      "Your job 5483000 (\"sim_Ehrlichia_ruminantium_GCF_000026005\") has been submitted\n",
      "Your job 5483001 (\"sim_Ehrlichia_sp_GCF_000632845\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/ehrlichia/cleaned_genomes/Ehrlichia_canis_GCF_000012565_formatted.fas path=results/phyluce/ehrlichia/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483002 (\"map_Ehrlichia_chaffeensis_GCF_000013145\") has been submitted\n",
      "Your job 5483003 (\"map_Ehrlichia_minasensis_GCF_000825765\") has been submitted\n",
      "Your job 5483004 (\"map_Ehrlichia_muris_GCF_000508225\") has been submitted\n",
      "Your job 5483005 (\"map_Ehrlichia_ruminantium_GCF_000026005\") has been submitted\n",
      "Your job 5483006 (\"map_Ehrlichia_sp_GCF_000632845\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483007 (\"merge_Ehrlichia_chaffeensis_GCF_000013145\") has been submitted\n",
      "Your job 5483008 (\"merge_Ehrlichia_minasensis_GCF_000825765\") has been submitted\n",
      "Your job 5483009 (\"merge_Ehrlichia_muris_GCF_000508225\") has been submitted\n",
      "Your job 5483010 (\"merge_Ehrlichia_ruminantium_GCF_000026005\") has been submitted\n",
      "Your job 5483011 (\"merge_Ehrlichia_sp_GCF_000632845\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 1367 sequences from Ehrlichia_chaffeensis_GCF_000013145_merged.bed.  Filtered 697 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 670.\n",
      "Screened 1144 sequences from Ehrlichia_minasensis_GCF_000825765_merged.bed.  Filtered 254 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 890.\n",
      "Screened 1479 sequences from Ehrlichia_muris_GCF_000508225_merged.bed.  Filtered 813 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 666.\n",
      "Screened 480 sequences from Ehrlichia_ruminantium_GCF_000026005_merged.bed.  Filtered 321 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 159.\n",
      "Screened 1426 sequences from Ehrlichia_sp_GCF_000632845_merged.bed.  Filtered 777 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 649.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/ehrlichia/initial_intervals/Ehrlichia_sp_GCF_000632845_merged.bed --twobit results/phyluce/ehrlichia/cleaned_genomes/Ehrlichia_canis_GCF_000012565_formatted.2bit --output results/phyluce/ehrlichia/initial_intervals/Ehrlichia_sp_GCF_000632845_stripped.bed;\n",
      "ehrlichia_chaffeensis_gcf_000013145.\n",
      "ehrlichia_minasensis_gcf_000825765.\n",
      "ehrlichia_muris_gcf_000508225.\n",
      "ehrlichia_ruminantium_gcf_000026005.\n",
      "ehrlichia_sp_gcf_000632845.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/ehrlichia/initial_intervals/Ehrlichia_sp_GCF_000632845_merged.bed --twobit results/phyluce/ehrlichia/cleaned_genomes/Ehrlichia_canis_GCF_000012565_formatted.2bit --output results/phyluce/ehrlichia/initial_intervals/Ehrlichia_sp_GCF_000632845_stripped.bed;\n",
      "Loci shared by Ehrlichia_canis_GCF_000012565 + 0 taxa:\t1,397.0\n",
      "Loci shared by Ehrlichia_canis_GCF_000012565 + 1 taxa:\t1,397.0\n",
      "Loci shared by Ehrlichia_canis_GCF_000012565 + 2 taxa:\t830.0\n",
      "Loci shared by Ehrlichia_canis_GCF_000012565 + 3 taxa:\t568.0\n",
      "Loci shared by Ehrlichia_canis_GCF_000012565 + 4 taxa:\t426.0\n",
      "Loci shared by Ehrlichia_canis_GCF_000012565 + 5 taxa:\t122.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/ehrlichia/initial_intervals/ehrlichia-to-Ehrlichia_canis_GCF_000012565.sqlite --base-taxon Ehrlichia_canis_GCF_000012565 --output results/phyluce/ehrlichia/initial_intervals/Ehrlichia_canis_GCF_000012565_+4.bed --specific-counts 4;\n",
      "Counter({'ehrlichia_minasensis_gcf_000825765': 426, 'ehrlichia_chaffeensis_gcf_000013145': 425, 'ehrlichia_sp_gcf_000632845': 423, 'ehrlichia_muris_gcf_000508225': 418, 'ehrlichia_ruminantium_gcf_000026005': 134})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 4\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/ehrlichia/initial_intervals/Ehrlichia_canis_GCF_000012565_+4.bed --twobit results/phyluce/ehrlichia/cleaned_genomes/Ehrlichia_canis_GCF_000012565_formatted.2bit --buffer-to 160 --output results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4.fasta;\n",
      "Screened 426 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 426.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4.fasta --probe-prefix uce_ehrlichia_ --design ehrlichia_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 345\n",
      "Probe Count = 611\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas --query results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  10:28:15\n",
      "Ended:  Fri Feb 07, 2020  10:28:15\n",
      "Time for execution:  0.00455739895503 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas                        --lastz results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes_vself.lastz                       --probe-prefix=uce_ehrlichia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 610 fasta sequences.  Filtered 2 duplicates. Kept 607.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/ehrlichia/validate_intervals/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas --scaffoldlist Ehrlichia_chaffeensis_GCF_000013145 Ehrlichia_minasensis_GCF_000825765 Ehrlichia_muris_GCF_000508225 Ehrlichia_ruminantium_GCF_000026005 Ehrlichia_sp_GCF_000632845 Ehrlichia_canis_GCF_000012565 --genome-base-path results/phyluce/ehrlichia/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/ehrlichia/validate_intervals/ehrlichia-to-Ehrlichia_canis_GCF_000012565.sqlite --output results/phyluce/ehrlichia/validate_intervals/lastz/;\n",
      "\n",
      "Running against Ehrlichia_chaffeensis_GCF_000013145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3B_2hh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaLFPwf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/ehrlichia/validate_intervals/lastz/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_Ehrlichia_chaffeensis_GCF_000013145.lastz\n",
      "Creating Ehrlichia_chaffeensis_GCF_000013145 table\n",
      "Inserting data to Ehrlichia_chaffeensis_GCF_000013145 table\n",
      "\n",
      "Running against Ehrlichia_minasensis_GCF_000825765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9zQI4U.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgmqvv0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/ehrlichia/validate_intervals/lastz/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_Ehrlichia_minasensis_GCF_000825765.lastz\n",
      "Creating Ehrlichia_minasensis_GCF_000825765 table\n",
      "Inserting data to Ehrlichia_minasensis_GCF_000825765 table\n",
      "\n",
      "Running against Ehrlichia_muris_GCF_000508225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGEbG3J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpraap7f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/ehrlichia/validate_intervals/lastz/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_Ehrlichia_muris_GCF_000508225.lastz\n",
      "Creating Ehrlichia_muris_GCF_000508225 table\n",
      "Inserting data to Ehrlichia_muris_GCF_000508225 table\n",
      "\n",
      "Running against Ehrlichia_ruminantium_GCF_000026005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDwg2Ll.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwEZ8e7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/ehrlichia/validate_intervals/lastz/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_Ehrlichia_ruminantium_GCF_000026005.lastz\n",
      "Creating Ehrlichia_ruminantium_GCF_000026005 table\n",
      "Inserting data to Ehrlichia_ruminantium_GCF_000026005 table\n",
      "\n",
      "Running against Ehrlichia_sp_GCF_000632845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpA6EFB9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvtj9_x.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/ehrlichia/validate_intervals/lastz/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_Ehrlichia_sp_GCF_000632845.lastz\n",
      "Creating Ehrlichia_sp_GCF_000632845 table\n",
      "Inserting data to Ehrlichia_sp_GCF_000632845 table\n",
      "\n",
      "Running against Ehrlichia_canis_GCF_000012565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1WMczD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfYGi02.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/ehrlichia/validate_intervals/lastz/Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_Ehrlichia_canis_GCF_000012565.lastz\n",
      "Creating Ehrlichia_canis_GCF_000012565 table\n",
      "Inserting data to Ehrlichia_canis_GCF_000012565 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/ehrlichia/extract_probes_from_group/ehrlichia_genome.conf --lastz results/phyluce/ehrlichia/validate_intervals/lastz --probes 120 --probe-prefix uce_ehrlichia_ --name-pattern \"Ehrlichia_canis_GCF_000012565_+4_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/ehrlichia/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 10:28:27,920 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 10:28:27,922 - Phyluce - INFO - ----- Working on Ehrlichia_chaffeensis_GCF_000013145 genome -----\n",
      "2020-02-07 10:28:27,923 - Phyluce - INFO - Reading Ehrlichia_chaffeensis_GCF_000013145 genome\n",
      "2020-02-07 10:28:29,336 - Phyluce - INFO - Ehrlichia_chaffeensis_GCF_000013145: 345 uces, 0 dupes, 345 non-dupes, 4 orient drop, 5 length drop, 336 written\n",
      "2020-02-07 10:28:29,337 - Phyluce - INFO - ------ Working on Ehrlichia_minasensis_GCF_000825765 genome -----\n",
      "2020-02-07 10:28:29,337 - Phyluce - INFO - Reading Ehrlichia_minasensis_GCF_000825765 genome\n",
      "2020-02-07 10:28:30,375 - Phyluce - INFO - Ehrlichia_minasensis_GCF_000825765: 345 uces, 10 dupes, 335 non-dupes, 1 orient drop, 3 length drop, 331 written\n",
      "2020-02-07 10:28:30,375 - Phyluce - INFO - -------- Working on Ehrlichia_muris_GCF_000508225 genome --------\n",
      "2020-02-07 10:28:30,376 - Phyluce - INFO - Reading Ehrlichia_muris_GCF_000508225 genome\n",
      "2020-02-07 10:28:31,413 - Phyluce - INFO - Ehrlichia_muris_GCF_000508225: 345 uces, 0 dupes, 345 non-dupes, 3 orient drop, 12 length drop, 330 written\n",
      "2020-02-07 10:28:31,413 - Phyluce - INFO - ----- Working on Ehrlichia_ruminantium_GCF_000026005 genome -----\n",
      "2020-02-07 10:28:31,414 - Phyluce - INFO - Reading Ehrlichia_ruminantium_GCF_000026005 genome\n",
      "2020-02-07 10:28:32,660 - Phyluce - INFO - Ehrlichia_ruminantium_GCF_000026005: 345 uces, 0 dupes, 345 non-dupes, 11 orient drop, 9 length drop, 325 written\n",
      "2020-02-07 10:28:32,661 - Phyluce - INFO - ---------- Working on Ehrlichia_sp_GCF_000632845 genome ---------\n",
      "2020-02-07 10:28:32,661 - Phyluce - INFO - Reading Ehrlichia_sp_GCF_000632845 genome\n",
      "2020-02-07 10:28:33,865 - Phyluce - INFO - Ehrlichia_sp_GCF_000632845: 345 uces, 0 dupes, 345 non-dupes, 4 orient drop, 7 length drop, 334 written\n",
      "2020-02-07 10:28:33,865 - Phyluce - INFO - -------- Working on Ehrlichia_canis_GCF_000012565 genome --------\n",
      "2020-02-07 10:28:33,870 - Phyluce - INFO - Reading Ehrlichia_canis_GCF_000012565 genome\n",
      "2020-02-07 10:28:35,054 - Phyluce - INFO - Ehrlichia_canis_GCF_000012565: 345 uces, 0 dupes, 345 non-dupes, 5 orient drop, 9 length drop, 331 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/ehrlichia/extract_probes_from_group/probe_fasta --output results/phyluce/ehrlichia/extract_probes_from_group/multifastas.sqlite --base-taxon Ehrlichia_canis_GCF_000012565;\n",
      "ehrlichia_chaffeensis_gcf_000013145.\n",
      "ehrlichia_minasensis_gcf_000825765.\n",
      "ehrlichia_muris_gcf_000508225.\n",
      "ehrlichia_ruminantium_gcf_000026005.\n",
      "ehrlichia_sp_gcf_000632845.\n",
      "ehrlichia_canis_gcf_000012565.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/ehrlichia/extract_probes_from_group/multifastas.sqlite --base-taxon Ehrlichia_canis_GCF_000012565;\n",
      "Loci shared by 0 taxa:\t339.0\n",
      "Loci shared by 1 taxa:\t339.0\n",
      "Loci shared by 2 taxa:\t337.0\n",
      "Loci shared by 3 taxa:\t335.0\n",
      "Loci shared by 4 taxa:\t335.0\n",
      "Loci shared by 5 taxa:\t329.0\n",
      "Loci shared by 6 taxa:\t312.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/ehrlichia/extract_probes_from_group/multifastas.sqlite  --base-taxon Ehrlichia_canis_GCF_000012565 --output results/phyluce/ehrlichia/extract_probes_from_group/Ehrlichia_canis_GCF_000012565+4-back-to-6.conf --specific-counts 6;\n",
      "Counter({'ehrlichia_ruminantium_gcf_000026005': 312, 'ehrlichia_minasensis_gcf_000825765': 312, 'ehrlichia_muris_gcf_000508225': 312, 'ehrlichia_chaffeensis_gcf_000013145': 312, 'ehrlichia_canis_gcf_000012565': 312, 'ehrlichia_sp_gcf_000632845': 312})\n",
      "Total loci = 312\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(6)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/ehrlichia/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/ehrlichia/extract_probes_from_group/Ehrlichia_canis_GCF_000012565+4-back-to-6.conf --probe-prefix uce_ehrlichia_ --designer rnplattii --design ehrlichia_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 312\n",
      "Probe Count = 3207\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list.fasta --query results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  10:28:47\n",
      "Ended:  Fri Feb 07, 2020  10:28:50\n",
      "Time for execution:  0.0384563008944 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list.fasta --lastz results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_ehrlichia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 3206 fasta sequences.  Filtered 0 duplicates. Kept 3207.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/ehrlichia/final_probe_design/ehrlichia_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 10:29:38 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 3207\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 256560\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3959542\n",
      "Max number of word counting entries: 89129304\n",
      "\n",
      "# comparing sequences from          0  to        534\n",
      "---------- new table with      449 representatives\n",
      "# comparing sequences from        534  to        979\n",
      "----------    142 remaining sequences to the next cycle\n",
      "---------- new table with      252 representatives\n",
      "# comparing sequences from        837  to       1232\n",
      "----------    149 remaining sequences to the next cycle\n",
      "---------- new table with      209 representatives\n",
      "# comparing sequences from       1083  to       1437\n",
      "----------     57 remaining sequences to the next cycle\n",
      "---------- new table with      239 representatives\n",
      "# comparing sequences from       1380  to       1684\n",
      "----------     33 remaining sequences to the next cycle\n",
      "---------- new table with      218 representatives\n",
      "# comparing sequences from       1651  to       1910\n",
      "----------     14 remaining sequences to the next cycle\n",
      "---------- new table with      197 representatives\n",
      "# comparing sequences from       1896  to       2114\n",
      "99.9%---------- new table with      182 representatives\n",
      "# comparing sequences from       2114  to       2296\n",
      "----------     53 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       2243  to       3207\n",
      ".....................---------- new table with      793 representatives\n",
      "\n",
      "     3207  finished       2639  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.80\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Francisella"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'francisella'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Francisella_adeliensis_GCF_003290445\",\n",
    "               \"Francisella_frigiditurris_GCF_001880225\",\n",
    "               \"Francisella_halioticida_GCF_002211785\",\n",
    "               \"Francisella_hispniensis_GCF_001885235\",\n",
    "               \"Francisella_marina_GCF_008369785\",\n",
    "               \"Francisella_noatunensis_GCF_008330245\",\n",
    "               \"Francisella_opportunistica_GCF_003347135\",\n",
    "               \"Francisella_persica_GCF_001653955\",\n",
    "               \"Francisella_philomiragia_GCF_000019285\",\n",
    "               \"Francisella_salina_GCF_000219045\",\n",
    "               \"Francisella_sp_GCF_000764555\",\n",
    "               \"Francisella_uliginis_GCF_001895265\" ]\n",
    "                    \n",
    "reference_taxon = \"Francisella_tularensis_GCF_000008985\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003290445.1_ASM329044v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597341 bytes  398255.33 bytes/sec\n",
      "total size is 597083  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001880225.1_ASM188022v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 536837 bytes  357919.33 bytes/sec\n",
      "total size is 536595  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002211785.1_ASM221178v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 620187 bytes  413486.00 bytes/sec\n",
      "total size is 619929  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001885235.1_ASM188523v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 561757 bytes  374532.67 bytes/sec\n",
      "total size is 561507  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008369785.1_ASM836978v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 601266 bytes  400872.00 bytes/sec\n",
      "total size is 601008  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008330245.1_ASM833024v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 542224 bytes  361510.67 bytes/sec\n",
      "total size is 541982  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003347135.1_ASM334713v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 533528 bytes  355713.33 bytes/sec\n",
      "total size is 533286  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001653955.1_ASM165395v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 444440 bytes  296321.33 bytes/sec\n",
      "total size is 444222  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000019285.1_ASM1928v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 593547 bytes  1187178.00 bytes/sec\n",
      "total size is 593291  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000219045.1_ASM21904v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 593045 bytes  395391.33 bytes/sec\n",
      "total size is 592788  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000764555.1_ASM76455v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 584089 bytes  389420.67 bytes/sec\n",
      "total size is 583840  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000008985.1_ASM898v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 547773 bytes  365210.00 bytes/sec\n",
      "total size is 547534  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001895265.1_ASM189526v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 644544 bytes  429724.00 bytes/sec\n",
      "total size is 644278  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483012 (\"sim_Francisella_adeliensis_GCF_003290445\") has been submitted\n",
      "Your job 5483013 (\"sim_Francisella_frigiditurris_GCF_001880225\") has been submitted\n",
      "Your job 5483014 (\"sim_Francisella_halioticida_GCF_002211785\") has been submitted\n",
      "Your job 5483015 (\"sim_Francisella_hispniensis_GCF_001885235\") has been submitted\n",
      "Your job 5483016 (\"sim_Francisella_marina_GCF_008369785\") has been submitted\n",
      "Your job 5483017 (\"sim_Francisella_noatunensis_GCF_008330245\") has been submitted\n",
      "Your job 5483018 (\"sim_Francisella_opportunistica_GCF_003347135\") has been submitted\n",
      "Your job 5483019 (\"sim_Francisella_persica_GCF_001653955\") has been submitted\n",
      "Your job 5483020 (\"sim_Francisella_philomiragia_GCF_000019285\") has been submitted\n",
      "Your job 5483021 (\"sim_Francisella_salina_GCF_000219045\") has been submitted\n",
      "Your job 5483022 (\"sim_Francisella_sp_GCF_000764555\") has been submitted\n",
      "Your job 5483023 (\"sim_Francisella_uliginis_GCF_001895265\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/francisella/cleaned_genomes/Francisella_tularensis_GCF_000008985_formatted.fas path=results/phyluce/francisella/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483024 (\"map_Francisella_adeliensis_GCF_003290445\") has been submitted\n",
      "Your job 5483025 (\"map_Francisella_frigiditurris_GCF_001880225\") has been submitted\n",
      "Your job 5483026 (\"map_Francisella_halioticida_GCF_002211785\") has been submitted\n",
      "Your job 5483027 (\"map_Francisella_hispniensis_GCF_001885235\") has been submitted\n",
      "Your job 5483028 (\"map_Francisella_marina_GCF_008369785\") has been submitted\n",
      "Your job 5483029 (\"map_Francisella_noatunensis_GCF_008330245\") has been submitted\n",
      "Your job 5483030 (\"map_Francisella_opportunistica_GCF_003347135\") has been submitted\n",
      "Your job 5483031 (\"map_Francisella_persica_GCF_001653955\") has been submitted\n",
      "Your job 5483032 (\"map_Francisella_philomiragia_GCF_000019285\") has been submitted\n",
      "Your job 5483033 (\"map_Francisella_salina_GCF_000219045\") has been submitted\n",
      "Your job 5483034 (\"map_Francisella_sp_GCF_000764555\") has been submitted\n",
      "Your job 5483035 (\"map_Francisella_uliginis_GCF_001895265\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483036 (\"merge_Francisella_adeliensis_GCF_003290445\") has been submitted\n",
      "Your job 5483037 (\"merge_Francisella_frigiditurris_GCF_001880225\") has been submitted\n",
      "Your job 5483038 (\"merge_Francisella_halioticida_GCF_002211785\") has been submitted\n",
      "Your job 5483039 (\"merge_Francisella_hispniensis_GCF_001885235\") has been submitted\n",
      "Your job 5483040 (\"merge_Francisella_marina_GCF_008369785\") has been submitted\n",
      "Your job 5483041 (\"merge_Francisella_noatunensis_GCF_008330245\") has been submitted\n",
      "Your job 5483042 (\"merge_Francisella_opportunistica_GCF_003347135\") has been submitted\n",
      "Your job 5483043 (\"merge_Francisella_persica_GCF_001653955\") has been submitted\n",
      "Your job 5483044 (\"merge_Francisella_philomiragia_GCF_000019285\") has been submitted\n",
      "Your job 5483045 (\"merge_Francisella_salina_GCF_000219045\") has been submitted\n",
      "Your job 5483046 (\"merge_Francisella_sp_GCF_000764555\") has been submitted\n",
      "Your job 5483047 (\"merge_Francisella_uliginis_GCF_001895265\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 295 sequences from Francisella_adeliensis_GCF_003290445_merged.bed.  Filtered 211 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 84.\n",
      "Screened 234 sequences from Francisella_frigiditurris_GCF_001880225_merged.bed.  Filtered 162 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 72.\n",
      "Screened 638 sequences from Francisella_halioticida_GCF_002211785_merged.bed.  Filtered 454 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 184.\n",
      "Screened 2181 sequences from Francisella_hispniensis_GCF_001885235_merged.bed.  Filtered 430 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1751.\n",
      "Screened 1263 sequences from Francisella_marina_GCF_008369785_merged.bed.  Filtered 734 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 529.\n",
      "Screened 1242 sequences from Francisella_noatunensis_GCF_008330245_merged.bed.  Filtered 688 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 554.\n",
      "Screened 2919 sequences from Francisella_opportunistica_GCF_003347135_merged.bed.  Filtered 1059 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1860.\n",
      "Screened 2406 sequences from Francisella_persica_GCF_001653955_merged.bed.  Filtered 844 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1562.\n",
      "Screened 1382 sequences from Francisella_philomiragia_GCF_000019285_merged.bed.  Filtered 746 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 636.\n",
      "Screened 1261 sequences from Francisella_salina_GCF_000219045_merged.bed.  Filtered 689 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 572.\n",
      "Screened 737 sequences from Francisella_sp_GCF_000764555_merged.bed.  Filtered 499 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 238.\n",
      "Screened 829 sequences from Francisella_uliginis_GCF_001895265_merged.bed.  Filtered 546 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 283.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/francisella/initial_intervals/Francisella_uliginis_GCF_001895265_merged.bed --twobit results/phyluce/francisella/cleaned_genomes/Francisella_tularensis_GCF_000008985_formatted.2bit --output results/phyluce/francisella/initial_intervals/Francisella_uliginis_GCF_001895265_stripped.bed;\n",
      "francisella_adeliensis_gcf_003290445.\n",
      "francisella_frigiditurris_gcf_001880225.\n",
      "francisella_halioticida_gcf_002211785.\n",
      "francisella_hispniensis_gcf_001885235..\n",
      "francisella_marina_gcf_008369785.\n",
      "francisella_noatunensis_gcf_008330245.\n",
      "francisella_opportunistica_gcf_003347135..\n",
      "francisella_persica_gcf_001653955..\n",
      "francisella_philomiragia_gcf_000019285.\n",
      "francisella_salina_gcf_000219045.\n",
      "francisella_sp_gcf_000764555.\n",
      "francisella_uliginis_gcf_001895265.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/francisella/initial_intervals/Francisella_uliginis_GCF_001895265_merged.bed --twobit results/phyluce/francisella/cleaned_genomes/Francisella_tularensis_GCF_000008985_formatted.2bit --output results/phyluce/francisella/initial_intervals/Francisella_uliginis_GCF_001895265_stripped.bed;\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 0 taxa:\t2,296.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 1 taxa:\t2,296.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 2 taxa:\t1,761.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 3 taxa:\t1,286.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 4 taxa:\t720.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 5 taxa:\t543.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 6 taxa:\t371.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 7 taxa:\t269.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 8 taxa:\t188.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 9 taxa:\t128.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 10 taxa:\t83.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 11 taxa:\t48.0\n",
      "Loci shared by Francisella_tularensis_GCF_000008985 + 12 taxa:\t27.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/francisella/initial_intervals/francisella-to-Francisella_tularensis_GCF_000008985.sqlite --base-taxon Francisella_tularensis_GCF_000008985 --output results/phyluce/francisella/initial_intervals/Francisella_tularensis_GCF_000008985_+7.bed --specific-counts 7;\n",
      "Counter({'francisella_philomiragia_gcf_000019285': 254, 'francisella_opportunistica_gcf_003347135': 252, 'francisella_hispniensis_gcf_001885235': 252, 'francisella_noatunensis_gcf_008330245': 247, 'francisella_persica_gcf_001653955': 244, 'francisella_salina_gcf_000219045': 242, 'francisella_marina_gcf_008369785': 241, 'francisella_uliginis_gcf_001895265': 198, 'francisella_sp_gcf_000764555': 168, 'francisella_halioticida_gcf_002211785': 148, 'francisella_adeliensis_gcf_003290445': 64, 'francisella_frigiditurris_gcf_001880225': 47})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 7\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/francisella/initial_intervals/Francisella_tularensis_GCF_000008985_+7.bed --twobit results/phyluce/francisella/cleaned_genomes/Francisella_tularensis_GCF_000008985_formatted.2bit --buffer-to 160 --output results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7.fasta;\n",
      "Screened 269 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 269.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7.fasta --probe-prefix uce_francisella_ --design francisella_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 253\n",
      "Probe Count = 470\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas --query results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  10:33:58\n",
      "Ended:  Fri Feb 07, 2020  10:34:00\n",
      "Time for execution:  0.0289617498716 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas                        --lastz results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes_vself.lastz                       --probe-prefix=uce_francisella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 469 fasta sequences.  Filtered 2 duplicates. Kept 466.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/francisella/validate_intervals/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas --scaffoldlist Francisella_adeliensis_GCF_003290445 Francisella_frigiditurris_GCF_001880225 Francisella_halioticida_GCF_002211785 Francisella_hispniensis_GCF_001885235 Francisella_marina_GCF_008369785 Francisella_noatunensis_GCF_008330245 Francisella_opportunistica_GCF_003347135 Francisella_persica_GCF_001653955 Francisella_philomiragia_GCF_000019285 Francisella_salina_GCF_000219045 Francisella_sp_GCF_000764555 Francisella_uliginis_GCF_001895265 Francisella_tularensis_GCF_000008985 --genome-base-path results/phyluce/francisella/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/francisella/validate_intervals/francisella-to-Francisella_tularensis_GCF_000008985.sqlite --output results/phyluce/francisella/validate_intervals/lastz/;\n",
      "\n",
      "Running against Francisella_adeliensis_GCF_003290445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnpfSee.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxSdnzM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_adeliensis_GCF_003290445.lastz\n",
      "Creating Francisella_adeliensis_GCF_003290445 table\n",
      "Inserting data to Francisella_adeliensis_GCF_003290445 table\n",
      "\n",
      "Running against Francisella_frigiditurris_GCF_001880225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8cwiT_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQQawu4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_frigiditurris_GCF_001880225.lastz\n",
      "Creating Francisella_frigiditurris_GCF_001880225 table\n",
      "Inserting data to Francisella_frigiditurris_GCF_001880225 table\n",
      "\n",
      "Running against Francisella_halioticida_GCF_002211785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuLLHZ0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_Cs_cD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_halioticida_GCF_002211785.lastz\n",
      "Creating Francisella_halioticida_GCF_002211785 table\n",
      "Inserting data to Francisella_halioticida_GCF_002211785 table\n",
      "\n",
      "Running against Francisella_hispniensis_GCF_001885235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3AkyCV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpODHBrs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_hispniensis_GCF_001885235.lastz\n",
      "Creating Francisella_hispniensis_GCF_001885235 table\n",
      "Inserting data to Francisella_hispniensis_GCF_001885235 table\n",
      "\n",
      "Running against Francisella_marina_GCF_008369785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkE3MaE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpW5_5Wu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_marina_GCF_008369785.lastz\n",
      "Creating Francisella_marina_GCF_008369785 table\n",
      "Inserting data to Francisella_marina_GCF_008369785 table\n",
      "\n",
      "Running against Francisella_noatunensis_GCF_008330245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjZBnxN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIlLwIT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_noatunensis_GCF_008330245.lastz\n",
      "Creating Francisella_noatunensis_GCF_008330245 table\n",
      "Inserting data to Francisella_noatunensis_GCF_008330245 table\n",
      "\n",
      "Running against Francisella_opportunistica_GCF_003347135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVMcnFP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0wP2b_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_opportunistica_GCF_003347135.lastz\n",
      "Creating Francisella_opportunistica_GCF_003347135 table\n",
      "Inserting data to Francisella_opportunistica_GCF_003347135 table\n",
      "\n",
      "Running against Francisella_persica_GCF_001653955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3NOW_E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqjqbBM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_persica_GCF_001653955.lastz\n",
      "Creating Francisella_persica_GCF_001653955 table\n",
      "Inserting data to Francisella_persica_GCF_001653955 table\n",
      "\n",
      "Running against Francisella_philomiragia_GCF_000019285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWrpHgN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphyiq9Z.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_philomiragia_GCF_000019285.lastz\n",
      "Creating Francisella_philomiragia_GCF_000019285 table\n",
      "Inserting data to Francisella_philomiragia_GCF_000019285 table\n",
      "\n",
      "Running against Francisella_salina_GCF_000219045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDhvtUH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVbY2CU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_salina_GCF_000219045.lastz\n",
      "Creating Francisella_salina_GCF_000219045 table\n",
      "Inserting data to Francisella_salina_GCF_000219045 table\n",
      "\n",
      "Running against Francisella_sp_GCF_000764555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQyEmkx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbVt1fc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_sp_GCF_000764555.lastz\n",
      "Creating Francisella_sp_GCF_000764555 table\n",
      "Inserting data to Francisella_sp_GCF_000764555 table\n",
      "\n",
      "Running against Francisella_uliginis_GCF_001895265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxOc8Bq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppL_XLQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_uliginis_GCF_001895265.lastz\n",
      "Creating Francisella_uliginis_GCF_001895265 table\n",
      "Inserting data to Francisella_uliginis_GCF_001895265 table\n",
      "\n",
      "Running against Francisella_tularensis_GCF_000008985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpT9tMoV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqegYi9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/francisella/validate_intervals/lastz/Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_Francisella_tularensis_GCF_000008985.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Francisella_tularensis_GCF_000008985 table\r\n",
      "Inserting data to Francisella_tularensis_GCF_000008985 table\r\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/francisella/extract_probes_from_group/francisella_genome.conf --lastz results/phyluce/francisella/validate_intervals/lastz --probes 120 --probe-prefix uce_francisella_ --name-pattern \"Francisella_tularensis_GCF_000008985_+7_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/francisella/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 10:34:23,743 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 10:34:23,745 - Phyluce - INFO - ----- Working on Francisella_adeliensis_GCF_003290445 genome ----\n",
      "2020-02-07 10:34:23,746 - Phyluce - INFO - Reading Francisella_adeliensis_GCF_003290445 genome\n",
      "2020-02-07 10:34:24,610 - Phyluce - INFO - Francisella_adeliensis_GCF_003290445: 245 uces, 0 dupes, 245 non-dupes, 7 orient drop, 2 length drop, 236 written\n",
      "2020-02-07 10:34:24,610 - Phyluce - INFO - --- Working on Francisella_frigiditurris_GCF_001880225 genome ---\n",
      "2020-02-07 10:34:24,611 - Phyluce - INFO - Reading Francisella_frigiditurris_GCF_001880225 genome\n",
      "2020-02-07 10:34:25,358 - Phyluce - INFO - Francisella_frigiditurris_GCF_001880225: 239 uces, 0 dupes, 239 non-dupes, 5 orient drop, 2 length drop, 232 written\n",
      "2020-02-07 10:34:25,359 - Phyluce - INFO - ---- Working on Francisella_halioticida_GCF_002211785 genome ----\n",
      "2020-02-07 10:34:25,359 - Phyluce - INFO - Reading Francisella_halioticida_GCF_002211785 genome\n",
      "2020-02-07 10:34:26,172 - Phyluce - INFO - Francisella_halioticida_GCF_002211785: 248 uces, 0 dupes, 248 non-dupes, 9 orient drop, 5 length drop, 234 written\n",
      "2020-02-07 10:34:26,172 - Phyluce - INFO - ---- Working on Francisella_hispniensis_GCF_001885235 genome ----\n",
      "2020-02-07 10:34:26,173 - Phyluce - INFO - Reading Francisella_hispniensis_GCF_001885235 genome\n",
      "2020-02-07 10:34:27,167 - Phyluce - INFO - Francisella_hispniensis_GCF_001885235: 253 uces, 0 dupes, 253 non-dupes, 9 orient drop, 3 length drop, 241 written\n",
      "2020-02-07 10:34:27,167 - Phyluce - INFO - ------- Working on Francisella_marina_GCF_008369785 genome ------\n",
      "2020-02-07 10:34:27,168 - Phyluce - INFO - Reading Francisella_marina_GCF_008369785 genome\n",
      "2020-02-07 10:34:28,123 - Phyluce - INFO - Francisella_marina_GCF_008369785: 251 uces, 0 dupes, 251 non-dupes, 10 orient drop, 2 length drop, 239 written\n",
      "2020-02-07 10:34:28,124 - Phyluce - INFO - ---- Working on Francisella_noatunensis_GCF_008330245 genome ----\n",
      "2020-02-07 10:34:28,142 - Phyluce - INFO - Reading Francisella_noatunensis_GCF_008330245 genome\n",
      "2020-02-07 10:34:28,989 - Phyluce - INFO - Francisella_noatunensis_GCF_008330245: 248 uces, 0 dupes, 248 non-dupes, 7 orient drop, 5 length drop, 236 written\n",
      "2020-02-07 10:34:28,990 - Phyluce - INFO - --- Working on Francisella_opportunistica_GCF_003347135 genome --\n",
      "2020-02-07 10:34:28,991 - Phyluce - INFO - Reading Francisella_opportunistica_GCF_003347135 genome\n",
      "2020-02-07 10:34:29,784 - Phyluce - INFO - Francisella_opportunistica_GCF_003347135: 242 uces, 0 dupes, 242 non-dupes, 10 orient drop, 4 length drop, 228 written\n",
      "2020-02-07 10:34:29,784 - Phyluce - INFO - ------ Working on Francisella_persica_GCF_001653955 genome ------\n",
      "2020-02-07 10:34:29,785 - Phyluce - INFO - Reading Francisella_persica_GCF_001653955 genome\n",
      "2020-02-07 10:34:30,636 - Phyluce - INFO - Francisella_persica_GCF_001653955: 243 uces, 0 dupes, 243 non-dupes, 6 orient drop, 5 length drop, 232 written\n",
      "2020-02-07 10:34:30,637 - Phyluce - INFO - ---- Working on Francisella_philomiragia_GCF_000019285 genome ---\n",
      "2020-02-07 10:34:30,637 - Phyluce - INFO - Reading Francisella_philomiragia_GCF_000019285 genome\n",
      "2020-02-07 10:34:31,568 - Phyluce - INFO - Francisella_philomiragia_GCF_000019285: 251 uces, 0 dupes, 251 non-dupes, 8 orient drop, 8 length drop, 235 written\n",
      "2020-02-07 10:34:31,569 - Phyluce - INFO - ------- Working on Francisella_salina_GCF_000219045 genome ------\n",
      "2020-02-07 10:34:31,569 - Phyluce - INFO - Reading Francisella_salina_GCF_000219045 genome\n",
      "2020-02-07 10:34:32,414 - Phyluce - INFO - Francisella_salina_GCF_000219045: 251 uces, 0 dupes, 251 non-dupes, 12 orient drop, 1 length drop, 238 written\n",
      "2020-02-07 10:34:32,414 - Phyluce - INFO - --------- Working on Francisella_sp_GCF_000764555 genome --------\n",
      "2020-02-07 10:34:32,415 - Phyluce - INFO - Reading Francisella_sp_GCF_000764555 genome\n",
      "2020-02-07 10:34:33,443 - Phyluce - INFO - Francisella_sp_GCF_000764555: 246 uces, 0 dupes, 246 non-dupes, 8 orient drop, 1 length drop, 237 written\n",
      "2020-02-07 10:34:33,444 - Phyluce - INFO - ------ Working on Francisella_uliginis_GCF_001895265 genome -----\n",
      "2020-02-07 10:34:33,475 - Phyluce - INFO - Reading Francisella_uliginis_GCF_001895265 genome\n",
      "2020-02-07 10:34:34,576 - Phyluce - INFO - Francisella_uliginis_GCF_001895265: 250 uces, 0 dupes, 250 non-dupes, 9 orient drop, 4 length drop, 237 written\n",
      "2020-02-07 10:34:34,577 - Phyluce - INFO - ----- Working on Francisella_tularensis_GCF_000008985 genome ----\n",
      "2020-02-07 10:34:34,577 - Phyluce - INFO - Reading Francisella_tularensis_GCF_000008985 genome\n",
      "2020-02-07 10:34:35,551 - Phyluce - INFO - Francisella_tularensis_GCF_000008985: 253 uces, 0 dupes, 253 non-dupes, 6 orient drop, 7 length drop, 240 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/francisella/extract_probes_from_group/probe_fasta --output results/phyluce/francisella/extract_probes_from_group/multifastas.sqlite --base-taxon Francisella_tularensis_GCF_000008985;\n",
      "francisella_adeliensis_gcf_003290445.\n",
      "francisella_frigiditurris_gcf_001880225.\n",
      "francisella_halioticida_gcf_002211785.\n",
      "francisella_hispniensis_gcf_001885235.\n",
      "francisella_marina_gcf_008369785.\n",
      "francisella_noatunensis_gcf_008330245.\n",
      "francisella_opportunistica_gcf_003347135.\n",
      "francisella_persica_gcf_001653955.\n",
      "francisella_philomiragia_gcf_000019285.\n",
      "francisella_salina_gcf_000219045.\n",
      "francisella_sp_gcf_000764555.\n",
      "francisella_uliginis_gcf_001895265.\n",
      "francisella_tularensis_gcf_000008985.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/francisella/extract_probes_from_group/multifastas.sqlite --base-taxon Francisella_tularensis_GCF_000008985;\n",
      "Loci shared by 0 taxa:\t248.0\n",
      "Loci shared by 1 taxa:\t248.0\n",
      "Loci shared by 2 taxa:\t248.0\n",
      "Loci shared by 3 taxa:\t243.0\n",
      "Loci shared by 4 taxa:\t243.0\n",
      "Loci shared by 5 taxa:\t243.0\n",
      "Loci shared by 6 taxa:\t242.0\n",
      "Loci shared by 7 taxa:\t242.0\n",
      "Loci shared by 8 taxa:\t239.0\n",
      "Loci shared by 9 taxa:\t238.0\n",
      "Loci shared by 10 taxa:\t232.0\n",
      "Loci shared by 11 taxa:\t225.0\n",
      "Loci shared by 12 taxa:\t221.0\n",
      "Loci shared by 13 taxa:\t201.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/francisella/extract_probes_from_group/multifastas.sqlite  --base-taxon Francisella_tularensis_GCF_000008985 --output results/phyluce/francisella/extract_probes_from_group/Francisella_tularensis_GCF_000008985+7-back-to-13.conf --specific-counts 13;\n",
      "Counter({'francisella_persica_gcf_001653955': 201, 'francisella_sp_gcf_000764555': 201, 'francisella_salina_gcf_000219045': 201, 'francisella_frigiditurris_gcf_001880225': 201, 'francisella_opportunistica_gcf_003347135': 201, 'francisella_noatunensis_gcf_008330245': 201, 'francisella_hispniensis_gcf_001885235': 201, 'francisella_marina_gcf_008369785': 201, 'francisella_philomiragia_gcf_000019285': 201, 'francisella_adeliensis_gcf_003290445': 201, 'francisella_halioticida_gcf_002211785': 201, 'francisella_tularensis_gcf_000008985': 201, 'francisella_uliginis_gcf_001895265': 201})\n",
      "Total loci = 201\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(13)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/francisella/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/francisella/extract_probes_from_group/Francisella_tularensis_GCF_000008985+7-back-to-13.conf --probe-prefix uce_francisella_ --designer rnplattii --design francisella_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 201\n",
      "Probe Count = 4671\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list.fasta --query results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  10:34:49\n",
      "Ended:  Fri Feb 07, 2020  10:34:56\n",
      "Time for execution:  0.113882136345 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list.fasta --lastz results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_francisella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 4670 fasta sequences.  Filtered 0 duplicates. Kept 4671.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/francisella/final_probe_design/francisella_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 10:37:22 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 4671\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 373680\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3957054\n",
      "Max number of word counting entries: 89073306\n",
      "\n",
      "# comparing sequences from          0  to        778\n",
      "---------- new table with      618 representatives\n",
      "# comparing sequences from        778  to       1426\n",
      "----------    212 remaining sequences to the next cycle\n",
      "---------- new table with      335 representatives\n",
      "# comparing sequences from       1214  to       1790\n",
      "----------    172 remaining sequences to the next cycle\n",
      "---------- new table with      312 representatives\n",
      "# comparing sequences from       1618  to       2126\n",
      "----------    120 remaining sequences to the next cycle\n",
      "---------- new table with      299 representatives\n",
      "# comparing sequences from       2006  to       2450\n",
      "----------    115 remaining sequences to the next cycle\n",
      "---------- new table with      234 representatives\n",
      "# comparing sequences from       2335  to       2724\n",
      "----------    127 remaining sequences to the next cycle\n",
      "---------- new table with      186 representatives\n",
      "# comparing sequences from       2597  to       2942\n",
      "----------     88 remaining sequences to the next cycle\n",
      "---------- new table with      184 representatives\n",
      "# comparing sequences from       2854  to       3156\n",
      "----------    136 remaining sequences to the next cycle\n",
      "---------- new table with      122 representatives\n",
      "# comparing sequences from       3020  to       3295\n",
      "----------     62 remaining sequences to the next cycle\n",
      "---------- new table with      163 representatives\n",
      "# comparing sequences from       3233  to       3472\n",
      "----------     39 remaining sequences to the next cycle\n",
      "---------- new table with      142 representatives\n",
      "# comparing sequences from       3433  to       3639\n",
      "----------     59 remaining sequences to the next cycle\n",
      "---------- new table with      115 representatives\n",
      "# comparing sequences from       3580  to       3761\n",
      "----------      6 remaining sequences to the next cycle\n",
      "---------- new table with      142 representatives\n",
      "# comparing sequences from       3755  to       4671\n",
      ".....................---------- new table with      603 representatives\n",
      "\n",
      "     4671  finished       3455  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.09\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Leptospira\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'leptospira'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Leptospira_adleri_GCF_002811985\",\n",
    "               \"Leptospira_alexanderi_GCF_000243815\",\n",
    "               \"Leptospira_alstonii_GCF_000347175\",\n",
    "               \"Leptospira_andrefontaineae_GCF_004770105\",\n",
    "               \"Leptospira_bandrabouensis_GCF_004770905\",\n",
    "               \"Leptospira_barantonii_GCF_002811925\",\n",
    "               \"Leptospira_biflexa_GCF_000017685\",\n",
    "               \"Leptospira_borgpetersenii_GCF_000013945\",\n",
    "               \"Leptospira_bourretii_GCF_004769285\",\n",
    "               \"Leptospira_bouyouniensis_GCF_004770625\",\n",
    "               \"Leptospira_brenneri_GCF_002812125\",\n",
    "               \"Leptospira_broomii_GCF_000243715\",\n",
    "               \"Leptospira_congkakensis_GCF_004770265\",\n",
    "               \"Leptospira_dzianensis_GCF_004770135\",\n",
    "               \"Leptospira_dzoumogneensis_GCF_004770895\",\n",
    "               \"Leptospira_ellinghausenii_GCF_003114815\",\n",
    "               \"Leptospira_elllisii_GCF_002811955\",\n",
    "               \"Leptospira_fainei_GCF_000306235\",\n",
    "               \"Leptospira_fletcheri_GCF_004769195\",\n",
    "               \"Leptospira_fluminis_GCF_004771275\",\n",
    "               \"Leptospira_gomenensis_GCF_004770155\",\n",
    "               \"Leptospira_haakeii_GCF_002812045\",\n",
    "               \"Leptospira_harrisiae_GCF_002811945\",\n",
    "               \"Leptospira_hartskeerlii_GCF_002811475\",\n",
    "               \"Leptospira_idonii_GCF_004770995\",\n",
    "               \"Leptospira_ilyithenensis_GCF_004771005\",\n",
    "               \"Leptospira_inadai_GCF_000243675\",\n",
    "               \"Leptospira_jelokensis_GCF_004769775\",\n",
    "               \"Leptospira_johnsonii_GCF_003112675\",\n",
    "               \"Leptospira_kanakyensis_GCF_004769235\",\n",
    "               \"Leptospira_kemamanensis_GCF_004769665\",\n",
    "               \"Leptospira_kirschneri_GCF_000243695\",\n",
    "               \"Leptospira_kmetyi_GCF_000243735\",\n",
    "               \"Leptospira_kobayashii_GCF_003114835\",\n",
    "               \"Leptospira_koniamboensis_GCF_004769555\",\n",
    "               \"Leptospira_langatensis_GCF_004770615\",\n",
    "               \"Leptospira_levettii_GCF_002812085\",\n",
    "               \"Leptospira_licerasiae_GCF_000244755\",\n",
    "               \"Leptospira_macculloughii_GCF_002811975\",\n",
    "               \"Leptospira_mayottensis_GCF_000306675\",\n",
    "               \"Leptospira_meyeri_GCF_000304275\",\n",
    "               \"Leptospira_montravelensis_GCF_004769455\",\n",
    "               \"Leptospira_mtsangambouensis_GCF_004770475\",\n",
    "               \"Leptospira_neocaledonica_GCF_002812205\",\n",
    "               \"Leptospira_noguchii_GCF_000306255\",\n",
    "               \"Leptospira_noumeaensis_GCF_004770765\",\n",
    "               \"Leptospira_ognonensis_GCF_004770745\",\n",
    "               \"Leptospira_perdikensis_GCF_004769575\",\n",
    "               \"Leptospira_perolatii_GCF_002811875\",\n",
    "               \"Leptospira_putramalaysiae_GCF_004770035\",\n",
    "               \"Leptospira_ryugenii_GCF_003114855\",\n",
    "               \"Leptospira_saintgironsiae_GCF_002811765\",\n",
    "               \"Leptospira_santarosai_GCF_000313175\",\n",
    "               \"Leptospira_sarikeiensis_GCF_004769615\",\n",
    "               \"Leptospira_selangorensis_GCF_004769405\",\n",
    "               \"Leptospira_semungkisensis_GCF_004770055\",\n",
    "               \"Leptospira_sp_GCF_000347035\",\n",
    "               \"Leptospira_stimsonii_GCF_003545885\",\n",
    "               \"Leptospira_terpstrae_GCF_000332495\",\n",
    "               \"Leptospira_tipperaryensis_GCF_001729245\",\n",
    "               \"Leptospira_vanthielii_GCF_000332455\",\n",
    "               \"Leptospira_venezuelensis_GCF_002150035\",\n",
    "               \"Leptospira_weilii_GCF_000244815\",\n",
    "               \"Leptospira_wolbachii_GCF_000332515\",\n",
    "               \"Leptospira_wolffii_GCF_000306115\",\n",
    "               \"Leptospira_yanagawae_GCF_000332475\",\n",
    "               \"Leptospira_yasudae_GCF_003545925\" ]\n",
    "                    \n",
    "reference_taxon = \"Leptospira_interrogans_GCF_000092565\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811985.1_ASM281198v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1422418 bytes  948306.67 bytes/sec\n",
      "total size is 1421960  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000243815.3_gls454062v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1245686 bytes  830485.33 bytes/sec\n",
      "total size is 1245267  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000347175.1_gls454194v01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1310884 bytes  873950.67 bytes/sec\n",
      "total size is 1310457  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770105.1_ASM477010v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1266828 bytes  506748.00 bytes/sec\n",
      "total size is 1266410  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770905.1_ASM477090v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1188743 bytes  792523.33 bytes/sec\n",
      "total size is 1188341  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811925.1_ASM281192v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1284483 bytes  856350.00 bytes/sec\n",
      "total size is 1284057  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017685.1_ASM1768v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1162081 bytes  774748.67 bytes/sec\n",
      "total size is 1161689  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000013945.1_ASM1394v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1143180 bytes  762148.00 bytes/sec\n",
      "total size is 1142796  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769285.1_ASM476928v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1248227 bytes  832179.33 bytes/sec\n",
      "total size is 1247809  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770625.1_ASM477062v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1213489 bytes  809020.67 bytes/sec\n",
      "total size is 1213079  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002812125.1_ASM281212v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1213988 bytes  809353.33 bytes/sec\n",
      "total size is 1213578  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000243715.3_gls454050v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1306348 bytes  870926.67 bytes/sec\n",
      "total size is 1305921  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770265.1_ASM477026v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1180525 bytes  787044.67 bytes/sec\n",
      "total size is 1180123  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770135.1_ASM477013v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1263524 bytes  842377.33 bytes/sec\n",
      "total size is 1263106  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770895.1_ASM477089v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1217354 bytes  811597.33 bytes/sec\n",
      "total size is 1216944  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003114815.1_ASM311481v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1233831 bytes  822582.00 bytes/sec\n",
      "total size is 1233421  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811955.1_ASM281195v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1275125 bytes  850111.33 bytes/sec\n",
      "total size is 1274707  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000306235.2_gls454058v2.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1273907 bytes  849299.33 bytes/sec\n",
      "total size is 1273487  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769195.1_ASM476919v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1107337 bytes  130279.88 bytes/sec\n",
      "total size is 1106959  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004771275.1_ASM477127v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1111588 bytes  741086.67 bytes/sec\n",
      "total size is 1111210  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770155.1_ASM477015v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1256745 bytes  837858.00 bytes/sec\n",
      "total size is 1256327  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GCA_002812045.1_ASM281204v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1240941 bytes  107911.57 bytes/sec\n",
      "total size is 1240531  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811945.1_ASM281194v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1164568 bytes  776406.67 bytes/sec\n",
      "total size is 1164174  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811475.1_ASM281147v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1199790 bytes  799888.00 bytes/sec\n",
      "total size is 1199388  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770995.1_ASM477099v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1210897 bytes  807292.67 bytes/sec\n",
      "total size is 1210495  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004771005.1_ASM477100v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1251153 bytes  834130.00 bytes/sec\n",
      "total size is 1250735  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000243675.3_gls454047v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1320651 bytes  880462.00 bytes/sec\n",
      "total size is 1320216  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000092565.1_ASM9256v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1361938 bytes  907986.67 bytes/sec\n",
      "total size is 1361498  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769775.1_ASM476977v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1214005 bytes  809364.67 bytes/sec\n",
      "total size is 1213595  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003112675.1_ASM311267v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1203203 bytes  802163.33 bytes/sec\n",
      "total size is 1202801  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769235.1_ASM476923v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1220145 bytes  813458.00 bytes/sec\n",
      "total size is 1219735  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769665.1_ASM476966v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1113001 bytes  742028.67 bytes/sec\n",
      "total size is 1112623  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000243695.3_gls454049v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1284980 bytes  856681.33 bytes/sec\n",
      "total size is 1284553  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000243735.3_gls454052v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1286415 bytes  857638.00 bytes/sec\n",
      "total size is 1285987  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003114835.1_ASM311483v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1268806 bytes  845898.67 bytes/sec\n",
      "total size is 1268388  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769555.1_ASM476955v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1273296 bytes  848892.00 bytes/sec\n",
      "total size is 1272878  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770615.1_ASM477061v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1214391 bytes  809622.00 bytes/sec\n",
      "total size is 1213981  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002812085.1_ASM281208v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1141066 bytes  760738.67 bytes/sec\n",
      "total size is 1140680  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000244755.3_ASM24475v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1247850 bytes  831928.00 bytes/sec\n",
      "total size is 1247433  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811975.1_ASM281197v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2051170 bytes  1367474.67 bytes/sec\n",
      "total size is 2050560  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000306675.3_ASM30667v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1222557 bytes  815066.00 bytes/sec\n",
      "total size is 1222148  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000304275.1_gls454017v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1234380 bytes  822948.00 bytes/sec\n",
      "total size is 1233968  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769455.1_ASM476945v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1202634 bytes  801784.00 bytes/sec\n",
      "total size is 1202232  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_004770475.1_ASM477047v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1203143 bytes  802123.33 bytes/sec\n",
      "total size is 1202741  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002812205.1_ASM281220v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1246331 bytes  830915.33 bytes/sec\n",
      "total size is 1245913  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000306255.2_gls454059v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1378200 bytes  918828.00 bytes/sec\n",
      "total size is 1377749  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770765.1_ASM477076v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1210528 bytes  807046.67 bytes/sec\n",
      "total size is 1210126  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770745.1_ASM477074v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1191241 bytes  794188.67 bytes/sec\n",
      "total size is 1190839  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769575.1_ASM476957v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1183039 bytes  788720.67 bytes/sec\n",
      "total size is 1182637  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811875.1_ASM281187v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1189581 bytes  793082.00 bytes/sec\n",
      "total size is 1189179  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770035.1_ASM477003v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1467280 bytes  978214.67 bytes/sec\n",
      "total size is 1466814  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003114855.1_ASM311485v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1173617 bytes  782439.33 bytes/sec\n",
      "total size is 1173223  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002811765.1_ASM281176v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1206094 bytes  804090.67 bytes/sec\n",
      "total size is 1205692  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000313175.2_ASM31317v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1163695 bytes  775824.67 bytes/sec\n",
      "total size is 1163302  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_004769615.1_ASM476961v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1293778 bytes  862546.67 bytes/sec\n",
      "total size is 1293352  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004769405.1_ASM476940v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1237738 bytes  825186.67 bytes/sec\n",
      "total size is 1237328  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004770055.1_ASM477005v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1169389 bytes  779620.67 bytes/sec\n",
      "total size is 1168995  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000347035.1_gls454192v01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1239149 bytes  826127.33 bytes/sec\n",
      "total size is 1238738  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003545885.1_ASM354588v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1420180 bytes  946814.67 bytes/sec\n",
      "total size is 1419722  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000332495.2_gls454203v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1209529 bytes  806380.67 bytes/sec\n",
      "total size is 1209126  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001729245.1_ASM172924v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1343708 bytes  895833.33 bytes/sec\n",
      "total size is 1343274  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000332455.2_gls454199v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1255096 bytes  836758.67 bytes/sec\n",
      "total size is 1254677  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002150035.1_ASM215003v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1273277 bytes  848879.33 bytes/sec\n",
      "total size is 1272859  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000244815.3_gls454188v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1273266 bytes  848872.00 bytes/sec\n",
      "total size is 1272847  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000332515.2_gls454195v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1206267 bytes  804206.00 bytes/sec\n",
      "total size is 1205864  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000306115.2_gls454061v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1301152 bytes  867462.67 bytes/sec\n",
      "total size is 1300725  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000332475.2_gls454202v02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1195500 bytes  797028.00 bytes/sec\n",
      "total size is 1195097  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003545925.1_ASM354592v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1303185 bytes  868818.00 bytes/sec\n",
      "total size is 1302759  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484051 (\"sim_Leptospira_adleri_GCF_002811985\") has been submitted\n",
      "Your job 5484052 (\"sim_Leptospira_alexanderi_GCF_000243815\") has been submitted\n",
      "Your job 5484053 (\"sim_Leptospira_alstonii_GCF_000347175\") has been submitted\n",
      "Your job 5484054 (\"sim_Leptospira_andrefontaineae_GCF_004770105\") has been submitted\n",
      "Your job 5484055 (\"sim_Leptospira_bandrabouensis_GCF_004770905\") has been submitted\n",
      "Your job 5484056 (\"sim_Leptospira_barantonii_GCF_002811925\") has been submitted\n",
      "Your job 5484057 (\"sim_Leptospira_biflexa_GCF_000017685\") has been submitted\n",
      "Your job 5484058 (\"sim_Leptospira_borgpetersenii_GCF_000013945\") has been submitted\n",
      "Your job 5484059 (\"sim_Leptospira_bourretii_GCF_004769285\") has been submitted\n",
      "Your job 5484060 (\"sim_Leptospira_bouyouniensis_GCF_004770625\") has been submitted\n",
      "Your job 5484061 (\"sim_Leptospira_brenneri_GCF_002812125\") has been submitted\n",
      "Your job 5484062 (\"sim_Leptospira_broomii_GCF_000243715\") has been submitted\n",
      "Your job 5484063 (\"sim_Leptospira_congkakensis_GCF_004770265\") has been submitted\n",
      "Your job 5484064 (\"sim_Leptospira_dzianensis_GCF_004770135\") has been submitted\n",
      "Your job 5484065 (\"sim_Leptospira_dzoumogneensis_GCF_004770895\") has been submitted\n",
      "Your job 5484066 (\"sim_Leptospira_ellinghausenii_GCF_003114815\") has been submitted\n",
      "Your job 5484067 (\"sim_Leptospira_elllisii_GCF_002811955\") has been submitted\n",
      "Your job 5484068 (\"sim_Leptospira_fainei_GCF_000306235\") has been submitted\n",
      "Your job 5484069 (\"sim_Leptospira_fletcheri_GCF_004769195\") has been submitted\n",
      "Your job 5484070 (\"sim_Leptospira_fluminis_GCF_004771275\") has been submitted\n",
      "Your job 5484071 (\"sim_Leptospira_gomenensis_GCF_004770155\") has been submitted\n",
      "Your job 5484072 (\"sim_Leptospira_haakeii_GCF_002812045\") has been submitted\n",
      "Your job 5484073 (\"sim_Leptospira_harrisiae_GCF_002811945\") has been submitted\n",
      "Your job 5484074 (\"sim_Leptospira_hartskeerlii_GCF_002811475\") has been submitted\n",
      "Your job 5484075 (\"sim_Leptospira_idonii_GCF_004770995\") has been submitted\n",
      "Your job 5484076 (\"sim_Leptospira_ilyithenensis_GCF_004771005\") has been submitted\n",
      "Your job 5484077 (\"sim_Leptospira_inadai_GCF_000243675\") has been submitted\n",
      "Your job 5484078 (\"sim_Leptospira_jelokensis_GCF_004769775\") has been submitted\n",
      "Your job 5484079 (\"sim_Leptospira_johnsonii_GCF_003112675\") has been submitted\n",
      "Your job 5484080 (\"sim_Leptospira_kanakyensis_GCF_004769235\") has been submitted\n",
      "Your job 5484081 (\"sim_Leptospira_kemamanensis_GCF_004769665\") has been submitted\n",
      "Your job 5484082 (\"sim_Leptospira_kirschneri_GCF_000243695\") has been submitted\n",
      "Your job 5484083 (\"sim_Leptospira_kmetyi_GCF_000243735\") has been submitted\n",
      "Your job 5484084 (\"sim_Leptospira_kobayashii_GCF_003114835\") has been submitted\n",
      "Your job 5484085 (\"sim_Leptospira_koniamboensis_GCF_004769555\") has been submitted\n",
      "Your job 5484086 (\"sim_Leptospira_langatensis_GCF_004770615\") has been submitted\n",
      "Your job 5484087 (\"sim_Leptospira_levettii_GCF_002812085\") has been submitted\n",
      "Your job 5484088 (\"sim_Leptospira_licerasiae_GCF_000244755\") has been submitted\n",
      "Your job 5484089 (\"sim_Leptospira_macculloughii_GCF_002811975\") has been submitted\n",
      "Your job 5484090 (\"sim_Leptospira_mayottensis_GCF_000306675\") has been submitted\n",
      "Your job 5484091 (\"sim_Leptospira_meyeri_GCF_000304275\") has been submitted\n",
      "Your job 5484092 (\"sim_Leptospira_montravelensis_GCF_004769455\") has been submitted\n",
      "Your job 5484093 (\"sim_Leptospira_mtsangambouensis_GCF_004770475\") has been submitted\n",
      "Your job 5484094 (\"sim_Leptospira_neocaledonica_GCF_002812205\") has been submitted\n",
      "Your job 5484095 (\"sim_Leptospira_noguchii_GCF_000306255\") has been submitted\n",
      "Your job 5484096 (\"sim_Leptospira_noumeaensis_GCF_004770765\") has been submitted\n",
      "Your job 5484097 (\"sim_Leptospira_ognonensis_GCF_004770745\") has been submitted\n",
      "Your job 5484098 (\"sim_Leptospira_perdikensis_GCF_004769575\") has been submitted\n",
      "Your job 5484099 (\"sim_Leptospira_perolatii_GCF_002811875\") has been submitted\n",
      "Your job 5484100 (\"sim_Leptospira_putramalaysiae_GCF_004770035\") has been submitted\n",
      "Your job 5484101 (\"sim_Leptospira_ryugenii_GCF_003114855\") has been submitted\n",
      "Your job 5484102 (\"sim_Leptospira_saintgironsiae_GCF_002811765\") has been submitted\n",
      "Your job 5484103 (\"sim_Leptospira_santarosai_GCF_000313175\") has been submitted\n",
      "Your job 5484104 (\"sim_Leptospira_sarikeiensis_GCF_004769615\") has been submitted\n",
      "Your job 5484105 (\"sim_Leptospira_selangorensis_GCF_004769405\") has been submitted\n",
      "Your job 5484106 (\"sim_Leptospira_semungkisensis_GCF_004770055\") has been submitted\n",
      "Your job 5484107 (\"sim_Leptospira_sp_GCF_000347035\") has been submitted\n",
      "Your job 5484108 (\"sim_Leptospira_stimsonii_GCF_003545885\") has been submitted\n",
      "Your job 5484109 (\"sim_Leptospira_terpstrae_GCF_000332495\") has been submitted\n",
      "Your job 5484110 (\"sim_Leptospira_tipperaryensis_GCF_001729245\") has been submitted\n",
      "Your job 5484111 (\"sim_Leptospira_vanthielii_GCF_000332455\") has been submitted\n",
      "Your job 5484112 (\"sim_Leptospira_venezuelensis_GCF_002150035\") has been submitted\n",
      "Your job 5484113 (\"sim_Leptospira_weilii_GCF_000244815\") has been submitted\n",
      "Your job 5484114 (\"sim_Leptospira_wolbachii_GCF_000332515\") has been submitted\n",
      "Your job 5484115 (\"sim_Leptospira_wolffii_GCF_000306115\") has been submitted\n",
      "Your job 5484116 (\"sim_Leptospira_yanagawae_GCF_000332475\") has been submitted\n",
      "Your job 5484117 (\"sim_Leptospira_yasudae_GCF_003545925\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/leptospira/cleaned_genomes/Leptospira_interrogans_GCF_000092565_formatted.fas path=results/phyluce/leptospira/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484118 (\"map_Leptospira_adleri_GCF_002811985\") has been submitted\n",
      "Your job 5484119 (\"map_Leptospira_alexanderi_GCF_000243815\") has been submitted\n",
      "Your job 5484120 (\"map_Leptospira_alstonii_GCF_000347175\") has been submitted\n",
      "Your job 5484121 (\"map_Leptospira_andrefontaineae_GCF_004770105\") has been submitted\n",
      "Your job 5484122 (\"map_Leptospira_bandrabouensis_GCF_004770905\") has been submitted\n",
      "Your job 5484123 (\"map_Leptospira_barantonii_GCF_002811925\") has been submitted\n",
      "Your job 5484124 (\"map_Leptospira_biflexa_GCF_000017685\") has been submitted\n",
      "Your job 5484125 (\"map_Leptospira_borgpetersenii_GCF_000013945\") has been submitted\n",
      "Your job 5484126 (\"map_Leptospira_bourretii_GCF_004769285\") has been submitted\n",
      "Your job 5484127 (\"map_Leptospira_bouyouniensis_GCF_004770625\") has been submitted\n",
      "Your job 5484128 (\"map_Leptospira_brenneri_GCF_002812125\") has been submitted\n",
      "Your job 5484129 (\"map_Leptospira_broomii_GCF_000243715\") has been submitted\n",
      "Your job 5484130 (\"map_Leptospira_congkakensis_GCF_004770265\") has been submitted\n",
      "Your job 5484131 (\"map_Leptospira_dzianensis_GCF_004770135\") has been submitted\n",
      "Your job 5484132 (\"map_Leptospira_dzoumogneensis_GCF_004770895\") has been submitted\n",
      "Your job 5484133 (\"map_Leptospira_ellinghausenii_GCF_003114815\") has been submitted\n",
      "Your job 5484134 (\"map_Leptospira_elllisii_GCF_002811955\") has been submitted\n",
      "Your job 5484135 (\"map_Leptospira_fainei_GCF_000306235\") has been submitted\n",
      "Your job 5484136 (\"map_Leptospira_fletcheri_GCF_004769195\") has been submitted\n",
      "Your job 5484137 (\"map_Leptospira_fluminis_GCF_004771275\") has been submitted\n",
      "Your job 5484138 (\"map_Leptospira_gomenensis_GCF_004770155\") has been submitted\n",
      "Your job 5484139 (\"map_Leptospira_haakeii_GCF_002812045\") has been submitted\n",
      "Your job 5484140 (\"map_Leptospira_harrisiae_GCF_002811945\") has been submitted\n",
      "Your job 5484141 (\"map_Leptospira_hartskeerlii_GCF_002811475\") has been submitted\n",
      "Your job 5484142 (\"map_Leptospira_idonii_GCF_004770995\") has been submitted\n",
      "Your job 5484143 (\"map_Leptospira_ilyithenensis_GCF_004771005\") has been submitted\n",
      "Your job 5484144 (\"map_Leptospira_inadai_GCF_000243675\") has been submitted\n",
      "Your job 5484145 (\"map_Leptospira_jelokensis_GCF_004769775\") has been submitted\n",
      "Your job 5484146 (\"map_Leptospira_johnsonii_GCF_003112675\") has been submitted\n",
      "Your job 5484147 (\"map_Leptospira_kanakyensis_GCF_004769235\") has been submitted\n",
      "Your job 5484148 (\"map_Leptospira_kemamanensis_GCF_004769665\") has been submitted\n",
      "Your job 5484149 (\"map_Leptospira_kirschneri_GCF_000243695\") has been submitted\n",
      "Your job 5484150 (\"map_Leptospira_kmetyi_GCF_000243735\") has been submitted\n",
      "Your job 5484151 (\"map_Leptospira_kobayashii_GCF_003114835\") has been submitted\n",
      "Your job 5484152 (\"map_Leptospira_koniamboensis_GCF_004769555\") has been submitted\n",
      "Your job 5484153 (\"map_Leptospira_langatensis_GCF_004770615\") has been submitted\n",
      "Your job 5484154 (\"map_Leptospira_levettii_GCF_002812085\") has been submitted\n",
      "Your job 5484155 (\"map_Leptospira_licerasiae_GCF_000244755\") has been submitted\n",
      "Your job 5484156 (\"map_Leptospira_macculloughii_GCF_002811975\") has been submitted\n",
      "Your job 5484157 (\"map_Leptospira_mayottensis_GCF_000306675\") has been submitted\n",
      "Your job 5484158 (\"map_Leptospira_meyeri_GCF_000304275\") has been submitted\n",
      "Your job 5484159 (\"map_Leptospira_montravelensis_GCF_004769455\") has been submitted\n",
      "Your job 5484160 (\"map_Leptospira_mtsangambouensis_GCF_004770475\") has been submitted\n",
      "Your job 5484161 (\"map_Leptospira_neocaledonica_GCF_002812205\") has been submitted\n",
      "Your job 5484162 (\"map_Leptospira_noguchii_GCF_000306255\") has been submitted\n",
      "Your job 5484163 (\"map_Leptospira_noumeaensis_GCF_004770765\") has been submitted\n",
      "Your job 5484164 (\"map_Leptospira_ognonensis_GCF_004770745\") has been submitted\n",
      "Your job 5484165 (\"map_Leptospira_perdikensis_GCF_004769575\") has been submitted\n",
      "Your job 5484166 (\"map_Leptospira_perolatii_GCF_002811875\") has been submitted\n",
      "Your job 5484167 (\"map_Leptospira_putramalaysiae_GCF_004770035\") has been submitted\n",
      "Your job 5484168 (\"map_Leptospira_ryugenii_GCF_003114855\") has been submitted\n",
      "Your job 5484169 (\"map_Leptospira_saintgironsiae_GCF_002811765\") has been submitted\n",
      "Your job 5484170 (\"map_Leptospira_santarosai_GCF_000313175\") has been submitted\n",
      "Your job 5484171 (\"map_Leptospira_sarikeiensis_GCF_004769615\") has been submitted\n",
      "Your job 5484172 (\"map_Leptospira_selangorensis_GCF_004769405\") has been submitted\n",
      "Your job 5484173 (\"map_Leptospira_semungkisensis_GCF_004770055\") has been submitted\n",
      "Your job 5484174 (\"map_Leptospira_sp_GCF_000347035\") has been submitted\n",
      "Your job 5484175 (\"map_Leptospira_stimsonii_GCF_003545885\") has been submitted\n",
      "Your job 5484176 (\"map_Leptospira_terpstrae_GCF_000332495\") has been submitted\n",
      "Your job 5484177 (\"map_Leptospira_tipperaryensis_GCF_001729245\") has been submitted\n",
      "Your job 5484178 (\"map_Leptospira_vanthielii_GCF_000332455\") has been submitted\n",
      "Your job 5484179 (\"map_Leptospira_venezuelensis_GCF_002150035\") has been submitted\n",
      "Your job 5484180 (\"map_Leptospira_weilii_GCF_000244815\") has been submitted\n",
      "Your job 5484181 (\"map_Leptospira_wolbachii_GCF_000332515\") has been submitted\n",
      "Your job 5484182 (\"map_Leptospira_wolffii_GCF_000306115\") has been submitted\n",
      "Your job 5484183 (\"map_Leptospira_yanagawae_GCF_000332475\") has been submitted\n",
      "Your job 5484184 (\"map_Leptospira_yasudae_GCF_003545925\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484185 (\"merge_Leptospira_adleri_GCF_002811985\") has been submitted\n",
      "Your job 5484186 (\"merge_Leptospira_alexanderi_GCF_000243815\") has been submitted\n",
      "Your job 5484187 (\"merge_Leptospira_alstonii_GCF_000347175\") has been submitted\n",
      "Your job 5484188 (\"merge_Leptospira_andrefontaineae_GCF_004770105\") has been submitted\n",
      "Your job 5484189 (\"merge_Leptospira_bandrabouensis_GCF_004770905\") has been submitted\n",
      "Your job 5484190 (\"merge_Leptospira_barantonii_GCF_002811925\") has been submitted\n",
      "Your job 5484191 (\"merge_Leptospira_biflexa_GCF_000017685\") has been submitted\n",
      "Your job 5484192 (\"merge_Leptospira_borgpetersenii_GCF_000013945\") has been submitted\n",
      "Your job 5484193 (\"merge_Leptospira_bourretii_GCF_004769285\") has been submitted\n",
      "Your job 5484194 (\"merge_Leptospira_bouyouniensis_GCF_004770625\") has been submitted\n",
      "Your job 5484195 (\"merge_Leptospira_brenneri_GCF_002812125\") has been submitted\n",
      "Your job 5484196 (\"merge_Leptospira_broomii_GCF_000243715\") has been submitted\n",
      "Your job 5484197 (\"merge_Leptospira_congkakensis_GCF_004770265\") has been submitted\n",
      "Your job 5484198 (\"merge_Leptospira_dzianensis_GCF_004770135\") has been submitted\n",
      "Your job 5484199 (\"merge_Leptospira_dzoumogneensis_GCF_004770895\") has been submitted\n",
      "Your job 5484200 (\"merge_Leptospira_ellinghausenii_GCF_003114815\") has been submitted\n",
      "Your job 5484201 (\"merge_Leptospira_elllisii_GCF_002811955\") has been submitted\n",
      "Your job 5484202 (\"merge_Leptospira_fainei_GCF_000306235\") has been submitted\n",
      "Your job 5484203 (\"merge_Leptospira_fletcheri_GCF_004769195\") has been submitted\n",
      "Your job 5484204 (\"merge_Leptospira_fluminis_GCF_004771275\") has been submitted\n",
      "Your job 5484205 (\"merge_Leptospira_gomenensis_GCF_004770155\") has been submitted\n",
      "Your job 5484206 (\"merge_Leptospira_haakeii_GCF_002812045\") has been submitted\n",
      "Your job 5484207 (\"merge_Leptospira_harrisiae_GCF_002811945\") has been submitted\n",
      "Your job 5484208 (\"merge_Leptospira_hartskeerlii_GCF_002811475\") has been submitted\n",
      "Your job 5484209 (\"merge_Leptospira_idonii_GCF_004770995\") has been submitted\n",
      "Your job 5484210 (\"merge_Leptospira_ilyithenensis_GCF_004771005\") has been submitted\n",
      "Your job 5484211 (\"merge_Leptospira_inadai_GCF_000243675\") has been submitted\n",
      "Your job 5484212 (\"merge_Leptospira_jelokensis_GCF_004769775\") has been submitted\n",
      "Your job 5484213 (\"merge_Leptospira_johnsonii_GCF_003112675\") has been submitted\n",
      "Your job 5484214 (\"merge_Leptospira_kanakyensis_GCF_004769235\") has been submitted\n",
      "Your job 5484215 (\"merge_Leptospira_kemamanensis_GCF_004769665\") has been submitted\n",
      "Your job 5484216 (\"merge_Leptospira_kirschneri_GCF_000243695\") has been submitted\n",
      "Your job 5484217 (\"merge_Leptospira_kmetyi_GCF_000243735\") has been submitted\n",
      "Your job 5484218 (\"merge_Leptospira_kobayashii_GCF_003114835\") has been submitted\n",
      "Your job 5484219 (\"merge_Leptospira_koniamboensis_GCF_004769555\") has been submitted\n",
      "Your job 5484220 (\"merge_Leptospira_langatensis_GCF_004770615\") has been submitted\n",
      "Your job 5484221 (\"merge_Leptospira_levettii_GCF_002812085\") has been submitted\n",
      "Your job 5484222 (\"merge_Leptospira_licerasiae_GCF_000244755\") has been submitted\n",
      "Your job 5484223 (\"merge_Leptospira_macculloughii_GCF_002811975\") has been submitted\n",
      "Your job 5484224 (\"merge_Leptospira_mayottensis_GCF_000306675\") has been submitted\n",
      "Your job 5484225 (\"merge_Leptospira_meyeri_GCF_000304275\") has been submitted\n",
      "Your job 5484226 (\"merge_Leptospira_montravelensis_GCF_004769455\") has been submitted\n",
      "Your job 5484227 (\"merge_Leptospira_mtsangambouensis_GCF_004770475\") has been submitted\n",
      "Your job 5484228 (\"merge_Leptospira_neocaledonica_GCF_002812205\") has been submitted\n",
      "Your job 5484229 (\"merge_Leptospira_noguchii_GCF_000306255\") has been submitted\n",
      "Your job 5484230 (\"merge_Leptospira_noumeaensis_GCF_004770765\") has been submitted\n",
      "Your job 5484231 (\"merge_Leptospira_ognonensis_GCF_004770745\") has been submitted\n",
      "Your job 5484232 (\"merge_Leptospira_perdikensis_GCF_004769575\") has been submitted\n",
      "Your job 5484233 (\"merge_Leptospira_perolatii_GCF_002811875\") has been submitted\n",
      "Your job 5484234 (\"merge_Leptospira_putramalaysiae_GCF_004770035\") has been submitted\n",
      "Your job 5484235 (\"merge_Leptospira_ryugenii_GCF_003114855\") has been submitted\n",
      "Your job 5484236 (\"merge_Leptospira_saintgironsiae_GCF_002811765\") has been submitted\n",
      "Your job 5484237 (\"merge_Leptospira_santarosai_GCF_000313175\") has been submitted\n",
      "Your job 5484238 (\"merge_Leptospira_sarikeiensis_GCF_004769615\") has been submitted\n",
      "Your job 5484239 (\"merge_Leptospira_selangorensis_GCF_004769405\") has been submitted\n",
      "Your job 5484240 (\"merge_Leptospira_semungkisensis_GCF_004770055\") has been submitted\n",
      "Your job 5484241 (\"merge_Leptospira_sp_GCF_000347035\") has been submitted\n",
      "Your job 5484242 (\"merge_Leptospira_stimsonii_GCF_003545885\") has been submitted\n",
      "Your job 5484243 (\"merge_Leptospira_terpstrae_GCF_000332495\") has been submitted\n",
      "Your job 5484244 (\"merge_Leptospira_tipperaryensis_GCF_001729245\") has been submitted\n",
      "Your job 5484245 (\"merge_Leptospira_vanthielii_GCF_000332455\") has been submitted\n",
      "Your job 5484246 (\"merge_Leptospira_venezuelensis_GCF_002150035\") has been submitted\n",
      "Your job 5484247 (\"merge_Leptospira_weilii_GCF_000244815\") has been submitted\n",
      "Your job 5484248 (\"merge_Leptospira_wolbachii_GCF_000332515\") has been submitted\n",
      "Your job 5484249 (\"merge_Leptospira_wolffii_GCF_000306115\") has been submitted\n",
      "Your job 5484250 (\"merge_Leptospira_yanagawae_GCF_000332475\") has been submitted\n",
      "Your job 5484251 (\"merge_Leptospira_yasudae_GCF_003545925\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 913 sequences from Leptospira_adleri_GCF_002811985_merged.bed.  Filtered 630 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 283.\n",
      "Screened 2402 sequences from Leptospira_alexanderi_GCF_000243815_merged.bed.  Filtered 1640 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 762.\n",
      "Screened 2609 sequences from Leptospira_alstonii_GCF_000347175_merged.bed.  Filtered 1806 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 803.\n",
      "Screened 50 sequences from Leptospira_andrefontaineae_GCF_004770105_merged.bed.  Filtered 44 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 24 sequences from Leptospira_bandrabouensis_GCF_004770905_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 1492 sequences from Leptospira_barantonii_GCF_002811925_merged.bed.  Filtered 1000 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 492.\n",
      "Screened 21 sequences from Leptospira_biflexa_GCF_000017685_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 2155 sequences from Leptospira_borgpetersenii_GCF_000013945_merged.bed.  Filtered 1510 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 645.\n",
      "Screened 19 sequences from Leptospira_bourretii_GCF_004769285_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 23 sequences from Leptospira_bouyouniensis_GCF_004770625_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 21 sequences from Leptospira_brenneri_GCF_002812125_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 36 sequences from Leptospira_broomii_GCF_000243715_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 20 sequences from Leptospira_congkakensis_GCF_004770265_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 1173 sequences from Leptospira_dzianensis_GCF_004770135_merged.bed.  Filtered 761 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 412.\n",
      "Screened 53 sequences from Leptospira_dzoumogneensis_GCF_004770895_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 22 sequences from Leptospira_ellinghausenii_GCF_003114815_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 639 sequences from Leptospira_elllisii_GCF_002811955_merged.bed.  Filtered 412 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 227.\n",
      "Screened 40 sequences from Leptospira_fainei_GCF_000306235_merged.bed.  Filtered 34 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 35 sequences from Leptospira_fletcheri_GCF_004769195_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 40 sequences from Leptospira_fluminis_GCF_004771275_merged.bed.  Filtered 33 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 679 sequences from Leptospira_gomenensis_GCF_004770155_merged.bed.  Filtered 455 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 224.\n",
      "Screened 45 sequences from Leptospira_haakeii_GCF_002812045_merged.bed.  Filtered 42 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 24 sequences from Leptospira_harrisiae_GCF_002811945_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 48 sequences from Leptospira_hartskeerlii_GCF_002811475_merged.bed.  Filtered 42 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 30 sequences from Leptospira_idonii_GCF_004770995_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 28 sequences from Leptospira_ilyithenensis_GCF_004771005_merged.bed.  Filtered 19 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 34 sequences from Leptospira_inadai_GCF_000243675_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 20 sequences from Leptospira_jelokensis_GCF_004769775_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 45 sequences from Leptospira_johnsonii_GCF_003112675_merged.bed.  Filtered 39 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 25 sequences from Leptospira_kanakyensis_GCF_004769235_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 23 sequences from Leptospira_kemamanensis_GCF_004769665_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 4862 sequences from Leptospira_kirschneri_GCF_000243695_merged.bed.  Filtered 1114 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3748.\n",
      "Screened 1428 sequences from Leptospira_kmetyi_GCF_000243735_merged.bed.  Filtered 934 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 494.\n",
      "Screened 28 sequences from Leptospira_kobayashii_GCF_003114835_merged.bed.  Filtered 23 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 39 sequences from Leptospira_koniamboensis_GCF_004769555_merged.bed.  Filtered 33 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 34 sequences from Leptospira_langatensis_GCF_004770615_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 19 sequences from Leptospira_levettii_GCF_002812085_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 48 sequences from Leptospira_licerasiae_GCF_000244755_merged.bed.  Filtered 41 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 27 sequences from Leptospira_macculloughii_GCF_002811975_merged.bed.  Filtered 21 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 2225 sequences from Leptospira_mayottensis_GCF_000306675_merged.bed.  Filtered 1531 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 694.\n",
      "Screened 25 sequences from Leptospira_meyeri_GCF_000304275_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 19 sequences from Leptospira_montravelensis_GCF_004769455_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 24 sequences from Leptospira_mtsangambouensis_GCF_004770475_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 51 sequences from Leptospira_neocaledonica_GCF_002812205_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 6267 sequences from Leptospira_noguchii_GCF_000306255_merged.bed.  Filtered 1625 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4642.\n",
      "Screened 22 sequences from Leptospira_noumeaensis_GCF_004770765_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 18 sequences from Leptospira_ognonensis_GCF_004770745_merged.bed.  Filtered 11 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 21 sequences from Leptospira_perdikensis_GCF_004769575_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 40 sequences from Leptospira_perolatii_GCF_002811875_merged.bed.  Filtered 36 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 886 sequences from Leptospira_putramalaysiae_GCF_004770035_merged.bed.  Filtered 607 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 279.\n",
      "Screened 16 sequences from Leptospira_ryugenii_GCF_003114855_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 52 sequences from Leptospira_saintgironsiae_GCF_002811765_merged.bed.  Filtered 47 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 2037 sequences from Leptospira_santarosai_GCF_000313175_merged.bed.  Filtered 1443 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 594.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 55 sequences from Leptospira_sarikeiensis_GCF_004769615_merged.bed.  Filtered 49 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 49 sequences from Leptospira_selangorensis_GCF_004769405_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 44 sequences from Leptospira_semungkisensis_GCF_004770055_merged.bed.  Filtered 37 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 41 sequences from Leptospira_sp_GCF_000347035_merged.bed.  Filtered 34 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 897 sequences from Leptospira_stimsonii_GCF_003545885_merged.bed.  Filtered 608 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 289.\n",
      "Screened 40 sequences from Leptospira_terpstrae_GCF_000332495_merged.bed.  Filtered 20 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 20.\n",
      "Screened 1035 sequences from Leptospira_tipperaryensis_GCF_001729245_merged.bed.  Filtered 737 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 298.\n",
      "Screened 24 sequences from Leptospira_vanthielii_GCF_000332455_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 40 sequences from Leptospira_venezuelensis_GCF_002150035_merged.bed.  Filtered 35 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 2215 sequences from Leptospira_weilii_GCF_000244815_merged.bed.  Filtered 1539 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 676.\n",
      "Screened 23 sequences from Leptospira_wolbachii_GCF_000332515_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 53 sequences from Leptospira_wolffii_GCF_000306115_merged.bed.  Filtered 40 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 47 sequences from Leptospira_yanagawae_GCF_000332475_merged.bed.  Filtered 29 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 18.\n",
      "Screened 1227 sequences from Leptospira_yasudae_GCF_003545925_merged.bed.  Filtered 815 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 412.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/leptospira/initial_intervals/Leptospira_yasudae_GCF_003545925_merged.bed --twobit results/phyluce/leptospira/cleaned_genomes/Leptospira_interrogans_GCF_000092565_formatted.2bit --output results/phyluce/leptospira/initial_intervals/Leptospira_yasudae_GCF_003545925_stripped.bed;\n",
      "leptospira_adleri_gcf_002811985.\n",
      "leptospira_alexanderi_gcf_000243815.\n",
      "leptospira_alstonii_gcf_000347175.\n",
      "leptospira_andrefontaineae_gcf_004770105.\n",
      "leptospira_bandrabouensis_gcf_004770905.\n",
      "leptospira_barantonii_gcf_002811925.\n",
      "leptospira_biflexa_gcf_000017685.\n",
      "leptospira_borgpetersenii_gcf_000013945.\n",
      "leptospira_bourretii_gcf_004769285.\n",
      "leptospira_bouyouniensis_gcf_004770625.\n",
      "leptospira_brenneri_gcf_002812125.\n",
      "leptospira_broomii_gcf_000243715.\n",
      "leptospira_congkakensis_gcf_004770265.\n",
      "leptospira_dzianensis_gcf_004770135.\n",
      "leptospira_dzoumogneensis_gcf_004770895.\n",
      "leptospira_ellinghausenii_gcf_003114815.\n",
      "leptospira_elllisii_gcf_002811955.\n",
      "leptospira_fainei_gcf_000306235.\n",
      "leptospira_fletcheri_gcf_004769195.\n",
      "leptospira_fluminis_gcf_004771275.\n",
      "leptospira_gomenensis_gcf_004770155.\n",
      "leptospira_haakeii_gcf_002812045.\n",
      "leptospira_harrisiae_gcf_002811945.\n",
      "leptospira_hartskeerlii_gcf_002811475.\n",
      "leptospira_idonii_gcf_004770995.\n",
      "leptospira_ilyithenensis_gcf_004771005.\n",
      "leptospira_inadai_gcf_000243675.\n",
      "leptospira_jelokensis_gcf_004769775.\n",
      "leptospira_johnsonii_gcf_003112675.\n",
      "leptospira_kanakyensis_gcf_004769235.\n",
      "leptospira_kemamanensis_gcf_004769665.\n",
      "leptospira_kirschneri_gcf_000243695....\n",
      "leptospira_kmetyi_gcf_000243735.\n",
      "leptospira_kobayashii_gcf_003114835.\n",
      "leptospira_koniamboensis_gcf_004769555.\n",
      "leptospira_langatensis_gcf_004770615.\n",
      "leptospira_levettii_gcf_002812085.\n",
      "leptospira_licerasiae_gcf_000244755.\n",
      "leptospira_macculloughii_gcf_002811975.\n",
      "leptospira_mayottensis_gcf_000306675.\n",
      "leptospira_meyeri_gcf_000304275.\n",
      "leptospira_montravelensis_gcf_004769455.\n",
      "leptospira_mtsangambouensis_gcf_004770475.\n",
      "leptospira_neocaledonica_gcf_002812205.\n",
      "leptospira_noguchii_gcf_000306255.....\n",
      "leptospira_noumeaensis_gcf_004770765.\n",
      "leptospira_ognonensis_gcf_004770745.\n",
      "leptospira_perdikensis_gcf_004769575.\n",
      "leptospira_perolatii_gcf_002811875.\n",
      "leptospira_putramalaysiae_gcf_004770035.\n",
      "leptospira_ryugenii_gcf_003114855.\n",
      "leptospira_saintgironsiae_gcf_002811765.\n",
      "leptospira_santarosai_gcf_000313175.\n",
      "leptospira_sarikeiensis_gcf_004769615.\n",
      "leptospira_selangorensis_gcf_004769405.\n",
      "leptospira_semungkisensis_gcf_004770055.\n",
      "leptospira_sp_gcf_000347035.\n",
      "leptospira_stimsonii_gcf_003545885.\n",
      "leptospira_terpstrae_gcf_000332495.\n",
      "leptospira_tipperaryensis_gcf_001729245.\n",
      "leptospira_vanthielii_gcf_000332455.\n",
      "leptospira_venezuelensis_gcf_002150035.\n",
      "leptospira_weilii_gcf_000244815.\n",
      "leptospira_wolbachii_gcf_000332515.\n",
      "leptospira_wolffii_gcf_000306115.\n",
      "leptospira_yanagawae_gcf_000332475.\n",
      "leptospira_yasudae_gcf_003545925.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/leptospira/initial_intervals/Leptospira_yasudae_GCF_003545925_merged.bed --twobit results/phyluce/leptospira/cleaned_genomes/Leptospira_interrogans_GCF_000092565_formatted.2bit --output results/phyluce/leptospira/initial_intervals/Leptospira_yasudae_GCF_003545925_stripped.bed;\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 0 taxa:\t5,335.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 1 taxa:\t5,335.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 2 taxa:\t3,863.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 3 taxa:\t1,553.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 4 taxa:\t1,072.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 5 taxa:\t812.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 6 taxa:\t660.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 7 taxa:\t570.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 8 taxa:\t466.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 9 taxa:\t400.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 10 taxa:\t356.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 11 taxa:\t320.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 12 taxa:\t281.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 13 taxa:\t240.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 14 taxa:\t208.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 15 taxa:\t182.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 16 taxa:\t158.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 17 taxa:\t127.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 18 taxa:\t87.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 19 taxa:\t10.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 20 taxa:\t7.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 21 taxa:\t6.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 22 taxa:\t6.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 23 taxa:\t6.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 24 taxa:\t4.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 25 taxa:\t4.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 26 taxa:\t4.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 27 taxa:\t3.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 28 taxa:\t3.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 29 taxa:\t3.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 30 taxa:\t3.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 31 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 32 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 33 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 34 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 35 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 36 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 37 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 38 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 39 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 40 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 41 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 42 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 43 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 44 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 45 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 46 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 47 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 48 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 49 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 50 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 51 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 52 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 53 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 54 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 55 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 56 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 57 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 58 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 59 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 60 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 61 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 62 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 63 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 64 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 65 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 66 taxa:\t2.0\n",
      "Loci shared by Leptospira_interrogans_GCF_000092565 + 67 taxa:\t2.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/leptospira/initial_intervals/leptospira-to-Leptospira_interrogans_GCF_000092565.sqlite --base-taxon Leptospira_interrogans_GCF_000092565 --output results/phyluce/leptospira/initial_intervals/Leptospira_interrogans_GCF_000092565_+10.bed --specific-counts 10;\n",
      "Counter({'leptospira_noguchii_gcf_000306255': 353, 'leptospira_kirschneri_gcf_000243695': 352, 'leptospira_alstonii_gcf_000347175': 318, 'leptospira_weilii_gcf_000244815': 315, 'leptospira_kmetyi_gcf_000243735': 314, 'leptospira_dzianensis_gcf_004770135': 312, 'leptospira_yasudae_gcf_003545925': 311, 'leptospira_mayottensis_gcf_000306675': 308, 'leptospira_alexanderi_gcf_000243815': 306, 'leptospira_barantonii_gcf_002811925': 301, 'leptospira_santarosai_gcf_000313175': 300, 'leptospira_borgpetersenii_gcf_000013945': 298, 'leptospira_putramalaysiae_gcf_004770035': 246, 'leptospira_stimsonii_gcf_003545885': 244, 'leptospira_tipperaryensis_gcf_001729245': 233, 'leptospira_adleri_gcf_002811985': 228, 'leptospira_elllisii_gcf_002811955': 208, 'leptospira_gomenensis_gcf_004770155': 196, 'leptospira_neocaledonica_gcf_002812205': 7, 'leptospira_licerasiae_gcf_000244755': 6, 'leptospira_langatensis_gcf_004770615': 6, 'leptospira_andrefontaineae_gcf_004770105': 5, 'leptospira_hartskeerlii_gcf_002811475': 5, 'leptospira_dzoumogneensis_gcf_004770895': 5, 'leptospira_johnsonii_gcf_003112675': 4, 'leptospira_sp_gcf_000347035': 4, 'leptospira_fletcheri_gcf_004769195': 4, 'leptospira_semungkisensis_gcf_004770055': 4, 'leptospira_meyeri_gcf_000304275': 4, 'leptospira_saintgironsiae_gcf_002811765': 4, 'leptospira_fainei_gcf_000306235': 4, 'leptospira_levettii_gcf_002812085': 4, 'leptospira_vanthielii_gcf_000332455': 4, 'leptospira_idonii_gcf_004770995': 4, 'leptospira_jelokensis_gcf_004769775': 3, 'leptospira_selangorensis_gcf_004769405': 3, 'leptospira_broomii_gcf_000243715': 3, 'leptospira_inadai_gcf_000243675': 3, 'leptospira_terpstrae_gcf_000332495': 3, 'leptospira_koniamboensis_gcf_004769555': 3, 'leptospira_wolbachii_gcf_000332515': 3, 'leptospira_perolatii_gcf_002811875': 3, 'leptospira_macculloughii_gcf_002811975': 3, 'leptospira_sarikeiensis_gcf_004769615': 3, 'leptospira_ilyithenensis_gcf_004771005': 3, 'leptospira_wolffii_gcf_000306115': 3, 'leptospira_fluminis_gcf_004771275': 3, 'leptospira_ellinghausenii_gcf_003114815': 2, 'leptospira_venezuelensis_gcf_002150035': 2, 'leptospira_bouyouniensis_gcf_004770625': 2, 'leptospira_ryugenii_gcf_003114855': 2, 'leptospira_kobayashii_gcf_003114835': 2, 'leptospira_yanagawae_gcf_000332475': 2, 'leptospira_congkakensis_gcf_004770265': 2, 'leptospira_ognonensis_gcf_004770745': 2, 'leptospira_mtsangambouensis_gcf_004770475': 2, 'leptospira_bourretii_gcf_004769285': 2, 'leptospira_kemamanensis_gcf_004769665': 2, 'leptospira_kanakyensis_gcf_004769235': 2, 'leptospira_haakeii_gcf_002812045': 2, 'leptospira_montravelensis_gcf_004769455': 2, 'leptospira_noumeaensis_gcf_004770765': 2, 'leptospira_biflexa_gcf_000017685': 2, 'leptospira_bandrabouensis_gcf_004770905': 2, 'leptospira_brenneri_gcf_002812125': 2, 'leptospira_harrisiae_gcf_002811945': 2, 'leptospira_perdikensis_gcf_004769575': 2})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 10\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/leptospira/initial_intervals/Leptospira_interrogans_GCF_000092565_+10.bed --twobit results/phyluce/leptospira/cleaned_genomes/Leptospira_interrogans_GCF_000092565_formatted.2bit --buffer-to 160 --output results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10.fasta;\n",
      "Screened 356 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 356.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10.fasta --probe-prefix uce_leptospira_ --design leptospira_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 354\n",
      "Probe Count = 702\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas --query results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  13:12:57\n",
      "Ended:  Fri Feb 07, 2020  13:12:57\n",
      "Time for execution:  0.00522816578547 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas                        --lastz results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes_vself.lastz                       --probe-prefix=uce_leptospira_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 701 fasta sequences.  Filtered 12 duplicates. Kept 678.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/leptospira/validate_intervals/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas --scaffoldlist Leptospira_adleri_GCF_002811985 Leptospira_alexanderi_GCF_000243815 Leptospira_alstonii_GCF_000347175 Leptospira_andrefontaineae_GCF_004770105 Leptospira_bandrabouensis_GCF_004770905 Leptospira_barantonii_GCF_002811925 Leptospira_biflexa_GCF_000017685 Leptospira_borgpetersenii_GCF_000013945 Leptospira_bourretii_GCF_004769285 Leptospira_bouyouniensis_GCF_004770625 Leptospira_brenneri_GCF_002812125 Leptospira_broomii_GCF_000243715 Leptospira_congkakensis_GCF_004770265 Leptospira_dzianensis_GCF_004770135 Leptospira_dzoumogneensis_GCF_004770895 Leptospira_ellinghausenii_GCF_003114815 Leptospira_elllisii_GCF_002811955 Leptospira_fainei_GCF_000306235 Leptospira_fletcheri_GCF_004769195 Leptospira_fluminis_GCF_004771275 Leptospira_gomenensis_GCF_004770155 Leptospira_haakeii_GCF_002812045 Leptospira_harrisiae_GCF_002811945 Leptospira_hartskeerlii_GCF_002811475 Leptospira_idonii_GCF_004770995 Leptospira_ilyithenensis_GCF_004771005 Leptospira_inadai_GCF_000243675 Leptospira_jelokensis_GCF_004769775 Leptospira_johnsonii_GCF_003112675 Leptospira_kanakyensis_GCF_004769235 Leptospira_kemamanensis_GCF_004769665 Leptospira_kirschneri_GCF_000243695 Leptospira_kmetyi_GCF_000243735 Leptospira_kobayashii_GCF_003114835 Leptospira_koniamboensis_GCF_004769555 Leptospira_langatensis_GCF_004770615 Leptospira_levettii_GCF_002812085 Leptospira_licerasiae_GCF_000244755 Leptospira_macculloughii_GCF_002811975 Leptospira_mayottensis_GCF_000306675 Leptospira_meyeri_GCF_000304275 Leptospira_montravelensis_GCF_004769455 Leptospira_mtsangambouensis_GCF_004770475 Leptospira_neocaledonica_GCF_002812205 Leptospira_noguchii_GCF_000306255 Leptospira_noumeaensis_GCF_004770765 Leptospira_ognonensis_GCF_004770745 Leptospira_perdikensis_GCF_004769575 Leptospira_perolatii_GCF_002811875 Leptospira_putramalaysiae_GCF_004770035 Leptospira_ryugenii_GCF_003114855 Leptospira_saintgironsiae_GCF_002811765 Leptospira_santarosai_GCF_000313175 Leptospira_sarikeiensis_GCF_004769615 Leptospira_selangorensis_GCF_004769405 Leptospira_semungkisensis_GCF_004770055 Leptospira_sp_GCF_000347035 Leptospira_stimsonii_GCF_003545885 Leptospira_terpstrae_GCF_000332495 Leptospira_tipperaryensis_GCF_001729245 Leptospira_vanthielii_GCF_000332455 Leptospira_venezuelensis_GCF_002150035 Leptospira_weilii_GCF_000244815 Leptospira_wolbachii_GCF_000332515 Leptospira_wolffii_GCF_000306115 Leptospira_yanagawae_GCF_000332475 Leptospira_yasudae_GCF_003545925 Leptospira_interrogans_GCF_000092565 --genome-base-path results/phyluce/leptospira/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/leptospira/validate_intervals/leptospira-to-Leptospira_interrogans_GCF_000092565.sqlite --output results/phyluce/leptospira/validate_intervals/lastz/;\n",
      "\n",
      "Running against Leptospira_adleri_GCF_002811985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1Mb19U.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmy7vT3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_adleri_GCF_002811985.lastz\n",
      "Creating Leptospira_adleri_GCF_002811985 table\n",
      "Inserting data to Leptospira_adleri_GCF_002811985 table\n",
      "\n",
      "Running against Leptospira_alexanderi_GCF_000243815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDguEYt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXUtSwR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_alexanderi_GCF_000243815.lastz\n",
      "Creating Leptospira_alexanderi_GCF_000243815 table\n",
      "Inserting data to Leptospira_alexanderi_GCF_000243815 table\n",
      "\n",
      "Running against Leptospira_alstonii_GCF_000347175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRZT8yj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpz2wIyE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_alstonii_GCF_000347175.lastz\n",
      "Creating Leptospira_alstonii_GCF_000347175 table\n",
      "Inserting data to Leptospira_alstonii_GCF_000347175 table\n",
      "\n",
      "Running against Leptospira_andrefontaineae_GCF_004770105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZ2T2_D.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpM88o7O.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_andrefontaineae_GCF_004770105.lastz\n",
      "Creating Leptospira_andrefontaineae_GCF_004770105 table\n",
      "Inserting data to Leptospira_andrefontaineae_GCF_004770105 table\n",
      "\n",
      "Running against Leptospira_bandrabouensis_GCF_004770905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpR7Yj_T.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVJqT2S.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_bandrabouensis_GCF_004770905.lastz\n",
      "Creating Leptospira_bandrabouensis_GCF_004770905 table\n",
      "Inserting data to Leptospira_bandrabouensis_GCF_004770905 table\n",
      "\n",
      "Running against Leptospira_barantonii_GCF_002811925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplqZSEb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsLb0oO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_barantonii_GCF_002811925.lastz\n",
      "Creating Leptospira_barantonii_GCF_002811925 table\n",
      "Inserting data to Leptospira_barantonii_GCF_002811925 table\n",
      "\n",
      "Running against Leptospira_biflexa_GCF_000017685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmB7adU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNz1brs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_biflexa_GCF_000017685.lastz\n",
      "Creating Leptospira_biflexa_GCF_000017685 table\n",
      "Inserting data to Leptospira_biflexa_GCF_000017685 table\n",
      "\n",
      "Running against Leptospira_borgpetersenii_GCF_000013945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphajgNU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeXzPIa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_borgpetersenii_GCF_000013945.lastz\n",
      "Creating Leptospira_borgpetersenii_GCF_000013945 table\n",
      "Inserting data to Leptospira_borgpetersenii_GCF_000013945 table\n",
      "\n",
      "Running against Leptospira_bourretii_GCF_004769285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFqTnyA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_qDwfL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_bourretii_GCF_004769285.lastz\n",
      "Creating Leptospira_bourretii_GCF_004769285 table\n",
      "Inserting data to Leptospira_bourretii_GCF_004769285 table\n",
      "\n",
      "Running against Leptospira_bouyouniensis_GCF_004770625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpVjo6lW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTJgfFk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_bouyouniensis_GCF_004770625.lastz\n",
      "Creating Leptospira_bouyouniensis_GCF_004770625 table\n",
      "Inserting data to Leptospira_bouyouniensis_GCF_004770625 table\n",
      "\n",
      "Running against Leptospira_brenneri_GCF_002812125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4u1Bej.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcNfbl7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_brenneri_GCF_002812125.lastz\n",
      "Creating Leptospira_brenneri_GCF_002812125 table\n",
      "Inserting data to Leptospira_brenneri_GCF_002812125 table\n",
      "\n",
      "Running against Leptospira_broomii_GCF_000243715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNFQgJn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnCFj7f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_broomii_GCF_000243715.lastz\n",
      "Creating Leptospira_broomii_GCF_000243715 table\n",
      "Inserting data to Leptospira_broomii_GCF_000243715 table\n",
      "\n",
      "Running against Leptospira_congkakensis_GCF_004770265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJDaUOL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpg6I_ib.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_congkakensis_GCF_004770265.lastz\n",
      "Creating Leptospira_congkakensis_GCF_004770265 table\n",
      "Inserting data to Leptospira_congkakensis_GCF_004770265 table\n",
      "\n",
      "Running against Leptospira_dzianensis_GCF_004770135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgbr2EM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcegLv0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_dzianensis_GCF_004770135.lastz\n",
      "Creating Leptospira_dzianensis_GCF_004770135 table\n",
      "Inserting data to Leptospira_dzianensis_GCF_004770135 table\n",
      "\n",
      "Running against Leptospira_dzoumogneensis_GCF_004770895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqrr3dZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuqxYo1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_dzoumogneensis_GCF_004770895.lastz\n",
      "Creating Leptospira_dzoumogneensis_GCF_004770895 table\n",
      "Inserting data to Leptospira_dzoumogneensis_GCF_004770895 table\n",
      "\n",
      "Running against Leptospira_ellinghausenii_GCF_003114815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQzBRuj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpambQXm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_ellinghausenii_GCF_003114815.lastz\n",
      "Creating Leptospira_ellinghausenii_GCF_003114815 table\n",
      "Inserting data to Leptospira_ellinghausenii_GCF_003114815 table\n",
      "\n",
      "Running against Leptospira_elllisii_GCF_002811955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpK5vlgw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0yjaiw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_elllisii_GCF_002811955.lastz\n",
      "Creating Leptospira_elllisii_GCF_002811955 table\n",
      "Inserting data to Leptospira_elllisii_GCF_002811955 table\n",
      "\n",
      "Running against Leptospira_fainei_GCF_000306235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPr2nZL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplVW4Ie.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_fainei_GCF_000306235.lastz\n",
      "Creating Leptospira_fainei_GCF_000306235 table\n",
      "Inserting data to Leptospira_fainei_GCF_000306235 table\n",
      "\n",
      "Running against Leptospira_fletcheri_GCF_004769195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptEzZuu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1VKBqQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_fletcheri_GCF_004769195.lastz\n",
      "Creating Leptospira_fletcheri_GCF_004769195 table\n",
      "Inserting data to Leptospira_fletcheri_GCF_004769195 table\n",
      "\n",
      "Running against Leptospira_fluminis_GCF_004771275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2gpDFG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFjfpdv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_fluminis_GCF_004771275.lastz\n",
      "Creating Leptospira_fluminis_GCF_004771275 table\n",
      "Inserting data to Leptospira_fluminis_GCF_004771275 table\n",
      "\n",
      "Running against Leptospira_gomenensis_GCF_004770155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_NACj9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7AP_QO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_gomenensis_GCF_004770155.lastz\n",
      "Creating Leptospira_gomenensis_GCF_004770155 table\n",
      "Inserting data to Leptospira_gomenensis_GCF_004770155 table\n",
      "\n",
      "Running against Leptospira_haakeii_GCF_002812045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpcR2bz6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmph5IuHJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_haakeii_GCF_002812045.lastz\n",
      "Creating Leptospira_haakeii_GCF_002812045 table\n",
      "Inserting data to Leptospira_haakeii_GCF_002812045 table\n",
      "\n",
      "Running against Leptospira_harrisiae_GCF_002811945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgPbBh4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCIPvca.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_harrisiae_GCF_002811945.lastz\n",
      "Creating Leptospira_harrisiae_GCF_002811945 table\n",
      "Inserting data to Leptospira_harrisiae_GCF_002811945 table\n",
      "\n",
      "Running against Leptospira_hartskeerlii_GCF_002811475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpi1_335.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5HIOkc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_hartskeerlii_GCF_002811475.lastz\n",
      "Creating Leptospira_hartskeerlii_GCF_002811475 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Leptospira_hartskeerlii_GCF_002811475 table\n",
      "\n",
      "Running against Leptospira_idonii_GCF_004770995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJ1gRbP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQWGWkS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_idonii_GCF_004770995.lastz\n",
      "Creating Leptospira_idonii_GCF_004770995 table\n",
      "Inserting data to Leptospira_idonii_GCF_004770995 table\n",
      "\n",
      "Running against Leptospira_ilyithenensis_GCF_004771005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpY7jAI7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3L_maL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_ilyithenensis_GCF_004771005.lastz\n",
      "Creating Leptospira_ilyithenensis_GCF_004771005 table\n",
      "Inserting data to Leptospira_ilyithenensis_GCF_004771005 table\n",
      "\n",
      "Running against Leptospira_inadai_GCF_000243675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFGjr9Z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpi9zvnh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_inadai_GCF_000243675.lastz\n",
      "Creating Leptospira_inadai_GCF_000243675 table\n",
      "Inserting data to Leptospira_inadai_GCF_000243675 table\n",
      "\n",
      "Running against Leptospira_jelokensis_GCF_004769775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpSLE4Jk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWDD4ng.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_jelokensis_GCF_004769775.lastz\n",
      "Creating Leptospira_jelokensis_GCF_004769775 table\n",
      "Inserting data to Leptospira_jelokensis_GCF_004769775 table\n",
      "\n",
      "Running against Leptospira_johnsonii_GCF_003112675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaHZITX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphsMLdA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_johnsonii_GCF_003112675.lastz\n",
      "Creating Leptospira_johnsonii_GCF_003112675 table\n",
      "Inserting data to Leptospira_johnsonii_GCF_003112675 table\n",
      "\n",
      "Running against Leptospira_kanakyensis_GCF_004769235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzAGwcu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOuJ52F.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_kanakyensis_GCF_004769235.lastz\n",
      "Creating Leptospira_kanakyensis_GCF_004769235 table\n",
      "Inserting data to Leptospira_kanakyensis_GCF_004769235 table\n",
      "\n",
      "Running against Leptospira_kemamanensis_GCF_004769665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpztBRFb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKC6Rsr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_kemamanensis_GCF_004769665.lastz\n",
      "Creating Leptospira_kemamanensis_GCF_004769665 table\n",
      "Inserting data to Leptospira_kemamanensis_GCF_004769665 table\n",
      "\n",
      "Running against Leptospira_kirschneri_GCF_000243695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMz4tU2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNhF3es.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_kirschneri_GCF_000243695.lastz\n",
      "Creating Leptospira_kirschneri_GCF_000243695 table\n",
      "Inserting data to Leptospira_kirschneri_GCF_000243695 table\n",
      "\n",
      "Running against Leptospira_kmetyi_GCF_000243735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7bwKvM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUG3WXV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_kmetyi_GCF_000243735.lastz\n",
      "Creating Leptospira_kmetyi_GCF_000243735 table\n",
      "Inserting data to Leptospira_kmetyi_GCF_000243735 table\n",
      "\n",
      "Running against Leptospira_kobayashii_GCF_003114835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpj2eJHK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9ImUCq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_kobayashii_GCF_003114835.lastz\n",
      "Creating Leptospira_kobayashii_GCF_003114835 table\n",
      "Inserting data to Leptospira_kobayashii_GCF_003114835 table\n",
      "\n",
      "Running against Leptospira_koniamboensis_GCF_004769555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJp1IgV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpy6H995.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_koniamboensis_GCF_004769555.lastz\n",
      "Creating Leptospira_koniamboensis_GCF_004769555 table\n",
      "Inserting data to Leptospira_koniamboensis_GCF_004769555 table\n",
      "\n",
      "Running against Leptospira_langatensis_GCF_004770615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp86J19_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTnaMmF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_langatensis_GCF_004770615.lastz\n",
      "Creating Leptospira_langatensis_GCF_004770615 table\n",
      "Inserting data to Leptospira_langatensis_GCF_004770615 table\n",
      "\n",
      "Running against Leptospira_levettii_GCF_002812085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpp7DGEn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpH2EWiq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_levettii_GCF_002812085.lastz\n",
      "Creating Leptospira_levettii_GCF_002812085 table\n",
      "Inserting data to Leptospira_levettii_GCF_002812085 table\n",
      "\n",
      "Running against Leptospira_licerasiae_GCF_000244755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiAr2X1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMTLVLH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_licerasiae_GCF_000244755.lastz\n",
      "Creating Leptospira_licerasiae_GCF_000244755 table\n",
      "Inserting data to Leptospira_licerasiae_GCF_000244755 table\n",
      "\n",
      "Running against Leptospira_macculloughii_GCF_002811975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmppNOxe3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxrEl_b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_macculloughii_GCF_002811975.lastz\n",
      "Creating Leptospira_macculloughii_GCF_002811975 table\n",
      "Inserting data to Leptospira_macculloughii_GCF_002811975 table\n",
      "\n",
      "Running against Leptospira_mayottensis_GCF_000306675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxHLVVn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHQZK1a.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_mayottensis_GCF_000306675.lastz\n",
      "Creating Leptospira_mayottensis_GCF_000306675 table\n",
      "Inserting data to Leptospira_mayottensis_GCF_000306675 table\n",
      "\n",
      "Running against Leptospira_meyeri_GCF_000304275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpml30jv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdXZbTD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_meyeri_GCF_000304275.lastz\n",
      "Creating Leptospira_meyeri_GCF_000304275 table\n",
      "Inserting data to Leptospira_meyeri_GCF_000304275 table\n",
      "\n",
      "Running against Leptospira_montravelensis_GCF_004769455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfHaCAD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgmqC4y.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_montravelensis_GCF_004769455.lastz\n",
      "Creating Leptospira_montravelensis_GCF_004769455 table\n",
      "Inserting data to Leptospira_montravelensis_GCF_004769455 table\n",
      "\n",
      "Running against Leptospira_mtsangambouensis_GCF_004770475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuaPqzo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpS5IDHI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_mtsangambouensis_GCF_004770475.lastz\n",
      "Creating Leptospira_mtsangambouensis_GCF_004770475 table\n",
      "Inserting data to Leptospira_mtsangambouensis_GCF_004770475 table\n",
      "\n",
      "Running against Leptospira_neocaledonica_GCF_002812205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmps8tN9i.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIoWC5t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_neocaledonica_GCF_002812205.lastz\n",
      "Creating Leptospira_neocaledonica_GCF_002812205 table\n",
      "Inserting data to Leptospira_neocaledonica_GCF_002812205 table\n",
      "\n",
      "Running against Leptospira_noguchii_GCF_000306255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpSpJFZV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6CwLsx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_noguchii_GCF_000306255.lastz\n",
      "Creating Leptospira_noguchii_GCF_000306255 table\n",
      "Inserting data to Leptospira_noguchii_GCF_000306255 table\n",
      "\n",
      "Running against Leptospira_noumeaensis_GCF_004770765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxttpuK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEOdsX1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_noumeaensis_GCF_004770765.lastz\n",
      "Creating Leptospira_noumeaensis_GCF_004770765 table\n",
      "Inserting data to Leptospira_noumeaensis_GCF_004770765 table\n",
      "\n",
      "Running against Leptospira_ognonensis_GCF_004770745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphIjLpn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBkLLJL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_ognonensis_GCF_004770745.lastz\n",
      "Creating Leptospira_ognonensis_GCF_004770745 table\n",
      "Inserting data to Leptospira_ognonensis_GCF_004770745 table\n",
      "\n",
      "Running against Leptospira_perdikensis_GCF_004769575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5UIuiz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp06SpV6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_perdikensis_GCF_004769575.lastz\n",
      "Creating Leptospira_perdikensis_GCF_004769575 table\n",
      "Inserting data to Leptospira_perdikensis_GCF_004769575 table\n",
      "\n",
      "Running against Leptospira_perolatii_GCF_002811875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDGhAig.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpama1kG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_perolatii_GCF_002811875.lastz\n",
      "Creating Leptospira_perolatii_GCF_002811875 table\n",
      "Inserting data to Leptospira_perolatii_GCF_002811875 table\n",
      "\n",
      "Running against Leptospira_putramalaysiae_GCF_004770035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp80Nr4p.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsfB36F.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_putramalaysiae_GCF_004770035.lastz\n",
      "Creating Leptospira_putramalaysiae_GCF_004770035 table\n",
      "Inserting data to Leptospira_putramalaysiae_GCF_004770035 table\n",
      "\n",
      "Running against Leptospira_ryugenii_GCF_003114855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpa1aL1R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNbka0s.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_ryugenii_GCF_003114855.lastz\n",
      "Creating Leptospira_ryugenii_GCF_003114855 table\n",
      "Inserting data to Leptospira_ryugenii_GCF_003114855 table\n",
      "\n",
      "Running against Leptospira_saintgironsiae_GCF_002811765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpm_bsXC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKQws94.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_saintgironsiae_GCF_002811765.lastz\n",
      "Creating Leptospira_saintgironsiae_GCF_002811765 table\n",
      "Inserting data to Leptospira_saintgironsiae_GCF_002811765 table\n",
      "\n",
      "Running against Leptospira_santarosai_GCF_000313175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpb4rDVq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3W_MFK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_santarosai_GCF_000313175.lastz\n",
      "Creating Leptospira_santarosai_GCF_000313175 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Leptospira_santarosai_GCF_000313175 table\n",
      "\n",
      "Running against Leptospira_sarikeiensis_GCF_004769615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOUJA2J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2hF7HY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_sarikeiensis_GCF_004769615.lastz\n",
      "Creating Leptospira_sarikeiensis_GCF_004769615 table\n",
      "Inserting data to Leptospira_sarikeiensis_GCF_004769615 table\n",
      "\n",
      "Running against Leptospira_selangorensis_GCF_004769405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfdOszP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5hPhHr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_selangorensis_GCF_004769405.lastz\n",
      "Creating Leptospira_selangorensis_GCF_004769405 table\n",
      "Inserting data to Leptospira_selangorensis_GCF_004769405 table\n",
      "\n",
      "Running against Leptospira_semungkisensis_GCF_004770055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmpLoAb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4Ei63r.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_semungkisensis_GCF_004770055.lastz\n",
      "Creating Leptospira_semungkisensis_GCF_004770055 table\n",
      "Inserting data to Leptospira_semungkisensis_GCF_004770055 table\n",
      "\n",
      "Running against Leptospira_sp_GCF_000347035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpttMtG5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa07UdZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_sp_GCF_000347035.lastz\n",
      "Creating Leptospira_sp_GCF_000347035 table\n",
      "Inserting data to Leptospira_sp_GCF_000347035 table\n",
      "\n",
      "Running against Leptospira_stimsonii_GCF_003545885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmjalGO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpX3OSBd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_stimsonii_GCF_003545885.lastz\n",
      "Creating Leptospira_stimsonii_GCF_003545885 table\n",
      "Inserting data to Leptospira_stimsonii_GCF_003545885 table\n",
      "\n",
      "Running against Leptospira_terpstrae_GCF_000332495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp76Fwcd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzs8DCB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_terpstrae_GCF_000332495.lastz\n",
      "Creating Leptospira_terpstrae_GCF_000332495 table\n",
      "Inserting data to Leptospira_terpstrae_GCF_000332495 table\n",
      "\n",
      "Running against Leptospira_tipperaryensis_GCF_001729245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8GwPiI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdb_Pgt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_tipperaryensis_GCF_001729245.lastz\n",
      "Creating Leptospira_tipperaryensis_GCF_001729245 table\n",
      "Inserting data to Leptospira_tipperaryensis_GCF_001729245 table\n",
      "\n",
      "Running against Leptospira_vanthielii_GCF_000332455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpcKqWJC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpI35_m7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_vanthielii_GCF_000332455.lastz\n",
      "Creating Leptospira_vanthielii_GCF_000332455 table\n",
      "Inserting data to Leptospira_vanthielii_GCF_000332455 table\n",
      "\n",
      "Running against Leptospira_venezuelensis_GCF_002150035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdLO0pc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVMMs7k.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_venezuelensis_GCF_002150035.lastz\n",
      "Creating Leptospira_venezuelensis_GCF_002150035 table\n",
      "Inserting data to Leptospira_venezuelensis_GCF_002150035 table\n",
      "\n",
      "Running against Leptospira_weilii_GCF_000244815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRH_dg9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLhgG0M.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_weilii_GCF_000244815.lastz\n",
      "Creating Leptospira_weilii_GCF_000244815 table\n",
      "Inserting data to Leptospira_weilii_GCF_000244815 table\n",
      "\n",
      "Running against Leptospira_wolbachii_GCF_000332515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmps2hVrE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFzMP0E.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_wolbachii_GCF_000332515.lastz\n",
      "Creating Leptospira_wolbachii_GCF_000332515 table\n",
      "Inserting data to Leptospira_wolbachii_GCF_000332515 table\n",
      "\n",
      "Running against Leptospira_wolffii_GCF_000306115.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyLg_VC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvSeoVa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_wolffii_GCF_000306115.lastz\n",
      "Creating Leptospira_wolffii_GCF_000306115 table\n",
      "Inserting data to Leptospira_wolffii_GCF_000306115 table\n",
      "\n",
      "Running against Leptospira_yanagawae_GCF_000332475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnP8OWa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpe3PsDB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_yanagawae_GCF_000332475.lastz\n",
      "Creating Leptospira_yanagawae_GCF_000332475 table\n",
      "Inserting data to Leptospira_yanagawae_GCF_000332475 table\n",
      "\n",
      "Running against Leptospira_yasudae_GCF_003545925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpn3WD1q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_yMmMM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_yasudae_GCF_003545925.lastz\n",
      "Creating Leptospira_yasudae_GCF_003545925 table\n",
      "Inserting data to Leptospira_yasudae_GCF_003545925 table\n",
      "\n",
      "Running against Leptospira_interrogans_GCF_000092565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpP9IoQC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpomayrZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/leptospira/validate_intervals/lastz/Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_Leptospira_interrogans_GCF_000092565.lastz\n",
      "Creating Leptospira_interrogans_GCF_000092565 table\n",
      "Inserting data to Leptospira_interrogans_GCF_000092565 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/leptospira/extract_probes_from_group/leptospira_genome.conf --lastz results/phyluce/leptospira/validate_intervals/lastz --probes 120 --probe-prefix uce_leptospira_ --name-pattern \"Leptospira_interrogans_GCF_000092565_+10_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/leptospira/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 13:16:05,345 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 13:16:05,349 - Phyluce - INFO - ------- Working on Leptospira_adleri_GCF_002811985 genome -------\n",
      "2020-02-07 13:16:05,350 - Phyluce - INFO - Reading Leptospira_adleri_GCF_002811985 genome\n",
      "2020-02-07 13:16:06,627 - Phyluce - INFO - Leptospira_adleri_GCF_002811985: 354 uces, 16 dupes, 338 non-dupes, 0 orient drop, 0 length drop, 338 written\n",
      "2020-02-07 13:16:06,627 - Phyluce - INFO - ----- Working on Leptospira_alexanderi_GCF_000243815 genome -----\n",
      "2020-02-07 13:16:06,628 - Phyluce - INFO - Reading Leptospira_alexanderi_GCF_000243815 genome\n",
      "2020-02-07 13:16:07,819 - Phyluce - INFO - Leptospira_alexanderi_GCF_000243815: 352 uces, 14 dupes, 338 non-dupes, 0 orient drop, 0 length drop, 338 written\n",
      "2020-02-07 13:16:07,819 - Phyluce - INFO - ------ Working on Leptospira_alstonii_GCF_000347175 genome ------\n",
      "2020-02-07 13:16:07,820 - Phyluce - INFO - Reading Leptospira_alstonii_GCF_000347175 genome\n",
      "2020-02-07 13:16:09,064 - Phyluce - INFO - Leptospira_alstonii_GCF_000347175: 352 uces, 21 dupes, 331 non-dupes, 0 orient drop, 0 length drop, 331 written\n",
      "2020-02-07 13:16:09,064 - Phyluce - INFO - --- Working on Leptospira_andrefontaineae_GCF_004770105 genome --\n",
      "2020-02-07 13:16:09,065 - Phyluce - INFO - Reading Leptospira_andrefontaineae_GCF_004770105 genome\n",
      "2020-02-07 13:16:10,016 - Phyluce - INFO - Leptospira_andrefontaineae_GCF_004770105: 300 uces, 5 dupes, 295 non-dupes, 4 orient drop, 7 length drop, 284 written\n",
      "2020-02-07 13:16:10,016 - Phyluce - INFO - --- Working on Leptospira_bandrabouensis_GCF_004770905 genome ---\n",
      "2020-02-07 13:16:10,029 - Phyluce - INFO - Reading Leptospira_bandrabouensis_GCF_004770905 genome\n",
      "2020-02-07 13:16:10,675 - Phyluce - INFO - Leptospira_bandrabouensis_GCF_004770905: 217 uces, 10 dupes, 207 non-dupes, 0 orient drop, 6 length drop, 201 written\n",
      "2020-02-07 13:16:10,675 - Phyluce - INFO - ----- Working on Leptospira_barantonii_GCF_002811925 genome -----\n",
      "2020-02-07 13:16:10,676 - Phyluce - INFO - Reading Leptospira_barantonii_GCF_002811925 genome\n",
      "2020-02-07 13:16:11,878 - Phyluce - INFO - Leptospira_barantonii_GCF_002811925: 354 uces, 13 dupes, 341 non-dupes, 1 orient drop, 1 length drop, 339 written\n",
      "2020-02-07 13:16:11,878 - Phyluce - INFO - ------- Working on Leptospira_biflexa_GCF_000017685 genome ------\n",
      "2020-02-07 13:16:11,890 - Phyluce - INFO - Reading Leptospira_biflexa_GCF_000017685 genome\n",
      "2020-02-07 13:16:12,526 - Phyluce - INFO - Leptospira_biflexa_GCF_000017685: 211 uces, 0 dupes, 211 non-dupes, 10 orient drop, 9 length drop, 192 written\n",
      "2020-02-07 13:16:12,527 - Phyluce - INFO - --- Working on Leptospira_borgpetersenii_GCF_000013945 genome ---\n",
      "2020-02-07 13:16:12,556 - Phyluce - INFO - Reading Leptospira_borgpetersenii_GCF_000013945 genome\n",
      "2020-02-07 13:16:13,858 - Phyluce - INFO - Leptospira_borgpetersenii_GCF_000013945: 353 uces, 0 dupes, 353 non-dupes, 8 orient drop, 42 length drop, 303 written\n",
      "2020-02-07 13:16:13,858 - Phyluce - INFO - ------ Working on Leptospira_bourretii_GCF_004769285 genome -----\n",
      "2020-02-07 13:16:13,859 - Phyluce - INFO - Reading Leptospira_bourretii_GCF_004769285 genome\n",
      "2020-02-07 13:16:14,502 - Phyluce - INFO - Leptospira_bourretii_GCF_004769285: 218 uces, 8 dupes, 210 non-dupes, 1 orient drop, 4 length drop, 205 written\n",
      "2020-02-07 13:16:14,502 - Phyluce - INFO - ---- Working on Leptospira_bouyouniensis_GCF_004770625 genome ---\n",
      "2020-02-07 13:16:14,503 - Phyluce - INFO - Reading Leptospira_bouyouniensis_GCF_004770625 genome\n",
      "2020-02-07 13:16:15,138 - Phyluce - INFO - Leptospira_bouyouniensis_GCF_004770625: 217 uces, 3 dupes, 214 non-dupes, 8 orient drop, 7 length drop, 199 written\n",
      "2020-02-07 13:16:15,138 - Phyluce - INFO - ------ Working on Leptospira_brenneri_GCF_002812125 genome ------\n",
      "2020-02-07 13:16:15,139 - Phyluce - INFO - Reading Leptospira_brenneri_GCF_002812125 genome\n",
      "2020-02-07 13:16:15,789 - Phyluce - INFO - Leptospira_brenneri_GCF_002812125: 223 uces, 12 dupes, 211 non-dupes, 0 orient drop, 2 length drop, 209 written\n",
      "2020-02-07 13:16:15,789 - Phyluce - INFO - ------- Working on Leptospira_broomii_GCF_000243715 genome ------\n",
      "2020-02-07 13:16:15,790 - Phyluce - INFO - Reading Leptospira_broomii_GCF_000243715 genome\n",
      "2020-02-07 13:16:16,752 - Phyluce - INFO - Leptospira_broomii_GCF_000243715: 307 uces, 4 dupes, 303 non-dupes, 3 orient drop, 8 length drop, 292 written\n",
      "2020-02-07 13:16:16,753 - Phyluce - INFO - ---- Working on Leptospira_congkakensis_GCF_004770265 genome ----\n",
      "2020-02-07 13:16:16,753 - Phyluce - INFO - Reading Leptospira_congkakensis_GCF_004770265 genome\n",
      "2020-02-07 13:16:17,384 - Phyluce - INFO - Leptospira_congkakensis_GCF_004770265: 212 uces, 6 dupes, 206 non-dupes, 6 orient drop, 2 length drop, 198 written\n",
      "2020-02-07 13:16:17,384 - Phyluce - INFO - ----- Working on Leptospira_dzianensis_GCF_004770135 genome -----\n",
      "2020-02-07 13:16:17,389 - Phyluce - INFO - Reading Leptospira_dzianensis_GCF_004770135 genome\n",
      "2020-02-07 13:16:18,624 - Phyluce - INFO - Leptospira_dzianensis_GCF_004770135: 354 uces, 13 dupes, 341 non-dupes, 1 orient drop, 3 length drop, 337 written\n",
      "2020-02-07 13:16:18,625 - Phyluce - INFO - --- Working on Leptospira_dzoumogneensis_GCF_004770895 genome ---\n",
      "2020-02-07 13:16:18,625 - Phyluce - INFO - Reading Leptospira_dzoumogneensis_GCF_004770895 genome\n",
      "2020-02-07 13:16:19,566 - Phyluce - INFO - Leptospira_dzoumogneensis_GCF_004770895: 303 uces, 5 dupes, 298 non-dupes, 4 orient drop, 6 length drop, 288 written\n",
      "2020-02-07 13:16:19,566 - Phyluce - INFO - --- Working on Leptospira_ellinghausenii_GCF_003114815 genome ---\n",
      "2020-02-07 13:16:19,567 - Phyluce - INFO - Reading Leptospira_ellinghausenii_GCF_003114815 genome\n",
      "2020-02-07 13:16:20,224 - Phyluce - INFO - Leptospira_ellinghausenii_GCF_003114815: 217 uces, 11 dupes, 206 non-dupes, 0 orient drop, 1 length drop, 205 written\n",
      "2020-02-07 13:16:20,224 - Phyluce - INFO - ------ Working on Leptospira_elllisii_GCF_002811955 genome ------\n",
      "2020-02-07 13:16:20,232 - Phyluce - INFO - Reading Leptospira_elllisii_GCF_002811955 genome\n",
      "2020-02-07 13:16:21,446 - Phyluce - INFO - Leptospira_elllisii_GCF_002811955: 350 uces, 15 dupes, 335 non-dupes, 0 orient drop, 0 length drop, 335 written\n",
      "2020-02-07 13:16:21,446 - Phyluce - INFO - ------- Working on Leptospira_fainei_GCF_000306235 genome -------\n",
      "2020-02-07 13:16:21,447 - Phyluce - INFO - Reading Leptospira_fainei_GCF_000306235 genome\n",
      "2020-02-07 13:16:22,423 - Phyluce - INFO - Leptospira_fainei_GCF_000306235: 306 uces, 10 dupes, 296 non-dupes, 0 orient drop, 6 length drop, 290 written\n",
      "2020-02-07 13:16:22,424 - Phyluce - INFO - ------ Working on Leptospira_fletcheri_GCF_004769195 genome -----\n",
      "2020-02-07 13:16:22,425 - Phyluce - INFO - Reading Leptospira_fletcheri_GCF_004769195 genome\n",
      "2020-02-07 13:16:23,398 - Phyluce - INFO - Leptospira_fletcheri_GCF_004769195: 306 uces, 11 dupes, 295 non-dupes, 0 orient drop, 3 length drop, 292 written\n",
      "2020-02-07 13:16:23,398 - Phyluce - INFO - ------ Working on Leptospira_fluminis_GCF_004771275 genome ------\n",
      "2020-02-07 13:16:23,399 - Phyluce - INFO - Reading Leptospira_fluminis_GCF_004771275 genome\n",
      "2020-02-07 13:16:24,369 - Phyluce - INFO - Leptospira_fluminis_GCF_004771275: 305 uces, 12 dupes, 293 non-dupes, 0 orient drop, 3 length drop, 290 written\n",
      "2020-02-07 13:16:24,369 - Phyluce - INFO - ----- Working on Leptospira_gomenensis_GCF_004770155 genome -----\n",
      "2020-02-07 13:16:24,376 - Phyluce - INFO - Reading Leptospira_gomenensis_GCF_004770155 genome\n",
      "2020-02-07 13:16:25,578 - Phyluce - INFO - Leptospira_gomenensis_GCF_004770155: 353 uces, 15 dupes, 338 non-dupes, 1 orient drop, 0 length drop, 337 written\n",
      "2020-02-07 13:16:25,578 - Phyluce - INFO - ------- Working on Leptospira_haakeii_GCF_002812045 genome ------\n",
      "2020-02-07 13:16:25,580 - Phyluce - INFO - Reading Leptospira_haakeii_GCF_002812045 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 13:16:26,545 - Phyluce - INFO - Leptospira_haakeii_GCF_002812045: 301 uces, 9 dupes, 292 non-dupes, 0 orient drop, 6 length drop, 286 written\n",
      "2020-02-07 13:16:26,546 - Phyluce - INFO - ------ Working on Leptospira_harrisiae_GCF_002811945 genome -----\n",
      "2020-02-07 13:16:26,546 - Phyluce - INFO - Reading Leptospira_harrisiae_GCF_002811945 genome\n",
      "2020-02-07 13:16:27,187 - Phyluce - INFO - Leptospira_harrisiae_GCF_002811945: 216 uces, 6 dupes, 210 non-dupes, 6 orient drop, 3 length drop, 201 written\n",
      "2020-02-07 13:16:27,187 - Phyluce - INFO - ---- Working on Leptospira_hartskeerlii_GCF_002811475 genome ----\n",
      "2020-02-07 13:16:27,188 - Phyluce - INFO - Reading Leptospira_hartskeerlii_GCF_002811475 genome\n",
      "2020-02-07 13:16:28,109 - Phyluce - INFO - Leptospira_hartskeerlii_GCF_002811475: 298 uces, 5 dupes, 293 non-dupes, 4 orient drop, 3 length drop, 286 written\n",
      "2020-02-07 13:16:28,109 - Phyluce - INFO - ------- Working on Leptospira_idonii_GCF_004770995 genome -------\n",
      "2020-02-07 13:16:28,110 - Phyluce - INFO - Reading Leptospira_idonii_GCF_004770995 genome\n",
      "2020-02-07 13:16:28,774 - Phyluce - INFO - Leptospira_idonii_GCF_004770995: 230 uces, 2 dupes, 228 non-dupes, 0 orient drop, 8 length drop, 220 written\n",
      "2020-02-07 13:16:28,774 - Phyluce - INFO - ---- Working on Leptospira_ilyithenensis_GCF_004771005 genome ---\n",
      "2020-02-07 13:16:28,775 - Phyluce - INFO - Reading Leptospira_ilyithenensis_GCF_004771005 genome\n",
      "2020-02-07 13:16:29,421 - Phyluce - INFO - Leptospira_ilyithenensis_GCF_004771005: 215 uces, 11 dupes, 204 non-dupes, 1 orient drop, 3 length drop, 200 written\n",
      "2020-02-07 13:16:29,421 - Phyluce - INFO - ------- Working on Leptospira_inadai_GCF_000243675 genome -------\n",
      "2020-02-07 13:16:29,422 - Phyluce - INFO - Reading Leptospira_inadai_GCF_000243675 genome\n",
      "2020-02-07 13:16:30,406 - Phyluce - INFO - Leptospira_inadai_GCF_000243675: 308 uces, 9 dupes, 299 non-dupes, 4 orient drop, 5 length drop, 290 written\n",
      "2020-02-07 13:16:30,407 - Phyluce - INFO - ----- Working on Leptospira_jelokensis_GCF_004769775 genome -----\n",
      "2020-02-07 13:16:30,407 - Phyluce - INFO - Reading Leptospira_jelokensis_GCF_004769775 genome\n",
      "2020-02-07 13:16:31,042 - Phyluce - INFO - Leptospira_jelokensis_GCF_004769775: 217 uces, 10 dupes, 207 non-dupes, 1 orient drop, 5 length drop, 201 written\n",
      "2020-02-07 13:16:31,043 - Phyluce - INFO - ------ Working on Leptospira_johnsonii_GCF_003112675 genome -----\n",
      "2020-02-07 13:16:31,043 - Phyluce - INFO - Reading Leptospira_johnsonii_GCF_003112675 genome\n",
      "2020-02-07 13:16:31,988 - Phyluce - INFO - Leptospira_johnsonii_GCF_003112675: 303 uces, 2 dupes, 301 non-dupes, 4 orient drop, 6 length drop, 291 written\n",
      "2020-02-07 13:16:31,988 - Phyluce - INFO - ----- Working on Leptospira_kanakyensis_GCF_004769235 genome ----\n",
      "2020-02-07 13:16:31,988 - Phyluce - INFO - Reading Leptospira_kanakyensis_GCF_004769235 genome\n",
      "2020-02-07 13:16:32,624 - Phyluce - INFO - Leptospira_kanakyensis_GCF_004769235: 220 uces, 4 dupes, 216 non-dupes, 6 orient drop, 2 length drop, 208 written\n",
      "2020-02-07 13:16:32,624 - Phyluce - INFO - ---- Working on Leptospira_kemamanensis_GCF_004769665 genome ----\n",
      "2020-02-07 13:16:32,631 - Phyluce - INFO - Reading Leptospira_kemamanensis_GCF_004769665 genome\n",
      "2020-02-07 13:16:33,275 - Phyluce - INFO - Leptospira_kemamanensis_GCF_004769665: 219 uces, 12 dupes, 207 non-dupes, 0 orient drop, 1 length drop, 206 written\n",
      "2020-02-07 13:16:33,275 - Phyluce - INFO - ----- Working on Leptospira_kirschneri_GCF_000243695 genome -----\n",
      "2020-02-07 13:16:33,287 - Phyluce - INFO - Reading Leptospira_kirschneri_GCF_000243695 genome\n",
      "2020-02-07 13:16:34,488 - Phyluce - INFO - Leptospira_kirschneri_GCF_000243695: 354 uces, 4 dupes, 350 non-dupes, 4 orient drop, 5 length drop, 341 written\n",
      "2020-02-07 13:16:34,488 - Phyluce - INFO - ------- Working on Leptospira_kmetyi_GCF_000243735 genome -------\n",
      "2020-02-07 13:16:34,489 - Phyluce - INFO - Reading Leptospira_kmetyi_GCF_000243735 genome\n",
      "2020-02-07 13:16:35,714 - Phyluce - INFO - Leptospira_kmetyi_GCF_000243735: 354 uces, 3 dupes, 351 non-dupes, 12 orient drop, 5 length drop, 334 written\n",
      "2020-02-07 13:16:35,714 - Phyluce - INFO - ----- Working on Leptospira_kobayashii_GCF_003114835 genome -----\n",
      "2020-02-07 13:16:35,715 - Phyluce - INFO - Reading Leptospira_kobayashii_GCF_003114835 genome\n",
      "2020-02-07 13:16:36,349 - Phyluce - INFO - Leptospira_kobayashii_GCF_003114835: 218 uces, 9 dupes, 209 non-dupes, 0 orient drop, 2 length drop, 207 written\n",
      "2020-02-07 13:16:36,349 - Phyluce - INFO - ---- Working on Leptospira_koniamboensis_GCF_004769555 genome ---\n",
      "2020-02-07 13:16:36,350 - Phyluce - INFO - Reading Leptospira_koniamboensis_GCF_004769555 genome\n",
      "2020-02-07 13:16:37,278 - Phyluce - INFO - Leptospira_koniamboensis_GCF_004769555: 300 uces, 3 dupes, 297 non-dupes, 4 orient drop, 6 length drop, 287 written\n",
      "2020-02-07 13:16:37,278 - Phyluce - INFO - ----- Working on Leptospira_langatensis_GCF_004770615 genome ----\n",
      "2020-02-07 13:16:37,293 - Phyluce - INFO - Reading Leptospira_langatensis_GCF_004770615 genome\n",
      "2020-02-07 13:16:38,243 - Phyluce - INFO - Leptospira_langatensis_GCF_004770615: 301 uces, 10 dupes, 291 non-dupes, 0 orient drop, 5 length drop, 286 written\n",
      "2020-02-07 13:16:38,243 - Phyluce - INFO - ------ Working on Leptospira_levettii_GCF_002812085 genome ------\n",
      "2020-02-07 13:16:38,244 - Phyluce - INFO - Reading Leptospira_levettii_GCF_002812085 genome\n",
      "2020-02-07 13:16:38,882 - Phyluce - INFO - Leptospira_levettii_GCF_002812085: 219 uces, 10 dupes, 209 non-dupes, 0 orient drop, 2 length drop, 207 written\n",
      "2020-02-07 13:16:38,882 - Phyluce - INFO - ----- Working on Leptospira_licerasiae_GCF_000244755 genome -----\n",
      "2020-02-07 13:16:38,883 - Phyluce - INFO - Reading Leptospira_licerasiae_GCF_000244755 genome\n",
      "2020-02-07 13:16:39,832 - Phyluce - INFO - Leptospira_licerasiae_GCF_000244755: 302 uces, 9 dupes, 293 non-dupes, 4 orient drop, 5 length drop, 284 written\n",
      "2020-02-07 13:16:39,832 - Phyluce - INFO - ---- Working on Leptospira_macculloughii_GCF_002811975 genome ---\n",
      "2020-02-07 13:16:39,833 - Phyluce - INFO - Reading Leptospira_macculloughii_GCF_002811975 genome\n",
      "2020-02-07 13:16:40,870 - Phyluce - INFO - Leptospira_macculloughii_GCF_002811975: 227 uces, 155 dupes, 72 non-dupes, 0 orient drop, 0 length drop, 72 written\n",
      "2020-02-07 13:16:40,870 - Phyluce - INFO - ----- Working on Leptospira_mayottensis_GCF_000306675 genome ----\n",
      "2020-02-07 13:16:40,880 - Phyluce - INFO - Reading Leptospira_mayottensis_GCF_000306675 genome\n",
      "2020-02-07 13:16:42,115 - Phyluce - INFO - Leptospira_mayottensis_GCF_000306675: 353 uces, 0 dupes, 353 non-dupes, 19 orient drop, 4 length drop, 330 written\n",
      "2020-02-07 13:16:42,115 - Phyluce - INFO - ------- Working on Leptospira_meyeri_GCF_000304275 genome -------\n",
      "2020-02-07 13:16:42,116 - Phyluce - INFO - Reading Leptospira_meyeri_GCF_000304275 genome\n",
      "2020-02-07 13:16:42,786 - Phyluce - INFO - Leptospira_meyeri_GCF_000304275: 223 uces, 8 dupes, 215 non-dupes, 0 orient drop, 7 length drop, 208 written\n",
      "2020-02-07 13:16:42,787 - Phyluce - INFO - --- Working on Leptospira_montravelensis_GCF_004769455 genome ---\n",
      "2020-02-07 13:16:42,787 - Phyluce - INFO - Reading Leptospira_montravelensis_GCF_004769455 genome\n",
      "2020-02-07 13:16:43,419 - Phyluce - INFO - Leptospira_montravelensis_GCF_004769455: 219 uces, 4 dupes, 215 non-dupes, 6 orient drop, 4 length drop, 205 written\n",
      "2020-02-07 13:16:43,419 - Phyluce - INFO - -- Working on Leptospira_mtsangambouensis_GCF_004770475 genome --\n",
      "2020-02-07 13:16:43,420 - Phyluce - INFO - Reading Leptospira_mtsangambouensis_GCF_004770475 genome\n",
      "2020-02-07 13:16:44,058 - Phyluce - INFO - Leptospira_mtsangambouensis_GCF_004770475: 219 uces, 3 dupes, 216 non-dupes, 6 orient drop, 3 length drop, 207 written\n",
      "2020-02-07 13:16:44,058 - Phyluce - INFO - ---- Working on Leptospira_neocaledonica_GCF_002812205 genome ---\n",
      "2020-02-07 13:16:44,059 - Phyluce - INFO - Reading Leptospira_neocaledonica_GCF_002812205 genome\n",
      "2020-02-07 13:16:44,979 - Phyluce - INFO - Leptospira_neocaledonica_GCF_002812205: 299 uces, 3 dupes, 296 non-dupes, 3 orient drop, 6 length drop, 287 written\n",
      "2020-02-07 13:16:44,979 - Phyluce - INFO - ------ Working on Leptospira_noguchii_GCF_000306255 genome ------\n",
      "2020-02-07 13:16:44,987 - Phyluce - INFO - Reading Leptospira_noguchii_GCF_000306255 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 13:16:46,217 - Phyluce - INFO - Leptospira_noguchii_GCF_000306255: 353 uces, 15 dupes, 338 non-dupes, 1 orient drop, 1 length drop, 336 written\n",
      "2020-02-07 13:16:46,217 - Phyluce - INFO - ----- Working on Leptospira_noumeaensis_GCF_004770765 genome ----\n",
      "2020-02-07 13:16:46,218 - Phyluce - INFO - Reading Leptospira_noumeaensis_GCF_004770765 genome\n",
      "2020-02-07 13:16:46,849 - Phyluce - INFO - Leptospira_noumeaensis_GCF_004770765: 216 uces, 12 dupes, 204 non-dupes, 0 orient drop, 2 length drop, 202 written\n",
      "2020-02-07 13:16:46,849 - Phyluce - INFO - ----- Working on Leptospira_ognonensis_GCF_004770745 genome -----\n",
      "2020-02-07 13:16:46,852 - Phyluce - INFO - Reading Leptospira_ognonensis_GCF_004770745 genome\n",
      "2020-02-07 13:16:47,482 - Phyluce - INFO - Leptospira_ognonensis_GCF_004770745: 216 uces, 10 dupes, 206 non-dupes, 0 orient drop, 0 length drop, 206 written\n",
      "2020-02-07 13:16:47,483 - Phyluce - INFO - ----- Working on Leptospira_perdikensis_GCF_004769575 genome ----\n",
      "2020-02-07 13:16:47,483 - Phyluce - INFO - Reading Leptospira_perdikensis_GCF_004769575 genome\n",
      "2020-02-07 13:16:48,115 - Phyluce - INFO - Leptospira_perdikensis_GCF_004769575: 216 uces, 11 dupes, 205 non-dupes, 0 orient drop, 1 length drop, 204 written\n",
      "2020-02-07 13:16:48,115 - Phyluce - INFO - ------ Working on Leptospira_perolatii_GCF_002811875 genome -----\n",
      "2020-02-07 13:16:48,116 - Phyluce - INFO - Reading Leptospira_perolatii_GCF_002811875 genome\n",
      "2020-02-07 13:16:49,065 - Phyluce - INFO - Leptospira_perolatii_GCF_002811875: 307 uces, 9 dupes, 298 non-dupes, 0 orient drop, 4 length drop, 294 written\n",
      "2020-02-07 13:16:49,065 - Phyluce - INFO - --- Working on Leptospira_putramalaysiae_GCF_004770035 genome ---\n",
      "2020-02-07 13:16:49,066 - Phyluce - INFO - Reading Leptospira_putramalaysiae_GCF_004770035 genome\n",
      "2020-02-07 13:16:50,271 - Phyluce - INFO - Leptospira_putramalaysiae_GCF_004770035: 353 uces, 16 dupes, 337 non-dupes, 1 orient drop, 1 length drop, 335 written\n",
      "2020-02-07 13:16:50,271 - Phyluce - INFO - ------ Working on Leptospira_ryugenii_GCF_003114855 genome ------\n",
      "2020-02-07 13:16:50,271 - Phyluce - INFO - Reading Leptospira_ryugenii_GCF_003114855 genome\n",
      "2020-02-07 13:16:50,886 - Phyluce - INFO - Leptospira_ryugenii_GCF_003114855: 214 uces, 3 dupes, 211 non-dupes, 0 orient drop, 9 length drop, 202 written\n",
      "2020-02-07 13:16:50,886 - Phyluce - INFO - --- Working on Leptospira_saintgironsiae_GCF_002811765 genome ---\n",
      "2020-02-07 13:16:50,887 - Phyluce - INFO - Reading Leptospira_saintgironsiae_GCF_002811765 genome\n",
      "2020-02-07 13:16:51,820 - Phyluce - INFO - Leptospira_saintgironsiae_GCF_002811765: 298 uces, 5 dupes, 293 non-dupes, 4 orient drop, 6 length drop, 283 written\n",
      "2020-02-07 13:16:51,821 - Phyluce - INFO - ----- Working on Leptospira_santarosai_GCF_000313175 genome -----\n",
      "2020-02-07 13:16:51,821 - Phyluce - INFO - Reading Leptospira_santarosai_GCF_000313175 genome\n",
      "2020-02-07 13:16:53,035 - Phyluce - INFO - Leptospira_santarosai_GCF_000313175: 353 uces, 1 dupes, 352 non-dupes, 15 orient drop, 3 length drop, 334 written\n",
      "2020-02-07 13:16:53,035 - Phyluce - INFO - ---- Working on Leptospira_sarikeiensis_GCF_004769615 genome ----\n",
      "2020-02-07 13:16:53,036 - Phyluce - INFO - Reading Leptospira_sarikeiensis_GCF_004769615 genome\n",
      "2020-02-07 13:16:53,979 - Phyluce - INFO - Leptospira_sarikeiensis_GCF_004769615: 302 uces, 11 dupes, 291 non-dupes, 0 orient drop, 3 length drop, 288 written\n",
      "2020-02-07 13:16:53,979 - Phyluce - INFO - ---- Working on Leptospira_selangorensis_GCF_004769405 genome ---\n",
      "2020-02-07 13:16:53,979 - Phyluce - INFO - Reading Leptospira_selangorensis_GCF_004769405 genome\n",
      "2020-02-07 13:16:54,952 - Phyluce - INFO - Leptospira_selangorensis_GCF_004769405: 301 uces, 5 dupes, 296 non-dupes, 4 orient drop, 6 length drop, 286 written\n",
      "2020-02-07 13:16:54,952 - Phyluce - INFO - --- Working on Leptospira_semungkisensis_GCF_004770055 genome ---\n",
      "2020-02-07 13:16:54,959 - Phyluce - INFO - Reading Leptospira_semungkisensis_GCF_004770055 genome\n",
      "2020-02-07 13:16:55,910 - Phyluce - INFO - Leptospira_semungkisensis_GCF_004770055: 301 uces, 4 dupes, 297 non-dupes, 4 orient drop, 7 length drop, 286 written\n",
      "2020-02-07 13:16:55,911 - Phyluce - INFO - --------- Working on Leptospira_sp_GCF_000347035 genome ---------\n",
      "2020-02-07 13:16:55,911 - Phyluce - INFO - Reading Leptospira_sp_GCF_000347035 genome\n",
      "2020-02-07 13:16:56,877 - Phyluce - INFO - Leptospira_sp_GCF_000347035: 302 uces, 6 dupes, 296 non-dupes, 4 orient drop, 5 length drop, 287 written\n",
      "2020-02-07 13:16:56,878 - Phyluce - INFO - ------ Working on Leptospira_stimsonii_GCF_003545885 genome -----\n",
      "2020-02-07 13:16:56,878 - Phyluce - INFO - Reading Leptospira_stimsonii_GCF_003545885 genome\n",
      "2020-02-07 13:16:58,100 - Phyluce - INFO - Leptospira_stimsonii_GCF_003545885: 353 uces, 10 dupes, 343 non-dupes, 4 orient drop, 5 length drop, 334 written\n",
      "2020-02-07 13:16:58,100 - Phyluce - INFO - ------ Working on Leptospira_terpstrae_GCF_000332495 genome -----\n",
      "2020-02-07 13:16:58,101 - Phyluce - INFO - Reading Leptospira_terpstrae_GCF_000332495 genome\n",
      "2020-02-07 13:16:58,773 - Phyluce - INFO - Leptospira_terpstrae_GCF_000332495: 223 uces, 10 dupes, 213 non-dupes, 6 orient drop, 1 length drop, 206 written\n",
      "2020-02-07 13:16:58,773 - Phyluce - INFO - --- Working on Leptospira_tipperaryensis_GCF_001729245 genome ---\n",
      "2020-02-07 13:16:58,773 - Phyluce - INFO - Reading Leptospira_tipperaryensis_GCF_001729245 genome\n",
      "2020-02-07 13:17:00,008 - Phyluce - INFO - Leptospira_tipperaryensis_GCF_001729245: 354 uces, 1 dupes, 353 non-dupes, 15 orient drop, 9 length drop, 329 written\n",
      "2020-02-07 13:17:00,008 - Phyluce - INFO - ----- Working on Leptospira_vanthielii_GCF_000332455 genome -----\n",
      "2020-02-07 13:17:00,011 - Phyluce - INFO - Reading Leptospira_vanthielii_GCF_000332455 genome\n",
      "2020-02-07 13:17:00,687 - Phyluce - INFO - Leptospira_vanthielii_GCF_000332455: 220 uces, 18 dupes, 202 non-dupes, 0 orient drop, 1 length drop, 201 written\n",
      "2020-02-07 13:17:00,687 - Phyluce - INFO - ---- Working on Leptospira_venezuelensis_GCF_002150035 genome ---\n",
      "2020-02-07 13:17:00,696 - Phyluce - INFO - Reading Leptospira_venezuelensis_GCF_002150035 genome\n",
      "2020-02-07 13:17:01,636 - Phyluce - INFO - Leptospira_venezuelensis_GCF_002150035: 303 uces, 5 dupes, 298 non-dupes, 4 orient drop, 6 length drop, 288 written\n",
      "2020-02-07 13:17:01,636 - Phyluce - INFO - ------- Working on Leptospira_weilii_GCF_000244815 genome -------\n",
      "2020-02-07 13:17:01,637 - Phyluce - INFO - Reading Leptospira_weilii_GCF_000244815 genome\n",
      "2020-02-07 13:17:02,860 - Phyluce - INFO - Leptospira_weilii_GCF_000244815: 352 uces, 18 dupes, 334 non-dupes, 0 orient drop, 0 length drop, 334 written\n",
      "2020-02-07 13:17:02,860 - Phyluce - INFO - ------ Working on Leptospira_wolbachii_GCF_000332515 genome -----\n",
      "2020-02-07 13:17:02,869 - Phyluce - INFO - Reading Leptospira_wolbachii_GCF_000332515 genome\n",
      "2020-02-07 13:17:03,541 - Phyluce - INFO - Leptospira_wolbachii_GCF_000332515: 219 uces, 15 dupes, 204 non-dupes, 0 orient drop, 3 length drop, 201 written\n",
      "2020-02-07 13:17:03,541 - Phyluce - INFO - ------- Working on Leptospira_wolffii_GCF_000306115 genome ------\n",
      "2020-02-07 13:17:03,550 - Phyluce - INFO - Reading Leptospira_wolffii_GCF_000306115 genome\n",
      "2020-02-07 13:17:04,509 - Phyluce - INFO - Leptospira_wolffii_GCF_000306115: 305 uces, 6 dupes, 299 non-dupes, 4 orient drop, 3 length drop, 292 written\n",
      "2020-02-07 13:17:04,509 - Phyluce - INFO - ------ Working on Leptospira_yanagawae_GCF_000332475 genome -----\n",
      "2020-02-07 13:17:04,510 - Phyluce - INFO - Reading Leptospira_yanagawae_GCF_000332475 genome\n",
      "2020-02-07 13:17:05,162 - Phyluce - INFO - Leptospira_yanagawae_GCF_000332475: 217 uces, 13 dupes, 204 non-dupes, 1 orient drop, 0 length drop, 203 written\n",
      "2020-02-07 13:17:05,163 - Phyluce - INFO - ------- Working on Leptospira_yasudae_GCF_003545925 genome ------\n",
      "2020-02-07 13:17:05,163 - Phyluce - INFO - Reading Leptospira_yasudae_GCF_003545925 genome\n",
      "2020-02-07 13:17:06,369 - Phyluce - INFO - Leptospira_yasudae_GCF_003545925: 354 uces, 8 dupes, 346 non-dupes, 3 orient drop, 6 length drop, 337 written\n",
      "2020-02-07 13:17:06,369 - Phyluce - INFO - ----- Working on Leptospira_interrogans_GCF_000092565 genome ----\n",
      "2020-02-07 13:17:06,370 - Phyluce - INFO - Reading Leptospira_interrogans_GCF_000092565 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 13:17:07,595 - Phyluce - INFO - Leptospira_interrogans_GCF_000092565: 354 uces, 0 dupes, 354 non-dupes, 16 orient drop, 5 length drop, 333 written\r\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/leptospira/extract_probes_from_group/probe_fasta --output results/phyluce/leptospira/extract_probes_from_group/multifastas.sqlite --base-taxon Leptospira_interrogans_GCF_000092565;\n",
      "leptospira_adleri_gcf_002811985.\n",
      "leptospira_alexanderi_gcf_000243815.\n",
      "leptospira_alstonii_gcf_000347175.\n",
      "leptospira_andrefontaineae_gcf_004770105.\n",
      "leptospira_bandrabouensis_gcf_004770905.\n",
      "leptospira_barantonii_gcf_002811925.\n",
      "leptospira_biflexa_gcf_000017685.\n",
      "leptospira_borgpetersenii_gcf_000013945.\n",
      "leptospira_bourretii_gcf_004769285.\n",
      "leptospira_bouyouniensis_gcf_004770625.\n",
      "leptospira_brenneri_gcf_002812125.\n",
      "leptospira_broomii_gcf_000243715.\n",
      "leptospira_congkakensis_gcf_004770265.\n",
      "leptospira_dzianensis_gcf_004770135.\n",
      "leptospira_dzoumogneensis_gcf_004770895.\n",
      "leptospira_ellinghausenii_gcf_003114815.\n",
      "leptospira_elllisii_gcf_002811955.\n",
      "leptospira_fainei_gcf_000306235.\n",
      "leptospira_fletcheri_gcf_004769195.\n",
      "leptospira_fluminis_gcf_004771275.\n",
      "leptospira_gomenensis_gcf_004770155.\n",
      "leptospira_haakeii_gcf_002812045.\n",
      "leptospira_harrisiae_gcf_002811945.\n",
      "leptospira_hartskeerlii_gcf_002811475.\n",
      "leptospira_idonii_gcf_004770995.\n",
      "leptospira_ilyithenensis_gcf_004771005.\n",
      "leptospira_inadai_gcf_000243675.\n",
      "leptospira_jelokensis_gcf_004769775.\n",
      "leptospira_johnsonii_gcf_003112675.\n",
      "leptospira_kanakyensis_gcf_004769235.\n",
      "leptospira_kemamanensis_gcf_004769665.\n",
      "leptospira_kirschneri_gcf_000243695.\n",
      "leptospira_kmetyi_gcf_000243735.\n",
      "leptospira_kobayashii_gcf_003114835.\n",
      "leptospira_koniamboensis_gcf_004769555.\n",
      "leptospira_langatensis_gcf_004770615.\n",
      "leptospira_levettii_gcf_002812085.\n",
      "leptospira_licerasiae_gcf_000244755.\n",
      "leptospira_macculloughii_gcf_002811975.\n",
      "leptospira_mayottensis_gcf_000306675.\n",
      "leptospira_meyeri_gcf_000304275.\n",
      "leptospira_montravelensis_gcf_004769455.\n",
      "leptospira_mtsangambouensis_gcf_004770475.\n",
      "leptospira_neocaledonica_gcf_002812205.\n",
      "leptospira_noguchii_gcf_000306255.\n",
      "leptospira_noumeaensis_gcf_004770765.\n",
      "leptospira_ognonensis_gcf_004770745.\n",
      "leptospira_perdikensis_gcf_004769575.\n",
      "leptospira_perolatii_gcf_002811875.\n",
      "leptospira_putramalaysiae_gcf_004770035.\n",
      "leptospira_ryugenii_gcf_003114855.\n",
      "leptospira_saintgironsiae_gcf_002811765.\n",
      "leptospira_santarosai_gcf_000313175.\n",
      "leptospira_sarikeiensis_gcf_004769615.\n",
      "leptospira_selangorensis_gcf_004769405.\n",
      "leptospira_semungkisensis_gcf_004770055.\n",
      "leptospira_sp_gcf_000347035.\n",
      "leptospira_stimsonii_gcf_003545885.\n",
      "leptospira_terpstrae_gcf_000332495.\n",
      "leptospira_tipperaryensis_gcf_001729245.\n",
      "leptospira_vanthielii_gcf_000332455.\n",
      "leptospira_venezuelensis_gcf_002150035.\n",
      "leptospira_weilii_gcf_000244815.\n",
      "leptospira_wolbachii_gcf_000332515.\n",
      "leptospira_wolffii_gcf_000306115.\n",
      "leptospira_yanagawae_gcf_000332475.\n",
      "leptospira_yasudae_gcf_003545925.\n",
      "leptospira_interrogans_gcf_000092565.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/leptospira/extract_probes_from_group/multifastas.sqlite --base-taxon Leptospira_interrogans_GCF_000092565;\n",
      "Loci shared by 0 taxa:\t348.0\n",
      "Loci shared by 1 taxa:\t348.0\n",
      "Loci shared by 2 taxa:\t346.0\n",
      "Loci shared by 3 taxa:\t346.0\n",
      "Loci shared by 4 taxa:\t346.0\n",
      "Loci shared by 5 taxa:\t346.0\n",
      "Loci shared by 6 taxa:\t346.0\n",
      "Loci shared by 7 taxa:\t346.0\n",
      "Loci shared by 8 taxa:\t345.0\n",
      "Loci shared by 9 taxa:\t345.0\n",
      "Loci shared by 10 taxa:\t345.0\n",
      "Loci shared by 11 taxa:\t345.0\n",
      "Loci shared by 12 taxa:\t344.0\n",
      "Loci shared by 13 taxa:\t342.0\n",
      "Loci shared by 14 taxa:\t341.0\n",
      "Loci shared by 15 taxa:\t341.0\n",
      "Loci shared by 16 taxa:\t341.0\n",
      "Loci shared by 17 taxa:\t341.0\n",
      "Loci shared by 18 taxa:\t339.0\n",
      "Loci shared by 19 taxa:\t337.0\n",
      "Loci shared by 20 taxa:\t316.0\n",
      "Loci shared by 21 taxa:\t313.0\n",
      "Loci shared by 22 taxa:\t311.0\n",
      "Loci shared by 23 taxa:\t309.0\n",
      "Loci shared by 24 taxa:\t306.0\n",
      "Loci shared by 25 taxa:\t305.0\n",
      "Loci shared by 26 taxa:\t304.0\n",
      "Loci shared by 27 taxa:\t303.0\n",
      "Loci shared by 28 taxa:\t302.0\n",
      "Loci shared by 29 taxa:\t301.0\n",
      "Loci shared by 30 taxa:\t299.0\n",
      "Loci shared by 31 taxa:\t298.0\n",
      "Loci shared by 32 taxa:\t297.0\n",
      "Loci shared by 33 taxa:\t297.0\n",
      "Loci shared by 34 taxa:\t296.0\n",
      "Loci shared by 35 taxa:\t294.0\n",
      "Loci shared by 36 taxa:\t292.0\n",
      "Loci shared by 37 taxa:\t290.0\n",
      "Loci shared by 38 taxa:\t289.0\n",
      "Loci shared by 39 taxa:\t286.0\n",
      "Loci shared by 40 taxa:\t283.0\n",
      "Loci shared by 41 taxa:\t273.0\n",
      "Loci shared by 42 taxa:\t240.0\n",
      "Loci shared by 43 taxa:\t230.0\n",
      "Loci shared by 44 taxa:\t225.0\n",
      "Loci shared by 45 taxa:\t223.0\n",
      "Loci shared by 46 taxa:\t219.0\n",
      "Loci shared by 47 taxa:\t215.0\n",
      "Loci shared by 48 taxa:\t209.0\n",
      "Loci shared by 49 taxa:\t207.0\n",
      "Loci shared by 50 taxa:\t206.0\n",
      "Loci shared by 51 taxa:\t200.0\n",
      "Loci shared by 52 taxa:\t195.0\n",
      "Loci shared by 53 taxa:\t191.0\n",
      "Loci shared by 54 taxa:\t190.0\n",
      "Loci shared by 55 taxa:\t190.0\n",
      "Loci shared by 56 taxa:\t187.0\n",
      "Loci shared by 57 taxa:\t185.0\n",
      "Loci shared by 58 taxa:\t184.0\n",
      "Loci shared by 59 taxa:\t181.0\n",
      "Loci shared by 60 taxa:\t176.0\n",
      "Loci shared by 61 taxa:\t173.0\n",
      "Loci shared by 62 taxa:\t170.0\n",
      "Loci shared by 63 taxa:\t166.0\n",
      "Loci shared by 64 taxa:\t160.0\n",
      "Loci shared by 65 taxa:\t151.0\n",
      "Loci shared by 66 taxa:\t141.0\n",
      "Loci shared by 67 taxa:\t115.0\n",
      "Loci shared by 68 taxa:\t25.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/leptospira/extract_probes_from_group/multifastas.sqlite  --base-taxon Leptospira_interrogans_GCF_000092565 --output results/phyluce/leptospira/extract_probes_from_group/Leptospira_interrogans_GCF_000092565+10-back-to-67.conf --specific-counts 67;\n",
      "Counter({'leptospira_tipperaryensis_gcf_001729245': 115, 'leptospira_kirschneri_gcf_000243695': 115, 'leptospira_jelokensis_gcf_004769775': 115, 'leptospira_selangorensis_gcf_004769405': 115, 'leptospira_neocaledonica_gcf_002812205': 115, 'leptospira_johnsonii_gcf_003112675': 115, 'leptospira_venezuelensis_gcf_002150035': 115, 'leptospira_broomii_gcf_000243715': 115, 'leptospira_inadai_gcf_000243675': 115, 'leptospira_gomenensis_gcf_004770155': 115, 'leptospira_bouyouniensis_gcf_004770625': 115, 'leptospira_sp_gcf_000347035': 115, 'leptospira_noguchii_gcf_000306255': 115, 'leptospira_ryugenii_gcf_003114855': 115, 'leptospira_fletcheri_gcf_004769195': 115, 'leptospira_brenneri_gcf_002812125': 115, 'leptospira_kobayashii_gcf_003114835': 115, 'leptospira_yanagawae_gcf_000332475': 115, 'leptospira_andrefontaineae_gcf_004770105': 115, 'leptospira_bandrabouensis_gcf_004770905': 115, 'leptospira_licerasiae_gcf_000244755': 115, 'leptospira_meyeri_gcf_000304275': 115, 'leptospira_alexanderi_gcf_000243815': 115, 'leptospira_hartskeerlii_gcf_002811475': 115, 'leptospira_terpstrae_gcf_000332495': 115, 'leptospira_koniamboensis_gcf_004769555': 115, 'leptospira_fluminis_gcf_004771275': 115, 'leptospira_weilii_gcf_000244815': 115, 'leptospira_saintgironsiae_gcf_002811765': 115, 'leptospira_fainei_gcf_000306235': 115, 'leptospira_santarosai_gcf_000313175': 115, 'leptospira_yasudae_gcf_003545925': 115, 'leptospira_elllisii_gcf_002811955': 115, 'leptospira_congkakensis_gcf_004770265': 115, 'leptospira_levettii_gcf_002812085': 115, 'leptospira_wolbachii_gcf_000332515': 115, 'leptospira_dzianensis_gcf_004770135': 115, 'leptospira_mayottensis_gcf_000306675': 115, 'leptospira_mtsangambouensis_gcf_004770475': 115, 'leptospira_bourretii_gcf_004769285': 115, 'leptospira_stimsonii_gcf_003545885': 115, 'leptospira_langatensis_gcf_004770615': 115, 'leptospira_kemamanensis_gcf_004769665': 115, 'leptospira_putramalaysiae_gcf_004770035': 115, 'leptospira_kmetyi_gcf_000243735': 115, 'leptospira_interrogans_gcf_000092565': 115, 'leptospira_haakeii_gcf_002812045': 115, 'leptospira_idonii_gcf_004770995': 115, 'leptospira_adleri_gcf_002811985': 115, 'leptospira_montravelensis_gcf_004769455': 115, 'leptospira_sarikeiensis_gcf_004769615': 115, 'leptospira_noumeaensis_gcf_004770765': 115, 'leptospira_biflexa_gcf_000017685': 115, 'leptospira_semungkisensis_gcf_004770055': 115, 'leptospira_ilyithenensis_gcf_004771005': 115, 'leptospira_alstonii_gcf_000347175': 115, 'leptospira_kanakyensis_gcf_004769235': 115, 'leptospira_wolffii_gcf_000306115': 115, 'leptospira_harrisiae_gcf_002811945': 115, 'leptospira_perdikensis_gcf_004769575': 115, 'leptospira_vanthielii_gcf_000332455': 115, 'leptospira_barantonii_gcf_002811925': 115, 'leptospira_dzoumogneensis_gcf_004770895': 115, 'leptospira_ellinghausenii_gcf_003114815': 114, 'leptospira_ognonensis_gcf_004770745': 114, 'leptospira_perolatii_gcf_002811875': 114, 'leptospira_borgpetersenii_gcf_000013945': 108, 'leptospira_macculloughii_gcf_002811975': 35})\n",
      "Total loci = 115\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(67)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/leptospira/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/leptospira/extract_probes_from_group/Leptospira_interrogans_GCF_000092565+10-back-to-67.conf --probe-prefix uce_leptospira_ --designer rnplattii --design leptospira_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 115\n",
      "Probe Count = 15390\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list.fasta --query results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  13:19:30\n",
      "Ended:  Fri Feb 07, 2020  13:20:54\n",
      "Time for execution:  1.39971166452 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list.fasta --lastz results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_leptospira_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 15389 fasta sequences.  Filtered 0 duplicates. Kept 15390.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/leptospira/final_probe_design/leptospira_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 13:54:26 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 15390\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1231200\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 90M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3938769\n",
      "Max number of word counting entries: 88661707\n",
      "\n",
      "# comparing sequences from          0  to       2565\n",
      "..---------- new table with     1422 representatives\n",
      "# comparing sequences from       2565  to       4702\n",
      "----------    894 remaining sequences to the next cycle\n",
      "---------- new table with      798 representatives\n",
      "# comparing sequences from       3808  to       5738\n",
      "----------    858 remaining sequences to the next cycle\n",
      "---------- new table with      585 representatives\n",
      "# comparing sequences from       4880  to       6631\n",
      "----------    827 remaining sequences to the next cycle\n",
      "---------- new table with      498 representatives\n",
      "# comparing sequences from       5804  to       7401\n",
      "----------    765 remaining sequences to the next cycle\n",
      "---------- new table with      490 representatives\n",
      "# comparing sequences from       6636  to       8095\n",
      "----------    692 remaining sequences to the next cycle\n",
      "---------- new table with      469 representatives\n",
      "# comparing sequences from       7403  to       8734\n",
      "----------    530 remaining sequences to the next cycle\n",
      "---------- new table with      434 representatives\n",
      "# comparing sequences from       8204  to       9401\n",
      "----------    450 remaining sequences to the next cycle\n",
      "---------- new table with      434 representatives\n",
      "# comparing sequences from       8951  to      10024\n",
      "----------    490 remaining sequences to the next cycle\n",
      "---------- new table with      312 representatives\n",
      "# comparing sequences from       9534  to      10510\n",
      "..........    10000  finished       5674  clusters\n",
      "----------    381 remaining sequences to the next cycle\n",
      "---------- new table with      274 representatives\n",
      "# comparing sequences from      10129  to      11005\n",
      "----------    321 remaining sequences to the next cycle\n",
      "---------- new table with      254 representatives\n",
      "# comparing sequences from      10684  to      11468\n",
      "----------    327 remaining sequences to the next cycle\n",
      "---------- new table with      275 representatives\n",
      "# comparing sequences from      11141  to      11849\n",
      "----------    435 remaining sequences to the next cycle\n",
      "---------- new table with      200 representatives\n",
      "# comparing sequences from      11414  to      12076\n",
      "----------    204 remaining sequences to the next cycle\n",
      "---------- new table with      285 representatives\n",
      "# comparing sequences from      11872  to      12458\n",
      "----------    265 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      12193  to      12725\n",
      "----------    126 remaining sequences to the next cycle\n",
      "---------- new table with      191 representatives\n",
      "# comparing sequences from      12599  to      13064\n",
      "----------    146 remaining sequences to the next cycle\n",
      "---------- new table with      181 representatives\n",
      "# comparing sequences from      12918  to      13330\n",
      "----------     24 remaining sequences to the next cycle\n",
      "---------- new table with      158 representatives\n",
      "# comparing sequences from      13306  to      13653\n",
      "----------    144 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      13509  to      13822\n",
      "----------    131 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      13691  to      13974\n",
      "99.9%---------- new table with      141 representatives\n",
      "# comparing sequences from      13974  to      14210\n",
      "----------     15 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      14195  to      14394\n",
      "----------     19 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      14375  to      14544\n",
      "99.9%---------- new table with       82 representatives\n",
      "# comparing sequences from      14544  to      15390\n",
      "....................---------- new table with      413 representatives\n",
      "\n",
      "    15390  finished       8396  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 91M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 2.69\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Listeria"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'listeria'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Listeria_aquatica_GCF_000525795\",\n",
    "               \"Listeria_booriae_GCF_000766865\",\n",
    "               \"Listeria_cornellensis_GCF_000525855\",\n",
    "               \"Listeria_costaricensis_GCF_900183385\",\n",
    "               \"Listeria_fleischmannii_GCF_000252625\",\n",
    "               \"Listeria_floridensis_GCF_000525875\",\n",
    "               \"Listeria_goaensis_GCF_900186125\",\n",
    "               \"Listeria_grandensis_GCF_000525835\",\n",
    "               \"Listeria_grayi_GCF_000148995\",\n",
    "               \"Listeria_innocua_GCF_000195795\",\n",
    "               \"Listeria_ivanovii_GCF_900637745\",\n",
    "               \"Listeria_kieliensis_GCF_003369925\",\n",
    "               \"Listeria_marthii_GCF_000183865\",          \n",
    "               \"Listeria_newyorkensis_GCF_900461625\",\n",
    "               \"Listeria_riparia_GCF_000525995\",\n",
    "               \"Listeria_rocourtiae_GCF_000525975\",\n",
    "               \"Listeria_seeligeri_GCF_000027145\",\n",
    "               \"Listeria_sp_GCF_003856515\",\n",
    "               \"Listeria_thailandensis_GCF_900576925\",\n",
    "               \"Listeria_weihenstephanensis_GCF_000525955\",\n",
    "               \"Listeria_welshimeri_GCF_900187315\" ]\n",
    "                    \n",
    "reference_taxon = \"Listeria_monocytogenes_GCF_000196035\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525795.1_Listeriaceae_bacterium_FSL_S10-1188_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 773875 bytes  515944.67 bytes/sec\n",
      "total size is 773553  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000766865.1_ASM76686v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1022888 bytes  681953.33 bytes/sec\n",
      "total size is 1022527  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525855.1_Listeriaceae_bacterium_FSL_F6-969_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1000469 bytes  400204.40 bytes/sec\n",
      "total size is 1000093  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900183385.1_CLIP_2016_00682_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 950880 bytes  633948.00 bytes/sec\n",
      "total size is 950530  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000252625.2_ASM25262v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 843010 bytes  562034.67 bytes/sec\n",
      "total size is 842697  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525875.1_Listeriaceae_bacterium_FSL_S10-1187_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 833819 bytes  555907.33 bytes/sec\n",
      "total size is 833481  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900186125.1_ILCC801.fasta_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 860462 bytes  573669.33 bytes/sec\n",
      "total size is 860138  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525835.1_Listeriaceaebacterium_FSL_F6-971_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 950991 bytes  634022.00 bytes/sec\n",
      "total size is 950624  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000148995.1_ASM14899v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 769862 bytes  513269.33 bytes/sec\n",
      "total size is 769565  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000195795.1_ASM19579v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 911779 bytes  607880.67 bytes/sec\n",
      "total size is 911450  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_900637745.1_52551_F01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 885882 bytes  590616.00 bytes/sec\n",
      "total size is 885554  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003369925.1_ASM336992v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 773749 bytes  515860.67 bytes/sec\n",
      "total size is 773451  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000183865.1_ASM18386v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 824466 bytes  549672.00 bytes/sec\n",
      "total size is 824153  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000196035.1_ASM19603v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 869721 bytes  579842.00 bytes/sec\n",
      "total size is 869400  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900461625.1_58116_G02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1042892 bytes  695289.33 bytes/sec\n",
      "total size is 1042532  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525995.1_PRIP_MIRA_assembly_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 981442 bytes  654322.67 bytes/sec\n",
      "total size is 981089  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525975.1_LrocourtiaeTYPE_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 960612 bytes  640436.00 bytes/sec\n",
      "total size is 960262  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000027145.1_ASM2714v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 823530 bytes  549048.00 bytes/sec\n",
      "total size is 823218  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003856515.1_ASM385651v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1068210 bytes  712168.00 bytes/sec\n",
      "total size is 1067840  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900576925.1_CLIP201500305T_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 801809 bytes  534567.33 bytes/sec\n",
      "total size is 801500  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000525955.1_Lweihenstaphanensis_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1008856 bytes  672598.67 bytes/sec\n",
      "total size is 1008494  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_900187315.1_52087_C01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 827718 bytes  551840.00 bytes/sec\n",
      "total size is 827406  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483096 (\"sim_Listeria_aquatica_GCF_000525795\") has been submitted\n",
      "Your job 5483097 (\"sim_Listeria_booriae_GCF_000766865\") has been submitted\n",
      "Your job 5483098 (\"sim_Listeria_cornellensis_GCF_000525855\") has been submitted\n",
      "Your job 5483099 (\"sim_Listeria_costaricensis_GCF_900183385\") has been submitted\n",
      "Your job 5483100 (\"sim_Listeria_fleischmannii_GCF_000252625\") has been submitted\n",
      "Your job 5483101 (\"sim_Listeria_floridensis_GCF_000525875\") has been submitted\n",
      "Your job 5483102 (\"sim_Listeria_goaensis_GCF_900186125\") has been submitted\n",
      "Your job 5483103 (\"sim_Listeria_grandensis_GCF_000525835\") has been submitted\n",
      "Your job 5483104 (\"sim_Listeria_grayi_GCF_000148995\") has been submitted\n",
      "Your job 5483105 (\"sim_Listeria_innocua_GCF_000195795\") has been submitted\n",
      "Your job 5483106 (\"sim_Listeria_ivanovii_GCF_900637745\") has been submitted\n",
      "Your job 5483107 (\"sim_Listeria_kieliensis_GCF_003369925\") has been submitted\n",
      "Your job 5483108 (\"sim_Listeria_marthii_GCF_000183865\") has been submitted\n",
      "Your job 5483109 (\"sim_Listeria_newyorkensis_GCF_900461625\") has been submitted\n",
      "Your job 5483110 (\"sim_Listeria_riparia_GCF_000525995\") has been submitted\n",
      "Your job 5483111 (\"sim_Listeria_rocourtiae_GCF_000525975\") has been submitted\n",
      "Your job 5483112 (\"sim_Listeria_seeligeri_GCF_000027145\") has been submitted\n",
      "Your job 5483113 (\"sim_Listeria_sp_GCF_003856515\") has been submitted\n",
      "Your job 5483114 (\"sim_Listeria_thailandensis_GCF_900576925\") has been submitted\n",
      "Your job 5483115 (\"sim_Listeria_weihenstephanensis_GCF_000525955\") has been submitted\n",
      "Your job 5483116 (\"sim_Listeria_welshimeri_GCF_900187315\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/listeria/cleaned_genomes/Listeria_monocytogenes_GCF_000196035_formatted.fas path=results/phyluce/listeria/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483117 (\"map_Listeria_aquatica_GCF_000525795\") has been submitted\n",
      "Your job 5483118 (\"map_Listeria_booriae_GCF_000766865\") has been submitted\n",
      "Your job 5483119 (\"map_Listeria_cornellensis_GCF_000525855\") has been submitted\n",
      "Your job 5483120 (\"map_Listeria_costaricensis_GCF_900183385\") has been submitted\n",
      "Your job 5483121 (\"map_Listeria_fleischmannii_GCF_000252625\") has been submitted\n",
      "Your job 5483122 (\"map_Listeria_floridensis_GCF_000525875\") has been submitted\n",
      "Your job 5483123 (\"map_Listeria_goaensis_GCF_900186125\") has been submitted\n",
      "Your job 5483124 (\"map_Listeria_grandensis_GCF_000525835\") has been submitted\n",
      "Your job 5483125 (\"map_Listeria_grayi_GCF_000148995\") has been submitted\n",
      "Your job 5483126 (\"map_Listeria_innocua_GCF_000195795\") has been submitted\n",
      "Your job 5483127 (\"map_Listeria_ivanovii_GCF_900637745\") has been submitted\n",
      "Your job 5483128 (\"map_Listeria_kieliensis_GCF_003369925\") has been submitted\n",
      "Your job 5483129 (\"map_Listeria_marthii_GCF_000183865\") has been submitted\n",
      "Your job 5483130 (\"map_Listeria_newyorkensis_GCF_900461625\") has been submitted\n",
      "Your job 5483131 (\"map_Listeria_riparia_GCF_000525995\") has been submitted\n",
      "Your job 5483132 (\"map_Listeria_rocourtiae_GCF_000525975\") has been submitted\n",
      "Your job 5483133 (\"map_Listeria_seeligeri_GCF_000027145\") has been submitted\n",
      "Your job 5483134 (\"map_Listeria_sp_GCF_003856515\") has been submitted\n",
      "Your job 5483135 (\"map_Listeria_thailandensis_GCF_900576925\") has been submitted\n",
      "Your job 5483136 (\"map_Listeria_weihenstephanensis_GCF_000525955\") has been submitted\n",
      "Your job 5483137 (\"map_Listeria_welshimeri_GCF_900187315\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483138 (\"merge_Listeria_aquatica_GCF_000525795\") has been submitted\n",
      "Your job 5483139 (\"merge_Listeria_booriae_GCF_000766865\") has been submitted\n",
      "Your job 5483140 (\"merge_Listeria_cornellensis_GCF_000525855\") has been submitted\n",
      "Your job 5483141 (\"merge_Listeria_costaricensis_GCF_900183385\") has been submitted\n",
      "Your job 5483142 (\"merge_Listeria_fleischmannii_GCF_000252625\") has been submitted\n",
      "Your job 5483143 (\"merge_Listeria_floridensis_GCF_000525875\") has been submitted\n",
      "Your job 5483144 (\"merge_Listeria_goaensis_GCF_900186125\") has been submitted\n",
      "Your job 5483145 (\"merge_Listeria_grandensis_GCF_000525835\") has been submitted\n",
      "Your job 5483146 (\"merge_Listeria_grayi_GCF_000148995\") has been submitted\n",
      "Your job 5483147 (\"merge_Listeria_innocua_GCF_000195795\") has been submitted\n",
      "Your job 5483148 (\"merge_Listeria_ivanovii_GCF_900637745\") has been submitted\n",
      "Your job 5483149 (\"merge_Listeria_kieliensis_GCF_003369925\") has been submitted\n",
      "Your job 5483150 (\"merge_Listeria_marthii_GCF_000183865\") has been submitted\n",
      "Your job 5483151 (\"merge_Listeria_newyorkensis_GCF_900461625\") has been submitted\n",
      "Your job 5483152 (\"merge_Listeria_riparia_GCF_000525995\") has been submitted\n",
      "Your job 5483153 (\"merge_Listeria_rocourtiae_GCF_000525975\") has been submitted\n",
      "Your job 5483154 (\"merge_Listeria_seeligeri_GCF_000027145\") has been submitted\n",
      "Your job 5483155 (\"merge_Listeria_sp_GCF_003856515\") has been submitted\n",
      "Your job 5483156 (\"merge_Listeria_thailandensis_GCF_900576925\") has been submitted\n",
      "Your job 5483157 (\"merge_Listeria_weihenstephanensis_GCF_000525955\") has been submitted\n",
      "Your job 5483158 (\"merge_Listeria_welshimeri_GCF_900187315\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 196 sequences from Listeria_aquatica_GCF_000525795_merged.bed.  Filtered 131 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 65.\n",
      "Screened 308 sequences from Listeria_booriae_GCF_000766865_merged.bed.  Filtered 205 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 103.\n",
      "Screened 220 sequences from Listeria_cornellensis_GCF_000525855_merged.bed.  Filtered 138 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 82.\n",
      "Screened 248 sequences from Listeria_costaricensis_GCF_900183385_merged.bed.  Filtered 164 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 84.\n",
      "Screened 268 sequences from Listeria_fleischmannii_GCF_000252625_merged.bed.  Filtered 176 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 92.\n",
      "Screened 242 sequences from Listeria_floridensis_GCF_000525875_merged.bed.  Filtered 167 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 263 sequences from Listeria_goaensis_GCF_900186125_merged.bed.  Filtered 167 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 96.\n",
      "Screened 231 sequences from Listeria_grandensis_GCF_000525835_merged.bed.  Filtered 156 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 228 sequences from Listeria_grayi_GCF_000148995_merged.bed.  Filtered 137 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 91.\n",
      "Screened 3801 sequences from Listeria_innocua_GCF_000195795_merged.bed.  Filtered 1357 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2444.\n",
      "Screened 2848 sequences from Listeria_ivanovii_GCF_900637745_merged.bed.  Filtered 1397 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1451.\n",
      "Screened 205 sequences from Listeria_kieliensis_GCF_003369925_merged.bed.  Filtered 146 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 59.\n",
      "Screened 4140 sequences from Listeria_marthii_GCF_000183865_merged.bed.  Filtered 1384 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2756.\n",
      "Screened 243 sequences from Listeria_newyorkensis_GCF_900461625_merged.bed.  Filtered 155 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 88.\n",
      "Screened 247 sequences from Listeria_riparia_GCF_000525995_merged.bed.  Filtered 177 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 70.\n",
      "Screened 207 sequences from Listeria_rocourtiae_GCF_000525975_merged.bed.  Filtered 139 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 68.\n",
      "Screened 2935 sequences from Listeria_seeligeri_GCF_000027145_merged.bed.  Filtered 1428 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1507.\n",
      "Screened 246 sequences from Listeria_sp_GCF_003856515_merged.bed.  Filtered 165 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 81.\n",
      "Screened 201 sequences from Listeria_thailandensis_GCF_900576925_merged.bed.  Filtered 148 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 53.\n",
      "Screened 253 sequences from Listeria_weihenstephanensis_GCF_000525955_merged.bed.  Filtered 167 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 86.\n",
      "Screened 3902 sequences from Listeria_welshimeri_GCF_900187315_merged.bed.  Filtered 1512 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2390.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/listeria/initial_intervals/Listeria_welshimeri_GCF_900187315_merged.bed --twobit results/phyluce/listeria/cleaned_genomes/Listeria_monocytogenes_GCF_000196035_formatted.2bit --output results/phyluce/listeria/initial_intervals/Listeria_welshimeri_GCF_900187315_stripped.bed;\n",
      "listeria_aquatica_gcf_000525795.\n",
      "listeria_booriae_gcf_000766865.\n",
      "listeria_cornellensis_gcf_000525855.\n",
      "listeria_costaricensis_gcf_900183385.\n",
      "listeria_fleischmannii_gcf_000252625.\n",
      "listeria_floridensis_gcf_000525875.\n",
      "listeria_goaensis_gcf_900186125.\n",
      "listeria_grandensis_gcf_000525835.\n",
      "listeria_grayi_gcf_000148995.\n",
      "listeria_innocua_gcf_000195795...\n",
      "listeria_ivanovii_gcf_900637745..\n",
      "listeria_kieliensis_gcf_003369925.\n",
      "listeria_marthii_gcf_000183865...\n",
      "listeria_newyorkensis_gcf_900461625.\n",
      "listeria_riparia_gcf_000525995.\n",
      "listeria_rocourtiae_gcf_000525975.\n",
      "listeria_seeligeri_gcf_000027145..\n",
      "listeria_sp_gcf_003856515.\n",
      "listeria_thailandensis_gcf_900576925.\n",
      "listeria_weihenstephanensis_gcf_000525955.\n",
      "listeria_welshimeri_gcf_900187315...\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/listeria/initial_intervals/Listeria_welshimeri_GCF_900187315_merged.bed --twobit results/phyluce/listeria/cleaned_genomes/Listeria_monocytogenes_GCF_000196035_formatted.2bit --output results/phyluce/listeria/initial_intervals/Listeria_welshimeri_GCF_900187315_stripped.bed;\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 0 taxa:\t3,661.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 1 taxa:\t3,661.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 2 taxa:\t2,643.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 3 taxa:\t2,008.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 4 taxa:\t1,323.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 5 taxa:\t913.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 6 taxa:\t214.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 7 taxa:\t156.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 8 taxa:\t132.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 9 taxa:\t121.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 10 taxa:\t109.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 11 taxa:\t92.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 12 taxa:\t85.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 13 taxa:\t71.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 14 taxa:\t62.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 15 taxa:\t52.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 16 taxa:\t43.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 17 taxa:\t39.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 18 taxa:\t30.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 19 taxa:\t27.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 20 taxa:\t24.0\n",
      "Loci shared by Listeria_monocytogenes_GCF_000196035 + 21 taxa:\t14.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/listeria/initial_intervals/listeria-to-Listeria_monocytogenes_GCF_000196035.sqlite --base-taxon Listeria_monocytogenes_GCF_000196035 --output results/phyluce/listeria/initial_intervals/Listeria_monocytogenes_GCF_000196035_+6.bed --specific-counts 6;\n",
      "Counter({'listeria_welshimeri_gcf_900187315': 213, 'listeria_seeligeri_gcf_000027145': 212, 'listeria_innocua_gcf_000195795': 211, 'listeria_ivanovii_gcf_900637745': 210, 'listeria_marthii_gcf_000183865': 199, 'listeria_goaensis_gcf_900186125': 100, 'listeria_grayi_gcf_000148995': 95, 'listeria_newyorkensis_gcf_900461625': 92, 'listeria_fleischmannii_gcf_000252625': 90, 'listeria_weihenstephanensis_gcf_000525955': 89, 'listeria_booriae_gcf_000766865': 89, 'listeria_sp_gcf_003856515': 87, 'listeria_cornellensis_gcf_000525855': 85, 'listeria_costaricensis_gcf_900183385': 83, 'listeria_grandensis_gcf_000525835': 80, 'listeria_rocourtiae_gcf_000525975': 78, 'listeria_riparia_gcf_000525995': 76, 'listeria_floridensis_gcf_000525875': 75, 'listeria_aquatica_gcf_000525795': 64, 'listeria_kieliensis_gcf_003369925': 61, 'listeria_thailandensis_gcf_900576925': 52})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 6\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/listeria/initial_intervals/Listeria_monocytogenes_GCF_000196035_+6.bed --twobit results/phyluce/listeria/cleaned_genomes/Listeria_monocytogenes_GCF_000196035_formatted.2bit --buffer-to 160 --output results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6.fasta;\n",
      "Screened 214 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 214.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6.fasta --probe-prefix uce_listeria_ --design listeria_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 213\n",
      "Probe Count = 424\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas --query results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  10:43:07\n",
      "Ended:  Fri Feb 07, 2020  10:43:07\n",
      "Time for execution:  0.00404454867045 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas                        --lastz results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes_vself.lastz                       --probe-prefix=uce_listeria_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 423 fasta sequences.  Filtered 19 duplicates. Kept 386.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/listeria/validate_intervals/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas --scaffoldlist Listeria_aquatica_GCF_000525795 Listeria_booriae_GCF_000766865 Listeria_cornellensis_GCF_000525855 Listeria_costaricensis_GCF_900183385 Listeria_fleischmannii_GCF_000252625 Listeria_floridensis_GCF_000525875 Listeria_goaensis_GCF_900186125 Listeria_grandensis_GCF_000525835 Listeria_grayi_GCF_000148995 Listeria_innocua_GCF_000195795 Listeria_ivanovii_GCF_900637745 Listeria_kieliensis_GCF_003369925 Listeria_marthii_GCF_000183865 Listeria_newyorkensis_GCF_900461625 Listeria_riparia_GCF_000525995 Listeria_rocourtiae_GCF_000525975 Listeria_seeligeri_GCF_000027145 Listeria_sp_GCF_003856515 Listeria_thailandensis_GCF_900576925 Listeria_weihenstephanensis_GCF_000525955 Listeria_welshimeri_GCF_900187315 Listeria_monocytogenes_GCF_000196035 --genome-base-path results/phyluce/listeria/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/listeria/validate_intervals/listeria-to-Listeria_monocytogenes_GCF_000196035.sqlite --output results/phyluce/listeria/validate_intervals/lastz/;\n",
      "\n",
      "Running against Listeria_aquatica_GCF_000525795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpr0gi0n.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4pQPy2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_aquatica_GCF_000525795.lastz\n",
      "Creating Listeria_aquatica_GCF_000525795 table\n",
      "Inserting data to Listeria_aquatica_GCF_000525795 table\n",
      "\n",
      "Running against Listeria_booriae_GCF_000766865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmHcBZj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKlhqX0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_booriae_GCF_000766865.lastz\n",
      "Creating Listeria_booriae_GCF_000766865 table\n",
      "Inserting data to Listeria_booriae_GCF_000766865 table\n",
      "\n",
      "Running against Listeria_cornellensis_GCF_000525855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbV3xDu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLzAjc4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_cornellensis_GCF_000525855.lastz\n",
      "Creating Listeria_cornellensis_GCF_000525855 table\n",
      "Inserting data to Listeria_cornellensis_GCF_000525855 table\n",
      "\n",
      "Running against Listeria_costaricensis_GCF_900183385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4sFenA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpW_j3Mf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_costaricensis_GCF_900183385.lastz\n",
      "Creating Listeria_costaricensis_GCF_900183385 table\n",
      "Inserting data to Listeria_costaricensis_GCF_900183385 table\n",
      "\n",
      "Running against Listeria_fleischmannii_GCF_000252625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwUEkkI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmKcyWF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_fleischmannii_GCF_000252625.lastz\n",
      "Creating Listeria_fleischmannii_GCF_000252625 table\n",
      "Inserting data to Listeria_fleischmannii_GCF_000252625 table\n",
      "\n",
      "Running against Listeria_floridensis_GCF_000525875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpv6ieIy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwyQ19a.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_floridensis_GCF_000525875.lastz\n",
      "Creating Listeria_floridensis_GCF_000525875 table\n",
      "Inserting data to Listeria_floridensis_GCF_000525875 table\n",
      "\n",
      "Running against Listeria_goaensis_GCF_900186125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsmcN57.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa4tJtY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_goaensis_GCF_900186125.lastz\n",
      "Creating Listeria_goaensis_GCF_900186125 table\n",
      "Inserting data to Listeria_goaensis_GCF_900186125 table\n",
      "\n",
      "Running against Listeria_grandensis_GCF_000525835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9fSHQt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPWKrBu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_grandensis_GCF_000525835.lastz\n",
      "Creating Listeria_grandensis_GCF_000525835 table\n",
      "Inserting data to Listeria_grandensis_GCF_000525835 table\n",
      "\n",
      "Running against Listeria_grayi_GCF_000148995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIWkPol.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdUH6qX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_grayi_GCF_000148995.lastz\n",
      "Creating Listeria_grayi_GCF_000148995 table\n",
      "Inserting data to Listeria_grayi_GCF_000148995 table\n",
      "\n",
      "Running against Listeria_innocua_GCF_000195795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpddjfH9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpV1LXHy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_innocua_GCF_000195795.lastz\n",
      "Creating Listeria_innocua_GCF_000195795 table\n",
      "Inserting data to Listeria_innocua_GCF_000195795 table\n",
      "\n",
      "Running against Listeria_ivanovii_GCF_900637745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyCdCrx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpA3z4Uj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_ivanovii_GCF_900637745.lastz\n",
      "Creating Listeria_ivanovii_GCF_900637745 table\n",
      "Inserting data to Listeria_ivanovii_GCF_900637745 table\n",
      "\n",
      "Running against Listeria_kieliensis_GCF_003369925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYAnE_S.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqMSdRX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_kieliensis_GCF_003369925.lastz\n",
      "Creating Listeria_kieliensis_GCF_003369925 table\n",
      "Inserting data to Listeria_kieliensis_GCF_003369925 table\n",
      "\n",
      "Running against Listeria_marthii_GCF_000183865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpB0NGc8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvP1t1O.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_marthii_GCF_000183865.lastz\n",
      "Creating Listeria_marthii_GCF_000183865 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Listeria_marthii_GCF_000183865 table\n",
      "\n",
      "Running against Listeria_newyorkensis_GCF_900461625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaTHJLW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoe7aR1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_newyorkensis_GCF_900461625.lastz\n",
      "Creating Listeria_newyorkensis_GCF_900461625 table\n",
      "Inserting data to Listeria_newyorkensis_GCF_900461625 table\n",
      "\n",
      "Running against Listeria_riparia_GCF_000525995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptsfym0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBH4ick.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_riparia_GCF_000525995.lastz\n",
      "Creating Listeria_riparia_GCF_000525995 table\n",
      "Inserting data to Listeria_riparia_GCF_000525995 table\n",
      "\n",
      "Running against Listeria_rocourtiae_GCF_000525975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpz8wgiJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNlZhgR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_rocourtiae_GCF_000525975.lastz\n",
      "Creating Listeria_rocourtiae_GCF_000525975 table\n",
      "Inserting data to Listeria_rocourtiae_GCF_000525975 table\n",
      "\n",
      "Running against Listeria_seeligeri_GCF_000027145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZZ8Uxu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpk62b0s.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_seeligeri_GCF_000027145.lastz\n",
      "Creating Listeria_seeligeri_GCF_000027145 table\n",
      "Inserting data to Listeria_seeligeri_GCF_000027145 table\n",
      "\n",
      "Running against Listeria_sp_GCF_003856515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsu1hg2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDhlGsC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_sp_GCF_003856515.lastz\n",
      "Creating Listeria_sp_GCF_003856515 table\n",
      "Inserting data to Listeria_sp_GCF_003856515 table\n",
      "\n",
      "Running against Listeria_thailandensis_GCF_900576925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDolh59.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQsUg9U.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_thailandensis_GCF_900576925.lastz\n",
      "Creating Listeria_thailandensis_GCF_900576925 table\n",
      "Inserting data to Listeria_thailandensis_GCF_900576925 table\n",
      "\n",
      "Running against Listeria_weihenstephanensis_GCF_000525955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp53Bi5K.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpn50LTE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_weihenstephanensis_GCF_000525955.lastz\n",
      "Creating Listeria_weihenstephanensis_GCF_000525955 table\n",
      "Inserting data to Listeria_weihenstephanensis_GCF_000525955 table\n",
      "\n",
      "Running against Listeria_welshimeri_GCF_900187315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpz9mcLb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxMolnK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_welshimeri_GCF_900187315.lastz\n",
      "Creating Listeria_welshimeri_GCF_900187315 table\n",
      "Inserting data to Listeria_welshimeri_GCF_900187315 table\n",
      "\n",
      "Running against Listeria_monocytogenes_GCF_000196035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_muMJ1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprLY3da.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/listeria/validate_intervals/lastz/Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_Listeria_monocytogenes_GCF_000196035.lastz\n",
      "Creating Listeria_monocytogenes_GCF_000196035 table\n",
      "Inserting data to Listeria_monocytogenes_GCF_000196035 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/listeria/extract_probes_from_group/listeria_genome.conf --lastz results/phyluce/listeria/validate_intervals/lastz --probes 120 --probe-prefix uce_listeria_ --name-pattern \"Listeria_monocytogenes_GCF_000196035_+6_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/listeria/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 10:43:50,288 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 10:43:50,291 - Phyluce - INFO - ------- Working on Listeria_aquatica_GCF_000525795 genome -------\n",
      "2020-02-07 10:43:50,292 - Phyluce - INFO - Reading Listeria_aquatica_GCF_000525795 genome\n",
      "2020-02-07 10:43:51,135 - Phyluce - INFO - Listeria_aquatica_GCF_000525795: 210 uces, 25 dupes, 185 non-dupes, 0 orient drop, 0 length drop, 185 written\n",
      "2020-02-07 10:43:51,135 - Phyluce - INFO - -------- Working on Listeria_booriae_GCF_000766865 genome -------\n",
      "2020-02-07 10:43:51,135 - Phyluce - INFO - Reading Listeria_booriae_GCF_000766865 genome\n",
      "2020-02-07 10:43:51,898 - Phyluce - INFO - Listeria_booriae_GCF_000766865: 211 uces, 19 dupes, 192 non-dupes, 0 orient drop, 2 length drop, 190 written\n",
      "2020-02-07 10:43:51,898 - Phyluce - INFO - ----- Working on Listeria_cornellensis_GCF_000525855 genome -----\n",
      "2020-02-07 10:43:51,898 - Phyluce - INFO - Reading Listeria_cornellensis_GCF_000525855 genome\n",
      "2020-02-07 10:43:52,670 - Phyluce - INFO - Listeria_cornellensis_GCF_000525855: 210 uces, 22 dupes, 188 non-dupes, 0 orient drop, 4 length drop, 184 written\n",
      "2020-02-07 10:43:52,670 - Phyluce - INFO - ----- Working on Listeria_costaricensis_GCF_900183385 genome ----\n",
      "2020-02-07 10:43:52,671 - Phyluce - INFO - Reading Listeria_costaricensis_GCF_900183385 genome\n",
      "2020-02-07 10:43:53,394 - Phyluce - INFO - Listeria_costaricensis_GCF_900183385: 203 uces, 18 dupes, 185 non-dupes, 0 orient drop, 1 length drop, 184 written\n",
      "2020-02-07 10:43:53,394 - Phyluce - INFO - ----- Working on Listeria_fleischmannii_GCF_000252625 genome ----\n",
      "2020-02-07 10:43:53,395 - Phyluce - INFO - Reading Listeria_fleischmannii_GCF_000252625 genome\n",
      "2020-02-07 10:43:54,126 - Phyluce - INFO - Listeria_fleischmannii_GCF_000252625: 208 uces, 19 dupes, 189 non-dupes, 0 orient drop, 0 length drop, 188 written\n",
      "2020-02-07 10:43:54,126 - Phyluce - INFO - ------ Working on Listeria_floridensis_GCF_000525875 genome -----\n",
      "2020-02-07 10:43:54,129 - Phyluce - INFO - Reading Listeria_floridensis_GCF_000525875 genome\n",
      "2020-02-07 10:43:54,909 - Phyluce - INFO - Listeria_floridensis_GCF_000525875: 211 uces, 24 dupes, 187 non-dupes, 0 orient drop, 0 length drop, 187 written\n",
      "2020-02-07 10:43:54,909 - Phyluce - INFO - ------- Working on Listeria_goaensis_GCF_900186125 genome -------\n",
      "2020-02-07 10:43:54,910 - Phyluce - INFO - Reading Listeria_goaensis_GCF_900186125 genome\n",
      "2020-02-07 10:43:55,657 - Phyluce - INFO - Listeria_goaensis_GCF_900186125: 205 uces, 19 dupes, 186 non-dupes, 0 orient drop, 0 length drop, 186 written\n",
      "2020-02-07 10:43:55,658 - Phyluce - INFO - ------ Working on Listeria_grandensis_GCF_000525835 genome ------\n",
      "2020-02-07 10:43:55,658 - Phyluce - INFO - Reading Listeria_grandensis_GCF_000525835 genome\n",
      "2020-02-07 10:43:56,451 - Phyluce - INFO - Listeria_grandensis_GCF_000525835: 210 uces, 28 dupes, 182 non-dupes, 0 orient drop, 1 length drop, 181 written\n",
      "2020-02-07 10:43:56,452 - Phyluce - INFO - --------- Working on Listeria_grayi_GCF_000148995 genome --------\n",
      "2020-02-07 10:43:56,462 - Phyluce - INFO - Reading Listeria_grayi_GCF_000148995 genome\n",
      "2020-02-07 10:43:57,293 - Phyluce - INFO - Listeria_grayi_GCF_000148995: 211 uces, 11 dupes, 200 non-dupes, 7 orient drop, 7 length drop, 186 written\n",
      "2020-02-07 10:43:57,293 - Phyluce - INFO - -------- Working on Listeria_innocua_GCF_000195795 genome -------\n",
      "2020-02-07 10:43:57,304 - Phyluce - INFO - Reading Listeria_innocua_GCF_000195795 genome\n",
      "2020-02-07 10:43:58,419 - Phyluce - INFO - Listeria_innocua_GCF_000195795: 213 uces, 2 dupes, 211 non-dupes, 39 orient drop, 9 length drop, 163 written\n",
      "2020-02-07 10:43:58,420 - Phyluce - INFO - ------- Working on Listeria_ivanovii_GCF_900637745 genome -------\n",
      "2020-02-07 10:43:58,421 - Phyluce - INFO - Reading Listeria_ivanovii_GCF_900637745 genome\n",
      "2020-02-07 10:43:59,517 - Phyluce - INFO - Listeria_ivanovii_GCF_900637745: 213 uces, 0 dupes, 213 non-dupes, 39 orient drop, 10 length drop, 164 written\n",
      "2020-02-07 10:43:59,517 - Phyluce - INFO - ------ Working on Listeria_kieliensis_GCF_003369925 genome ------\n",
      "2020-02-07 10:43:59,518 - Phyluce - INFO - Reading Listeria_kieliensis_GCF_003369925 genome\n",
      "2020-02-07 10:44:00,285 - Phyluce - INFO - Listeria_kieliensis_GCF_003369925: 209 uces, 17 dupes, 192 non-dupes, 5 orient drop, 2 length drop, 185 written\n",
      "2020-02-07 10:44:00,285 - Phyluce - INFO - -------- Working on Listeria_marthii_GCF_000183865 genome -------\n",
      "2020-02-07 10:44:00,286 - Phyluce - INFO - Reading Listeria_marthii_GCF_000183865 genome\n",
      "2020-02-07 10:44:00,984 - Phyluce - INFO - Listeria_marthii_GCF_000183865: 202 uces, 0 dupes, 202 non-dupes, 14 orient drop, 12 length drop, 176 written\n",
      "2020-02-07 10:44:00,984 - Phyluce - INFO - ----- Working on Listeria_newyorkensis_GCF_900461625 genome -----\n",
      "2020-02-07 10:44:00,991 - Phyluce - INFO - Reading Listeria_newyorkensis_GCF_900461625 genome\n",
      "2020-02-07 10:44:02,074 - Phyluce - INFO - Listeria_newyorkensis_GCF_900461625: 210 uces, 40 dupes, 170 non-dupes, 0 orient drop, 1 length drop, 169 written\n",
      "2020-02-07 10:44:02,074 - Phyluce - INFO - -------- Working on Listeria_riparia_GCF_000525995 genome -------\n",
      "2020-02-07 10:44:02,075 - Phyluce - INFO - Reading Listeria_riparia_GCF_000525995 genome\n",
      "2020-02-07 10:44:02,848 - Phyluce - INFO - Listeria_riparia_GCF_000525995: 210 uces, 23 dupes, 187 non-dupes, 0 orient drop, 2 length drop, 185 written\n",
      "2020-02-07 10:44:02,849 - Phyluce - INFO - ------ Working on Listeria_rocourtiae_GCF_000525975 genome ------\n",
      "2020-02-07 10:44:02,866 - Phyluce - INFO - Reading Listeria_rocourtiae_GCF_000525975 genome\n",
      "2020-02-07 10:44:03,644 - Phyluce - INFO - Listeria_rocourtiae_GCF_000525975: 211 uces, 20 dupes, 191 non-dupes, 0 orient drop, 0 length drop, 191 written\n",
      "2020-02-07 10:44:03,644 - Phyluce - INFO - ------- Working on Listeria_seeligeri_GCF_000027145 genome ------\n",
      "2020-02-07 10:44:03,645 - Phyluce - INFO - Reading Listeria_seeligeri_GCF_000027145 genome\n",
      "2020-02-07 10:44:04,733 - Phyluce - INFO - Listeria_seeligeri_GCF_000027145: 213 uces, 0 dupes, 213 non-dupes, 42 orient drop, 9 length drop, 162 written\n",
      "2020-02-07 10:44:04,733 - Phyluce - INFO - ---------- Working on Listeria_sp_GCF_003856515 genome ----------\n",
      "2020-02-07 10:44:04,734 - Phyluce - INFO - Reading Listeria_sp_GCF_003856515 genome\n",
      "2020-02-07 10:44:05,486 - Phyluce - INFO - Listeria_sp_GCF_003856515: 210 uces, 17 dupes, 193 non-dupes, 0 orient drop, 3 length drop, 190 written\n",
      "2020-02-07 10:44:05,486 - Phyluce - INFO - ----- Working on Listeria_thailandensis_GCF_900576925 genome ----\n",
      "2020-02-07 10:44:05,487 - Phyluce - INFO - Reading Listeria_thailandensis_GCF_900576925 genome\n",
      "2020-02-07 10:44:06,257 - Phyluce - INFO - Listeria_thailandensis_GCF_900576925: 208 uces, 21 dupes, 187 non-dupes, 2 orient drop, 0 length drop, 185 written\n",
      "2020-02-07 10:44:06,257 - Phyluce - INFO - -- Working on Listeria_weihenstephanensis_GCF_000525955 genome --\n",
      "2020-02-07 10:44:06,258 - Phyluce - INFO - Reading Listeria_weihenstephanensis_GCF_000525955 genome\n",
      "2020-02-07 10:44:07,067 - Phyluce - INFO - Listeria_weihenstephanensis_GCF_000525955: 211 uces, 30 dupes, 181 non-dupes, 0 orient drop, 1 length drop, 180 written\n",
      "2020-02-07 10:44:07,067 - Phyluce - INFO - ------ Working on Listeria_welshimeri_GCF_900187315 genome ------\n",
      "2020-02-07 10:44:07,068 - Phyluce - INFO - Reading Listeria_welshimeri_GCF_900187315 genome\n",
      "2020-02-07 10:44:08,170 - Phyluce - INFO - Listeria_welshimeri_GCF_900187315: 213 uces, 0 dupes, 213 non-dupes, 41 orient drop, 11 length drop, 161 written\n",
      "2020-02-07 10:44:08,170 - Phyluce - INFO - ----- Working on Listeria_monocytogenes_GCF_000196035 genome ----\n",
      "2020-02-07 10:44:08,172 - Phyluce - INFO - Reading Listeria_monocytogenes_GCF_000196035 genome\n",
      "2020-02-07 10:44:09,271 - Phyluce - INFO - Listeria_monocytogenes_GCF_000196035: 213 uces, 0 dupes, 213 non-dupes, 40 orient drop, 11 length drop, 162 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/listeria/extract_probes_from_group/probe_fasta --output results/phyluce/listeria/extract_probes_from_group/multifastas.sqlite --base-taxon Listeria_monocytogenes_GCF_000196035;\n",
      "listeria_aquatica_gcf_000525795.\n",
      "listeria_booriae_gcf_000766865.\n",
      "listeria_cornellensis_gcf_000525855.\n",
      "listeria_costaricensis_gcf_900183385.\n",
      "listeria_fleischmannii_gcf_000252625.\n",
      "listeria_floridensis_gcf_000525875.\n",
      "listeria_goaensis_gcf_900186125.\n",
      "listeria_grandensis_gcf_000525835.\n",
      "listeria_grayi_gcf_000148995.\n",
      "listeria_innocua_gcf_000195795.\n",
      "listeria_ivanovii_gcf_900637745.\n",
      "listeria_kieliensis_gcf_003369925.\n",
      "listeria_marthii_gcf_000183865.\n",
      "listeria_newyorkensis_gcf_900461625.\n",
      "listeria_riparia_gcf_000525995.\n",
      "listeria_rocourtiae_gcf_000525975.\n",
      "listeria_seeligeri_gcf_000027145.\n",
      "listeria_sp_gcf_003856515.\n",
      "listeria_thailandensis_gcf_900576925.\n",
      "listeria_weihenstephanensis_gcf_000525955.\n",
      "listeria_welshimeri_gcf_900187315.\n",
      "listeria_monocytogenes_gcf_000196035.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/listeria/extract_probes_from_group/multifastas.sqlite --base-taxon Listeria_monocytogenes_GCF_000196035;\n",
      "Loci shared by 0 taxa:\t208.0\n",
      "Loci shared by 1 taxa:\t208.0\n",
      "Loci shared by 2 taxa:\t204.0\n",
      "Loci shared by 3 taxa:\t202.0\n",
      "Loci shared by 4 taxa:\t202.0\n",
      "Loci shared by 5 taxa:\t200.0\n",
      "Loci shared by 6 taxa:\t200.0\n",
      "Loci shared by 7 taxa:\t198.0\n",
      "Loci shared by 8 taxa:\t196.0\n",
      "Loci shared by 9 taxa:\t194.0\n",
      "Loci shared by 10 taxa:\t192.0\n",
      "Loci shared by 11 taxa:\t189.0\n",
      "Loci shared by 12 taxa:\t184.0\n",
      "Loci shared by 13 taxa:\t181.0\n",
      "Loci shared by 14 taxa:\t177.0\n",
      "Loci shared by 15 taxa:\t175.0\n",
      "Loci shared by 16 taxa:\t163.0\n",
      "Loci shared by 17 taxa:\t157.0\n",
      "Loci shared by 18 taxa:\t152.0\n",
      "Loci shared by 19 taxa:\t150.0\n",
      "Loci shared by 20 taxa:\t145.0\n",
      "Loci shared by 21 taxa:\t144.0\n",
      "Loci shared by 22 taxa:\t131.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/listeria/extract_probes_from_group/multifastas.sqlite  --base-taxon Listeria_monocytogenes_GCF_000196035 --output results/phyluce/listeria/extract_probes_from_group/Listeria_monocytogenes_GCF_000196035+6-back-to-20.conf --specific-counts 20;\n",
      "Counter({'listeria_goaensis_gcf_900186125': 145, 'listeria_floridensis_gcf_000525875': 145, 'listeria_fleischmannii_gcf_000252625': 145, 'listeria_welshimeri_gcf_900187315': 145, 'listeria_aquatica_gcf_000525795': 145, 'listeria_cornellensis_gcf_000525855': 145, 'listeria_weihenstephanensis_gcf_000525955': 145, 'listeria_kieliensis_gcf_003369925': 145, 'listeria_thailandensis_gcf_900576925': 145, 'listeria_rocourtiae_gcf_000525975': 145, 'listeria_grayi_gcf_000148995': 145, 'listeria_riparia_gcf_000525995': 145, 'listeria_sp_gcf_003856515': 145, 'listeria_monocytogenes_gcf_000196035': 145, 'listeria_newyorkensis_gcf_900461625': 145, 'listeria_seeligeri_gcf_000027145': 144, 'listeria_ivanovii_gcf_900637745': 144, 'listeria_booriae_gcf_000766865': 144, 'listeria_grandensis_gcf_000525835': 144, 'listeria_innocua_gcf_000195795': 143, 'listeria_costaricensis_gcf_900183385': 141, 'listeria_marthii_gcf_000183865': 140})\n",
      "Total loci = 145\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(20)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/listeria/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/listeria/extract_probes_from_group/Listeria_monocytogenes_GCF_000196035+6-back-to-20.conf --probe-prefix uce_listeria_ --designer rnplattii --design listeria_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGNNGGGGGNGGGNNNNNGGGGGGNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 145\n",
      "Probe Count = 6264\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list.fasta --query results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  10:44:32\n",
      "Ended:  Fri Feb 07, 2020  10:44:47\n",
      "Time for execution:  0.244520934423 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list.fasta --lastz results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_listeria_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6263 fasta sequences.  Filtered 2 duplicates. Kept 6180.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/listeria/final_probe_design/listeria_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 10:50:30 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 6264\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 501120\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3954331\n",
      "Max number of word counting entries: 89011995\n",
      "\n",
      "# comparing sequences from          0  to       1044\n",
      ".---------- new table with      645 representatives\n",
      "# comparing sequences from       1044  to       1914\n",
      "----------    386 remaining sequences to the next cycle\n",
      "---------- new table with      299 representatives\n",
      "# comparing sequences from       1528  to       2317\n",
      "----------    326 remaining sequences to the next cycle\n",
      "---------- new table with      264 representatives\n",
      "# comparing sequences from       1991  to       2703\n",
      "----------    335 remaining sequences to the next cycle\n",
      "---------- new table with      149 representatives\n",
      "# comparing sequences from       2368  to       3017\n",
      "----------    344 remaining sequences to the next cycle\n",
      "---------- new table with      131 representatives\n",
      "# comparing sequences from       2673  to       3271\n",
      "----------    194 remaining sequences to the next cycle\n",
      "---------- new table with      225 representatives\n",
      "# comparing sequences from       3077  to       3608\n",
      "----------    252 remaining sequences to the next cycle\n",
      "---------- new table with      141 representatives\n",
      "# comparing sequences from       3356  to       3840\n",
      "----------    132 remaining sequences to the next cycle\n",
      "---------- new table with      201 representatives\n",
      "# comparing sequences from       3708  to       4134\n",
      "----------    207 remaining sequences to the next cycle\n",
      "---------- new table with      152 representatives\n",
      "# comparing sequences from       3927  to       4316\n",
      "----------    172 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       4144  to       4497\n",
      "----------     19 remaining sequences to the next cycle\n",
      "---------- new table with      148 representatives\n",
      "# comparing sequences from       4478  to       4775\n",
      "----------    110 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       4665  to       4931\n",
      "----------    106 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       4825  to       5064\n",
      "----------     66 remaining sequences to the next cycle\n",
      "---------- new table with      106 representatives\n",
      "# comparing sequences from       4998  to       5209\n",
      "...................---------- new table with       86 representatives\n",
      "# comparing sequences from       5209  to       5384\n",
      "100.0%---------- new table with       91 representatives\n",
      "# comparing sequences from       5384  to       6264\n",
      ".....................---------- new table with      507 representatives\n",
      "\n",
      "     6264  finished       3445  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.38\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "319px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
