{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import shutil\n",
    "from shutil import copy\n",
    "import time\n",
    "from collections import defaultdict\n",
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "os.chdir(\"/master/nplatt/pathogen_probes/\")\n",
    "\n",
    "\n",
    "def wait_on_running_jobs():\n",
    "   \n",
    "    num_jobs = 1\n",
    "    \n",
    "    while num_jobs > 0:\n",
    "        num_jobs = len(subprocess.check_output('qstat', shell=True).split(\"\\n\")) - 2\n",
    "        time.sleep(60)\n",
    "        print(\".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clade 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'nematoda-clade1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Romanomermis_culicivorax_GCA_001039655\",\n",
    "               \"Trichinella_britovi_GCA_001447585\",\n",
    "               \"Trichinella_murrelli_GCA_002221485\",\n",
    "               \"Trichinella_nativa_GCA_001447565\",\n",
    "               \"Trichinella_nelsoni_GCA_001447455\",\n",
    "               \"Trichinella_papuae_GCA_001447755\",\n",
    "               \"Trichinella_patagoniensis_GCA_001447655\",\n",
    "               \"Trichinella_pseudospralis_GCA_001447645\",\n",
    "               \"Trichinella_sp_GCA_001447745\",\n",
    "               \"Trichinella_zimbabwensis_GCA_001447665\",\n",
    "               \"Soboliphyme_baturini_GCA_900618415\" ]\n",
    "                    \n",
    "reference_taxon = \"Trichinella_spiralis_GCA_000181795\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001039655.1_nRc.2.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 91333130 bytes  8698397.33 bytes/sec\n",
      "total size is 91310731  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447585.1_T3_ISS120_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16427426 bytes  6570987.20 bytes/sec\n",
      "total size is 16423300  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002221485.1_ASM222148v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 19176830 bytes  5479106.29 bytes/sec\n",
      "total size is 19172035  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447565.1_T2_ISS10_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15390515 bytes  4397302.00 bytes/sec\n",
      "total size is 15386646  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447455.1_T7_ISS37_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15152254 bytes  6060918.40 bytes/sec\n",
      "total size is 15148441  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447755.1_T10_ISS1980_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 14986146 bytes  4281768.00 bytes/sec\n",
      "total size is 14982370  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447655.1_T12_ISS2496_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15964294 bytes  6385734.40 bytes/sec\n",
      "total size is 15960278  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447645.1_T4_ISS176_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15412248 bytes  6164916.00 bytes/sec\n",
      "total size is 15408370  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447745.1_T8_ISS272_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15753623 bytes  4501047.14 bytes/sec\n",
      "total size is 15749665  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000181795.3_ASM18179v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 18782658 bytes  5366485.71 bytes/sec\n",
      "total size is 18777960  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001447665.1_T11_ISS1029_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16144165 bytes  4612630.57 bytes/sec\n",
      "total size is 16140109  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900618415.1_S_baturini_Dall_Island_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 70050859 bytes  6091382.70 bytes/sec\n",
      "total size is 70033628  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486908 (\"sim_Romanomermis_culicivorax_GCA_001039655\") has been submitted\n",
      "Your job 5486909 (\"sim_Trichinella_britovi_GCA_001447585\") has been submitted\n",
      "Your job 5486910 (\"sim_Trichinella_murrelli_GCA_002221485\") has been submitted\n",
      "Your job 5486911 (\"sim_Trichinella_nativa_GCA_001447565\") has been submitted\n",
      "Your job 5486912 (\"sim_Trichinella_nelsoni_GCA_001447455\") has been submitted\n",
      "Your job 5486913 (\"sim_Trichinella_papuae_GCA_001447755\") has been submitted\n",
      "Your job 5486914 (\"sim_Trichinella_patagoniensis_GCA_001447655\") has been submitted\n",
      "Your job 5486915 (\"sim_Trichinella_pseudospralis_GCA_001447645\") has been submitted\n",
      "Your job 5486916 (\"sim_Trichinella_sp_GCA_001447745\") has been submitted\n",
      "Your job 5486917 (\"sim_Trichinella_zimbabwensis_GCA_001447665\") has been submitted\n",
      "Your job 5486918 (\"sim_Soboliphyme_baturini_GCA_900618415\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/nematoda-clade1/cleaned_genomes/Trichinella_spiralis_GCA_000181795_formatted.fas path=results/phyluce/nematoda-clade1/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486919 (\"map_Romanomermis_culicivorax_GCA_001039655\") has been submitted\n",
      "Your job 5486920 (\"map_Trichinella_britovi_GCA_001447585\") has been submitted\n",
      "Your job 5486921 (\"map_Trichinella_murrelli_GCA_002221485\") has been submitted\n",
      "Your job 5486922 (\"map_Trichinella_nativa_GCA_001447565\") has been submitted\n",
      "Your job 5486923 (\"map_Trichinella_nelsoni_GCA_001447455\") has been submitted\n",
      "Your job 5486924 (\"map_Trichinella_papuae_GCA_001447755\") has been submitted\n",
      "Your job 5486925 (\"map_Trichinella_patagoniensis_GCA_001447655\") has been submitted\n",
      "Your job 5486926 (\"map_Trichinella_pseudospralis_GCA_001447645\") has been submitted\n",
      "Your job 5486927 (\"map_Trichinella_sp_GCA_001447745\") has been submitted\n",
      "Your job 5486928 (\"map_Trichinella_zimbabwensis_GCA_001447665\") has been submitted\n",
      "Your job 5486929 (\"map_Soboliphyme_baturini_GCA_900618415\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486930 (\"merge_Romanomermis_culicivorax_GCA_001039655\") has been submitted\n",
      "Your job 5486931 (\"merge_Trichinella_britovi_GCA_001447585\") has been submitted\n",
      "Your job 5486932 (\"merge_Trichinella_murrelli_GCA_002221485\") has been submitted\n",
      "Your job 5486933 (\"merge_Trichinella_nativa_GCA_001447565\") has been submitted\n",
      "Your job 5486934 (\"merge_Trichinella_nelsoni_GCA_001447455\") has been submitted\n",
      "Your job 5486935 (\"merge_Trichinella_papuae_GCA_001447755\") has been submitted\n",
      "Your job 5486936 (\"merge_Trichinella_patagoniensis_GCA_001447655\") has been submitted\n",
      "Your job 5486937 (\"merge_Trichinella_pseudospralis_GCA_001447645\") has been submitted\n",
      "Your job 5486938 (\"merge_Trichinella_sp_GCA_001447745\") has been submitted\n",
      "Your job 5486939 (\"merge_Trichinella_zimbabwensis_GCA_001447665\") has been submitted\n",
      "Your job 5486940 (\"merge_Soboliphyme_baturini_GCA_900618415\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 43 sequences from Romanomermis_culicivorax_GCA_001039655_merged.bed.  Filtered 35 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 54380 sequences from Trichinella_britovi_GCA_001447585_merged.bed.  Filtered 43771 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10609.\n",
      "Screened 54970 sequences from Trichinella_murrelli_GCA_002221485_merged.bed.  Filtered 44133 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10837.\n",
      "Screened 54561 sequences from Trichinella_nativa_GCA_001447565_merged.bed.  Filtered 43886 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10675.\n",
      "Screened 53128 sequences from Trichinella_nelsoni_GCA_001447455_merged.bed.  Filtered 42986 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10142.\n",
      "Screened 95574 sequences from Trichinella_papuae_GCA_001447755_merged.bed.  Filtered 69845 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 25729.\n",
      "Screened 54336 sequences from Trichinella_patagoniensis_GCA_001447655_merged.bed.  Filtered 43775 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10561.\n",
      "Screened 87687 sequences from Trichinella_pseudospralis_GCA_001447645_merged.bed.  Filtered 60185 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 27502.\n",
      "Screened 53706 sequences from Trichinella_sp_GCA_001447745_merged.bed.  Filtered 43207 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10499.\n",
      "Screened 96113 sequences from Trichinella_zimbabwensis_GCA_001447665_merged.bed.  Filtered 70333 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 25780.\n",
      "Screened 31 sequences from Soboliphyme_baturini_GCA_900618415_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/nematoda-clade1/initial_intervals/Soboliphyme_baturini_GCA_900618415_merged.bed --twobit results/phyluce/nematoda-clade1/cleaned_genomes/Trichinella_spiralis_GCA_000181795_formatted.2bit --output results/phyluce/nematoda-clade1/initial_intervals/Soboliphyme_baturini_GCA_900618415_stripped.bed;\n",
      "romanomermis_culicivorax_gca_001039655.\n",
      "trichinella_britovi_gca_001447585...........\n",
      "trichinella_murrelli_gca_002221485...........\n",
      "trichinella_nativa_gca_001447565...........\n",
      "trichinella_nelsoni_gca_001447455...........\n",
      "trichinella_papuae_gca_001447755..........................\n",
      "trichinella_patagoniensis_gca_001447655...........\n",
      "trichinella_pseudospralis_gca_001447645............................\n",
      "trichinella_sp_gca_001447745...........\n",
      "trichinella_zimbabwensis_gca_001447665..........................\n",
      "soboliphyme_baturini_gca_900618415.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade1/initial_intervals/nematoda-clade1-to-Trichinella_spiralis_GCA_000181795.sqlite --base-taxon Trichinella_spiralis_GCA_000181795\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 0 taxa:\t38,179.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 1 taxa:\t38,179.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 2 taxa:\t28,986.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 3 taxa:\t22,754.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 4 taxa:\t12,320.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 5 taxa:\t9,391.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 6 taxa:\t6,992.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 7 taxa:\t4,120.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 8 taxa:\t3,029.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 9 taxa:\t1,845.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 10 taxa:\t9.0\n",
      "Loci shared by Trichinella_spiralis_GCA_000181795 + 11 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade1/initial_intervals/nematoda-clade1-to-Trichinella_spiralis_GCA_000181795.sqlite --base-taxon Trichinella_spiralis_GCA_000181795 --output results/phyluce/nematoda-clade1/initial_intervals/Trichinella_spiralis_GCA_000181795_+9.bed --specific-counts 9;\n",
      "Counter({'trichinella_papuae_gca_001447755': 1845, 'trichinella_patagoniensis_gca_001447655': 1845, 'trichinella_sp_gca_001447745': 1845, 'trichinella_murrelli_gca_002221485': 1845, 'trichinella_nelsoni_gca_001447455': 1845, 'trichinella_nativa_gca_001447565': 1845, 'trichinella_zimbabwensis_gca_001447665': 1845, 'trichinella_pseudospralis_gca_001447645': 1845, 'trichinella_britovi_gca_001447585': 1845, 'romanomermis_culicivorax_gca_001039655': 6, 'soboliphyme_baturini_gca_900618415': 3})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 9\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/nematoda-clade1/initial_intervals/Trichinella_spiralis_GCA_000181795_+9.bed --twobit results/phyluce/nematoda-clade1/cleaned_genomes/Trichinella_spiralis_GCA_000181795_formatted.2bit --buffer-to 160 --output results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9.fasta;\n",
      "Screened 1845 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 1845.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9.fasta --probe-prefix uce_nematoda-clade1_ --design nematoda-clade1_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 1790\n",
      "Probe Count = 3482\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas --query results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  08:40:50\n",
      "Ended:  Tue Feb 11, 2020  08:40:52\n",
      "Time for execution:  0.0266137997309 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas                        --lastz results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes_vself.lastz                       --probe-prefix=uce_nematoda-clade1_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 3481 fasta sequences.  Filtered 139 duplicates. Kept 3204.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/nematoda-clade1/validate_intervals/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas --scaffoldlist Romanomermis_culicivorax_GCA_001039655 Trichinella_britovi_GCA_001447585 Trichinella_murrelli_GCA_002221485 Trichinella_nativa_GCA_001447565 Trichinella_nelsoni_GCA_001447455 Trichinella_papuae_GCA_001447755 Trichinella_patagoniensis_GCA_001447655 Trichinella_pseudospralis_GCA_001447645 Trichinella_sp_GCA_001447745 Trichinella_zimbabwensis_GCA_001447665 Soboliphyme_baturini_GCA_900618415 Trichinella_spiralis_GCA_000181795 --genome-base-path results/phyluce/nematoda-clade1/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/nematoda-clade1/validate_intervals/nematoda-clade1-to-Trichinella_spiralis_GCA_000181795.sqlite --output results/phyluce/nematoda-clade1/validate_intervals/lastz/;\n",
      "\n",
      "Running against Romanomermis_culicivorax_GCA_001039655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 33 queries...\n",
      "\t/tmp/tmpfuf6R2.fasta\n",
      "\t/tmp/tmpGQq4BC.fasta\n",
      "\t/tmp/tmp5cRPYf.fasta\n",
      "\t/tmp/tmp5fKalP.fasta\n",
      "\t/tmp/tmpYaAi45.fasta\n",
      "\t/tmp/tmpYYebK2.fasta\n",
      "\t/tmp/tmpEBlzaW.fasta\n",
      "\t/tmp/tmpf25wy7.fasta\n",
      "\t/tmp/tmpQR5vKw.fasta\n",
      "\t/tmp/tmpwhJK6z.fasta\n",
      "\t/tmp/tmpHtsoBz.fasta\n",
      "\t/tmp/tmpSkYKRY.fasta\n",
      "\t/tmp/tmpSBlgcQ.fasta\n",
      "\t/tmp/tmp7FwVzR.fasta\n",
      "\t/tmp/tmpZJDNMr.fasta\n",
      "\t/tmp/tmpgr5jKT.fasta\n",
      "\t/tmp/tmpur6hJ1.fasta\n",
      "\t/tmp/tmp55tFaA.fasta\n",
      "\t/tmp/tmpq2oT3E.fasta\n",
      "\t/tmp/tmpn7yXTG.fasta\n",
      "\t/tmp/tmpFlV1CD.fasta\n",
      "\t/tmp/tmpzYgAcK.fasta\n",
      "\t/tmp/tmpjcnE_F.fasta\n",
      "\t/tmp/tmp34RaSm.fasta\n",
      "\t/tmp/tmp1UBWde.fasta\n",
      "\t/tmp/tmpTJ_zCo.fasta\n",
      "\t/tmp/tmpVNyzio.fasta\n",
      "\t/tmp/tmp9uuGzk.fasta\n",
      "\t/tmp/tmpmUOQFk.fasta\n",
      "\t/tmp/tmpS6eujg.fasta\n",
      "\t/tmp/tmpbPgUHv.fasta\n",
      "\t/tmp/tmpsCJ_Pj.fasta\n",
      "\t/tmp/tmpO2NXaj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpc5gyJ1.lastz\n",
      "\t/tmp/tmpl6J7Iw.lastz\n",
      "\t/tmp/tmp_kdL8y.lastz\n",
      "\t/tmp/tmpdBOUN4.lastz\n",
      "\t/tmp/tmpYsOKwS.lastz\n",
      "\t/tmp/tmpGuGpkT.lastz\n",
      "\t/tmp/tmpZK8tEn.lastz\n",
      "\t/tmp/tmpfHicql.lastz\n",
      "\t/tmp/tmpvPk1iH.lastz\n",
      "\t/tmp/tmppbpQcB.lastz\n",
      "\t/tmp/tmpfy1k30.lastz\n",
      "\t/tmp/tmpl9ezUR.lastz\n",
      "\t/tmp/tmpeMhTro.lastz\n",
      "\t/tmp/tmp2hhgKf.lastz\n",
      "\t/tmp/tmpnWxKkZ.lastz\n",
      "\t/tmp/tmp4JnHiG.lastz\n",
      "\t/tmp/tmpRdhh4W.lastz\n",
      "\t/tmp/tmp2iIOEt.lastz\n",
      "\t/tmp/tmpfj4Mj0.lastz\n",
      "\t/tmp/tmpP8OXbx.lastz\n",
      "\t/tmp/tmpe7TviB.lastz\n",
      "\t/tmp/tmpWOkPQW.lastz\n",
      "\t/tmp/tmp5j0Dqs.lastz\n",
      "\t/tmp/tmppx0Hj2.lastz\n",
      "\t/tmp/tmpan6trS.lastz\n",
      "\t/tmp/tmpB49sZI.lastz\n",
      "\t/tmp/tmpYQ1EQH.lastz\n",
      "\t/tmp/tmpImCuRB.lastz\n",
      "\t/tmp/tmpqw1Koj.lastz\n",
      "\t/tmp/tmpa3nMsG.lastz\n",
      "\t/tmp/tmpcdxmCa.lastz\n",
      "\t/tmp/tmpfG9F_i.lastz\n",
      "\t/tmp/tmpoJ0haa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Romanomermis_culicivorax_GCA_001039655.lastz\n",
      "Creating Romanomermis_culicivorax_GCA_001039655 table\n",
      "Inserting data to Romanomermis_culicivorax_GCA_001039655 table\n",
      "\n",
      "Running against Trichinella_britovi_GCA_001447585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmp7NEt_3.fasta\n",
      "\t/tmp/tmpUQDcCL.fasta\n",
      "\t/tmp/tmpc9f_Hq.fasta\n",
      "\t/tmp/tmpZeNyjx.fasta\n",
      "\t/tmp/tmpPVSgeQ.fasta\n",
      "\t/tmp/tmpsNYdv6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiBS6xL.lastz\n",
      "\t/tmp/tmpWNC7qR.lastz\n",
      "\t/tmp/tmpbPKesz.lastz\n",
      "\t/tmp/tmp6gZNxb.lastz\n",
      "\t/tmp/tmpeAtfUl.lastz\n",
      "\t/tmp/tmpnMMBAQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_britovi_GCA_001447585.lastz\n",
      "Creating Trichinella_britovi_GCA_001447585 table\n",
      "Inserting data to Trichinella_britovi_GCA_001447585 table\n",
      "\n",
      "Running against Trichinella_murrelli_GCA_002221485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpxYo2n8.fasta\n",
      "\t/tmp/tmpiospoO.fasta\n",
      "\t/tmp/tmplFXzmI.fasta\n",
      "\t/tmp/tmpnIwhl_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDfxUCe.lastz\n",
      "\t/tmp/tmpP40BUP.lastz\n",
      "\t/tmp/tmpKk3LpO.lastz\n",
      "\t/tmp/tmpHbueO0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_murrelli_GCA_002221485.lastz\n",
      "Creating Trichinella_murrelli_GCA_002221485 table\n",
      "Inserting data to Trichinella_murrelli_GCA_002221485 table\n",
      "\n",
      "Running against Trichinella_nativa_GCA_001447565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmppoa5RM.fasta\n",
      "\t/tmp/tmptyYwWR.fasta\n",
      "\t/tmp/tmpMa11fr.fasta\n",
      "\t/tmp/tmpUv2ohL.fasta\n",
      "\t/tmp/tmpf3rWBT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDWvBu0.lastz\n",
      "\t/tmp/tmpOWSdCh.lastz\n",
      "\t/tmp/tmpyp8v_j.lastz\n",
      "\t/tmp/tmpXjXBL8.lastz\n",
      "\t/tmp/tmpFNgSSO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_nativa_GCA_001447565.lastz\n",
      "Creating Trichinella_nativa_GCA_001447565 table\n",
      "Inserting data to Trichinella_nativa_GCA_001447565 table\n",
      "\n",
      "Running against Trichinella_nelsoni_GCA_001447455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmp8E5NhT.fasta\n",
      "\t/tmp/tmp9PZVaR.fasta\n",
      "\t/tmp/tmp9og_qJ.fasta\n",
      "\t/tmp/tmpvWJOlp.fasta\n",
      "\t/tmp/tmpoQlurp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpntfZDr.lastz\n",
      "\t/tmp/tmpOCQ6c3.lastz\n",
      "\t/tmp/tmpEP80yA.lastz\n",
      "\t/tmp/tmptuQQ_8.lastz\n",
      "\t/tmp/tmpZzeqBQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_nelsoni_GCA_001447455.lastz\n",
      "Creating Trichinella_nelsoni_GCA_001447455 table\n",
      "Inserting data to Trichinella_nelsoni_GCA_001447455 table\n",
      "\n",
      "Running against Trichinella_papuae_GCA_001447755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpotrT4_.fasta\n",
      "\t/tmp/tmpLpr0YQ.fasta\n",
      "\t/tmp/tmpxtrj6g.fasta\n",
      "\t/tmp/tmpny_jiZ.fasta\n",
      "\t/tmp/tmpMsz9eK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5u3caF.lastz\n",
      "\t/tmp/tmpfO7kq2.lastz\n",
      "\t/tmp/tmp210OQV.lastz\n",
      "\t/tmp/tmporf26N.lastz\n",
      "\t/tmp/tmpgey7nc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_papuae_GCA_001447755.lastz\n",
      "Creating Trichinella_papuae_GCA_001447755 table\n",
      "Inserting data to Trichinella_papuae_GCA_001447755 table\n",
      "\n",
      "Running against Trichinella_patagoniensis_GCA_001447655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmp5tybmI.fasta\n",
      "\t/tmp/tmpO7dLlu.fasta\n",
      "\t/tmp/tmp3SN7s_.fasta\n",
      "\t/tmp/tmpBEWmXP.fasta\n",
      "\t/tmp/tmpwuVEAb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6G3IKj.lastz\n",
      "\t/tmp/tmpwUNzfp.lastz\n",
      "\t/tmp/tmpf1DSg_.lastz\n",
      "\t/tmp/tmplRwZpL.lastz\n",
      "\t/tmp/tmpZPODZ1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_patagoniensis_GCA_001447655.lastz\n",
      "Creating Trichinella_patagoniensis_GCA_001447655 table\n",
      "Inserting data to Trichinella_patagoniensis_GCA_001447655 table\n",
      "\n",
      "Running against Trichinella_pseudospralis_GCA_001447645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpYcdgGT.fasta\n",
      "\t/tmp/tmpQrgDvR.fasta\n",
      "\t/tmp/tmpDRdHxr.fasta\n",
      "\t/tmp/tmpGewLxY.fasta\n",
      "\t/tmp/tmpPNaqpj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprzRYUy.lastz\n",
      "\t/tmp/tmpFGuqAF.lastz\n",
      "\t/tmp/tmptVsiHO.lastz\n",
      "\t/tmp/tmpgyqq54.lastz\n",
      "\t/tmp/tmpsqzxqC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_pseudospralis_GCA_001447645.lastz\n",
      "Creating Trichinella_pseudospralis_GCA_001447645 table\n",
      "Inserting data to Trichinella_pseudospralis_GCA_001447645 table\n",
      "\n",
      "Running against Trichinella_sp_GCA_001447745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpyZOwfs.fasta\n",
      "\t/tmp/tmpMfASHk.fasta\n",
      "\t/tmp/tmp43valE.fasta\n",
      "\t/tmp/tmpAdo8Vs.fasta\n",
      "\t/tmp/tmpWjubzF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPb2jqZ.lastz\n",
      "\t/tmp/tmp1OdhtZ.lastz\n",
      "\t/tmp/tmpRWnAkb.lastz\n",
      "\t/tmp/tmpChxl2G.lastz\n",
      "\t/tmp/tmpu5pCWY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_sp_GCA_001447745.lastz\n",
      "Creating Trichinella_sp_GCA_001447745 table\n",
      "Inserting data to Trichinella_sp_GCA_001447745 table\n",
      "\n",
      "Running against Trichinella_zimbabwensis_GCA_001447665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpsC2lKT.fasta\n",
      "\t/tmp/tmphOYXu9.fasta\n",
      "\t/tmp/tmpji2w7S.fasta\n",
      "\t/tmp/tmpDSTUXx.fasta\n",
      "\t/tmp/tmp5dJ3eJ.fasta\n",
      "\t/tmp/tmp3fWVHK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7W80Tp.lastz\n",
      "\t/tmp/tmpJPTF01.lastz\n",
      "\t/tmp/tmpRrF7eK.lastz\n",
      "\t/tmp/tmpooaHf3.lastz\n",
      "\t/tmp/tmphmiUEi.lastz\n",
      "\t/tmp/tmp9EaGoF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_zimbabwensis_GCA_001447665.lastz\n",
      "Creating Trichinella_zimbabwensis_GCA_001447665 table\n",
      "Inserting data to Trichinella_zimbabwensis_GCA_001447665 table\n",
      "\n",
      "Running against Soboliphyme_baturini_GCA_900618415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 22 queries...\n",
      "\t/tmp/tmp73iNxi.fasta\n",
      "\t/tmp/tmp7NCPrf.fasta\n",
      "\t/tmp/tmptE7NLR.fasta\n",
      "\t/tmp/tmpgukJMp.fasta\n",
      "\t/tmp/tmp1_P7O6.fasta\n",
      "\t/tmp/tmph9huDN.fasta\n",
      "\t/tmp/tmpw6WnPt.fasta\n",
      "\t/tmp/tmphhzi7x.fasta\n",
      "\t/tmp/tmpuX9aIR.fasta\n",
      "\t/tmp/tmpPF7JSj.fasta\n",
      "\t/tmp/tmpRbLTdZ.fasta\n",
      "\t/tmp/tmp4frqAL.fasta\n",
      "\t/tmp/tmpzIcBCT.fasta\n",
      "\t/tmp/tmpWO4S_K.fasta\n",
      "\t/tmp/tmpCGBm1w.fasta\n",
      "\t/tmp/tmpuEAP0w.fasta\n",
      "\t/tmp/tmp0qGDR4.fasta\n",
      "\t/tmp/tmpZ0wPJ5.fasta\n",
      "\t/tmp/tmp9yOFDW.fasta\n",
      "\t/tmp/tmpVCuPxo.fasta\n",
      "\t/tmp/tmp52ugUw.fasta\n",
      "\t/tmp/tmp5LaXkh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJF049r.lastz\n",
      "\t/tmp/tmp_ZsEEk.lastz\n",
      "\t/tmp/tmpiCVufC.lastz\n",
      "\t/tmp/tmpw1EKBj.lastz\n",
      "\t/tmp/tmpcyZSNR.lastz\n",
      "\t/tmp/tmphcqMMP.lastz\n",
      "\t/tmp/tmppMx8gv.lastz\n",
      "\t/tmp/tmpcCm9Jm.lastz\n",
      "\t/tmp/tmp4T2C61.lastz\n",
      "\t/tmp/tmpcsAKWS.lastz\n",
      "\t/tmp/tmpodK4A8.lastz\n",
      "\t/tmp/tmpwH_Xk4.lastz\n",
      "\t/tmp/tmpx88mWK.lastz\n",
      "\t/tmp/tmp2AL2kH.lastz\n",
      "\t/tmp/tmp6C7gff.lastz\n",
      "\t/tmp/tmpUU9coZ.lastz\n",
      "\t/tmp/tmpC8MgYG.lastz\n",
      "\t/tmp/tmplPGhqR.lastz\n",
      "\t/tmp/tmpMPj9bl.lastz\n",
      "\t/tmp/tmpOeYC_j.lastz\n",
      "\t/tmp/tmpiw1Tp8.lastz\n",
      "\t/tmp/tmpIOa_Hn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Soboliphyme_baturini_GCA_900618415.lastz\n",
      "Creating Soboliphyme_baturini_GCA_900618415 table\n",
      "Inserting data to Soboliphyme_baturini_GCA_900618415 table\n",
      "\n",
      "Running against Trichinella_spiralis_GCA_000181795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpFRUafR.fasta\n",
      "\t/tmp/tmpUiXZPC.fasta\n",
      "\t/tmp/tmpAHhTZM.fasta\n",
      "\t/tmp/tmp69naEK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdwTfbq.lastz\n",
      "\t/tmp/tmpZGI9oB.lastz\n",
      "\t/tmp/tmpV7ODVE.lastz\n",
      "\t/tmp/tmp94IHkD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade1/validate_intervals/lastz/Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_Trichinella_spiralis_GCA_000181795.lastz\n",
      "Creating Trichinella_spiralis_GCA_000181795 table\n",
      "Inserting data to Trichinella_spiralis_GCA_000181795 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/nematoda-clade1/extract_probes_from_group/nematoda-clade1_genome.conf --lastz results/phyluce/nematoda-clade1/validate_intervals/lastz --probes 120 --probe-prefix uce_nematoda-clade1_ --name-pattern \"Trichinella_spiralis_GCA_000181795_+9_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/nematoda-clade1/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 08:45:14,967 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 08:45:14,969 - Phyluce - INFO - ---- Working on Romanomermis_culicivorax_GCA_001039655 genome ---\n",
      "2020-02-11 08:45:14,970 - Phyluce - INFO - Reading Romanomermis_culicivorax_GCA_001039655 genome\n",
      "2020-02-11 08:45:17,654 - Phyluce - INFO - Romanomermis_culicivorax_GCA_001039655: 185 uces, 68 dupes, 117 non-dupes, 1 orient drop, 2 length drop, 114 written\n",
      "2020-02-11 08:45:17,654 - Phyluce - INFO - ------ Working on Trichinella_britovi_GCA_001447585 genome ------\n",
      "2020-02-11 08:45:17,666 - Phyluce - INFO - Reading Trichinella_britovi_GCA_001447585 genome\n",
      "2020-02-11 08:45:38,894 - Phyluce - INFO - Trichinella_britovi_GCA_001447585: 1785 uces, 389 dupes, 1396 non-dupes, 2 orient drop, 15 length drop, 1379 written\n",
      "2020-02-11 08:45:38,894 - Phyluce - INFO - ------ Working on Trichinella_murrelli_GCA_002221485 genome -----\n",
      "2020-02-11 08:45:38,895 - Phyluce - INFO - Reading Trichinella_murrelli_GCA_002221485 genome\n",
      "2020-02-11 08:46:10,883 - Phyluce - INFO - Trichinella_murrelli_GCA_002221485: 1786 uces, 379 dupes, 1407 non-dupes, 40 orient drop, 77 length drop, 1290 written\n",
      "2020-02-11 08:46:10,884 - Phyluce - INFO - ------- Working on Trichinella_nativa_GCA_001447565 genome ------\n",
      "2020-02-11 08:46:10,885 - Phyluce - INFO - Reading Trichinella_nativa_GCA_001447565 genome\n",
      "2020-02-11 08:46:27,117 - Phyluce - INFO - Trichinella_nativa_GCA_001447565: 1784 uces, 373 dupes, 1411 non-dupes, 0 orient drop, 9 length drop, 1402 written\n",
      "2020-02-11 08:46:27,117 - Phyluce - INFO - ------ Working on Trichinella_nelsoni_GCA_001447455 genome ------\n",
      "2020-02-11 08:46:27,118 - Phyluce - INFO - Reading Trichinella_nelsoni_GCA_001447455 genome\n",
      "2020-02-11 08:46:43,309 - Phyluce - INFO - Trichinella_nelsoni_GCA_001447455: 1786 uces, 364 dupes, 1422 non-dupes, 4 orient drop, 12 length drop, 1406 written\n",
      "2020-02-11 08:46:43,310 - Phyluce - INFO - ------- Working on Trichinella_papuae_GCA_001447755 genome ------\n",
      "2020-02-11 08:46:43,311 - Phyluce - INFO - Reading Trichinella_papuae_GCA_001447755 genome\n",
      "2020-02-11 08:46:57,617 - Phyluce - INFO - Trichinella_papuae_GCA_001447755: 1741 uces, 348 dupes, 1393 non-dupes, 20 orient drop, 11 length drop, 1362 written\n",
      "2020-02-11 08:46:57,617 - Phyluce - INFO - --- Working on Trichinella_patagoniensis_GCA_001447655 genome ---\n",
      "2020-02-11 08:46:57,627 - Phyluce - INFO - Reading Trichinella_patagoniensis_GCA_001447655 genome\n",
      "2020-02-11 08:47:16,628 - Phyluce - INFO - Trichinella_patagoniensis_GCA_001447655: 1787 uces, 373 dupes, 1414 non-dupes, 2 orient drop, 17 length drop, 1395 written\n",
      "2020-02-11 08:47:16,628 - Phyluce - INFO - --- Working on Trichinella_pseudospralis_GCA_001447645 genome ---\n",
      "2020-02-11 08:47:16,629 - Phyluce - INFO - Reading Trichinella_pseudospralis_GCA_001447645 genome\n",
      "2020-02-11 08:47:29,318 - Phyluce - INFO - Trichinella_pseudospralis_GCA_001447645: 1720 uces, 335 dupes, 1385 non-dupes, 8 orient drop, 16 length drop, 1361 written\n",
      "2020-02-11 08:47:29,318 - Phyluce - INFO - --------- Working on Trichinella_sp_GCA_001447745 genome --------\n",
      "2020-02-11 08:47:29,320 - Phyluce - INFO - Reading Trichinella_sp_GCA_001447745 genome\n",
      "2020-02-11 08:47:48,196 - Phyluce - INFO - Trichinella_sp_GCA_001447745: 1785 uces, 377 dupes, 1408 non-dupes, 5 orient drop, 14 length drop, 1389 written\n",
      "2020-02-11 08:47:48,196 - Phyluce - INFO - ---- Working on Trichinella_zimbabwensis_GCA_001447665 genome ---\n",
      "2020-02-11 08:47:48,204 - Phyluce - INFO - Reading Trichinella_zimbabwensis_GCA_001447665 genome\n",
      "2020-02-11 08:48:04,178 - Phyluce - INFO - Trichinella_zimbabwensis_GCA_001447665: 1740 uces, 357 dupes, 1383 non-dupes, 5 orient drop, 23 length drop, 1355 written\n",
      "2020-02-11 08:48:04,178 - Phyluce - INFO - ------ Working on Soboliphyme_baturini_GCA_900618415 genome -----\n",
      "2020-02-11 08:48:04,179 - Phyluce - INFO - Reading Soboliphyme_baturini_GCA_900618415 genome\n",
      "2020-02-11 08:48:05,562 - Phyluce - INFO - Soboliphyme_baturini_GCA_900618415: 245 uces, 47 dupes, 198 non-dupes, 0 orient drop, 3 length drop, 195 written\n",
      "2020-02-11 08:48:05,563 - Phyluce - INFO - ------ Working on Trichinella_spiralis_GCA_000181795 genome -----\n",
      "2020-02-11 08:48:05,563 - Phyluce - INFO - Reading Trichinella_spiralis_GCA_000181795 genome\n",
      "2020-02-11 08:48:32,874 - Phyluce - INFO - Trichinella_spiralis_GCA_000181795: 1790 uces, 823 dupes, 967 non-dupes, 46 orient drop, 297 length drop, 624 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/nematoda-clade1/extract_probes_from_group/probe_fasta --output results/phyluce/nematoda-clade1/extract_probes_from_group/multifastas.sqlite --base-taxon Trichinella_spiralis_GCA_000181795;\n",
      "romanomermis_culicivorax_gca_001039655.\n",
      "trichinella_britovi_gca_001447585..\n",
      "trichinella_murrelli_gca_002221485..\n",
      "trichinella_nativa_gca_001447565..\n",
      "trichinella_nelsoni_gca_001447455..\n",
      "trichinella_papuae_gca_001447755..\n",
      "trichinella_patagoniensis_gca_001447655..\n",
      "trichinella_pseudospralis_gca_001447645..\n",
      "trichinella_sp_gca_001447745..\n",
      "trichinella_zimbabwensis_gca_001447665..\n",
      "soboliphyme_baturini_gca_900618415.\n",
      "trichinella_spiralis_gca_000181795.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade1/extract_probes_from_group/multifastas.sqlite --base-taxon Trichinella_spiralis_GCA_000181795;\n",
      "Loci shared by 0 taxa:\t1,517.0\n",
      "Loci shared by 1 taxa:\t1,517.0\n",
      "Loci shared by 2 taxa:\t1,466.0\n",
      "Loci shared by 3 taxa:\t1,429.0\n",
      "Loci shared by 4 taxa:\t1,409.0\n",
      "Loci shared by 5 taxa:\t1,398.0\n",
      "Loci shared by 6 taxa:\t1,386.0\n",
      "Loci shared by 7 taxa:\t1,373.0\n",
      "Loci shared by 8 taxa:\t1,329.0\n",
      "Loci shared by 9 taxa:\t1,225.0\n",
      "Loci shared by 10 taxa:\t614.0\n",
      "Loci shared by 11 taxa:\t102.0\n",
      "Loci shared by 12 taxa:\t24.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade1/extract_probes_from_group/multifastas.sqlite  --base-taxon Trichinella_spiralis_GCA_000181795 --output results/phyluce/nematoda-clade1/extract_probes_from_group/Trichinella_spiralis_GCA_000181795+9-back-to-11.conf --specific-counts 11;\n",
      "Counter({'trichinella_papuae_gca_001447755': 102, 'trichinella_patagoniensis_gca_001447655': 102, 'trichinella_sp_gca_001447745': 102, 'trichinella_murrelli_gca_002221485': 102, 'trichinella_nelsoni_gca_001447455': 102, 'trichinella_nativa_gca_001447565': 102, 'trichinella_zimbabwensis_gca_001447665': 102, 'trichinella_pseudospralis_gca_001447645': 102, 'trichinella_britovi_gca_001447585': 102, 'soboliphyme_baturini_gca_900618415': 97, 'trichinella_spiralis_gca_000181795': 71, 'romanomermis_culicivorax_gca_001039655': 60})\n",
      "Total loci = 102\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(11)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/nematoda-clade1/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/nematoda-clade1/extract_probes_from_group/Trichinella_spiralis_GCA_000181795+9-back-to-11.conf --probe-prefix uce_nematoda-clade1_ --designer rnplattii --design nematoda-clade1_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGNN\n",
      "\n",
      "\n",
      "Conserved locus count = 102\n",
      "Probe Count = 2267\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list.fasta --query results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  08:53:07\n",
      "Ended:  Tue Feb 11, 2020  08:53:10\n",
      "Time for execution:  0.0412437637647 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list.fasta --lastz results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_nematoda-clade1_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 2266 fasta sequences.  Filtered 0 duplicates. Kept 2267.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/nematoda-clade1/final_probe_design/nematoda-clade1_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 08:54:07 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 2267\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 181360\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3961142\n",
      "Max number of word counting entries: 89165324\n",
      "\n",
      "# comparing sequences from          0  to        377\n",
      "---------- new table with      158 representatives\n",
      "# comparing sequences from        377  to        692\n",
      "99.8%---------- new table with      119 representatives\n",
      "# comparing sequences from        692  to        954\n",
      "99.9%---------- new table with      101 representatives\n",
      "# comparing sequences from        954  to       1172\n",
      "99.6%---------- new table with       89 representatives\n",
      "# comparing sequences from       1172  to       1354\n",
      "..................---------- new table with       73 representatives\n",
      "# comparing sequences from       1354  to       2267\n",
      ".....................---------- new table with      380 representatives\n",
      "\n",
      "     2267  finished        920  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 86M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.57\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clade 3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'nematoda-clade3'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Acanthocheilonema_viteae_GCA_900537255\",\n",
    "               \"Anisakis_simplex_GCA_900617985\",\n",
    "               \"Ascaris_suum_GCA_000187025\",\n",
    "               \"Baylisascaris_schroederi_GCA_006503575\",\n",
    "               \"Brugia_pahangi_GCA_001280985\",\n",
    "               \"Brugia_timori_GCA_900618025\",\n",
    "               \"Dirofilaria_immitis_GCA_009829315\",\n",
    "               \"Dirofilaria_repens_GCA_008729115\",\n",
    "               \"Dracunculus_medinensis_GCA_900625125\",\n",
    "               \"Elaeophora_elaphi_GCA_000499685\",\n",
    "               \"Enterobius_vermicularis_GCA_900576705\",\n",
    "               \"Gongylonema_pulchrum_GCA_900617915\",\n",
    "               \"Litomosoides_sigmodontis_GCA_900537275\",\n",
    "               \"Loa_loa_GCF_000183805\",\n",
    "               \"Onchocerca_flexuosa_GCA_002249935\",\n",
    "               \"Onchocerca_ochengi_GCA_000950515\",\n",
    "               \"Onchocerca_volvulus_GCA_000499405\",\n",
    "               \"Parascaris_univalens_GCA_002259215\",\n",
    "               \"Plectus_murrayi_GCA_004785735\",\n",
    "               \"Plectus_sambesii_GCA_002796945\",\n",
    "               \"Thelazia_callipaeda_GCA_900618365\",\n",
    "               \"Toxocara_canis_GCA_000803305\",\n",
    "               \"Wuchereria_bancrofti_GCA_005281725\" ]\n",
    "                    \n",
    "reference_taxon = \"Brugia_malayi_GCA_000002995\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900537255.1_ASM90053725v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24974073 bytes  5549803.33 bytes/sec\n",
      "total size is 24967868  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617985.1_A_simplex_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 40203686 bytes  7309768.73 bytes/sec\n",
      "total size is 40193756  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187025.3_ASM18702v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 93179338 bytes  8102554.78 bytes/sec\n",
      "total size is 93156488  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006503575.1_ASM650357v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 74453449 bytes  8759234.24 bytes/sec\n",
      "total size is 74435166  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002995.5_B_malayi-4.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 26449967 bytes  10580003.60 bytes/sec\n",
      "total size is 26443403  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001280985.1_Brugia_pa_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 26060416 bytes  5791212.89 bytes/sec\n",
      "total size is 26053939  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900618025.1_B_timori_Indonesia_v1_0_4_001_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20932259 bytes  5980657.43 bytes/sec\n",
      "total size is 20927018  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009829315.1_ASM982931v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 28561198 bytes  6346942.22 bytes/sec\n",
      "total size is 28554115  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008729115.1_FGCZ_Drep_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 31011571 bytes  6891469.56 bytes/sec\n",
      "total size is 31003886  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900625125.1_D_medinensis_Ghana_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 33399425 bytes  7422103.78 bytes/sec\n",
      "total size is 33391142  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499685.1_EEL001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 443227 bytes  295512.67 bytes/sec\n",
      "total size is 443014  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900576705.1_E_vermicularis_Canary_Islands_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 47227259 bytes  8586782.00 bytes/sec\n",
      "total size is 47215602  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617915.1_G_pulchrum_Hokkaido_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 99855847 bytes  7988471.12 bytes/sec\n",
      "total size is 99831347  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900537275.1_ASM90053727v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20827271 bytes  5950660.86 bytes/sec\n",
      "total size is 20822074  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000183805.3_Loa_loa_V3.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 27855285 bytes  7958664.86 bytes/sec\n",
      "total size is 27848377  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002249935.1_O_flexuosa_1.0.allpaths.pg.lrna_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20301941 bytes  5800566.57 bytes/sec\n",
      "total size is 20296854  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000950515.2_O_ochengi_Ngaoundere_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 29545480 bytes  8441577.71 bytes/sec\n",
      "total size is 29538148  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499405.2_ASM49940v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 29647934 bytes  6588439.11 bytes/sec\n",
      "total size is 29640588  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002259215.1_ASM225921v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 75600979 bytes  8894237.76 bytes/sec\n",
      "total size is 75582416  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004785735.1_ASM478573v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 72225908 bytes  8497170.59 bytes/sec\n",
      "total size is 72208169  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002796945.1_Psam_v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 60430076 bytes  8057349.07 bytes/sec\n",
      "total size is 60415219  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900618365.1_T_callipaeda_Ticino_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24131626 bytes  6894762.29 bytes/sec\n",
      "total size is 24125606  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000803305.1_Toxocara_canis_adult_r1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 93406305 bytes  8122291.04 bytes/sec\n",
      "total size is 93383384  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005281725.1_ASM528172v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 27103806 bytes  4927972.36 bytes/sec\n",
      "total size is 27097083  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486941 (\"sim_Acanthocheilonema_viteae_GCA_900537255\") has been submitted\n",
      "Your job 5486942 (\"sim_Anisakis_simplex_GCA_900617985\") has been submitted\n",
      "Your job 5486943 (\"sim_Ascaris_suum_GCA_000187025\") has been submitted\n",
      "Your job 5486944 (\"sim_Baylisascaris_schroederi_GCA_006503575\") has been submitted\n",
      "Your job 5486945 (\"sim_Brugia_pahangi_GCA_001280985\") has been submitted\n",
      "Your job 5486946 (\"sim_Brugia_timori_GCA_900618025\") has been submitted\n",
      "Your job 5486947 (\"sim_Dirofilaria_immitis_GCA_009829315\") has been submitted\n",
      "Your job 5486948 (\"sim_Dirofilaria_repens_GCA_008729115\") has been submitted\n",
      "Your job 5486949 (\"sim_Dracunculus_medinensis_GCA_900625125\") has been submitted\n",
      "Your job 5486950 (\"sim_Elaeophora_elaphi_GCA_000499685\") has been submitted\n",
      "Your job 5486951 (\"sim_Enterobius_vermicularis_GCA_900576705\") has been submitted\n",
      "Your job 5486952 (\"sim_Gongylonema_pulchrum_GCA_900617915\") has been submitted\n",
      "Your job 5486953 (\"sim_Litomosoides_sigmodontis_GCA_900537275\") has been submitted\n",
      "Your job 5486954 (\"sim_Loa_loa_GCF_000183805\") has been submitted\n",
      "Your job 5486955 (\"sim_Onchocerca_flexuosa_GCA_002249935\") has been submitted\n",
      "Your job 5486956 (\"sim_Onchocerca_ochengi_GCA_000950515\") has been submitted\n",
      "Your job 5486957 (\"sim_Onchocerca_volvulus_GCA_000499405\") has been submitted\n",
      "Your job 5486958 (\"sim_Parascaris_univalens_GCA_002259215\") has been submitted\n",
      "Your job 5486959 (\"sim_Plectus_murrayi_GCA_004785735\") has been submitted\n",
      "Your job 5486960 (\"sim_Plectus_sambesii_GCA_002796945\") has been submitted\n",
      "Your job 5486961 (\"sim_Thelazia_callipaeda_GCA_900618365\") has been submitted\n",
      "Your job 5486962 (\"sim_Toxocara_canis_GCA_000803305\") has been submitted\n",
      "Your job 5486963 (\"sim_Wuchereria_bancrofti_GCA_005281725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/nematoda-clade3/cleaned_genomes/Brugia_malayi_GCA_000002995_formatted.fas path=results/phyluce/nematoda-clade3/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486964 (\"map_Acanthocheilonema_viteae_GCA_900537255\") has been submitted\n",
      "Your job 5486965 (\"map_Anisakis_simplex_GCA_900617985\") has been submitted\n",
      "Your job 5486966 (\"map_Ascaris_suum_GCA_000187025\") has been submitted\n",
      "Your job 5486967 (\"map_Baylisascaris_schroederi_GCA_006503575\") has been submitted\n",
      "Your job 5486968 (\"map_Brugia_pahangi_GCA_001280985\") has been submitted\n",
      "Your job 5486969 (\"map_Brugia_timori_GCA_900618025\") has been submitted\n",
      "Your job 5486970 (\"map_Dirofilaria_immitis_GCA_009829315\") has been submitted\n",
      "Your job 5486971 (\"map_Dirofilaria_repens_GCA_008729115\") has been submitted\n",
      "Your job 5486972 (\"map_Dracunculus_medinensis_GCA_900625125\") has been submitted\n",
      "Your job 5486973 (\"map_Elaeophora_elaphi_GCA_000499685\") has been submitted\n",
      "Your job 5486974 (\"map_Enterobius_vermicularis_GCA_900576705\") has been submitted\n",
      "Your job 5486975 (\"map_Gongylonema_pulchrum_GCA_900617915\") has been submitted\n",
      "Your job 5486976 (\"map_Litomosoides_sigmodontis_GCA_900537275\") has been submitted\n",
      "Your job 5486977 (\"map_Loa_loa_GCF_000183805\") has been submitted\n",
      "Your job 5486978 (\"map_Onchocerca_flexuosa_GCA_002249935\") has been submitted\n",
      "Your job 5486979 (\"map_Onchocerca_ochengi_GCA_000950515\") has been submitted\n",
      "Your job 5486980 (\"map_Onchocerca_volvulus_GCA_000499405\") has been submitted\n",
      "Your job 5486981 (\"map_Parascaris_univalens_GCA_002259215\") has been submitted\n",
      "Your job 5486982 (\"map_Plectus_murrayi_GCA_004785735\") has been submitted\n",
      "Your job 5486983 (\"map_Plectus_sambesii_GCA_002796945\") has been submitted\n",
      "Your job 5486984 (\"map_Thelazia_callipaeda_GCA_900618365\") has been submitted\n",
      "Your job 5486985 (\"map_Toxocara_canis_GCA_000803305\") has been submitted\n",
      "Your job 5486986 (\"map_Wuchereria_bancrofti_GCA_005281725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5486987 (\"merge_Acanthocheilonema_viteae_GCA_900537255\") has been submitted\n",
      "Your job 5486988 (\"merge_Anisakis_simplex_GCA_900617985\") has been submitted\n",
      "Your job 5486989 (\"merge_Ascaris_suum_GCA_000187025\") has been submitted\n",
      "Your job 5486990 (\"merge_Baylisascaris_schroederi_GCA_006503575\") has been submitted\n",
      "Your job 5486991 (\"merge_Brugia_pahangi_GCA_001280985\") has been submitted\n",
      "Your job 5486992 (\"merge_Brugia_timori_GCA_900618025\") has been submitted\n",
      "Your job 5486993 (\"merge_Dirofilaria_immitis_GCA_009829315\") has been submitted\n",
      "Your job 5486994 (\"merge_Dirofilaria_repens_GCA_008729115\") has been submitted\n",
      "Your job 5486995 (\"merge_Dracunculus_medinensis_GCA_900625125\") has been submitted\n",
      "Your job 5486996 (\"merge_Elaeophora_elaphi_GCA_000499685\") has been submitted\n",
      "Your job 5486997 (\"merge_Enterobius_vermicularis_GCA_900576705\") has been submitted\n",
      "Your job 5486998 (\"merge_Gongylonema_pulchrum_GCA_900617915\") has been submitted\n",
      "Your job 5486999 (\"merge_Litomosoides_sigmodontis_GCA_900537275\") has been submitted\n",
      "Your job 5487000 (\"merge_Loa_loa_GCF_000183805\") has been submitted\n",
      "Your job 5487001 (\"merge_Onchocerca_flexuosa_GCA_002249935\") has been submitted\n",
      "Your job 5487002 (\"merge_Onchocerca_ochengi_GCA_000950515\") has been submitted\n",
      "Your job 5487003 (\"merge_Onchocerca_volvulus_GCA_000499405\") has been submitted\n",
      "Your job 5487004 (\"merge_Parascaris_univalens_GCA_002259215\") has been submitted\n",
      "Your job 5487005 (\"merge_Plectus_murrayi_GCA_004785735\") has been submitted\n",
      "Your job 5487006 (\"merge_Plectus_sambesii_GCA_002796945\") has been submitted\n",
      "Your job 5487007 (\"merge_Thelazia_callipaeda_GCA_900618365\") has been submitted\n",
      "Your job 5487008 (\"merge_Toxocara_canis_GCA_000803305\") has been submitted\n",
      "Your job 5487009 (\"merge_Wuchereria_bancrofti_GCA_005281725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 40159 sequences from Acanthocheilonema_viteae_GCA_900537255_merged.bed.  Filtered 28615 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11544.\n",
      "Screened 428 sequences from Anisakis_simplex_GCA_900617985_merged.bed.  Filtered 418 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 918 sequences from Ascaris_suum_GCA_000187025_merged.bed.  Filtered 890 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 28.\n",
      "Screened 513 sequences from Baylisascaris_schroederi_GCA_006503575_merged.bed.  Filtered 498 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15.\n",
      "Screened 56981 sequences from Brugia_pahangi_GCA_001280985_merged.bed.  Filtered 42537 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 14444.\n",
      "Screened 35463 sequences from Brugia_timori_GCA_900618025_merged.bed.  Filtered 24747 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10716.\n",
      "Screened 2145 sequences from Dirofilaria_immitis_GCA_009829315_merged.bed.  Filtered 2019 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 126.\n",
      "Screened 29499 sequences from Dirofilaria_repens_GCA_008729115_merged.bed.  Filtered 22617 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6882.\n",
      "Screened 480 sequences from Dracunculus_medinensis_GCA_900625125_merged.bed.  Filtered 458 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 22.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 637 sequences from Enterobius_vermicularis_GCA_900576705_merged.bed.  Filtered 637 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 0.\n",
      "Screened 845 sequences from Gongylonema_pulchrum_GCA_900617915_merged.bed.  Filtered 803 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 42.\n",
      "Screened 22913 sequences from Litomosoides_sigmodontis_GCA_900537275_merged.bed.  Filtered 17838 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5075.\n",
      "Screened 51492 sequences from Loa_loa_GCF_000183805_merged.bed.  Filtered 34518 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 16974.\n",
      "Screened 27121 sequences from Onchocerca_flexuosa_GCA_002249935_merged.bed.  Filtered 19585 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7536.\n",
      "Screened 35112 sequences from Onchocerca_ochengi_GCA_000950515_merged.bed.  Filtered 25594 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9518.\n",
      "Screened 36061 sequences from Onchocerca_volvulus_GCA_000499405_merged.bed.  Filtered 26445 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9616.\n",
      "Screened 776 sequences from Parascaris_univalens_GCA_002259215_merged.bed.  Filtered 746 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 30.\n",
      "Screened 86 sequences from Plectus_murrayi_GCA_004785735_merged.bed.  Filtered 74 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 97 sequences from Plectus_sambesii_GCA_002796945_merged.bed.  Filtered 81 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 16.\n",
      "Screened 2156 sequences from Thelazia_callipaeda_GCA_900618365_merged.bed.  Filtered 2023 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 133.\n",
      "Screened 531 sequences from Toxocara_canis_GCA_000803305_merged.bed.  Filtered 506 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 25.\n",
      "Screened 177983 sequences from Wuchereria_bancrofti_GCA_005281725_merged.bed.  Filtered 120781 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 57202.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/nematoda-clade3/initial_intervals/Wuchereria_bancrofti_GCA_005281725_merged.bed --twobit results/phyluce/nematoda-clade3/cleaned_genomes/Brugia_malayi_GCA_000002995_formatted.2bit --output results/phyluce/nematoda-clade3/initial_intervals/Wuchereria_bancrofti_GCA_005281725_stripped.bed;\n",
      "acanthocheilonema_viteae_gca_900537255............\n",
      "anisakis_simplex_gca_900617985.\n",
      "ascaris_suum_gca_000187025.\n",
      "baylisascaris_schroederi_gca_006503575.\n",
      "brugia_pahangi_gca_001280985...............\n",
      "brugia_timori_gca_900618025...........\n",
      "dirofilaria_immitis_gca_009829315.\n",
      "dirofilaria_repens_gca_008729115.......\n",
      "dracunculus_medinensis_gca_900625125.\n",
      "elaeophora_elaphi_gca_000499685\n",
      "enterobius_vermicularis_gca_900576705\n",
      "gongylonema_pulchrum_gca_900617915.\n",
      "litomosoides_sigmodontis_gca_900537275......\n",
      "loa_loa_gcf_000183805.................\n",
      "onchocerca_flexuosa_gca_002249935........\n",
      "onchocerca_ochengi_gca_000950515..........\n",
      "onchocerca_volvulus_gca_000499405..........\n",
      "parascaris_univalens_gca_002259215.\n",
      "plectus_murrayi_gca_004785735.\n",
      "plectus_sambesii_gca_002796945.\n",
      "thelazia_callipaeda_gca_900618365.\n",
      "toxocara_canis_gca_000803305.\n",
      "wuchereria_bancrofti_gca_005281725..........................................................\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade3/initial_intervals/nematoda-clade3-to-Brugia_malayi_GCA_000002995.sqlite --base-taxon Brugia_malayi_GCA_000002995\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 0 taxa:\t55,713.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 1 taxa:\t55,713.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 2 taxa:\t28,748.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 3 taxa:\t20,510.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 4 taxa:\t13,407.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 5 taxa:\t10,136.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 6 taxa:\t7,450.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 7 taxa:\t5,473.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 8 taxa:\t3,888.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 9 taxa:\t2,394.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 10 taxa:\t989.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 11 taxa:\t94.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 12 taxa:\t36.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 13 taxa:\t21.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 14 taxa:\t17.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 15 taxa:\t15.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 16 taxa:\t13.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 17 taxa:\t9.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 18 taxa:\t3.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 19 taxa:\t2.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 20 taxa:\t1.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 21 taxa:\t1.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 22 taxa:\t0.0\n",
      "Loci shared by Brugia_malayi_GCA_000002995 + 23 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade3/initial_intervals/nematoda-clade3-to-Brugia_malayi_GCA_000002995.sqlite --base-taxon Brugia_malayi_GCA_000002995 --output results/phyluce/nematoda-clade3/initial_intervals/Brugia_malayi_GCA_000002995_+9.bed --specific-counts 9;\n",
      "Counter({'onchocerca_volvulus_gca_000499405': 2368, 'loa_loa_gcf_000183805': 2356, 'onchocerca_ochengi_gca_000950515': 2351, 'wuchereria_bancrofti_gca_005281725': 2350, 'acanthocheilonema_viteae_gca_900537255': 2348, 'brugia_pahangi_gca_001280985': 2284, 'brugia_timori_gca_900618025': 2192, 'onchocerca_flexuosa_gca_002249935': 2178, 'dirofilaria_repens_gca_008729115': 2085, 'litomosoides_sigmodontis_gca_900537275': 1879, 'thelazia_callipaeda_gca_900618365': 91, 'dirofilaria_immitis_gca_009829315': 87, 'gongylonema_pulchrum_gca_900617915': 39, 'ascaris_suum_gca_000187025': 25, 'toxocara_canis_gca_000803305': 24, 'parascaris_univalens_gca_002259215': 23, 'dracunculus_medinensis_gca_900625125': 19, 'plectus_sambesii_gca_002796945': 16, 'baylisascaris_schroederi_gca_006503575': 11, 'plectus_murrayi_gca_004785735': 11, 'anisakis_simplex_gca_900617985': 10})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 9\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/nematoda-clade3/initial_intervals/Brugia_malayi_GCA_000002995_+9.bed --twobit results/phyluce/nematoda-clade3/cleaned_genomes/Brugia_malayi_GCA_000002995_formatted.2bit --buffer-to 160 --output results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9.fasta;\n",
      "Screened 2394 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 2394.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9.fasta --probe-prefix uce_nematoda-clade3_ --design nematoda-clade3_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 2385\n",
      "Probe Count = 4755\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes.fas --query results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  08:56:42\n",
      "Ended:  Tue Feb 11, 2020  08:56:44\n",
      "Time for execution:  0.0361920833588 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes.fas                        --lastz results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes_vself.lastz                       --probe-prefix=uce_nematoda-clade3_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 4754 fasta sequences.  Filtered 72 duplicates. Kept 4612.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/nematoda-clade3/validate_intervals/Brugia_malayi_GCA_000002995_+9_temp_probes.fas --scaffoldlist Acanthocheilonema_viteae_GCA_900537255 Anisakis_simplex_GCA_900617985 Ascaris_suum_GCA_000187025 Baylisascaris_schroederi_GCA_006503575 Brugia_pahangi_GCA_001280985 Brugia_timori_GCA_900618025 Dirofilaria_immitis_GCA_009829315 Dirofilaria_repens_GCA_008729115 Dracunculus_medinensis_GCA_900625125 Elaeophora_elaphi_GCA_000499685 Enterobius_vermicularis_GCA_900576705 Gongylonema_pulchrum_GCA_900617915 Litomosoides_sigmodontis_GCA_900537275 Loa_loa_GCF_000183805 Onchocerca_flexuosa_GCA_002249935 Onchocerca_ochengi_GCA_000950515 Onchocerca_volvulus_GCA_000499405 Parascaris_univalens_GCA_002259215 Plectus_murrayi_GCA_004785735 Plectus_sambesii_GCA_002796945 Thelazia_callipaeda_GCA_900618365 Toxocara_canis_GCA_000803305 Wuchereria_bancrofti_GCA_005281725 Brugia_malayi_GCA_000002995 --genome-base-path results/phyluce/nematoda-clade3/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/nematoda-clade3/validate_intervals/nematoda-clade3-to-Brugia_malayi_GCA_000002995.sqlite --output results/phyluce/nematoda-clade3/validate_intervals/lastz/;\n",
      "\n",
      "Running against Acanthocheilonema_viteae_GCA_900537255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmpxcYIxi.fasta\n",
      "\t/tmp/tmpn0vr3Y.fasta\n",
      "\t/tmp/tmpUDvhG2.fasta\n",
      "\t/tmp/tmpHBhCwJ.fasta\n",
      "\t/tmp/tmpfZVT6f.fasta\n",
      "\t/tmp/tmp2iCC4K.fasta\n",
      "\t/tmp/tmpySPjG9.fasta\n",
      "\t/tmp/tmpbd1xNq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7w4mfk.lastz\n",
      "\t/tmp/tmpOfB4TY.lastz\n",
      "\t/tmp/tmpz262fU.lastz\n",
      "\t/tmp/tmpsQpC3c.lastz\n",
      "\t/tmp/tmptqci0i.lastz\n",
      "\t/tmp/tmpVaUqGS.lastz\n",
      "\t/tmp/tmpceZXMe.lastz\n",
      "\t/tmp/tmpRDxso9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Acanthocheilonema_viteae_GCA_900537255.lastz\n",
      "Creating Acanthocheilonema_viteae_GCA_900537255 table\n",
      "Inserting data to Acanthocheilonema_viteae_GCA_900537255 table\n",
      "\n",
      "Running against Anisakis_simplex_GCA_900617985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 13 queries...\n",
      "\t/tmp/tmpsgL8VU.fasta\n",
      "\t/tmp/tmpiGY6Qf.fasta\n",
      "\t/tmp/tmp02kivp.fasta\n",
      "\t/tmp/tmpbVFevt.fasta\n",
      "\t/tmp/tmpVxZhp6.fasta\n",
      "\t/tmp/tmp5OXv92.fasta\n",
      "\t/tmp/tmp0uKteb.fasta\n",
      "\t/tmp/tmpU4wcRL.fasta\n",
      "\t/tmp/tmpjhbLOa.fasta\n",
      "\t/tmp/tmpCx7WHW.fasta\n",
      "\t/tmp/tmpwyAFNe.fasta\n",
      "\t/tmp/tmpdfxkEZ.fasta\n",
      "\t/tmp/tmp3loIka.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIBYWEc.lastz\n",
      "\t/tmp/tmpc6KMgk.lastz\n",
      "\t/tmp/tmpBnGWie.lastz\n",
      "\t/tmp/tmppxOmBw.lastz\n",
      "\t/tmp/tmpwqhZeF.lastz\n",
      "\t/tmp/tmpIWkqKS.lastz\n",
      "\t/tmp/tmpWQO1om.lastz\n",
      "\t/tmp/tmpovHtR4.lastz\n",
      "\t/tmp/tmpcWqv6X.lastz\n",
      "\t/tmp/tmpyid6iC.lastz\n",
      "\t/tmp/tmpQkYW5G.lastz\n",
      "\t/tmp/tmpDWu79H.lastz\n",
      "\t/tmp/tmpQZ_OWy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Anisakis_simplex_GCA_900617985.lastz\n",
      "Creating Anisakis_simplex_GCA_900617985 table\n",
      "Inserting data to Anisakis_simplex_GCA_900617985 table\n",
      "\n",
      "Running against Ascaris_suum_GCA_000187025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 26 queries...\n",
      "\t/tmp/tmpIPG04H.fasta\n",
      "\t/tmp/tmp_KqfO6.fasta\n",
      "\t/tmp/tmpvZfPci.fasta\n",
      "\t/tmp/tmpqRAcIk.fasta\n",
      "\t/tmp/tmp3sjW0h.fasta\n",
      "\t/tmp/tmpoUK9QG.fasta\n",
      "\t/tmp/tmpuNPNRX.fasta\n",
      "\t/tmp/tmp3eog34.fasta\n",
      "\t/tmp/tmp8xbNTw.fasta\n",
      "\t/tmp/tmp9Bdi5F.fasta\n",
      "\t/tmp/tmplmxCDF.fasta\n",
      "\t/tmp/tmp_LyMUI.fasta\n",
      "\t/tmp/tmpoA5G7R.fasta\n",
      "\t/tmp/tmpsu_vdo.fasta\n",
      "\t/tmp/tmpXIC05N.fasta\n",
      "\t/tmp/tmpbZu0u3.fasta\n",
      "\t/tmp/tmpFwG6zG.fasta\n",
      "\t/tmp/tmpyPUlQi.fasta\n",
      "\t/tmp/tmpkmhbj4.fasta\n",
      "\t/tmp/tmpEcSm63.fasta\n",
      "\t/tmp/tmpl8qwBk.fasta\n",
      "\t/tmp/tmpjcxBsd.fasta\n",
      "\t/tmp/tmpDWIWoB.fasta\n",
      "\t/tmp/tmp6r6CCH.fasta\n",
      "\t/tmp/tmpWGQ4w7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0sg7yB.lastz\n",
      "\t/tmp/tmpt0Ly0a.lastz\n",
      "\t/tmp/tmpxRTpJ5.lastz\n",
      "\t/tmp/tmpIVts1m.lastz\n",
      "\t/tmp/tmpO63dpv.lastz\n",
      "\t/tmp/tmpMbtuxp.lastz\n",
      "\t/tmp/tmplcYvlG.lastz\n",
      "\t/tmp/tmpARqMQY.lastz\n",
      "\t/tmp/tmpQ8BVqY.lastz\n",
      "\t/tmp/tmpgI9ZQn.lastz\n",
      "\t/tmp/tmpgQaL6V.lastz\n",
      "\t/tmp/tmp1sznY8.lastz\n",
      "\t/tmp/tmpyvYEQP.lastz\n",
      "\t/tmp/tmp0QCx3e.lastz\n",
      "\t/tmp/tmp3fvJuX.lastz\n",
      "\t/tmp/tmpLuJGI_.lastz\n",
      "\t/tmp/tmpHecNYS.lastz\n",
      "\t/tmp/tmplrnLaU.lastz\n",
      "\t/tmp/tmp9Qxthz.lastz\n",
      "\t/tmp/tmpMcf2lc.lastz\n",
      "\t/tmp/tmpirlCmd.lastz\n",
      "\t/tmp/tmp4P_ALQ.lastz\n",
      "\t/tmp/tmpZ93zu7.lastz\n",
      "\t/tmp/tmpZX6rYf.lastz\n",
      "\t/tmp/tmpOOUoxK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Ascaris_suum_GCA_000187025.lastz\n",
      "Creating Ascaris_suum_GCA_000187025 table\n",
      "Inserting data to Ascaris_suum_GCA_000187025 table\n",
      "\n",
      "Running against Baylisascaris_schroederi_GCA_006503575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 23 queries...\n",
      "\t/tmp/tmpvT4CPm.fasta\n",
      "\t/tmp/tmp9Eh2ll.fasta\n",
      "\t/tmp/tmpPWLYpt.fasta\n",
      "\t/tmp/tmpmRsqR7.fasta\n",
      "\t/tmp/tmp96F9B8.fasta\n",
      "\t/tmp/tmp0Upnrd.fasta\n",
      "\t/tmp/tmpVCvKpJ.fasta\n",
      "\t/tmp/tmpTkl5To.fasta\n",
      "\t/tmp/tmpXldaR5.fasta\n",
      "\t/tmp/tmpRsPu1v.fasta\n",
      "\t/tmp/tmpBKe4SO.fasta\n",
      "\t/tmp/tmpALiL_9.fasta\n",
      "\t/tmp/tmpQAHi8z.fasta\n",
      "\t/tmp/tmpv7FrY6.fasta\n",
      "\t/tmp/tmpyXdtBX.fasta\n",
      "\t/tmp/tmpaskDsM.fasta\n",
      "\t/tmp/tmpYwtkMl.fasta\n",
      "\t/tmp/tmpCTI9Zw.fasta\n",
      "\t/tmp/tmphHcDxY.fasta\n",
      "\t/tmp/tmpJttssV.fasta\n",
      "\t/tmp/tmpaYU1cZ.fasta\n",
      "\t/tmp/tmpQ2bcWi.fasta\n",
      "\t/tmp/tmpbWaBUt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxZWJTO.lastz\n",
      "\t/tmp/tmp1ifYCC.lastz\n",
      "\t/tmp/tmp4QLr8J.lastz\n",
      "\t/tmp/tmpYRZcwu.lastz\n",
      "\t/tmp/tmpVwSdLe.lastz\n",
      "\t/tmp/tmp99fUTt.lastz\n",
      "\t/tmp/tmpsp1xpJ.lastz\n",
      "\t/tmp/tmpdEmSzz.lastz\n",
      "\t/tmp/tmple057z.lastz\n",
      "\t/tmp/tmpftmqrT.lastz\n",
      "\t/tmp/tmpYrYCfS.lastz\n",
      "\t/tmp/tmpTquHEP.lastz\n",
      "\t/tmp/tmpjkhW1I.lastz\n",
      "\t/tmp/tmpfubjy1.lastz\n",
      "\t/tmp/tmp1m9t9v.lastz\n",
      "\t/tmp/tmp4vZFtm.lastz\n",
      "\t/tmp/tmptGO3wB.lastz\n",
      "\t/tmp/tmplUVyu7.lastz\n",
      "\t/tmp/tmpvnOrPB.lastz\n",
      "\t/tmp/tmprauqal.lastz\n",
      "\t/tmp/tmpSk7bSm.lastz\n",
      "\t/tmp/tmpVMG1n3.lastz\n",
      "\t/tmp/tmpbmCd_h.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Baylisascaris_schroederi_GCA_006503575.lastz\n",
      "Creating Baylisascaris_schroederi_GCA_006503575 table\n",
      "Inserting data to Baylisascaris_schroederi_GCA_006503575 table\n",
      "\n",
      "Running against Brugia_pahangi_GCA_001280985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmprLN9tb.fasta\n",
      "\t/tmp/tmpN2lDDe.fasta\n",
      "\t/tmp/tmpWEtZiL.fasta\n",
      "\t/tmp/tmpKXcQsV.fasta\n",
      "\t/tmp/tmpuJuLoS.fasta\n",
      "\t/tmp/tmp7mYibS.fasta\n",
      "\t/tmp/tmpNJDhOD.fasta\n",
      "\t/tmp/tmpqHuDdO.fasta\n",
      "\t/tmp/tmp2UFW5u.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxPpzsl.lastz\n",
      "\t/tmp/tmpOGtV2T.lastz\n",
      "\t/tmp/tmpwoYJrH.lastz\n",
      "\t/tmp/tmpHSsatw.lastz\n",
      "\t/tmp/tmpdVdFiL.lastz\n",
      "\t/tmp/tmphZfJDz.lastz\n",
      "\t/tmp/tmpQmVLtZ.lastz\n",
      "\t/tmp/tmpTUiYFF.lastz\n",
      "\t/tmp/tmper83dB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Brugia_pahangi_GCA_001280985.lastz\n",
      "Creating Brugia_pahangi_GCA_001280985 table\n",
      "Inserting data to Brugia_pahangi_GCA_001280985 table\n",
      "\n",
      "Running against Brugia_timori_GCA_900618025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpOTODpY.fasta\n",
      "\t/tmp/tmpVpOi5z.fasta\n",
      "\t/tmp/tmp_pSLUo.fasta\n",
      "\t/tmp/tmpep14Eq.fasta\n",
      "\t/tmp/tmp8dzd_3.fasta\n",
      "\t/tmp/tmp4RIM0b.fasta\n",
      "\t/tmp/tmpkvoIax.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJupTmT.lastz\n",
      "\t/tmp/tmpGAvmoo.lastz\n",
      "\t/tmp/tmptQ63iB.lastz\n",
      "\t/tmp/tmpUWsE2D.lastz\n",
      "\t/tmp/tmpAaKFoQ.lastz\n",
      "\t/tmp/tmpG04NTm.lastz\n",
      "\t/tmp/tmpYA3ZNF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Brugia_timori_GCA_900618025.lastz\n",
      "Creating Brugia_timori_GCA_900618025 table\n",
      "Inserting data to Brugia_timori_GCA_900618025 table\n",
      "\n",
      "Running against Dirofilaria_immitis_GCA_009829315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpKzNF2z.fasta\n",
      "\t/tmp/tmpObtcit.fasta\n",
      "\t/tmp/tmp1zqUwn.fasta\n",
      "\t/tmp/tmp3NxrIq.fasta\n",
      "\t/tmp/tmpj6c54m.fasta\n",
      "\t/tmp/tmpvkAHCt.fasta\n",
      "\t/tmp/tmpbaLpwV.fasta\n",
      "\t/tmp/tmpK7eI5m.fasta\n",
      "\t/tmp/tmpKYlhp8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpR66kCi.lastz\n",
      "\t/tmp/tmpM08rp7.lastz\n",
      "\t/tmp/tmpOCh2HR.lastz\n",
      "\t/tmp/tmp1UunQ6.lastz\n",
      "\t/tmp/tmpXXYjt8.lastz\n",
      "\t/tmp/tmpl0Ywrj.lastz\n",
      "\t/tmp/tmpX4KbWM.lastz\n",
      "\t/tmp/tmpaynvLv.lastz\n",
      "\t/tmp/tmp1T5SVs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Dirofilaria_immitis_GCA_009829315.lastz\n",
      "Creating Dirofilaria_immitis_GCA_009829315 table\n",
      "Inserting data to Dirofilaria_immitis_GCA_009829315 table\n",
      "\n",
      "Running against Dirofilaria_repens_GCA_008729115.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmpiGCxp9.fasta\n",
      "\t/tmp/tmp8lrSOI.fasta\n",
      "\t/tmp/tmpRaoCHU.fasta\n",
      "\t/tmp/tmpGwwk34.fasta\n",
      "\t/tmp/tmpAzN9AH.fasta\n",
      "\t/tmp/tmpX0UDu7.fasta\n",
      "\t/tmp/tmpOZnZme.fasta\n",
      "\t/tmp/tmp2qHtfG.fasta\n",
      "\t/tmp/tmpxveyw8.fasta\n",
      "\t/tmp/tmpmPocTb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLFZsrO.lastz\n",
      "\t/tmp/tmpgKEVZo.lastz\n",
      "\t/tmp/tmpBeZfni.lastz\n",
      "\t/tmp/tmpLJDwYn.lastz\n",
      "\t/tmp/tmpCNTsHB.lastz\n",
      "\t/tmp/tmpfXYkq0.lastz\n",
      "\t/tmp/tmpk9xEf_.lastz\n",
      "\t/tmp/tmpSK7dRu.lastz\n",
      "\t/tmp/tmpkodUUs.lastz\n",
      "\t/tmp/tmpueoPDq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Dirofilaria_repens_GCA_008729115.lastz\n",
      "Creating Dirofilaria_repens_GCA_008729115 table\n",
      "Inserting data to Dirofilaria_repens_GCA_008729115 table\n",
      "\n",
      "Running against Dracunculus_medinensis_GCA_900625125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmplR6sor.fasta\n",
      "\t/tmp/tmp6IJe8P.fasta\n",
      "\t/tmp/tmpSXLGWJ.fasta\n",
      "\t/tmp/tmpPY84w3.fasta\n",
      "\t/tmp/tmpWfx2xv.fasta\n",
      "\t/tmp/tmpA1xebt.fasta\n",
      "\t/tmp/tmpEQer4e.fasta\n",
      "\t/tmp/tmp8APCRz.fasta\n",
      "\t/tmp/tmpJl1uzD.fasta\n",
      "\t/tmp/tmp6Z_QQM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyj79dD.lastz\n",
      "\t/tmp/tmppRaWJm.lastz\n",
      "\t/tmp/tmp6yGk0D.lastz\n",
      "\t/tmp/tmpcDAyjW.lastz\n",
      "\t/tmp/tmpqupXq5.lastz\n",
      "\t/tmp/tmpIoAGEv.lastz\n",
      "\t/tmp/tmpF3nKaO.lastz\n",
      "\t/tmp/tmpD28e10.lastz\n",
      "\t/tmp/tmpNszS6B.lastz\n",
      "\t/tmp/tmpTEHuz7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Dracunculus_medinensis_GCA_900625125.lastz\n",
      "Creating Dracunculus_medinensis_GCA_900625125 table\n",
      "Inserting data to Dracunculus_medinensis_GCA_900625125 table\n",
      "\n",
      "Running against Elaeophora_elaphi_GCA_000499685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEkVVTu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGYvUZO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Elaeophora_elaphi_GCA_000499685.lastz\n",
      "Creating Elaeophora_elaphi_GCA_000499685 table\n",
      "Inserting data to Elaeophora_elaphi_GCA_000499685 table\n",
      "\n",
      "Running against Enterobius_vermicularis_GCA_900576705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 15 queries...\n",
      "\t/tmp/tmpb7BOfM.fasta\n",
      "\t/tmp/tmpZrFSNT.fasta\n",
      "\t/tmp/tmpmH8dnd.fasta\n",
      "\t/tmp/tmp3yhsJi.fasta\n",
      "\t/tmp/tmpP0eAkX.fasta\n",
      "\t/tmp/tmphcJXqb.fasta\n",
      "\t/tmp/tmpjk0lHK.fasta\n",
      "\t/tmp/tmp5PACJT.fasta\n",
      "\t/tmp/tmpCGq0F9.fasta\n",
      "\t/tmp/tmpyLJ2qF.fasta\n",
      "\t/tmp/tmpCX3EsZ.fasta\n",
      "\t/tmp/tmpoyVXL6.fasta\n",
      "\t/tmp/tmpAU1_wQ.fasta\n",
      "\t/tmp/tmphuho7N.fasta\n",
      "\t/tmp/tmpzIkGxe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpu4teku.lastz\n",
      "\t/tmp/tmpHkOXVB.lastz\n",
      "\t/tmp/tmpL5wwGS.lastz\n",
      "\t/tmp/tmptTL4c7.lastz\n",
      "\t/tmp/tmpQZBc7x.lastz\n",
      "\t/tmp/tmp2p9p_I.lastz\n",
      "\t/tmp/tmpWV0NnC.lastz\n",
      "\t/tmp/tmpP_6XaJ.lastz\n",
      "\t/tmp/tmpzmrcz4.lastz\n",
      "\t/tmp/tmpmgu6wp.lastz\n",
      "\t/tmp/tmpydtGo1.lastz\n",
      "\t/tmp/tmp1EZFGX.lastz\n",
      "\t/tmp/tmpZGoacB.lastz\n",
      "\t/tmp/tmpQcnOof.lastz\n",
      "\t/tmp/tmpp1jHIN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Enterobius_vermicularis_GCA_900576705.lastz\n",
      "Creating Enterobius_vermicularis_GCA_900576705 table\n",
      "Inserting data to Enterobius_vermicularis_GCA_900576705 table\n",
      "\n",
      "Running against Gongylonema_pulchrum_GCA_900617915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 33 queries...\n",
      "\t/tmp/tmp3qkHDI.fasta\n",
      "\t/tmp/tmpGzunCx.fasta\n",
      "\t/tmp/tmpOYh7WS.fasta\n",
      "\t/tmp/tmp_BSNAi.fasta\n",
      "\t/tmp/tmpj4YTHQ.fasta\n",
      "\t/tmp/tmp2_Cttn.fasta\n",
      "\t/tmp/tmpCfPoa7.fasta\n",
      "\t/tmp/tmpOiDGO8.fasta\n",
      "\t/tmp/tmpv60JqK.fasta\n",
      "\t/tmp/tmplMOBjr.fasta\n",
      "\t/tmp/tmpT_WdZ3.fasta\n",
      "\t/tmp/tmpODYlne.fasta\n",
      "\t/tmp/tmpQBS5cm.fasta\n",
      "\t/tmp/tmpIQMWg2.fasta\n",
      "\t/tmp/tmpnvNLrB.fasta\n",
      "\t/tmp/tmpBnOSN9.fasta\n",
      "\t/tmp/tmpSROH5C.fasta\n",
      "\t/tmp/tmpWw_EDP.fasta\n",
      "\t/tmp/tmpI1F1Xi.fasta\n",
      "\t/tmp/tmpwn7aCv.fasta\n",
      "\t/tmp/tmpqL4Y6T.fasta\n",
      "\t/tmp/tmptlYw0Z.fasta\n",
      "\t/tmp/tmpOxWb70.fasta\n",
      "\t/tmp/tmpcyW8SP.fasta\n",
      "\t/tmp/tmpu9yhyA.fasta\n",
      "\t/tmp/tmpEw0O_s.fasta\n",
      "\t/tmp/tmpYhJ2gD.fasta\n",
      "\t/tmp/tmpt9wjBa.fasta\n",
      "\t/tmp/tmpznbjDD.fasta\n",
      "\t/tmp/tmpbk_j25.fasta\n",
      "\t/tmp/tmp4H_O5T.fasta\n",
      "\t/tmp/tmpCWy0xB.fasta\n",
      "\t/tmp/tmplXEWwy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEzGlbc.lastz\n",
      "\t/tmp/tmpvAdvAL.lastz\n",
      "\t/tmp/tmp5_ZEW3.lastz\n",
      "\t/tmp/tmpmdEN8q.lastz\n",
      "\t/tmp/tmpNNi9a4.lastz\n",
      "\t/tmp/tmpcSMvPA.lastz\n",
      "\t/tmp/tmp2AVYKY.lastz\n",
      "\t/tmp/tmpXJnnZf.lastz\n",
      "\t/tmp/tmpNsshml.lastz\n",
      "\t/tmp/tmpe3PSxV.lastz\n",
      "\t/tmp/tmpQSm1zH.lastz\n",
      "\t/tmp/tmpsy1bEW.lastz\n",
      "\t/tmp/tmpkId5Xs.lastz\n",
      "\t/tmp/tmpkcmi0k.lastz\n",
      "\t/tmp/tmpXZmlk0.lastz\n",
      "\t/tmp/tmpskF9PU.lastz\n",
      "\t/tmp/tmpV01lsp.lastz\n",
      "\t/tmp/tmp0vxf5Q.lastz\n",
      "\t/tmp/tmp82o2uJ.lastz\n",
      "\t/tmp/tmpB6SPoC.lastz\n",
      "\t/tmp/tmpxlhdHc.lastz\n",
      "\t/tmp/tmpwPJASX.lastz\n",
      "\t/tmp/tmpmYzwd4.lastz\n",
      "\t/tmp/tmpA8iX3V.lastz\n",
      "\t/tmp/tmpLcjHmo.lastz\n",
      "\t/tmp/tmpys7ceW.lastz\n",
      "\t/tmp/tmpA7fok8.lastz\n",
      "\t/tmp/tmpidjMYn.lastz\n",
      "\t/tmp/tmp9ZZIUG.lastz\n",
      "\t/tmp/tmpqhOC1M.lastz\n",
      "\t/tmp/tmp6qFq0y.lastz\n",
      "\t/tmp/tmpgSmgzk.lastz\n",
      "\t/tmp/tmpvzODdD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Gongylonema_pulchrum_GCA_900617915.lastz\n",
      "Creating Gongylonema_pulchrum_GCA_900617915 table\n",
      "Inserting data to Gongylonema_pulchrum_GCA_900617915 table\n",
      "\n",
      "Running against Litomosoides_sigmodontis_GCA_900537275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmp8Oa6j8.fasta\n",
      "\t/tmp/tmpbq1tKz.fasta\n",
      "\t/tmp/tmpwEuTZV.fasta\n",
      "\t/tmp/tmp54h91w.fasta\n",
      "\t/tmp/tmpPRDdrp.fasta\n",
      "\t/tmp/tmpc2xONh.fasta\n",
      "\t/tmp/tmpzyyZN6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpriEyc9.lastz\n",
      "\t/tmp/tmpX4QOYp.lastz\n",
      "\t/tmp/tmpDRpZLx.lastz\n",
      "\t/tmp/tmpG0SaQz.lastz\n",
      "\t/tmp/tmprfhd99.lastz\n",
      "\t/tmp/tmpdhvJoQ.lastz\n",
      "\t/tmp/tmpJuNnCX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Litomosoides_sigmodontis_GCA_900537275.lastz\n",
      "Creating Litomosoides_sigmodontis_GCA_900537275 table\n",
      "Inserting data to Litomosoides_sigmodontis_GCA_900537275 table\n",
      "\n",
      "Running against Loa_loa_GCF_000183805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpl96tXd.fasta\n",
      "\t/tmp/tmp4zktzX.fasta\n",
      "\t/tmp/tmpwXOaqN.fasta\n",
      "\t/tmp/tmp67auRH.fasta\n",
      "\t/tmp/tmpvsYmWm.fasta\n",
      "\t/tmp/tmpAM2Z8f.fasta\n",
      "\t/tmp/tmpSQXht2.fasta\n",
      "\t/tmp/tmpEvDZ8M.fasta\n",
      "\t/tmp/tmpmy9ruv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr1oLsn.lastz\n",
      "\t/tmp/tmp6dnn84.lastz\n",
      "\t/tmp/tmp6WUjnu.lastz\n",
      "\t/tmp/tmpE3WszG.lastz\n",
      "\t/tmp/tmppSFZGN.lastz\n",
      "\t/tmp/tmp2KNvFq.lastz\n",
      "\t/tmp/tmpf4oreE.lastz\n",
      "\t/tmp/tmpwJGFUI.lastz\n",
      "\t/tmp/tmpyp9oSE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Loa_loa_GCF_000183805.lastz\n",
      "Creating Loa_loa_GCF_000183805 table\n",
      "Inserting data to Loa_loa_GCF_000183805 table\n",
      "\n",
      "Running against Onchocerca_flexuosa_GCA_002249935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpjIAoD6.fasta\n",
      "\t/tmp/tmpasoOtc.fasta\n",
      "\t/tmp/tmpSOz43g.fasta\n",
      "\t/tmp/tmpg7MmPV.fasta\n",
      "\t/tmp/tmpjthVeU.fasta\n",
      "\t/tmp/tmpTiK7S7.fasta\n",
      "\t/tmp/tmpL6yGYJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE1JSUM.lastz\n",
      "\t/tmp/tmpuncd_5.lastz\n",
      "\t/tmp/tmpnAPqHC.lastz\n",
      "\t/tmp/tmpAx9hfp.lastz\n",
      "\t/tmp/tmp6s6Ijb.lastz\n",
      "\t/tmp/tmpmlrjMy.lastz\n",
      "\t/tmp/tmp4XyVGR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Onchocerca_flexuosa_GCA_002249935.lastz\n",
      "Creating Onchocerca_flexuosa_GCA_002249935 table\n",
      "Inserting data to Onchocerca_flexuosa_GCA_002249935 table\n",
      "\n",
      "Running against Onchocerca_ochengi_GCA_000950515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmp12u6Dr.fasta\n",
      "\t/tmp/tmpveFihQ.fasta\n",
      "\t/tmp/tmp7CM0Ut.fasta\n",
      "\t/tmp/tmpvzC7M6.fasta\n",
      "\t/tmp/tmpky_QnS.fasta\n",
      "\t/tmp/tmpBoWs2u.fasta\n",
      "\t/tmp/tmpsr7x6z.fasta\n",
      "\t/tmp/tmpzbyYZq.fasta\n",
      "\t/tmp/tmpoQCoTY.fasta\n",
      "\t/tmp/tmpl9lmUr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKOUgxk.lastz\n",
      "\t/tmp/tmpUZnAz4.lastz\n",
      "\t/tmp/tmpS9JtMP.lastz\n",
      "\t/tmp/tmpvbkttp.lastz\n",
      "\t/tmp/tmpbJBriQ.lastz\n",
      "\t/tmp/tmp5IsSCt.lastz\n",
      "\t/tmp/tmpDFbBcN.lastz\n",
      "\t/tmp/tmphVIZ09.lastz\n",
      "\t/tmp/tmp0N5Mpw.lastz\n",
      "\t/tmp/tmpYZMO2q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Onchocerca_ochengi_GCA_000950515.lastz\n",
      "Creating Onchocerca_ochengi_GCA_000950515 table\n",
      "Inserting data to Onchocerca_ochengi_GCA_000950515 table\n",
      "\n",
      "Running against Onchocerca_volvulus_GCA_000499405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmp1yefVj.fasta\n",
      "\t/tmp/tmpmrjjk8.fasta\n",
      "\t/tmp/tmpRDKOY0.fasta\n",
      "\t/tmp/tmpNTfOtj.fasta\n",
      "\t/tmp/tmp1p4q6G.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr1P4Kf.lastz\n",
      "\t/tmp/tmpgP3eFY.lastz\n",
      "\t/tmp/tmpicd0X2.lastz\n",
      "\t/tmp/tmpHyhQ_h.lastz\n",
      "\t/tmp/tmpi4HtU1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Onchocerca_volvulus_GCA_000499405.lastz\n",
      "Creating Onchocerca_volvulus_GCA_000499405 table\n",
      "Inserting data to Onchocerca_volvulus_GCA_000499405 table\n",
      "\n",
      "Running against Parascaris_univalens_GCA_002259215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 22 queries...\n",
      "\t/tmp/tmpDUklwY.fasta\n",
      "\t/tmp/tmp18FZaS.fasta\n",
      "\t/tmp/tmptOXq9l.fasta\n",
      "\t/tmp/tmpAARtUh.fasta\n",
      "\t/tmp/tmpg2ukev.fasta\n",
      "\t/tmp/tmpR9hWHV.fasta\n",
      "\t/tmp/tmp0r2jkg.fasta\n",
      "\t/tmp/tmp9f5RFu.fasta\n",
      "\t/tmp/tmpf1JomY.fasta\n",
      "\t/tmp/tmpCeo9Jy.fasta\n",
      "\t/tmp/tmpffaplx.fasta\n",
      "\t/tmp/tmp76lPz9.fasta\n",
      "\t/tmp/tmpHFhcmt.fasta\n",
      "\t/tmp/tmpJvgOjI.fasta\n",
      "\t/tmp/tmp8MGB8V.fasta\n",
      "\t/tmp/tmpkbG1jx.fasta\n",
      "\t/tmp/tmpQ_1F0L.fasta\n",
      "\t/tmp/tmp8iGFlG.fasta\n",
      "\t/tmp/tmpgfqA9i.fasta\n",
      "\t/tmp/tmphtSBNQ.fasta\n",
      "\t/tmp/tmpX1wuDd.fasta\n",
      "\t/tmp/tmpgeZ4SN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYNaAC8.lastz\n",
      "\t/tmp/tmp4TCObg.lastz\n",
      "\t/tmp/tmpFPaA3E.lastz\n",
      "\t/tmp/tmpryp_zX.lastz\n",
      "\t/tmp/tmpNwwoDE.lastz\n",
      "\t/tmp/tmpvxzlaw.lastz\n",
      "\t/tmp/tmpGplq69.lastz\n",
      "\t/tmp/tmpJ8ZciB.lastz\n",
      "\t/tmp/tmpwI6jsR.lastz\n",
      "\t/tmp/tmpeAFseP.lastz\n",
      "\t/tmp/tmpVFN4RE.lastz\n",
      "\t/tmp/tmpb4fXAP.lastz\n",
      "\t/tmp/tmpuhX4gI.lastz\n",
      "\t/tmp/tmpEWc9BB.lastz\n",
      "\t/tmp/tmp0Nr5BT.lastz\n",
      "\t/tmp/tmp1fgKcl.lastz\n",
      "\t/tmp/tmpUyz7HE.lastz\n",
      "\t/tmp/tmpNrq8_X.lastz\n",
      "\t/tmp/tmpKG7Prk.lastz\n",
      "\t/tmp/tmpgFfLQq.lastz\n",
      "\t/tmp/tmpjYKPVl.lastz\n",
      "\t/tmp/tmpk8mArc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Parascaris_univalens_GCA_002259215.lastz\n",
      "Creating Parascaris_univalens_GCA_002259215 table\n",
      "Inserting data to Parascaris_univalens_GCA_002259215 table\n",
      "\n",
      "Running against Plectus_murrayi_GCA_004785735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 23 queries...\n",
      "\t/tmp/tmpbFHO71.fasta\n",
      "\t/tmp/tmpQD2app.fasta\n",
      "\t/tmp/tmpGCvtoj.fasta\n",
      "\t/tmp/tmpS5DY1E.fasta\n",
      "\t/tmp/tmprnxAAw.fasta\n",
      "\t/tmp/tmprWuqj4.fasta\n",
      "\t/tmp/tmpz4XI2_.fasta\n",
      "\t/tmp/tmprqVADY.fasta\n",
      "\t/tmp/tmpKcsOPM.fasta\n",
      "\t/tmp/tmppgbOH1.fasta\n",
      "\t/tmp/tmpbpILM3.fasta\n",
      "\t/tmp/tmpLCbzat.fasta\n",
      "\t/tmp/tmp29oyzF.fasta\n",
      "\t/tmp/tmpy_yqGo.fasta\n",
      "\t/tmp/tmpWlbobS.fasta\n",
      "\t/tmp/tmpzzg51z.fasta\n",
      "\t/tmp/tmpzTqImS.fasta\n",
      "\t/tmp/tmpzf_dtC.fasta\n",
      "\t/tmp/tmpvOp7ir.fasta\n",
      "\t/tmp/tmpcp387u.fasta\n",
      "\t/tmp/tmpn6e2Io.fasta\n",
      "\t/tmp/tmpTCugYW.fasta\n",
      "\t/tmp/tmpJG0gmP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHe_h_J.lastz\n",
      "\t/tmp/tmpxCgZRg.lastz\n",
      "\t/tmp/tmp8WTvU_.lastz\n",
      "\t/tmp/tmpAeJyG2.lastz\n",
      "\t/tmp/tmp9jSmGO.lastz\n",
      "\t/tmp/tmpPKTAK1.lastz\n",
      "\t/tmp/tmpBUDRsU.lastz\n",
      "\t/tmp/tmpMczEcm.lastz\n",
      "\t/tmp/tmp7_YIDI.lastz\n",
      "\t/tmp/tmpsMoAOp.lastz\n",
      "\t/tmp/tmpNMoYHX.lastz\n",
      "\t/tmp/tmpErKXx9.lastz\n",
      "\t/tmp/tmpxlyYAv.lastz\n",
      "\t/tmp/tmpkt82IS.lastz\n",
      "\t/tmp/tmpbzkZSF.lastz\n",
      "\t/tmp/tmpWsKahe.lastz\n",
      "\t/tmp/tmpg4SI8_.lastz\n",
      "\t/tmp/tmptF2WxE.lastz\n",
      "\t/tmp/tmpo17IaY.lastz\n",
      "\t/tmp/tmpKa138q.lastz\n",
      "\t/tmp/tmpfsjrV4.lastz\n",
      "\t/tmp/tmpSYng_C.lastz\n",
      "\t/tmp/tmph0QzYN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Plectus_murrayi_GCA_004785735.lastz\n",
      "Creating Plectus_murrayi_GCA_004785735 table\n",
      "Inserting data to Plectus_murrayi_GCA_004785735 table\n",
      "\n",
      "Running against Plectus_sambesii_GCA_002796945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 19 queries...\n",
      "\t/tmp/tmpq84_bs.fasta\n",
      "\t/tmp/tmp7nwT8J.fasta\n",
      "\t/tmp/tmpK36vxs.fasta\n",
      "\t/tmp/tmpq8rZW8.fasta\n",
      "\t/tmp/tmpZnzSly.fasta\n",
      "\t/tmp/tmpFG6pEv.fasta\n",
      "\t/tmp/tmpWy_u9F.fasta\n",
      "\t/tmp/tmpuYBOhg.fasta\n",
      "\t/tmp/tmpEJapon.fasta\n",
      "\t/tmp/tmpkptH6C.fasta\n",
      "\t/tmp/tmpIjDxHK.fasta\n",
      "\t/tmp/tmp1Drm_k.fasta\n",
      "\t/tmp/tmp2PjiAx.fasta\n",
      "\t/tmp/tmpQpy1qP.fasta\n",
      "\t/tmp/tmpp5aWzD.fasta\n",
      "\t/tmp/tmpo2fKmM.fasta\n",
      "\t/tmp/tmpw3FiPo.fasta\n",
      "\t/tmp/tmpN9rYgI.fasta\n",
      "\t/tmp/tmpueB2F1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvpBdOg.lastz\n",
      "\t/tmp/tmp9yX0Po.lastz\n",
      "\t/tmp/tmpxAfFNU.lastz\n",
      "\t/tmp/tmpBX5Dmm.lastz\n",
      "\t/tmp/tmpJVzmSG.lastz\n",
      "\t/tmp/tmpcW_uaF.lastz\n",
      "\t/tmp/tmpiXZjS9.lastz\n",
      "\t/tmp/tmpmHuJxl.lastz\n",
      "\t/tmp/tmplDZo9S.lastz\n",
      "\t/tmp/tmp8os0QK.lastz\n",
      "\t/tmp/tmpkMwYLJ.lastz\n",
      "\t/tmp/tmpaZrNku.lastz\n",
      "\t/tmp/tmpqVgfxj.lastz\n",
      "\t/tmp/tmpeAqkj3.lastz\n",
      "\t/tmp/tmp0UnbxR.lastz\n",
      "\t/tmp/tmpxMrtCS.lastz\n",
      "\t/tmp/tmpyjSajb.lastz\n",
      "\t/tmp/tmpMw3tKp.lastz\n",
      "\t/tmp/tmpEvDz4_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Plectus_sambesii_GCA_002796945.lastz\n",
      "Creating Plectus_sambesii_GCA_002796945 table\n",
      "Inserting data to Plectus_sambesii_GCA_002796945 table\n",
      "\n",
      "Running against Thelazia_callipaeda_GCA_900618365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmpBfW7eH.fasta\n",
      "\t/tmp/tmpx0OlMc.fasta\n",
      "\t/tmp/tmpIp2gJF.fasta\n",
      "\t/tmp/tmpoMtNZy.fasta\n",
      "\t/tmp/tmpRpb1dD.fasta\n",
      "\t/tmp/tmpavIIKg.fasta\n",
      "\t/tmp/tmpJpRQXG.fasta\n",
      "\t/tmp/tmpOskM8l.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0ybyu0.lastz\n",
      "\t/tmp/tmpgyirIz.lastz\n",
      "\t/tmp/tmp4R0R6V.lastz\n",
      "\t/tmp/tmpXEiXKH.lastz\n",
      "\t/tmp/tmprfFyET.lastz\n",
      "\t/tmp/tmp2XimEi.lastz\n",
      "\t/tmp/tmpQls18f.lastz\n",
      "\t/tmp/tmp6e0jSo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Thelazia_callipaeda_GCA_900618365.lastz\n",
      "Creating Thelazia_callipaeda_GCA_900618365 table\n",
      "Inserting data to Thelazia_callipaeda_GCA_900618365 table\n",
      "\n",
      "Running against Toxocara_canis_GCA_000803305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 31 queries...\n",
      "\t/tmp/tmpr1GRS4.fasta\n",
      "\t/tmp/tmph5hA2A.fasta\n",
      "\t/tmp/tmp_f6SyG.fasta\n",
      "\t/tmp/tmpGmFcEs.fasta\n",
      "\t/tmp/tmpncYCPN.fasta\n",
      "\t/tmp/tmpKQyaWN.fasta\n",
      "\t/tmp/tmpORAx6w.fasta\n",
      "\t/tmp/tmpptay6K.fasta\n",
      "\t/tmp/tmp_27ZrX.fasta\n",
      "\t/tmp/tmpqIYxPL.fasta\n",
      "\t/tmp/tmpX12r_L.fasta\n",
      "\t/tmp/tmpTTlnSA.fasta\n",
      "\t/tmp/tmp32iRGt.fasta\n",
      "\t/tmp/tmpl2X580.fasta\n",
      "\t/tmp/tmpG3cs2t.fasta\n",
      "\t/tmp/tmpTyEzWX.fasta\n",
      "\t/tmp/tmp0nDqHz.fasta\n",
      "\t/tmp/tmpQjoLhz.fasta\n",
      "\t/tmp/tmpIhUQAU.fasta\n",
      "\t/tmp/tmpH5RMCA.fasta\n",
      "\t/tmp/tmpo8kRML.fasta\n",
      "\t/tmp/tmpC1Dp4V.fasta\n",
      "\t/tmp/tmpEY3VIr.fasta\n",
      "\t/tmp/tmpTdc_NT.fasta\n",
      "\t/tmp/tmp5etBtQ.fasta\n",
      "\t/tmp/tmp4f_UH4.fasta\n",
      "\t/tmp/tmpCnb95K.fasta\n",
      "\t/tmp/tmpXt4eK1.fasta\n",
      "\t/tmp/tmpgOYRIH.fasta\n",
      "\t/tmp/tmpPeNKzw.fasta\n",
      "\t/tmp/tmpWgsk7T.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkIxDX2.lastz\n",
      "\t/tmp/tmpy2rLY0.lastz\n",
      "\t/tmp/tmpJqRSUS.lastz\n",
      "\t/tmp/tmppWmeZs.lastz\n",
      "\t/tmp/tmpaYXtkM.lastz\n",
      "\t/tmp/tmpkszhG3.lastz\n",
      "\t/tmp/tmp_n8ush.lastz\n",
      "\t/tmp/tmphw4beJ.lastz\n",
      "\t/tmp/tmp3dnzsY.lastz\n",
      "\t/tmp/tmprnOkVp.lastz\n",
      "\t/tmp/tmpuOIabK.lastz\n",
      "\t/tmp/tmpJB9eZJ.lastz\n",
      "\t/tmp/tmpOMz9hF.lastz\n",
      "\t/tmp/tmp6xaiy1.lastz\n",
      "\t/tmp/tmpT84fUu.lastz\n",
      "\t/tmp/tmpBU7otO.lastz\n",
      "\t/tmp/tmplvPdwD.lastz\n",
      "\t/tmp/tmpeqibNA.lastz\n",
      "\t/tmp/tmpKoAnYT.lastz\n",
      "\t/tmp/tmp0WGLer.lastz\n",
      "\t/tmp/tmpy0l7Jh.lastz\n",
      "\t/tmp/tmpEn_VbX.lastz\n",
      "\t/tmp/tmpQlqbNp.lastz\n",
      "\t/tmp/tmp9ImzxI.lastz\n",
      "\t/tmp/tmp6Pepw4.lastz\n",
      "\t/tmp/tmpbqBihI.lastz\n",
      "\t/tmp/tmpbgOQP9.lastz\n",
      "\t/tmp/tmpjNReT2.lastz\n",
      "\t/tmp/tmpuQcvVM.lastz\n",
      "\t/tmp/tmpCkLM2T.lastz\n",
      "\t/tmp/tmpAJjRHF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Toxocara_canis_GCA_000803305.lastz\n",
      "Creating Toxocara_canis_GCA_000803305 table\n",
      "Inserting data to Toxocara_canis_GCA_000803305 table\n",
      "\n",
      "Running against Wuchereria_bancrofti_GCA_005281725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmp7Fw8L_.fasta\n",
      "\t/tmp/tmpg89TmP.fasta\n",
      "\t/tmp/tmpGjopVW.fasta\n",
      "\t/tmp/tmp4f2FN5.fasta\n",
      "\t/tmp/tmpvXz6Mh.fasta\n",
      "\t/tmp/tmpmbAHT6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZSadgX.lastz\n",
      "\t/tmp/tmp60EEAG.lastz\n",
      "\t/tmp/tmpNxTdBk.lastz\n",
      "\t/tmp/tmp2idbxz.lastz\n",
      "\t/tmp/tmp7JcU9U.lastz\n",
      "\t/tmp/tmpwRHoCG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Wuchereria_bancrofti_GCA_005281725.lastz\n",
      "Creating Wuchereria_bancrofti_GCA_005281725 table\n",
      "Inserting data to Wuchereria_bancrofti_GCA_005281725 table\n",
      "\n",
      "Running against Brugia_malayi_GCA_000002995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpn2edTh.fasta\n",
      "\t/tmp/tmpPOa4wZ.fasta\n",
      "\t/tmp/tmporyDpf.fasta\n",
      "\t/tmp/tmpF9ASA5.fasta\n",
      "\t/tmp/tmpKvwVrY.fasta\n",
      "\t/tmp/tmpvTKhCM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0pNYIg.lastz\n",
      "\t/tmp/tmpNQQAFQ.lastz\n",
      "\t/tmp/tmpSGALVZ.lastz\n",
      "\t/tmp/tmp09daMS.lastz\n",
      "\t/tmp/tmpDk16Us.lastz\n",
      "\t/tmp/tmpEfeXDR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade3/validate_intervals/lastz/Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_Brugia_malayi_GCA_000002995.lastz\n",
      "Creating Brugia_malayi_GCA_000002995 table\n",
      "Inserting data to Brugia_malayi_GCA_000002995 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/nematoda-clade3/extract_probes_from_group/nematoda-clade3_genome.conf --lastz results/phyluce/nematoda-clade3/validate_intervals/lastz --probes 120 --probe-prefix uce_nematoda-clade3_ --name-pattern \"Brugia_malayi_GCA_000002995_+9_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/nematoda-clade3/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 09:11:20,502 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 09:11:20,506 - Phyluce - INFO - ---- Working on Acanthocheilonema_viteae_GCA_900537255 genome ---\n",
      "2020-02-11 09:11:20,508 - Phyluce - INFO - Reading Acanthocheilonema_viteae_GCA_900537255 genome\n",
      "2020-02-11 09:11:31,064 - Phyluce - INFO - Acanthocheilonema_viteae_GCA_900537255: 2362 uces, 181 dupes, 2181 non-dupes, 0 orient drop, 11 length drop, 2169 written\n",
      "2020-02-11 09:11:31,065 - Phyluce - INFO - -------- Working on Anisakis_simplex_GCA_900617985 genome -------\n",
      "2020-02-11 09:11:31,066 - Phyluce - INFO - Reading Anisakis_simplex_GCA_900617985 genome\n",
      "2020-02-11 09:11:38,312 - Phyluce - INFO - Anisakis_simplex_GCA_900617985: 1754 uces, 293 dupes, 1461 non-dupes, 2 orient drop, 4 length drop, 1452 written\n",
      "2020-02-11 09:11:38,312 - Phyluce - INFO - ---------- Working on Ascaris_suum_GCA_000187025 genome ---------\n",
      "2020-02-11 09:11:38,327 - Phyluce - INFO - Reading Ascaris_suum_GCA_000187025 genome\n",
      "2020-02-11 09:11:46,797 - Phyluce - INFO - Ascaris_suum_GCA_000187025: 1954 uces, 331 dupes, 1623 non-dupes, 22 orient drop, 51 length drop, 1550 written\n",
      "2020-02-11 09:11:46,797 - Phyluce - INFO - ---- Working on Baylisascaris_schroederi_GCA_006503575 genome ---\n",
      "2020-02-11 09:11:46,798 - Phyluce - INFO - Reading Baylisascaris_schroederi_GCA_006503575 genome\n",
      "2020-02-11 09:11:55,141 - Phyluce - INFO - Baylisascaris_schroederi_GCA_006503575: 1920 uces, 345 dupes, 1575 non-dupes, 0 orient drop, 22 length drop, 1553 written\n",
      "2020-02-11 09:11:55,142 - Phyluce - INFO - --------- Working on Brugia_pahangi_GCA_001280985 genome --------\n",
      "2020-02-11 09:11:55,142 - Phyluce - INFO - Reading Brugia_pahangi_GCA_001280985 genome\n",
      "2020-02-11 09:12:04,937 - Phyluce - INFO - Brugia_pahangi_GCA_001280985: 2377 uces, 314 dupes, 2063 non-dupes, 6 orient drop, 63 length drop, 1991 written\n",
      "2020-02-11 09:12:04,937 - Phyluce - INFO - --------- Working on Brugia_timori_GCA_900618025 genome ---------\n",
      "2020-02-11 09:12:04,954 - Phyluce - INFO - Reading Brugia_timori_GCA_900618025 genome\n",
      "2020-02-11 09:12:12,881 - Phyluce - INFO - Brugia_timori_GCA_900618025: 2086 uces, 168 dupes, 1918 non-dupes, 0 orient drop, 1 length drop, 1897 written\n",
      "2020-02-11 09:12:12,881 - Phyluce - INFO - ------ Working on Dirofilaria_immitis_GCA_009829315 genome ------\n",
      "2020-02-11 09:12:12,882 - Phyluce - INFO - Reading Dirofilaria_immitis_GCA_009829315 genome\n",
      "2020-02-11 09:12:17,479 - Phyluce - INFO - Dirofilaria_immitis_GCA_009829315: 1428 uces, 70 dupes, 1358 non-dupes, 1 orient drop, 1 length drop, 1356 written\n",
      "2020-02-11 09:12:17,479 - Phyluce - INFO - ------- Working on Dirofilaria_repens_GCA_008729115 genome ------\n",
      "2020-02-11 09:12:17,480 - Phyluce - INFO - Reading Dirofilaria_repens_GCA_008729115 genome\n",
      "2020-02-11 09:12:27,133 - Phyluce - INFO - Dirofilaria_repens_GCA_008729115: 2367 uces, 258 dupes, 2109 non-dupes, 4 orient drop, 15 length drop, 2090 written\n",
      "2020-02-11 09:12:27,134 - Phyluce - INFO - ----- Working on Dracunculus_medinensis_GCA_900625125 genome ----\n",
      "2020-02-11 09:12:27,135 - Phyluce - INFO - Reading Dracunculus_medinensis_GCA_900625125 genome\n",
      "2020-02-11 09:12:33,308 - Phyluce - INFO - Dracunculus_medinensis_GCA_900625125: 1783 uces, 174 dupes, 1609 non-dupes, 2 orient drop, 9 length drop, 1598 written\n",
      "2020-02-11 09:12:33,308 - Phyluce - INFO - ------- Working on Elaeophora_elaphi_GCA_000499685 genome -------\n",
      "2020-02-11 09:12:33,309 - Phyluce - INFO - Reading Elaeophora_elaphi_GCA_000499685 genome\n",
      "2020-02-11 09:12:33,329 - Phyluce - INFO - Elaeophora_elaphi_GCA_000499685: 6 uces, 0 dupes, 6 non-dupes, 0 orient drop, 0 length drop, 6 written\n",
      "2020-02-11 09:12:33,329 - Phyluce - INFO - ---- Working on Enterobius_vermicularis_GCA_900576705 genome ----\n",
      "2020-02-11 09:12:33,330 - Phyluce - INFO - Reading Enterobius_vermicularis_GCA_900576705 genome\n",
      "2020-02-11 09:12:39,530 - Phyluce - INFO - Enterobius_vermicularis_GCA_900576705: 1691 uces, 175 dupes, 1516 non-dupes, 3 orient drop, 7 length drop, 1506 written\n",
      "2020-02-11 09:12:39,530 - Phyluce - INFO - ------ Working on Gongylonema_pulchrum_GCA_900617915 genome -----\n",
      "2020-02-11 09:12:39,531 - Phyluce - INFO - Reading Gongylonema_pulchrum_GCA_900617915 genome\n",
      "2020-02-11 09:12:52,449 - Phyluce - INFO - Gongylonema_pulchrum_GCA_900617915: 2155 uces, 753 dupes, 1402 non-dupes, 1 orient drop, 30 length drop, 1370 written\n",
      "2020-02-11 09:12:52,449 - Phyluce - INFO - ---- Working on Litomosoides_sigmodontis_GCA_900537275 genome ---\n",
      "2020-02-11 09:12:52,450 - Phyluce - INFO - Reading Litomosoides_sigmodontis_GCA_900537275 genome\n",
      "2020-02-11 09:13:01,197 - Phyluce - INFO - Litomosoides_sigmodontis_GCA_900537275: 2361 uces, 191 dupes, 2170 non-dupes, 26 orient drop, 6 length drop, 2138 written\n",
      "2020-02-11 09:13:01,197 - Phyluce - INFO - ------------ Working on Loa_loa_GCF_000183805 genome ------------\n",
      "2020-02-11 09:13:01,198 - Phyluce - INFO - Reading Loa_loa_GCF_000183805 genome\n",
      "2020-02-11 09:13:10,946 - Phyluce - INFO - Loa_loa_GCF_000183805: 2371 uces, 203 dupes, 2168 non-dupes, 0 orient drop, 28 length drop, 2140 written\n",
      "2020-02-11 09:13:10,947 - Phyluce - INFO - ------ Working on Onchocerca_flexuosa_GCA_002249935 genome ------\n",
      "2020-02-11 09:13:10,947 - Phyluce - INFO - Reading Onchocerca_flexuosa_GCA_002249935 genome\n",
      "2020-02-11 09:13:19,249 - Phyluce - INFO - Onchocerca_flexuosa_GCA_002249935: 2227 uces, 157 dupes, 2070 non-dupes, 4 orient drop, 22 length drop, 2044 written\n",
      "2020-02-11 09:13:19,249 - Phyluce - INFO - ------- Working on Onchocerca_ochengi_GCA_000950515 genome ------\n",
      "2020-02-11 09:13:19,254 - Phyluce - INFO - Reading Onchocerca_ochengi_GCA_000950515 genome\n",
      "2020-02-11 09:13:28,599 - Phyluce - INFO - Onchocerca_ochengi_GCA_000950515: 2364 uces, 224 dupes, 2140 non-dupes, 0 orient drop, 13 length drop, 2127 written\n",
      "2020-02-11 09:13:28,599 - Phyluce - INFO - ------ Working on Onchocerca_volvulus_GCA_000499405 genome ------\n",
      "2020-02-11 09:13:28,600 - Phyluce - INFO - Reading Onchocerca_volvulus_GCA_000499405 genome\n",
      "2020-02-11 09:13:38,021 - Phyluce - INFO - Onchocerca_volvulus_GCA_000499405: 2372 uces, 172 dupes, 2200 non-dupes, 28 orient drop, 49 length drop, 2123 written\n",
      "2020-02-11 09:13:38,021 - Phyluce - INFO - ------ Working on Parascaris_univalens_GCA_002259215 genome -----\n",
      "2020-02-11 09:13:38,022 - Phyluce - INFO - Reading Parascaris_univalens_GCA_002259215 genome\n",
      "2020-02-11 09:13:46,549 - Phyluce - INFO - Parascaris_univalens_GCA_002259215: 1941 uces, 245 dupes, 1696 non-dupes, 11 orient drop, 32 length drop, 1653 written\n",
      "2020-02-11 09:13:46,549 - Phyluce - INFO - -------- Working on Plectus_murrayi_GCA_004785735 genome --------\n",
      "2020-02-11 09:13:46,550 - Phyluce - INFO - Reading Plectus_murrayi_GCA_004785735 genome\n",
      "2020-02-11 09:13:53,553 - Phyluce - INFO - Plectus_murrayi_GCA_004785735: 1193 uces, 241 dupes, 952 non-dupes, 0 orient drop, 6 length drop, 946 written\n",
      "2020-02-11 09:13:53,553 - Phyluce - INFO - -------- Working on Plectus_sambesii_GCA_002796945 genome -------\n",
      "2020-02-11 09:13:53,554 - Phyluce - INFO - Reading Plectus_sambesii_GCA_002796945 genome\n",
      "2020-02-11 09:13:58,843 - Phyluce - INFO - Plectus_sambesii_GCA_002796945: 1210 uces, 499 dupes, 711 non-dupes, 0 orient drop, 3 length drop, 707 written\n",
      "2020-02-11 09:13:58,843 - Phyluce - INFO - ------ Working on Thelazia_callipaeda_GCA_900618365 genome ------\n",
      "2020-02-11 09:13:58,844 - Phyluce - INFO - Reading Thelazia_callipaeda_GCA_900618365 genome\n",
      "2020-02-11 09:14:07,871 - Phyluce - INFO - Thelazia_callipaeda_GCA_900618365: 2296 uces, 163 dupes, 2133 non-dupes, 0 orient drop, 15 length drop, 2118 written\n",
      "2020-02-11 09:14:07,872 - Phyluce - INFO - --------- Working on Toxocara_canis_GCA_000803305 genome --------\n",
      "2020-02-11 09:14:07,874 - Phyluce - INFO - Reading Toxocara_canis_GCA_000803305 genome\n",
      "2020-02-11 09:14:16,939 - Phyluce - INFO - Toxocara_canis_GCA_000803305: 1959 uces, 314 dupes, 1645 non-dupes, 1 orient drop, 84 length drop, 1560 written\n",
      "2020-02-11 09:14:16,939 - Phyluce - INFO - ------ Working on Wuchereria_bancrofti_GCA_005281725 genome -----\n",
      "2020-02-11 09:14:16,940 - Phyluce - INFO - Reading Wuchereria_bancrofti_GCA_005281725 genome\n",
      "2020-02-11 09:14:26,655 - Phyluce - INFO - Wuchereria_bancrofti_GCA_005281725: 2382 uces, 229 dupes, 2153 non-dupes, 25 orient drop, 70 length drop, 2058 written\n",
      "2020-02-11 09:14:26,655 - Phyluce - INFO - --------- Working on Brugia_malayi_GCA_000002995 genome ---------\n",
      "2020-02-11 09:14:26,656 - Phyluce - INFO - Reading Brugia_malayi_GCA_000002995 genome\n",
      "2020-02-11 09:14:36,412 - Phyluce - INFO - Brugia_malayi_GCA_000002995: 2385 uces, 185 dupes, 2200 non-dupes, 40 orient drop, 31 length drop, 2129 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/nematoda-clade3/extract_probes_from_group/probe_fasta --output results/phyluce/nematoda-clade3/extract_probes_from_group/multifastas.sqlite --base-taxon Brugia_malayi_GCA_000002995;\n",
      "acanthocheilonema_viteae_gca_900537255...\n",
      "anisakis_simplex_gca_900617985..\n",
      "ascaris_suum_gca_000187025..\n",
      "baylisascaris_schroederi_gca_006503575..\n",
      "brugia_pahangi_gca_001280985..\n",
      "brugia_timori_gca_900618025..\n",
      "dirofilaria_immitis_gca_009829315..\n",
      "dirofilaria_repens_gca_008729115...\n",
      "dracunculus_medinensis_gca_900625125..\n",
      "elaeophora_elaphi_gca_000499685.\n",
      "enterobius_vermicularis_gca_900576705..\n",
      "gongylonema_pulchrum_gca_900617915..\n",
      "litomosoides_sigmodontis_gca_900537275...\n",
      "loa_loa_gcf_000183805...\n",
      "onchocerca_flexuosa_gca_002249935...\n",
      "onchocerca_ochengi_gca_000950515...\n",
      "onchocerca_volvulus_gca_000499405...\n",
      "parascaris_univalens_gca_002259215..\n",
      "plectus_murrayi_gca_004785735.\n",
      "plectus_sambesii_gca_002796945.\n",
      "thelazia_callipaeda_gca_900618365...\n",
      "toxocara_canis_gca_000803305..\n",
      "wuchereria_bancrofti_gca_005281725...\n",
      "brugia_malayi_gca_000002995...\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade3/extract_probes_from_group/multifastas.sqlite --base-taxon Brugia_malayi_GCA_000002995;\n",
      "Loci shared by 0 taxa:\t2,375.0\n",
      "Loci shared by 1 taxa:\t2,375.0\n",
      "Loci shared by 2 taxa:\t2,345.0\n",
      "Loci shared by 3 taxa:\t2,329.0\n",
      "Loci shared by 4 taxa:\t2,293.0\n",
      "Loci shared by 5 taxa:\t2,263.0\n",
      "Loci shared by 6 taxa:\t2,237.0\n",
      "Loci shared by 7 taxa:\t2,225.0\n",
      "Loci shared by 8 taxa:\t2,214.0\n",
      "Loci shared by 9 taxa:\t2,203.0\n",
      "Loci shared by 10 taxa:\t2,184.0\n",
      "Loci shared by 11 taxa:\t2,160.0\n",
      "Loci shared by 12 taxa:\t2,118.0\n",
      "Loci shared by 13 taxa:\t2,012.0\n",
      "Loci shared by 14 taxa:\t1,913.0\n",
      "Loci shared by 15 taxa:\t1,820.0\n",
      "Loci shared by 16 taxa:\t1,732.0\n",
      "Loci shared by 17 taxa:\t1,612.0\n",
      "Loci shared by 18 taxa:\t1,440.0\n",
      "Loci shared by 19 taxa:\t1,187.0\n",
      "Loci shared by 20 taxa:\t878.0\n",
      "Loci shared by 21 taxa:\t493.0\n",
      "Loci shared by 22 taxa:\t203.0\n",
      "Loci shared by 23 taxa:\t45.0\n",
      "Loci shared by 24 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade3/extract_probes_from_group/multifastas.sqlite  --base-taxon Brugia_malayi_GCA_000002995 --output results/phyluce/nematoda-clade3/extract_probes_from_group/Brugia_malayi_GCA_000002995+9-back-to-22.conf --specific-counts 22;\n",
      "Counter({'acanthocheilonema_viteae_gca_900537255': 203, 'brugia_malayi_gca_000002995': 203, 'onchocerca_ochengi_gca_000950515': 203, 'onchocerca_volvulus_gca_000499405': 202, 'loa_loa_gcf_000183805': 202, 'litomosoides_sigmodontis_gca_900537275': 202, 'thelazia_callipaeda_gca_900618365': 202, 'dirofilaria_repens_gca_008729115': 201, 'onchocerca_flexuosa_gca_002249935': 201, 'ascaris_suum_gca_000187025': 200, 'parascaris_univalens_gca_002259215': 200, 'baylisascaris_schroederi_gca_006503575': 200, 'dracunculus_medinensis_gca_900625125': 199, 'enterobius_vermicularis_gca_900576705': 199, 'wuchereria_bancrofti_gca_005281725': 198, 'toxocara_canis_gca_000803305': 198, 'brugia_pahangi_gca_001280985': 198, 'brugia_timori_gca_900618025': 195, 'anisakis_simplex_gca_900617985': 194, 'plectus_murrayi_gca_004785735': 192, 'gongylonema_pulchrum_gca_900617915': 181, 'dirofilaria_immitis_gca_009829315': 174, 'plectus_sambesii_gca_002796945': 162, 'elaeophora_elaphi_gca_000499685': 2})\n",
      "Total loci = 203\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(22)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/nematoda-clade3/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/nematoda-clade3/extract_probes_from_group/Brugia_malayi_GCA_000002995+9-back-to-22.conf --probe-prefix uce_nematoda-clade3_ --designer rnplattii --design nematoda-clade3_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGNNGGGGGNNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGGGGGGNNGGGGGGGGGGGGNNNNGG\n",
      "\n",
      "\n",
      "Conserved locus count = 203\n",
      "Probe Count = 8938\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list.fasta --query results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  09:15:08\n",
      "Ended:  Tue Feb 11, 2020  09:15:25\n",
      "Time for execution:  0.291917149226 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list.fasta --lastz results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_nematoda-clade3_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 8937 fasta sequences.  Filtered 0 duplicates. Kept 8938.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/nematoda-clade3/final_probe_design/nematoda-clade3_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 09:21:43 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 8938\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 715040\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3949766\n",
      "Max number of word counting entries: 88909241\n",
      "\n",
      "# comparing sequences from          0  to       1489\n",
      ".---------- new table with     1132 representatives\n",
      "# comparing sequences from       1489  to       2730\n",
      "----------    232 remaining sequences to the next cycle\n",
      "---------- new table with      757 representatives\n",
      "# comparing sequences from       2498  to       3571\n",
      "----------    228 remaining sequences to the next cycle\n",
      "---------- new table with      657 representatives\n",
      "# comparing sequences from       3343  to       4275\n",
      "----------     93 remaining sequences to the next cycle\n",
      "---------- new table with      643 representatives\n",
      "# comparing sequences from       4182  to       4974\n",
      "----------    125 remaining sequences to the next cycle\n",
      "---------- new table with      521 representatives\n",
      "# comparing sequences from       4849  to       5530\n",
      "----------    204 remaining sequences to the next cycle\n",
      "---------- new table with      350 representatives\n",
      "# comparing sequences from       5326  to       5928\n",
      "----------    224 remaining sequences to the next cycle\n",
      "---------- new table with      282 representatives\n",
      "# comparing sequences from       5704  to       6243\n",
      "----------    193 remaining sequences to the next cycle\n",
      "---------- new table with      277 representatives\n",
      "# comparing sequences from       6050  to       6531\n",
      "----------     25 remaining sequences to the next cycle\n",
      "---------- new table with      345 representatives\n",
      "# comparing sequences from       6506  to       6911\n",
      "99.9%---------- new table with      293 representatives\n",
      "# comparing sequences from       6911  to       7248\n",
      "----------    140 remaining sequences to the next cycle\n",
      "---------- new table with      154 representatives\n",
      "# comparing sequences from       7108  to       7413\n",
      "99.9%---------- new table with      239 representatives\n",
      "# comparing sequences from       7413  to       7667\n",
      "99.9%---------- new table with      200 representatives\n",
      "# comparing sequences from       7667  to       7878\n",
      "----------     27 remaining sequences to the next cycle\n",
      "---------- new table with      148 representatives\n",
      "# comparing sequences from       7851  to       8032\n",
      "99.9%---------- new table with      144 representatives\n",
      "# comparing sequences from       8032  to       8938\n",
      "....................---------- new table with      704 representatives\n",
      "\n",
      "     8938  finished       6846  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 89M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.59\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clade 4a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'nematoda-clade4a'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Parastrongyloides_trichosuri_GCA_000941615\",\n",
    "               \"Rhabditophanes_sp_GCA_000944355\", \n",
    "               \"Strongyloides_papillosus_GCA_005656395\",\n",
    "               \"Strongyloides_ratti_GCF_001040885\",\n",
    "               \"Strongyloides_venezuelensis_GCA_001028725\" ]\n",
    "                    \n",
    "reference_taxon = \"Strongyloides_stercoralis_GCA_000947215\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000941615.1_P_trichosuri_KNP_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 13271380 bytes  5308568.80 bytes/sec\n",
      "total size is 13268028  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000944355.1_Rhabditophanes_sp_KR3021_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 14896511 bytes  4256158.00 bytes/sec\n",
      "total size is 14892751  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005656395.1_S_papillosus_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 18192319 bytes  5197817.43 bytes/sec\n",
      "total size is 18187760  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001040885.1_S_ratti_ED321_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12971180 bytes  5188488.80 bytes/sec\n",
      "total size is 12967903  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000947215.1_S_stercoralis_PV0001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12998104 bytes  5199258.40 bytes/sec\n",
      "total size is 12994812  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001028725.1_S_venezuelensis_HH1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16270365 bytes  4648687.71 bytes/sec\n",
      "total size is 16266274  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487852 (\"sim_Parastrongyloides_trichosuri_GCA_000941615\") has been submitted\n",
      "Your job 5487853 (\"sim_Rhabditophanes_sp_GCA_000944355\") has been submitted\n",
      "Your job 5487854 (\"sim_Strongyloides_papillosus_GCA_005656395\") has been submitted\n",
      "Your job 5487855 (\"sim_Strongyloides_ratti_GCF_001040885\") has been submitted\n",
      "Your job 5487856 (\"sim_Strongyloides_venezuelensis_GCA_001028725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/nematoda-clade4a/cleaned_genomes/Strongyloides_stercoralis_GCA_000947215_formatted.fas path=results/phyluce/nematoda-clade4a/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487857 (\"map_Parastrongyloides_trichosuri_GCA_000941615\") has been submitted\n",
      "Your job 5487858 (\"map_Rhabditophanes_sp_GCA_000944355\") has been submitted\n",
      "Your job 5487859 (\"map_Strongyloides_papillosus_GCA_005656395\") has been submitted\n",
      "Your job 5487860 (\"map_Strongyloides_ratti_GCF_001040885\") has been submitted\n",
      "Your job 5487861 (\"map_Strongyloides_venezuelensis_GCA_001028725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487862 (\"merge_Parastrongyloides_trichosuri_GCA_000941615\") has been submitted\n",
      "Your job 5487863 (\"merge_Rhabditophanes_sp_GCA_000944355\") has been submitted\n",
      "Your job 5487864 (\"merge_Strongyloides_papillosus_GCA_005656395\") has been submitted\n",
      "Your job 5487865 (\"merge_Strongyloides_ratti_GCF_001040885\") has been submitted\n",
      "Your job 5487866 (\"merge_Strongyloides_venezuelensis_GCA_001028725\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 12706 sequences from Parastrongyloides_trichosuri_GCA_000941615_merged.bed.  Filtered 9665 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3041.\n",
      "Screened 174 sequences from Rhabditophanes_sp_GCA_000944355_merged.bed.  Filtered 159 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15.\n",
      "Screened 28730 sequences from Strongyloides_papillosus_GCA_005656395_merged.bed.  Filtered 21237 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7493.\n",
      "Screened 48398 sequences from Strongyloides_ratti_GCF_001040885_merged.bed.  Filtered 38225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10173.\n",
      "Screened 27998 sequences from Strongyloides_venezuelensis_GCA_001028725_merged.bed.  Filtered 20567 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7431.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/nematoda-clade4a/initial_intervals/Strongyloides_venezuelensis_GCA_001028725_merged.bed --twobit results/phyluce/nematoda-clade4a/cleaned_genomes/Strongyloides_stercoralis_GCA_000947215_formatted.2bit --output results/phyluce/nematoda-clade4a/initial_intervals/Strongyloides_venezuelensis_GCA_001028725_stripped.bed;\n",
      "parastrongyloides_trichosuri_gca_000941615....\n",
      "rhabditophanes_sp_gca_000944355.\n",
      "strongyloides_papillosus_gca_005656395........\n",
      "strongyloides_ratti_gcf_001040885...........\n",
      "strongyloides_venezuelensis_gca_001028725........\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade4a/initial_intervals/nematoda-clade4a-to-Strongyloides_stercoralis_GCA_000947215.sqlite --base-taxon Strongyloides_stercoralis_GCA_000947215\n",
      "Loci shared by Strongyloides_stercoralis_GCA_000947215 + 0 taxa:\t15,077.0\n",
      "Loci shared by Strongyloides_stercoralis_GCA_000947215 + 1 taxa:\t15,077.0\n",
      "Loci shared by Strongyloides_stercoralis_GCA_000947215 + 2 taxa:\t8,081.0\n",
      "Loci shared by Strongyloides_stercoralis_GCA_000947215 + 3 taxa:\t5,023.0\n",
      "Loci shared by Strongyloides_stercoralis_GCA_000947215 + 4 taxa:\t1,657.0\n",
      "Loci shared by Strongyloides_stercoralis_GCA_000947215 + 5 taxa:\t9.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade4a/initial_intervals/nematoda-clade4a-to-Strongyloides_stercoralis_GCA_000947215.sqlite --base-taxon Strongyloides_stercoralis_GCA_000947215 --output results/phyluce/nematoda-clade4a/initial_intervals/Strongyloides_stercoralis_GCA_000947215_+4.bed --specific-counts 4;\n",
      "Counter({'parastrongyloides_trichosuri_gca_000941615': 1657, 'strongyloides_ratti_gcf_001040885': 1657, 'strongyloides_papillosus_gca_005656395': 1657, 'strongyloides_venezuelensis_gca_001028725': 1653, 'rhabditophanes_sp_gca_000944355': 13})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 4\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/nematoda-clade4a/initial_intervals/Strongyloides_stercoralis_GCA_000947215_+4.bed --twobit results/phyluce/nematoda-clade4a/cleaned_genomes/Strongyloides_stercoralis_GCA_000947215_formatted.2bit --buffer-to 160 --output results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4.fasta;\n",
      "Screened 1657 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 1657.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4.fasta --probe-prefix uce_nematoda-clade4a_ --design nematoda-clade4a_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 1425\n",
      "Probe Count = 2546\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas --query results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  14:26:52\n",
      "Ended:  Tue Feb 11, 2020  14:26:54\n",
      "Time for execution:  0.0234659830729 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas                        --lastz results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes_vself.lastz                       --probe-prefix=uce_nematoda-clade4a_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 2545 fasta sequences.  Filtered 143 duplicates. Kept 2279.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/nematoda-clade4a/validate_intervals/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas --scaffoldlist Parastrongyloides_trichosuri_GCA_000941615 Rhabditophanes_sp_GCA_000944355 Strongyloides_papillosus_GCA_005656395 Strongyloides_ratti_GCF_001040885 Strongyloides_venezuelensis_GCA_001028725 Strongyloides_stercoralis_GCA_000947215 --genome-base-path results/phyluce/nematoda-clade4a/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/nematoda-clade4a/validate_intervals/nematoda-clade4a-to-Strongyloides_stercoralis_GCA_000947215.sqlite --output results/phyluce/nematoda-clade4a/validate_intervals/lastz/;\n",
      "\n",
      "Running against Parastrongyloides_trichosuri_GCA_000941615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpST3thk.fasta\n",
      "\t/tmp/tmpkH08pC.fasta\n",
      "\t/tmp/tmpOODDfg.fasta\n",
      "\t/tmp/tmpgvn99j.fasta\n",
      "\t/tmp/tmpnfAgFi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_y46_d.lastz\n",
      "\t/tmp/tmp93YStQ.lastz\n",
      "\t/tmp/tmpGEe3kw.lastz\n",
      "\t/tmp/tmpEbdQXK.lastz\n",
      "\t/tmp/tmps8hSwn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4a/validate_intervals/lastz/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_Parastrongyloides_trichosuri_GCA_000941615.lastz\n",
      "Creating Parastrongyloides_trichosuri_GCA_000941615 table\n",
      "Inserting data to Parastrongyloides_trichosuri_GCA_000941615 table\n",
      "\n",
      "Running against Rhabditophanes_sp_GCA_000944355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpLbMvLg.fasta\n",
      "\t/tmp/tmpd0AKIp.fasta\n",
      "\t/tmp/tmpK1bV5f.fasta\n",
      "\t/tmp/tmpSeiJk9.fasta\n",
      "\t/tmp/tmpMgJmqm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzf_qg1.lastz\n",
      "\t/tmp/tmpDDyQ6j.lastz\n",
      "\t/tmp/tmpKOdmTY.lastz\n",
      "\t/tmp/tmp3o2WDB.lastz\n",
      "\t/tmp/tmpCYyRML.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4a/validate_intervals/lastz/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_Rhabditophanes_sp_GCA_000944355.lastz\n",
      "Creating Rhabditophanes_sp_GCA_000944355 table\n",
      "Inserting data to Rhabditophanes_sp_GCA_000944355 table\n",
      "\n",
      "Running against Strongyloides_papillosus_GCA_005656395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpIQuSop.fasta\n",
      "\t/tmp/tmp9AT6K0.fasta\n",
      "\t/tmp/tmplYQ9NY.fasta\n",
      "\t/tmp/tmpB6fIdA.fasta\n",
      "\t/tmp/tmpAXsZdz.fasta\n",
      "\t/tmp/tmpM7Y1zx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdfm3_s.lastz\n",
      "\t/tmp/tmp_GtegL.lastz\n",
      "\t/tmp/tmpLlKEJt.lastz\n",
      "\t/tmp/tmp2XPvy_.lastz\n",
      "\t/tmp/tmpE1hFfo.lastz\n",
      "\t/tmp/tmpumerZi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4a/validate_intervals/lastz/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_Strongyloides_papillosus_GCA_005656395.lastz\n",
      "Creating Strongyloides_papillosus_GCA_005656395 table\n",
      "Inserting data to Strongyloides_papillosus_GCA_005656395 table\n",
      "\n",
      "Running against Strongyloides_ratti_GCF_001040885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpSdTbGL.fasta\n",
      "\t/tmp/tmpREARsD.fasta\n",
      "\t/tmp/tmpSOWJu9.fasta\n",
      "\t/tmp/tmpkiu3JL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprMZ4hr.lastz\n",
      "\t/tmp/tmpQe0XCF.lastz\n",
      "\t/tmp/tmpFno9br.lastz\n",
      "\t/tmp/tmpYrifzm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4a/validate_intervals/lastz/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_Strongyloides_ratti_GCF_001040885.lastz\n",
      "Creating Strongyloides_ratti_GCF_001040885 table\n",
      "Inserting data to Strongyloides_ratti_GCF_001040885 table\n",
      "\n",
      "Running against Strongyloides_venezuelensis_GCA_001028725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpg7Svsy.fasta\n",
      "\t/tmp/tmpTgbxvG.fasta\n",
      "\t/tmp/tmp_H2Xvp.fasta\n",
      "\t/tmp/tmpyezsUW.fasta\n",
      "\t/tmp/tmppH7s6e.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_GPjUs.lastz\n",
      "\t/tmp/tmpDRSYXY.lastz\n",
      "\t/tmp/tmpV2xASQ.lastz\n",
      "\t/tmp/tmpm32PXX.lastz\n",
      "\t/tmp/tmpjZxIQa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4a/validate_intervals/lastz/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_Strongyloides_venezuelensis_GCA_001028725.lastz\n",
      "Creating Strongyloides_venezuelensis_GCA_001028725 table\n",
      "Inserting data to Strongyloides_venezuelensis_GCA_001028725 table\n",
      "\n",
      "Running against Strongyloides_stercoralis_GCA_000947215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpB2EK_b.fasta\n",
      "\t/tmp/tmp3sm9xq.fasta\n",
      "\t/tmp/tmpuOpkcE.fasta\n",
      "\t/tmp/tmp4grhi2.fasta\n",
      "\t/tmp/tmpNfJaOm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprkNOIv.lastz\n",
      "\t/tmp/tmpTtBpB_.lastz\n",
      "\t/tmp/tmpsGyGGo.lastz\n",
      "\t/tmp/tmpZFnZm3.lastz\n",
      "\t/tmp/tmpr7iBmI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4a/validate_intervals/lastz/Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_Strongyloides_stercoralis_GCA_000947215.lastz\n",
      "Creating Strongyloides_stercoralis_GCA_000947215 table\n",
      "Inserting data to Strongyloides_stercoralis_GCA_000947215 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/nematoda-clade4a/extract_probes_from_group/nematoda-clade4a_genome.conf --lastz results/phyluce/nematoda-clade4a/validate_intervals/lastz --probes 120 --probe-prefix uce_nematoda-clade4a_ --name-pattern \"Strongyloides_stercoralis_GCA_000947215_+4_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/nematoda-clade4a/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 14:28:09,923 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 14:28:09,930 - Phyluce - INFO - -- Working on Parastrongyloides_trichosuri_GCA_000941615 genome -\n",
      "2020-02-11 14:28:09,934 - Phyluce - INFO - Reading Parastrongyloides_trichosuri_GCA_000941615 genome\n",
      "2020-02-11 14:28:18,432 - Phyluce - INFO - Parastrongyloides_trichosuri_GCA_000941615: 1416 uces, 389 dupes, 1027 non-dupes, 9 orient drop, 30 length drop, 988 written\n",
      "2020-02-11 14:28:18,432 - Phyluce - INFO - ------- Working on Rhabditophanes_sp_GCA_000944355 genome -------\n",
      "2020-02-11 14:28:18,438 - Phyluce - INFO - Reading Rhabditophanes_sp_GCA_000944355 genome\n",
      "2020-02-11 14:28:24,365 - Phyluce - INFO - Rhabditophanes_sp_GCA_000944355: 1208 uces, 335 dupes, 873 non-dupes, 7 orient drop, 21 length drop, 845 written\n",
      "2020-02-11 14:28:24,366 - Phyluce - INFO - ---- Working on Strongyloides_papillosus_GCA_005656395 genome ---\n",
      "2020-02-11 14:28:24,378 - Phyluce - INFO - Reading Strongyloides_papillosus_GCA_005656395 genome\n",
      "2020-02-11 14:28:33,029 - Phyluce - INFO - Strongyloides_papillosus_GCA_005656395: 1421 uces, 392 dupes, 1029 non-dupes, 8 orient drop, 34 length drop, 987 written\n",
      "2020-02-11 14:28:33,029 - Phyluce - INFO - ------ Working on Strongyloides_ratti_GCF_001040885 genome ------\n",
      "2020-02-11 14:28:33,036 - Phyluce - INFO - Reading Strongyloides_ratti_GCF_001040885 genome\n",
      "2020-02-11 14:28:42,147 - Phyluce - INFO - Strongyloides_ratti_GCF_001040885: 1423 uces, 348 dupes, 1075 non-dupes, 42 orient drop, 52 length drop, 981 written\n",
      "2020-02-11 14:28:42,147 - Phyluce - INFO - -- Working on Strongyloides_venezuelensis_GCA_001028725 genome --\n",
      "2020-02-11 14:28:42,149 - Phyluce - INFO - Reading Strongyloides_venezuelensis_GCA_001028725 genome\n",
      "2020-02-11 14:28:50,333 - Phyluce - INFO - Strongyloides_venezuelensis_GCA_001028725: 1417 uces, 378 dupes, 1039 non-dupes, 10 orient drop, 27 length drop, 1002 written\n",
      "2020-02-11 14:28:50,333 - Phyluce - INFO - --- Working on Strongyloides_stercoralis_GCA_000947215 genome ---\n",
      "2020-02-11 14:28:50,337 - Phyluce - INFO - Reading Strongyloides_stercoralis_GCA_000947215 genome\n",
      "2020-02-11 14:28:59,536 - Phyluce - INFO - Strongyloides_stercoralis_GCA_000947215: 1425 uces, 389 dupes, 1036 non-dupes, 4 orient drop, 30 length drop, 1002 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/nematoda-clade4a/extract_probes_from_group/probe_fasta --output results/phyluce/nematoda-clade4a/extract_probes_from_group/multifastas.sqlite --base-taxon Strongyloides_stercoralis_GCA_000947215;\n",
      "parastrongyloides_trichosuri_gca_000941615.\n",
      "rhabditophanes_sp_gca_000944355.\n",
      "strongyloides_papillosus_gca_005656395.\n",
      "strongyloides_ratti_gcf_001040885.\n",
      "strongyloides_venezuelensis_gca_001028725..\n",
      "strongyloides_stercoralis_gca_000947215..\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade4a/extract_probes_from_group/multifastas.sqlite --base-taxon Strongyloides_stercoralis_GCA_000947215;\n",
      "Loci shared by 0 taxa:\t1,175.0\n",
      "Loci shared by 1 taxa:\t1,175.0\n",
      "Loci shared by 2 taxa:\t1,078.0\n",
      "Loci shared by 3 taxa:\t1,018.0\n",
      "Loci shared by 4 taxa:\t980.0\n",
      "Loci shared by 5 taxa:\t928.0\n",
      "Loci shared by 6 taxa:\t626.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade4a/extract_probes_from_group/multifastas.sqlite  --base-taxon Strongyloides_stercoralis_GCA_000947215 --output results/phyluce/nematoda-clade4a/extract_probes_from_group/Strongyloides_stercoralis_GCA_000947215+4-back-to-6.conf --specific-counts 6;\n",
      "Counter({'strongyloides_venezuelensis_gca_001028725': 626, 'parastrongyloides_trichosuri_gca_000941615': 626, 'strongyloides_ratti_gcf_001040885': 626, 'strongyloides_stercoralis_gca_000947215': 626, 'strongyloides_papillosus_gca_005656395': 626, 'rhabditophanes_sp_gca_000944355': 626})\n",
      "Total loci = 626\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(6)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/nematoda-clade4a/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/nematoda-clade4a/extract_probes_from_group/Strongyloides_stercoralis_GCA_000947215+4-back-to-6.conf --probe-prefix uce_nematoda-clade4a_ --designer rnplattii --design nematoda-clade4a_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 626\n",
      "Probe Count = 6784\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list.fasta --query results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  14:29:21\n",
      "Ended:  Tue Feb 11, 2020  14:29:26\n",
      "Time for execution:  0.0882786830266 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list.fasta --lastz results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_nematoda-clade4a_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6783 fasta sequences.  Filtered 0 duplicates. Kept 6784.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/nematoda-clade4a/final_probe_design/nematoda-clade4a_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 14:31:09 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 6784\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 542720\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3953465\n",
      "Max number of word counting entries: 88992499\n",
      "\n",
      "# comparing sequences from          0  to       1130\n",
      ".---------- new table with      931 representatives\n",
      "# comparing sequences from       1130  to       2072\n",
      "----------    188 remaining sequences to the next cycle\n",
      "---------- new table with      626 representatives\n",
      "# comparing sequences from       1884  to       2700\n",
      "----------    140 remaining sequences to the next cycle\n",
      "---------- new table with      561 representatives\n",
      "# comparing sequences from       2560  to       3264\n",
      "----------    142 remaining sequences to the next cycle\n",
      "---------- new table with      467 representatives\n",
      "# comparing sequences from       3122  to       3732\n",
      "----------     15 remaining sequences to the next cycle\n",
      "---------- new table with      494 representatives\n",
      "# comparing sequences from       3717  to       4228\n",
      "----------    131 remaining sequences to the next cycle\n",
      "---------- new table with      306 representatives\n",
      "# comparing sequences from       4097  to       4544\n",
      "----------    108 remaining sequences to the next cycle\n",
      "---------- new table with      285 representatives\n",
      "# comparing sequences from       4436  to       4827\n",
      "----------     41 remaining sequences to the next cycle\n",
      "---------- new table with      288 representatives\n",
      "# comparing sequences from       4786  to       5119\n",
      "99.9%---------- new table with      279 representatives\n",
      "# comparing sequences from       5119  to       5396\n",
      "----------      9 remaining sequences to the next cycle\n",
      "---------- new table with      223 representatives\n",
      "# comparing sequences from       5387  to       5619\n",
      "99.9%---------- new table with      206 representatives\n",
      "# comparing sequences from       5619  to       5813\n",
      "----------     49 remaining sequences to the next cycle\n",
      "---------- new table with      119 representatives\n",
      "# comparing sequences from       5764  to       5934\n",
      "100.0%---------- new table with      136 representatives\n",
      "# comparing sequences from       5934  to       6784\n",
      ".....................---------- new table with      724 representatives\n",
      "\n",
      "     6784  finished       5645  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.32\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clade 4b"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'nematoda-clade4b'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Acrobeloides_nanus_GCA_900406225\",\n",
    "               \"Bursaphelenchus_xylophilus_GCA_004128815\",\n",
    "               \"Deladenus_siricidicola_GCA_009724625\",\n",
    "               \"Ditylenchus_destructor_GCA_001579705\",\n",
    "               \"Ditylenchus_dipsaci_GCA_004194705\",\n",
    "               \"Globodera_ellingtonae_GCA_001723225\",\n",
    "               \"Globodera_pallida_GCA_000724045\",\n",
    "               \"Globodera_rostochiensis_GCA_900079975\",\n",
    "               \"Halicephalobus_mephisto_GCA_009193035\",\n",
    "               \"Halicephalobus_sp_GCA_009761265\",\n",
    "               \"Heterodera_glycines_GCA_004148225\",\n",
    "               \"Heterorhabditis_bacteriophora_GCA_000223415\",\n",
    "               \"Meloidogyne_arenaria_GCA_003133805\",\n",
    "               \"Meloidogyne_enterolobii_GCA_003693675\",\n",
    "               \"Meloidogyne_floridensis_GCA_003693605\",\n",
    "               \"Meloidogyne_graminicola_GCA_002778205\",\n",
    "               \"Meloidogyne_hapla_GCA_000172435\",\n",
    "               \"Meloidogyne_incognita_GCA_900182535\",\n",
    "               \"Meloidogyne_javanica_GCA_003693625\",\n",
    "               \"Meloidogyne_luci_GCA_902706615\",\n",
    "               \"Panagrellus_redivivus_GCA_000341325\",\n",
    "               \"Panagrolaimus_davidi_GCA_901779475\",\n",
    "               \"Panagrolaimus_sp_GCA_901766855\",\n",
    "               \"Panagrolaimus_superbus_GCA_901766145\",\n",
    "               \"Radopholus_similis_GCA_004764675\",\n",
    "               \"Rotylenchulus_reniformis_GCA_001026735\",\n",
    "               \"Steinernema_feltiae_GCA_007213375\",\n",
    "               \"Steinernema_glaseri_GCA_000757755\",\n",
    "               \"Steinernema_monticolum_GCA_000505645\",\n",
    "               \"Steinernema_scapterisci_GCA_000757745\",\n",
    "               \"Subanguina_moxae_GCA_000981365\", ]\n",
    "                    \n",
    "reference_taxon = \"Steinernema_carpocapsae_GCA_000757645\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900406225.1_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 79686543 bytes  8388061.58 bytes/sec\n",
      "total size is 79666989  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004128815.1_ASM412881v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 23556743 bytes  6730510.00 bytes/sec\n",
      "total size is 23550884  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009724625.1_UP_Dsir_3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 30203492 bytes  6711896.44 bytes/sec\n",
      "total size is 30196011  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001579705.1_ASM157970v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 35053786 bytes  7789739.56 bytes/sec\n",
      "total size is 35045119  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004194705.1_D.dipsaci.v1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 73004671 bytes  8588789.76 bytes/sec\n",
      "total size is 72986737  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001723225.1_ASM172322v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 33082049 bytes  6014925.64 bytes/sec\n",
      "total size is 33073862  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000724045.1_GPAL001_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 33316490 bytes  7403673.78 bytes/sec\n",
      "total size is 33308251  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900079975.1_nGr_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 28633713 bytes  6363056.67 bytes/sec\n",
      "total size is 28626622  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009193035.1_mephisto_omega_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 18955942 bytes  5415995.43 bytes/sec\n",
      "total size is 18951200  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009761265.1_ASM976126v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 15185582 bytes  6074249.60 bytes/sec\n",
      "total size is 15181763  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004148225.1_Hetgly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 37309953 bytes  6783635.45 bytes/sec\n",
      "total size is 37300737  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000223415.1_Heterorhabditis_bacteriophora-7.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24086736 bytes  6881936.57 bytes/sec\n",
      "total size is 24080727  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003133805.1_ASM313380v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 84700265 bytes  8066695.90 bytes/sec\n",
      "total size is 84679478  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003693675.1_ASM369367v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 51996639 bytes  6932890.80 bytes/sec\n",
      "total size is 51983836  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003693605.1_ASM369360v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 23762442 bytes  6789281.14 bytes/sec\n",
      "total size is 23756535  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002778205.1_Mgraminicola_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 11626235 bytes  4650510.80 bytes/sec\n",
      "total size is 11623284  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000172435.1_Freeze_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16483474 bytes  4709576.00 bytes/sec\n",
      "total size is 16479346  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900182535.1_Meloidogyne_incognita_V3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 56982013 bytes  5998111.05 bytes/sec\n",
      "total size is 56967981  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003693625.1_ASM369362v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 48030076 bytes  7389248.92 bytes/sec\n",
      "total size is 48018241  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902706615.1_V13_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 62683592 bytes  7374545.18 bytes/sec\n",
      "total size is 62668189  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000341325.1_Pred3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 19919841 bytes  5691395.14 bytes/sec\n",
      "total size is 19914876  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_901779475.1_PDA_v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 37997800 bytes  8443964.89 bytes/sec\n",
      "total size is 37988418  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_901766855.1_ES5_v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 28521588 bytes  6338140.00 bytes/sec\n",
      "total size is 28514518  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_901766145.1_PSU_v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24886975 bytes  7110576.29 bytes/sec\n",
      "total size is 24880793  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004764675.1_ASM476467v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20848089 bytes  5956608.86 bytes/sec\n",
      "total size is 20842886  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001026735.1_RREN1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 100472246 bytes  7442391.70 bytes/sec\n",
      "total size is 100447615  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000757645.3_ASM75764v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 26441046 bytes  7554596.57 bytes/sec\n",
      "total size is 26434484  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007213375.1_S_feltiae_NW_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 37891350 bytes  8420309.33 bytes/sec\n",
      "total size is 37881982  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000757755.1_S_glas_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 28395917 bytes  6310213.11 bytes/sec\n",
      "total size is 28388876  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000505645.1_S_monti_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 27576020 bytes  4242471.08 bytes/sec\n",
      "total size is 27569178  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000757745.1_S_scapt_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 25076794 bytes  7164810.29 bytes/sec\n",
      "total size is 25070560  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900624965.1_S_vulgaris_Kentucky_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 89298697 bytes  3369763.74 bytes/sec\n",
      "total size is 89276773  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000981365.1_SAMX_assembly_v0.8_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 29290336 bytes  6508972.89 bytes/sec\n",
      "total size is 29283070  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487756 (\"sim_Acrobeloides_nanus_GCA_900406225\") has been submitted\n",
      "Your job 5487757 (\"sim_Bursaphelenchus_xylophilus_GCA_004128815\") has been submitted\n",
      "Your job 5487758 (\"sim_Deladenus_siricidicola_GCA_009724625\") has been submitted\n",
      "Your job 5487759 (\"sim_Ditylenchus_destructor_GCA_001579705\") has been submitted\n",
      "Your job 5487760 (\"sim_Ditylenchus_dipsaci_GCA_004194705\") has been submitted\n",
      "Your job 5487761 (\"sim_Globodera_ellingtonae_GCA_001723225\") has been submitted\n",
      "Your job 5487762 (\"sim_Globodera_pallida_GCA_000724045\") has been submitted\n",
      "Your job 5487763 (\"sim_Globodera_rostochiensis_GCA_900079975\") has been submitted\n",
      "Your job 5487764 (\"sim_Halicephalobus_mephisto_GCA_009193035\") has been submitted\n",
      "Your job 5487765 (\"sim_Halicephalobus_sp_GCA_009761265\") has been submitted\n",
      "Your job 5487766 (\"sim_Heterodera_glycines_GCA_004148225\") has been submitted\n",
      "Your job 5487767 (\"sim_Heterorhabditis_bacteriophora_GCA_000223415\") has been submitted\n",
      "Your job 5487768 (\"sim_Meloidogyne_arenaria_GCA_003133805\") has been submitted\n",
      "Your job 5487769 (\"sim_Meloidogyne_enterolobii_GCA_003693675\") has been submitted\n",
      "Your job 5487770 (\"sim_Meloidogyne_floridensis_GCA_003693605\") has been submitted\n",
      "Your job 5487771 (\"sim_Meloidogyne_graminicola_GCA_002778205\") has been submitted\n",
      "Your job 5487772 (\"sim_Meloidogyne_hapla_GCA_000172435\") has been submitted\n",
      "Your job 5487773 (\"sim_Meloidogyne_incognita_GCA_900182535\") has been submitted\n",
      "Your job 5487774 (\"sim_Meloidogyne_javanica_GCA_003693625\") has been submitted\n",
      "Your job 5487775 (\"sim_Meloidogyne_luci_GCA_902706615\") has been submitted\n",
      "Your job 5487776 (\"sim_Panagrellus_redivivus_GCA_000341325\") has been submitted\n",
      "Your job 5487777 (\"sim_Panagrolaimus_davidi_GCA_901779475\") has been submitted\n",
      "Your job 5487778 (\"sim_Panagrolaimus_sp_GCA_901766855\") has been submitted\n",
      "Your job 5487779 (\"sim_Panagrolaimus_superbus_GCA_901766145\") has been submitted\n",
      "Your job 5487780 (\"sim_Radopholus_similis_GCA_004764675\") has been submitted\n",
      "Your job 5487781 (\"sim_Rotylenchulus_reniformis_GCA_001026735\") has been submitted\n",
      "Your job 5487782 (\"sim_Steinernema_carpocapsae_GCA_000757645\") has been submitted\n",
      "Your job 5487783 (\"sim_Steinernema_feltiae_GCA_007213375\") has been submitted\n",
      "Your job 5487784 (\"sim_Steinernema_glaseri_GCA_000757755\") has been submitted\n",
      "Your job 5487785 (\"sim_Steinernema_monticolum_GCA_000505645\") has been submitted\n",
      "Your job 5487786 (\"sim_Steinernema_scapterisci_GCA_000757745\") has been submitted\n",
      "Your job 5487787 (\"sim_Subanguina_moxae_GCA_000981365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/nematoda-clade4b/cleaned_genomes/Steinernema_carpocapsae_GCA_000757645_formatted.fas path=results/phyluce/nematoda-clade4b/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5488099 (\"map_Acrobeloides_nanus_GCA_900406225\") has been submitted\n",
      "Your job 5488100 (\"map_Bursaphelenchus_xylophilus_GCA_004128815\") has been submitted\n",
      "Your job 5488101 (\"map_Deladenus_siricidicola_GCA_009724625\") has been submitted\n",
      "Your job 5488102 (\"map_Ditylenchus_destructor_GCA_001579705\") has been submitted\n",
      "Your job 5488103 (\"map_Ditylenchus_dipsaci_GCA_004194705\") has been submitted\n",
      "Your job 5488104 (\"map_Globodera_ellingtonae_GCA_001723225\") has been submitted\n",
      "Your job 5488105 (\"map_Globodera_pallida_GCA_000724045\") has been submitted\n",
      "Your job 5488106 (\"map_Globodera_rostochiensis_GCA_900079975\") has been submitted\n",
      "Your job 5488107 (\"map_Halicephalobus_mephisto_GCA_009193035\") has been submitted\n",
      "Your job 5488108 (\"map_Halicephalobus_sp_GCA_009761265\") has been submitted\n",
      "Your job 5488109 (\"map_Heterodera_glycines_GCA_004148225\") has been submitted\n",
      "Your job 5488110 (\"map_Heterorhabditis_bacteriophora_GCA_000223415\") has been submitted\n",
      "Your job 5488111 (\"map_Meloidogyne_arenaria_GCA_003133805\") has been submitted\n",
      "Your job 5488112 (\"map_Meloidogyne_enterolobii_GCA_003693675\") has been submitted\n",
      "Your job 5488113 (\"map_Meloidogyne_floridensis_GCA_003693605\") has been submitted\n",
      "Your job 5488114 (\"map_Meloidogyne_graminicola_GCA_002778205\") has been submitted\n",
      "Your job 5488115 (\"map_Meloidogyne_hapla_GCA_000172435\") has been submitted\n",
      "Your job 5488116 (\"map_Meloidogyne_incognita_GCA_900182535\") has been submitted\n",
      "Your job 5488117 (\"map_Meloidogyne_javanica_GCA_003693625\") has been submitted\n",
      "Your job 5488118 (\"map_Meloidogyne_luci_GCA_902706615\") has been submitted\n",
      "Your job 5488119 (\"map_Panagrellus_redivivus_GCA_000341325\") has been submitted\n",
      "Your job 5488120 (\"map_Panagrolaimus_davidi_GCA_901779475\") has been submitted\n",
      "Your job 5488121 (\"map_Panagrolaimus_sp_GCA_901766855\") has been submitted\n",
      "Your job 5488122 (\"map_Panagrolaimus_superbus_GCA_901766145\") has been submitted\n",
      "Your job 5488123 (\"map_Radopholus_similis_GCA_004764675\") has been submitted\n",
      "Your job 5488124 (\"map_Rotylenchulus_reniformis_GCA_001026735\") has been submitted\n",
      "Your job 5488125 (\"map_Steinernema_carpocapsae_GCA_000757645\") has been submitted\n",
      "Your job 5488126 (\"map_Steinernema_feltiae_GCA_007213375\") has been submitted\n",
      "Your job 5488127 (\"map_Steinernema_glaseri_GCA_000757755\") has been submitted\n",
      "Your job 5488128 (\"map_Steinernema_monticolum_GCA_000505645\") has been submitted\n",
      "Your job 5488129 (\"map_Steinernema_scapterisci_GCA_000757745\") has been submitted\n",
      "Your job 5488130 (\"map_Subanguina_moxae_GCA_000981365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5488131 (\"merge_Acrobeloides_nanus_GCA_900406225\") has been submitted\n",
      "Your job 5488132 (\"merge_Bursaphelenchus_xylophilus_GCA_004128815\") has been submitted\n",
      "Your job 5488133 (\"merge_Deladenus_siricidicola_GCA_009724625\") has been submitted\n",
      "Your job 5488134 (\"merge_Ditylenchus_destructor_GCA_001579705\") has been submitted\n",
      "Your job 5488135 (\"merge_Ditylenchus_dipsaci_GCA_004194705\") has been submitted\n",
      "Your job 5488136 (\"merge_Globodera_ellingtonae_GCA_001723225\") has been submitted\n",
      "Your job 5488137 (\"merge_Globodera_pallida_GCA_000724045\") has been submitted\n",
      "Your job 5488138 (\"merge_Globodera_rostochiensis_GCA_900079975\") has been submitted\n",
      "Your job 5488139 (\"merge_Halicephalobus_mephisto_GCA_009193035\") has been submitted\n",
      "Your job 5488140 (\"merge_Halicephalobus_sp_GCA_009761265\") has been submitted\n",
      "Your job 5488141 (\"merge_Heterodera_glycines_GCA_004148225\") has been submitted\n",
      "Your job 5488142 (\"merge_Heterorhabditis_bacteriophora_GCA_000223415\") has been submitted\n",
      "Your job 5488143 (\"merge_Meloidogyne_arenaria_GCA_003133805\") has been submitted\n",
      "Your job 5488144 (\"merge_Meloidogyne_enterolobii_GCA_003693675\") has been submitted\n",
      "Your job 5488145 (\"merge_Meloidogyne_floridensis_GCA_003693605\") has been submitted\n",
      "Your job 5488146 (\"merge_Meloidogyne_graminicola_GCA_002778205\") has been submitted\n",
      "Your job 5488147 (\"merge_Meloidogyne_hapla_GCA_000172435\") has been submitted\n",
      "Your job 5488148 (\"merge_Meloidogyne_incognita_GCA_900182535\") has been submitted\n",
      "Your job 5488149 (\"merge_Meloidogyne_javanica_GCA_003693625\") has been submitted\n",
      "Your job 5488150 (\"merge_Meloidogyne_luci_GCA_902706615\") has been submitted\n",
      "Your job 5488151 (\"merge_Panagrellus_redivivus_GCA_000341325\") has been submitted\n",
      "Your job 5488152 (\"merge_Panagrolaimus_davidi_GCA_901779475\") has been submitted\n",
      "Your job 5488153 (\"merge_Panagrolaimus_sp_GCA_901766855\") has been submitted\n",
      "Your job 5488154 (\"merge_Panagrolaimus_superbus_GCA_901766145\") has been submitted\n",
      "Your job 5488155 (\"merge_Radopholus_similis_GCA_004764675\") has been submitted\n",
      "Your job 5488156 (\"merge_Rotylenchulus_reniformis_GCA_001026735\") has been submitted\n",
      "Your job 5488157 (\"merge_Steinernema_carpocapsae_GCA_000757645\") has been submitted\n",
      "Your job 5488158 (\"merge_Steinernema_feltiae_GCA_007213375\") has been submitted\n",
      "Your job 5488159 (\"merge_Steinernema_glaseri_GCA_000757755\") has been submitted\n",
      "Your job 5488160 (\"merge_Steinernema_monticolum_GCA_000505645\") has been submitted\n",
      "Your job 5488161 (\"merge_Steinernema_scapterisci_GCA_000757745\") has been submitted\n",
      "Your job 5488162 (\"merge_Subanguina_moxae_GCA_000981365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 161 sequences from Acrobeloides_nanus_GCA_900406225_merged.bed.  Filtered 150 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 241 sequences from Bursaphelenchus_xylophilus_GCA_004128815_merged.bed.  Filtered 228 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 220 sequences from Deladenus_siricidicola_GCA_009724625_merged.bed.  Filtered 207 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 48 sequences from Ditylenchus_destructor_GCA_001579705_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 92 sequences from Ditylenchus_dipsaci_GCA_004194705_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 72 sequences from Globodera_ellingtonae_GCA_001723225_merged.bed.  Filtered 66 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 77 sequences from Globodera_pallida_GCA_000724045_merged.bed.  Filtered 70 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 71 sequences from Globodera_rostochiensis_GCA_900079975_merged.bed.  Filtered 67 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 93 sequences from Halicephalobus_mephisto_GCA_009193035_merged.bed.  Filtered 82 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 93 sequences from Halicephalobus_sp_GCA_009761265_merged.bed.  Filtered 79 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 14.\n",
      "Screened 78 sequences from Heterodera_glycines_GCA_004148225_merged.bed.  Filtered 68 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 78 sequences from Heterorhabditis_bacteriophora_GCA_000223415_merged.bed.  Filtered 70 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 57 sequences from Meloidogyne_arenaria_GCA_003133805_merged.bed.  Filtered 49 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 44 sequences from Meloidogyne_enterolobii_GCA_003693675_merged.bed.  Filtered 40 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 44 sequences from Meloidogyne_floridensis_GCA_003693605_merged.bed.  Filtered 38 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 29 sequences from Meloidogyne_graminicola_GCA_002778205_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 39 sequences from Meloidogyne_hapla_GCA_000172435_merged.bed.  Filtered 30 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 51 sequences from Meloidogyne_incognita_GCA_900182535_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 42 sequences from Meloidogyne_javanica_GCA_003693625_merged.bed.  Filtered 37 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 64 sequences from Meloidogyne_luci_GCA_902706615_merged.bed.  Filtered 55 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 277 sequences from Panagrellus_redivivus_GCA_000341325_merged.bed.  Filtered 234 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 28 sequences from Panagrolaimus_davidi_GCA_901779475_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 0.\n",
      "Screened 34 sequences from Panagrolaimus_sp_GCA_901766855_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2.\n",
      "Screened 34 sequences from Panagrolaimus_superbus_GCA_901766145_merged.bed.  Filtered 31 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 92 sequences from Radopholus_similis_GCA_004764675_merged.bed.  Filtered 80 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 139 sequences from Rotylenchulus_reniformis_GCA_001026735_merged.bed.  Filtered 123 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 16.\n",
      "Screened 4972 sequences from Steinernema_carpocapsae_GCA_000757645_merged.bed.  Filtered 1818 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3154.\n",
      "Screened 4296 sequences from Steinernema_feltiae_GCA_007213375_merged.bed.  Filtered 3492 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 804.\n",
      "Screened 3633 sequences from Steinernema_glaseri_GCA_000757755_merged.bed.  Filtered 2925 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 708.\n",
      "Screened 3283 sequences from Steinernema_monticolum_GCA_000505645_merged.bed.  Filtered 2745 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 538.\n",
      "Screened 41742 sequences from Steinernema_scapterisci_GCA_000757745_merged.bed.  Filtered 26342 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15400.\n",
      "Screened 65 sequences from Subanguina_moxae_GCA_000981365_merged.bed.  Filtered 61 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/nematoda-clade4b/initial_intervals/Subanguina_moxae_GCA_000981365_merged.bed --twobit results/phyluce/nematoda-clade4b/cleaned_genomes/Steinernema_carpocapsae_GCA_000757645_formatted.2bit --output results/phyluce/nematoda-clade4b/initial_intervals/Subanguina_moxae_GCA_000981365_stripped.bed;\n",
      "acrobeloides_nanus_gca_900406225.\n",
      "bursaphelenchus_xylophilus_gca_004128815.\n",
      "deladenus_siricidicola_gca_009724625.\n",
      "ditylenchus_destructor_gca_001579705.\n",
      "ditylenchus_dipsaci_gca_004194705.\n",
      "globodera_ellingtonae_gca_001723225.\n",
      "globodera_pallida_gca_000724045.\n",
      "globodera_rostochiensis_gca_900079975.\n",
      "halicephalobus_mephisto_gca_009193035.\n",
      "halicephalobus_sp_gca_009761265.\n",
      "heterodera_glycines_gca_004148225.\n",
      "heterorhabditis_bacteriophora_gca_000223415.\n",
      "meloidogyne_arenaria_gca_003133805.\n",
      "meloidogyne_enterolobii_gca_003693675.\n",
      "meloidogyne_floridensis_gca_003693605.\n",
      "meloidogyne_graminicola_gca_002778205.\n",
      "meloidogyne_hapla_gca_000172435.\n",
      "meloidogyne_incognita_gca_900182535.\n",
      "meloidogyne_javanica_gca_003693625.\n",
      "meloidogyne_luci_gca_902706615.\n",
      "panagrellus_redivivus_gca_000341325.\n",
      "panagrolaimus_davidi_gca_901779475\n",
      "panagrolaimus_sp_gca_901766855.\n",
      "panagrolaimus_superbus_gca_901766145.\n",
      "radopholus_similis_gca_004764675.\n",
      "rotylenchulus_reniformis_gca_001026735.\n",
      "steinernema_carpocapsae_gca_000757645....\n",
      "steinernema_feltiae_gca_007213375.\n",
      "steinernema_glaseri_gca_000757755.\n",
      "steinernema_monticolum_gca_000505645.\n",
      "steinernema_scapterisci_gca_000757745................\n",
      "subanguina_moxae_gca_000981365.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade4b/initial_intervals/nematoda-clade4b-to-Steinernema_carpocapsae_GCA_000757645.sqlite --base-taxon Steinernema_carpocapsae_GCA_000757645\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 0 taxa:\t6,132.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 1 taxa:\t6,132.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 2 taxa:\t1,243.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 3 taxa:\t547.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 4 taxa:\t318.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 5 taxa:\t145.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 6 taxa:\t35.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 7 taxa:\t18.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 8 taxa:\t14.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 9 taxa:\t10.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 10 taxa:\t10.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 11 taxa:\t10.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 12 taxa:\t9.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 13 taxa:\t8.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 14 taxa:\t6.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 15 taxa:\t6.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 16 taxa:\t6.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 17 taxa:\t5.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 18 taxa:\t3.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 19 taxa:\t3.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 20 taxa:\t3.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 21 taxa:\t3.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 22 taxa:\t3.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 23 taxa:\t3.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 24 taxa:\t1.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 25 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 26 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 27 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 28 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 29 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 30 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 31 taxa:\t0.0\n",
      "Loci shared by Steinernema_carpocapsae_GCA_000757645 + 32 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade4b/initial_intervals/nematoda-clade4b-to-Steinernema_carpocapsae_GCA_000757645.sqlite --base-taxon Steinernema_carpocapsae_GCA_000757645 --output results/phyluce/nematoda-clade4b/initial_intervals/Steinernema_carpocapsae_GCA_000757645_+2.bed --specific-counts 2;\n",
      "Counter({'steinernema_scapterisci_gca_000757745': 1209, 'steinernema_carpocapsae_gca_000757645': 970, 'steinernema_feltiae_gca_007213375': 475, 'steinernema_glaseri_gca_000757755': 434, 'steinernema_monticolum_gca_000505645': 325, 'panagrellus_redivivus_gca_000341325': 43, 'rotylenchulus_reniformis_gca_001026735': 16, 'halicephalobus_sp_gca_009761265': 14, 'deladenus_siricidicola_gca_009724625': 13, 'bursaphelenchus_xylophilus_gca_004128815': 13, 'radopholus_similis_gca_004764675': 13, 'acrobeloides_nanus_gca_900406225': 11, 'heterodera_glycines_gca_004148225': 11, 'halicephalobus_mephisto_gca_009193035': 11, 'meloidogyne_luci_gca_902706615': 9, 'meloidogyne_hapla_gca_000172435': 9, 'meloidogyne_arenaria_gca_003133805': 8, 'meloidogyne_incognita_gca_900182535': 8, 'heterorhabditis_bacteriophora_gca_000223415': 8, 'globodera_pallida_gca_000724045': 7, 'globodera_ellingtonae_gca_001723225': 7, 'meloidogyne_floridensis_gca_003693605': 6, 'meloidogyne_javanica_gca_003693625': 5, 'meloidogyne_enterolobii_gca_003693675': 4, 'globodera_rostochiensis_gca_900079975': 4, 'ditylenchus_dipsaci_gca_004194705': 4, 'subanguina_moxae_gca_000981365': 4, 'meloidogyne_graminicola_gca_002778205': 4, 'panagrolaimus_superbus_gca_901766145': 3, 'panagrolaimus_sp_gca_901766855': 2, 'ditylenchus_destructor_gca_001579705': 2})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 2\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/nematoda-clade4b/initial_intervals/Steinernema_carpocapsae_GCA_000757645_+2.bed --twobit results/phyluce/nematoda-clade4b/cleaned_genomes/Steinernema_carpocapsae_GCA_000757645_formatted.2bit --buffer-to 160 --output results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2.fasta;\n",
      "Screened 1243 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 1243.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2.fasta --probe-prefix uce_nematoda-clade4b_ --design nematoda-clade4b_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 1231\n",
      "Probe Count = 2438\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas --query results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  19:08:06\n",
      "Ended:  Tue Feb 11, 2020  19:08:07\n",
      "Time for execution:  0.0178224802017 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas                        --lastz results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes_vself.lastz                       --probe-prefix=uce_nematoda-clade4b_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 2437 fasta sequences.  Filtered 92 duplicates. Kept 2256.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/nematoda-clade4b/validate_intervals/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas --scaffoldlist Acrobeloides_nanus_GCA_900406225 Bursaphelenchus_xylophilus_GCA_004128815 Deladenus_siricidicola_GCA_009724625 Ditylenchus_destructor_GCA_001579705 Ditylenchus_dipsaci_GCA_004194705 Globodera_ellingtonae_GCA_001723225 Globodera_pallida_GCA_000724045 Globodera_rostochiensis_GCA_900079975 Halicephalobus_mephisto_GCA_009193035 Halicephalobus_sp_GCA_009761265 Heterodera_glycines_GCA_004148225 Heterorhabditis_bacteriophora_GCA_000223415 Meloidogyne_arenaria_GCA_003133805 Meloidogyne_enterolobii_GCA_003693675 Meloidogyne_floridensis_GCA_003693605 Meloidogyne_graminicola_GCA_002778205 Meloidogyne_hapla_GCA_000172435 Meloidogyne_incognita_GCA_900182535 Meloidogyne_javanica_GCA_003693625 Meloidogyne_luci_GCA_902706615 Panagrellus_redivivus_GCA_000341325 Panagrolaimus_davidi_GCA_901779475 Panagrolaimus_sp_GCA_901766855 Panagrolaimus_superbus_GCA_901766145 Radopholus_similis_GCA_004764675 Rotylenchulus_reniformis_GCA_001026735 Steinernema_feltiae_GCA_007213375 Steinernema_glaseri_GCA_000757755 Steinernema_monticolum_GCA_000505645 Steinernema_scapterisci_GCA_000757745 Subanguina_moxae_GCA_000981365 Steinernema_carpocapsae_GCA_000757645 --genome-base-path results/phyluce/nematoda-clade4b/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/nematoda-clade4b/extract_probes_from_group/nematoda-clade4b-to-Steinernema_carpocapsae_GCA_000757645.sqlite --output results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/;\n",
      "\n",
      "Running against Acrobeloides_nanus_GCA_900406225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 25 queries...\n",
      "\t/tmp/tmpGNShqI.fasta\n",
      "\t/tmp/tmpkdCcuT.fasta\n",
      "\t/tmp/tmpKyufn4.fasta\n",
      "\t/tmp/tmpH6T_Ft.fasta\n",
      "\t/tmp/tmp6J9weK.fasta\n",
      "\t/tmp/tmp65IQru.fasta\n",
      "\t/tmp/tmpe3El9e.fasta\n",
      "\t/tmp/tmpiVPFps.fasta\n",
      "\t/tmp/tmpy40ucS.fasta\n",
      "\t/tmp/tmpOnk9qY.fasta\n",
      "\t/tmp/tmpb1EaeM.fasta\n",
      "\t/tmp/tmprh1yD8.fasta\n",
      "\t/tmp/tmp6q9cB2.fasta\n",
      "\t/tmp/tmpMyHYRi.fasta\n",
      "\t/tmp/tmpUcTqJy.fasta\n",
      "\t/tmp/tmpVGaiFG.fasta\n",
      "\t/tmp/tmp_Ua3qH.fasta\n",
      "\t/tmp/tmpBJVGL7.fasta\n",
      "\t/tmp/tmpAxSODe.fasta\n",
      "\t/tmp/tmpZHvVJ2.fasta\n",
      "\t/tmp/tmp_UIL_t.fasta\n",
      "\t/tmp/tmpAnDnAF.fasta\n",
      "\t/tmp/tmpMj09lw.fasta\n",
      "\t/tmp/tmpQcgSRx.fasta\n",
      "\t/tmp/tmpev7IIT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOXEKQZ.lastz\n",
      "\t/tmp/tmprQihK0.lastz\n",
      "\t/tmp/tmpoFxT9V.lastz\n",
      "\t/tmp/tmpOYRP6p.lastz\n",
      "\t/tmp/tmpJ9vKsc.lastz\n",
      "\t/tmp/tmpiflcKV.lastz\n",
      "\t/tmp/tmpPHFh_R.lastz\n",
      "\t/tmp/tmp5I2eja.lastz\n",
      "\t/tmp/tmpf9Nptk.lastz\n",
      "\t/tmp/tmp7u9guU.lastz\n",
      "\t/tmp/tmpXu1PGC.lastz\n",
      "\t/tmp/tmp53rmUI.lastz\n",
      "\t/tmp/tmpEXuULM.lastz\n",
      "\t/tmp/tmpVbtqnG.lastz\n",
      "\t/tmp/tmpuPAaxr.lastz\n",
      "\t/tmp/tmptSMp4L.lastz\n",
      "\t/tmp/tmpz8t9H9.lastz\n",
      "\t/tmp/tmpcfPjef.lastz\n",
      "\t/tmp/tmpUlXwH8.lastz\n",
      "\t/tmp/tmpiUd2LR.lastz\n",
      "\t/tmp/tmp2Y9BwY.lastz\n",
      "\t/tmp/tmpkE2eoD.lastz\n",
      "\t/tmp/tmp511n1D.lastz\n",
      "\t/tmp/tmpa4NoyO.lastz\n",
      "\t/tmp/tmp_Cd3_W.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Acrobeloides_nanus_GCA_900406225.lastz\n",
      "Creating Acrobeloides_nanus_GCA_900406225 table\n",
      "Inserting data to Acrobeloides_nanus_GCA_900406225 table\n",
      "\n",
      "Running against Bursaphelenchus_xylophilus_GCA_004128815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmprqZoG_.fasta\n",
      "\t/tmp/tmp2HF9We.fasta\n",
      "\t/tmp/tmpXvSZm0.fasta\n",
      "\t/tmp/tmpE98Zg2.fasta\n",
      "\t/tmp/tmpOQyfFU.fasta\n",
      "\t/tmp/tmp39kDmj.fasta\n",
      "\t/tmp/tmp5_4h7H.fasta\n",
      "\t/tmp/tmpWK46lh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpk_7nnD.lastz\n",
      "\t/tmp/tmpoar3bu.lastz\n",
      "\t/tmp/tmpD85Vlv.lastz\n",
      "\t/tmp/tmph8OT4j.lastz\n",
      "\t/tmp/tmpu4bSpe.lastz\n",
      "\t/tmp/tmpFmenlI.lastz\n",
      "\t/tmp/tmpW7YozN.lastz\n",
      "\t/tmp/tmpag5wFP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Bursaphelenchus_xylophilus_GCA_004128815.lastz\n",
      "Creating Bursaphelenchus_xylophilus_GCA_004128815 table\n",
      "Inserting data to Bursaphelenchus_xylophilus_GCA_004128815 table\n",
      "\n",
      "Running against Deladenus_siricidicola_GCA_009724625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmpMjIll0.fasta\n",
      "\t/tmp/tmpX9lvoN.fasta\n",
      "\t/tmp/tmpuhDwBl.fasta\n",
      "\t/tmp/tmpmIPo0P.fasta\n",
      "\t/tmp/tmpan9ukO.fasta\n",
      "\t/tmp/tmpzJnXRg.fasta\n",
      "\t/tmp/tmpnWWS7Y.fasta\n",
      "\t/tmp/tmpsglyBP.fasta\n",
      "\t/tmp/tmp00uL6L.fasta\n",
      "\t/tmp/tmpkbz2KE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWJMIs4.lastz\n",
      "\t/tmp/tmpI2Vl9n.lastz\n",
      "\t/tmp/tmpTzl2nW.lastz\n",
      "\t/tmp/tmpJ3X4vN.lastz\n",
      "\t/tmp/tmpA61Q5d.lastz\n",
      "\t/tmp/tmp2JK31A.lastz\n",
      "\t/tmp/tmpmQUvAN.lastz\n",
      "\t/tmp/tmpgwsHHX.lastz\n",
      "\t/tmp/tmpeZ0Ddl.lastz\n",
      "\t/tmp/tmpglTUqi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Deladenus_siricidicola_GCA_009724625.lastz\n",
      "Creating Deladenus_siricidicola_GCA_009724625 table\n",
      "Inserting data to Deladenus_siricidicola_GCA_009724625 table\n",
      "\n",
      "Running against Ditylenchus_destructor_GCA_001579705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmpPfTHfv.fasta\n",
      "\t/tmp/tmpbJ5Dbt.fasta\n",
      "\t/tmp/tmp7278fz.fasta\n",
      "\t/tmp/tmpZF3lkV.fasta\n",
      "\t/tmp/tmpvfDu9j.fasta\n",
      "\t/tmp/tmp4bAg1U.fasta\n",
      "\t/tmp/tmp2IRBt9.fasta\n",
      "\t/tmp/tmpMvgeF_.fasta\n",
      "\t/tmp/tmpmxBT4u.fasta\n",
      "\t/tmp/tmpW03qSr.fasta\n",
      "\t/tmp/tmpnwuUOv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKnaRAQ.lastz\n",
      "\t/tmp/tmppwZ5R8.lastz\n",
      "\t/tmp/tmpYVydgF.lastz\n",
      "\t/tmp/tmpVQrXGZ.lastz\n",
      "\t/tmp/tmpDYxYfD.lastz\n",
      "\t/tmp/tmpFXuk4S.lastz\n",
      "\t/tmp/tmpiRFc89.lastz\n",
      "\t/tmp/tmpYi_n9a.lastz\n",
      "\t/tmp/tmpk_ju54.lastz\n",
      "\t/tmp/tmpGxIBV9.lastz\n",
      "\t/tmp/tmpWNhzkp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Ditylenchus_destructor_GCA_001579705.lastz\n",
      "Creating Ditylenchus_destructor_GCA_001579705 table\n",
      "Inserting data to Ditylenchus_destructor_GCA_001579705 table\n",
      "\n",
      "Running against Ditylenchus_dipsaci_GCA_004194705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 23 queries...\n",
      "\t/tmp/tmpcfInFe.fasta\n",
      "\t/tmp/tmpaJIu7F.fasta\n",
      "\t/tmp/tmpURYDbA.fasta\n",
      "\t/tmp/tmpYohyGe.fasta\n",
      "\t/tmp/tmpJRxzVo.fasta\n",
      "\t/tmp/tmpyREpLQ.fasta\n",
      "\t/tmp/tmpw2jDe4.fasta\n",
      "\t/tmp/tmpLdXfCR.fasta\n",
      "\t/tmp/tmpOraNzz.fasta\n",
      "\t/tmp/tmpwjygc0.fasta\n",
      "\t/tmp/tmpqgKH3N.fasta\n",
      "\t/tmp/tmpQR_kfM.fasta\n",
      "\t/tmp/tmpCGeeu_.fasta\n",
      "\t/tmp/tmpp7cvbZ.fasta\n",
      "\t/tmp/tmpdwDwuy.fasta\n",
      "\t/tmp/tmpPtSGQh.fasta\n",
      "\t/tmp/tmpyFmmvI.fasta\n",
      "\t/tmp/tmpAzflUK.fasta\n",
      "\t/tmp/tmpf3rTHP.fasta\n",
      "\t/tmp/tmp4sEO9G.fasta\n",
      "\t/tmp/tmp9ZqNUm.fasta\n",
      "\t/tmp/tmpXnSWwO.fasta\n",
      "\t/tmp/tmpEA47tL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzyIC7k.lastz\n",
      "\t/tmp/tmpwDZK_o.lastz\n",
      "\t/tmp/tmpCi2ABc.lastz\n",
      "\t/tmp/tmpPXZfSo.lastz\n",
      "\t/tmp/tmp38zYtH.lastz\n",
      "\t/tmp/tmpcy6HFH.lastz\n",
      "\t/tmp/tmpDvjsiR.lastz\n",
      "\t/tmp/tmpqLBZmA.lastz\n",
      "\t/tmp/tmpaN_ewz.lastz\n",
      "\t/tmp/tmpYtbuIv.lastz\n",
      "\t/tmp/tmphG54t6.lastz\n",
      "\t/tmp/tmp4yOXi3.lastz\n",
      "\t/tmp/tmpQkGmBS.lastz\n",
      "\t/tmp/tmpE0xJ3S.lastz\n",
      "\t/tmp/tmp95YfAU.lastz\n",
      "\t/tmp/tmpdEksMy.lastz\n",
      "\t/tmp/tmpyrN8lt.lastz\n",
      "\t/tmp/tmpGFZoPQ.lastz\n",
      "\t/tmp/tmpHmwOif.lastz\n",
      "\t/tmp/tmpoloCcB.lastz\n",
      "\t/tmp/tmpvIKV8r.lastz\n",
      "\t/tmp/tmpRRYjDL.lastz\n",
      "\t/tmp/tmpNkihfO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Ditylenchus_dipsaci_GCA_004194705.lastz\n",
      "Creating Ditylenchus_dipsaci_GCA_004194705 table\n",
      "Inserting data to Ditylenchus_dipsaci_GCA_004194705 table\n",
      "\n",
      "Running against Globodera_ellingtonae_GCA_001723225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 11 queries...\n",
      "\t/tmp/tmp0ewD7c.fasta\n",
      "\t/tmp/tmpcbKQly.fasta\n",
      "\t/tmp/tmpfazK0_.fasta\n",
      "\t/tmp/tmpNDeXE9.fasta\n",
      "\t/tmp/tmpiFQZVP.fasta\n",
      "\t/tmp/tmpJe80FM.fasta\n",
      "\t/tmp/tmprJCip2.fasta\n",
      "\t/tmp/tmp0Mfn24.fasta\n",
      "\t/tmp/tmpXumn6O.fasta\n",
      "\t/tmp/tmpa7Tiql.fasta\n",
      "\t/tmp/tmpV3npXA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCJYTXu.lastz\n",
      "\t/tmp/tmpS4DFa0.lastz\n",
      "\t/tmp/tmpNwTMna.lastz\n",
      "\t/tmp/tmp3sSsbd.lastz\n",
      "\t/tmp/tmpMbwS7v.lastz\n",
      "\t/tmp/tmpvFWA1O.lastz\n",
      "\t/tmp/tmpkCFo_q.lastz\n",
      "\t/tmp/tmpRyc3Qe.lastz\n",
      "\t/tmp/tmpZuDHdB.lastz\n",
      "\t/tmp/tmp_dOV3h.lastz\n",
      "\t/tmp/tmpJOQ83t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Globodera_ellingtonae_GCA_001723225.lastz\n",
      "Creating Globodera_ellingtonae_GCA_001723225 table\n",
      "Inserting data to Globodera_ellingtonae_GCA_001723225 table\n",
      "\n",
      "Running against Globodera_pallida_GCA_000724045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 13 queries...\n",
      "\t/tmp/tmpGfOmbP.fasta\n",
      "\t/tmp/tmpOyEv2Y.fasta\n",
      "\t/tmp/tmpeWPKqM.fasta\n",
      "\t/tmp/tmpkWxL5Z.fasta\n",
      "\t/tmp/tmpPZGWG0.fasta\n",
      "\t/tmp/tmpUQEtae.fasta\n",
      "\t/tmp/tmpL5Fo5N.fasta\n",
      "\t/tmp/tmpu4bU4c.fasta\n",
      "\t/tmp/tmpvFOeEW.fasta\n",
      "\t/tmp/tmpxcRW9w.fasta\n",
      "\t/tmp/tmptQ6Vg5.fasta\n",
      "\t/tmp/tmpT5bREd.fasta\n",
      "\t/tmp/tmpHCrBt9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyosK8E.lastz\n",
      "\t/tmp/tmpexqMkt.lastz\n",
      "\t/tmp/tmpM8iMvt.lastz\n",
      "\t/tmp/tmpHko6as.lastz\n",
      "\t/tmp/tmpUssRH9.lastz\n",
      "\t/tmp/tmpWFYtTl.lastz\n",
      "\t/tmp/tmp7X6GXK.lastz\n",
      "\t/tmp/tmptcGLGi.lastz\n",
      "\t/tmp/tmp5bnSLC.lastz\n",
      "\t/tmp/tmpHnmZzV.lastz\n",
      "\t/tmp/tmpNDOLZh.lastz\n",
      "\t/tmp/tmpx1wj3p.lastz\n",
      "\t/tmp/tmpkr92_I.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Globodera_pallida_GCA_000724045.lastz\n",
      "Creating Globodera_pallida_GCA_000724045 table\n",
      "Inserting data to Globodera_pallida_GCA_000724045 table\n",
      "\n",
      "Running against Globodera_rostochiensis_GCA_900079975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmpmwD_R8.fasta\n",
      "\t/tmp/tmpeTocZj.fasta\n",
      "\t/tmp/tmpHztiqm.fasta\n",
      "\t/tmp/tmpCVoaHf.fasta\n",
      "\t/tmp/tmpBFXnRX.fasta\n",
      "\t/tmp/tmpAeTgnn.fasta\n",
      "\t/tmp/tmpxU_hgH.fasta\n",
      "\t/tmp/tmpbza9gn.fasta\n",
      "\t/tmp/tmp_SB1sY.fasta\n",
      "\t/tmp/tmpSxGypc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplDvDN2.lastz\n",
      "\t/tmp/tmp1OnxLt.lastz\n",
      "\t/tmp/tmpj81Mwe.lastz\n",
      "\t/tmp/tmpVF4s1e.lastz\n",
      "\t/tmp/tmp_Pfi2K.lastz\n",
      "\t/tmp/tmpgyszrC.lastz\n",
      "\t/tmp/tmpSMuzrT.lastz\n",
      "\t/tmp/tmph4ox64.lastz\n",
      "\t/tmp/tmp9aoCx2.lastz\n",
      "\t/tmp/tmpuwn16T.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Globodera_rostochiensis_GCA_900079975.lastz\n",
      "Creating Globodera_rostochiensis_GCA_900079975 table\n",
      "Inserting data to Globodera_rostochiensis_GCA_900079975 table\n",
      "\n",
      "Running against Halicephalobus_mephisto_GCA_009193035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpoc0tad.fasta\n",
      "\t/tmp/tmpPQNGMP.fasta\n",
      "\t/tmp/tmp1zTbtH.fasta\n",
      "\t/tmp/tmpvs0G8Y.fasta\n",
      "\t/tmp/tmpwh5Fxh.fasta\n",
      "\t/tmp/tmpLyiiZR.fasta\n",
      "\t/tmp/tmp2HMgC_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRzRMcm.lastz\n",
      "\t/tmp/tmp1Q7q8T.lastz\n",
      "\t/tmp/tmpvuZYm5.lastz\n",
      "\t/tmp/tmpgmxEcm.lastz\n",
      "\t/tmp/tmpMjSABJ.lastz\n",
      "\t/tmp/tmp_ZOPji.lastz\n",
      "\t/tmp/tmp7aRnf3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Halicephalobus_mephisto_GCA_009193035.lastz\n",
      "Creating Halicephalobus_mephisto_GCA_009193035 table\n",
      "Inserting data to Halicephalobus_mephisto_GCA_009193035 table\n",
      "\n",
      "Running against Halicephalobus_sp_GCA_009761265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmp2j8nD4.fasta\n",
      "\t/tmp/tmpKRsj4k.fasta\n",
      "\t/tmp/tmpHIELur.fasta\n",
      "\t/tmp/tmpoA6snR.fasta\n",
      "\t/tmp/tmpsh15Rv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCV8Xxa.lastz\n",
      "\t/tmp/tmp9NJyQn.lastz\n",
      "\t/tmp/tmp0nBR8V.lastz\n",
      "\t/tmp/tmpopKH_z.lastz\n",
      "\t/tmp/tmpXRuRiZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Halicephalobus_sp_GCA_009761265.lastz\n",
      "Creating Halicephalobus_sp_GCA_009761265 table\n",
      "Inserting data to Halicephalobus_sp_GCA_009761265 table\n",
      "\n",
      "Running against Heterodera_glycines_GCA_004148225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 13 queries...\n",
      "\t/tmp/tmpyOJNvr.fasta\n",
      "\t/tmp/tmpOykLTf.fasta\n",
      "\t/tmp/tmpGCgXYL.fasta\n",
      "\t/tmp/tmpz7P0Zr.fasta\n",
      "\t/tmp/tmpPqCM6y.fasta\n",
      "\t/tmp/tmpIQinBU.fasta\n",
      "\t/tmp/tmprYdp3M.fasta\n",
      "\t/tmp/tmp88D9s5.fasta\n",
      "\t/tmp/tmpuQlY7U.fasta\n",
      "\t/tmp/tmp_bYfF0.fasta\n",
      "\t/tmp/tmp2Qb5zp.fasta\n",
      "\t/tmp/tmp3UDTQS.fasta\n",
      "\t/tmp/tmp3076_q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptq3VqI.lastz\n",
      "\t/tmp/tmpINpLWC.lastz\n",
      "\t/tmp/tmpGyLUy4.lastz\n",
      "\t/tmp/tmpzTBKKk.lastz\n",
      "\t/tmp/tmpNmzlCn.lastz\n",
      "\t/tmp/tmpH8wny2.lastz\n",
      "\t/tmp/tmpfJF2N0.lastz\n",
      "\t/tmp/tmpd8eNpY.lastz\n",
      "\t/tmp/tmpUrctal.lastz\n",
      "\t/tmp/tmpmQdIds.lastz\n",
      "\t/tmp/tmpPmxKIl.lastz\n",
      "\t/tmp/tmpjI4mwz.lastz\n",
      "\t/tmp/tmpWqngWk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Heterodera_glycines_GCA_004148225.lastz\n",
      "Creating Heterodera_glycines_GCA_004148225 table\n",
      "Inserting data to Heterodera_glycines_GCA_004148225 table\n",
      "\n",
      "Running against Heterorhabditis_bacteriophora_GCA_000223415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmpyudogu.fasta\n",
      "\t/tmp/tmp2zYD8S.fasta\n",
      "\t/tmp/tmpizFitN.fasta\n",
      "\t/tmp/tmplv5YIO.fasta\n",
      "\t/tmp/tmpDI0ag9.fasta\n",
      "\t/tmp/tmpSF3had.fasta\n",
      "\t/tmp/tmp_JETvB.fasta\n",
      "\t/tmp/tmpFS_vNZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1z_9IY.lastz\n",
      "\t/tmp/tmpWQhBmn.lastz\n",
      "\t/tmp/tmpZ8XIqs.lastz\n",
      "\t/tmp/tmptlV2z5.lastz\n",
      "\t/tmp/tmpaR9uEZ.lastz\n",
      "\t/tmp/tmprzTpN0.lastz\n",
      "\t/tmp/tmpImbqhQ.lastz\n",
      "\t/tmp/tmpxhSq9t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Heterorhabditis_bacteriophora_GCA_000223415.lastz\n",
      "Creating Heterorhabditis_bacteriophora_GCA_000223415 table\n",
      "Inserting data to Heterorhabditis_bacteriophora_GCA_000223415 table\n",
      "\n",
      "Running against Meloidogyne_arenaria_GCA_003133805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 28 queries...\n",
      "\t/tmp/tmpTXkGnK.fasta\n",
      "\t/tmp/tmpK1r6Yt.fasta\n",
      "\t/tmp/tmpL738J0.fasta\n",
      "\t/tmp/tmpPxT8SK.fasta\n",
      "\t/tmp/tmphFJVHu.fasta\n",
      "\t/tmp/tmp5ygyaB.fasta\n",
      "\t/tmp/tmpMGeQoR.fasta\n",
      "\t/tmp/tmp8W1IZ0.fasta\n",
      "\t/tmp/tmpX4jCPn.fasta\n",
      "\t/tmp/tmpqq9Ser.fasta\n",
      "\t/tmp/tmp0u5th2.fasta\n",
      "\t/tmp/tmpQTDAn8.fasta\n",
      "\t/tmp/tmpwOzuop.fasta\n",
      "\t/tmp/tmpMgl9yh.fasta\n",
      "\t/tmp/tmpVcMseQ.fasta\n",
      "\t/tmp/tmpWGe5vF.fasta\n",
      "\t/tmp/tmpWuSK_N.fasta\n",
      "\t/tmp/tmpe5GtT6.fasta\n",
      "\t/tmp/tmpuvyHYX.fasta\n",
      "\t/tmp/tmpalh3u7.fasta\n",
      "\t/tmp/tmpEv5dog.fasta\n",
      "\t/tmp/tmpGzVvji.fasta\n",
      "\t/tmp/tmpIaVgic.fasta\n",
      "\t/tmp/tmpP5OU2M.fasta\n",
      "\t/tmp/tmpvYb1Cn.fasta\n",
      "\t/tmp/tmpNNk5IZ.fasta\n",
      "\t/tmp/tmp9mfLOb.fasta\n",
      "\t/tmp/tmpDga_eO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxFva12.lastz\n",
      "\t/tmp/tmp2KUQ4d.lastz\n",
      "\t/tmp/tmpI1_k9a.lastz\n",
      "\t/tmp/tmpqMrDDM.lastz\n",
      "\t/tmp/tmpqKv2fN.lastz\n",
      "\t/tmp/tmp_dWMzu.lastz\n",
      "\t/tmp/tmpaWC_Rn.lastz\n",
      "\t/tmp/tmpdzdE8J.lastz\n",
      "\t/tmp/tmpsvYyjM.lastz\n",
      "\t/tmp/tmp0nkuFI.lastz\n",
      "\t/tmp/tmpdmOewj.lastz\n",
      "\t/tmp/tmpy90vP7.lastz\n",
      "\t/tmp/tmphAo_8Y.lastz\n",
      "\t/tmp/tmpURrRv9.lastz\n",
      "\t/tmp/tmpgafgtR.lastz\n",
      "\t/tmp/tmpLioD6O.lastz\n",
      "\t/tmp/tmpTQu3jT.lastz\n",
      "\t/tmp/tmpoSyT6N.lastz\n",
      "\t/tmp/tmp7rpNbE.lastz\n",
      "\t/tmp/tmpRyFhTi.lastz\n",
      "\t/tmp/tmpMv2NV_.lastz\n",
      "\t/tmp/tmpHjPA2h.lastz\n",
      "\t/tmp/tmpyNy1nQ.lastz\n",
      "\t/tmp/tmpPDCu7S.lastz\n",
      "\t/tmp/tmpBSO_Zw.lastz\n",
      "\t/tmp/tmpeubP0T.lastz\n",
      "\t/tmp/tmppXnATq.lastz\n",
      "\t/tmp/tmp8h8rX7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_arenaria_GCA_003133805.lastz\n",
      "Creating Meloidogyne_arenaria_GCA_003133805 table\n",
      "Inserting data to Meloidogyne_arenaria_GCA_003133805 table\n",
      "\n",
      "Running against Meloidogyne_enterolobii_GCA_003693675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 17 queries...\n",
      "\t/tmp/tmpO2Mzrd.fasta\n",
      "\t/tmp/tmpezBJEP.fasta\n",
      "\t/tmp/tmpbACjAs.fasta\n",
      "\t/tmp/tmpk0AtTB.fasta\n",
      "\t/tmp/tmpgumtFV.fasta\n",
      "\t/tmp/tmp9f7610.fasta\n",
      "\t/tmp/tmpvjkLtW.fasta\n",
      "\t/tmp/tmpAniSdO.fasta\n",
      "\t/tmp/tmpvqTv4Q.fasta\n",
      "\t/tmp/tmp_QIrLo.fasta\n",
      "\t/tmp/tmpZp_Q0O.fasta\n",
      "\t/tmp/tmpxJSPtx.fasta\n",
      "\t/tmp/tmpFCLfGr.fasta\n",
      "\t/tmp/tmp16hyou.fasta\n",
      "\t/tmp/tmpK2kORu.fasta\n",
      "\t/tmp/tmpx5RxTM.fasta\n",
      "\t/tmp/tmprpqthd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSFN2Jl.lastz\n",
      "\t/tmp/tmp1LathJ.lastz\n",
      "\t/tmp/tmp19P_RC.lastz\n",
      "\t/tmp/tmpMBWUjJ.lastz\n",
      "\t/tmp/tmpsmShn3.lastz\n",
      "\t/tmp/tmptv9nQW.lastz\n",
      "\t/tmp/tmpJkfOOm.lastz\n",
      "\t/tmp/tmpzqgmop.lastz\n",
      "\t/tmp/tmpGHuX5R.lastz\n",
      "\t/tmp/tmp0l_f9G.lastz\n",
      "\t/tmp/tmpnVmdfc.lastz\n",
      "\t/tmp/tmp7u7EX5.lastz\n",
      "\t/tmp/tmp70i1a9.lastz\n",
      "\t/tmp/tmpQ4Qs5J.lastz\n",
      "\t/tmp/tmpDiEVAI.lastz\n",
      "\t/tmp/tmpJTMyPj.lastz\n",
      "\t/tmp/tmpAVtAGg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_enterolobii_GCA_003693675.lastz\n",
      "Creating Meloidogyne_enterolobii_GCA_003693675 table\n",
      "Inserting data to Meloidogyne_enterolobii_GCA_003693675 table\n",
      "\n",
      "Running against Meloidogyne_floridensis_GCA_003693605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmpd89C2i.fasta\n",
      "\t/tmp/tmp_LH474.fasta\n",
      "\t/tmp/tmpJJLjcY.fasta\n",
      "\t/tmp/tmpEJL9tP.fasta\n",
      "\t/tmp/tmpTvSMJI.fasta\n",
      "\t/tmp/tmpO7fEr8.fasta\n",
      "\t/tmp/tmp59GFoL.fasta\n",
      "\t/tmp/tmpNZCanC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWmAqiR.lastz\n",
      "\t/tmp/tmpl8rjwy.lastz\n",
      "\t/tmp/tmpgX0mTG.lastz\n",
      "\t/tmp/tmpe8Im0a.lastz\n",
      "\t/tmp/tmpp3wJ62.lastz\n",
      "\t/tmp/tmpuns0z4.lastz\n",
      "\t/tmp/tmpg6YewE.lastz\n",
      "\t/tmp/tmpU0Vmci.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_floridensis_GCA_003693605.lastz\n",
      "Creating Meloidogyne_floridensis_GCA_003693605 table\n",
      "Inserting data to Meloidogyne_floridensis_GCA_003693605 table\n",
      "\n",
      "Running against Meloidogyne_graminicola_GCA_002778205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpwgvVjb.fasta\n",
      "\t/tmp/tmpEjEnHY.fasta\n",
      "\t/tmp/tmpMuOzee.fasta\n",
      "\t/tmp/tmpNZ_Nq6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTLqwPG.lastz\n",
      "\t/tmp/tmpICrl1e.lastz\n",
      "\t/tmp/tmpfN5gZj.lastz\n",
      "\t/tmp/tmpX2rTHB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_graminicola_GCA_002778205.lastz\n",
      "Creating Meloidogyne_graminicola_GCA_002778205 table\n",
      "Inserting data to Meloidogyne_graminicola_GCA_002778205 table\n",
      "\n",
      "Running against Meloidogyne_hapla_GCA_000172435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpV2Pimm.fasta\n",
      "\t/tmp/tmpxbKoCC.fasta\n",
      "\t/tmp/tmpoSkzTc.fasta\n",
      "\t/tmp/tmp51hoJ9.fasta\n",
      "\t/tmp/tmpLhklfc.fasta\n",
      "\t/tmp/tmp2yC5mB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNP2U3G.lastz\n",
      "\t/tmp/tmpoqg2bS.lastz\n",
      "\t/tmp/tmpa8GXqW.lastz\n",
      "\t/tmp/tmpIz_j3N.lastz\n",
      "\t/tmp/tmpTZWpfj.lastz\n",
      "\t/tmp/tmpamNlL_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_hapla_GCA_000172435.lastz\n",
      "Creating Meloidogyne_hapla_GCA_000172435 table\n",
      "Inserting data to Meloidogyne_hapla_GCA_000172435 table\n",
      "\n",
      "Running against Meloidogyne_incognita_GCA_900182535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 19 queries...\n",
      "\t/tmp/tmpFjw4Fr.fasta\n",
      "\t/tmp/tmp1WFdpF.fasta\n",
      "\t/tmp/tmpF1mPCl.fasta\n",
      "\t/tmp/tmpwWto0E.fasta\n",
      "\t/tmp/tmpTQdtrq.fasta\n",
      "\t/tmp/tmpXrKs5v.fasta\n",
      "\t/tmp/tmpOHRE4Y.fasta\n",
      "\t/tmp/tmpa6V7Ik.fasta\n",
      "\t/tmp/tmplmJE22.fasta\n",
      "\t/tmp/tmp7AI5Fq.fasta\n",
      "\t/tmp/tmpJj_sof.fasta\n",
      "\t/tmp/tmpVerVPg.fasta\n",
      "\t/tmp/tmpZzOqvh.fasta\n",
      "\t/tmp/tmp6H02hB.fasta\n",
      "\t/tmp/tmpGpCf_R.fasta\n",
      "\t/tmp/tmp95yNzj.fasta\n",
      "\t/tmp/tmpymHttC.fasta\n",
      "\t/tmp/tmpdvZBMs.fasta\n",
      "\t/tmp/tmpBIXCPV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpA_SPiz.lastz\n",
      "\t/tmp/tmpQDVxED.lastz\n",
      "\t/tmp/tmpintBEu.lastz\n",
      "\t/tmp/tmpIL4sL6.lastz\n",
      "\t/tmp/tmpKQU_Dx.lastz\n",
      "\t/tmp/tmpRJix8n.lastz\n",
      "\t/tmp/tmpJQnCkJ.lastz\n",
      "\t/tmp/tmpOm9VKy.lastz\n",
      "\t/tmp/tmpn5fDVI.lastz\n",
      "\t/tmp/tmp91OTxR.lastz\n",
      "\t/tmp/tmplL39ED.lastz\n",
      "\t/tmp/tmpo6cac3.lastz\n",
      "\t/tmp/tmpDRB2zH.lastz\n",
      "\t/tmp/tmpPJ09q6.lastz\n",
      "\t/tmp/tmpCK9BC1.lastz\n",
      "\t/tmp/tmpLy8YSv.lastz\n",
      "\t/tmp/tmpW5qhqf.lastz\n",
      "\t/tmp/tmphdoBSz.lastz\n",
      "\t/tmp/tmpLv1Eur.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_incognita_GCA_900182535.lastz\n",
      "Creating Meloidogyne_incognita_GCA_900182535 table\n",
      "Inserting data to Meloidogyne_incognita_GCA_900182535 table\n",
      "\n",
      "Running against Meloidogyne_javanica_GCA_003693625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 16 queries...\n",
      "\t/tmp/tmp6msav7.fasta\n",
      "\t/tmp/tmp2lkQID.fasta\n",
      "\t/tmp/tmpvBQuDc.fasta\n",
      "\t/tmp/tmp4e3cwp.fasta\n",
      "\t/tmp/tmpZvCt9L.fasta\n",
      "\t/tmp/tmpJMwoTo.fasta\n",
      "\t/tmp/tmplVkpJn.fasta\n",
      "\t/tmp/tmpcgkVmB.fasta\n",
      "\t/tmp/tmpUqqxnz.fasta\n",
      "\t/tmp/tmpZfvs5G.fasta\n",
      "\t/tmp/tmpMT4zJG.fasta\n",
      "\t/tmp/tmpoKgXdX.fasta\n",
      "\t/tmp/tmpqr3q2E.fasta\n",
      "\t/tmp/tmpN6O6q3.fasta\n",
      "\t/tmp/tmpSkt4Gu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLkSFoL.lastz\n",
      "\t/tmp/tmpO45p3I.lastz\n",
      "\t/tmp/tmpDntys7.lastz\n",
      "\t/tmp/tmppBH80I.lastz\n",
      "\t/tmp/tmpHWM2JK.lastz\n",
      "\t/tmp/tmpFwm1oP.lastz\n",
      "\t/tmp/tmpG7L5Oz.lastz\n",
      "\t/tmp/tmp1_m9xD.lastz\n",
      "\t/tmp/tmpMJw6FT.lastz\n",
      "\t/tmp/tmpIGOy_h.lastz\n",
      "\t/tmp/tmpMa6tVl.lastz\n",
      "\t/tmp/tmp7tIqed.lastz\n",
      "\t/tmp/tmpJzEiwG.lastz\n",
      "\t/tmp/tmpWAHdzZ.lastz\n",
      "\t/tmp/tmp5ZXNCk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_javanica_GCA_003693625.lastz\n",
      "Creating Meloidogyne_javanica_GCA_003693625 table\n",
      "Inserting data to Meloidogyne_javanica_GCA_003693625 table\n",
      "\n",
      "Running against Meloidogyne_luci_GCA_902706615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 19 queries...\n",
      "\t/tmp/tmpX6TFN8.fasta\n",
      "\t/tmp/tmpuY98aV.fasta\n",
      "\t/tmp/tmpqZMWgl.fasta\n",
      "\t/tmp/tmp1rDEyB.fasta\n",
      "\t/tmp/tmpMgTzuR.fasta\n",
      "\t/tmp/tmpUag0ye.fasta\n",
      "\t/tmp/tmpgX46HO.fasta\n",
      "\t/tmp/tmpDiXp4H.fasta\n",
      "\t/tmp/tmpQEEoIG.fasta\n",
      "\t/tmp/tmpJbLpck.fasta\n",
      "\t/tmp/tmp2oTTPd.fasta\n",
      "\t/tmp/tmpWzSvXu.fasta\n",
      "\t/tmp/tmpa49pgP.fasta\n",
      "\t/tmp/tmpgFJPTl.fasta\n",
      "\t/tmp/tmpgKpQUC.fasta\n",
      "\t/tmp/tmphyDCI7.fasta\n",
      "\t/tmp/tmpFYQLeb.fasta\n",
      "\t/tmp/tmpDGOQSy.fasta\n",
      "\t/tmp/tmpTyG7Mf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8un57N.lastz\n",
      "\t/tmp/tmp8NiwBc.lastz\n",
      "\t/tmp/tmp3M4UQ0.lastz\n",
      "\t/tmp/tmpCjMk2q.lastz\n",
      "\t/tmp/tmpjLL8Qk.lastz\n",
      "\t/tmp/tmpdfzt7k.lastz\n",
      "\t/tmp/tmpKsNWOT.lastz\n",
      "\t/tmp/tmpU5GzSs.lastz\n",
      "\t/tmp/tmpSzDPx_.lastz\n",
      "\t/tmp/tmpBkRKLH.lastz\n",
      "\t/tmp/tmpV3ajD9.lastz\n",
      "\t/tmp/tmpkgHL46.lastz\n",
      "\t/tmp/tmprZiITT.lastz\n",
      "\t/tmp/tmpkwx1fy.lastz\n",
      "\t/tmp/tmpFCrlSE.lastz\n",
      "\t/tmp/tmpo6f312.lastz\n",
      "\t/tmp/tmpIUzaN8.lastz\n",
      "\t/tmp/tmpILyTYS.lastz\n",
      "\t/tmp/tmp4_x2pd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Meloidogyne_luci_GCA_902706615.lastz\n",
      "Creating Meloidogyne_luci_GCA_902706615 table\n",
      "Inserting data to Meloidogyne_luci_GCA_902706615 table\n",
      "\n",
      "Running against Panagrellus_redivivus_GCA_000341325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpXus5jo.fasta\n",
      "\t/tmp/tmpg8fBio.fasta\n",
      "\t/tmp/tmpdJiiEm.fasta\n",
      "\t/tmp/tmpVJc1ws.fasta\n",
      "\t/tmp/tmpP0Kiph.fasta\n",
      "\t/tmp/tmppGlZYk.fasta\n",
      "\t/tmp/tmpgEVlbq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVmX6B9.lastz\n",
      "\t/tmp/tmpaMGCHP.lastz\n",
      "\t/tmp/tmpTyXANJ.lastz\n",
      "\t/tmp/tmptwT4Zf.lastz\n",
      "\t/tmp/tmpnxbWSf.lastz\n",
      "\t/tmp/tmptIwVxe.lastz\n",
      "\t/tmp/tmp8K4PHG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Panagrellus_redivivus_GCA_000341325.lastz\n",
      "Creating Panagrellus_redivivus_GCA_000341325 table\n",
      "Inserting data to Panagrellus_redivivus_GCA_000341325 table\n",
      "\n",
      "Running against Panagrolaimus_davidi_GCA_901779475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 12 queries...\n",
      "\t/tmp/tmp7ZZIjP.fasta\n",
      "\t/tmp/tmpMKPqAK.fasta\n",
      "\t/tmp/tmpYTyWzi.fasta\n",
      "\t/tmp/tmpWor5AS.fasta\n",
      "\t/tmp/tmpuFEpT6.fasta\n",
      "\t/tmp/tmpTVqXX0.fasta\n",
      "\t/tmp/tmpNmUf5T.fasta\n",
      "\t/tmp/tmpfgnNtv.fasta\n",
      "\t/tmp/tmpv_Wlor.fasta\n",
      "\t/tmp/tmpVFXr9x.fasta\n",
      "\t/tmp/tmpekf_UX.fasta\n",
      "\t/tmp/tmpWx6P_t.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBiiQv1.lastz\n",
      "\t/tmp/tmprNfnBG.lastz\n",
      "\t/tmp/tmpT386h_.lastz\n",
      "\t/tmp/tmpzo5hza.lastz\n",
      "\t/tmp/tmp3IImEh.lastz\n",
      "\t/tmp/tmpvrrNHC.lastz\n",
      "\t/tmp/tmpxKRT6w.lastz\n",
      "\t/tmp/tmpeGZjOJ.lastz\n",
      "\t/tmp/tmpDm3WtU.lastz\n",
      "\t/tmp/tmpeRLT2d.lastz\n",
      "\t/tmp/tmpK6SHLB.lastz\n",
      "\t/tmp/tmppKzbu5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Panagrolaimus_davidi_GCA_901779475.lastz\n",
      "Creating Panagrolaimus_davidi_GCA_901779475 table\n",
      "Inserting data to Panagrolaimus_davidi_GCA_901779475 table\n",
      "\n",
      "Running against Panagrolaimus_sp_GCA_901766855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmpH7ZZmY.fasta\n",
      "\t/tmp/tmptP7J5P.fasta\n",
      "\t/tmp/tmpSeP5g8.fasta\n",
      "\t/tmp/tmpGD_h_d.fasta\n",
      "\t/tmp/tmptjFYfd.fasta\n",
      "\t/tmp/tmpsL8SNl.fasta\n",
      "\t/tmp/tmpvXTS4P.fasta\n",
      "\t/tmp/tmpNPcCD6.fasta\n",
      "\t/tmp/tmpLXu0n1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphlh2UT.lastz\n",
      "\t/tmp/tmpU0BfSE.lastz\n",
      "\t/tmp/tmpguckkn.lastz\n",
      "\t/tmp/tmpxiH4oy.lastz\n",
      "\t/tmp/tmp43Oxec.lastz\n",
      "\t/tmp/tmpnLL2Z_.lastz\n",
      "\t/tmp/tmpuHdGVn.lastz\n",
      "\t/tmp/tmp06dvun.lastz\n",
      "\t/tmp/tmpDMLCzM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Panagrolaimus_sp_GCA_901766855.lastz\n",
      "Creating Panagrolaimus_sp_GCA_901766855 table\n",
      "Inserting data to Panagrolaimus_sp_GCA_901766855 table\n",
      "\n",
      "Running against Panagrolaimus_superbus_GCA_901766145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmp8_Km1P.fasta\n",
      "\t/tmp/tmpZB9uWU.fasta\n",
      "\t/tmp/tmplyHU0b.fasta\n",
      "\t/tmp/tmp5EqgOa.fasta\n",
      "\t/tmp/tmp4GCfII.fasta\n",
      "\t/tmp/tmpswTNpB.fasta\n",
      "\t/tmp/tmpa_V041.fasta\n",
      "\t/tmp/tmpiheGqP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgpn4Un.lastz\n",
      "\t/tmp/tmpBMBIGq.lastz\n",
      "\t/tmp/tmp6WI61Y.lastz\n",
      "\t/tmp/tmpUHnOEa.lastz\n",
      "\t/tmp/tmp3Ntkz_.lastz\n",
      "\t/tmp/tmp836K3u.lastz\n",
      "\t/tmp/tmp7SOlHK.lastz\n",
      "\t/tmp/tmpQQXVcQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Panagrolaimus_superbus_GCA_901766145.lastz\n",
      "Creating Panagrolaimus_superbus_GCA_901766145 table\n",
      "Inserting data to Panagrolaimus_superbus_GCA_901766145 table\n",
      "\n",
      "Running against Radopholus_similis_GCA_004764675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpNdcRC4.fasta\n",
      "\t/tmp/tmpNxI_N6.fasta\n",
      "\t/tmp/tmpC54QHN.fasta\n",
      "\t/tmp/tmpuIiEcR.fasta\n",
      "\t/tmp/tmp6u8ru_.fasta\n",
      "\t/tmp/tmpG7ds6l.fasta\n",
      "\t/tmp/tmpGKACqc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwEkNHZ.lastz\n",
      "\t/tmp/tmp244mtK.lastz\n",
      "\t/tmp/tmpiVxW8u.lastz\n",
      "\t/tmp/tmpUASghS.lastz\n",
      "\t/tmp/tmpeMFvNb.lastz\n",
      "\t/tmp/tmp85hkIj.lastz\n",
      "\t/tmp/tmpmtpJCT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Radopholus_similis_GCA_004764675.lastz\n",
      "Creating Radopholus_similis_GCA_004764675 table\n",
      "Inserting data to Radopholus_similis_GCA_004764675 table\n",
      "\n",
      "Running against Rotylenchulus_reniformis_GCA_001026735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 32 queries...\n",
      "\t/tmp/tmpEknTT_.fasta\n",
      "\t/tmp/tmp5PNptm.fasta\n",
      "\t/tmp/tmpTYWltd.fasta\n",
      "\t/tmp/tmpmbdP87.fasta\n",
      "\t/tmp/tmpULroAu.fasta\n",
      "\t/tmp/tmp65CByE.fasta\n",
      "\t/tmp/tmpnCjc2c.fasta\n",
      "\t/tmp/tmpOPbPOo.fasta\n",
      "\t/tmp/tmp1EFkID.fasta\n",
      "\t/tmp/tmp4e3U7j.fasta\n",
      "\t/tmp/tmpfiQh_B.fasta\n",
      "\t/tmp/tmplutKbj.fasta\n",
      "\t/tmp/tmpI8mEvw.fasta\n",
      "\t/tmp/tmp2fIqeO.fasta\n",
      "\t/tmp/tmp0yyl8v.fasta\n",
      "\t/tmp/tmpJNWbT4.fasta\n",
      "\t/tmp/tmpU8QEhs.fasta\n",
      "\t/tmp/tmp5wioao.fasta\n",
      "\t/tmp/tmpQQbs_8.fasta\n",
      "\t/tmp/tmp3NX77z.fasta\n",
      "\t/tmp/tmppZbaAj.fasta\n",
      "\t/tmp/tmpGf4yBl.fasta\n",
      "\t/tmp/tmpbPPxIM.fasta\n",
      "\t/tmp/tmp0cYlZ2.fasta\n",
      "\t/tmp/tmpgDG6Ri.fasta\n",
      "\t/tmp/tmp596kd0.fasta\n",
      "\t/tmp/tmpvuyGdo.fasta\n",
      "\t/tmp/tmprM9XzE.fasta\n",
      "\t/tmp/tmpw0DsZ5.fasta\n",
      "\t/tmp/tmpH93Bzn.fasta\n",
      "\t/tmp/tmpR9SFcw.fasta\n",
      "\t/tmp/tmpT0ogby.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsGLixr.lastz\n",
      "\t/tmp/tmp5UP7u7.lastz\n",
      "\t/tmp/tmpg46Uzl.lastz\n",
      "\t/tmp/tmphnqdmU.lastz\n",
      "\t/tmp/tmpZ8_aoP.lastz\n",
      "\t/tmp/tmpQP5S2S.lastz\n",
      "\t/tmp/tmpqIrf3A.lastz\n",
      "\t/tmp/tmpAo8ogJ.lastz\n",
      "\t/tmp/tmpvUADc1.lastz\n",
      "\t/tmp/tmpWpMuFD.lastz\n",
      "\t/tmp/tmpOb3wob.lastz\n",
      "\t/tmp/tmpBVu7IB.lastz\n",
      "\t/tmp/tmpUnsLis.lastz\n",
      "\t/tmp/tmpB080Pp.lastz\n",
      "\t/tmp/tmpMCm7Dh.lastz\n",
      "\t/tmp/tmpd_wK0u.lastz\n",
      "\t/tmp/tmpeI8AH2.lastz\n",
      "\t/tmp/tmpxXp8GS.lastz\n",
      "\t/tmp/tmpOyXvUT.lastz\n",
      "\t/tmp/tmpcEYJQ6.lastz\n",
      "\t/tmp/tmpwOGh6v.lastz\n",
      "\t/tmp/tmp75D08s.lastz\n",
      "\t/tmp/tmpsNKAh3.lastz\n",
      "\t/tmp/tmp2gigW6.lastz\n",
      "\t/tmp/tmpZlriYF.lastz\n",
      "\t/tmp/tmpvc5qvP.lastz\n",
      "\t/tmp/tmp7EZHF3.lastz\n",
      "\t/tmp/tmpqXb5ch.lastz\n",
      "\t/tmp/tmppzqAA1.lastz\n",
      "\t/tmp/tmpiyO7FN.lastz\n",
      "\t/tmp/tmpMnlS87.lastz\n",
      "\t/tmp/tmpPHhmMt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Rotylenchulus_reniformis_GCA_001026735.lastz\n",
      "Creating Rotylenchulus_reniformis_GCA_001026735 table\n",
      "Inserting data to Rotylenchulus_reniformis_GCA_001026735 table\n",
      "\n",
      "Running against Steinernema_feltiae_GCA_007213375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 13 queries...\n",
      "\t/tmp/tmppGgLXo.fasta\n",
      "\t/tmp/tmp_QJwzK.fasta\n",
      "\t/tmp/tmpZWsE9O.fasta\n",
      "\t/tmp/tmpv7GYlB.fasta\n",
      "\t/tmp/tmpF3Q4Bl.fasta\n",
      "\t/tmp/tmpDYArhE.fasta\n",
      "\t/tmp/tmpYBhTPP.fasta\n",
      "\t/tmp/tmp22kUol.fasta\n",
      "\t/tmp/tmp0SAaUe.fasta\n",
      "\t/tmp/tmpwfRJM3.fasta\n",
      "\t/tmp/tmpL5KxCa.fasta\n",
      "\t/tmp/tmpnklHzk.fasta\n",
      "\t/tmp/tmpu9Y09z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpymrObj.lastz\n",
      "\t/tmp/tmpXJrjiw.lastz\n",
      "\t/tmp/tmpuRY1DS.lastz\n",
      "\t/tmp/tmp9_aZ2h.lastz\n",
      "\t/tmp/tmpnkahQf.lastz\n",
      "\t/tmp/tmpNzXhMj.lastz\n",
      "\t/tmp/tmpwrOTrY.lastz\n",
      "\t/tmp/tmpApvSKo.lastz\n",
      "\t/tmp/tmpu7yMB3.lastz\n",
      "\t/tmp/tmpP2_jJg.lastz\n",
      "\t/tmp/tmp1TLC7y.lastz\n",
      "\t/tmp/tmpT1zo8n.lastz\n",
      "\t/tmp/tmpmOI1aY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Steinernema_feltiae_GCA_007213375.lastz\n",
      "Creating Steinernema_feltiae_GCA_007213375 table\n",
      "Inserting data to Steinernema_feltiae_GCA_007213375 table\n",
      "\n",
      "Running against Steinernema_glaseri_GCA_000757755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmp204C2Q.fasta\n",
      "\t/tmp/tmp3iC8_r.fasta\n",
      "\t/tmp/tmpJP76Ae.fasta\n",
      "\t/tmp/tmptGQ12X.fasta\n",
      "\t/tmp/tmpsx7QS8.fasta\n",
      "\t/tmp/tmp0txlLr.fasta\n",
      "\t/tmp/tmp_lsmrE.fasta\n",
      "\t/tmp/tmpN0MzdZ.fasta\n",
      "\t/tmp/tmpruC6oc.fasta\n",
      "\t/tmp/tmphJNoec.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUOHLTJ.lastz\n",
      "\t/tmp/tmpYB7zI3.lastz\n",
      "\t/tmp/tmpZJ65Me.lastz\n",
      "\t/tmp/tmpw4JW17.lastz\n",
      "\t/tmp/tmp1cH_Wx.lastz\n",
      "\t/tmp/tmpBgnhP1.lastz\n",
      "\t/tmp/tmpyigzy0.lastz\n",
      "\t/tmp/tmpGHOaTJ.lastz\n",
      "\t/tmp/tmpZ8NIN2.lastz\n",
      "\t/tmp/tmp2MflOY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Steinernema_glaseri_GCA_000757755.lastz\n",
      "Creating Steinernema_glaseri_GCA_000757755 table\n",
      "Inserting data to Steinernema_glaseri_GCA_000757755 table\n",
      "\n",
      "Running against Steinernema_monticolum_GCA_000505645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmpCELH16.fasta\n",
      "\t/tmp/tmpxKYLjW.fasta\n",
      "\t/tmp/tmpxJoFbE.fasta\n",
      "\t/tmp/tmpNX4Jws.fasta\n",
      "\t/tmp/tmpr64Mdx.fasta\n",
      "\t/tmp/tmpP0sArX.fasta\n",
      "\t/tmp/tmpNObeUO.fasta\n",
      "\t/tmp/tmpTKCsrY.fasta\n",
      "\t/tmp/tmpZ_jWAq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpi5HIVb.lastz\n",
      "\t/tmp/tmpMFlKoe.lastz\n",
      "\t/tmp/tmp3s9okV.lastz\n",
      "\t/tmp/tmp3M3Ts3.lastz\n",
      "\t/tmp/tmpPixad5.lastz\n",
      "\t/tmp/tmpmZi1xQ.lastz\n",
      "\t/tmp/tmp7JFr9h.lastz\n",
      "\t/tmp/tmpkxq_Vv.lastz\n",
      "\t/tmp/tmp22VLsW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Steinernema_monticolum_GCA_000505645.lastz\n",
      "Creating Steinernema_monticolum_GCA_000505645 table\n",
      "Inserting data to Steinernema_monticolum_GCA_000505645 table\n",
      "\n",
      "Running against Steinernema_scapterisci_GCA_000757745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmp4vcXr5.fasta\n",
      "\t/tmp/tmpOIcAs6.fasta\n",
      "\t/tmp/tmpEqeFXh.fasta\n",
      "\t/tmp/tmpR7dqKQ.fasta\n",
      "\t/tmp/tmpRocti_.fasta\n",
      "\t/tmp/tmpimwcku.fasta\n",
      "\t/tmp/tmpXtwzzY.fasta\n",
      "\t/tmp/tmpYP2Eex.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPy7TYT.lastz\n",
      "\t/tmp/tmpYbHUFd.lastz\n",
      "\t/tmp/tmp_maP7A.lastz\n",
      "\t/tmp/tmpASBhpl.lastz\n",
      "\t/tmp/tmpMT2EUH.lastz\n",
      "\t/tmp/tmp7AFBX_.lastz\n",
      "\t/tmp/tmplsdscP.lastz\n",
      "\t/tmp/tmp4QfhrK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Steinernema_scapterisci_GCA_000757745.lastz\n",
      "Creating Steinernema_scapterisci_GCA_000757745 table\n",
      "Inserting data to Steinernema_scapterisci_GCA_000757745 table\n",
      "\n",
      "Running against Subanguina_moxae_GCA_000981365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmpXsQpLw.fasta\n",
      "\t/tmp/tmpNR_qtT.fasta\n",
      "\t/tmp/tmpYqKlEz.fasta\n",
      "\t/tmp/tmpycqUsA.fasta\n",
      "\t/tmp/tmpiBj2Go.fasta\n",
      "\t/tmp/tmp1QXVxV.fasta\n",
      "\t/tmp/tmpVnA5wr.fasta\n",
      "\t/tmp/tmpu7raTz.fasta\n",
      "\t/tmp/tmpehFq92.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8OgLmS.lastz\n",
      "\t/tmp/tmpOaKfki.lastz\n",
      "\t/tmp/tmpmRUhbP.lastz\n",
      "\t/tmp/tmpIsl20J.lastz\n",
      "\t/tmp/tmpPgC_vO.lastz\n",
      "\t/tmp/tmpSdCoIj.lastz\n",
      "\t/tmp/tmpeXpTUh.lastz\n",
      "\t/tmp/tmpc34crr.lastz\n",
      "\t/tmp/tmpR314ZF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Subanguina_moxae_GCA_000981365.lastz\n",
      "Creating Subanguina_moxae_GCA_000981365 table\n",
      "Inserting data to Subanguina_moxae_GCA_000981365 table\n",
      "\n",
      "Running against Steinernema_carpocapsae_GCA_000757645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpdbCyqv.fasta\n",
      "\t/tmp/tmpoPpLTz.fasta\n",
      "\t/tmp/tmp5b66QQ.fasta\n",
      "\t/tmp/tmpn9duxV.fasta\n",
      "\t/tmp/tmpHBwD8f.fasta\n",
      "\t/tmp/tmpt_7kPf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4k0bAj.lastz\n",
      "\t/tmp/tmpQdJrzn.lastz\n",
      "\t/tmp/tmpeEGA4q.lastz\n",
      "\t/tmp/tmpv07osG.lastz\n",
      "\t/tmp/tmpE9D8Ep.lastz\n",
      "\t/tmp/tmpwN3AuY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade4b/extract_probes_from_group/lastz/Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_Steinernema_carpocapsae_GCA_000757645.lastz\n",
      "Creating Steinernema_carpocapsae_GCA_000757645 table\n",
      "Inserting data to Steinernema_carpocapsae_GCA_000757645 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/nematoda-clade4b/extract_probes_from_group/nematoda-clade4b_genome.conf --lastz results/phyluce/nematoda-clade4b/validate_intervals/lastz --probes 120 --probe-prefix uce_nematoda-clade4b_ --name-pattern \"Steinernema_carpocapsae_GCA_000757645_+2_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/nematoda-clade4b/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 19:27:32,251 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 19:27:32,256 - Phyluce - INFO - ------- Working on Acrobeloides_nanus_GCA_900406225 genome ------\n",
      "2020-02-11 19:27:32,260 - Phyluce - INFO - Reading Acrobeloides_nanus_GCA_900406225 genome\n",
      "2020-02-11 19:27:36,433 - Phyluce - INFO - Acrobeloides_nanus_GCA_900406225: 268 uces, 176 dupes, 92 non-dupes, 0 orient drop, 2 length drop, 90 written\n",
      "2020-02-11 19:27:36,433 - Phyluce - INFO - --- Working on Bursaphelenchus_xylophilus_GCA_004128815 genome --\n",
      "2020-02-11 19:27:36,434 - Phyluce - INFO - Reading Bursaphelenchus_xylophilus_GCA_004128815 genome\n",
      "2020-02-11 19:27:38,617 - Phyluce - INFO - Bursaphelenchus_xylophilus_GCA_004128815: 264 uces, 110 dupes, 154 non-dupes, 6 orient drop, 12 length drop, 136 written\n",
      "2020-02-11 19:27:38,617 - Phyluce - INFO - ----- Working on Deladenus_siricidicola_GCA_009724625 genome ----\n",
      "2020-02-11 19:27:38,651 - Phyluce - INFO - Reading Deladenus_siricidicola_GCA_009724625 genome\n",
      "2020-02-11 19:27:40,501 - Phyluce - INFO - Deladenus_siricidicola_GCA_009724625: 274 uces, 127 dupes, 147 non-dupes, 0 orient drop, 4 length drop, 143 written\n",
      "2020-02-11 19:27:40,501 - Phyluce - INFO - ----- Working on Ditylenchus_destructor_GCA_001579705 genome ----\n",
      "2020-02-11 19:27:40,515 - Phyluce - INFO - Reading Ditylenchus_destructor_GCA_001579705 genome\n",
      "2020-02-11 19:27:42,094 - Phyluce - INFO - Ditylenchus_destructor_GCA_001579705: 226 uces, 95 dupes, 131 non-dupes, 2 orient drop, 6 length drop, 123 written\n",
      "2020-02-11 19:27:42,094 - Phyluce - INFO - ------ Working on Ditylenchus_dipsaci_GCA_004194705 genome ------\n",
      "2020-02-11 19:27:42,130 - Phyluce - INFO - Reading Ditylenchus_dipsaci_GCA_004194705 genome\n",
      "2020-02-11 19:27:46,184 - Phyluce - INFO - Ditylenchus_dipsaci_GCA_004194705: 239 uces, 143 dupes, 96 non-dupes, 1 orient drop, 4 length drop, 91 written\n",
      "2020-02-11 19:27:46,184 - Phyluce - INFO - ----- Working on Globodera_ellingtonae_GCA_001723225 genome -----\n",
      "2020-02-11 19:27:46,208 - Phyluce - INFO - Reading Globodera_ellingtonae_GCA_001723225 genome\n",
      "2020-02-11 19:27:48,116 - Phyluce - INFO - Globodera_ellingtonae_GCA_001723225: 224 uces, 88 dupes, 136 non-dupes, 4 orient drop, 9 length drop, 123 written\n",
      "2020-02-11 19:27:48,117 - Phyluce - INFO - ------- Working on Globodera_pallida_GCA_000724045 genome -------\n",
      "2020-02-11 19:27:48,124 - Phyluce - INFO - Reading Globodera_pallida_GCA_000724045 genome\n",
      "2020-02-11 19:27:49,928 - Phyluce - INFO - Globodera_pallida_GCA_000724045: 208 uces, 113 dupes, 95 non-dupes, 1 orient drop, 3 length drop, 91 written\n",
      "2020-02-11 19:27:49,928 - Phyluce - INFO - ---- Working on Globodera_rostochiensis_GCA_900079975 genome ----\n",
      "2020-02-11 19:27:49,929 - Phyluce - INFO - Reading Globodera_rostochiensis_GCA_900079975 genome\n",
      "2020-02-11 19:27:51,530 - Phyluce - INFO - Globodera_rostochiensis_GCA_900079975: 230 uces, 84 dupes, 146 non-dupes, 3 orient drop, 9 length drop, 134 written\n",
      "2020-02-11 19:27:51,531 - Phyluce - INFO - ---- Working on Halicephalobus_mephisto_GCA_009193035 genome ----\n",
      "2020-02-11 19:27:51,572 - Phyluce - INFO - Reading Halicephalobus_mephisto_GCA_009193035 genome\n",
      "2020-02-11 19:27:53,638 - Phyluce - INFO - Halicephalobus_mephisto_GCA_009193035: 283 uces, 129 dupes, 154 non-dupes, 0 orient drop, 8 length drop, 146 written\n",
      "2020-02-11 19:27:53,638 - Phyluce - INFO - ------- Working on Halicephalobus_sp_GCA_009761265 genome -------\n",
      "2020-02-11 19:27:53,646 - Phyluce - INFO - Reading Halicephalobus_sp_GCA_009761265 genome\n",
      "2020-02-11 19:27:55,479 - Phyluce - INFO - Halicephalobus_sp_GCA_009761265: 277 uces, 119 dupes, 158 non-dupes, 0 orient drop, 0 length drop, 158 written\n",
      "2020-02-11 19:27:55,479 - Phyluce - INFO - ------ Working on Heterodera_glycines_GCA_004148225 genome ------\n",
      "2020-02-11 19:27:55,482 - Phyluce - INFO - Reading Heterodera_glycines_GCA_004148225 genome\n",
      "2020-02-11 19:27:57,671 - Phyluce - INFO - Heterodera_glycines_GCA_004148225: 216 uces, 119 dupes, 97 non-dupes, 7 orient drop, 3 length drop, 87 written\n",
      "2020-02-11 19:27:57,672 - Phyluce - INFO - - Working on Heterorhabditis_bacteriophora_GCA_000223415 genome -\n",
      "2020-02-11 19:27:57,685 - Phyluce - INFO - Reading Heterorhabditis_bacteriophora_GCA_000223415 genome\n",
      "2020-02-11 19:28:00,350 - Phyluce - INFO - Heterorhabditis_bacteriophora_GCA_000223415: 220 uces, 105 dupes, 115 non-dupes, 0 orient drop, 2 length drop, 113 written\n",
      "2020-02-11 19:28:00,350 - Phyluce - INFO - ------ Working on Meloidogyne_arenaria_GCA_003133805 genome -----\n",
      "2020-02-11 19:28:00,351 - Phyluce - INFO - Reading Meloidogyne_arenaria_GCA_003133805 genome\n",
      "2020-02-11 19:28:08,123 - Phyluce - INFO - Meloidogyne_arenaria_GCA_003133805: 202 uces, 185 dupes, 17 non-dupes, 1 orient drop, 0 length drop, 16 written\n",
      "2020-02-11 19:28:08,123 - Phyluce - INFO - ---- Working on Meloidogyne_enterolobii_GCA_003693675 genome ----\n",
      "2020-02-11 19:28:08,133 - Phyluce - INFO - Reading Meloidogyne_enterolobii_GCA_003693675 genome\n",
      "2020-02-11 19:28:10,836 - Phyluce - INFO - Meloidogyne_enterolobii_GCA_003693675: 190 uces, 141 dupes, 49 non-dupes, 1 orient drop, 0 length drop, 48 written\n",
      "2020-02-11 19:28:10,836 - Phyluce - INFO - ---- Working on Meloidogyne_floridensis_GCA_003693605 genome ----\n",
      "2020-02-11 19:28:10,837 - Phyluce - INFO - Reading Meloidogyne_floridensis_GCA_003693605 genome\n",
      "2020-02-11 19:28:12,002 - Phyluce - INFO - Meloidogyne_floridensis_GCA_003693605: 169 uces, 81 dupes, 88 non-dupes, 1 orient drop, 2 length drop, 85 written\n",
      "2020-02-11 19:28:12,003 - Phyluce - INFO - ---- Working on Meloidogyne_graminicola_GCA_002778205 genome ----\n",
      "2020-02-11 19:28:12,012 - Phyluce - INFO - Reading Meloidogyne_graminicola_GCA_002778205 genome\n",
      "2020-02-11 19:28:12,943 - Phyluce - INFO - Meloidogyne_graminicola_GCA_002778205: 163 uces, 53 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-11 19:28:12,943 - Phyluce - INFO - ------- Working on Meloidogyne_hapla_GCA_000172435 genome -------\n",
      "2020-02-11 19:28:12,960 - Phyluce - INFO - Reading Meloidogyne_hapla_GCA_000172435 genome\n",
      "2020-02-11 19:28:14,333 - Phyluce - INFO - Meloidogyne_hapla_GCA_000172435: 180 uces, 88 dupes, 92 non-dupes, 1 orient drop, 1 length drop, 90 written\n",
      "2020-02-11 19:28:14,333 - Phyluce - INFO - ----- Working on Meloidogyne_incognita_GCA_900182535 genome -----\n",
      "2020-02-11 19:28:14,335 - Phyluce - INFO - Reading Meloidogyne_incognita_GCA_900182535 genome\n",
      "2020-02-11 19:28:18,501 - Phyluce - INFO - Meloidogyne_incognita_GCA_900182535: 197 uces, 184 dupes, 13 non-dupes, 0 orient drop, 0 length drop, 13 written\n",
      "2020-02-11 19:28:18,501 - Phyluce - INFO - ------ Working on Meloidogyne_javanica_GCA_003693625 genome -----\n",
      "2020-02-11 19:28:18,506 - Phyluce - INFO - Reading Meloidogyne_javanica_GCA_003693625 genome\n",
      "2020-02-11 19:28:21,412 - Phyluce - INFO - Meloidogyne_javanica_GCA_003693625: 189 uces, 167 dupes, 22 non-dupes, 0 orient drop, 0 length drop, 22 written\n",
      "2020-02-11 19:28:21,413 - Phyluce - INFO - -------- Working on Meloidogyne_luci_GCA_902706615 genome -------\n",
      "2020-02-11 19:28:21,440 - Phyluce - INFO - Reading Meloidogyne_luci_GCA_902706615 genome\n",
      "2020-02-11 19:28:25,997 - Phyluce - INFO - Meloidogyne_luci_GCA_902706615: 190 uces, 170 dupes, 20 non-dupes, 0 orient drop, 0 length drop, 20 written\n",
      "2020-02-11 19:28:25,998 - Phyluce - INFO - ----- Working on Panagrellus_redivivus_GCA_000341325 genome -----\n",
      "2020-02-11 19:28:26,012 - Phyluce - INFO - Reading Panagrellus_redivivus_GCA_000341325 genome\n",
      "2020-02-11 19:28:28,318 - Phyluce - INFO - Panagrellus_redivivus_GCA_000341325: 289 uces, 145 dupes, 144 non-dupes, 0 orient drop, 1 length drop, 143 written\n",
      "2020-02-11 19:28:28,318 - Phyluce - INFO - ------ Working on Panagrolaimus_davidi_GCA_901779475 genome -----\n",
      "2020-02-11 19:28:28,328 - Phyluce - INFO - Reading Panagrolaimus_davidi_GCA_901779475 genome\n",
      "2020-02-11 19:28:30,327 - Phyluce - INFO - Panagrolaimus_davidi_GCA_901779475: 139 uces, 72 dupes, 67 non-dupes, 1 orient drop, 0 length drop, 66 written\n",
      "2020-02-11 19:28:30,327 - Phyluce - INFO - -------- Working on Panagrolaimus_sp_GCA_901766855 genome -------\n",
      "2020-02-11 19:28:30,330 - Phyluce - INFO - Reading Panagrolaimus_sp_GCA_901766855 genome\n",
      "2020-02-11 19:28:32,079 - Phyluce - INFO - Panagrolaimus_sp_GCA_901766855: 218 uces, 90 dupes, 128 non-dupes, 1 orient drop, 1 length drop, 126 written\n",
      "2020-02-11 19:28:32,079 - Phyluce - INFO - ----- Working on Panagrolaimus_superbus_GCA_901766145 genome ----\n",
      "2020-02-11 19:28:32,089 - Phyluce - INFO - Reading Panagrolaimus_superbus_GCA_901766145 genome\n",
      "2020-02-11 19:28:34,159 - Phyluce - INFO - Panagrolaimus_superbus_GCA_901766145: 219 uces, 122 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-11 19:28:34,159 - Phyluce - INFO - ------- Working on Radopholus_similis_GCA_004764675 genome ------\n",
      "2020-02-11 19:28:34,160 - Phyluce - INFO - Reading Radopholus_similis_GCA_004764675 genome\n",
      "2020-02-11 19:28:35,773 - Phyluce - INFO - Radopholus_similis_GCA_004764675: 240 uces, 111 dupes, 129 non-dupes, 0 orient drop, 0 length drop, 129 written\n",
      "2020-02-11 19:28:35,774 - Phyluce - INFO - ---- Working on Rotylenchulus_reniformis_GCA_001026735 genome ---\n",
      "2020-02-11 19:28:35,774 - Phyluce - INFO - Reading Rotylenchulus_reniformis_GCA_001026735 genome\n",
      "2020-02-11 19:28:42,310 - Phyluce - INFO - Rotylenchulus_reniformis_GCA_001026735: 240 uces, 190 dupes, 50 non-dupes, 0 orient drop, 2 length drop, 48 written\n",
      "2020-02-11 19:28:42,310 - Phyluce - INFO - ------ Working on Steinernema_feltiae_GCA_007213375 genome ------\n",
      "2020-02-11 19:28:42,311 - Phyluce - INFO - Reading Steinernema_feltiae_GCA_007213375 genome\n",
      "2020-02-11 19:28:48,210 - Phyluce - INFO - Steinernema_feltiae_GCA_007213375: 521 uces, 217 dupes, 304 non-dupes, 3 orient drop, 8 length drop, 293 written\n",
      "2020-02-11 19:28:48,210 - Phyluce - INFO - ------ Working on Steinernema_glaseri_GCA_000757755 genome ------\n",
      "2020-02-11 19:28:48,214 - Phyluce - INFO - Reading Steinernema_glaseri_GCA_000757755 genome\n",
      "2020-02-11 19:28:51,393 - Phyluce - INFO - Steinernema_glaseri_GCA_000757755: 478 uces, 160 dupes, 318 non-dupes, 3 orient drop, 5 length drop, 310 written\n",
      "2020-02-11 19:28:51,393 - Phyluce - INFO - ----- Working on Steinernema_monticolum_GCA_000505645 genome ----\n",
      "2020-02-11 19:28:51,404 - Phyluce - INFO - Reading Steinernema_monticolum_GCA_000505645 genome\n",
      "2020-02-11 19:28:55,746 - Phyluce - INFO - Steinernema_monticolum_GCA_000505645: 509 uces, 180 dupes, 329 non-dupes, 1 orient drop, 2 length drop, 326 written\n",
      "2020-02-11 19:28:55,746 - Phyluce - INFO - ---- Working on Steinernema_scapterisci_GCA_000757745 genome ----\n",
      "2020-02-11 19:28:55,755 - Phyluce - INFO - Reading Steinernema_scapterisci_GCA_000757745 genome\n",
      "2020-02-11 19:29:00,346 - Phyluce - INFO - Steinernema_scapterisci_GCA_000757745: 700 uces, 198 dupes, 502 non-dupes, 3 orient drop, 9 length drop, 490 written\n",
      "2020-02-11 19:29:00,346 - Phyluce - INFO - -------- Working on Subanguina_moxae_GCA_000981365 genome -------\n",
      "2020-02-11 19:29:00,374 - Phyluce - INFO - Reading Subanguina_moxae_GCA_000981365 genome\n",
      "2020-02-11 19:29:03,151 - Phyluce - INFO - Subanguina_moxae_GCA_000981365: 249 uces, 144 dupes, 105 non-dupes, 0 orient drop, 0 length drop, 104 written\n",
      "2020-02-11 19:29:03,152 - Phyluce - INFO - ---- Working on Steinernema_carpocapsae_GCA_000757645 genome ----\n",
      "2020-02-11 19:29:03,153 - Phyluce - INFO - Reading Steinernema_carpocapsae_GCA_000757645 genome\n",
      "2020-02-11 19:29:12,000 - Phyluce - INFO - Steinernema_carpocapsae_GCA_000757645: 1228 uces, 272 dupes, 956 non-dupes, 51 orient drop, 43 length drop, 862 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/nematoda-clade4b/extract_probes_from_group/probe_fasta --output results/phyluce/nematoda-clade4b/extract_probes_from_group/multifastas.sqlite --base-taxon Steinernema_carpocapsae_GCA_000757645;\n",
      "acrobeloides_nanus_gca_900406225.\n",
      "bursaphelenchus_xylophilus_gca_004128815.\n",
      "deladenus_siricidicola_gca_009724625.\n",
      "ditylenchus_destructor_gca_001579705.\n",
      "ditylenchus_dipsaci_gca_004194705.\n",
      "globodera_ellingtonae_gca_001723225.\n",
      "globodera_pallida_gca_000724045.\n",
      "globodera_rostochiensis_gca_900079975.\n",
      "halicephalobus_mephisto_gca_009193035.\n",
      "halicephalobus_sp_gca_009761265.\n",
      "heterodera_glycines_gca_004148225.\n",
      "heterorhabditis_bacteriophora_gca_000223415.\n",
      "meloidogyne_arenaria_gca_003133805.\n",
      "meloidogyne_enterolobii_gca_003693675.\n",
      "meloidogyne_floridensis_gca_003693605.\n",
      "meloidogyne_graminicola_gca_002778205.\n",
      "meloidogyne_hapla_gca_000172435.\n",
      "meloidogyne_incognita_gca_900182535.\n",
      "meloidogyne_javanica_gca_003693625.\n",
      "meloidogyne_luci_gca_902706615.\n",
      "panagrellus_redivivus_gca_000341325.\n",
      "panagrolaimus_davidi_gca_901779475.\n",
      "panagrolaimus_sp_gca_901766855.\n",
      "panagrolaimus_superbus_gca_901766145.\n",
      "radopholus_similis_gca_004764675.\n",
      "rotylenchulus_reniformis_gca_001026735.\n",
      "steinernema_feltiae_gca_007213375.\n",
      "steinernema_glaseri_gca_000757755.\n",
      "steinernema_monticolum_gca_000505645.\n",
      "steinernema_scapterisci_gca_000757745.\n",
      "subanguina_moxae_gca_000981365.\n",
      "steinernema_carpocapsae_gca_000757645.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade4b/extract_probes_from_group/multifastas.sqlite --base-taxon Steinernema_carpocapsae_GCA_000757645;\n",
      "Loci shared by 0 taxa:\t1,095.0\n",
      "Loci shared by 1 taxa:\t1,095.0\n",
      "Loci shared by 2 taxa:\t641.0\n",
      "Loci shared by 3 taxa:\t484.0\n",
      "Loci shared by 4 taxa:\t408.0\n",
      "Loci shared by 5 taxa:\t349.0\n",
      "Loci shared by 6 taxa:\t233.0\n",
      "Loci shared by 7 taxa:\t213.0\n",
      "Loci shared by 8 taxa:\t196.0\n",
      "Loci shared by 9 taxa:\t173.0\n",
      "Loci shared by 10 taxa:\t159.0\n",
      "Loci shared by 11 taxa:\t143.0\n",
      "Loci shared by 12 taxa:\t132.0\n",
      "Loci shared by 13 taxa:\t114.0\n",
      "Loci shared by 14 taxa:\t103.0\n",
      "Loci shared by 15 taxa:\t93.0\n",
      "Loci shared by 16 taxa:\t77.0\n",
      "Loci shared by 17 taxa:\t59.0\n",
      "Loci shared by 18 taxa:\t44.0\n",
      "Loci shared by 19 taxa:\t36.0\n",
      "Loci shared by 20 taxa:\t30.0\n",
      "Loci shared by 21 taxa:\t24.0\n",
      "Loci shared by 22 taxa:\t17.0\n",
      "Loci shared by 23 taxa:\t8.0\n",
      "Loci shared by 24 taxa:\t1.0\n",
      "Loci shared by 25 taxa:\t1.0\n",
      "Loci shared by 26 taxa:\t0.0\n",
      "Loci shared by 27 taxa:\t0.0\n",
      "Loci shared by 28 taxa:\t0.0\n",
      "Loci shared by 29 taxa:\t0.0\n",
      "Loci shared by 30 taxa:\t0.0\n",
      "Loci shared by 31 taxa:\t0.0\n",
      "Loci shared by 32 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade4b/extract_probes_from_group/multifastas.sqlite  --base-taxon Steinernema_carpocapsae_GCA_000757645 --output results/phyluce/nematoda-clade4b/extract_probes_from_group/Steinernema_carpocapsae_GCA_000757645+2-back-to-17.conf --specific-counts 17;\n",
      "Counter({'halicephalobus_mephisto_gca_009193035': 54, 'halicephalobus_sp_gca_009761265': 53, 'steinernema_monticolum_gca_000505645': 53, 'deladenus_siricidicola_gca_009724625': 52, 'steinernema_feltiae_gca_007213375': 50, 'bursaphelenchus_xylophilus_gca_004128815': 50, 'steinernema_carpocapsae_gca_000757645': 49, 'steinernema_glaseri_gca_000757755': 49, 'panagrellus_redivivus_gca_000341325': 48, 'steinernema_scapterisci_gca_000757745': 48, 'globodera_rostochiensis_gca_900079975': 47, 'globodera_ellingtonae_gca_001723225': 44, 'ditylenchus_destructor_gca_001579705': 44, 'panagrolaimus_sp_gca_901766855': 42, 'heterorhabditis_bacteriophora_gca_000223415': 41, 'radopholus_similis_gca_004764675': 39, 'subanguina_moxae_gca_000981365': 39, 'acrobeloides_nanus_gca_900406225': 38, 'heterodera_glycines_gca_004148225': 37, 'panagrolaimus_superbus_gca_901766145': 35, 'meloidogyne_graminicola_gca_002778205': 34, 'globodera_pallida_gca_000724045': 33, 'ditylenchus_dipsaci_gca_004194705': 32, 'meloidogyne_hapla_gca_000172435': 32, 'meloidogyne_floridensis_gca_003693605': 26, 'panagrolaimus_davidi_gca_901779475': 25, 'meloidogyne_enterolobii_gca_003693675': 18, 'rotylenchulus_reniformis_gca_001026735': 18, 'meloidogyne_luci_gca_902706615': 12, 'meloidogyne_javanica_gca_003693625': 9, 'meloidogyne_arenaria_gca_003133805': 7, 'meloidogyne_incognita_gca_900182535': 6})\n",
      "Total loci = 59\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(17)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/nematoda-clade4b/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/nematoda-clade4b/extract_probes_from_group/Steinernema_carpocapsae_GCA_000757645+2-back-to-17.conf --probe-prefix uce_nematoda-clade4b_ --designer rnplattii --design nematoda-clade4b_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list.fasta;\n",
      "NGGGGGGGGGGGGGNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGNNGGGNNGGGGGGGGGGGGGGGNGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 59\n",
      "Probe Count = 2237\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list.fasta --query results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  19:30:02\n",
      "Ended:  Tue Feb 11, 2020  19:30:05\n",
      "Time for execution:  0.0525559147199 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list.fasta --lastz results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_nematoda-clade4b_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 2236 fasta sequences.  Filtered 0 duplicates. Kept 2237.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/nematoda-clade4b/final_probe_design/nematoda-clade4b_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 19:31:13 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 2237\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 178960\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3961192\n",
      "Max number of word counting entries: 89166441\n",
      "\n",
      "# comparing sequences from          0  to        372\n",
      "---------- new table with      325 representatives\n",
      "# comparing sequences from        372  to        682\n",
      "99.9%---------- new table with      286 representatives\n",
      "# comparing sequences from        682  to        941\n",
      "99.9%---------- new table with      240 representatives\n",
      "# comparing sequences from        941  to       1157\n",
      "----------     16 remaining sequences to the next cycle\n",
      "---------- new table with      178 representatives\n",
      "# comparing sequences from       1141  to       1323\n",
      "99.8%---------- new table with      164 representatives\n",
      "# comparing sequences from       1323  to       2237\n",
      ".....................---------- new table with      811 representatives\n",
      "\n",
      "     2237  finished       2004  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.62\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clade 5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'nematoda-clade5'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Ancylostoma_caninum_GCA_003336725\",\n",
    "               \"Ancylostoma_ceylanicum_GCA_000688135\",\n",
    "               \"Ancylostoma_duodenale_GCA_000816745\",\n",
    "               \"Angiostrongylus_cantonensis_GCA_009735665\",\n",
    "               \"Angiostrongylus_costaricensis_GCA_900624975\",\n",
    "               \"Caenorhabditis_angaria_GCA_000165025\",\n",
    "               \"Caenorhabditis_becei_GCA_900536315\",\n",
    "               \"Caenorhabditis_bovis_GCA_902636285\",\n",
    "               \"Caenorhabditis_brenneri_GCA_000143925\",\n",
    "               \"Caenorhabditis_briggsae_GCA_000004555\",\n",
    "               \"Caenorhabditis_elegans_GCF_000002985\",\n",
    "               \"Caenorhabditis_inopinata_GCA_003052745\",\n",
    "               \"Caenorhabditis_japonica_GCA_000147155\",\n",
    "               \"Caenorhabditis_latens_GCA_002259235\",\n",
    "               \"Caenorhabditis_nigoni_GCA_002742825\",\n",
    "               \"Caenorhabditis_panamensis_GCA_900536275\",\n",
    "               \"Caenorhabditis_remanei_GCF_000149515\",\n",
    "               \"Caenorhabditis_sp_GCA_900536305\",\n",
    "               \"Caenorhabditis_tropicalis_GCA_000186765\",\n",
    "               \"Cylicostephanus_goldi_GCA_900617965\",\n",
    "               \"Dictyocaulus_viviparus_GCA_000816705\",\n",
    "               \"Diploscapter_coronatus_GCA_002207785\",\n",
    "               \"Diploscapter_pachys_GCA_002287525\",\n",
    "               \"Haemonchus_placei_GCA_900617895\",\n",
    "               \"Heligmosomoides_polygyrus_GCA_900096555\",\n",
    "               \"Heterorhabditis_bacteriophora_GCA_000223415\",\n",
    "               \"Mesorhabditis_belari_GCA_900631915\",\n",
    "               \"Micoletzkya_japonica_GCA_900490955\",\n",
    "               \"Necator_americanus_GCF_000507365\",\n",
    "               \"Nippostrongylus_brasiliensis_GCA_900200055\",\n",
    "               \"Oesophagostomum_dentatum_GCA_000797555\",\n",
    "               \"Oscheius_sp_GCA_000934875\",\n",
    "               \"Oscheius_tipulae_GCA_900184235\",\n",
    "               \"Parapristionchus_giblindavisi_GCA_900491355\",\n",
    "               \"Pristionchus_arcanus_GCA_900490705\",\n",
    "               \"Pristionchus_entomophagus_GCA_900490825\",\n",
    "               \"Pristionchus_exspctatus_GCA_900380275\",\n",
    "               \"Pristionchus_fissidentatus_GCA_900490895\",\n",
    "               \"Pristionchus_japonicus_GCA_900490845\",\n",
    "               \"Pristionchus_maxplancki_GCA_900490775\",\n",
    "               \"Pristionchus_mayeri_GCA_900490875\",\n",
    "               \"Pristionchus_pacificus_GCA_000180635\",\n",
    "               \"Rhabditida_sp_GCA_004026265\",\n",
    "               \"Strongylus_vulgaris_GCA_900624965\",\n",
    "               \"Teladorsagia_circumcincta_GCA_002352805\" ]\n",
    "                    \n",
    "reference_taxon = \"Haemonchus_contortus_GCA_007637855\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003336725.1_A_caninum_9.3.2.ec.cg.pg_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 134450487 bytes  9959298.44 bytes/sec\n",
      "total size is 134417543  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000688135.1_Acey_2013.11.30.genDNA_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 98298572 bytes  8547705.57 bytes/sec\n",
      "total size is 98274454  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000816745.1_A_duodenale_2.2.ec.cg.pg_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 106813771 bytes  9288157.65 bytes/sec\n",
      "total size is 106787579  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009735665.1_ACgenome_NGS_v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 88230720 bytes  8402929.71 bytes/sec\n",
      "total size is 88209073  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900624975.1_A_costaricensis_Costa_Rica_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 85949047 bytes  9047272.53 bytes/sec\n",
      "total size is 85927932  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000165025.1_ps1010rel4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24295732 bytes  6941649.71 bytes/sec\n",
      "total size is 24289690  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900536315.3_CBECE_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 27341369 bytes  7811831.71 bytes/sec\n",
      "total size is 27334585  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902636285.1_CBOVIS_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 20089305 bytes  4464299.33 bytes/sec\n",
      "total size is 20084296  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000143925.2_C_brenneri-6.0.1b_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 54942723 bytes  7325702.00 bytes/sec\n",
      "total size is 54929194  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000004555.3_CB4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 33440158 bytes  7431155.56 bytes/sec\n",
      "total size is 33431891  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000002985.3_WBcel235_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 31714525 bytes  7047681.56 bytes/sec\n",
      "total size is 31706677  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003052745.1_Sp34_v7_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 38129117 bytes  5083887.87 bytes/sec\n",
      "total size is 38119702  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000147155.1_C_japonica-7.0.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 49969620 bytes  6662621.60 bytes/sec\n",
      "total size is 49957308  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002259235.1_CaeLat1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 34145402 bytes  6208262.55 bytes/sec\n",
      "total size is 34136961  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002742825.1_nigoni.pc_2016.07.14_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 39784401 bytes  5304592.40 bytes/sec\n",
      "total size is 39774573  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900536275.1_CPANA_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24860891 bytes  4520169.64 bytes/sec\n",
      "total size is 24854715  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000149515.1_ASM14951v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 43997469 bytes  5176177.76 bytes/sec\n",
      "total size is 43986619  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900536305.3_CTRIB_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 32339992 bytes  3804709.88 bytes/sec\n",
      "total size is 32331992  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000186765.1_Caenorhabditis_sp11_JU1373-3.0.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24684802 bytes  5485520.89 bytes/sec\n",
      "total size is 24678642  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617965.1_C_goldi_Cheshire_0011_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 56453459 bytes  5376523.90 bytes/sec\n",
      "total size is 56439558  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000816705.1_D_viviparus_9.2.1.ec.pg_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 50861713 bytes  5353868.95 bytes/sec\n",
      "total size is 50849178  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002207785.1_ASM220778v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 54918119 bytes  5780859.05 bytes/sec\n",
      "total size is 54904604  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002287525.1_DipSp1Ass11Ann3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 51161194 bytes  6821498.13 bytes/sec\n",
      "total size is 51148595  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007637855.2_ASM763785v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 136058126 bytes  6977341.95 bytes/sec\n",
      "total size is 136024803  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900617895.1_H_placei_MHpl1_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 81656624 bytes  8595438.53 bytes/sec\n",
      "total size is 81636569  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900096555.1_nHp_v2.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 227191567 bytes  7701410.47 bytes/sec\n",
      "total size is 227136007  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000223415.1_Heterorhabditis_bacteriophora-7.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 24086736 bytes  5352617.33 bytes/sec\n",
      "total size is 24080727  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900631915.1_M.Belari_Hybrid_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 51322027 bytes  7895702.92 bytes/sec\n",
      "total size is 51309381  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490955.1_Micoletzkya_japonica_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 58822765 bytes  7843040.93 bytes/sec\n",
      "total size is 58808282  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000507365.1_N_americanus_v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 69689467 bytes  8198765.76 bytes/sec\n",
      "total size is 69672340  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900200055.1_NbL5_MIMR_Canu1.5_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 112749604 bytes  8351825.63 bytes/sec\n",
      "total size is 112721963  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000797555.1_O_dentatum_10.0.ec.cg.pg_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 119672073 bytes  9573769.20 bytes/sec\n",
      "total size is 119642737  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000934875.1_ASM93487v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 12524411 bytes  5009781.20 bytes/sec\n",
      "total size is 12521241  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900184235.1_Oscheius_tipulae_assembly_v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 19005746 bytes  5430225.14 bytes/sec\n",
      "total size is 19000982  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900491355.1_Parapristionchus_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 57680218 bytes  7690701.33 bytes/sec\n",
      "total size is 57666019  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490705.1_P._arcanus_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 62842641 bytes  8379024.40 bytes/sec\n",
      "total size is 62827184  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490825.1_P._entomophagus_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 79728163 bytes  8392442.63 bytes/sec\n",
      "total size is 79708581  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900380275.1_Pristionchus_exspectatus_de_novo_assembly_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 54062191 bytes  8317266.62 bytes/sec\n",
      "total size is 54048854  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490895.1_Pristionchus_fissidentatus_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 76305065 bytes  6635226.70 bytes/sec\n",
      "total size is 76286304  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490845.1_Pristionchus_japonicus_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 64658450 bytes  8621132.27 bytes/sec\n",
      "total size is 64642541  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490775.1_Prisstionchus_maxplancki_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 73361403 bytes  6986804.29 bytes/sec\n",
      "total size is 73343364  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900490875.1_Pristionchus_mayeri_genome_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 88370897 bytes  8416279.90 bytes/sec\n",
      "total size is 88349199  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000180635.3_El_Paco_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 48561295 bytes  7470974.92 bytes/sec\n",
      "total size is 48549336  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004026265.1_ASM402626v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 16438252 bytes  4696655.43 bytes/sec\n",
      "total size is 16434129  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900624965.1_S_vulgaris_Kentucky_0011_upd_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 89298697 bytes  7765107.74 bytes/sec\n",
      "total size is 89276773  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002352805.1_T_circumcincta.14.0.ec.cg.pg_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 196947150 bytes  8753208.53 bytes/sec\n",
      "total size is 196898954  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if os.path.exists(\"data/genomes/\" + group):\n",
    "    shutil.rmtree(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "os.makedirs(\"data/genomes/\" + group + \"/\")\n",
    "\n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 25x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487883 (\"sim_Ancylostoma_caninum_GCA_003336725\") has been submitted\n",
      "Your job 5487884 (\"sim_Ancylostoma_ceylanicum_GCA_000688135\") has been submitted\n",
      "Your job 5487885 (\"sim_Ancylostoma_duodenale_GCA_000816745\") has been submitted\n",
      "Your job 5487886 (\"sim_Angiostrongylus_cantonensis_GCA_009735665\") has been submitted\n",
      "Your job 5487887 (\"sim_Angiostrongylus_costaricensis_GCA_900624975\") has been submitted\n",
      "Your job 5487888 (\"sim_Caenorhabditis_angaria_GCA_000165025\") has been submitted\n",
      "Your job 5487889 (\"sim_Caenorhabditis_becei_GCA_900536315\") has been submitted\n",
      "Your job 5487890 (\"sim_Caenorhabditis_bovis_GCA_902636285\") has been submitted\n",
      "Your job 5487891 (\"sim_Caenorhabditis_brenneri_GCA_000143925\") has been submitted\n",
      "Your job 5487892 (\"sim_Caenorhabditis_briggsae_GCA_000004555\") has been submitted\n",
      "Your job 5487893 (\"sim_Caenorhabditis_elegans_GCF_000002985\") has been submitted\n",
      "Your job 5487894 (\"sim_Caenorhabditis_inopinata_GCA_003052745\") has been submitted\n",
      "Your job 5487895 (\"sim_Caenorhabditis_japonica_GCA_000147155\") has been submitted\n",
      "Your job 5487896 (\"sim_Caenorhabditis_latens_GCA_002259235\") has been submitted\n",
      "Your job 5487897 (\"sim_Caenorhabditis_nigoni_GCA_002742825\") has been submitted\n",
      "Your job 5487898 (\"sim_Caenorhabditis_panamensis_GCA_900536275\") has been submitted\n",
      "Your job 5487899 (\"sim_Caenorhabditis_remanei_GCF_000149515\") has been submitted\n",
      "Your job 5487900 (\"sim_Caenorhabditis_sp_GCA_900536305\") has been submitted\n",
      "Your job 5487901 (\"sim_Caenorhabditis_tropicalis_GCA_000186765\") has been submitted\n",
      "Your job 5487902 (\"sim_Cylicostephanus_goldi_GCA_900617965\") has been submitted\n",
      "Your job 5487903 (\"sim_Dictyocaulus_viviparus_GCA_000816705\") has been submitted\n",
      "Your job 5487904 (\"sim_Diploscapter_coronatus_GCA_002207785\") has been submitted\n",
      "Your job 5487905 (\"sim_Diploscapter_pachys_GCA_002287525\") has been submitted\n",
      "Your job 5487906 (\"sim_Haemonchus_placei_GCA_900617895\") has been submitted\n",
      "Your job 5487907 (\"sim_Heligmosomoides_polygyrus_GCA_900096555\") has been submitted\n",
      "Your job 5487908 (\"sim_Heterorhabditis_bacteriophora_GCA_000223415\") has been submitted\n",
      "Your job 5487909 (\"sim_Mesorhabditis_belari_GCA_900631915\") has been submitted\n",
      "Your job 5487910 (\"sim_Micoletzkya_japonica_GCA_900490955\") has been submitted\n",
      "Your job 5487911 (\"sim_Necator_americanus_GCF_000507365\") has been submitted\n",
      "Your job 5487912 (\"sim_Nippostrongylus_brasiliensis_GCA_900200055\") has been submitted\n",
      "Your job 5487913 (\"sim_Oesophagostomum_dentatum_GCA_000797555\") has been submitted\n",
      "Your job 5487914 (\"sim_Oscheius_sp_GCA_000934875\") has been submitted\n",
      "Your job 5487915 (\"sim_Oscheius_tipulae_GCA_900184235\") has been submitted\n",
      "Your job 5487916 (\"sim_Parapristionchus_giblindavisi_GCA_900491355\") has been submitted\n",
      "Your job 5487917 (\"sim_Pristionchus_arcanus_GCA_900490705\") has been submitted\n",
      "Your job 5487918 (\"sim_Pristionchus_entomophagus_GCA_900490825\") has been submitted\n",
      "Your job 5487919 (\"sim_Pristionchus_exspctatus_GCA_900380275\") has been submitted\n",
      "Your job 5487920 (\"sim_Pristionchus_fissidentatus_GCA_900490895\") has been submitted\n",
      "Your job 5487921 (\"sim_Pristionchus_japonicus_GCA_900490845\") has been submitted\n",
      "Your job 5487922 (\"sim_Pristionchus_maxplancki_GCA_900490775\") has been submitted\n",
      "Your job 5487923 (\"sim_Pristionchus_mayeri_GCA_900490875\") has been submitted\n",
      "Your job 5487924 (\"sim_Pristionchus_pacificus_GCA_000180635\") has been submitted\n",
      "Your job 5487925 (\"sim_Rhabditida_sp_GCA_004026265\") has been submitted\n",
      "Your job 5487926 (\"sim_Strongylus_vulgaris_GCA_900624965\") has been submitted\n",
      "Your job 5487927 (\"sim_Teladorsagia_circumcincta_GCA_002352805\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 25 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/nematoda-clade5/cleaned_genomes/Haemonchus_contortus_GCA_007637855_formatted.fas path=results/phyluce/nematoda-clade5/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487928 (\"map_Ancylostoma_caninum_GCA_003336725\") has been submitted\n",
      "Your job 5487929 (\"map_Ancylostoma_ceylanicum_GCA_000688135\") has been submitted\n",
      "Your job 5487930 (\"map_Ancylostoma_duodenale_GCA_000816745\") has been submitted\n",
      "Your job 5487931 (\"map_Angiostrongylus_cantonensis_GCA_009735665\") has been submitted\n",
      "Your job 5487932 (\"map_Angiostrongylus_costaricensis_GCA_900624975\") has been submitted\n",
      "Your job 5487933 (\"map_Caenorhabditis_angaria_GCA_000165025\") has been submitted\n",
      "Your job 5487934 (\"map_Caenorhabditis_becei_GCA_900536315\") has been submitted\n",
      "Your job 5487935 (\"map_Caenorhabditis_bovis_GCA_902636285\") has been submitted\n",
      "Your job 5487936 (\"map_Caenorhabditis_brenneri_GCA_000143925\") has been submitted\n",
      "Your job 5487937 (\"map_Caenorhabditis_briggsae_GCA_000004555\") has been submitted\n",
      "Your job 5487938 (\"map_Caenorhabditis_elegans_GCF_000002985\") has been submitted\n",
      "Your job 5487939 (\"map_Caenorhabditis_inopinata_GCA_003052745\") has been submitted\n",
      "Your job 5487940 (\"map_Caenorhabditis_japonica_GCA_000147155\") has been submitted\n",
      "Your job 5487941 (\"map_Caenorhabditis_latens_GCA_002259235\") has been submitted\n",
      "Your job 5487942 (\"map_Caenorhabditis_nigoni_GCA_002742825\") has been submitted\n",
      "Your job 5487943 (\"map_Caenorhabditis_panamensis_GCA_900536275\") has been submitted\n",
      "Your job 5487944 (\"map_Caenorhabditis_remanei_GCF_000149515\") has been submitted\n",
      "Your job 5487945 (\"map_Caenorhabditis_sp_GCA_900536305\") has been submitted\n",
      "Your job 5487946 (\"map_Caenorhabditis_tropicalis_GCA_000186765\") has been submitted\n",
      "Your job 5487947 (\"map_Cylicostephanus_goldi_GCA_900617965\") has been submitted\n",
      "Your job 5487948 (\"map_Dictyocaulus_viviparus_GCA_000816705\") has been submitted\n",
      "Your job 5487949 (\"map_Diploscapter_coronatus_GCA_002207785\") has been submitted\n",
      "Your job 5487950 (\"map_Diploscapter_pachys_GCA_002287525\") has been submitted\n",
      "Your job 5487951 (\"map_Haemonchus_placei_GCA_900617895\") has been submitted\n",
      "Your job 5487952 (\"map_Heligmosomoides_polygyrus_GCA_900096555\") has been submitted\n",
      "Your job 5487953 (\"map_Heterorhabditis_bacteriophora_GCA_000223415\") has been submitted\n",
      "Your job 5487954 (\"map_Mesorhabditis_belari_GCA_900631915\") has been submitted\n",
      "Your job 5487955 (\"map_Micoletzkya_japonica_GCA_900490955\") has been submitted\n",
      "Your job 5487956 (\"map_Necator_americanus_GCF_000507365\") has been submitted\n",
      "Your job 5487957 (\"map_Nippostrongylus_brasiliensis_GCA_900200055\") has been submitted\n",
      "Your job 5487958 (\"map_Oesophagostomum_dentatum_GCA_000797555\") has been submitted\n",
      "Your job 5487959 (\"map_Oscheius_sp_GCA_000934875\") has been submitted\n",
      "Your job 5487960 (\"map_Oscheius_tipulae_GCA_900184235\") has been submitted\n",
      "Your job 5487961 (\"map_Parapristionchus_giblindavisi_GCA_900491355\") has been submitted\n",
      "Your job 5487962 (\"map_Pristionchus_arcanus_GCA_900490705\") has been submitted\n",
      "Your job 5487963 (\"map_Pristionchus_entomophagus_GCA_900490825\") has been submitted\n",
      "Your job 5487964 (\"map_Pristionchus_exspctatus_GCA_900380275\") has been submitted\n",
      "Your job 5487965 (\"map_Pristionchus_fissidentatus_GCA_900490895\") has been submitted\n",
      "Your job 5487966 (\"map_Pristionchus_japonicus_GCA_900490845\") has been submitted\n",
      "Your job 5487967 (\"map_Pristionchus_maxplancki_GCA_900490775\") has been submitted\n",
      "Your job 5487968 (\"map_Pristionchus_mayeri_GCA_900490875\") has been submitted\n",
      "Your job 5487969 (\"map_Pristionchus_pacificus_GCA_000180635\") has been submitted\n",
      "Your job 5487970 (\"map_Rhabditida_sp_GCA_004026265\") has been submitted\n",
      "Your job 5487971 (\"map_Strongylus_vulgaris_GCA_900624965\") has been submitted\n",
      "Your job 5487972 (\"map_Teladorsagia_circumcincta_GCA_002352805\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5487973 (\"merge_Ancylostoma_caninum_GCA_003336725\") has been submitted\n",
      "Your job 5487974 (\"merge_Ancylostoma_ceylanicum_GCA_000688135\") has been submitted\n",
      "Your job 5487975 (\"merge_Ancylostoma_duodenale_GCA_000816745\") has been submitted\n",
      "Your job 5487976 (\"merge_Angiostrongylus_cantonensis_GCA_009735665\") has been submitted\n",
      "Your job 5487977 (\"merge_Angiostrongylus_costaricensis_GCA_900624975\") has been submitted\n",
      "Your job 5487978 (\"merge_Caenorhabditis_angaria_GCA_000165025\") has been submitted\n",
      "Your job 5487979 (\"merge_Caenorhabditis_becei_GCA_900536315\") has been submitted\n",
      "Your job 5487980 (\"merge_Caenorhabditis_bovis_GCA_902636285\") has been submitted\n",
      "Your job 5487981 (\"merge_Caenorhabditis_brenneri_GCA_000143925\") has been submitted\n",
      "Your job 5487982 (\"merge_Caenorhabditis_briggsae_GCA_000004555\") has been submitted\n",
      "Your job 5487983 (\"merge_Caenorhabditis_elegans_GCF_000002985\") has been submitted\n",
      "Your job 5487984 (\"merge_Caenorhabditis_inopinata_GCA_003052745\") has been submitted\n",
      "Your job 5487985 (\"merge_Caenorhabditis_japonica_GCA_000147155\") has been submitted\n",
      "Your job 5487986 (\"merge_Caenorhabditis_latens_GCA_002259235\") has been submitted\n",
      "Your job 5487987 (\"merge_Caenorhabditis_nigoni_GCA_002742825\") has been submitted\n",
      "Your job 5487988 (\"merge_Caenorhabditis_panamensis_GCA_900536275\") has been submitted\n",
      "Your job 5487989 (\"merge_Caenorhabditis_remanei_GCF_000149515\") has been submitted\n",
      "Your job 5487990 (\"merge_Caenorhabditis_sp_GCA_900536305\") has been submitted\n",
      "Your job 5487991 (\"merge_Caenorhabditis_tropicalis_GCA_000186765\") has been submitted\n",
      "Your job 5487992 (\"merge_Cylicostephanus_goldi_GCA_900617965\") has been submitted\n",
      "Your job 5487993 (\"merge_Dictyocaulus_viviparus_GCA_000816705\") has been submitted\n",
      "Your job 5487994 (\"merge_Diploscapter_coronatus_GCA_002207785\") has been submitted\n",
      "Your job 5487995 (\"merge_Diploscapter_pachys_GCA_002287525\") has been submitted\n",
      "Your job 5487996 (\"merge_Haemonchus_placei_GCA_900617895\") has been submitted\n",
      "Your job 5487997 (\"merge_Heligmosomoides_polygyrus_GCA_900096555\") has been submitted\n",
      "Your job 5487998 (\"merge_Heterorhabditis_bacteriophora_GCA_000223415\") has been submitted\n",
      "Your job 5487999 (\"merge_Mesorhabditis_belari_GCA_900631915\") has been submitted\n",
      "Your job 5488000 (\"merge_Micoletzkya_japonica_GCA_900490955\") has been submitted\n",
      "Your job 5488001 (\"merge_Necator_americanus_GCF_000507365\") has been submitted\n",
      "Your job 5488002 (\"merge_Nippostrongylus_brasiliensis_GCA_900200055\") has been submitted\n",
      "Your job 5488003 (\"merge_Oesophagostomum_dentatum_GCA_000797555\") has been submitted\n",
      "Your job 5488004 (\"merge_Oscheius_sp_GCA_000934875\") has been submitted\n",
      "Your job 5488005 (\"merge_Oscheius_tipulae_GCA_900184235\") has been submitted\n",
      "Your job 5488006 (\"merge_Parapristionchus_giblindavisi_GCA_900491355\") has been submitted\n",
      "Your job 5488007 (\"merge_Pristionchus_arcanus_GCA_900490705\") has been submitted\n",
      "Your job 5488008 (\"merge_Pristionchus_entomophagus_GCA_900490825\") has been submitted\n",
      "Your job 5488009 (\"merge_Pristionchus_exspctatus_GCA_900380275\") has been submitted\n",
      "Your job 5488010 (\"merge_Pristionchus_fissidentatus_GCA_900490895\") has been submitted\n",
      "Your job 5488011 (\"merge_Pristionchus_japonicus_GCA_900490845\") has been submitted\n",
      "Your job 5488012 (\"merge_Pristionchus_maxplancki_GCA_900490775\") has been submitted\n",
      "Your job 5488013 (\"merge_Pristionchus_mayeri_GCA_900490875\") has been submitted\n",
      "Your job 5488014 (\"merge_Pristionchus_pacificus_GCA_000180635\") has been submitted\n",
      "Your job 5488015 (\"merge_Rhabditida_sp_GCA_004026265\") has been submitted\n",
      "Your job 5488016 (\"merge_Strongylus_vulgaris_GCA_900624965\") has been submitted\n",
      "Your job 5488017 (\"merge_Teladorsagia_circumcincta_GCA_002352805\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 1581 sequences from Ancylostoma_caninum_GCA_003336725_merged.bed.  Filtered 1433 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 148.\n",
      "Screened 1547 sequences from Ancylostoma_ceylanicum_GCA_000688135_merged.bed.  Filtered 1413 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 134.\n",
      "Screened 1478 sequences from Ancylostoma_duodenale_GCA_000816745_merged.bed.  Filtered 1345 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 133.\n",
      "Screened 903 sequences from Angiostrongylus_cantonensis_GCA_009735665_merged.bed.  Filtered 852 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 51.\n",
      "Screened 753 sequences from Angiostrongylus_costaricensis_GCA_900624975_merged.bed.  Filtered 706 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 47.\n",
      "Screened 123 sequences from Caenorhabditis_angaria_GCA_000165025_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 146 sequences from Caenorhabditis_becei_GCA_900536315_merged.bed.  Filtered 134 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 170 sequences from Caenorhabditis_bovis_GCA_902636285_merged.bed.  Filtered 159 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 305 sequences from Caenorhabditis_brenneri_GCA_000143925_merged.bed.  Filtered 294 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 168 sequences from Caenorhabditis_briggsae_GCA_000004555_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 181 sequences from Caenorhabditis_elegans_GCF_000002985_merged.bed.  Filtered 168 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 288 sequences from Caenorhabditis_inopinata_GCA_003052745_merged.bed.  Filtered 278 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 227 sequences from Caenorhabditis_japonica_GCA_000147155_merged.bed.  Filtered 214 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 13.\n",
      "Screened 128 sequences from Caenorhabditis_latens_GCA_002259235_merged.bed.  Filtered 124 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 184 sequences from Caenorhabditis_nigoni_GCA_002742825_merged.bed.  Filtered 172 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 166 sequences from Caenorhabditis_panamensis_GCA_900536275_merged.bed.  Filtered 157 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 192 sequences from Caenorhabditis_remanei_GCF_000149515_merged.bed.  Filtered 180 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 170 sequences from Caenorhabditis_sp_GCA_900536305_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 142 sequences from Caenorhabditis_tropicalis_GCA_000186765_merged.bed.  Filtered 130 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 464 sequences from Cylicostephanus_goldi_GCA_900617965_merged.bed.  Filtered 417 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 47.\n",
      "Screened 430 sequences from Dictyocaulus_viviparus_GCA_000816705_merged.bed.  Filtered 390 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 40.\n",
      "Screened 47 sequences from Diploscapter_coronatus_GCA_002207785_merged.bed.  Filtered 39 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 36 sequences from Diploscapter_pachys_GCA_002287525_merged.bed.  Filtered 33 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 94, in main\n",
      "    for cnt, line in enumerate(open(args.bed, 'rU')):\n",
      "IOError: [Errno 2] No such file or directory: 'results/phyluce/nematoda-clade5/initial_intervals/Haemonchus_placei_GCA_900617895_merged.bed'\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 94, in main\n",
      "    for cnt, line in enumerate(open(args.bed, 'rU')):\n",
      "IOError: [Errno 2] No such file or directory: 'results/phyluce/nematoda-clade5/initial_intervals/Heligmosomoides_polygyrus_GCA_900096555_merged.bed'\n",
      "Screened 172 sequences from Heterorhabditis_bacteriophora_GCA_000223415_merged.bed.  Filtered 164 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 356 sequences from Mesorhabditis_belari_GCA_900631915_merged.bed.  Filtered 351 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 243 sequences from Micoletzkya_japonica_GCA_900490955_merged.bed.  Filtered 234 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 1400 sequences from Necator_americanus_GCF_000507365_merged.bed.  Filtered 1292 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 108.\n",
      "Screened 1864 sequences from Nippostrongylus_brasiliensis_GCA_900200055_merged.bed.  Filtered 1713 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 151.\n",
      "Screened 1005 sequences from Oesophagostomum_dentatum_GCA_000797555_merged.bed.  Filtered 918 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 87.\n",
      "Screened 174 sequences from Oscheius_sp_GCA_000934875_merged.bed.  Filtered 166 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 176 sequences from Oscheius_tipulae_GCA_900184235_merged.bed.  Filtered 161 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15.\n",
      "Screened 256 sequences from Parapristionchus_giblindavisi_GCA_900491355_merged.bed.  Filtered 251 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 168 sequences from Pristionchus_arcanus_GCA_900490705_merged.bed.  Filtered 165 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 192 sequences from Pristionchus_entomophagus_GCA_900490825_merged.bed.  Filtered 181 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 173 sequences from Pristionchus_exspctatus_GCA_900380275_merged.bed.  Filtered 167 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 254 sequences from Pristionchus_fissidentatus_GCA_900490895_merged.bed.  Filtered 242 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 186 sequences from Pristionchus_japonicus_GCA_900490845_merged.bed.  Filtered 175 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 206 sequences from Pristionchus_maxplancki_GCA_900490775_merged.bed.  Filtered 195 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 249 sequences from Pristionchus_mayeri_GCA_900490875_merged.bed.  Filtered 239 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 197 sequences from Pristionchus_pacificus_GCA_000180635_merged.bed.  Filtered 187 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Traceback (most recent call last):\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 122, in <module>\n",
      "    main()\n",
      "  File \"/master/nplatt/anaconda3/envs/pathogen_probes-phyluce/bin/phyluce_probe_strip_masked_loci_from_set\", line 111, in main\n",
      "    cnt + 1,\n",
      "UnboundLocalError: local variable 'cnt' referenced before assignment\n",
      "Screened 944 sequences from Strongylus_vulgaris_GCA_900624965_merged.bed.  Filtered 869 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 20812 sequences from Teladorsagia_circumcincta_GCA_002352805_merged.bed.  Filtered 18230 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2582.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/nematoda-clade5/initial_intervals/Teladorsagia_circumcincta_GCA_002352805_merged.bed --twobit results/phyluce/nematoda-clade5/cleaned_genomes/Haemonchus_contortus_GCA_007637855_formatted.2bit --output results/phyluce/nematoda-clade5/initial_intervals/Teladorsagia_circumcincta_GCA_002352805_stripped.bed;\n",
      "ancylostoma_caninum_gca_003336725.\n",
      "ancylostoma_ceylanicum_gca_000688135.\n",
      "ancylostoma_duodenale_gca_000816745.\n",
      "angiostrongylus_cantonensis_gca_009735665.\n",
      "angiostrongylus_costaricensis_gca_900624975.\n",
      "caenorhabditis_angaria_gca_000165025.\n",
      "caenorhabditis_becei_gca_900536315.\n",
      "caenorhabditis_bovis_gca_902636285.\n",
      "caenorhabditis_brenneri_gca_000143925.\n",
      "caenorhabditis_briggsae_gca_000004555.\n",
      "caenorhabditis_elegans_gcf_000002985.\n",
      "caenorhabditis_inopinata_gca_003052745.\n",
      "caenorhabditis_japonica_gca_000147155.\n",
      "caenorhabditis_latens_gca_002259235.\n",
      "caenorhabditis_nigoni_gca_002742825.\n",
      "caenorhabditis_panamensis_gca_900536275.\n",
      "caenorhabditis_remanei_gcf_000149515.\n",
      "caenorhabditis_sp_gca_900536305.\n",
      "caenorhabditis_tropicalis_gca_000186765.\n",
      "cylicostephanus_goldi_gca_900617965.\n",
      "dictyocaulus_viviparus_gca_000816705.\n",
      "diploscapter_coronatus_gca_002207785.\n",
      "diploscapter_pachys_gca_002287525.\n",
      "haemonchus_placei_gca_900617895\n",
      "heligmosomoides_polygyrus_gca_900096555\n",
      "heterorhabditis_bacteriophora_gca_000223415.\n",
      "mesorhabditis_belari_gca_900631915.\n",
      "micoletzkya_japonica_gca_900490955.\n",
      "necator_americanus_gcf_000507365.\n",
      "nippostrongylus_brasiliensis_gca_900200055.\n",
      "oesophagostomum_dentatum_gca_000797555.\n",
      "oscheius_sp_gca_000934875.\n",
      "oscheius_tipulae_gca_900184235.\n",
      "parapristionchus_giblindavisi_gca_900491355.\n",
      "pristionchus_arcanus_gca_900490705.\n",
      "pristionchus_entomophagus_gca_900490825.\n",
      "pristionchus_exspctatus_gca_900380275.\n",
      "pristionchus_fissidentatus_gca_900490895.\n",
      "pristionchus_japonicus_gca_900490845.\n",
      "pristionchus_maxplancki_gca_900490775.\n",
      "pristionchus_mayeri_gca_900490875.\n",
      "pristionchus_pacificus_gca_000180635.\n",
      "rhabditida_sp_gca_004026265\n",
      "strongylus_vulgaris_gca_900624965.\n",
      "teladorsagia_circumcincta_gca_002352805...\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade5/initial_intervals/nematoda-clade5-to-Haemonchus_contortus_GCA_007637855.sqlite --base-taxon Haemonchus_contortus_GCA_007637855\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 0 taxa:\t2,710.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 1 taxa:\t2,710.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 2 taxa:\t272.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 3 taxa:\t182.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 4 taxa:\t140.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 5 taxa:\t108.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 6 taxa:\t90.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 7 taxa:\t75.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 8 taxa:\t56.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 9 taxa:\t43.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 10 taxa:\t29.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 11 taxa:\t24.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 12 taxa:\t17.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 13 taxa:\t14.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 14 taxa:\t11.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 15 taxa:\t11.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 16 taxa:\t11.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 17 taxa:\t10.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 18 taxa:\t10.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 19 taxa:\t10.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 20 taxa:\t9.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 21 taxa:\t9.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 22 taxa:\t9.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 23 taxa:\t8.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 24 taxa:\t6.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 25 taxa:\t5.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 26 taxa:\t4.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 27 taxa:\t4.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 28 taxa:\t3.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 29 taxa:\t3.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 30 taxa:\t2.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 31 taxa:\t2.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 32 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 33 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 34 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 35 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 36 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 37 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 38 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 39 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 40 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 41 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 42 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 43 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 44 taxa:\t0.0\n",
      "Loci shared by Haemonchus_contortus_GCA_007637855 + 45 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_query_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/nematoda-clade5/initial_intervals/nematoda-clade5-to-Haemonchus_contortus_GCA_007637855.sqlite --base-taxon Haemonchus_contortus_GCA_007637855 --output results/phyluce/nematoda-clade5/initial_intervals/Haemonchus_contortus_GCA_007637855_+1.bed --specific-counts 1;\n",
      "Counter({'teladorsagia_circumcincta_gca_002352805': 2588, 'nippostrongylus_brasiliensis_gca_900200055': 153, 'ancylostoma_caninum_gca_003336725': 148, 'ancylostoma_duodenale_gca_000816745': 134, 'ancylostoma_ceylanicum_gca_000688135': 134, 'necator_americanus_gcf_000507365': 108, 'oesophagostomum_dentatum_gca_000797555': 87, 'strongylus_vulgaris_gca_900624965': 75, 'angiostrongylus_cantonensis_gca_009735665': 51, 'cylicostephanus_goldi_gca_900617965': 47, 'angiostrongylus_costaricensis_gca_900624975': 47, 'dictyocaulus_viviparus_gca_000816705': 40, 'oscheius_tipulae_gca_900184235': 15, 'caenorhabditis_elegans_gcf_000002985': 14, 'caenorhabditis_japonica_gca_000147155': 13, 'caenorhabditis_remanei_gcf_000149515': 12, 'caenorhabditis_nigoni_gca_002742825': 12, 'caenorhabditis_becei_gca_900536315': 12, 'caenorhabditis_bovis_gca_902636285': 12, 'caenorhabditis_tropicalis_gca_000186765': 12, 'caenorhabditis_sp_gca_900536305': 12, 'caenorhabditis_brenneri_gca_000143925': 11, 'pristionchus_maxplancki_gca_900490775': 11, 'pristionchus_fissidentatus_gca_900490895': 11, 'pristionchus_japonicus_gca_900490845': 10, 'pristionchus_entomophagus_gca_900490825': 10, 'pristionchus_pacificus_gca_000180635': 10, 'caenorhabditis_inopinata_gca_003052745': 10, 'caenorhabditis_briggsae_gca_000004555': 10, 'pristionchus_mayeri_gca_900490875': 9, 'caenorhabditis_panamensis_gca_900536275': 8, 'micoletzkya_japonica_gca_900490955': 8, 'oscheius_sp_gca_000934875': 8, 'heterorhabditis_bacteriophora_gca_000223415': 8, 'diploscapter_coronatus_gca_002207785': 8, 'pristionchus_exspctatus_gca_900380275': 6, 'parapristionchus_giblindavisi_gca_900491355': 5, 'mesorhabditis_belari_gca_900631915': 5, 'caenorhabditis_latens_gca_002259235': 4, 'caenorhabditis_angaria_gca_000165025': 3, 'diploscapter_pachys_gca_002287525': 3, 'pristionchus_arcanus_gca_900490705': 3})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 1\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/nematoda-clade5/initial_intervals/Haemonchus_contortus_GCA_007637855_+1.bed --twobit results/phyluce/nematoda-clade5/cleaned_genomes/Haemonchus_contortus_GCA_007637855_formatted.2bit --buffer-to 160 --output results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1.fasta;\n",
      "Screened 2710 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 2710.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1.fasta --probe-prefix uce_nematoda-clade5_ --design nematoda-clade5_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 2707\n",
      "Probe Count = 5397\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas --query results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes_vself.lastz;\n",
      "Started:  Tue Feb 11, 2020  15:53:13\n",
      "Ended:  Tue Feb 11, 2020  15:53:15\n",
      "Time for execution:  0.0468909343084 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas                        --lastz results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes_vself.lastz                       --probe-prefix=uce_nematoda-clade5_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 5396 fasta sequences.  Filtered 336 duplicates. Kept 4725.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/nematoda-clade5/validate_intervals/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas --scaffoldlist Ancylostoma_caninum_GCA_003336725 Ancylostoma_ceylanicum_GCA_000688135 Ancylostoma_duodenale_GCA_000816745 Angiostrongylus_cantonensis_GCA_009735665 Angiostrongylus_costaricensis_GCA_900624975 Caenorhabditis_angaria_GCA_000165025 Caenorhabditis_becei_GCA_900536315 Caenorhabditis_bovis_GCA_902636285 Caenorhabditis_brenneri_GCA_000143925 Caenorhabditis_briggsae_GCA_000004555 Caenorhabditis_elegans_GCF_000002985 Caenorhabditis_inopinata_GCA_003052745 Caenorhabditis_japonica_GCA_000147155 Caenorhabditis_latens_GCA_002259235 Caenorhabditis_nigoni_GCA_002742825 Caenorhabditis_panamensis_GCA_900536275 Caenorhabditis_remanei_GCF_000149515 Caenorhabditis_sp_GCA_900536305 Caenorhabditis_tropicalis_GCA_000186765 Cylicostephanus_goldi_GCA_900617965 Dictyocaulus_viviparus_GCA_000816705 Diploscapter_coronatus_GCA_002207785 Diploscapter_pachys_GCA_002287525 Haemonchus_placei_GCA_900617895 Heligmosomoides_polygyrus_GCA_900096555 Heterorhabditis_bacteriophora_GCA_000223415 Mesorhabditis_belari_GCA_900631915 Micoletzkya_japonica_GCA_900490955 Necator_americanus_GCF_000507365 Nippostrongylus_brasiliensis_GCA_900200055 Oesophagostomum_dentatum_GCA_000797555 Oscheius_sp_GCA_000934875 Oscheius_tipulae_GCA_900184235 Parapristionchus_giblindavisi_GCA_900491355 Pristionchus_arcanus_GCA_900490705 Pristionchus_entomophagus_GCA_900490825 Pristionchus_exspctatus_GCA_900380275 Pristionchus_fissidentatus_GCA_900490895 Pristionchus_japonicus_GCA_900490845 Pristionchus_maxplancki_GCA_900490775 Pristionchus_mayeri_GCA_900490875 Pristionchus_pacificus_GCA_000180635 Rhabditida_sp_GCA_004026265 Strongylus_vulgaris_GCA_900624965 Teladorsagia_circumcincta_GCA_002352805 Haemonchus_contortus_GCA_007637855 --genome-base-path results/phyluce/nematoda-clade5/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/nematoda-clade5/validate_intervals/nematoda-clade5-to-Haemonchus_contortus_GCA_007637855.sqlite --output results/phyluce/nematoda-clade5/validate_intervals/lastz/;\n",
      "\n",
      "Running against Ancylostoma_caninum_GCA_003336725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 46 queries...\n",
      "\t/tmp/tmpUKgVa_.fasta\n",
      "\t/tmp/tmpxMmK4_.fasta\n",
      "\t/tmp/tmpyH3XaD.fasta\n",
      "\t/tmp/tmp0Po7h8.fasta\n",
      "\t/tmp/tmppDGFFj.fasta\n",
      "\t/tmp/tmpCKkaz8.fasta\n",
      "\t/tmp/tmpqZ2MuV.fasta\n",
      "\t/tmp/tmpNLt0D9.fasta\n",
      "\t/tmp/tmpKeufIp.fasta\n",
      "\t/tmp/tmpctUZlB.fasta\n",
      "\t/tmp/tmpLMDCQ0.fasta\n",
      "\t/tmp/tmpAgeagl.fasta\n",
      "\t/tmp/tmpXtL1Wp.fasta\n",
      "\t/tmp/tmpYSwQEG.fasta\n",
      "\t/tmp/tmpERDRP7.fasta\n",
      "\t/tmp/tmp0Ln_w5.fasta\n",
      "\t/tmp/tmpvjgS7A.fasta\n",
      "\t/tmp/tmpTsyHMA.fasta\n",
      "\t/tmp/tmpBFUswa.fasta\n",
      "\t/tmp/tmp_zrUCD.fasta\n",
      "\t/tmp/tmp1GUmxo.fasta\n",
      "\t/tmp/tmpdIMYnD.fasta\n",
      "\t/tmp/tmp8IJeSR.fasta\n",
      "\t/tmp/tmp5md91Q.fasta\n",
      "\t/tmp/tmp_rZxNo.fasta\n",
      "\t/tmp/tmpEyUPSM.fasta\n",
      "\t/tmp/tmpBw1TFX.fasta\n",
      "\t/tmp/tmp6fZFGy.fasta\n",
      "\t/tmp/tmpjuKjKc.fasta\n",
      "\t/tmp/tmp1p04Ki.fasta\n",
      "\t/tmp/tmpp5xXtx.fasta\n",
      "\t/tmp/tmpJkJskI.fasta\n",
      "\t/tmp/tmpe6dtIr.fasta\n",
      "\t/tmp/tmpnUjNpj.fasta\n",
      "\t/tmp/tmpm4JmXH.fasta\n",
      "\t/tmp/tmp79l146.fasta\n",
      "\t/tmp/tmpBAYoj9.fasta\n",
      "\t/tmp/tmpZ8ql2B.fasta\n",
      "\t/tmp/tmpBtm9rB.fasta\n",
      "\t/tmp/tmpQTHIjY.fasta\n",
      "\t/tmp/tmpnU6hMm.fasta\n",
      "\t/tmp/tmpMgUz1s.fasta\n",
      "\t/tmp/tmpivYhf4.fasta\n",
      "\t/tmp/tmpH_gBkn.fasta\n",
      "\t/tmp/tmplO_zQL.fasta\n",
      "\t/tmp/tmpwY_6ZG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdVcHfm.lastz\n",
      "\t/tmp/tmpO_NPzq.lastz\n",
      "\t/tmp/tmpcRnDnA.lastz\n",
      "\t/tmp/tmpBs1OAI.lastz\n",
      "\t/tmp/tmpGuvCQs.lastz\n",
      "\t/tmp/tmpE3DveH.lastz\n",
      "\t/tmp/tmpJy9EIg.lastz\n",
      "\t/tmp/tmpvWt0Rc.lastz\n",
      "\t/tmp/tmpBcLKrg.lastz\n",
      "\t/tmp/tmp15irfn.lastz\n",
      "\t/tmp/tmpYPcUli.lastz\n",
      "\t/tmp/tmp8qecDc.lastz\n",
      "\t/tmp/tmpRJxEuZ.lastz\n",
      "\t/tmp/tmpT_mgVb.lastz\n",
      "\t/tmp/tmpbP4_ul.lastz\n",
      "\t/tmp/tmpdtdy4E.lastz\n",
      "\t/tmp/tmpcXcr9_.lastz\n",
      "\t/tmp/tmpQZt1VS.lastz\n",
      "\t/tmp/tmpCN_44j.lastz\n",
      "\t/tmp/tmpZvmejg.lastz\n",
      "\t/tmp/tmpsLX268.lastz\n",
      "\t/tmp/tmpqW2xQh.lastz\n",
      "\t/tmp/tmpmzYIau.lastz\n",
      "\t/tmp/tmpW1sHA_.lastz\n",
      "\t/tmp/tmpnMLJKn.lastz\n",
      "\t/tmp/tmpepbIpg.lastz\n",
      "\t/tmp/tmpORn3F0.lastz\n",
      "\t/tmp/tmpiU0wsT.lastz\n",
      "\t/tmp/tmpsLLgiF.lastz\n",
      "\t/tmp/tmpByiiNv.lastz\n",
      "\t/tmp/tmpSsIF0C.lastz\n",
      "\t/tmp/tmpaHTots.lastz\n",
      "\t/tmp/tmpiuNNvN.lastz\n",
      "\t/tmp/tmp7O6MzH.lastz\n",
      "\t/tmp/tmp8EC62o.lastz\n",
      "\t/tmp/tmpGO1tfd.lastz\n",
      "\t/tmp/tmpnHAn91.lastz\n",
      "\t/tmp/tmpIWsekt.lastz\n",
      "\t/tmp/tmpFg_ZVJ.lastz\n",
      "\t/tmp/tmpA8cF5c.lastz\n",
      "\t/tmp/tmptj1JAm.lastz\n",
      "\t/tmp/tmpY508i1.lastz\n",
      "\t/tmp/tmpPsjbIH.lastz\n",
      "\t/tmp/tmpCc1FIv.lastz\n",
      "\t/tmp/tmp1CCsX8.lastz\n",
      "\t/tmp/tmp6wjnHg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Ancylostoma_caninum_GCA_003336725.lastz\n",
      "Creating Ancylostoma_caninum_GCA_003336725 table\n",
      "Inserting data to Ancylostoma_caninum_GCA_003336725 table\n",
      "\n",
      "Running against Ancylostoma_ceylanicum_GCA_000688135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 30 queries...\n",
      "\t/tmp/tmp1xVLCC.fasta\n",
      "\t/tmp/tmpVYhHzp.fasta\n",
      "\t/tmp/tmpBxIYnl.fasta\n",
      "\t/tmp/tmp2D9H34.fasta\n",
      "\t/tmp/tmpo76Ope.fasta\n",
      "\t/tmp/tmpuUNF6Q.fasta\n",
      "\t/tmp/tmpcefGri.fasta\n",
      "\t/tmp/tmpdI5nSv.fasta\n",
      "\t/tmp/tmpXdKfA5.fasta\n",
      "\t/tmp/tmpXPJlb8.fasta\n",
      "\t/tmp/tmpCZUL0Y.fasta\n",
      "\t/tmp/tmpfiqdTV.fasta\n",
      "\t/tmp/tmpVL2fWO.fasta\n",
      "\t/tmp/tmp4N0aPt.fasta\n",
      "\t/tmp/tmp2YKVs_.fasta\n",
      "\t/tmp/tmpXFpZhg.fasta\n",
      "\t/tmp/tmpcVCjo5.fasta\n",
      "\t/tmp/tmpIvrLW9.fasta\n",
      "\t/tmp/tmpDNhZcR.fasta\n",
      "\t/tmp/tmpwQB0u9.fasta\n",
      "\t/tmp/tmpmEmcMh.fasta\n",
      "\t/tmp/tmp43MKXE.fasta\n",
      "\t/tmp/tmpapYbWp.fasta\n",
      "\t/tmp/tmpDJqaep.fasta\n",
      "\t/tmp/tmpfvTopO.fasta\n",
      "\t/tmp/tmpZMf0a3.fasta\n",
      "\t/tmp/tmpOyKaZJ.fasta\n",
      "\t/tmp/tmpm1pbji.fasta\n",
      "\t/tmp/tmp5pU6pj.fasta\n",
      "\t/tmp/tmpXVnuis.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXB6DAV.lastz\n",
      "\t/tmp/tmpw0O5NN.lastz\n",
      "\t/tmp/tmpi0sflI.lastz\n",
      "\t/tmp/tmpLziEXX.lastz\n",
      "\t/tmp/tmp3OYA2J.lastz\n",
      "\t/tmp/tmp0bUKte.lastz\n",
      "\t/tmp/tmpQrrLuP.lastz\n",
      "\t/tmp/tmpbINmyQ.lastz\n",
      "\t/tmp/tmpD6KNDI.lastz\n",
      "\t/tmp/tmpmnBMt3.lastz\n",
      "\t/tmp/tmp8nRGTk.lastz\n",
      "\t/tmp/tmpB1Zlvt.lastz\n",
      "\t/tmp/tmpPe3ahf.lastz\n",
      "\t/tmp/tmpiiQPy6.lastz\n",
      "\t/tmp/tmpAXrac4.lastz\n",
      "\t/tmp/tmpI0OsAC.lastz\n",
      "\t/tmp/tmp8ywDpF.lastz\n",
      "\t/tmp/tmpd4b6Ie.lastz\n",
      "\t/tmp/tmpTd3x9P.lastz\n",
      "\t/tmp/tmpofaMeO.lastz\n",
      "\t/tmp/tmpKAl5Np.lastz\n",
      "\t/tmp/tmpfqC8_k.lastz\n",
      "\t/tmp/tmpiiypYv.lastz\n",
      "\t/tmp/tmpUnUqtk.lastz\n",
      "\t/tmp/tmpNS2HFH.lastz\n",
      "\t/tmp/tmpbaVYFZ.lastz\n",
      "\t/tmp/tmpPpT7Ur.lastz\n",
      "\t/tmp/tmpotIWfv.lastz\n",
      "\t/tmp/tmpjKceEb.lastz\n",
      "\t/tmp/tmpXqJw5B.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Ancylostoma_ceylanicum_GCA_000688135.lastz\n",
      "Creating Ancylostoma_ceylanicum_GCA_000688135 table\n",
      "Inserting data to Ancylostoma_ceylanicum_GCA_000688135 table\n",
      "\n",
      "Running against Ancylostoma_duodenale_GCA_000816745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 34 queries...\n",
      "\t/tmp/tmpMfO5wx.fasta\n",
      "\t/tmp/tmp67mhE8.fasta\n",
      "\t/tmp/tmpyvYwqu.fasta\n",
      "\t/tmp/tmpkMdUm2.fasta\n",
      "\t/tmp/tmp5VoXPW.fasta\n",
      "\t/tmp/tmpKKWQKn.fasta\n",
      "\t/tmp/tmpL6FLVb.fasta\n",
      "\t/tmp/tmp69_9bB.fasta\n",
      "\t/tmp/tmpaNZ4IQ.fasta\n",
      "\t/tmp/tmp9EUZSP.fasta\n",
      "\t/tmp/tmpCDYk4R.fasta\n",
      "\t/tmp/tmpGqqHNx.fasta\n",
      "\t/tmp/tmpcPaTWy.fasta\n",
      "\t/tmp/tmpoMFiaY.fasta\n",
      "\t/tmp/tmp1aIgaM.fasta\n",
      "\t/tmp/tmpHjZPPB.fasta\n",
      "\t/tmp/tmpPjuKkv.fasta\n",
      "\t/tmp/tmpdAlBbt.fasta\n",
      "\t/tmp/tmp1x_EJM.fasta\n",
      "\t/tmp/tmp85Og6Z.fasta\n",
      "\t/tmp/tmpLWrEAF.fasta\n",
      "\t/tmp/tmpzgyarU.fasta\n",
      "\t/tmp/tmpjiOa1Q.fasta\n",
      "\t/tmp/tmpjhGT9E.fasta\n",
      "\t/tmp/tmpvM6sdl.fasta\n",
      "\t/tmp/tmpy7oNVb.fasta\n",
      "\t/tmp/tmpA0sZLm.fasta\n",
      "\t/tmp/tmpl0i1V3.fasta\n",
      "\t/tmp/tmpisX4MR.fasta\n",
      "\t/tmp/tmpIi__wd.fasta\n",
      "\t/tmp/tmppYUMHq.fasta\n",
      "\t/tmp/tmppWdrwb.fasta\n",
      "\t/tmp/tmpCMpEcy.fasta\n",
      "\t/tmp/tmphSjyzH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuTmntu.lastz\n",
      "\t/tmp/tmp2csbRL.lastz\n",
      "\t/tmp/tmpBtWona.lastz\n",
      "\t/tmp/tmp8GdjNE.lastz\n",
      "\t/tmp/tmpXbxHJw.lastz\n",
      "\t/tmp/tmpLOkCxz.lastz\n",
      "\t/tmp/tmp0WJQ0K.lastz\n",
      "\t/tmp/tmpsrLjWr.lastz\n",
      "\t/tmp/tmpDLoqbU.lastz\n",
      "\t/tmp/tmpefvivB.lastz\n",
      "\t/tmp/tmp8rw7hi.lastz\n",
      "\t/tmp/tmpKEsTME.lastz\n",
      "\t/tmp/tmprA7PZl.lastz\n",
      "\t/tmp/tmpk8cy46.lastz\n",
      "\t/tmp/tmpbbUXiC.lastz\n",
      "\t/tmp/tmp3zPyLe.lastz\n",
      "\t/tmp/tmpJia161.lastz\n",
      "\t/tmp/tmpixXPW9.lastz\n",
      "\t/tmp/tmpeG3Rug.lastz\n",
      "\t/tmp/tmpClDJKz.lastz\n",
      "\t/tmp/tmpyHSmiF.lastz\n",
      "\t/tmp/tmp9aRxqy.lastz\n",
      "\t/tmp/tmp30WcDJ.lastz\n",
      "\t/tmp/tmpwMHbsn.lastz\n",
      "\t/tmp/tmpLBDEUH.lastz\n",
      "\t/tmp/tmpSggJO6.lastz\n",
      "\t/tmp/tmphE25iq.lastz\n",
      "\t/tmp/tmp9Moj8v.lastz\n",
      "\t/tmp/tmpMCJdre.lastz\n",
      "\t/tmp/tmpcyIyKL.lastz\n",
      "\t/tmp/tmp6HJEBD.lastz\n",
      "\t/tmp/tmpY4JTZa.lastz\n",
      "\t/tmp/tmp2Rutj1.lastz\n",
      "\t/tmp/tmpciYUZQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Ancylostoma_duodenale_GCA_000816745.lastz\n",
      "Creating Ancylostoma_duodenale_GCA_000816745 table\n",
      "Inserting data to Ancylostoma_duodenale_GCA_000816745 table\n",
      "\n",
      "Running against Angiostrongylus_cantonensis_GCA_009735665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 28 queries...\n",
      "\t/tmp/tmpQ9rkHM.fasta\n",
      "\t/tmp/tmp2iKDbU.fasta\n",
      "\t/tmp/tmpxOsEoh.fasta\n",
      "\t/tmp/tmpliQK6L.fasta\n",
      "\t/tmp/tmpqMI_xV.fasta\n",
      "\t/tmp/tmpZtE63W.fasta\n",
      "\t/tmp/tmphhlm5F.fasta\n",
      "\t/tmp/tmpVyrTNj.fasta\n",
      "\t/tmp/tmpAZOlxf.fasta\n",
      "\t/tmp/tmprma_Qo.fasta\n",
      "\t/tmp/tmpt1gGAv.fasta\n",
      "\t/tmp/tmpDeNg59.fasta\n",
      "\t/tmp/tmpAaAHEF.fasta\n",
      "\t/tmp/tmpABxeCI.fasta\n",
      "\t/tmp/tmpDpleHV.fasta\n",
      "\t/tmp/tmplMA6Lc.fasta\n",
      "\t/tmp/tmpRereN3.fasta\n",
      "\t/tmp/tmpY8Dqyz.fasta\n",
      "\t/tmp/tmpf714DA.fasta\n",
      "\t/tmp/tmpx7ut0O.fasta\n",
      "\t/tmp/tmphS00Ok.fasta\n",
      "\t/tmp/tmpjzVFAY.fasta\n",
      "\t/tmp/tmpqR0X62.fasta\n",
      "\t/tmp/tmp4yMxRn.fasta\n",
      "\t/tmp/tmpPCRZAi.fasta\n",
      "\t/tmp/tmpde4LMT.fasta\n",
      "\t/tmp/tmpy4xUWN.fasta\n",
      "\t/tmp/tmphRrYqZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptGU5oz.lastz\n",
      "\t/tmp/tmpV0f6DK.lastz\n",
      "\t/tmp/tmpFHDoSA.lastz\n",
      "\t/tmp/tmpouUqU8.lastz\n",
      "\t/tmp/tmpr8VZ9T.lastz\n",
      "\t/tmp/tmpJe1Ao6.lastz\n",
      "\t/tmp/tmpW3ahN6.lastz\n",
      "\t/tmp/tmpufOTrP.lastz\n",
      "\t/tmp/tmpFUbrPG.lastz\n",
      "\t/tmp/tmp5OauYP.lastz\n",
      "\t/tmp/tmpvAwy2_.lastz\n",
      "\t/tmp/tmpRawW5M.lastz\n",
      "\t/tmp/tmpkZChHE.lastz\n",
      "\t/tmp/tmp2uSHXO.lastz\n",
      "\t/tmp/tmpfRDWPp.lastz\n",
      "\t/tmp/tmpL9ueL_.lastz\n",
      "\t/tmp/tmpK6CxpQ.lastz\n",
      "\t/tmp/tmpbTspsA.lastz\n",
      "\t/tmp/tmpge4dQ7.lastz\n",
      "\t/tmp/tmpznJrlp.lastz\n",
      "\t/tmp/tmpbnMncc.lastz\n",
      "\t/tmp/tmpBdZxis.lastz\n",
      "\t/tmp/tmp3OGLiz.lastz\n",
      "\t/tmp/tmpRP0dTj.lastz\n",
      "\t/tmp/tmpfXsYY4.lastz\n",
      "\t/tmp/tmpT0v09L.lastz\n",
      "\t/tmp/tmpKkBEcK.lastz\n",
      "\t/tmp/tmpsfqGkE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Angiostrongylus_cantonensis_GCA_009735665.lastz\n",
      "Creating Angiostrongylus_cantonensis_GCA_009735665 table\n",
      "Inserting data to Angiostrongylus_cantonensis_GCA_009735665 table\n",
      "\n",
      "Running against Angiostrongylus_costaricensis_GCA_900624975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 27 queries...\n",
      "\t/tmp/tmp8gurfR.fasta\n",
      "\t/tmp/tmpkwZf65.fasta\n",
      "\t/tmp/tmpnKIfy3.fasta\n",
      "\t/tmp/tmppAPPFI.fasta\n",
      "\t/tmp/tmpxYnNZd.fasta\n",
      "\t/tmp/tmpinr8Kc.fasta\n",
      "\t/tmp/tmp2H2aKS.fasta\n",
      "\t/tmp/tmpGRWV15.fasta\n",
      "\t/tmp/tmpC5n6GO.fasta\n",
      "\t/tmp/tmpkEo0jM.fasta\n",
      "\t/tmp/tmpxaRkuG.fasta\n",
      "\t/tmp/tmp0VSmaM.fasta\n",
      "\t/tmp/tmp7kws6f.fasta\n",
      "\t/tmp/tmpjs2D22.fasta\n",
      "\t/tmp/tmpA_Q938.fasta\n",
      "\t/tmp/tmpQlxtMf.fasta\n",
      "\t/tmp/tmpuE0C9f.fasta\n",
      "\t/tmp/tmphFxqY0.fasta\n",
      "\t/tmp/tmpaex7K6.fasta\n",
      "\t/tmp/tmpxHRLSH.fasta\n",
      "\t/tmp/tmpW6aLLu.fasta\n",
      "\t/tmp/tmpU54QVr.fasta\n",
      "\t/tmp/tmpzDTV8H.fasta\n",
      "\t/tmp/tmp4cFdiO.fasta\n",
      "\t/tmp/tmprM4Akj.fasta\n",
      "\t/tmp/tmpBOPFMy.fasta\n",
      "\t/tmp/tmpAO8KHY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpp3mQIH.lastz\n",
      "\t/tmp/tmpbegyzi.lastz\n",
      "\t/tmp/tmpns5C1P.lastz\n",
      "\t/tmp/tmpv59C6Z.lastz\n",
      "\t/tmp/tmpAl_w_f.lastz\n",
      "\t/tmp/tmpuUEK2q.lastz\n",
      "\t/tmp/tmp8ht7wS.lastz\n",
      "\t/tmp/tmp1Efo_e.lastz\n",
      "\t/tmp/tmpccETfD.lastz\n",
      "\t/tmp/tmpg4eE0X.lastz\n",
      "\t/tmp/tmpkqnkFG.lastz\n",
      "\t/tmp/tmpA_giPG.lastz\n",
      "\t/tmp/tmp4cIc32.lastz\n",
      "\t/tmp/tmp7D6O1B.lastz\n",
      "\t/tmp/tmpdep0vS.lastz\n",
      "\t/tmp/tmp2lSyov.lastz\n",
      "\t/tmp/tmpC16Pbi.lastz\n",
      "\t/tmp/tmpF9mLkL.lastz\n",
      "\t/tmp/tmpZNTl6B.lastz\n",
      "\t/tmp/tmp7enCwo.lastz\n",
      "\t/tmp/tmpJndgvy.lastz\n",
      "\t/tmp/tmpiPw_cM.lastz\n",
      "\t/tmp/tmpp7l0X9.lastz\n",
      "\t/tmp/tmp3axbcH.lastz\n",
      "\t/tmp/tmpBoMOYT.lastz\n",
      "\t/tmp/tmpbq3Jhl.lastz\n",
      "\t/tmp/tmp7pquRS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Angiostrongylus_costaricensis_GCA_900624975.lastz\n",
      "Creating Angiostrongylus_costaricensis_GCA_900624975 table\n",
      "Inserting data to Angiostrongylus_costaricensis_GCA_900624975 table\n",
      "\n",
      "Running against Caenorhabditis_angaria_GCA_000165025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmp5Sqvza.fasta\n",
      "\t/tmp/tmpXISpAY.fasta\n",
      "\t/tmp/tmpneipJI.fasta\n",
      "\t/tmp/tmpCzPDde.fasta\n",
      "\t/tmp/tmpAzygge.fasta\n",
      "\t/tmp/tmpyIZUzY.fasta\n",
      "\t/tmp/tmp3tzm85.fasta\n",
      "\t/tmp/tmp8DuJdb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzv281F.lastz\n",
      "\t/tmp/tmpnNEcdA.lastz\n",
      "\t/tmp/tmpIatLbQ.lastz\n",
      "\t/tmp/tmp68nVik.lastz\n",
      "\t/tmp/tmpXAVXQ3.lastz\n",
      "\t/tmp/tmprzFSlE.lastz\n",
      "\t/tmp/tmpD3i9fl.lastz\n",
      "\t/tmp/tmpe2hIhX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_angaria_GCA_000165025.lastz\n",
      "Creating Caenorhabditis_angaria_GCA_000165025 table\n",
      "Inserting data to Caenorhabditis_angaria_GCA_000165025 table\n",
      "\n",
      "Running against Caenorhabditis_becei_GCA_900536315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 9 queries...\n",
      "\t/tmp/tmp9PrN7W.fasta\n",
      "\t/tmp/tmpApZnUn.fasta\n",
      "\t/tmp/tmpfKpb7X.fasta\n",
      "\t/tmp/tmp3wWTB8.fasta\n",
      "\t/tmp/tmphUt_sZ.fasta\n",
      "\t/tmp/tmpvGUuUv.fasta\n",
      "\t/tmp/tmpgsr0Kp.fasta\n",
      "\t/tmp/tmpfO4mRy.fasta\n",
      "\t/tmp/tmp7_o1dy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4l2I0m.lastz\n",
      "\t/tmp/tmpMkruR3.lastz\n",
      "\t/tmp/tmpHoLvMy.lastz\n",
      "\t/tmp/tmpBRFMJB.lastz\n",
      "\t/tmp/tmpleARlU.lastz\n",
      "\t/tmp/tmpjKnmxq.lastz\n",
      "\t/tmp/tmph2Rg3m.lastz\n",
      "\t/tmp/tmpG6P9ld.lastz\n",
      "\t/tmp/tmpjFQorX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_becei_GCA_900536315.lastz\n",
      "Creating Caenorhabditis_becei_GCA_900536315 table\n",
      "Inserting data to Caenorhabditis_becei_GCA_900536315 table\n",
      "\n",
      "Running against Caenorhabditis_bovis_GCA_902636285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpo_qu09.fasta\n",
      "\t/tmp/tmp475CNj.fasta\n",
      "\t/tmp/tmpM2RdNF.fasta\n",
      "\t/tmp/tmpDDrULs.fasta\n",
      "\t/tmp/tmplTV4K7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNO2WhA.lastz\n",
      "\t/tmp/tmpCekNi2.lastz\n",
      "\t/tmp/tmptIs77v.lastz\n",
      "\t/tmp/tmpxmXou2.lastz\n",
      "\t/tmp/tmpLTRQnJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_bovis_GCA_902636285.lastz\n",
      "Creating Caenorhabditis_bovis_GCA_902636285 table\n",
      "Inserting data to Caenorhabditis_bovis_GCA_902636285 table\n",
      "\n",
      "Running against Caenorhabditis_brenneri_GCA_000143925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 19 queries...\n",
      "\t/tmp/tmpALHJ5L.fasta\n",
      "\t/tmp/tmpwP2J1z.fasta\n",
      "\t/tmp/tmpz37EmN.fasta\n",
      "\t/tmp/tmp8lINGi.fasta\n",
      "\t/tmp/tmp8eBF_R.fasta\n",
      "\t/tmp/tmp4lc_nr.fasta\n",
      "\t/tmp/tmpcvwJOt.fasta\n",
      "\t/tmp/tmpJDuYfO.fasta\n",
      "\t/tmp/tmpTRtN9l.fasta\n",
      "\t/tmp/tmp3almxa.fasta\n",
      "\t/tmp/tmpvFSNyD.fasta\n",
      "\t/tmp/tmpBm_PmZ.fasta\n",
      "\t/tmp/tmpfWw4FF.fasta\n",
      "\t/tmp/tmpPtoJK0.fasta\n",
      "\t/tmp/tmpYfAJS7.fasta\n",
      "\t/tmp/tmp1b6TxX.fasta\n",
      "\t/tmp/tmpehFBcu.fasta\n",
      "\t/tmp/tmp_aIjnw.fasta\n",
      "\t/tmp/tmpBD49uk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcMh1Nl.lastz\n",
      "\t/tmp/tmp3Oquar.lastz\n",
      "\t/tmp/tmpaJcnK2.lastz\n",
      "\t/tmp/tmpfxbQpI.lastz\n",
      "\t/tmp/tmpkZUaK1.lastz\n",
      "\t/tmp/tmpsTrgSu.lastz\n",
      "\t/tmp/tmplqzrC9.lastz\n",
      "\t/tmp/tmpy2ZBS6.lastz\n",
      "\t/tmp/tmpTuk9ZW.lastz\n",
      "\t/tmp/tmpV6Jqkc.lastz\n",
      "\t/tmp/tmpow8yn_.lastz\n",
      "\t/tmp/tmpjejDBw.lastz\n",
      "\t/tmp/tmpzBP0fh.lastz\n",
      "\t/tmp/tmp6LZ0M8.lastz\n",
      "\t/tmp/tmp60wsHg.lastz\n",
      "\t/tmp/tmpqk0NI2.lastz\n",
      "\t/tmp/tmp9zaIU1.lastz\n",
      "\t/tmp/tmpBATuEd.lastz\n",
      "\t/tmp/tmpVO0wOU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_brenneri_GCA_000143925.lastz\n",
      "Creating Caenorhabditis_brenneri_GCA_000143925 table\n",
      "Inserting data to Caenorhabditis_brenneri_GCA_000143925 table\n",
      "\n",
      "Running against Caenorhabditis_briggsae_GCA_000004555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpYYJaBy.fasta\n",
      "\t/tmp/tmpUD1jwG.fasta\n",
      "\t/tmp/tmpx9_FQ3.fasta\n",
      "\t/tmp/tmp0G9g12.fasta\n",
      "\t/tmp/tmp1gIbLT.fasta\n",
      "\t/tmp/tmp7VmZsh.fasta\n",
      "\t/tmp/tmpWNmpJ7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpphGF7u.lastz\n",
      "\t/tmp/tmpfRsS_6.lastz\n",
      "\t/tmp/tmp8VFTID.lastz\n",
      "\t/tmp/tmptIGcjf.lastz\n",
      "\t/tmp/tmpHjV92R.lastz\n",
      "\t/tmp/tmpaWUqrt.lastz\n",
      "\t/tmp/tmpTYgCep.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_briggsae_GCA_000004555.lastz\n",
      "Creating Caenorhabditis_briggsae_GCA_000004555 table\n",
      "Inserting data to Caenorhabditis_briggsae_GCA_000004555 table\n",
      "\n",
      "Running against Caenorhabditis_elegans_GCF_000002985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpU1DViF.fasta\n",
      "\t/tmp/tmpb62OOd.fasta\n",
      "\t/tmp/tmpigBj5P.fasta\n",
      "\t/tmp/tmpIlNbKE.fasta\n",
      "\t/tmp/tmp36S_ce.fasta\n",
      "\t/tmp/tmpFkhI1S.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAU3nb8.lastz\n",
      "\t/tmp/tmp1R_czK.lastz\n",
      "\t/tmp/tmpJptqIM.lastz\n",
      "\t/tmp/tmpbE33Ao.lastz\n",
      "\t/tmp/tmphCgzEa.lastz\n",
      "\t/tmp/tmpJ7UpcF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_elegans_GCF_000002985.lastz\n",
      "Creating Caenorhabditis_elegans_GCF_000002985 table\n",
      "Inserting data to Caenorhabditis_elegans_GCF_000002985 table\n",
      "\n",
      "Running against Caenorhabditis_inopinata_GCA_003052745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpXQmlL5.fasta\n",
      "\t/tmp/tmpd5zlSu.fasta\n",
      "\t/tmp/tmp9EG9nq.fasta\n",
      "\t/tmp/tmpXMwpST.fasta\n",
      "\t/tmp/tmpxMYJDO.fasta\n",
      "\t/tmp/tmpgrBwkt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0LuMMk.lastz\n",
      "\t/tmp/tmp4GEOtw.lastz\n",
      "\t/tmp/tmp8lFyOl.lastz\n",
      "\t/tmp/tmpkIaXba.lastz\n",
      "\t/tmp/tmpjSYRtD.lastz\n",
      "\t/tmp/tmpgGhzke.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_inopinata_GCA_003052745.lastz\n",
      "Creating Caenorhabditis_inopinata_GCA_003052745 table\n",
      "Inserting data to Caenorhabditis_inopinata_GCA_003052745 table\n",
      "\n",
      "Running against Caenorhabditis_japonica_GCA_000147155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 17 queries...\n",
      "\t/tmp/tmpvPLz36.fasta\n",
      "\t/tmp/tmpo7CGKo.fasta\n",
      "\t/tmp/tmpEWPvyU.fasta\n",
      "\t/tmp/tmpsHJUoj.fasta\n",
      "\t/tmp/tmpkhDUqA.fasta\n",
      "\t/tmp/tmpxBQZ7m.fasta\n",
      "\t/tmp/tmp0LAp2m.fasta\n",
      "\t/tmp/tmpz4mYrA.fasta\n",
      "\t/tmp/tmpfSrGAJ.fasta\n",
      "\t/tmp/tmpevu3on.fasta\n",
      "\t/tmp/tmpPT8EV4.fasta\n",
      "\t/tmp/tmpS9ilHl.fasta\n",
      "\t/tmp/tmpNi4NPn.fasta\n",
      "\t/tmp/tmpe9BuH1.fasta\n",
      "\t/tmp/tmpHvnGCh.fasta\n",
      "\t/tmp/tmp2Xn2oX.fasta\n",
      "\t/tmp/tmphVD18U.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaVMJ8Y.lastz\n",
      "\t/tmp/tmpY8qe0_.lastz\n",
      "\t/tmp/tmpn8CopE.lastz\n",
      "\t/tmp/tmpfJakbj.lastz\n",
      "\t/tmp/tmpF8ZADx.lastz\n",
      "\t/tmp/tmp_hLvvh.lastz\n",
      "\t/tmp/tmpiHyjgq.lastz\n",
      "\t/tmp/tmpbRtN2n.lastz\n",
      "\t/tmp/tmpBwoPGz.lastz\n",
      "\t/tmp/tmpYZ_qnB.lastz\n",
      "\t/tmp/tmpXJiXop.lastz\n",
      "\t/tmp/tmp2FDvgF.lastz\n",
      "\t/tmp/tmpPulv1D.lastz\n",
      "\t/tmp/tmpzEZ7ZU.lastz\n",
      "\t/tmp/tmpXuKFOi.lastz\n",
      "\t/tmp/tmp6qB2jW.lastz\n",
      "\t/tmp/tmpUd4Och.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_japonica_GCA_000147155.lastz\n",
      "Creating Caenorhabditis_japonica_GCA_000147155 table\n",
      "Inserting data to Caenorhabditis_japonica_GCA_000147155 table\n",
      "\n",
      "Running against Caenorhabditis_latens_GCA_002259235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 12 queries...\n",
      "\t/tmp/tmpw2Jgbe.fasta\n",
      "\t/tmp/tmpPWe0lB.fasta\n",
      "\t/tmp/tmpTdjPGg.fasta\n",
      "\t/tmp/tmpfTTeAe.fasta\n",
      "\t/tmp/tmpa8UZs0.fasta\n",
      "\t/tmp/tmp9MOZTg.fasta\n",
      "\t/tmp/tmpM8fIqc.fasta\n",
      "\t/tmp/tmpaL2_p_.fasta\n",
      "\t/tmp/tmpQQPekC.fasta\n",
      "\t/tmp/tmpjQkl6M.fasta\n",
      "\t/tmp/tmpOV7RVI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHD5S83.lastz\n",
      "\t/tmp/tmpaVSChr.lastz\n",
      "\t/tmp/tmpes8lQA.lastz\n",
      "\t/tmp/tmpNfTVYM.lastz\n",
      "\t/tmp/tmpwi3FEJ.lastz\n",
      "\t/tmp/tmpZkDkQa.lastz\n",
      "\t/tmp/tmprrWdhl.lastz\n",
      "\t/tmp/tmpopm4_h.lastz\n",
      "\t/tmp/tmp_AFEC4.lastz\n",
      "\t/tmp/tmp5OFV3n.lastz\n",
      "\t/tmp/tmp2aMU6z.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_latens_GCA_002259235.lastz\n",
      "Creating Caenorhabditis_latens_GCA_002259235 table\n",
      "Inserting data to Caenorhabditis_latens_GCA_002259235 table\n",
      "\n",
      "Running against Caenorhabditis_nigoni_GCA_002742825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 7 queries...\n",
      "\t/tmp/tmpYTORj9.fasta\n",
      "\t/tmp/tmp9VG5hv.fasta\n",
      "\t/tmp/tmp3ACzpK.fasta\n",
      "\t/tmp/tmpZWlRpy.fasta\n",
      "\t/tmp/tmpSv11ed.fasta\n",
      "\t/tmp/tmpTcg1t7.fasta\n",
      "\t/tmp/tmpoRutqL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpk4Jlq5.lastz\n",
      "\t/tmp/tmpr7GbcD.lastz\n",
      "\t/tmp/tmpvAOaJX.lastz\n",
      "\t/tmp/tmpA9nCJu.lastz\n",
      "\t/tmp/tmpEzgaDi.lastz\n",
      "\t/tmp/tmpTiiMcj.lastz\n",
      "\t/tmp/tmpDkdh8v.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_nigoni_GCA_002742825.lastz\n",
      "Creating Caenorhabditis_nigoni_GCA_002742825 table\n",
      "Inserting data to Caenorhabditis_nigoni_GCA_002742825 table\n",
      "\n",
      "Running against Caenorhabditis_panamensis_GCA_900536275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmpVpjBhy.fasta\n",
      "\t/tmp/tmpxCJMoG.fasta\n",
      "\t/tmp/tmpfWtME9.fasta\n",
      "\t/tmp/tmpGIqKnZ.fasta\n",
      "\t/tmp/tmpNMReCA.fasta\n",
      "\t/tmp/tmpPQKKp3.fasta\n",
      "\t/tmp/tmpJ3J0n6.fasta\n",
      "\t/tmp/tmpCmp_1H.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGBNomN.lastz\n",
      "\t/tmp/tmpYJblXP.lastz\n",
      "\t/tmp/tmpdbXkSW.lastz\n",
      "\t/tmp/tmpEqZ9do.lastz\n",
      "\t/tmp/tmpfAFOVr.lastz\n",
      "\t/tmp/tmpMTyBFE.lastz\n",
      "\t/tmp/tmpCz9kuH.lastz\n",
      "\t/tmp/tmpMS5Nge.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_panamensis_GCA_900536275.lastz\n",
      "Creating Caenorhabditis_panamensis_GCA_900536275 table\n",
      "Inserting data to Caenorhabditis_panamensis_GCA_900536275 table\n",
      "\n",
      "Running against Caenorhabditis_remanei_GCF_000149515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 15 queries...\n",
      "\t/tmp/tmpi1fopm.fasta\n",
      "\t/tmp/tmpYxJ5uv.fasta\n",
      "\t/tmp/tmpjQB393.fasta\n",
      "\t/tmp/tmpSRMRKf.fasta\n",
      "\t/tmp/tmp3cYugP.fasta\n",
      "\t/tmp/tmpHYq8X_.fasta\n",
      "\t/tmp/tmp2JtI0W.fasta\n",
      "\t/tmp/tmpOE1VSN.fasta\n",
      "\t/tmp/tmpdIhzgq.fasta\n",
      "\t/tmp/tmpWPPB59.fasta\n",
      "\t/tmp/tmpCIEcC7.fasta\n",
      "\t/tmp/tmpC43N8c.fasta\n",
      "\t/tmp/tmpYDsTSj.fasta\n",
      "\t/tmp/tmp9NvWiL.fasta\n",
      "\t/tmp/tmpwfNnRn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpT7Q3nz.lastz\n",
      "\t/tmp/tmpmGMTx6.lastz\n",
      "\t/tmp/tmpEL_l8T.lastz\n",
      "\t/tmp/tmp_i78VL.lastz\n",
      "\t/tmp/tmpAYc2hk.lastz\n",
      "\t/tmp/tmp8wJHbt.lastz\n",
      "\t/tmp/tmpOK6TXW.lastz\n",
      "\t/tmp/tmpwfyXuk.lastz\n",
      "\t/tmp/tmpVazIBC.lastz\n",
      "\t/tmp/tmp7DMGPT.lastz\n",
      "\t/tmp/tmpxx7B_a.lastz\n",
      "\t/tmp/tmpRfjR8g.lastz\n",
      "\t/tmp/tmpOnitfk.lastz\n",
      "\t/tmp/tmpzBiheW.lastz\n",
      "\t/tmp/tmpYbn6gh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_remanei_GCF_000149515.lastz\n",
      "Creating Caenorhabditis_remanei_GCF_000149515 table\n",
      "Inserting data to Caenorhabditis_remanei_GCF_000149515 table\n",
      "\n",
      "Running against Caenorhabditis_sp_GCA_900536305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 10 queries...\n",
      "\t/tmp/tmpViel68.fasta\n",
      "\t/tmp/tmpte2PAa.fasta\n",
      "\t/tmp/tmp5Dih_L.fasta\n",
      "\t/tmp/tmp7v0Qme.fasta\n",
      "\t/tmp/tmp0es_Th.fasta\n",
      "\t/tmp/tmpmDZG0H.fasta\n",
      "\t/tmp/tmpLaVxU7.fasta\n",
      "\t/tmp/tmpHd_s_f.fasta\n",
      "\t/tmp/tmp5vVOrw.fasta\n",
      "\t/tmp/tmpr2cHE6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjWFt3z.lastz\n",
      "\t/tmp/tmp1uGiS2.lastz\n",
      "\t/tmp/tmpe_Spl3.lastz\n",
      "\t/tmp/tmpoZTv2u.lastz\n",
      "\t/tmp/tmpGwImcj.lastz\n",
      "\t/tmp/tmpUciDu0.lastz\n",
      "\t/tmp/tmpWpd8kd.lastz\n",
      "\t/tmp/tmpHEZEfN.lastz\n",
      "\t/tmp/tmpIKYrTX.lastz\n",
      "\t/tmp/tmp85dvko.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_sp_GCA_900536305.lastz\n",
      "Creating Caenorhabditis_sp_GCA_900536305 table\n",
      "Inserting data to Caenorhabditis_sp_GCA_900536305 table\n",
      "\n",
      "Running against Caenorhabditis_tropicalis_GCA_000186765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 5 queries...\n",
      "\t/tmp/tmpPkUNQs.fasta\n",
      "\t/tmp/tmp0CEoEX.fasta\n",
      "\t/tmp/tmpr7PKrn.fasta\n",
      "\t/tmp/tmpyXMQzt.fasta\n",
      "\t/tmp/tmpyMJjba.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpov1gn3.lastz\n",
      "\t/tmp/tmpsBrFT4.lastz\n",
      "\t/tmp/tmpI9TFdX.lastz\n",
      "\t/tmp/tmpimE6oS.lastz\n",
      "\t/tmp/tmpfP1Jbe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Caenorhabditis_tropicalis_GCA_000186765.lastz\n",
      "Creating Caenorhabditis_tropicalis_GCA_000186765 table\n",
      "Inserting data to Caenorhabditis_tropicalis_GCA_000186765 table\n",
      "\n",
      "Running against Cylicostephanus_goldi_GCA_900617965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 18 queries...\n",
      "\t/tmp/tmpZn1O42.fasta\n",
      "\t/tmp/tmpaq2uER.fasta\n",
      "\t/tmp/tmp2lGY01.fasta\n",
      "\t/tmp/tmpqqo3AH.fasta\n",
      "\t/tmp/tmpkw5X1h.fasta\n",
      "\t/tmp/tmpTiBFMX.fasta\n",
      "\t/tmp/tmpiT5Eyb.fasta\n",
      "\t/tmp/tmpNfcn3C.fasta\n",
      "\t/tmp/tmp4kLuRj.fasta\n",
      "\t/tmp/tmpiJPso8.fasta\n",
      "\t/tmp/tmpCMiogO.fasta\n",
      "\t/tmp/tmpz6WGLd.fasta\n",
      "\t/tmp/tmpi67o3L.fasta\n",
      "\t/tmp/tmpIIYXVe.fasta\n",
      "\t/tmp/tmp1dQW6N.fasta\n",
      "\t/tmp/tmpHeAYDR.fasta\n",
      "\t/tmp/tmpUDXFlV.fasta\n",
      "\t/tmp/tmpzZMc02.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_3VCq2.lastz\n",
      "\t/tmp/tmpzRd_hJ.lastz\n",
      "\t/tmp/tmpiOP7uY.lastz\n",
      "\t/tmp/tmp6Y8QDJ.lastz\n",
      "\t/tmp/tmpHraciZ.lastz\n",
      "\t/tmp/tmpospkn4.lastz\n",
      "\t/tmp/tmpJAlfOt.lastz\n",
      "\t/tmp/tmpxipewM.lastz\n",
      "\t/tmp/tmpOnDQQx.lastz\n",
      "\t/tmp/tmpIvq2F1.lastz\n",
      "\t/tmp/tmpGLIikA.lastz\n",
      "\t/tmp/tmpOmar0x.lastz\n",
      "\t/tmp/tmpcrOlfV.lastz\n",
      "\t/tmp/tmpEsNuZp.lastz\n",
      "\t/tmp/tmpGwvlTm.lastz\n",
      "\t/tmp/tmpVf1uKt.lastz\n",
      "\t/tmp/tmp1xf703.lastz\n",
      "\t/tmp/tmpsL8qBJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Cylicostephanus_goldi_GCA_900617965.lastz\n",
      "Creating Cylicostephanus_goldi_GCA_900617965 table\n",
      "Inserting data to Cylicostephanus_goldi_GCA_900617965 table\n",
      "\n",
      "Running against Dictyocaulus_viviparus_GCA_000816705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 16 queries...\n",
      "\t/tmp/tmpkst35U.fasta\n",
      "\t/tmp/tmpLmBkNh.fasta\n",
      "\t/tmp/tmpWHV9bV.fasta\n",
      "\t/tmp/tmpbNbG4G.fasta\n",
      "\t/tmp/tmpC08zcD.fasta\n",
      "\t/tmp/tmpUDnBSh.fasta\n",
      "\t/tmp/tmpS0KtHu.fasta\n",
      "\t/tmp/tmpva1Er6.fasta\n",
      "\t/tmp/tmpItM6y7.fasta\n",
      "\t/tmp/tmpkoeOVu.fasta\n",
      "\t/tmp/tmpBFgdxO.fasta\n",
      "\t/tmp/tmp1VmlIL.fasta\n",
      "\t/tmp/tmprBCv6N.fasta\n",
      "\t/tmp/tmpFtKNsP.fasta\n",
      "\t/tmp/tmpW9FIZm.fasta\n",
      "\t/tmp/tmpAogfWS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpisdtHM.lastz\n",
      "\t/tmp/tmp4fD4mT.lastz\n",
      "\t/tmp/tmpi6H4BU.lastz\n",
      "\t/tmp/tmpSvcigQ.lastz\n",
      "\t/tmp/tmpH3E_oQ.lastz\n",
      "\t/tmp/tmp0c_iX3.lastz\n",
      "\t/tmp/tmpz54Sv5.lastz\n",
      "\t/tmp/tmpotxJTe.lastz\n",
      "\t/tmp/tmpyICmRv.lastz\n",
      "\t/tmp/tmpei3IRJ.lastz\n",
      "\t/tmp/tmp9dwf96.lastz\n",
      "\t/tmp/tmpZ3eugn.lastz\n",
      "\t/tmp/tmpiWdXo6.lastz\n",
      "\t/tmp/tmpm6TTaR.lastz\n",
      "\t/tmp/tmpBGQ2zM.lastz\n",
      "\t/tmp/tmpyA_NEL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Dictyocaulus_viviparus_GCA_000816705.lastz\n",
      "Creating Dictyocaulus_viviparus_GCA_000816705 table\n",
      "Inserting data to Dictyocaulus_viviparus_GCA_000816705 table\n",
      "\n",
      "Running against Diploscapter_coronatus_GCA_002207785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 17 queries...\n",
      "\t/tmp/tmpYHagsr.fasta\n",
      "\t/tmp/tmpQPrbLV.fasta\n",
      "\t/tmp/tmpMuWBNx.fasta\n",
      "\t/tmp/tmp0dUNFx.fasta\n",
      "\t/tmp/tmpWbknrh.fasta\n",
      "\t/tmp/tmpc1P54X.fasta\n",
      "\t/tmp/tmpqQGHW5.fasta\n",
      "\t/tmp/tmpy2Spe4.fasta\n",
      "\t/tmp/tmphunyZn.fasta\n",
      "\t/tmp/tmp7PzOaX.fasta\n",
      "\t/tmp/tmp2VhNA4.fasta\n",
      "\t/tmp/tmp7M9xnr.fasta\n",
      "\t/tmp/tmpDimPUz.fasta\n",
      "\t/tmp/tmpoQCGFW.fasta\n",
      "\t/tmp/tmpjvoTrK.fasta\n",
      "\t/tmp/tmpyfIe2q.fasta\n",
      "\t/tmp/tmpFu9iKz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpao4GAX.lastz\n",
      "\t/tmp/tmpGHHoe3.lastz\n",
      "\t/tmp/tmp0uG1Hw.lastz\n",
      "\t/tmp/tmpBPjD1L.lastz\n",
      "\t/tmp/tmpIWQ_dQ.lastz\n",
      "\t/tmp/tmpyYs5KI.lastz\n",
      "\t/tmp/tmpN9r_3F.lastz\n",
      "\t/tmp/tmp0VeZEo.lastz\n",
      "\t/tmp/tmpqPcqH6.lastz\n",
      "\t/tmp/tmpqRyzym.lastz\n",
      "\t/tmp/tmpJmKV_S.lastz\n",
      "\t/tmp/tmpSsFSwI.lastz\n",
      "\t/tmp/tmp6Bljs1.lastz\n",
      "\t/tmp/tmpEOIzwH.lastz\n",
      "\t/tmp/tmpLw8Kot.lastz\n",
      "\t/tmp/tmp4ExiTS.lastz\n",
      "\t/tmp/tmpbkR9Zp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Diploscapter_coronatus_GCA_002207785.lastz\n",
      "Creating Diploscapter_coronatus_GCA_002207785 table\n",
      "Inserting data to Diploscapter_coronatus_GCA_002207785 table\n",
      "\n",
      "Running against Diploscapter_pachys_GCA_002287525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 16 queries...\n",
      "\t/tmp/tmptA2ONo.fasta\n",
      "\t/tmp/tmpmaOCZd.fasta\n",
      "\t/tmp/tmpPaWM33.fasta\n",
      "\t/tmp/tmp_Zrmnm.fasta\n",
      "\t/tmp/tmpOVk6Vs.fasta\n",
      "\t/tmp/tmpyRTPDO.fasta\n",
      "\t/tmp/tmp7JQSQI.fasta\n",
      "\t/tmp/tmpH43GSm.fasta\n",
      "\t/tmp/tmpnLjwxJ.fasta\n",
      "\t/tmp/tmpQqz6nX.fasta\n",
      "\t/tmp/tmpPLOiMg.fasta\n",
      "\t/tmp/tmpM0K8eV.fasta\n",
      "\t/tmp/tmpZno0ii.fasta\n",
      "\t/tmp/tmp3BnBAo.fasta\n",
      "\t/tmp/tmpp7frdl.fasta\n",
      "\t/tmp/tmpuvc1WT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGNPTm2.lastz\n",
      "\t/tmp/tmpCndkq2.lastz\n",
      "\t/tmp/tmpwLgtIp.lastz\n",
      "\t/tmp/tmp_RQMRg.lastz\n",
      "\t/tmp/tmppqSqOn.lastz\n",
      "\t/tmp/tmphMJuwq.lastz\n",
      "\t/tmp/tmpkCiNDy.lastz\n",
      "\t/tmp/tmpigppi1.lastz\n",
      "\t/tmp/tmp4XDJFh.lastz\n",
      "\t/tmp/tmprzgTU3.lastz\n",
      "\t/tmp/tmpFYhksI.lastz\n",
      "\t/tmp/tmpTpnCGF.lastz\n",
      "\t/tmp/tmpJYkBqs.lastz\n",
      "\t/tmp/tmpT_IMOj.lastz\n",
      "\t/tmp/tmpByu3_H.lastz\n",
      "\t/tmp/tmpEI8v9v.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Diploscapter_pachys_GCA_002287525.lastz\n",
      "Creating Diploscapter_pachys_GCA_002287525 table\n",
      "Inserting data to Diploscapter_pachys_GCA_002287525 table\n",
      "\n",
      "Running against Haemonchus_placei_GCA_900617895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 26 queries...\n",
      "\t/tmp/tmpGaOTrN.fasta\n",
      "\t/tmp/tmpmYC7cE.fasta\n",
      "\t/tmp/tmpuNZDg6.fasta\n",
      "\t/tmp/tmpujxIW9.fasta\n",
      "\t/tmp/tmphdLU8Z.fasta\n",
      "\t/tmp/tmpnxIp2i.fasta\n",
      "\t/tmp/tmpkCuE_t.fasta\n",
      "\t/tmp/tmpwklRsh.fasta\n",
      "\t/tmp/tmpGbAXRa.fasta\n",
      "\t/tmp/tmp02RFCT.fasta\n",
      "\t/tmp/tmpsIeD0M.fasta\n",
      "\t/tmp/tmps1OnNi.fasta\n",
      "\t/tmp/tmpFR4Zry.fasta\n",
      "\t/tmp/tmpG_JYKx.fasta\n",
      "\t/tmp/tmp2PDb_a.fasta\n",
      "\t/tmp/tmpWYpB08.fasta\n",
      "\t/tmp/tmpkVsM4J.fasta\n",
      "\t/tmp/tmp6QGLK7.fasta\n",
      "\t/tmp/tmpf5PnCl.fasta\n",
      "\t/tmp/tmpo8v_2H.fasta\n",
      "\t/tmp/tmpMcmKW4.fasta\n",
      "\t/tmp/tmpEgxDyW.fasta\n",
      "\t/tmp/tmpVtFwY7.fasta\n",
      "\t/tmp/tmptinrCK.fasta\n",
      "\t/tmp/tmpp7kQ1q.fasta\n",
      "\t/tmp/tmpGf0Myo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsLxWWm.lastz\n",
      "\t/tmp/tmpqNVRpz.lastz\n",
      "\t/tmp/tmpfdmwvM.lastz\n",
      "\t/tmp/tmpMobIBu.lastz\n",
      "\t/tmp/tmpmSMWC4.lastz\n",
      "\t/tmp/tmpYj7_RG.lastz\n",
      "\t/tmp/tmphK49KY.lastz\n",
      "\t/tmp/tmpwHw7Pf.lastz\n",
      "\t/tmp/tmpSg8wCC.lastz\n",
      "\t/tmp/tmpnDRL1_.lastz\n",
      "\t/tmp/tmpoAApA9.lastz\n",
      "\t/tmp/tmphpLsXX.lastz\n",
      "\t/tmp/tmpGZvcMT.lastz\n",
      "\t/tmp/tmpB0lU7s.lastz\n",
      "\t/tmp/tmpfdWCT8.lastz\n",
      "\t/tmp/tmp1rAzXb.lastz\n",
      "\t/tmp/tmpjU1b28.lastz\n",
      "\t/tmp/tmpEydZa8.lastz\n",
      "\t/tmp/tmpyNR5mr.lastz\n",
      "\t/tmp/tmpRVAwZY.lastz\n",
      "\t/tmp/tmp_r3Ob6.lastz\n",
      "\t/tmp/tmpRpkJAj.lastz\n",
      "\t/tmp/tmpJ5O0_N.lastz\n",
      "\t/tmp/tmphGJAV_.lastz\n",
      "\t/tmp/tmpcGZKVR.lastz\n",
      "\t/tmp/tmptCfPCq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Haemonchus_placei_GCA_900617895.lastz\n",
      "Creating Haemonchus_placei_GCA_900617895 table\n",
      "Inserting data to Haemonchus_placei_GCA_900617895 table\n",
      "\n",
      "Running against Heligmosomoides_polygyrus_GCA_900096555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 70 queries...\n",
      "\t/tmp/tmpOMoHS2.fasta\n",
      "\t/tmp/tmp9WkBVI.fasta\n",
      "\t/tmp/tmp6xjg7e.fasta\n",
      "\t/tmp/tmpQPHarU.fasta\n",
      "\t/tmp/tmpKa8ui5.fasta\n",
      "\t/tmp/tmp4xB8DD.fasta\n",
      "\t/tmp/tmpEBnQHb.fasta\n",
      "\t/tmp/tmpMuOs1H.fasta\n",
      "\t/tmp/tmp3yEqqC.fasta\n",
      "\t/tmp/tmpaLXjal.fasta\n",
      "\t/tmp/tmpjUkNr1.fasta\n",
      "\t/tmp/tmp_wxI4u.fasta\n",
      "\t/tmp/tmpzXGnDM.fasta\n",
      "\t/tmp/tmpRGBkex.fasta\n",
      "\t/tmp/tmpM3GCW0.fasta\n",
      "\t/tmp/tmpXGUZvj.fasta\n",
      "\t/tmp/tmpPfKUiA.fasta\n",
      "\t/tmp/tmpcqKN9W.fasta\n",
      "\t/tmp/tmpgMktQ6.fasta\n",
      "\t/tmp/tmpvn836J.fasta\n",
      "\t/tmp/tmp3An6ay.fasta\n",
      "\t/tmp/tmpAAVdLe.fasta\n",
      "\t/tmp/tmpBfTdyJ.fasta\n",
      "\t/tmp/tmp8nF2pZ.fasta\n",
      "\t/tmp/tmpRWYm_M.fasta\n",
      "\t/tmp/tmpZF2h5g.fasta\n",
      "\t/tmp/tmpPWTTsC.fasta\n",
      "\t/tmp/tmpGiCJrB.fasta\n",
      "\t/tmp/tmpXA6_Cp.fasta\n",
      "\t/tmp/tmpdL9BmC.fasta\n",
      "\t/tmp/tmpzw1TFq.fasta\n",
      "\t/tmp/tmpoRYs6P.fasta\n",
      "\t/tmp/tmpFUgCMx.fasta\n",
      "\t/tmp/tmpFBr7yb.fasta\n",
      "\t/tmp/tmpx9iF4l.fasta\n",
      "\t/tmp/tmpIcD5k9.fasta\n",
      "\t/tmp/tmpvy_Hwu.fasta\n",
      "\t/tmp/tmpsvcTbW.fasta\n",
      "\t/tmp/tmpkENijj.fasta\n",
      "\t/tmp/tmp8Zbjm6.fasta\n",
      "\t/tmp/tmpmh82bN.fasta\n",
      "\t/tmp/tmpMxv2Td.fasta\n",
      "\t/tmp/tmpxQDtLR.fasta\n",
      "\t/tmp/tmp_gcvwt.fasta\n",
      "\t/tmp/tmp_H2L2P.fasta\n",
      "\t/tmp/tmpSi1EBO.fasta\n",
      "\t/tmp/tmpLYRJFD.fasta\n",
      "\t/tmp/tmp8rEjHc.fasta\n",
      "\t/tmp/tmp6CCxSO.fasta\n",
      "\t/tmp/tmpDqjGwK.fasta\n",
      "\t/tmp/tmpq7KPdA.fasta\n",
      "\t/tmp/tmpWSTGGw.fasta\n",
      "\t/tmp/tmp4YTt5v.fasta\n",
      "\t/tmp/tmpx6ldOZ.fasta\n",
      "\t/tmp/tmpXlvgiW.fasta\n",
      "\t/tmp/tmpiOI9z6.fasta\n",
      "\t/tmp/tmpEguB7Q.fasta\n",
      "\t/tmp/tmpubO7vp.fasta\n",
      "\t/tmp/tmpOADG1z.fasta\n",
      "\t/tmp/tmplxKTSD.fasta\n",
      "\t/tmp/tmpi8AQen.fasta\n",
      "\t/tmp/tmpxkClXm.fasta\n",
      "\t/tmp/tmptF3Ff4.fasta\n",
      "\t/tmp/tmpOWumTC.fasta\n",
      "\t/tmp/tmp5fwBGn.fasta\n",
      "\t/tmp/tmpK5_1vM.fasta\n",
      "\t/tmp/tmp9MSYHD.fasta\n",
      "\t/tmp/tmpBERmTu.fasta\n",
      "\t/tmp/tmpbpyc38.fasta\n",
      "\t/tmp/tmpV5y8eW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp24smhY.lastz\n",
      "\t/tmp/tmpNKQ7kW.lastz\n",
      "\t/tmp/tmpFSllHq.lastz\n",
      "\t/tmp/tmp3crA8W.lastz\n",
      "\t/tmp/tmpYi3VDO.lastz\n",
      "\t/tmp/tmpkGVkH9.lastz\n",
      "\t/tmp/tmp30Ut0z.lastz\n",
      "\t/tmp/tmp1rCTHw.lastz\n",
      "\t/tmp/tmpxGViMZ.lastz\n",
      "\t/tmp/tmpyK50Hr.lastz\n",
      "\t/tmp/tmpwAFGbC.lastz\n",
      "\t/tmp/tmpJMmZNy.lastz\n",
      "\t/tmp/tmpQiOEPr.lastz\n",
      "\t/tmp/tmpS6k3Fv.lastz\n",
      "\t/tmp/tmp3N4dJa.lastz\n",
      "\t/tmp/tmpLO0VFr.lastz\n",
      "\t/tmp/tmpRIcdAO.lastz\n",
      "\t/tmp/tmpVUb3gE.lastz\n",
      "\t/tmp/tmpfqu_Rx.lastz\n",
      "\t/tmp/tmpoW7Rsl.lastz\n",
      "\t/tmp/tmpe2S2Ye.lastz\n",
      "\t/tmp/tmpaDRZO0.lastz\n",
      "\t/tmp/tmpWXi6cx.lastz\n",
      "\t/tmp/tmp09sm07.lastz\n",
      "\t/tmp/tmpikXR9d.lastz\n",
      "\t/tmp/tmpJfMsiA.lastz\n",
      "\t/tmp/tmp7Y8q4V.lastz\n",
      "\t/tmp/tmpRB11Hf.lastz\n",
      "\t/tmp/tmpcXnU_X.lastz\n",
      "\t/tmp/tmpmeQXFv.lastz\n",
      "\t/tmp/tmp_EYsoT.lastz\n",
      "\t/tmp/tmpzTcIWG.lastz\n",
      "\t/tmp/tmp5Nzp7J.lastz\n",
      "\t/tmp/tmpsTNdLb.lastz\n",
      "\t/tmp/tmpho1qIL.lastz\n",
      "\t/tmp/tmp7MWe0g.lastz\n",
      "\t/tmp/tmpX5RlmH.lastz\n",
      "\t/tmp/tmpZ362_l.lastz\n",
      "\t/tmp/tmp4jagql.lastz\n",
      "\t/tmp/tmp3Dv1bv.lastz\n",
      "\t/tmp/tmpUDMHGP.lastz\n",
      "\t/tmp/tmpF4cK1t.lastz\n",
      "\t/tmp/tmpLxLruc.lastz\n",
      "\t/tmp/tmpAP2hBO.lastz\n",
      "\t/tmp/tmpaQxnJv.lastz\n",
      "\t/tmp/tmp1JOfqa.lastz\n",
      "\t/tmp/tmppOeIrn.lastz\n",
      "\t/tmp/tmp5MVIDh.lastz\n",
      "\t/tmp/tmpaaHw9e.lastz\n",
      "\t/tmp/tmpPZYQSV.lastz\n",
      "\t/tmp/tmpP9FbEM.lastz\n",
      "\t/tmp/tmp0U_WyN.lastz\n",
      "\t/tmp/tmpiMd1R7.lastz\n",
      "\t/tmp/tmp75xYZE.lastz\n",
      "\t/tmp/tmpfMVtXN.lastz\n",
      "\t/tmp/tmpepy5wB.lastz\n",
      "\t/tmp/tmpCYd485.lastz\n",
      "\t/tmp/tmp5hcXsr.lastz\n",
      "\t/tmp/tmphoOpnA.lastz\n",
      "\t/tmp/tmpjVmJQD.lastz\n",
      "\t/tmp/tmpTlQN0O.lastz\n",
      "\t/tmp/tmpV8s4HQ.lastz\n",
      "\t/tmp/tmpqftpzt.lastz\n",
      "\t/tmp/tmpK7GF7J.lastz\n",
      "\t/tmp/tmpTibNBn.lastz\n",
      "\t/tmp/tmpIvLJAo.lastz\n",
      "\t/tmp/tmppPfrEH.lastz\n",
      "\t/tmp/tmpVKTsS1.lastz\n",
      "\t/tmp/tmpKzJAw3.lastz\n",
      "\t/tmp/tmppqaAZu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Heligmosomoides_polygyrus_GCA_900096555.lastz\n",
      "Creating Heligmosomoides_polygyrus_GCA_900096555 table\n",
      "Inserting data to Heligmosomoides_polygyrus_GCA_900096555 table\n",
      "\n",
      "Running against Heterorhabditis_bacteriophora_GCA_000223415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 8 queries...\n",
      "\t/tmp/tmp7xkNdh.fasta\n",
      "\t/tmp/tmpbOkjBm.fasta\n",
      "\t/tmp/tmp7DfEAL.fasta\n",
      "\t/tmp/tmp4soS7E.fasta\n",
      "\t/tmp/tmpbnun_U.fasta\n",
      "\t/tmp/tmp7tVzzT.fasta\n",
      "\t/tmp/tmpBz7Xjr.fasta\n",
      "\t/tmp/tmppDboXa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2elPHe.lastz\n",
      "\t/tmp/tmpAqaHKe.lastz\n",
      "\t/tmp/tmpa1yo97.lastz\n",
      "\t/tmp/tmpABJtWt.lastz\n",
      "\t/tmp/tmpU1myhE.lastz\n",
      "\t/tmp/tmp84JdCu.lastz\n",
      "\t/tmp/tmp1uZUJt.lastz\n",
      "\t/tmp/tmpW5inla.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Heterorhabditis_bacteriophora_GCA_000223415.lastz\n",
      "Creating Heterorhabditis_bacteriophora_GCA_000223415 table\n",
      "Inserting data to Heterorhabditis_bacteriophora_GCA_000223415 table\n",
      "\n",
      "Running against Mesorhabditis_belari_GCA_900631915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 16 queries...\n",
      "\t/tmp/tmpWoKA48.fasta\n",
      "\t/tmp/tmpXhHBmL.fasta\n",
      "\t/tmp/tmp15fkkF.fasta\n",
      "\t/tmp/tmpN5km_f.fasta\n",
      "\t/tmp/tmpCJHYWO.fasta\n",
      "\t/tmp/tmpGxAc10.fasta\n",
      "\t/tmp/tmpkOSEog.fasta\n",
      "\t/tmp/tmp5U1TXx.fasta\n",
      "\t/tmp/tmpSism0C.fasta\n",
      "\t/tmp/tmpJ1PlOQ.fasta\n",
      "\t/tmp/tmptI8kZI.fasta\n",
      "\t/tmp/tmplECOAe.fasta\n",
      "\t/tmp/tmpGRRvlh.fasta\n",
      "\t/tmp/tmpyyk9Jp.fasta\n",
      "\t/tmp/tmps4u1jM.fasta\n",
      "\t/tmp/tmp5UnNSU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpj1BaPp.lastz\n",
      "\t/tmp/tmpF7oc1A.lastz\n",
      "\t/tmp/tmp5zbyLk.lastz\n",
      "\t/tmp/tmp4oFK3I.lastz\n",
      "\t/tmp/tmpV40fnD.lastz\n",
      "\t/tmp/tmpWLx7oQ.lastz\n",
      "\t/tmp/tmpBavrEK.lastz\n",
      "\t/tmp/tmp7riIdY.lastz\n",
      "\t/tmp/tmpO7vj1t.lastz\n",
      "\t/tmp/tmpt_qrTc.lastz\n",
      "\t/tmp/tmpduzKCp.lastz\n",
      "\t/tmp/tmptv3D7z.lastz\n",
      "\t/tmp/tmp4sLa9N.lastz\n",
      "\t/tmp/tmphPYj2C.lastz\n",
      "\t/tmp/tmpUdgBI4.lastz\n",
      "\t/tmp/tmpUenydP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Mesorhabditis_belari_GCA_900631915.lastz\n",
      "Creating Mesorhabditis_belari_GCA_900631915 table\n",
      "Inserting data to Mesorhabditis_belari_GCA_900631915 table\n",
      "\n",
      "Running against Micoletzkya_japonica_GCA_900490955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 21 queries...\n",
      "\t/tmp/tmpLsoYeR.fasta\n",
      "\t/tmp/tmpc_sLq5.fasta\n",
      "\t/tmp/tmpvNdvNy.fasta\n",
      "\t/tmp/tmpcDZVGO.fasta\n",
      "\t/tmp/tmpFj95sa.fasta\n",
      "\t/tmp/tmpTKcecC.fasta\n",
      "\t/tmp/tmpPAqdww.fasta\n",
      "\t/tmp/tmpBEWwbu.fasta\n",
      "\t/tmp/tmp68QKka.fasta\n",
      "\t/tmp/tmpZlbtBj.fasta\n",
      "\t/tmp/tmpDLui2S.fasta\n",
      "\t/tmp/tmpceNvk8.fasta\n",
      "\t/tmp/tmp82cr2_.fasta\n",
      "\t/tmp/tmp6hIM7m.fasta\n",
      "\t/tmp/tmplARpWW.fasta\n",
      "\t/tmp/tmpKdoc6P.fasta\n",
      "\t/tmp/tmpxPlQTr.fasta\n",
      "\t/tmp/tmpE3peYW.fasta\n",
      "\t/tmp/tmpCJ5chR.fasta\n",
      "\t/tmp/tmpsEkQOt.fasta\n",
      "\t/tmp/tmpMIECGx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppiCHQT.lastz\n",
      "\t/tmp/tmpolVJ3y.lastz\n",
      "\t/tmp/tmp7jpnLv.lastz\n",
      "\t/tmp/tmpAQVEcm.lastz\n",
      "\t/tmp/tmpJ9LxiP.lastz\n",
      "\t/tmp/tmpZDpHOe.lastz\n",
      "\t/tmp/tmpAAXpjH.lastz\n",
      "\t/tmp/tmpYhJmRz.lastz\n",
      "\t/tmp/tmpgeKkcV.lastz\n",
      "\t/tmp/tmpsGvtne.lastz\n",
      "\t/tmp/tmpQGeOBF.lastz\n",
      "\t/tmp/tmpktwfA6.lastz\n",
      "\t/tmp/tmpLH3Jzt.lastz\n",
      "\t/tmp/tmpUOMuJJ.lastz\n",
      "\t/tmp/tmp0qQXsj.lastz\n",
      "\t/tmp/tmpWw1jWZ.lastz\n",
      "\t/tmp/tmp2wRCDV.lastz\n",
      "\t/tmp/tmp3LvplC.lastz\n",
      "\t/tmp/tmpqkY9ir.lastz\n",
      "\t/tmp/tmplyk223.lastz\n",
      "\t/tmp/tmpwXpia3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Micoletzkya_japonica_GCA_900490955.lastz\n",
      "Creating Micoletzkya_japonica_GCA_900490955 table\n",
      "Inserting data to Micoletzkya_japonica_GCA_900490955 table\n",
      "\n",
      "Running against Necator_americanus_GCF_000507365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 25 queries...\n",
      "\t/tmp/tmpYsRrCY.fasta\n",
      "\t/tmp/tmpUePmEH.fasta\n",
      "\t/tmp/tmpThQ8ef.fasta\n",
      "\t/tmp/tmpqsnD6u.fasta\n",
      "\t/tmp/tmpbDunT0.fasta\n",
      "\t/tmp/tmpSZsC7G.fasta\n",
      "\t/tmp/tmp6TbCow.fasta\n",
      "\t/tmp/tmpnTA7ms.fasta\n",
      "\t/tmp/tmpOtKSl6.fasta\n",
      "\t/tmp/tmpm_Fz_B.fasta\n",
      "\t/tmp/tmpMACd7H.fasta\n",
      "\t/tmp/tmpMRceoL.fasta\n",
      "\t/tmp/tmpvhBISO.fasta\n",
      "\t/tmp/tmpVvGdro.fasta\n",
      "\t/tmp/tmpRdcQsW.fasta\n",
      "\t/tmp/tmpFGiSGH.fasta\n",
      "\t/tmp/tmp2teV9R.fasta\n",
      "\t/tmp/tmpSl0Nvm.fasta\n",
      "\t/tmp/tmppNkSgv.fasta\n",
      "\t/tmp/tmpDhUNy3.fasta\n",
      "\t/tmp/tmpXD3jAg.fasta\n",
      "\t/tmp/tmpKWtK8o.fasta\n",
      "\t/tmp/tmpcJGGLc.fasta\n",
      "\t/tmp/tmp36uvZ0.fasta\n",
      "\t/tmp/tmpFIWxpk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBugVvq.lastz\n",
      "\t/tmp/tmp6uio4l.lastz\n",
      "\t/tmp/tmpN1bBPN.lastz\n",
      "\t/tmp/tmpveFARq.lastz\n",
      "\t/tmp/tmpCHfMNQ.lastz\n",
      "\t/tmp/tmpzLKkyK.lastz\n",
      "\t/tmp/tmpN3U1o0.lastz\n",
      "\t/tmp/tmpBaQ2Il.lastz\n",
      "\t/tmp/tmpxQDrFP.lastz\n",
      "\t/tmp/tmpPqDu5m.lastz\n",
      "\t/tmp/tmph43ww6.lastz\n",
      "\t/tmp/tmpuyGeKE.lastz\n",
      "\t/tmp/tmpCdRG0N.lastz\n",
      "\t/tmp/tmpUgj9SA.lastz\n",
      "\t/tmp/tmpZegT20.lastz\n",
      "\t/tmp/tmpBEi6Z9.lastz\n",
      "\t/tmp/tmpwYxRjo.lastz\n",
      "\t/tmp/tmp643JyX.lastz\n",
      "\t/tmp/tmpMs_rkh.lastz\n",
      "\t/tmp/tmp5pZIk2.lastz\n",
      "\t/tmp/tmpCWFDWf.lastz\n",
      "\t/tmp/tmpJJxElL.lastz\n",
      "\t/tmp/tmpU7i_oF.lastz\n",
      "\t/tmp/tmpmj2Woe.lastz\n",
      "\t/tmp/tmpczzYAu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Necator_americanus_GCF_000507365.lastz\n",
      "Creating Necator_americanus_GCF_000507365 table\n",
      "Inserting data to Necator_americanus_GCF_000507365 table\n",
      "\n",
      "Running against Nippostrongylus_brasiliensis_GCA_900200055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 35 queries...\n",
      "\t/tmp/tmpJCyybx.fasta\n",
      "\t/tmp/tmpCHI44r.fasta\n",
      "\t/tmp/tmpntRDpG.fasta\n",
      "\t/tmp/tmpz9z3Wt.fasta\n",
      "\t/tmp/tmpO9DCBc.fasta\n",
      "\t/tmp/tmp66W5L6.fasta\n",
      "\t/tmp/tmpCzlxcs.fasta\n",
      "\t/tmp/tmpa4b7gG.fasta\n",
      "\t/tmp/tmpDOrBxh.fasta\n",
      "\t/tmp/tmpJRxyne.fasta\n",
      "\t/tmp/tmpW11Pgu.fasta\n",
      "\t/tmp/tmplvaaX1.fasta\n",
      "\t/tmp/tmptVXHtU.fasta\n",
      "\t/tmp/tmp9WMb08.fasta\n",
      "\t/tmp/tmpRQUHY8.fasta\n",
      "\t/tmp/tmpjDG5yI.fasta\n",
      "\t/tmp/tmpciFzrS.fasta\n",
      "\t/tmp/tmpa6A0lU.fasta\n",
      "\t/tmp/tmpdOwzfr.fasta\n",
      "\t/tmp/tmpzn_7VR.fasta\n",
      "\t/tmp/tmpUfujOa.fasta\n",
      "\t/tmp/tmpbnO_Fw.fasta\n",
      "\t/tmp/tmpMJkrU3.fasta\n",
      "\t/tmp/tmp34JmdP.fasta\n",
      "\t/tmp/tmpFGzkKq.fasta\n",
      "\t/tmp/tmpk2Sbx4.fasta\n",
      "\t/tmp/tmpzpPYIW.fasta\n",
      "\t/tmp/tmpFBTJeH.fasta\n",
      "\t/tmp/tmp3JEVqZ.fasta\n",
      "\t/tmp/tmpmIjGR8.fasta\n",
      "\t/tmp/tmpiOgiTF.fasta\n",
      "\t/tmp/tmpOf0h1W.fasta\n",
      "\t/tmp/tmpZrgHp9.fasta\n",
      "\t/tmp/tmp4jFE12.fasta\n",
      "\t/tmp/tmppeFfdh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZRg4Gm.lastz\n",
      "\t/tmp/tmpINkvoe.lastz\n",
      "\t/tmp/tmp84hVzQ.lastz\n",
      "\t/tmp/tmpP30G7v.lastz\n",
      "\t/tmp/tmpm7UBDN.lastz\n",
      "\t/tmp/tmpT17RsF.lastz\n",
      "\t/tmp/tmpqNRP_h.lastz\n",
      "\t/tmp/tmp1FGHd3.lastz\n",
      "\t/tmp/tmpfgt_6s.lastz\n",
      "\t/tmp/tmp9flFiC.lastz\n",
      "\t/tmp/tmpjH2Oob.lastz\n",
      "\t/tmp/tmpbTGDNf.lastz\n",
      "\t/tmp/tmpn9LMzp.lastz\n",
      "\t/tmp/tmppxAsTo.lastz\n",
      "\t/tmp/tmpzZmCmr.lastz\n",
      "\t/tmp/tmp1oa3NW.lastz\n",
      "\t/tmp/tmphbhXwK.lastz\n",
      "\t/tmp/tmpw0JhVq.lastz\n",
      "\t/tmp/tmpH40kfy.lastz\n",
      "\t/tmp/tmpoKl5jG.lastz\n",
      "\t/tmp/tmp2m1xvT.lastz\n",
      "\t/tmp/tmpKMDhS8.lastz\n",
      "\t/tmp/tmpvx3gT9.lastz\n",
      "\t/tmp/tmpigvo8G.lastz\n",
      "\t/tmp/tmpU1E97r.lastz\n",
      "\t/tmp/tmpswTYn1.lastz\n",
      "\t/tmp/tmpQpRF6z.lastz\n",
      "\t/tmp/tmpudTr0O.lastz\n",
      "\t/tmp/tmpI6XVdI.lastz\n",
      "\t/tmp/tmpLfzceL.lastz\n",
      "\t/tmp/tmp6r2iIr.lastz\n",
      "\t/tmp/tmprZhE_z.lastz\n",
      "\t/tmp/tmpWbVUD7.lastz\n",
      "\t/tmp/tmpx6oxZ4.lastz\n",
      "\t/tmp/tmpjVJrb7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Nippostrongylus_brasiliensis_GCA_900200055.lastz\n",
      "Creating Nippostrongylus_brasiliensis_GCA_900200055 table\n",
      "Inserting data to Nippostrongylus_brasiliensis_GCA_900200055 table\n",
      "\n",
      "Running against Oesophagostomum_dentatum_GCA_000797555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 45 queries...\n",
      "\t/tmp/tmpEgThzB.fasta\n",
      "\t/tmp/tmp34ISUs.fasta\n",
      "\t/tmp/tmpiaxlo3.fasta\n",
      "\t/tmp/tmpeOBIRx.fasta\n",
      "\t/tmp/tmpZvfiw9.fasta\n",
      "\t/tmp/tmpMjiSMp.fasta\n",
      "\t/tmp/tmpx9t_CV.fasta\n",
      "\t/tmp/tmpmDy418.fasta\n",
      "\t/tmp/tmpgE48JZ.fasta\n",
      "\t/tmp/tmpInHu97.fasta\n",
      "\t/tmp/tmpjbaIJP.fasta\n",
      "\t/tmp/tmpHrrc2O.fasta\n",
      "\t/tmp/tmpwo07S2.fasta\n",
      "\t/tmp/tmpkobQyW.fasta\n",
      "\t/tmp/tmpkQXJ_O.fasta\n",
      "\t/tmp/tmphxPiLP.fasta\n",
      "\t/tmp/tmpRZRIb9.fasta\n",
      "\t/tmp/tmpQWq9Xr.fasta\n",
      "\t/tmp/tmpAlTO62.fasta\n",
      "\t/tmp/tmp_c3MBP.fasta\n",
      "\t/tmp/tmpad4row.fasta\n",
      "\t/tmp/tmpdSb0l4.fasta\n",
      "\t/tmp/tmp6TFSxh.fasta\n",
      "\t/tmp/tmpkB8VA6.fasta\n",
      "\t/tmp/tmp3gwi1q.fasta\n",
      "\t/tmp/tmpRrBzyS.fasta\n",
      "\t/tmp/tmpHIDboK.fasta\n",
      "\t/tmp/tmp8r04Wf.fasta\n",
      "\t/tmp/tmpqMGHBr.fasta\n",
      "\t/tmp/tmpP2IcOO.fasta\n",
      "\t/tmp/tmp1hpRqj.fasta\n",
      "\t/tmp/tmpWTD1xx.fasta\n",
      "\t/tmp/tmp8v68dB.fasta\n",
      "\t/tmp/tmpl_Qa4C.fasta\n",
      "\t/tmp/tmpciogqm.fasta\n",
      "\t/tmp/tmpbq82k1.fasta\n",
      "\t/tmp/tmp6xckUX.fasta\n",
      "\t/tmp/tmpdiJSkM.fasta\n",
      "\t/tmp/tmp7eZI__.fasta\n",
      "\t/tmp/tmpVeR5L2.fasta\n",
      "\t/tmp/tmpaHqfGo.fasta\n",
      "\t/tmp/tmpOYT8P9.fasta\n",
      "\t/tmp/tmpJ6RBZK.fasta\n",
      "\t/tmp/tmpVA2ynr.fasta\n",
      "\t/tmp/tmpE81cDd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSCseiB.lastz\n",
      "\t/tmp/tmpRXBgyr.lastz\n",
      "\t/tmp/tmprAmolr.lastz\n",
      "\t/tmp/tmptRZFQN.lastz\n",
      "\t/tmp/tmpfguniE.lastz\n",
      "\t/tmp/tmpYYpfgC.lastz\n",
      "\t/tmp/tmp98Bt_d.lastz\n",
      "\t/tmp/tmplYyAbV.lastz\n",
      "\t/tmp/tmp5a4sEC.lastz\n",
      "\t/tmp/tmphS3xrE.lastz\n",
      "\t/tmp/tmp5d8AUH.lastz\n",
      "\t/tmp/tmp8LSs3b.lastz\n",
      "\t/tmp/tmpoxxMTu.lastz\n",
      "\t/tmp/tmppN047G.lastz\n",
      "\t/tmp/tmpLsEB4u.lastz\n",
      "\t/tmp/tmpzg0AeE.lastz\n",
      "\t/tmp/tmpq5znFf.lastz\n",
      "\t/tmp/tmp3W2NQ9.lastz\n",
      "\t/tmp/tmp4tDcoO.lastz\n",
      "\t/tmp/tmpt61AVU.lastz\n",
      "\t/tmp/tmpj9vddJ.lastz\n",
      "\t/tmp/tmpXkdNkx.lastz\n",
      "\t/tmp/tmpksadcB.lastz\n",
      "\t/tmp/tmpMzlMPj.lastz\n",
      "\t/tmp/tmpkkWA8s.lastz\n",
      "\t/tmp/tmpHQc72z.lastz\n",
      "\t/tmp/tmpmtR1Rb.lastz\n",
      "\t/tmp/tmpsnJlSD.lastz\n",
      "\t/tmp/tmpzqd2MW.lastz\n",
      "\t/tmp/tmp0l9BXg.lastz\n",
      "\t/tmp/tmpAp5ye6.lastz\n",
      "\t/tmp/tmpLHesnx.lastz\n",
      "\t/tmp/tmpTIvn7I.lastz\n",
      "\t/tmp/tmpdrWc0G.lastz\n",
      "\t/tmp/tmpWmb8GZ.lastz\n",
      "\t/tmp/tmpO1_AFm.lastz\n",
      "\t/tmp/tmpbtRmNr.lastz\n",
      "\t/tmp/tmp5vENXq.lastz\n",
      "\t/tmp/tmpoLj2Qq.lastz\n",
      "\t/tmp/tmpdnOUQj.lastz\n",
      "\t/tmp/tmpmOvUBj.lastz\n",
      "\t/tmp/tmpC_ahEA.lastz\n",
      "\t/tmp/tmp2LzqMp.lastz\n",
      "\t/tmp/tmpy550M7.lastz\n",
      "\t/tmp/tmpuY7Fwd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Oesophagostomum_dentatum_GCA_000797555.lastz\n",
      "Creating Oesophagostomum_dentatum_GCA_000797555 table\n",
      "Inserting data to Oesophagostomum_dentatum_GCA_000797555 table\n",
      "\n",
      "Running against Oscheius_sp_GCA_000934875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 4 queries...\n",
      "\t/tmp/tmpF6d8dh.fasta\n",
      "\t/tmp/tmpmPGkb3.fasta\n",
      "\t/tmp/tmpEo3ARE.fasta\n",
      "\t/tmp/tmpaFW8pm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9DddNX.lastz\n",
      "\t/tmp/tmp8umbcy.lastz\n",
      "\t/tmp/tmp4KR3uE.lastz\n",
      "\t/tmp/tmpIMPD48.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Oscheius_sp_GCA_000934875.lastz\n",
      "Creating Oscheius_sp_GCA_000934875 table\n",
      "Inserting data to Oscheius_sp_GCA_000934875 table\n",
      "\n",
      "Running against Oscheius_tipulae_GCA_900184235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpkx8EY0.fasta\n",
      "\t/tmp/tmpchkI6m.fasta\n",
      "\t/tmp/tmpwBV8l9.fasta\n",
      "\t/tmp/tmprBZj5u.fasta\n",
      "\t/tmp/tmpLXCDy5.fasta\n",
      "\t/tmp/tmpMd7mVe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJrZtK1.lastz\n",
      "\t/tmp/tmpEWfwHT.lastz\n",
      "\t/tmp/tmpz7Gfsz.lastz\n",
      "\t/tmp/tmpQsymru.lastz\n",
      "\t/tmp/tmpChWUxn.lastz\n",
      "\t/tmp/tmprFVA9v.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Oscheius_tipulae_GCA_900184235.lastz\n",
      "Creating Oscheius_tipulae_GCA_900184235 table\n",
      "Inserting data to Oscheius_tipulae_GCA_900184235 table\n",
      "\n",
      "Running against Parapristionchus_giblindavisi_GCA_900491355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 20 queries...\n",
      "\t/tmp/tmpvbCLGV.fasta\n",
      "\t/tmp/tmp93CckR.fasta\n",
      "\t/tmp/tmpeSkHLU.fasta\n",
      "\t/tmp/tmpuxKXnr.fasta\n",
      "\t/tmp/tmpE4mrdd.fasta\n",
      "\t/tmp/tmpbFwCUi.fasta\n",
      "\t/tmp/tmpnJAcJv.fasta\n",
      "\t/tmp/tmp1n22Dg.fasta\n",
      "\t/tmp/tmpuBkk45.fasta\n",
      "\t/tmp/tmpQEVyh_.fasta\n",
      "\t/tmp/tmp5Dfs8X.fasta\n",
      "\t/tmp/tmpWDiamm.fasta\n",
      "\t/tmp/tmpYu7eut.fasta\n",
      "\t/tmp/tmpF7j5Ci.fasta\n",
      "\t/tmp/tmp_B6QIb.fasta\n",
      "\t/tmp/tmpiyiJ62.fasta\n",
      "\t/tmp/tmp2pTXv3.fasta\n",
      "\t/tmp/tmpirhtkX.fasta\n",
      "\t/tmp/tmpKjRI8L.fasta\n",
      "\t/tmp/tmp3kFYZJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd8yGeP.lastz\n",
      "\t/tmp/tmpX6TqhU.lastz\n",
      "\t/tmp/tmpc_nzWn.lastz\n",
      "\t/tmp/tmp_1OEkk.lastz\n",
      "\t/tmp/tmpMpf0YW.lastz\n",
      "\t/tmp/tmp_4h8vS.lastz\n",
      "\t/tmp/tmpb8KhtI.lastz\n",
      "\t/tmp/tmpjCmprE.lastz\n",
      "\t/tmp/tmpPU0A8O.lastz\n",
      "\t/tmp/tmpqMce7d.lastz\n",
      "\t/tmp/tmpjZjFxJ.lastz\n",
      "\t/tmp/tmpZ95juJ.lastz\n",
      "\t/tmp/tmp4WctKh.lastz\n",
      "\t/tmp/tmpFYZ89i.lastz\n",
      "\t/tmp/tmpDy25s4.lastz\n",
      "\t/tmp/tmpoKRDdj.lastz\n",
      "\t/tmp/tmp7OiCJe.lastz\n",
      "\t/tmp/tmp7cnP5N.lastz\n",
      "\t/tmp/tmpsuQGQn.lastz\n",
      "\t/tmp/tmp2aqNdx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Parapristionchus_giblindavisi_GCA_900491355.lastz\n",
      "Creating Parapristionchus_giblindavisi_GCA_900491355 table\n",
      "Inserting data to Parapristionchus_giblindavisi_GCA_900491355 table\n",
      "\n",
      "Running against Pristionchus_arcanus_GCA_900490705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 20 queries...\n",
      "\t/tmp/tmpPxrXNB.fasta\n",
      "\t/tmp/tmppYvsnP.fasta\n",
      "\t/tmp/tmpttSKGl.fasta\n",
      "\t/tmp/tmp1suZ1t.fasta\n",
      "\t/tmp/tmpvt8w3l.fasta\n",
      "\t/tmp/tmp7Ymtdd.fasta\n",
      "\t/tmp/tmpYko_6f.fasta\n",
      "\t/tmp/tmpmMNjcx.fasta\n",
      "\t/tmp/tmp26YiKU.fasta\n",
      "\t/tmp/tmppJ4MTt.fasta\n",
      "\t/tmp/tmpUedHgm.fasta\n",
      "\t/tmp/tmpvIxWFq.fasta\n",
      "\t/tmp/tmpV0ywS6.fasta\n",
      "\t/tmp/tmpsOJOfY.fasta\n",
      "\t/tmp/tmp9pHwdb.fasta\n",
      "\t/tmp/tmpOXUof6.fasta\n",
      "\t/tmp/tmpPm00Jb.fasta\n",
      "\t/tmp/tmp1PBmFH.fasta\n",
      "\t/tmp/tmpsvWWCG.fasta\n",
      "\t/tmp/tmpb_S8aW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpR2_VvZ.lastz\n",
      "\t/tmp/tmpskeRqK.lastz\n",
      "\t/tmp/tmpBiH7UP.lastz\n",
      "\t/tmp/tmpdlD5md.lastz\n",
      "\t/tmp/tmpMU4ky0.lastz\n",
      "\t/tmp/tmpb_YoCq.lastz\n",
      "\t/tmp/tmp7Es4Qc.lastz\n",
      "\t/tmp/tmpe6zwDl.lastz\n",
      "\t/tmp/tmpkN7k6w.lastz\n",
      "\t/tmp/tmp9MwwJb.lastz\n",
      "\t/tmp/tmpX2rUwC.lastz\n",
      "\t/tmp/tmpociDjN.lastz\n",
      "\t/tmp/tmpdZXJO7.lastz\n",
      "\t/tmp/tmpwPtlbV.lastz\n",
      "\t/tmp/tmp9ugUnc.lastz\n",
      "\t/tmp/tmp1IcKii.lastz\n",
      "\t/tmp/tmp1D4Q4J.lastz\n",
      "\t/tmp/tmpTdUVJb.lastz\n",
      "\t/tmp/tmpZkxpH4.lastz\n",
      "\t/tmp/tmphXYNEP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_arcanus_GCA_900490705.lastz\n",
      "Creating Pristionchus_arcanus_GCA_900490705 table\n",
      "Inserting data to Pristionchus_arcanus_GCA_900490705 table\n",
      "\n",
      "Running against Pristionchus_entomophagus_GCA_900490825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 26 queries...\n",
      "\t/tmp/tmpa7QXgk.fasta\n",
      "\t/tmp/tmpVitaD9.fasta\n",
      "\t/tmp/tmpdktC9Z.fasta\n",
      "\t/tmp/tmpvUYYXP.fasta\n",
      "\t/tmp/tmpm7vGL1.fasta\n",
      "\t/tmp/tmp90TFnc.fasta\n",
      "\t/tmp/tmpwsV4Nf.fasta\n",
      "\t/tmp/tmpmU6XlJ.fasta\n",
      "\t/tmp/tmprtn_k7.fasta\n",
      "\t/tmp/tmpROzP2I.fasta\n",
      "\t/tmp/tmpUhiVps.fasta\n",
      "\t/tmp/tmpdy28ra.fasta\n",
      "\t/tmp/tmpzwB7l_.fasta\n",
      "\t/tmp/tmpP6Xmvw.fasta\n",
      "\t/tmp/tmpj0TOCO.fasta\n",
      "\t/tmp/tmpFOJNyA.fasta\n",
      "\t/tmp/tmpD3FclI.fasta\n",
      "\t/tmp/tmp5tDI0F.fasta\n",
      "\t/tmp/tmp_5tuO3.fasta\n",
      "\t/tmp/tmpNFM59d.fasta\n",
      "\t/tmp/tmpLHKg47.fasta\n",
      "\t/tmp/tmpPcUq_k.fasta\n",
      "\t/tmp/tmpilvXgR.fasta\n",
      "\t/tmp/tmpuXMdaw.fasta\n",
      "\t/tmp/tmpLUEgNV.fasta\n",
      "\t/tmp/tmpnwDAdY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2i1gXW.lastz\n",
      "\t/tmp/tmpYNEEFW.lastz\n",
      "\t/tmp/tmpBG5a4W.lastz\n",
      "\t/tmp/tmpefy9Ga.lastz\n",
      "\t/tmp/tmplBWaIw.lastz\n",
      "\t/tmp/tmpf8j_5k.lastz\n",
      "\t/tmp/tmpVAwDGH.lastz\n",
      "\t/tmp/tmpBHqvOx.lastz\n",
      "\t/tmp/tmpxQn7Li.lastz\n",
      "\t/tmp/tmpYQba8j.lastz\n",
      "\t/tmp/tmpGzH_Hz.lastz\n",
      "\t/tmp/tmpH1lxtI.lastz\n",
      "\t/tmp/tmpLtK2rR.lastz\n",
      "\t/tmp/tmpf4e4vT.lastz\n",
      "\t/tmp/tmpOlrMdm.lastz\n",
      "\t/tmp/tmpOoq4d5.lastz\n",
      "\t/tmp/tmpyJ6yGI.lastz\n",
      "\t/tmp/tmpo3Iiat.lastz\n",
      "\t/tmp/tmpXdTxsH.lastz\n",
      "\t/tmp/tmpr6Od3E.lastz\n",
      "\t/tmp/tmpVcHfl8.lastz\n",
      "\t/tmp/tmpzPKnzK.lastz\n",
      "\t/tmp/tmpm163lJ.lastz\n",
      "\t/tmp/tmpSW0UI2.lastz\n",
      "\t/tmp/tmplohPpS.lastz\n",
      "\t/tmp/tmpEyEsbE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_entomophagus_GCA_900490825.lastz\n",
      "Creating Pristionchus_entomophagus_GCA_900490825 table\n",
      "Inserting data to Pristionchus_entomophagus_GCA_900490825 table\n",
      "\n",
      "Running against Pristionchus_exspctatus_GCA_900380275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 18 queries...\n",
      "\t/tmp/tmpT0lNhc.fasta\n",
      "\t/tmp/tmpzGzQ20.fasta\n",
      "\t/tmp/tmpoKjySx.fasta\n",
      "\t/tmp/tmprw6bQy.fasta\n",
      "\t/tmp/tmpuOQgcw.fasta\n",
      "\t/tmp/tmpfAdrBX.fasta\n",
      "\t/tmp/tmp9rBlZc.fasta\n",
      "\t/tmp/tmpzjl1RB.fasta\n",
      "\t/tmp/tmpNyYpCU.fasta\n",
      "\t/tmp/tmpOjJZZ_.fasta\n",
      "\t/tmp/tmpbwhoBm.fasta\n",
      "\t/tmp/tmppcIeNs.fasta\n",
      "\t/tmp/tmptzyNfd.fasta\n",
      "\t/tmp/tmplBrTKo.fasta\n",
      "\t/tmp/tmpXZ106p.fasta\n",
      "\t/tmp/tmp2Wbw0L.fasta\n",
      "\t/tmp/tmpWIfiJC.fasta\n",
      "\t/tmp/tmpkrrk0x.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJP1wmW.lastz\n",
      "\t/tmp/tmp_QRuIv.lastz\n",
      "\t/tmp/tmphlTsEs.lastz\n",
      "\t/tmp/tmplbqVfi.lastz\n",
      "\t/tmp/tmpNSIbsw.lastz\n",
      "\t/tmp/tmpKlQ5MT.lastz\n",
      "\t/tmp/tmpwZy1yf.lastz\n",
      "\t/tmp/tmpprdZLu.lastz\n",
      "\t/tmp/tmpfJ8gx1.lastz\n",
      "\t/tmp/tmpQIZmkG.lastz\n",
      "\t/tmp/tmptCKf2W.lastz\n",
      "\t/tmp/tmpB3uJd1.lastz\n",
      "\t/tmp/tmp_llBvV.lastz\n",
      "\t/tmp/tmpxlV3sU.lastz\n",
      "\t/tmp/tmpZe1sOm.lastz\n",
      "\t/tmp/tmpRV_4lm.lastz\n",
      "\t/tmp/tmpjKOa5V.lastz\n",
      "\t/tmp/tmpcfrPZh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_exspctatus_GCA_900380275.lastz\n",
      "Creating Pristionchus_exspctatus_GCA_900380275 table\n",
      "Inserting data to Pristionchus_exspctatus_GCA_900380275 table\n",
      "\n",
      "Running against Pristionchus_fissidentatus_GCA_900490895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 24 queries...\n",
      "\t/tmp/tmpQYrGwt.fasta\n",
      "\t/tmp/tmp7nzfjK.fasta\n",
      "\t/tmp/tmpFxl7Ee.fasta\n",
      "\t/tmp/tmpYvS1XD.fasta\n",
      "\t/tmp/tmpP3Tc80.fasta\n",
      "\t/tmp/tmpOX0ji6.fasta\n",
      "\t/tmp/tmp1xR5si.fasta\n",
      "\t/tmp/tmpfr_XZB.fasta\n",
      "\t/tmp/tmpODKVrO.fasta\n",
      "\t/tmp/tmpMO8gZ0.fasta\n",
      "\t/tmp/tmpMgiDHH.fasta\n",
      "\t/tmp/tmplTyKM4.fasta\n",
      "\t/tmp/tmpgIogp_.fasta\n",
      "\t/tmp/tmpBnyt6s.fasta\n",
      "\t/tmp/tmpHCnBa3.fasta\n",
      "\t/tmp/tmplIlxDb.fasta\n",
      "\t/tmp/tmpWm_9tg.fasta\n",
      "\t/tmp/tmpamTiN_.fasta\n",
      "\t/tmp/tmpuqDzRp.fasta\n",
      "\t/tmp/tmp9cISdh.fasta\n",
      "\t/tmp/tmphcPdxA.fasta\n",
      "\t/tmp/tmpL0My7w.fasta\n",
      "\t/tmp/tmp_eX6Ho.fasta\n",
      "\t/tmp/tmpEfMJxM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpY9iFlg.lastz\n",
      "\t/tmp/tmpt1fqtA.lastz\n",
      "\t/tmp/tmpzCXumo.lastz\n",
      "\t/tmp/tmp8Dt8om.lastz\n",
      "\t/tmp/tmpCWdRw6.lastz\n",
      "\t/tmp/tmpWsvegU.lastz\n",
      "\t/tmp/tmpcpf9tN.lastz\n",
      "\t/tmp/tmp3d7tjR.lastz\n",
      "\t/tmp/tmp1ZHNex.lastz\n",
      "\t/tmp/tmpgqOudv.lastz\n",
      "\t/tmp/tmpJ8IISg.lastz\n",
      "\t/tmp/tmpyp0hv3.lastz\n",
      "\t/tmp/tmpecP26u.lastz\n",
      "\t/tmp/tmpZchi7b.lastz\n",
      "\t/tmp/tmpT84AjO.lastz\n",
      "\t/tmp/tmpSndJSf.lastz\n",
      "\t/tmp/tmpWOB0yJ.lastz\n",
      "\t/tmp/tmp2wuv3r.lastz\n",
      "\t/tmp/tmpvnodN8.lastz\n",
      "\t/tmp/tmprRWNYB.lastz\n",
      "\t/tmp/tmpT0bWW7.lastz\n",
      "\t/tmp/tmpe9qvUr.lastz\n",
      "\t/tmp/tmpfNrdcA.lastz\n",
      "\t/tmp/tmp4DF84L.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_fissidentatus_GCA_900490895.lastz\n",
      "Creating Pristionchus_fissidentatus_GCA_900490895 table\n",
      "Inserting data to Pristionchus_fissidentatus_GCA_900490895 table\n",
      "\n",
      "Running against Pristionchus_japonicus_GCA_900490845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 22 queries...\n",
      "\t/tmp/tmp3Sf2Tb.fasta\n",
      "\t/tmp/tmp4DfnKe.fasta\n",
      "\t/tmp/tmp4dOrZO.fasta\n",
      "\t/tmp/tmpYMseLG.fasta\n",
      "\t/tmp/tmp8ccj5t.fasta\n",
      "\t/tmp/tmpqvYHpe.fasta\n",
      "\t/tmp/tmp4dJQkS.fasta\n",
      "\t/tmp/tmpDWx11o.fasta\n",
      "\t/tmp/tmpyLSwJD.fasta\n",
      "\t/tmp/tmpskYupW.fasta\n",
      "\t/tmp/tmpmKSbjL.fasta\n",
      "\t/tmp/tmpovSYXT.fasta\n",
      "\t/tmp/tmphbCDMn.fasta\n",
      "\t/tmp/tmptuCjOc.fasta\n",
      "\t/tmp/tmpdPF0ew.fasta\n",
      "\t/tmp/tmppkr3NA.fasta\n",
      "\t/tmp/tmpye7rjm.fasta\n",
      "\t/tmp/tmpfQjKny.fasta\n",
      "\t/tmp/tmpmleJ4Y.fasta\n",
      "\t/tmp/tmpodjxfI.fasta\n",
      "\t/tmp/tmpSWeZDn.fasta\n",
      "\t/tmp/tmps5T_eI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnnsOdL.lastz\n",
      "\t/tmp/tmp41ZWJY.lastz\n",
      "\t/tmp/tmpnM4emO.lastz\n",
      "\t/tmp/tmp1gfTwK.lastz\n",
      "\t/tmp/tmpwWLM5a.lastz\n",
      "\t/tmp/tmpWsziXn.lastz\n",
      "\t/tmp/tmpemZKyK.lastz\n",
      "\t/tmp/tmpSUMKRS.lastz\n",
      "\t/tmp/tmpt1gtbn.lastz\n",
      "\t/tmp/tmp8Hx3uS.lastz\n",
      "\t/tmp/tmpOKaXrT.lastz\n",
      "\t/tmp/tmpWGda27.lastz\n",
      "\t/tmp/tmpLIqVbN.lastz\n",
      "\t/tmp/tmpvg_1dq.lastz\n",
      "\t/tmp/tmprvm3yH.lastz\n",
      "\t/tmp/tmp9cQAUL.lastz\n",
      "\t/tmp/tmpXG38q_.lastz\n",
      "\t/tmp/tmplLN7WD.lastz\n",
      "\t/tmp/tmpTvSXdk.lastz\n",
      "\t/tmp/tmp4gpyc0.lastz\n",
      "\t/tmp/tmp8WbXFg.lastz\n",
      "\t/tmp/tmpSVKB3w.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_japonicus_GCA_900490845.lastz\n",
      "Creating Pristionchus_japonicus_GCA_900490845 table\n",
      "Inserting data to Pristionchus_japonicus_GCA_900490845 table\n",
      "\n",
      "Running against Pristionchus_maxplancki_GCA_900490775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 27 queries...\n",
      "\t/tmp/tmpnFKicZ.fasta\n",
      "\t/tmp/tmp5I_PJB.fasta\n",
      "\t/tmp/tmpo795LW.fasta\n",
      "\t/tmp/tmp7o0FNl.fasta\n",
      "\t/tmp/tmpMbF03p.fasta\n",
      "\t/tmp/tmp2WevQW.fasta\n",
      "\t/tmp/tmpvdbrqL.fasta\n",
      "\t/tmp/tmpIpusUY.fasta\n",
      "\t/tmp/tmpl455C2.fasta\n",
      "\t/tmp/tmpIar3bi.fasta\n",
      "\t/tmp/tmpxXvMGN.fasta\n",
      "\t/tmp/tmpSS5KFF.fasta\n",
      "\t/tmp/tmpMOFiD8.fasta\n",
      "\t/tmp/tmpk0SiQw.fasta\n",
      "\t/tmp/tmpapIPdo.fasta\n",
      "\t/tmp/tmp8Aalqm.fasta\n",
      "\t/tmp/tmpkvuOn6.fasta\n",
      "\t/tmp/tmpSCpIMk.fasta\n",
      "\t/tmp/tmpVa8O7r.fasta\n",
      "\t/tmp/tmpNRYJQD.fasta\n",
      "\t/tmp/tmpz3LRy0.fasta\n",
      "\t/tmp/tmp9gG1r7.fasta\n",
      "\t/tmp/tmpolpS94.fasta\n",
      "\t/tmp/tmpy0C4og.fasta\n",
      "\t/tmp/tmpoD0IPm.fasta\n",
      "\t/tmp/tmpkSFJZO.fasta\n",
      "\t/tmp/tmpUx1Xk9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpe0kdrj.lastz\n",
      "\t/tmp/tmp3tRWfs.lastz\n",
      "\t/tmp/tmpV5Myd4.lastz\n",
      "\t/tmp/tmplzKcU9.lastz\n",
      "\t/tmp/tmpY35xuc.lastz\n",
      "\t/tmp/tmpGsJm9_.lastz\n",
      "\t/tmp/tmpWT404Q.lastz\n",
      "\t/tmp/tmpfiOtfF.lastz\n",
      "\t/tmp/tmpo8TbnF.lastz\n",
      "\t/tmp/tmpYw3fWd.lastz\n",
      "\t/tmp/tmpluaV9Y.lastz\n",
      "\t/tmp/tmpeKBVT3.lastz\n",
      "\t/tmp/tmpRWUNrG.lastz\n",
      "\t/tmp/tmpnLkPEi.lastz\n",
      "\t/tmp/tmpVOZFQM.lastz\n",
      "\t/tmp/tmpkKeHN9.lastz\n",
      "\t/tmp/tmpbiiI_N.lastz\n",
      "\t/tmp/tmp84PTnk.lastz\n",
      "\t/tmp/tmpG2H8q3.lastz\n",
      "\t/tmp/tmpnLhqYy.lastz\n",
      "\t/tmp/tmpA8bFiJ.lastz\n",
      "\t/tmp/tmpJK5Lu2.lastz\n",
      "\t/tmp/tmpAHtNQm.lastz\n",
      "\t/tmp/tmp4nKrNB.lastz\n",
      "\t/tmp/tmpFyL_0U.lastz\n",
      "\t/tmp/tmpb0R8er.lastz\n",
      "\t/tmp/tmpd1KmkW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_maxplancki_GCA_900490775.lastz\n",
      "Creating Pristionchus_maxplancki_GCA_900490775 table\n",
      "Inserting data to Pristionchus_maxplancki_GCA_900490775 table\n",
      "\n",
      "Running against Pristionchus_mayeri_GCA_900490875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 30 queries...\n",
      "\t/tmp/tmpGbZjHZ.fasta\n",
      "\t/tmp/tmpSOgySs.fasta\n",
      "\t/tmp/tmpWw1Nw2.fasta\n",
      "\t/tmp/tmpKqx6Ka.fasta\n",
      "\t/tmp/tmpopvDvM.fasta\n",
      "\t/tmp/tmpFFuITo.fasta\n",
      "\t/tmp/tmpnPOekv.fasta\n",
      "\t/tmp/tmpEIzWns.fasta\n",
      "\t/tmp/tmptpQq7z.fasta\n",
      "\t/tmp/tmpLisES8.fasta\n",
      "\t/tmp/tmp8K_jo7.fasta\n",
      "\t/tmp/tmpgRBfhx.fasta\n",
      "\t/tmp/tmpxBPFq5.fasta\n",
      "\t/tmp/tmpzKzH3f.fasta\n",
      "\t/tmp/tmpObfjTz.fasta\n",
      "\t/tmp/tmp4FFOGi.fasta\n",
      "\t/tmp/tmpeSZ8tT.fasta\n",
      "\t/tmp/tmpWa_aja.fasta\n",
      "\t/tmp/tmpzvU_fr.fasta\n",
      "\t/tmp/tmpE0Hyuz.fasta\n",
      "\t/tmp/tmpgq7Qgy.fasta\n",
      "\t/tmp/tmpx4HRg5.fasta\n",
      "\t/tmp/tmpUJnJw9.fasta\n",
      "\t/tmp/tmpgrvNkX.fasta\n",
      "\t/tmp/tmpLYfMpT.fasta\n",
      "\t/tmp/tmp7b5Awx.fasta\n",
      "\t/tmp/tmp7IQxHe.fasta\n",
      "\t/tmp/tmpj_GkmR.fasta\n",
      "\t/tmp/tmpJuSdo_.fasta\n",
      "\t/tmp/tmpVy0KGx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8jkXpK.lastz\n",
      "\t/tmp/tmpYBAyF4.lastz\n",
      "\t/tmp/tmpm1vpyB.lastz\n",
      "\t/tmp/tmpPSwQPV.lastz\n",
      "\t/tmp/tmpEWJG2i.lastz\n",
      "\t/tmp/tmpZisuMx.lastz\n",
      "\t/tmp/tmpCZwCVo.lastz\n",
      "\t/tmp/tmpcsOx7h.lastz\n",
      "\t/tmp/tmpTJV7uq.lastz\n",
      "\t/tmp/tmpididqQ.lastz\n",
      "\t/tmp/tmpSIyjc_.lastz\n",
      "\t/tmp/tmpXfVxt9.lastz\n",
      "\t/tmp/tmpgU88rl.lastz\n",
      "\t/tmp/tmpCYhYOd.lastz\n",
      "\t/tmp/tmpliZNDX.lastz\n",
      "\t/tmp/tmpCuQdsn.lastz\n",
      "\t/tmp/tmplwDUKC.lastz\n",
      "\t/tmp/tmpUTeOAm.lastz\n",
      "\t/tmp/tmpZ0L7VT.lastz\n",
      "\t/tmp/tmp6QxGDS.lastz\n",
      "\t/tmp/tmpODhktk.lastz\n",
      "\t/tmp/tmpCRug1z.lastz\n",
      "\t/tmp/tmpCXsbgL.lastz\n",
      "\t/tmp/tmp_xHuNL.lastz\n",
      "\t/tmp/tmp3xTX5Z.lastz\n",
      "\t/tmp/tmpfoKQPb.lastz\n",
      "\t/tmp/tmprOp_RV.lastz\n",
      "\t/tmp/tmpVNmVd7.lastz\n",
      "\t/tmp/tmpqtWwrj.lastz\n",
      "\t/tmp/tmplSyAUf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_mayeri_GCA_900490875.lastz\n",
      "Creating Pristionchus_mayeri_GCA_900490875 table\n",
      "Inserting data to Pristionchus_mayeri_GCA_900490875 table\n",
      "\n",
      "Running against Pristionchus_pacificus_GCA_000180635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 12 queries...\n",
      "\t/tmp/tmp85FEms.fasta\n",
      "\t/tmp/tmpHy5z_8.fasta\n",
      "\t/tmp/tmpXZRuAF.fasta\n",
      "\t/tmp/tmpVwYv3G.fasta\n",
      "\t/tmp/tmpBtXWaq.fasta\n",
      "\t/tmp/tmpy8pwKd.fasta\n",
      "\t/tmp/tmpRIT7jf.fasta\n",
      "\t/tmp/tmp7fD0TZ.fasta\n",
      "\t/tmp/tmpZygRUJ.fasta\n",
      "\t/tmp/tmpwv4yJK.fasta\n",
      "\t/tmp/tmp3Tp48I.fasta\n",
      "\t/tmp/tmphvjLZG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnUfp1I.lastz\n",
      "\t/tmp/tmpLLbWqA.lastz\n",
      "\t/tmp/tmpWZv1c3.lastz\n",
      "\t/tmp/tmpVd2149.lastz\n",
      "\t/tmp/tmp_GCsaP.lastz\n",
      "\t/tmp/tmpLZuwyn.lastz\n",
      "\t/tmp/tmpF8L7OY.lastz\n",
      "\t/tmp/tmpFTcVID.lastz\n",
      "\t/tmp/tmptC70qz.lastz\n",
      "\t/tmp/tmp86xJ7d.lastz\n",
      "\t/tmp/tmphsOpVa.lastz\n",
      "\t/tmp/tmpZ_gYCp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Pristionchus_pacificus_GCA_000180635.lastz\n",
      "Creating Pristionchus_pacificus_GCA_000180635 table\n",
      "Inserting data to Pristionchus_pacificus_GCA_000180635 table\n",
      "\n",
      "Running against Rhabditida_sp_GCA_004026265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpVuvSdh.fasta\n",
      "\t/tmp/tmpUVUOMr.fasta\n",
      "\t/tmp/tmpvn8Rwe.fasta\n",
      "\t/tmp/tmp_AOmGb.fasta\n",
      "\t/tmp/tmpXLVUlS.fasta\n",
      "\t/tmp/tmphiG9Ih.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHM0onY.lastz\n",
      "\t/tmp/tmpBjDRXR.lastz\n",
      "\t/tmp/tmpxTXzLc.lastz\n",
      "\t/tmp/tmp2Ffzi1.lastz\n",
      "\t/tmp/tmp95i7BS.lastz\n",
      "\t/tmp/tmpe6yhSr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Rhabditida_sp_GCA_004026265.lastz\n",
      "Creating Rhabditida_sp_GCA_004026265 table\n",
      "Inserting data to Rhabditida_sp_GCA_004026265 table\n",
      "\n",
      "Running against Strongylus_vulgaris_GCA_900624965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 30 queries...\n",
      "\t/tmp/tmptL0rBN.fasta\n",
      "\t/tmp/tmpdho8qK.fasta\n",
      "\t/tmp/tmplQayPy.fasta\n",
      "\t/tmp/tmpmtoBiM.fasta\n",
      "\t/tmp/tmpF1fTAq.fasta\n",
      "\t/tmp/tmpxpufc9.fasta\n",
      "\t/tmp/tmpdlawBH.fasta\n",
      "\t/tmp/tmp9i6niq.fasta\n",
      "\t/tmp/tmpXHTYPh.fasta\n",
      "\t/tmp/tmpNJtBBO.fasta\n",
      "\t/tmp/tmpfB75Wy.fasta\n",
      "\t/tmp/tmp_3l095.fasta\n",
      "\t/tmp/tmph1aJvg.fasta\n",
      "\t/tmp/tmpbA7_E_.fasta\n",
      "\t/tmp/tmpKhT9Jr.fasta\n",
      "\t/tmp/tmpiRiUl2.fasta\n",
      "\t/tmp/tmpEfgZDX.fasta\n",
      "\t/tmp/tmpipTbWz.fasta\n",
      "\t/tmp/tmploXGNK.fasta\n",
      "\t/tmp/tmp5AsU4s.fasta\n",
      "\t/tmp/tmpQor41Y.fasta\n",
      "\t/tmp/tmps6xZQa.fasta\n",
      "\t/tmp/tmpkwMtVo.fasta\n",
      "\t/tmp/tmpBdOxb7.fasta\n",
      "\t/tmp/tmpk0YcnO.fasta\n",
      "\t/tmp/tmpdfWoRc.fasta\n",
      "\t/tmp/tmpVf4aJi.fasta\n",
      "\t/tmp/tmp1sQRM9.fasta\n",
      "\t/tmp/tmpwt_VIs.fasta\n",
      "\t/tmp/tmpVVn5X4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqzkGVl.lastz\n",
      "\t/tmp/tmpClYXjA.lastz\n",
      "\t/tmp/tmpWlarHG.lastz\n",
      "\t/tmp/tmpRHUM19.lastz\n",
      "\t/tmp/tmpP3SRgN.lastz\n",
      "\t/tmp/tmp0lIs9U.lastz\n",
      "\t/tmp/tmp34ildn.lastz\n",
      "\t/tmp/tmpUMuSO7.lastz\n",
      "\t/tmp/tmpTaDDov.lastz\n",
      "\t/tmp/tmpYMV8lH.lastz\n",
      "\t/tmp/tmp9Ckzyb.lastz\n",
      "\t/tmp/tmpBXmBcV.lastz\n",
      "\t/tmp/tmpesgnhz.lastz\n",
      "\t/tmp/tmpa71vwQ.lastz\n",
      "\t/tmp/tmpxv3EI8.lastz\n",
      "\t/tmp/tmpiOOpeV.lastz\n",
      "\t/tmp/tmpokUgp3.lastz\n",
      "\t/tmp/tmpOInJCo.lastz\n",
      "\t/tmp/tmp1nHd87.lastz\n",
      "\t/tmp/tmpZF4nAp.lastz\n",
      "\t/tmp/tmpMKs93S.lastz\n",
      "\t/tmp/tmpgiPeIP.lastz\n",
      "\t/tmp/tmpKWiNm8.lastz\n",
      "\t/tmp/tmpPI_eA7.lastz\n",
      "\t/tmp/tmprNf2jv.lastz\n",
      "\t/tmp/tmpjhWF1P.lastz\n",
      "\t/tmp/tmphDin3a.lastz\n",
      "\t/tmp/tmpX40o2y.lastz\n",
      "\t/tmp/tmpzbKVLC.lastz\n",
      "\t/tmp/tmpBsHtkd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Strongylus_vulgaris_GCA_900624965.lastz\n",
      "Creating Strongylus_vulgaris_GCA_900624965 table\n",
      "Inserting data to Strongylus_vulgaris_GCA_900624965 table\n",
      "\n",
      "Running against Teladorsagia_circumcincta_GCA_002352805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 70 queries...\n",
      "\t/tmp/tmpUavGLY.fasta\n",
      "\t/tmp/tmpGE4ANP.fasta\n",
      "\t/tmp/tmpiMKCyw.fasta\n",
      "\t/tmp/tmpiu4Aci.fasta\n",
      "\t/tmp/tmpCqWWB5.fasta\n",
      "\t/tmp/tmpUOQAjU.fasta\n",
      "\t/tmp/tmpzspg8P.fasta\n",
      "\t/tmp/tmpfa1VPX.fasta\n",
      "\t/tmp/tmpSVNG0w.fasta\n",
      "\t/tmp/tmpgFvq6y.fasta\n",
      "\t/tmp/tmpC04jwP.fasta\n",
      "\t/tmp/tmpsFzoNz.fasta\n",
      "\t/tmp/tmpFhPmlB.fasta\n",
      "\t/tmp/tmp6NS0iL.fasta\n",
      "\t/tmp/tmp60l5JZ.fasta\n",
      "\t/tmp/tmpFBIj3Q.fasta\n",
      "\t/tmp/tmpPOyVL0.fasta\n",
      "\t/tmp/tmpznOIWJ.fasta\n",
      "\t/tmp/tmpNMayQq.fasta\n",
      "\t/tmp/tmpUT5r3g.fasta\n",
      "\t/tmp/tmp_F0WKv.fasta\n",
      "\t/tmp/tmpc_x1_3.fasta\n",
      "\t/tmp/tmpGlX0bW.fasta\n",
      "\t/tmp/tmp5yD7rT.fasta\n",
      "\t/tmp/tmpD9ungS.fasta\n",
      "\t/tmp/tmphtryof.fasta\n",
      "\t/tmp/tmpY7YSgr.fasta\n",
      "\t/tmp/tmpNKZBLC.fasta\n",
      "\t/tmp/tmpMBS_YR.fasta\n",
      "\t/tmp/tmpD2_DFL.fasta\n",
      "\t/tmp/tmpfGTkZV.fasta\n",
      "\t/tmp/tmp7WDRa8.fasta\n",
      "\t/tmp/tmph0JfuS.fasta\n",
      "\t/tmp/tmpcPD2dv.fasta\n",
      "\t/tmp/tmp8BooQb.fasta\n",
      "\t/tmp/tmpn96NKV.fasta\n",
      "\t/tmp/tmpnVjGav.fasta\n",
      "\t/tmp/tmpniiWri.fasta\n",
      "\t/tmp/tmp49j86P.fasta\n",
      "\t/tmp/tmpsbBslq.fasta\n",
      "\t/tmp/tmpOsvJMq.fasta\n",
      "\t/tmp/tmpT0tjUY.fasta\n",
      "\t/tmp/tmp9oV75n.fasta\n",
      "\t/tmp/tmpSAfpQA.fasta\n",
      "\t/tmp/tmpCH4Ib7.fasta\n",
      "\t/tmp/tmp4U4hSx.fasta\n",
      "\t/tmp/tmpPkf2QS.fasta\n",
      "\t/tmp/tmp9Dz2_c.fasta\n",
      "\t/tmp/tmp0FrnFu.fasta\n",
      "\t/tmp/tmpZSMQEm.fasta\n",
      "\t/tmp/tmp0D4uCS.fasta\n",
      "\t/tmp/tmpMa7D_b.fasta\n",
      "\t/tmp/tmpGbC8YO.fasta\n",
      "\t/tmp/tmpuoahbz.fasta\n",
      "\t/tmp/tmp8zDPcL.fasta\n",
      "\t/tmp/tmpAQHZYH.fasta\n",
      "\t/tmp/tmpaglsQ0.fasta\n",
      "\t/tmp/tmpvoyPRt.fasta\n",
      "\t/tmp/tmpkJF2vL.fasta\n",
      "\t/tmp/tmpETeZNG.fasta\n",
      "\t/tmp/tmpxpK1Fq.fasta\n",
      "\t/tmp/tmpZzVsMb.fasta\n",
      "\t/tmp/tmpLwRLvy.fasta\n",
      "\t/tmp/tmpvy87Dt.fasta\n",
      "\t/tmp/tmp4bKL87.fasta\n",
      "\t/tmp/tmpZhFXXw.fasta\n",
      "\t/tmp/tmpO2r1Mp.fasta\n",
      "\t/tmp/tmpskKwWA.fasta\n",
      "\t/tmp/tmpPHMGlp.fasta\n",
      "\t/tmp/tmpiN_Jcv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAQcDmN.lastz\n",
      "\t/tmp/tmp19nvzW.lastz\n",
      "\t/tmp/tmpBTbr08.lastz\n",
      "\t/tmp/tmpBMxaqk.lastz\n",
      "\t/tmp/tmpSeJCoq.lastz\n",
      "\t/tmp/tmpRHiJFW.lastz\n",
      "\t/tmp/tmpMJlQp2.lastz\n",
      "\t/tmp/tmpN5P5Kt.lastz\n",
      "\t/tmp/tmp2NnJEE.lastz\n",
      "\t/tmp/tmpMVW08_.lastz\n",
      "\t/tmp/tmpGbeCJ_.lastz\n",
      "\t/tmp/tmpaxAssa.lastz\n",
      "\t/tmp/tmpgQM5yP.lastz\n",
      "\t/tmp/tmpmLq57o.lastz\n",
      "\t/tmp/tmp_DQdY7.lastz\n",
      "\t/tmp/tmpw5dwxb.lastz\n",
      "\t/tmp/tmpFvW4Ay.lastz\n",
      "\t/tmp/tmpeL8Cyw.lastz\n",
      "\t/tmp/tmpoHMdvm.lastz\n",
      "\t/tmp/tmpRCxCjB.lastz\n",
      "\t/tmp/tmpklMv2L.lastz\n",
      "\t/tmp/tmplh4sqw.lastz\n",
      "\t/tmp/tmppd8bRX.lastz\n",
      "\t/tmp/tmpHQKwEw.lastz\n",
      "\t/tmp/tmpsTYaOl.lastz\n",
      "\t/tmp/tmpJ5mTz9.lastz\n",
      "\t/tmp/tmpBQX8jJ.lastz\n",
      "\t/tmp/tmp1797nS.lastz\n",
      "\t/tmp/tmpaQ61lM.lastz\n",
      "\t/tmp/tmpdjDJfn.lastz\n",
      "\t/tmp/tmpjs_O4S.lastz\n",
      "\t/tmp/tmpckmuv8.lastz\n",
      "\t/tmp/tmp_2Wm6N.lastz\n",
      "\t/tmp/tmpSqk53S.lastz\n",
      "\t/tmp/tmpf9TsQL.lastz\n",
      "\t/tmp/tmpePFsNr.lastz\n",
      "\t/tmp/tmpLtuYLs.lastz\n",
      "\t/tmp/tmpE8iMdH.lastz\n",
      "\t/tmp/tmpucroeR.lastz\n",
      "\t/tmp/tmpqmDr1E.lastz\n",
      "\t/tmp/tmpo0fY_R.lastz\n",
      "\t/tmp/tmpKaMlTo.lastz\n",
      "\t/tmp/tmpSxZX2F.lastz\n",
      "\t/tmp/tmpNQxb2K.lastz\n",
      "\t/tmp/tmpcZ08E4.lastz\n",
      "\t/tmp/tmp6ddTM_.lastz\n",
      "\t/tmp/tmprBmD4q.lastz\n",
      "\t/tmp/tmpb8a5WN.lastz\n",
      "\t/tmp/tmpjirc91.lastz\n",
      "\t/tmp/tmpVi7kFO.lastz\n",
      "\t/tmp/tmpPGlhFs.lastz\n",
      "\t/tmp/tmpeFDbER.lastz\n",
      "\t/tmp/tmpjfmk8q.lastz\n",
      "\t/tmp/tmpiaW5oA.lastz\n",
      "\t/tmp/tmp8py77n.lastz\n",
      "\t/tmp/tmpGe8URD.lastz\n",
      "\t/tmp/tmp3mbNCD.lastz\n",
      "\t/tmp/tmpHiXFzW.lastz\n",
      "\t/tmp/tmpfcIG5K.lastz\n",
      "\t/tmp/tmpawtSiq.lastz\n",
      "\t/tmp/tmpkXNPyT.lastz\n",
      "\t/tmp/tmpQ8cUim.lastz\n",
      "\t/tmp/tmpzuE08Q.lastz\n",
      "\t/tmp/tmpHNmfhV.lastz\n",
      "\t/tmp/tmpysP7vN.lastz\n",
      "\t/tmp/tmp4fwCJt.lastz\n",
      "\t/tmp/tmpiiIXiE.lastz\n",
      "\t/tmp/tmpTONyEZ.lastz\n",
      "\t/tmp/tmp0lRHMX.lastz\n",
      "\t/tmp/tmpvn4OFi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Teladorsagia_circumcincta_GCA_002352805.lastz\n",
      "Creating Teladorsagia_circumcincta_GCA_002352805 table\n",
      "Inserting data to Teladorsagia_circumcincta_GCA_002352805 table\n",
      "\n",
      "Running against Haemonchus_contortus_GCA_007637855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 6 queries...\n",
      "\t/tmp/tmpM5R5eH.fasta\n",
      "\t/tmp/tmpPANy8p.fasta\n",
      "\t/tmp/tmp3aPNRu.fasta\n",
      "\t/tmp/tmp7XZW7T.fasta\n",
      "\t/tmp/tmp5m5A0r.fasta\n",
      "\t/tmp/tmpA2ezoU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3dgsUP.lastz\n",
      "\t/tmp/tmpZPADYf.lastz\n",
      "\t/tmp/tmppK9a4E.lastz\n",
      "\t/tmp/tmpsaGsTT.lastz\n",
      "\t/tmp/tmppCEuXd.lastz\n",
      "\t/tmp/tmpdtWoc9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/nematoda-clade5/validate_intervals/lastz/Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_Haemonchus_contortus_GCA_007637855.lastz\n",
      "Creating Haemonchus_contortus_GCA_007637855 table\n",
      "Inserting data to Haemonchus_contortus_GCA_007637855 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/nematoda-clade5/extract_probes_from_group/nematoda-clade5_genome.conf --lastz results/phyluce/nematoda-clade5/validate_intervals/lastz --probes 120 --probe-prefix uce_nematoda-clade5_ --name-pattern \"Haemonchus_contortus_GCA_007637855_+1_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/nematoda-clade5/extract_probes_from_group/probe_fasta;\n",
      "2020-02-11 16:38:16,056 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-11 16:38:16,061 - Phyluce - INFO - ------ Working on Ancylostoma_caninum_GCA_003336725 genome ------\n",
      "2020-02-11 16:38:16,070 - Phyluce - INFO - Reading Ancylostoma_caninum_GCA_003336725 genome\n",
      "2020-02-11 16:38:49,995 - Phyluce - INFO - Ancylostoma_caninum_GCA_003336725: 2196 uces, 690 dupes, 1506 non-dupes, 3 orient drop, 15 length drop, 1488 written\n",
      "2020-02-11 16:38:49,995 - Phyluce - INFO - ----- Working on Ancylostoma_ceylanicum_GCA_000688135 genome ----\n",
      "2020-02-11 16:38:49,998 - Phyluce - INFO - Reading Ancylostoma_ceylanicum_GCA_000688135 genome\n",
      "2020-02-11 16:39:26,909 - Phyluce - INFO - Ancylostoma_ceylanicum_GCA_000688135: 2159 uces, 467 dupes, 1692 non-dupes, 2 orient drop, 12 length drop, 1678 written\n",
      "2020-02-11 16:39:26,909 - Phyluce - INFO - ----- Working on Ancylostoma_duodenale_GCA_000816745 genome -----\n",
      "2020-02-11 16:39:26,911 - Phyluce - INFO - Reading Ancylostoma_duodenale_GCA_000816745 genome\n",
      "2020-02-11 16:39:56,049 - Phyluce - INFO - Ancylostoma_duodenale_GCA_000816745: 2067 uces, 640 dupes, 1427 non-dupes, 2 orient drop, 15 length drop, 1406 written\n",
      "2020-02-11 16:39:56,049 - Phyluce - INFO - -- Working on Angiostrongylus_cantonensis_GCA_009735665 genome --\n",
      "2020-02-11 16:39:56,051 - Phyluce - INFO - Reading Angiostrongylus_cantonensis_GCA_009735665 genome\n",
      "2020-02-11 16:40:09,463 - Phyluce - INFO - Angiostrongylus_cantonensis_GCA_009735665: 1735 uces, 269 dupes, 1466 non-dupes, 3 orient drop, 37 length drop, 1426 written\n",
      "2020-02-11 16:40:09,463 - Phyluce - INFO - - Working on Angiostrongylus_costaricensis_GCA_900624975 genome -\n",
      "2020-02-11 16:40:09,477 - Phyluce - INFO - Reading Angiostrongylus_costaricensis_GCA_900624975 genome\n",
      "2020-02-11 16:40:21,749 - Phyluce - INFO - Angiostrongylus_costaricensis_GCA_900624975: 1738 uces, 246 dupes, 1492 non-dupes, 3 orient drop, 11 length drop, 1478 written\n",
      "2020-02-11 16:40:21,749 - Phyluce - INFO - ----- Working on Caenorhabditis_angaria_GCA_000165025 genome ----\n",
      "2020-02-11 16:40:21,759 - Phyluce - INFO - Reading Caenorhabditis_angaria_GCA_000165025 genome\n",
      "2020-02-11 16:40:27,118 - Phyluce - INFO - Caenorhabditis_angaria_GCA_000165025: 646 uces, 164 dupes, 482 non-dupes, 1 orient drop, 24 length drop, 457 written\n",
      "2020-02-11 16:40:27,118 - Phyluce - INFO - ------ Working on Caenorhabditis_becei_GCA_900536315 genome -----\n",
      "2020-02-11 16:40:27,119 - Phyluce - INFO - Reading Caenorhabditis_becei_GCA_900536315 genome\n",
      "2020-02-11 16:40:32,442 - Phyluce - INFO - Caenorhabditis_becei_GCA_900536315: 689 uces, 213 dupes, 476 non-dupes, 3 orient drop, 7 length drop, 466 written\n",
      "2020-02-11 16:40:32,443 - Phyluce - INFO - ------ Working on Caenorhabditis_bovis_GCA_902636285 genome -----\n",
      "2020-02-11 16:40:32,444 - Phyluce - INFO - Reading Caenorhabditis_bovis_GCA_902636285 genome\n",
      "2020-02-11 16:40:38,594 - Phyluce - INFO - Caenorhabditis_bovis_GCA_902636285: 742 uces, 197 dupes, 545 non-dupes, 9 orient drop, 16 length drop, 520 written\n",
      "2020-02-11 16:40:38,594 - Phyluce - INFO - ---- Working on Caenorhabditis_brenneri_GCA_000143925 genome ----\n",
      "2020-02-11 16:40:38,601 - Phyluce - INFO - Reading Caenorhabditis_brenneri_GCA_000143925 genome\n",
      "2020-02-11 16:40:47,539 - Phyluce - INFO - Caenorhabditis_brenneri_GCA_000143925: 748 uces, 414 dupes, 334 non-dupes, 4 orient drop, 15 length drop, 315 written\n",
      "2020-02-11 16:40:47,539 - Phyluce - INFO - ---- Working on Caenorhabditis_briggsae_GCA_000004555 genome ----\n",
      "2020-02-11 16:40:47,541 - Phyluce - INFO - Reading Caenorhabditis_briggsae_GCA_000004555 genome\n",
      "2020-02-11 16:40:54,241 - Phyluce - INFO - Caenorhabditis_briggsae_GCA_000004555: 721 uces, 223 dupes, 498 non-dupes, 15 orient drop, 19 length drop, 464 written\n",
      "2020-02-11 16:40:54,242 - Phyluce - INFO - ----- Working on Caenorhabditis_elegans_GCF_000002985 genome ----\n",
      "2020-02-11 16:40:54,248 - Phyluce - INFO - Reading Caenorhabditis_elegans_GCF_000002985 genome\n",
      "2020-02-11 16:40:59,998 - Phyluce - INFO - Caenorhabditis_elegans_GCF_000002985: 698 uces, 172 dupes, 526 non-dupes, 17 orient drop, 22 length drop, 487 written\n",
      "2020-02-11 16:40:59,998 - Phyluce - INFO - ---- Working on Caenorhabditis_inopinata_GCA_003052745 genome ---\n",
      "2020-02-11 16:41:00,003 - Phyluce - INFO - Reading Caenorhabditis_inopinata_GCA_003052745 genome\n",
      "2020-02-11 16:41:10,462 - Phyluce - INFO - Caenorhabditis_inopinata_GCA_003052745: 770 uces, 234 dupes, 536 non-dupes, 27 orient drop, 14 length drop, 495 written\n",
      "2020-02-11 16:41:10,462 - Phyluce - INFO - ---- Working on Caenorhabditis_japonica_GCA_000147155 genome ----\n",
      "2020-02-11 16:41:10,463 - Phyluce - INFO - Reading Caenorhabditis_japonica_GCA_000147155 genome\n",
      "2020-02-11 16:41:23,450 - Phyluce - INFO - Caenorhabditis_japonica_GCA_000147155: 775 uces, 325 dupes, 450 non-dupes, 1 orient drop, 8 length drop, 441 written\n",
      "2020-02-11 16:41:23,450 - Phyluce - INFO - ----- Working on Caenorhabditis_latens_GCA_002259235 genome -----\n",
      "2020-02-11 16:41:23,451 - Phyluce - INFO - Reading Caenorhabditis_latens_GCA_002259235 genome\n",
      "2020-02-11 16:41:30,176 - Phyluce - INFO - Caenorhabditis_latens_GCA_002259235: 721 uces, 250 dupes, 471 non-dupes, 1 orient drop, 10 length drop, 460 written\n",
      "2020-02-11 16:41:30,177 - Phyluce - INFO - ----- Working on Caenorhabditis_nigoni_GCA_002742825 genome -----\n",
      "2020-02-11 16:41:30,234 - Phyluce - INFO - Reading Caenorhabditis_nigoni_GCA_002742825 genome\n",
      "2020-02-11 16:41:39,809 - Phyluce - INFO - Caenorhabditis_nigoni_GCA_002742825: 735 uces, 247 dupes, 488 non-dupes, 15 orient drop, 22 length drop, 451 written\n",
      "2020-02-11 16:41:39,809 - Phyluce - INFO - --- Working on Caenorhabditis_panamensis_GCA_900536275 genome ---\n",
      "2020-02-11 16:41:39,810 - Phyluce - INFO - Reading Caenorhabditis_panamensis_GCA_900536275 genome\n",
      "2020-02-11 16:41:45,628 - Phyluce - INFO - Caenorhabditis_panamensis_GCA_900536275: 704 uces, 206 dupes, 498 non-dupes, 4 orient drop, 8 length drop, 486 written\n",
      "2020-02-11 16:41:45,628 - Phyluce - INFO - ----- Working on Caenorhabditis_remanei_GCF_000149515 genome ----\n",
      "2020-02-11 16:41:45,629 - Phyluce - INFO - Reading Caenorhabditis_remanei_GCF_000149515 genome\n",
      "2020-02-11 16:41:53,518 - Phyluce - INFO - Caenorhabditis_remanei_GCF_000149515: 750 uces, 305 dupes, 445 non-dupes, 5 orient drop, 12 length drop, 428 written\n",
      "2020-02-11 16:41:53,518 - Phyluce - INFO - ------- Working on Caenorhabditis_sp_GCA_900536305 genome -------\n",
      "2020-02-11 16:41:53,519 - Phyluce - INFO - Reading Caenorhabditis_sp_GCA_900536305 genome\n",
      "2020-02-11 16:42:00,675 - Phyluce - INFO - Caenorhabditis_sp_GCA_900536305: 723 uces, 234 dupes, 489 non-dupes, 4 orient drop, 9 length drop, 476 written\n",
      "2020-02-11 16:42:00,676 - Phyluce - INFO - --- Working on Caenorhabditis_tropicalis_GCA_000186765 genome ---\n",
      "2020-02-11 16:42:00,724 - Phyluce - INFO - Reading Caenorhabditis_tropicalis_GCA_000186765 genome\n",
      "2020-02-11 16:42:05,807 - Phyluce - INFO - Caenorhabditis_tropicalis_GCA_000186765: 692 uces, 167 dupes, 525 non-dupes, 19 orient drop, 4 length drop, 502 written\n",
      "2020-02-11 16:42:05,807 - Phyluce - INFO - ----- Working on Cylicostephanus_goldi_GCA_900617965 genome -----\n",
      "2020-02-11 16:42:05,808 - Phyluce - INFO - Reading Cylicostephanus_goldi_GCA_900617965 genome\n",
      "2020-02-11 16:42:14,915 - Phyluce - INFO - Cylicostephanus_goldi_GCA_900617965: 1350 uces, 286 dupes, 1064 non-dupes, 0 orient drop, 1 length drop, 1058 written\n",
      "2020-02-11 16:42:14,915 - Phyluce - INFO - ----- Working on Dictyocaulus_viviparus_GCA_000816705 genome ----\n",
      "2020-02-11 16:42:14,957 - Phyluce - INFO - Reading Dictyocaulus_viviparus_GCA_000816705 genome\n",
      "2020-02-11 16:42:25,544 - Phyluce - INFO - Dictyocaulus_viviparus_GCA_000816705: 1609 uces, 207 dupes, 1402 non-dupes, 4 orient drop, 14 length drop, 1384 written\n",
      "2020-02-11 16:42:25,545 - Phyluce - INFO - ----- Working on Diploscapter_coronatus_GCA_002207785 genome ----\n",
      "2020-02-11 16:42:25,567 - Phyluce - INFO - Reading Diploscapter_coronatus_GCA_002207785 genome\n",
      "2020-02-11 16:42:34,067 - Phyluce - INFO - Diploscapter_coronatus_GCA_002207785: 685 uces, 610 dupes, 75 non-dupes, 0 orient drop, 5 length drop, 70 written\n",
      "2020-02-11 16:42:34,067 - Phyluce - INFO - ------ Working on Diploscapter_pachys_GCA_002287525 genome ------\n",
      "2020-02-11 16:42:34,068 - Phyluce - INFO - Reading Diploscapter_pachys_GCA_002287525 genome\n",
      "2020-02-11 16:42:40,082 - Phyluce - INFO - Diploscapter_pachys_GCA_002287525: 656 uces, 483 dupes, 173 non-dupes, 0 orient drop, 1 length drop, 172 written\n",
      "2020-02-11 16:42:40,082 - Phyluce - INFO - ------- Working on Haemonchus_placei_GCA_900617895 genome -------\n",
      "2020-02-11 16:42:40,083 - Phyluce - INFO - Reading Haemonchus_placei_GCA_900617895 genome\n",
      "2020-02-11 16:43:18,467 - Phyluce - INFO - Haemonchus_placei_GCA_900617895: 2679 uces, 641 dupes, 2038 non-dupes, 3 orient drop, 11 length drop, 2024 written\n",
      "2020-02-11 16:43:18,467 - Phyluce - INFO - --- Working on Heligmosomoides_polygyrus_GCA_900096555 genome ---\n",
      "2020-02-11 16:43:18,468 - Phyluce - INFO - Reading Heligmosomoides_polygyrus_GCA_900096555 genome\n",
      "2020-02-11 16:44:25,794 - Phyluce - INFO - Heligmosomoides_polygyrus_GCA_900096555: 2387 uces, 583 dupes, 1804 non-dupes, 18 orient drop, 35 length drop, 1751 written\n",
      "2020-02-11 16:44:25,795 - Phyluce - INFO - - Working on Heterorhabditis_bacteriophora_GCA_000223415 genome -\n",
      "2020-02-11 16:44:25,796 - Phyluce - INFO - Reading Heterorhabditis_bacteriophora_GCA_000223415 genome\n",
      "2020-02-11 16:44:35,198 - Phyluce - INFO - Heterorhabditis_bacteriophora_GCA_000223415: 1146 uces, 248 dupes, 898 non-dupes, 6 orient drop, 5 length drop, 887 written\n",
      "2020-02-11 16:44:35,199 - Phyluce - INFO - ------ Working on Mesorhabditis_belari_GCA_900631915 genome -----\n",
      "2020-02-11 16:44:35,200 - Phyluce - INFO - Reading Mesorhabditis_belari_GCA_900631915 genome\n",
      "2020-02-11 16:44:43,851 - Phyluce - INFO - Mesorhabditis_belari_GCA_900631915: 571 uces, 242 dupes, 329 non-dupes, 3 orient drop, 9 length drop, 317 written\n",
      "2020-02-11 16:44:43,851 - Phyluce - INFO - ------ Working on Micoletzkya_japonica_GCA_900490955 genome -----\n",
      "2020-02-11 16:44:43,852 - Phyluce - INFO - Reading Micoletzkya_japonica_GCA_900490955 genome\n",
      "2020-02-11 16:44:54,428 - Phyluce - INFO - Micoletzkya_japonica_GCA_900490955: 535 uces, 198 dupes, 337 non-dupes, 1 orient drop, 7 length drop, 329 written\n",
      "2020-02-11 16:44:54,428 - Phyluce - INFO - ------- Working on Necator_americanus_GCF_000507365 genome ------\n",
      "2020-02-11 16:44:54,434 - Phyluce - INFO - Reading Necator_americanus_GCF_000507365 genome\n",
      "2020-02-11 16:45:14,026 - Phyluce - INFO - Necator_americanus_GCF_000507365: 1932 uces, 351 dupes, 1581 non-dupes, 3 orient drop, 11 length drop, 1566 written\n",
      "2020-02-11 16:45:14,026 - Phyluce - INFO - -- Working on Nippostrongylus_brasiliensis_GCA_900200055 genome -\n",
      "2020-02-11 16:45:14,041 - Phyluce - INFO - Reading Nippostrongylus_brasiliensis_GCA_900200055 genome\n",
      "2020-02-11 16:45:48,333 - Phyluce - INFO - Nippostrongylus_brasiliensis_GCA_900200055: 2162 uces, 909 dupes, 1253 non-dupes, 6 orient drop, 8 length drop, 1238 written\n",
      "2020-02-11 16:45:48,333 - Phyluce - INFO - ---- Working on Oesophagostomum_dentatum_GCA_000797555 genome ---\n",
      "2020-02-11 16:45:48,337 - Phyluce - INFO - Reading Oesophagostomum_dentatum_GCA_000797555 genome\n",
      "2020-02-11 16:46:06,311 - Phyluce - INFO - Oesophagostomum_dentatum_GCA_000797555: 1778 uces, 494 dupes, 1284 non-dupes, 2 orient drop, 15 length drop, 1267 written\n",
      "2020-02-11 16:46:06,312 - Phyluce - INFO - ---------- Working on Oscheius_sp_GCA_000934875 genome ----------\n",
      "2020-02-11 16:46:06,324 - Phyluce - INFO - Reading Oscheius_sp_GCA_000934875 genome\n",
      "2020-02-11 16:46:08,467 - Phyluce - INFO - Oscheius_sp_GCA_000934875: 308 uces, 109 dupes, 199 non-dupes, 2 orient drop, 0 length drop, 197 written\n",
      "2020-02-11 16:46:08,467 - Phyluce - INFO - -------- Working on Oscheius_tipulae_GCA_900184235 genome -------\n",
      "2020-02-11 16:46:08,468 - Phyluce - INFO - Reading Oscheius_tipulae_GCA_900184235 genome\n",
      "2020-02-11 16:46:14,523 - Phyluce - INFO - Oscheius_tipulae_GCA_900184235: 796 uces, 209 dupes, 587 non-dupes, 7 orient drop, 18 length drop, 562 written\n",
      "2020-02-11 16:46:14,523 - Phyluce - INFO - - Working on Parapristionchus_giblindavisi_GCA_900491355 genome -\n",
      "2020-02-11 16:46:14,527 - Phyluce - INFO - Reading Parapristionchus_giblindavisi_GCA_900491355 genome\n",
      "2020-02-11 16:46:21,049 - Phyluce - INFO - Parapristionchus_giblindavisi_GCA_900491355: 507 uces, 167 dupes, 340 non-dupes, 2 orient drop, 5 length drop, 333 written\n",
      "2020-02-11 16:46:21,049 - Phyluce - INFO - ------ Working on Pristionchus_arcanus_GCA_900490705 genome -----\n",
      "2020-02-11 16:46:21,050 - Phyluce - INFO - Reading Pristionchus_arcanus_GCA_900490705 genome\n",
      "2020-02-11 16:46:25,422 - Phyluce - INFO - Pristionchus_arcanus_GCA_900490705: 496 uces, 190 dupes, 306 non-dupes, 4 orient drop, 9 length drop, 292 written\n",
      "2020-02-11 16:46:25,422 - Phyluce - INFO - --- Working on Pristionchus_entomophagus_GCA_900490825 genome ---\n",
      "2020-02-11 16:46:25,430 - Phyluce - INFO - Reading Pristionchus_entomophagus_GCA_900490825 genome\n",
      "2020-02-11 16:46:32,960 - Phyluce - INFO - Pristionchus_entomophagus_GCA_900490825: 461 uces, 197 dupes, 264 non-dupes, 7 orient drop, 4 length drop, 253 written\n",
      "2020-02-11 16:46:32,961 - Phyluce - INFO - ---- Working on Pristionchus_exspctatus_GCA_900380275 genome ----\n",
      "2020-02-11 16:46:32,973 - Phyluce - INFO - Reading Pristionchus_exspctatus_GCA_900380275 genome\n",
      "2020-02-11 16:46:36,247 - Phyluce - INFO - Pristionchus_exspctatus_GCA_900380275: 458 uces, 172 dupes, 286 non-dupes, 1 orient drop, 12 length drop, 273 written\n",
      "2020-02-11 16:46:36,247 - Phyluce - INFO - --- Working on Pristionchus_fissidentatus_GCA_900490895 genome --\n",
      "2020-02-11 16:46:36,254 - Phyluce - INFO - Reading Pristionchus_fissidentatus_GCA_900490895 genome\n",
      "2020-02-11 16:46:44,073 - Phyluce - INFO - Pristionchus_fissidentatus_GCA_900490895: 511 uces, 170 dupes, 341 non-dupes, 6 orient drop, 4 length drop, 331 written\n",
      "2020-02-11 16:46:44,073 - Phyluce - INFO - ----- Working on Pristionchus_japonicus_GCA_900490845 genome ----\n",
      "2020-02-11 16:46:44,074 - Phyluce - INFO - Reading Pristionchus_japonicus_GCA_900490845 genome\n",
      "2020-02-11 16:46:49,813 - Phyluce - INFO - Pristionchus_japonicus_GCA_900490845: 485 uces, 199 dupes, 286 non-dupes, 3 orient drop, 10 length drop, 273 written\n",
      "2020-02-11 16:46:49,813 - Phyluce - INFO - ---- Working on Pristionchus_maxplancki_GCA_900490775 genome ----\n",
      "2020-02-11 16:46:49,821 - Phyluce - INFO - Reading Pristionchus_maxplancki_GCA_900490775 genome\n",
      "2020-02-11 16:46:58,135 - Phyluce - INFO - Pristionchus_maxplancki_GCA_900490775: 502 uces, 226 dupes, 276 non-dupes, 5 orient drop, 9 length drop, 262 written\n",
      "2020-02-11 16:46:58,136 - Phyluce - INFO - ------ Working on Pristionchus_mayeri_GCA_900490875 genome ------\n",
      "2020-02-11 16:46:58,141 - Phyluce - INFO - Reading Pristionchus_mayeri_GCA_900490875 genome\n",
      "2020-02-11 16:47:07,668 - Phyluce - INFO - Pristionchus_mayeri_GCA_900490875: 504 uces, 226 dupes, 278 non-dupes, 3 orient drop, 7 length drop, 268 written\n",
      "2020-02-11 16:47:07,668 - Phyluce - INFO - ----- Working on Pristionchus_pacificus_GCA_000180635 genome ----\n",
      "2020-02-11 16:47:07,669 - Phyluce - INFO - Reading Pristionchus_pacificus_GCA_000180635 genome\n",
      "2020-02-11 16:47:13,434 - Phyluce - INFO - Pristionchus_pacificus_GCA_000180635: 492 uces, 176 dupes, 316 non-dupes, 5 orient drop, 10 length drop, 301 written\n",
      "2020-02-11 16:47:13,435 - Phyluce - INFO - --------- Working on Rhabditida_sp_GCA_004026265 genome ---------\n",
      "2020-02-11 16:47:13,441 - Phyluce - INFO - Reading Rhabditida_sp_GCA_004026265 genome\n",
      "2020-02-11 16:47:18,261 - Phyluce - INFO - Rhabditida_sp_GCA_004026265: 667 uces, 183 dupes, 484 non-dupes, 0 orient drop, 1 length drop, 482 written\n",
      "2020-02-11 16:47:18,262 - Phyluce - INFO - ------ Working on Strongylus_vulgaris_GCA_900624965 genome ------\n",
      "2020-02-11 16:47:18,268 - Phyluce - INFO - Reading Strongylus_vulgaris_GCA_900624965 genome\n",
      "2020-02-11 16:47:30,442 - Phyluce - INFO - Strongylus_vulgaris_GCA_900624965: 1626 uces, 353 dupes, 1273 non-dupes, 0 orient drop, 2 length drop, 1266 written\n",
      "2020-02-11 16:47:30,443 - Phyluce - INFO - --- Working on Teladorsagia_circumcincta_GCA_002352805 genome ---\n",
      "2020-02-11 16:47:30,465 - Phyluce - INFO - Reading Teladorsagia_circumcincta_GCA_002352805 genome\n",
      "2020-02-11 16:48:39,320 - Phyluce - INFO - Teladorsagia_circumcincta_GCA_002352805: 2665 uces, 1161 dupes, 1504 non-dupes, 2 orient drop, 20 length drop, 1482 written\n",
      "2020-02-11 16:48:39,320 - Phyluce - INFO - ------ Working on Haemonchus_contortus_GCA_007637855 genome -----\n",
      "2020-02-11 16:48:39,325 - Phyluce - INFO - Reading Haemonchus_contortus_GCA_007637855 genome\n",
      "2020-02-11 16:49:20,390 - Phyluce - INFO - Haemonchus_contortus_GCA_007637855: 2707 uces, 599 dupes, 2108 non-dupes, 22 orient drop, 381 length drop, 1705 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/nematoda-clade5/extract_probes_from_group/probe_fasta --output results/phyluce/nematoda-clade5/extract_probes_from_group/multifastas.sqlite --base-taxon Haemonchus_contortus_GCA_007637855;\n",
      "ancylostoma_caninum_gca_003336725..\n",
      "ancylostoma_ceylanicum_gca_000688135..\n",
      "ancylostoma_duodenale_gca_000816745..\n",
      "angiostrongylus_cantonensis_gca_009735665..\n",
      "angiostrongylus_costaricensis_gca_900624975..\n",
      "caenorhabditis_angaria_gca_000165025.\n",
      "caenorhabditis_becei_gca_900536315.\n",
      "caenorhabditis_bovis_gca_902636285.\n",
      "caenorhabditis_brenneri_gca_000143925.\n",
      "caenorhabditis_briggsae_gca_000004555.\n",
      "caenorhabditis_elegans_gcf_000002985.\n",
      "caenorhabditis_inopinata_gca_003052745.\n",
      "caenorhabditis_japonica_gca_000147155.\n",
      "caenorhabditis_latens_gca_002259235.\n",
      "caenorhabditis_nigoni_gca_002742825.\n",
      "caenorhabditis_panamensis_gca_900536275.\n",
      "caenorhabditis_remanei_gcf_000149515.\n",
      "caenorhabditis_sp_gca_900536305.\n",
      "caenorhabditis_tropicalis_gca_000186765.\n",
      "cylicostephanus_goldi_gca_900617965..\n",
      "dictyocaulus_viviparus_gca_000816705..\n",
      "diploscapter_coronatus_gca_002207785.\n",
      "diploscapter_pachys_gca_002287525.\n",
      "haemonchus_placei_gca_900617895...\n",
      "heligmosomoides_polygyrus_gca_900096555..\n",
      "heterorhabditis_bacteriophora_gca_000223415.\n",
      "mesorhabditis_belari_gca_900631915.\n",
      "micoletzkya_japonica_gca_900490955.\n",
      "necator_americanus_gcf_000507365..\n",
      "nippostrongylus_brasiliensis_gca_900200055..\n",
      "oesophagostomum_dentatum_gca_000797555..\n",
      "oscheius_sp_gca_000934875.\n",
      "oscheius_tipulae_gca_900184235.\n",
      "parapristionchus_giblindavisi_gca_900491355.\n",
      "pristionchus_arcanus_gca_900490705.\n",
      "pristionchus_entomophagus_gca_900490825.\n",
      "pristionchus_exspctatus_gca_900380275.\n",
      "pristionchus_fissidentatus_gca_900490895.\n",
      "pristionchus_japonicus_gca_900490845.\n",
      "pristionchus_maxplancki_gca_900490775.\n",
      "pristionchus_mayeri_gca_900490875.\n",
      "pristionchus_pacificus_gca_000180635.\n",
      "rhabditida_sp_gca_004026265.\n",
      "strongylus_vulgaris_gca_900624965..\n",
      "teladorsagia_circumcincta_gca_002352805..\n",
      "haemonchus_contortus_gca_007637855..\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade5/extract_probes_from_group/multifastas.sqlite --base-taxon Haemonchus_contortus_GCA_007637855;\n",
      "Loci shared by 0 taxa:\t2,529.0\n",
      "Loci shared by 1 taxa:\t2,529.0\n",
      "Loci shared by 2 taxa:\t2,408.0\n",
      "Loci shared by 3 taxa:\t2,276.0\n",
      "Loci shared by 4 taxa:\t2,104.0\n",
      "Loci shared by 5 taxa:\t1,962.0\n",
      "Loci shared by 6 taxa:\t1,878.0\n",
      "Loci shared by 7 taxa:\t1,819.0\n",
      "Loci shared by 8 taxa:\t1,755.0\n",
      "Loci shared by 9 taxa:\t1,679.0\n",
      "Loci shared by 10 taxa:\t1,575.0\n",
      "Loci shared by 11 taxa:\t1,454.0\n",
      "Loci shared by 12 taxa:\t1,323.0\n",
      "Loci shared by 13 taxa:\t1,154.0\n",
      "Loci shared by 14 taxa:\t991.0\n",
      "Loci shared by 15 taxa:\t819.0\n",
      "Loci shared by 16 taxa:\t695.0\n",
      "Loci shared by 17 taxa:\t619.0\n",
      "Loci shared by 18 taxa:\t593.0\n",
      "Loci shared by 19 taxa:\t564.0\n",
      "Loci shared by 20 taxa:\t544.0\n",
      "Loci shared by 21 taxa:\t528.0\n",
      "Loci shared by 22 taxa:\t513.0\n",
      "Loci shared by 23 taxa:\t493.0\n",
      "Loci shared by 24 taxa:\t474.0\n",
      "Loci shared by 25 taxa:\t437.0\n",
      "Loci shared by 26 taxa:\t408.0\n",
      "Loci shared by 27 taxa:\t370.0\n",
      "Loci shared by 28 taxa:\t346.0\n",
      "Loci shared by 29 taxa:\t314.0\n",
      "Loci shared by 30 taxa:\t289.0\n",
      "Loci shared by 31 taxa:\t258.0\n",
      "Loci shared by 32 taxa:\t235.0\n",
      "Loci shared by 33 taxa:\t202.0\n",
      "Loci shared by 34 taxa:\t167.0\n",
      "Loci shared by 35 taxa:\t146.0\n",
      "Loci shared by 36 taxa:\t104.0\n",
      "Loci shared by 37 taxa:\t81.0\n",
      "Loci shared by 38 taxa:\t60.0\n",
      "Loci shared by 39 taxa:\t44.0\n",
      "Loci shared by 40 taxa:\t31.0\n",
      "Loci shared by 41 taxa:\t17.0\n",
      "Loci shared by 42 taxa:\t8.0\n",
      "Loci shared by 43 taxa:\t1.0\n",
      "Loci shared by 44 taxa:\t0.0\n",
      "Loci shared by 45 taxa:\t0.0\n",
      "Loci shared by 46 taxa:\t0.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/nematoda-clade5/extract_probes_from_group/multifastas.sqlite  --base-taxon Haemonchus_contortus_GCA_007637855 --output results/phyluce/nematoda-clade5/extract_probes_from_group/Haemonchus_contortus_GCA_007637855+1-back-to-38.conf --specific-counts 38;\n",
      "Counter({'caenorhabditis_nigoni_gca_002742825': 60, 'caenorhabditis_panamensis_gca_900536275': 60, 'dictyocaulus_viviparus_gca_000816705': 60, 'haemonchus_placei_gca_900617895': 60, 'caenorhabditis_bovis_gca_902636285': 60, 'caenorhabditis_inopinata_gca_003052745': 60, 'ancylostoma_ceylanicum_gca_000688135': 60, 'necator_americanus_gcf_000507365': 59, 'angiostrongylus_costaricensis_gca_900624975': 59, 'heterorhabditis_bacteriophora_gca_000223415': 59, 'caenorhabditis_tropicalis_gca_000186765': 59, 'caenorhabditis_becei_gca_900536315': 58, 'caenorhabditis_latens_gca_002259235': 58, 'heligmosomoides_polygyrus_gca_900096555': 58, 'oscheius_tipulae_gca_900184235': 58, 'caenorhabditis_sp_gca_900536305': 58, 'caenorhabditis_briggsae_gca_000004555': 58, 'pristionchus_pacificus_gca_000180635': 57, 'caenorhabditis_elegans_gcf_000002985': 57, 'angiostrongylus_cantonensis_gca_009735665': 57, 'caenorhabditis_remanei_gcf_000149515': 56, 'pristionchus_arcanus_gca_900490705': 56, 'ancylostoma_caninum_gca_003336725': 56, 'caenorhabditis_japonica_gca_000147155': 55, 'pristionchus_fissidentatus_gca_900490895': 55, 'haemonchus_contortus_gca_007637855': 54, 'pristionchus_mayeri_gca_900490875': 54, 'caenorhabditis_angaria_gca_000165025': 53, 'micoletzkya_japonica_gca_900490955': 53, 'pristionchus_japonicus_gca_900490845': 53, 'cylicostephanus_goldi_gca_900617965': 52, 'strongylus_vulgaris_gca_900624965': 51, 'pristionchus_maxplancki_gca_900490775': 51, 'oesophagostomum_dentatum_gca_000797555': 50, 'pristionchus_exspctatus_gca_900380275': 50, 'ancylostoma_duodenale_gca_000816745': 50, 'parapristionchus_giblindavisi_gca_900491355': 49, 'nippostrongylus_brasiliensis_gca_900200055': 48, 'teladorsagia_circumcincta_gca_002352805': 48, 'rhabditida_sp_gca_004026265': 48, 'pristionchus_entomophagus_gca_900490825': 46, 'caenorhabditis_brenneri_gca_000143925': 45, 'mesorhabditis_belari_gca_900631915': 40, 'diploscapter_pachys_gca_002287525': 21, 'oscheius_sp_gca_000934875': 10, 'diploscapter_coronatus_gca_002207785': 2})\n",
      "Total loci = 60\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(38)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/nematoda-clade5/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/nematoda-clade5/extract_probes_from_group/Haemonchus_contortus_GCA_007637855+1-back-to-38.conf --probe-prefix uce_nematoda-clade5_ --designer rnplattii --design nematoda-clade5_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list.fasta;\n",
      "NGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 60\n",
      "Probe Count = 4725\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list.fasta --query results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Tue Feb 11, 2020  16:50:40\n",
      "Ended:  Tue Feb 11, 2020  16:50:55\n",
      "Time for execution:  0.256766732534 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list.fasta --lastz results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_nematoda-clade5_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 4724 fasta sequences.  Filtered 0 duplicates. Kept 4725.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/nematoda-clade5/final_probe_design/nematoda-clade5_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Tue Feb 11 16:56:43 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 4725\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 378000\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3956965\n",
      "Max number of word counting entries: 89071294\n",
      "\n",
      "# comparing sequences from          0  to        787\n",
      "---------- new table with      615 representatives\n",
      "# comparing sequences from        787  to       1443\n",
      "----------    131 remaining sequences to the next cycle\n",
      "---------- new table with      454 representatives\n",
      "# comparing sequences from       1312  to       1880\n",
      "----------    196 remaining sequences to the next cycle\n",
      "---------- new table with      315 representatives\n",
      "# comparing sequences from       1684  to       2190\n",
      "----------     25 remaining sequences to the next cycle\n",
      "---------- new table with      424 representatives\n",
      "# comparing sequences from       2165  to       2591\n",
      "----------    184 remaining sequences to the next cycle\n",
      "---------- new table with      193 representatives\n",
      "# comparing sequences from       2407  to       2793\n",
      "----------    147 remaining sequences to the next cycle\n",
      "---------- new table with      201 representatives\n",
      "# comparing sequences from       2646  to       2992\n",
      "----------    175 remaining sequences to the next cycle\n",
      "---------- new table with      136 representatives\n",
      "# comparing sequences from       2817  to       3135\n",
      "----------     22 remaining sequences to the next cycle\n",
      "---------- new table with      254 representatives\n",
      "# comparing sequences from       3113  to       3381\n",
      "----------     89 remaining sequences to the next cycle\n",
      "---------- new table with      154 representatives\n",
      "# comparing sequences from       3292  to       3530\n",
      "----------     94 remaining sequences to the next cycle\n",
      "---------- new table with      135 representatives\n",
      "# comparing sequences from       3436  to       3650\n",
      "----------     91 remaining sequences to the next cycle\n",
      "---------- new table with      111 representatives\n",
      "# comparing sequences from       3559  to       3753\n",
      "----------     70 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       3683  to       3856\n",
      "----------     49 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       3807  to       4725\n",
      ".....................---------- new table with      781 representatives\n",
      "\n",
      "     4725  finished       3973  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.06\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "jupyterlab_ancestry",
   "language": "python",
   "name": "jupyterlab_ancestry"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "350px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
