{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import shutil\n",
    "from shutil import copy\n",
    "import time\n",
    "from collections import defaultdict\n",
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "os.chdir(\"/master/nplatt/pathogen_probes/\")\n",
    "\n",
    "\n",
    "def wait_on_running_jobs():\n",
    "   \n",
    "    num_jobs = 1\n",
    "    \n",
    "    while num_jobs > 0:\n",
    "        num_jobs = len(subprocess.check_output('qstat', shell=True).split(\"\\n\")) - 2\n",
    "        time.sleep(60)\n",
    "        print(\".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mycobacterium"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'mycobacterium'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Mycobacterium_ahvazicum_GCF_900176255\",\n",
    "               \"Mycobacterium_alsense_GCF_002086635\",\n",
    "               \"Mycobacterium_angelicum_GCF_002086155\",\n",
    "               \"Mycobacterium_aquaticum_GCF_002086485\",\n",
    "               \"Mycobacterium_arosiense_GCF_002086125\",\n",
    "               \"Mycobacterium_asiaticum_GCF_000613245\",\n",
    "               \"Mycobacterium_attenuatum_GCF_900566085\",\n",
    "               \"Mycobacterium_avium_GCF_000007865\",\n",
    "               \"Mycobacterium_basiliense_GCF_900292015\",\n",
    "               \"Mycobacterium_bohemicum_GCF_001053185\",\n",
    "               \"Mycobacterium_bouchedurhonense_GCF_002086165\",\n",
    "               \"Mycobacterium_branderi_GCF_002086575\",\n",
    "               \"Mycobacterium_canettii_GCF_000253375\",\n",
    "               \"Mycobacterium_celatum_GCF_002101595\",\n",
    "               \"Mycobacterium_chimaera_GCF_002219285\",\n",
    "               \"Mycobacterium_colombiense_GCF_002105755\",\n",
    "               \"Mycobacterium_conspcuum_GCF_002102095\",\n",
    "               \"Mycobacterium_decipiens_GCF_002104675\",\n",
    "               \"Mycobacterium_dioxanotrophicus_GCF_002157835\",\n",
    "               \"Mycobacterium_eburneum_GCF_004354905\",\n",
    "               \"Mycobacterium_europaeum_GCF_001373515\",\n",
    "               \"Mycobacterium_florentinum_GCF_002101635\",\n",
    "               \"Mycobacterium_fragae_GCF_002102185\",\n",
    "               \"Mycobacterium_gastri_GCF_002102175\",\n",
    "               \"Mycobacterium_genavense_GCF_000526915\",\n",
    "               \"Mycobacterium_gordonae_GCF_002101675\",\n",
    "               \"Mycobacterium_grossiae_GCF_008329645\",\n",
    "               \"Mycobacterium_haemophilum_GCF_000340435\",\n",
    "               \"Mycobacterium_heckeshornense_GCF_001881585\",\n",
    "               \"Mycobacterium_heidelbergense_GCF_002086215\",\n",
    "               \"Mycobacterium_innocens_GCF_900566055\",\n",
    "               \"Mycobacterium_interjectum_GCF_900078675\",\n",
    "               \"Mycobacterium_intermedium_GCF_002086275\",\n",
    "               \"Mycobacterium_intracellulare_GCF_000277125\",\n",
    "               \"Mycobacterium_kansasii_GCF_000157895\",\n",
    "               \"Mycobacterium_kubicae_GCF_002101745\",\n",
    "               \"Mycobacterium_kyorinense_GCF_000759695\",\n",
    "               \"Mycobacterium_lacus_GCF_002102215\",\n",
    "               \"Mycobacterium_lehmannii_GCF_002245535\",\n",
    "               \"Mycobacterium_lentiflavum_GCF_001373395\",\n",
    "               \"Mycobacterium_leprae_GCF_000195855\",\n",
    "               \"Mycobacterium_lepraemurium_GCF_002291465\",\n",
    "               \"Mycobacterium_lepromatosis_GCF_000966355\",\n",
    "               \"Mycobacterium_liflandii_GCF_000026445\",\n",
    "               \"Mycobacterium_malmoense_GCF_002086305\",\n",
    "               \"Mycobacterium_mantenii_GCF_002086335\",\n",
    "               \"Mycobacterium_marinum_GCF_000723425\",\n",
    "               \"Mycobacterium_marseillense_GCF_002285715\",\n",
    "               \"Mycobacterium_montefiorense_GCF_003112775\",\n",
    "               \"Mycobacterium_mungi_GCF_001652545\",\n",
    "               \"Mycobacterium_nebraskense_GCF_002102255\",\n",
    "               \"Mycobacterium_neumannii_GCF_002245615\",\n",
    "               \"Mycobacterium_noviomagense_GCF_002086415\",\n",
    "               \"Mycobacterium_orygis_GCF_006385035\",\n",
    "               \"Mycobacterium_palustre_GCF_002101785\",\n",
    "               \"Mycobacterium_paraense_GCF_002101815\",\n",
    "               \"Mycobacterium_paraffinicum_GCF_001907675\",\n",
    "               \"Mycobacterium_paragordonae_GCF_003614435\",\n",
    "               \"Mycobacterium_paraintracellulare_GCF_000276825\",\n",
    "               \"Mycobacterium_parascrofulaceum_GCF_000164135\",\n",
    "               \"Mycobacterium_paraseoulense_GCF_002086475\",\n",
    "               \"Mycobacterium_parmense_GCF_002102335\",\n",
    "               \"Mycobacterium_persicum_GCF_002086675\",\n",
    "               \"Mycobacterium_pseudokansasii_GCF_900566075\",\n",
    "               \"Mycobacterium_pseudoshottsii_GCF_003584745\",\n",
    "               \"Mycobacterium_riyadhense_GCF_002101845\",\n",
    "               \"Mycobacterium_saskatchewanense_GCF_002101875\",\n",
    "               \"Mycobacterium_scrofulaceum_GCF_002086735\",\n",
    "               \"Mycobacterium_sherrisii_GCF_002102355\",\n",
    "               \"Mycobacterium_shigaense_GCF_002356315\",\n",
    "               \"Mycobacterium_shimoidei_GCF_002101905\",\n",
    "               \"Mycobacterium_shinjukuense_GCF_002086755\",\n",
    "               \"Mycobacterium_simiae_GCF_002093075\",\n",
    "               \"Mycobacterium_sp_GCF_000328565\",\n",
    "               \"Mycobacterium_syngnathidarum_GCF_001942625\",\n",
    "               \"Mycobacterium_szulgai_GCF_002116635\",\n",
    "               \"Mycobacterium_talmoniae_GCF_002967005\",\n",
    "               \"Mycobacterium_tilburgii_GCF_902168065\",\n",
    "               \"Mycobacterium_timonense_GCF_002086775\",\n",
    "               \"Mycobacterium_triplex_GCF_000689255\",\n",
    "               \"Mycobacterium_uberis_GCF_003408705\",\n",
    "               \"Mycobacterium_ulcerans_GCF_000013925\",\n",
    "               \"Mycobacterium_vulneris_GCF_002104765\"  ]\n",
    "                    \n",
    "reference_taxon = \"Mycobacterium_tuberculosis_GCF_000195955\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900176255.2_PRJEB20293_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1770222 bytes  1180176.00 bytes/sec\n",
      "total size is 1769677  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086635.1_ASM208663v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1635491 bytes  1090355.33 bytes/sec\n",
      "total size is 1634985  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086155.1_ASM208615v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1951850 bytes  1301261.33 bytes/sec\n",
      "total size is 1951264  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086485.1_ASM208648v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2298949 bytes  1532660.67 bytes/sec\n",
      "total size is 2298275  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086125.1_ASM208612v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1745103 bytes  1163430.00 bytes/sec\n",
      "total size is 1744565  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000613245.1_PRJEB5743_assembly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1712431 bytes  684989.20 bytes/sec\n",
      "total size is 1711892  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900566085.1_LAUMK41_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1908108 bytes  1272100.00 bytes/sec\n",
      "total size is 1907534  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007865.1_ASM786v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1369018 bytes  912706.67 bytes/sec\n",
      "total size is 1368579  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900292015.1_MB901379_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1610469 bytes  1073674.00 bytes/sec\n",
      "total size is 1609966  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001053185.1_PRJEB5741_assembly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1453799 bytes  969227.33 bytes/sec\n",
      "total size is 1453324  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002086165.1_ASM208616v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1697891 bytes  1131955.33 bytes/sec\n",
      "total size is 1697369  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086575.1_ASM208657v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1709243 bytes  1139523.33 bytes/sec\n",
      "total size is 1708713  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000253375.1_ASM25337v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1295041 bytes  863388.67 bytes/sec\n",
      "total size is 1294616  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002101595.1_ASM210159v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1364061 bytes  909402.00 bytes/sec\n",
      "total size is 1363619  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002219285.1_ASM221928v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1758558 bytes  1172400.00 bytes/sec\n",
      "total size is 1758020  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002105755.1_ASM210575v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1595622 bytes  1063776.00 bytes/sec\n",
      "total size is 1595124  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102095.1_ASM210209v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1789438 bytes  1192986.67 bytes/sec\n",
      "total size is 1788892  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002104675.1_ASM210467v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1530440 bytes  1020321.33 bytes/sec\n",
      "total size is 1529958  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002157835.1_ASM215783v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2328109 bytes  1552100.67 bytes/sec\n",
      "total size is 2327427  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004354905.1_ASM435490v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1631700 bytes  1087828.00 bytes/sec\n",
      "total size is 1631194  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001373515.1_PRJEB8464_assembly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1759729 bytes  703908.40 bytes/sec\n",
      "total size is 1759182  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002101635.1_ASM210163v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1788132 bytes  1192116.00 bytes/sec\n",
      "total size is 1787586  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102185.1_ASM210218v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1372234 bytes  914850.67 bytes/sec\n",
      "total size is 1371792  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102175.1_ASM210217v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1693539 bytes  1129054.00 bytes/sec\n",
      "total size is 1693017  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000526915.1_ASM52691v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1424719 bytes  569904.40 bytes/sec\n",
      "total size is 1424262  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002101675.1_ASM210167v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2211606 bytes  1474432.00 bytes/sec\n",
      "total size is 2210956  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008329645.1_ASM832964v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1585901 bytes  1057295.33 bytes/sec\n",
      "total size is 1585403  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000340435.3_ASM34043v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1239663 bytes  826470.00 bytes/sec\n",
      "total size is 1239254  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001881585.2_ASM188158v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1403715 bytes  935838.00 bytes/sec\n",
      "total size is 1403265  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086215.1_ASM208621v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1445378 bytes  963613.33 bytes/sec\n",
      "total size is 1444912  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900566055.1_LAUMK13_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1806699 bytes  1204494.00 bytes/sec\n",
      "total size is 1806149  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900078675.2_PRJEB13236_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1664877 bytes  1109946.00 bytes/sec\n",
      "total size is 1664364  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002086275.1_ASM208627v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2000766 bytes  1333872.00 bytes/sec\n",
      "total size is 2000164  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000277125.1_ASM27712v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1545607 bytes  1030432.67 bytes/sec\n",
      "total size is 1545118  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000157895.2_ASM15789v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1892523 bytes  757026.00 bytes/sec\n",
      "total size is 1891954  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002101745.1_ASM210174v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1698429 bytes  1132314.00 bytes/sec\n",
      "total size is 1697907  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000759695.1_ASM75969v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1540962 bytes  1027336.00 bytes/sec\n",
      "total size is 1540473  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102215.1_ASM210221v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1426849 bytes  951260.67 bytes/sec\n",
      "total size is 1426391  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002245535.1_ASM224553v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1587737 bytes  635111.60 bytes/sec\n",
      "total size is 1587239  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001373395.1_PRJEB8430_assembly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1977951 bytes  1318662.00 bytes/sec\n",
      "total size is 1977348  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000195855.1_ASM19585v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 976185 bytes  650818.00 bytes/sec\n",
      "total size is 975840  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002291465.1_ASM229146v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1151368 bytes  767606.67 bytes/sec\n",
      "total size is 1150974  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000966355.1_ASM96635v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 967232 bytes  644849.33 bytes/sec\n",
      "total size is 966887  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000026445.2_ASM2644v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1811799 bytes  1207894.00 bytes/sec\n",
      "total size is 1811247  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086305.1_ASM208630v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1537825 bytes  1025244.67 bytes/sec\n",
      "total size is 1537343  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086335.1_ASM208633v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1777860 bytes  1185268.00 bytes/sec\n",
      "total size is 1777314  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000723425.2_E11_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1844134 bytes  1229450.67 bytes/sec\n",
      "total size is 1843580  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002285715.1_ASM228571v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1515107 bytes  1010099.33 bytes/sec\n",
      "total size is 1514625  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003112775.1_ASM311277v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1674527 bytes  1116379.33 bytes/sec\n",
      "total size is 1674005  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001652545.1_ASM165254v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1270925 bytes  847311.33 bytes/sec\n",
      "total size is 1270507  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102255.1_ASM210225v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1958656 bytes  783479.20 bytes/sec\n",
      "total size is 1958070  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002245615.1_ASM224561v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1551896 bytes  1034625.33 bytes/sec\n",
      "total size is 1551406  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086415.1_ASM208641v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1390627 bytes  927112.67 bytes/sec\n",
      "total size is 1390177  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006385035.1_ASM638503v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1257348 bytes  838260.00 bytes/sec\n",
      "total size is 1256930  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002101785.1_ASM210178v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1738558 bytes  1159066.67 bytes/sec\n",
      "total size is 1738020  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002101815.1_ASM210181v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1604399 bytes  641776.40 bytes/sec\n",
      "total size is 1603901  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001907675.1_ASM190767v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1875947 bytes  1250659.33 bytes/sec\n",
      "total size is 1875377  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003614435.1_ASM361443v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2072662 bytes  1381802.67 bytes/sec\n",
      "total size is 2072044  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000276825.1_ASM27682v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1573779 bytes  629528.40 bytes/sec\n",
      "total size is 1573282  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000164135.1_ASM16413v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1814443 bytes  1209656.67 bytes/sec\n",
      "total size is 1813890  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086475.1_ASM208647v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1761776 bytes  1174545.33 bytes/sec\n",
      "total size is 1761238  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102335.1_ASM210233v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1694927 bytes  1129979.33 bytes/sec\n",
      "total size is 1694405  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086675.1_ASM208667v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1810871 bytes  724365.20 bytes/sec\n",
      "total size is 1810317  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900566075.1_MK142_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1852812 bytes  1235236.00 bytes/sec\n",
      "total size is 1852256  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003584745.1_ASM358474v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1739012 bytes  1159369.33 bytes/sec\n",
      "total size is 1738474  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002101845.1_ASM210184v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1836158 bytes  1224133.33 bytes/sec\n",
      "total size is 1835596  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002101875.1_ASM210187v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1711662 bytes  1141136.00 bytes/sec\n",
      "total size is 1711132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086735.1_ASM208673v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1782428 bytes  1188313.33 bytes/sec\n",
      "total size is 1781882  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102355.1_ASM210235v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1611172 bytes  1074142.67 bytes/sec\n",
      "total size is 1610666  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002356315.1_ASM235631v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1500743 bytes  600314.00 bytes/sec\n",
      "total size is 1500269  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002101905.1_ASM210190v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1372101 bytes  914762.00 bytes/sec\n",
      "total size is 1371659  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086755.1_ASM208675v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1272660 bytes  848468.00 bytes/sec\n",
      "total size is 1272242  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002093075.1_ASM209307v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1941873 bytes  1294610.00 bytes/sec\n",
      "total size is 1941287  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000328565.1_ASM32856v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2098125 bytes  1398778.00 bytes/sec\n",
      "total size is 2097500  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001942625.1_ASM194262v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1841706 bytes  1227832.00 bytes/sec\n",
      "total size is 1841144  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002116635.1_ASM211663v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1943748 bytes  1295860.00 bytes/sec\n",
      "total size is 1943162  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002967005.1_ASM296700v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1481308 bytes  987566.67 bytes/sec\n",
      "total size is 1480834  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902168065.1_PRJEB33515_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 945889 bytes  630620.67 bytes/sec\n",
      "total size is 945552  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002086775.1_ASM208677v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1760186 bytes  1173485.33 bytes/sec\n",
      "total size is 1759648  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000689255.1_BN973_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1839507 bytes  1226366.00 bytes/sec\n",
      "total size is 1838949  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000195955.2_ASM19595v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1277919 bytes  851974.00 bytes/sec\n",
      "total size is 1277502  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003408705.1_ASM340870v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 938918 bytes  625973.33 bytes/sec\n",
      "total size is 938580  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000013925.2_ASM1392v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1640649 bytes  1093794.00 bytes/sec\n",
      "total size is 1640137  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002104765.1_ASM210476v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1821168 bytes  1214140.00 bytes/sec\n",
      "total size is 1820614  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002102015.1_ASM210201v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1440407 bytes  960299.33 bytes/sec\n",
      "total size is 1439949  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480936 (\"sim_Mycobacterium_ahvazicum_GCF_900176255\") has been submitted\n",
      "Your job 5480937 (\"sim_Mycobacterium_alsense_GCF_002086635\") has been submitted\n",
      "Your job 5480938 (\"sim_Mycobacterium_angelicum_GCF_002086155\") has been submitted\n",
      "Your job 5480939 (\"sim_Mycobacterium_aquaticum_GCF_002086485\") has been submitted\n",
      "Your job 5480940 (\"sim_Mycobacterium_arosiense_GCF_002086125\") has been submitted\n",
      "Your job 5480941 (\"sim_Mycobacterium_asiaticum_GCF_000613245\") has been submitted\n",
      "Your job 5480942 (\"sim_Mycobacterium_attenuatum_GCF_900566085\") has been submitted\n",
      "Your job 5480943 (\"sim_Mycobacterium_avium_GCF_000007865\") has been submitted\n",
      "Your job 5480944 (\"sim_Mycobacterium_basiliense_GCF_900292015\") has been submitted\n",
      "Your job 5480945 (\"sim_Mycobacterium_bohemicum_GCF_001053185\") has been submitted\n",
      "Your job 5480946 (\"sim_Mycobacterium_bouchedurhonense_GCF_002086165\") has been submitted\n",
      "Your job 5480947 (\"sim_Mycobacterium_branderi_GCF_002086575\") has been submitted\n",
      "Your job 5480948 (\"sim_Mycobacterium_canettii_GCF_000253375\") has been submitted\n",
      "Your job 5480949 (\"sim_Mycobacterium_celatum_GCF_002101595\") has been submitted\n",
      "Your job 5480950 (\"sim_Mycobacterium_chimaera_GCF_002219285\") has been submitted\n",
      "Your job 5480951 (\"sim_Mycobacterium_colombiense_GCF_002105755\") has been submitted\n",
      "Your job 5480952 (\"sim_Mycobacterium_conspcuum_GCF_002102095\") has been submitted\n",
      "Your job 5480953 (\"sim_Mycobacterium_decipiens_GCF_002104675\") has been submitted\n",
      "Your job 5480954 (\"sim_Mycobacterium_dioxanotrophicus_GCF_002157835\") has been submitted\n",
      "Your job 5480955 (\"sim_Mycobacterium_eburneum_GCF_004354905\") has been submitted\n",
      "Your job 5480956 (\"sim_Mycobacterium_europaeum_GCF_001373515\") has been submitted\n",
      "Your job 5480957 (\"sim_Mycobacterium_florentinum_GCF_002101635\") has been submitted\n",
      "Your job 5480958 (\"sim_Mycobacterium_fragae_GCF_002102185\") has been submitted\n",
      "Your job 5480959 (\"sim_Mycobacterium_gastri_GCF_002102175\") has been submitted\n",
      "Your job 5480960 (\"sim_Mycobacterium_genavense_GCF_000526915\") has been submitted\n",
      "Your job 5480961 (\"sim_Mycobacterium_gordonae_GCF_002101675\") has been submitted\n",
      "Your job 5480962 (\"sim_Mycobacterium_grossiae_GCF_008329645\") has been submitted\n",
      "Your job 5480963 (\"sim_Mycobacterium_haemophilum_GCF_000340435\") has been submitted\n",
      "Your job 5480964 (\"sim_Mycobacterium_heckeshornense_GCF_001881585\") has been submitted\n",
      "Your job 5480965 (\"sim_Mycobacterium_heidelbergense_GCF_002086215\") has been submitted\n",
      "Your job 5480966 (\"sim_Mycobacterium_innocens_GCF_900566055\") has been submitted\n",
      "Your job 5480967 (\"sim_Mycobacterium_interjectum_GCF_900078675\") has been submitted\n",
      "Your job 5480968 (\"sim_Mycobacterium_intermedium_GCF_002086275\") has been submitted\n",
      "Your job 5480969 (\"sim_Mycobacterium_intracellulare_GCF_000277125\") has been submitted\n",
      "Your job 5480970 (\"sim_Mycobacterium_kansasii_GCF_000157895\") has been submitted\n",
      "Your job 5480971 (\"sim_Mycobacterium_kubicae_GCF_002101745\") has been submitted\n",
      "Your job 5480972 (\"sim_Mycobacterium_kyorinense_GCF_000759695\") has been submitted\n",
      "Your job 5480973 (\"sim_Mycobacterium_lacus_GCF_002102215\") has been submitted\n",
      "Your job 5480974 (\"sim_Mycobacterium_lehmannii_GCF_002245535\") has been submitted\n",
      "Your job 5480975 (\"sim_Mycobacterium_lentiflavum_GCF_001373395\") has been submitted\n",
      "Your job 5480976 (\"sim_Mycobacterium_leprae_GCF_000195855\") has been submitted\n",
      "Your job 5480977 (\"sim_Mycobacterium_lepraemurium_GCF_002291465\") has been submitted\n",
      "Your job 5480978 (\"sim_Mycobacterium_lepromatosis_GCF_000966355\") has been submitted\n",
      "Your job 5480979 (\"sim_Mycobacterium_liflandii_GCF_000026445\") has been submitted\n",
      "Your job 5480980 (\"sim_Mycobacterium_malmoense_GCF_002086305\") has been submitted\n",
      "Your job 5480981 (\"sim_Mycobacterium_mantenii_GCF_002086335\") has been submitted\n",
      "Your job 5480982 (\"sim_Mycobacterium_marinum_GCF_000723425\") has been submitted\n",
      "Your job 5480983 (\"sim_Mycobacterium_marseillense_GCF_002285715\") has been submitted\n",
      "Your job 5480984 (\"sim_Mycobacterium_montefiorense_GCF_003112775\") has been submitted\n",
      "Your job 5480985 (\"sim_Mycobacterium_mungi_GCF_001652545\") has been submitted\n",
      "Your job 5480986 (\"sim_Mycobacterium_nebraskense_GCF_002102255\") has been submitted\n",
      "Your job 5480987 (\"sim_Mycobacterium_neumannii_GCF_002245615\") has been submitted\n",
      "Your job 5480988 (\"sim_Mycobacterium_noviomagense_GCF_002086415\") has been submitted\n",
      "Your job 5480989 (\"sim_Mycobacterium_orygis_GCF_006385035\") has been submitted\n",
      "Your job 5480990 (\"sim_Mycobacterium_palustre_GCF_002101785\") has been submitted\n",
      "Your job 5480991 (\"sim_Mycobacterium_paraense_GCF_002101815\") has been submitted\n",
      "Your job 5480992 (\"sim_Mycobacterium_paraffinicum_GCF_001907675\") has been submitted\n",
      "Your job 5480993 (\"sim_Mycobacterium_paragordonae_GCF_003614435\") has been submitted\n",
      "Your job 5480994 (\"sim_Mycobacterium_paraintracellulare_GCF_000276825\") has been submitted\n",
      "Your job 5480995 (\"sim_Mycobacterium_parascrofulaceum_GCF_000164135\") has been submitted\n",
      "Your job 5480996 (\"sim_Mycobacterium_paraseoulense_GCF_002086475\") has been submitted\n",
      "Your job 5480997 (\"sim_Mycobacterium_parmense_GCF_002102335\") has been submitted\n",
      "Your job 5480998 (\"sim_Mycobacterium_persicum_GCF_002086675\") has been submitted\n",
      "Your job 5480999 (\"sim_Mycobacterium_pseudokansasii_GCF_900566075\") has been submitted\n",
      "Your job 5481000 (\"sim_Mycobacterium_pseudoshottsii_GCF_003584745\") has been submitted\n",
      "Your job 5481001 (\"sim_Mycobacterium_riyadhense_GCF_002101845\") has been submitted\n",
      "Your job 5481002 (\"sim_Mycobacterium_saskatchewanense_GCF_002101875\") has been submitted\n",
      "Your job 5481003 (\"sim_Mycobacterium_scrofulaceum_GCF_002086735\") has been submitted\n",
      "Your job 5481004 (\"sim_Mycobacterium_sherrisii_GCF_002102355\") has been submitted\n",
      "Your job 5481005 (\"sim_Mycobacterium_shigaense_GCF_002356315\") has been submitted\n",
      "Your job 5481006 (\"sim_Mycobacterium_shimoidei_GCF_002101905\") has been submitted\n",
      "Your job 5481007 (\"sim_Mycobacterium_shinjukuense_GCF_002086755\") has been submitted\n",
      "Your job 5481008 (\"sim_Mycobacterium_simiae_GCF_002093075\") has been submitted\n",
      "Your job 5481009 (\"sim_Mycobacterium_sp_GCF_000328565\") has been submitted\n",
      "Your job 5481010 (\"sim_Mycobacterium_syngnathidarum_GCF_001942625\") has been submitted\n",
      "Your job 5481011 (\"sim_Mycobacterium_szulgai_GCF_002116635\") has been submitted\n",
      "Your job 5481012 (\"sim_Mycobacterium_talmoniae_GCF_002967005\") has been submitted\n",
      "Your job 5481013 (\"sim_Mycobacterium_tilburgii_GCF_902168065\") has been submitted\n",
      "Your job 5481014 (\"sim_Mycobacterium_timonense_GCF_002086775\") has been submitted\n",
      "Your job 5481015 (\"sim_Mycobacterium_triplex_GCF_000689255\") has been submitted\n",
      "Your job 5481016 (\"sim_Mycobacterium_uberis_GCF_003408705\") has been submitted\n",
      "Your job 5481017 (\"sim_Mycobacterium_ulcerans_GCF_000013925\") has been submitted\n",
      "Your job 5481018 (\"sim_Mycobacterium_vulneris_GCF_002104765\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/mycobacterium/cleaned_genomes/Mycobacterium_tuberculosis_GCF_000195955_formatted.fas path=results/phyluce/mycobacterium/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481019 (\"map_Mycobacterium_ahvazicum_GCF_900176255\") has been submitted\n",
      "Your job 5481020 (\"map_Mycobacterium_alsense_GCF_002086635\") has been submitted\n",
      "Your job 5481021 (\"map_Mycobacterium_angelicum_GCF_002086155\") has been submitted\n",
      "Your job 5481022 (\"map_Mycobacterium_aquaticum_GCF_002086485\") has been submitted\n",
      "Your job 5481023 (\"map_Mycobacterium_arosiense_GCF_002086125\") has been submitted\n",
      "Your job 5481024 (\"map_Mycobacterium_asiaticum_GCF_000613245\") has been submitted\n",
      "Your job 5481025 (\"map_Mycobacterium_attenuatum_GCF_900566085\") has been submitted\n",
      "Your job 5481026 (\"map_Mycobacterium_avium_GCF_000007865\") has been submitted\n",
      "Your job 5481027 (\"map_Mycobacterium_basiliense_GCF_900292015\") has been submitted\n",
      "Your job 5481028 (\"map_Mycobacterium_bohemicum_GCF_001053185\") has been submitted\n",
      "Your job 5481029 (\"map_Mycobacterium_bouchedurhonense_GCF_002086165\") has been submitted\n",
      "Your job 5481030 (\"map_Mycobacterium_branderi_GCF_002086575\") has been submitted\n",
      "Your job 5481031 (\"map_Mycobacterium_canettii_GCF_000253375\") has been submitted\n",
      "Your job 5481032 (\"map_Mycobacterium_celatum_GCF_002101595\") has been submitted\n",
      "Your job 5481033 (\"map_Mycobacterium_chimaera_GCF_002219285\") has been submitted\n",
      "Your job 5481034 (\"map_Mycobacterium_colombiense_GCF_002105755\") has been submitted\n",
      "Your job 5481035 (\"map_Mycobacterium_conspcuum_GCF_002102095\") has been submitted\n",
      "Your job 5481036 (\"map_Mycobacterium_decipiens_GCF_002104675\") has been submitted\n",
      "Your job 5481037 (\"map_Mycobacterium_dioxanotrophicus_GCF_002157835\") has been submitted\n",
      "Your job 5481038 (\"map_Mycobacterium_eburneum_GCF_004354905\") has been submitted\n",
      "Your job 5481039 (\"map_Mycobacterium_europaeum_GCF_001373515\") has been submitted\n",
      "Your job 5481040 (\"map_Mycobacterium_florentinum_GCF_002101635\") has been submitted\n",
      "Your job 5481041 (\"map_Mycobacterium_fragae_GCF_002102185\") has been submitted\n",
      "Your job 5481042 (\"map_Mycobacterium_gastri_GCF_002102175\") has been submitted\n",
      "Your job 5481043 (\"map_Mycobacterium_genavense_GCF_000526915\") has been submitted\n",
      "Your job 5481044 (\"map_Mycobacterium_gordonae_GCF_002101675\") has been submitted\n",
      "Your job 5481045 (\"map_Mycobacterium_grossiae_GCF_008329645\") has been submitted\n",
      "Your job 5481046 (\"map_Mycobacterium_haemophilum_GCF_000340435\") has been submitted\n",
      "Your job 5481047 (\"map_Mycobacterium_heckeshornense_GCF_001881585\") has been submitted\n",
      "Your job 5481048 (\"map_Mycobacterium_heidelbergense_GCF_002086215\") has been submitted\n",
      "Your job 5481049 (\"map_Mycobacterium_innocens_GCF_900566055\") has been submitted\n",
      "Your job 5481050 (\"map_Mycobacterium_interjectum_GCF_900078675\") has been submitted\n",
      "Your job 5481051 (\"map_Mycobacterium_intermedium_GCF_002086275\") has been submitted\n",
      "Your job 5481052 (\"map_Mycobacterium_intracellulare_GCF_000277125\") has been submitted\n",
      "Your job 5481053 (\"map_Mycobacterium_kansasii_GCF_000157895\") has been submitted\n",
      "Your job 5481054 (\"map_Mycobacterium_kubicae_GCF_002101745\") has been submitted\n",
      "Your job 5481055 (\"map_Mycobacterium_kyorinense_GCF_000759695\") has been submitted\n",
      "Your job 5481056 (\"map_Mycobacterium_lacus_GCF_002102215\") has been submitted\n",
      "Your job 5481057 (\"map_Mycobacterium_lehmannii_GCF_002245535\") has been submitted\n",
      "Your job 5481058 (\"map_Mycobacterium_lentiflavum_GCF_001373395\") has been submitted\n",
      "Your job 5481059 (\"map_Mycobacterium_leprae_GCF_000195855\") has been submitted\n",
      "Your job 5481060 (\"map_Mycobacterium_lepraemurium_GCF_002291465\") has been submitted\n",
      "Your job 5481061 (\"map_Mycobacterium_lepromatosis_GCF_000966355\") has been submitted\n",
      "Your job 5481062 (\"map_Mycobacterium_liflandii_GCF_000026445\") has been submitted\n",
      "Your job 5481063 (\"map_Mycobacterium_malmoense_GCF_002086305\") has been submitted\n",
      "Your job 5481064 (\"map_Mycobacterium_mantenii_GCF_002086335\") has been submitted\n",
      "Your job 5481065 (\"map_Mycobacterium_marinum_GCF_000723425\") has been submitted\n",
      "Your job 5481066 (\"map_Mycobacterium_marseillense_GCF_002285715\") has been submitted\n",
      "Your job 5481067 (\"map_Mycobacterium_montefiorense_GCF_003112775\") has been submitted\n",
      "Your job 5481068 (\"map_Mycobacterium_mungi_GCF_001652545\") has been submitted\n",
      "Your job 5481069 (\"map_Mycobacterium_nebraskense_GCF_002102255\") has been submitted\n",
      "Your job 5481070 (\"map_Mycobacterium_neumannii_GCF_002245615\") has been submitted\n",
      "Your job 5481071 (\"map_Mycobacterium_noviomagense_GCF_002086415\") has been submitted\n",
      "Your job 5481072 (\"map_Mycobacterium_orygis_GCF_006385035\") has been submitted\n",
      "Your job 5481073 (\"map_Mycobacterium_palustre_GCF_002101785\") has been submitted\n",
      "Your job 5481074 (\"map_Mycobacterium_paraense_GCF_002101815\") has been submitted\n",
      "Your job 5481075 (\"map_Mycobacterium_paraffinicum_GCF_001907675\") has been submitted\n",
      "Your job 5481076 (\"map_Mycobacterium_paragordonae_GCF_003614435\") has been submitted\n",
      "Your job 5481077 (\"map_Mycobacterium_paraintracellulare_GCF_000276825\") has been submitted\n",
      "Your job 5481078 (\"map_Mycobacterium_parascrofulaceum_GCF_000164135\") has been submitted\n",
      "Your job 5481079 (\"map_Mycobacterium_paraseoulense_GCF_002086475\") has been submitted\n",
      "Your job 5481080 (\"map_Mycobacterium_parmense_GCF_002102335\") has been submitted\n",
      "Your job 5481081 (\"map_Mycobacterium_persicum_GCF_002086675\") has been submitted\n",
      "Your job 5481082 (\"map_Mycobacterium_pseudokansasii_GCF_900566075\") has been submitted\n",
      "Your job 5481083 (\"map_Mycobacterium_pseudoshottsii_GCF_003584745\") has been submitted\n",
      "Your job 5481084 (\"map_Mycobacterium_riyadhense_GCF_002101845\") has been submitted\n",
      "Your job 5481085 (\"map_Mycobacterium_saskatchewanense_GCF_002101875\") has been submitted\n",
      "Your job 5481086 (\"map_Mycobacterium_scrofulaceum_GCF_002086735\") has been submitted\n",
      "Your job 5481087 (\"map_Mycobacterium_sherrisii_GCF_002102355\") has been submitted\n",
      "Your job 5481088 (\"map_Mycobacterium_shigaense_GCF_002356315\") has been submitted\n",
      "Your job 5481089 (\"map_Mycobacterium_shimoidei_GCF_002101905\") has been submitted\n",
      "Your job 5481090 (\"map_Mycobacterium_shinjukuense_GCF_002086755\") has been submitted\n",
      "Your job 5481091 (\"map_Mycobacterium_simiae_GCF_002093075\") has been submitted\n",
      "Your job 5481092 (\"map_Mycobacterium_sp_GCF_000328565\") has been submitted\n",
      "Your job 5481093 (\"map_Mycobacterium_syngnathidarum_GCF_001942625\") has been submitted\n",
      "Your job 5481094 (\"map_Mycobacterium_szulgai_GCF_002116635\") has been submitted\n",
      "Your job 5481095 (\"map_Mycobacterium_talmoniae_GCF_002967005\") has been submitted\n",
      "Your job 5481096 (\"map_Mycobacterium_tilburgii_GCF_902168065\") has been submitted\n",
      "Your job 5481097 (\"map_Mycobacterium_timonense_GCF_002086775\") has been submitted\n",
      "Your job 5481098 (\"map_Mycobacterium_triplex_GCF_000689255\") has been submitted\n",
      "Your job 5481099 (\"map_Mycobacterium_uberis_GCF_003408705\") has been submitted\n",
      "Your job 5481100 (\"map_Mycobacterium_ulcerans_GCF_000013925\") has been submitted\n",
      "Your job 5481101 (\"map_Mycobacterium_vulneris_GCF_002104765\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481102 (\"merge_Mycobacterium_ahvazicum_GCF_900176255\") has been submitted\n",
      "Your job 5481103 (\"merge_Mycobacterium_alsense_GCF_002086635\") has been submitted\n",
      "Your job 5481104 (\"merge_Mycobacterium_angelicum_GCF_002086155\") has been submitted\n",
      "Your job 5481105 (\"merge_Mycobacterium_aquaticum_GCF_002086485\") has been submitted\n",
      "Your job 5481106 (\"merge_Mycobacterium_arosiense_GCF_002086125\") has been submitted\n",
      "Your job 5481107 (\"merge_Mycobacterium_asiaticum_GCF_000613245\") has been submitted\n",
      "Your job 5481108 (\"merge_Mycobacterium_attenuatum_GCF_900566085\") has been submitted\n",
      "Your job 5481109 (\"merge_Mycobacterium_avium_GCF_000007865\") has been submitted\n",
      "Your job 5481110 (\"merge_Mycobacterium_basiliense_GCF_900292015\") has been submitted\n",
      "Your job 5481111 (\"merge_Mycobacterium_bohemicum_GCF_001053185\") has been submitted\n",
      "Your job 5481112 (\"merge_Mycobacterium_bouchedurhonense_GCF_002086165\") has been submitted\n",
      "Your job 5481113 (\"merge_Mycobacterium_branderi_GCF_002086575\") has been submitted\n",
      "Your job 5481114 (\"merge_Mycobacterium_canettii_GCF_000253375\") has been submitted\n",
      "Your job 5481115 (\"merge_Mycobacterium_celatum_GCF_002101595\") has been submitted\n",
      "Your job 5481116 (\"merge_Mycobacterium_chimaera_GCF_002219285\") has been submitted\n",
      "Your job 5481117 (\"merge_Mycobacterium_colombiense_GCF_002105755\") has been submitted\n",
      "Your job 5481118 (\"merge_Mycobacterium_conspcuum_GCF_002102095\") has been submitted\n",
      "Your job 5481119 (\"merge_Mycobacterium_decipiens_GCF_002104675\") has been submitted\n",
      "Your job 5481120 (\"merge_Mycobacterium_dioxanotrophicus_GCF_002157835\") has been submitted\n",
      "Your job 5481121 (\"merge_Mycobacterium_eburneum_GCF_004354905\") has been submitted\n",
      "Your job 5481122 (\"merge_Mycobacterium_europaeum_GCF_001373515\") has been submitted\n",
      "Your job 5481123 (\"merge_Mycobacterium_florentinum_GCF_002101635\") has been submitted\n",
      "Your job 5481124 (\"merge_Mycobacterium_fragae_GCF_002102185\") has been submitted\n",
      "Your job 5481125 (\"merge_Mycobacterium_gastri_GCF_002102175\") has been submitted\n",
      "Your job 5481126 (\"merge_Mycobacterium_genavense_GCF_000526915\") has been submitted\n",
      "Your job 5481127 (\"merge_Mycobacterium_gordonae_GCF_002101675\") has been submitted\n",
      "Your job 5481128 (\"merge_Mycobacterium_grossiae_GCF_008329645\") has been submitted\n",
      "Your job 5481129 (\"merge_Mycobacterium_haemophilum_GCF_000340435\") has been submitted\n",
      "Your job 5481130 (\"merge_Mycobacterium_heckeshornense_GCF_001881585\") has been submitted\n",
      "Your job 5481131 (\"merge_Mycobacterium_heidelbergense_GCF_002086215\") has been submitted\n",
      "Your job 5481132 (\"merge_Mycobacterium_innocens_GCF_900566055\") has been submitted\n",
      "Your job 5481133 (\"merge_Mycobacterium_interjectum_GCF_900078675\") has been submitted\n",
      "Your job 5481134 (\"merge_Mycobacterium_intermedium_GCF_002086275\") has been submitted\n",
      "Your job 5481135 (\"merge_Mycobacterium_intracellulare_GCF_000277125\") has been submitted\n",
      "Your job 5481136 (\"merge_Mycobacterium_kansasii_GCF_000157895\") has been submitted\n",
      "Your job 5481137 (\"merge_Mycobacterium_kubicae_GCF_002101745\") has been submitted\n",
      "Your job 5481138 (\"merge_Mycobacterium_kyorinense_GCF_000759695\") has been submitted\n",
      "Your job 5481139 (\"merge_Mycobacterium_lacus_GCF_002102215\") has been submitted\n",
      "Your job 5481140 (\"merge_Mycobacterium_lehmannii_GCF_002245535\") has been submitted\n",
      "Your job 5481141 (\"merge_Mycobacterium_lentiflavum_GCF_001373395\") has been submitted\n",
      "Your job 5481142 (\"merge_Mycobacterium_leprae_GCF_000195855\") has been submitted\n",
      "Your job 5481143 (\"merge_Mycobacterium_lepraemurium_GCF_002291465\") has been submitted\n",
      "Your job 5481144 (\"merge_Mycobacterium_lepromatosis_GCF_000966355\") has been submitted\n",
      "Your job 5481145 (\"merge_Mycobacterium_liflandii_GCF_000026445\") has been submitted\n",
      "Your job 5481146 (\"merge_Mycobacterium_malmoense_GCF_002086305\") has been submitted\n",
      "Your job 5481147 (\"merge_Mycobacterium_mantenii_GCF_002086335\") has been submitted\n",
      "Your job 5481148 (\"merge_Mycobacterium_marinum_GCF_000723425\") has been submitted\n",
      "Your job 5481149 (\"merge_Mycobacterium_marseillense_GCF_002285715\") has been submitted\n",
      "Your job 5481150 (\"merge_Mycobacterium_montefiorense_GCF_003112775\") has been submitted\n",
      "Your job 5481151 (\"merge_Mycobacterium_mungi_GCF_001652545\") has been submitted\n",
      "Your job 5481152 (\"merge_Mycobacterium_nebraskense_GCF_002102255\") has been submitted\n",
      "Your job 5481153 (\"merge_Mycobacterium_neumannii_GCF_002245615\") has been submitted\n",
      "Your job 5481154 (\"merge_Mycobacterium_noviomagense_GCF_002086415\") has been submitted\n",
      "Your job 5481155 (\"merge_Mycobacterium_orygis_GCF_006385035\") has been submitted\n",
      "Your job 5481156 (\"merge_Mycobacterium_palustre_GCF_002101785\") has been submitted\n",
      "Your job 5481157 (\"merge_Mycobacterium_paraense_GCF_002101815\") has been submitted\n",
      "Your job 5481158 (\"merge_Mycobacterium_paraffinicum_GCF_001907675\") has been submitted\n",
      "Your job 5481159 (\"merge_Mycobacterium_paragordonae_GCF_003614435\") has been submitted\n",
      "Your job 5481160 (\"merge_Mycobacterium_paraintracellulare_GCF_000276825\") has been submitted\n",
      "Your job 5481161 (\"merge_Mycobacterium_parascrofulaceum_GCF_000164135\") has been submitted\n",
      "Your job 5481162 (\"merge_Mycobacterium_paraseoulense_GCF_002086475\") has been submitted\n",
      "Your job 5481163 (\"merge_Mycobacterium_parmense_GCF_002102335\") has been submitted\n",
      "Your job 5481164 (\"merge_Mycobacterium_persicum_GCF_002086675\") has been submitted\n",
      "Your job 5481165 (\"merge_Mycobacterium_pseudokansasii_GCF_900566075\") has been submitted\n",
      "Your job 5481166 (\"merge_Mycobacterium_pseudoshottsii_GCF_003584745\") has been submitted\n",
      "Your job 5481167 (\"merge_Mycobacterium_riyadhense_GCF_002101845\") has been submitted\n",
      "Your job 5481168 (\"merge_Mycobacterium_saskatchewanense_GCF_002101875\") has been submitted\n",
      "Your job 5481169 (\"merge_Mycobacterium_scrofulaceum_GCF_002086735\") has been submitted\n",
      "Your job 5481170 (\"merge_Mycobacterium_sherrisii_GCF_002102355\") has been submitted\n",
      "Your job 5481171 (\"merge_Mycobacterium_shigaense_GCF_002356315\") has been submitted\n",
      "Your job 5481172 (\"merge_Mycobacterium_shimoidei_GCF_002101905\") has been submitted\n",
      "Your job 5481173 (\"merge_Mycobacterium_shinjukuense_GCF_002086755\") has been submitted\n",
      "Your job 5481174 (\"merge_Mycobacterium_simiae_GCF_002093075\") has been submitted\n",
      "Your job 5481175 (\"merge_Mycobacterium_sp_GCF_000328565\") has been submitted\n",
      "Your job 5481176 (\"merge_Mycobacterium_syngnathidarum_GCF_001942625\") has been submitted\n",
      "Your job 5481177 (\"merge_Mycobacterium_szulgai_GCF_002116635\") has been submitted\n",
      "Your job 5481178 (\"merge_Mycobacterium_talmoniae_GCF_002967005\") has been submitted\n",
      "Your job 5481179 (\"merge_Mycobacterium_tilburgii_GCF_902168065\") has been submitted\n",
      "Your job 5481180 (\"merge_Mycobacterium_timonense_GCF_002086775\") has been submitted\n",
      "Your job 5481181 (\"merge_Mycobacterium_triplex_GCF_000689255\") has been submitted\n",
      "Your job 5481182 (\"merge_Mycobacterium_uberis_GCF_003408705\") has been submitted\n",
      "Your job 5481183 (\"merge_Mycobacterium_ulcerans_GCF_000013925\") has been submitted\n",
      "Your job 5481184 (\"merge_Mycobacterium_vulneris_GCF_002104765\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 2386 sequences from Mycobacterium_ahvazicum_GCF_900176255_merged.bed.  Filtered 1589 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 797.\n",
      "Screened 3113 sequences from Mycobacterium_alsense_GCF_002086635_merged.bed.  Filtered 2002 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1111.\n",
      "Screened 3568 sequences from Mycobacterium_angelicum_GCF_002086155_merged.bed.  Filtered 2226 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1342.\n",
      "Screened 850 sequences from Mycobacterium_aquaticum_GCF_002086485_merged.bed.  Filtered 623 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 227.\n",
      "Screened 2374 sequences from Mycobacterium_arosiense_GCF_002086125_merged.bed.  Filtered 1603 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 771.\n",
      "Screened 2180 sequences from Mycobacterium_asiaticum_GCF_000613245_merged.bed.  Filtered 1524 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 656.\n",
      "Screened 3164 sequences from Mycobacterium_attenuatum_GCF_900566085_merged.bed.  Filtered 2072 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1092.\n",
      "Screened 2793 sequences from Mycobacterium_avium_GCF_000007865_merged.bed.  Filtered 1795 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 998.\n",
      "Screened 2655 sequences from Mycobacterium_basiliense_GCF_900292015_merged.bed.  Filtered 1801 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 854.\n",
      "Screened 2445 sequences from Mycobacterium_bohemicum_GCF_001053185_merged.bed.  Filtered 1635 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 810.\n",
      "Screened 2814 sequences from Mycobacterium_bouchedurhonense_GCF_002086165_merged.bed.  Filtered 1794 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1020.\n",
      "Screened 2032 sequences from Mycobacterium_branderi_GCF_002086575_merged.bed.  Filtered 1376 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 656.\n",
      "Screened 418 sequences from Mycobacterium_canettii_GCF_000253375_merged.bed.  Filtered 49 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 369.\n",
      "Screened 2061 sequences from Mycobacterium_celatum_GCF_002101595_merged.bed.  Filtered 1380 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 681.\n",
      "Screened 2663 sequences from Mycobacterium_chimaera_GCF_002219285_merged.bed.  Filtered 1739 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 924.\n",
      "Screened 2584 sequences from Mycobacterium_colombiense_GCF_002105755_merged.bed.  Filtered 1712 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 872.\n",
      "Screened 2992 sequences from Mycobacterium_conspcuum_GCF_002102095_merged.bed.  Filtered 1918 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1074.\n",
      "Screened 7290 sequences from Mycobacterium_decipiens_GCF_002104675_merged.bed.  Filtered 2842 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4448.\n",
      "Screened 822 sequences from Mycobacterium_dioxanotrophicus_GCF_002157835_merged.bed.  Filtered 620 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 202.\n",
      "Screened 1796 sequences from Mycobacterium_eburneum_GCF_004354905_merged.bed.  Filtered 1234 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 562.\n",
      "Screened 2735 sequences from Mycobacterium_europaeum_GCF_001373515_merged.bed.  Filtered 1775 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 960.\n",
      "Screened 2646 sequences from Mycobacterium_florentinum_GCF_002101635_merged.bed.  Filtered 1742 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 904.\n",
      "Screened 1870 sequences from Mycobacterium_fragae_GCF_002102185_merged.bed.  Filtered 1309 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 561.\n",
      "Screened 3731 sequences from Mycobacterium_gastri_GCF_002102175_merged.bed.  Filtered 2311 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1420.\n",
      "Screened 2425 sequences from Mycobacterium_genavense_GCF_000526915_merged.bed.  Filtered 1613 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 812.\n",
      "Screened 2366 sequences from Mycobacterium_gordonae_GCF_002101675_merged.bed.  Filtered 1607 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 759.\n",
      "Screened 822 sequences from Mycobacterium_grossiae_GCF_008329645_merged.bed.  Filtered 605 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 217.\n",
      "Screened 3069 sequences from Mycobacterium_haemophilum_GCF_000340435_merged.bed.  Filtered 1998 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1071.\n",
      "Screened 1558 sequences from Mycobacterium_heckeshornense_GCF_001881585_merged.bed.  Filtered 1109 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 449.\n",
      "Screened 3666 sequences from Mycobacterium_heidelbergense_GCF_002086215_merged.bed.  Filtered 2210 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1456.\n",
      "Screened 3758 sequences from Mycobacterium_innocens_GCF_900566055_merged.bed.  Filtered 2387 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1371.\n",
      "Screened 3029 sequences from Mycobacterium_interjectum_GCF_900078675_merged.bed.  Filtered 1921 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1108.\n",
      "Screened 2371 sequences from Mycobacterium_intermedium_GCF_002086275_merged.bed.  Filtered 1567 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 804.\n",
      "Screened 2624 sequences from Mycobacterium_intracellulare_GCF_000277125_merged.bed.  Filtered 1722 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 902.\n",
      "Screened 3612 sequences from Mycobacterium_kansasii_GCF_000157895_merged.bed.  Filtered 2284 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1328.\n",
      "Screened 2470 sequences from Mycobacterium_kubicae_GCF_002101745_merged.bed.  Filtered 1644 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 826.\n",
      "Screened 2099 sequences from Mycobacterium_kyorinense_GCF_000759695_merged.bed.  Filtered 1421 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 678.\n",
      "Screened 4983 sequences from Mycobacterium_lacus_GCF_002102215_merged.bed.  Filtered 2884 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2099.\n",
      "Screened 813 sequences from Mycobacterium_lehmannii_GCF_002245535_merged.bed.  Filtered 602 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 211.\n",
      "Screened 2365 sequences from Mycobacterium_lentiflavum_GCF_001373395_merged.bed.  Filtered 1565 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 800.\n",
      "Screened 909 sequences from Mycobacterium_leprae_GCF_000195855_merged.bed.  Filtered 688 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 221.\n",
      "Screened 2435 sequences from Mycobacterium_lepraemurium_GCF_002291465_merged.bed.  Filtered 1584 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 851.\n",
      "Screened 879 sequences from Mycobacterium_lepromatosis_GCF_000966355_merged.bed.  Filtered 662 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 217.\n",
      "Screened 2638 sequences from Mycobacterium_liflandii_GCF_000026445_merged.bed.  Filtered 1785 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 853.\n",
      "Screened 3804 sequences from Mycobacterium_malmoense_GCF_002086305_merged.bed.  Filtered 2298 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1506.\n",
      "Screened 2508 sequences from Mycobacterium_mantenii_GCF_002086335_merged.bed.  Filtered 1667 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 841.\n",
      "Screened 2692 sequences from Mycobacterium_marinum_GCF_000723425_merged.bed.  Filtered 1804 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 888.\n",
      "Screened 2484 sequences from Mycobacterium_marseillense_GCF_002285715_merged.bed.  Filtered 1640 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 844.\n",
      "Screened 2333 sequences from Mycobacterium_montefiorense_GCF_003112775_merged.bed.  Filtered 1551 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 782.\n",
      "Screened 240 sequences from Mycobacterium_mungi_GCF_001652545_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 224.\n",
      "Screened 2937 sequences from Mycobacterium_nebraskense_GCF_002102255_merged.bed.  Filtered 1928 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1009.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 767 sequences from Mycobacterium_neumannii_GCF_002245615_merged.bed.  Filtered 566 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 201.\n",
      "Screened 1687 sequences from Mycobacterium_noviomagense_GCF_002086415_merged.bed.  Filtered 1184 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 503.\n",
      "Screened 215 sequences from Mycobacterium_orygis_GCF_006385035_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 198.\n",
      "Screened 3052 sequences from Mycobacterium_palustre_GCF_002101785_merged.bed.  Filtered 1958 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1094.\n",
      "Screened 3154 sequences from Mycobacterium_paraense_GCF_002101815_merged.bed.  Filtered 1972 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1182.\n",
      "Screened 2543 sequences from Mycobacterium_paraffinicum_GCF_001907675_merged.bed.  Filtered 1659 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 884.\n",
      "Screened 2423 sequences from Mycobacterium_paragordonae_GCF_003614435_merged.bed.  Filtered 1620 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 803.\n",
      "Screened 2634 sequences from Mycobacterium_paraintracellulare_GCF_000276825_merged.bed.  Filtered 1726 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 908.\n",
      "Screened 2786 sequences from Mycobacterium_parascrofulaceum_GCF_000164135_merged.bed.  Filtered 1799 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 987.\n",
      "Screened 2749 sequences from Mycobacterium_paraseoulense_GCF_002086475_merged.bed.  Filtered 1815 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 934.\n",
      "Screened 2402 sequences from Mycobacterium_parmense_GCF_002102335_merged.bed.  Filtered 1576 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 826.\n",
      "Screened 3598 sequences from Mycobacterium_persicum_GCF_002086675_merged.bed.  Filtered 2291 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1307.\n",
      "Screened 3571 sequences from Mycobacterium_pseudokansasii_GCF_900566075_merged.bed.  Filtered 2251 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1320.\n",
      "Screened 2606 sequences from Mycobacterium_pseudoshottsii_GCF_003584745_merged.bed.  Filtered 1748 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 858.\n",
      "Screened 4426 sequences from Mycobacterium_riyadhense_GCF_002101845_merged.bed.  Filtered 2701 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1725.\n",
      "Screened 2389 sequences from Mycobacterium_saskatchewanense_GCF_002101875_merged.bed.  Filtered 1596 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 793.\n",
      "Screened 2699 sequences from Mycobacterium_scrofulaceum_GCF_002086735_merged.bed.  Filtered 1775 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 924.\n",
      "Screened 2043 sequences from Mycobacterium_sherrisii_GCF_002102355_merged.bed.  Filtered 1383 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 660.\n",
      "Screened 2268 sequences from Mycobacterium_shigaense_GCF_002356315_merged.bed.  Filtered 1544 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 724.\n",
      "Screened 1502 sequences from Mycobacterium_shimoidei_GCF_002101905_merged.bed.  Filtered 1047 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 455.\n",
      "Screened 5041 sequences from Mycobacterium_shinjukuense_GCF_002086755_merged.bed.  Filtered 2686 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2355.\n",
      "Screened 1934 sequences from Mycobacterium_simiae_GCF_002093075_merged.bed.  Filtered 1315 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 619.\n",
      "Screened 750 sequences from Mycobacterium_sp_GCF_000328565_merged.bed.  Filtered 571 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 179.\n",
      "Screened 802 sequences from Mycobacterium_syngnathidarum_GCF_001942625_merged.bed.  Filtered 599 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 203.\n",
      "Screened 3266 sequences from Mycobacterium_szulgai_GCF_002116635_merged.bed.  Filtered 2155 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1111.\n",
      "Screened 1686 sequences from Mycobacterium_talmoniae_GCF_002967005_merged.bed.  Filtered 1146 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 540.\n",
      "Screened 1548 sequences from Mycobacterium_tilburgii_GCF_902168065_merged.bed.  Filtered 1039 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 509.\n",
      "Screened 2792 sequences from Mycobacterium_timonense_GCF_002086775_merged.bed.  Filtered 1814 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 978.\n",
      "Screened 2661 sequences from Mycobacterium_triplex_GCF_000689255_merged.bed.  Filtered 1715 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 946.\n",
      "Screened 857 sequences from Mycobacterium_uberis_GCF_003408705_merged.bed.  Filtered 661 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 196.\n",
      "Screened 2492 sequences from Mycobacterium_ulcerans_GCF_000013925_merged.bed.  Filtered 1686 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 806.\n",
      "Screened 2484 sequences from Mycobacterium_vulneris_GCF_002104765_merged.bed.  Filtered 1648 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 836.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/mycobacterium/initial_intervals/Mycobacterium_vulneris_GCF_002104765_merged.bed --twobit results/phyluce/mycobacterium/cleaned_genomes/Mycobacterium_tuberculosis_GCF_000195955_formatted.2bit --output results/phyluce/mycobacterium/initial_intervals/Mycobacterium_vulneris_GCF_002104765_stripped.bed;\n",
      "mycobacterium_ahvazicum_gcf_900176255.\n",
      "mycobacterium_alsense_gcf_002086635..\n",
      "mycobacterium_angelicum_gcf_002086155..\n",
      "mycobacterium_aquaticum_gcf_002086485.\n",
      "mycobacterium_arosiense_gcf_002086125.\n",
      "mycobacterium_asiaticum_gcf_000613245.\n",
      "mycobacterium_attenuatum_gcf_900566085..\n",
      "mycobacterium_avium_gcf_000007865.\n",
      "mycobacterium_basiliense_gcf_900292015.\n",
      "mycobacterium_bohemicum_gcf_001053185.\n",
      "mycobacterium_bouchedurhonense_gcf_002086165..\n",
      "mycobacterium_branderi_gcf_002086575.\n",
      "mycobacterium_canettii_gcf_000253375.\n",
      "mycobacterium_celatum_gcf_002101595.\n",
      "mycobacterium_chimaera_gcf_002219285.\n",
      "mycobacterium_colombiense_gcf_002105755.\n",
      "mycobacterium_conspcuum_gcf_002102095..\n",
      "mycobacterium_decipiens_gcf_002104675.....\n",
      "mycobacterium_dioxanotrophicus_gcf_002157835.\n",
      "mycobacterium_eburneum_gcf_004354905.\n",
      "mycobacterium_europaeum_gcf_001373515.\n",
      "mycobacterium_florentinum_gcf_002101635.\n",
      "mycobacterium_fragae_gcf_002102185.\n",
      "mycobacterium_gastri_gcf_002102175..\n",
      "mycobacterium_genavense_gcf_000526915.\n",
      "mycobacterium_gordonae_gcf_002101675.\n",
      "mycobacterium_grossiae_gcf_008329645.\n",
      "mycobacterium_haemophilum_gcf_000340435..\n",
      "mycobacterium_heckeshornense_gcf_001881585.\n",
      "mycobacterium_heidelbergense_gcf_002086215..\n",
      "mycobacterium_innocens_gcf_900566055..\n",
      "mycobacterium_interjectum_gcf_900078675..\n",
      "mycobacterium_intermedium_gcf_002086275.\n",
      "mycobacterium_intracellulare_gcf_000277125.\n",
      "mycobacterium_kansasii_gcf_000157895..\n",
      "mycobacterium_kubicae_gcf_002101745.\n",
      "mycobacterium_kyorinense_gcf_000759695.\n",
      "mycobacterium_lacus_gcf_002102215...\n",
      "mycobacterium_lehmannii_gcf_002245535.\n",
      "mycobacterium_lentiflavum_gcf_001373395.\n",
      "mycobacterium_leprae_gcf_000195855.\n",
      "mycobacterium_lepraemurium_gcf_002291465.\n",
      "mycobacterium_lepromatosis_gcf_000966355.\n",
      "mycobacterium_liflandii_gcf_000026445.\n",
      "mycobacterium_malmoense_gcf_002086305..\n",
      "mycobacterium_mantenii_gcf_002086335.\n",
      "mycobacterium_marinum_gcf_000723425.\n",
      "mycobacterium_marseillense_gcf_002285715.\n",
      "mycobacterium_montefiorense_gcf_003112775.\n",
      "mycobacterium_mungi_gcf_001652545.\n",
      "mycobacterium_nebraskense_gcf_002102255..\n",
      "mycobacterium_neumannii_gcf_002245615.\n",
      "mycobacterium_noviomagense_gcf_002086415.\n",
      "mycobacterium_orygis_gcf_006385035.\n",
      "mycobacterium_palustre_gcf_002101785..\n",
      "mycobacterium_paraense_gcf_002101815..\n",
      "mycobacterium_paraffinicum_gcf_001907675.\n",
      "mycobacterium_paragordonae_gcf_003614435.\n",
      "mycobacterium_paraintracellulare_gcf_000276825.\n",
      "mycobacterium_parascrofulaceum_gcf_000164135.\n",
      "mycobacterium_paraseoulense_gcf_002086475.\n",
      "mycobacterium_parmense_gcf_002102335.\n",
      "mycobacterium_persicum_gcf_002086675..\n",
      "mycobacterium_pseudokansasii_gcf_900566075..\n",
      "mycobacterium_pseudoshottsii_gcf_003584745.\n",
      "mycobacterium_riyadhense_gcf_002101845..\n",
      "mycobacterium_saskatchewanense_gcf_002101875.\n",
      "mycobacterium_scrofulaceum_gcf_002086735.\n",
      "mycobacterium_sherrisii_gcf_002102355.\n",
      "mycobacterium_shigaense_gcf_002356315.\n",
      "mycobacterium_shimoidei_gcf_002101905.\n",
      "mycobacterium_shinjukuense_gcf_002086755...\n",
      "mycobacterium_simiae_gcf_002093075.\n",
      "mycobacterium_sp_gcf_000328565.\n",
      "mycobacterium_syngnathidarum_gcf_001942625.\n",
      "mycobacterium_szulgai_gcf_002116635..\n",
      "mycobacterium_talmoniae_gcf_002967005.\n",
      "mycobacterium_tilburgii_gcf_902168065.\n",
      "mycobacterium_timonense_gcf_002086775.\n",
      "mycobacterium_triplex_gcf_000689255.\n",
      "mycobacterium_uberis_gcf_003408705.\n",
      "mycobacterium_ulcerans_gcf_000013925.\n",
      "mycobacterium_vulneris_gcf_002104765.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/mycobacterium/initial_intervals/Mycobacterium_vulneris_GCF_002104765_merged.bed --twobit results/phyluce/mycobacterium/cleaned_genomes/Mycobacterium_tuberculosis_GCF_000195955_formatted.2bit --output results/phyluce/mycobacterium/initial_intervals/Mycobacterium_vulneris_GCF_002104765_stripped.bed;\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 0 taxa:\t6,832.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 1 taxa:\t6,832.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 2 taxa:\t6,370.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 3 taxa:\t6,243.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 4 taxa:\t4,460.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 5 taxa:\t3,759.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 6 taxa:\t3,326.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 7 taxa:\t3,066.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 8 taxa:\t2,827.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 9 taxa:\t2,618.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 10 taxa:\t2,450.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 11 taxa:\t2,313.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 12 taxa:\t2,188.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 13 taxa:\t2,063.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 14 taxa:\t1,950.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 15 taxa:\t1,864.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 16 taxa:\t1,783.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 17 taxa:\t1,707.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 18 taxa:\t1,636.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 19 taxa:\t1,566.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 20 taxa:\t1,512.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 21 taxa:\t1,441.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 22 taxa:\t1,393.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 23 taxa:\t1,341.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 24 taxa:\t1,293.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 25 taxa:\t1,245.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 26 taxa:\t1,189.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 27 taxa:\t1,139.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 28 taxa:\t1,100.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 29 taxa:\t1,065.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 30 taxa:\t1,028.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 31 taxa:\t991.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 32 taxa:\t953.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 33 taxa:\t924.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 34 taxa:\t902.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 35 taxa:\t879.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 36 taxa:\t851.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 37 taxa:\t826.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 38 taxa:\t800.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 39 taxa:\t768.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 40 taxa:\t731.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 41 taxa:\t710.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 42 taxa:\t682.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 43 taxa:\t655.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 44 taxa:\t632.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 45 taxa:\t606.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 46 taxa:\t589.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 47 taxa:\t565.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 48 taxa:\t545.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 49 taxa:\t519.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 50 taxa:\t501.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 51 taxa:\t482.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 52 taxa:\t467.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 53 taxa:\t453.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 54 taxa:\t433.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 55 taxa:\t415.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 56 taxa:\t394.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 57 taxa:\t377.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 58 taxa:\t350.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 59 taxa:\t332.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 60 taxa:\t321.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 61 taxa:\t311.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 62 taxa:\t299.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 63 taxa:\t284.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 64 taxa:\t268.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 65 taxa:\t256.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 66 taxa:\t239.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 67 taxa:\t224.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 68 taxa:\t208.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 69 taxa:\t191.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 70 taxa:\t182.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 71 taxa:\t170.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 72 taxa:\t156.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 73 taxa:\t144.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 74 taxa:\t134.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 75 taxa:\t126.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 76 taxa:\t111.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 77 taxa:\t95.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 78 taxa:\t86.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 79 taxa:\t72.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 80 taxa:\t57.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 81 taxa:\t47.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 82 taxa:\t37.0\n",
      "Loci shared by Mycobacterium_tuberculosis_GCF_000195955 + 83 taxa:\t18.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/mycobacterium/initial_intervals/mycobacterium-to-Mycobacterium_tuberculosis_GCF_000195955.sqlite --base-taxon Mycobacterium_tuberculosis_GCF_000195955 --output results/phyluce/mycobacterium/initial_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70.bed --specific-counts 70;\n",
      "Counter({'mycobacterium_vulneris_gcf_002104765': 182, 'mycobacterium_paraffinicum_gcf_001907675': 182, 'mycobacterium_colombiense_gcf_002105755': 182, 'mycobacterium_avium_gcf_000007865': 182, 'mycobacterium_orygis_gcf_006385035': 182, 'mycobacterium_decipiens_gcf_002104675': 182, 'mycobacterium_intracellulare_gcf_000277125': 182, 'mycobacterium_lacus_gcf_002102215': 182, 'mycobacterium_malmoense_gcf_002086305': 182, 'mycobacterium_mungi_gcf_001652545': 182, 'mycobacterium_canettii_gcf_000253375': 182, 'mycobacterium_paraense_gcf_002101815': 182, 'mycobacterium_pseudokansasii_gcf_900566075': 181, 'mycobacterium_palustre_gcf_002101785': 181, 'mycobacterium_shinjukuense_gcf_002086755': 181, 'mycobacterium_innocens_gcf_900566055': 181, 'mycobacterium_genavense_gcf_000526915': 181, 'mycobacterium_heidelbergense_gcf_002086215': 181, 'mycobacterium_gastri_gcf_002102175': 181, 'mycobacterium_bouchedurhonense_gcf_002086165': 181, 'mycobacterium_conspcuum_gcf_002102095': 181, 'mycobacterium_interjectum_gcf_900078675': 180, 'mycobacterium_timonense_gcf_002086775': 180, 'mycobacterium_mantenii_gcf_002086335': 180, 'mycobacterium_kansasii_gcf_000157895': 180, 'mycobacterium_parascrofulaceum_gcf_000164135': 180, 'mycobacterium_nebraskense_gcf_002102255': 180, 'mycobacterium_chimaera_gcf_002219285': 180, 'mycobacterium_lepraemurium_gcf_002291465': 180, 'mycobacterium_marseillense_gcf_002285715': 180, 'mycobacterium_paraintracellulare_gcf_000276825': 179, 'mycobacterium_angelicum_gcf_002086155': 179, 'mycobacterium_scrofulaceum_gcf_002086735': 179, 'mycobacterium_triplex_gcf_000689255': 179, 'mycobacterium_persicum_gcf_002086675': 179, 'mycobacterium_europaeum_gcf_001373515': 178, 'mycobacterium_kubicae_gcf_002101745': 178, 'mycobacterium_arosiense_gcf_002086125': 178, 'mycobacterium_attenuatum_gcf_900566085': 178, 'mycobacterium_alsense_gcf_002086635': 178, 'mycobacterium_gordonae_gcf_002101675': 177, 'mycobacterium_florentinum_gcf_002101635': 177, 'mycobacterium_saskatchewanense_gcf_002101875': 177, 'mycobacterium_riyadhense_gcf_002101845': 177, 'mycobacterium_bohemicum_gcf_001053185': 177, 'mycobacterium_intermedium_gcf_002086275': 177, 'mycobacterium_pseudoshottsii_gcf_003584745': 176, 'mycobacterium_montefiorense_gcf_003112775': 176, 'mycobacterium_marinum_gcf_000723425': 176, 'mycobacterium_parmense_gcf_002102335': 176, 'mycobacterium_liflandii_gcf_000026445': 176, 'mycobacterium_szulgai_gcf_002116635': 176, 'mycobacterium_paraseoulense_gcf_002086475': 175, 'mycobacterium_paragordonae_gcf_003614435': 175, 'mycobacterium_lentiflavum_gcf_001373395': 174, 'mycobacterium_simiae_gcf_002093075': 174, 'mycobacterium_ulcerans_gcf_000013925': 173, 'mycobacterium_celatum_gcf_002101595': 173, 'mycobacterium_sherrisii_gcf_002102355': 173, 'mycobacterium_haemophilum_gcf_000340435': 172, 'mycobacterium_shigaense_gcf_002356315': 172, 'mycobacterium_branderi_gcf_002086575': 172, 'mycobacterium_ahvazicum_gcf_900176255': 171, 'mycobacterium_kyorinense_gcf_000759695': 169, 'mycobacterium_asiaticum_gcf_000613245': 168, 'mycobacterium_fragae_gcf_002102185': 168, 'mycobacterium_basiliense_gcf_900292015': 166, 'mycobacterium_tilburgii_gcf_902168065': 166, 'mycobacterium_eburneum_gcf_004354905': 163, 'mycobacterium_talmoniae_gcf_002967005': 160, 'mycobacterium_shimoidei_gcf_002101905': 157, 'mycobacterium_noviomagense_gcf_002086415': 155, 'mycobacterium_heckeshornense_gcf_001881585': 142, 'mycobacterium_aquaticum_gcf_002086485': 127, 'mycobacterium_lehmannii_gcf_002245535': 125, 'mycobacterium_neumannii_gcf_002245615': 123, 'mycobacterium_sp_gcf_000328565': 122, 'mycobacterium_dioxanotrophicus_gcf_002157835': 121, 'mycobacterium_grossiae_gcf_008329645': 118, 'mycobacterium_syngnathidarum_gcf_001942625': 118, 'mycobacterium_leprae_gcf_000195855': 103, 'mycobacterium_lepromatosis_gcf_000966355': 97, 'mycobacterium_uberis_gcf_003408705': 85})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 70\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/mycobacterium/initial_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70.bed --twobit results/phyluce/mycobacterium/cleaned_genomes/Mycobacterium_tuberculosis_GCF_000195955_formatted.2bit --buffer-to 160 --output results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70.fasta;\n",
      "Screened 182 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 182.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70.fasta --probe-prefix uce_mycobacterium_ --design mycobacterium_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 172\n",
      "Probe Count = 333\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas --query results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  14:20:49\n",
      "Ended:  Thu Feb 06, 2020  14:20:49\n",
      "Time for execution:  0.00333928267161 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas                        --lastz results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes_vself.lastz                       --probe-prefix=uce_mycobacterium_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 332 fasta sequences.  Filtered 4 duplicates. Kept 326.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/mycobacterium/validate_intervals/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas --scaffoldlist Mycobacterium_ahvazicum_GCF_900176255 Mycobacterium_alsense_GCF_002086635 Mycobacterium_angelicum_GCF_002086155 Mycobacterium_aquaticum_GCF_002086485 Mycobacterium_arosiense_GCF_002086125 Mycobacterium_asiaticum_GCF_000613245 Mycobacterium_attenuatum_GCF_900566085 Mycobacterium_avium_GCF_000007865 Mycobacterium_basiliense_GCF_900292015 Mycobacterium_bohemicum_GCF_001053185 Mycobacterium_bouchedurhonense_GCF_002086165 Mycobacterium_branderi_GCF_002086575 Mycobacterium_canettii_GCF_000253375 Mycobacterium_celatum_GCF_002101595 Mycobacterium_chimaera_GCF_002219285 Mycobacterium_colombiense_GCF_002105755 Mycobacterium_conspcuum_GCF_002102095 Mycobacterium_decipiens_GCF_002104675 Mycobacterium_dioxanotrophicus_GCF_002157835 Mycobacterium_eburneum_GCF_004354905 Mycobacterium_europaeum_GCF_001373515 Mycobacterium_florentinum_GCF_002101635 Mycobacterium_fragae_GCF_002102185 Mycobacterium_gastri_GCF_002102175 Mycobacterium_genavense_GCF_000526915 Mycobacterium_gordonae_GCF_002101675 Mycobacterium_grossiae_GCF_008329645 Mycobacterium_haemophilum_GCF_000340435 Mycobacterium_heckeshornense_GCF_001881585 Mycobacterium_heidelbergense_GCF_002086215 Mycobacterium_innocens_GCF_900566055 Mycobacterium_interjectum_GCF_900078675 Mycobacterium_intermedium_GCF_002086275 Mycobacterium_intracellulare_GCF_000277125 Mycobacterium_kansasii_GCF_000157895 Mycobacterium_kubicae_GCF_002101745 Mycobacterium_kyorinense_GCF_000759695 Mycobacterium_lacus_GCF_002102215 Mycobacterium_lehmannii_GCF_002245535 Mycobacterium_lentiflavum_GCF_001373395 Mycobacterium_leprae_GCF_000195855 Mycobacterium_lepraemurium_GCF_002291465 Mycobacterium_lepromatosis_GCF_000966355 Mycobacterium_liflandii_GCF_000026445 Mycobacterium_malmoense_GCF_002086305 Mycobacterium_mantenii_GCF_002086335 Mycobacterium_marinum_GCF_000723425 Mycobacterium_marseillense_GCF_002285715 Mycobacterium_montefiorense_GCF_003112775 Mycobacterium_mungi_GCF_001652545 Mycobacterium_nebraskense_GCF_002102255 Mycobacterium_neumannii_GCF_002245615 Mycobacterium_noviomagense_GCF_002086415 Mycobacterium_orygis_GCF_006385035 Mycobacterium_palustre_GCF_002101785 Mycobacterium_paraense_GCF_002101815 Mycobacterium_paraffinicum_GCF_001907675 Mycobacterium_paragordonae_GCF_003614435 Mycobacterium_paraintracellulare_GCF_000276825 Mycobacterium_parascrofulaceum_GCF_000164135 Mycobacterium_paraseoulense_GCF_002086475 Mycobacterium_parmense_GCF_002102335 Mycobacterium_persicum_GCF_002086675 Mycobacterium_pseudokansasii_GCF_900566075 Mycobacterium_pseudoshottsii_GCF_003584745 Mycobacterium_riyadhense_GCF_002101845 Mycobacterium_saskatchewanense_GCF_002101875 Mycobacterium_scrofulaceum_GCF_002086735 Mycobacterium_sherrisii_GCF_002102355 Mycobacterium_shigaense_GCF_002356315 Mycobacterium_shimoidei_GCF_002101905 Mycobacterium_shinjukuense_GCF_002086755 Mycobacterium_simiae_GCF_002093075 Mycobacterium_sp_GCF_000328565 Mycobacterium_syngnathidarum_GCF_001942625 Mycobacterium_szulgai_GCF_002116635 Mycobacterium_talmoniae_GCF_002967005 Mycobacterium_tilburgii_GCF_902168065 Mycobacterium_timonense_GCF_002086775 Mycobacterium_triplex_GCF_000689255 Mycobacterium_uberis_GCF_003408705 Mycobacterium_ulcerans_GCF_000013925 Mycobacterium_vulneris_GCF_002104765 Mycobacterium_tuberculosis_GCF_000195955 --genome-base-path results/phyluce/mycobacterium/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/mycobacterium/validate_intervals/mycobacterium-to-Mycobacterium_tuberculosis_GCF_000195955.sqlite --output results/phyluce/mycobacterium/validate_intervals/lastz/;\n",
      "\n",
      "Running against Mycobacterium_ahvazicum_GCF_900176255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmph19tLP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2rBR4E.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_ahvazicum_GCF_900176255.lastz\n",
      "Creating Mycobacterium_ahvazicum_GCF_900176255 table\n",
      "Inserting data to Mycobacterium_ahvazicum_GCF_900176255 table\n",
      "\n",
      "Running against Mycobacterium_alsense_GCF_002086635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvQfF4R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuj7DHO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_alsense_GCF_002086635.lastz\n",
      "Creating Mycobacterium_alsense_GCF_002086635 table\n",
      "Inserting data to Mycobacterium_alsense_GCF_002086635 table\n",
      "\n",
      "Running against Mycobacterium_angelicum_GCF_002086155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_OqWdW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxtfir4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_angelicum_GCF_002086155.lastz\n",
      "Creating Mycobacterium_angelicum_GCF_002086155 table\n",
      "Inserting data to Mycobacterium_angelicum_GCF_002086155 table\n",
      "\n",
      "Running against Mycobacterium_aquaticum_GCF_002086485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZl8whj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpozyLV9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_aquaticum_GCF_002086485.lastz\n",
      "Creating Mycobacterium_aquaticum_GCF_002086485 table\n",
      "Inserting data to Mycobacterium_aquaticum_GCF_002086485 table\n",
      "\n",
      "Running against Mycobacterium_arosiense_GCF_002086125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpysjw14.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8tNDzL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_arosiense_GCF_002086125.lastz\n",
      "Creating Mycobacterium_arosiense_GCF_002086125 table\n",
      "Inserting data to Mycobacterium_arosiense_GCF_002086125 table\n",
      "\n",
      "Running against Mycobacterium_asiaticum_GCF_000613245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIBYNT5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaqKFxX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_asiaticum_GCF_000613245.lastz\n",
      "Creating Mycobacterium_asiaticum_GCF_000613245 table\n",
      "Inserting data to Mycobacterium_asiaticum_GCF_000613245 table\n",
      "\n",
      "Running against Mycobacterium_attenuatum_GCF_900566085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpI0rYtv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmps78lDk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_attenuatum_GCF_900566085.lastz\n",
      "Creating Mycobacterium_attenuatum_GCF_900566085 table\n",
      "Inserting data to Mycobacterium_attenuatum_GCF_900566085 table\n",
      "\n",
      "Running against Mycobacterium_avium_GCF_000007865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPFnU1r.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEIANW7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_avium_GCF_000007865.lastz\n",
      "Creating Mycobacterium_avium_GCF_000007865 table\n",
      "Inserting data to Mycobacterium_avium_GCF_000007865 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Mycobacterium_basiliense_GCF_900292015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbAscj9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2QMCbd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_basiliense_GCF_900292015.lastz\n",
      "Creating Mycobacterium_basiliense_GCF_900292015 table\n",
      "Inserting data to Mycobacterium_basiliense_GCF_900292015 table\n",
      "\n",
      "Running against Mycobacterium_bohemicum_GCF_001053185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7YGlnG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwGiqYL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_bohemicum_GCF_001053185.lastz\n",
      "Creating Mycobacterium_bohemicum_GCF_001053185 table\n",
      "Inserting data to Mycobacterium_bohemicum_GCF_001053185 table\n",
      "\n",
      "Running against Mycobacterium_bouchedurhonense_GCF_002086165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfNZR7a.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEcHMWu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_bouchedurhonense_GCF_002086165.lastz\n",
      "Creating Mycobacterium_bouchedurhonense_GCF_002086165 table\n",
      "Inserting data to Mycobacterium_bouchedurhonense_GCF_002086165 table\n",
      "\n",
      "Running against Mycobacterium_branderi_GCF_002086575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEZPguM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptbRf_5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_branderi_GCF_002086575.lastz\n",
      "Creating Mycobacterium_branderi_GCF_002086575 table\n",
      "Inserting data to Mycobacterium_branderi_GCF_002086575 table\n",
      "\n",
      "Running against Mycobacterium_canettii_GCF_000253375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPKtw9V.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRn2LHI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_canettii_GCF_000253375.lastz\n",
      "Creating Mycobacterium_canettii_GCF_000253375 table\n",
      "Inserting data to Mycobacterium_canettii_GCF_000253375 table\n",
      "\n",
      "Running against Mycobacterium_celatum_GCF_002101595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpXbrQDn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJWjQVy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_celatum_GCF_002101595.lastz\n",
      "Creating Mycobacterium_celatum_GCF_002101595 table\n",
      "Inserting data to Mycobacterium_celatum_GCF_002101595 table\n",
      "\n",
      "Running against Mycobacterium_chimaera_GCF_002219285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPu2xgJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZJM1ar.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_chimaera_GCF_002219285.lastz\n",
      "Creating Mycobacterium_chimaera_GCF_002219285 table\n",
      "Inserting data to Mycobacterium_chimaera_GCF_002219285 table\n",
      "\n",
      "Running against Mycobacterium_colombiense_GCF_002105755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5MyDQh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpz8keTs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_colombiense_GCF_002105755.lastz\n",
      "Creating Mycobacterium_colombiense_GCF_002105755 table\n",
      "Inserting data to Mycobacterium_colombiense_GCF_002105755 table\n",
      "\n",
      "Running against Mycobacterium_conspcuum_GCF_002102095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprJ6WFF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr8wZHo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_conspcuum_GCF_002102095.lastz\n",
      "Creating Mycobacterium_conspcuum_GCF_002102095 table\n",
      "Inserting data to Mycobacterium_conspcuum_GCF_002102095 table\n",
      "\n",
      "Running against Mycobacterium_decipiens_GCF_002104675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOQnz0q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5QpJ0P.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_decipiens_GCF_002104675.lastz\n",
      "Creating Mycobacterium_decipiens_GCF_002104675 table\n",
      "Inserting data to Mycobacterium_decipiens_GCF_002104675 table\n",
      "\n",
      "Running against Mycobacterium_dioxanotrophicus_GCF_002157835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpSen9WX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpv2l_TX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_dioxanotrophicus_GCF_002157835.lastz\n",
      "Creating Mycobacterium_dioxanotrophicus_GCF_002157835 table\n",
      "Inserting data to Mycobacterium_dioxanotrophicus_GCF_002157835 table\n",
      "\n",
      "Running against Mycobacterium_eburneum_GCF_004354905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmh5ENG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeMM7ty.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_eburneum_GCF_004354905.lastz\n",
      "Creating Mycobacterium_eburneum_GCF_004354905 table\n",
      "Inserting data to Mycobacterium_eburneum_GCF_004354905 table\n",
      "\n",
      "Running against Mycobacterium_europaeum_GCF_001373515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIfLD4u.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3d2tGz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_europaeum_GCF_001373515.lastz\n",
      "Creating Mycobacterium_europaeum_GCF_001373515 table\n",
      "Inserting data to Mycobacterium_europaeum_GCF_001373515 table\n",
      "\n",
      "Running against Mycobacterium_florentinum_GCF_002101635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1e30Zi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphxZOJj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_florentinum_GCF_002101635.lastz\n",
      "Creating Mycobacterium_florentinum_GCF_002101635 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Mycobacterium_florentinum_GCF_002101635 table\n",
      "\n",
      "Running against Mycobacterium_fragae_GCF_002102185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpXpkzrq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTziuKf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_fragae_GCF_002102185.lastz\n",
      "Creating Mycobacterium_fragae_GCF_002102185 table\n",
      "Inserting data to Mycobacterium_fragae_GCF_002102185 table\n",
      "\n",
      "Running against Mycobacterium_gastri_GCF_002102175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3pe4kc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZSuRd3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_gastri_GCF_002102175.lastz\n",
      "Creating Mycobacterium_gastri_GCF_002102175 table\n",
      "Inserting data to Mycobacterium_gastri_GCF_002102175 table\n",
      "\n",
      "Running against Mycobacterium_genavense_GCF_000526915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpog8qtP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa7OCAJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_genavense_GCF_000526915.lastz\n",
      "Creating Mycobacterium_genavense_GCF_000526915 table\n",
      "Inserting data to Mycobacterium_genavense_GCF_000526915 table\n",
      "\n",
      "Running against Mycobacterium_gordonae_GCF_002101675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKEqgdz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNJ0OZr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_gordonae_GCF_002101675.lastz\n",
      "Creating Mycobacterium_gordonae_GCF_002101675 table\n",
      "Inserting data to Mycobacterium_gordonae_GCF_002101675 table\n",
      "\n",
      "Running against Mycobacterium_grossiae_GCF_008329645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMuCnSE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwVB6EI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_grossiae_GCF_008329645.lastz\n",
      "Creating Mycobacterium_grossiae_GCF_008329645 table\n",
      "Inserting data to Mycobacterium_grossiae_GCF_008329645 table\n",
      "\n",
      "Running against Mycobacterium_haemophilum_GCF_000340435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5ssGag.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4Y2XJz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_haemophilum_GCF_000340435.lastz\n",
      "Creating Mycobacterium_haemophilum_GCF_000340435 table\n",
      "Inserting data to Mycobacterium_haemophilum_GCF_000340435 table\n",
      "\n",
      "Running against Mycobacterium_heckeshornense_GCF_001881585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1NfpeH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLttbjV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_heckeshornense_GCF_001881585.lastz\n",
      "Creating Mycobacterium_heckeshornense_GCF_001881585 table\n",
      "Inserting data to Mycobacterium_heckeshornense_GCF_001881585 table\n",
      "\n",
      "Running against Mycobacterium_heidelbergense_GCF_002086215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3219rh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpizLAJt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_heidelbergense_GCF_002086215.lastz\n",
      "Creating Mycobacterium_heidelbergense_GCF_002086215 table\n",
      "Inserting data to Mycobacterium_heidelbergense_GCF_002086215 table\n",
      "\n",
      "Running against Mycobacterium_innocens_GCF_900566055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWY6jEL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfbfqfN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_innocens_GCF_900566055.lastz\n",
      "Creating Mycobacterium_innocens_GCF_900566055 table\n",
      "Inserting data to Mycobacterium_innocens_GCF_900566055 table\n",
      "\n",
      "Running against Mycobacterium_interjectum_GCF_900078675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYEJz61.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpimM6hn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_interjectum_GCF_900078675.lastz\n",
      "Creating Mycobacterium_interjectum_GCF_900078675 table\n",
      "Inserting data to Mycobacterium_interjectum_GCF_900078675 table\n",
      "\n",
      "Running against Mycobacterium_intermedium_GCF_002086275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsY91ft.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7bZehz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_intermedium_GCF_002086275.lastz\n",
      "Creating Mycobacterium_intermedium_GCF_002086275 table\n",
      "Inserting data to Mycobacterium_intermedium_GCF_002086275 table\n",
      "\n",
      "Running against Mycobacterium_intracellulare_GCF_000277125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpUHUVII.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprc0ls1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_intracellulare_GCF_000277125.lastz\n",
      "Creating Mycobacterium_intracellulare_GCF_000277125 table\n",
      "Inserting data to Mycobacterium_intracellulare_GCF_000277125 table\n",
      "\n",
      "Running against Mycobacterium_kansasii_GCF_000157895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_nv7ju.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGE_6Hr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_kansasii_GCF_000157895.lastz\n",
      "Creating Mycobacterium_kansasii_GCF_000157895 table\n",
      "Inserting data to Mycobacterium_kansasii_GCF_000157895 table\n",
      "\n",
      "Running against Mycobacterium_kubicae_GCF_002101745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0lTKA5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9_6Wja.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_kubicae_GCF_002101745.lastz\n",
      "Creating Mycobacterium_kubicae_GCF_002101745 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Mycobacterium_kubicae_GCF_002101745 table\n",
      "\n",
      "Running against Mycobacterium_kyorinense_GCF_000759695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxFyehe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppJ6dJN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_kyorinense_GCF_000759695.lastz\n",
      "Creating Mycobacterium_kyorinense_GCF_000759695 table\n",
      "Inserting data to Mycobacterium_kyorinense_GCF_000759695 table\n",
      "\n",
      "Running against Mycobacterium_lacus_GCF_002102215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8CTHoA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoGzyfl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_lacus_GCF_002102215.lastz\n",
      "Creating Mycobacterium_lacus_GCF_002102215 table\n",
      "Inserting data to Mycobacterium_lacus_GCF_002102215 table\n",
      "\n",
      "Running against Mycobacterium_lehmannii_GCF_002245535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRR3od8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJb9Evn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_lehmannii_GCF_002245535.lastz\n",
      "Creating Mycobacterium_lehmannii_GCF_002245535 table\n",
      "Inserting data to Mycobacterium_lehmannii_GCF_002245535 table\n",
      "\n",
      "Running against Mycobacterium_lentiflavum_GCF_001373395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8MJHr8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjD2FgA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_lentiflavum_GCF_001373395.lastz\n",
      "Creating Mycobacterium_lentiflavum_GCF_001373395 table\n",
      "Inserting data to Mycobacterium_lentiflavum_GCF_001373395 table\n",
      "\n",
      "Running against Mycobacterium_leprae_GCF_000195855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiOzzR6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzmYPOd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_leprae_GCF_000195855.lastz\n",
      "Creating Mycobacterium_leprae_GCF_000195855 table\n",
      "Inserting data to Mycobacterium_leprae_GCF_000195855 table\n",
      "\n",
      "Running against Mycobacterium_lepraemurium_GCF_002291465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTz8Ido.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2O3q0b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_lepraemurium_GCF_002291465.lastz\n",
      "Creating Mycobacterium_lepraemurium_GCF_002291465 table\n",
      "Inserting data to Mycobacterium_lepraemurium_GCF_002291465 table\n",
      "\n",
      "Running against Mycobacterium_lepromatosis_GCF_000966355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpoxBc16.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkFD5Qf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_lepromatosis_GCF_000966355.lastz\n",
      "Creating Mycobacterium_lepromatosis_GCF_000966355 table\n",
      "Inserting data to Mycobacterium_lepromatosis_GCF_000966355 table\n",
      "\n",
      "Running against Mycobacterium_liflandii_GCF_000026445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMnitr0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFmqx2q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_liflandii_GCF_000026445.lastz\n",
      "Creating Mycobacterium_liflandii_GCF_000026445 table\n",
      "Inserting data to Mycobacterium_liflandii_GCF_000026445 table\n",
      "\n",
      "Running against Mycobacterium_malmoense_GCF_002086305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpM5a1TZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3Yojng.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_malmoense_GCF_002086305.lastz\n",
      "Creating Mycobacterium_malmoense_GCF_002086305 table\n",
      "Inserting data to Mycobacterium_malmoense_GCF_002086305 table\n",
      "\n",
      "Running against Mycobacterium_mantenii_GCF_002086335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBfcApd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOCK9A0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_mantenii_GCF_002086335.lastz\n",
      "Creating Mycobacterium_mantenii_GCF_002086335 table\n",
      "Inserting data to Mycobacterium_mantenii_GCF_002086335 table\n",
      "\n",
      "Running against Mycobacterium_marinum_GCF_000723425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnQzbqC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKzOarT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_marinum_GCF_000723425.lastz\n",
      "Creating Mycobacterium_marinum_GCF_000723425 table\n",
      "Inserting data to Mycobacterium_marinum_GCF_000723425 table\n",
      "\n",
      "Running against Mycobacterium_marseillense_GCF_002285715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6pRrMu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVpk8of.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_marseillense_GCF_002285715.lastz\n",
      "Creating Mycobacterium_marseillense_GCF_002285715 table\n",
      "Inserting data to Mycobacterium_marseillense_GCF_002285715 table\n",
      "\n",
      "Running against Mycobacterium_montefiorense_GCF_003112775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpa5Cc1O.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCA8Ute.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_montefiorense_GCF_003112775.lastz\n",
      "Creating Mycobacterium_montefiorense_GCF_003112775 table\n",
      "Inserting data to Mycobacterium_montefiorense_GCF_003112775 table\n",
      "\n",
      "Running against Mycobacterium_mungi_GCF_001652545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDG3DfV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOs6dmL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_mungi_GCF_001652545.lastz\n",
      "Creating Mycobacterium_mungi_GCF_001652545 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Mycobacterium_mungi_GCF_001652545 table\n",
      "\n",
      "Running against Mycobacterium_nebraskense_GCF_002102255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7nh_1z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOycUsL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_nebraskense_GCF_002102255.lastz\n",
      "Creating Mycobacterium_nebraskense_GCF_002102255 table\n",
      "Inserting data to Mycobacterium_nebraskense_GCF_002102255 table\n",
      "\n",
      "Running against Mycobacterium_neumannii_GCF_002245615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLpPRuf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNAynNn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_neumannii_GCF_002245615.lastz\n",
      "Creating Mycobacterium_neumannii_GCF_002245615 table\n",
      "Inserting data to Mycobacterium_neumannii_GCF_002245615 table\n",
      "\n",
      "Running against Mycobacterium_noviomagense_GCF_002086415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRY2TOh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0gQTyj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_noviomagense_GCF_002086415.lastz\n",
      "Creating Mycobacterium_noviomagense_GCF_002086415 table\n",
      "Inserting data to Mycobacterium_noviomagense_GCF_002086415 table\n",
      "\n",
      "Running against Mycobacterium_orygis_GCF_006385035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpz6MYI0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpq3MsXm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_orygis_GCF_006385035.lastz\n",
      "Creating Mycobacterium_orygis_GCF_006385035 table\n",
      "Inserting data to Mycobacterium_orygis_GCF_006385035 table\n",
      "\n",
      "Running against Mycobacterium_palustre_GCF_002101785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpv85Q24.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpI0_XVa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_palustre_GCF_002101785.lastz\n",
      "Creating Mycobacterium_palustre_GCF_002101785 table\n",
      "Inserting data to Mycobacterium_palustre_GCF_002101785 table\n",
      "\n",
      "Running against Mycobacterium_paraense_GCF_002101815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVq5gjW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4HWsWE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_paraense_GCF_002101815.lastz\n",
      "Creating Mycobacterium_paraense_GCF_002101815 table\n",
      "Inserting data to Mycobacterium_paraense_GCF_002101815 table\n",
      "\n",
      "Running against Mycobacterium_paraffinicum_GCF_001907675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBLSDZK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMYTwTb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_paraffinicum_GCF_001907675.lastz\n",
      "Creating Mycobacterium_paraffinicum_GCF_001907675 table\n",
      "Inserting data to Mycobacterium_paraffinicum_GCF_001907675 table\n",
      "\n",
      "Running against Mycobacterium_paragordonae_GCF_003614435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgnKTr7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1fKKEs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_paragordonae_GCF_003614435.lastz\n",
      "Creating Mycobacterium_paragordonae_GCF_003614435 table\n",
      "Inserting data to Mycobacterium_paragordonae_GCF_003614435 table\n",
      "\n",
      "Running against Mycobacterium_paraintracellulare_GCF_000276825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGL4hIP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3vHbz3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_paraintracellulare_GCF_000276825.lastz\n",
      "Creating Mycobacterium_paraintracellulare_GCF_000276825 table\n",
      "Inserting data to Mycobacterium_paraintracellulare_GCF_000276825 table\n",
      "\n",
      "Running against Mycobacterium_parascrofulaceum_GCF_000164135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpp7ea7e.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNV03Rg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_parascrofulaceum_GCF_000164135.lastz\n",
      "Creating Mycobacterium_parascrofulaceum_GCF_000164135 table\n",
      "Inserting data to Mycobacterium_parascrofulaceum_GCF_000164135 table\n",
      "\n",
      "Running against Mycobacterium_paraseoulense_GCF_002086475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfU89VY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQnPmCX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_paraseoulense_GCF_002086475.lastz\n",
      "Creating Mycobacterium_paraseoulense_GCF_002086475 table\n",
      "Inserting data to Mycobacterium_paraseoulense_GCF_002086475 table\n",
      "\n",
      "Running against Mycobacterium_parmense_GCF_002102335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEHNYRK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8OwwLC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_parmense_GCF_002102335.lastz\n",
      "Creating Mycobacterium_parmense_GCF_002102335 table\n",
      "Inserting data to Mycobacterium_parmense_GCF_002102335 table\n",
      "\n",
      "Running against Mycobacterium_persicum_GCF_002086675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpL09MEs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeZbiTa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_persicum_GCF_002086675.lastz\n",
      "Creating Mycobacterium_persicum_GCF_002086675 table\n",
      "Inserting data to Mycobacterium_persicum_GCF_002086675 table\n",
      "\n",
      "Running against Mycobacterium_pseudokansasii_GCF_900566075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpv4trIc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9FueDw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_pseudokansasii_GCF_900566075.lastz\n",
      "Creating Mycobacterium_pseudokansasii_GCF_900566075 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Mycobacterium_pseudokansasii_GCF_900566075 table\n",
      "\n",
      "Running against Mycobacterium_pseudoshottsii_GCF_003584745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvGkav_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9PPyK9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_pseudoshottsii_GCF_003584745.lastz\n",
      "Creating Mycobacterium_pseudoshottsii_GCF_003584745 table\n",
      "Inserting data to Mycobacterium_pseudoshottsii_GCF_003584745 table\n",
      "\n",
      "Running against Mycobacterium_riyadhense_GCF_002101845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpI3WgTu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpC0aCnW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_riyadhense_GCF_002101845.lastz\n",
      "Creating Mycobacterium_riyadhense_GCF_002101845 table\n",
      "Inserting data to Mycobacterium_riyadhense_GCF_002101845 table\n",
      "\n",
      "Running against Mycobacterium_saskatchewanense_GCF_002101875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMeu0hP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpemEmhP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_saskatchewanense_GCF_002101875.lastz\n",
      "Creating Mycobacterium_saskatchewanense_GCF_002101875 table\n",
      "Inserting data to Mycobacterium_saskatchewanense_GCF_002101875 table\n",
      "\n",
      "Running against Mycobacterium_scrofulaceum_GCF_002086735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpF4VaxS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKBqCrV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_scrofulaceum_GCF_002086735.lastz\n",
      "Creating Mycobacterium_scrofulaceum_GCF_002086735 table\n",
      "Inserting data to Mycobacterium_scrofulaceum_GCF_002086735 table\n",
      "\n",
      "Running against Mycobacterium_sherrisii_GCF_002102355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLqnxfw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdRs7oh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_sherrisii_GCF_002102355.lastz\n",
      "Creating Mycobacterium_sherrisii_GCF_002102355 table\n",
      "Inserting data to Mycobacterium_sherrisii_GCF_002102355 table\n",
      "\n",
      "Running against Mycobacterium_shigaense_GCF_002356315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCpBQr7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZduP2P.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_shigaense_GCF_002356315.lastz\n",
      "Creating Mycobacterium_shigaense_GCF_002356315 table\n",
      "Inserting data to Mycobacterium_shigaense_GCF_002356315 table\n",
      "\n",
      "Running against Mycobacterium_shimoidei_GCF_002101905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxvISTr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0E3eUI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_shimoidei_GCF_002101905.lastz\n",
      "Creating Mycobacterium_shimoidei_GCF_002101905 table\n",
      "Inserting data to Mycobacterium_shimoidei_GCF_002101905 table\n",
      "\n",
      "Running against Mycobacterium_shinjukuense_GCF_002086755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpik_alk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphbNrqO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_shinjukuense_GCF_002086755.lastz\n",
      "Creating Mycobacterium_shinjukuense_GCF_002086755 table\n",
      "Inserting data to Mycobacterium_shinjukuense_GCF_002086755 table\n",
      "\n",
      "Running against Mycobacterium_simiae_GCF_002093075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1OjY5n.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxALwMK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_simiae_GCF_002093075.lastz\n",
      "Creating Mycobacterium_simiae_GCF_002093075 table\n",
      "Inserting data to Mycobacterium_simiae_GCF_002093075 table\n",
      "\n",
      "Running against Mycobacterium_sp_GCF_000328565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyLKw1Q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpesFbSo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_sp_GCF_000328565.lastz\n",
      "Creating Mycobacterium_sp_GCF_000328565 table\n",
      "Inserting data to Mycobacterium_sp_GCF_000328565 table\n",
      "\n",
      "Running against Mycobacterium_syngnathidarum_GCF_001942625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplCbL97.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcg1huB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_syngnathidarum_GCF_001942625.lastz\n",
      "Creating Mycobacterium_syngnathidarum_GCF_001942625 table\n",
      "Inserting data to Mycobacterium_syngnathidarum_GCF_001942625 table\n",
      "\n",
      "Running against Mycobacterium_szulgai_GCF_002116635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpirws3x.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSLH2VF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_szulgai_GCF_002116635.lastz\n",
      "Creating Mycobacterium_szulgai_GCF_002116635 table\n",
      "Inserting data to Mycobacterium_szulgai_GCF_002116635 table\n",
      "\n",
      "Running against Mycobacterium_talmoniae_GCF_002967005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyRYTyZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzmaqv7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_talmoniae_GCF_002967005.lastz\n",
      "Creating Mycobacterium_talmoniae_GCF_002967005 table\n",
      "Inserting data to Mycobacterium_talmoniae_GCF_002967005 table\n",
      "\n",
      "Running against Mycobacterium_tilburgii_GCF_902168065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMjL4dH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsc_J_l.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_tilburgii_GCF_902168065.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Mycobacterium_tilburgii_GCF_902168065 table\n",
      "Inserting data to Mycobacterium_tilburgii_GCF_902168065 table\n",
      "\n",
      "Running against Mycobacterium_timonense_GCF_002086775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDDz2uc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvKiEqx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_timonense_GCF_002086775.lastz\n",
      "Creating Mycobacterium_timonense_GCF_002086775 table\n",
      "Inserting data to Mycobacterium_timonense_GCF_002086775 table\n",
      "\n",
      "Running against Mycobacterium_triplex_GCF_000689255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3oGfQR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNNxzCS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_triplex_GCF_000689255.lastz\n",
      "Creating Mycobacterium_triplex_GCF_000689255 table\n",
      "Inserting data to Mycobacterium_triplex_GCF_000689255 table\n",
      "\n",
      "Running against Mycobacterium_uberis_GCF_003408705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptIRbNu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUBvQlN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_uberis_GCF_003408705.lastz\n",
      "Creating Mycobacterium_uberis_GCF_003408705 table\n",
      "Inserting data to Mycobacterium_uberis_GCF_003408705 table\n",
      "\n",
      "Running against Mycobacterium_ulcerans_GCF_000013925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpx47ldD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSO1FzE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_ulcerans_GCF_000013925.lastz\n",
      "Creating Mycobacterium_ulcerans_GCF_000013925 table\n",
      "Inserting data to Mycobacterium_ulcerans_GCF_000013925 table\n",
      "\n",
      "Running against Mycobacterium_vulneris_GCF_002104765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMtdPgU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr80TzN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_vulneris_GCF_002104765.lastz\n",
      "Creating Mycobacterium_vulneris_GCF_002104765 table\n",
      "Inserting data to Mycobacterium_vulneris_GCF_002104765 table\n",
      "\n",
      "Running against Mycobacterium_tuberculosis_GCF_000195955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmppP4sdY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdIqbrx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/mycobacterium/validate_intervals/lastz/Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_Mycobacterium_tuberculosis_GCF_000195955.lastz\n",
      "Creating Mycobacterium_tuberculosis_GCF_000195955 table\n",
      "Inserting data to Mycobacterium_tuberculosis_GCF_000195955 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/mycobacterium/extract_probes_from_group/mycobacterium_genome.conf --lastz results/phyluce/mycobacterium/validate_intervals/lastz --probes 120 --probe-prefix uce_mycobacterium_ --name-pattern \"Mycobacterium_tuberculosis_GCF_000195955_+70_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/mycobacterium/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 14:25:42,916 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 14:25:42,923 - Phyluce - INFO - ---- Working on Mycobacterium_ahvazicum_GCF_900176255 genome ----\n",
      "2020-02-06 14:25:42,931 - Phyluce - INFO - Reading Mycobacterium_ahvazicum_GCF_900176255 genome\n",
      "2020-02-06 14:25:43,677 - Phyluce - INFO - Mycobacterium_ahvazicum_GCF_900176255: 172 uces, 18 dupes, 154 non-dupes, 4 orient drop, 16 length drop, 134 written\n",
      "2020-02-06 14:25:43,678 - Phyluce - INFO - ----- Working on Mycobacterium_alsense_GCF_002086635 genome -----\n",
      "2020-02-06 14:25:43,686 - Phyluce - INFO - Reading Mycobacterium_alsense_GCF_002086635 genome\n",
      "2020-02-06 14:25:44,341 - Phyluce - INFO - Mycobacterium_alsense_GCF_002086635: 172 uces, 29 dupes, 143 non-dupes, 0 orient drop, 5 length drop, 138 written\n",
      "2020-02-06 14:25:44,341 - Phyluce - INFO - ---- Working on Mycobacterium_angelicum_GCF_002086155 genome ----\n",
      "2020-02-06 14:25:44,349 - Phyluce - INFO - Reading Mycobacterium_angelicum_GCF_002086155 genome\n",
      "2020-02-06 14:25:45,009 - Phyluce - INFO - Mycobacterium_angelicum_GCF_002086155: 172 uces, 22 dupes, 150 non-dupes, 0 orient drop, 5 length drop, 145 written\n",
      "2020-02-06 14:25:45,009 - Phyluce - INFO - ---- Working on Mycobacterium_aquaticum_GCF_002086485 genome ----\n",
      "2020-02-06 14:25:45,015 - Phyluce - INFO - Reading Mycobacterium_aquaticum_GCF_002086485 genome\n",
      "2020-02-06 14:25:45,717 - Phyluce - INFO - Mycobacterium_aquaticum_GCF_002086485: 172 uces, 36 dupes, 136 non-dupes, 0 orient drop, 4 length drop, 132 written\n",
      "2020-02-06 14:25:45,717 - Phyluce - INFO - ---- Working on Mycobacterium_arosiense_GCF_002086125 genome ----\n",
      "2020-02-06 14:25:45,718 - Phyluce - INFO - Reading Mycobacterium_arosiense_GCF_002086125 genome\n",
      "2020-02-06 14:25:46,402 - Phyluce - INFO - Mycobacterium_arosiense_GCF_002086125: 172 uces, 30 dupes, 142 non-dupes, 0 orient drop, 4 length drop, 138 written\n",
      "2020-02-06 14:25:46,402 - Phyluce - INFO - ---- Working on Mycobacterium_asiaticum_GCF_000613245 genome ----\n",
      "2020-02-06 14:25:46,413 - Phyluce - INFO - Reading Mycobacterium_asiaticum_GCF_000613245 genome\n",
      "2020-02-06 14:25:47,080 - Phyluce - INFO - Mycobacterium_asiaticum_GCF_000613245: 172 uces, 21 dupes, 151 non-dupes, 2 orient drop, 9 length drop, 140 written\n",
      "2020-02-06 14:25:47,080 - Phyluce - INFO - ---- Working on Mycobacterium_attenuatum_GCF_900566085 genome ---\n",
      "2020-02-06 14:25:47,081 - Phyluce - INFO - Reading Mycobacterium_attenuatum_GCF_900566085 genome\n",
      "2020-02-06 14:25:47,733 - Phyluce - INFO - Mycobacterium_attenuatum_GCF_900566085: 172 uces, 23 dupes, 149 non-dupes, 0 orient drop, 3 length drop, 146 written\n",
      "2020-02-06 14:25:47,733 - Phyluce - INFO - ------ Working on Mycobacterium_avium_GCF_000007865 genome ------\n",
      "2020-02-06 14:25:47,734 - Phyluce - INFO - Reading Mycobacterium_avium_GCF_000007865 genome\n",
      "2020-02-06 14:25:48,397 - Phyluce - INFO - Mycobacterium_avium_GCF_000007865: 172 uces, 0 dupes, 172 non-dupes, 13 orient drop, 23 length drop, 136 written\n",
      "2020-02-06 14:25:48,398 - Phyluce - INFO - ---- Working on Mycobacterium_basiliense_GCF_900292015 genome ---\n",
      "2020-02-06 14:25:48,415 - Phyluce - INFO - Reading Mycobacterium_basiliense_GCF_900292015 genome\n",
      "2020-02-06 14:25:49,052 - Phyluce - INFO - Mycobacterium_basiliense_GCF_900292015: 172 uces, 0 dupes, 172 non-dupes, 13 orient drop, 11 length drop, 148 written\n",
      "2020-02-06 14:25:49,052 - Phyluce - INFO - ---- Working on Mycobacterium_bohemicum_GCF_001053185 genome ----\n",
      "2020-02-06 14:25:49,053 - Phyluce - INFO - Reading Mycobacterium_bohemicum_GCF_001053185 genome\n",
      "2020-02-06 14:25:49,725 - Phyluce - INFO - Mycobacterium_bohemicum_GCF_001053185: 171 uces, 27 dupes, 144 non-dupes, 2 orient drop, 5 length drop, 137 written\n",
      "2020-02-06 14:25:49,725 - Phyluce - INFO - - Working on Mycobacterium_bouchedurhonense_GCF_002086165 genome \n",
      "2020-02-06 14:25:49,726 - Phyluce - INFO - Reading Mycobacterium_bouchedurhonense_GCF_002086165 genome\n",
      "2020-02-06 14:25:50,421 - Phyluce - INFO - Mycobacterium_bouchedurhonense_GCF_002086165: 172 uces, 31 dupes, 141 non-dupes, 1 orient drop, 4 length drop, 136 written\n",
      "2020-02-06 14:25:50,422 - Phyluce - INFO - ----- Working on Mycobacterium_branderi_GCF_002086575 genome ----\n",
      "2020-02-06 14:25:50,443 - Phyluce - INFO - Reading Mycobacterium_branderi_GCF_002086575 genome\n",
      "2020-02-06 14:25:51,116 - Phyluce - INFO - Mycobacterium_branderi_GCF_002086575: 172 uces, 26 dupes, 146 non-dupes, 0 orient drop, 5 length drop, 141 written\n",
      "2020-02-06 14:25:51,116 - Phyluce - INFO - ----- Working on Mycobacterium_canettii_GCF_000253375 genome ----\n",
      "2020-02-06 14:25:51,123 - Phyluce - INFO - Reading Mycobacterium_canettii_GCF_000253375 genome\n",
      "2020-02-06 14:25:51,754 - Phyluce - INFO - Mycobacterium_canettii_GCF_000253375: 172 uces, 0 dupes, 172 non-dupes, 10 orient drop, 15 length drop, 147 written\n",
      "2020-02-06 14:25:51,754 - Phyluce - INFO - ----- Working on Mycobacterium_celatum_GCF_002101595 genome -----\n",
      "2020-02-06 14:25:51,755 - Phyluce - INFO - Reading Mycobacterium_celatum_GCF_002101595 genome\n",
      "2020-02-06 14:25:52,401 - Phyluce - INFO - Mycobacterium_celatum_GCF_002101595: 172 uces, 25 dupes, 147 non-dupes, 0 orient drop, 3 length drop, 144 written\n",
      "2020-02-06 14:25:52,401 - Phyluce - INFO - ----- Working on Mycobacterium_chimaera_GCF_002219285 genome ----\n",
      "2020-02-06 14:25:52,413 - Phyluce - INFO - Reading Mycobacterium_chimaera_GCF_002219285 genome\n",
      "2020-02-06 14:25:53,110 - Phyluce - INFO - Mycobacterium_chimaera_GCF_002219285: 172 uces, 1 dupes, 171 non-dupes, 20 orient drop, 11 length drop, 140 written\n",
      "2020-02-06 14:25:53,111 - Phyluce - INFO - --- Working on Mycobacterium_colombiense_GCF_002105755 genome ---\n",
      "2020-02-06 14:25:53,111 - Phyluce - INFO - Reading Mycobacterium_colombiense_GCF_002105755 genome\n",
      "2020-02-06 14:25:53,766 - Phyluce - INFO - Mycobacterium_colombiense_GCF_002105755: 172 uces, 0 dupes, 172 non-dupes, 17 orient drop, 11 length drop, 144 written\n",
      "2020-02-06 14:25:53,766 - Phyluce - INFO - ---- Working on Mycobacterium_conspcuum_GCF_002102095 genome ----\n",
      "2020-02-06 14:25:53,780 - Phyluce - INFO - Reading Mycobacterium_conspcuum_GCF_002102095 genome\n",
      "2020-02-06 14:25:54,457 - Phyluce - INFO - Mycobacterium_conspcuum_GCF_002102095: 172 uces, 29 dupes, 143 non-dupes, 0 orient drop, 4 length drop, 139 written\n",
      "2020-02-06 14:25:54,457 - Phyluce - INFO - ---- Working on Mycobacterium_decipiens_GCF_002104675 genome ----\n",
      "2020-02-06 14:25:54,467 - Phyluce - INFO - Reading Mycobacterium_decipiens_GCF_002104675 genome\n",
      "2020-02-06 14:25:55,123 - Phyluce - INFO - Mycobacterium_decipiens_GCF_002104675: 172 uces, 25 dupes, 147 non-dupes, 0 orient drop, 4 length drop, 143 written\n",
      "2020-02-06 14:25:55,123 - Phyluce - INFO - - Working on Mycobacterium_dioxanotrophicus_GCF_002157835 genome \n",
      "2020-02-06 14:25:55,133 - Phyluce - INFO - Reading Mycobacterium_dioxanotrophicus_GCF_002157835 genome\n",
      "2020-02-06 14:25:55,912 - Phyluce - INFO - Mycobacterium_dioxanotrophicus_GCF_002157835: 171 uces, 2 dupes, 169 non-dupes, 31 orient drop, 12 length drop, 126 written\n",
      "2020-02-06 14:25:55,912 - Phyluce - INFO - ----- Working on Mycobacterium_eburneum_GCF_004354905 genome ----\n",
      "2020-02-06 14:25:55,923 - Phyluce - INFO - Reading Mycobacterium_eburneum_GCF_004354905 genome\n",
      "2020-02-06 14:25:56,601 - Phyluce - INFO - Mycobacterium_eburneum_GCF_004354905: 172 uces, 28 dupes, 144 non-dupes, 1 orient drop, 3 length drop, 140 written\n",
      "2020-02-06 14:25:56,601 - Phyluce - INFO - ---- Working on Mycobacterium_europaeum_GCF_001373515 genome ----\n",
      "2020-02-06 14:25:56,609 - Phyluce - INFO - Reading Mycobacterium_europaeum_GCF_001373515 genome\n",
      "2020-02-06 14:25:57,310 - Phyluce - INFO - Mycobacterium_europaeum_GCF_001373515: 172 uces, 12 dupes, 160 non-dupes, 13 orient drop, 9 length drop, 138 written\n",
      "2020-02-06 14:25:57,310 - Phyluce - INFO - --- Working on Mycobacterium_florentinum_GCF_002101635 genome ---\n",
      "2020-02-06 14:25:57,319 - Phyluce - INFO - Reading Mycobacterium_florentinum_GCF_002101635 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 14:25:58,009 - Phyluce - INFO - Mycobacterium_florentinum_GCF_002101635: 172 uces, 26 dupes, 146 non-dupes, 2 orient drop, 5 length drop, 139 written\n",
      "2020-02-06 14:25:58,009 - Phyluce - INFO - ------ Working on Mycobacterium_fragae_GCF_002102185 genome -----\n",
      "2020-02-06 14:25:58,023 - Phyluce - INFO - Reading Mycobacterium_fragae_GCF_002102185 genome\n",
      "2020-02-06 14:25:58,684 - Phyluce - INFO - Mycobacterium_fragae_GCF_002102185: 172 uces, 25 dupes, 147 non-dupes, 0 orient drop, 4 length drop, 143 written\n",
      "2020-02-06 14:25:58,684 - Phyluce - INFO - ------ Working on Mycobacterium_gastri_GCF_002102175 genome -----\n",
      "2020-02-06 14:25:58,685 - Phyluce - INFO - Reading Mycobacterium_gastri_GCF_002102175 genome\n",
      "2020-02-06 14:25:59,341 - Phyluce - INFO - Mycobacterium_gastri_GCF_002102175: 172 uces, 26 dupes, 146 non-dupes, 0 orient drop, 4 length drop, 142 written\n",
      "2020-02-06 14:25:59,342 - Phyluce - INFO - ---- Working on Mycobacterium_genavense_GCF_000526915 genome ----\n",
      "2020-02-06 14:25:59,342 - Phyluce - INFO - Reading Mycobacterium_genavense_GCF_000526915 genome\n",
      "2020-02-06 14:25:59,982 - Phyluce - INFO - Mycobacterium_genavense_GCF_000526915: 172 uces, 16 dupes, 156 non-dupes, 2 orient drop, 9 length drop, 145 written\n",
      "2020-02-06 14:25:59,982 - Phyluce - INFO - ----- Working on Mycobacterium_gordonae_GCF_002101675 genome ----\n",
      "2020-02-06 14:25:59,982 - Phyluce - INFO - Reading Mycobacterium_gordonae_GCF_002101675 genome\n",
      "2020-02-06 14:26:00,694 - Phyluce - INFO - Mycobacterium_gordonae_GCF_002101675: 172 uces, 29 dupes, 143 non-dupes, 0 orient drop, 3 length drop, 140 written\n",
      "2020-02-06 14:26:00,694 - Phyluce - INFO - ----- Working on Mycobacterium_grossiae_GCF_008329645 genome ----\n",
      "2020-02-06 14:26:00,695 - Phyluce - INFO - Reading Mycobacterium_grossiae_GCF_008329645 genome\n",
      "2020-02-06 14:26:01,387 - Phyluce - INFO - Mycobacterium_grossiae_GCF_008329645: 171 uces, 0 dupes, 171 non-dupes, 26 orient drop, 17 length drop, 128 written\n",
      "2020-02-06 14:26:01,388 - Phyluce - INFO - --- Working on Mycobacterium_haemophilum_GCF_000340435 genome ---\n",
      "2020-02-06 14:26:01,400 - Phyluce - INFO - Reading Mycobacterium_haemophilum_GCF_000340435 genome\n",
      "2020-02-06 14:26:02,031 - Phyluce - INFO - Mycobacterium_haemophilum_GCF_000340435: 170 uces, 0 dupes, 170 non-dupes, 11 orient drop, 14 length drop, 145 written\n",
      "2020-02-06 14:26:02,031 - Phyluce - INFO - -- Working on Mycobacterium_heckeshornense_GCF_001881585 genome -\n",
      "2020-02-06 14:26:02,041 - Phyluce - INFO - Reading Mycobacterium_heckeshornense_GCF_001881585 genome\n",
      "2020-02-06 14:26:02,687 - Phyluce - INFO - Mycobacterium_heckeshornense_GCF_001881585: 169 uces, 21 dupes, 148 non-dupes, 0 orient drop, 9 length drop, 139 written\n",
      "2020-02-06 14:26:02,688 - Phyluce - INFO - -- Working on Mycobacterium_heidelbergense_GCF_002086215 genome -\n",
      "2020-02-06 14:26:02,702 - Phyluce - INFO - Reading Mycobacterium_heidelbergense_GCF_002086215 genome\n",
      "2020-02-06 14:26:03,377 - Phyluce - INFO - Mycobacterium_heidelbergense_GCF_002086215: 172 uces, 29 dupes, 143 non-dupes, 0 orient drop, 6 length drop, 137 written\n",
      "2020-02-06 14:26:03,377 - Phyluce - INFO - ----- Working on Mycobacterium_innocens_GCF_900566055 genome ----\n",
      "2020-02-06 14:26:03,378 - Phyluce - INFO - Reading Mycobacterium_innocens_GCF_900566055 genome\n",
      "2020-02-06 14:26:04,035 - Phyluce - INFO - Mycobacterium_innocens_GCF_900566055: 172 uces, 24 dupes, 148 non-dupes, 0 orient drop, 3 length drop, 145 written\n",
      "2020-02-06 14:26:04,036 - Phyluce - INFO - --- Working on Mycobacterium_interjectum_GCF_900078675 genome ---\n",
      "2020-02-06 14:26:04,036 - Phyluce - INFO - Reading Mycobacterium_interjectum_GCF_900078675 genome\n",
      "2020-02-06 14:26:04,714 - Phyluce - INFO - Mycobacterium_interjectum_GCF_900078675: 171 uces, 19 dupes, 152 non-dupes, 11 orient drop, 9 length drop, 132 written\n",
      "2020-02-06 14:26:04,714 - Phyluce - INFO - --- Working on Mycobacterium_intermedium_GCF_002086275 genome ---\n",
      "2020-02-06 14:26:04,715 - Phyluce - INFO - Reading Mycobacterium_intermedium_GCF_002086275 genome\n",
      "2020-02-06 14:26:05,395 - Phyluce - INFO - Mycobacterium_intermedium_GCF_002086275: 172 uces, 36 dupes, 136 non-dupes, 1 orient drop, 2 length drop, 133 written\n",
      "2020-02-06 14:26:05,395 - Phyluce - INFO - -- Working on Mycobacterium_intracellulare_GCF_000277125 genome -\n",
      "2020-02-06 14:26:05,395 - Phyluce - INFO - Reading Mycobacterium_intracellulare_GCF_000277125 genome\n",
      "2020-02-06 14:26:06,067 - Phyluce - INFO - Mycobacterium_intracellulare_GCF_000277125: 172 uces, 0 dupes, 172 non-dupes, 19 orient drop, 14 length drop, 139 written\n",
      "2020-02-06 14:26:06,067 - Phyluce - INFO - ----- Working on Mycobacterium_kansasii_GCF_000157895 genome ----\n",
      "2020-02-06 14:26:06,068 - Phyluce - INFO - Reading Mycobacterium_kansasii_GCF_000157895 genome\n",
      "2020-02-06 14:26:06,713 - Phyluce - INFO - Mycobacterium_kansasii_GCF_000157895: 172 uces, 1 dupes, 171 non-dupes, 12 orient drop, 12 length drop, 147 written\n",
      "2020-02-06 14:26:06,713 - Phyluce - INFO - ----- Working on Mycobacterium_kubicae_GCF_002101745 genome -----\n",
      "2020-02-06 14:26:06,714 - Phyluce - INFO - Reading Mycobacterium_kubicae_GCF_002101745 genome\n",
      "2020-02-06 14:26:07,371 - Phyluce - INFO - Mycobacterium_kubicae_GCF_002101745: 172 uces, 27 dupes, 145 non-dupes, 0 orient drop, 4 length drop, 141 written\n",
      "2020-02-06 14:26:07,371 - Phyluce - INFO - ---- Working on Mycobacterium_kyorinense_GCF_000759695 genome ---\n",
      "2020-02-06 14:26:07,372 - Phyluce - INFO - Reading Mycobacterium_kyorinense_GCF_000759695 genome\n",
      "2020-02-06 14:26:08,039 - Phyluce - INFO - Mycobacterium_kyorinense_GCF_000759695: 172 uces, 26 dupes, 146 non-dupes, 0 orient drop, 4 length drop, 142 written\n",
      "2020-02-06 14:26:08,039 - Phyluce - INFO - ------ Working on Mycobacterium_lacus_GCF_002102215 genome ------\n",
      "2020-02-06 14:26:08,054 - Phyluce - INFO - Reading Mycobacterium_lacus_GCF_002102215 genome\n",
      "2020-02-06 14:26:08,718 - Phyluce - INFO - Mycobacterium_lacus_GCF_002102215: 172 uces, 28 dupes, 144 non-dupes, 0 orient drop, 1 length drop, 143 written\n",
      "2020-02-06 14:26:08,718 - Phyluce - INFO - ---- Working on Mycobacterium_lehmannii_GCF_002245535 genome ----\n",
      "2020-02-06 14:26:08,724 - Phyluce - INFO - Reading Mycobacterium_lehmannii_GCF_002245535 genome\n",
      "2020-02-06 14:26:09,400 - Phyluce - INFO - Mycobacterium_lehmannii_GCF_002245535: 171 uces, 26 dupes, 145 non-dupes, 0 orient drop, 9 length drop, 136 written\n",
      "2020-02-06 14:26:09,400 - Phyluce - INFO - --- Working on Mycobacterium_lentiflavum_GCF_001373395 genome ---\n",
      "2020-02-06 14:26:09,401 - Phyluce - INFO - Reading Mycobacterium_lentiflavum_GCF_001373395 genome\n",
      "2020-02-06 14:26:10,094 - Phyluce - INFO - Mycobacterium_lentiflavum_GCF_001373395: 172 uces, 3 dupes, 169 non-dupes, 17 orient drop, 14 length drop, 138 written\n",
      "2020-02-06 14:26:10,095 - Phyluce - INFO - ------ Working on Mycobacterium_leprae_GCF_000195855 genome -----\n",
      "2020-02-06 14:26:10,095 - Phyluce - INFO - Reading Mycobacterium_leprae_GCF_000195855 genome\n",
      "2020-02-06 14:26:10,660 - Phyluce - INFO - Mycobacterium_leprae_GCF_000195855: 164 uces, 0 dupes, 164 non-dupes, 12 orient drop, 7 length drop, 145 written\n",
      "2020-02-06 14:26:10,660 - Phyluce - INFO - --- Working on Mycobacterium_lepraemurium_GCF_002291465 genome --\n",
      "2020-02-06 14:26:10,675 - Phyluce - INFO - Reading Mycobacterium_lepraemurium_GCF_002291465 genome\n",
      "2020-02-06 14:26:11,310 - Phyluce - INFO - Mycobacterium_lepraemurium_GCF_002291465: 171 uces, 0 dupes, 171 non-dupes, 14 orient drop, 14 length drop, 143 written\n",
      "2020-02-06 14:26:11,310 - Phyluce - INFO - --- Working on Mycobacterium_lepromatosis_GCF_000966355 genome --\n",
      "2020-02-06 14:26:11,311 - Phyluce - INFO - Reading Mycobacterium_lepromatosis_GCF_000966355 genome\n",
      "2020-02-06 14:26:11,884 - Phyluce - INFO - Mycobacterium_lepromatosis_GCF_000966355: 165 uces, 15 dupes, 150 non-dupes, 0 orient drop, 5 length drop, 145 written\n",
      "2020-02-06 14:26:11,884 - Phyluce - INFO - ---- Working on Mycobacterium_liflandii_GCF_000026445 genome ----\n",
      "2020-02-06 14:26:11,885 - Phyluce - INFO - Reading Mycobacterium_liflandii_GCF_000026445 genome\n",
      "2020-02-06 14:26:12,536 - Phyluce - INFO - Mycobacterium_liflandii_GCF_000026445: 172 uces, 0 dupes, 172 non-dupes, 14 orient drop, 13 length drop, 145 written\n",
      "2020-02-06 14:26:12,536 - Phyluce - INFO - ---- Working on Mycobacterium_malmoense_GCF_002086305 genome ----\n",
      "2020-02-06 14:26:12,537 - Phyluce - INFO - Reading Mycobacterium_malmoense_GCF_002086305 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 14:26:13,206 - Phyluce - INFO - Mycobacterium_malmoense_GCF_002086305: 172 uces, 29 dupes, 143 non-dupes, 0 orient drop, 7 length drop, 136 written\n",
      "2020-02-06 14:26:13,206 - Phyluce - INFO - ----- Working on Mycobacterium_mantenii_GCF_002086335 genome ----\n",
      "2020-02-06 14:26:13,207 - Phyluce - INFO - Reading Mycobacterium_mantenii_GCF_002086335 genome\n",
      "2020-02-06 14:26:13,867 - Phyluce - INFO - Mycobacterium_mantenii_GCF_002086335: 172 uces, 27 dupes, 145 non-dupes, 0 orient drop, 4 length drop, 141 written\n",
      "2020-02-06 14:26:13,868 - Phyluce - INFO - ----- Working on Mycobacterium_marinum_GCF_000723425 genome -----\n",
      "2020-02-06 14:26:13,884 - Phyluce - INFO - Reading Mycobacterium_marinum_GCF_000723425 genome\n",
      "2020-02-06 14:26:14,559 - Phyluce - INFO - Mycobacterium_marinum_GCF_000723425: 172 uces, 0 dupes, 172 non-dupes, 14 orient drop, 20 length drop, 138 written\n",
      "2020-02-06 14:26:14,559 - Phyluce - INFO - --- Working on Mycobacterium_marseillense_GCF_002285715 genome --\n",
      "2020-02-06 14:26:14,560 - Phyluce - INFO - Reading Mycobacterium_marseillense_GCF_002285715 genome\n",
      "2020-02-06 14:26:15,249 - Phyluce - INFO - Mycobacterium_marseillense_GCF_002285715: 172 uces, 0 dupes, 172 non-dupes, 21 orient drop, 16 length drop, 135 written\n",
      "2020-02-06 14:26:15,250 - Phyluce - INFO - -- Working on Mycobacterium_montefiorense_GCF_003112775 genome --\n",
      "2020-02-06 14:26:15,250 - Phyluce - INFO - Reading Mycobacterium_montefiorense_GCF_003112775 genome\n",
      "2020-02-06 14:26:15,915 - Phyluce - INFO - Mycobacterium_montefiorense_GCF_003112775: 172 uces, 24 dupes, 148 non-dupes, 2 orient drop, 5 length drop, 141 written\n",
      "2020-02-06 14:26:15,916 - Phyluce - INFO - ------ Working on Mycobacterium_mungi_GCF_001652545 genome ------\n",
      "2020-02-06 14:26:15,916 - Phyluce - INFO - Reading Mycobacterium_mungi_GCF_001652545 genome\n",
      "2020-02-06 14:26:16,550 - Phyluce - INFO - Mycobacterium_mungi_GCF_001652545: 172 uces, 22 dupes, 150 non-dupes, 0 orient drop, 3 length drop, 147 written\n",
      "2020-02-06 14:26:16,550 - Phyluce - INFO - --- Working on Mycobacterium_nebraskense_GCF_002102255 genome ---\n",
      "2020-02-06 14:26:16,551 - Phyluce - INFO - Reading Mycobacterium_nebraskense_GCF_002102255 genome\n",
      "2020-02-06 14:26:17,221 - Phyluce - INFO - Mycobacterium_nebraskense_GCF_002102255: 172 uces, 28 dupes, 144 non-dupes, 0 orient drop, 4 length drop, 140 written\n",
      "2020-02-06 14:26:17,221 - Phyluce - INFO - ---- Working on Mycobacterium_neumannii_GCF_002245615 genome ----\n",
      "2020-02-06 14:26:17,230 - Phyluce - INFO - Reading Mycobacterium_neumannii_GCF_002245615 genome\n",
      "2020-02-06 14:26:17,909 - Phyluce - INFO - Mycobacterium_neumannii_GCF_002245615: 171 uces, 26 dupes, 145 non-dupes, 0 orient drop, 9 length drop, 136 written\n",
      "2020-02-06 14:26:17,909 - Phyluce - INFO - --- Working on Mycobacterium_noviomagense_GCF_002086415 genome --\n",
      "2020-02-06 14:26:17,919 - Phyluce - INFO - Reading Mycobacterium_noviomagense_GCF_002086415 genome\n",
      "2020-02-06 14:26:18,573 - Phyluce - INFO - Mycobacterium_noviomagense_GCF_002086415: 172 uces, 24 dupes, 148 non-dupes, 2 orient drop, 5 length drop, 141 written\n",
      "2020-02-06 14:26:18,574 - Phyluce - INFO - ------ Working on Mycobacterium_orygis_GCF_006385035 genome -----\n",
      "2020-02-06 14:26:18,574 - Phyluce - INFO - Reading Mycobacterium_orygis_GCF_006385035 genome\n",
      "2020-02-06 14:26:19,201 - Phyluce - INFO - Mycobacterium_orygis_GCF_006385035: 172 uces, 21 dupes, 151 non-dupes, 0 orient drop, 5 length drop, 146 written\n",
      "2020-02-06 14:26:19,201 - Phyluce - INFO - ----- Working on Mycobacterium_palustre_GCF_002101785 genome ----\n",
      "2020-02-06 14:26:19,202 - Phyluce - INFO - Reading Mycobacterium_palustre_GCF_002101785 genome\n",
      "2020-02-06 14:26:19,910 - Phyluce - INFO - Mycobacterium_palustre_GCF_002101785: 171 uces, 29 dupes, 142 non-dupes, 4 orient drop, 6 length drop, 132 written\n",
      "2020-02-06 14:26:19,910 - Phyluce - INFO - ----- Working on Mycobacterium_paraense_GCF_002101815 genome ----\n",
      "2020-02-06 14:26:19,911 - Phyluce - INFO - Reading Mycobacterium_paraense_GCF_002101815 genome\n",
      "2020-02-06 14:26:20,601 - Phyluce - INFO - Mycobacterium_paraense_GCF_002101815: 172 uces, 34 dupes, 138 non-dupes, 0 orient drop, 4 length drop, 134 written\n",
      "2020-02-06 14:26:20,601 - Phyluce - INFO - --- Working on Mycobacterium_paraffinicum_GCF_001907675 genome --\n",
      "2020-02-06 14:26:20,628 - Phyluce - INFO - Reading Mycobacterium_paraffinicum_GCF_001907675 genome\n",
      "2020-02-06 14:26:21,302 - Phyluce - INFO - Mycobacterium_paraffinicum_GCF_001907675: 172 uces, 27 dupes, 145 non-dupes, 1 orient drop, 4 length drop, 140 written\n",
      "2020-02-06 14:26:21,303 - Phyluce - INFO - --- Working on Mycobacterium_paragordonae_GCF_003614435 genome --\n",
      "2020-02-06 14:26:21,321 - Phyluce - INFO - Reading Mycobacterium_paragordonae_GCF_003614435 genome\n",
      "2020-02-06 14:26:22,005 - Phyluce - INFO - Mycobacterium_paragordonae_GCF_003614435: 172 uces, 3 dupes, 169 non-dupes, 20 orient drop, 11 length drop, 138 written\n",
      "2020-02-06 14:26:22,005 - Phyluce - INFO -  Working on Mycobacterium_paraintracellulare_GCF_000276825 genome \n",
      "2020-02-06 14:26:22,006 - Phyluce - INFO - Reading Mycobacterium_paraintracellulare_GCF_000276825 genome\n",
      "2020-02-06 14:26:22,691 - Phyluce - INFO - Mycobacterium_paraintracellulare_GCF_000276825: 172 uces, 0 dupes, 172 non-dupes, 23 orient drop, 12 length drop, 137 written\n",
      "2020-02-06 14:26:22,691 - Phyluce - INFO - - Working on Mycobacterium_parascrofulaceum_GCF_000164135 genome \n",
      "2020-02-06 14:26:22,692 - Phyluce - INFO - Reading Mycobacterium_parascrofulaceum_GCF_000164135 genome\n",
      "2020-02-06 14:26:23,387 - Phyluce - INFO - Mycobacterium_parascrofulaceum_GCF_000164135: 172 uces, 16 dupes, 156 non-dupes, 7 orient drop, 8 length drop, 141 written\n",
      "2020-02-06 14:26:23,387 - Phyluce - INFO - -- Working on Mycobacterium_paraseoulense_GCF_002086475 genome --\n",
      "2020-02-06 14:26:23,397 - Phyluce - INFO - Reading Mycobacterium_paraseoulense_GCF_002086475 genome\n",
      "2020-02-06 14:26:24,075 - Phyluce - INFO - Mycobacterium_paraseoulense_GCF_002086475: 172 uces, 27 dupes, 145 non-dupes, 0 orient drop, 4 length drop, 141 written\n",
      "2020-02-06 14:26:24,075 - Phyluce - INFO - ----- Working on Mycobacterium_parmense_GCF_002102335 genome ----\n",
      "2020-02-06 14:26:24,075 - Phyluce - INFO - Reading Mycobacterium_parmense_GCF_002102335 genome\n",
      "2020-02-06 14:26:24,767 - Phyluce - INFO - Mycobacterium_parmense_GCF_002102335: 171 uces, 25 dupes, 146 non-dupes, 1 orient drop, 8 length drop, 137 written\n",
      "2020-02-06 14:26:24,768 - Phyluce - INFO - ----- Working on Mycobacterium_persicum_GCF_002086675 genome ----\n",
      "2020-02-06 14:26:24,768 - Phyluce - INFO - Reading Mycobacterium_persicum_GCF_002086675 genome\n",
      "2020-02-06 14:26:25,419 - Phyluce - INFO - Mycobacterium_persicum_GCF_002086675: 172 uces, 26 dupes, 146 non-dupes, 0 orient drop, 2 length drop, 144 written\n",
      "2020-02-06 14:26:25,419 - Phyluce - INFO - -- Working on Mycobacterium_pseudokansasii_GCF_900566075 genome -\n",
      "2020-02-06 14:26:25,420 - Phyluce - INFO - Reading Mycobacterium_pseudokansasii_GCF_900566075 genome\n",
      "2020-02-06 14:26:26,072 - Phyluce - INFO - Mycobacterium_pseudokansasii_GCF_900566075: 172 uces, 0 dupes, 172 non-dupes, 15 orient drop, 13 length drop, 144 written\n",
      "2020-02-06 14:26:26,072 - Phyluce - INFO - -- Working on Mycobacterium_pseudoshottsii_GCF_003584745 genome -\n",
      "2020-02-06 14:26:26,079 - Phyluce - INFO - Reading Mycobacterium_pseudoshottsii_GCF_003584745 genome\n",
      "2020-02-06 14:26:26,723 - Phyluce - INFO - Mycobacterium_pseudoshottsii_GCF_003584745: 172 uces, 0 dupes, 172 non-dupes, 8 orient drop, 19 length drop, 145 written\n",
      "2020-02-06 14:26:26,723 - Phyluce - INFO - ---- Working on Mycobacterium_riyadhense_GCF_002101845 genome ---\n",
      "2020-02-06 14:26:26,723 - Phyluce - INFO - Reading Mycobacterium_riyadhense_GCF_002101845 genome\n",
      "2020-02-06 14:26:27,378 - Phyluce - INFO - Mycobacterium_riyadhense_GCF_002101845: 172 uces, 27 dupes, 145 non-dupes, 0 orient drop, 4 length drop, 141 written\n",
      "2020-02-06 14:26:27,378 - Phyluce - INFO - - Working on Mycobacterium_saskatchewanense_GCF_002101875 genome \n",
      "2020-02-06 14:26:27,379 - Phyluce - INFO - Reading Mycobacterium_saskatchewanense_GCF_002101875 genome\n",
      "2020-02-06 14:26:28,052 - Phyluce - INFO - Mycobacterium_saskatchewanense_GCF_002101875: 172 uces, 31 dupes, 141 non-dupes, 0 orient drop, 4 length drop, 137 written\n",
      "2020-02-06 14:26:28,053 - Phyluce - INFO - --- Working on Mycobacterium_scrofulaceum_GCF_002086735 genome --\n",
      "2020-02-06 14:26:28,053 - Phyluce - INFO - Reading Mycobacterium_scrofulaceum_GCF_002086735 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 14:26:28,727 - Phyluce - INFO - Mycobacterium_scrofulaceum_GCF_002086735: 172 uces, 26 dupes, 146 non-dupes, 0 orient drop, 4 length drop, 142 written\n",
      "2020-02-06 14:26:28,727 - Phyluce - INFO - ---- Working on Mycobacterium_sherrisii_GCF_002102355 genome ----\n",
      "2020-02-06 14:26:28,736 - Phyluce - INFO - Reading Mycobacterium_sherrisii_GCF_002102355 genome\n",
      "2020-02-06 14:26:29,409 - Phyluce - INFO - Mycobacterium_sherrisii_GCF_002102355: 172 uces, 26 dupes, 146 non-dupes, 1 orient drop, 4 length drop, 141 written\n",
      "2020-02-06 14:26:29,409 - Phyluce - INFO - ---- Working on Mycobacterium_shigaense_GCF_002356315 genome ----\n",
      "2020-02-06 14:26:29,430 - Phyluce - INFO - Reading Mycobacterium_shigaense_GCF_002356315 genome\n",
      "2020-02-06 14:26:30,095 - Phyluce - INFO - Mycobacterium_shigaense_GCF_002356315: 172 uces, 0 dupes, 172 non-dupes, 16 orient drop, 16 length drop, 140 written\n",
      "2020-02-06 14:26:30,096 - Phyluce - INFO - ---- Working on Mycobacterium_shimoidei_GCF_002101905 genome ----\n",
      "2020-02-06 14:26:30,096 - Phyluce - INFO - Reading Mycobacterium_shimoidei_GCF_002101905 genome\n",
      "2020-02-06 14:26:30,746 - Phyluce - INFO - Mycobacterium_shimoidei_GCF_002101905: 172 uces, 26 dupes, 146 non-dupes, 0 orient drop, 4 length drop, 142 written\n",
      "2020-02-06 14:26:30,746 - Phyluce - INFO - --- Working on Mycobacterium_shinjukuense_GCF_002086755 genome --\n",
      "2020-02-06 14:26:30,747 - Phyluce - INFO - Reading Mycobacterium_shinjukuense_GCF_002086755 genome\n",
      "2020-02-06 14:26:31,397 - Phyluce - INFO - Mycobacterium_shinjukuense_GCF_002086755: 172 uces, 24 dupes, 148 non-dupes, 1 orient drop, 5 length drop, 142 written\n",
      "2020-02-06 14:26:31,397 - Phyluce - INFO - ------ Working on Mycobacterium_simiae_GCF_002093075 genome -----\n",
      "2020-02-06 14:26:31,398 - Phyluce - INFO - Reading Mycobacterium_simiae_GCF_002093075 genome\n",
      "2020-02-06 14:26:32,096 - Phyluce - INFO - Mycobacterium_simiae_GCF_002093075: 172 uces, 27 dupes, 145 non-dupes, 2 orient drop, 7 length drop, 136 written\n",
      "2020-02-06 14:26:32,096 - Phyluce - INFO - -------- Working on Mycobacterium_sp_GCF_000328565 genome -------\n",
      "2020-02-06 14:26:32,096 - Phyluce - INFO - Reading Mycobacterium_sp_GCF_000328565 genome\n",
      "2020-02-06 14:26:32,802 - Phyluce - INFO - Mycobacterium_sp_GCF_000328565: 171 uces, 2 dupes, 169 non-dupes, 26 orient drop, 9 length drop, 134 written\n",
      "2020-02-06 14:26:32,802 - Phyluce - INFO - -- Working on Mycobacterium_syngnathidarum_GCF_001942625 genome -\n",
      "2020-02-06 14:26:32,803 - Phyluce - INFO - Reading Mycobacterium_syngnathidarum_GCF_001942625 genome\n",
      "2020-02-06 14:26:33,447 - Phyluce - INFO - Mycobacterium_syngnathidarum_GCF_001942625: 171 uces, 20 dupes, 151 non-dupes, 1 orient drop, 5 length drop, 145 written\n",
      "2020-02-06 14:26:33,448 - Phyluce - INFO - ----- Working on Mycobacterium_szulgai_GCF_002116635 genome -----\n",
      "2020-02-06 14:26:33,448 - Phyluce - INFO - Reading Mycobacterium_szulgai_GCF_002116635 genome\n",
      "2020-02-06 14:26:34,102 - Phyluce - INFO - Mycobacterium_szulgai_GCF_002116635: 172 uces, 22 dupes, 150 non-dupes, 0 orient drop, 5 length drop, 145 written\n",
      "2020-02-06 14:26:34,102 - Phyluce - INFO - ---- Working on Mycobacterium_talmoniae_GCF_002967005 genome ----\n",
      "2020-02-06 14:26:34,103 - Phyluce - INFO - Reading Mycobacterium_talmoniae_GCF_002967005 genome\n",
      "2020-02-06 14:26:34,785 - Phyluce - INFO - Mycobacterium_talmoniae_GCF_002967005: 172 uces, 33 dupes, 139 non-dupes, 0 orient drop, 0 length drop, 139 written\n",
      "2020-02-06 14:26:34,785 - Phyluce - INFO - ---- Working on Mycobacterium_tilburgii_GCF_902168065 genome ----\n",
      "2020-02-06 14:26:34,786 - Phyluce - INFO - Reading Mycobacterium_tilburgii_GCF_902168065 genome\n",
      "2020-02-06 14:26:35,418 - Phyluce - INFO - Mycobacterium_tilburgii_GCF_902168065: 165 uces, 20 dupes, 145 non-dupes, 4 orient drop, 4 length drop, 137 written\n",
      "2020-02-06 14:26:35,418 - Phyluce - INFO - ---- Working on Mycobacterium_timonense_GCF_002086775 genome ----\n",
      "2020-02-06 14:26:35,424 - Phyluce - INFO - Reading Mycobacterium_timonense_GCF_002086775 genome\n",
      "2020-02-06 14:26:36,120 - Phyluce - INFO - Mycobacterium_timonense_GCF_002086775: 170 uces, 32 dupes, 138 non-dupes, 1 orient drop, 4 length drop, 133 written\n",
      "2020-02-06 14:26:36,120 - Phyluce - INFO - ----- Working on Mycobacterium_triplex_GCF_000689255 genome -----\n",
      "2020-02-06 14:26:36,121 - Phyluce - INFO - Reading Mycobacterium_triplex_GCF_000689255 genome\n",
      "2020-02-06 14:26:36,805 - Phyluce - INFO - Mycobacterium_triplex_GCF_000689255: 172 uces, 1 dupes, 171 non-dupes, 16 orient drop, 16 length drop, 139 written\n",
      "2020-02-06 14:26:36,805 - Phyluce - INFO - ------ Working on Mycobacterium_uberis_GCF_003408705 genome -----\n",
      "2020-02-06 14:26:36,820 - Phyluce - INFO - Reading Mycobacterium_uberis_GCF_003408705 genome\n",
      "2020-02-06 14:26:37,389 - Phyluce - INFO - Mycobacterium_uberis_GCF_003408705: 160 uces, 17 dupes, 143 non-dupes, 0 orient drop, 1 length drop, 142 written\n",
      "2020-02-06 14:26:37,389 - Phyluce - INFO - ----- Working on Mycobacterium_ulcerans_GCF_000013925 genome ----\n",
      "2020-02-06 14:26:37,390 - Phyluce - INFO - Reading Mycobacterium_ulcerans_GCF_000013925 genome\n",
      "2020-02-06 14:26:38,025 - Phyluce - INFO - Mycobacterium_ulcerans_GCF_000013925: 172 uces, 0 dupes, 172 non-dupes, 8 orient drop, 18 length drop, 146 written\n",
      "2020-02-06 14:26:38,025 - Phyluce - INFO - ----- Working on Mycobacterium_vulneris_GCF_002104765 genome ----\n",
      "2020-02-06 14:26:38,041 - Phyluce - INFO - Reading Mycobacterium_vulneris_GCF_002104765 genome\n",
      "2020-02-06 14:26:38,758 - Phyluce - INFO - Mycobacterium_vulneris_GCF_002104765: 172 uces, 31 dupes, 141 non-dupes, 1 orient drop, 4 length drop, 136 written\n",
      "2020-02-06 14:26:38,758 - Phyluce - INFO - --- Working on Mycobacterium_tuberculosis_GCF_000195955 genome --\n",
      "2020-02-06 14:26:38,767 - Phyluce - INFO - Reading Mycobacterium_tuberculosis_GCF_000195955 genome\n",
      "2020-02-06 14:26:39,409 - Phyluce - INFO - Mycobacterium_tuberculosis_GCF_000195955: 172 uces, 0 dupes, 172 non-dupes, 11 orient drop, 14 length drop, 147 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/mycobacterium/extract_probes_from_group/probe_fasta --output results/phyluce/mycobacterium/extract_probes_from_group/multifastas.sqlite --base-taxon Mycobacterium_tuberculosis_GCF_000195955;\n",
      "mycobacterium_ahvazicum_gcf_900176255.\n",
      "mycobacterium_alsense_gcf_002086635.\n",
      "mycobacterium_angelicum_gcf_002086155.\n",
      "mycobacterium_aquaticum_gcf_002086485.\n",
      "mycobacterium_arosiense_gcf_002086125.\n",
      "mycobacterium_asiaticum_gcf_000613245.\n",
      "mycobacterium_attenuatum_gcf_900566085.\n",
      "mycobacterium_avium_gcf_000007865.\n",
      "mycobacterium_basiliense_gcf_900292015.\n",
      "mycobacterium_bohemicum_gcf_001053185.\n",
      "mycobacterium_bouchedurhonense_gcf_002086165.\n",
      "mycobacterium_branderi_gcf_002086575.\n",
      "mycobacterium_canettii_gcf_000253375.\n",
      "mycobacterium_celatum_gcf_002101595.\n",
      "mycobacterium_chimaera_gcf_002219285.\n",
      "mycobacterium_colombiense_gcf_002105755.\n",
      "mycobacterium_conspcuum_gcf_002102095.\n",
      "mycobacterium_decipiens_gcf_002104675.\n",
      "mycobacterium_dioxanotrophicus_gcf_002157835.\n",
      "mycobacterium_eburneum_gcf_004354905.\n",
      "mycobacterium_europaeum_gcf_001373515.\n",
      "mycobacterium_florentinum_gcf_002101635.\n",
      "mycobacterium_fragae_gcf_002102185.\n",
      "mycobacterium_gastri_gcf_002102175.\n",
      "mycobacterium_genavense_gcf_000526915.\n",
      "mycobacterium_gordonae_gcf_002101675.\n",
      "mycobacterium_grossiae_gcf_008329645.\n",
      "mycobacterium_haemophilum_gcf_000340435.\n",
      "mycobacterium_heckeshornense_gcf_001881585.\n",
      "mycobacterium_heidelbergense_gcf_002086215.\n",
      "mycobacterium_innocens_gcf_900566055.\n",
      "mycobacterium_interjectum_gcf_900078675.\n",
      "mycobacterium_intermedium_gcf_002086275.\n",
      "mycobacterium_intracellulare_gcf_000277125.\n",
      "mycobacterium_kansasii_gcf_000157895.\n",
      "mycobacterium_kubicae_gcf_002101745.\n",
      "mycobacterium_kyorinense_gcf_000759695.\n",
      "mycobacterium_lacus_gcf_002102215.\n",
      "mycobacterium_lehmannii_gcf_002245535.\n",
      "mycobacterium_lentiflavum_gcf_001373395.\n",
      "mycobacterium_leprae_gcf_000195855.\n",
      "mycobacterium_lepraemurium_gcf_002291465.\n",
      "mycobacterium_lepromatosis_gcf_000966355.\n",
      "mycobacterium_liflandii_gcf_000026445.\n",
      "mycobacterium_malmoense_gcf_002086305.\n",
      "mycobacterium_mantenii_gcf_002086335.\n",
      "mycobacterium_marinum_gcf_000723425.\n",
      "mycobacterium_marseillense_gcf_002285715.\n",
      "mycobacterium_montefiorense_gcf_003112775.\n",
      "mycobacterium_mungi_gcf_001652545.\n",
      "mycobacterium_nebraskense_gcf_002102255.\n",
      "mycobacterium_neumannii_gcf_002245615.\n",
      "mycobacterium_noviomagense_gcf_002086415.\n",
      "mycobacterium_orygis_gcf_006385035.\n",
      "mycobacterium_palustre_gcf_002101785.\n",
      "mycobacterium_paraense_gcf_002101815.\n",
      "mycobacterium_paraffinicum_gcf_001907675.\n",
      "mycobacterium_paragordonae_gcf_003614435.\n",
      "mycobacterium_paraintracellulare_gcf_000276825.\n",
      "mycobacterium_parascrofulaceum_gcf_000164135.\n",
      "mycobacterium_paraseoulense_gcf_002086475.\n",
      "mycobacterium_parmense_gcf_002102335.\n",
      "mycobacterium_persicum_gcf_002086675.\n",
      "mycobacterium_pseudokansasii_gcf_900566075.\n",
      "mycobacterium_pseudoshottsii_gcf_003584745.\n",
      "mycobacterium_riyadhense_gcf_002101845.\n",
      "mycobacterium_saskatchewanense_gcf_002101875.\n",
      "mycobacterium_scrofulaceum_gcf_002086735.\n",
      "mycobacterium_sherrisii_gcf_002102355.\n",
      "mycobacterium_shigaense_gcf_002356315.\n",
      "mycobacterium_shimoidei_gcf_002101905.\n",
      "mycobacterium_shinjukuense_gcf_002086755.\n",
      "mycobacterium_simiae_gcf_002093075.\n",
      "mycobacterium_sp_gcf_000328565.\n",
      "mycobacterium_syngnathidarum_gcf_001942625.\n",
      "mycobacterium_szulgai_gcf_002116635.\n",
      "mycobacterium_talmoniae_gcf_002967005.\n",
      "mycobacterium_tilburgii_gcf_902168065.\n",
      "mycobacterium_timonense_gcf_002086775.\n",
      "mycobacterium_triplex_gcf_000689255.\n",
      "mycobacterium_uberis_gcf_003408705.\n",
      "mycobacterium_ulcerans_gcf_000013925.\n",
      "mycobacterium_vulneris_gcf_002104765.\n",
      "mycobacterium_tuberculosis_gcf_000195955.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/mycobacterium/extract_probes_from_group/multifastas.sqlite --base-taxon Mycobacterium_tuberculosis_GCF_000195955;\n",
      "Loci shared by 0 taxa:\t163.0\n",
      "Loci shared by 1 taxa:\t163.0\n",
      "Loci shared by 2 taxa:\t160.0\n",
      "Loci shared by 3 taxa:\t159.0\n",
      "Loci shared by 4 taxa:\t158.0\n",
      "Loci shared by 5 taxa:\t157.0\n",
      "Loci shared by 6 taxa:\t157.0\n",
      "Loci shared by 7 taxa:\t157.0\n",
      "Loci shared by 8 taxa:\t157.0\n",
      "Loci shared by 9 taxa:\t154.0\n",
      "Loci shared by 10 taxa:\t154.0\n",
      "Loci shared by 11 taxa:\t152.0\n",
      "Loci shared by 12 taxa:\t152.0\n",
      "Loci shared by 13 taxa:\t152.0\n",
      "Loci shared by 14 taxa:\t152.0\n",
      "Loci shared by 15 taxa:\t152.0\n",
      "Loci shared by 16 taxa:\t152.0\n",
      "Loci shared by 17 taxa:\t152.0\n",
      "Loci shared by 18 taxa:\t150.0\n",
      "Loci shared by 19 taxa:\t150.0\n",
      "Loci shared by 20 taxa:\t149.0\n",
      "Loci shared by 21 taxa:\t149.0\n",
      "Loci shared by 22 taxa:\t149.0\n",
      "Loci shared by 23 taxa:\t149.0\n",
      "Loci shared by 24 taxa:\t148.0\n",
      "Loci shared by 25 taxa:\t148.0\n",
      "Loci shared by 26 taxa:\t148.0\n",
      "Loci shared by 27 taxa:\t148.0\n",
      "Loci shared by 28 taxa:\t147.0\n",
      "Loci shared by 29 taxa:\t147.0\n",
      "Loci shared by 30 taxa:\t147.0\n",
      "Loci shared by 31 taxa:\t147.0\n",
      "Loci shared by 32 taxa:\t147.0\n",
      "Loci shared by 33 taxa:\t147.0\n",
      "Loci shared by 34 taxa:\t147.0\n",
      "Loci shared by 35 taxa:\t147.0\n",
      "Loci shared by 36 taxa:\t147.0\n",
      "Loci shared by 37 taxa:\t147.0\n",
      "Loci shared by 38 taxa:\t146.0\n",
      "Loci shared by 39 taxa:\t144.0\n",
      "Loci shared by 40 taxa:\t143.0\n",
      "Loci shared by 41 taxa:\t143.0\n",
      "Loci shared by 42 taxa:\t143.0\n",
      "Loci shared by 43 taxa:\t143.0\n",
      "Loci shared by 44 taxa:\t143.0\n",
      "Loci shared by 45 taxa:\t143.0\n",
      "Loci shared by 46 taxa:\t142.0\n",
      "Loci shared by 47 taxa:\t142.0\n",
      "Loci shared by 48 taxa:\t142.0\n",
      "Loci shared by 49 taxa:\t142.0\n",
      "Loci shared by 50 taxa:\t142.0\n",
      "Loci shared by 51 taxa:\t142.0\n",
      "Loci shared by 52 taxa:\t142.0\n",
      "Loci shared by 53 taxa:\t142.0\n",
      "Loci shared by 54 taxa:\t142.0\n",
      "Loci shared by 55 taxa:\t142.0\n",
      "Loci shared by 56 taxa:\t140.0\n",
      "Loci shared by 57 taxa:\t139.0\n",
      "Loci shared by 58 taxa:\t139.0\n",
      "Loci shared by 59 taxa:\t139.0\n",
      "Loci shared by 60 taxa:\t139.0\n",
      "Loci shared by 61 taxa:\t139.0\n",
      "Loci shared by 62 taxa:\t138.0\n",
      "Loci shared by 63 taxa:\t138.0\n",
      "Loci shared by 64 taxa:\t138.0\n",
      "Loci shared by 65 taxa:\t138.0\n",
      "Loci shared by 66 taxa:\t137.0\n",
      "Loci shared by 67 taxa:\t135.0\n",
      "Loci shared by 68 taxa:\t133.0\n",
      "Loci shared by 69 taxa:\t132.0\n",
      "Loci shared by 70 taxa:\t132.0\n",
      "Loci shared by 71 taxa:\t129.0\n",
      "Loci shared by 72 taxa:\t127.0\n",
      "Loci shared by 73 taxa:\t126.0\n",
      "Loci shared by 74 taxa:\t125.0\n",
      "Loci shared by 75 taxa:\t124.0\n",
      "Loci shared by 76 taxa:\t122.0\n",
      "Loci shared by 77 taxa:\t121.0\n",
      "Loci shared by 78 taxa:\t119.0\n",
      "Loci shared by 79 taxa:\t111.0\n",
      "Loci shared by 80 taxa:\t104.0\n",
      "Loci shared by 81 taxa:\t96.0\n",
      "Loci shared by 82 taxa:\t90.0\n",
      "Loci shared by 83 taxa:\t83.0\n",
      "Loci shared by 84 taxa:\t63.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/mycobacterium/extract_probes_from_group/multifastas.sqlite  --base-taxon Mycobacterium_tuberculosis_GCF_000195955 --output results/phyluce/mycobacterium/extract_probes_from_group/Mycobacterium_tuberculosis_GCF_000195955+70-back-to-80.conf --specific-counts 80;\n",
      "Counter({'mycobacterium_basiliense_gcf_900292015': 104, 'mycobacterium_attenuatum_gcf_900566085': 104, 'mycobacterium_shinjukuense_gcf_002086755': 104, 'mycobacterium_innocens_gcf_900566055': 104, 'mycobacterium_europaeum_gcf_001373515': 104, 'mycobacterium_kubicae_gcf_002101745': 104, 'mycobacterium_kansasii_gcf_000157895': 104, 'mycobacterium_orygis_gcf_006385035': 104, 'mycobacterium_decipiens_gcf_002104675': 104, 'mycobacterium_genavense_gcf_000526915': 104, 'mycobacterium_celatum_gcf_002101595': 104, 'mycobacterium_montefiorense_gcf_003112775': 104, 'mycobacterium_shimoidei_gcf_002101905': 104, 'mycobacterium_noviomagense_gcf_002086415': 104, 'mycobacterium_tuberculosis_gcf_000195955': 104, 'mycobacterium_lacus_gcf_002102215': 104, 'mycobacterium_fragae_gcf_002102185': 104, 'mycobacterium_parascrofulaceum_gcf_000164135': 104, 'mycobacterium_gastri_gcf_002102175': 104, 'mycobacterium_vulneris_gcf_002104765': 104, 'mycobacterium_scrofulaceum_gcf_002086735': 104, 'mycobacterium_simiae_gcf_002093075': 104, 'mycobacterium_shigaense_gcf_002356315': 104, 'mycobacterium_nebraskense_gcf_002102255': 104, 'mycobacterium_parmense_gcf_002102335': 104, 'mycobacterium_mungi_gcf_001652545': 104, 'mycobacterium_neumannii_gcf_002245615': 104, 'mycobacterium_conspcuum_gcf_002102095': 104, 'mycobacterium_canettii_gcf_000253375': 104, 'mycobacterium_lepraemurium_gcf_002291465': 104, 'mycobacterium_branderi_gcf_002086575': 104, 'mycobacterium_pseudokansasii_gcf_900566075': 104, 'mycobacterium_alsense_gcf_002086635': 104, 'mycobacterium_sherrisii_gcf_002102355': 104, 'mycobacterium_persicum_gcf_002086675': 104, 'mycobacterium_szulgai_gcf_002116635': 104, 'mycobacterium_eburneum_gcf_004354905': 103, 'mycobacterium_gordonae_gcf_002101675': 103, 'mycobacterium_asiaticum_gcf_000613245': 103, 'mycobacterium_talmoniae_gcf_002967005': 103, 'mycobacterium_paraffinicum_gcf_001907675': 103, 'mycobacterium_lentiflavum_gcf_001373395': 103, 'mycobacterium_palustre_gcf_002101785': 103, 'mycobacterium_mantenii_gcf_002086335': 103, 'mycobacterium_colombiense_gcf_002105755': 103, 'mycobacterium_avium_gcf_000007865': 103, 'mycobacterium_ulcerans_gcf_000013925': 103, 'mycobacterium_chimaera_gcf_002219285': 103, 'mycobacterium_florentinum_gcf_002101635': 103, 'mycobacterium_intracellulare_gcf_000277125': 103, 'mycobacterium_saskatchewanense_gcf_002101875': 103, 'mycobacterium_marinum_gcf_000723425': 103, 'mycobacterium_angelicum_gcf_002086155': 103, 'mycobacterium_heidelbergense_gcf_002086215': 103, 'mycobacterium_sp_gcf_000328565': 103, 'mycobacterium_triplex_gcf_000689255': 103, 'mycobacterium_riyadhense_gcf_002101845': 103, 'mycobacterium_malmoense_gcf_002086305': 103, 'mycobacterium_bohemicum_gcf_001053185': 103, 'mycobacterium_paraintracellulare_gcf_000276825': 103, 'mycobacterium_heckeshornense_gcf_001881585': 103, 'mycobacterium_liflandii_gcf_000026445': 103, 'mycobacterium_kyorinense_gcf_000759695': 103, 'mycobacterium_pseudoshottsii_gcf_003584745': 103, 'mycobacterium_paragordonae_gcf_003614435': 103, 'mycobacterium_lehmannii_gcf_002245535': 103, 'mycobacterium_syngnathidarum_gcf_001942625': 103, 'mycobacterium_marseillense_gcf_002285715': 103, 'mycobacterium_haemophilum_gcf_000340435': 102, 'mycobacterium_paraseoulense_gcf_002086475': 102, 'mycobacterium_paraense_gcf_002101815': 102, 'mycobacterium_uberis_gcf_003408705': 102, 'mycobacterium_bouchedurhonense_gcf_002086165': 102, 'mycobacterium_arosiense_gcf_002086125': 102, 'mycobacterium_interjectum_gcf_900078675': 101, 'mycobacterium_timonense_gcf_002086775': 101, 'mycobacterium_tilburgii_gcf_902168065': 101, 'mycobacterium_grossiae_gcf_008329645': 101, 'mycobacterium_intermedium_gcf_002086275': 101, 'mycobacterium_leprae_gcf_000195855': 100, 'mycobacterium_aquaticum_gcf_002086485': 100, 'mycobacterium_dioxanotrophicus_gcf_002157835': 99, 'mycobacterium_lepromatosis_gcf_000966355': 98, 'mycobacterium_ahvazicum_gcf_900176255': 98})\n",
      "Total loci = 104\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(80)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/mycobacterium/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/mycobacterium/extract_probes_from_group/Mycobacterium_tuberculosis_GCF_000195955+70-back-to-80.conf --probe-prefix uce_mycobacterium_ --designer rnplattii --design mycobacterium_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 104\n",
      "Probe Count = 15607\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list.fasta --query results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  14:29:39\n",
      "Ended:  Thu Feb 06, 2020  14:31:44\n",
      "Time for execution:  2.08234694799 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list.fasta --lastz results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_mycobacterium_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 15606 fasta sequences.  Filtered 2 duplicates. Kept 15291.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/mycobacterium/final_probe_design/mycobacterium_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 15:25:41 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 15607\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1248560\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 90M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3938410\n",
      "Max number of word counting entries: 88653624\n",
      "\n",
      "# comparing sequences from          0  to       2601\n",
      "..---------- new table with     1427 representatives\n",
      "# comparing sequences from       2601  to       4768\n",
      "----------   1382 remaining sequences to the next cycle\n",
      "---------- new table with      369 representatives\n",
      "# comparing sequences from       3386  to       5422\n",
      "----------   1360 remaining sequences to the next cycle\n",
      "---------- new table with      295 representatives\n",
      "# comparing sequences from       4062  to       5986\n",
      "----------   1110 remaining sequences to the next cycle\n",
      "---------- new table with      255 representatives\n",
      "# comparing sequences from       4876  to       6664\n",
      "----------   1158 remaining sequences to the next cycle\n",
      "---------- new table with      259 representatives\n",
      "# comparing sequences from       5506  to       7189\n",
      "----------   1080 remaining sequences to the next cycle\n",
      "---------- new table with      173 representatives\n",
      "# comparing sequences from       6109  to       7692\n",
      "----------    823 remaining sequences to the next cycle\n",
      "---------- new table with      261 representatives\n",
      "# comparing sequences from       6869  to       8325\n",
      "----------    900 remaining sequences to the next cycle\n",
      "---------- new table with      281 representatives\n",
      "# comparing sequences from       7425  to       8788\n",
      "----------    780 remaining sequences to the next cycle\n",
      "---------- new table with      269 representatives\n",
      "# comparing sequences from       8008  to       9274\n",
      "----------    648 remaining sequences to the next cycle\n",
      "---------- new table with      227 representatives\n",
      "# comparing sequences from       8626  to       9789\n",
      "----------    724 remaining sequences to the next cycle\n",
      "---------- new table with      251 representatives\n",
      "# comparing sequences from       9065  to      10155\n",
      "----------    726 remaining sequences to the next cycle\n",
      "---------- new table with      129 representatives\n",
      "# comparing sequences from       9429  to      10458\n",
      "----------    676 remaining sequences to the next cycle\n",
      "---------- new table with      123 representatives\n",
      "# comparing sequences from       9782  to      10752\n",
      "..........    10000  finished       4427  clusters\n",
      "----------    557 remaining sequences to the next cycle\n",
      "---------- new table with      216 representatives\n",
      "# comparing sequences from      10195  to      11097\n",
      "----------    582 remaining sequences to the next cycle\n",
      "---------- new table with      194 representatives\n",
      "# comparing sequences from      10515  to      11363\n",
      "----------    615 remaining sequences to the next cycle\n",
      "---------- new table with      137 representatives\n",
      "# comparing sequences from      10748  to      11557\n",
      "----------    405 remaining sequences to the next cycle\n",
      "---------- new table with      191 representatives\n",
      "# comparing sequences from      11152  to      11894\n",
      "----------    425 remaining sequences to the next cycle\n",
      "---------- new table with      159 representatives\n",
      "# comparing sequences from      11469  to      12158\n",
      "----------    432 remaining sequences to the next cycle\n",
      "---------- new table with      151 representatives\n",
      "# comparing sequences from      11726  to      12372\n",
      "----------    365 remaining sequences to the next cycle\n",
      "---------- new table with      131 representatives\n",
      "# comparing sequences from      12007  to      12607\n",
      "----------    419 remaining sequences to the next cycle\n",
      "---------- new table with      127 representatives\n",
      "# comparing sequences from      12188  to      12757\n",
      "----------    262 remaining sequences to the next cycle\n",
      "---------- new table with      115 representatives\n",
      "# comparing sequences from      12495  to      13013\n",
      "----------    271 remaining sequences to the next cycle\n",
      "---------- new table with      135 representatives\n",
      "# comparing sequences from      12742  to      13219\n",
      "----------    224 remaining sequences to the next cycle\n",
      "---------- new table with      113 representatives\n",
      "# comparing sequences from      12995  to      13430\n",
      "----------    209 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      13221  to      13618\n",
      "----------     90 remaining sequences to the next cycle\n",
      "---------- new table with      191 representatives\n",
      "# comparing sequences from      13528  to      13874\n",
      "----------    164 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      13710  to      14026\n",
      "----------    131 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      13895  to      14180\n",
      "----------     78 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      14102  to      14352\n",
      "----------     48 remaining sequences to the next cycle\n",
      "---------- new table with      111 representatives\n",
      "# comparing sequences from      14304  to      14521\n",
      "....................---------- new table with       75 representatives\n",
      "# comparing sequences from      14521  to      14702\n",
      "..................---------- new table with       93 representatives\n",
      "# comparing sequences from      14702  to      15607\n",
      ".....................---------- new table with      453 representatives\n",
      "\n",
      "    15607  finished       7311  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 91M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 3.48\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pasteurella"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'pasteurella'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Pasteurella_bettyae_GCF_000262245\",\n",
    "               \"Pasteurella_caecimuris_GCF_004793515\",\n",
    "               \"Pasteurella_canis_GCF_900454865\",\n",
    "               \"Pasteurella_dagmatis_GCF_900186835\",\n",
    "               \"Pasteurella_langaaensis_GCF_003096995\",\n",
    "               \"Pasteurella_oralis_GCF_002850605\",\n",
    "               \"Pasteurella_skyensis_GCF_900109845\",\n",
    "               \"Pasteurella_sp_GCF_004570945\",\n",
    "               \"Pasteurella_testudinis_GCF_900454705\" ]\n",
    "                    \n",
    "reference_taxon =  \"Pasteurella_multocida_GCF_000754275\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000262245.1_ASM26224v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 666562 bytes  444402.67 bytes/sec\n",
      "total size is 666289  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004793515.1_ASM479351v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 767347 bytes  511592.67 bytes/sec\n",
      "total size is 767049  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900454865.1_50305_A01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 692055 bytes  461398.00 bytes/sec\n",
      "total size is 691775  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900186835.1_49950_F01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 673548 bytes  449060.00 bytes/sec\n",
      "total size is 673276  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003096995.1_ASM309699v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 589347 bytes  392926.00 bytes/sec\n",
      "total size is 589097  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000754275.1_ASM75427v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 675091 bytes  450088.67 bytes/sec\n",
      "total size is 674818  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002850605.1_ASM285060v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 706065 bytes  1412214.00 bytes/sec\n",
      "total size is 705783  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900109845.1_IMG-taxon_2675903214_annotated_assembly_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 646314 bytes  1292712.00 bytes/sec\n",
      "total size is 646020  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004570945.1_ASM457094v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 672160 bytes  448134.67 bytes/sec\n",
      "total size is 671886  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900454705.1_50465_G01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 886839 bytes  591254.00 bytes/sec\n",
      "total size is 886511  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480620 (\"sim_Pasteurella_bettyae_GCF_000262245\") has been submitted\n",
      "Your job 5480621 (\"sim_Pasteurella_caecimuris_GCF_004793515\") has been submitted\n",
      "Your job 5480622 (\"sim_Pasteurella_canis_GCF_900454865\") has been submitted\n",
      "Your job 5480623 (\"sim_Pasteurella_dagmatis_GCF_900186835\") has been submitted\n",
      "Your job 5480624 (\"sim_Pasteurella_langaaensis_GCF_003096995\") has been submitted\n",
      "Your job 5480625 (\"sim_Pasteurella_oralis_GCF_002850605\") has been submitted\n",
      "Your job 5480626 (\"sim_Pasteurella_skyensis_GCF_900109845\") has been submitted\n",
      "Your job 5480627 (\"sim_Pasteurella_sp_GCF_004570945\") has been submitted\n",
      "Your job 5480628 (\"sim_Pasteurella_testudinis_GCF_900454705\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/pasteurella/cleaned_genomes/Pasteurella_multocida_GCF_000754275_formatted.fas path=results/phyluce/pasteurella/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480629 (\"map_Pasteurella_bettyae_GCF_000262245\") has been submitted\n",
      "Your job 5480630 (\"map_Pasteurella_caecimuris_GCF_004793515\") has been submitted\n",
      "Your job 5480631 (\"map_Pasteurella_canis_GCF_900454865\") has been submitted\n",
      "Your job 5480632 (\"map_Pasteurella_dagmatis_GCF_900186835\") has been submitted\n",
      "Your job 5480633 (\"map_Pasteurella_langaaensis_GCF_003096995\") has been submitted\n",
      "Your job 5480634 (\"map_Pasteurella_oralis_GCF_002850605\") has been submitted\n",
      "Your job 5480635 (\"map_Pasteurella_skyensis_GCF_900109845\") has been submitted\n",
      "Your job 5480636 (\"map_Pasteurella_sp_GCF_004570945\") has been submitted\n",
      "Your job 5480637 (\"map_Pasteurella_testudinis_GCF_900454705\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480638 (\"merge_Pasteurella_bettyae_GCF_000262245\") has been submitted\n",
      "Your job 5480639 (\"merge_Pasteurella_caecimuris_GCF_004793515\") has been submitted\n",
      "Your job 5480640 (\"merge_Pasteurella_canis_GCF_900454865\") has been submitted\n",
      "Your job 5480641 (\"merge_Pasteurella_dagmatis_GCF_900186835\") has been submitted\n",
      "Your job 5480642 (\"merge_Pasteurella_langaaensis_GCF_003096995\") has been submitted\n",
      "Your job 5480643 (\"merge_Pasteurella_oralis_GCF_002850605\") has been submitted\n",
      "Your job 5480644 (\"merge_Pasteurella_skyensis_GCF_900109845\") has been submitted\n",
      "Your job 5480645 (\"merge_Pasteurella_sp_GCF_004570945\") has been submitted\n",
      "Your job 5480646 (\"merge_Pasteurella_testudinis_GCF_900454705\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 263 sequences from Pasteurella_bettyae_GCF_000262245_merged.bed.  Filtered 189 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 74.\n",
      "Screened 263 sequences from Pasteurella_caecimuris_GCF_004793515_merged.bed.  Filtered 186 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 77.\n",
      "Screened 1116 sequences from Pasteurella_canis_GCF_900454865_merged.bed.  Filtered 702 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 414.\n",
      "Screened 1066 sequences from Pasteurella_dagmatis_GCF_900186835_merged.bed.  Filtered 682 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 384.\n",
      "Screened 268 sequences from Pasteurella_langaaensis_GCF_003096995_merged.bed.  Filtered 191 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 77.\n",
      "Screened 1203 sequences from Pasteurella_oralis_GCF_002850605_merged.bed.  Filtered 764 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 439.\n",
      "Screened 183 sequences from Pasteurella_skyensis_GCF_900109845_merged.bed.  Filtered 124 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 59.\n",
      "Screened 262 sequences from Pasteurella_sp_GCF_004570945_merged.bed.  Filtered 188 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 74.\n",
      "Screened 150 sequences from Pasteurella_testudinis_GCF_900454705_merged.bed.  Filtered 93 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 57.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/pasteurella/initial_intervals/Pasteurella_testudinis_GCF_900454705_merged.bed --twobit results/phyluce/pasteurella/cleaned_genomes/Pasteurella_multocida_GCF_000754275_formatted.2bit --output results/phyluce/pasteurella/initial_intervals/Pasteurella_testudinis_GCF_900454705_stripped.bed;\n",
      "pasteurella_bettyae_gcf_000262245.\n",
      "pasteurella_caecimuris_gcf_004793515.\n",
      "pasteurella_canis_gcf_900454865.\n",
      "pasteurella_dagmatis_gcf_900186835.\n",
      "pasteurella_langaaensis_gcf_003096995.\n",
      "pasteurella_oralis_gcf_002850605.\n",
      "pasteurella_skyensis_gcf_900109845.\n",
      "pasteurella_sp_gcf_004570945.\n",
      "pasteurella_testudinis_gcf_900454705.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/pasteurella/initial_intervals/Pasteurella_testudinis_GCF_900454705_merged.bed --twobit results/phyluce/pasteurella/cleaned_genomes/Pasteurella_multocida_GCF_000754275_formatted.2bit --output results/phyluce/pasteurella/initial_intervals/Pasteurella_testudinis_GCF_900454705_stripped.bed;\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 0 taxa:\t751.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 1 taxa:\t751.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 2 taxa:\t404.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 3 taxa:\t228.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 4 taxa:\t95.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 5 taxa:\t70.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 6 taxa:\t59.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 7 taxa:\t49.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 8 taxa:\t36.0\n",
      "Loci shared by Pasteurella_multocida_GCF_000754275 + 9 taxa:\t20.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/pasteurella/initial_intervals/pasteurella-to-Pasteurella_multocida_GCF_000754275.sqlite --base-taxon Pasteurella_multocida_GCF_000754275 --output results/phyluce/pasteurella/initial_intervals/Pasteurella_multocida_GCF_000754275_+3.bed --specific-counts 3;\n",
      "Counter({'pasteurella_dagmatis_gcf_900186835': 221, 'pasteurella_canis_gcf_900454865': 219, 'pasteurella_oralis_gcf_002850605': 217, 'pasteurella_langaaensis_gcf_003096995': 70, 'pasteurella_caecimuris_gcf_004793515': 68, 'pasteurella_bettyae_gcf_000262245': 67, 'pasteurella_sp_gcf_004570945': 63, 'pasteurella_testudinis_gcf_900454705': 45, 'pasteurella_skyensis_gcf_900109845': 43})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 3\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/pasteurella/initial_intervals/Pasteurella_multocida_GCF_000754275_+3.bed --twobit results/phyluce/pasteurella/cleaned_genomes/Pasteurella_multocida_GCF_000754275_formatted.2bit --buffer-to 160 --output results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3.fasta;\n",
      "Screened 228 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 228.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3.fasta --probe-prefix uce_pasteurella_ --design pasteurella_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 227\n",
      "Probe Count = 446\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas --query results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  11:50:51\n",
      "Ended:  Thu Feb 06, 2020  11:50:51\n",
      "Time for execution:  0.00381955305735 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas                        --lastz results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes_vself.lastz                       --probe-prefix=uce_pasteurella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 445 fasta sequences.  Filtered 10 duplicates. Kept 426.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/pasteurella/validate_intervals/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas --scaffoldlist Pasteurella_bettyae_GCF_000262245 Pasteurella_caecimuris_GCF_004793515 Pasteurella_canis_GCF_900454865 Pasteurella_dagmatis_GCF_900186835 Pasteurella_langaaensis_GCF_003096995 Pasteurella_oralis_GCF_002850605 Pasteurella_skyensis_GCF_900109845 Pasteurella_sp_GCF_004570945 Pasteurella_testudinis_GCF_900454705 Pasteurella_multocida_GCF_000754275 --genome-base-path results/phyluce/pasteurella/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/pasteurella/validate_intervals/pasteurella-to-Pasteurella_multocida_GCF_000754275.sqlite --output results/phyluce/pasteurella/validate_intervals/lastz/;\n",
      "\n",
      "Running against Pasteurella_bettyae_GCF_000262245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpv4tqUR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWjcLEm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_bettyae_GCF_000262245.lastz\n",
      "Creating Pasteurella_bettyae_GCF_000262245 table\n",
      "Inserting data to Pasteurella_bettyae_GCF_000262245 table\n",
      "\n",
      "Running against Pasteurella_caecimuris_GCF_004793515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDdF4fh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5pyyPc.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_caecimuris_GCF_004793515.lastz\n",
      "Creating Pasteurella_caecimuris_GCF_004793515 table\n",
      "Inserting data to Pasteurella_caecimuris_GCF_004793515 table\n",
      "\n",
      "Running against Pasteurella_canis_GCF_900454865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzoyBDm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjvR0R2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_canis_GCF_900454865.lastz\n",
      "Creating Pasteurella_canis_GCF_900454865 table\n",
      "Inserting data to Pasteurella_canis_GCF_900454865 table\n",
      "\n",
      "Running against Pasteurella_dagmatis_GCF_900186835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKWfQYg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJwhunA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_dagmatis_GCF_900186835.lastz\n",
      "Creating Pasteurella_dagmatis_GCF_900186835 table\n",
      "Inserting data to Pasteurella_dagmatis_GCF_900186835 table\n",
      "\n",
      "Running against Pasteurella_langaaensis_GCF_003096995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpXMTBOX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpseFP4l.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_langaaensis_GCF_003096995.lastz\n",
      "Creating Pasteurella_langaaensis_GCF_003096995 table\n",
      "Inserting data to Pasteurella_langaaensis_GCF_003096995 table\n",
      "\n",
      "Running against Pasteurella_oralis_GCF_002850605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpoU2h9L.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp27flhp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_oralis_GCF_002850605.lastz\n",
      "Creating Pasteurella_oralis_GCF_002850605 table\n",
      "Inserting data to Pasteurella_oralis_GCF_002850605 table\n",
      "\n",
      "Running against Pasteurella_skyensis_GCF_900109845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsbn6N6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJ4WCvI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_skyensis_GCF_900109845.lastz\n",
      "Creating Pasteurella_skyensis_GCF_900109845 table\n",
      "Inserting data to Pasteurella_skyensis_GCF_900109845 table\n",
      "\n",
      "Running against Pasteurella_sp_GCF_004570945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZGdVFO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaCLwvb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_sp_GCF_004570945.lastz\n",
      "Creating Pasteurella_sp_GCF_004570945 table\n",
      "Inserting data to Pasteurella_sp_GCF_004570945 table\n",
      "\n",
      "Running against Pasteurella_testudinis_GCF_900454705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpoS13Zo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdNgPnP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_testudinis_GCF_900454705.lastz\n",
      "Creating Pasteurella_testudinis_GCF_900454705 table\n",
      "Inserting data to Pasteurella_testudinis_GCF_900454705 table\n",
      "\n",
      "Running against Pasteurella_multocida_GCF_000754275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzpnVrL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpj7UMuU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/pasteurella/validate_intervals/lastz/Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_Pasteurella_multocida_GCF_000754275.lastz\n",
      "Creating Pasteurella_multocida_GCF_000754275 table\n",
      "Inserting data to Pasteurella_multocida_GCF_000754275 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/pasteurella/extract_probes_from_group/pasteurella_genome.conf --lastz results/phyluce/pasteurella/validate_intervals/lastz --probes 120 --probe-prefix uce_pasteurella_ --name-pattern \"Pasteurella_multocida_GCF_000754275_+3_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/pasteurella/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 11:53:01,077 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 11:53:01,078 - Phyluce - INFO - ------ Working on Pasteurella_bettyae_GCF_000262245 genome ------\n",
      "2020-02-06 11:53:01,092 - Phyluce - INFO - Reading Pasteurella_bettyae_GCF_000262245 genome\n",
      "2020-02-06 11:53:01,856 - Phyluce - INFO - Pasteurella_bettyae_GCF_000262245: 213 uces, 10 dupes, 203 non-dupes, 0 orient drop, 0 length drop, 203 written\n",
      "2020-02-06 11:53:01,856 - Phyluce - INFO - ----- Working on Pasteurella_caecimuris_GCF_004793515 genome ----\n",
      "2020-02-06 11:53:01,857 - Phyluce - INFO - Reading Pasteurella_caecimuris_GCF_004793515 genome\n",
      "2020-02-06 11:53:02,541 - Phyluce - INFO - Pasteurella_caecimuris_GCF_004793515: 213 uces, 6 dupes, 207 non-dupes, 2 orient drop, 0 length drop, 205 written\n",
      "2020-02-06 11:53:02,541 - Phyluce - INFO - ------- Working on Pasteurella_canis_GCF_900454865 genome -------\n",
      "2020-02-06 11:53:02,542 - Phyluce - INFO - Reading Pasteurella_canis_GCF_900454865 genome\n",
      "2020-02-06 11:53:03,587 - Phyluce - INFO - Pasteurella_canis_GCF_900454865: 227 uces, 19 dupes, 208 non-dupes, 5 orient drop, 9 length drop, 194 written\n",
      "2020-02-06 11:53:03,587 - Phyluce - INFO - ------ Working on Pasteurella_dagmatis_GCF_900186835 genome -----\n",
      "2020-02-06 11:53:03,588 - Phyluce - INFO - Reading Pasteurella_dagmatis_GCF_900186835 genome\n",
      "2020-02-06 11:53:04,530 - Phyluce - INFO - Pasteurella_dagmatis_GCF_900186835: 226 uces, 0 dupes, 226 non-dupes, 20 orient drop, 4 length drop, 202 written\n",
      "2020-02-06 11:53:04,531 - Phyluce - INFO - ---- Working on Pasteurella_langaaensis_GCF_003096995 genome ----\n",
      "2020-02-06 11:53:04,531 - Phyluce - INFO - Reading Pasteurella_langaaensis_GCF_003096995 genome\n",
      "2020-02-06 11:53:05,209 - Phyluce - INFO - Pasteurella_langaaensis_GCF_003096995: 211 uces, 6 dupes, 205 non-dupes, 0 orient drop, 0 length drop, 205 written\n",
      "2020-02-06 11:53:05,209 - Phyluce - INFO - ------- Working on Pasteurella_oralis_GCF_002850605 genome ------\n",
      "2020-02-06 11:53:05,229 - Phyluce - INFO - Reading Pasteurella_oralis_GCF_002850605 genome\n",
      "2020-02-06 11:53:05,987 - Phyluce - INFO - Pasteurella_oralis_GCF_002850605: 226 uces, 7 dupes, 219 non-dupes, 0 orient drop, 0 length drop, 219 written\n",
      "2020-02-06 11:53:05,987 - Phyluce - INFO - ------ Working on Pasteurella_skyensis_GCF_900109845 genome -----\n",
      "2020-02-06 11:53:05,992 - Phyluce - INFO - Reading Pasteurella_skyensis_GCF_900109845 genome\n",
      "2020-02-06 11:53:06,618 - Phyluce - INFO - Pasteurella_skyensis_GCF_900109845: 190 uces, 8 dupes, 182 non-dupes, 0 orient drop, 0 length drop, 182 written\n",
      "2020-02-06 11:53:06,618 - Phyluce - INFO - --------- Working on Pasteurella_sp_GCF_004570945 genome --------\n",
      "2020-02-06 11:53:06,629 - Phyluce - INFO - Reading Pasteurella_sp_GCF_004570945 genome\n",
      "2020-02-06 11:53:07,333 - Phyluce - INFO - Pasteurella_sp_GCF_004570945: 212 uces, 4 dupes, 208 non-dupes, 0 orient drop, 4 length drop, 204 written\n",
      "2020-02-06 11:53:07,334 - Phyluce - INFO - ----- Working on Pasteurella_testudinis_GCF_900454705 genome ----\n",
      "2020-02-06 11:53:07,334 - Phyluce - INFO - Reading Pasteurella_testudinis_GCF_900454705 genome\n",
      "2020-02-06 11:53:08,217 - Phyluce - INFO - Pasteurella_testudinis_GCF_900454705: 208 uces, 1 dupes, 207 non-dupes, 17 orient drop, 6 length drop, 184 written\n",
      "2020-02-06 11:53:08,218 - Phyluce - INFO - ----- Working on Pasteurella_multocida_GCF_000754275 genome -----\n",
      "2020-02-06 11:53:08,228 - Phyluce - INFO - Reading Pasteurella_multocida_GCF_000754275 genome\n",
      "2020-02-06 11:53:09,200 - Phyluce - INFO - Pasteurella_multocida_GCF_000754275: 227 uces, 0 dupes, 227 non-dupes, 15 orient drop, 7 length drop, 205 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/pasteurella/extract_probes_from_group/probe_fasta --output results/phyluce/pasteurella/extract_probes_from_group/multifastas.sqlite --base-taxon Pasteurella_multocida_GCF_000754275;\n",
      "pasteurella_bettyae_gcf_000262245.\n",
      "pasteurella_caecimuris_gcf_004793515.\n",
      "pasteurella_canis_gcf_900454865.\n",
      "pasteurella_dagmatis_gcf_900186835.\n",
      "pasteurella_langaaensis_gcf_003096995.\n",
      "pasteurella_oralis_gcf_002850605.\n",
      "pasteurella_skyensis_gcf_900109845.\n",
      "pasteurella_sp_gcf_004570945.\n",
      "pasteurella_testudinis_gcf_900454705.\n",
      "pasteurella_multocida_gcf_000754275.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/pasteurella/extract_probes_from_group/multifastas.sqlite --base-taxon Pasteurella_multocida_GCF_000754275;\n",
      "Loci shared by 0 taxa:\t226.0\n",
      "Loci shared by 1 taxa:\t226.0\n",
      "Loci shared by 2 taxa:\t224.0\n",
      "Loci shared by 3 taxa:\t223.0\n",
      "Loci shared by 4 taxa:\t221.0\n",
      "Loci shared by 5 taxa:\t212.0\n",
      "Loci shared by 6 taxa:\t206.0\n",
      "Loci shared by 7 taxa:\t191.0\n",
      "Loci shared by 8 taxa:\t182.0\n",
      "Loci shared by 9 taxa:\t169.0\n",
      "Loci shared by 10 taxa:\t149.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/pasteurella/extract_probes_from_group/multifastas.sqlite  --base-taxon Pasteurella_multocida_GCF_000754275 --output results/phyluce/pasteurella/extract_probes_from_group/Pasteurella_multocida_GCF_000754275+3-back-to-10.conf --specific-counts 10;\n",
      "Counter({'pasteurella_bettyae_gcf_000262245': 149, 'pasteurella_langaaensis_gcf_003096995': 149, 'pasteurella_dagmatis_gcf_900186835': 149, 'pasteurella_oralis_gcf_002850605': 149, 'pasteurella_canis_gcf_900454865': 149, 'pasteurella_caecimuris_gcf_004793515': 149, 'pasteurella_multocida_gcf_000754275': 149, 'pasteurella_sp_gcf_004570945': 149, 'pasteurella_skyensis_gcf_900109845': 149, 'pasteurella_testudinis_gcf_900454705': 149})\n",
      "Total loci = 149\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(10)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/pasteurella/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/pasteurella/extract_probes_from_group/Pasteurella_multocida_GCF_000754275+3-back-to-10.conf --probe-prefix uce_pasteurella_ --designer rnplattii --design pasteurella_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 149\n",
      "Probe Count = 2932\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list.fasta --query results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  11:53:18\n",
      "Ended:  Thu Feb 06, 2020  11:53:21\n",
      "Time for execution:  0.0533509174983 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list.fasta --lastz results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_pasteurella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 2931 fasta sequences.  Filtered 2 duplicates. Kept 2892.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/pasteurella/final_probe_design/pasteurella_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 11:54:31 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 2932\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 234560\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3959997\n",
      "Max number of word counting entries: 89139541\n",
      "\n",
      "# comparing sequences from          0  to        488\n",
      "---------- new table with      378 representatives\n",
      "# comparing sequences from        488  to        895\n",
      "----------      1 remaining sequences to the next cycle\n",
      "---------- new table with      322 representatives\n",
      "# comparing sequences from        894  to       1233\n",
      "----------     49 remaining sequences to the next cycle\n",
      "---------- new table with      236 representatives\n",
      "# comparing sequences from       1184  to       1475\n",
      "----------     39 remaining sequences to the next cycle\n",
      "---------- new table with      218 representatives\n",
      "# comparing sequences from       1436  to       1685\n",
      "----------    123 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       1562  to       1790\n",
      "----------     46 remaining sequences to the next cycle\n",
      "---------- new table with      153 representatives\n",
      "# comparing sequences from       1744  to       1942\n",
      "99.8%---------- new table with      175 representatives\n",
      "# comparing sequences from       1942  to       2932\n",
      ".....................---------- new table with      880 representatives\n",
      "\n",
      "     2932  finished       2462  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.73\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Rickettsia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'rickettsia'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Rickettsia_aeschlimannii_GCF_001051325\",\n",
    "               \"Rickettsia_africae_GCF_000023005\",\n",
    "               \"Rickettsia_akari_GCF_000018205\",\n",
    "               \"Rickettsia_amblyommatis_GCF_001273795\",\n",
    "               \"Rickettsia_argasii_GCF_000965185\",\n",
    "               \"Rickettsia_asembonensis_GCF_000828125\",\n",
    "               \"Rickettsia_asiatica_GCF_007989425\",\n",
    "               \"Rickettsia_australis_GCF_000284155\",\n",
    "               \"Rickettsia_bellii_GCF_000012385\",\n",
    "               \"Rickettsia_buchneri_GCF_000696365\",\n",
    "               \"Rickettsia_canadensis_GCF_000014345\",\n",
    "               \"Rickettsia_conorii_GCF_000007025\",\n",
    "               \"Rickettsia_endosymbiont_GCF_002285905\",\n",
    "               \"Rickettsia_felis_GCF_000012145\",\n",
    "               \"Rickettsia_fournieri_GCF_900243065\",\n",
    "               \"Rickettsia_gravesii_GCF_000485845\",\n",
    "               \"Rickettsia_heilongjiangensis_GCF_000221205\",\n",
    "               \"Rickettsia_helvetica_GCF_000255355\",\n",
    "               \"Rickettsia_honei_GCF_000263055\",\n",
    "               \"Rickettsia_hoogstraalii_GCF_000825685\",\n",
    "               \"Rickettsia_japonica_GCF_000283595\",\n",
    "               \"Rickettsia_massiliae_GCF_000016625\",\n",
    "               \"Rickettsia_monacensis_GCF_000499665\",\n",
    "               \"Rickettsia_montanensis_GCF_000284175\",\n",
    "               \"Rickettsia_parkeri_GCF_005549115\",\n",
    "               \"Rickettsia_peacockii_GCF_000021525\",\n",
    "               \"Rickettsia_philipii_GCF_000283995\",\n",
    "               \"Rickettsia_prowazekii_GCF_000195735\",\n",
    "               \"Rickettsia_raoultii_GCF_001975185\",\n",
    "               \"Rickettsia_rhipicephali_GCF_000284075\",\n",
    "               \"Rickettsia_sibirica_GCF_000166935\",\n",
    "               \"Rickettsia_slovaca_GCF_000237845\",\n",
    "               \"Rickettsia_sp_GCF_001653015\",\n",
    "               \"Rickettsia_tamurae_GCF_000751075\",\n",
    "               \"Rickettsia_typhi_GCF_000008045\"  ]\n",
    "                    \n",
    "reference_taxon = \"Rickettsia_rickettsii_GCF_001951015\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001051325.1_Rickettsia_aeschlimannii_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 383879 bytes  255947.33 bytes/sec\n",
      "total size is 383664  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000023005.1_ASM2300v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 375986 bytes  250685.33 bytes/sec\n",
      "total size is 375786  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000018205.1_ASM1820v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 359240 bytes  239521.33 bytes/sec\n",
      "total size is 359048  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001273795.1_ASM127379v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 424119 bytes  282774.00 bytes/sec\n",
      "total size is 423909  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000965185.1_ASM96518v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 420396 bytes  280292.00 bytes/sec\n",
      "total size is 420187  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000828125.2_ASM82812v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 401066 bytes  802216.00 bytes/sec\n",
      "total size is 400857  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007989425.1_ASM798942v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 412561 bytes  275068.67 bytes/sec\n",
      "total size is 412351  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000284155.1_ASM28415v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 384377 bytes  256279.33 bytes/sec\n",
      "total size is 384176  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000012385.1_ASM1238v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 439599 bytes  293094.00 bytes/sec\n",
      "total size is 439383  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000696365.1_REISMNv1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 483808 bytes  322566.67 bytes/sec\n",
      "total size is 483585  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000014345.1_ASM1434v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 337308 bytes  224900.00 bytes/sec\n",
      "total size is 337116  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007025.1_ASM702v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 369605 bytes  246431.33 bytes/sec\n",
      "total size is 369406  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002285905.1_ASM228590v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 374529 bytes  249714.00 bytes/sec\n",
      "total size is 374327  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000012145.1_ASM1214v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 459959 bytes  306667.33 bytes/sec\n",
      "total size is 459735  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900243065.1_PRJEB23962_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 418182 bytes  278816.00 bytes/sec\n",
      "total size is 417973  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000485845.1_RicGra1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 395381 bytes  263615.33 bytes/sec\n",
      "total size is 395173  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000221205.1_ASM22120v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 372368 bytes  248273.33 bytes/sec\n",
      "total size is 372167  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000255355.1_ASM25535v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 412294 bytes  274890.67 bytes/sec\n",
      "total size is 412085  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000263055.1_Rho1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 371213 bytes  247503.33 bytes/sec\n",
      "total size is 371016  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000825685.1_Rickettsia_hoogstraalii_Croatica_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 429784 bytes  286550.67 bytes/sec\n",
      "total size is 429545  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000283595.1_ASM28359v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 373656 bytes  747396.00 bytes/sec\n",
      "total size is 373455  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000016625.1_ASM1662v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 400457 bytes  800998.00 bytes/sec\n",
      "total size is 400249  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000499665.2_RMONA_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 377742 bytes  755568.00 bytes/sec\n",
      "total size is 377544  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000284175.1_ASM28417v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 372568 bytes  745220.00 bytes/sec\n",
      "total size is 372367  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005549115.1_ASM554911v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 390849 bytes  781782.00 bytes/sec\n",
      "total size is 390647  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000021525.1_ASM2152v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 380913 bytes  761910.00 bytes/sec\n",
      "total size is 380713  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000283995.1_ASM28399v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 375066 bytes  750216.00 bytes/sec\n",
      "total size is 374865  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000195735.1_ASM19573v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 321266 bytes  642616.00 bytes/sec\n",
      "total size is 321081  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001975185.1_ASM197518v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 391464 bytes  783012.00 bytes/sec\n",
      "total size is 391262  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000284075.1_ASM28407v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 380492 bytes  761068.00 bytes/sec\n",
      "total size is 380291  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001951015.1_ASM195101v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 369793 bytes  739670.00 bytes/sec\n",
      "total size is 369591  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000166935.1_ASM16693v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 364719 bytes  243174.00 bytes/sec\n",
      "total size is 364518  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000237845.1_ASM23784v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 371207 bytes  247499.33 bytes/sec\n",
      "total size is 371006  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001653015.1_ASM165301v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 368825 bytes  245911.33 bytes/sec\n",
      "total size is 368623  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000751075.1_Rickettsia_tamurae_AT-1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 424447 bytes  282992.67 bytes/sec\n",
      "total size is 424225  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000008045.1_ASM804v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 321115 bytes  214104.67 bytes/sec\n",
      "total size is 320932  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480647 (\"sim_Rickettsia_aeschlimannii_GCF_001051325\") has been submitted\n",
      "Your job 5480648 (\"sim_Rickettsia_africae_GCF_000023005\") has been submitted\n",
      "Your job 5480649 (\"sim_Rickettsia_akari_GCF_000018205\") has been submitted\n",
      "Your job 5480650 (\"sim_Rickettsia_amblyommatis_GCF_001273795\") has been submitted\n",
      "Your job 5480651 (\"sim_Rickettsia_argasii_GCF_000965185\") has been submitted\n",
      "Your job 5480652 (\"sim_Rickettsia_asembonensis_GCF_000828125\") has been submitted\n",
      "Your job 5480653 (\"sim_Rickettsia_asiatica_GCF_007989425\") has been submitted\n",
      "Your job 5480654 (\"sim_Rickettsia_australis_GCF_000284155\") has been submitted\n",
      "Your job 5480655 (\"sim_Rickettsia_bellii_GCF_000012385\") has been submitted\n",
      "Your job 5480656 (\"sim_Rickettsia_buchneri_GCF_000696365\") has been submitted\n",
      "Your job 5480657 (\"sim_Rickettsia_canadensis_GCF_000014345\") has been submitted\n",
      "Your job 5480658 (\"sim_Rickettsia_conorii_GCF_000007025\") has been submitted\n",
      "Your job 5480659 (\"sim_Rickettsia_endosymbiont_GCF_002285905\") has been submitted\n",
      "Your job 5480660 (\"sim_Rickettsia_felis_GCF_000012145\") has been submitted\n",
      "Your job 5480661 (\"sim_Rickettsia_fournieri_GCF_900243065\") has been submitted\n",
      "Your job 5480662 (\"sim_Rickettsia_gravesii_GCF_000485845\") has been submitted\n",
      "Your job 5480663 (\"sim_Rickettsia_heilongjiangensis_GCF_000221205\") has been submitted\n",
      "Your job 5480664 (\"sim_Rickettsia_helvetica_GCF_000255355\") has been submitted\n",
      "Your job 5480665 (\"sim_Rickettsia_honei_GCF_000263055\") has been submitted\n",
      "Your job 5480666 (\"sim_Rickettsia_hoogstraalii_GCF_000825685\") has been submitted\n",
      "Your job 5480667 (\"sim_Rickettsia_japonica_GCF_000283595\") has been submitted\n",
      "Your job 5480668 (\"sim_Rickettsia_massiliae_GCF_000016625\") has been submitted\n",
      "Your job 5480669 (\"sim_Rickettsia_monacensis_GCF_000499665\") has been submitted\n",
      "Your job 5480670 (\"sim_Rickettsia_montanensis_GCF_000284175\") has been submitted\n",
      "Your job 5480671 (\"sim_Rickettsia_parkeri_GCF_005549115\") has been submitted\n",
      "Your job 5480672 (\"sim_Rickettsia_peacockii_GCF_000021525\") has been submitted\n",
      "Your job 5480673 (\"sim_Rickettsia_philipii_GCF_000283995\") has been submitted\n",
      "Your job 5480674 (\"sim_Rickettsia_prowazekii_GCF_000195735\") has been submitted\n",
      "Your job 5480675 (\"sim_Rickettsia_raoultii_GCF_001975185\") has been submitted\n",
      "Your job 5480676 (\"sim_Rickettsia_rhipicephali_GCF_000284075\") has been submitted\n",
      "Your job 5480677 (\"sim_Rickettsia_sibirica_GCF_000166935\") has been submitted\n",
      "Your job 5480678 (\"sim_Rickettsia_slovaca_GCF_000237845\") has been submitted\n",
      "Your job 5480679 (\"sim_Rickettsia_sp_GCF_001653015\") has been submitted\n",
      "Your job 5480680 (\"sim_Rickettsia_tamurae_GCF_000751075\") has been submitted\n",
      "Your job 5480681 (\"sim_Rickettsia_typhi_GCF_000008045\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/rickettsia/cleaned_genomes/Rickettsia_rickettsii_GCF_001951015_formatted.fas path=results/phyluce/rickettsia/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480682 (\"map_Rickettsia_aeschlimannii_GCF_001051325\") has been submitted\n",
      "Your job 5480683 (\"map_Rickettsia_africae_GCF_000023005\") has been submitted\n",
      "Your job 5480684 (\"map_Rickettsia_akari_GCF_000018205\") has been submitted\n",
      "Your job 5480685 (\"map_Rickettsia_amblyommatis_GCF_001273795\") has been submitted\n",
      "Your job 5480686 (\"map_Rickettsia_argasii_GCF_000965185\") has been submitted\n",
      "Your job 5480687 (\"map_Rickettsia_asembonensis_GCF_000828125\") has been submitted\n",
      "Your job 5480688 (\"map_Rickettsia_asiatica_GCF_007989425\") has been submitted\n",
      "Your job 5480689 (\"map_Rickettsia_australis_GCF_000284155\") has been submitted\n",
      "Your job 5480690 (\"map_Rickettsia_bellii_GCF_000012385\") has been submitted\n",
      "Your job 5480691 (\"map_Rickettsia_buchneri_GCF_000696365\") has been submitted\n",
      "Your job 5480692 (\"map_Rickettsia_canadensis_GCF_000014345\") has been submitted\n",
      "Your job 5480693 (\"map_Rickettsia_conorii_GCF_000007025\") has been submitted\n",
      "Your job 5480694 (\"map_Rickettsia_endosymbiont_GCF_002285905\") has been submitted\n",
      "Your job 5480695 (\"map_Rickettsia_felis_GCF_000012145\") has been submitted\n",
      "Your job 5480696 (\"map_Rickettsia_fournieri_GCF_900243065\") has been submitted\n",
      "Your job 5480697 (\"map_Rickettsia_gravesii_GCF_000485845\") has been submitted\n",
      "Your job 5480698 (\"map_Rickettsia_heilongjiangensis_GCF_000221205\") has been submitted\n",
      "Your job 5480699 (\"map_Rickettsia_helvetica_GCF_000255355\") has been submitted\n",
      "Your job 5480700 (\"map_Rickettsia_honei_GCF_000263055\") has been submitted\n",
      "Your job 5480701 (\"map_Rickettsia_hoogstraalii_GCF_000825685\") has been submitted\n",
      "Your job 5480702 (\"map_Rickettsia_japonica_GCF_000283595\") has been submitted\n",
      "Your job 5480703 (\"map_Rickettsia_massiliae_GCF_000016625\") has been submitted\n",
      "Your job 5480704 (\"map_Rickettsia_monacensis_GCF_000499665\") has been submitted\n",
      "Your job 5480705 (\"map_Rickettsia_montanensis_GCF_000284175\") has been submitted\n",
      "Your job 5480706 (\"map_Rickettsia_parkeri_GCF_005549115\") has been submitted\n",
      "Your job 5480707 (\"map_Rickettsia_peacockii_GCF_000021525\") has been submitted\n",
      "Your job 5480708 (\"map_Rickettsia_philipii_GCF_000283995\") has been submitted\n",
      "Your job 5480709 (\"map_Rickettsia_prowazekii_GCF_000195735\") has been submitted\n",
      "Your job 5480710 (\"map_Rickettsia_raoultii_GCF_001975185\") has been submitted\n",
      "Your job 5480711 (\"map_Rickettsia_rhipicephali_GCF_000284075\") has been submitted\n",
      "Your job 5480712 (\"map_Rickettsia_sibirica_GCF_000166935\") has been submitted\n",
      "Your job 5480713 (\"map_Rickettsia_slovaca_GCF_000237845\") has been submitted\n",
      "Your job 5480714 (\"map_Rickettsia_sp_GCF_001653015\") has been submitted\n",
      "Your job 5480715 (\"map_Rickettsia_tamurae_GCF_000751075\") has been submitted\n",
      "Your job 5480716 (\"map_Rickettsia_typhi_GCF_000008045\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480717 (\"merge_Rickettsia_aeschlimannii_GCF_001051325\") has been submitted\n",
      "Your job 5480718 (\"merge_Rickettsia_africae_GCF_000023005\") has been submitted\n",
      "Your job 5480719 (\"merge_Rickettsia_akari_GCF_000018205\") has been submitted\n",
      "Your job 5480720 (\"merge_Rickettsia_amblyommatis_GCF_001273795\") has been submitted\n",
      "Your job 5480721 (\"merge_Rickettsia_argasii_GCF_000965185\") has been submitted\n",
      "Your job 5480722 (\"merge_Rickettsia_asembonensis_GCF_000828125\") has been submitted\n",
      "Your job 5480723 (\"merge_Rickettsia_asiatica_GCF_007989425\") has been submitted\n",
      "Your job 5480724 (\"merge_Rickettsia_australis_GCF_000284155\") has been submitted\n",
      "Your job 5480725 (\"merge_Rickettsia_bellii_GCF_000012385\") has been submitted\n",
      "Your job 5480726 (\"merge_Rickettsia_buchneri_GCF_000696365\") has been submitted\n",
      "Your job 5480727 (\"merge_Rickettsia_canadensis_GCF_000014345\") has been submitted\n",
      "Your job 5480728 (\"merge_Rickettsia_conorii_GCF_000007025\") has been submitted\n",
      "Your job 5480729 (\"merge_Rickettsia_endosymbiont_GCF_002285905\") has been submitted\n",
      "Your job 5480730 (\"merge_Rickettsia_felis_GCF_000012145\") has been submitted\n",
      "Your job 5480731 (\"merge_Rickettsia_fournieri_GCF_900243065\") has been submitted\n",
      "Your job 5480732 (\"merge_Rickettsia_gravesii_GCF_000485845\") has been submitted\n",
      "Your job 5480733 (\"merge_Rickettsia_heilongjiangensis_GCF_000221205\") has been submitted\n",
      "Your job 5480734 (\"merge_Rickettsia_helvetica_GCF_000255355\") has been submitted\n",
      "Your job 5480735 (\"merge_Rickettsia_honei_GCF_000263055\") has been submitted\n",
      "Your job 5480736 (\"merge_Rickettsia_hoogstraalii_GCF_000825685\") has been submitted\n",
      "Your job 5480737 (\"merge_Rickettsia_japonica_GCF_000283595\") has been submitted\n",
      "Your job 5480738 (\"merge_Rickettsia_massiliae_GCF_000016625\") has been submitted\n",
      "Your job 5480739 (\"merge_Rickettsia_monacensis_GCF_000499665\") has been submitted\n",
      "Your job 5480740 (\"merge_Rickettsia_montanensis_GCF_000284175\") has been submitted\n",
      "Your job 5480741 (\"merge_Rickettsia_parkeri_GCF_005549115\") has been submitted\n",
      "Your job 5480742 (\"merge_Rickettsia_peacockii_GCF_000021525\") has been submitted\n",
      "Your job 5480743 (\"merge_Rickettsia_philipii_GCF_000283995\") has been submitted\n",
      "Your job 5480744 (\"merge_Rickettsia_prowazekii_GCF_000195735\") has been submitted\n",
      "Your job 5480745 (\"merge_Rickettsia_raoultii_GCF_001975185\") has been submitted\n",
      "Your job 5480746 (\"merge_Rickettsia_rhipicephali_GCF_000284075\") has been submitted\n",
      "Your job 5480747 (\"merge_Rickettsia_sibirica_GCF_000166935\") has been submitted\n",
      "Your job 5480748 (\"merge_Rickettsia_slovaca_GCF_000237845\") has been submitted\n",
      "Your job 5480749 (\"merge_Rickettsia_sp_GCF_001653015\") has been submitted\n",
      "Your job 5480750 (\"merge_Rickettsia_tamurae_GCF_000751075\") has been submitted\n",
      "Your job 5480751 (\"merge_Rickettsia_typhi_GCF_000008045\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 578 sequences from Rickettsia_aeschlimannii_GCF_001051325_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 535.\n",
      "Screened 209 sequences from Rickettsia_africae_GCF_000023005_merged.bed.  Filtered 11 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 198.\n",
      "Screened 1575 sequences from Rickettsia_akari_GCF_000018205_merged.bed.  Filtered 362 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1213.\n",
      "Screened 542 sequences from Rickettsia_amblyommatis_GCF_001273795_merged.bed.  Filtered 64 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 478.\n",
      "Screened 389 sequences from Rickettsia_argasii_GCF_000965185_merged.bed.  Filtered 34 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 355.\n",
      "Screened 1387 sequences from Rickettsia_asembonensis_GCF_000828125_merged.bed.  Filtered 253 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1134.\n",
      "Screened 1392 sequences from Rickettsia_asiatica_GCF_007989425_merged.bed.  Filtered 281 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1111.\n",
      "Screened 1572 sequences from Rickettsia_australis_GCF_000284155_merged.bed.  Filtered 336 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1236.\n",
      "Screened 886 sequences from Rickettsia_bellii_GCF_000012385_merged.bed.  Filtered 636 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 250.\n",
      "Screened 1110 sequences from Rickettsia_buchneri_GCF_000696365_merged.bed.  Filtered 191 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 919.\n",
      "Screened 1818 sequences from Rickettsia_canadensis_GCF_000014345_merged.bed.  Filtered 555 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1263.\n",
      "Screened 231 sequences from Rickettsia_conorii_GCF_000007025_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 218.\n",
      "Screened 871 sequences from Rickettsia_endosymbiont_GCF_002285905_merged.bed.  Filtered 624 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 247.\n",
      "Screened 1554 sequences from Rickettsia_felis_GCF_000012145_merged.bed.  Filtered 324 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1230.\n",
      "Screened 316 sequences from Rickettsia_fournieri_GCF_900243065_merged.bed.  Filtered 23 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 293.\n",
      "Screened 449 sequences from Rickettsia_gravesii_GCF_000485845_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 417.\n",
      "Screened 365 sequences from Rickettsia_heilongjiangensis_GCF_000221205_merged.bed.  Filtered 19 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 346.\n",
      "Screened 1395 sequences from Rickettsia_helvetica_GCF_000255355_merged.bed.  Filtered 300 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1095.\n",
      "Screened 247 sequences from Rickettsia_honei_GCF_000263055_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 235.\n",
      "Screened 1444 sequences from Rickettsia_hoogstraalii_GCF_000825685_merged.bed.  Filtered 281 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1163.\n",
      "Screened 355 sequences from Rickettsia_japonica_GCF_000283595_merged.bed.  Filtered 20 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 335.\n",
      "Screened 549 sequences from Rickettsia_massiliae_GCF_000016625_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 503.\n",
      "Screened 1125 sequences from Rickettsia_monacensis_GCF_000499665_merged.bed.  Filtered 178 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 947.\n",
      "Screened 511 sequences from Rickettsia_montanensis_GCF_000284175_merged.bed.  Filtered 46 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 465.\n",
      "Screened 234 sequences from Rickettsia_parkeri_GCF_005549115_merged.bed.  Filtered 11 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 223.\n",
      "Screened 168 sequences from Rickettsia_peacockii_GCF_000021525_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 158.\n",
      "Screened 74 sequences from Rickettsia_philipii_GCF_000283995_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 72.\n",
      "Screened 1911 sequences from Rickettsia_prowazekii_GCF_000195735_merged.bed.  Filtered 719 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1192.\n",
      "Screened 394 sequences from Rickettsia_raoultii_GCF_001975185_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 369.\n",
      "Screened 506 sequences from Rickettsia_rhipicephali_GCF_000284075_merged.bed.  Filtered 38 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 468.\n",
      "Screened 235 sequences from Rickettsia_sibirica_GCF_000166935_merged.bed.  Filtered 10 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 225.\n",
      "Screened 174 sequences from Rickettsia_slovaca_GCF_000237845_merged.bed.  Filtered 11 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 163.\n",
      "Screened 311 sequences from Rickettsia_sp_GCF_001653015_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 296.\n",
      "Screened 1221 sequences from Rickettsia_tamurae_GCF_000751075_merged.bed.  Filtered 222 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 999.\n",
      "Screened 1923 sequences from Rickettsia_typhi_GCF_000008045_merged.bed.  Filtered 725 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1198.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/rickettsia/initial_intervals/Rickettsia_typhi_GCF_000008045_merged.bed --twobit results/phyluce/rickettsia/cleaned_genomes/Rickettsia_rickettsii_GCF_001951015_formatted.2bit --output results/phyluce/rickettsia/initial_intervals/Rickettsia_typhi_GCF_000008045_stripped.bed;\n",
      "rickettsia_aeschlimannii_gcf_001051325.\n",
      "rickettsia_africae_gcf_000023005.\n",
      "rickettsia_akari_gcf_000018205..\n",
      "rickettsia_amblyommatis_gcf_001273795.\n",
      "rickettsia_argasii_gcf_000965185.\n",
      "rickettsia_asembonensis_gcf_000828125..\n",
      "rickettsia_asiatica_gcf_007989425..\n",
      "rickettsia_australis_gcf_000284155..\n",
      "rickettsia_bellii_gcf_000012385.\n",
      "rickettsia_buchneri_gcf_000696365.\n",
      "rickettsia_canadensis_gcf_000014345..\n",
      "rickettsia_conorii_gcf_000007025.\n",
      "rickettsia_endosymbiont_gcf_002285905.\n",
      "rickettsia_felis_gcf_000012145..\n",
      "rickettsia_fournieri_gcf_900243065.\n",
      "rickettsia_gravesii_gcf_000485845.\n",
      "rickettsia_heilongjiangensis_gcf_000221205.\n",
      "rickettsia_helvetica_gcf_000255355..\n",
      "rickettsia_honei_gcf_000263055.\n",
      "rickettsia_hoogstraalii_gcf_000825685..\n",
      "rickettsia_japonica_gcf_000283595.\n",
      "rickettsia_massiliae_gcf_000016625.\n",
      "rickettsia_monacensis_gcf_000499665.\n",
      "rickettsia_montanensis_gcf_000284175.\n",
      "rickettsia_parkeri_gcf_005549115.\n",
      "rickettsia_peacockii_gcf_000021525.\n",
      "rickettsia_philipii_gcf_000283995.\n",
      "rickettsia_prowazekii_gcf_000195735..\n",
      "rickettsia_raoultii_gcf_001975185.\n",
      "rickettsia_rhipicephali_gcf_000284075.\n",
      "rickettsia_sibirica_gcf_000166935.\n",
      "rickettsia_slovaca_gcf_000237845.\n",
      "rickettsia_sp_gcf_001653015.\n",
      "rickettsia_tamurae_gcf_000751075.\n",
      "rickettsia_typhi_gcf_000008045..\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/rickettsia/initial_intervals/Rickettsia_typhi_GCF_000008045_merged.bed --twobit results/phyluce/rickettsia/cleaned_genomes/Rickettsia_rickettsii_GCF_001951015_formatted.2bit --output results/phyluce/rickettsia/initial_intervals/Rickettsia_typhi_GCF_000008045_stripped.bed;\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 0 taxa:\t648.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 1 taxa:\t648.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 2 taxa:\t645.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 3 taxa:\t644.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 4 taxa:\t643.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 5 taxa:\t643.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 6 taxa:\t642.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 7 taxa:\t641.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 8 taxa:\t640.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 9 taxa:\t638.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 10 taxa:\t632.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 11 taxa:\t627.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 12 taxa:\t623.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 13 taxa:\t618.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 14 taxa:\t615.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 15 taxa:\t611.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 16 taxa:\t604.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 17 taxa:\t594.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 18 taxa:\t581.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 19 taxa:\t569.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 20 taxa:\t546.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 21 taxa:\t525.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 22 taxa:\t513.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 23 taxa:\t489.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 24 taxa:\t456.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 25 taxa:\t440.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 26 taxa:\t424.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 27 taxa:\t407.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 28 taxa:\t380.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 29 taxa:\t350.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 30 taxa:\t319.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 31 taxa:\t298.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 32 taxa:\t269.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 33 taxa:\t254.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 34 taxa:\t148.0\n",
      "Loci shared by Rickettsia_rickettsii_GCF_001951015 + 35 taxa:\t102.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/rickettsia/initial_intervals/rickettsia-to-Rickettsia_rickettsii_GCF_001951015.sqlite --base-taxon Rickettsia_rickettsii_GCF_001951015 --output results/phyluce/rickettsia/initial_intervals/Rickettsia_rickettsii_GCF_001951015_+33.bed --specific-counts 33;\n",
      "Counter({'rickettsia_japonica_gcf_000283595': 254, 'rickettsia_gravesii_gcf_000485845': 254, 'rickettsia_argasii_gcf_000965185': 254, 'rickettsia_sibirica_gcf_000166935': 254, 'rickettsia_tamurae_gcf_000751075': 254, 'rickettsia_conorii_gcf_000007025': 254, 'rickettsia_honei_gcf_000263055': 254, 'rickettsia_heilongjiangensis_gcf_000221205': 254, 'rickettsia_monacensis_gcf_000499665': 254, 'rickettsia_amblyommatis_gcf_001273795': 254, 'rickettsia_fournieri_gcf_900243065': 254, 'rickettsia_philipii_gcf_000283995': 254, 'rickettsia_asembonensis_gcf_000828125': 254, 'rickettsia_sp_gcf_001653015': 254, 'rickettsia_buchneri_gcf_000696365': 254, 'rickettsia_rhipicephali_gcf_000284075': 254, 'rickettsia_aeschlimannii_gcf_001051325': 254, 'rickettsia_asiatica_gcf_007989425': 254, 'rickettsia_akari_gcf_000018205': 254, 'rickettsia_montanensis_gcf_000284175': 254, 'rickettsia_slovaca_gcf_000237845': 254, 'rickettsia_australis_gcf_000284155': 254, 'rickettsia_helvetica_gcf_000255355': 254, 'rickettsia_raoultii_gcf_001975185': 254, 'rickettsia_parkeri_gcf_005549115': 254, 'rickettsia_hoogstraalii_gcf_000825685': 254, 'rickettsia_massiliae_gcf_000016625': 254, 'rickettsia_felis_gcf_000012145': 254, 'rickettsia_peacockii_gcf_000021525': 254, 'rickettsia_canadensis_gcf_000014345': 254, 'rickettsia_africae_gcf_000023005': 254, 'rickettsia_prowazekii_gcf_000195735': 252, 'rickettsia_typhi_gcf_000008045': 252, 'rickettsia_bellii_gcf_000012385': 132, 'rickettsia_endosymbiont_gcf_002285905': 122})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 33\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/rickettsia/initial_intervals/Rickettsia_rickettsii_GCF_001951015_+33.bed --twobit results/phyluce/rickettsia/cleaned_genomes/Rickettsia_rickettsii_GCF_001951015_formatted.2bit --buffer-to 160 --output results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33.fasta;\n",
      "Screened 254 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 254.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33.fasta --probe-prefix uce_rickettsia_ --design rickettsia_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 197\n",
      "Probe Count = 354\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas --query results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  12:00:43\n",
      "Ended:  Thu Feb 06, 2020  12:00:43\n",
      "Time for execution:  0.00349261760712 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas                        --lastz results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes_vself.lastz                       --probe-prefix=uce_rickettsia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 353 fasta sequences.  Filtered 2 duplicates. Kept 351.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/rickettsia/validate_intervals/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas --scaffoldlist Rickettsia_aeschlimannii_GCF_001051325 Rickettsia_africae_GCF_000023005 Rickettsia_akari_GCF_000018205 Rickettsia_amblyommatis_GCF_001273795 Rickettsia_argasii_GCF_000965185 Rickettsia_asembonensis_GCF_000828125 Rickettsia_asiatica_GCF_007989425 Rickettsia_australis_GCF_000284155 Rickettsia_bellii_GCF_000012385 Rickettsia_buchneri_GCF_000696365 Rickettsia_canadensis_GCF_000014345 Rickettsia_conorii_GCF_000007025 Rickettsia_endosymbiont_GCF_002285905 Rickettsia_felis_GCF_000012145 Rickettsia_fournieri_GCF_900243065 Rickettsia_gravesii_GCF_000485845 Rickettsia_heilongjiangensis_GCF_000221205 Rickettsia_helvetica_GCF_000255355 Rickettsia_honei_GCF_000263055 Rickettsia_hoogstraalii_GCF_000825685 Rickettsia_japonica_GCF_000283595 Rickettsia_massiliae_GCF_000016625 Rickettsia_monacensis_GCF_000499665 Rickettsia_montanensis_GCF_000284175 Rickettsia_parkeri_GCF_005549115 Rickettsia_peacockii_GCF_000021525 Rickettsia_philipii_GCF_000283995 Rickettsia_prowazekii_GCF_000195735 Rickettsia_raoultii_GCF_001975185 Rickettsia_rhipicephali_GCF_000284075 Rickettsia_sibirica_GCF_000166935 Rickettsia_slovaca_GCF_000237845 Rickettsia_sp_GCF_001653015 Rickettsia_tamurae_GCF_000751075 Rickettsia_typhi_GCF_000008045 Rickettsia_rickettsii_GCF_001951015 --genome-base-path results/phyluce/rickettsia/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/rickettsia/validate_intervals/rickettsia-to-Rickettsia_rickettsii_GCF_001951015.sqlite --output results/phyluce/rickettsia/validate_intervals/lastz/;\n",
      "\n",
      "Running against Rickettsia_aeschlimannii_GCF_001051325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5wXgRP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPlKQwg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_aeschlimannii_GCF_001051325.lastz\n",
      "Creating Rickettsia_aeschlimannii_GCF_001051325 table\n",
      "Inserting data to Rickettsia_aeschlimannii_GCF_001051325 table\n",
      "\n",
      "Running against Rickettsia_africae_GCF_000023005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjULQqS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBXvDjk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_africae_GCF_000023005.lastz\n",
      "Creating Rickettsia_africae_GCF_000023005 table\n",
      "Inserting data to Rickettsia_africae_GCF_000023005 table\n",
      "\n",
      "Running against Rickettsia_akari_GCF_000018205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHFzhBt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxTbDXD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_akari_GCF_000018205.lastz\n",
      "Creating Rickettsia_akari_GCF_000018205 table\n",
      "Inserting data to Rickettsia_akari_GCF_000018205 table\n",
      "\n",
      "Running against Rickettsia_amblyommatis_GCF_001273795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpoMJ8Qb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTyqA60.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_amblyommatis_GCF_001273795.lastz\n",
      "Creating Rickettsia_amblyommatis_GCF_001273795 table\n",
      "Inserting data to Rickettsia_amblyommatis_GCF_001273795 table\n",
      "\n",
      "Running against Rickettsia_argasii_GCF_000965185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCDLtHR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpF_JRPi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_argasii_GCF_000965185.lastz\n",
      "Creating Rickettsia_argasii_GCF_000965185 table\n",
      "Inserting data to Rickettsia_argasii_GCF_000965185 table\n",
      "\n",
      "Running against Rickettsia_asembonensis_GCF_000828125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpB1JuUz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyuDPjO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_asembonensis_GCF_000828125.lastz\n",
      "Creating Rickettsia_asembonensis_GCF_000828125 table\n",
      "Inserting data to Rickettsia_asembonensis_GCF_000828125 table\n",
      "\n",
      "Running against Rickettsia_asiatica_GCF_007989425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHRxg4c.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJc2v7B.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_asiatica_GCF_007989425.lastz\n",
      "Creating Rickettsia_asiatica_GCF_007989425 table\n",
      "Inserting data to Rickettsia_asiatica_GCF_007989425 table\n",
      "\n",
      "Running against Rickettsia_australis_GCF_000284155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjtNh7J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphShItu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_australis_GCF_000284155.lastz\n",
      "Creating Rickettsia_australis_GCF_000284155 table\n",
      "Inserting data to Rickettsia_australis_GCF_000284155 table\n",
      "\n",
      "Running against Rickettsia_bellii_GCF_000012385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplI3ASS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp94c47B.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_bellii_GCF_000012385.lastz\n",
      "Creating Rickettsia_bellii_GCF_000012385 table\n",
      "Inserting data to Rickettsia_bellii_GCF_000012385 table\n",
      "\n",
      "Running against Rickettsia_buchneri_GCF_000696365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpm6tEis.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyWreeK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_buchneri_GCF_000696365.lastz\n",
      "Creating Rickettsia_buchneri_GCF_000696365 table\n",
      "Inserting data to Rickettsia_buchneri_GCF_000696365 table\n",
      "\n",
      "Running against Rickettsia_canadensis_GCF_000014345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHhpjEF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp89FEB3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_canadensis_GCF_000014345.lastz\n",
      "Creating Rickettsia_canadensis_GCF_000014345 table\n",
      "Inserting data to Rickettsia_canadensis_GCF_000014345 table\n",
      "\n",
      "Running against Rickettsia_conorii_GCF_000007025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4dQAJf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwoW0ia.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_conorii_GCF_000007025.lastz\n",
      "Creating Rickettsia_conorii_GCF_000007025 table\n",
      "Inserting data to Rickettsia_conorii_GCF_000007025 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Rickettsia_endosymbiont_GCF_002285905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpAzTzWg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFQmGVK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_endosymbiont_GCF_002285905.lastz\n",
      "Creating Rickettsia_endosymbiont_GCF_002285905 table\n",
      "Inserting data to Rickettsia_endosymbiont_GCF_002285905 table\n",
      "\n",
      "Running against Rickettsia_felis_GCF_000012145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvfRvlO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpP8kqDw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_felis_GCF_000012145.lastz\n",
      "Creating Rickettsia_felis_GCF_000012145 table\n",
      "Inserting data to Rickettsia_felis_GCF_000012145 table\n",
      "\n",
      "Running against Rickettsia_fournieri_GCF_900243065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxR9FPG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeleGEO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_fournieri_GCF_900243065.lastz\n",
      "Creating Rickettsia_fournieri_GCF_900243065 table\n",
      "Inserting data to Rickettsia_fournieri_GCF_900243065 table\n",
      "\n",
      "Running against Rickettsia_gravesii_GCF_000485845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpODkO4_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcQSFZA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_gravesii_GCF_000485845.lastz\n",
      "Creating Rickettsia_gravesii_GCF_000485845 table\n",
      "Inserting data to Rickettsia_gravesii_GCF_000485845 table\n",
      "\n",
      "Running against Rickettsia_heilongjiangensis_GCF_000221205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEybg92.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp52LZWd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_heilongjiangensis_GCF_000221205.lastz\n",
      "Creating Rickettsia_heilongjiangensis_GCF_000221205 table\n",
      "Inserting data to Rickettsia_heilongjiangensis_GCF_000221205 table\n",
      "\n",
      "Running against Rickettsia_helvetica_GCF_000255355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4Fe10y.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAGuxtW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_helvetica_GCF_000255355.lastz\n",
      "Creating Rickettsia_helvetica_GCF_000255355 table\n",
      "Inserting data to Rickettsia_helvetica_GCF_000255355 table\n",
      "\n",
      "Running against Rickettsia_honei_GCF_000263055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpL2sP0r.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFhVOyj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_honei_GCF_000263055.lastz\n",
      "Creating Rickettsia_honei_GCF_000263055 table\n",
      "Inserting data to Rickettsia_honei_GCF_000263055 table\n",
      "\n",
      "Running against Rickettsia_hoogstraalii_GCF_000825685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLUggc2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuxieic.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_hoogstraalii_GCF_000825685.lastz\n",
      "Creating Rickettsia_hoogstraalii_GCF_000825685 table\n",
      "Inserting data to Rickettsia_hoogstraalii_GCF_000825685 table\n",
      "\n",
      "Running against Rickettsia_japonica_GCF_000283595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnlFTuH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8EtcvZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_japonica_GCF_000283595.lastz\n",
      "Creating Rickettsia_japonica_GCF_000283595 table\n",
      "Inserting data to Rickettsia_japonica_GCF_000283595 table\n",
      "\n",
      "Running against Rickettsia_massiliae_GCF_000016625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvvDOfn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwYSdmn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_massiliae_GCF_000016625.lastz\n",
      "Creating Rickettsia_massiliae_GCF_000016625 table\n",
      "Inserting data to Rickettsia_massiliae_GCF_000016625 table\n",
      "\n",
      "Running against Rickettsia_monacensis_GCF_000499665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgkfdGU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppXLaZR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_monacensis_GCF_000499665.lastz\n",
      "Creating Rickettsia_monacensis_GCF_000499665 table\n",
      "Inserting data to Rickettsia_monacensis_GCF_000499665 table\n",
      "\n",
      "Running against Rickettsia_montanensis_GCF_000284175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpe4Xcpu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmmoQEu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_montanensis_GCF_000284175.lastz\n",
      "Creating Rickettsia_montanensis_GCF_000284175 table\n",
      "Inserting data to Rickettsia_montanensis_GCF_000284175 table\n",
      "\n",
      "Running against Rickettsia_parkeri_GCF_005549115.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIHIAt0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppJf0Jj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_parkeri_GCF_005549115.lastz\n",
      "Creating Rickettsia_parkeri_GCF_005549115 table\n",
      "Inserting data to Rickettsia_parkeri_GCF_005549115 table\n",
      "\n",
      "Running against Rickettsia_peacockii_GCF_000021525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphzFA3R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8IAREA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_peacockii_GCF_000021525.lastz\n",
      "Creating Rickettsia_peacockii_GCF_000021525 table\n",
      "Inserting data to Rickettsia_peacockii_GCF_000021525 table\n",
      "\n",
      "Running against Rickettsia_philipii_GCF_000283995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmptWLdbG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXE2Qeu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_philipii_GCF_000283995.lastz\n",
      "Creating Rickettsia_philipii_GCF_000283995 table\n",
      "Inserting data to Rickettsia_philipii_GCF_000283995 table\n",
      "\n",
      "Running against Rickettsia_prowazekii_GCF_000195735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIxEith.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpshMUDx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_prowazekii_GCF_000195735.lastz\n",
      "Creating Rickettsia_prowazekii_GCF_000195735 table\n",
      "Inserting data to Rickettsia_prowazekii_GCF_000195735 table\n",
      "\n",
      "Running against Rickettsia_raoultii_GCF_001975185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfsasGA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpiEbGNB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_raoultii_GCF_001975185.lastz\n",
      "Creating Rickettsia_raoultii_GCF_001975185 table\n",
      "Inserting data to Rickettsia_raoultii_GCF_001975185 table\n",
      "\n",
      "Running against Rickettsia_rhipicephali_GCF_000284075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYaeaRl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcSHv4E.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_rhipicephali_GCF_000284075.lastz\n",
      "Creating Rickettsia_rhipicephali_GCF_000284075 table\n",
      "Inserting data to Rickettsia_rhipicephali_GCF_000284075 table\n",
      "\n",
      "Running against Rickettsia_sibirica_GCF_000166935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjvEoSW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWIMJQ5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_sibirica_GCF_000166935.lastz\n",
      "Creating Rickettsia_sibirica_GCF_000166935 table\n",
      "Inserting data to Rickettsia_sibirica_GCF_000166935 table\n",
      "\n",
      "Running against Rickettsia_slovaca_GCF_000237845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4b2zy1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0K2Rsm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_slovaca_GCF_000237845.lastz\n",
      "Creating Rickettsia_slovaca_GCF_000237845 table\n",
      "Inserting data to Rickettsia_slovaca_GCF_000237845 table\n",
      "\n",
      "Running against Rickettsia_sp_GCF_001653015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMoTO1d.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQPkyxJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_sp_GCF_001653015.lastz\n",
      "Creating Rickettsia_sp_GCF_001653015 table\n",
      "Inserting data to Rickettsia_sp_GCF_001653015 table\n",
      "\n",
      "Running against Rickettsia_tamurae_GCF_000751075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfRerWf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpA9Df1j.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_tamurae_GCF_000751075.lastz\n",
      "Creating Rickettsia_tamurae_GCF_000751075 table\n",
      "Inserting data to Rickettsia_tamurae_GCF_000751075 table\n",
      "\n",
      "Running against Rickettsia_typhi_GCF_000008045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdZ_P5q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYtkgct.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_typhi_GCF_000008045.lastz\n",
      "Creating Rickettsia_typhi_GCF_000008045 table\n",
      "Inserting data to Rickettsia_typhi_GCF_000008045 table\n",
      "\n",
      "Running against Rickettsia_rickettsii_GCF_001951015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9u9Vv0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7m_R7b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/rickettsia/validate_intervals/lastz/Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_Rickettsia_rickettsii_GCF_001951015.lastz\n",
      "Creating Rickettsia_rickettsii_GCF_001951015 table\n",
      "Inserting data to Rickettsia_rickettsii_GCF_001951015 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/rickettsia/extract_probes_from_group/rickettsia_genome.conf --lastz results/phyluce/rickettsia/validate_intervals/lastz --probes 120 --probe-prefix uce_rickettsia_ --name-pattern \"Rickettsia_rickettsii_GCF_001951015_+33_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/rickettsia/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 12:01:49,366 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 12:01:49,369 - Phyluce - INFO - ---- Working on Rickettsia_aeschlimannii_GCF_001051325 genome ---\n",
      "2020-02-06 12:01:49,370 - Phyluce - INFO - Reading Rickettsia_aeschlimannii_GCF_001051325 genome\n",
      "2020-02-06 12:01:50,072 - Phyluce - INFO - Rickettsia_aeschlimannii_GCF_001051325: 197 uces, 4 dupes, 193 non-dupes, 0 orient drop, 1 length drop, 192 written\n",
      "2020-02-06 12:01:50,072 - Phyluce - INFO - ------- Working on Rickettsia_africae_GCF_000023005 genome ------\n",
      "2020-02-06 12:01:50,073 - Phyluce - INFO - Reading Rickettsia_africae_GCF_000023005 genome\n",
      "2020-02-06 12:01:50,686 - Phyluce - INFO - Rickettsia_africae_GCF_000023005: 197 uces, 0 dupes, 197 non-dupes, 4 orient drop, 2 length drop, 191 written\n",
      "2020-02-06 12:01:50,686 - Phyluce - INFO - -------- Working on Rickettsia_akari_GCF_000018205 genome -------\n",
      "2020-02-06 12:01:50,704 - Phyluce - INFO - Reading Rickettsia_akari_GCF_000018205 genome\n",
      "2020-02-06 12:01:51,293 - Phyluce - INFO - Rickettsia_akari_GCF_000018205: 188 uces, 0 dupes, 188 non-dupes, 3 orient drop, 2 length drop, 183 written\n",
      "2020-02-06 12:01:51,293 - Phyluce - INFO - ---- Working on Rickettsia_amblyommatis_GCF_001273795 genome ----\n",
      "2020-02-06 12:01:51,294 - Phyluce - INFO - Reading Rickettsia_amblyommatis_GCF_001273795 genome\n",
      "2020-02-06 12:01:51,905 - Phyluce - INFO - Rickettsia_amblyommatis_GCF_001273795: 195 uces, 0 dupes, 195 non-dupes, 4 orient drop, 2 length drop, 189 written\n",
      "2020-02-06 12:01:51,905 - Phyluce - INFO - ------- Working on Rickettsia_argasii_GCF_000965185 genome ------\n",
      "2020-02-06 12:01:51,924 - Phyluce - INFO - Reading Rickettsia_argasii_GCF_000965185 genome\n",
      "2020-02-06 12:01:52,533 - Phyluce - INFO - Rickettsia_argasii_GCF_000965185: 196 uces, 4 dupes, 192 non-dupes, 0 orient drop, 1 length drop, 191 written\n",
      "2020-02-06 12:01:52,533 - Phyluce - INFO - ---- Working on Rickettsia_asembonensis_GCF_000828125 genome ----\n",
      "2020-02-06 12:01:52,545 - Phyluce - INFO - Reading Rickettsia_asembonensis_GCF_000828125 genome\n",
      "2020-02-06 12:01:53,215 - Phyluce - INFO - Rickettsia_asembonensis_GCF_000828125: 191 uces, 9 dupes, 182 non-dupes, 0 orient drop, 1 length drop, 181 written\n",
      "2020-02-06 12:01:53,215 - Phyluce - INFO - ------ Working on Rickettsia_asiatica_GCF_007989425 genome ------\n",
      "2020-02-06 12:01:53,225 - Phyluce - INFO - Reading Rickettsia_asiatica_GCF_007989425 genome\n",
      "2020-02-06 12:01:53,823 - Phyluce - INFO - Rickettsia_asiatica_GCF_007989425: 192 uces, 1 dupes, 191 non-dupes, 4 orient drop, 3 length drop, 184 written\n",
      "2020-02-06 12:01:53,823 - Phyluce - INFO - ------ Working on Rickettsia_australis_GCF_000284155 genome -----\n",
      "2020-02-06 12:01:53,823 - Phyluce - INFO - Reading Rickettsia_australis_GCF_000284155 genome\n",
      "2020-02-06 12:01:54,432 - Phyluce - INFO - Rickettsia_australis_GCF_000284155: 190 uces, 0 dupes, 190 non-dupes, 2 orient drop, 2 length drop, 186 written\n",
      "2020-02-06 12:01:54,432 - Phyluce - INFO - ------- Working on Rickettsia_bellii_GCF_000012385 genome -------\n",
      "2020-02-06 12:01:54,433 - Phyluce - INFO - Reading Rickettsia_bellii_GCF_000012385 genome\n",
      "2020-02-06 12:01:55,038 - Phyluce - INFO - Rickettsia_bellii_GCF_000012385: 185 uces, 0 dupes, 185 non-dupes, 2 orient drop, 0 length drop, 183 written\n",
      "2020-02-06 12:01:55,038 - Phyluce - INFO - ------ Working on Rickettsia_buchneri_GCF_000696365 genome ------\n",
      "2020-02-06 12:01:55,047 - Phyluce - INFO - Reading Rickettsia_buchneri_GCF_000696365 genome\n",
      "2020-02-06 12:01:55,640 - Phyluce - INFO - Rickettsia_buchneri_GCF_000696365: 190 uces, 10 dupes, 180 non-dupes, 0 orient drop, 0 length drop, 180 written\n",
      "2020-02-06 12:01:55,640 - Phyluce - INFO - ----- Working on Rickettsia_canadensis_GCF_000014345 genome -----\n",
      "2020-02-06 12:01:55,641 - Phyluce - INFO - Reading Rickettsia_canadensis_GCF_000014345 genome\n",
      "2020-02-06 12:01:56,189 - Phyluce - INFO - Rickettsia_canadensis_GCF_000014345: 185 uces, 0 dupes, 185 non-dupes, 2 orient drop, 0 length drop, 183 written\n",
      "2020-02-06 12:01:56,189 - Phyluce - INFO - ------- Working on Rickettsia_conorii_GCF_000007025 genome ------\n",
      "2020-02-06 12:01:56,196 - Phyluce - INFO - Reading Rickettsia_conorii_GCF_000007025 genome\n",
      "2020-02-06 12:01:56,810 - Phyluce - INFO - Rickettsia_conorii_GCF_000007025: 197 uces, 0 dupes, 197 non-dupes, 3 orient drop, 3 length drop, 191 written\n",
      "2020-02-06 12:01:56,811 - Phyluce - INFO - ---- Working on Rickettsia_endosymbiont_GCF_002285905 genome ----\n",
      "2020-02-06 12:01:56,813 - Phyluce - INFO - Reading Rickettsia_endosymbiont_GCF_002285905 genome\n",
      "2020-02-06 12:01:57,440 - Phyluce - INFO - Rickettsia_endosymbiont_GCF_002285905: 181 uces, 0 dupes, 181 non-dupes, 3 orient drop, 3 length drop, 175 written\n",
      "2020-02-06 12:01:57,440 - Phyluce - INFO - -------- Working on Rickettsia_felis_GCF_000012145 genome -------\n",
      "2020-02-06 12:01:57,441 - Phyluce - INFO - Reading Rickettsia_felis_GCF_000012145 genome\n",
      "2020-02-06 12:01:58,073 - Phyluce - INFO - Rickettsia_felis_GCF_000012145: 190 uces, 0 dupes, 190 non-dupes, 3 orient drop, 2 length drop, 185 written\n",
      "2020-02-06 12:01:58,073 - Phyluce - INFO - ------ Working on Rickettsia_fournieri_GCF_900243065 genome -----\n",
      "2020-02-06 12:01:58,083 - Phyluce - INFO - Reading Rickettsia_fournieri_GCF_900243065 genome\n",
      "2020-02-06 12:01:58,709 - Phyluce - INFO - Rickettsia_fournieri_GCF_900243065: 196 uces, 4 dupes, 192 non-dupes, 0 orient drop, 1 length drop, 191 written\n",
      "2020-02-06 12:01:58,709 - Phyluce - INFO - ------ Working on Rickettsia_gravesii_GCF_000485845 genome ------\n",
      "2020-02-06 12:01:58,710 - Phyluce - INFO - Reading Rickettsia_gravesii_GCF_000485845 genome\n",
      "2020-02-06 12:01:59,323 - Phyluce - INFO - Rickettsia_gravesii_GCF_000485845: 195 uces, 4 dupes, 191 non-dupes, 0 orient drop, 1 length drop, 190 written\n",
      "2020-02-06 12:01:59,323 - Phyluce - INFO - -- Working on Rickettsia_heilongjiangensis_GCF_000221205 genome -\n",
      "2020-02-06 12:01:59,324 - Phyluce - INFO - Reading Rickettsia_heilongjiangensis_GCF_000221205 genome\n",
      "2020-02-06 12:01:59,946 - Phyluce - INFO - Rickettsia_heilongjiangensis_GCF_000221205: 197 uces, 0 dupes, 197 non-dupes, 4 orient drop, 1 length drop, 192 written\n",
      "2020-02-06 12:01:59,946 - Phyluce - INFO - ------ Working on Rickettsia_helvetica_GCF_000255355 genome -----\n",
      "2020-02-06 12:01:59,947 - Phyluce - INFO - Reading Rickettsia_helvetica_GCF_000255355 genome\n",
      "2020-02-06 12:02:00,547 - Phyluce - INFO - Rickettsia_helvetica_GCF_000255355: 190 uces, 1 dupes, 189 non-dupes, 3 orient drop, 5 length drop, 181 written\n",
      "2020-02-06 12:02:00,547 - Phyluce - INFO - -------- Working on Rickettsia_honei_GCF_000263055 genome -------\n",
      "2020-02-06 12:02:00,556 - Phyluce - INFO - Reading Rickettsia_honei_GCF_000263055 genome\n",
      "2020-02-06 12:02:01,179 - Phyluce - INFO - Rickettsia_honei_GCF_000263055: 197 uces, 5 dupes, 192 non-dupes, 0 orient drop, 1 length drop, 191 written\n",
      "2020-02-06 12:02:01,179 - Phyluce - INFO - ---- Working on Rickettsia_hoogstraalii_GCF_000825685 genome ----\n",
      "2020-02-06 12:02:01,201 - Phyluce - INFO - Reading Rickettsia_hoogstraalii_GCF_000825685 genome\n",
      "2020-02-06 12:02:01,815 - Phyluce - INFO - Rickettsia_hoogstraalii_GCF_000825685: 191 uces, 0 dupes, 191 non-dupes, 2 orient drop, 3 length drop, 186 written\n",
      "2020-02-06 12:02:01,815 - Phyluce - INFO - ------ Working on Rickettsia_japonica_GCF_000283595 genome ------\n",
      "2020-02-06 12:02:01,824 - Phyluce - INFO - Reading Rickettsia_japonica_GCF_000283595 genome\n",
      "2020-02-06 12:02:02,452 - Phyluce - INFO - Rickettsia_japonica_GCF_000283595: 197 uces, 0 dupes, 197 non-dupes, 3 orient drop, 3 length drop, 191 written\n",
      "2020-02-06 12:02:02,452 - Phyluce - INFO - ------ Working on Rickettsia_massiliae_GCF_000016625 genome -----\n",
      "2020-02-06 12:02:02,469 - Phyluce - INFO - Reading Rickettsia_massiliae_GCF_000016625 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 12:02:03,095 - Phyluce - INFO - Rickettsia_massiliae_GCF_000016625: 196 uces, 0 dupes, 196 non-dupes, 3 orient drop, 3 length drop, 190 written\n",
      "2020-02-06 12:02:03,095 - Phyluce - INFO - ----- Working on Rickettsia_monacensis_GCF_000499665 genome -----\n",
      "2020-02-06 12:02:03,096 - Phyluce - INFO - Reading Rickettsia_monacensis_GCF_000499665 genome\n",
      "2020-02-06 12:02:03,662 - Phyluce - INFO - Rickettsia_monacensis_GCF_000499665: 188 uces, 0 dupes, 188 non-dupes, 4 orient drop, 0 length drop, 184 written\n",
      "2020-02-06 12:02:03,662 - Phyluce - INFO - ----- Working on Rickettsia_montanensis_GCF_000284175 genome ----\n",
      "2020-02-06 12:02:03,662 - Phyluce - INFO - Reading Rickettsia_montanensis_GCF_000284175 genome\n",
      "2020-02-06 12:02:04,288 - Phyluce - INFO - Rickettsia_montanensis_GCF_000284175: 197 uces, 0 dupes, 197 non-dupes, 3 orient drop, 5 length drop, 189 written\n",
      "2020-02-06 12:02:04,289 - Phyluce - INFO - ------- Working on Rickettsia_parkeri_GCF_005549115 genome ------\n",
      "2020-02-06 12:02:04,294 - Phyluce - INFO - Reading Rickettsia_parkeri_GCF_005549115 genome\n",
      "2020-02-06 12:02:04,919 - Phyluce - INFO - Rickettsia_parkeri_GCF_005549115: 197 uces, 0 dupes, 197 non-dupes, 4 orient drop, 2 length drop, 191 written\n",
      "2020-02-06 12:02:04,919 - Phyluce - INFO - ------ Working on Rickettsia_peacockii_GCF_000021525 genome -----\n",
      "2020-02-06 12:02:04,926 - Phyluce - INFO - Reading Rickettsia_peacockii_GCF_000021525 genome\n",
      "2020-02-06 12:02:05,562 - Phyluce - INFO - Rickettsia_peacockii_GCF_000021525: 197 uces, 0 dupes, 197 non-dupes, 3 orient drop, 3 length drop, 191 written\n",
      "2020-02-06 12:02:05,563 - Phyluce - INFO - ------ Working on Rickettsia_philipii_GCF_000283995 genome ------\n",
      "2020-02-06 12:02:05,570 - Phyluce - INFO - Reading Rickettsia_philipii_GCF_000283995 genome\n",
      "2020-02-06 12:02:06,189 - Phyluce - INFO - Rickettsia_philipii_GCF_000283995: 196 uces, 0 dupes, 196 non-dupes, 4 orient drop, 1 length drop, 191 written\n",
      "2020-02-06 12:02:06,189 - Phyluce - INFO - ----- Working on Rickettsia_prowazekii_GCF_000195735 genome -----\n",
      "2020-02-06 12:02:06,190 - Phyluce - INFO - Reading Rickettsia_prowazekii_GCF_000195735 genome\n",
      "2020-02-06 12:02:06,715 - Phyluce - INFO - Rickettsia_prowazekii_GCF_000195735: 179 uces, 0 dupes, 179 non-dupes, 2 orient drop, 1 length drop, 176 written\n",
      "2020-02-06 12:02:06,715 - Phyluce - INFO - ------ Working on Rickettsia_raoultii_GCF_001975185 genome ------\n",
      "2020-02-06 12:02:06,716 - Phyluce - INFO - Reading Rickettsia_raoultii_GCF_001975185 genome\n",
      "2020-02-06 12:02:07,343 - Phyluce - INFO - Rickettsia_raoultii_GCF_001975185: 197 uces, 0 dupes, 197 non-dupes, 4 orient drop, 1 length drop, 192 written\n",
      "2020-02-06 12:02:07,344 - Phyluce - INFO - ---- Working on Rickettsia_rhipicephali_GCF_000284075 genome ----\n",
      "2020-02-06 12:02:07,344 - Phyluce - INFO - Reading Rickettsia_rhipicephali_GCF_000284075 genome\n",
      "2020-02-06 12:02:07,976 - Phyluce - INFO - Rickettsia_rhipicephali_GCF_000284075: 197 uces, 0 dupes, 197 non-dupes, 3 orient drop, 4 length drop, 190 written\n",
      "2020-02-06 12:02:07,976 - Phyluce - INFO - ------ Working on Rickettsia_sibirica_GCF_000166935 genome ------\n",
      "2020-02-06 12:02:07,977 - Phyluce - INFO - Reading Rickettsia_sibirica_GCF_000166935 genome\n",
      "2020-02-06 12:02:08,593 - Phyluce - INFO - Rickettsia_sibirica_GCF_000166935: 197 uces, 0 dupes, 197 non-dupes, 4 orient drop, 2 length drop, 191 written\n",
      "2020-02-06 12:02:08,593 - Phyluce - INFO - ------- Working on Rickettsia_slovaca_GCF_000237845 genome ------\n",
      "2020-02-06 12:02:08,605 - Phyluce - INFO - Reading Rickettsia_slovaca_GCF_000237845 genome\n",
      "2020-02-06 12:02:09,223 - Phyluce - INFO - Rickettsia_slovaca_GCF_000237845: 197 uces, 0 dupes, 197 non-dupes, 4 orient drop, 1 length drop, 192 written\n",
      "2020-02-06 12:02:09,223 - Phyluce - INFO - --------- Working on Rickettsia_sp_GCF_001653015 genome ---------\n",
      "2020-02-06 12:02:09,235 - Phyluce - INFO - Reading Rickettsia_sp_GCF_001653015 genome\n",
      "2020-02-06 12:02:09,853 - Phyluce - INFO - Rickettsia_sp_GCF_001653015: 196 uces, 5 dupes, 191 non-dupes, 0 orient drop, 1 length drop, 190 written\n",
      "2020-02-06 12:02:09,853 - Phyluce - INFO - ------- Working on Rickettsia_tamurae_GCF_000751075 genome ------\n",
      "2020-02-06 12:02:09,864 - Phyluce - INFO - Reading Rickettsia_tamurae_GCF_000751075 genome\n",
      "2020-02-06 12:02:10,450 - Phyluce - INFO - Rickettsia_tamurae_GCF_000751075: 190 uces, 5 dupes, 185 non-dupes, 0 orient drop, 0 length drop, 185 written\n",
      "2020-02-06 12:02:10,450 - Phyluce - INFO - -------- Working on Rickettsia_typhi_GCF_000008045 genome -------\n",
      "2020-02-06 12:02:10,462 - Phyluce - INFO - Reading Rickettsia_typhi_GCF_000008045 genome\n",
      "2020-02-06 12:02:11,003 - Phyluce - INFO - Rickettsia_typhi_GCF_000008045: 179 uces, 0 dupes, 179 non-dupes, 2 orient drop, 0 length drop, 177 written\n",
      "2020-02-06 12:02:11,003 - Phyluce - INFO - ----- Working on Rickettsia_rickettsii_GCF_001951015 genome -----\n",
      "2020-02-06 12:02:11,003 - Phyluce - INFO - Reading Rickettsia_rickettsii_GCF_001951015 genome\n",
      "2020-02-06 12:02:11,619 - Phyluce - INFO - Rickettsia_rickettsii_GCF_001951015: 197 uces, 0 dupes, 197 non-dupes, 3 orient drop, 2 length drop, 192 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/rickettsia/extract_probes_from_group/probe_fasta --output results/phyluce/rickettsia/extract_probes_from_group/multifastas.sqlite --base-taxon Rickettsia_rickettsii_GCF_001951015;\n",
      "rickettsia_aeschlimannii_gcf_001051325.\n",
      "rickettsia_africae_gcf_000023005.\n",
      "rickettsia_akari_gcf_000018205.\n",
      "rickettsia_amblyommatis_gcf_001273795.\n",
      "rickettsia_argasii_gcf_000965185.\n",
      "rickettsia_asembonensis_gcf_000828125.\n",
      "rickettsia_asiatica_gcf_007989425.\n",
      "rickettsia_australis_gcf_000284155.\n",
      "rickettsia_bellii_gcf_000012385.\n",
      "rickettsia_buchneri_gcf_000696365.\n",
      "rickettsia_canadensis_gcf_000014345.\n",
      "rickettsia_conorii_gcf_000007025.\n",
      "rickettsia_endosymbiont_gcf_002285905.\n",
      "rickettsia_felis_gcf_000012145.\n",
      "rickettsia_fournieri_gcf_900243065.\n",
      "rickettsia_gravesii_gcf_000485845.\n",
      "rickettsia_heilongjiangensis_gcf_000221205.\n",
      "rickettsia_helvetica_gcf_000255355.\n",
      "rickettsia_honei_gcf_000263055.\n",
      "rickettsia_hoogstraalii_gcf_000825685.\n",
      "rickettsia_japonica_gcf_000283595.\n",
      "rickettsia_massiliae_gcf_000016625.\n",
      "rickettsia_monacensis_gcf_000499665.\n",
      "rickettsia_montanensis_gcf_000284175.\n",
      "rickettsia_parkeri_gcf_005549115.\n",
      "rickettsia_peacockii_gcf_000021525.\n",
      "rickettsia_philipii_gcf_000283995.\n",
      "rickettsia_prowazekii_gcf_000195735.\n",
      "rickettsia_raoultii_gcf_001975185.\n",
      "rickettsia_rhipicephali_gcf_000284075.\n",
      "rickettsia_sibirica_gcf_000166935.\n",
      "rickettsia_slovaca_gcf_000237845.\n",
      "rickettsia_sp_gcf_001653015.\n",
      "rickettsia_tamurae_gcf_000751075.\n",
      "rickettsia_typhi_gcf_000008045.\n",
      "rickettsia_rickettsii_gcf_001951015.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/rickettsia/extract_probes_from_group/multifastas.sqlite --base-taxon Rickettsia_rickettsii_GCF_001951015;\n",
      "Loci shared by 0 taxa:\t196.0\n",
      "Loci shared by 1 taxa:\t196.0\n",
      "Loci shared by 2 taxa:\t196.0\n",
      "Loci shared by 3 taxa:\t195.0\n",
      "Loci shared by 4 taxa:\t194.0\n",
      "Loci shared by 5 taxa:\t194.0\n",
      "Loci shared by 6 taxa:\t194.0\n",
      "Loci shared by 7 taxa:\t193.0\n",
      "Loci shared by 8 taxa:\t193.0\n",
      "Loci shared by 9 taxa:\t193.0\n",
      "Loci shared by 10 taxa:\t193.0\n",
      "Loci shared by 11 taxa:\t193.0\n",
      "Loci shared by 12 taxa:\t192.0\n",
      "Loci shared by 13 taxa:\t192.0\n",
      "Loci shared by 14 taxa:\t192.0\n",
      "Loci shared by 15 taxa:\t192.0\n",
      "Loci shared by 16 taxa:\t191.0\n",
      "Loci shared by 17 taxa:\t191.0\n",
      "Loci shared by 18 taxa:\t191.0\n",
      "Loci shared by 19 taxa:\t191.0\n",
      "Loci shared by 20 taxa:\t191.0\n",
      "Loci shared by 21 taxa:\t190.0\n",
      "Loci shared by 22 taxa:\t188.0\n",
      "Loci shared by 23 taxa:\t187.0\n",
      "Loci shared by 24 taxa:\t186.0\n",
      "Loci shared by 25 taxa:\t185.0\n",
      "Loci shared by 26 taxa:\t185.0\n",
      "Loci shared by 27 taxa:\t184.0\n",
      "Loci shared by 28 taxa:\t184.0\n",
      "Loci shared by 29 taxa:\t182.0\n",
      "Loci shared by 30 taxa:\t182.0\n",
      "Loci shared by 31 taxa:\t181.0\n",
      "Loci shared by 32 taxa:\t179.0\n",
      "Loci shared by 33 taxa:\t177.0\n",
      "Loci shared by 34 taxa:\t173.0\n",
      "Loci shared by 35 taxa:\t168.0\n",
      "Loci shared by 36 taxa:\t149.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/rickettsia/extract_probes_from_group/multifastas.sqlite  --base-taxon Rickettsia_rickettsii_GCF_001951015 --output results/phyluce/rickettsia/extract_probes_from_group/Rickettsia_rickettsii_GCF_001951015+33-back-to-35.conf --specific-counts 35;\n",
      "Counter({'rickettsia_japonica_gcf_000283595': 168, 'rickettsia_gravesii_gcf_000485845': 168, 'rickettsia_argasii_gcf_000965185': 168, 'rickettsia_sibirica_gcf_000166935': 168, 'rickettsia_tamurae_gcf_000751075': 168, 'rickettsia_conorii_gcf_000007025': 168, 'rickettsia_honei_gcf_000263055': 168, 'rickettsia_heilongjiangensis_gcf_000221205': 168, 'rickettsia_monacensis_gcf_000499665': 168, 'rickettsia_fournieri_gcf_900243065': 168, 'rickettsia_philipii_gcf_000283995': 168, 'rickettsia_rhipicephali_gcf_000284075': 168, 'rickettsia_aeschlimannii_gcf_001051325': 168, 'rickettsia_asiatica_gcf_007989425': 168, 'rickettsia_prowazekii_gcf_000195735': 168, 'rickettsia_akari_gcf_000018205': 168, 'rickettsia_rickettsii_gcf_001951015': 168, 'rickettsia_montanensis_gcf_000284175': 168, 'rickettsia_slovaca_gcf_000237845': 168, 'rickettsia_australis_gcf_000284155': 168, 'rickettsia_helvetica_gcf_000255355': 168, 'rickettsia_raoultii_gcf_001975185': 168, 'rickettsia_parkeri_gcf_005549115': 168, 'rickettsia_hoogstraalii_gcf_000825685': 168, 'rickettsia_massiliae_gcf_000016625': 168, 'rickettsia_felis_gcf_000012145': 168, 'rickettsia_africae_gcf_000023005': 168, 'rickettsia_amblyommatis_gcf_001273795': 167, 'rickettsia_sp_gcf_001653015': 167, 'rickettsia_typhi_gcf_000008045': 167, 'rickettsia_peacockii_gcf_000021525': 167, 'rickettsia_canadensis_gcf_000014345': 167, 'rickettsia_bellii_gcf_000012385': 167, 'rickettsia_buchneri_gcf_000696365': 165, 'rickettsia_asembonensis_gcf_000828125': 164, 'rickettsia_endosymbiont_gcf_002285905': 162})\n",
      "Total loci = 168\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(35)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/rickettsia/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/rickettsia/extract_probes_from_group/Rickettsia_rickettsii_GCF_001951015+33-back-to-35.conf --probe-prefix uce_rickettsia_ --designer rnplattii --design rickettsia_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 168\n",
      "Probe Count = 10563\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list.fasta --query results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  12:02:45\n",
      "Ended:  Thu Feb 06, 2020  12:03:22\n",
      "Time for execution:  0.628028416634 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list.fasta --lastz results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_rickettsia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 10562 fasta sequences.  Filtered 0 duplicates. Kept 10563.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/rickettsia/final_probe_design/rickettsia_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 12:20:34 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 10563\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 845040\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 2M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 89M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3946989\n",
      "Max number of word counting entries: 88846733\n",
      "\n",
      "# comparing sequences from          0  to       1760\n",
      ".---------- new table with      383 representatives\n",
      "# comparing sequences from       1760  to       3227\n",
      "----------    544 remaining sequences to the next cycle\n",
      "---------- new table with      176 representatives\n",
      "# comparing sequences from       2683  to       3996\n",
      "----------    719 remaining sequences to the next cycle\n",
      "---------- new table with      163 representatives\n",
      "# comparing sequences from       3277  to       4491\n",
      "----------    476 remaining sequences to the next cycle\n",
      "---------- new table with      143 representatives\n",
      "# comparing sequences from       4015  to       5106\n",
      "----------    513 remaining sequences to the next cycle\n",
      "---------- new table with      126 representatives\n",
      "# comparing sequences from       4593  to       5588\n",
      "----------    365 remaining sequences to the next cycle\n",
      "---------- new table with      148 representatives\n",
      "# comparing sequences from       5223  to       6113\n",
      "----------    450 remaining sequences to the next cycle\n",
      "---------- new table with      121 representatives\n",
      "# comparing sequences from       5663  to       6479\n",
      "----------    239 remaining sequences to the next cycle\n",
      "---------- new table with      136 representatives\n",
      "# comparing sequences from       6240  to       6960\n",
      "----------    211 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       6749  to       7384\n",
      "----------    135 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       7249  to       7801\n",
      "----------    145 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       7656  to       8140\n",
      "----------     80 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       8060  to       8477\n",
      "----------      8 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       8469  to       8818\n",
      "...................---------- new table with       93 representatives\n",
      "# comparing sequences from       8818  to       9108\n",
      ".....................---------- new table with       76 representatives\n",
      "# comparing sequences from       9108  to       9350\n",
      "99.9%---------- new table with       50 representatives\n",
      "# comparing sequences from       9350  to       9552\n",
      "99.9%---------- new table with       47 representatives\n",
      "# comparing sequences from       9552  to       9720\n",
      "..................---------- new table with       43 representatives\n",
      "# comparing sequences from       9720  to      10563\n",
      "..........    10000  finished       2266  clusters\n",
      "---------- new table with      190 representatives\n",
      "\n",
      "    10563  finished       2395  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 89M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.73\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Salmonella "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'salmonella'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Salmonella_bongori_GCF_000252995\",\n",
    "               \"Salmonella_bongori_GCF_002035475\",\n",
    "               \"Salmonella_bongori_GCA_003497195\",\n",
    "               \"Salmonella_bongori_GCF_902387725\",\n",
    "               \"Salmonella_bongori_GCA_003452365\",\n",
    "               \"Salmonella_bongori_GCF_900635485\",\n",
    "               \"Salmonella_bongori_GCF_002211925\",\n",
    "               \"Salmonella_bongori_GCF_000709535\",\n",
    "               \"Salmonella_bongori_GCA_003522735\",\n",
    "               \"Salmonella_bongori_GCF_002035285\",\n",
    "               \"Salmonella_arizonae_GCF_001951595\",\n",
    "               \"Salmonella_arizonae_GCF_001951445\",\n",
    "               \"Salmonella_arizonae_GCF_000018625\",\n",
    "               \"Salmonella_arizonae_GCF_001970905\",\n",
    "               \"Salmonella_diarizonae_GCF_002241435\",\n",
    "               \"Salmonella_diarizonae_GCF_001832255\",\n",
    "               \"Salmonella_diarizonae_GCF_002063885\",\n",
    "               \"Salmonella_diarizonae_GCF_002035715\",\n",
    "               \"Salmonella_enterica_GCF_009208155\",\n",
    "               \"Salmonella_enterica_GCF_005938475\",\n",
    "               \"Salmonella_enterica_GCF_002064535\",\n",
    "               \"Salmonella_houtenae_GCF_002047425\",\n",
    "               \"Salmonella_houtenae_GCF_002045905\",\n",
    "               \"Salmonella_houtenae_GCF_002127815\",\n",
    "               \"Salmonella_houtenae_GCF_002106115\",\n",
    "               \"Salmonella_indica_GCF_002066855\",\n",
    "               \"Salmonella_indica_GCF_002035225\",\n",
    "               \"Salmonella_indica_GCF_002066875\",\n",
    "               \"Salmonella_indica_GCF_002266165\",\n",
    "               \"Salmonella_salamae_GCF_002036105\",\n",
    "               \"Salmonella_salamae_GCF_900478195\",\n",
    "               \"Salmonella_salamae_GCF_003994675\",\n",
    "               \"Salmonella_salamae_GCF_002036155\" ]\n",
    "                    \n",
    "reference_taxon =  \"Salmonella_enterica_GCF_001159405\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000252995.1_ASM25299v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1327500 bytes  885028.00 bytes/sec\n",
      "total size is 1327067  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002035475.1_ASM203547v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1325640 bytes  883788.00 bytes/sec\n",
      "total size is 1325206  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003497195.1_ASM349719v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1288886 bytes  859285.33 bytes/sec\n",
      "total size is 1288460  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902387725.1_UHGG_MGYG-HGUT-02496_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1368888 bytes  547572.00 bytes/sec\n",
      "total size is 1368437  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003452365.1_ASM345236v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 743485 bytes  495684.67 bytes/sec\n",
      "total size is 743195  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900635485.1_29751_D01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1329801 bytes  886562.00 bytes/sec\n",
      "total size is 1329369  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002211925.1_ASM221192v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1328644 bytes  885790.67 bytes/sec\n",
      "total size is 1328210  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000709535.1_ASM70953v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1307795 bytes  871891.33 bytes/sec\n",
      "total size is 1307370  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003522735.1_ASM352273v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1292446 bytes  861658.67 bytes/sec\n",
      "total size is 1292020  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002035285.1_ASM203528v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1327801 bytes  885228.67 bytes/sec\n",
      "total size is 1327367  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001951595.1_ASM195159v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1450258 bytes  966866.67 bytes/sec\n",
      "total size is 1449792  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001951445.1_ASM195144v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1504894 bytes  1003290.67 bytes/sec\n",
      "total size is 1504420  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000018625.1_ASM1862v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1368367 bytes  912272.67 bytes/sec\n",
      "total size is 1367927  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001970905.1_ASM197090v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1343226 bytes  895512.00 bytes/sec\n",
      "total size is 1342792  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002241435.2_ASM224143v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1499999 bytes  1000027.33 bytes/sec\n",
      "total size is 1499525  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001832255.1_ASM183225v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1529287 bytes  1019552.67 bytes/sec\n",
      "total size is 1528805  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002063885.1_ASM206388v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1440902 bytes  960629.33 bytes/sec\n",
      "total size is 1440444  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002035715.1_ASM203571v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1388306 bytes  925565.33 bytes/sec\n",
      "total size is 1387856  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009208155.1_ASM920815v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1432754 bytes  955197.33 bytes/sec\n",
      "total size is 1432296  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005938475.1_ASM593847v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1426808 bytes  951233.33 bytes/sec\n",
      "total size is 1426350  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001159405.1_8490_2_88_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1493212 bytes  995502.67 bytes/sec\n",
      "total size is 1492740  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002064535.1_ASM206453v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1374955 bytes  916664.67 bytes/sec\n",
      "total size is 1374513  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002047425.1_ASM204742v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1377820 bytes  918574.67 bytes/sec\n",
      "total size is 1377370  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002045905.1_ASM204590v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1453452 bytes  968996.00 bytes/sec\n",
      "total size is 1452986  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002127815.1_ASM212781v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1408385 bytes  938951.33 bytes/sec\n",
      "total size is 1407935  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002106115.1_ASM210611v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1397486 bytes  931685.33 bytes/sec\n",
      "total size is 1397036  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002066855.1_ASM206685v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1394663 bytes  929803.33 bytes/sec\n",
      "total size is 1394213  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002035225.1_ASM203522v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1370125 bytes  913444.67 bytes/sec\n",
      "total size is 1369683  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002066875.1_ASM206687v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1412564 bytes  941737.33 bytes/sec\n",
      "total size is 1412106  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002266165.1_ASM226616v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1401074 bytes  934077.33 bytes/sec\n",
      "total size is 1400624  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002036105.1_ASM203610v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1498532 bytes  999049.33 bytes/sec\n",
      "total size is 1498058  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900478195.1_47297_E02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1405005 bytes  936698.00 bytes/sec\n",
      "total size is 1404557  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003994675.1_ASM399467v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1445111 bytes  963435.33 bytes/sec\n",
      "total size is 1444645  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002036155.1_ASM203615v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1532138 bytes  1021453.33 bytes/sec\n",
      "total size is 1531656  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482531 (\"sim_Salmonella_bongori_GCF_000252995\") has been submitted\n",
      "Your job 5482532 (\"sim_Salmonella_bongori_GCF_002035475\") has been submitted\n",
      "Your job 5482533 (\"sim_Salmonella_bongori_GCA_003497195\") has been submitted\n",
      "Your job 5482534 (\"sim_Salmonella_bongori_GCF_902387725\") has been submitted\n",
      "Your job 5482535 (\"sim_Salmonella_bongori_GCA_003452365\") has been submitted\n",
      "Your job 5482536 (\"sim_Salmonella_bongori_GCF_900635485\") has been submitted\n",
      "Your job 5482537 (\"sim_Salmonella_bongori_GCF_002211925\") has been submitted\n",
      "Your job 5482538 (\"sim_Salmonella_bongori_GCF_000709535\") has been submitted\n",
      "Your job 5482539 (\"sim_Salmonella_bongori_GCA_003522735\") has been submitted\n",
      "Your job 5482540 (\"sim_Salmonella_bongori_GCF_002035285\") has been submitted\n",
      "Your job 5482541 (\"sim_Salmonella_arizonae_GCF_001951595\") has been submitted\n",
      "Your job 5482542 (\"sim_Salmonella_arizonae_GCF_001951445\") has been submitted\n",
      "Your job 5482543 (\"sim_Salmonella_arizonae_GCF_000018625\") has been submitted\n",
      "Your job 5482544 (\"sim_Salmonella_arizonae_GCF_001970905\") has been submitted\n",
      "Your job 5482545 (\"sim_Salmonella_diarizonae_GCF_002241435\") has been submitted\n",
      "Your job 5482546 (\"sim_Salmonella_diarizonae_GCF_001832255\") has been submitted\n",
      "Your job 5482547 (\"sim_Salmonella_diarizonae_GCF_002063885\") has been submitted\n",
      "Your job 5482548 (\"sim_Salmonella_diarizonae_GCF_002035715\") has been submitted\n",
      "Your job 5482549 (\"sim_Salmonella_enterica_GCF_009208155\") has been submitted\n",
      "Your job 5482550 (\"sim_Salmonella_enterica_GCF_005938475\") has been submitted\n",
      "Your job 5482551 (\"sim_Salmonella_enterica_GCF_002064535\") has been submitted\n",
      "Your job 5482552 (\"sim_Salmonella_houtenae_GCF_002047425\") has been submitted\n",
      "Your job 5482553 (\"sim_Salmonella_houtenae_GCF_002045905\") has been submitted\n",
      "Your job 5482554 (\"sim_Salmonella_houtenae_GCF_002127815\") has been submitted\n",
      "Your job 5482555 (\"sim_Salmonella_houtenae_GCF_002106115\") has been submitted\n",
      "Your job 5482556 (\"sim_Salmonella_indica_GCF_002066855\") has been submitted\n",
      "Your job 5482557 (\"sim_Salmonella_indica_GCF_002035225\") has been submitted\n",
      "Your job 5482558 (\"sim_Salmonella_indica_GCF_002066875\") has been submitted\n",
      "Your job 5482559 (\"sim_Salmonella_indica_GCF_002266165\") has been submitted\n",
      "Your job 5482560 (\"sim_Salmonella_salamae_GCF_002036105\") has been submitted\n",
      "Your job 5482561 (\"sim_Salmonella_salamae_GCF_900478195\") has been submitted\n",
      "Your job 5482562 (\"sim_Salmonella_salamae_GCF_003994675\") has been submitted\n",
      "Your job 5482563 (\"sim_Salmonella_salamae_GCF_002036155\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/salmonella/cleaned_genomes/Salmonella_enterica_GCF_001159405_formatted.fas path=results/phyluce/salmonella/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482564 (\"map_Salmonella_bongori_GCF_000252995\") has been submitted\n",
      "Your job 5482565 (\"map_Salmonella_bongori_GCF_002035475\") has been submitted\n",
      "Your job 5482566 (\"map_Salmonella_bongori_GCA_003497195\") has been submitted\n",
      "Your job 5482567 (\"map_Salmonella_bongori_GCF_902387725\") has been submitted\n",
      "Your job 5482568 (\"map_Salmonella_bongori_GCA_003452365\") has been submitted\n",
      "Your job 5482569 (\"map_Salmonella_bongori_GCF_900635485\") has been submitted\n",
      "Your job 5482570 (\"map_Salmonella_bongori_GCF_002211925\") has been submitted\n",
      "Your job 5482571 (\"map_Salmonella_bongori_GCF_000709535\") has been submitted\n",
      "Your job 5482572 (\"map_Salmonella_bongori_GCA_003522735\") has been submitted\n",
      "Your job 5482573 (\"map_Salmonella_bongori_GCF_002035285\") has been submitted\n",
      "Your job 5482574 (\"map_Salmonella_arizonae_GCF_001951595\") has been submitted\n",
      "Your job 5482575 (\"map_Salmonella_arizonae_GCF_001951445\") has been submitted\n",
      "Your job 5482576 (\"map_Salmonella_arizonae_GCF_000018625\") has been submitted\n",
      "Your job 5482577 (\"map_Salmonella_arizonae_GCF_001970905\") has been submitted\n",
      "Your job 5482578 (\"map_Salmonella_diarizonae_GCF_002241435\") has been submitted\n",
      "Your job 5482579 (\"map_Salmonella_diarizonae_GCF_001832255\") has been submitted\n",
      "Your job 5482580 (\"map_Salmonella_diarizonae_GCF_002063885\") has been submitted\n",
      "Your job 5482581 (\"map_Salmonella_diarizonae_GCF_002035715\") has been submitted\n",
      "Your job 5482582 (\"map_Salmonella_enterica_GCF_009208155\") has been submitted\n",
      "Your job 5482583 (\"map_Salmonella_enterica_GCF_005938475\") has been submitted\n",
      "Your job 5482584 (\"map_Salmonella_enterica_GCF_002064535\") has been submitted\n",
      "Your job 5482585 (\"map_Salmonella_houtenae_GCF_002047425\") has been submitted\n",
      "Your job 5482586 (\"map_Salmonella_houtenae_GCF_002045905\") has been submitted\n",
      "Your job 5482587 (\"map_Salmonella_houtenae_GCF_002127815\") has been submitted\n",
      "Your job 5482588 (\"map_Salmonella_houtenae_GCF_002106115\") has been submitted\n",
      "Your job 5482589 (\"map_Salmonella_indica_GCF_002066855\") has been submitted\n",
      "Your job 5482590 (\"map_Salmonella_indica_GCF_002035225\") has been submitted\n",
      "Your job 5482591 (\"map_Salmonella_indica_GCF_002066875\") has been submitted\n",
      "Your job 5482592 (\"map_Salmonella_indica_GCF_002266165\") has been submitted\n",
      "Your job 5482593 (\"map_Salmonella_salamae_GCF_002036105\") has been submitted\n",
      "Your job 5482594 (\"map_Salmonella_salamae_GCF_900478195\") has been submitted\n",
      "Your job 5482595 (\"map_Salmonella_salamae_GCF_003994675\") has been submitted\n",
      "Your job 5482596 (\"map_Salmonella_salamae_GCF_002036155\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5482597 (\"merge_Salmonella_bongori_GCF_000252995\") has been submitted\n",
      "Your job 5482598 (\"merge_Salmonella_bongori_GCF_002035475\") has been submitted\n",
      "Your job 5482599 (\"merge_Salmonella_bongori_GCA_003497195\") has been submitted\n",
      "Your job 5482600 (\"merge_Salmonella_bongori_GCF_902387725\") has been submitted\n",
      "Your job 5482601 (\"merge_Salmonella_bongori_GCA_003452365\") has been submitted\n",
      "Your job 5482602 (\"merge_Salmonella_bongori_GCF_900635485\") has been submitted\n",
      "Your job 5482603 (\"merge_Salmonella_bongori_GCF_002211925\") has been submitted\n",
      "Your job 5482604 (\"merge_Salmonella_bongori_GCF_000709535\") has been submitted\n",
      "Your job 5482605 (\"merge_Salmonella_bongori_GCA_003522735\") has been submitted\n",
      "Your job 5482606 (\"merge_Salmonella_bongori_GCF_002035285\") has been submitted\n",
      "Your job 5482607 (\"merge_Salmonella_arizonae_GCF_001951595\") has been submitted\n",
      "Your job 5482608 (\"merge_Salmonella_arizonae_GCF_001951445\") has been submitted\n",
      "Your job 5482609 (\"merge_Salmonella_arizonae_GCF_000018625\") has been submitted\n",
      "Your job 5482610 (\"merge_Salmonella_arizonae_GCF_001970905\") has been submitted\n",
      "Your job 5482611 (\"merge_Salmonella_diarizonae_GCF_002241435\") has been submitted\n",
      "Your job 5482612 (\"merge_Salmonella_diarizonae_GCF_001832255\") has been submitted\n",
      "Your job 5482613 (\"merge_Salmonella_diarizonae_GCF_002063885\") has been submitted\n",
      "Your job 5482614 (\"merge_Salmonella_diarizonae_GCF_002035715\") has been submitted\n",
      "Your job 5482615 (\"merge_Salmonella_enterica_GCF_009208155\") has been submitted\n",
      "Your job 5482616 (\"merge_Salmonella_enterica_GCF_005938475\") has been submitted\n",
      "Your job 5482617 (\"merge_Salmonella_enterica_GCF_002064535\") has been submitted\n",
      "Your job 5482618 (\"merge_Salmonella_houtenae_GCF_002047425\") has been submitted\n",
      "Your job 5482619 (\"merge_Salmonella_houtenae_GCF_002045905\") has been submitted\n",
      "Your job 5482620 (\"merge_Salmonella_houtenae_GCF_002127815\") has been submitted\n",
      "Your job 5482621 (\"merge_Salmonella_houtenae_GCF_002106115\") has been submitted\n",
      "Your job 5482622 (\"merge_Salmonella_indica_GCF_002066855\") has been submitted\n",
      "Your job 5482623 (\"merge_Salmonella_indica_GCF_002035225\") has been submitted\n",
      "Your job 5482624 (\"merge_Salmonella_indica_GCF_002066875\") has been submitted\n",
      "Your job 5482625 (\"merge_Salmonella_indica_GCF_002266165\") has been submitted\n",
      "Your job 5482626 (\"merge_Salmonella_salamae_GCF_002036105\") has been submitted\n",
      "Your job 5482627 (\"merge_Salmonella_salamae_GCF_900478195\") has been submitted\n",
      "Your job 5482628 (\"merge_Salmonella_salamae_GCF_003994675\") has been submitted\n",
      "Your job 5482629 (\"merge_Salmonella_salamae_GCF_002036155\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 6426 sequences from Salmonella_bongori_GCF_000252995_merged.bed.  Filtered 1734 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4692.\n",
      "Screened 6343 sequences from Salmonella_bongori_GCF_002035475_merged.bed.  Filtered 1645 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4698.\n",
      "Screened 6297 sequences from Salmonella_bongori_GCA_003497195_merged.bed.  Filtered 1677 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4620.\n",
      "Screened 6400 sequences from Salmonella_bongori_GCF_902387725_merged.bed.  Filtered 1682 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4718.\n",
      "Screened 3850 sequences from Salmonella_bongori_GCA_003452365_merged.bed.  Filtered 1017 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2833.\n",
      "Screened 6516 sequences from Salmonella_bongori_GCF_900635485_merged.bed.  Filtered 1765 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4751.\n",
      "Screened 6377 sequences from Salmonella_bongori_GCF_002211925_merged.bed.  Filtered 1670 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4707.\n",
      "Screened 6275 sequences from Salmonella_bongori_GCF_000709535_merged.bed.  Filtered 1662 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4613.\n",
      "Screened 6302 sequences from Salmonella_bongori_GCA_003522735_merged.bed.  Filtered 1648 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4654.\n",
      "Screened 6374 sequences from Salmonella_bongori_GCF_002035285_merged.bed.  Filtered 1681 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4693.\n",
      "Screened 189 sequences from Salmonella_arizonae_GCF_001951595_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 162.\n",
      "Screened 189 sequences from Salmonella_arizonae_GCF_001951445_merged.bed.  Filtered 19 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 170.\n",
      "Screened 2670 sequences from Salmonella_arizonae_GCF_000018625_merged.bed.  Filtered 419 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2251.\n",
      "Screened 2751 sequences from Salmonella_arizonae_GCF_001970905_merged.bed.  Filtered 426 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2325.\n",
      "Screened 1509 sequences from Salmonella_diarizonae_GCF_002241435_merged.bed.  Filtered 220 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1289.\n",
      "Screened 1557 sequences from Salmonella_diarizonae_GCF_001832255_merged.bed.  Filtered 228 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1329.\n",
      "Screened 1781 sequences from Salmonella_diarizonae_GCF_002063885_merged.bed.  Filtered 231 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1550.\n",
      "Screened 474 sequences from Salmonella_diarizonae_GCF_002035715_merged.bed.  Filtered 57 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 417.\n",
      "Screened 284 sequences from Salmonella_enterica_GCF_009208155_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 259.\n",
      "Screened 456 sequences from Salmonella_enterica_GCF_005938475_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 431.\n",
      "Screened 297 sequences from Salmonella_enterica_GCF_002064535_merged.bed.  Filtered 26 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 271.\n",
      "Screened 1789 sequences from Salmonella_houtenae_GCF_002047425_merged.bed.  Filtered 268 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1521.\n",
      "Screened 1886 sequences from Salmonella_houtenae_GCF_002045905_merged.bed.  Filtered 299 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1587.\n",
      "Screened 1908 sequences from Salmonella_houtenae_GCF_002127815_merged.bed.  Filtered 305 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1603.\n",
      "Screened 1834 sequences from Salmonella_houtenae_GCF_002106115_merged.bed.  Filtered 288 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1546.\n",
      "Screened 1368 sequences from Salmonella_indica_GCF_002066855_merged.bed.  Filtered 188 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1180.\n",
      "Screened 1380 sequences from Salmonella_indica_GCF_002035225_merged.bed.  Filtered 191 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1189.\n",
      "Screened 1385 sequences from Salmonella_indica_GCF_002066875_merged.bed.  Filtered 192 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1193.\n",
      "Screened 521 sequences from Salmonella_indica_GCF_002266165_merged.bed.  Filtered 83 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 438.\n",
      "Screened 1349 sequences from Salmonella_salamae_GCF_002036105_merged.bed.  Filtered 207 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1142.\n",
      "Screened 1150 sequences from Salmonella_salamae_GCF_900478195_merged.bed.  Filtered 157 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 993.\n",
      "Screened 1280 sequences from Salmonella_salamae_GCF_003994675_merged.bed.  Filtered 232 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1048.\n",
      "Screened 1262 sequences from Salmonella_salamae_GCF_002036155_merged.bed.  Filtered 192 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1070.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/salmonella/initial_intervals/Salmonella_salamae_GCF_002036155_merged.bed --twobit results/phyluce/salmonella/cleaned_genomes/Salmonella_enterica_GCF_001159405_formatted.2bit --output results/phyluce/salmonella/initial_intervals/Salmonella_salamae_GCF_002036155_stripped.bed;\n",
      "salmonella_bongori_gcf_000252995.....\n",
      "salmonella_bongori_gcf_002035475.....\n",
      "salmonella_bongori_gca_003497195.....\n",
      "salmonella_bongori_gcf_902387725.....\n",
      "salmonella_bongori_gca_003452365...\n",
      "salmonella_bongori_gcf_900635485.....\n",
      "salmonella_bongori_gcf_002211925.....\n",
      "salmonella_bongori_gcf_000709535.....\n",
      "salmonella_bongori_gca_003522735.....\n",
      "salmonella_bongori_gcf_002035285.....\n",
      "salmonella_arizonae_gcf_001951595.\n",
      "salmonella_arizonae_gcf_001951445.\n",
      "salmonella_arizonae_gcf_000018625...\n",
      "salmonella_arizonae_gcf_001970905...\n",
      "salmonella_diarizonae_gcf_002241435..\n",
      "salmonella_diarizonae_gcf_001832255..\n",
      "salmonella_diarizonae_gcf_002063885..\n",
      "salmonella_diarizonae_gcf_002035715.\n",
      "salmonella_enterica_gcf_009208155.\n",
      "salmonella_enterica_gcf_005938475.\n",
      "salmonella_enterica_gcf_002064535.\n",
      "salmonella_houtenae_gcf_002047425..\n",
      "salmonella_houtenae_gcf_002045905..\n",
      "salmonella_houtenae_gcf_002127815..\n",
      "salmonella_houtenae_gcf_002106115..\n",
      "salmonella_indica_gcf_002066855..\n",
      "salmonella_indica_gcf_002035225..\n",
      "salmonella_indica_gcf_002066875..\n",
      "salmonella_indica_gcf_002266165.\n",
      "salmonella_salamae_gcf_002036105..\n",
      "salmonella_salamae_gcf_900478195.\n",
      "salmonella_salamae_gcf_003994675..\n",
      "salmonella_salamae_gcf_002036155..\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/salmonella/initial_intervals/Salmonella_salamae_GCF_002036155_merged.bed --twobit results/phyluce/salmonella/cleaned_genomes/Salmonella_enterica_GCF_001159405_formatted.2bit --output results/phyluce/salmonella/initial_intervals/Salmonella_salamae_GCF_002036155_stripped.bed;\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 0 taxa:\t6,437.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 1 taxa:\t6,437.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 2 taxa:\t6,337.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 3 taxa:\t6,271.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 4 taxa:\t6,202.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 5 taxa:\t6,152.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 6 taxa:\t6,058.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 7 taxa:\t6,005.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 8 taxa:\t5,951.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 9 taxa:\t5,899.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 10 taxa:\t5,862.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 11 taxa:\t5,831.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 12 taxa:\t5,774.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 13 taxa:\t5,722.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 14 taxa:\t5,687.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 15 taxa:\t5,652.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 16 taxa:\t5,620.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 17 taxa:\t5,572.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 18 taxa:\t5,511.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 19 taxa:\t5,457.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 20 taxa:\t5,393.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 21 taxa:\t5,280.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 22 taxa:\t5,082.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 23 taxa:\t5,051.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 24 taxa:\t4,994.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 25 taxa:\t4,861.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 26 taxa:\t4,745.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 27 taxa:\t4,620.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 28 taxa:\t4,477.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 29 taxa:\t4,308.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 30 taxa:\t4,128.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 31 taxa:\t3,861.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 32 taxa:\t3,358.0\n",
      "Loci shared by Salmonella_enterica_GCF_001159405 + 33 taxa:\t1,864.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/salmonella/initial_intervals/salmonella-to-Salmonella_enterica_GCF_001159405.sqlite --base-taxon Salmonella_enterica_GCF_001159405 --output results/phyluce/salmonella/initial_intervals/Salmonella_enterica_GCF_001159405_+33.bed --specific-counts 33;\n",
      "Counter({'salmonella_arizonae_gcf_001970905': 1864, 'salmonella_diarizonae_gcf_002241435': 1864, 'salmonella_arizonae_gcf_001951445': 1864, 'salmonella_bongori_gcf_002035475': 1864, 'salmonella_indica_gcf_002035225': 1864, 'salmonella_salamae_gcf_900478195': 1864, 'salmonella_diarizonae_gcf_001832255': 1864, 'salmonella_enterica_gcf_009208155': 1864, 'salmonella_enterica_gcf_002064535': 1864, 'salmonella_houtenae_gcf_002047425': 1864, 'salmonella_bongori_gca_003522735': 1864, 'salmonella_arizonae_gcf_001951595': 1864, 'salmonella_salamae_gcf_002036105': 1864, 'salmonella_bongori_gca_003497195': 1864, 'salmonella_bongori_gcf_000252995': 1864, 'salmonella_houtenae_gcf_002045905': 1864, 'salmonella_indica_gcf_002066875': 1864, 'salmonella_bongori_gcf_902387725': 1864, 'salmonella_indica_gcf_002266165': 1864, 'salmonella_houtenae_gcf_002106115': 1864, 'salmonella_salamae_gcf_002036155': 1864, 'salmonella_bongori_gcf_002035285': 1864, 'salmonella_bongori_gcf_002211925': 1864, 'salmonella_salamae_gcf_003994675': 1864, 'salmonella_indica_gcf_002066855': 1864, 'salmonella_bongori_gcf_900635485': 1864, 'salmonella_bongori_gca_003452365': 1864, 'salmonella_enterica_gcf_005938475': 1864, 'salmonella_houtenae_gcf_002127815': 1864, 'salmonella_bongori_gcf_000709535': 1864, 'salmonella_diarizonae_gcf_002063885': 1864, 'salmonella_arizonae_gcf_000018625': 1864, 'salmonella_diarizonae_gcf_002035715': 1864})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 33\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/salmonella/initial_intervals/Salmonella_enterica_GCF_001159405_+33.bed --twobit results/phyluce/salmonella/cleaned_genomes/Salmonella_enterica_GCF_001159405_formatted.2bit --buffer-to 160 --output results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33.fasta;\n",
      "Screened 1864 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 1864.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33.fasta --probe-prefix uce_salmonella_ --design salmonella_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 1864\n",
      "Probe Count = 3723\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas --query results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  08:59:35\n",
      "Ended:  Fri Feb 07, 2020  08:59:36\n",
      "Time for execution:  0.0295583804448 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas                        --lastz results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes_vself.lastz                       --probe-prefix=uce_salmonella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 3722 fasta sequences.  Filtered 20 duplicates. Kept 3683.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/salmonella/validate_intervals/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas --scaffoldlist Salmonella_bongori_GCF_000252995 Salmonella_bongori_GCF_002035475 Salmonella_bongori_GCA_003497195 Salmonella_bongori_GCF_902387725 Salmonella_bongori_GCA_003452365 Salmonella_bongori_GCF_900635485 Salmonella_bongori_GCF_002211925 Salmonella_bongori_GCF_000709535 Salmonella_bongori_GCA_003522735 Salmonella_bongori_GCF_002035285 Salmonella_arizonae_GCF_001951595 Salmonella_arizonae_GCF_001951445 Salmonella_arizonae_GCF_000018625 Salmonella_arizonae_GCF_001970905 Salmonella_diarizonae_GCF_002241435 Salmonella_diarizonae_GCF_001832255 Salmonella_diarizonae_GCF_002063885 Salmonella_diarizonae_GCF_002035715 Salmonella_enterica_GCF_009208155 Salmonella_enterica_GCF_005938475 Salmonella_enterica_GCF_002064535 Salmonella_houtenae_GCF_002047425 Salmonella_houtenae_GCF_002045905 Salmonella_houtenae_GCF_002127815 Salmonella_houtenae_GCF_002106115 Salmonella_indica_GCF_002066855 Salmonella_indica_GCF_002035225 Salmonella_indica_GCF_002066875 Salmonella_indica_GCF_002266165 Salmonella_salamae_GCF_002036105 Salmonella_salamae_GCF_900478195 Salmonella_salamae_GCF_003994675 Salmonella_salamae_GCF_002036155 Salmonella_enterica_GCF_001159405 --genome-base-path results/phyluce/salmonella/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/salmonella/validate_intervals/salmonella-to-Salmonella_enterica_GCF_001159405.sqlite --output results/phyluce/salmonella/validate_intervals/lastz/;\n",
      "\n",
      "Running against Salmonella_bongori_GCF_000252995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDNZy5I.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcMIk1o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_000252995.lastz\n",
      "Creating Salmonella_bongori_GCF_000252995 table\n",
      "Inserting data to Salmonella_bongori_GCF_000252995 table\n",
      "\n",
      "Running against Salmonella_bongori_GCF_002035475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpN6pvQx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2RiTah.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_002035475.lastz\n",
      "Creating Salmonella_bongori_GCF_002035475 table\n",
      "Inserting data to Salmonella_bongori_GCF_002035475 table\n",
      "\n",
      "Running against Salmonella_bongori_GCA_003497195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGLqKba.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8PKWBK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCA_003497195.lastz\n",
      "Creating Salmonella_bongori_GCA_003497195 table\n",
      "Inserting data to Salmonella_bongori_GCA_003497195 table\n",
      "\n",
      "Running against Salmonella_bongori_GCF_902387725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpAE_KjE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptsJAg4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_902387725.lastz\n",
      "Creating Salmonella_bongori_GCF_902387725 table\n",
      "Inserting data to Salmonella_bongori_GCF_902387725 table\n",
      "\n",
      "Running against Salmonella_bongori_GCA_003452365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGcqBBt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvhuwSo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCA_003452365.lastz\n",
      "Creating Salmonella_bongori_GCA_003452365 table\n",
      "Inserting data to Salmonella_bongori_GCA_003452365 table\n",
      "\n",
      "Running against Salmonella_bongori_GCF_900635485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTxEiCp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4TstwB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_900635485.lastz\n",
      "Creating Salmonella_bongori_GCF_900635485 table\n",
      "Inserting data to Salmonella_bongori_GCF_900635485 table\n",
      "\n",
      "Running against Salmonella_bongori_GCF_002211925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptcSxuT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwv_3HZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_002211925.lastz\n",
      "Creating Salmonella_bongori_GCF_002211925 table\n",
      "Inserting data to Salmonella_bongori_GCF_002211925 table\n",
      "\n",
      "Running against Salmonella_bongori_GCF_000709535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWVjmWC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOJPo3j.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_000709535.lastz\n",
      "Creating Salmonella_bongori_GCF_000709535 table\n",
      "Inserting data to Salmonella_bongori_GCF_000709535 table\n",
      "\n",
      "Running against Salmonella_bongori_GCA_003522735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyer5Ng.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxSya26.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCA_003522735.lastz\n",
      "Creating Salmonella_bongori_GCA_003522735 table\n",
      "Inserting data to Salmonella_bongori_GCA_003522735 table\n",
      "\n",
      "Running against Salmonella_bongori_GCF_002035285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYMnaG7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQqt10N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_bongori_GCF_002035285.lastz\n",
      "Creating Salmonella_bongori_GCF_002035285 table\n",
      "Inserting data to Salmonella_bongori_GCF_002035285 table\n",
      "\n",
      "Running against Salmonella_arizonae_GCF_001951595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmyj81R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptvDFeH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_arizonae_GCF_001951595.lastz\n",
      "Creating Salmonella_arizonae_GCF_001951595 table\n",
      "Inserting data to Salmonella_arizonae_GCF_001951595 table\n",
      "\n",
      "Running against Salmonella_arizonae_GCF_001951445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHOFWqK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9TH6ol.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_arizonae_GCF_001951445.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Salmonella_arizonae_GCF_001951445 table\n",
      "Inserting data to Salmonella_arizonae_GCF_001951445 table\n",
      "\n",
      "Running against Salmonella_arizonae_GCF_000018625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvLABVB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKRn9oN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_arizonae_GCF_000018625.lastz\n",
      "Creating Salmonella_arizonae_GCF_000018625 table\n",
      "Inserting data to Salmonella_arizonae_GCF_000018625 table\n",
      "\n",
      "Running against Salmonella_arizonae_GCF_001970905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpv2lUlg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphl4NuV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_arizonae_GCF_001970905.lastz\n",
      "Creating Salmonella_arizonae_GCF_001970905 table\n",
      "Inserting data to Salmonella_arizonae_GCF_001970905 table\n",
      "\n",
      "Running against Salmonella_diarizonae_GCF_002241435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpl2YKEM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB114f3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_diarizonae_GCF_002241435.lastz\n",
      "Creating Salmonella_diarizonae_GCF_002241435 table\n",
      "Inserting data to Salmonella_diarizonae_GCF_002241435 table\n",
      "\n",
      "Running against Salmonella_diarizonae_GCF_001832255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYd1wLS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2Qr_wN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_diarizonae_GCF_001832255.lastz\n",
      "Creating Salmonella_diarizonae_GCF_001832255 table\n",
      "Inserting data to Salmonella_diarizonae_GCF_001832255 table\n",
      "\n",
      "Running against Salmonella_diarizonae_GCF_002063885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpP041ba.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1XrRB5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_diarizonae_GCF_002063885.lastz\n",
      "Creating Salmonella_diarizonae_GCF_002063885 table\n",
      "Inserting data to Salmonella_diarizonae_GCF_002063885 table\n",
      "\n",
      "Running against Salmonella_diarizonae_GCF_002035715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpXKO_9M.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWvZE0t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_diarizonae_GCF_002035715.lastz\n",
      "Creating Salmonella_diarizonae_GCF_002035715 table\n",
      "Inserting data to Salmonella_diarizonae_GCF_002035715 table\n",
      "\n",
      "Running against Salmonella_enterica_GCF_009208155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMWS7NM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfEC48R.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_enterica_GCF_009208155.lastz\n",
      "Creating Salmonella_enterica_GCF_009208155 table\n",
      "Inserting data to Salmonella_enterica_GCF_009208155 table\n",
      "\n",
      "Running against Salmonella_enterica_GCF_005938475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYCFbh4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWfoRMm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_enterica_GCF_005938475.lastz\n",
      "Creating Salmonella_enterica_GCF_005938475 table\n",
      "Inserting data to Salmonella_enterica_GCF_005938475 table\n",
      "\n",
      "Running against Salmonella_enterica_GCF_002064535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHMdLq9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOomoOF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_enterica_GCF_002064535.lastz\n",
      "Creating Salmonella_enterica_GCF_002064535 table\n",
      "Inserting data to Salmonella_enterica_GCF_002064535 table\n",
      "\n",
      "Running against Salmonella_houtenae_GCF_002047425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3oORDR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaRrEBO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_houtenae_GCF_002047425.lastz\n",
      "Creating Salmonella_houtenae_GCF_002047425 table\n",
      "Inserting data to Salmonella_houtenae_GCF_002047425 table\n",
      "\n",
      "Running against Salmonella_houtenae_GCF_002045905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp58weyJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpO_RK2b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_houtenae_GCF_002045905.lastz\n",
      "Creating Salmonella_houtenae_GCF_002045905 table\n",
      "Inserting data to Salmonella_houtenae_GCF_002045905 table\n",
      "\n",
      "Running against Salmonella_houtenae_GCF_002127815.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphbJ7a_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5cLSA4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_houtenae_GCF_002127815.lastz\n",
      "Creating Salmonella_houtenae_GCF_002127815 table\n",
      "Inserting data to Salmonella_houtenae_GCF_002127815 table\n",
      "\n",
      "Running against Salmonella_houtenae_GCF_002106115.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbdWobs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzAFbeB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_houtenae_GCF_002106115.lastz\n",
      "Creating Salmonella_houtenae_GCF_002106115 table\n",
      "Inserting data to Salmonella_houtenae_GCF_002106115 table\n",
      "\n",
      "Running against Salmonella_indica_GCF_002066855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBzrK30.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkYHqI0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_indica_GCF_002066855.lastz\n",
      "Creating Salmonella_indica_GCF_002066855 table\n",
      "Inserting data to Salmonella_indica_GCF_002066855 table\n",
      "\n",
      "Running against Salmonella_indica_GCF_002035225.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpVKHyyd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHbMOZa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_indica_GCF_002035225.lastz\n",
      "Creating Salmonella_indica_GCF_002035225 table\n",
      "Inserting data to Salmonella_indica_GCF_002035225 table\n",
      "\n",
      "Running against Salmonella_indica_GCF_002066875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjl6xpV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNzpmIZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_indica_GCF_002066875.lastz\n",
      "Creating Salmonella_indica_GCF_002066875 table\n",
      "Inserting data to Salmonella_indica_GCF_002066875 table\n",
      "\n",
      "Running against Salmonella_indica_GCF_002266165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpapy4Bv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2cMlDe.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_indica_GCF_002266165.lastz\n",
      "Creating Salmonella_indica_GCF_002266165 table\n",
      "Inserting data to Salmonella_indica_GCF_002266165 table\n",
      "\n",
      "Running against Salmonella_salamae_GCF_002036105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsWoWaz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuFFTRX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_salamae_GCF_002036105.lastz\n",
      "Creating Salmonella_salamae_GCF_002036105 table\n",
      "Inserting data to Salmonella_salamae_GCF_002036105 table\n",
      "\n",
      "Running against Salmonella_salamae_GCF_900478195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpX9qybw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPgKxsM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_salamae_GCF_900478195.lastz\n",
      "Creating Salmonella_salamae_GCF_900478195 table\n",
      "Inserting data to Salmonella_salamae_GCF_900478195 table\n",
      "\n",
      "Running against Salmonella_salamae_GCF_003994675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpW0blf4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcrFs9b.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_salamae_GCF_003994675.lastz\n",
      "Creating Salmonella_salamae_GCF_003994675 table\n",
      "Inserting data to Salmonella_salamae_GCF_003994675 table\n",
      "\n",
      "Running against Salmonella_salamae_GCF_002036155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpj1KGMr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLfAhiS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_salamae_GCF_002036155.lastz\n",
      "Creating Salmonella_salamae_GCF_002036155 table\n",
      "Inserting data to Salmonella_salamae_GCF_002036155 table\n",
      "\n",
      "Running against Salmonella_enterica_GCF_001159405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdkwJL1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGCI6ci.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/salmonella/validate_intervals/lastz/Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_Salmonella_enterica_GCF_001159405.lastz\n",
      "Creating Salmonella_enterica_GCF_001159405 table\n",
      "Inserting data to Salmonella_enterica_GCF_001159405 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/salmonella/extract_probes_from_group/salmonella_genome.conf --lastz results/phyluce/salmonella/validate_intervals/lastz --probes 120 --probe-prefix uce_salmonella_ --name-pattern \"Salmonella_enterica_GCF_001159405_+33_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/salmonella/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 09:03:33,419 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 09:03:33,423 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_000252995 genome ------\n",
      "2020-02-07 09:03:33,424 - Phyluce - INFO - Reading Salmonella_bongori_GCF_000252995 genome\n",
      "2020-02-07 09:03:39,996 - Phyluce - INFO - Salmonella_bongori_GCF_000252995: 1864 uces, 0 dupes, 1864 non-dupes, 79 orient drop, 38 length drop, 1747 written\n",
      "2020-02-07 09:03:39,996 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_002035475 genome ------\n",
      "2020-02-07 09:03:39,997 - Phyluce - INFO - Reading Salmonella_bongori_GCF_002035475 genome\n",
      "2020-02-07 09:03:46,485 - Phyluce - INFO - Salmonella_bongori_GCF_002035475: 1863 uces, 108 dupes, 1755 non-dupes, 3 orient drop, 6 length drop, 1746 written\n",
      "2020-02-07 09:03:46,486 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCA_003497195 genome ------\n",
      "2020-02-07 09:03:46,487 - Phyluce - INFO - Reading Salmonella_bongori_GCA_003497195 genome\n",
      "2020-02-07 09:03:53,044 - Phyluce - INFO - Salmonella_bongori_GCA_003497195: 1861 uces, 109 dupes, 1752 non-dupes, 1 orient drop, 5 length drop, 1745 written\n",
      "2020-02-07 09:03:53,044 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_902387725 genome ------\n",
      "2020-02-07 09:03:53,059 - Phyluce - INFO - Reading Salmonella_bongori_GCF_902387725 genome\n",
      "2020-02-07 09:03:59,687 - Phyluce - INFO - Salmonella_bongori_GCF_902387725: 1864 uces, 111 dupes, 1753 non-dupes, 4 orient drop, 4 length drop, 1745 written\n",
      "2020-02-07 09:03:59,687 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCA_003452365 genome ------\n",
      "2020-02-07 09:03:59,688 - Phyluce - INFO - Reading Salmonella_bongori_GCA_003452365 genome\n",
      "2020-02-07 09:04:05,849 - Phyluce - INFO - Salmonella_bongori_GCA_003452365: 1784 uces, 74 dupes, 1710 non-dupes, 0 orient drop, 1 length drop, 1705 written\n",
      "2020-02-07 09:04:05,850 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_900635485 genome ------\n",
      "2020-02-07 09:04:05,850 - Phyluce - INFO - Reading Salmonella_bongori_GCF_900635485 genome\n",
      "2020-02-07 09:04:12,502 - Phyluce - INFO - Salmonella_bongori_GCF_900635485: 1864 uces, 0 dupes, 1864 non-dupes, 77 orient drop, 38 length drop, 1749 written\n",
      "2020-02-07 09:04:12,502 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_002211925 genome ------\n",
      "2020-02-07 09:04:12,503 - Phyluce - INFO - Reading Salmonella_bongori_GCF_002211925 genome\n",
      "2020-02-07 09:04:19,189 - Phyluce - INFO - Salmonella_bongori_GCF_002211925: 1864 uces, 0 dupes, 1864 non-dupes, 80 orient drop, 48 length drop, 1736 written\n",
      "2020-02-07 09:04:19,189 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_000709535 genome ------\n",
      "2020-02-07 09:04:19,190 - Phyluce - INFO - Reading Salmonella_bongori_GCF_000709535 genome\n",
      "2020-02-07 09:04:25,828 - Phyluce - INFO - Salmonella_bongori_GCF_000709535: 1864 uces, 0 dupes, 1864 non-dupes, 77 orient drop, 40 length drop, 1747 written\n",
      "2020-02-07 09:04:25,828 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCA_003522735 genome ------\n",
      "2020-02-07 09:04:25,829 - Phyluce - INFO - Reading Salmonella_bongori_GCA_003522735 genome\n",
      "2020-02-07 09:04:32,493 - Phyluce - INFO - Salmonella_bongori_GCA_003522735: 1863 uces, 99 dupes, 1764 non-dupes, 12 orient drop, 4 length drop, 1748 written\n",
      "2020-02-07 09:04:32,493 - Phyluce - INFO - ------- Working on Salmonella_bongori_GCF_002035285 genome ------\n",
      "2020-02-07 09:04:32,504 - Phyluce - INFO - Reading Salmonella_bongori_GCF_002035285 genome\n",
      "2020-02-07 09:04:39,112 - Phyluce - INFO - Salmonella_bongori_GCF_002035285: 1864 uces, 112 dupes, 1752 non-dupes, 1 orient drop, 2 length drop, 1749 written\n",
      "2020-02-07 09:04:39,113 - Phyluce - INFO - ------ Working on Salmonella_arizonae_GCF_001951595 genome ------\n",
      "2020-02-07 09:04:39,113 - Phyluce - INFO - Reading Salmonella_arizonae_GCF_001951595 genome\n",
      "2020-02-07 09:04:45,803 - Phyluce - INFO - Salmonella_arizonae_GCF_001951595: 1864 uces, 111 dupes, 1753 non-dupes, 10 orient drop, 7 length drop, 1736 written\n",
      "2020-02-07 09:04:45,803 - Phyluce - INFO - ------ Working on Salmonella_arizonae_GCF_001951445 genome ------\n",
      "2020-02-07 09:04:45,804 - Phyluce - INFO - Reading Salmonella_arizonae_GCF_001951445 genome\n",
      "2020-02-07 09:04:52,487 - Phyluce - INFO - Salmonella_arizonae_GCF_001951445: 1863 uces, 119 dupes, 1744 non-dupes, 2 orient drop, 8 length drop, 1734 written\n",
      "2020-02-07 09:04:52,487 - Phyluce - INFO - ------ Working on Salmonella_arizonae_GCF_000018625 genome ------\n",
      "2020-02-07 09:04:52,488 - Phyluce - INFO - Reading Salmonella_arizonae_GCF_000018625 genome\n",
      "2020-02-07 09:04:59,100 - Phyluce - INFO - Salmonella_arizonae_GCF_000018625: 1854 uces, 0 dupes, 1854 non-dupes, 68 orient drop, 49 length drop, 1737 written\n",
      "2020-02-07 09:04:59,100 - Phyluce - INFO - ------ Working on Salmonella_arizonae_GCF_001970905 genome ------\n",
      "2020-02-07 09:04:59,110 - Phyluce - INFO - Reading Salmonella_arizonae_GCF_001970905 genome\n",
      "2020-02-07 09:05:05,664 - Phyluce - INFO - Salmonella_arizonae_GCF_001970905: 1853 uces, 112 dupes, 1741 non-dupes, 0 orient drop, 6 length drop, 1735 written\n",
      "2020-02-07 09:05:05,664 - Phyluce - INFO - ----- Working on Salmonella_diarizonae_GCF_002241435 genome -----\n",
      "2020-02-07 09:05:05,665 - Phyluce - INFO - Reading Salmonella_diarizonae_GCF_002241435 genome\n",
      "2020-02-07 09:05:12,354 - Phyluce - INFO - Salmonella_diarizonae_GCF_002241435: 1861 uces, 120 dupes, 1741 non-dupes, 3 orient drop, 6 length drop, 1732 written\n",
      "2020-02-07 09:05:12,355 - Phyluce - INFO - ----- Working on Salmonella_diarizonae_GCF_001832255 genome -----\n",
      "2020-02-07 09:05:12,356 - Phyluce - INFO - Reading Salmonella_diarizonae_GCF_001832255 genome\n",
      "2020-02-07 09:05:19,046 - Phyluce - INFO - Salmonella_diarizonae_GCF_001832255: 1861 uces, 118 dupes, 1743 non-dupes, 4 orient drop, 7 length drop, 1732 written\n",
      "2020-02-07 09:05:19,047 - Phyluce - INFO - ----- Working on Salmonella_diarizonae_GCF_002063885 genome -----\n",
      "2020-02-07 09:05:19,064 - Phyluce - INFO - Reading Salmonella_diarizonae_GCF_002063885 genome\n",
      "2020-02-07 09:05:25,687 - Phyluce - INFO - Salmonella_diarizonae_GCF_002063885: 1845 uces, 123 dupes, 1722 non-dupes, 0 orient drop, 3 length drop, 1719 written\n",
      "2020-02-07 09:05:25,687 - Phyluce - INFO - ----- Working on Salmonella_diarizonae_GCF_002035715 genome -----\n",
      "2020-02-07 09:05:25,688 - Phyluce - INFO - Reading Salmonella_diarizonae_GCF_002035715 genome\n",
      "2020-02-07 09:05:32,366 - Phyluce - INFO - Salmonella_diarizonae_GCF_002035715: 1864 uces, 120 dupes, 1744 non-dupes, 2 orient drop, 4 length drop, 1738 written\n",
      "2020-02-07 09:05:32,366 - Phyluce - INFO - ------ Working on Salmonella_enterica_GCF_009208155 genome ------\n",
      "2020-02-07 09:05:32,367 - Phyluce - INFO - Reading Salmonella_enterica_GCF_009208155 genome\n",
      "2020-02-07 09:05:39,073 - Phyluce - INFO - Salmonella_enterica_GCF_009208155: 1864 uces, 113 dupes, 1751 non-dupes, 11 orient drop, 8 length drop, 1732 written\n",
      "2020-02-07 09:05:39,074 - Phyluce - INFO - ------ Working on Salmonella_enterica_GCF_005938475 genome ------\n",
      "2020-02-07 09:05:39,075 - Phyluce - INFO - Reading Salmonella_enterica_GCF_005938475 genome\n",
      "2020-02-07 09:05:45,620 - Phyluce - INFO - Salmonella_enterica_GCF_005938475: 1856 uces, 119 dupes, 1737 non-dupes, 0 orient drop, 3 length drop, 1733 written\n",
      "2020-02-07 09:05:45,620 - Phyluce - INFO - ------ Working on Salmonella_enterica_GCF_002064535 genome ------\n",
      "2020-02-07 09:05:45,621 - Phyluce - INFO - Reading Salmonella_enterica_GCF_002064535 genome\n",
      "2020-02-07 09:05:52,335 - Phyluce - INFO - Salmonella_enterica_GCF_002064535: 1864 uces, 120 dupes, 1744 non-dupes, 3 orient drop, 7 length drop, 1734 written\n",
      "2020-02-07 09:05:52,335 - Phyluce - INFO - ------ Working on Salmonella_houtenae_GCF_002047425 genome ------\n",
      "2020-02-07 09:05:52,343 - Phyluce - INFO - Reading Salmonella_houtenae_GCF_002047425 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 09:05:58,968 - Phyluce - INFO - Salmonella_houtenae_GCF_002047425: 1858 uces, 116 dupes, 1742 non-dupes, 3 orient drop, 7 length drop, 1732 written\n",
      "2020-02-07 09:05:58,968 - Phyluce - INFO - ------ Working on Salmonella_houtenae_GCF_002045905 genome ------\n",
      "2020-02-07 09:05:58,969 - Phyluce - INFO - Reading Salmonella_houtenae_GCF_002045905 genome\n",
      "2020-02-07 09:06:05,655 - Phyluce - INFO - Salmonella_houtenae_GCF_002045905: 1858 uces, 124 dupes, 1734 non-dupes, 2 orient drop, 6 length drop, 1726 written\n",
      "2020-02-07 09:06:05,655 - Phyluce - INFO - ------ Working on Salmonella_houtenae_GCF_002127815 genome ------\n",
      "2020-02-07 09:06:05,656 - Phyluce - INFO - Reading Salmonella_houtenae_GCF_002127815 genome\n",
      "2020-02-07 09:06:12,268 - Phyluce - INFO - Salmonella_houtenae_GCF_002127815: 1857 uces, 113 dupes, 1744 non-dupes, 2 orient drop, 8 length drop, 1734 written\n",
      "2020-02-07 09:06:12,268 - Phyluce - INFO - ------ Working on Salmonella_houtenae_GCF_002106115 genome ------\n",
      "2020-02-07 09:06:12,269 - Phyluce - INFO - Reading Salmonella_houtenae_GCF_002106115 genome\n",
      "2020-02-07 09:06:18,949 - Phyluce - INFO - Salmonella_houtenae_GCF_002106115: 1858 uces, 116 dupes, 1742 non-dupes, 4 orient drop, 7 length drop, 1731 written\n",
      "2020-02-07 09:06:18,950 - Phyluce - INFO - ------- Working on Salmonella_indica_GCF_002066855 genome -------\n",
      "2020-02-07 09:06:18,963 - Phyluce - INFO - Reading Salmonella_indica_GCF_002066855 genome\n",
      "2020-02-07 09:06:25,576 - Phyluce - INFO - Salmonella_indica_GCF_002066855: 1860 uces, 106 dupes, 1754 non-dupes, 14 orient drop, 5 length drop, 1735 written\n",
      "2020-02-07 09:06:25,576 - Phyluce - INFO - ------- Working on Salmonella_indica_GCF_002035225 genome -------\n",
      "2020-02-07 09:06:25,577 - Phyluce - INFO - Reading Salmonella_indica_GCF_002035225 genome\n",
      "2020-02-07 09:06:32,203 - Phyluce - INFO - Salmonella_indica_GCF_002035225: 1860 uces, 110 dupes, 1750 non-dupes, 11 orient drop, 6 length drop, 1733 written\n",
      "2020-02-07 09:06:32,203 - Phyluce - INFO - ------- Working on Salmonella_indica_GCF_002066875 genome -------\n",
      "2020-02-07 09:06:32,203 - Phyluce - INFO - Reading Salmonella_indica_GCF_002066875 genome\n",
      "2020-02-07 09:06:38,774 - Phyluce - INFO - Salmonella_indica_GCF_002066875: 1861 uces, 114 dupes, 1747 non-dupes, 4 orient drop, 6 length drop, 1737 written\n",
      "2020-02-07 09:06:38,774 - Phyluce - INFO - ------- Working on Salmonella_indica_GCF_002266165 genome -------\n",
      "2020-02-07 09:06:38,781 - Phyluce - INFO - Reading Salmonella_indica_GCF_002266165 genome\n",
      "2020-02-07 09:06:45,436 - Phyluce - INFO - Salmonella_indica_GCF_002266165: 1864 uces, 120 dupes, 1744 non-dupes, 2 orient drop, 6 length drop, 1736 written\n",
      "2020-02-07 09:06:45,436 - Phyluce - INFO - ------- Working on Salmonella_salamae_GCF_002036105 genome ------\n",
      "2020-02-07 09:06:45,437 - Phyluce - INFO - Reading Salmonella_salamae_GCF_002036105 genome\n",
      "2020-02-07 09:06:52,040 - Phyluce - INFO - Salmonella_salamae_GCF_002036105: 1862 uces, 116 dupes, 1746 non-dupes, 2 orient drop, 6 length drop, 1738 written\n",
      "2020-02-07 09:06:52,040 - Phyluce - INFO - ------- Working on Salmonella_salamae_GCF_900478195 genome ------\n",
      "2020-02-07 09:06:52,041 - Phyluce - INFO - Reading Salmonella_salamae_GCF_900478195 genome\n",
      "2020-02-07 09:06:58,686 - Phyluce - INFO - Salmonella_salamae_GCF_900478195: 1860 uces, 0 dupes, 1860 non-dupes, 83 orient drop, 47 length drop, 1730 written\n",
      "2020-02-07 09:06:58,686 - Phyluce - INFO - ------- Working on Salmonella_salamae_GCF_003994675 genome ------\n",
      "2020-02-07 09:06:58,696 - Phyluce - INFO - Reading Salmonella_salamae_GCF_003994675 genome\n",
      "2020-02-07 09:07:05,256 - Phyluce - INFO - Salmonella_salamae_GCF_003994675: 1863 uces, 0 dupes, 1863 non-dupes, 86 orient drop, 45 length drop, 1732 written\n",
      "2020-02-07 09:07:05,256 - Phyluce - INFO - ------- Working on Salmonella_salamae_GCF_002036155 genome ------\n",
      "2020-02-07 09:07:05,257 - Phyluce - INFO - Reading Salmonella_salamae_GCF_002036155 genome\n",
      "2020-02-07 09:07:11,824 - Phyluce - INFO - Salmonella_salamae_GCF_002036155: 1862 uces, 121 dupes, 1741 non-dupes, 2 orient drop, 7 length drop, 1732 written\n",
      "2020-02-07 09:07:11,824 - Phyluce - INFO - ------ Working on Salmonella_enterica_GCF_001159405 genome ------\n",
      "2020-02-07 09:07:11,825 - Phyluce - INFO - Reading Salmonella_enterica_GCF_001159405 genome\n",
      "2020-02-07 09:07:18,409 - Phyluce - INFO - Salmonella_enterica_GCF_001159405: 1864 uces, 116 dupes, 1748 non-dupes, 6 orient drop, 9 length drop, 1733 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/salmonella/extract_probes_from_group/probe_fasta --output results/phyluce/salmonella/extract_probes_from_group/multifastas.sqlite --base-taxon Salmonella_enterica_GCF_001159405;\n",
      "salmonella_bongori_gcf_000252995..\n",
      "salmonella_bongori_gcf_002035475..\n",
      "salmonella_bongori_gca_003497195..\n",
      "salmonella_bongori_gcf_902387725..\n",
      "salmonella_bongori_gca_003452365..\n",
      "salmonella_bongori_gcf_900635485..\n",
      "salmonella_bongori_gcf_002211925..\n",
      "salmonella_bongori_gcf_000709535..\n",
      "salmonella_bongori_gca_003522735..\n",
      "salmonella_bongori_gcf_002035285..\n",
      "salmonella_arizonae_gcf_001951595..\n",
      "salmonella_arizonae_gcf_001951445..\n",
      "salmonella_arizonae_gcf_000018625..\n",
      "salmonella_arizonae_gcf_001970905..\n",
      "salmonella_diarizonae_gcf_002241435..\n",
      "salmonella_diarizonae_gcf_001832255..\n",
      "salmonella_diarizonae_gcf_002063885..\n",
      "salmonella_diarizonae_gcf_002035715..\n",
      "salmonella_enterica_gcf_009208155..\n",
      "salmonella_enterica_gcf_005938475..\n",
      "salmonella_enterica_gcf_002064535..\n",
      "salmonella_houtenae_gcf_002047425..\n",
      "salmonella_houtenae_gcf_002045905..\n",
      "salmonella_houtenae_gcf_002127815..\n",
      "salmonella_houtenae_gcf_002106115..\n",
      "salmonella_indica_gcf_002066855..\n",
      "salmonella_indica_gcf_002035225..\n",
      "salmonella_indica_gcf_002066875..\n",
      "salmonella_indica_gcf_002266165..\n",
      "salmonella_salamae_gcf_002036105..\n",
      "salmonella_salamae_gcf_900478195..\n",
      "salmonella_salamae_gcf_003994675..\n",
      "salmonella_salamae_gcf_002036155..\n",
      "salmonella_enterica_gcf_001159405..\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/salmonella/extract_probes_from_group/multifastas.sqlite --base-taxon Salmonella_enterica_GCF_001159405;\n",
      "Loci shared by 0 taxa:\t1,813.0\n",
      "Loci shared by 1 taxa:\t1,813.0\n",
      "Loci shared by 2 taxa:\t1,787.0\n",
      "Loci shared by 3 taxa:\t1,781.0\n",
      "Loci shared by 4 taxa:\t1,780.0\n",
      "Loci shared by 5 taxa:\t1,778.0\n",
      "Loci shared by 6 taxa:\t1,775.0\n",
      "Loci shared by 7 taxa:\t1,773.0\n",
      "Loci shared by 8 taxa:\t1,771.0\n",
      "Loci shared by 9 taxa:\t1,769.0\n",
      "Loci shared by 10 taxa:\t1,765.0\n",
      "Loci shared by 11 taxa:\t1,762.0\n",
      "Loci shared by 12 taxa:\t1,760.0\n",
      "Loci shared by 13 taxa:\t1,754.0\n",
      "Loci shared by 14 taxa:\t1,752.0\n",
      "Loci shared by 15 taxa:\t1,750.0\n",
      "Loci shared by 16 taxa:\t1,750.0\n",
      "Loci shared by 17 taxa:\t1,748.0\n",
      "Loci shared by 18 taxa:\t1,747.0\n",
      "Loci shared by 19 taxa:\t1,746.0\n",
      "Loci shared by 20 taxa:\t1,745.0\n",
      "Loci shared by 21 taxa:\t1,737.0\n",
      "Loci shared by 22 taxa:\t1,734.0\n",
      "Loci shared by 23 taxa:\t1,731.0\n",
      "Loci shared by 24 taxa:\t1,722.0\n",
      "Loci shared by 25 taxa:\t1,719.0\n",
      "Loci shared by 26 taxa:\t1,710.0\n",
      "Loci shared by 27 taxa:\t1,707.0\n",
      "Loci shared by 28 taxa:\t1,702.0\n",
      "Loci shared by 29 taxa:\t1,699.0\n",
      "Loci shared by 30 taxa:\t1,689.0\n",
      "Loci shared by 31 taxa:\t1,683.0\n",
      "Loci shared by 32 taxa:\t1,673.0\n",
      "Loci shared by 33 taxa:\t1,661.0\n",
      "Loci shared by 34 taxa:\t1,535.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/salmonella/extract_probes_from_group/multifastas.sqlite  --base-taxon Salmonella_enterica_GCF_001159405 --output results/phyluce/salmonella/extract_probes_from_group/Salmonella_enterica_GCF_001159405+33-back-to-34.conf --specific-counts 34;\n",
      "Counter({'salmonella_arizonae_gcf_001970905': 1535, 'salmonella_diarizonae_gcf_002241435': 1535, 'salmonella_arizonae_gcf_001951445': 1535, 'salmonella_bongori_gcf_002035475': 1535, 'salmonella_indica_gcf_002035225': 1535, 'salmonella_houtenae_gcf_002045905': 1535, 'salmonella_diarizonae_gcf_001832255': 1535, 'salmonella_enterica_gcf_009208155': 1535, 'salmonella_enterica_gcf_002064535': 1535, 'salmonella_houtenae_gcf_002047425': 1535, 'salmonella_bongori_gca_003522735': 1535, 'salmonella_arizonae_gcf_001951595': 1535, 'salmonella_salamae_gcf_002036105': 1535, 'salmonella_bongori_gca_003497195': 1535, 'salmonella_bongori_gcf_000252995': 1535, 'salmonella_salamae_gcf_900478195': 1535, 'salmonella_indica_gcf_002066875': 1535, 'salmonella_bongori_gcf_902387725': 1535, 'salmonella_indica_gcf_002266165': 1535, 'salmonella_houtenae_gcf_002106115': 1535, 'salmonella_salamae_gcf_002036155': 1535, 'salmonella_bongori_gcf_002035285': 1535, 'salmonella_bongori_gcf_002211925': 1535, 'salmonella_salamae_gcf_003994675': 1535, 'salmonella_indica_gcf_002066855': 1535, 'salmonella_bongori_gcf_900635485': 1535, 'salmonella_bongori_gca_003452365': 1535, 'salmonella_enterica_gcf_005938475': 1535, 'salmonella_enterica_gcf_001159405': 1535, 'salmonella_houtenae_gcf_002127815': 1535, 'salmonella_bongori_gcf_000709535': 1535, 'salmonella_diarizonae_gcf_002063885': 1535, 'salmonella_arizonae_gcf_000018625': 1535, 'salmonella_diarizonae_gcf_002035715': 1535})\n",
      "Total loci = 1535\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(34)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/salmonella/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/salmonella/extract_probes_from_group/Salmonella_enterica_GCF_001159405+33-back-to-34.conf --probe-prefix uce_salmonella_ --designer rnplattii --design salmonella_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGNNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 1535\n",
      "Probe Count = 103901\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list.fasta --query results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  09:09:20\n",
      "Ended:  Fri Feb 07, 2020  09:18:57\n",
      "Time for execution:  9.62158638239 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list.fasta --lastz results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_salmonella_;\n",
      "Parsing lastz file...\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/salmonella/final_probe_design/salmonella_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 13:11:16 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 103901\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 8312080\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 22M\n",
      "Buffer          : 4 X 13M = 53M\n",
      "Table           : 2 X 18M = 36M\n",
      "Miscellaneous   : 5M\n",
      "Total           : 117M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3788359\n",
      "Max number of word counting entries: 85275978\n",
      "\n",
      "# comparing sequences from          0  to      17316\n",
      "..........    10000  finished       1400  clusters\n",
      ".......---------- new table with     2396 representatives\n",
      "# comparing sequences from      17316  to      31746\n",
      "..........    20000  finished       2731  clusters\n",
      "----------   6423 remaining sequences to the next cycle\n",
      "---------- new table with     1031 representatives\n",
      "# comparing sequences from      25323  to      38419\n",
      "..........    30000  finished       4017  clusters\n",
      "----------   5615 remaining sequences to the next cycle\n",
      "---------- new table with      980 representatives\n",
      "# comparing sequences from      32804  to      44653\n",
      "----------   5351 remaining sequences to the next cycle\n",
      "---------- new table with      921 representatives\n",
      "# comparing sequences from      39302  to      50068\n",
      "..........    40000  finished       5422  clusters\n",
      "----------   5226 remaining sequences to the next cycle\n",
      "---------- new table with      786 representatives\n",
      "# comparing sequences from      44842  to      54685\n",
      "----------   4853 remaining sequences to the next cycle\n",
      "---------- new table with      686 representatives\n",
      "# comparing sequences from      49832  to      58843\n",
      "..........    50000  finished       6833  clusters\n",
      "----------   4193 remaining sequences to the next cycle\n",
      "---------- new table with      663 representatives\n",
      "# comparing sequences from      54650  to      62858\n",
      "----------   3667 remaining sequences to the next cycle\n",
      "---------- new table with      652 representatives\n",
      "# comparing sequences from      59191  to      66642\n",
      "..........    60000  finished       8215  clusters\n",
      "----------   3526 remaining sequences to the next cycle\n",
      "---------- new table with      525 representatives\n",
      "# comparing sequences from      63116  to      69913\n",
      "----------   3439 remaining sequences to the next cycle\n",
      "---------- new table with      428 representatives\n",
      "# comparing sequences from      66474  to      72711\n",
      "----------   2950 remaining sequences to the next cycle\n",
      "---------- new table with      445 representatives\n",
      "# comparing sequences from      69761  to      75451\n",
      "..........    70000  finished       9554  clusters\n",
      "----------   2592 remaining sequences to the next cycle\n",
      "---------- new table with      395 representatives\n",
      "# comparing sequences from      72859  to      78032\n",
      "----------   2455 remaining sequences to the next cycle\n",
      "---------- new table with      357 representatives\n",
      "# comparing sequences from      75577  to      80297\n",
      "----------   2254 remaining sequences to the next cycle\n",
      "---------- new table with      328 representatives\n",
      "# comparing sequences from      78043  to      82352\n",
      "..........    80000  finished      10847  clusters\n",
      "----------   2007 remaining sequences to the next cycle\n",
      "---------- new table with      299 representatives\n",
      "# comparing sequences from      80345  to      84271\n",
      "----------   1931 remaining sequences to the next cycle\n",
      "---------- new table with      253 representatives\n",
      "# comparing sequences from      82340  to      85933\n",
      "----------   1547 remaining sequences to the next cycle\n",
      "---------- new table with      255 representatives\n",
      "# comparing sequences from      84386  to      87638\n",
      "----------   1520 remaining sequences to the next cycle\n",
      "---------- new table with      252 representatives\n",
      "# comparing sequences from      86118  to      89081\n",
      "----------   1322 remaining sequences to the next cycle\n",
      "---------- new table with      238 representatives\n",
      "# comparing sequences from      87759  to      90449\n",
      "----------   1224 remaining sequences to the next cycle\n",
      "---------- new table with      162 representatives\n",
      "# comparing sequences from      89225  to      91671\n",
      "..........    90000  finished      12150  clusters\n",
      "----------   1176 remaining sequences to the next cycle\n",
      "---------- new table with      150 representatives\n",
      "# comparing sequences from      90495  to      92729\n",
      "----------   1171 remaining sequences to the next cycle\n",
      "---------- new table with      120 representatives\n",
      "# comparing sequences from      91558  to      93615\n",
      "----------    963 remaining sequences to the next cycle\n",
      "---------- new table with      133 representatives\n",
      "# comparing sequences from      92652  to      94526\n",
      "----------    769 remaining sequences to the next cycle\n",
      "---------- new table with      161 representatives\n",
      "# comparing sequences from      93757  to      95447\n",
      "----------    592 remaining sequences to the next cycle\n",
      "---------- new table with      117 representatives\n",
      "# comparing sequences from      94855  to      96362\n",
      "----------    625 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      95737  to      97097\n",
      "----------    566 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      96531  to      97759\n",
      "----------    411 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      97348  to      98440\n",
      "----------    506 remaining sequences to the next cycle\n",
      "---------- new table with      103 representatives\n",
      "# comparing sequences from      97934  to      98928\n",
      "----------    287 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      98641  to      99517\n",
      "----------    147 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      99370  to     100125\n",
      "..........   100000  finished      13433  clusters\n",
      "----------    120 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from     100005  to     100654\n",
      "....................---------- new table with       85 representatives\n",
      "# comparing sequences from     100654  to     101195\n",
      ".....................---------- new table with       60 representatives\n",
      "# comparing sequences from     101195  to     101646\n",
      "....................---------- new table with       68 representatives\n",
      "# comparing sequences from     101646  to     102021\n",
      ".....................---------- new table with       63 representatives\n",
      "# comparing sequences from     102021  to     102334\n",
      "...................---------- new table with       41 representatives\n",
      "# comparing sequences from     102334  to     102595\n",
      "99.9%---------- new table with       38 representatives\n",
      "# comparing sequences from     102595  to     102812\n",
      "100.0%---------- new table with       23 representatives\n",
      "# comparing sequences from     102812  to     102993\n",
      "..................---------- new table with       18 representatives\n",
      "# comparing sequences from     102993  to     103901\n",
      "....................---------- new table with      138 representatives\n",
      "\n",
      "   103901  finished      13970  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 119M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 13.08\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Streptobacillus"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'streptobacillus'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Streptobacillus_felis_GCF_001559775\",\n",
    "               \"Streptobacillus_hongkongensis_GCF_001559795\",\n",
    "               \"Streptobacillus_notomytis_GCF_001902575\",\n",
    "               \"Streptobacillus_ratti_GCF_001891165\",\n",
    "               \"Streptobacillus_sp_GCF_009733925\"  ]\n",
    "                    \n",
    "reference_taxon = \"Streptobacillus_moniliformis_GCF_000024565\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001559775.1_ASM155977v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 451685 bytes  903454.00 bytes/sec\n",
      "total size is 451467  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001559795.1_ASM155979v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 413015 bytes  826114.00 bytes/sec\n",
      "total size is 412805  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000024565.1_ASM2456v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 462610 bytes  925304.00 bytes/sec\n",
      "total size is 462386  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001902575.1_ASM190257v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 387197 bytes  774478.00 bytes/sec\n",
      "total size is 386995  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001891165.1_ASM189116v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 403132 bytes  806348.00 bytes/sec\n",
      "total size is 402922  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009733925.1_ASM973392v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 453090 bytes  906264.00 bytes/sec\n",
      "total size is 452872  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480861 (\"sim_Streptobacillus_felis_GCF_001559775\") has been submitted\n",
      "Your job 5480862 (\"sim_Streptobacillus_hongkongensis_GCF_001559795\") has been submitted\n",
      "Your job 5480863 (\"sim_Streptobacillus_notomytis_GCF_001902575\") has been submitted\n",
      "Your job 5480864 (\"sim_Streptobacillus_ratti_GCF_001891165\") has been submitted\n",
      "Your job 5480865 (\"sim_Streptobacillus_sp_GCF_009733925\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/streptobacillus/cleaned_genomes/Streptobacillus_moniliformis_GCF_000024565_formatted.fas path=results/phyluce/streptobacillus/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480866 (\"map_Streptobacillus_felis_GCF_001559775\") has been submitted\n",
      "Your job 5480867 (\"map_Streptobacillus_hongkongensis_GCF_001559795\") has been submitted\n",
      "Your job 5480868 (\"map_Streptobacillus_notomytis_GCF_001902575\") has been submitted\n",
      "Your job 5480869 (\"map_Streptobacillus_ratti_GCF_001891165\") has been submitted\n",
      "Your job 5480870 (\"map_Streptobacillus_sp_GCF_009733925\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480871 (\"merge_Streptobacillus_felis_GCF_001559775\") has been submitted\n",
      "Your job 5480872 (\"merge_Streptobacillus_hongkongensis_GCF_001559795\") has been submitted\n",
      "Your job 5480873 (\"merge_Streptobacillus_notomytis_GCF_001902575\") has been submitted\n",
      "Your job 5480874 (\"merge_Streptobacillus_ratti_GCF_001891165\") has been submitted\n",
      "Your job 5480875 (\"merge_Streptobacillus_sp_GCF_009733925\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 1554 sequences from Streptobacillus_felis_GCF_001559775_merged.bed.  Filtered 871 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 683.\n",
      "Screened 514 sequences from Streptobacillus_hongkongensis_GCF_001559795_merged.bed.  Filtered 318 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 196.\n",
      "Screened 2204 sequences from Streptobacillus_notomytis_GCF_001902575_merged.bed.  Filtered 728 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1476.\n",
      "Screened 2041 sequences from Streptobacillus_ratti_GCF_001891165_merged.bed.  Filtered 569 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1472.\n",
      "Screened 1656 sequences from Streptobacillus_sp_GCF_009733925_merged.bed.  Filtered 905 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 751.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/streptobacillus/initial_intervals/Streptobacillus_sp_GCF_009733925_merged.bed --twobit results/phyluce/streptobacillus/cleaned_genomes/Streptobacillus_moniliformis_GCF_000024565_formatted.2bit --output results/phyluce/streptobacillus/initial_intervals/Streptobacillus_sp_GCF_009733925_stripped.bed;\n",
      "streptobacillus_felis_gcf_001559775.\n",
      "streptobacillus_hongkongensis_gcf_001559795.\n",
      "streptobacillus_notomytis_gcf_001902575..\n",
      "streptobacillus_ratti_gcf_001891165..\n",
      "streptobacillus_sp_gcf_009733925.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/streptobacillus/initial_intervals/Streptobacillus_sp_GCF_009733925_merged.bed --twobit results/phyluce/streptobacillus/cleaned_genomes/Streptobacillus_moniliformis_GCF_000024565_formatted.2bit --output results/phyluce/streptobacillus/initial_intervals/Streptobacillus_sp_GCF_009733925_stripped.bed;\n",
      "Loci shared by Streptobacillus_moniliformis_GCF_000024565 + 0 taxa:\t2,153.0\n",
      "Loci shared by Streptobacillus_moniliformis_GCF_000024565 + 1 taxa:\t2,153.0\n",
      "Loci shared by Streptobacillus_moniliformis_GCF_000024565 + 2 taxa:\t1,598.0\n",
      "Loci shared by Streptobacillus_moniliformis_GCF_000024565 + 3 taxa:\t811.0\n",
      "Loci shared by Streptobacillus_moniliformis_GCF_000024565 + 4 taxa:\t480.0\n",
      "Loci shared by Streptobacillus_moniliformis_GCF_000024565 + 5 taxa:\t130.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/streptobacillus/initial_intervals/streptobacillus-to-Streptobacillus_moniliformis_GCF_000024565.sqlite --base-taxon Streptobacillus_moniliformis_GCF_000024565 --output results/phyluce/streptobacillus/initial_intervals/Streptobacillus_moniliformis_GCF_000024565_+4.bed --specific-counts 4;\n",
      "Counter({'streptobacillus_notomytis_gcf_001902575': 480, 'streptobacillus_ratti_gcf_001891165': 480, 'streptobacillus_sp_gcf_009733925': 473, 'streptobacillus_felis_gcf_001559775': 467, 'streptobacillus_hongkongensis_gcf_001559795': 150})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 4\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/streptobacillus/initial_intervals/Streptobacillus_moniliformis_GCF_000024565_+4.bed --twobit results/phyluce/streptobacillus/cleaned_genomes/Streptobacillus_moniliformis_GCF_000024565_formatted.2bit --buffer-to 160 --output results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4.fasta;\n",
      "Screened 480 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 480.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4.fasta --probe-prefix uce_streptobacillus_ --design streptobacillus_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 267\n",
      "Probe Count = 440\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas --query results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  12:34:01\n",
      "Ended:  Thu Feb 06, 2020  12:34:02\n",
      "Time for execution:  0.0038468003273 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas                        --lastz results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes_vself.lastz                       --probe-prefix=uce_streptobacillus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 439 fasta sequences.  Filtered 6 duplicates. Kept 428.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/streptobacillus/validate_intervals/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas --scaffoldlist Streptobacillus_felis_GCF_001559775 Streptobacillus_hongkongensis_GCF_001559795 Streptobacillus_notomytis_GCF_001902575 Streptobacillus_ratti_GCF_001891165 Streptobacillus_sp_GCF_009733925 Streptobacillus_moniliformis_GCF_000024565 --genome-base-path results/phyluce/streptobacillus/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/streptobacillus/validate_intervals/streptobacillus-to-Streptobacillus_moniliformis_GCF_000024565.sqlite --output results/phyluce/streptobacillus/validate_intervals/lastz/;\n",
      "\n",
      "Running against Streptobacillus_felis_GCF_001559775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpga3iKI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjhJyyM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptobacillus/validate_intervals/lastz/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_Streptobacillus_felis_GCF_001559775.lastz\n",
      "Creating Streptobacillus_felis_GCF_001559775 table\n",
      "Inserting data to Streptobacillus_felis_GCF_001559775 table\n",
      "\n",
      "Running against Streptobacillus_hongkongensis_GCF_001559795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjr2Can.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLI7PhH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptobacillus/validate_intervals/lastz/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_Streptobacillus_hongkongensis_GCF_001559795.lastz\n",
      "Creating Streptobacillus_hongkongensis_GCF_001559795 table\n",
      "Inserting data to Streptobacillus_hongkongensis_GCF_001559795 table\n",
      "\n",
      "Running against Streptobacillus_notomytis_GCF_001902575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPRBpAv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd9P0GD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptobacillus/validate_intervals/lastz/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_Streptobacillus_notomytis_GCF_001902575.lastz\n",
      "Creating Streptobacillus_notomytis_GCF_001902575 table\n",
      "Inserting data to Streptobacillus_notomytis_GCF_001902575 table\n",
      "\n",
      "Running against Streptobacillus_ratti_GCF_001891165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHCpLuY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE6w5QU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptobacillus/validate_intervals/lastz/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_Streptobacillus_ratti_GCF_001891165.lastz\n",
      "Creating Streptobacillus_ratti_GCF_001891165 table\n",
      "Inserting data to Streptobacillus_ratti_GCF_001891165 table\n",
      "\n",
      "Running against Streptobacillus_sp_GCF_009733925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZ8Mjj1.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphZ9No3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptobacillus/validate_intervals/lastz/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_Streptobacillus_sp_GCF_009733925.lastz\n",
      "Creating Streptobacillus_sp_GCF_009733925 table\n",
      "Inserting data to Streptobacillus_sp_GCF_009733925 table\n",
      "\n",
      "Running against Streptobacillus_moniliformis_GCF_000024565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNfmQ9E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJu0O90.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptobacillus/validate_intervals/lastz/Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_Streptobacillus_moniliformis_GCF_000024565.lastz\n",
      "Creating Streptobacillus_moniliformis_GCF_000024565 table\n",
      "Inserting data to Streptobacillus_moniliformis_GCF_000024565 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/streptobacillus/extract_probes_from_group/streptobacillus_genome.conf --lastz results/phyluce/streptobacillus/validate_intervals/lastz --probes 120 --probe-prefix uce_streptobacillus_ --name-pattern \"Streptobacillus_moniliformis_GCF_000024565_+4_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/streptobacillus/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 12:34:16,985 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 12:34:16,987 - Phyluce - INFO - ----- Working on Streptobacillus_felis_GCF_001559775 genome -----\n",
      "2020-02-06 12:34:16,988 - Phyluce - INFO - Reading Streptobacillus_felis_GCF_001559775 genome\n",
      "2020-02-06 12:34:17,899 - Phyluce - INFO - Streptobacillus_felis_GCF_001559775: 267 uces, 30 dupes, 237 non-dupes, 0 orient drop, 0 length drop, 237 written\n",
      "2020-02-06 12:34:17,899 - Phyluce - INFO - - Working on Streptobacillus_hongkongensis_GCF_001559795 genome -\n",
      "2020-02-06 12:34:17,900 - Phyluce - INFO - Reading Streptobacillus_hongkongensis_GCF_001559795 genome\n",
      "2020-02-06 12:34:18,629 - Phyluce - INFO - Streptobacillus_hongkongensis_GCF_001559795: 248 uces, 22 dupes, 226 non-dupes, 0 orient drop, 3 length drop, 223 written\n",
      "2020-02-06 12:34:18,629 - Phyluce - INFO - --- Working on Streptobacillus_notomytis_GCF_001902575 genome ---\n",
      "2020-02-06 12:34:18,630 - Phyluce - INFO - Reading Streptobacillus_notomytis_GCF_001902575 genome\n",
      "2020-02-06 12:34:19,439 - Phyluce - INFO - Streptobacillus_notomytis_GCF_001902575: 267 uces, 22 dupes, 245 non-dupes, 0 orient drop, 3 length drop, 242 written\n",
      "2020-02-06 12:34:19,439 - Phyluce - INFO - ----- Working on Streptobacillus_ratti_GCF_001891165 genome -----\n",
      "2020-02-06 12:34:19,459 - Phyluce - INFO - Reading Streptobacillus_ratti_GCF_001891165 genome\n",
      "2020-02-06 12:34:20,264 - Phyluce - INFO - Streptobacillus_ratti_GCF_001891165: 267 uces, 21 dupes, 246 non-dupes, 0 orient drop, 2 length drop, 244 written\n",
      "2020-02-06 12:34:20,265 - Phyluce - INFO - ------- Working on Streptobacillus_sp_GCF_009733925 genome ------\n",
      "2020-02-06 12:34:20,272 - Phyluce - INFO - Reading Streptobacillus_sp_GCF_009733925 genome\n",
      "2020-02-06 12:34:21,085 - Phyluce - INFO - Streptobacillus_sp_GCF_009733925: 267 uces, 23 dupes, 244 non-dupes, 0 orient drop, 2 length drop, 242 written\n",
      "2020-02-06 12:34:21,085 - Phyluce - INFO - -- Working on Streptobacillus_moniliformis_GCF_000024565 genome -\n",
      "2020-02-06 12:34:21,099 - Phyluce - INFO - Reading Streptobacillus_moniliformis_GCF_000024565 genome\n",
      "2020-02-06 12:34:22,054 - Phyluce - INFO - Streptobacillus_moniliformis_GCF_000024565: 267 uces, 0 dupes, 267 non-dupes, 25 orient drop, 10 length drop, 232 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/streptobacillus/extract_probes_from_group/probe_fasta --output results/phyluce/streptobacillus/extract_probes_from_group/multifastas.sqlite --base-taxon Streptobacillus_moniliformis_GCF_000024565;\n",
      "streptobacillus_felis_gcf_001559775.\n",
      "streptobacillus_hongkongensis_gcf_001559795.\n",
      "streptobacillus_notomytis_gcf_001902575.\n",
      "streptobacillus_ratti_gcf_001891165.\n",
      "streptobacillus_sp_gcf_009733925.\n",
      "streptobacillus_moniliformis_gcf_000024565.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/streptobacillus/extract_probes_from_group/multifastas.sqlite --base-taxon Streptobacillus_moniliformis_GCF_000024565;\n",
      "Loci shared by 0 taxa:\t257.0\n",
      "Loci shared by 1 taxa:\t257.0\n",
      "Loci shared by 2 taxa:\t250.0\n",
      "Loci shared by 3 taxa:\t247.0\n",
      "Loci shared by 4 taxa:\t242.0\n",
      "Loci shared by 5 taxa:\t234.0\n",
      "Loci shared by 6 taxa:\t190.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/streptobacillus/extract_probes_from_group/multifastas.sqlite  --base-taxon Streptobacillus_moniliformis_GCF_000024565 --output results/phyluce/streptobacillus/extract_probes_from_group/Streptobacillus_moniliformis_GCF_000024565+4-back-to-6.conf --specific-counts 6;\n",
      "Counter({'streptobacillus_hongkongensis_gcf_001559795': 190, 'streptobacillus_ratti_gcf_001891165': 190, 'streptobacillus_moniliformis_gcf_000024565': 190, 'streptobacillus_felis_gcf_001559775': 190, 'streptobacillus_notomytis_gcf_001902575': 190, 'streptobacillus_sp_gcf_009733925': 190})\n",
      "Total loci = 190\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(6)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/streptobacillus/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/streptobacillus/extract_probes_from_group/Streptobacillus_moniliformis_GCF_000024565+4-back-to-6.conf --probe-prefix uce_streptobacillus_ --designer rnplattii --design streptobacillus_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 190\n",
      "Probe Count = 1852\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list.fasta --query results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  12:34:33\n",
      "Ended:  Thu Feb 06, 2020  12:34:35\n",
      "Time for execution:  0.022558816274 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list.fasta --lastz results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_streptobacillus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 1851 fasta sequences.  Filtered 0 duplicates. Kept 1852.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/streptobacillus/final_probe_design/streptobacillus_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 12:35:02 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 1852\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 148160\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3961846\n",
      "Max number of word counting entries: 89181166\n",
      "\n",
      "# comparing sequences from          0  to        308\n",
      "---------- new table with      261 representatives\n",
      "# comparing sequences from        308  to        565\n",
      "99.9%---------- new table with      224 representatives\n",
      "# comparing sequences from        565  to        779\n",
      "----------     40 remaining sequences to the next cycle\n",
      "---------- new table with      139 representatives\n",
      "# comparing sequences from        739  to        924\n",
      "99.8%---------- new table with      159 representatives\n",
      "# comparing sequences from        924  to       1852\n",
      ".....................---------- new table with      766 representatives\n",
      "\n",
      "     1852  finished       1549  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 87M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.56\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Streptococcus"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'streptococcus'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Streptococcus_acidominimus_GCF_900187245\",\n",
    "               \"Streptococcus_agalactiae_GCF_000007265\",\n",
    "               \"Streptococcus_alactolyticus_GCF_009695625\",\n",
    "               \"Streptococcus_anginosus_GCF_000463505\",\n",
    "               \"Streptococcus_australis_GCF_900476055\",\n",
    "               \"Streptococcus_azizii_GCF_001984715\",\n",
    "               \"Streptococcus_bovimastitidis_GCF_001885095\",\n",
    "               \"Streptococcus_caballi_GCF_000379985\",\n",
    "               \"Streptococcus_canis_GCF_900636575\",\n",
    "               \"Streptococcus_castoreus_GCF_000425025\",\n",
    "               \"Streptococcus_constellatus_GCF_000463425\",\n",
    "               \"Streptococcus_criceti_GCF_000187975\",\n",
    "               \"Streptococcus_cristatus_GCF_000385925\",\n",
    "               \"Streptococcus_cuniculi_GCF_001921845\",\n",
    "               \"Streptococcus_danieliae_GCF_009767945\",\n",
    "               \"Streptococcus_devriesei_GCF_000423725\",\n",
    "               \"Streptococcus_didelphis_GCF_000380005\",\n",
    "               \"Streptococcus_downei_GCF_900459175\",\n",
    "               \"Streptococcus_dysgalactiae_GCF_000317855\",\n",
    "               \"Streptococcus_entericus_GCF_000380025\",\n",
    "               \"Streptococcus_equi_GCF_000026605\",\n",
    "               \"Streptococcus_equinus_GCF_000964315\",\n",
    "               \"Streptococcus_ferus_GCF_000372425\",\n",
    "               \"Streptococcus_gallolyticus_GCF_002000985\",\n",
    "               \"Streptococcus_gordonii_GCF_000017005\",\n",
    "               \"Streptococcus_halichoeri_GCF_009870755\",\n",
    "               \"Streptococcus_halitosis_GCF_003143695\",\n",
    "               \"Streptococcus_halotolerans_GCF_001598035\",\n",
    "               \"Streptococcus_henryi_GCF_000376985\",\n",
    "               \"Streptococcus_himalayensis_GCF_001708305\",\n",
    "               \"Streptococcus_hongkongensis_GCF_000785845\",\n",
    "               \"Streptococcus_hyointestinalis_GCF_900459405\",\n",
    "               \"Streptococcus_hyovaginalis_GCF_000420785\",\n",
    "               \"Streptococcus_ictaluri_GCF_000188015\",\n",
    "               \"Streptococcus_infantarius_GCF_000246835\",\n",
    "               \"Streptococcus_infantis_GCF_000187465\",\n",
    "               \"Streptococcus_iniae_GCF_000831485\",\n",
    "               \"Streptococcus_intermedius_GCF_000463355\",\n",
    "               \"Streptococcus_lutetiensis_GCF_900475675\",\n",
    "               \"Streptococcus_macacae_GCF_000187995\",\n",
    "               \"Streptococcus_macedonicus_GCF_000283635\",\n",
    "               \"Streptococcus_marimammalium_GCF_000380045\",\n",
    "               \"Streptococcus_marmotae_GCF_001623565\",\n",
    "               \"Streptococcus_massiliensis_GCF_000380065\",\n",
    "               \"Streptococcus_merionis_GCF_000380085\",\n",
    "               \"Streptococcus_milleri_GCF_900636715\",\n",
    "               \"Streptococcus_minor_GCF_000377005\",\n",
    "               \"Streptococcus_mitis_GCF_000027165\",\n",
    "               \"Streptococcus_mutans_GCF_000007465\",\n",
    "               \"Streptococcus_oralis_GCF_900637025\",\n",
    "               \"Streptococcus_orisasini_GCF_001431045\",\n",
    "               \"Streptococcus_orisratti_GCF_000380105\",\n",
    "               \"Streptococcus_ovis_GCF_000380125\",\n",
    "               \"Streptococcus_pantholopis_GCF_001642085\",\n",
    "               \"Streptococcus_parasanguinis_GCF_000164675\",\n",
    "               \"Streptococcus_parasuis_GCF_004283785\",\n",
    "               \"Streptococcus_parauberis_GCF_000213825\",\n",
    "               \"Streptococcus_pasteurianus_GCF_900478025\",\n",
    "               \"Streptococcus_penaeicida_GCF_002887775\",\n",
    "               \"Streptococcus_peroris_GCF_000187585\",\n",
    "               \"Streptococcus_pharyngis_GCF_007859195\",\n",
    "               \"Streptococcus_phocae_GCF_000772915\",\n",
    "               \"Streptococcus_pluranimalium_GCF_003352995\",\n",
    "               \"Streptococcus_plurextorum_GCF_000423745\",              \n",
    "               \"Streptococcus_porci_GCF_000423765\",\n",
    "               \"Streptococcus_porcinus_GCF_900475415\",\n",
    "               \"Streptococcus_pseudopneumoniae_GCF_000221985\",\n",
    "               \"Streptococcus_pseudoporcinus_GCF_000188035\",\n",
    "               \"Streptococcus_pyogenes_GCF_000006785\",\n",
    "               \"Streptococcus_ratti_GCF_000286075\",\n",
    "               \"Streptococcus_respraculi_GCF_003595525\",\n",
    "               \"Streptococcus_rubneri_GCF_004785935\",\n",
    "               \"Streptococcus_ruminantium_GCF_003609975\",\n",
    "               \"Streptococcus_salivarius_GCF_000785515\",\n",
    "               \"Streptococcus_sanguinis_GCF_000014205\",\n",
    "               \"Streptococcus_sinensis_GCF_000767835\",\n",
    "               \"Streptococcus_sobrinus_GCF_000686605\",\n",
    "               \"Streptococcus_sp_GCF_003086355\",\n",
    "               \"Streptococcus_suis_GCF_000026745\",\n",
    "               \"Streptococcus_thermophilus_GCF_000253395\",\n",
    "               \"Streptococcus_thoraltensis_GCF_000380145\",\n",
    "               \"Streptococcus_timonensis_GCF_900095845\",\n",
    "               \"Streptococcus_troglodytae_GCF_002355215\",\n",
    "               \"Streptococcus_uberis_GCF_000009545\",\n",
    "               \"Streptococcus_urinalis_GCF_000188055\",\n",
    "               \"Streptococcus_varani_GCF_001375655\",\n",
    "               \"Streptococcus_vestibularis_GCF_900636445\",\n",
    "               \"Streptococcus_viridans_GCF_900636365\" ]\n",
    "                    \n",
    "reference_taxon = \"Streptococcus_pneumoniae_GCF_000007045\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900187245.1_51470_D02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 687125 bytes  196333.43 bytes/sec\n",
      "total size is 686853  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007265.1_ASM726v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 634787 bytes  423219.33 bytes/sec\n",
      "total size is 634524  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009695625.1_ASM969562v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 522123 bytes  348110.00 bytes/sec\n",
      "total size is 521889  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000463505.1_ASM46350v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 659465 bytes  439671.33 bytes/sec\n",
      "total size is 659192  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900476055.1_53750_F01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 636295 bytes  1272674.00 bytes/sec\n",
      "total size is 636031  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001984715.1_ASM198471v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 691194 bytes  460824.00 bytes/sec\n",
      "total size is 690912  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001885095.1_ASM188509v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 613156 bytes  408798.67 bytes/sec\n",
      "total size is 612898  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000379985.1_ASM37998v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 635318 bytes  423573.33 bytes/sec\n",
      "total size is 635053  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900636575.1_42912_C01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 613705 bytes  409164.67 bytes/sec\n",
      "total size is 613449  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000425025.1_ASM42502v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 561215 bytes  374171.33 bytes/sec\n",
      "total size is 560966  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000463425.1_ASM46342v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 588864 bytes  392604.00 bytes/sec\n",
      "total size is 588615  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187975.3_ASM18797v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 718440 bytes  478988.00 bytes/sec\n",
      "total size is 718159  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000385925.1_ASM38592v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 634315 bytes  422904.67 bytes/sec\n",
      "total size is 634050  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001921845.1_ASM192184v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 622464 bytes  415004.00 bytes/sec\n",
      "total size is 622206  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009767945.1_ASM976794v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 559301 bytes  372895.33 bytes/sec\n",
      "total size is 559051  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000423725.1_ASM42372v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 631010 bytes  420701.33 bytes/sec\n",
      "total size is 630745  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380005.1_ASM38000v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 552950 bytes  368661.33 bytes/sec\n",
      "total size is 552709  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900459175.1_52087_B01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 662540 bytes  441721.33 bytes/sec\n",
      "total size is 662268  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000317855.1_ASM31785v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 645532 bytes  430382.67 bytes/sec\n",
      "total size is 645267  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380025.1_ASM38002v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 609952 bytes  406662.67 bytes/sec\n",
      "total size is 609695  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000026605.1_ASM2660v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 634515 bytes  423038.00 bytes/sec\n",
      "total size is 634251  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000964315.1_ASM96431v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 565160 bytes  376801.33 bytes/sec\n",
      "total size is 564911  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000372425.1_ASM37242v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 551834 bytes  367917.33 bytes/sec\n",
      "total size is 551593  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002000985.1_ASM200098v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 731886 bytes  487952.00 bytes/sec\n",
      "total size is 731596  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017005.1_ASM1700v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 650116 bytes  433438.67 bytes/sec\n",
      "total size is 649852  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009870755.1_ASM987075v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 569389 bytes  379620.67 bytes/sec\n",
      "total size is 569139  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003143695.2_ASM314369v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 560341 bytes  373588.67 bytes/sec\n",
      "total size is 560091  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001598035.1_ASM159803v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 639476 bytes  426345.33 bytes/sec\n",
      "total size is 639210  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000376985.1_ASM37698v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 721784 bytes  481217.33 bytes/sec\n",
      "total size is 721495  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001708305.1_ASM170830v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 666402 bytes  444296.00 bytes/sec\n",
      "total size is 666128  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000785845.1_CAIM1895T_spades_scaffold_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 590058 bytes  393400.00 bytes/sec\n",
      "total size is 589794  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900459405.1_41965_B01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 746015 bytes  497371.33 bytes/sec\n",
      "total size is 745727  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000420785.1_ASM42078v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 621677 bytes  414479.33 bytes/sec\n",
      "total size is 621420  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000188015.3_ASM18801v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657139 bytes  438120.67 bytes/sec\n",
      "total size is 656866  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000246835.1_ASM24683v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 590879 bytes  393947.33 bytes/sec\n",
      "total size is 590622  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187465.1_ASM18746v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 558304 bytes  372230.67 bytes/sec\n",
      "total size is 558055  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000831485.1_ASM83148v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 613389 bytes  408954.00 bytes/sec\n",
      "total size is 613132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000463355.1_ASM46335v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 589479 bytes  393014.00 bytes/sec\n",
      "total size is 589230  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900475675.1_45473_D02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 525723 bytes  350510.00 bytes/sec\n",
      "total size is 525483  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187995.3_ASM18799v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 564124 bytes  376110.67 bytes/sec\n",
      "total size is 563875  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000283635.1_ASM28363v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 625437 bytes  416986.00 bytes/sec\n",
      "total size is 625172  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380045.1_ASM38004v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 439459 bytes  293000.67 bytes/sec\n",
      "total size is 439242  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001623565.1_ASM162356v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 684540 bytes  456388.00 bytes/sec\n",
      "total size is 684266  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000380065.1_ASM38006v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 557536 bytes  371718.67 bytes/sec\n",
      "total size is 557287  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380085.1_ASM38008v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 705605 bytes  470431.33 bytes/sec\n",
      "total size is 705324  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900636715.1_43908_E01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 557131 bytes  371448.67 bytes/sec\n",
      "total size is 556891  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000377005.1_ASM37700v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 576037 bytes  384052.67 bytes/sec\n",
      "total size is 575788  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000027165.1_ASM2716v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 629208 bytes  419500.00 bytes/sec\n",
      "total size is 628944  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007465.2_ASM746v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597171 bytes  398142.00 bytes/sec\n",
      "total size is 596916  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900637025.1_46338_H01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 571143 bytes  380790.00 bytes/sec\n",
      "total size is 570895  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001431045.1_ASM143104v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 685495 bytes  457024.67 bytes/sec\n",
      "total size is 685221  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380105.1_ASM38010v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 722804 bytes  481897.33 bytes/sec\n",
      "total size is 722515  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380125.1_ASM38012v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 703797 bytes  469226.00 bytes/sec\n",
      "total size is 703516  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001642085.1_ASM164208v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657218 bytes  438173.33 bytes/sec\n",
      "total size is 656944  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000164675.2_ASM16467v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 637082 bytes  424749.33 bytes/sec\n",
      "total size is 636817  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004283785.1_ASM428378v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 562323 bytes  374910.00 bytes/sec\n",
      "total size is 562073  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000213825.1_ASM21382v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 631468 bytes  421006.67 bytes/sec\n",
      "total size is 631203  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900478025.1_48128_C02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 639261 bytes  426202.00 bytes/sec\n",
      "total size is 638997  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002887775.1_ASM288777v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 606631 bytes  404448.67 bytes/sec\n",
      "total size is 606373  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187585.1_ASM18758v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 485318 bytes  970720.00 bytes/sec\n",
      "total size is 485093  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007859195.1_ASM785919v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 449091 bytes  299422.00 bytes/sec\n",
      "total size is 448873  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000772915.1_ASM77291v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 497553 bytes  331730.00 bytes/sec\n",
      "total size is 497320  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003352995.1_ASM335299v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657971 bytes  438675.33 bytes/sec\n",
      "total size is 657697  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000423745.1_ASM42374v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 627729 bytes  1255542.00 bytes/sec\n",
      "total size is 627464  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007045.1_ASM704v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602545 bytes  401724.67 bytes/sec\n",
      "total size is 602290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000423765.1_ASM42376v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 684563 bytes  456403.33 bytes/sec\n",
      "total size is 684290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900475415.1_42206_H01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 600001 bytes  400028.67 bytes/sec\n",
      "total size is 599745  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000221985.1_ASM22198v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 649335 bytes  432918.00 bytes/sec\n",
      "total size is 649070  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000188035.3_ASM18803v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 629490 bytes  419688.00 bytes/sec\n",
      "total size is 629225  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006785.2_ASM678v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 548272 bytes  365542.67 bytes/sec\n",
      "total size is 548033  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000286075.1_ASM28607v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 612045 bytes  1224174.00 bytes/sec\n",
      "total size is 611788  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003595525.1_ASM359552v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 612875 bytes  408611.33 bytes/sec\n",
      "total size is 612617  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004785935.1_ASM478593v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 639521 bytes  426375.33 bytes/sec\n",
      "total size is 639255  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003609975.1_ASM360997v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 617986 bytes  412018.67 bytes/sec\n",
      "total size is 617728  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000785515.1_ASM78551v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 646181 bytes  430815.33 bytes/sec\n",
      "total size is 645916  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000014205.1_ASM1420v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 708385 bytes  472284.67 bytes/sec\n",
      "total size is 708105  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000767835.1_ASM76783v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 618228 bytes  412180.00 bytes/sec\n",
      "total size is 617971  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000686605.1_ASM68660v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 641205 bytes  427498.00 bytes/sec\n",
      "total size is 640940  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003086355.2_ASM308635v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 720458 bytes  1441000.00 bytes/sec\n",
      "total size is 720176  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000026745.1_ASM2674v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 644781 bytes  429882.00 bytes/sec\n",
      "total size is 644517  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000253395.1_ASM25339v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 570272 bytes  380209.33 bytes/sec\n",
      "total size is 570023  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380145.1_ASM38014v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 708641 bytes  472455.33 bytes/sec\n",
      "total size is 708360  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900095845.1_PRJEB15309_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 569860 bytes  379934.67 bytes/sec\n",
      "total size is 569611  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002355215.1_ASM235521v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 617097 bytes  411426.00 bytes/sec\n",
      "total size is 616839  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000009545.1_ASM954v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 546418 bytes  364306.67 bytes/sec\n",
      "total size is 546179  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000188055.3_ASM18805v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 622696 bytes  415158.67 bytes/sec\n",
      "total size is 622439  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001375655.1_PRJEB8936_assembly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 728228 bytes  485513.33 bytes/sec\n",
      "total size is 727929  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GCA_900636445.1_41965_G01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 575065 bytes  383404.67 bytes/sec\n",
      "total size is 574817  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900636365.1_41559_G01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 592864 bytes  395270.67 bytes/sec\n",
      "total size is 592608  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481185 (\"sim_Streptococcus_acidominimus_GCF_900187245\") has been submitted\n",
      "Your job 5481186 (\"sim_Streptococcus_agalactiae_GCF_000007265\") has been submitted\n",
      "Your job 5481187 (\"sim_Streptococcus_alactolyticus_GCF_009695625\") has been submitted\n",
      "Your job 5481188 (\"sim_Streptococcus_anginosus_GCF_000463505\") has been submitted\n",
      "Your job 5481189 (\"sim_Streptococcus_australis_GCF_900476055\") has been submitted\n",
      "Your job 5481190 (\"sim_Streptococcus_azizii_GCF_001984715\") has been submitted\n",
      "Your job 5481191 (\"sim_Streptococcus_bovimastitidis_GCF_001885095\") has been submitted\n",
      "Your job 5481192 (\"sim_Streptococcus_caballi_GCF_000379985\") has been submitted\n",
      "Your job 5481193 (\"sim_Streptococcus_canis_GCF_900636575\") has been submitted\n",
      "Your job 5481194 (\"sim_Streptococcus_castoreus_GCF_000425025\") has been submitted\n",
      "Your job 5481195 (\"sim_Streptococcus_constellatus_GCF_000463425\") has been submitted\n",
      "Your job 5481196 (\"sim_Streptococcus_criceti_GCF_000187975\") has been submitted\n",
      "Your job 5481197 (\"sim_Streptococcus_cristatus_GCF_000385925\") has been submitted\n",
      "Your job 5481198 (\"sim_Streptococcus_cuniculi_GCF_001921845\") has been submitted\n",
      "Your job 5481199 (\"sim_Streptococcus_danieliae_GCF_009767945\") has been submitted\n",
      "Your job 5481200 (\"sim_Streptococcus_devriesei_GCF_000423725\") has been submitted\n",
      "Your job 5481201 (\"sim_Streptococcus_didelphis_GCF_000380005\") has been submitted\n",
      "Your job 5481202 (\"sim_Streptococcus_downei_GCF_900459175\") has been submitted\n",
      "Your job 5481203 (\"sim_Streptococcus_dysgalactiae_GCF_000317855\") has been submitted\n",
      "Your job 5481204 (\"sim_Streptococcus_entericus_GCF_000380025\") has been submitted\n",
      "Your job 5481205 (\"sim_Streptococcus_equi_GCF_000026605\") has been submitted\n",
      "Your job 5481206 (\"sim_Streptococcus_equinus_GCF_000964315\") has been submitted\n",
      "Your job 5481207 (\"sim_Streptococcus_ferus_GCF_000372425\") has been submitted\n",
      "Your job 5481208 (\"sim_Streptococcus_gallolyticus_GCF_002000985\") has been submitted\n",
      "Your job 5481209 (\"sim_Streptococcus_gordonii_GCF_000017005\") has been submitted\n",
      "Your job 5481210 (\"sim_Streptococcus_halichoeri_GCF_009870755\") has been submitted\n",
      "Your job 5481211 (\"sim_Streptococcus_halitosis_GCF_003143695\") has been submitted\n",
      "Your job 5481212 (\"sim_Streptococcus_halotolerans_GCF_001598035\") has been submitted\n",
      "Your job 5481213 (\"sim_Streptococcus_henryi_GCF_000376985\") has been submitted\n",
      "Your job 5481214 (\"sim_Streptococcus_himalayensis_GCF_001708305\") has been submitted\n",
      "Your job 5481215 (\"sim_Streptococcus_hongkongensis_GCF_000785845\") has been submitted\n",
      "Your job 5481216 (\"sim_Streptococcus_hyointestinalis_GCF_900459405\") has been submitted\n",
      "Your job 5481217 (\"sim_Streptococcus_hyovaginalis_GCF_000420785\") has been submitted\n",
      "Your job 5481218 (\"sim_Streptococcus_ictaluri_GCF_000188015\") has been submitted\n",
      "Your job 5481219 (\"sim_Streptococcus_infantarius_GCF_000246835\") has been submitted\n",
      "Your job 5481220 (\"sim_Streptococcus_infantis_GCF_000187465\") has been submitted\n",
      "Your job 5481221 (\"sim_Streptococcus_iniae_GCF_000831485\") has been submitted\n",
      "Your job 5481222 (\"sim_Streptococcus_intermedius_GCF_000463355\") has been submitted\n",
      "Your job 5481223 (\"sim_Streptococcus_lutetiensis_GCF_900475675\") has been submitted\n",
      "Your job 5481224 (\"sim_Streptococcus_macacae_GCF_000187995\") has been submitted\n",
      "Your job 5481225 (\"sim_Streptococcus_macedonicus_GCF_000283635\") has been submitted\n",
      "Your job 5481226 (\"sim_Streptococcus_marimammalium_GCF_000380045\") has been submitted\n",
      "Your job 5481227 (\"sim_Streptococcus_marmotae_GCF_001623565\") has been submitted\n",
      "Your job 5481228 (\"sim_Streptococcus_massiliensis_GCF_000380065\") has been submitted\n",
      "Your job 5481229 (\"sim_Streptococcus_merionis_GCF_000380085\") has been submitted\n",
      "Your job 5481230 (\"sim_Streptococcus_milleri_GCF_900636715\") has been submitted\n",
      "Your job 5481231 (\"sim_Streptococcus_minor_GCF_000377005\") has been submitted\n",
      "Your job 5481232 (\"sim_Streptococcus_mitis_GCF_000027165\") has been submitted\n",
      "Your job 5481233 (\"sim_Streptococcus_mutans_GCF_000007465\") has been submitted\n",
      "Your job 5481234 (\"sim_Streptococcus_oralis_GCF_900637025\") has been submitted\n",
      "Your job 5481235 (\"sim_Streptococcus_orisasini_GCF_001431045\") has been submitted\n",
      "Your job 5481236 (\"sim_Streptococcus_orisratti_GCF_000380105\") has been submitted\n",
      "Your job 5481237 (\"sim_Streptococcus_ovis_GCF_000380125\") has been submitted\n",
      "Your job 5481238 (\"sim_Streptococcus_pantholopis_GCF_001642085\") has been submitted\n",
      "Your job 5481239 (\"sim_Streptococcus_parasanguinis_GCF_000164675\") has been submitted\n",
      "Your job 5481240 (\"sim_Streptococcus_parasuis_GCF_004283785\") has been submitted\n",
      "Your job 5481241 (\"sim_Streptococcus_parauberis_GCF_000213825\") has been submitted\n",
      "Your job 5481242 (\"sim_Streptococcus_pasteurianus_GCF_900478025\") has been submitted\n",
      "Your job 5481243 (\"sim_Streptococcus_penaeicida_GCF_002887775\") has been submitted\n",
      "Your job 5481244 (\"sim_Streptococcus_peroris_GCF_000187585\") has been submitted\n",
      "Your job 5481245 (\"sim_Streptococcus_pharyngis_GCF_007859195\") has been submitted\n",
      "Your job 5481246 (\"sim_Streptococcus_phocae_GCF_000772915\") has been submitted\n",
      "Your job 5481247 (\"sim_Streptococcus_pluranimalium_GCF_003352995\") has been submitted\n",
      "Your job 5481248 (\"sim_Streptococcus_plurextorum_GCF_000423745\") has been submitted\n",
      "Your job 5481249 (\"sim_Streptococcus_porci_GCF_000423765\") has been submitted\n",
      "Your job 5481250 (\"sim_Streptococcus_porcinus_GCF_900475415\") has been submitted\n",
      "Your job 5481251 (\"sim_Streptococcus_pseudopneumoniae_GCF_000221985\") has been submitted\n",
      "Your job 5481252 (\"sim_Streptococcus_pseudoporcinus_GCF_000188035\") has been submitted\n",
      "Your job 5481253 (\"sim_Streptococcus_pyogenes_GCF_000006785\") has been submitted\n",
      "Your job 5481254 (\"sim_Streptococcus_ratti_GCF_000286075\") has been submitted\n",
      "Your job 5481255 (\"sim_Streptococcus_respraculi_GCF_003595525\") has been submitted\n",
      "Your job 5481256 (\"sim_Streptococcus_rubneri_GCF_004785935\") has been submitted\n",
      "Your job 5481257 (\"sim_Streptococcus_ruminantium_GCF_003609975\") has been submitted\n",
      "Your job 5481258 (\"sim_Streptococcus_salivarius_GCF_000785515\") has been submitted\n",
      "Your job 5481259 (\"sim_Streptococcus_sanguinis_GCF_000014205\") has been submitted\n",
      "Your job 5481260 (\"sim_Streptococcus_sinensis_GCF_000767835\") has been submitted\n",
      "Your job 5481261 (\"sim_Streptococcus_sobrinus_GCF_000686605\") has been submitted\n",
      "Your job 5481262 (\"sim_Streptococcus_sp_GCF_003086355\") has been submitted\n",
      "Your job 5481263 (\"sim_Streptococcus_suis_GCF_000026745\") has been submitted\n",
      "Your job 5481264 (\"sim_Streptococcus_thermophilus_GCF_000253395\") has been submitted\n",
      "Your job 5481265 (\"sim_Streptococcus_thoraltensis_GCF_000380145\") has been submitted\n",
      "Your job 5481266 (\"sim_Streptococcus_timonensis_GCF_900095845\") has been submitted\n",
      "Your job 5481267 (\"sim_Streptococcus_troglodytae_GCF_002355215\") has been submitted\n",
      "Your job 5481268 (\"sim_Streptococcus_uberis_GCF_000009545\") has been submitted\n",
      "Your job 5481269 (\"sim_Streptococcus_urinalis_GCF_000188055\") has been submitted\n",
      "Your job 5481270 (\"sim_Streptococcus_varani_GCF_001375655\") has been submitted\n",
      "Your job 5481271 (\"sim_Streptococcus_vestibularis_GCF_900636445\") has been submitted\n",
      "Your job 5481272 (\"sim_Streptococcus_viridans_GCF_900636365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.fas path=results/phyluce/streptococcus/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481273 (\"map_Streptococcus_acidominimus_GCF_900187245\") has been submitted\n",
      "Your job 5481274 (\"map_Streptococcus_agalactiae_GCF_000007265\") has been submitted\n",
      "Your job 5481275 (\"map_Streptococcus_alactolyticus_GCF_009695625\") has been submitted\n",
      "Your job 5481276 (\"map_Streptococcus_anginosus_GCF_000463505\") has been submitted\n",
      "Your job 5481277 (\"map_Streptococcus_australis_GCF_900476055\") has been submitted\n",
      "Your job 5481278 (\"map_Streptococcus_azizii_GCF_001984715\") has been submitted\n",
      "Your job 5481279 (\"map_Streptococcus_bovimastitidis_GCF_001885095\") has been submitted\n",
      "Your job 5481280 (\"map_Streptococcus_caballi_GCF_000379985\") has been submitted\n",
      "Your job 5481281 (\"map_Streptococcus_canis_GCF_900636575\") has been submitted\n",
      "Your job 5481282 (\"map_Streptococcus_castoreus_GCF_000425025\") has been submitted\n",
      "Your job 5481283 (\"map_Streptococcus_constellatus_GCF_000463425\") has been submitted\n",
      "Your job 5481284 (\"map_Streptococcus_criceti_GCF_000187975\") has been submitted\n",
      "Your job 5481285 (\"map_Streptococcus_cristatus_GCF_000385925\") has been submitted\n",
      "Your job 5481286 (\"map_Streptococcus_cuniculi_GCF_001921845\") has been submitted\n",
      "Your job 5481287 (\"map_Streptococcus_danieliae_GCF_009767945\") has been submitted\n",
      "Your job 5481288 (\"map_Streptococcus_devriesei_GCF_000423725\") has been submitted\n",
      "Your job 5481289 (\"map_Streptococcus_didelphis_GCF_000380005\") has been submitted\n",
      "Your job 5481290 (\"map_Streptococcus_downei_GCF_900459175\") has been submitted\n",
      "Your job 5481291 (\"map_Streptococcus_dysgalactiae_GCF_000317855\") has been submitted\n",
      "Your job 5481292 (\"map_Streptococcus_entericus_GCF_000380025\") has been submitted\n",
      "Your job 5481293 (\"map_Streptococcus_equi_GCF_000026605\") has been submitted\n",
      "Your job 5481294 (\"map_Streptococcus_equinus_GCF_000964315\") has been submitted\n",
      "Your job 5481295 (\"map_Streptococcus_ferus_GCF_000372425\") has been submitted\n",
      "Your job 5481296 (\"map_Streptococcus_gallolyticus_GCF_002000985\") has been submitted\n",
      "Your job 5481297 (\"map_Streptococcus_gordonii_GCF_000017005\") has been submitted\n",
      "Your job 5481298 (\"map_Streptococcus_halichoeri_GCF_009870755\") has been submitted\n",
      "Your job 5481299 (\"map_Streptococcus_halitosis_GCF_003143695\") has been submitted\n",
      "Your job 5481300 (\"map_Streptococcus_halotolerans_GCF_001598035\") has been submitted\n",
      "Your job 5481301 (\"map_Streptococcus_henryi_GCF_000376985\") has been submitted\n",
      "Your job 5481302 (\"map_Streptococcus_himalayensis_GCF_001708305\") has been submitted\n",
      "Your job 5481303 (\"map_Streptococcus_hongkongensis_GCF_000785845\") has been submitted\n",
      "Your job 5481304 (\"map_Streptococcus_hyointestinalis_GCF_900459405\") has been submitted\n",
      "Your job 5481305 (\"map_Streptococcus_hyovaginalis_GCF_000420785\") has been submitted\n",
      "Your job 5481306 (\"map_Streptococcus_ictaluri_GCF_000188015\") has been submitted\n",
      "Your job 5481307 (\"map_Streptococcus_infantarius_GCF_000246835\") has been submitted\n",
      "Your job 5481308 (\"map_Streptococcus_infantis_GCF_000187465\") has been submitted\n",
      "Your job 5481309 (\"map_Streptococcus_iniae_GCF_000831485\") has been submitted\n",
      "Your job 5481310 (\"map_Streptococcus_intermedius_GCF_000463355\") has been submitted\n",
      "Your job 5481311 (\"map_Streptococcus_lutetiensis_GCF_900475675\") has been submitted\n",
      "Your job 5481312 (\"map_Streptococcus_macacae_GCF_000187995\") has been submitted\n",
      "Your job 5481313 (\"map_Streptococcus_macedonicus_GCF_000283635\") has been submitted\n",
      "Your job 5481314 (\"map_Streptococcus_marimammalium_GCF_000380045\") has been submitted\n",
      "Your job 5481315 (\"map_Streptococcus_marmotae_GCF_001623565\") has been submitted\n",
      "Your job 5481316 (\"map_Streptococcus_massiliensis_GCF_000380065\") has been submitted\n",
      "Your job 5481317 (\"map_Streptococcus_merionis_GCF_000380085\") has been submitted\n",
      "Your job 5481318 (\"map_Streptococcus_milleri_GCF_900636715\") has been submitted\n",
      "Your job 5481319 (\"map_Streptococcus_minor_GCF_000377005\") has been submitted\n",
      "Your job 5481320 (\"map_Streptococcus_mitis_GCF_000027165\") has been submitted\n",
      "Your job 5481321 (\"map_Streptococcus_mutans_GCF_000007465\") has been submitted\n",
      "Your job 5481322 (\"map_Streptococcus_oralis_GCF_900637025\") has been submitted\n",
      "Your job 5481323 (\"map_Streptococcus_orisasini_GCF_001431045\") has been submitted\n",
      "Your job 5481324 (\"map_Streptococcus_orisratti_GCF_000380105\") has been submitted\n",
      "Your job 5481325 (\"map_Streptococcus_ovis_GCF_000380125\") has been submitted\n",
      "Your job 5481326 (\"map_Streptococcus_pantholopis_GCF_001642085\") has been submitted\n",
      "Your job 5481327 (\"map_Streptococcus_parasanguinis_GCF_000164675\") has been submitted\n",
      "Your job 5481328 (\"map_Streptococcus_parasuis_GCF_004283785\") has been submitted\n",
      "Your job 5481329 (\"map_Streptococcus_parauberis_GCF_000213825\") has been submitted\n",
      "Your job 5481330 (\"map_Streptococcus_pasteurianus_GCF_900478025\") has been submitted\n",
      "Your job 5481331 (\"map_Streptococcus_penaeicida_GCF_002887775\") has been submitted\n",
      "Your job 5481332 (\"map_Streptococcus_peroris_GCF_000187585\") has been submitted\n",
      "Your job 5481333 (\"map_Streptococcus_pharyngis_GCF_007859195\") has been submitted\n",
      "Your job 5481334 (\"map_Streptococcus_phocae_GCF_000772915\") has been submitted\n",
      "Your job 5481335 (\"map_Streptococcus_pluranimalium_GCF_003352995\") has been submitted\n",
      "Your job 5481336 (\"map_Streptococcus_plurextorum_GCF_000423745\") has been submitted\n",
      "Your job 5481337 (\"map_Streptococcus_porci_GCF_000423765\") has been submitted\n",
      "Your job 5481338 (\"map_Streptococcus_porcinus_GCF_900475415\") has been submitted\n",
      "Your job 5481339 (\"map_Streptococcus_pseudopneumoniae_GCF_000221985\") has been submitted\n",
      "Your job 5481340 (\"map_Streptococcus_pseudoporcinus_GCF_000188035\") has been submitted\n",
      "Your job 5481341 (\"map_Streptococcus_pyogenes_GCF_000006785\") has been submitted\n",
      "Your job 5481342 (\"map_Streptococcus_ratti_GCF_000286075\") has been submitted\n",
      "Your job 5481343 (\"map_Streptococcus_respraculi_GCF_003595525\") has been submitted\n",
      "Your job 5481344 (\"map_Streptococcus_rubneri_GCF_004785935\") has been submitted\n",
      "Your job 5481345 (\"map_Streptococcus_ruminantium_GCF_003609975\") has been submitted\n",
      "Your job 5481346 (\"map_Streptococcus_salivarius_GCF_000785515\") has been submitted\n",
      "Your job 5481347 (\"map_Streptococcus_sanguinis_GCF_000014205\") has been submitted\n",
      "Your job 5481348 (\"map_Streptococcus_sinensis_GCF_000767835\") has been submitted\n",
      "Your job 5481349 (\"map_Streptococcus_sobrinus_GCF_000686605\") has been submitted\n",
      "Your job 5481350 (\"map_Streptococcus_sp_GCF_003086355\") has been submitted\n",
      "Your job 5481351 (\"map_Streptococcus_suis_GCF_000026745\") has been submitted\n",
      "Your job 5481352 (\"map_Streptococcus_thermophilus_GCF_000253395\") has been submitted\n",
      "Your job 5481353 (\"map_Streptococcus_thoraltensis_GCF_000380145\") has been submitted\n",
      "Your job 5481354 (\"map_Streptococcus_timonensis_GCF_900095845\") has been submitted\n",
      "Your job 5481355 (\"map_Streptococcus_troglodytae_GCF_002355215\") has been submitted\n",
      "Your job 5481356 (\"map_Streptococcus_uberis_GCF_000009545\") has been submitted\n",
      "Your job 5481357 (\"map_Streptococcus_urinalis_GCF_000188055\") has been submitted\n",
      "Your job 5481358 (\"map_Streptococcus_varani_GCF_001375655\") has been submitted\n",
      "Your job 5481359 (\"map_Streptococcus_vestibularis_GCF_900636445\") has been submitted\n",
      "Your job 5481360 (\"map_Streptococcus_viridans_GCF_900636365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481361 (\"merge_Streptococcus_acidominimus_GCF_900187245\") has been submitted\n",
      "Your job 5481362 (\"merge_Streptococcus_agalactiae_GCF_000007265\") has been submitted\n",
      "Your job 5481363 (\"merge_Streptococcus_alactolyticus_GCF_009695625\") has been submitted\n",
      "Your job 5481364 (\"merge_Streptococcus_anginosus_GCF_000463505\") has been submitted\n",
      "Your job 5481365 (\"merge_Streptococcus_australis_GCF_900476055\") has been submitted\n",
      "Your job 5481366 (\"merge_Streptococcus_azizii_GCF_001984715\") has been submitted\n",
      "Your job 5481367 (\"merge_Streptococcus_bovimastitidis_GCF_001885095\") has been submitted\n",
      "Your job 5481368 (\"merge_Streptococcus_caballi_GCF_000379985\") has been submitted\n",
      "Your job 5481369 (\"merge_Streptococcus_canis_GCF_900636575\") has been submitted\n",
      "Your job 5481370 (\"merge_Streptococcus_castoreus_GCF_000425025\") has been submitted\n",
      "Your job 5481371 (\"merge_Streptococcus_constellatus_GCF_000463425\") has been submitted\n",
      "Your job 5481372 (\"merge_Streptococcus_criceti_GCF_000187975\") has been submitted\n",
      "Your job 5481373 (\"merge_Streptococcus_cristatus_GCF_000385925\") has been submitted\n",
      "Your job 5481374 (\"merge_Streptococcus_cuniculi_GCF_001921845\") has been submitted\n",
      "Your job 5481375 (\"merge_Streptococcus_danieliae_GCF_009767945\") has been submitted\n",
      "Your job 5481376 (\"merge_Streptococcus_devriesei_GCF_000423725\") has been submitted\n",
      "Your job 5481377 (\"merge_Streptococcus_didelphis_GCF_000380005\") has been submitted\n",
      "Your job 5481378 (\"merge_Streptococcus_downei_GCF_900459175\") has been submitted\n",
      "Your job 5481379 (\"merge_Streptococcus_dysgalactiae_GCF_000317855\") has been submitted\n",
      "Your job 5481380 (\"merge_Streptococcus_entericus_GCF_000380025\") has been submitted\n",
      "Your job 5481381 (\"merge_Streptococcus_equi_GCF_000026605\") has been submitted\n",
      "Your job 5481382 (\"merge_Streptococcus_equinus_GCF_000964315\") has been submitted\n",
      "Your job 5481383 (\"merge_Streptococcus_ferus_GCF_000372425\") has been submitted\n",
      "Your job 5481384 (\"merge_Streptococcus_gallolyticus_GCF_002000985\") has been submitted\n",
      "Your job 5481385 (\"merge_Streptococcus_gordonii_GCF_000017005\") has been submitted\n",
      "Your job 5481386 (\"merge_Streptococcus_halichoeri_GCF_009870755\") has been submitted\n",
      "Your job 5481387 (\"merge_Streptococcus_halitosis_GCF_003143695\") has been submitted\n",
      "Your job 5481388 (\"merge_Streptococcus_halotolerans_GCF_001598035\") has been submitted\n",
      "Your job 5481389 (\"merge_Streptococcus_henryi_GCF_000376985\") has been submitted\n",
      "Your job 5481390 (\"merge_Streptococcus_himalayensis_GCF_001708305\") has been submitted\n",
      "Your job 5481391 (\"merge_Streptococcus_hongkongensis_GCF_000785845\") has been submitted\n",
      "Your job 5481392 (\"merge_Streptococcus_hyointestinalis_GCF_900459405\") has been submitted\n",
      "Your job 5481393 (\"merge_Streptococcus_hyovaginalis_GCF_000420785\") has been submitted\n",
      "Your job 5481394 (\"merge_Streptococcus_ictaluri_GCF_000188015\") has been submitted\n",
      "Your job 5481395 (\"merge_Streptococcus_infantarius_GCF_000246835\") has been submitted\n",
      "Your job 5481396 (\"merge_Streptococcus_infantis_GCF_000187465\") has been submitted\n",
      "Your job 5481397 (\"merge_Streptococcus_iniae_GCF_000831485\") has been submitted\n",
      "Your job 5481398 (\"merge_Streptococcus_intermedius_GCF_000463355\") has been submitted\n",
      "Your job 5481399 (\"merge_Streptococcus_lutetiensis_GCF_900475675\") has been submitted\n",
      "Your job 5481400 (\"merge_Streptococcus_macacae_GCF_000187995\") has been submitted\n",
      "Your job 5481401 (\"merge_Streptococcus_macedonicus_GCF_000283635\") has been submitted\n",
      "Your job 5481402 (\"merge_Streptococcus_marimammalium_GCF_000380045\") has been submitted\n",
      "Your job 5481403 (\"merge_Streptococcus_marmotae_GCF_001623565\") has been submitted\n",
      "Your job 5481404 (\"merge_Streptococcus_massiliensis_GCF_000380065\") has been submitted\n",
      "Your job 5481405 (\"merge_Streptococcus_merionis_GCF_000380085\") has been submitted\n",
      "Your job 5481406 (\"merge_Streptococcus_milleri_GCF_900636715\") has been submitted\n",
      "Your job 5481407 (\"merge_Streptococcus_minor_GCF_000377005\") has been submitted\n",
      "Your job 5481408 (\"merge_Streptococcus_mitis_GCF_000027165\") has been submitted\n",
      "Your job 5481409 (\"merge_Streptococcus_mutans_GCF_000007465\") has been submitted\n",
      "Your job 5481410 (\"merge_Streptococcus_oralis_GCF_900637025\") has been submitted\n",
      "Your job 5481411 (\"merge_Streptococcus_orisasini_GCF_001431045\") has been submitted\n",
      "Your job 5481412 (\"merge_Streptococcus_orisratti_GCF_000380105\") has been submitted\n",
      "Your job 5481413 (\"merge_Streptococcus_ovis_GCF_000380125\") has been submitted\n",
      "Your job 5481414 (\"merge_Streptococcus_pantholopis_GCF_001642085\") has been submitted\n",
      "Your job 5481415 (\"merge_Streptococcus_parasanguinis_GCF_000164675\") has been submitted\n",
      "Your job 5481416 (\"merge_Streptococcus_parasuis_GCF_004283785\") has been submitted\n",
      "Your job 5481417 (\"merge_Streptococcus_parauberis_GCF_000213825\") has been submitted\n",
      "Your job 5481418 (\"merge_Streptococcus_pasteurianus_GCF_900478025\") has been submitted\n",
      "Your job 5481419 (\"merge_Streptococcus_penaeicida_GCF_002887775\") has been submitted\n",
      "Your job 5481420 (\"merge_Streptococcus_peroris_GCF_000187585\") has been submitted\n",
      "Your job 5481421 (\"merge_Streptococcus_pharyngis_GCF_007859195\") has been submitted\n",
      "Your job 5481422 (\"merge_Streptococcus_phocae_GCF_000772915\") has been submitted\n",
      "Your job 5481423 (\"merge_Streptococcus_pluranimalium_GCF_003352995\") has been submitted\n",
      "Your job 5481424 (\"merge_Streptococcus_plurextorum_GCF_000423745\") has been submitted\n",
      "Your job 5481425 (\"merge_Streptococcus_porci_GCF_000423765\") has been submitted\n",
      "Your job 5481426 (\"merge_Streptococcus_porcinus_GCF_900475415\") has been submitted\n",
      "Your job 5481427 (\"merge_Streptococcus_pseudopneumoniae_GCF_000221985\") has been submitted\n",
      "Your job 5481428 (\"merge_Streptococcus_pseudoporcinus_GCF_000188035\") has been submitted\n",
      "Your job 5481429 (\"merge_Streptococcus_pyogenes_GCF_000006785\") has been submitted\n",
      "Your job 5481430 (\"merge_Streptococcus_ratti_GCF_000286075\") has been submitted\n",
      "Your job 5481431 (\"merge_Streptococcus_respraculi_GCF_003595525\") has been submitted\n",
      "Your job 5481432 (\"merge_Streptococcus_rubneri_GCF_004785935\") has been submitted\n",
      "Your job 5481433 (\"merge_Streptococcus_ruminantium_GCF_003609975\") has been submitted\n",
      "Your job 5481434 (\"merge_Streptococcus_salivarius_GCF_000785515\") has been submitted\n",
      "Your job 5481435 (\"merge_Streptococcus_sanguinis_GCF_000014205\") has been submitted\n",
      "Your job 5481436 (\"merge_Streptococcus_sinensis_GCF_000767835\") has been submitted\n",
      "Your job 5481437 (\"merge_Streptococcus_sobrinus_GCF_000686605\") has been submitted\n",
      "Your job 5481438 (\"merge_Streptococcus_sp_GCF_003086355\") has been submitted\n",
      "Your job 5481439 (\"merge_Streptococcus_suis_GCF_000026745\") has been submitted\n",
      "Your job 5481440 (\"merge_Streptococcus_thermophilus_GCF_000253395\") has been submitted\n",
      "Your job 5481441 (\"merge_Streptococcus_thoraltensis_GCF_000380145\") has been submitted\n",
      "Your job 5481442 (\"merge_Streptococcus_timonensis_GCF_900095845\") has been submitted\n",
      "Your job 5481443 (\"merge_Streptococcus_troglodytae_GCF_002355215\") has been submitted\n",
      "Your job 5481444 (\"merge_Streptococcus_uberis_GCF_000009545\") has been submitted\n",
      "Your job 5481445 (\"merge_Streptococcus_urinalis_GCF_000188055\") has been submitted\n",
      "Your job 5481446 (\"merge_Streptococcus_varani_GCF_001375655\") has been submitted\n",
      "Your job 5481447 (\"merge_Streptococcus_vestibularis_GCF_900636445\") has been submitted\n",
      "Your job 5481448 (\"merge_Streptococcus_viridans_GCF_900636365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 205 sequences from Streptococcus_acidominimus_GCF_900187245_merged.bed.  Filtered 117 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 88.\n",
      "Screened 203 sequences from Streptococcus_agalactiae_GCF_000007265_merged.bed.  Filtered 114 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 211 sequences from Streptococcus_alactolyticus_GCF_009695625_merged.bed.  Filtered 113 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 98.\n",
      "Screened 275 sequences from Streptococcus_anginosus_GCF_000463505_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 578 sequences from Streptococcus_australis_GCF_900476055_merged.bed.  Filtered 298 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 280.\n",
      "Screened 162 sequences from Streptococcus_azizii_GCF_001984715_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 74.\n",
      "Screened 174 sequences from Streptococcus_bovimastitidis_GCF_001885095_merged.bed.  Filtered 98 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 76.\n",
      "Screened 220 sequences from Streptococcus_caballi_GCF_000379985_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 100.\n",
      "Screened 193 sequences from Streptococcus_canis_GCF_900636575_merged.bed.  Filtered 102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 91.\n",
      "Screened 168 sequences from Streptococcus_castoreus_GCF_000425025_merged.bed.  Filtered 99 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 285 sequences from Streptococcus_constellatus_GCF_000463425_merged.bed.  Filtered 167 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 118.\n",
      "Screened 134 sequences from Streptococcus_criceti_GCF_000187975_merged.bed.  Filtered 86 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 48.\n",
      "Screened 497 sequences from Streptococcus_cristatus_GCF_000385925_merged.bed.  Filtered 241 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 256.\n",
      "Screened 217 sequences from Streptococcus_cuniculi_GCF_001921845_merged.bed.  Filtered 123 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 178 sequences from Streptococcus_danieliae_GCF_009767945_merged.bed.  Filtered 99 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 111 sequences from Streptococcus_devriesei_GCF_000423725_merged.bed.  Filtered 62 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 49.\n",
      "Screened 147 sequences from Streptococcus_didelphis_GCF_000380005_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 59.\n",
      "Screened 152 sequences from Streptococcus_downei_GCF_900459175_merged.bed.  Filtered 95 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 57.\n",
      "Screened 223 sequences from Streptococcus_dysgalactiae_GCF_000317855_merged.bed.  Filtered 129 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 134 sequences from Streptococcus_entericus_GCF_000380025_merged.bed.  Filtered 83 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 51.\n",
      "Screened 155 sequences from Streptococcus_equi_GCF_000026605_merged.bed.  Filtered 82 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 73.\n",
      "Screened 235 sequences from Streptococcus_equinus_GCF_000964315_merged.bed.  Filtered 125 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 135 sequences from Streptococcus_ferus_GCF_000372425_merged.bed.  Filtered 89 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 46.\n",
      "Screened 236 sequences from Streptococcus_gallolyticus_GCF_002000985_merged.bed.  Filtered 123 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 435 sequences from Streptococcus_gordonii_GCF_000017005_merged.bed.  Filtered 220 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 215.\n",
      "Screened 132 sequences from Streptococcus_halichoeri_GCF_009870755_merged.bed.  Filtered 80 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 2296 sequences from Streptococcus_halitosis_GCF_003143695_merged.bed.  Filtered 860 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1436.\n",
      "Screened 170 sequences from Streptococcus_halotolerans_GCF_001598035_merged.bed.  Filtered 85 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 270 sequences from Streptococcus_henryi_GCF_000376985_merged.bed.  Filtered 168 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102.\n",
      "Screened 249 sequences from Streptococcus_himalayensis_GCF_001708305_merged.bed.  Filtered 140 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 163 sequences from Streptococcus_hongkongensis_GCF_000785845_merged.bed.  Filtered 94 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 200 sequences from Streptococcus_hyointestinalis_GCF_900459405_merged.bed.  Filtered 102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 98.\n",
      "Screened 182 sequences from Streptococcus_hyovaginalis_GCF_000420785_merged.bed.  Filtered 103 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 165 sequences from Streptococcus_ictaluri_GCF_000188015_merged.bed.  Filtered 86 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 242 sequences from Streptococcus_infantarius_GCF_000246835_merged.bed.  Filtered 137 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 105.\n",
      "Screened 1375 sequences from Streptococcus_infantis_GCF_000187465_merged.bed.  Filtered 707 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 668.\n",
      "Screened 169 sequences from Streptococcus_iniae_GCF_000831485_merged.bed.  Filtered 90 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 253 sequences from Streptococcus_intermedius_GCF_000463355_merged.bed.  Filtered 133 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 120.\n",
      "Screened 240 sequences from Streptococcus_lutetiensis_GCF_900475675_merged.bed.  Filtered 129 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 98 sequences from Streptococcus_macacae_GCF_000187995_merged.bed.  Filtered 55 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 238 sequences from Streptococcus_macedonicus_GCF_000283635_merged.bed.  Filtered 125 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 98 sequences from Streptococcus_marimammalium_GCF_000380045_merged.bed.  Filtered 54 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 44.\n",
      "Screened 217 sequences from Streptococcus_marmotae_GCF_001623565_merged.bed.  Filtered 116 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 101.\n",
      "Screened 225 sequences from Streptococcus_massiliensis_GCF_000380065_merged.bed.  Filtered 115 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 178 sequences from Streptococcus_merionis_GCF_000380085_merged.bed.  Filtered 109 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 291 sequences from Streptococcus_milleri_GCF_900636715_merged.bed.  Filtered 160 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 166 sequences from Streptococcus_minor_GCF_000377005_merged.bed.  Filtered 101 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 65.\n",
      "Screened 1639 sequences from Streptococcus_mitis_GCF_000027165_merged.bed.  Filtered 304 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1335.\n",
      "Screened 159 sequences from Streptococcus_mutans_GCF_000007465_merged.bed.  Filtered 98 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 61.\n",
      "Screened 2276 sequences from Streptococcus_oralis_GCF_900637025_merged.bed.  Filtered 812 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1464.\n",
      "Screened 161 sequences from Streptococcus_orisasini_GCF_001431045_merged.bed.  Filtered 100 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 61.\n",
      "Screened 221 sequences from Streptococcus_orisratti_GCF_000380105_merged.bed.  Filtered 131 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 90.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 233 sequences from Streptococcus_ovis_GCF_000380125_merged.bed.  Filtered 127 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 106.\n",
      "Screened 70 sequences from Streptococcus_pantholopis_GCF_001642085_merged.bed.  Filtered 39 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 31.\n",
      "Screened 505 sequences from Streptococcus_parasanguinis_GCF_000164675_merged.bed.  Filtered 260 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 245.\n",
      "Screened 245 sequences from Streptococcus_parasuis_GCF_004283785_merged.bed.  Filtered 138 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 157 sequences from Streptococcus_parauberis_GCF_000213825_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 245 sequences from Streptococcus_pasteurianus_GCF_900478025_merged.bed.  Filtered 130 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 115.\n",
      "Screened 186 sequences from Streptococcus_penaeicida_GCF_002887775_merged.bed.  Filtered 111 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 1261 sequences from Streptococcus_peroris_GCF_000187585_merged.bed.  Filtered 645 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 616.\n",
      "Screened 160 sequences from Streptococcus_pharyngis_GCF_007859195_merged.bed.  Filtered 91 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 145 sequences from Streptococcus_phocae_GCF_000772915_merged.bed.  Filtered 73 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 72.\n",
      "Screened 211 sequences from Streptococcus_pluranimalium_GCF_003352995_merged.bed.  Filtered 112 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 99.\n",
      "Screened 208 sequences from Streptococcus_plurextorum_GCF_000423745_merged.bed.  Filtered 128 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 219 sequences from Streptococcus_porci_GCF_000423765_merged.bed.  Filtered 134 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 143 sequences from Streptococcus_porcinus_GCF_900475415_merged.bed.  Filtered 74 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 1299 sequences from Streptococcus_pseudopneumoniae_GCF_000221985_merged.bed.  Filtered 225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1074.\n",
      "Screened 160 sequences from Streptococcus_pseudoporcinus_GCF_000188035_merged.bed.  Filtered 92 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 68.\n",
      "Screened 213 sequences from Streptococcus_pyogenes_GCF_000006785_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 93.\n",
      "Screened 148 sequences from Streptococcus_ratti_GCF_000286075_merged.bed.  Filtered 86 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 62.\n",
      "Screened 224 sequences from Streptococcus_respraculi_GCF_003595525_merged.bed.  Filtered 130 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 581 sequences from Streptococcus_rubneri_GCF_004785935_merged.bed.  Filtered 296 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 285.\n",
      "Screened 197 sequences from Streptococcus_ruminantium_GCF_003609975_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 77.\n",
      "Screened 313 sequences from Streptococcus_salivarius_GCF_000785515_merged.bed.  Filtered 154 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 159.\n",
      "Screened 363 sequences from Streptococcus_sanguinis_GCF_000014205_merged.bed.  Filtered 199 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 164.\n",
      "Screened 434 sequences from Streptococcus_sinensis_GCF_000767835_merged.bed.  Filtered 236 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 198.\n",
      "Screened 124 sequences from Streptococcus_sobrinus_GCF_000686605_merged.bed.  Filtered 74 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 50.\n",
      "Screened 77 sequences from Streptococcus_sp_GCF_003086355_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 34.\n",
      "Screened 252 sequences from Streptococcus_suis_GCF_000026745_merged.bed.  Filtered 135 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 289 sequences from Streptococcus_thermophilus_GCF_000253395_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 187 sequences from Streptococcus_thoraltensis_GCF_000380145_merged.bed.  Filtered 102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 1252 sequences from Streptococcus_timonensis_GCF_900095845_merged.bed.  Filtered 644 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 608.\n",
      "Screened 143 sequences from Streptococcus_troglodytae_GCF_002355215_merged.bed.  Filtered 87 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 56.\n",
      "Screened 158 sequences from Streptococcus_uberis_GCF_000009545_merged.bed.  Filtered 80 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 78.\n",
      "Screened 166 sequences from Streptococcus_urinalis_GCF_000188055_merged.bed.  Filtered 91 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 211 sequences from Streptococcus_varani_GCF_001375655_merged.bed.  Filtered 133 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 78.\n",
      "Screened 346 sequences from Streptococcus_vestibularis_GCF_900636445_merged.bed.  Filtered 181 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 165.\n",
      "Screened 510 sequences from Streptococcus_viridans_GCF_900636365_merged.bed.  Filtered 254 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 256.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_merged.bed --twobit results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.2bit --output results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_stripped.bed;\n",
      "streptococcus_acidominimus_gcf_900187245.\n",
      "streptococcus_agalactiae_gcf_000007265.\n",
      "streptococcus_alactolyticus_gcf_009695625.\n",
      "streptococcus_anginosus_gcf_000463505.\n",
      "streptococcus_australis_gcf_900476055.\n",
      "streptococcus_azizii_gcf_001984715.\n",
      "streptococcus_bovimastitidis_gcf_001885095.\n",
      "streptococcus_caballi_gcf_000379985.\n",
      "streptococcus_canis_gcf_900636575.\n",
      "streptococcus_castoreus_gcf_000425025.\n",
      "streptococcus_constellatus_gcf_000463425.\n",
      "streptococcus_criceti_gcf_000187975.\n",
      "streptococcus_cristatus_gcf_000385925.\n",
      "streptococcus_cuniculi_gcf_001921845.\n",
      "streptococcus_danieliae_gcf_009767945.\n",
      "streptococcus_devriesei_gcf_000423725.\n",
      "streptococcus_didelphis_gcf_000380005.\n",
      "streptococcus_downei_gcf_900459175.\n",
      "streptococcus_dysgalactiae_gcf_000317855.\n",
      "streptococcus_entericus_gcf_000380025.\n",
      "streptococcus_equi_gcf_000026605.\n",
      "streptococcus_equinus_gcf_000964315.\n",
      "streptococcus_ferus_gcf_000372425.\n",
      "streptococcus_gallolyticus_gcf_002000985.\n",
      "streptococcus_gordonii_gcf_000017005.\n",
      "streptococcus_halichoeri_gcf_009870755.\n",
      "streptococcus_halitosis_gcf_003143695..\n",
      "streptococcus_halotolerans_gcf_001598035.\n",
      "streptococcus_henryi_gcf_000376985.\n",
      "streptococcus_himalayensis_gcf_001708305.\n",
      "streptococcus_hongkongensis_gcf_000785845.\n",
      "streptococcus_hyointestinalis_gcf_900459405.\n",
      "streptococcus_hyovaginalis_gcf_000420785.\n",
      "streptococcus_ictaluri_gcf_000188015.\n",
      "streptococcus_infantarius_gcf_000246835.\n",
      "streptococcus_infantis_gcf_000187465.\n",
      "streptococcus_iniae_gcf_000831485.\n",
      "streptococcus_intermedius_gcf_000463355.\n",
      "streptococcus_lutetiensis_gcf_900475675.\n",
      "streptococcus_macacae_gcf_000187995.\n",
      "streptococcus_macedonicus_gcf_000283635.\n",
      "streptococcus_marimammalium_gcf_000380045.\n",
      "streptococcus_marmotae_gcf_001623565.\n",
      "streptococcus_massiliensis_gcf_000380065.\n",
      "streptococcus_merionis_gcf_000380085.\n",
      "streptococcus_milleri_gcf_900636715.\n",
      "streptococcus_minor_gcf_000377005.\n",
      "streptococcus_mitis_gcf_000027165..\n",
      "streptococcus_mutans_gcf_000007465.\n",
      "streptococcus_oralis_gcf_900637025..\n",
      "streptococcus_orisasini_gcf_001431045.\n",
      "streptococcus_orisratti_gcf_000380105.\n",
      "streptococcus_ovis_gcf_000380125.\n",
      "streptococcus_pantholopis_gcf_001642085.\n",
      "streptococcus_parasanguinis_gcf_000164675.\n",
      "streptococcus_parasuis_gcf_004283785.\n",
      "streptococcus_parauberis_gcf_000213825.\n",
      "streptococcus_pasteurianus_gcf_900478025.\n",
      "streptococcus_penaeicida_gcf_002887775.\n",
      "streptococcus_peroris_gcf_000187585.\n",
      "streptococcus_pharyngis_gcf_007859195.\n",
      "streptococcus_phocae_gcf_000772915.\n",
      "streptococcus_pluranimalium_gcf_003352995.\n",
      "streptococcus_plurextorum_gcf_000423745.\n",
      "streptococcus_porci_gcf_000423765.\n",
      "streptococcus_porcinus_gcf_900475415.\n",
      "streptococcus_pseudopneumoniae_gcf_000221985..\n",
      "streptococcus_pseudoporcinus_gcf_000188035.\n",
      "streptococcus_pyogenes_gcf_000006785.\n",
      "streptococcus_ratti_gcf_000286075.\n",
      "streptococcus_respraculi_gcf_003595525.\n",
      "streptococcus_rubneri_gcf_004785935.\n",
      "streptococcus_ruminantium_gcf_003609975.\n",
      "streptococcus_salivarius_gcf_000785515.\n",
      "streptococcus_sanguinis_gcf_000014205.\n",
      "streptococcus_sinensis_gcf_000767835.\n",
      "streptococcus_sobrinus_gcf_000686605.\n",
      "streptococcus_sp_gcf_003086355.\n",
      "streptococcus_suis_gcf_000026745.\n",
      "streptococcus_thermophilus_gcf_000253395.\n",
      "streptococcus_thoraltensis_gcf_000380145.\n",
      "streptococcus_timonensis_gcf_900095845.\n",
      "streptococcus_troglodytae_gcf_002355215.\n",
      "streptococcus_uberis_gcf_000009545.\n",
      "streptococcus_urinalis_gcf_000188055.\n",
      "streptococcus_varani_gcf_001375655.\n",
      "streptococcus_vestibularis_gcf_900636445.\n",
      "streptococcus_viridans_gcf_900636365.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_merged.bed --twobit results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.2bit --output results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_stripped.bed;\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 0 taxa:\t2,656.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 1 taxa:\t2,656.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 2 taxa:\t2,359.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 3 taxa:\t1,812.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 4 taxa:\t1,506.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 5 taxa:\t908.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 6 taxa:\t723.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 7 taxa:\t583.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 8 taxa:\t424.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 9 taxa:\t373.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 10 taxa:\t336.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 11 taxa:\t292.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 12 taxa:\t264.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 13 taxa:\t242.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 14 taxa:\t224.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 15 taxa:\t213.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 16 taxa:\t203.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 17 taxa:\t196.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 18 taxa:\t190.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 19 taxa:\t177.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 20 taxa:\t168.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 21 taxa:\t166.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 22 taxa:\t161.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 23 taxa:\t154.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 24 taxa:\t149.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 25 taxa:\t146.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 26 taxa:\t142.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 27 taxa:\t137.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 28 taxa:\t135.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 29 taxa:\t132.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 30 taxa:\t127.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 31 taxa:\t127.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 32 taxa:\t125.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 33 taxa:\t123.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 34 taxa:\t120.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 35 taxa:\t115.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 36 taxa:\t114.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 37 taxa:\t113.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 38 taxa:\t110.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 39 taxa:\t108.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 40 taxa:\t106.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 41 taxa:\t100.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 42 taxa:\t99.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 43 taxa:\t97.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 44 taxa:\t92.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 45 taxa:\t90.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 46 taxa:\t89.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 47 taxa:\t88.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 48 taxa:\t84.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 49 taxa:\t82.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 50 taxa:\t82.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 51 taxa:\t78.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 52 taxa:\t76.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 53 taxa:\t73.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 54 taxa:\t72.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 55 taxa:\t69.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 56 taxa:\t69.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 57 taxa:\t65.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 58 taxa:\t63.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 59 taxa:\t61.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 60 taxa:\t59.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 61 taxa:\t57.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 62 taxa:\t55.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 63 taxa:\t54.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 64 taxa:\t53.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 65 taxa:\t51.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 66 taxa:\t51.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 67 taxa:\t49.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 68 taxa:\t48.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 69 taxa:\t44.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 70 taxa:\t42.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 71 taxa:\t40.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 72 taxa:\t39.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 73 taxa:\t36.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 74 taxa:\t34.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 75 taxa:\t31.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 76 taxa:\t31.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 77 taxa:\t30.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 78 taxa:\t27.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 79 taxa:\t24.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 80 taxa:\t22.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 81 taxa:\t18.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 82 taxa:\t17.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 83 taxa:\t17.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 84 taxa:\t16.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 85 taxa:\t14.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 86 taxa:\t9.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 87 taxa:\t6.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 88 taxa:\t3.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/streptococcus/initial_intervals/streptococcus-to-Streptococcus_pneumoniae_GCF_000007045.sqlite --base-taxon Streptococcus_pneumoniae_GCF_000007045 --output results/phyluce/streptococcus/initial_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.bed --specific-counts 41;\n",
      "Counter({'streptococcus_oralis_gcf_900637025': 100, 'streptococcus_pseudopneumoniae_gcf_000221985': 100, 'streptococcus_mitis_gcf_000027165': 100, 'streptococcus_timonensis_gcf_900095845': 99, 'streptococcus_halitosis_gcf_003143695': 99, 'streptococcus_parasanguinis_gcf_000164675': 99, 'streptococcus_infantis_gcf_000187465': 98, 'streptococcus_gordonii_gcf_000017005': 98, 'streptococcus_viridans_gcf_900636365': 97, 'streptococcus_cristatus_gcf_000385925': 96, 'streptococcus_salivarius_gcf_000785515': 95, 'streptococcus_peroris_gcf_000187585': 95, 'streptococcus_gallolyticus_gcf_002000985': 94, 'streptococcus_infantarius_gcf_000246835': 94, 'streptococcus_sanguinis_gcf_000014205': 94, 'streptococcus_australis_gcf_900476055': 93, 'streptococcus_rubneri_gcf_004785935': 93, 'streptococcus_sinensis_gcf_000767835': 92, 'streptococcus_thermophilus_gcf_000253395': 92, 'streptococcus_vestibularis_gcf_900636445': 91, 'streptococcus_macedonicus_gcf_000283635': 91, 'streptococcus_himalayensis_gcf_001708305': 89, 'streptococcus_equinus_gcf_000964315': 88, 'streptococcus_suis_gcf_000026745': 88, 'streptococcus_marmotae_gcf_001623565': 87, 'streptococcus_milleri_gcf_900636715': 86, 'streptococcus_pasteurianus_gcf_900478025': 86, 'streptococcus_lutetiensis_gcf_900475675': 86, 'streptococcus_anginosus_gcf_000463505': 85, 'streptococcus_constellatus_gcf_000463425': 85, 'streptococcus_parasuis_gcf_004283785': 84, 'streptococcus_pyogenes_gcf_000006785': 83, 'streptococcus_ovis_gcf_000380125': 82, 'streptococcus_intermedius_gcf_000463355': 82, 'streptococcus_dysgalactiae_gcf_000317855': 82, 'streptococcus_pluranimalium_gcf_003352995': 82, 'streptococcus_canis_gcf_900636575': 82, 'streptococcus_respraculi_gcf_003595525': 81, 'streptococcus_hyointestinalis_gcf_900459405': 80, 'streptococcus_henryi_gcf_000376985': 80, 'streptococcus_cuniculi_gcf_001921845': 79, 'streptococcus_caballi_gcf_000379985': 78, 'streptococcus_ictaluri_gcf_000188015': 77, 'streptococcus_massiliensis_gcf_000380065': 76, 'streptococcus_orisratti_gcf_000380105': 76, 'streptococcus_plurextorum_gcf_000423745': 75, 'streptococcus_porci_gcf_000423765': 74, 'streptococcus_uberis_gcf_000009545': 72, 'streptococcus_iniae_gcf_000831485': 72, 'streptococcus_ruminantium_gcf_003609975': 71, 'streptococcus_alactolyticus_gcf_009695625': 71, 'streptococcus_halotolerans_gcf_001598035': 71, 'streptococcus_agalactiae_gcf_000007265': 70, 'streptococcus_acidominimus_gcf_900187245': 70, 'streptococcus_urinalis_gcf_000188055': 68, 'streptococcus_hyovaginalis_gcf_000420785': 68, 'streptococcus_varani_gcf_001375655': 67, 'streptococcus_phocae_gcf_000772915': 67, 'streptococcus_penaeicida_gcf_002887775': 67, 'streptococcus_pharyngis_gcf_007859195': 66, 'streptococcus_thoraltensis_gcf_000380145': 65, 'streptococcus_bovimastitidis_gcf_001885095': 65, 'streptococcus_danieliae_gcf_009767945': 64, 'streptococcus_azizii_gcf_001984715': 63, 'streptococcus_castoreus_gcf_000425025': 63, 'streptococcus_parauberis_gcf_000213825': 61, 'streptococcus_pseudoporcinus_gcf_000188035': 61, 'streptococcus_hongkongensis_gcf_000785845': 61, 'streptococcus_porcinus_gcf_900475415': 61, 'streptococcus_equi_gcf_000026605': 60, 'streptococcus_minor_gcf_000377005': 60, 'streptococcus_merionis_gcf_000380085': 59, 'streptococcus_didelphis_gcf_000380005': 58, 'streptococcus_orisasini_gcf_001431045': 56, 'streptococcus_sobrinus_gcf_000686605': 54, 'streptococcus_downei_gcf_900459175': 52, 'streptococcus_mutans_gcf_000007465': 51, 'streptococcus_devriesei_gcf_000423725': 51, 'streptococcus_troglodytae_gcf_002355215': 48, 'streptococcus_ratti_gcf_000286075': 48, 'streptococcus_halichoeri_gcf_009870755': 47, 'streptococcus_entericus_gcf_000380025': 46, 'streptococcus_ferus_gcf_000372425': 45, 'streptococcus_criceti_gcf_000187975': 43, 'streptococcus_marimammalium_gcf_000380045': 41, 'streptococcus_macacae_gcf_000187995': 33, 'streptococcus_sp_gcf_003086355': 25, 'streptococcus_pantholopis_gcf_001642085': 22})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 41\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/streptococcus/initial_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.bed --twobit results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.2bit --buffer-to 160 --output results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.fasta;\n",
      "Screened 100 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 100.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.fasta --probe-prefix uce_streptococcus_ --design streptococcus_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "\n",
      "\n",
      "\n",
      "Conserved locus count = 100\n",
      "Probe Count = 200\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas --query results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  15:38:46\n",
      "Ended:  Thu Feb 06, 2020  15:38:46\n",
      "Time for execution:  0.00281981627146 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas                        --lastz results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes_vself.lastz                       --probe-prefix=uce_streptococcus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 199 fasta sequences.  Filtered 9 duplicates. Kept 182.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas --scaffoldlist Streptococcus_acidominimus_GCF_900187245 Streptococcus_agalactiae_GCF_000007265 Streptococcus_alactolyticus_GCF_009695625 Streptococcus_anginosus_GCF_000463505 Streptococcus_australis_GCF_900476055 Streptococcus_azizii_GCF_001984715 Streptococcus_bovimastitidis_GCF_001885095 Streptococcus_caballi_GCF_000379985 Streptococcus_canis_GCF_900636575 Streptococcus_castoreus_GCF_000425025 Streptococcus_constellatus_GCF_000463425 Streptococcus_criceti_GCF_000187975 Streptococcus_cristatus_GCF_000385925 Streptococcus_cuniculi_GCF_001921845 Streptococcus_danieliae_GCF_009767945 Streptococcus_devriesei_GCF_000423725 Streptococcus_didelphis_GCF_000380005 Streptococcus_downei_GCF_900459175 Streptococcus_dysgalactiae_GCF_000317855 Streptococcus_entericus_GCF_000380025 Streptococcus_equi_GCF_000026605 Streptococcus_equinus_GCF_000964315 Streptococcus_ferus_GCF_000372425 Streptococcus_gallolyticus_GCF_002000985 Streptococcus_gordonii_GCF_000017005 Streptococcus_halichoeri_GCF_009870755 Streptococcus_halitosis_GCF_003143695 Streptococcus_halotolerans_GCF_001598035 Streptococcus_henryi_GCF_000376985 Streptococcus_himalayensis_GCF_001708305 Streptococcus_hongkongensis_GCF_000785845 Streptococcus_hyointestinalis_GCF_900459405 Streptococcus_hyovaginalis_GCF_000420785 Streptococcus_ictaluri_GCF_000188015 Streptococcus_infantarius_GCF_000246835 Streptococcus_infantis_GCF_000187465 Streptococcus_iniae_GCF_000831485 Streptococcus_intermedius_GCF_000463355 Streptococcus_lutetiensis_GCF_900475675 Streptococcus_macacae_GCF_000187995 Streptococcus_macedonicus_GCF_000283635 Streptococcus_marimammalium_GCF_000380045 Streptococcus_marmotae_GCF_001623565 Streptococcus_massiliensis_GCF_000380065 Streptococcus_merionis_GCF_000380085 Streptococcus_milleri_GCF_900636715 Streptococcus_minor_GCF_000377005 Streptococcus_mitis_GCF_000027165 Streptococcus_mutans_GCF_000007465 Streptococcus_oralis_GCF_900637025 Streptococcus_orisasini_GCF_001431045 Streptococcus_orisratti_GCF_000380105 Streptococcus_ovis_GCF_000380125 Streptococcus_pantholopis_GCF_001642085 Streptococcus_parasanguinis_GCF_000164675 Streptococcus_parasuis_GCF_004283785 Streptococcus_parauberis_GCF_000213825 Streptococcus_pasteurianus_GCF_900478025 Streptococcus_penaeicida_GCF_002887775 Streptococcus_peroris_GCF_000187585 Streptococcus_pharyngis_GCF_007859195 Streptococcus_phocae_GCF_000772915 Streptococcus_pluranimalium_GCF_003352995 Streptococcus_plurextorum_GCF_000423745 Streptococcus_porci_GCF_000423765 Streptococcus_porcinus_GCF_900475415 Streptococcus_pseudopneumoniae_GCF_000221985 Streptococcus_pseudoporcinus_GCF_000188035 Streptococcus_pyogenes_GCF_000006785 Streptococcus_ratti_GCF_000286075 Streptococcus_respraculi_GCF_003595525 Streptococcus_rubneri_GCF_004785935 Streptococcus_ruminantium_GCF_003609975 Streptococcus_salivarius_GCF_000785515 Streptococcus_sanguinis_GCF_000014205 Streptococcus_sinensis_GCF_000767835 Streptococcus_sobrinus_GCF_000686605 Streptococcus_sp_GCF_003086355 Streptococcus_suis_GCF_000026745 Streptococcus_thermophilus_GCF_000253395 Streptococcus_thoraltensis_GCF_000380145 Streptococcus_timonensis_GCF_900095845 Streptococcus_troglodytae_GCF_002355215 Streptococcus_uberis_GCF_000009545 Streptococcus_urinalis_GCF_000188055 Streptococcus_varani_GCF_001375655 Streptococcus_vestibularis_GCF_900636445 Streptococcus_viridans_GCF_900636365 Streptococcus_pneumoniae_GCF_000007045 --genome-base-path results/phyluce/streptococcus/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/streptococcus/validate_intervals/streptococcus-to-Streptococcus_pneumoniae_GCF_000007045.sqlite --output results/phyluce/streptococcus/validate_intervals/lastz/;\n",
      "\n",
      "Running against Streptococcus_acidominimus_GCF_900187245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNzOKsa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7DKdqk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_acidominimus_GCF_900187245.lastz\n",
      "Creating Streptococcus_acidominimus_GCF_900187245 table\n",
      "Inserting data to Streptococcus_acidominimus_GCF_900187245 table\n",
      "\n",
      "Running against Streptococcus_agalactiae_GCF_000007265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVKB9xh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbz9Kiv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_agalactiae_GCF_000007265.lastz\n",
      "Creating Streptococcus_agalactiae_GCF_000007265 table\n",
      "Inserting data to Streptococcus_agalactiae_GCF_000007265 table\n",
      "\n",
      "Running against Streptococcus_alactolyticus_GCF_009695625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptwi_Q4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmph5A5bG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_alactolyticus_GCF_009695625.lastz\n",
      "Creating Streptococcus_alactolyticus_GCF_009695625 table\n",
      "Inserting data to Streptococcus_alactolyticus_GCF_009695625 table\n",
      "\n",
      "Running against Streptococcus_anginosus_GCF_000463505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpA7ZXrc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1NW4IF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_anginosus_GCF_000463505.lastz\n",
      "Creating Streptococcus_anginosus_GCF_000463505 table\n",
      "Inserting data to Streptococcus_anginosus_GCF_000463505 table\n",
      "\n",
      "Running against Streptococcus_australis_GCF_900476055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaS_uFs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEUYUM7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_australis_GCF_900476055.lastz\n",
      "Creating Streptococcus_australis_GCF_900476055 table\n",
      "Inserting data to Streptococcus_australis_GCF_900476055 table\n",
      "\n",
      "Running against Streptococcus_azizii_GCF_001984715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2vtEsx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdp4in_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_azizii_GCF_001984715.lastz\n",
      "Creating Streptococcus_azizii_GCF_001984715 table\n",
      "Inserting data to Streptococcus_azizii_GCF_001984715 table\n",
      "\n",
      "Running against Streptococcus_bovimastitidis_GCF_001885095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7jVi6Y.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9hKhnl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_bovimastitidis_GCF_001885095.lastz\n",
      "Creating Streptococcus_bovimastitidis_GCF_001885095 table\n",
      "Inserting data to Streptococcus_bovimastitidis_GCF_001885095 table\n",
      "\n",
      "Running against Streptococcus_caballi_GCF_000379985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpD6wm2v.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdn8uJC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_caballi_GCF_000379985.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Streptococcus_caballi_GCF_000379985 table\n",
      "Inserting data to Streptococcus_caballi_GCF_000379985 table\n",
      "\n",
      "Running against Streptococcus_canis_GCF_900636575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpz6O8Vl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppVZ5hF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_canis_GCF_900636575.lastz\n",
      "Creating Streptococcus_canis_GCF_900636575 table\n",
      "Inserting data to Streptococcus_canis_GCF_900636575 table\n",
      "\n",
      "Running against Streptococcus_castoreus_GCF_000425025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZxyIro.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8ERmAr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_castoreus_GCF_000425025.lastz\n",
      "Creating Streptococcus_castoreus_GCF_000425025 table\n",
      "Inserting data to Streptococcus_castoreus_GCF_000425025 table\n",
      "\n",
      "Running against Streptococcus_constellatus_GCF_000463425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpq6ZcN3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVqXCUI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_constellatus_GCF_000463425.lastz\n",
      "Creating Streptococcus_constellatus_GCF_000463425 table\n",
      "Inserting data to Streptococcus_constellatus_GCF_000463425 table\n",
      "\n",
      "Running against Streptococcus_criceti_GCF_000187975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpy_7qW_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRR8rhC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_criceti_GCF_000187975.lastz\n",
      "Creating Streptococcus_criceti_GCF_000187975 table\n",
      "Inserting data to Streptococcus_criceti_GCF_000187975 table\n",
      "\n",
      "Running against Streptococcus_cristatus_GCF_000385925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3zPxUf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4XwGIN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_cristatus_GCF_000385925.lastz\n",
      "Creating Streptococcus_cristatus_GCF_000385925 table\n",
      "Inserting data to Streptococcus_cristatus_GCF_000385925 table\n",
      "\n",
      "Running against Streptococcus_cuniculi_GCF_001921845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwKJZEQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpboS5Qu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_cuniculi_GCF_001921845.lastz\n",
      "Creating Streptococcus_cuniculi_GCF_001921845 table\n",
      "Inserting data to Streptococcus_cuniculi_GCF_001921845 table\n",
      "\n",
      "Running against Streptococcus_danieliae_GCF_009767945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6FbUFL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnUf2S1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_danieliae_GCF_009767945.lastz\n",
      "Creating Streptococcus_danieliae_GCF_009767945 table\n",
      "Inserting data to Streptococcus_danieliae_GCF_009767945 table\n",
      "\n",
      "Running against Streptococcus_devriesei_GCF_000423725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQLCnYR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfwrlU1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_devriesei_GCF_000423725.lastz\n",
      "Creating Streptococcus_devriesei_GCF_000423725 table\n",
      "Inserting data to Streptococcus_devriesei_GCF_000423725 table\n",
      "\n",
      "Running against Streptococcus_didelphis_GCF_000380005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxvXgwU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGNRjYN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_didelphis_GCF_000380005.lastz\n",
      "Creating Streptococcus_didelphis_GCF_000380005 table\n",
      "Inserting data to Streptococcus_didelphis_GCF_000380005 table\n",
      "\n",
      "Running against Streptococcus_downei_GCF_900459175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwtLGZx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdNVIww.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_downei_GCF_900459175.lastz\n",
      "Creating Streptococcus_downei_GCF_900459175 table\n",
      "Inserting data to Streptococcus_downei_GCF_900459175 table\n",
      "\n",
      "Running against Streptococcus_dysgalactiae_GCF_000317855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaHMb6K.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp75i_NE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_dysgalactiae_GCF_000317855.lastz\n",
      "Creating Streptococcus_dysgalactiae_GCF_000317855 table\n",
      "Inserting data to Streptococcus_dysgalactiae_GCF_000317855 table\n",
      "\n",
      "Running against Streptococcus_entericus_GCF_000380025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuB8QIB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4HepAP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_entericus_GCF_000380025.lastz\n",
      "Creating Streptococcus_entericus_GCF_000380025 table\n",
      "Inserting data to Streptococcus_entericus_GCF_000380025 table\n",
      "\n",
      "Running against Streptococcus_equi_GCF_000026605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwZEbKm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnqtrPf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_equi_GCF_000026605.lastz\n",
      "Creating Streptococcus_equi_GCF_000026605 table\n",
      "Inserting data to Streptococcus_equi_GCF_000026605 table\n",
      "\n",
      "Running against Streptococcus_equinus_GCF_000964315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOfv_P2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHKdws3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_equinus_GCF_000964315.lastz\n",
      "Creating Streptococcus_equinus_GCF_000964315 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Streptococcus_equinus_GCF_000964315 table\n",
      "\n",
      "Running against Streptococcus_ferus_GCF_000372425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9s3guk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprpK1wS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ferus_GCF_000372425.lastz\n",
      "Creating Streptococcus_ferus_GCF_000372425 table\n",
      "Inserting data to Streptococcus_ferus_GCF_000372425 table\n",
      "\n",
      "Running against Streptococcus_gallolyticus_GCF_002000985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpN6PSFq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_tCpfa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_gallolyticus_GCF_002000985.lastz\n",
      "Creating Streptococcus_gallolyticus_GCF_002000985 table\n",
      "Inserting data to Streptococcus_gallolyticus_GCF_002000985 table\n",
      "\n",
      "Running against Streptococcus_gordonii_GCF_000017005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0tANUP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBAIJy9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_gordonii_GCF_000017005.lastz\n",
      "Creating Streptococcus_gordonii_GCF_000017005 table\n",
      "Inserting data to Streptococcus_gordonii_GCF_000017005 table\n",
      "\n",
      "Running against Streptococcus_halichoeri_GCF_009870755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpeVXY9M.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3B2lHF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_halichoeri_GCF_009870755.lastz\n",
      "Creating Streptococcus_halichoeri_GCF_009870755 table\n",
      "Inserting data to Streptococcus_halichoeri_GCF_009870755 table\n",
      "\n",
      "Running against Streptococcus_halitosis_GCF_003143695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpT77s0k.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVPqXZb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_halitosis_GCF_003143695.lastz\n",
      "Creating Streptococcus_halitosis_GCF_003143695 table\n",
      "Inserting data to Streptococcus_halitosis_GCF_003143695 table\n",
      "\n",
      "Running against Streptococcus_halotolerans_GCF_001598035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7MuMl8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwpW2AH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_halotolerans_GCF_001598035.lastz\n",
      "Creating Streptococcus_halotolerans_GCF_001598035 table\n",
      "Inserting data to Streptococcus_halotolerans_GCF_001598035 table\n",
      "\n",
      "Running against Streptococcus_henryi_GCF_000376985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpC4ILJS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHl3w5K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_henryi_GCF_000376985.lastz\n",
      "Creating Streptococcus_henryi_GCF_000376985 table\n",
      "Inserting data to Streptococcus_henryi_GCF_000376985 table\n",
      "\n",
      "Running against Streptococcus_himalayensis_GCF_001708305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWSNU6O.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdgT8G4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_himalayensis_GCF_001708305.lastz\n",
      "Creating Streptococcus_himalayensis_GCF_001708305 table\n",
      "Inserting data to Streptococcus_himalayensis_GCF_001708305 table\n",
      "\n",
      "Running against Streptococcus_hongkongensis_GCF_000785845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpd75k59.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpX93aYd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_hongkongensis_GCF_000785845.lastz\n",
      "Creating Streptococcus_hongkongensis_GCF_000785845 table\n",
      "Inserting data to Streptococcus_hongkongensis_GCF_000785845 table\n",
      "\n",
      "Running against Streptococcus_hyointestinalis_GCF_900459405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBT8kZp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpplm72N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_hyointestinalis_GCF_900459405.lastz\n",
      "Creating Streptococcus_hyointestinalis_GCF_900459405 table\n",
      "Inserting data to Streptococcus_hyointestinalis_GCF_900459405 table\n",
      "\n",
      "Running against Streptococcus_hyovaginalis_GCF_000420785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0FRcSZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpP3CisV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_hyovaginalis_GCF_000420785.lastz\n",
      "Creating Streptococcus_hyovaginalis_GCF_000420785 table\n",
      "Inserting data to Streptococcus_hyovaginalis_GCF_000420785 table\n",
      "\n",
      "Running against Streptococcus_ictaluri_GCF_000188015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGApkus.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9AOqXT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ictaluri_GCF_000188015.lastz\n",
      "Creating Streptococcus_ictaluri_GCF_000188015 table\n",
      "Inserting data to Streptococcus_ictaluri_GCF_000188015 table\n",
      "\n",
      "Running against Streptococcus_infantarius_GCF_000246835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmVGpg4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp27GAZu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_infantarius_GCF_000246835.lastz\n",
      "Creating Streptococcus_infantarius_GCF_000246835 table\n",
      "Inserting data to Streptococcus_infantarius_GCF_000246835 table\n",
      "\n",
      "Running against Streptococcus_infantis_GCF_000187465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2MzRWQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXJKlyJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_infantis_GCF_000187465.lastz\n",
      "Creating Streptococcus_infantis_GCF_000187465 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Streptococcus_infantis_GCF_000187465 table\n",
      "\n",
      "Running against Streptococcus_iniae_GCF_000831485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJLsAGi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJb2aaC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_iniae_GCF_000831485.lastz\n",
      "Creating Streptococcus_iniae_GCF_000831485 table\n",
      "Inserting data to Streptococcus_iniae_GCF_000831485 table\n",
      "\n",
      "Running against Streptococcus_intermedius_GCF_000463355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp81yovK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzE3xC3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_intermedius_GCF_000463355.lastz\n",
      "Creating Streptococcus_intermedius_GCF_000463355 table\n",
      "Inserting data to Streptococcus_intermedius_GCF_000463355 table\n",
      "\n",
      "Running against Streptococcus_lutetiensis_GCF_900475675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0hSFEV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpY9d9Sf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_lutetiensis_GCF_900475675.lastz\n",
      "Creating Streptococcus_lutetiensis_GCF_900475675 table\n",
      "Inserting data to Streptococcus_lutetiensis_GCF_900475675 table\n",
      "\n",
      "Running against Streptococcus_macacae_GCF_000187995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprLPlb_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAXlZDM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_macacae_GCF_000187995.lastz\n",
      "Creating Streptococcus_macacae_GCF_000187995 table\n",
      "Inserting data to Streptococcus_macacae_GCF_000187995 table\n",
      "\n",
      "Running against Streptococcus_macedonicus_GCF_000283635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpcht_vS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDWfIR3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_macedonicus_GCF_000283635.lastz\n",
      "Creating Streptococcus_macedonicus_GCF_000283635 table\n",
      "Inserting data to Streptococcus_macedonicus_GCF_000283635 table\n",
      "\n",
      "Running against Streptococcus_marimammalium_GCF_000380045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFW7m9y.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl7uQ8Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_marimammalium_GCF_000380045.lastz\n",
      "Creating Streptococcus_marimammalium_GCF_000380045 table\n",
      "Inserting data to Streptococcus_marimammalium_GCF_000380045 table\n",
      "\n",
      "Running against Streptococcus_marmotae_GCF_001623565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0E5Y5L.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfred0D.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_marmotae_GCF_001623565.lastz\n",
      "Creating Streptococcus_marmotae_GCF_001623565 table\n",
      "Inserting data to Streptococcus_marmotae_GCF_001623565 table\n",
      "\n",
      "Running against Streptococcus_massiliensis_GCF_000380065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJkfsv7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9gLuGX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_massiliensis_GCF_000380065.lastz\n",
      "Creating Streptococcus_massiliensis_GCF_000380065 table\n",
      "Inserting data to Streptococcus_massiliensis_GCF_000380065 table\n",
      "\n",
      "Running against Streptococcus_merionis_GCF_000380085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWFKgqd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpo3jV5W.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_merionis_GCF_000380085.lastz\n",
      "Creating Streptococcus_merionis_GCF_000380085 table\n",
      "Inserting data to Streptococcus_merionis_GCF_000380085 table\n",
      "\n",
      "Running against Streptococcus_milleri_GCF_900636715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdGgH4g.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpImEctX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_milleri_GCF_900636715.lastz\n",
      "Creating Streptococcus_milleri_GCF_900636715 table\n",
      "Inserting data to Streptococcus_milleri_GCF_900636715 table\n",
      "\n",
      "Running against Streptococcus_minor_GCF_000377005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpADUrS9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFe2CT5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_minor_GCF_000377005.lastz\n",
      "Creating Streptococcus_minor_GCF_000377005 table\n",
      "Inserting data to Streptococcus_minor_GCF_000377005 table\n",
      "\n",
      "Running against Streptococcus_mitis_GCF_000027165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKyR3bS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7DuTQ5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_mitis_GCF_000027165.lastz\n",
      "Creating Streptococcus_mitis_GCF_000027165 table\n",
      "Inserting data to Streptococcus_mitis_GCF_000027165 table\n",
      "\n",
      "Running against Streptococcus_mutans_GCF_000007465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgcxmhZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE1tRja.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_mutans_GCF_000007465.lastz\n",
      "Creating Streptococcus_mutans_GCF_000007465 table\n",
      "Inserting data to Streptococcus_mutans_GCF_000007465 table\n",
      "\n",
      "Running against Streptococcus_oralis_GCF_900637025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZAJ2Qm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpx0FX5d.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_oralis_GCF_900637025.lastz\n",
      "Creating Streptococcus_oralis_GCF_900637025 table\n",
      "Inserting data to Streptococcus_oralis_GCF_900637025 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Streptococcus_orisasini_GCF_001431045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCxQQss.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxcLn4K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_orisasini_GCF_001431045.lastz\n",
      "Creating Streptococcus_orisasini_GCF_001431045 table\n",
      "Inserting data to Streptococcus_orisasini_GCF_001431045 table\n",
      "\n",
      "Running against Streptococcus_orisratti_GCF_000380105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp09d6mv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAcPUrC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_orisratti_GCF_000380105.lastz\n",
      "Creating Streptococcus_orisratti_GCF_000380105 table\n",
      "Inserting data to Streptococcus_orisratti_GCF_000380105 table\n",
      "\n",
      "Running against Streptococcus_ovis_GCF_000380125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphTkQu6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPkzfTb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ovis_GCF_000380125.lastz\n",
      "Creating Streptococcus_ovis_GCF_000380125 table\n",
      "Inserting data to Streptococcus_ovis_GCF_000380125 table\n",
      "\n",
      "Running against Streptococcus_pantholopis_GCF_001642085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqE0YUX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0pRspH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pantholopis_GCF_001642085.lastz\n",
      "Creating Streptococcus_pantholopis_GCF_001642085 table\n",
      "Inserting data to Streptococcus_pantholopis_GCF_001642085 table\n",
      "\n",
      "Running against Streptococcus_parasanguinis_GCF_000164675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfnJ8vB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQx_SxU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_parasanguinis_GCF_000164675.lastz\n",
      "Creating Streptococcus_parasanguinis_GCF_000164675 table\n",
      "Inserting data to Streptococcus_parasanguinis_GCF_000164675 table\n",
      "\n",
      "Running against Streptococcus_parasuis_GCF_004283785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOuGFhK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6uDiHY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_parasuis_GCF_004283785.lastz\n",
      "Creating Streptococcus_parasuis_GCF_004283785 table\n",
      "Inserting data to Streptococcus_parasuis_GCF_004283785 table\n",
      "\n",
      "Running against Streptococcus_parauberis_GCF_000213825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkOF43V.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpw7exCr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_parauberis_GCF_000213825.lastz\n",
      "Creating Streptococcus_parauberis_GCF_000213825 table\n",
      "Inserting data to Streptococcus_parauberis_GCF_000213825 table\n",
      "\n",
      "Running against Streptococcus_pasteurianus_GCF_900478025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRoeSw4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpg5c8E8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pasteurianus_GCF_900478025.lastz\n",
      "Creating Streptococcus_pasteurianus_GCF_900478025 table\n",
      "Inserting data to Streptococcus_pasteurianus_GCF_900478025 table\n",
      "\n",
      "Running against Streptococcus_penaeicida_GCF_002887775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBd8P8E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcaFefw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_penaeicida_GCF_002887775.lastz\n",
      "Creating Streptococcus_penaeicida_GCF_002887775 table\n",
      "Inserting data to Streptococcus_penaeicida_GCF_002887775 table\n",
      "\n",
      "Running against Streptococcus_peroris_GCF_000187585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHHcIqP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTgFuDJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_peroris_GCF_000187585.lastz\n",
      "Creating Streptococcus_peroris_GCF_000187585 table\n",
      "Inserting data to Streptococcus_peroris_GCF_000187585 table\n",
      "\n",
      "Running against Streptococcus_pharyngis_GCF_007859195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6GqUnU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3vnlZY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pharyngis_GCF_007859195.lastz\n",
      "Creating Streptococcus_pharyngis_GCF_007859195 table\n",
      "Inserting data to Streptococcus_pharyngis_GCF_007859195 table\n",
      "\n",
      "Running against Streptococcus_phocae_GCF_000772915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpneZN_E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcflk93.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_phocae_GCF_000772915.lastz\n",
      "Creating Streptococcus_phocae_GCF_000772915 table\n",
      "Inserting data to Streptococcus_phocae_GCF_000772915 table\n",
      "\n",
      "Running against Streptococcus_pluranimalium_GCF_003352995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpw4zukz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSHbDMS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pluranimalium_GCF_003352995.lastz\n",
      "Creating Streptococcus_pluranimalium_GCF_003352995 table\n",
      "Inserting data to Streptococcus_pluranimalium_GCF_003352995 table\n",
      "\n",
      "Running against Streptococcus_plurextorum_GCF_000423745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpC2QTrj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd1W95L.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_plurextorum_GCF_000423745.lastz\n",
      "Creating Streptococcus_plurextorum_GCF_000423745 table\n",
      "Inserting data to Streptococcus_plurextorum_GCF_000423745 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Streptococcus_porci_GCF_000423765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZPMOyW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6Getaa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_porci_GCF_000423765.lastz\n",
      "Creating Streptococcus_porci_GCF_000423765 table\n",
      "Inserting data to Streptococcus_porci_GCF_000423765 table\n",
      "\n",
      "Running against Streptococcus_porcinus_GCF_900475415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwRfBqW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2bBfpT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_porcinus_GCF_900475415.lastz\n",
      "Creating Streptococcus_porcinus_GCF_900475415 table\n",
      "Inserting data to Streptococcus_porcinus_GCF_900475415 table\n",
      "\n",
      "Running against Streptococcus_pseudopneumoniae_GCF_000221985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpm7w8hr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGHuGdq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pseudopneumoniae_GCF_000221985.lastz\n",
      "Creating Streptococcus_pseudopneumoniae_GCF_000221985 table\n",
      "Inserting data to Streptococcus_pseudopneumoniae_GCF_000221985 table\n",
      "\n",
      "Running against Streptococcus_pseudoporcinus_GCF_000188035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJaurvE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6Nogec.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pseudoporcinus_GCF_000188035.lastz\n",
      "Creating Streptococcus_pseudoporcinus_GCF_000188035 table\n",
      "Inserting data to Streptococcus_pseudoporcinus_GCF_000188035 table\n",
      "\n",
      "Running against Streptococcus_pyogenes_GCF_000006785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7ibKoz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptobWDU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pyogenes_GCF_000006785.lastz\n",
      "Creating Streptococcus_pyogenes_GCF_000006785 table\n",
      "Inserting data to Streptococcus_pyogenes_GCF_000006785 table\n",
      "\n",
      "Running against Streptococcus_ratti_GCF_000286075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxG50tH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKxk9ek.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ratti_GCF_000286075.lastz\n",
      "Creating Streptococcus_ratti_GCF_000286075 table\n",
      "Inserting data to Streptococcus_ratti_GCF_000286075 table\n",
      "\n",
      "Running against Streptococcus_respraculi_GCF_003595525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJu95zM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmps5YHP9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_respraculi_GCF_003595525.lastz\n",
      "Creating Streptococcus_respraculi_GCF_003595525 table\n",
      "Inserting data to Streptococcus_respraculi_GCF_003595525 table\n",
      "\n",
      "Running against Streptococcus_rubneri_GCF_004785935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8i2YFR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpREVpeh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_rubneri_GCF_004785935.lastz\n",
      "Creating Streptococcus_rubneri_GCF_004785935 table\n",
      "Inserting data to Streptococcus_rubneri_GCF_004785935 table\n",
      "\n",
      "Running against Streptococcus_ruminantium_GCF_003609975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprzEeZo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCYkYxR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ruminantium_GCF_003609975.lastz\n",
      "Creating Streptococcus_ruminantium_GCF_003609975 table\n",
      "Inserting data to Streptococcus_ruminantium_GCF_003609975 table\n",
      "\n",
      "Running against Streptococcus_salivarius_GCF_000785515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsaDi2A.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLBZP5f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_salivarius_GCF_000785515.lastz\n",
      "Creating Streptococcus_salivarius_GCF_000785515 table\n",
      "Inserting data to Streptococcus_salivarius_GCF_000785515 table\n",
      "\n",
      "Running against Streptococcus_sanguinis_GCF_000014205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplYdboE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIQjj7C.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sanguinis_GCF_000014205.lastz\n",
      "Creating Streptococcus_sanguinis_GCF_000014205 table\n",
      "Inserting data to Streptococcus_sanguinis_GCF_000014205 table\n",
      "\n",
      "Running against Streptococcus_sinensis_GCF_000767835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_mgLyq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpy2Yhvs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sinensis_GCF_000767835.lastz\n",
      "Creating Streptococcus_sinensis_GCF_000767835 table\n",
      "Inserting data to Streptococcus_sinensis_GCF_000767835 table\n",
      "\n",
      "Running against Streptococcus_sobrinus_GCF_000686605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWfXxeB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqeaCzK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sobrinus_GCF_000686605.lastz\n",
      "Creating Streptococcus_sobrinus_GCF_000686605 table\n",
      "Inserting data to Streptococcus_sobrinus_GCF_000686605 table\n",
      "\n",
      "Running against Streptococcus_sp_GCF_003086355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxnAnPb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8stMo9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sp_GCF_003086355.lastz\n",
      "Creating Streptococcus_sp_GCF_003086355 table\n",
      "Inserting data to Streptococcus_sp_GCF_003086355 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Streptococcus_suis_GCF_000026745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvukvvc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbRqZo3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_suis_GCF_000026745.lastz\n",
      "Creating Streptococcus_suis_GCF_000026745 table\n",
      "Inserting data to Streptococcus_suis_GCF_000026745 table\n",
      "\n",
      "Running against Streptococcus_thermophilus_GCF_000253395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpakGhf3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmRYu_r.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_thermophilus_GCF_000253395.lastz\n",
      "Creating Streptococcus_thermophilus_GCF_000253395 table\n",
      "Inserting data to Streptococcus_thermophilus_GCF_000253395 table\n",
      "\n",
      "Running against Streptococcus_thoraltensis_GCF_000380145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3nAiXI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTttoCw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_thoraltensis_GCF_000380145.lastz\n",
      "Creating Streptococcus_thoraltensis_GCF_000380145 table\n",
      "Inserting data to Streptococcus_thoraltensis_GCF_000380145 table\n",
      "\n",
      "Running against Streptococcus_timonensis_GCF_900095845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTfEaLI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHOLqH2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_timonensis_GCF_900095845.lastz\n",
      "Creating Streptococcus_timonensis_GCF_900095845 table\n",
      "Inserting data to Streptococcus_timonensis_GCF_900095845 table\n",
      "\n",
      "Running against Streptococcus_troglodytae_GCF_002355215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6plCxl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5gLQwD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_troglodytae_GCF_002355215.lastz\n",
      "Creating Streptococcus_troglodytae_GCF_002355215 table\n",
      "Inserting data to Streptococcus_troglodytae_GCF_002355215 table\n",
      "\n",
      "Running against Streptococcus_uberis_GCF_000009545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaC0TID.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJ4zn02.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_uberis_GCF_000009545.lastz\n",
      "Creating Streptococcus_uberis_GCF_000009545 table\n",
      "Inserting data to Streptococcus_uberis_GCF_000009545 table\n",
      "\n",
      "Running against Streptococcus_urinalis_GCF_000188055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmzXlA0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqN8Uiv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_urinalis_GCF_000188055.lastz\n",
      "Creating Streptococcus_urinalis_GCF_000188055 table\n",
      "Inserting data to Streptococcus_urinalis_GCF_000188055 table\n",
      "\n",
      "Running against Streptococcus_varani_GCF_001375655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQEkcuJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGSaf6N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_varani_GCF_001375655.lastz\n",
      "Creating Streptococcus_varani_GCF_001375655 table\n",
      "Inserting data to Streptococcus_varani_GCF_001375655 table\n",
      "\n",
      "Running against Streptococcus_vestibularis_GCF_900636445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnN8F57.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKsbtK5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_vestibularis_GCF_900636445.lastz\n",
      "Creating Streptococcus_vestibularis_GCF_900636445 table\n",
      "Inserting data to Streptococcus_vestibularis_GCF_900636445 table\n",
      "\n",
      "Running against Streptococcus_viridans_GCF_900636365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnuhvtL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfOoUFh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_viridans_GCF_900636365.lastz\n",
      "Creating Streptococcus_viridans_GCF_900636365 table\n",
      "Inserting data to Streptococcus_viridans_GCF_900636365 table\n",
      "\n",
      "Running against Streptococcus_pneumoniae_GCF_000007045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQaOQwB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUUiNM4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pneumoniae_GCF_000007045.lastz\n",
      "Creating Streptococcus_pneumoniae_GCF_000007045 table\n",
      "Inserting data to Streptococcus_pneumoniae_GCF_000007045 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/streptococcus/extract_probes_from_group/streptococcus_genome.conf --lastz results/phyluce/streptococcus/validate_intervals/lastz --probes 120 --probe-prefix uce_streptococcus_ --name-pattern \"Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/streptococcus/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 15:40:38,165 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 15:40:38,174 - Phyluce - INFO - --- Working on Streptococcus_acidominimus_GCF_900187245 genome --\n",
      "2020-02-06 15:40:38,174 - Phyluce - INFO - Reading Streptococcus_acidominimus_GCF_900187245 genome\n",
      "2020-02-06 15:40:39,054 - Phyluce - INFO - Streptococcus_acidominimus_GCF_900187245: 100 uces, 0 dupes, 100 non-dupes, 18 orient drop, 1 length drop, 81 written\n",
      "2020-02-06 15:40:39,054 - Phyluce - INFO - ---- Working on Streptococcus_agalactiae_GCF_000007265 genome ---\n",
      "2020-02-06 15:40:39,055 - Phyluce - INFO - Reading Streptococcus_agalactiae_GCF_000007265 genome\n",
      "2020-02-06 15:40:39,641 - Phyluce - INFO - Streptococcus_agalactiae_GCF_000007265: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 19 length drop, 81 written\n",
      "2020-02-06 15:40:39,641 - Phyluce - INFO - -- Working on Streptococcus_alactolyticus_GCF_009695625 genome --\n",
      "2020-02-06 15:40:39,642 - Phyluce - INFO - Reading Streptococcus_alactolyticus_GCF_009695625 genome\n",
      "2020-02-06 15:40:39,968 - Phyluce - INFO - Streptococcus_alactolyticus_GCF_009695625: 98 uces, 2 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:40:39,968 - Phyluce - INFO - ---- Working on Streptococcus_anginosus_GCF_000463505 genome ----\n",
      "2020-02-06 15:40:39,968 - Phyluce - INFO - Reading Streptococcus_anginosus_GCF_000463505 genome\n",
      "2020-02-06 15:40:40,422 - Phyluce - INFO - Streptococcus_anginosus_GCF_000463505: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:40,422 - Phyluce - INFO - ---- Working on Streptococcus_australis_GCF_900476055 genome ----\n",
      "2020-02-06 15:40:40,438 - Phyluce - INFO - Reading Streptococcus_australis_GCF_900476055 genome\n",
      "2020-02-06 15:40:40,901 - Phyluce - INFO - Streptococcus_australis_GCF_900476055: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:40,901 - Phyluce - INFO - ------ Working on Streptococcus_azizii_GCF_001984715 genome -----\n",
      "2020-02-06 15:40:40,908 - Phyluce - INFO - Reading Streptococcus_azizii_GCF_001984715 genome\n",
      "2020-02-06 15:40:41,252 - Phyluce - INFO - Streptococcus_azizii_GCF_001984715: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:41,252 - Phyluce - INFO - -- Working on Streptococcus_bovimastitidis_GCF_001885095 genome -\n",
      "2020-02-06 15:40:41,252 - Phyluce - INFO - Reading Streptococcus_bovimastitidis_GCF_001885095 genome\n",
      "2020-02-06 15:40:41,605 - Phyluce - INFO - Streptococcus_bovimastitidis_GCF_001885095: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 2 length drop, 91 written\n",
      "2020-02-06 15:40:41,605 - Phyluce - INFO - ----- Working on Streptococcus_caballi_GCF_000379985 genome -----\n",
      "2020-02-06 15:40:41,606 - Phyluce - INFO - Reading Streptococcus_caballi_GCF_000379985 genome\n",
      "2020-02-06 15:40:41,930 - Phyluce - INFO - Streptococcus_caballi_GCF_000379985: 98 uces, 1 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-06 15:40:41,930 - Phyluce - INFO - ------ Working on Streptococcus_canis_GCF_900636575 genome ------\n",
      "2020-02-06 15:40:41,931 - Phyluce - INFO - Reading Streptococcus_canis_GCF_900636575 genome\n",
      "2020-02-06 15:40:42,473 - Phyluce - INFO - Streptococcus_canis_GCF_900636575: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:42,474 - Phyluce - INFO - ---- Working on Streptococcus_castoreus_GCF_000425025 genome ----\n",
      "2020-02-06 15:40:42,474 - Phyluce - INFO - Reading Streptococcus_castoreus_GCF_000425025 genome\n",
      "2020-02-06 15:40:42,778 - Phyluce - INFO - Streptococcus_castoreus_GCF_000425025: 93 uces, 1 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-06 15:40:42,778 - Phyluce - INFO - --- Working on Streptococcus_constellatus_GCF_000463425 genome --\n",
      "2020-02-06 15:40:42,779 - Phyluce - INFO - Reading Streptococcus_constellatus_GCF_000463425 genome\n",
      "2020-02-06 15:40:43,240 - Phyluce - INFO - Streptococcus_constellatus_GCF_000463425: 100 uces, 0 dupes, 100 non-dupes, 17 orient drop, 1 length drop, 82 written\n",
      "2020-02-06 15:40:43,240 - Phyluce - INFO - ----- Working on Streptococcus_criceti_GCF_000187975 genome -----\n",
      "2020-02-06 15:40:43,241 - Phyluce - INFO - Reading Streptococcus_criceti_GCF_000187975 genome\n",
      "2020-02-06 15:40:43,731 - Phyluce - INFO - Streptococcus_criceti_GCF_000187975: 99 uces, 1 dupes, 98 non-dupes, 12 orient drop, 6 length drop, 80 written\n",
      "2020-02-06 15:40:43,731 - Phyluce - INFO - ---- Working on Streptococcus_cristatus_GCF_000385925 genome ----\n",
      "2020-02-06 15:40:43,746 - Phyluce - INFO - Reading Streptococcus_cristatus_GCF_000385925 genome\n",
      "2020-02-06 15:40:44,199 - Phyluce - INFO - Streptococcus_cristatus_GCF_000385925: 100 uces, 0 dupes, 100 non-dupes, 14 orient drop, 3 length drop, 83 written\n",
      "2020-02-06 15:40:44,199 - Phyluce - INFO - ----- Working on Streptococcus_cuniculi_GCF_001921845 genome ----\n",
      "2020-02-06 15:40:44,212 - Phyluce - INFO - Reading Streptococcus_cuniculi_GCF_001921845 genome\n",
      "2020-02-06 15:40:44,549 - Phyluce - INFO - Streptococcus_cuniculi_GCF_001921845: 98 uces, 4 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-06 15:40:44,549 - Phyluce - INFO - ---- Working on Streptococcus_danieliae_GCF_009767945 genome ----\n",
      "2020-02-06 15:40:44,550 - Phyluce - INFO - Reading Streptococcus_danieliae_GCF_009767945 genome\n",
      "2020-02-06 15:40:44,885 - Phyluce - INFO - Streptococcus_danieliae_GCF_009767945: 100 uces, 2 dupes, 98 non-dupes, 0 orient drop, 0 length drop, 98 written\n",
      "2020-02-06 15:40:44,885 - Phyluce - INFO - ---- Working on Streptococcus_devriesei_GCF_000423725 genome ----\n",
      "2020-02-06 15:40:44,885 - Phyluce - INFO - Reading Streptococcus_devriesei_GCF_000423725 genome\n",
      "2020-02-06 15:40:45,187 - Phyluce - INFO - Streptococcus_devriesei_GCF_000423725: 91 uces, 4 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-06 15:40:45,188 - Phyluce - INFO - ---- Working on Streptococcus_didelphis_GCF_000380005 genome ----\n",
      "2020-02-06 15:40:45,209 - Phyluce - INFO - Reading Streptococcus_didelphis_GCF_000380005 genome\n",
      "2020-02-06 15:40:45,561 - Phyluce - INFO - Streptococcus_didelphis_GCF_000380005: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:45,561 - Phyluce - INFO - ------ Working on Streptococcus_downei_GCF_900459175 genome -----\n",
      "2020-02-06 15:40:45,562 - Phyluce - INFO - Reading Streptococcus_downei_GCF_900459175 genome\n",
      "2020-02-06 15:40:46,127 - Phyluce - INFO - Streptococcus_downei_GCF_900459175: 99 uces, 0 dupes, 99 non-dupes, 16 orient drop, 5 length drop, 78 written\n",
      "2020-02-06 15:40:46,127 - Phyluce - INFO - --- Working on Streptococcus_dysgalactiae_GCF_000317855 genome --\n",
      "2020-02-06 15:40:46,143 - Phyluce - INFO - Reading Streptococcus_dysgalactiae_GCF_000317855 genome\n",
      "2020-02-06 15:40:46,648 - Phyluce - INFO - Streptococcus_dysgalactiae_GCF_000317855: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:46,648 - Phyluce - INFO - ---- Working on Streptococcus_entericus_GCF_000380025 genome ----\n",
      "2020-02-06 15:40:46,648 - Phyluce - INFO - Reading Streptococcus_entericus_GCF_000380025 genome\n",
      "2020-02-06 15:40:46,968 - Phyluce - INFO - Streptococcus_entericus_GCF_000380025: 96 uces, 3 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:40:46,968 - Phyluce - INFO - ------- Working on Streptococcus_equi_GCF_000026605 genome ------\n",
      "2020-02-06 15:40:46,968 - Phyluce - INFO - Reading Streptococcus_equi_GCF_000026605 genome\n",
      "2020-02-06 15:40:47,463 - Phyluce - INFO - Streptococcus_equi_GCF_000026605: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:47,463 - Phyluce - INFO - ----- Working on Streptococcus_equinus_GCF_000964315 genome -----\n",
      "2020-02-06 15:40:47,473 - Phyluce - INFO - Reading Streptococcus_equinus_GCF_000964315 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 15:40:48,061 - Phyluce - INFO - Streptococcus_equinus_GCF_000964315: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:48,061 - Phyluce - INFO - ------ Working on Streptococcus_ferus_GCF_000372425 genome ------\n",
      "2020-02-06 15:40:48,061 - Phyluce - INFO - Reading Streptococcus_ferus_GCF_000372425 genome\n",
      "2020-02-06 15:40:48,393 - Phyluce - INFO - Streptococcus_ferus_GCF_000372425: 99 uces, 3 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:40:48,394 - Phyluce - INFO - --- Working on Streptococcus_gallolyticus_GCF_002000985 genome --\n",
      "2020-02-06 15:40:48,394 - Phyluce - INFO - Reading Streptococcus_gallolyticus_GCF_002000985 genome\n",
      "2020-02-06 15:40:48,942 - Phyluce - INFO - Streptococcus_gallolyticus_GCF_002000985: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:48,943 - Phyluce - INFO - ----- Working on Streptococcus_gordonii_GCF_000017005 genome ----\n",
      "2020-02-06 15:40:48,948 - Phyluce - INFO - Reading Streptococcus_gordonii_GCF_000017005 genome\n",
      "2020-02-06 15:40:49,414 - Phyluce - INFO - Streptococcus_gordonii_GCF_000017005: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:49,414 - Phyluce - INFO - ---- Working on Streptococcus_halichoeri_GCF_009870755 genome ---\n",
      "2020-02-06 15:40:49,415 - Phyluce - INFO - Reading Streptococcus_halichoeri_GCF_009870755 genome\n",
      "2020-02-06 15:40:49,745 - Phyluce - INFO - Streptococcus_halichoeri_GCF_009870755: 99 uces, 3 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:40:49,745 - Phyluce - INFO - ---- Working on Streptococcus_halitosis_GCF_003143695 genome ----\n",
      "2020-02-06 15:40:49,745 - Phyluce - INFO - Reading Streptococcus_halitosis_GCF_003143695 genome\n",
      "2020-02-06 15:40:50,079 - Phyluce - INFO - Streptococcus_halitosis_GCF_003143695: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:50,079 - Phyluce - INFO - --- Working on Streptococcus_halotolerans_GCF_001598035 genome --\n",
      "2020-02-06 15:40:50,079 - Phyluce - INFO - Reading Streptococcus_halotolerans_GCF_001598035 genome\n",
      "2020-02-06 15:40:50,613 - Phyluce - INFO - Streptococcus_halotolerans_GCF_001598035: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:50,613 - Phyluce - INFO - ------ Working on Streptococcus_henryi_GCF_000376985 genome -----\n",
      "2020-02-06 15:40:50,632 - Phyluce - INFO - Reading Streptococcus_henryi_GCF_000376985 genome\n",
      "2020-02-06 15:40:50,961 - Phyluce - INFO - Streptococcus_henryi_GCF_000376985: 97 uces, 0 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-06 15:40:50,961 - Phyluce - INFO - --- Working on Streptococcus_himalayensis_GCF_001708305 genome --\n",
      "2020-02-06 15:40:50,962 - Phyluce - INFO - Reading Streptococcus_himalayensis_GCF_001708305 genome\n",
      "2020-02-06 15:40:51,419 - Phyluce - INFO - Streptococcus_himalayensis_GCF_001708305: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n",
      "2020-02-06 15:40:51,419 - Phyluce - INFO - -- Working on Streptococcus_hongkongensis_GCF_000785845 genome --\n",
      "2020-02-06 15:40:51,420 - Phyluce - INFO - Reading Streptococcus_hongkongensis_GCF_000785845 genome\n",
      "2020-02-06 15:40:51,788 - Phyluce - INFO - Streptococcus_hongkongensis_GCF_000785845: 100 uces, 9 dupes, 91 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-06 15:40:51,788 - Phyluce - INFO - - Working on Streptococcus_hyointestinalis_GCF_900459405 genome -\n",
      "2020-02-06 15:40:51,804 - Phyluce - INFO - Reading Streptococcus_hyointestinalis_GCF_900459405 genome\n",
      "2020-02-06 15:40:52,356 - Phyluce - INFO - Streptococcus_hyointestinalis_GCF_900459405: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 4 length drop, 81 written\n",
      "2020-02-06 15:40:52,357 - Phyluce - INFO - --- Working on Streptococcus_hyovaginalis_GCF_000420785 genome --\n",
      "2020-02-06 15:40:52,364 - Phyluce - INFO - Reading Streptococcus_hyovaginalis_GCF_000420785 genome\n",
      "2020-02-06 15:40:52,696 - Phyluce - INFO - Streptococcus_hyovaginalis_GCF_000420785: 97 uces, 4 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:40:52,696 - Phyluce - INFO - ----- Working on Streptococcus_ictaluri_GCF_000188015 genome ----\n",
      "2020-02-06 15:40:52,697 - Phyluce - INFO - Reading Streptococcus_ictaluri_GCF_000188015 genome\n",
      "2020-02-06 15:40:53,109 - Phyluce - INFO - Streptococcus_ictaluri_GCF_000188015: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 16 length drop, 84 written\n",
      "2020-02-06 15:40:53,109 - Phyluce - INFO - --- Working on Streptococcus_infantarius_GCF_000246835 genome ---\n",
      "2020-02-06 15:40:53,109 - Phyluce - INFO - Reading Streptococcus_infantarius_GCF_000246835 genome\n",
      "2020-02-06 15:40:53,643 - Phyluce - INFO - Streptococcus_infantarius_GCF_000246835: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:53,643 - Phyluce - INFO - ----- Working on Streptococcus_infantis_GCF_000187465 genome ----\n",
      "2020-02-06 15:40:53,643 - Phyluce - INFO - Reading Streptococcus_infantis_GCF_000187465 genome\n",
      "2020-02-06 15:40:53,975 - Phyluce - INFO - Streptococcus_infantis_GCF_000187465: 99 uces, 1 dupes, 98 non-dupes, 3 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:53,976 - Phyluce - INFO - ------ Working on Streptococcus_iniae_GCF_000831485 genome ------\n",
      "2020-02-06 15:40:53,996 - Phyluce - INFO - Reading Streptococcus_iniae_GCF_000831485 genome\n",
      "2020-02-06 15:40:54,499 - Phyluce - INFO - Streptococcus_iniae_GCF_000831485: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 5 length drop, 82 written\n",
      "2020-02-06 15:40:54,500 - Phyluce - INFO - --- Working on Streptococcus_intermedius_GCF_000463355 genome ---\n",
      "2020-02-06 15:40:54,500 - Phyluce - INFO - Reading Streptococcus_intermedius_GCF_000463355 genome\n",
      "2020-02-06 15:40:54,954 - Phyluce - INFO - Streptococcus_intermedius_GCF_000463355: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:40:54,954 - Phyluce - INFO - --- Working on Streptococcus_lutetiensis_GCF_900475675 genome ---\n",
      "2020-02-06 15:40:54,963 - Phyluce - INFO - Reading Streptococcus_lutetiensis_GCF_900475675 genome\n",
      "2020-02-06 15:40:55,547 - Phyluce - INFO - Streptococcus_lutetiensis_GCF_900475675: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 19 length drop, 81 written\n",
      "2020-02-06 15:40:55,548 - Phyluce - INFO - ----- Working on Streptococcus_macacae_GCF_000187995 genome -----\n",
      "2020-02-06 15:40:55,548 - Phyluce - INFO - Reading Streptococcus_macacae_GCF_000187995 genome\n",
      "2020-02-06 15:40:56,059 - Phyluce - INFO - Streptococcus_macacae_GCF_000187995: 100 uces, 0 dupes, 100 non-dupes, 18 orient drop, 3 length drop, 79 written\n",
      "2020-02-06 15:40:56,059 - Phyluce - INFO - --- Working on Streptococcus_macedonicus_GCF_000283635 genome ---\n",
      "2020-02-06 15:40:56,060 - Phyluce - INFO - Reading Streptococcus_macedonicus_GCF_000283635 genome\n",
      "2020-02-06 15:40:56,601 - Phyluce - INFO - Streptococcus_macedonicus_GCF_000283635: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 19 length drop, 81 written\n",
      "2020-02-06 15:40:56,601 - Phyluce - INFO - -- Working on Streptococcus_marimammalium_GCF_000380045 genome --\n",
      "2020-02-06 15:40:56,613 - Phyluce - INFO - Reading Streptococcus_marimammalium_GCF_000380045 genome\n",
      "2020-02-06 15:40:56,976 - Phyluce - INFO - Streptococcus_marimammalium_GCF_000380045: 99 uces, 1 dupes, 98 non-dupes, 0 orient drop, 3 length drop, 95 written\n",
      "2020-02-06 15:40:56,977 - Phyluce - INFO - ----- Working on Streptococcus_marmotae_GCF_001623565 genome ----\n",
      "2020-02-06 15:40:56,977 - Phyluce - INFO - Reading Streptococcus_marmotae_GCF_001623565 genome\n",
      "2020-02-06 15:40:57,441 - Phyluce - INFO - Streptococcus_marmotae_GCF_001623565: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 2 length drop, 83 written\n",
      "2020-02-06 15:40:57,441 - Phyluce - INFO - --- Working on Streptococcus_massiliensis_GCF_000380065 genome --\n",
      "2020-02-06 15:40:57,441 - Phyluce - INFO - Reading Streptococcus_massiliensis_GCF_000380065 genome\n",
      "2020-02-06 15:40:57,743 - Phyluce - INFO - Streptococcus_massiliensis_GCF_000380065: 93 uces, 1 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:40:57,743 - Phyluce - INFO - ----- Working on Streptococcus_merionis_GCF_000380085 genome ----\n",
      "2020-02-06 15:40:57,744 - Phyluce - INFO - Reading Streptococcus_merionis_GCF_000380085 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 15:40:58,080 - Phyluce - INFO - Streptococcus_merionis_GCF_000380085: 99 uces, 5 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-06 15:40:58,080 - Phyluce - INFO - ----- Working on Streptococcus_milleri_GCF_900636715 genome -----\n",
      "2020-02-06 15:40:58,081 - Phyluce - INFO - Reading Streptococcus_milleri_GCF_900636715 genome\n",
      "2020-02-06 15:40:58,537 - Phyluce - INFO - Streptococcus_milleri_GCF_900636715: 100 uces, 0 dupes, 100 non-dupes, 17 orient drop, 1 length drop, 82 written\n",
      "2020-02-06 15:40:58,537 - Phyluce - INFO - ------ Working on Streptococcus_minor_GCF_000377005 genome ------\n",
      "2020-02-06 15:40:58,538 - Phyluce - INFO - Reading Streptococcus_minor_GCF_000377005 genome\n",
      "2020-02-06 15:40:58,868 - Phyluce - INFO - Streptococcus_minor_GCF_000377005: 99 uces, 5 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-06 15:40:58,868 - Phyluce - INFO - ------ Working on Streptococcus_mitis_GCF_000027165 genome ------\n",
      "2020-02-06 15:40:58,868 - Phyluce - INFO - Reading Streptococcus_mitis_GCF_000027165 genome\n",
      "2020-02-06 15:40:59,328 - Phyluce - INFO - Streptococcus_mitis_GCF_000027165: 100 uces, 0 dupes, 100 non-dupes, 2 orient drop, 17 length drop, 81 written\n",
      "2020-02-06 15:40:59,328 - Phyluce - INFO - ------ Working on Streptococcus_mutans_GCF_000007465 genome -----\n",
      "2020-02-06 15:40:59,329 - Phyluce - INFO - Reading Streptococcus_mutans_GCF_000007465 genome\n",
      "2020-02-06 15:40:59,833 - Phyluce - INFO - Streptococcus_mutans_GCF_000007465: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:40:59,833 - Phyluce - INFO - ------ Working on Streptococcus_oralis_GCF_900637025 genome -----\n",
      "2020-02-06 15:40:59,839 - Phyluce - INFO - Reading Streptococcus_oralis_GCF_900637025 genome\n",
      "2020-02-06 15:41:00,315 - Phyluce - INFO - Streptococcus_oralis_GCF_900637025: 100 uces, 0 dupes, 100 non-dupes, 19 orient drop, 0 length drop, 81 written\n",
      "2020-02-06 15:41:00,315 - Phyluce - INFO - ---- Working on Streptococcus_orisasini_GCF_001431045 genome ----\n",
      "2020-02-06 15:41:00,316 - Phyluce - INFO - Reading Streptococcus_orisasini_GCF_001431045 genome\n",
      "2020-02-06 15:41:00,628 - Phyluce - INFO - Streptococcus_orisasini_GCF_001431045: 97 uces, 1 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:41:00,628 - Phyluce - INFO - ---- Working on Streptococcus_orisratti_GCF_000380105 genome ----\n",
      "2020-02-06 15:41:00,654 - Phyluce - INFO - Reading Streptococcus_orisratti_GCF_000380105 genome\n",
      "2020-02-06 15:41:00,982 - Phyluce - INFO - Streptococcus_orisratti_GCF_000380105: 98 uces, 2 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:41:00,982 - Phyluce - INFO - ------- Working on Streptococcus_ovis_GCF_000380125 genome ------\n",
      "2020-02-06 15:41:00,982 - Phyluce - INFO - Reading Streptococcus_ovis_GCF_000380125 genome\n",
      "2020-02-06 15:41:01,337 - Phyluce - INFO - Streptococcus_ovis_GCF_000380125: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 2 length drop, 91 written\n",
      "2020-02-06 15:41:01,337 - Phyluce - INFO - --- Working on Streptococcus_pantholopis_GCF_001642085 genome ---\n",
      "2020-02-06 15:41:01,337 - Phyluce - INFO - Reading Streptococcus_pantholopis_GCF_001642085 genome\n",
      "2020-02-06 15:41:01,834 - Phyluce - INFO - Streptococcus_pantholopis_GCF_001642085: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:41:01,835 - Phyluce - INFO - -- Working on Streptococcus_parasanguinis_GCF_000164675 genome --\n",
      "2020-02-06 15:41:01,845 - Phyluce - INFO - Reading Streptococcus_parasanguinis_GCF_000164675 genome\n",
      "2020-02-06 15:41:02,310 - Phyluce - INFO - Streptococcus_parasanguinis_GCF_000164675: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:41:02,310 - Phyluce - INFO - ----- Working on Streptococcus_parasuis_GCF_004283785 genome ----\n",
      "2020-02-06 15:41:02,345 - Phyluce - INFO - Reading Streptococcus_parasuis_GCF_004283785 genome\n",
      "2020-02-06 15:41:02,688 - Phyluce - INFO - Streptococcus_parasuis_GCF_004283785: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:02,688 - Phyluce - INFO - ---- Working on Streptococcus_parauberis_GCF_000213825 genome ---\n",
      "2020-02-06 15:41:02,688 - Phyluce - INFO - Reading Streptococcus_parauberis_GCF_000213825 genome\n",
      "2020-02-06 15:41:03,188 - Phyluce - INFO - Streptococcus_parauberis_GCF_000213825: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 2 length drop, 83 written\n",
      "2020-02-06 15:41:03,188 - Phyluce - INFO - --- Working on Streptococcus_pasteurianus_GCF_900478025 genome --\n",
      "2020-02-06 15:41:03,189 - Phyluce - INFO - Reading Streptococcus_pasteurianus_GCF_900478025 genome\n",
      "2020-02-06 15:41:03,733 - Phyluce - INFO - Streptococcus_pasteurianus_GCF_900478025: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 20 length drop, 80 written\n",
      "2020-02-06 15:41:03,734 - Phyluce - INFO - ---- Working on Streptococcus_penaeicida_GCF_002887775 genome ---\n",
      "2020-02-06 15:41:03,734 - Phyluce - INFO - Reading Streptococcus_penaeicida_GCF_002887775 genome\n",
      "2020-02-06 15:41:04,131 - Phyluce - INFO - Streptococcus_penaeicida_GCF_002887775: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:04,132 - Phyluce - INFO - ----- Working on Streptococcus_peroris_GCF_000187585 genome -----\n",
      "2020-02-06 15:41:04,132 - Phyluce - INFO - Reading Streptococcus_peroris_GCF_000187585 genome\n",
      "2020-02-06 15:41:04,453 - Phyluce - INFO - Streptococcus_peroris_GCF_000187585: 96 uces, 2 dupes, 94 non-dupes, 1 orient drop, 2 length drop, 91 written\n",
      "2020-02-06 15:41:04,453 - Phyluce - INFO - ---- Working on Streptococcus_pharyngis_GCF_007859195 genome ----\n",
      "2020-02-06 15:41:04,453 - Phyluce - INFO - Reading Streptococcus_pharyngis_GCF_007859195 genome\n",
      "2020-02-06 15:41:04,798 - Phyluce - INFO - Streptococcus_pharyngis_GCF_007859195: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:41:04,798 - Phyluce - INFO - ------ Working on Streptococcus_phocae_GCF_000772915 genome -----\n",
      "2020-02-06 15:41:04,799 - Phyluce - INFO - Reading Streptococcus_phocae_GCF_000772915 genome\n",
      "2020-02-06 15:41:05,140 - Phyluce - INFO - Streptococcus_phocae_GCF_000772915: 99 uces, 6 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:05,141 - Phyluce - INFO - -- Working on Streptococcus_pluranimalium_GCF_003352995 genome --\n",
      "2020-02-06 15:41:05,142 - Phyluce - INFO - Reading Streptococcus_pluranimalium_GCF_003352995 genome\n",
      "2020-02-06 15:41:05,691 - Phyluce - INFO - Streptococcus_pluranimalium_GCF_003352995: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:41:05,691 - Phyluce - INFO - --- Working on Streptococcus_plurextorum_GCF_000423745 genome ---\n",
      "2020-02-06 15:41:05,691 - Phyluce - INFO - Reading Streptococcus_plurextorum_GCF_000423745 genome\n",
      "2020-02-06 15:41:06,011 - Phyluce - INFO - Streptococcus_plurextorum_GCF_000423745: 96 uces, 4 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:41:06,011 - Phyluce - INFO - ------ Working on Streptococcus_porci_GCF_000423765 genome ------\n",
      "2020-02-06 15:41:06,012 - Phyluce - INFO - Reading Streptococcus_porci_GCF_000423765 genome\n",
      "2020-02-06 15:41:06,322 - Phyluce - INFO - Streptococcus_porci_GCF_000423765: 94 uces, 1 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:06,322 - Phyluce - INFO - ----- Working on Streptococcus_porcinus_GCF_900475415 genome ----\n",
      "2020-02-06 15:41:06,330 - Phyluce - INFO - Reading Streptococcus_porcinus_GCF_900475415 genome\n",
      "2020-02-06 15:41:06,879 - Phyluce - INFO - Streptococcus_porcinus_GCF_900475415: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n",
      "2020-02-06 15:41:06,879 - Phyluce - INFO - - Working on Streptococcus_pseudopneumoniae_GCF_000221985 genome \n",
      "2020-02-06 15:41:06,891 - Phyluce - INFO - Reading Streptococcus_pseudopneumoniae_GCF_000221985 genome\n",
      "2020-02-06 15:41:07,232 - Phyluce - INFO - Streptococcus_pseudopneumoniae_GCF_000221985: 99 uces, 0 dupes, 99 non-dupes, 0 orient drop, 4 length drop, 95 written\n",
      "2020-02-06 15:41:07,232 - Phyluce - INFO - -- Working on Streptococcus_pseudoporcinus_GCF_000188035 genome -\n",
      "2020-02-06 15:41:07,233 - Phyluce - INFO - Reading Streptococcus_pseudoporcinus_GCF_000188035 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 15:41:07,723 - Phyluce - INFO - Streptococcus_pseudoporcinus_GCF_000188035: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n",
      "2020-02-06 15:41:07,723 - Phyluce - INFO - ----- Working on Streptococcus_pyogenes_GCF_000006785 genome ----\n",
      "2020-02-06 15:41:07,724 - Phyluce - INFO - Reading Streptococcus_pyogenes_GCF_000006785 genome\n",
      "2020-02-06 15:41:08,250 - Phyluce - INFO - Streptococcus_pyogenes_GCF_000006785: 100 uces, 0 dupes, 100 non-dupes, 14 orient drop, 4 length drop, 82 written\n",
      "2020-02-06 15:41:08,250 - Phyluce - INFO - ------ Working on Streptococcus_ratti_GCF_000286075 genome ------\n",
      "2020-02-06 15:41:08,251 - Phyluce - INFO - Reading Streptococcus_ratti_GCF_000286075 genome\n",
      "2020-02-06 15:41:08,565 - Phyluce - INFO - Streptococcus_ratti_GCF_000286075: 95 uces, 0 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:41:08,565 - Phyluce - INFO - ---- Working on Streptococcus_respraculi_GCF_003595525 genome ---\n",
      "2020-02-06 15:41:08,565 - Phyluce - INFO - Reading Streptococcus_respraculi_GCF_003595525 genome\n",
      "2020-02-06 15:41:09,018 - Phyluce - INFO - Streptococcus_respraculi_GCF_003595525: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 5 length drop, 82 written\n",
      "2020-02-06 15:41:09,018 - Phyluce - INFO - ----- Working on Streptococcus_rubneri_GCF_004785935 genome -----\n",
      "2020-02-06 15:41:09,019 - Phyluce - INFO - Reading Streptococcus_rubneri_GCF_004785935 genome\n",
      "2020-02-06 15:41:09,345 - Phyluce - INFO - Streptococcus_rubneri_GCF_004785935: 99 uces, 2 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-06 15:41:09,346 - Phyluce - INFO - --- Working on Streptococcus_ruminantium_GCF_003609975 genome ---\n",
      "2020-02-06 15:41:09,346 - Phyluce - INFO - Reading Streptococcus_ruminantium_GCF_003609975 genome\n",
      "2020-02-06 15:41:09,800 - Phyluce - INFO - Streptococcus_ruminantium_GCF_003609975: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:41:09,800 - Phyluce - INFO - ---- Working on Streptococcus_salivarius_GCF_000785515 genome ---\n",
      "2020-02-06 15:41:09,800 - Phyluce - INFO - Reading Streptococcus_salivarius_GCF_000785515 genome\n",
      "2020-02-06 15:41:10,338 - Phyluce - INFO - Streptococcus_salivarius_GCF_000785515: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 6 length drop, 81 written\n",
      "2020-02-06 15:41:10,338 - Phyluce - INFO - ---- Working on Streptococcus_sanguinis_GCF_000014205 genome ----\n",
      "2020-02-06 15:41:10,346 - Phyluce - INFO - Reading Streptococcus_sanguinis_GCF_000014205 genome\n",
      "2020-02-06 15:41:10,806 - Phyluce - INFO - Streptococcus_sanguinis_GCF_000014205: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:41:10,806 - Phyluce - INFO - ----- Working on Streptococcus_sinensis_GCF_000767835 genome ----\n",
      "2020-02-06 15:41:10,814 - Phyluce - INFO - Reading Streptococcus_sinensis_GCF_000767835 genome\n",
      "2020-02-06 15:41:11,169 - Phyluce - INFO - Streptococcus_sinensis_GCF_000767835: 100 uces, 4 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:41:11,169 - Phyluce - INFO - ----- Working on Streptococcus_sobrinus_GCF_000686605 genome ----\n",
      "2020-02-06 15:41:11,169 - Phyluce - INFO - Reading Streptococcus_sobrinus_GCF_000686605 genome\n",
      "2020-02-06 15:41:11,497 - Phyluce - INFO - Streptococcus_sobrinus_GCF_000686605: 96 uces, 4 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:41:11,497 - Phyluce - INFO - -------- Working on Streptococcus_sp_GCF_003086355 genome -------\n",
      "2020-02-06 15:41:11,510 - Phyluce - INFO - Reading Streptococcus_sp_GCF_003086355 genome\n",
      "2020-02-06 15:41:12,019 - Phyluce - INFO - Streptococcus_sp_GCF_003086355: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:41:12,019 - Phyluce - INFO - ------- Working on Streptococcus_suis_GCF_000026745 genome ------\n",
      "2020-02-06 15:41:12,020 - Phyluce - INFO - Reading Streptococcus_suis_GCF_000026745 genome\n",
      "2020-02-06 15:41:12,482 - Phyluce - INFO - Streptococcus_suis_GCF_000026745: 100 uces, 0 dupes, 100 non-dupes, 1 orient drop, 18 length drop, 81 written\n",
      "2020-02-06 15:41:12,482 - Phyluce - INFO - --- Working on Streptococcus_thermophilus_GCF_000253395 genome --\n",
      "2020-02-06 15:41:12,483 - Phyluce - INFO - Reading Streptococcus_thermophilus_GCF_000253395 genome\n",
      "2020-02-06 15:41:13,022 - Phyluce - INFO - Streptococcus_thermophilus_GCF_000253395: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 6 length drop, 81 written\n",
      "2020-02-06 15:41:13,023 - Phyluce - INFO - --- Working on Streptococcus_thoraltensis_GCF_000380145 genome --\n",
      "2020-02-06 15:41:13,023 - Phyluce - INFO - Reading Streptococcus_thoraltensis_GCF_000380145 genome\n",
      "2020-02-06 15:41:13,344 - Phyluce - INFO - Streptococcus_thoraltensis_GCF_000380145: 95 uces, 3 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-06 15:41:13,344 - Phyluce - INFO - ---- Working on Streptococcus_timonensis_GCF_900095845 genome ---\n",
      "2020-02-06 15:41:13,345 - Phyluce - INFO - Reading Streptococcus_timonensis_GCF_900095845 genome\n",
      "2020-02-06 15:41:13,878 - Phyluce - INFO - Streptococcus_timonensis_GCF_900095845: 100 uces, 15 dupes, 85 non-dupes, 3 orient drop, 0 length drop, 82 written\n",
      "2020-02-06 15:41:13,878 - Phyluce - INFO - --- Working on Streptococcus_troglodytae_GCF_002355215 genome ---\n",
      "2020-02-06 15:41:13,879 - Phyluce - INFO - Reading Streptococcus_troglodytae_GCF_002355215 genome\n",
      "2020-02-06 15:41:14,389 - Phyluce - INFO - Streptococcus_troglodytae_GCF_002355215: 100 uces, 0 dupes, 100 non-dupes, 18 orient drop, 0 length drop, 82 written\n",
      "2020-02-06 15:41:14,389 - Phyluce - INFO - ------ Working on Streptococcus_uberis_GCF_000009545 genome -----\n",
      "2020-02-06 15:41:14,390 - Phyluce - INFO - Reading Streptococcus_uberis_GCF_000009545 genome\n",
      "2020-02-06 15:41:14,883 - Phyluce - INFO - Streptococcus_uberis_GCF_000009545: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:41:14,884 - Phyluce - INFO - ----- Working on Streptococcus_urinalis_GCF_000188055 genome ----\n",
      "2020-02-06 15:41:14,884 - Phyluce - INFO - Reading Streptococcus_urinalis_GCF_000188055 genome\n",
      "2020-02-06 15:41:15,375 - Phyluce - INFO - Streptococcus_urinalis_GCF_000188055: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 1 length drop, 83 written\n",
      "2020-02-06 15:41:15,375 - Phyluce - INFO - ------ Working on Streptococcus_varani_GCF_001375655 genome -----\n",
      "2020-02-06 15:41:15,375 - Phyluce - INFO - Reading Streptococcus_varani_GCF_001375655 genome\n",
      "2020-02-06 15:41:15,715 - Phyluce - INFO - Streptococcus_varani_GCF_001375655: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:41:15,715 - Phyluce - INFO - --- Working on Streptococcus_vestibularis_GCF_900636445 genome --\n",
      "2020-02-06 15:41:15,716 - Phyluce - INFO - Reading Streptococcus_vestibularis_GCF_900636445 genome\n",
      "2020-02-06 15:41:16,261 - Phyluce - INFO - Streptococcus_vestibularis_GCF_900636445: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 7 length drop, 80 written\n",
      "2020-02-06 15:41:16,261 - Phyluce - INFO - ----- Working on Streptococcus_viridans_GCF_900636365 genome ----\n",
      "2020-02-06 15:41:16,262 - Phyluce - INFO - Reading Streptococcus_viridans_GCF_900636365 genome\n",
      "2020-02-06 15:41:16,724 - Phyluce - INFO - Streptococcus_viridans_GCF_900636365: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:41:16,724 - Phyluce - INFO - ---- Working on Streptococcus_pneumoniae_GCF_000007045 genome ---\n",
      "2020-02-06 15:41:16,725 - Phyluce - INFO - Reading Streptococcus_pneumoniae_GCF_000007045 genome\n",
      "2020-02-06 15:41:17,179 - Phyluce - INFO - Streptococcus_pneumoniae_GCF_000007045: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/streptococcus/extract_probes_from_group/probe_fasta --output results/phyluce/streptococcus/extract_probes_from_group/multifastas.sqlite --base-taxon Streptococcus_pneumoniae_GCF_000007045;\n",
      "streptococcus_acidominimus_gcf_900187245.\n",
      "streptococcus_agalactiae_gcf_000007265.\n",
      "streptococcus_alactolyticus_gcf_009695625.\n",
      "streptococcus_anginosus_gcf_000463505.\n",
      "streptococcus_australis_gcf_900476055.\n",
      "streptococcus_azizii_gcf_001984715.\n",
      "streptococcus_bovimastitidis_gcf_001885095.\n",
      "streptococcus_caballi_gcf_000379985.\n",
      "streptococcus_canis_gcf_900636575.\n",
      "streptococcus_castoreus_gcf_000425025.\n",
      "streptococcus_constellatus_gcf_000463425.\n",
      "streptococcus_criceti_gcf_000187975.\n",
      "streptococcus_cristatus_gcf_000385925.\n",
      "streptococcus_cuniculi_gcf_001921845.\n",
      "streptococcus_danieliae_gcf_009767945.\n",
      "streptococcus_devriesei_gcf_000423725.\n",
      "streptococcus_didelphis_gcf_000380005.\n",
      "streptococcus_downei_gcf_900459175.\n",
      "streptococcus_dysgalactiae_gcf_000317855.\n",
      "streptococcus_entericus_gcf_000380025.\n",
      "streptococcus_equi_gcf_000026605.\n",
      "streptococcus_equinus_gcf_000964315.\n",
      "streptococcus_ferus_gcf_000372425.\n",
      "streptococcus_gallolyticus_gcf_002000985.\n",
      "streptococcus_gordonii_gcf_000017005.\n",
      "streptococcus_halichoeri_gcf_009870755.\n",
      "streptococcus_halitosis_gcf_003143695.\n",
      "streptococcus_halotolerans_gcf_001598035.\n",
      "streptococcus_henryi_gcf_000376985.\n",
      "streptococcus_himalayensis_gcf_001708305.\n",
      "streptococcus_hongkongensis_gcf_000785845.\n",
      "streptococcus_hyointestinalis_gcf_900459405.\n",
      "streptococcus_hyovaginalis_gcf_000420785.\n",
      "streptococcus_ictaluri_gcf_000188015.\n",
      "streptococcus_infantarius_gcf_000246835.\n",
      "streptococcus_infantis_gcf_000187465.\n",
      "streptococcus_iniae_gcf_000831485.\n",
      "streptococcus_intermedius_gcf_000463355.\n",
      "streptococcus_lutetiensis_gcf_900475675.\n",
      "streptococcus_macacae_gcf_000187995.\n",
      "streptococcus_macedonicus_gcf_000283635.\n",
      "streptococcus_marimammalium_gcf_000380045.\n",
      "streptococcus_marmotae_gcf_001623565.\n",
      "streptococcus_massiliensis_gcf_000380065.\n",
      "streptococcus_merionis_gcf_000380085.\n",
      "streptococcus_milleri_gcf_900636715.\n",
      "streptococcus_minor_gcf_000377005.\n",
      "streptococcus_mitis_gcf_000027165.\n",
      "streptococcus_mutans_gcf_000007465.\n",
      "streptococcus_oralis_gcf_900637025.\n",
      "streptococcus_orisasini_gcf_001431045.\n",
      "streptococcus_orisratti_gcf_000380105.\n",
      "streptococcus_ovis_gcf_000380125.\n",
      "streptococcus_pantholopis_gcf_001642085.\n",
      "streptococcus_parasanguinis_gcf_000164675.\n",
      "streptococcus_parasuis_gcf_004283785.\n",
      "streptococcus_parauberis_gcf_000213825.\n",
      "streptococcus_pasteurianus_gcf_900478025.\n",
      "streptococcus_penaeicida_gcf_002887775.\n",
      "streptococcus_peroris_gcf_000187585.\n",
      "streptococcus_pharyngis_gcf_007859195.\n",
      "streptococcus_phocae_gcf_000772915.\n",
      "streptococcus_pluranimalium_gcf_003352995.\n",
      "streptococcus_plurextorum_gcf_000423745.\n",
      "streptococcus_porci_gcf_000423765.\n",
      "streptococcus_porcinus_gcf_900475415.\n",
      "streptococcus_pseudopneumoniae_gcf_000221985.\n",
      "streptococcus_pseudoporcinus_gcf_000188035.\n",
      "streptococcus_pyogenes_gcf_000006785.\n",
      "streptococcus_ratti_gcf_000286075.\n",
      "streptococcus_respraculi_gcf_003595525.\n",
      "streptococcus_rubneri_gcf_004785935.\n",
      "streptococcus_ruminantium_gcf_003609975.\n",
      "streptococcus_salivarius_gcf_000785515.\n",
      "streptococcus_sanguinis_gcf_000014205.\n",
      "streptococcus_sinensis_gcf_000767835.\n",
      "streptococcus_sobrinus_gcf_000686605.\n",
      "streptococcus_sp_gcf_003086355.\n",
      "streptococcus_suis_gcf_000026745.\n",
      "streptococcus_thermophilus_gcf_000253395.\n",
      "streptococcus_thoraltensis_gcf_000380145.\n",
      "streptococcus_timonensis_gcf_900095845.\n",
      "streptococcus_troglodytae_gcf_002355215.\n",
      "streptococcus_uberis_gcf_000009545.\n",
      "streptococcus_urinalis_gcf_000188055.\n",
      "streptococcus_varani_gcf_001375655.\n",
      "streptococcus_vestibularis_gcf_900636445.\n",
      "streptococcus_viridans_gcf_900636365.\n",
      "streptococcus_pneumoniae_gcf_000007045.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/streptococcus/extract_probes_from_group/multifastas.sqlite --base-taxon Streptococcus_pneumoniae_GCF_000007045;\n",
      "Loci shared by 0 taxa:\t100.0\n",
      "Loci shared by 1 taxa:\t100.0\n",
      "Loci shared by 2 taxa:\t100.0\n",
      "Loci shared by 3 taxa:\t100.0\n",
      "Loci shared by 4 taxa:\t100.0\n",
      "Loci shared by 5 taxa:\t100.0\n",
      "Loci shared by 6 taxa:\t100.0\n",
      "Loci shared by 7 taxa:\t100.0\n",
      "Loci shared by 8 taxa:\t100.0\n",
      "Loci shared by 9 taxa:\t99.0\n",
      "Loci shared by 10 taxa:\t98.0\n",
      "Loci shared by 11 taxa:\t97.0\n",
      "Loci shared by 12 taxa:\t97.0\n",
      "Loci shared by 13 taxa:\t97.0\n",
      "Loci shared by 14 taxa:\t97.0\n",
      "Loci shared by 15 taxa:\t97.0\n",
      "Loci shared by 16 taxa:\t96.0\n",
      "Loci shared by 17 taxa:\t95.0\n",
      "Loci shared by 18 taxa:\t95.0\n",
      "Loci shared by 19 taxa:\t95.0\n",
      "Loci shared by 20 taxa:\t95.0\n",
      "Loci shared by 21 taxa:\t95.0\n",
      "Loci shared by 22 taxa:\t95.0\n",
      "Loci shared by 23 taxa:\t95.0\n",
      "Loci shared by 24 taxa:\t95.0\n",
      "Loci shared by 25 taxa:\t94.0\n",
      "Loci shared by 26 taxa:\t94.0\n",
      "Loci shared by 27 taxa:\t94.0\n",
      "Loci shared by 28 taxa:\t94.0\n",
      "Loci shared by 29 taxa:\t94.0\n",
      "Loci shared by 30 taxa:\t94.0\n",
      "Loci shared by 31 taxa:\t94.0\n",
      "Loci shared by 32 taxa:\t92.0\n",
      "Loci shared by 33 taxa:\t92.0\n",
      "Loci shared by 34 taxa:\t91.0\n",
      "Loci shared by 35 taxa:\t91.0\n",
      "Loci shared by 36 taxa:\t89.0\n",
      "Loci shared by 37 taxa:\t86.0\n",
      "Loci shared by 38 taxa:\t83.0\n",
      "Loci shared by 39 taxa:\t83.0\n",
      "Loci shared by 40 taxa:\t82.0\n",
      "Loci shared by 41 taxa:\t82.0\n",
      "Loci shared by 42 taxa:\t82.0\n",
      "Loci shared by 43 taxa:\t82.0\n",
      "Loci shared by 44 taxa:\t82.0\n",
      "Loci shared by 45 taxa:\t82.0\n",
      "Loci shared by 46 taxa:\t82.0\n",
      "Loci shared by 47 taxa:\t82.0\n",
      "Loci shared by 48 taxa:\t82.0\n",
      "Loci shared by 49 taxa:\t82.0\n",
      "Loci shared by 50 taxa:\t82.0\n",
      "Loci shared by 51 taxa:\t82.0\n",
      "Loci shared by 52 taxa:\t82.0\n",
      "Loci shared by 53 taxa:\t82.0\n",
      "Loci shared by 54 taxa:\t82.0\n",
      "Loci shared by 55 taxa:\t82.0\n",
      "Loci shared by 56 taxa:\t82.0\n",
      "Loci shared by 57 taxa:\t82.0\n",
      "Loci shared by 58 taxa:\t82.0\n",
      "Loci shared by 59 taxa:\t82.0\n",
      "Loci shared by 60 taxa:\t82.0\n",
      "Loci shared by 61 taxa:\t82.0\n",
      "Loci shared by 62 taxa:\t82.0\n",
      "Loci shared by 63 taxa:\t81.0\n",
      "Loci shared by 64 taxa:\t81.0\n",
      "Loci shared by 65 taxa:\t81.0\n",
      "Loci shared by 66 taxa:\t81.0\n",
      "Loci shared by 67 taxa:\t81.0\n",
      "Loci shared by 68 taxa:\t81.0\n",
      "Loci shared by 69 taxa:\t81.0\n",
      "Loci shared by 70 taxa:\t81.0\n",
      "Loci shared by 71 taxa:\t81.0\n",
      "Loci shared by 72 taxa:\t81.0\n",
      "Loci shared by 73 taxa:\t81.0\n",
      "Loci shared by 74 taxa:\t81.0\n",
      "Loci shared by 75 taxa:\t81.0\n",
      "Loci shared by 76 taxa:\t81.0\n",
      "Loci shared by 77 taxa:\t81.0\n",
      "Loci shared by 78 taxa:\t80.0\n",
      "Loci shared by 79 taxa:\t80.0\n",
      "Loci shared by 80 taxa:\t80.0\n",
      "Loci shared by 81 taxa:\t80.0\n",
      "Loci shared by 82 taxa:\t80.0\n",
      "Loci shared by 83 taxa:\t80.0\n",
      "Loci shared by 84 taxa:\t80.0\n",
      "Loci shared by 85 taxa:\t78.0\n",
      "Loci shared by 86 taxa:\t76.0\n",
      "Loci shared by 87 taxa:\t76.0\n",
      "Loci shared by 88 taxa:\t76.0\n",
      "Loci shared by 89 taxa:\t67.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/streptococcus/extract_probes_from_group/multifastas.sqlite  --base-taxon Streptococcus_pneumoniae_GCF_000007045 --output results/phyluce/streptococcus/extract_probes_from_group/Streptococcus_pneumoniae_GCF_000007045+41-back-to-77.conf --specific-counts 77;\n",
      "Counter({'streptococcus_varani_gcf_001375655': 81, 'streptococcus_australis_gcf_900476055': 81, 'streptococcus_anginosus_gcf_000463505': 81, 'streptococcus_timonensis_gcf_900095845': 81, 'streptococcus_gallolyticus_gcf_002000985': 81, 'streptococcus_mutans_gcf_000007465': 81, 'streptococcus_intermedius_gcf_000463355': 81, 'streptococcus_infantarius_gcf_000246835': 81, 'streptococcus_equi_gcf_000026605': 81, 'streptococcus_respraculi_gcf_003595525': 81, 'streptococcus_pluranimalium_gcf_003352995': 81, 'streptococcus_uberis_gcf_000009545': 81, 'streptococcus_danieliae_gcf_009767945': 81, 'streptococcus_constellatus_gcf_000463425': 81, 'streptococcus_parauberis_gcf_000213825': 81, 'streptococcus_sinensis_gcf_000767835': 81, 'streptococcus_halichoeri_gcf_009870755': 81, 'streptococcus_infantis_gcf_000187465': 81, 'streptococcus_dysgalactiae_gcf_000317855': 81, 'streptococcus_gordonii_gcf_000017005': 81, 'streptococcus_troglodytae_gcf_002355215': 81, 'streptococcus_ruminantium_gcf_003609975': 81, 'streptococcus_viridans_gcf_900636365': 81, 'streptococcus_equinus_gcf_000964315': 81, 'streptococcus_minor_gcf_000377005': 81, 'streptococcus_rubneri_gcf_004785935': 81, 'streptococcus_milleri_gcf_900636715': 81, 'streptococcus_cuniculi_gcf_001921845': 81, 'streptococcus_thermophilus_gcf_000253395': 81, 'streptococcus_pyogenes_gcf_000006785': 81, 'streptococcus_pseudopneumoniae_gcf_000221985': 81, 'streptococcus_penaeicida_gcf_002887775': 81, 'streptococcus_azizii_gcf_001984715': 81, 'streptococcus_marmotae_gcf_001623565': 81, 'streptococcus_sp_gcf_003086355': 81, 'streptococcus_urinalis_gcf_000188055': 81, 'streptococcus_ictaluri_gcf_000188015': 81, 'streptococcus_himalayensis_gcf_001708305': 81, 'streptococcus_sanguinis_gcf_000014205': 81, 'streptococcus_ratti_gcf_000286075': 81, 'streptococcus_ferus_gcf_000372425': 81, 'streptococcus_didelphis_gcf_000380005': 81, 'streptococcus_entericus_gcf_000380025': 81, 'streptococcus_castoreus_gcf_000425025': 81, 'streptococcus_pneumoniae_gcf_000007045': 81, 'streptococcus_halitosis_gcf_003143695': 81, 'streptococcus_salivarius_gcf_000785515': 81, 'streptococcus_caballi_gcf_000379985': 81, 'streptococcus_plurextorum_gcf_000423745': 81, 'streptococcus_pseudoporcinus_gcf_000188035': 81, 'streptococcus_suis_gcf_000026745': 81, 'streptococcus_canis_gcf_900636575': 81, 'streptococcus_cristatus_gcf_000385925': 81, 'streptococcus_pantholopis_gcf_001642085': 81, 'streptococcus_iniae_gcf_000831485': 81, 'streptococcus_halotolerans_gcf_001598035': 81, 'streptococcus_parasanguinis_gcf_000164675': 81, 'streptococcus_marimammalium_gcf_000380045': 81, 'streptococcus_orisasini_gcf_001431045': 81, 'streptococcus_pharyngis_gcf_007859195': 81, 'streptococcus_devriesei_gcf_000423725': 81, 'streptococcus_porcinus_gcf_900475415': 81, 'streptococcus_parasuis_gcf_004283785': 81, 'streptococcus_porci_gcf_000423765': 81, 'streptococcus_agalactiae_gcf_000007265': 80, 'streptococcus_oralis_gcf_900637025': 80, 'streptococcus_alactolyticus_gcf_009695625': 80, 'streptococcus_merionis_gcf_000380085': 80, 'streptococcus_acidominimus_gcf_900187245': 80, 'streptococcus_hyovaginalis_gcf_000420785': 80, 'streptococcus_hyointestinalis_gcf_900459405': 80, 'streptococcus_mitis_gcf_000027165': 80, 'streptococcus_orisratti_gcf_000380105': 80, 'streptococcus_henryi_gcf_000376985': 80, 'streptococcus_pasteurianus_gcf_900478025': 80, 'streptococcus_lutetiensis_gcf_900475675': 80, 'streptococcus_bovimastitidis_gcf_001885095': 80, 'streptococcus_macedonicus_gcf_000283635': 80, 'streptococcus_ovis_gcf_000380125': 79, 'streptococcus_massiliensis_gcf_000380065': 79, 'streptococcus_vestibularis_gcf_900636445': 79, 'streptococcus_macacae_gcf_000187995': 79, 'streptococcus_sobrinus_gcf_000686605': 79, 'streptococcus_phocae_gcf_000772915': 79, 'streptococcus_hongkongensis_gcf_000785845': 79, 'streptococcus_criceti_gcf_000187975': 79, 'streptococcus_downei_gcf_900459175': 78, 'streptococcus_thoraltensis_gcf_000380145': 78, 'streptococcus_peroris_gcf_000187585': 78})\n",
      "Total loci = 81\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(77)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/streptococcus/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/streptococcus/extract_probes_from_group/Streptococcus_pneumoniae_GCF_000007045+41-back-to-77.conf --probe-prefix uce_streptococcus_ --designer rnplattii --design streptococcus_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 81\n",
      "Probe Count = 14298\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta --query results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  15:42:14\n",
      "Ended:  Thu Feb 06, 2020  15:44:20\n",
      "Time for execution:  2.11134323279 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta --lastz results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_streptococcus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 14297 fasta sequences.  Filtered 0 duplicates. Kept 14298.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 16:37:39 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 14298\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1143840\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 90M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3940576\n",
      "Max number of word counting entries: 88702384\n",
      "\n",
      "# comparing sequences from          0  to       2383\n",
      "..---------- new table with     1206 representatives\n",
      "# comparing sequences from       2383  to       4368\n",
      "----------   1283 remaining sequences to the next cycle\n",
      "---------- new table with      293 representatives\n",
      "# comparing sequences from       3085  to       4953\n",
      "----------   1166 remaining sequences to the next cycle\n",
      "---------- new table with      303 representatives\n",
      "# comparing sequences from       3787  to       5538\n",
      "----------   1151 remaining sequences to the next cycle\n",
      "---------- new table with      322 representatives\n",
      "# comparing sequences from       4387  to       6038\n",
      "----------   1028 remaining sequences to the next cycle\n",
      "---------- new table with      378 representatives\n",
      "# comparing sequences from       5010  to       6558\n",
      "----------    996 remaining sequences to the next cycle\n",
      "---------- new table with      328 representatives\n",
      "# comparing sequences from       5562  to       7018\n",
      "----------    851 remaining sequences to the next cycle\n",
      "---------- new table with      424 representatives\n",
      "# comparing sequences from       6167  to       7522\n",
      "----------    973 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       6549  to       7840\n",
      "----------    798 remaining sequences to the next cycle\n",
      "---------- new table with      334 representatives\n",
      "# comparing sequences from       7042  to       8251\n",
      "----------    823 remaining sequences to the next cycle\n",
      "---------- new table with      136 representatives\n",
      "# comparing sequences from       7428  to       8573\n",
      "----------    644 remaining sequences to the next cycle\n",
      "---------- new table with      294 representatives\n",
      "# comparing sequences from       7929  to       8990\n",
      "----------    715 remaining sequences to the next cycle\n",
      "---------- new table with      209 representatives\n",
      "# comparing sequences from       8275  to       9278\n",
      "----------    708 remaining sequences to the next cycle\n",
      "---------- new table with      108 representatives\n",
      "# comparing sequences from       8570  to       9524\n",
      "----------    521 remaining sequences to the next cycle\n",
      "---------- new table with      307 representatives\n",
      "# comparing sequences from       9003  to       9885\n",
      "----------    404 remaining sequences to the next cycle\n",
      "---------- new table with      165 representatives\n",
      "# comparing sequences from       9481  to      10283\n",
      "----------    530 remaining sequences to the next cycle\n",
      "---------- new table with      163 representatives\n",
      "# comparing sequences from       9753  to      10510\n",
      "..........    10000  finished       5197  clusters\n",
      "----------    359 remaining sequences to the next cycle\n",
      "---------- new table with      199 representatives\n",
      "# comparing sequences from      10151  to      10842\n",
      "----------    423 remaining sequences to the next cycle\n",
      "---------- new table with      130 representatives\n",
      "# comparing sequences from      10419  to      11065\n",
      "----------    373 remaining sequences to the next cycle\n",
      "---------- new table with      122 representatives\n",
      "# comparing sequences from      10692  to      11293\n",
      "----------    271 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      11022  to      11568\n",
      "----------    250 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      11318  to      11814\n",
      "----------    205 remaining sequences to the next cycle\n",
      "---------- new table with      151 representatives\n",
      "# comparing sequences from      11609  to      12057\n",
      "----------    226 remaining sequences to the next cycle\n",
      "---------- new table with      139 representatives\n",
      "# comparing sequences from      11831  to      12242\n",
      "----------    159 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      12083  to      12452\n",
      "----------     72 remaining sequences to the next cycle\n",
      "---------- new table with      163 representatives\n",
      "# comparing sequences from      12380  to      12699\n",
      "----------     97 remaining sequences to the next cycle\n",
      "---------- new table with      160 representatives\n",
      "# comparing sequences from      12602  to      12884\n",
      "----------     21 remaining sequences to the next cycle\n",
      "---------- new table with      182 representatives\n",
      "# comparing sequences from      12863  to      13102\n",
      ".....................---------- new table with       92 representatives\n",
      "# comparing sequences from      13102  to      13301\n",
      "....................---------- new table with       61 representatives\n",
      "# comparing sequences from      13301  to      14298\n",
      "....................---------- new table with      502 representatives\n",
      "\n",
      "    14298  finished       7271  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 91M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 3.16\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Yersinia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'yersinia'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Yersinia_aldovae_GCF_000834395\",\n",
    "               \"Yersinia_aleksiciae_GCF_001047675\",\n",
    "               \"Yersinia_bercovieri_GCF_000167975\",\n",
    "               \"Yersinia_canariae_GCF_009831415\",\n",
    "               \"Yersinia_enterocolitica_GCF_000009345\",\n",
    "               \"Yersinia_entomophaga_GCF_001656035\",\n",
    "               \"Yersinia_frederiksenii_GCF_002591095\",\n",
    "               \"Yersinia_hibernica_GCF_004124235\",\n",
    "               \"Yersinia_intermedia_GCF_900635455\",\n",
    "               \"Yersinia_kristensenii_GCF_000834865\",\n",
    "               \"Yersinia_massiliensis_GCF_003048255\",\n",
    "               \"Yersinia_mollaretii_GCF_000167995\",\n",
    "               \"Yersinia_nurmii_GCF_001112925\",\n",
    "               \"Yersinia_pekkanenii_GCF_001244635\",\n",
    "               \"Yersinia_pseudotuberculosis_GCF_900637475\",\n",
    "               \"Yersinia_rohdei_GCF_000834455\",\n",
    "               \"Yersinia_ruckeri_GCF_000964565\",\n",
    "               \"Yersinia_similis_GCF_000582515\",\n",
    "               \"Yersinia_sp_GCF_002073315\",\n",
    "               \"Yersinia_wautersii_GCF_000493415\"  ]\n",
    "                    \n",
    "reference_taxon = \"Yersinia_pestis_GCF_000009065\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000834395.1_ASM83439v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1331888 bytes  887953.33 bytes/sec\n",
      "total size is 1331455  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001047675.1_ASM104767v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1191961 bytes  794668.67 bytes/sec\n",
      "total size is 1191559  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000167975.1_ASM16797v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1302758 bytes  521120.00 bytes/sec\n",
      "total size is 1302333  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009831415.1_ASM983141v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1406656 bytes  937798.67 bytes/sec\n",
      "total size is 1406206  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000009345.1_ASM934v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1395203 bytes  930163.33 bytes/sec\n",
      "total size is 1394756  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001656035.1_ASM165603v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1275525 bytes  850378.00 bytes/sec\n",
      "total size is 1275107  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002591095.1_ASM259109v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1494224 bytes  996177.33 bytes/sec\n",
      "total size is 1493750  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004124235.1_ASM412423v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1431244 bytes  954190.67 bytes/sec\n",
      "total size is 1430786  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900635455.1_28307_A01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1420455 bytes  946998.00 bytes/sec\n",
      "total size is 1419999  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000834865.1_ASM83486v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1340001 bytes  893362.00 bytes/sec\n",
      "total size is 1339568  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003048255.1_ASM304825v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1543284 bytes  1028884.00 bytes/sec\n",
      "total size is 1542794  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000167995.1_ASM16799v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1367469 bytes  911674.00 bytes/sec\n",
      "total size is 1367028  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001112925.1_8016_2_91_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1240583 bytes  496250.00 bytes/sec\n",
      "total size is 1240175  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001244635.1_8016_2_92_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1527169 bytes  1018140.67 bytes/sec\n",
      "total size is 1526689  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000009065.1_ASM906v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1438349 bytes  958927.33 bytes/sec\n",
      "total size is 1437894  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900637475.1_51108_B01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1371500 bytes  914361.33 bytes/sec\n",
      "total size is 1371060  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000834455.1_ASM83445v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1301846 bytes  867925.33 bytes/sec\n",
      "total size is 1301421  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964565.1_ASM96456v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1103377 bytes  735612.67 bytes/sec\n",
      "total size is 1103000  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000582515.1_ASM58251v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1473926 bytes  982645.33 bytes/sec\n",
      "total size is 1473461  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002073315.2_ASM207331v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1480915 bytes  987304.67 bytes/sec\n",
      "total size is 1480441  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000493415.1_1_WP-931205_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1437622 bytes  958442.67 bytes/sec\n",
      "total size is 1437164  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480876 (\"sim_Yersinia_aldovae_GCF_000834395\") has been submitted\n",
      "Your job 5480877 (\"sim_Yersinia_aleksiciae_GCF_001047675\") has been submitted\n",
      "Your job 5480878 (\"sim_Yersinia_bercovieri_GCF_000167975\") has been submitted\n",
      "Your job 5480879 (\"sim_Yersinia_canariae_GCF_009831415\") has been submitted\n",
      "Your job 5480880 (\"sim_Yersinia_enterocolitica_GCF_000009345\") has been submitted\n",
      "Your job 5480881 (\"sim_Yersinia_entomophaga_GCF_001656035\") has been submitted\n",
      "Your job 5480882 (\"sim_Yersinia_frederiksenii_GCF_002591095\") has been submitted\n",
      "Your job 5480883 (\"sim_Yersinia_hibernica_GCF_004124235\") has been submitted\n",
      "Your job 5480884 (\"sim_Yersinia_intermedia_GCF_900635455\") has been submitted\n",
      "Your job 5480885 (\"sim_Yersinia_kristensenii_GCF_000834865\") has been submitted\n",
      "Your job 5480886 (\"sim_Yersinia_massiliensis_GCF_003048255\") has been submitted\n",
      "Your job 5480887 (\"sim_Yersinia_mollaretii_GCF_000167995\") has been submitted\n",
      "Your job 5480888 (\"sim_Yersinia_nurmii_GCF_001112925\") has been submitted\n",
      "Your job 5480889 (\"sim_Yersinia_pekkanenii_GCF_001244635\") has been submitted\n",
      "Your job 5480890 (\"sim_Yersinia_pseudotuberculosis_GCF_900637475\") has been submitted\n",
      "Your job 5480891 (\"sim_Yersinia_rohdei_GCF_000834455\") has been submitted\n",
      "Your job 5480892 (\"sim_Yersinia_ruckeri_GCF_000964565\") has been submitted\n",
      "Your job 5480893 (\"sim_Yersinia_similis_GCF_000582515\") has been submitted\n",
      "Your job 5480894 (\"sim_Yersinia_sp_GCF_002073315\") has been submitted\n",
      "Your job 5480895 (\"sim_Yersinia_wautersii_GCF_000493415\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/yersinia/cleaned_genomes/Yersinia_pestis_GCF_000009065_formatted.fas path=results/phyluce/yersinia/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480896 (\"map_Yersinia_aldovae_GCF_000834395\") has been submitted\n",
      "Your job 5480897 (\"map_Yersinia_aleksiciae_GCF_001047675\") has been submitted\n",
      "Your job 5480898 (\"map_Yersinia_bercovieri_GCF_000167975\") has been submitted\n",
      "Your job 5480899 (\"map_Yersinia_canariae_GCF_009831415\") has been submitted\n",
      "Your job 5480900 (\"map_Yersinia_enterocolitica_GCF_000009345\") has been submitted\n",
      "Your job 5480901 (\"map_Yersinia_entomophaga_GCF_001656035\") has been submitted\n",
      "Your job 5480902 (\"map_Yersinia_frederiksenii_GCF_002591095\") has been submitted\n",
      "Your job 5480903 (\"map_Yersinia_hibernica_GCF_004124235\") has been submitted\n",
      "Your job 5480904 (\"map_Yersinia_intermedia_GCF_900635455\") has been submitted\n",
      "Your job 5480905 (\"map_Yersinia_kristensenii_GCF_000834865\") has been submitted\n",
      "Your job 5480906 (\"map_Yersinia_massiliensis_GCF_003048255\") has been submitted\n",
      "Your job 5480907 (\"map_Yersinia_mollaretii_GCF_000167995\") has been submitted\n",
      "Your job 5480908 (\"map_Yersinia_nurmii_GCF_001112925\") has been submitted\n",
      "Your job 5480909 (\"map_Yersinia_pekkanenii_GCF_001244635\") has been submitted\n",
      "Your job 5480910 (\"map_Yersinia_pseudotuberculosis_GCF_900637475\") has been submitted\n",
      "Your job 5480911 (\"map_Yersinia_rohdei_GCF_000834455\") has been submitted\n",
      "Your job 5480912 (\"map_Yersinia_ruckeri_GCF_000964565\") has been submitted\n",
      "Your job 5480913 (\"map_Yersinia_similis_GCF_000582515\") has been submitted\n",
      "Your job 5480914 (\"map_Yersinia_sp_GCF_002073315\") has been submitted\n",
      "Your job 5480915 (\"map_Yersinia_wautersii_GCF_000493415\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5480916 (\"merge_Yersinia_aldovae_GCF_000834395\") has been submitted\n",
      "Your job 5480917 (\"merge_Yersinia_aleksiciae_GCF_001047675\") has been submitted\n",
      "Your job 5480918 (\"merge_Yersinia_bercovieri_GCF_000167975\") has been submitted\n",
      "Your job 5480919 (\"merge_Yersinia_canariae_GCF_009831415\") has been submitted\n",
      "Your job 5480920 (\"merge_Yersinia_enterocolitica_GCF_000009345\") has been submitted\n",
      "Your job 5480921 (\"merge_Yersinia_entomophaga_GCF_001656035\") has been submitted\n",
      "Your job 5480922 (\"merge_Yersinia_frederiksenii_GCF_002591095\") has been submitted\n",
      "Your job 5480923 (\"merge_Yersinia_hibernica_GCF_004124235\") has been submitted\n",
      "Your job 5480924 (\"merge_Yersinia_intermedia_GCF_900635455\") has been submitted\n",
      "Your job 5480925 (\"merge_Yersinia_kristensenii_GCF_000834865\") has been submitted\n",
      "Your job 5480926 (\"merge_Yersinia_massiliensis_GCF_003048255\") has been submitted\n",
      "Your job 5480927 (\"merge_Yersinia_mollaretii_GCF_000167995\") has been submitted\n",
      "Your job 5480928 (\"merge_Yersinia_nurmii_GCF_001112925\") has been submitted\n",
      "Your job 5480929 (\"merge_Yersinia_pekkanenii_GCF_001244635\") has been submitted\n",
      "Your job 5480930 (\"merge_Yersinia_pseudotuberculosis_GCF_900637475\") has been submitted\n",
      "Your job 5480931 (\"merge_Yersinia_rohdei_GCF_000834455\") has been submitted\n",
      "Your job 5480932 (\"merge_Yersinia_ruckeri_GCF_000964565\") has been submitted\n",
      "Your job 5480933 (\"merge_Yersinia_similis_GCF_000582515\") has been submitted\n",
      "Your job 5480934 (\"merge_Yersinia_sp_GCF_002073315\") has been submitted\n",
      "Your job 5480935 (\"merge_Yersinia_wautersii_GCF_000493415\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 3082 sequences from Yersinia_aldovae_GCF_000834395_merged.bed.  Filtered 1707 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1375.\n",
      "Screened 3023 sequences from Yersinia_aleksiciae_GCF_001047675_merged.bed.  Filtered 1661 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1362.\n",
      "Screened 3237 sequences from Yersinia_bercovieri_GCF_000167975_merged.bed.  Filtered 1796 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1441.\n",
      "Screened 3066 sequences from Yersinia_canariae_GCF_009831415_merged.bed.  Filtered 1694 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1372.\n",
      "Screened 3162 sequences from Yersinia_enterocolitica_GCF_000009345_merged.bed.  Filtered 1747 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1415.\n",
      "Screened 1445 sequences from Yersinia_entomophaga_GCF_001656035_merged.bed.  Filtered 829 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 616.\n",
      "Screened 2720 sequences from Yersinia_frederiksenii_GCF_002591095_merged.bed.  Filtered 1512 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1208.\n",
      "Screened 2910 sequences from Yersinia_hibernica_GCF_004124235_merged.bed.  Filtered 1627 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1283.\n",
      "Screened 3302 sequences from Yersinia_intermedia_GCF_900635455_merged.bed.  Filtered 1831 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1471.\n",
      "Screened 3156 sequences from Yersinia_kristensenii_GCF_000834865_merged.bed.  Filtered 1756 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1400.\n",
      "Screened 2798 sequences from Yersinia_massiliensis_GCF_003048255_merged.bed.  Filtered 1614 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1184.\n",
      "Screened 3259 sequences from Yersinia_mollaretii_GCF_000167995_merged.bed.  Filtered 1790 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1469.\n",
      "Screened 1438 sequences from Yersinia_nurmii_GCF_001112925_merged.bed.  Filtered 815 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 623.\n",
      "Screened 3660 sequences from Yersinia_pekkanenii_GCF_001244635_merged.bed.  Filtered 2074 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1586.\n",
      "Screened 577 sequences from Yersinia_pseudotuberculosis_GCF_900637475_merged.bed.  Filtered 59 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 518.\n",
      "Screened 2838 sequences from Yersinia_rohdei_GCF_000834455_merged.bed.  Filtered 1564 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1274.\n",
      "Screened 1386 sequences from Yersinia_ruckeri_GCF_000964565_merged.bed.  Filtered 805 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 581.\n",
      "Screened 2355 sequences from Yersinia_similis_GCF_000582515_merged.bed.  Filtered 328 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2027.\n",
      "Screened 3360 sequences from Yersinia_sp_GCF_002073315_merged.bed.  Filtered 1861 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1499.\n",
      "Screened 1455 sequences from Yersinia_wautersii_GCF_000493415_merged.bed.  Filtered 178 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1277.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/yersinia/initial_intervals/Yersinia_wautersii_GCF_000493415_merged.bed --twobit results/phyluce/yersinia/cleaned_genomes/Yersinia_pestis_GCF_000009065_formatted.2bit --output results/phyluce/yersinia/initial_intervals/Yersinia_wautersii_GCF_000493415_stripped.bed;\n",
      "yersinia_aldovae_gcf_000834395..\n",
      "yersinia_aleksiciae_gcf_001047675..\n",
      "yersinia_bercovieri_gcf_000167975..\n",
      "yersinia_canariae_gcf_009831415..\n",
      "yersinia_enterocolitica_gcf_000009345..\n",
      "yersinia_entomophaga_gcf_001656035.\n",
      "yersinia_frederiksenii_gcf_002591095..\n",
      "yersinia_hibernica_gcf_004124235..\n",
      "yersinia_intermedia_gcf_900635455..\n",
      "yersinia_kristensenii_gcf_000834865..\n",
      "yersinia_massiliensis_gcf_003048255..\n",
      "yersinia_mollaretii_gcf_000167995..\n",
      "yersinia_nurmii_gcf_001112925.\n",
      "yersinia_pekkanenii_gcf_001244635..\n",
      "yersinia_pseudotuberculosis_gcf_900637475.\n",
      "yersinia_rohdei_gcf_000834455..\n",
      "yersinia_ruckeri_gcf_000964565.\n",
      "yersinia_similis_gcf_000582515...\n",
      "yersinia_sp_gcf_002073315..\n",
      "yersinia_wautersii_gcf_000493415..\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/yersinia/initial_intervals/Yersinia_wautersii_GCF_000493415_merged.bed --twobit results/phyluce/yersinia/cleaned_genomes/Yersinia_pestis_GCF_000009065_formatted.2bit --output results/phyluce/yersinia/initial_intervals/Yersinia_wautersii_GCF_000493415_stripped.bed;\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 0 taxa:\t4,353.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 1 taxa:\t4,353.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 2 taxa:\t4,127.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 3 taxa:\t3,103.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 4 taxa:\t2,913.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 5 taxa:\t2,327.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 6 taxa:\t2,009.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 7 taxa:\t1,770.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 8 taxa:\t1,585.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 9 taxa:\t1,450.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 10 taxa:\t1,329.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 11 taxa:\t1,213.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 12 taxa:\t1,121.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 13 taxa:\t1,036.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 14 taxa:\t944.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 15 taxa:\t853.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 16 taxa:\t767.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 17 taxa:\t662.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 18 taxa:\t488.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 19 taxa:\t416.0\n",
      "Loci shared by Yersinia_pestis_GCF_000009065 + 20 taxa:\t293.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/yersinia/initial_intervals/yersinia-to-Yersinia_pestis_GCF_000009065.sqlite --base-taxon Yersinia_pestis_GCF_000009065 --output results/phyluce/yersinia/initial_intervals/Yersinia_pestis_GCF_000009065_+20.bed --specific-counts 20;\n",
      "Counter({'yersinia_kristensenii_gcf_000834865': 293, 'yersinia_intermedia_gcf_900635455': 293, 'yersinia_pekkanenii_gcf_001244635': 293, 'yersinia_similis_gcf_000582515': 293, 'yersinia_bercovieri_gcf_000167975': 293, 'yersinia_rohdei_gcf_000834455': 293, 'yersinia_sp_gcf_002073315': 293, 'yersinia_hibernica_gcf_004124235': 293, 'yersinia_enterocolitica_gcf_000009345': 293, 'yersinia_aleksiciae_gcf_001047675': 293, 'yersinia_massiliensis_gcf_003048255': 293, 'yersinia_wautersii_gcf_000493415': 293, 'yersinia_aldovae_gcf_000834395': 293, 'yersinia_frederiksenii_gcf_002591095': 293, 'yersinia_pseudotuberculosis_gcf_900637475': 293, 'yersinia_mollaretii_gcf_000167995': 293, 'yersinia_entomophaga_gcf_001656035': 293, 'yersinia_nurmii_gcf_001112925': 293, 'yersinia_ruckeri_gcf_000964565': 293, 'yersinia_canariae_gcf_009831415': 293})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 20\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/yersinia/initial_intervals/Yersinia_pestis_GCF_000009065_+20.bed --twobit results/phyluce/yersinia/cleaned_genomes/Yersinia_pestis_GCF_000009065_formatted.2bit --buffer-to 160 --output results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20.fasta;\n",
      "Screened 293 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 293.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20.fasta --probe-prefix uce_yersinia_ --design yersinia_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GG\n",
      "\n",
      "\n",
      "Conserved locus count = 292\n",
      "Probe Count = 584\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas --query results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  12:42:09\n",
      "Ended:  Thu Feb 06, 2020  12:42:09\n",
      "Time for execution:  0.00489437977473 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas                        --lastz results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes_vself.lastz                       --probe-prefix=uce_yersinia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 583 fasta sequences.  Filtered 8 duplicates. Kept 568.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/yersinia/validate_intervals/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas --scaffoldlist Yersinia_aldovae_GCF_000834395 Yersinia_aleksiciae_GCF_001047675 Yersinia_bercovieri_GCF_000167975 Yersinia_canariae_GCF_009831415 Yersinia_enterocolitica_GCF_000009345 Yersinia_entomophaga_GCF_001656035 Yersinia_frederiksenii_GCF_002591095 Yersinia_hibernica_GCF_004124235 Yersinia_intermedia_GCF_900635455 Yersinia_kristensenii_GCF_000834865 Yersinia_massiliensis_GCF_003048255 Yersinia_mollaretii_GCF_000167995 Yersinia_nurmii_GCF_001112925 Yersinia_pekkanenii_GCF_001244635 Yersinia_pseudotuberculosis_GCF_900637475 Yersinia_rohdei_GCF_000834455 Yersinia_ruckeri_GCF_000964565 Yersinia_similis_GCF_000582515 Yersinia_sp_GCF_002073315 Yersinia_wautersii_GCF_000493415 Yersinia_pestis_GCF_000009065 --genome-base-path results/phyluce/yersinia/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/yersinia/validate_intervals/yersinia-to-Yersinia_pestis_GCF_000009065.sqlite --output results/phyluce/yersinia/validate_intervals/lastz/;\n",
      "\n",
      "Running against Yersinia_aldovae_GCF_000834395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8oICsp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRlqOHR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_aldovae_GCF_000834395.lastz\n",
      "Creating Yersinia_aldovae_GCF_000834395 table\n",
      "Inserting data to Yersinia_aldovae_GCF_000834395 table\n",
      "\n",
      "Running against Yersinia_aleksiciae_GCF_001047675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphDnM0A.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkLiLq3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_aleksiciae_GCF_001047675.lastz\n",
      "Creating Yersinia_aleksiciae_GCF_001047675 table\n",
      "Inserting data to Yersinia_aleksiciae_GCF_001047675 table\n",
      "\n",
      "Running against Yersinia_bercovieri_GCF_000167975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOarGEI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNHFxrp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_bercovieri_GCF_000167975.lastz\n",
      "Creating Yersinia_bercovieri_GCF_000167975 table\n",
      "Inserting data to Yersinia_bercovieri_GCF_000167975 table\n",
      "\n",
      "Running against Yersinia_canariae_GCF_009831415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpA_f2sF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpocqwi2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_canariae_GCF_009831415.lastz\n",
      "Creating Yersinia_canariae_GCF_009831415 table\n",
      "Inserting data to Yersinia_canariae_GCF_009831415 table\n",
      "\n",
      "Running against Yersinia_enterocolitica_GCF_000009345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnTTLDf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2DoYMn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_enterocolitica_GCF_000009345.lastz\n",
      "Creating Yersinia_enterocolitica_GCF_000009345 table\n",
      "Inserting data to Yersinia_enterocolitica_GCF_000009345 table\n",
      "\n",
      "Running against Yersinia_entomophaga_GCF_001656035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5bba9J.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLRdWeN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_entomophaga_GCF_001656035.lastz\n",
      "Creating Yersinia_entomophaga_GCF_001656035 table\n",
      "Inserting data to Yersinia_entomophaga_GCF_001656035 table\n",
      "\n",
      "Running against Yersinia_frederiksenii_GCF_002591095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprSbQlG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgRELFn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_frederiksenii_GCF_002591095.lastz\n",
      "Creating Yersinia_frederiksenii_GCF_002591095 table\n",
      "Inserting data to Yersinia_frederiksenii_GCF_002591095 table\n",
      "\n",
      "Running against Yersinia_hibernica_GCF_004124235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiCwpm8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqDJqQy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_hibernica_GCF_004124235.lastz\n",
      "Creating Yersinia_hibernica_GCF_004124235 table\n",
      "Inserting data to Yersinia_hibernica_GCF_004124235 table\n",
      "\n",
      "Running against Yersinia_intermedia_GCF_900635455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprastBQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIRE8MY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_intermedia_GCF_900635455.lastz\n",
      "Creating Yersinia_intermedia_GCF_900635455 table\n",
      "Inserting data to Yersinia_intermedia_GCF_900635455 table\n",
      "\n",
      "Running against Yersinia_kristensenii_GCF_000834865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvswa4v.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJ9q3AG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_kristensenii_GCF_000834865.lastz\n",
      "Creating Yersinia_kristensenii_GCF_000834865 table\n",
      "Inserting data to Yersinia_kristensenii_GCF_000834865 table\n",
      "\n",
      "Running against Yersinia_massiliensis_GCF_003048255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5cNd3f.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvSvDUs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_massiliensis_GCF_003048255.lastz\n",
      "Creating Yersinia_massiliensis_GCF_003048255 table\n",
      "Inserting data to Yersinia_massiliensis_GCF_003048255 table\n",
      "\n",
      "Running against Yersinia_mollaretii_GCF_000167995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNY0Jyk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAJvqMr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_mollaretii_GCF_000167995.lastz\n",
      "Creating Yersinia_mollaretii_GCF_000167995 table\n",
      "Inserting data to Yersinia_mollaretii_GCF_000167995 table\n",
      "\n",
      "Running against Yersinia_nurmii_GCF_001112925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWbQzGo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl3mXN9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_nurmii_GCF_001112925.lastz\n",
      "Creating Yersinia_nurmii_GCF_001112925 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Yersinia_nurmii_GCF_001112925 table\n",
      "\n",
      "Running against Yersinia_pekkanenii_GCF_001244635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwAgoaA.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpI6vDoo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_pekkanenii_GCF_001244635.lastz\n",
      "Creating Yersinia_pekkanenii_GCF_001244635 table\n",
      "Inserting data to Yersinia_pekkanenii_GCF_001244635 table\n",
      "\n",
      "Running against Yersinia_pseudotuberculosis_GCF_900637475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp82dvin.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeaRd2t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_pseudotuberculosis_GCF_900637475.lastz\n",
      "Creating Yersinia_pseudotuberculosis_GCF_900637475 table\n",
      "Inserting data to Yersinia_pseudotuberculosis_GCF_900637475 table\n",
      "\n",
      "Running against Yersinia_rohdei_GCF_000834455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9zpWEr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpERM4Py.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_rohdei_GCF_000834455.lastz\n",
      "Creating Yersinia_rohdei_GCF_000834455 table\n",
      "Inserting data to Yersinia_rohdei_GCF_000834455 table\n",
      "\n",
      "Running against Yersinia_ruckeri_GCF_000964565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjrhphi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMZu3t3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_ruckeri_GCF_000964565.lastz\n",
      "Creating Yersinia_ruckeri_GCF_000964565 table\n",
      "Inserting data to Yersinia_ruckeri_GCF_000964565 table\n",
      "\n",
      "Running against Yersinia_similis_GCF_000582515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmTHl0Z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppicSf_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_similis_GCF_000582515.lastz\n",
      "Creating Yersinia_similis_GCF_000582515 table\n",
      "Inserting data to Yersinia_similis_GCF_000582515 table\n",
      "\n",
      "Running against Yersinia_sp_GCF_002073315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpR5mLJM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgEkQhC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_sp_GCF_002073315.lastz\n",
      "Creating Yersinia_sp_GCF_002073315 table\n",
      "Inserting data to Yersinia_sp_GCF_002073315 table\n",
      "\n",
      "Running against Yersinia_wautersii_GCF_000493415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6Yi78X.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp33pBZB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_wautersii_GCF_000493415.lastz\n",
      "Creating Yersinia_wautersii_GCF_000493415 table\n",
      "Inserting data to Yersinia_wautersii_GCF_000493415 table\n",
      "\n",
      "Running against Yersinia_pestis_GCF_000009065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVz1sp7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIQ3Swj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/yersinia/validate_intervals/lastz/Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_Yersinia_pestis_GCF_000009065.lastz\n",
      "Creating Yersinia_pestis_GCF_000009065 table\n",
      "Inserting data to Yersinia_pestis_GCF_000009065 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/yersinia/extract_probes_from_group/yersinia_genome.conf --lastz results/phyluce/yersinia/validate_intervals/lastz --probes 120 --probe-prefix uce_yersinia_ --name-pattern \"Yersinia_pestis_GCF_000009065_+20_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/yersinia/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 12:43:00,630 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 12:43:00,633 - Phyluce - INFO - -------- Working on Yersinia_aldovae_GCF_000834395 genome -------\n",
      "2020-02-06 12:43:00,634 - Phyluce - INFO - Reading Yersinia_aldovae_GCF_000834395 genome\n",
      "2020-02-06 12:43:01,712 - Phyluce - INFO - Yersinia_aldovae_GCF_000834395: 292 uces, 0 dupes, 292 non-dupes, 12 orient drop, 5 length drop, 275 written\n",
      "2020-02-06 12:43:01,712 - Phyluce - INFO - ------ Working on Yersinia_aleksiciae_GCF_001047675 genome ------\n",
      "2020-02-06 12:43:01,713 - Phyluce - INFO - Reading Yersinia_aleksiciae_GCF_001047675 genome\n",
      "2020-02-06 12:43:02,737 - Phyluce - INFO - Yersinia_aleksiciae_GCF_001047675: 292 uces, 0 dupes, 292 non-dupes, 12 orient drop, 9 length drop, 271 written\n",
      "2020-02-06 12:43:02,737 - Phyluce - INFO - ------ Working on Yersinia_bercovieri_GCF_000167975 genome ------\n",
      "2020-02-06 12:43:02,738 - Phyluce - INFO - Reading Yersinia_bercovieri_GCF_000167975 genome\n",
      "2020-02-06 12:43:03,758 - Phyluce - INFO - Yersinia_bercovieri_GCF_000167975: 292 uces, 20 dupes, 272 non-dupes, 0 orient drop, 0 length drop, 272 written\n",
      "2020-02-06 12:43:03,758 - Phyluce - INFO - ------- Working on Yersinia_canariae_GCF_009831415 genome -------\n",
      "2020-02-06 12:43:03,759 - Phyluce - INFO - Reading Yersinia_canariae_GCF_009831415 genome\n",
      "2020-02-06 12:43:04,778 - Phyluce - INFO - Yersinia_canariae_GCF_009831415: 292 uces, 0 dupes, 292 non-dupes, 8 orient drop, 10 length drop, 274 written\n",
      "2020-02-06 12:43:04,778 - Phyluce - INFO - ---- Working on Yersinia_enterocolitica_GCF_000009345 genome ----\n",
      "2020-02-06 12:43:04,779 - Phyluce - INFO - Reading Yersinia_enterocolitica_GCF_000009345 genome\n",
      "2020-02-06 12:43:05,824 - Phyluce - INFO - Yersinia_enterocolitica_GCF_000009345: 292 uces, 0 dupes, 292 non-dupes, 11 orient drop, 10 length drop, 271 written\n",
      "2020-02-06 12:43:05,824 - Phyluce - INFO - ------ Working on Yersinia_entomophaga_GCF_001656035 genome -----\n",
      "2020-02-06 12:43:05,825 - Phyluce - INFO - Reading Yersinia_entomophaga_GCF_001656035 genome\n",
      "2020-02-06 12:43:06,834 - Phyluce - INFO - Yersinia_entomophaga_GCF_001656035: 292 uces, 0 dupes, 292 non-dupes, 4 orient drop, 12 length drop, 276 written\n",
      "2020-02-06 12:43:06,835 - Phyluce - INFO - ----- Working on Yersinia_frederiksenii_GCF_002591095 genome ----\n",
      "2020-02-06 12:43:06,835 - Phyluce - INFO - Reading Yersinia_frederiksenii_GCF_002591095 genome\n",
      "2020-02-06 12:43:07,851 - Phyluce - INFO - Yersinia_frederiksenii_GCF_002591095: 292 uces, 0 dupes, 292 non-dupes, 14 orient drop, 8 length drop, 270 written\n",
      "2020-02-06 12:43:07,851 - Phyluce - INFO - ------- Working on Yersinia_hibernica_GCF_004124235 genome ------\n",
      "2020-02-06 12:43:07,855 - Phyluce - INFO - Reading Yersinia_hibernica_GCF_004124235 genome\n",
      "2020-02-06 12:43:08,887 - Phyluce - INFO - Yersinia_hibernica_GCF_004124235: 292 uces, 0 dupes, 292 non-dupes, 14 orient drop, 5 length drop, 273 written\n",
      "2020-02-06 12:43:08,887 - Phyluce - INFO - ------ Working on Yersinia_intermedia_GCF_900635455 genome ------\n",
      "2020-02-06 12:43:08,899 - Phyluce - INFO - Reading Yersinia_intermedia_GCF_900635455 genome\n",
      "2020-02-06 12:43:09,939 - Phyluce - INFO - Yersinia_intermedia_GCF_900635455: 292 uces, 0 dupes, 292 non-dupes, 13 orient drop, 7 length drop, 272 written\n",
      "2020-02-06 12:43:09,939 - Phyluce - INFO - ----- Working on Yersinia_kristensenii_GCF_000834865 genome -----\n",
      "2020-02-06 12:43:09,955 - Phyluce - INFO - Reading Yersinia_kristensenii_GCF_000834865 genome\n",
      "2020-02-06 12:43:10,994 - Phyluce - INFO - Yersinia_kristensenii_GCF_000834865: 292 uces, 0 dupes, 292 non-dupes, 17 orient drop, 2 length drop, 273 written\n",
      "2020-02-06 12:43:10,994 - Phyluce - INFO - ----- Working on Yersinia_massiliensis_GCF_003048255 genome -----\n",
      "2020-02-06 12:43:11,005 - Phyluce - INFO - Reading Yersinia_massiliensis_GCF_003048255 genome\n",
      "2020-02-06 12:43:12,031 - Phyluce - INFO - Yersinia_massiliensis_GCF_003048255: 292 uces, 0 dupes, 292 non-dupes, 18 orient drop, 4 length drop, 270 written\n",
      "2020-02-06 12:43:12,031 - Phyluce - INFO - ------ Working on Yersinia_mollaretii_GCF_000167995 genome ------\n",
      "2020-02-06 12:43:12,042 - Phyluce - INFO - Reading Yersinia_mollaretii_GCF_000167995 genome\n",
      "2020-02-06 12:43:13,057 - Phyluce - INFO - Yersinia_mollaretii_GCF_000167995: 290 uces, 21 dupes, 269 non-dupes, 0 orient drop, 0 length drop, 269 written\n",
      "2020-02-06 12:43:13,057 - Phyluce - INFO - -------- Working on Yersinia_nurmii_GCF_001112925 genome --------\n",
      "2020-02-06 12:43:13,058 - Phyluce - INFO - Reading Yersinia_nurmii_GCF_001112925 genome\n",
      "2020-02-06 12:43:14,075 - Phyluce - INFO - Yersinia_nurmii_GCF_001112925: 292 uces, 15 dupes, 277 non-dupes, 0 orient drop, 1 length drop, 276 written\n",
      "2020-02-06 12:43:14,075 - Phyluce - INFO - ------ Working on Yersinia_pekkanenii_GCF_001244635 genome ------\n",
      "2020-02-06 12:43:14,081 - Phyluce - INFO - Reading Yersinia_pekkanenii_GCF_001244635 genome\n",
      "2020-02-06 12:43:15,097 - Phyluce - INFO - Yersinia_pekkanenii_GCF_001244635: 292 uces, 17 dupes, 275 non-dupes, 0 orient drop, 0 length drop, 275 written\n",
      "2020-02-06 12:43:15,097 - Phyluce - INFO - -- Working on Yersinia_pseudotuberculosis_GCF_900637475 genome --\n",
      "2020-02-06 12:43:15,098 - Phyluce - INFO - Reading Yersinia_pseudotuberculosis_GCF_900637475 genome\n",
      "2020-02-06 12:43:16,116 - Phyluce - INFO - Yersinia_pseudotuberculosis_GCF_900637475: 292 uces, 0 dupes, 292 non-dupes, 8 orient drop, 5 length drop, 279 written\n",
      "2020-02-06 12:43:16,116 - Phyluce - INFO - -------- Working on Yersinia_rohdei_GCF_000834455 genome --------\n",
      "2020-02-06 12:43:16,117 - Phyluce - INFO - Reading Yersinia_rohdei_GCF_000834455 genome\n",
      "2020-02-06 12:43:17,143 - Phyluce - INFO - Yersinia_rohdei_GCF_000834455: 292 uces, 0 dupes, 292 non-dupes, 9 orient drop, 7 length drop, 276 written\n",
      "2020-02-06 12:43:17,143 - Phyluce - INFO - -------- Working on Yersinia_ruckeri_GCF_000964565 genome -------\n",
      "2020-02-06 12:43:17,144 - Phyluce - INFO - Reading Yersinia_ruckeri_GCF_000964565 genome\n",
      "2020-02-06 12:43:18,166 - Phyluce - INFO - Yersinia_ruckeri_GCF_000964565: 292 uces, 0 dupes, 292 non-dupes, 11 orient drop, 6 length drop, 275 written\n",
      "2020-02-06 12:43:18,166 - Phyluce - INFO - -------- Working on Yersinia_similis_GCF_000582515 genome -------\n",
      "2020-02-06 12:43:18,167 - Phyluce - INFO - Reading Yersinia_similis_GCF_000582515 genome\n",
      "2020-02-06 12:43:19,197 - Phyluce - INFO - Yersinia_similis_GCF_000582515: 292 uces, 0 dupes, 292 non-dupes, 9 orient drop, 7 length drop, 276 written\n",
      "2020-02-06 12:43:19,197 - Phyluce - INFO - ---------- Working on Yersinia_sp_GCF_002073315 genome ----------\n",
      "2020-02-06 12:43:19,198 - Phyluce - INFO - Reading Yersinia_sp_GCF_002073315 genome\n",
      "2020-02-06 12:43:20,219 - Phyluce - INFO - Yersinia_sp_GCF_002073315: 292 uces, 0 dupes, 292 non-dupes, 12 orient drop, 8 length drop, 272 written\n",
      "2020-02-06 12:43:20,219 - Phyluce - INFO - ------- Working on Yersinia_wautersii_GCF_000493415 genome ------\n",
      "2020-02-06 12:43:20,220 - Phyluce - INFO - Reading Yersinia_wautersii_GCF_000493415 genome\n",
      "2020-02-06 12:43:21,201 - Phyluce - INFO - Yersinia_wautersii_GCF_000493415: 288 uces, 13 dupes, 275 non-dupes, 0 orient drop, 0 length drop, 275 written\n",
      "2020-02-06 12:43:21,202 - Phyluce - INFO - -------- Working on Yersinia_pestis_GCF_000009065 genome --------\n",
      "2020-02-06 12:43:21,202 - Phyluce - INFO - Reading Yersinia_pestis_GCF_000009065 genome\n",
      "2020-02-06 12:43:22,213 - Phyluce - INFO - Yersinia_pestis_GCF_000009065: 292 uces, 0 dupes, 292 non-dupes, 6 orient drop, 7 length drop, 279 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/yersinia/extract_probes_from_group/probe_fasta --output results/phyluce/yersinia/extract_probes_from_group/multifastas.sqlite --base-taxon Yersinia_pestis_GCF_000009065;\n",
      "yersinia_aldovae_gcf_000834395.\n",
      "yersinia_aleksiciae_gcf_001047675.\n",
      "yersinia_bercovieri_gcf_000167975.\n",
      "yersinia_canariae_gcf_009831415.\n",
      "yersinia_enterocolitica_gcf_000009345.\n",
      "yersinia_entomophaga_gcf_001656035.\n",
      "yersinia_frederiksenii_gcf_002591095.\n",
      "yersinia_hibernica_gcf_004124235.\n",
      "yersinia_intermedia_gcf_900635455.\n",
      "yersinia_kristensenii_gcf_000834865.\n",
      "yersinia_massiliensis_gcf_003048255.\n",
      "yersinia_mollaretii_gcf_000167995.\n",
      "yersinia_nurmii_gcf_001112925.\n",
      "yersinia_pekkanenii_gcf_001244635.\n",
      "yersinia_pseudotuberculosis_gcf_900637475.\n",
      "yersinia_rohdei_gcf_000834455.\n",
      "yersinia_ruckeri_gcf_000964565.\n",
      "yersinia_similis_gcf_000582515.\n",
      "yersinia_sp_gcf_002073315.\n",
      "yersinia_wautersii_gcf_000493415.\n",
      "yersinia_pestis_gcf_000009065.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/yersinia/extract_probes_from_group/multifastas.sqlite --base-taxon Yersinia_pestis_GCF_000009065;\n",
      "Loci shared by 0 taxa:\t284.0\n",
      "Loci shared by 1 taxa:\t284.0\n",
      "Loci shared by 2 taxa:\t283.0\n",
      "Loci shared by 3 taxa:\t282.0\n",
      "Loci shared by 4 taxa:\t282.0\n",
      "Loci shared by 5 taxa:\t282.0\n",
      "Loci shared by 6 taxa:\t282.0\n",
      "Loci shared by 7 taxa:\t281.0\n",
      "Loci shared by 8 taxa:\t281.0\n",
      "Loci shared by 9 taxa:\t280.0\n",
      "Loci shared by 10 taxa:\t277.0\n",
      "Loci shared by 11 taxa:\t275.0\n",
      "Loci shared by 12 taxa:\t274.0\n",
      "Loci shared by 13 taxa:\t273.0\n",
      "Loci shared by 14 taxa:\t272.0\n",
      "Loci shared by 15 taxa:\t271.0\n",
      "Loci shared by 16 taxa:\t270.0\n",
      "Loci shared by 17 taxa:\t265.0\n",
      "Loci shared by 18 taxa:\t265.0\n",
      "Loci shared by 19 taxa:\t260.0\n",
      "Loci shared by 20 taxa:\t257.0\n",
      "Loci shared by 21 taxa:\t253.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/yersinia/extract_probes_from_group/multifastas.sqlite  --base-taxon Yersinia_pestis_GCF_000009065 --output results/phyluce/yersinia/extract_probes_from_group/Yersinia_pestis_GCF_000009065+20-back-to-21.conf --specific-counts 21;\n",
      "Counter({'yersinia_kristensenii_gcf_000834865': 253, 'yersinia_intermedia_gcf_900635455': 253, 'yersinia_pekkanenii_gcf_001244635': 253, 'yersinia_similis_gcf_000582515': 253, 'yersinia_pestis_gcf_000009065': 253, 'yersinia_bercovieri_gcf_000167975': 253, 'yersinia_rohdei_gcf_000834455': 253, 'yersinia_sp_gcf_002073315': 253, 'yersinia_hibernica_gcf_004124235': 253, 'yersinia_enterocolitica_gcf_000009345': 253, 'yersinia_aleksiciae_gcf_001047675': 253, 'yersinia_massiliensis_gcf_003048255': 253, 'yersinia_wautersii_gcf_000493415': 253, 'yersinia_aldovae_gcf_000834395': 253, 'yersinia_frederiksenii_gcf_002591095': 253, 'yersinia_pseudotuberculosis_gcf_900637475': 253, 'yersinia_mollaretii_gcf_000167995': 253, 'yersinia_entomophaga_gcf_001656035': 253, 'yersinia_nurmii_gcf_001112925': 253, 'yersinia_ruckeri_gcf_000964565': 253, 'yersinia_canariae_gcf_009831415': 253})\n",
      "Total loci = 253\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(21)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/yersinia/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/yersinia/extract_probes_from_group/Yersinia_pestis_GCF_000009065+20-back-to-21.conf --probe-prefix uce_yersinia_ --designer rnplattii --design yersinia_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list.fasta;\n",
      "GGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 253\n",
      "Probe Count = 10622\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list.fasta --query results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  12:43:38\n",
      "Ended:  Thu Feb 06, 2020  12:44:03\n",
      "Time for execution:  0.407919581731 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list.fasta --lastz results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_yersinia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 10621 fasta sequences.  Filtered 0 duplicates. Kept 10622.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/yersinia/final_probe_design/yersinia_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 14:07:22 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 10622\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 849760\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 2M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 89M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3946898\n",
      "Max number of word counting entries: 88844689\n",
      "\n",
      "# comparing sequences from          0  to       1770\n",
      ".---------- new table with      633 representatives\n",
      "# comparing sequences from       1770  to       3245\n",
      "----------    659 remaining sequences to the next cycle\n",
      "---------- new table with      214 representatives\n",
      "# comparing sequences from       2586  to       3925\n",
      "----------    601 remaining sequences to the next cycle\n",
      "---------- new table with      260 representatives\n",
      "# comparing sequences from       3324  to       4540\n",
      "----------    551 remaining sequences to the next cycle\n",
      "---------- new table with      197 representatives\n",
      "# comparing sequences from       3989  to       5094\n",
      "----------    458 remaining sequences to the next cycle\n",
      "---------- new table with      167 representatives\n",
      "# comparing sequences from       4636  to       5633\n",
      "----------    391 remaining sequences to the next cycle\n",
      "---------- new table with      147 representatives\n",
      "# comparing sequences from       5242  to       6138\n",
      "----------    306 remaining sequences to the next cycle\n",
      "---------- new table with      212 representatives\n",
      "# comparing sequences from       5832  to       6630\n",
      "----------    415 remaining sequences to the next cycle\n",
      "---------- new table with      108 representatives\n",
      "# comparing sequences from       6215  to       6949\n",
      "----------    318 remaining sequences to the next cycle\n",
      "---------- new table with      107 representatives\n",
      "# comparing sequences from       6631  to       7296\n",
      "----------    156 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       7140  to       7720\n",
      "----------    232 remaining sequences to the next cycle\n",
      "---------- new table with      133 representatives\n",
      "# comparing sequences from       7488  to       8010\n",
      "----------    201 remaining sequences to the next cycle\n",
      "---------- new table with      101 representatives\n",
      "# comparing sequences from       7809  to       8277\n",
      "----------    162 remaining sequences to the next cycle\n",
      "---------- new table with      112 representatives\n",
      "# comparing sequences from       8115  to       8532\n",
      "----------    144 remaining sequences to the next cycle\n",
      "---------- new table with      126 representatives\n",
      "# comparing sequences from       8388  to       8760\n",
      "....................---------- new table with       91 representatives\n",
      "# comparing sequences from       8760  to       9070\n",
      "....................---------- new table with       72 representatives\n",
      "# comparing sequences from       9070  to       9328\n",
      "100.0%---------- new table with       81 representatives\n",
      "# comparing sequences from       9328  to       9543\n",
      "....................---------- new table with       99 representatives\n",
      "# comparing sequences from       9543  to       9722\n",
      "....................---------- new table with       98 representatives\n",
      "# comparing sequences from       9722  to      10622\n",
      "..........    10000  finished       3116  clusters\n",
      "---------- new table with      215 representatives\n",
      "\n",
      "    10622  finished       3273  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 89M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.82\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "319px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
