{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import shutil\n",
    "from shutil import copy\n",
    "import time\n",
    "from collections import defaultdict\n",
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "os.chdir(\"/master/nplatt/pathogen_probes/\")\n",
    "\n",
    "\n",
    "def wait_on_running_jobs():\n",
    "   \n",
    "    num_jobs = 1\n",
    "    \n",
    "    while num_jobs > 0:\n",
    "        num_jobs = len(subprocess.check_output('qstat', shell=True).split(\"\\n\")) - 2\n",
    "        time.sleep(60)\n",
    "        print(\".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Anaplasma"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'anaplasma'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Anaplasma_marginale_GCF_000011945\",\n",
    "               \"Anaplasma_marginale_GCF_000020305\",\n",
    "               \"Anaplasma_marginale_GCF_000172475\",\n",
    "               \"Anaplasma_marginale_GCF_000172495\",\n",
    "               \"Anaplasma_marginale_GCF_000172515\",\n",
    "               \"Anaplasma_marginale_GCF_000215485\",\n",
    "               \"Anaplasma_marginale_GCF_000215505\",\n",
    "               \"Anaplasma_marginale_GCF_000215525\",\n",
    "               \"Anaplasma_marginale_GCF_000215545\",\n",
    "               \"Anaplasma_marginale_GCF_000215565\",\n",
    "               \"Anaplasma_marginale_GCF_000215585\",\n",
    "               \"Anaplasma_marginale_GCF_000215605\",\n",
    "               \"Anaplasma_marginale_GCF_000495495\",\n",
    "               \"Anaplasma_marginale_GCF_000495535\",\n",
    "               \"Anaplasma_marginale_GCF_002849365\",\n",
    "               \"Anaplasma_marginale_GCF_003331125\",\n",
    "               \"Anaplasma_marginale_GCF_003515675\",\n",
    "               \"Anaplasma_marginale_GCF_003515735\",\n",
    "               \"Anaplasma_marginale_GCF_008274665\",\n",
    "               \"Anaplasma_marginale_GCF_008690255\",\n",
    "               \"Anaplasma_marginale_GCF_008690265\",\n",
    "               \"Anaplasma_marginale_GCF_008801275\",\n",
    "               \"Anaplasma_marginale_GCF_008801305\",\n",
    "               \"Anaplasma_marginale_GCF_008801325\",\n",
    "               \"Anaplasma_ovis_GCF_002214625\",\n",
    "               \"Anaplasma_ovis_GCF_002849345\",\n",
    "               \"Anaplasma_phagocytophilum_GCA_900088675\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000439755\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000439775\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000439795\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000478425\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000478445\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000689615\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000689635\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000689655\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964685\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964725\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964745\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964785\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964915\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964935\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964945\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000964985\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000965125\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000968455\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_000968465\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_002849375\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900000025\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900078505\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900088605\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900088615\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900088625\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900088645\",\n",
    "               \"Anaplasma_phagocytophilum_GCF_900088665\" ]\n",
    "\n",
    "reference_taxon = \"Anaplasma_phagocytophilum_GCF_000013125\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000024505.1_ASM2450v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 360892 bytes  42462.82 bytes/sec\n",
      "total size is 360692  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000011945.1_ASM1194v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 358841 bytes  239255.33 bytes/sec\n",
      "total size is 358649  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000020305.1_ASM2030v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 359789 bytes  239887.33 bytes/sec\n",
      "total size is 359597  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000172475.1_ASM17247v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 346813 bytes  231236.67 bytes/sec\n",
      "total size is 346620  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000172495.1_ASM17249v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 350932 bytes  701948.00 bytes/sec\n",
      "total size is 350739  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000172515.1_ASM17251v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 350097 bytes  233426.00 bytes/sec\n",
      "total size is 349904  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000215485.2_ASM21548v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 346390 bytes  53297.23 bytes/sec\n",
      "total size is 346197  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000215505.2_ASM21550v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 351276 bytes  234212.00 bytes/sec\n",
      "total size is 351083  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000215525.2_ASM21552v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 351551 bytes  234395.33 bytes/sec\n",
      "total size is 351358  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000215545.2_ASM21554v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 421281 bytes  280882.00 bytes/sec\n",
      "total size is 421072  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000215565.2_ASM21556v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 351375 bytes  702834.00 bytes/sec\n",
      "total size is 351182  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000215585.2_ASM21558v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 429018 bytes  286040.00 bytes/sec\n",
      "total size is 428801  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000215605.2_ASM21560v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 425341 bytes  283588.67 bytes/sec\n",
      "total size is 425132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000495495.1_ASM49549v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 347107 bytes  231432.67 bytes/sec\n",
      "total size is 346914  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000495535.1_ASM49553v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 338029 bytes  135228.40 bytes/sec\n",
      "total size is 337836  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002849365.1_ASM284936v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 353351 bytes  141357.20 bytes/sec\n",
      "total size is 353157  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003331125.1_ASM333112v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 357598 bytes  238426.67 bytes/sec\n",
      "total size is 357404  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003515675.1_ASM351567v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 360417 bytes  240306.00 bytes/sec\n",
      "total size is 360223  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003515735.1_ASM351573v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 359685 bytes  239818.00 bytes/sec\n",
      "total size is 359491  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008274665.1_ASM827466v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 356067 bytes  237406.00 bytes/sec\n",
      "total size is 355873  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008690255.1_ASM869025v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 354993 bytes  236690.00 bytes/sec\n",
      "total size is 354799  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_008690265.1_ASM869026v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 357949 bytes  715982.00 bytes/sec\n",
      "total size is 357755  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008801275.1_ASM880127v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 356136 bytes  237452.00 bytes/sec\n",
      "total size is 355942  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008801305.1_ASM880130v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 357759 bytes  238534.00 bytes/sec\n",
      "total size is 357565  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008801325.1_ASM880132v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 354946 bytes  236658.67 bytes/sec\n",
      "total size is 354752  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002214625.2_ASM221462v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 359938 bytes  239986.67 bytes/sec\n",
      "total size is 359744  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002849345.1_ASM284934v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 352803 bytes  235230.00 bytes/sec\n",
      "total size is 352609  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900088675.2_ANAPH1_2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 353307 bytes  235566.00 bytes/sec\n",
      "total size is 353116  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000013125.1_ASM1312v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 430037 bytes  286719.33 bytes/sec\n",
      "total size is 429821  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000439755.1_ASM43975v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 430765 bytes  287204.67 bytes/sec\n",
      "total size is 430548  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000439775.1_ASM43977v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 430904 bytes  45362.74 bytes/sec\n",
      "total size is 430687  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000439795.1_ASM43979v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 430652 bytes  287129.33 bytes/sec\n",
      "total size is 430435  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000478425.1_ASM47842v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 430437 bytes  860958.00 bytes/sec\n",
      "total size is 430220  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000478445.1_ASM47844v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 442389 bytes  294954.00 bytes/sec\n",
      "total size is 442172  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000689615.1_ApCRT35_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 426909 bytes  284634.00 bytes/sec\n",
      "total size is 426691  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000689635.2_ASM68963v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 445172 bytes  296809.33 bytes/sec\n",
      "total size is 444955  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000689655.1_MRK1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 432733 bytes  288516.67 bytes/sec\n",
      "total size is 432520  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964685.1_ASM96468v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 431651 bytes  287795.33 bytes/sec\n",
      "total size is 431434  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964725.1_ASM96472v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 440829 bytes  293914.00 bytes/sec\n",
      "total size is 440612  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964745.1_ASM96474v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 438875 bytes  292611.33 bytes/sec\n",
      "total size is 438658  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964785.1_ASM96478v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 442153 bytes  294796.67 bytes/sec\n",
      "total size is 441936  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964915.1_ASM96491v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 448008 bytes  298700.00 bytes/sec\n",
      "total size is 447791  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964935.1_ASM96493v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 431792 bytes  287889.33 bytes/sec\n",
      "total size is 431575  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000964945.1_ASM96494v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 435834 bytes  290584.00 bytes/sec\n",
      "total size is 435617  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000964985.1_ASM96498v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 439469 bytes  175804.40 bytes/sec\n",
      "total size is 439252  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000965125.1_ASM96512v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 441930 bytes  176788.80 bytes/sec\n",
      "total size is 441713  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000968455.1_ASM96845v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 434033 bytes  289383.33 bytes/sec\n",
      "total size is 433816  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000968465.1_ASM96846v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 434577 bytes  289746.00 bytes/sec\n",
      "total size is 434360  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002849375.1_ASM284937v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 418822 bytes  279242.67 bytes/sec\n",
      "total size is 418612  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900000025.1_AnapV1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 412337 bytes  164951.60 bytes/sec\n",
      "total size is 412132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900078505.1_ANAP_C1_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 500456 bytes  200199.20 bytes/sec\n",
      "total size is 500223  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900088605.1_ANAPC2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 487341 bytes  324922.00 bytes/sec\n",
      "total size is 487120  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900088615.1_ANAPC4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 478097 bytes  956278.00 bytes/sec\n",
      "total size is 477876  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900088625.1_ANAPC3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 465028 bytes  310046.67 bytes/sec\n",
      "total size is 464807  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GCA_900088645.2_ANAPRD1_2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 478411 bytes  50363.47 bytes/sec\n",
      "total size is 478187  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900088665.2_ANAPC5_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 519887 bytes  346619.33 bytes/sec\n",
      "total size is 519658  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip -f data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483454 (\"sim_Anaplasma_marginale_GCF_000011945\") has been submitted\n",
      "Your job 5483456 (\"sim_Anaplasma_marginale_GCF_000020305\") has been submitted\n",
      "Your job 5483457 (\"sim_Anaplasma_marginale_GCF_000172475\") has been submitted\n",
      "Your job 5483459 (\"sim_Anaplasma_marginale_GCF_000172495\") has been submitted\n",
      "Your job 5483461 (\"sim_Anaplasma_marginale_GCF_000172515\") has been submitted\n",
      "Your job 5483463 (\"sim_Anaplasma_marginale_GCF_000215485\") has been submitted\n",
      "Your job 5483465 (\"sim_Anaplasma_marginale_GCF_000215505\") has been submitted\n",
      "Your job 5483467 (\"sim_Anaplasma_marginale_GCF_000215525\") has been submitted\n",
      "Your job 5483469 (\"sim_Anaplasma_marginale_GCF_000215545\") has been submitted\n",
      "Your job 5483471 (\"sim_Anaplasma_marginale_GCF_000215565\") has been submitted\n",
      "Your job 5483473 (\"sim_Anaplasma_marginale_GCF_000215585\") has been submitted\n",
      "Your job 5483475 (\"sim_Anaplasma_marginale_GCF_000215605\") has been submitted\n",
      "Your job 5483477 (\"sim_Anaplasma_marginale_GCF_000495495\") has been submitted\n",
      "Your job 5483479 (\"sim_Anaplasma_marginale_GCF_000495535\") has been submitted\n",
      "Your job 5483481 (\"sim_Anaplasma_marginale_GCF_002849365\") has been submitted\n",
      "Your job 5483483 (\"sim_Anaplasma_marginale_GCF_003331125\") has been submitted\n",
      "Your job 5483485 (\"sim_Anaplasma_marginale_GCF_003515675\") has been submitted\n",
      "Your job 5483487 (\"sim_Anaplasma_marginale_GCF_003515735\") has been submitted\n",
      "Your job 5483489 (\"sim_Anaplasma_marginale_GCF_008274665\") has been submitted\n",
      "Your job 5483491 (\"sim_Anaplasma_marginale_GCF_008690255\") has been submitted\n",
      "Your job 5483493 (\"sim_Anaplasma_marginale_GCF_008690265\") has been submitted\n",
      "Your job 5483495 (\"sim_Anaplasma_marginale_GCF_008801275\") has been submitted\n",
      "Your job 5483497 (\"sim_Anaplasma_marginale_GCF_008801305\") has been submitted\n",
      "Your job 5483499 (\"sim_Anaplasma_marginale_GCF_008801325\") has been submitted\n",
      "Your job 5483501 (\"sim_Anaplasma_ovis_GCF_002214625\") has been submitted\n",
      "Your job 5483503 (\"sim_Anaplasma_ovis_GCF_002849345\") has been submitted\n",
      "Your job 5483505 (\"sim_Anaplasma_phagocytophilum_GCA_900088675\") has been submitted\n",
      "Your job 5483507 (\"sim_Anaplasma_phagocytophilum_GCF_000439755\") has been submitted\n",
      "Your job 5483509 (\"sim_Anaplasma_phagocytophilum_GCF_000439775\") has been submitted\n",
      "Your job 5483511 (\"sim_Anaplasma_phagocytophilum_GCF_000439795\") has been submitted\n",
      "Your job 5483513 (\"sim_Anaplasma_phagocytophilum_GCF_000478425\") has been submitted\n",
      "Your job 5483515 (\"sim_Anaplasma_phagocytophilum_GCF_000478445\") has been submitted\n",
      "Your job 5483517 (\"sim_Anaplasma_phagocytophilum_GCF_000689615\") has been submitted\n",
      "Your job 5483519 (\"sim_Anaplasma_phagocytophilum_GCF_000689635\") has been submitted\n",
      "Your job 5483521 (\"sim_Anaplasma_phagocytophilum_GCF_000689655\") has been submitted\n",
      "Your job 5483523 (\"sim_Anaplasma_phagocytophilum_GCF_000964685\") has been submitted\n",
      "Your job 5483525 (\"sim_Anaplasma_phagocytophilum_GCF_000964725\") has been submitted\n",
      "Your job 5483527 (\"sim_Anaplasma_phagocytophilum_GCF_000964745\") has been submitted\n",
      "Your job 5483529 (\"sim_Anaplasma_phagocytophilum_GCF_000964785\") has been submitted\n",
      "Your job 5483531 (\"sim_Anaplasma_phagocytophilum_GCF_000964915\") has been submitted\n",
      "Your job 5483533 (\"sim_Anaplasma_phagocytophilum_GCF_000964935\") has been submitted\n",
      "Your job 5483535 (\"sim_Anaplasma_phagocytophilum_GCF_000964945\") has been submitted\n",
      "Your job 5483537 (\"sim_Anaplasma_phagocytophilum_GCF_000964985\") has been submitted\n",
      "Your job 5483539 (\"sim_Anaplasma_phagocytophilum_GCF_000965125\") has been submitted\n",
      "Your job 5483541 (\"sim_Anaplasma_phagocytophilum_GCF_000968455\") has been submitted\n",
      "Your job 5483543 (\"sim_Anaplasma_phagocytophilum_GCF_000968465\") has been submitted\n",
      "Your job 5483545 (\"sim_Anaplasma_phagocytophilum_GCF_002849375\") has been submitted\n",
      "Your job 5483547 (\"sim_Anaplasma_phagocytophilum_GCF_900000025\") has been submitted\n",
      "Your job 5483549 (\"sim_Anaplasma_phagocytophilum_GCF_900078505\") has been submitted\n",
      "Your job 5483551 (\"sim_Anaplasma_phagocytophilum_GCF_900088605\") has been submitted\n",
      "Your job 5483553 (\"sim_Anaplasma_phagocytophilum_GCF_900088615\") has been submitted\n",
      "Your job 5483555 (\"sim_Anaplasma_phagocytophilum_GCF_900088625\") has been submitted\n",
      "Your job 5483557 (\"sim_Anaplasma_phagocytophilum_GCF_900088645\") has been submitted\n",
      "Your job 5483559 (\"sim_Anaplasma_phagocytophilum_GCF_900088665\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/anaplasma/cleaned_genomes/Anaplasma_phagocytophilum_GCF_000013125_formatted.fas path=results/phyluce/anaplasma/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5483598 (\"map_Anaplasma_marginale_GCF_000011945\") has been submitted\n",
      "Your job 5483600 (\"map_Anaplasma_marginale_GCF_000020305\") has been submitted\n",
      "Your job 5483602 (\"map_Anaplasma_marginale_GCF_000172475\") has been submitted\n",
      "Your job 5483604 (\"map_Anaplasma_marginale_GCF_000172495\") has been submitted\n",
      "Your job 5483606 (\"map_Anaplasma_marginale_GCF_000172515\") has been submitted\n",
      "Your job 5483608 (\"map_Anaplasma_marginale_GCF_000215485\") has been submitted\n",
      "Your job 5483610 (\"map_Anaplasma_marginale_GCF_000215505\") has been submitted\n",
      "Your job 5483612 (\"map_Anaplasma_marginale_GCF_000215525\") has been submitted\n",
      "Your job 5483614 (\"map_Anaplasma_marginale_GCF_000215545\") has been submitted\n",
      "Your job 5483616 (\"map_Anaplasma_marginale_GCF_000215565\") has been submitted\n",
      "Your job 5483618 (\"map_Anaplasma_marginale_GCF_000215585\") has been submitted\n",
      "Your job 5483620 (\"map_Anaplasma_marginale_GCF_000215605\") has been submitted\n",
      "Your job 5483622 (\"map_Anaplasma_marginale_GCF_000495495\") has been submitted\n",
      "Your job 5483624 (\"map_Anaplasma_marginale_GCF_000495535\") has been submitted\n",
      "Your job 5483626 (\"map_Anaplasma_marginale_GCF_002849365\") has been submitted\n",
      "Your job 5483628 (\"map_Anaplasma_marginale_GCF_003331125\") has been submitted\n",
      "Your job 5483630 (\"map_Anaplasma_marginale_GCF_003515675\") has been submitted\n",
      "Your job 5483632 (\"map_Anaplasma_marginale_GCF_003515735\") has been submitted\n",
      "Your job 5483634 (\"map_Anaplasma_marginale_GCF_008274665\") has been submitted\n",
      "Your job 5483636 (\"map_Anaplasma_marginale_GCF_008690255\") has been submitted\n",
      "Your job 5483638 (\"map_Anaplasma_marginale_GCF_008690265\") has been submitted\n",
      "Your job 5483640 (\"map_Anaplasma_marginale_GCF_008801275\") has been submitted\n",
      "Your job 5483642 (\"map_Anaplasma_marginale_GCF_008801305\") has been submitted\n",
      "Your job 5483644 (\"map_Anaplasma_marginale_GCF_008801325\") has been submitted\n",
      "Your job 5483646 (\"map_Anaplasma_ovis_GCF_002214625\") has been submitted\n",
      "Your job 5483648 (\"map_Anaplasma_ovis_GCF_002849345\") has been submitted\n",
      "Your job 5483650 (\"map_Anaplasma_phagocytophilum_GCA_900088675\") has been submitted\n",
      "Your job 5483652 (\"map_Anaplasma_phagocytophilum_GCF_000439755\") has been submitted\n",
      "Your job 5483654 (\"map_Anaplasma_phagocytophilum_GCF_000439775\") has been submitted\n",
      "Your job 5483656 (\"map_Anaplasma_phagocytophilum_GCF_000439795\") has been submitted\n",
      "Your job 5483658 (\"map_Anaplasma_phagocytophilum_GCF_000478425\") has been submitted\n",
      "Your job 5483660 (\"map_Anaplasma_phagocytophilum_GCF_000478445\") has been submitted\n",
      "Your job 5483662 (\"map_Anaplasma_phagocytophilum_GCF_000689615\") has been submitted\n",
      "Your job 5483664 (\"map_Anaplasma_phagocytophilum_GCF_000689635\") has been submitted\n",
      "Your job 5483666 (\"map_Anaplasma_phagocytophilum_GCF_000689655\") has been submitted\n",
      "Your job 5483668 (\"map_Anaplasma_phagocytophilum_GCF_000964685\") has been submitted\n",
      "Your job 5483670 (\"map_Anaplasma_phagocytophilum_GCF_000964725\") has been submitted\n",
      "Your job 5483672 (\"map_Anaplasma_phagocytophilum_GCF_000964745\") has been submitted\n",
      "Your job 5483674 (\"map_Anaplasma_phagocytophilum_GCF_000964785\") has been submitted\n",
      "Your job 5483676 (\"map_Anaplasma_phagocytophilum_GCF_000964915\") has been submitted\n",
      "Your job 5483678 (\"map_Anaplasma_phagocytophilum_GCF_000964935\") has been submitted\n",
      "Your job 5483680 (\"map_Anaplasma_phagocytophilum_GCF_000964945\") has been submitted\n",
      "Your job 5483682 (\"map_Anaplasma_phagocytophilum_GCF_000964985\") has been submitted\n",
      "Your job 5483684 (\"map_Anaplasma_phagocytophilum_GCF_000965125\") has been submitted\n",
      "Your job 5483686 (\"map_Anaplasma_phagocytophilum_GCF_000968455\") has been submitted\n",
      "Your job 5483688 (\"map_Anaplasma_phagocytophilum_GCF_000968465\") has been submitted\n",
      "Your job 5483690 (\"map_Anaplasma_phagocytophilum_GCF_002849375\") has been submitted\n",
      "Your job 5483692 (\"map_Anaplasma_phagocytophilum_GCF_900000025\") has been submitted\n",
      "Your job 5483694 (\"map_Anaplasma_phagocytophilum_GCF_900078505\") has been submitted\n",
      "Your job 5483696 (\"map_Anaplasma_phagocytophilum_GCF_900088605\") has been submitted\n",
      "Your job 5483698 (\"map_Anaplasma_phagocytophilum_GCF_900088615\") has been submitted\n",
      "Your job 5483700 (\"map_Anaplasma_phagocytophilum_GCF_900088625\") has been submitted\n",
      "Your job 5483702 (\"map_Anaplasma_phagocytophilum_GCF_900088645\") has been submitted\n",
      "Your job 5483704 (\"map_Anaplasma_phagocytophilum_GCF_900088665\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5484252 (\"merge_Anaplasma_marginale_GCF_000011945\") has been submitted\n",
      "Your job 5484253 (\"merge_Anaplasma_marginale_GCF_000020305\") has been submitted\n",
      "Your job 5484254 (\"merge_Anaplasma_marginale_GCF_000172475\") has been submitted\n",
      "Your job 5484255 (\"merge_Anaplasma_marginale_GCF_000172495\") has been submitted\n",
      "Your job 5484256 (\"merge_Anaplasma_marginale_GCF_000172515\") has been submitted\n",
      "Your job 5484257 (\"merge_Anaplasma_marginale_GCF_000215485\") has been submitted\n",
      "Your job 5484258 (\"merge_Anaplasma_marginale_GCF_000215505\") has been submitted\n",
      "Your job 5484259 (\"merge_Anaplasma_marginale_GCF_000215525\") has been submitted\n",
      "Your job 5484260 (\"merge_Anaplasma_marginale_GCF_000215545\") has been submitted\n",
      "Your job 5484261 (\"merge_Anaplasma_marginale_GCF_000215565\") has been submitted\n",
      "Your job 5484262 (\"merge_Anaplasma_marginale_GCF_000215585\") has been submitted\n",
      "Your job 5484263 (\"merge_Anaplasma_marginale_GCF_000215605\") has been submitted\n",
      "Your job 5484264 (\"merge_Anaplasma_marginale_GCF_000495495\") has been submitted\n",
      "Your job 5484265 (\"merge_Anaplasma_marginale_GCF_000495535\") has been submitted\n",
      "Your job 5484266 (\"merge_Anaplasma_marginale_GCF_002849365\") has been submitted\n",
      "Your job 5484267 (\"merge_Anaplasma_marginale_GCF_003331125\") has been submitted\n",
      "Your job 5484268 (\"merge_Anaplasma_marginale_GCF_003515675\") has been submitted\n",
      "Your job 5484269 (\"merge_Anaplasma_marginale_GCF_003515735\") has been submitted\n",
      "Your job 5484270 (\"merge_Anaplasma_marginale_GCF_008274665\") has been submitted\n",
      "Your job 5484271 (\"merge_Anaplasma_marginale_GCF_008690255\") has been submitted\n",
      "Your job 5484272 (\"merge_Anaplasma_marginale_GCF_008690265\") has been submitted\n",
      "Your job 5484273 (\"merge_Anaplasma_marginale_GCF_008801275\") has been submitted\n",
      "Your job 5484274 (\"merge_Anaplasma_marginale_GCF_008801305\") has been submitted\n",
      "Your job 5484275 (\"merge_Anaplasma_marginale_GCF_008801325\") has been submitted\n",
      "Your job 5484276 (\"merge_Anaplasma_ovis_GCF_002214625\") has been submitted\n",
      "Your job 5484277 (\"merge_Anaplasma_ovis_GCF_002849345\") has been submitted\n",
      "Your job 5484278 (\"merge_Anaplasma_phagocytophilum_GCA_900088675\") has been submitted\n",
      "Your job 5484279 (\"merge_Anaplasma_phagocytophilum_GCF_000439755\") has been submitted\n",
      "Your job 5484280 (\"merge_Anaplasma_phagocytophilum_GCF_000439775\") has been submitted\n",
      "Your job 5484281 (\"merge_Anaplasma_phagocytophilum_GCF_000439795\") has been submitted\n",
      "Your job 5484282 (\"merge_Anaplasma_phagocytophilum_GCF_000478425\") has been submitted\n",
      "Your job 5484283 (\"merge_Anaplasma_phagocytophilum_GCF_000478445\") has been submitted\n",
      "Your job 5484284 (\"merge_Anaplasma_phagocytophilum_GCF_000689615\") has been submitted\n",
      "Your job 5484285 (\"merge_Anaplasma_phagocytophilum_GCF_000689635\") has been submitted\n",
      "Your job 5484286 (\"merge_Anaplasma_phagocytophilum_GCF_000689655\") has been submitted\n",
      "Your job 5484287 (\"merge_Anaplasma_phagocytophilum_GCF_000964685\") has been submitted\n",
      "Your job 5484288 (\"merge_Anaplasma_phagocytophilum_GCF_000964725\") has been submitted\n",
      "Your job 5484289 (\"merge_Anaplasma_phagocytophilum_GCF_000964745\") has been submitted\n",
      "Your job 5484290 (\"merge_Anaplasma_phagocytophilum_GCF_000964785\") has been submitted\n",
      "Your job 5484291 (\"merge_Anaplasma_phagocytophilum_GCF_000964915\") has been submitted\n",
      "Your job 5484292 (\"merge_Anaplasma_phagocytophilum_GCF_000964935\") has been submitted\n",
      "Your job 5484293 (\"merge_Anaplasma_phagocytophilum_GCF_000964945\") has been submitted\n",
      "Your job 5484294 (\"merge_Anaplasma_phagocytophilum_GCF_000964985\") has been submitted\n",
      "Your job 5484295 (\"merge_Anaplasma_phagocytophilum_GCF_000965125\") has been submitted\n",
      "Your job 5484296 (\"merge_Anaplasma_phagocytophilum_GCF_000968455\") has been submitted\n",
      "Your job 5484297 (\"merge_Anaplasma_phagocytophilum_GCF_000968465\") has been submitted\n",
      "Your job 5484298 (\"merge_Anaplasma_phagocytophilum_GCF_002849375\") has been submitted\n",
      "Your job 5484299 (\"merge_Anaplasma_phagocytophilum_GCF_900000025\") has been submitted\n",
      "Your job 5484300 (\"merge_Anaplasma_phagocytophilum_GCF_900078505\") has been submitted\n",
      "Your job 5484301 (\"merge_Anaplasma_phagocytophilum_GCF_900088605\") has been submitted\n",
      "Your job 5484302 (\"merge_Anaplasma_phagocytophilum_GCF_900088615\") has been submitted\n",
      "Your job 5484303 (\"merge_Anaplasma_phagocytophilum_GCF_900088625\") has been submitted\n",
      "Your job 5484304 (\"merge_Anaplasma_phagocytophilum_GCF_900088645\") has been submitted\n",
      "Your job 5484305 (\"merge_Anaplasma_phagocytophilum_GCF_900088665\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 18 sequences from Anaplasma_marginale_GCF_000011945_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 21 sequences from Anaplasma_marginale_GCF_000020305_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 24 sequences from Anaplasma_marginale_GCF_000172475_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 23 sequences from Anaplasma_marginale_GCF_000172495_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 20 sequences from Anaplasma_marginale_GCF_000172515_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 23 sequences from Anaplasma_marginale_GCF_000215485_merged.bed.  Filtered 19 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 21 sequences from Anaplasma_marginale_GCF_000215505_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 23 sequences from Anaplasma_marginale_GCF_000215525_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 24 sequences from Anaplasma_marginale_GCF_000215545_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 21 sequences from Anaplasma_marginale_GCF_000215565_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 25 sequences from Anaplasma_marginale_GCF_000215585_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 19 sequences from Anaplasma_marginale_GCF_000215605_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 24 sequences from Anaplasma_marginale_GCF_000495495_merged.bed.  Filtered 21 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3.\n",
      "Screened 27 sequences from Anaplasma_marginale_GCF_000495535_merged.bed.  Filtered 23 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 21 sequences from Anaplasma_marginale_GCF_002849365_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 19 sequences from Anaplasma_marginale_GCF_003331125_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 22 sequences from Anaplasma_marginale_GCF_003515675_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4.\n",
      "Screened 23 sequences from Anaplasma_marginale_GCF_003515735_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 19 sequences from Anaplasma_marginale_GCF_008274665_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 23 sequences from Anaplasma_marginale_GCF_008690255_merged.bed.  Filtered 18 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 22 sequences from Anaplasma_marginale_GCF_008690265_merged.bed.  Filtered 17 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 20 sequences from Anaplasma_marginale_GCF_008801275_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 20 sequences from Anaplasma_marginale_GCF_008801305_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5.\n",
      "Screened 23 sequences from Anaplasma_marginale_GCF_008801325_merged.bed.  Filtered 16 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 21 sequences from Anaplasma_ovis_GCF_002214625_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 23 sequences from Anaplasma_ovis_GCF_002849345_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 829 sequences from Anaplasma_phagocytophilum_GCA_900088675_merged.bed.  Filtered 212 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 617.\n",
      "Screened 165 sequences from Anaplasma_phagocytophilum_GCF_000439755_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 152.\n",
      "Screened 225 sequences from Anaplasma_phagocytophilum_GCF_000439775_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 200.\n",
      "Screened 215 sequences from Anaplasma_phagocytophilum_GCF_000439795_merged.bed.  Filtered 30 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 185.\n",
      "Screened 223 sequences from Anaplasma_phagocytophilum_GCF_000478425_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 196.\n",
      "Screened 717 sequences from Anaplasma_phagocytophilum_GCF_000478445_merged.bed.  Filtered 159 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 558.\n",
      "Screened 752 sequences from Anaplasma_phagocytophilum_GCF_000689615_merged.bed.  Filtered 182 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 570.\n",
      "Screened 784 sequences from Anaplasma_phagocytophilum_GCF_000689635_merged.bed.  Filtered 177 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 607.\n",
      "Screened 450 sequences from Anaplasma_phagocytophilum_GCF_000689655_merged.bed.  Filtered 106 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 344.\n",
      "Screened 230 sequences from Anaplasma_phagocytophilum_GCF_000964685_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 205.\n",
      "Screened 169 sequences from Anaplasma_phagocytophilum_GCF_000964725_merged.bed.  Filtered 12 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 157.\n",
      "Screened 812 sequences from Anaplasma_phagocytophilum_GCF_000964745_merged.bed.  Filtered 192 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 620.\n",
      "Screened 805 sequences from Anaplasma_phagocytophilum_GCF_000964785_merged.bed.  Filtered 183 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 622.\n",
      "Screened 737 sequences from Anaplasma_phagocytophilum_GCF_000964915_merged.bed.  Filtered 179 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 558.\n",
      "Screened 219 sequences from Anaplasma_phagocytophilum_GCF_000964935_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 192.\n",
      "Screened 212 sequences from Anaplasma_phagocytophilum_GCF_000964945_merged.bed.  Filtered 26 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 186.\n",
      "Screened 171 sequences from Anaplasma_phagocytophilum_GCF_000964985_merged.bed.  Filtered 9 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 162.\n",
      "Screened 224 sequences from Anaplasma_phagocytophilum_GCF_000965125_merged.bed.  Filtered 33 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 191.\n",
      "Screened 223 sequences from Anaplasma_phagocytophilum_GCF_000968455_merged.bed.  Filtered 30 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 193.\n",
      "Screened 219 sequences from Anaplasma_phagocytophilum_GCF_000968465_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 191.\n",
      "Screened 274 sequences from Anaplasma_phagocytophilum_GCF_002849375_merged.bed.  Filtered 42 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 232.\n",
      "Screened 761 sequences from Anaplasma_phagocytophilum_GCF_900000025_merged.bed.  Filtered 180 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 581.\n",
      "Screened 930 sequences from Anaplasma_phagocytophilum_GCF_900078505_merged.bed.  Filtered 233 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 697.\n",
      "Screened 962 sequences from Anaplasma_phagocytophilum_GCF_900088605_merged.bed.  Filtered 245 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 717.\n",
      "Screened 926 sequences from Anaplasma_phagocytophilum_GCF_900088615_merged.bed.  Filtered 252 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 674.\n",
      "Screened 923 sequences from Anaplasma_phagocytophilum_GCF_900088625_merged.bed.  Filtered 226 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 697.\n",
      "Screened 836 sequences from Anaplasma_phagocytophilum_GCF_900088645_merged.bed.  Filtered 225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 611.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 882 sequences from Anaplasma_phagocytophilum_GCF_900088665_merged.bed.  Filtered 225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 657.\r\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/anaplasma/initial_intervals/Anaplasma_phagocytophilum_GCF_900088665_merged.bed --twobit results/phyluce/anaplasma/cleaned_genomes/Anaplasma_phagocytophilum_GCF_000013125_formatted.2bit --output results/phyluce/anaplasma/initial_intervals/Anaplasma_phagocytophilum_GCF_900088665_stripped.bed;\n",
      "anaplasma_marginale_gcf_000011945.\n",
      "anaplasma_marginale_gcf_000020305.\n",
      "anaplasma_marginale_gcf_000172475.\n",
      "anaplasma_marginale_gcf_000172495.\n",
      "anaplasma_marginale_gcf_000172515.\n",
      "anaplasma_marginale_gcf_000215485.\n",
      "anaplasma_marginale_gcf_000215505.\n",
      "anaplasma_marginale_gcf_000215525.\n",
      "anaplasma_marginale_gcf_000215545.\n",
      "anaplasma_marginale_gcf_000215565.\n",
      "anaplasma_marginale_gcf_000215585.\n",
      "anaplasma_marginale_gcf_000215605.\n",
      "anaplasma_marginale_gcf_000495495.\n",
      "anaplasma_marginale_gcf_000495535.\n",
      "anaplasma_marginale_gcf_002849365.\n",
      "anaplasma_marginale_gcf_003331125.\n",
      "anaplasma_marginale_gcf_003515675.\n",
      "anaplasma_marginale_gcf_003515735.\n",
      "anaplasma_marginale_gcf_008274665.\n",
      "anaplasma_marginale_gcf_008690255.\n",
      "anaplasma_marginale_gcf_008690265.\n",
      "anaplasma_marginale_gcf_008801275.\n",
      "anaplasma_marginale_gcf_008801305.\n",
      "anaplasma_marginale_gcf_008801325.\n",
      "anaplasma_ovis_gcf_002214625.\n",
      "anaplasma_ovis_gcf_002849345.\n",
      "anaplasma_phagocytophilum_gca_900088675.\n",
      "anaplasma_phagocytophilum_gcf_000439755.\n",
      "anaplasma_phagocytophilum_gcf_000439775.\n",
      "anaplasma_phagocytophilum_gcf_000439795.\n",
      "anaplasma_phagocytophilum_gcf_000478425.\n",
      "anaplasma_phagocytophilum_gcf_000478445.\n",
      "anaplasma_phagocytophilum_gcf_000689615.\n",
      "anaplasma_phagocytophilum_gcf_000689635.\n",
      "anaplasma_phagocytophilum_gcf_000689655.\n",
      "anaplasma_phagocytophilum_gcf_000964685.\n",
      "anaplasma_phagocytophilum_gcf_000964725.\n",
      "anaplasma_phagocytophilum_gcf_000964745.\n",
      "anaplasma_phagocytophilum_gcf_000964785.\n",
      "anaplasma_phagocytophilum_gcf_000964915.\n",
      "anaplasma_phagocytophilum_gcf_000964935.\n",
      "anaplasma_phagocytophilum_gcf_000964945.\n",
      "anaplasma_phagocytophilum_gcf_000964985.\n",
      "anaplasma_phagocytophilum_gcf_000965125.\n",
      "anaplasma_phagocytophilum_gcf_000968455.\n",
      "anaplasma_phagocytophilum_gcf_000968465.\n",
      "anaplasma_phagocytophilum_gcf_002849375.\n",
      "anaplasma_phagocytophilum_gcf_900000025.\n",
      "anaplasma_phagocytophilum_gcf_900078505.\n",
      "anaplasma_phagocytophilum_gcf_900088605.\n",
      "anaplasma_phagocytophilum_gcf_900088615.\n",
      "anaplasma_phagocytophilum_gcf_900088625.\n",
      "anaplasma_phagocytophilum_gcf_900088645.\n",
      "anaplasma_phagocytophilum_gcf_900088665.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/anaplasma/initial_intervals/Anaplasma_phagocytophilum_GCF_900088665_merged.bed --twobit results/phyluce/anaplasma/cleaned_genomes/Anaplasma_phagocytophilum_GCF_000013125_formatted.2bit --output results/phyluce/anaplasma/initial_intervals/Anaplasma_phagocytophilum_GCF_900088665_stripped.bed;\n",
      "Screened 882 sequences from Anaplasma_phagocytophilum_GCF_900088665_merged.bed.  Filtered 225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 657.\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/anaplasma/initial_intervals/anaplasma-to-Anaplasma_phagocytophilum_GCF_000013125.sqlite --base-taxon Anaplasma_phagocytophilum_GCF_000013125 --output results/phyluce/anaplasma/initial_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18.bed --specific-counts 18;\n",
      "Counter({'anaplasma_phagocytophilum_gcf_000964685': 776, 'anaplasma_phagocytophilum_gcf_000964985': 776, 'anaplasma_phagocytophilum_gcf_000964945': 775, 'anaplasma_phagocytophilum_gcf_000968465': 775, 'anaplasma_phagocytophilum_gcf_000968455': 775, 'anaplasma_phagocytophilum_gcf_000964935': 775, 'anaplasma_phagocytophilum_gcf_000964725': 774, 'anaplasma_phagocytophilum_gcf_000965125': 774, 'anaplasma_phagocytophilum_gcf_000689655': 761, 'anaplasma_phagocytophilum_gcf_002849375': 756, 'anaplasma_phagocytophilum_gcf_000964785': 751, 'anaplasma_phagocytophilum_gcf_000964745': 748, 'anaplasma_phagocytophilum_gcf_900088625': 744, 'anaplasma_phagocytophilum_gcf_000689635': 743, 'anaplasma_phagocytophilum_gcf_900088605': 740, 'anaplasma_phagocytophilum_gcf_900078505': 739, 'anaplasma_phagocytophilum_gcf_900088615': 733, 'anaplasma_phagocytophilum_gcf_000964915': 727, 'anaplasma_phagocytophilum_gcf_900088665': 710, 'anaplasma_phagocytophilum_gcf_000689615': 683, 'anaplasma_phagocytophilum_gcf_000478445': 682, 'anaplasma_phagocytophilum_gcf_900000025': 682, 'anaplasma_phagocytophilum_gcf_900088645': 680, 'anaplasma_phagocytophilum_gcf_000478425': 641, 'anaplasma_phagocytophilum_gcf_000439795': 639, 'anaplasma_phagocytophilum_gcf_000439775': 634, 'anaplasma_phagocytophilum_gcf_000439755': 630, 'anaplasma_phagocytophilum_gca_900088675': 611, 'anaplasma_ovis_gcf_002214625': 7, 'anaplasma_marginale_gcf_000172495': 7, 'anaplasma_marginale_gcf_003331125': 7, 'anaplasma_ovis_gcf_002849345': 7, 'anaplasma_marginale_gcf_000020305': 7, 'anaplasma_marginale_gcf_002849365': 7, 'anaplasma_marginale_gcf_000215545': 7, 'anaplasma_marginale_gcf_000172475': 7, 'anaplasma_marginale_gcf_000215525': 7, 'anaplasma_marginale_gcf_008801305': 6, 'anaplasma_marginale_gcf_000215605': 6, 'anaplasma_marginale_gcf_008690265': 6, 'anaplasma_marginale_gcf_008801325': 6, 'anaplasma_marginale_gcf_000011945': 6, 'anaplasma_marginale_gcf_000215585': 6, 'anaplasma_marginale_gcf_008274665': 6, 'anaplasma_marginale_gcf_000215565': 6, 'anaplasma_marginale_gcf_003515735': 6, 'anaplasma_marginale_gcf_000495535': 5, 'anaplasma_marginale_gcf_008801275': 5, 'anaplasma_marginale_gcf_000172515': 5, 'anaplasma_marginale_gcf_000215505': 5, 'anaplasma_marginale_gcf_000215485': 5, 'anaplasma_marginale_gcf_003515675': 5, 'anaplasma_marginale_gcf_008690255': 5, 'anaplasma_marginale_gcf_000495495': 4})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = str(18)\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/anaplasma/initial_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18.bed --twobit results/phyluce/anaplasma/cleaned_genomes/Anaplasma_phagocytophilum_GCF_000013125_formatted.2bit --buffer-to 160 --output results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18.fasta;\n",
      "Screened 777 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 777.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18.fasta --probe-prefix uce_anaplasma_ --design anaplasma_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 773\n",
      "Probe Count = 1540\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas --query results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  13:23:54\n",
      "Ended:  Fri Feb 07, 2020  13:23:55\n",
      "Time for execution:  0.0193908691406 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas                        --lastz results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes_vself.lastz                       --probe-prefix=uce_anaplasma_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 1539 fasta sequences.  Filtered 148 duplicates. Kept 1244.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/anaplasma/validate_intervals/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas --scaffoldlist Anaplasma_marginale_GCF_000011945 Anaplasma_marginale_GCF_000020305 Anaplasma_marginale_GCF_000172475 Anaplasma_marginale_GCF_000172495 Anaplasma_marginale_GCF_000172515 Anaplasma_marginale_GCF_000215485 Anaplasma_marginale_GCF_000215505 Anaplasma_marginale_GCF_000215525 Anaplasma_marginale_GCF_000215545 Anaplasma_marginale_GCF_000215565 Anaplasma_marginale_GCF_000215585 Anaplasma_marginale_GCF_000215605 Anaplasma_marginale_GCF_000495495 Anaplasma_marginale_GCF_000495535 Anaplasma_marginale_GCF_002849365 Anaplasma_marginale_GCF_003331125 Anaplasma_marginale_GCF_003515675 Anaplasma_marginale_GCF_003515735 Anaplasma_marginale_GCF_008274665 Anaplasma_marginale_GCF_008690255 Anaplasma_marginale_GCF_008690265 Anaplasma_marginale_GCF_008801275 Anaplasma_marginale_GCF_008801305 Anaplasma_marginale_GCF_008801325 Anaplasma_ovis_GCF_002214625 Anaplasma_ovis_GCF_002849345 Anaplasma_phagocytophilum_GCA_900088675 Anaplasma_phagocytophilum_GCF_000439755 Anaplasma_phagocytophilum_GCF_000439775 Anaplasma_phagocytophilum_GCF_000439795 Anaplasma_phagocytophilum_GCF_000478425 Anaplasma_phagocytophilum_GCF_000478445 Anaplasma_phagocytophilum_GCF_000689615 Anaplasma_phagocytophilum_GCF_000689635 Anaplasma_phagocytophilum_GCF_000689655 Anaplasma_phagocytophilum_GCF_000964685 Anaplasma_phagocytophilum_GCF_000964725 Anaplasma_phagocytophilum_GCF_000964745 Anaplasma_phagocytophilum_GCF_000964785 Anaplasma_phagocytophilum_GCF_000964915 Anaplasma_phagocytophilum_GCF_000964935 Anaplasma_phagocytophilum_GCF_000964945 Anaplasma_phagocytophilum_GCF_000964985 Anaplasma_phagocytophilum_GCF_000965125 Anaplasma_phagocytophilum_GCF_000968455 Anaplasma_phagocytophilum_GCF_000968465 Anaplasma_phagocytophilum_GCF_002849375 Anaplasma_phagocytophilum_GCF_900000025 Anaplasma_phagocytophilum_GCF_900078505 Anaplasma_phagocytophilum_GCF_900088605 Anaplasma_phagocytophilum_GCF_900088615 Anaplasma_phagocytophilum_GCF_900088625 Anaplasma_phagocytophilum_GCF_900088645 Anaplasma_phagocytophilum_GCF_900088665 Anaplasma_phagocytophilum_GCF_000013125 --genome-base-path results/phyluce/anaplasma/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/anaplasma/validate_intervals/anaplasma-to-Anaplasma_phagocytophilum_GCF_000013125.sqlite --output results/phyluce/anaplasma/validate_intervals/lastz/;\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000011945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp89oa0q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeDmm9j.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000011945.lastz\n",
      "Creating Anaplasma_marginale_GCF_000011945 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000011945 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000020305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpcVCKpB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwHo3YA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000020305.lastz\n",
      "Creating Anaplasma_marginale_GCF_000020305 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000020305 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000172475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzGtvJz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEpIIn3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000172475.lastz\n",
      "Creating Anaplasma_marginale_GCF_000172475 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000172475 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000172495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWNeYF4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqmfykt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000172495.lastz\n",
      "Creating Anaplasma_marginale_GCF_000172495 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000172495 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000172515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJPA4YD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpR4nqls.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000172515.lastz\n",
      "Creating Anaplasma_marginale_GCF_000172515 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000172515 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuSxZEr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZzK1s2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215485.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215485 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215485 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWlIgA5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxlb3lk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215505.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215505 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215505 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaIZDQ3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwWTL04.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215525.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215525 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215525 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphVycQ0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWo9pVQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215545.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215545 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215545 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpog1QGl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpya88Hi.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215565.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215565 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215565 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0NgLLp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8N611q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215585.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215585 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215585 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000215605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp522Hks.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuptKFH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000215605.lastz\n",
      "Creating Anaplasma_marginale_GCF_000215605 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000215605 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000495495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWHf5cB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGHoe0Z.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000495495.lastz\n",
      "Creating Anaplasma_marginale_GCF_000495495 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000495495 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_000495535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgEb7O7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp72I404.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_000495535.lastz\n",
      "Creating Anaplasma_marginale_GCF_000495535 table\n",
      "Inserting data to Anaplasma_marginale_GCF_000495535 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_002849365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkq1bgM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEgmJCC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_002849365.lastz\n",
      "Creating Anaplasma_marginale_GCF_002849365 table\n",
      "Inserting data to Anaplasma_marginale_GCF_002849365 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_003331125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5ZC5ov.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHUkMyU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_003331125.lastz\n",
      "Creating Anaplasma_marginale_GCF_003331125 table\n",
      "Inserting data to Anaplasma_marginale_GCF_003331125 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_003515675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiECtPi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfxjIiM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_003515675.lastz\n",
      "Creating Anaplasma_marginale_GCF_003515675 table\n",
      "Inserting data to Anaplasma_marginale_GCF_003515675 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_003515735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpv5AXeE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHgj8fa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_003515735.lastz\n",
      "Creating Anaplasma_marginale_GCF_003515735 table\n",
      "Inserting data to Anaplasma_marginale_GCF_003515735 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_008274665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVshSk0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmAxn9c.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_008274665.lastz\n",
      "Creating Anaplasma_marginale_GCF_008274665 table\n",
      "Inserting data to Anaplasma_marginale_GCF_008274665 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_008690255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_KesKI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB09ptw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_008690255.lastz\n",
      "Creating Anaplasma_marginale_GCF_008690255 table\n",
      "Inserting data to Anaplasma_marginale_GCF_008690255 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_008690265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPGX8mY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpe4ACdP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_008690265.lastz\n",
      "Creating Anaplasma_marginale_GCF_008690265 table\n",
      "Inserting data to Anaplasma_marginale_GCF_008690265 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_008801275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptEKiFD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIozyNG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_008801275.lastz\n",
      "Creating Anaplasma_marginale_GCF_008801275 table\n",
      "Inserting data to Anaplasma_marginale_GCF_008801275 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_008801305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuvEa0R.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAW8KUV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_008801305.lastz\n",
      "Creating Anaplasma_marginale_GCF_008801305 table\n",
      "Inserting data to Anaplasma_marginale_GCF_008801305 table\n",
      "\n",
      "Running against Anaplasma_marginale_GCF_008801325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7gTiJt.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdoBwh3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_marginale_GCF_008801325.lastz\n",
      "Creating Anaplasma_marginale_GCF_008801325 table\n",
      "Inserting data to Anaplasma_marginale_GCF_008801325 table\n",
      "\n",
      "Running against Anaplasma_ovis_GCF_002214625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptOht3q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0PDpAn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_ovis_GCF_002214625.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Anaplasma_ovis_GCF_002214625 table\n",
      "Inserting data to Anaplasma_ovis_GCF_002214625 table\n",
      "\n",
      "Running against Anaplasma_ovis_GCF_002849345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp85GxPP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjsYbTD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_ovis_GCF_002849345.lastz\n",
      "Creating Anaplasma_ovis_GCF_002849345 table\n",
      "Inserting data to Anaplasma_ovis_GCF_002849345 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCA_900088675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyotXoG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzlVLge.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCA_900088675.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCA_900088675 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCA_900088675 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000439755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyMJfwu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpujBC4m.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000439755.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000439755 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000439755 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000439775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZ_umaO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKKD81l.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000439775.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000439775 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000439775 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000439795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZfsLwW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxSH9ae.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000439795.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000439795 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000439795 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000478425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRgadaD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3zX11d.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000478425.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000478425 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000478425 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000478445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGibDBR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5W2slp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000478445.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000478445 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000478445 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000689615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgLpbUQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppXNvV3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000689615.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000689615 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000689615 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000689635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8Hq3Hc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfOKTiz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000689635.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000689635 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000689635 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000689655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKfpSmq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE1DXyX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000689655.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000689655 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000689655 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkaWs38.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzElVWg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964685.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964685 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964685 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpy5pLas.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQa8XOG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964725.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964725 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964725 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJSMv3X.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9hyhU2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964745.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964745 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964745 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKUjbxr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBI0xJo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964785.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Anaplasma_phagocytophilum_GCF_000964785 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964785 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdG990K.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphRaFFY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964915.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964915 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964915 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFwewBI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpj4eYNQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964935.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964935 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964935 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqU_TFJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB8HrrP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964945.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964945 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964945 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000964985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKQxNfH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdIFTOW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000964985.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000964985 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000964985 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000965125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpplX9CI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6ueNHW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000965125.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000965125 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000965125 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000968455.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpXYiBkl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprsYq_T.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000968455.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000968455 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000968455 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000968465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdlfpxf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpk6f1mC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000968465.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000968465 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000968465 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_002849375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpD2veWG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCXw8zG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_002849375.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_002849375 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_002849375 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900000025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2tSwgT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGSuTdL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900000025.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_900000025 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900000025 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900078505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzTFfp9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjktVLj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900078505.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_900078505 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900078505 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900088605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpncrWIP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVs9cV6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900088605.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_900088605 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900088605 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900088615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJUvcSQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3wzI9t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900088615.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_900088615 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900088615 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900088625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqcUiqw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZ2s6NV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900088625.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_900088625 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900088625 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900088645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiyumrd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZyu5u4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900088645.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Anaplasma_phagocytophilum_GCF_900088645 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900088645 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_900088665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpiCkIAJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpto8D8I.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_900088665.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_900088665 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_900088665 table\n",
      "\n",
      "Running against Anaplasma_phagocytophilum_GCF_000013125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLalAcS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNoSVJl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/anaplasma/validate_intervals/lastz/Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_Anaplasma_phagocytophilum_GCF_000013125.lastz\n",
      "Creating Anaplasma_phagocytophilum_GCF_000013125 table\n",
      "Inserting data to Anaplasma_phagocytophilum_GCF_000013125 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/anaplasma/extract_probes_from_group/anaplasma_genome.conf --lastz results/phyluce/anaplasma/validate_intervals/lastz --probes 120 --probe-prefix uce_anaplasma_ --name-pattern \"Anaplasma_phagocytophilum_GCF_000013125_+18_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/anaplasma/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 13:29:43,192 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 13:29:43,212 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000011945 genome ------\n",
      "2020-02-07 13:29:43,219 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000011945 genome\n",
      "2020-02-07 13:29:44,546 - Phyluce - INFO - Anaplasma_marginale_GCF_000011945: 236 uces, 0 dupes, 236 non-dupes, 41 orient drop, 1 length drop, 194 written\n",
      "2020-02-07 13:29:44,546 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000020305 genome ------\n",
      "2020-02-07 13:29:44,787 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000020305 genome\n",
      "2020-02-07 13:29:45,960 - Phyluce - INFO - Anaplasma_marginale_GCF_000020305: 236 uces, 0 dupes, 236 non-dupes, 18 orient drop, 25 length drop, 193 written\n",
      "2020-02-07 13:29:45,960 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000172475 genome ------\n",
      "2020-02-07 13:29:45,977 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000172475 genome\n",
      "2020-02-07 13:29:46,772 - Phyluce - INFO - Anaplasma_marginale_GCF_000172475: 219 uces, 13 dupes, 206 non-dupes, 0 orient drop, 0 length drop, 206 written\n",
      "2020-02-07 13:29:46,773 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000172495 genome ------\n",
      "2020-02-07 13:29:47,105 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000172495 genome\n",
      "2020-02-07 13:29:47,981 - Phyluce - INFO - Anaplasma_marginale_GCF_000172495: 233 uces, 20 dupes, 213 non-dupes, 0 orient drop, 0 length drop, 213 written\n",
      "2020-02-07 13:29:47,982 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000172515 genome ------\n",
      "2020-02-07 13:29:47,995 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000172515 genome\n",
      "2020-02-07 13:29:48,754 - Phyluce - INFO - Anaplasma_marginale_GCF_000172515: 232 uces, 19 dupes, 213 non-dupes, 0 orient drop, 0 length drop, 213 written\n",
      "2020-02-07 13:29:48,755 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215485 genome ------\n",
      "2020-02-07 13:29:48,756 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215485 genome\n",
      "2020-02-07 13:29:49,494 - Phyluce - INFO - Anaplasma_marginale_GCF_000215485: 206 uces, 12 dupes, 194 non-dupes, 0 orient drop, 0 length drop, 194 written\n",
      "2020-02-07 13:29:49,495 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215505 genome ------\n",
      "2020-02-07 13:29:49,505 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215505 genome\n",
      "2020-02-07 13:29:50,170 - Phyluce - INFO - Anaplasma_marginale_GCF_000215505: 205 uces, 13 dupes, 192 non-dupes, 0 orient drop, 0 length drop, 192 written\n",
      "2020-02-07 13:29:50,170 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215525 genome ------\n",
      "2020-02-07 13:29:50,171 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215525 genome\n",
      "2020-02-07 13:29:51,082 - Phyluce - INFO - Anaplasma_marginale_GCF_000215525: 206 uces, 4 dupes, 202 non-dupes, 0 orient drop, 2 length drop, 200 written\n",
      "2020-02-07 13:29:51,082 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215545 genome ------\n",
      "2020-02-07 13:29:51,355 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215545 genome\n",
      "2020-02-07 13:29:51,920 - Phyluce - INFO - Anaplasma_marginale_GCF_000215545: 197 uces, 5 dupes, 192 non-dupes, 0 orient drop, 0 length drop, 191 written\n",
      "2020-02-07 13:29:51,921 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215565 genome ------\n",
      "2020-02-07 13:29:52,243 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215565 genome\n",
      "2020-02-07 13:29:53,083 - Phyluce - INFO - Anaplasma_marginale_GCF_000215565: 210 uces, 3 dupes, 207 non-dupes, 0 orient drop, 2 length drop, 205 written\n",
      "2020-02-07 13:29:53,083 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215585 genome ------\n",
      "2020-02-07 13:29:53,382 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215585 genome\n",
      "2020-02-07 13:29:54,313 - Phyluce - INFO - Anaplasma_marginale_GCF_000215585: 197 uces, 6 dupes, 191 non-dupes, 0 orient drop, 0 length drop, 190 written\n",
      "2020-02-07 13:29:54,314 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000215605 genome ------\n",
      "2020-02-07 13:29:54,688 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000215605 genome\n",
      "2020-02-07 13:29:55,574 - Phyluce - INFO - Anaplasma_marginale_GCF_000215605: 209 uces, 8 dupes, 201 non-dupes, 0 orient drop, 0 length drop, 201 written\n",
      "2020-02-07 13:29:55,575 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000495495 genome ------\n",
      "2020-02-07 13:29:55,576 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000495495 genome\n",
      "2020-02-07 13:29:56,585 - Phyluce - INFO - Anaplasma_marginale_GCF_000495495: 229 uces, 0 dupes, 229 non-dupes, 40 orient drop, 0 length drop, 189 written\n",
      "2020-02-07 13:29:56,585 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_000495535 genome ------\n",
      "2020-02-07 13:29:56,593 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_000495535 genome\n",
      "2020-02-07 13:29:57,372 - Phyluce - INFO - Anaplasma_marginale_GCF_000495535: 227 uces, 0 dupes, 227 non-dupes, 38 orient drop, 0 length drop, 189 written\n",
      "2020-02-07 13:29:57,372 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_002849365 genome ------\n",
      "2020-02-07 13:29:57,569 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_002849365 genome\n",
      "2020-02-07 13:29:58,400 - Phyluce - INFO - Anaplasma_marginale_GCF_002849365: 221 uces, 16 dupes, 205 non-dupes, 0 orient drop, 0 length drop, 205 written\n",
      "2020-02-07 13:29:58,400 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_003331125 genome ------\n",
      "2020-02-07 13:29:58,611 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_003331125 genome\n",
      "2020-02-07 13:29:59,231 - Phyluce - INFO - Anaplasma_marginale_GCF_003331125: 208 uces, 11 dupes, 197 non-dupes, 0 orient drop, 2 length drop, 195 written\n",
      "2020-02-07 13:29:59,232 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_003515675 genome ------\n",
      "2020-02-07 13:29:59,541 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_003515675 genome\n",
      "2020-02-07 13:30:00,702 - Phyluce - INFO - Anaplasma_marginale_GCF_003515675: 232 uces, 0 dupes, 232 non-dupes, 16 orient drop, 24 length drop, 192 written\n",
      "2020-02-07 13:30:00,702 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_003515735 genome ------\n",
      "2020-02-07 13:30:01,126 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_003515735 genome\n",
      "2020-02-07 13:30:02,392 - Phyluce - INFO - Anaplasma_marginale_GCF_003515735: 232 uces, 0 dupes, 232 non-dupes, 16 orient drop, 24 length drop, 192 written\n",
      "2020-02-07 13:30:02,393 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_008274665 genome ------\n",
      "2020-02-07 13:30:02,394 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_008274665 genome\n",
      "2020-02-07 13:30:03,248 - Phyluce - INFO - Anaplasma_marginale_GCF_008274665: 232 uces, 16 dupes, 216 non-dupes, 0 orient drop, 2 length drop, 214 written\n",
      "2020-02-07 13:30:03,248 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_008690255 genome ------\n",
      "2020-02-07 13:30:03,250 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_008690255 genome\n",
      "2020-02-07 13:30:04,226 - Phyluce - INFO - Anaplasma_marginale_GCF_008690255: 236 uces, 20 dupes, 216 non-dupes, 0 orient drop, 0 length drop, 216 written\n",
      "2020-02-07 13:30:04,226 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_008690265 genome ------\n",
      "2020-02-07 13:30:04,409 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_008690265 genome\n",
      "2020-02-07 13:30:05,323 - Phyluce - INFO - Anaplasma_marginale_GCF_008690265: 220 uces, 16 dupes, 204 non-dupes, 0 orient drop, 2 length drop, 202 written\n",
      "2020-02-07 13:30:05,323 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_008801275 genome ------\n",
      "2020-02-07 13:30:05,328 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_008801275 genome\n",
      "2020-02-07 13:30:06,117 - Phyluce - INFO - Anaplasma_marginale_GCF_008801275: 230 uces, 15 dupes, 215 non-dupes, 0 orient drop, 2 length drop, 213 written\n",
      "2020-02-07 13:30:06,117 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_008801305 genome ------\n",
      "2020-02-07 13:30:06,125 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_008801305 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 13:30:06,909 - Phyluce - INFO - Anaplasma_marginale_GCF_008801305: 233 uces, 31 dupes, 202 non-dupes, 0 orient drop, 2 length drop, 200 written\n",
      "2020-02-07 13:30:06,909 - Phyluce - INFO - ------ Working on Anaplasma_marginale_GCF_008801325 genome ------\n",
      "2020-02-07 13:30:06,920 - Phyluce - INFO - Reading Anaplasma_marginale_GCF_008801325 genome\n",
      "2020-02-07 13:30:07,823 - Phyluce - INFO - Anaplasma_marginale_GCF_008801325: 232 uces, 25 dupes, 207 non-dupes, 0 orient drop, 0 length drop, 207 written\n",
      "2020-02-07 13:30:07,824 - Phyluce - INFO - --------- Working on Anaplasma_ovis_GCF_002214625 genome --------\n",
      "2020-02-07 13:30:07,825 - Phyluce - INFO - Reading Anaplasma_ovis_GCF_002214625 genome\n",
      "2020-02-07 13:30:08,979 - Phyluce - INFO - Anaplasma_ovis_GCF_002214625: 230 uces, 0 dupes, 230 non-dupes, 39 orient drop, 4 length drop, 187 written\n",
      "2020-02-07 13:30:08,979 - Phyluce - INFO - --------- Working on Anaplasma_ovis_GCF_002849345 genome --------\n",
      "2020-02-07 13:30:08,999 - Phyluce - INFO - Reading Anaplasma_ovis_GCF_002849345 genome\n",
      "2020-02-07 13:30:10,010 - Phyluce - INFO - Anaplasma_ovis_GCF_002849345: 220 uces, 26 dupes, 194 non-dupes, 0 orient drop, 0 length drop, 194 written\n",
      "2020-02-07 13:30:10,011 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCA_900088675 genome ---\n",
      "2020-02-07 13:30:10,012 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCA_900088675 genome\n",
      "2020-02-07 13:30:18,893 - Phyluce - INFO - Anaplasma_phagocytophilum_GCA_900088675: 670 uces, 209 dupes, 461 non-dupes, 1 orient drop, 4 length drop, 456 written\n",
      "2020-02-07 13:30:18,893 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000439755 genome ---\n",
      "2020-02-07 13:30:18,905 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000439755 genome\n",
      "2020-02-07 13:30:35,564 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000439755: 773 uces, 0 dupes, 773 non-dupes, 181 orient drop, 76 length drop, 516 written\n",
      "2020-02-07 13:30:35,565 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000439775 genome ---\n",
      "2020-02-07 13:30:35,965 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000439775 genome\n",
      "2020-02-07 13:30:52,515 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000439775: 773 uces, 0 dupes, 773 non-dupes, 180 orient drop, 78 length drop, 515 written\n",
      "2020-02-07 13:30:52,515 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000439795 genome ---\n",
      "2020-02-07 13:30:52,516 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000439795 genome\n",
      "2020-02-07 13:31:09,721 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000439795: 773 uces, 0 dupes, 773 non-dupes, 179 orient drop, 78 length drop, 516 written\n",
      "2020-02-07 13:31:09,722 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000478425 genome ---\n",
      "2020-02-07 13:31:09,731 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000478425 genome\n",
      "2020-02-07 13:31:26,615 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000478425: 773 uces, 122 dupes, 651 non-dupes, 57 orient drop, 78 length drop, 516 written\n",
      "2020-02-07 13:31:26,615 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000478445 genome ---\n",
      "2020-02-07 13:31:26,616 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000478445 genome\n",
      "2020-02-07 13:31:45,663 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000478445: 759 uces, 120 dupes, 639 non-dupes, 58 orient drop, 75 length drop, 506 written\n",
      "2020-02-07 13:31:45,664 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000689615 genome ---\n",
      "2020-02-07 13:31:45,683 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000689615 genome\n",
      "2020-02-07 13:32:00,953 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000689615: 758 uces, 153 dupes, 605 non-dupes, 33 orient drop, 56 length drop, 516 written\n",
      "2020-02-07 13:32:00,953 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000689635 genome ---\n",
      "2020-02-07 13:32:00,954 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000689635 genome\n",
      "2020-02-07 13:32:20,150 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000689635: 765 uces, 0 dupes, 765 non-dupes, 174 orient drop, 80 length drop, 511 written\n",
      "2020-02-07 13:32:20,151 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000689655 genome ---\n",
      "2020-02-07 13:32:20,157 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000689655 genome\n",
      "2020-02-07 13:32:38,349 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000689655: 767 uces, 141 dupes, 626 non-dupes, 48 orient drop, 69 length drop, 509 written\n",
      "2020-02-07 13:32:38,349 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964685 genome ---\n",
      "2020-02-07 13:32:38,357 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964685 genome\n",
      "2020-02-07 13:32:55,451 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964685: 773 uces, 0 dupes, 773 non-dupes, 178 orient drop, 81 length drop, 514 written\n",
      "2020-02-07 13:32:55,451 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964725 genome ---\n",
      "2020-02-07 13:32:55,452 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964725 genome\n",
      "2020-02-07 13:33:13,587 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964725: 771 uces, 180 dupes, 591 non-dupes, 32 orient drop, 59 length drop, 500 written\n",
      "2020-02-07 13:33:13,588 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964745 genome ---\n",
      "2020-02-07 13:33:13,591 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964745 genome\n",
      "2020-02-07 13:33:31,505 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964745: 767 uces, 0 dupes, 767 non-dupes, 176 orient drop, 82 length drop, 509 written\n",
      "2020-02-07 13:33:31,506 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964785 genome ---\n",
      "2020-02-07 13:33:31,819 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964785 genome\n",
      "2020-02-07 13:33:49,369 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964785: 767 uces, 0 dupes, 767 non-dupes, 173 orient drop, 86 length drop, 508 written\n",
      "2020-02-07 13:33:49,370 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964915 genome ---\n",
      "2020-02-07 13:33:49,379 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964915 genome\n",
      "2020-02-07 13:34:06,761 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964915: 743 uces, 229 dupes, 514 non-dupes, 16 orient drop, 22 length drop, 476 written\n",
      "2020-02-07 13:34:06,761 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964935 genome ---\n",
      "2020-02-07 13:34:06,762 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964935 genome\n",
      "2020-02-07 13:34:23,980 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964935: 773 uces, 0 dupes, 773 non-dupes, 177 orient drop, 80 length drop, 516 written\n",
      "2020-02-07 13:34:23,980 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964945 genome ---\n",
      "2020-02-07 13:34:23,989 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964945 genome\n",
      "2020-02-07 13:34:41,612 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964945: 773 uces, 0 dupes, 773 non-dupes, 177 orient drop, 86 length drop, 510 written\n",
      "2020-02-07 13:34:41,612 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000964985 genome ---\n",
      "2020-02-07 13:34:41,614 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000964985 genome\n",
      "2020-02-07 13:34:58,537 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000964985: 772 uces, 191 dupes, 581 non-dupes, 33 orient drop, 44 length drop, 504 written\n",
      "2020-02-07 13:34:58,537 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000965125 genome ---\n",
      "2020-02-07 13:34:58,539 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000965125 genome\n",
      "2020-02-07 13:35:16,135 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000965125: 772 uces, 185 dupes, 587 non-dupes, 32 orient drop, 51 length drop, 504 written\n",
      "2020-02-07 13:35:16,136 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000968455 genome ---\n",
      "2020-02-07 13:35:16,147 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000968455 genome\n",
      "2020-02-07 13:35:32,766 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000968455: 773 uces, 17 dupes, 756 non-dupes, 172 orient drop, 75 length drop, 509 written\n",
      "2020-02-07 13:35:32,766 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000968465 genome ---\n",
      "2020-02-07 13:35:32,767 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000968465 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 13:35:51,253 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000968465: 772 uces, 134 dupes, 638 non-dupes, 69 orient drop, 67 length drop, 502 written\n",
      "2020-02-07 13:35:51,254 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_002849375 genome ---\n",
      "2020-02-07 13:35:51,254 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_002849375 genome\n",
      "2020-02-07 13:36:04,064 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_002849375: 759 uces, 189 dupes, 570 non-dupes, 2 orient drop, 14 length drop, 553 written\n",
      "2020-02-07 13:36:04,064 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900000025 genome ---\n",
      "2020-02-07 13:36:04,114 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900000025 genome\n",
      "2020-02-07 13:36:14,030 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900000025: 729 uces, 199 dupes, 530 non-dupes, 2 orient drop, 9 length drop, 515 written\n",
      "2020-02-07 13:36:14,031 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900078505 genome ---\n",
      "2020-02-07 13:36:14,031 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900078505 genome\n",
      "2020-02-07 13:36:34,086 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900078505: 754 uces, 265 dupes, 489 non-dupes, 3 orient drop, 17 length drop, 469 written\n",
      "2020-02-07 13:36:34,086 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900088605 genome ---\n",
      "2020-02-07 13:36:34,092 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900088605 genome\n",
      "2020-02-07 13:36:52,035 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900088605: 753 uces, 211 dupes, 542 non-dupes, 6 orient drop, 23 length drop, 513 written\n",
      "2020-02-07 13:36:52,036 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900088615 genome ---\n",
      "2020-02-07 13:36:52,071 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900088615 genome\n",
      "2020-02-07 13:37:08,058 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900088615: 747 uces, 231 dupes, 516 non-dupes, 3 orient drop, 21 length drop, 492 written\n",
      "2020-02-07 13:37:08,058 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900088625 genome ---\n",
      "2020-02-07 13:37:08,070 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900088625 genome\n",
      "2020-02-07 13:37:25,416 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900088625: 751 uces, 218 dupes, 533 non-dupes, 4 orient drop, 21 length drop, 508 written\n",
      "2020-02-07 13:37:25,416 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900088645 genome ---\n",
      "2020-02-07 13:37:25,421 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900088645 genome\n",
      "2020-02-07 13:37:35,757 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900088645: 720 uces, 296 dupes, 424 non-dupes, 3 orient drop, 7 length drop, 414 written\n",
      "2020-02-07 13:37:35,758 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_900088665 genome ---\n",
      "2020-02-07 13:37:35,807 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_900088665 genome\n",
      "2020-02-07 13:37:45,731 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_900088665: 732 uces, 200 dupes, 532 non-dupes, 2 orient drop, 20 length drop, 510 written\n",
      "2020-02-07 13:37:45,732 - Phyluce - INFO - --- Working on Anaplasma_phagocytophilum_GCF_000013125 genome ---\n",
      "2020-02-07 13:37:45,732 - Phyluce - INFO - Reading Anaplasma_phagocytophilum_GCF_000013125 genome\n",
      "2020-02-07 13:38:03,337 - Phyluce - INFO - Anaplasma_phagocytophilum_GCF_000013125: 773 uces, 0 dupes, 773 non-dupes, 181 orient drop, 76 length drop, 516 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/anaplasma/extract_probes_from_group/probe_fasta --output results/phyluce/anaplasma/extract_probes_from_group/multifastas.sqlite --base-taxon Anaplasma_phagocytophilum_GCF_000013125;\n",
      "anaplasma_marginale_gcf_000011945.\n",
      "anaplasma_marginale_gcf_000020305.\n",
      "anaplasma_marginale_gcf_000172475.\n",
      "anaplasma_marginale_gcf_000172495.\n",
      "anaplasma_marginale_gcf_000172515.\n",
      "anaplasma_marginale_gcf_000215485.\n",
      "anaplasma_marginale_gcf_000215505.\n",
      "anaplasma_marginale_gcf_000215525.\n",
      "anaplasma_marginale_gcf_000215545.\n",
      "anaplasma_marginale_gcf_000215565.\n",
      "anaplasma_marginale_gcf_000215585.\n",
      "anaplasma_marginale_gcf_000215605.\n",
      "anaplasma_marginale_gcf_000495495.\n",
      "anaplasma_marginale_gcf_000495535.\n",
      "anaplasma_marginale_gcf_002849365.\n",
      "anaplasma_marginale_gcf_003331125.\n",
      "anaplasma_marginale_gcf_003515675.\n",
      "anaplasma_marginale_gcf_003515735.\n",
      "anaplasma_marginale_gcf_008274665.\n",
      "anaplasma_marginale_gcf_008690255.\n",
      "anaplasma_marginale_gcf_008690265.\n",
      "anaplasma_marginale_gcf_008801275.\n",
      "anaplasma_marginale_gcf_008801305.\n",
      "anaplasma_marginale_gcf_008801325.\n",
      "anaplasma_ovis_gcf_002214625.\n",
      "anaplasma_ovis_gcf_002849345.\n",
      "anaplasma_phagocytophilum_gca_900088675.\n",
      "anaplasma_phagocytophilum_gcf_000439755.\n",
      "anaplasma_phagocytophilum_gcf_000439775.\n",
      "anaplasma_phagocytophilum_gcf_000439795.\n",
      "anaplasma_phagocytophilum_gcf_000478425.\n",
      "anaplasma_phagocytophilum_gcf_000478445.\n",
      "anaplasma_phagocytophilum_gcf_000689615.\n",
      "anaplasma_phagocytophilum_gcf_000689635.\n",
      "anaplasma_phagocytophilum_gcf_000689655.\n",
      "anaplasma_phagocytophilum_gcf_000964685.\n",
      "anaplasma_phagocytophilum_gcf_000964725.\n",
      "anaplasma_phagocytophilum_gcf_000964745.\n",
      "anaplasma_phagocytophilum_gcf_000964785.\n",
      "anaplasma_phagocytophilum_gcf_000964915.\n",
      "anaplasma_phagocytophilum_gcf_000964935.\n",
      "anaplasma_phagocytophilum_gcf_000964945.\n",
      "anaplasma_phagocytophilum_gcf_000964985.\n",
      "anaplasma_phagocytophilum_gcf_000965125.\n",
      "anaplasma_phagocytophilum_gcf_000968455.\n",
      "anaplasma_phagocytophilum_gcf_000968465.\n",
      "anaplasma_phagocytophilum_gcf_002849375.\n",
      "anaplasma_phagocytophilum_gcf_900000025.\n",
      "anaplasma_phagocytophilum_gcf_900078505.\n",
      "anaplasma_phagocytophilum_gcf_900088605.\n",
      "anaplasma_phagocytophilum_gcf_900088615.\n",
      "anaplasma_phagocytophilum_gcf_900088625.\n",
      "anaplasma_phagocytophilum_gcf_900088645.\n",
      "anaplasma_phagocytophilum_gcf_900088665.\n",
      "anaplasma_phagocytophilum_gcf_000013125.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/anaplasma/extract_probes_from_group/multifastas.sqlite --base-taxon Anaplasma_phagocytophilum_GCF_000013125;\n",
      "Loci shared by 0 taxa:\t684.0\n",
      "Loci shared by 1 taxa:\t684.0\n",
      "Loci shared by 2 taxa:\t654.0\n",
      "Loci shared by 3 taxa:\t639.0\n",
      "Loci shared by 4 taxa:\t618.0\n",
      "Loci shared by 5 taxa:\t608.0\n",
      "Loci shared by 6 taxa:\t593.0\n",
      "Loci shared by 7 taxa:\t582.0\n",
      "Loci shared by 8 taxa:\t571.0\n",
      "Loci shared by 9 taxa:\t553.0\n",
      "Loci shared by 10 taxa:\t551.0\n",
      "Loci shared by 11 taxa:\t548.0\n",
      "Loci shared by 12 taxa:\t546.0\n",
      "Loci shared by 13 taxa:\t546.0\n",
      "Loci shared by 14 taxa:\t545.0\n",
      "Loci shared by 15 taxa:\t539.0\n",
      "Loci shared by 16 taxa:\t537.0\n",
      "Loci shared by 17 taxa:\t536.0\n",
      "Loci shared by 18 taxa:\t534.0\n",
      "Loci shared by 19 taxa:\t532.0\n",
      "Loci shared by 20 taxa:\t530.0\n",
      "Loci shared by 21 taxa:\t527.0\n",
      "Loci shared by 22 taxa:\t525.0\n",
      "Loci shared by 23 taxa:\t523.0\n",
      "Loci shared by 24 taxa:\t516.0\n",
      "Loci shared by 25 taxa:\t507.0\n",
      "Loci shared by 26 taxa:\t492.0\n",
      "Loci shared by 27 taxa:\t455.0\n",
      "Loci shared by 28 taxa:\t378.0\n",
      "Loci shared by 29 taxa:\t287.0\n",
      "Loci shared by 30 taxa:\t182.0\n",
      "Loci shared by 31 taxa:\t180.0\n",
      "Loci shared by 32 taxa:\t178.0\n",
      "Loci shared by 33 taxa:\t178.0\n",
      "Loci shared by 34 taxa:\t175.0\n",
      "Loci shared by 35 taxa:\t174.0\n",
      "Loci shared by 36 taxa:\t172.0\n",
      "Loci shared by 37 taxa:\t171.0\n",
      "Loci shared by 38 taxa:\t169.0\n",
      "Loci shared by 39 taxa:\t166.0\n",
      "Loci shared by 40 taxa:\t166.0\n",
      "Loci shared by 41 taxa:\t166.0\n",
      "Loci shared by 42 taxa:\t165.0\n",
      "Loci shared by 43 taxa:\t165.0\n",
      "Loci shared by 44 taxa:\t163.0\n",
      "Loci shared by 45 taxa:\t163.0\n",
      "Loci shared by 46 taxa:\t161.0\n",
      "Loci shared by 47 taxa:\t158.0\n",
      "Loci shared by 48 taxa:\t156.0\n",
      "Loci shared by 49 taxa:\t154.0\n",
      "Loci shared by 50 taxa:\t152.0\n",
      "Loci shared by 51 taxa:\t151.0\n",
      "Loci shared by 52 taxa:\t149.0\n",
      "Loci shared by 53 taxa:\t143.0\n",
      "Loci shared by 54 taxa:\t116.0\n",
      "Loci shared by 55 taxa:\t61.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/anaplasma/extract_probes_from_group/multifastas.sqlite  --base-taxon Anaplasma_phagocytophilum_GCF_000013125 --output results/phyluce/anaplasma/extract_probes_from_group/Anaplasma_phagocytophilum_GCF_000013125+18-back-to-55.conf --specific-counts 55;\n",
      "Counter({'anaplasma_marginale_gcf_008801305': 61, 'anaplasma_ovis_gcf_002214625': 61, 'anaplasma_marginale_gcf_008274665': 61, 'anaplasma_ovis_gcf_002849345': 61, 'anaplasma_marginale_gcf_008801325': 61, 'anaplasma_phagocytophilum_gcf_000964945': 61, 'anaplasma_phagocytophilum_gcf_000478425': 61, 'anaplasma_phagocytophilum_gcf_000478445': 61, 'anaplasma_phagocytophilum_gcf_000964785': 61, 'anaplasma_phagocytophilum_gcf_000968465': 61, 'anaplasma_marginale_gcf_000495535': 61, 'anaplasma_marginale_gcf_000215545': 61, 'anaplasma_phagocytophilum_gcf_000964685': 61, 'anaplasma_marginale_gcf_008801275': 61, 'anaplasma_marginale_gcf_000172475': 61, 'anaplasma_marginale_gcf_003331125': 61, 'anaplasma_phagocytophilum_gca_900088675': 61, 'anaplasma_marginale_gcf_002849365': 61, 'anaplasma_phagocytophilum_gcf_900088615': 61, 'anaplasma_marginale_gcf_008690265': 61, 'anaplasma_phagocytophilum_gcf_000964985': 61, 'anaplasma_marginale_gcf_000215605': 61, 'anaplasma_marginale_gcf_000011945': 61, 'anaplasma_marginale_gcf_000172515': 61, 'anaplasma_phagocytophilum_gcf_002849375': 61, 'anaplasma_phagocytophilum_gcf_000689615': 61, 'anaplasma_phagocytophilum_gcf_000964935': 61, 'anaplasma_phagocytophilum_gcf_000689635': 61, 'anaplasma_marginale_gcf_003515675': 61, 'anaplasma_marginale_gcf_000215585': 61, 'anaplasma_phagocytophilum_gcf_000964725': 61, 'anaplasma_phagocytophilum_gcf_000439755': 61, 'anaplasma_phagocytophilum_gcf_000965125': 61, 'anaplasma_marginale_gcf_000215525': 61, 'anaplasma_marginale_gcf_000172495': 61, 'anaplasma_marginale_gcf_000215565': 61, 'anaplasma_phagocytophilum_gcf_000689655': 61, 'anaplasma_phagocytophilum_gcf_000968455': 61, 'anaplasma_phagocytophilum_gcf_000964915': 61, 'anaplasma_marginale_gcf_000495495': 61, 'anaplasma_phagocytophilum_gcf_900088665': 61, 'anaplasma_phagocytophilum_gcf_000013125': 61, 'anaplasma_marginale_gcf_008690255': 61, 'anaplasma_marginale_gcf_000215505': 61, 'anaplasma_phagocytophilum_gcf_900088605': 61, 'anaplasma_marginale_gcf_000215485': 61, 'anaplasma_phagocytophilum_gcf_000964745': 61, 'anaplasma_phagocytophilum_gcf_900078505': 61, 'anaplasma_phagocytophilum_gcf_000439795': 61, 'anaplasma_phagocytophilum_gcf_000439775': 61, 'anaplasma_marginale_gcf_000020305': 61, 'anaplasma_phagocytophilum_gcf_900088625': 61, 'anaplasma_phagocytophilum_gcf_900088645': 61, 'anaplasma_phagocytophilum_gcf_900000025': 61, 'anaplasma_marginale_gcf_003515735': 61})\n",
      "Total loci = 61\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(55)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/anaplasma/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/anaplasma/extract_probes_from_group/Anaplasma_phagocytophilum_GCF_000013125+18-back-to-55.conf --probe-prefix uce_anaplasma_ --designer rnplattii --design anaplasma_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 61\n",
      "Probe Count = 6693\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list.fasta --query results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  13:38:30\n",
      "Ended:  Fri Feb 07, 2020  13:39:00\n",
      "Time for execution:  0.506612567107 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list.fasta --lastz results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_anaplasma_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6692 fasta sequences.  Filtered 0 duplicates. Kept 6693.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/anaplasma/final_probe_design/anaplasma_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 13:50:40 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 6693\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 535440\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3953615\n",
      "Max number of word counting entries: 88995891\n",
      "\n",
      "# comparing sequences from          0  to       1115\n",
      ".---------- new table with       83 representatives\n",
      "# comparing sequences from       1115  to       2044\n",
      ".....................---------- new table with       70 representatives\n",
      "# comparing sequences from       2044  to       2818\n",
      "....................---------- new table with       54 representatives\n",
      "# comparing sequences from       2818  to       3463\n",
      "100.0%---------- new table with       49 representatives\n",
      "# comparing sequences from       3463  to       4001\n",
      ".....................---------- new table with       35 representatives\n",
      "# comparing sequences from       4001  to       4449\n",
      "....................---------- new table with       38 representatives\n",
      "# comparing sequences from       4449  to       4823\n",
      "99.9%---------- new table with       28 representatives\n",
      "# comparing sequences from       4823  to       5134\n",
      "....................---------- new table with       22 representatives\n",
      "# comparing sequences from       5134  to       5393\n",
      "99.9%---------- new table with       20 representatives\n",
      "# comparing sequences from       5393  to       5609\n",
      "...................---------- new table with       18 representatives\n",
      "# comparing sequences from       5609  to       5789\n",
      "100.0%---------- new table with       11 representatives\n",
      "# comparing sequences from       5789  to       6693\n",
      ".....................---------- new table with       60 representatives\n",
      "\n",
      "     6693  finished        488  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.03\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bacillus cereus group"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'bacillus'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Bacillus_albus_GCF_001884185\",\n",
    "               \"Bacillus_albus_GCF_003966295\",\n",
    "               \"Bacillus_albus_GCF_004116025\",\n",
    "               \"Bacillus_albus_GCF_004116035\",\n",
    "               \"Bacillus_albus_GCF_004116065\",\n",
    "               \"Bacillus_albus_GCF_004116085\",\n",
    "               \"Bacillus_albus_GCF_004116415\",\n",
    "               \"Bacillus_albus_GCF_004153665\",\n",
    "               \"Bacillus_albus_GCF_009740005\",\n",
    "               \"Bacillus_anthracis_GCF_000558965\",\n",
    "               \"Bacillus_anthracis_GCF_000559005\",\n",
    "               \"Bacillus_anthracis_GCF_000697555\",\n",
    "               \"Bacillus_anthracis_GCF_000742875\",\n",
    "               \"Bacillus_anthracis_GCF_000783055\",\n",
    "               \"Bacillus_anthracis_GCF_001883895\",\n",
    "               \"Bacillus_anthracis_GCF_002233635\",\n",
    "               \"Bacillus_anthracis_GCF_002525695\",\n",
    "               \"Bacillus_anthracis_GCF_002525705\",\n",
    "               \"Bacillus_bombysepticus_GCF_000831065\",\n",
    "               \"Bacillus_cereus_GCF_000161315\",\n",
    "               \"Bacillus_cereus_GCF_000290795\",\n",
    "               \"Bacillus_cereus_GCF_000291075\",\n",
    "               \"Bacillus_cereus_GCF_001583875\",\n",
    "               \"Bacillus_cereus_GCF_001584025\",\n",
    "               \"Bacillus_cereus_GCF_001683695\",\n",
    "               \"Bacillus_cereus_GCF_002530565\",\n",
    "               \"Bacillus_cereus_GCF_002552015\",\n",
    "               \"Bacillus_cereus_GCF_002560715\",\n",
    "               \"Bacillus_cereus_GCF_002565765\",\n",
    "               \"Bacillus_cytotoxicus_GCF_000017425\",\n",
    "               \"Bacillus_cytotoxicus_GCF_001050335\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002250885\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002250905\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002250925\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002250945\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002250965\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002251005\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002251025\",\n",
    "               \"Bacillus_cytotoxicus_GCF_002251045\",\n",
    "               \"Bacillus_gaemokensis_GCF_000712615\",\n",
    "               \"Bacillus_gaemokensis_GCF_001590835\",\n",
    "               \"Bacillus_luti_GCF_001884105\",\n",
    "               \"Bacillus_luti_GCF_008923665\",\n",
    "               \"Bacillus_luti_GCF_009739945\",\n",
    "               \"Bacillus_manliponensis_GCF_000712595\",\n",
    "               \"Bacillus_mobilis_GCF_001884045\",\n",
    "               \"Bacillus_mobilis_GCF_003612955\",\n",
    "               \"Bacillus_mobilis_GCF_007681185\",\n",
    "               \"Bacillus_mobilis_GCF_007681195\",\n",
    "               \"Bacillus_mobilis_GCF_007682195\",\n",
    "               \"Bacillus_mycoides_GCF_000018825\",\n",
    "               \"Bacillus_mycoides_GCF_000290695\",\n",
    "               \"Bacillus_mycoides_GCF_000517985\",\n",
    "               \"Bacillus_mycoides_GCF_001044935\",\n",
    "               \"Bacillus_mycoides_GCF_001541985\",\n",
    "               \"Bacillus_mycoides_GCF_001757825\",\n",
    "               \"Bacillus_mycoides_GCF_001757915\",\n",
    "               \"Bacillus_mycoides_GCF_001757985\",\n",
    "               \"Bacillus_mycoides_GCF_002014505\",\n",
    "               \"Bacillus_mycoides_GCF_002014695\",\n",
    "               \"Bacillus_nitratireducens_GCF_001884135\",\n",
    "               \"Bacillus_nitratireducens_GCF_007676595\",\n",
    "               \"Bacillus_nitratireducens_GCF_007676605\",\n",
    "               \"Bacillus_nitratireducens_GCF_007681065\",\n",
    "               \"Bacillus_nitratireducens_GCF_007681365\",\n",
    "               \"Bacillus_nitratireducens_GCF_007682355\",\n",
    "               \"Bacillus_pacificus_GCA_009884315\",\n",
    "               \"Bacillus_pacificus_GCF_001884025\",\n",
    "               \"Bacillus_pacificus_GCF_003858675\",\n",
    "               \"Bacillus_pacificus_GCF_006349595\",\n",
    "               \"Bacillus_paramycoides_GCF_001884235\",\n",
    "               \"Bacillus_paramycoides_GCF_007682005\",\n",
    "               \"Bacillus_paranthracis_GCA_009648955\",\n",
    "               \"Bacillus_paranthracis_GCA_009873395\",\n",
    "               \"Bacillus_paranthracis_GCA_009884335\",\n",
    "               \"Bacillus_paranthracis_GCF_001883995\",\n",
    "               \"Bacillus_paranthracis_GCF_004307985\",\n",
    "               \"Bacillus_paranthracis_GCF_005117095\",\n",
    "               \"Bacillus_paranthracis_GCF_005848985\",\n",
    "               \"Bacillus_paranthracis_GCF_007682135\",\n",
    "               \"Bacillus_paranthracis_GCF_007682155\",\n",
    "               \"Bacillus_paranthracis_GCF_009498675\",\n",
    "               \"Bacillus_paranthracis_GCF_009799965\",\n",
    "               \"Bacillus_proteolyticus_GCF_001884065\",\n",
    "               \"Bacillus_pseudomycoides_GCF_000161415\",\n",
    "               \"Bacillus_pseudomycoides_GCF_000161435\",\n",
    "               \"Bacillus_pseudomycoides_GCF_000746965\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002550095\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002555615\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002555685\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002555995\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002556205\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002557395\",\n",
    "               \"Bacillus_pseudomycoides_GCF_002569075\",\n",
    "               \"Bacillus_thuringiensis_GCF_000161735\",\n",
    "               \"Bacillus_thuringiensis_GCF_000710255\",\n",
    "               \"Bacillus_thuringiensis_GCF_001402735\",\n",
    "               \"Bacillus_thuringiensis_GCF_001675515\",\n",
    "               \"Bacillus_thuringiensis_GCF_002148135\",\n",
    "               \"Bacillus_thuringiensis_GCF_002551895\",\n",
    "               \"Bacillus_thuringiensis_GCF_002559825\",\n",
    "               \"Bacillus_thuringiensis_GCF_002562665\",\n",
    "               \"Bacillus_thuringiensis_GCF_002564505\",\n",
    "               \"Bacillus_thuringiensis_GCF_002570925\",\n",
    "               \"Bacillus_toyonensis_GCA_009799785\",\n",
    "               \"Bacillus_toyonensis_GCF_002550925\",\n",
    "               \"Bacillus_toyonensis_GCF_002551965\",\n",
    "               \"Bacillus_toyonensis_GCF_002554865\",\n",
    "               \"Bacillus_toyonensis_GCF_002555905\",\n",
    "               \"Bacillus_toyonensis_GCF_002556675\",\n",
    "               \"Bacillus_toyonensis_GCF_002570155\",\n",
    "               \"Bacillus_toyonensis_GCF_002580355\",\n",
    "               \"Bacillus_toyonensis_GCF_002581325\",\n",
    "               \"Bacillus_toyonensis_GCF_002581955\",\n",
    "               \"Bacillus_tropicus_GCF_001884035\",\n",
    "               \"Bacillus_tropicus_GCF_006349625\",\n",
    "               \"Bacillus_tropicus_GCF_006349645\",\n",
    "               \"Bacillus_tropicus_GCF_006457285\",\n",
    "               \"Bacillus_tropicus_GCF_007672275\",\n",
    "               \"Bacillus_tropicus_GCF_007676425\",\n",
    "               \"Bacillus_tropicus_GCF_007682035\",\n",
    "               \"Bacillus_tropicus_GCF_007682405\",\n",
    "               \"Bacillus_wiedmannii_GCF_000160955\",\n",
    "               \"Bacillus_wiedmannii_GCF_001645395\",\n",
    "               \"Bacillus_wiedmannii_GCF_002014635\",\n",
    "               \"Bacillus_wiedmannii_GCF_002555555\",\n",
    "               \"Bacillus_wiedmannii_GCF_002555735\",\n",
    "               \"Bacillus_wiedmannii_GCF_002555855\",\n",
    "               \"Bacillus_wiedmannii_GCF_002555945\",\n",
    "               \"Bacillus_wiedmannii_GCF_002556175\",\n",
    "               \"Bacillus_wiedmannii_GCF_002556405\",\n",
    "               \"Bacillus_wiedmannii_GCF_002569595\" ]\n",
    "                    \n",
    "reference_taxon = \"Bacillus_anthracis_GCF_000008165\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884185.1_ASM188418v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1711674 bytes  380381.33 bytes/sec\n",
      "total size is 1711144  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003966295.1_ASM396629v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1446707 bytes  222576.77 bytes/sec\n",
      "total size is 1446241  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004116025.1_ASM411602v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1731706 bytes  692699.20 bytes/sec\n",
      "total size is 1731176  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004116035.1_ASM411603v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1732190 bytes  384940.44 bytes/sec\n",
      "total size is 1731660  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004116065.1_ASM411606v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1731054 bytes  1154064.00 bytes/sec\n",
      "total size is 1730524  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004116085.1_ASM411608v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1733240 bytes  1155521.33 bytes/sec\n",
      "total size is 1732710  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004116415.1_ASM411641v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1731994 bytes  164955.81 bytes/sec\n",
      "total size is 1731464  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004153665.1_ASM415366v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1912498 bytes  546440.00 bytes/sec\n",
      "total size is 1911920  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009740005.1_ASM974000v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1709868 bytes  1139940.00 bytes/sec\n",
      "total size is 1709338  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000008165.1_ASM816v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1531260 bytes  612520.80 bytes/sec\n",
      "total size is 1530781  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000558965.1_Ba8903G_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1612544 bytes  460738.86 bytes/sec\n",
      "total size is 1612038  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000559005.1_Ba52G_1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1613112 bytes  460901.14 bytes/sec\n",
      "total size is 1612608  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000697555.2_ASM69755v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1598675 bytes  639486.80 bytes/sec\n",
      "total size is 1598178  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000742875.1_ASM74287v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1612268 bytes  248047.69 bytes/sec\n",
      "total size is 1611763  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000783055.1_ASM78305v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1602807 bytes  457956.86 bytes/sec\n",
      "total size is 1602310  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001883895.1_ASM188389v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1594039 bytes  455451.71 bytes/sec\n",
      "total size is 1593541  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002233635.1_ASM223363v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1603733 bytes  356394.44 bytes/sec\n",
      "total size is 1603235  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002525695.1_ASM252569v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1573407 bytes  629379.60 bytes/sec\n",
      "total size is 1572909  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002525705.1_ASM252570v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1577817 bytes  350635.33 bytes/sec\n",
      "total size is 1577319  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000831065.1_ASM83106v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1713506 bytes  489585.14 bytes/sec\n",
      "total size is 1712977  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000161315.1_ASM16131v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1794336 bytes  398750.67 bytes/sec\n",
      "total size is 1793791  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000290795.1_Baci_cere_VD154_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1814030 bytes  213420.24 bytes/sec\n",
      "total size is 1813469  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000291075.1_Baci_cere_HuA4-10_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1693351 bytes  483826.57 bytes/sec\n",
      "total size is 1692820  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001583875.1_ASM158387v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1730080 bytes  164773.52 bytes/sec\n",
      "total size is 1729550  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001584025.1_ASM158402v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1758790 bytes  167507.81 bytes/sec\n",
      "total size is 1758252  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001683695.1_ASM168369v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1543278 bytes  342960.00 bytes/sec\n",
      "total size is 1542788  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002530565.1_ASM253056v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1684866 bytes  481402.29 bytes/sec\n",
      "total size is 1684344  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002552015.1_ASM255201v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1745386 bytes  139634.24 bytes/sec\n",
      "total size is 1744848  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002560715.1_ASM256071v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1727781 bytes  314149.64 bytes/sec\n",
      "total size is 1727251  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002565765.1_ASM256576v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1755215 bytes  501502.00 bytes/sec\n",
      "total size is 1754677  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017425.1_ASM1742v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1196507 bytes  478619.60 bytes/sec\n",
      "total size is 1196107  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001050335.1_ASM105033v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1211050 bytes  346026.29 bytes/sec\n",
      "total size is 1210648  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002250885.2_ASM225088v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1271918 bytes  363417.14 bytes/sec\n",
      "total size is 1271500  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002250905.2_ASM225090v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1271118 bytes  363188.57 bytes/sec\n",
      "total size is 1270700  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002250925.2_ASM225092v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1262568 bytes  280580.00 bytes/sec\n",
      "total size is 1262150  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002250945.2_ASM225094v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1217399 bytes  270542.44 bytes/sec\n",
      "total size is 1216989  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002250965.2_ASM225096v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1256771 bytes  147860.35 bytes/sec\n",
      "total size is 1256353  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002251005.2_ASM225100v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1266928 bytes  361991.43 bytes/sec\n",
      "total size is 1266510  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002251025.2_ASM225102v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1256779 bytes  193357.08 bytes/sec\n",
      "total size is 1256361  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002251045.2_ASM225104v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1282330 bytes  284971.56 bytes/sec\n",
      "total size is 1281904  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000712615.1_ASM71261v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1655153 bytes  300944.55 bytes/sec\n",
      "total size is 1654640  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001590835.1_ASM159083v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1654510 bytes  220606.93 bytes/sec\n",
      "total size is 1653996  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884105.1_ASM188410v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1504013 bytes  273464.55 bytes/sec\n",
      "total size is 1503539  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_008923665.1_ASM892366v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1632111 bytes  652861.20 bytes/sec\n",
      "total size is 1631605  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009739945.1_ASM973994v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1550593 bytes  443038.57 bytes/sec\n",
      "total size is 1550103  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000712595.1_ASM71259v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1368454 bytes  390998.86 bytes/sec\n",
      "total size is 1368013  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884045.1_ASM188404v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1670986 bytes  477436.57 bytes/sec\n",
      "total size is 1670472  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003612955.1_ASM361295v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1597405 bytes  354988.22 bytes/sec\n",
      "total size is 1596907  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007681185.1_ASM768118v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1575602 bytes  150061.33 bytes/sec\n",
      "total size is 1575104  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007681195.1_ASM768119v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1764832 bytes  392194.22 bytes/sec\n",
      "total size is 1764294  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682195.1_ASM768219v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1706127 bytes  310212.55 bytes/sec\n",
      "total size is 1705597  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000018825.1_ASM1882v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1715599 bytes  201840.12 bytes/sec\n",
      "total size is 1715071  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000290695.1_Baci_cere_VDM022_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1800303 bytes  514384.29 bytes/sec\n",
      "total size is 1799749  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000517985.1_ASM51798v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1672652 bytes  477912.57 bytes/sec\n",
      "total size is 1672131  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001044935.1_ASM104493v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1785317 bytes  510102.57 bytes/sec\n",
      "total size is 1784771  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001541985.1_ASM154198v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1748306 bytes  499528.00 bytes/sec\n",
      "total size is 1747768  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001757825.1_ASM175782v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1641848 bytes  469111.43 bytes/sec\n",
      "total size is 1641334  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001757915.1_ASM175791v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1660514 bytes  369012.44 bytes/sec\n",
      "total size is 1660000  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001757985.1_ASM175798v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1668095 bytes  370697.11 bytes/sec\n",
      "total size is 1667581  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002014505.1_ASM201450v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1701319 bytes  680544.40 bytes/sec\n",
      "total size is 1700797  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002014695.1_ASM201469v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1595671 bytes  455918.00 bytes/sec\n",
      "total size is 1595173  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884135.1_ASM188413v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1619625 bytes  462762.00 bytes/sec\n",
      "total size is 1619119  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007676595.1_ASM767659v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1782258 bytes  96340.54 bytes/sec\n",
      "total size is 1781712  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007676605.1_ASM767660v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1789253 bytes  715718.00 bytes/sec\n",
      "total size is 1788707  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007681065.1_ASM768106v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1647185 bytes  470636.29 bytes/sec\n",
      "total size is 1646671  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_007681365.1_ASM768136v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1719053 bytes  382021.11 bytes/sec\n",
      "total size is 1718523  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682355.1_ASM768235v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1730813 bytes  203630.00 bytes/sec\n",
      "total size is 1730283  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009884315.1_ASM988431v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1594942 bytes  138694.26 bytes/sec\n",
      "total size is 1594444  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884025.1_ASM188402v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1597444 bytes  638994.40 bytes/sec\n",
      "total size is 1596946  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003858675.1_ASM385867v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1582096 bytes  452039.43 bytes/sec\n",
      "total size is 1581598  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006349595.1_ASM634959v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1601210 bytes  213500.27 bytes/sec\n",
      "total size is 1600712  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884235.1_ASM188423v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1608859 bytes  459686.00 bytes/sec\n",
      "total size is 1608353  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682005.1_ASM768200v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1709512 bytes  488444.00 bytes/sec\n",
      "total size is 1708982  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009648955.1_ASM964895v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1649550 bytes  366576.00 bytes/sec\n",
      "total size is 1649036  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009873395.1_ASM987339v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1493890 bytes  597572.80 bytes/sec\n",
      "total size is 1493416  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009884335.1_ASM988433v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1609649 bytes  643876.40 bytes/sec\n",
      "total size is 1609143  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001883995.1_ASM188399v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1624670 bytes  361047.11 bytes/sec\n",
      "total size is 1624164  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004307985.1_ASM430798v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1219474 bytes  128370.11 bytes/sec\n",
      "total size is 1219064  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005117095.1_ASM511709v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1531448 bytes  278452.73 bytes/sec\n",
      "total size is 1530966  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005848985.1_ASM584898v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1553705 bytes  345277.11 bytes/sec\n",
      "total size is 1553215  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682135.1_ASM768213v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1624011 bytes  360900.67 bytes/sec\n",
      "total size is 1623505  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682155.1_ASM768215v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1618973 bytes  462575.71 bytes/sec\n",
      "total size is 1618467  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009498675.1_ASM949867v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1675599 bytes  197134.24 bytes/sec\n",
      "total size is 1675077  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009799965.1_ASM979996v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1658659 bytes  473914.57 bytes/sec\n",
      "total size is 1658145  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884065.1_ASM188406v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1729783 bytes  691930.00 bytes/sec\n",
      "total size is 1729253  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000161415.1_ASM16141v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1706512 bytes  310282.55 bytes/sec\n",
      "total size is 1705983  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000161435.1_ASM16143v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1675248 bytes  670116.00 bytes/sec\n",
      "total size is 1674727  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000746965.1_BHU_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1590023 bytes  244625.38 bytes/sec\n",
      "total size is 1589531  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002550095.1_ASM255009v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1680194 bytes  305497.45 bytes/sec\n",
      "total size is 1679672  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555615.1_ASM255561v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1686473 bytes  481861.43 bytes/sec\n",
      "total size is 1685951  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555685.1_ASM255568v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1715350 bytes  686156.80 bytes/sec\n",
      "total size is 1714820  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555995.1_ASM255599v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1634344 bytes  1089590.67 bytes/sec\n",
      "total size is 1633838  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002556205.1_ASM255620v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1661369 bytes  664564.40 bytes/sec\n",
      "total size is 1660855  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002557395.1_ASM255739v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1683207 bytes  673299.60 bytes/sec\n",
      "total size is 1682685  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002569075.1_ASM256907v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1703252 bytes  309689.82 bytes/sec\n",
      "total size is 1702730  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000161735.1_ASM16173v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1933318 bytes  351520.00 bytes/sec\n",
      "total size is 1932741  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000710255.1_BTK_A5_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1884743 bytes  163894.35 bytes/sec\n",
      "total size is 1884178  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001402735.1_ASM140273v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1874175 bytes  749686.80 bytes/sec\n",
      "total size is 1873605  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001675515.1_ASM167551v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1834626 bytes  333576.00 bytes/sec\n",
      "total size is 1834072  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002148135.1_ASM214813v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1766187 bytes  271727.54 bytes/sec\n",
      "total size is 1765649  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002551895.1_ASM255189v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1742493 bytes  387230.00 bytes/sec\n",
      "total size is 1741955  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002559825.1_ASM255982v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1798916 bytes  276762.77 bytes/sec\n",
      "total size is 1798370  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002562665.1_ASM256266v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1791329 bytes  716548.40 bytes/sec\n",
      "total size is 1790783  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002564505.1_ASM256450v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1913250 bytes  255105.60 bytes/sec\n",
      "total size is 1912672  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002570925.1_ASM257092v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1737537 bytes  496451.14 bytes/sec\n",
      "total size is 1736999  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009799785.1_ASM979978v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1562281 bytes  347182.89 bytes/sec\n",
      "total size is 1561791  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002550925.1_ASM255092v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1779311 bytes  273746.62 bytes/sec\n",
      "total size is 1778765  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002551965.1_ASM255196v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1770035 bytes  505736.29 bytes/sec\n",
      "total size is 1769489  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002554865.1_ASM255486v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1600673 bytes  246263.85 bytes/sec\n",
      "total size is 1600175  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002555905.1_ASM255590v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1756276 bytes  206625.65 bytes/sec\n",
      "total size is 1755738  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002556675.1_ASM255667v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1745341 bytes  387862.89 bytes/sec\n",
      "total size is 1744803  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002570155.1_ASM257015v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1784865 bytes  274601.08 bytes/sec\n",
      "total size is 1784319  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002580355.1_ASM258035v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1710472 bytes  263156.00 bytes/sec\n",
      "total size is 1709942  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002581325.1_ASM258132v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1812048 bytes  517740.00 bytes/sec\n",
      "total size is 1811494  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002581955.1_ASM258195v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1649187 bytes  366495.33 bytes/sec\n",
      "total size is 1648673  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001884035.1_ASM188403v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1536552 bytes  341465.33 bytes/sec\n",
      "total size is 1536070  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006349625.1_ASM634962v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1570144 bytes  285488.36 bytes/sec\n",
      "total size is 1569654  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006349645.1_ASM634964v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1755956 bytes  270153.54 bytes/sec\n",
      "total size is 1755418  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006457285.1_ASM645728v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1799489 bytes  1199687.33 bytes/sec\n",
      "total size is 1798943  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007672275.1_ASM767227v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1506359 bytes  1004267.33 bytes/sec\n",
      "total size is 1505885  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_007676425.1_ASM767642v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1531947 bytes  1021326.00 bytes/sec\n",
      "total size is 1531465  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682035.1_ASM768203v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1781098 bytes  1187426.67 bytes/sec\n",
      "total size is 1780552  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007682405.1_ASM768240v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1549697 bytes  619895.60 bytes/sec\n",
      "total size is 1549207  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000160955.1_ASM16095v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1626692 bytes  650693.60 bytes/sec\n",
      "total size is 1626187  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001645395.1_ASM164539v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1653864 bytes  1102604.00 bytes/sec\n",
      "total size is 1653350  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002014635.1_ASM201463v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1624137 bytes  1082786.00 bytes/sec\n",
      "total size is 1623631  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555555.1_ASM255555v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1615180 bytes  1076814.67 bytes/sec\n",
      "total size is 1614674  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555735.1_ASM255573v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1767759 bytes  707120.40 bytes/sec\n",
      "total size is 1767221  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555855.1_ASM255585v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1595214 bytes  1063504.00 bytes/sec\n",
      "total size is 1594716  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002555945.1_ASM255594v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1784238 bytes  1189520.00 bytes/sec\n",
      "total size is 1783692  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002556175.1_ASM255617v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1643907 bytes  657579.60 bytes/sec\n",
      "total size is 1643393  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_002556405.1_ASM255640v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1542952 bytes  1028662.67 bytes/sec\n",
      "total size is 1542462  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002569595.1_ASM256959v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1774754 bytes  709918.40 bytes/sec\n",
      "total size is 1774208  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485056 (\"sim_Bacillus_albus_GCF_001884185\") has been submitted\n",
      "Your job 5485057 (\"sim_Bacillus_albus_GCF_003966295\") has been submitted\n",
      "Your job 5485058 (\"sim_Bacillus_albus_GCF_004116025\") has been submitted\n",
      "Your job 5485059 (\"sim_Bacillus_albus_GCF_004116035\") has been submitted\n",
      "Your job 5485060 (\"sim_Bacillus_albus_GCF_004116065\") has been submitted\n",
      "Your job 5485061 (\"sim_Bacillus_albus_GCF_004116085\") has been submitted\n",
      "Your job 5485062 (\"sim_Bacillus_albus_GCF_004116415\") has been submitted\n",
      "Your job 5485063 (\"sim_Bacillus_albus_GCF_004153665\") has been submitted\n",
      "Your job 5485064 (\"sim_Bacillus_albus_GCF_009740005\") has been submitted\n",
      "Your job 5485065 (\"sim_Bacillus_anthracis_GCF_000558965\") has been submitted\n",
      "Your job 5485066 (\"sim_Bacillus_anthracis_GCF_000559005\") has been submitted\n",
      "Your job 5485067 (\"sim_Bacillus_anthracis_GCF_000697555\") has been submitted\n",
      "Your job 5485068 (\"sim_Bacillus_anthracis_GCF_000742875\") has been submitted\n",
      "Your job 5485069 (\"sim_Bacillus_anthracis_GCF_000783055\") has been submitted\n",
      "Your job 5485070 (\"sim_Bacillus_anthracis_GCF_001883895\") has been submitted\n",
      "Your job 5485071 (\"sim_Bacillus_anthracis_GCF_002233635\") has been submitted\n",
      "Your job 5485072 (\"sim_Bacillus_anthracis_GCF_002525695\") has been submitted\n",
      "Your job 5485073 (\"sim_Bacillus_anthracis_GCF_002525705\") has been submitted\n",
      "Your job 5485074 (\"sim_Bacillus_bombysepticus_GCF_000831065\") has been submitted\n",
      "Your job 5485075 (\"sim_Bacillus_cereus_GCF_000161315\") has been submitted\n",
      "Your job 5485076 (\"sim_Bacillus_cereus_GCF_000290795\") has been submitted\n",
      "Your job 5485077 (\"sim_Bacillus_cereus_GCF_000291075\") has been submitted\n",
      "Your job 5485078 (\"sim_Bacillus_cereus_GCF_001583875\") has been submitted\n",
      "Your job 5485079 (\"sim_Bacillus_cereus_GCF_001584025\") has been submitted\n",
      "Your job 5485080 (\"sim_Bacillus_cereus_GCF_001683695\") has been submitted\n",
      "Your job 5485081 (\"sim_Bacillus_cereus_GCF_002530565\") has been submitted\n",
      "Your job 5485082 (\"sim_Bacillus_cereus_GCF_002552015\") has been submitted\n",
      "Your job 5485083 (\"sim_Bacillus_cereus_GCF_002560715\") has been submitted\n",
      "Your job 5485084 (\"sim_Bacillus_cereus_GCF_002565765\") has been submitted\n",
      "Your job 5485085 (\"sim_Bacillus_cytotoxicus_GCF_000017425\") has been submitted\n",
      "Your job 5485086 (\"sim_Bacillus_cytotoxicus_GCF_001050335\") has been submitted\n",
      "Your job 5485087 (\"sim_Bacillus_cytotoxicus_GCF_002250885\") has been submitted\n",
      "Your job 5485088 (\"sim_Bacillus_cytotoxicus_GCF_002250905\") has been submitted\n",
      "Your job 5485089 (\"sim_Bacillus_cytotoxicus_GCF_002250925\") has been submitted\n",
      "Your job 5485090 (\"sim_Bacillus_cytotoxicus_GCF_002250945\") has been submitted\n",
      "Your job 5485091 (\"sim_Bacillus_cytotoxicus_GCF_002250965\") has been submitted\n",
      "Your job 5485092 (\"sim_Bacillus_cytotoxicus_GCF_002251005\") has been submitted\n",
      "Your job 5485093 (\"sim_Bacillus_cytotoxicus_GCF_002251025\") has been submitted\n",
      "Your job 5485094 (\"sim_Bacillus_cytotoxicus_GCF_002251045\") has been submitted\n",
      "Your job 5485095 (\"sim_Bacillus_gaemokensis_GCF_000712615\") has been submitted\n",
      "Your job 5485096 (\"sim_Bacillus_gaemokensis_GCF_001590835\") has been submitted\n",
      "Your job 5485097 (\"sim_Bacillus_luti_GCF_001884105\") has been submitted\n",
      "Your job 5485098 (\"sim_Bacillus_luti_GCF_008923665\") has been submitted\n",
      "Your job 5485099 (\"sim_Bacillus_luti_GCF_009739945\") has been submitted\n",
      "Your job 5485100 (\"sim_Bacillus_manliponensis_GCF_000712595\") has been submitted\n",
      "Your job 5485101 (\"sim_Bacillus_mobilis_GCF_001884045\") has been submitted\n",
      "Your job 5485102 (\"sim_Bacillus_mobilis_GCF_003612955\") has been submitted\n",
      "Your job 5485103 (\"sim_Bacillus_mobilis_GCF_007681185\") has been submitted\n",
      "Your job 5485104 (\"sim_Bacillus_mobilis_GCF_007681195\") has been submitted\n",
      "Your job 5485105 (\"sim_Bacillus_mobilis_GCF_007682195\") has been submitted\n",
      "Your job 5485106 (\"sim_Bacillus_mycoides_GCF_000018825\") has been submitted\n",
      "Your job 5485107 (\"sim_Bacillus_mycoides_GCF_000290695\") has been submitted\n",
      "Your job 5485108 (\"sim_Bacillus_mycoides_GCF_000517985\") has been submitted\n",
      "Your job 5485109 (\"sim_Bacillus_mycoides_GCF_001044935\") has been submitted\n",
      "Your job 5485110 (\"sim_Bacillus_mycoides_GCF_001541985\") has been submitted\n",
      "Your job 5485111 (\"sim_Bacillus_mycoides_GCF_001757825\") has been submitted\n",
      "Your job 5485112 (\"sim_Bacillus_mycoides_GCF_001757915\") has been submitted\n",
      "Your job 5485113 (\"sim_Bacillus_mycoides_GCF_001757985\") has been submitted\n",
      "Your job 5485114 (\"sim_Bacillus_mycoides_GCF_002014505\") has been submitted\n",
      "Your job 5485115 (\"sim_Bacillus_mycoides_GCF_002014695\") has been submitted\n",
      "Your job 5485116 (\"sim_Bacillus_nitratireducens_GCF_001884135\") has been submitted\n",
      "Your job 5485117 (\"sim_Bacillus_nitratireducens_GCF_007676595\") has been submitted\n",
      "Your job 5485118 (\"sim_Bacillus_nitratireducens_GCF_007676605\") has been submitted\n",
      "Your job 5485119 (\"sim_Bacillus_nitratireducens_GCF_007681065\") has been submitted\n",
      "Your job 5485120 (\"sim_Bacillus_nitratireducens_GCF_007681365\") has been submitted\n",
      "Your job 5485121 (\"sim_Bacillus_nitratireducens_GCF_007682355\") has been submitted\n",
      "Your job 5485122 (\"sim_Bacillus_pacificus_GCA_009884315\") has been submitted\n",
      "Your job 5485123 (\"sim_Bacillus_pacificus_GCF_001884025\") has been submitted\n",
      "Your job 5485124 (\"sim_Bacillus_pacificus_GCF_003858675\") has been submitted\n",
      "Your job 5485125 (\"sim_Bacillus_pacificus_GCF_006349595\") has been submitted\n",
      "Your job 5485126 (\"sim_Bacillus_paramycoides_GCF_001884235\") has been submitted\n",
      "Your job 5485127 (\"sim_Bacillus_paramycoides_GCF_007682005\") has been submitted\n",
      "Your job 5485128 (\"sim_Bacillus_paranthracis_GCA_009648955\") has been submitted\n",
      "Your job 5485129 (\"sim_Bacillus_paranthracis_GCA_009873395\") has been submitted\n",
      "Your job 5485130 (\"sim_Bacillus_paranthracis_GCA_009884335\") has been submitted\n",
      "Your job 5485131 (\"sim_Bacillus_paranthracis_GCF_001883995\") has been submitted\n",
      "Your job 5485132 (\"sim_Bacillus_paranthracis_GCF_004307985\") has been submitted\n",
      "Your job 5485133 (\"sim_Bacillus_paranthracis_GCF_005117095\") has been submitted\n",
      "Your job 5485134 (\"sim_Bacillus_paranthracis_GCF_005848985\") has been submitted\n",
      "Your job 5485135 (\"sim_Bacillus_paranthracis_GCF_007682135\") has been submitted\n",
      "Your job 5485136 (\"sim_Bacillus_paranthracis_GCF_007682155\") has been submitted\n",
      "Your job 5485137 (\"sim_Bacillus_paranthracis_GCF_009498675\") has been submitted\n",
      "Your job 5485138 (\"sim_Bacillus_paranthracis_GCF_009799965\") has been submitted\n",
      "Your job 5485139 (\"sim_Bacillus_proteolyticus_GCF_001884065\") has been submitted\n",
      "Your job 5485140 (\"sim_Bacillus_pseudomycoides_GCF_000161415\") has been submitted\n",
      "Your job 5485141 (\"sim_Bacillus_pseudomycoides_GCF_000161435\") has been submitted\n",
      "Your job 5485142 (\"sim_Bacillus_pseudomycoides_GCF_000746965\") has been submitted\n",
      "Your job 5485143 (\"sim_Bacillus_pseudomycoides_GCF_002550095\") has been submitted\n",
      "Your job 5485144 (\"sim_Bacillus_pseudomycoides_GCF_002555615\") has been submitted\n",
      "Your job 5485145 (\"sim_Bacillus_pseudomycoides_GCF_002555685\") has been submitted\n",
      "Your job 5485146 (\"sim_Bacillus_pseudomycoides_GCF_002555995\") has been submitted\n",
      "Your job 5485147 (\"sim_Bacillus_pseudomycoides_GCF_002556205\") has been submitted\n",
      "Your job 5485148 (\"sim_Bacillus_pseudomycoides_GCF_002557395\") has been submitted\n",
      "Your job 5485149 (\"sim_Bacillus_pseudomycoides_GCF_002569075\") has been submitted\n",
      "Your job 5485150 (\"sim_Bacillus_thuringiensis_GCF_000161735\") has been submitted\n",
      "Your job 5485151 (\"sim_Bacillus_thuringiensis_GCF_000710255\") has been submitted\n",
      "Your job 5485152 (\"sim_Bacillus_thuringiensis_GCF_001402735\") has been submitted\n",
      "Your job 5485153 (\"sim_Bacillus_thuringiensis_GCF_001675515\") has been submitted\n",
      "Your job 5485154 (\"sim_Bacillus_thuringiensis_GCF_002148135\") has been submitted\n",
      "Your job 5485155 (\"sim_Bacillus_thuringiensis_GCF_002551895\") has been submitted\n",
      "Your job 5485156 (\"sim_Bacillus_thuringiensis_GCF_002559825\") has been submitted\n",
      "Your job 5485157 (\"sim_Bacillus_thuringiensis_GCF_002562665\") has been submitted\n",
      "Your job 5485158 (\"sim_Bacillus_thuringiensis_GCF_002564505\") has been submitted\n",
      "Your job 5485159 (\"sim_Bacillus_thuringiensis_GCF_002570925\") has been submitted\n",
      "Your job 5485160 (\"sim_Bacillus_toyonensis_GCA_009799785\") has been submitted\n",
      "Your job 5485161 (\"sim_Bacillus_toyonensis_GCF_002550925\") has been submitted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485162 (\"sim_Bacillus_toyonensis_GCF_002551965\") has been submitted\n",
      "Your job 5485163 (\"sim_Bacillus_toyonensis_GCF_002554865\") has been submitted\n",
      "Your job 5485164 (\"sim_Bacillus_toyonensis_GCF_002555905\") has been submitted\n",
      "Your job 5485165 (\"sim_Bacillus_toyonensis_GCF_002556675\") has been submitted\n",
      "Your job 5485166 (\"sim_Bacillus_toyonensis_GCF_002570155\") has been submitted\n",
      "Your job 5485167 (\"sim_Bacillus_toyonensis_GCF_002580355\") has been submitted\n",
      "Your job 5485168 (\"sim_Bacillus_toyonensis_GCF_002581325\") has been submitted\n",
      "Your job 5485169 (\"sim_Bacillus_toyonensis_GCF_002581955\") has been submitted\n",
      "Your job 5485170 (\"sim_Bacillus_tropicus_GCF_001884035\") has been submitted\n",
      "Your job 5485171 (\"sim_Bacillus_tropicus_GCF_006349625\") has been submitted\n",
      "Your job 5485172 (\"sim_Bacillus_tropicus_GCF_006349645\") has been submitted\n",
      "Your job 5485173 (\"sim_Bacillus_tropicus_GCF_006457285\") has been submitted\n",
      "Your job 5485174 (\"sim_Bacillus_tropicus_GCF_007672275\") has been submitted\n",
      "Your job 5485175 (\"sim_Bacillus_tropicus_GCF_007676425\") has been submitted\n",
      "Your job 5485176 (\"sim_Bacillus_tropicus_GCF_007682035\") has been submitted\n",
      "Your job 5485177 (\"sim_Bacillus_tropicus_GCF_007682405\") has been submitted\n",
      "Your job 5485178 (\"sim_Bacillus_wiedmannii_GCF_000160955\") has been submitted\n",
      "Your job 5485179 (\"sim_Bacillus_wiedmannii_GCF_001645395\") has been submitted\n",
      "Your job 5485180 (\"sim_Bacillus_wiedmannii_GCF_002014635\") has been submitted\n",
      "Your job 5485181 (\"sim_Bacillus_wiedmannii_GCF_002555555\") has been submitted\n",
      "Your job 5485182 (\"sim_Bacillus_wiedmannii_GCF_002555735\") has been submitted\n",
      "Your job 5485183 (\"sim_Bacillus_wiedmannii_GCF_002555855\") has been submitted\n",
      "Your job 5485184 (\"sim_Bacillus_wiedmannii_GCF_002555945\") has been submitted\n",
      "Your job 5485185 (\"sim_Bacillus_wiedmannii_GCF_002556175\") has been submitted\n",
      "Your job 5485186 (\"sim_Bacillus_wiedmannii_GCF_002556405\") has been submitted\n",
      "Your job 5485187 (\"sim_Bacillus_wiedmannii_GCF_002569595\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/bacillus/cleaned_genomes/Bacillus_anthracis_GCF_000008165_formatted.fas path=results/phyluce/bacillus/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485188 (\"map_Bacillus_albus_GCF_001884185\") has been submitted\n",
      "Your job 5485189 (\"map_Bacillus_albus_GCF_003966295\") has been submitted\n",
      "Your job 5485190 (\"map_Bacillus_albus_GCF_004116025\") has been submitted\n",
      "Your job 5485191 (\"map_Bacillus_albus_GCF_004116035\") has been submitted\n",
      "Your job 5485192 (\"map_Bacillus_albus_GCF_004116065\") has been submitted\n",
      "Your job 5485193 (\"map_Bacillus_albus_GCF_004116085\") has been submitted\n",
      "Your job 5485194 (\"map_Bacillus_albus_GCF_004116415\") has been submitted\n",
      "Your job 5485195 (\"map_Bacillus_albus_GCF_004153665\") has been submitted\n",
      "Your job 5485196 (\"map_Bacillus_albus_GCF_009740005\") has been submitted\n",
      "Your job 5485197 (\"map_Bacillus_anthracis_GCF_000558965\") has been submitted\n",
      "Your job 5485198 (\"map_Bacillus_anthracis_GCF_000559005\") has been submitted\n",
      "Your job 5485199 (\"map_Bacillus_anthracis_GCF_000697555\") has been submitted\n",
      "Your job 5485200 (\"map_Bacillus_anthracis_GCF_000742875\") has been submitted\n",
      "Your job 5485201 (\"map_Bacillus_anthracis_GCF_000783055\") has been submitted\n",
      "Your job 5485202 (\"map_Bacillus_anthracis_GCF_001883895\") has been submitted\n",
      "Your job 5485203 (\"map_Bacillus_anthracis_GCF_002233635\") has been submitted\n",
      "Your job 5485204 (\"map_Bacillus_anthracis_GCF_002525695\") has been submitted\n",
      "Your job 5485205 (\"map_Bacillus_anthracis_GCF_002525705\") has been submitted\n",
      "Your job 5485206 (\"map_Bacillus_bombysepticus_GCF_000831065\") has been submitted\n",
      "Your job 5485207 (\"map_Bacillus_cereus_GCF_000161315\") has been submitted\n",
      "Your job 5485208 (\"map_Bacillus_cereus_GCF_000290795\") has been submitted\n",
      "Your job 5485209 (\"map_Bacillus_cereus_GCF_000291075\") has been submitted\n",
      "Your job 5485210 (\"map_Bacillus_cereus_GCF_001583875\") has been submitted\n",
      "Your job 5485211 (\"map_Bacillus_cereus_GCF_001584025\") has been submitted\n",
      "Your job 5485212 (\"map_Bacillus_cereus_GCF_001683695\") has been submitted\n",
      "Your job 5485213 (\"map_Bacillus_cereus_GCF_002530565\") has been submitted\n",
      "Your job 5485214 (\"map_Bacillus_cereus_GCF_002552015\") has been submitted\n",
      "Your job 5485215 (\"map_Bacillus_cereus_GCF_002560715\") has been submitted\n",
      "Your job 5485216 (\"map_Bacillus_cereus_GCF_002565765\") has been submitted\n",
      "Your job 5485217 (\"map_Bacillus_cytotoxicus_GCF_000017425\") has been submitted\n",
      "Your job 5485218 (\"map_Bacillus_cytotoxicus_GCF_001050335\") has been submitted\n",
      "Your job 5485219 (\"map_Bacillus_cytotoxicus_GCF_002250885\") has been submitted\n",
      "Your job 5485220 (\"map_Bacillus_cytotoxicus_GCF_002250905\") has been submitted\n",
      "Your job 5485221 (\"map_Bacillus_cytotoxicus_GCF_002250925\") has been submitted\n",
      "Your job 5485222 (\"map_Bacillus_cytotoxicus_GCF_002250945\") has been submitted\n",
      "Your job 5485223 (\"map_Bacillus_cytotoxicus_GCF_002250965\") has been submitted\n",
      "Your job 5485224 (\"map_Bacillus_cytotoxicus_GCF_002251005\") has been submitted\n",
      "Your job 5485225 (\"map_Bacillus_cytotoxicus_GCF_002251025\") has been submitted\n",
      "Your job 5485226 (\"map_Bacillus_cytotoxicus_GCF_002251045\") has been submitted\n",
      "Your job 5485227 (\"map_Bacillus_gaemokensis_GCF_000712615\") has been submitted\n",
      "Your job 5485228 (\"map_Bacillus_gaemokensis_GCF_001590835\") has been submitted\n",
      "Your job 5485229 (\"map_Bacillus_luti_GCF_001884105\") has been submitted\n",
      "Your job 5485230 (\"map_Bacillus_luti_GCF_008923665\") has been submitted\n",
      "Your job 5485231 (\"map_Bacillus_luti_GCF_009739945\") has been submitted\n",
      "Your job 5485232 (\"map_Bacillus_manliponensis_GCF_000712595\") has been submitted\n",
      "Your job 5485233 (\"map_Bacillus_mobilis_GCF_001884045\") has been submitted\n",
      "Your job 5485234 (\"map_Bacillus_mobilis_GCF_003612955\") has been submitted\n",
      "Your job 5485235 (\"map_Bacillus_mobilis_GCF_007681185\") has been submitted\n",
      "Your job 5485236 (\"map_Bacillus_mobilis_GCF_007681195\") has been submitted\n",
      "Your job 5485237 (\"map_Bacillus_mobilis_GCF_007682195\") has been submitted\n",
      "Your job 5485238 (\"map_Bacillus_mycoides_GCF_000018825\") has been submitted\n",
      "Your job 5485239 (\"map_Bacillus_mycoides_GCF_000290695\") has been submitted\n",
      "Your job 5485240 (\"map_Bacillus_mycoides_GCF_000517985\") has been submitted\n",
      "Your job 5485241 (\"map_Bacillus_mycoides_GCF_001044935\") has been submitted\n",
      "Your job 5485242 (\"map_Bacillus_mycoides_GCF_001541985\") has been submitted\n",
      "Your job 5485243 (\"map_Bacillus_mycoides_GCF_001757825\") has been submitted\n",
      "Your job 5485244 (\"map_Bacillus_mycoides_GCF_001757915\") has been submitted\n",
      "Your job 5485245 (\"map_Bacillus_mycoides_GCF_001757985\") has been submitted\n",
      "Your job 5485246 (\"map_Bacillus_mycoides_GCF_002014505\") has been submitted\n",
      "Your job 5485247 (\"map_Bacillus_mycoides_GCF_002014695\") has been submitted\n",
      "Your job 5485248 (\"map_Bacillus_nitratireducens_GCF_001884135\") has been submitted\n",
      "Your job 5485249 (\"map_Bacillus_nitratireducens_GCF_007676595\") has been submitted\n",
      "Your job 5485250 (\"map_Bacillus_nitratireducens_GCF_007676605\") has been submitted\n",
      "Your job 5485251 (\"map_Bacillus_nitratireducens_GCF_007681065\") has been submitted\n",
      "Your job 5485252 (\"map_Bacillus_nitratireducens_GCF_007681365\") has been submitted\n",
      "Your job 5485253 (\"map_Bacillus_nitratireducens_GCF_007682355\") has been submitted\n",
      "Your job 5485254 (\"map_Bacillus_pacificus_GCA_009884315\") has been submitted\n",
      "Your job 5485255 (\"map_Bacillus_pacificus_GCF_001884025\") has been submitted\n",
      "Your job 5485256 (\"map_Bacillus_pacificus_GCF_003858675\") has been submitted\n",
      "Your job 5485257 (\"map_Bacillus_pacificus_GCF_006349595\") has been submitted\n",
      "Your job 5485258 (\"map_Bacillus_paramycoides_GCF_001884235\") has been submitted\n",
      "Your job 5485259 (\"map_Bacillus_paramycoides_GCF_007682005\") has been submitted\n",
      "Your job 5485260 (\"map_Bacillus_paranthracis_GCA_009648955\") has been submitted\n",
      "Your job 5485261 (\"map_Bacillus_paranthracis_GCA_009873395\") has been submitted\n",
      "Your job 5485262 (\"map_Bacillus_paranthracis_GCA_009884335\") has been submitted\n",
      "Your job 5485263 (\"map_Bacillus_paranthracis_GCF_001883995\") has been submitted\n",
      "Your job 5485264 (\"map_Bacillus_paranthracis_GCF_004307985\") has been submitted\n",
      "Your job 5485265 (\"map_Bacillus_paranthracis_GCF_005117095\") has been submitted\n",
      "Your job 5485266 (\"map_Bacillus_paranthracis_GCF_005848985\") has been submitted\n",
      "Your job 5485267 (\"map_Bacillus_paranthracis_GCF_007682135\") has been submitted\n",
      "Your job 5485268 (\"map_Bacillus_paranthracis_GCF_007682155\") has been submitted\n",
      "Your job 5485269 (\"map_Bacillus_paranthracis_GCF_009498675\") has been submitted\n",
      "Your job 5485270 (\"map_Bacillus_paranthracis_GCF_009799965\") has been submitted\n",
      "Your job 5485271 (\"map_Bacillus_proteolyticus_GCF_001884065\") has been submitted\n",
      "Your job 5485272 (\"map_Bacillus_pseudomycoides_GCF_000161415\") has been submitted\n",
      "Your job 5485273 (\"map_Bacillus_pseudomycoides_GCF_000161435\") has been submitted\n",
      "Your job 5485274 (\"map_Bacillus_pseudomycoides_GCF_000746965\") has been submitted\n",
      "Your job 5485275 (\"map_Bacillus_pseudomycoides_GCF_002550095\") has been submitted\n",
      "Your job 5485276 (\"map_Bacillus_pseudomycoides_GCF_002555615\") has been submitted\n",
      "Your job 5485277 (\"map_Bacillus_pseudomycoides_GCF_002555685\") has been submitted\n",
      "Your job 5485278 (\"map_Bacillus_pseudomycoides_GCF_002555995\") has been submitted\n",
      "Your job 5485279 (\"map_Bacillus_pseudomycoides_GCF_002556205\") has been submitted\n",
      "Your job 5485280 (\"map_Bacillus_pseudomycoides_GCF_002557395\") has been submitted\n",
      "Your job 5485281 (\"map_Bacillus_pseudomycoides_GCF_002569075\") has been submitted\n",
      "Your job 5485282 (\"map_Bacillus_thuringiensis_GCF_000161735\") has been submitted\n",
      "Your job 5485283 (\"map_Bacillus_thuringiensis_GCF_000710255\") has been submitted\n",
      "Your job 5485284 (\"map_Bacillus_thuringiensis_GCF_001402735\") has been submitted\n",
      "Your job 5485285 (\"map_Bacillus_thuringiensis_GCF_001675515\") has been submitted\n",
      "Your job 5485286 (\"map_Bacillus_thuringiensis_GCF_002148135\") has been submitted\n",
      "Your job 5485287 (\"map_Bacillus_thuringiensis_GCF_002551895\") has been submitted\n",
      "Your job 5485288 (\"map_Bacillus_thuringiensis_GCF_002559825\") has been submitted\n",
      "Your job 5485289 (\"map_Bacillus_thuringiensis_GCF_002562665\") has been submitted\n",
      "Your job 5485290 (\"map_Bacillus_thuringiensis_GCF_002564505\") has been submitted\n",
      "Your job 5485291 (\"map_Bacillus_thuringiensis_GCF_002570925\") has been submitted\n",
      "Your job 5485292 (\"map_Bacillus_toyonensis_GCA_009799785\") has been submitted\n",
      "Your job 5485293 (\"map_Bacillus_toyonensis_GCF_002550925\") has been submitted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485294 (\"map_Bacillus_toyonensis_GCF_002551965\") has been submitted\n",
      "Your job 5485295 (\"map_Bacillus_toyonensis_GCF_002554865\") has been submitted\n",
      "Your job 5485296 (\"map_Bacillus_toyonensis_GCF_002555905\") has been submitted\n",
      "Your job 5485297 (\"map_Bacillus_toyonensis_GCF_002556675\") has been submitted\n",
      "Your job 5485298 (\"map_Bacillus_toyonensis_GCF_002570155\") has been submitted\n",
      "Your job 5485299 (\"map_Bacillus_toyonensis_GCF_002580355\") has been submitted\n",
      "Your job 5485300 (\"map_Bacillus_toyonensis_GCF_002581325\") has been submitted\n",
      "Your job 5485301 (\"map_Bacillus_toyonensis_GCF_002581955\") has been submitted\n",
      "Your job 5485302 (\"map_Bacillus_tropicus_GCF_001884035\") has been submitted\n",
      "Your job 5485303 (\"map_Bacillus_tropicus_GCF_006349625\") has been submitted\n",
      "Your job 5485304 (\"map_Bacillus_tropicus_GCF_006349645\") has been submitted\n",
      "Your job 5485305 (\"map_Bacillus_tropicus_GCF_006457285\") has been submitted\n",
      "Your job 5485306 (\"map_Bacillus_tropicus_GCF_007672275\") has been submitted\n",
      "Your job 5485307 (\"map_Bacillus_tropicus_GCF_007676425\") has been submitted\n",
      "Your job 5485308 (\"map_Bacillus_tropicus_GCF_007682035\") has been submitted\n",
      "Your job 5485309 (\"map_Bacillus_tropicus_GCF_007682405\") has been submitted\n",
      "Your job 5485310 (\"map_Bacillus_wiedmannii_GCF_000160955\") has been submitted\n",
      "Your job 5485311 (\"map_Bacillus_wiedmannii_GCF_001645395\") has been submitted\n",
      "Your job 5485312 (\"map_Bacillus_wiedmannii_GCF_002014635\") has been submitted\n",
      "Your job 5485313 (\"map_Bacillus_wiedmannii_GCF_002555555\") has been submitted\n",
      "Your job 5485314 (\"map_Bacillus_wiedmannii_GCF_002555735\") has been submitted\n",
      "Your job 5485315 (\"map_Bacillus_wiedmannii_GCF_002555855\") has been submitted\n",
      "Your job 5485316 (\"map_Bacillus_wiedmannii_GCF_002555945\") has been submitted\n",
      "Your job 5485317 (\"map_Bacillus_wiedmannii_GCF_002556175\") has been submitted\n",
      "Your job 5485318 (\"map_Bacillus_wiedmannii_GCF_002556405\") has been submitted\n",
      "Your job 5485319 (\"map_Bacillus_wiedmannii_GCF_002569595\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485320 (\"merge_Bacillus_albus_GCF_001884185\") has been submitted\n",
      "Your job 5485321 (\"merge_Bacillus_albus_GCF_003966295\") has been submitted\n",
      "Your job 5485322 (\"merge_Bacillus_albus_GCF_004116025\") has been submitted\n",
      "Your job 5485323 (\"merge_Bacillus_albus_GCF_004116035\") has been submitted\n",
      "Your job 5485324 (\"merge_Bacillus_albus_GCF_004116065\") has been submitted\n",
      "Your job 5485325 (\"merge_Bacillus_albus_GCF_004116085\") has been submitted\n",
      "Your job 5485326 (\"merge_Bacillus_albus_GCF_004116415\") has been submitted\n",
      "Your job 5485327 (\"merge_Bacillus_albus_GCF_004153665\") has been submitted\n",
      "Your job 5485328 (\"merge_Bacillus_albus_GCF_009740005\") has been submitted\n",
      "Your job 5485329 (\"merge_Bacillus_anthracis_GCF_000558965\") has been submitted\n",
      "Your job 5485330 (\"merge_Bacillus_anthracis_GCF_000559005\") has been submitted\n",
      "Your job 5485331 (\"merge_Bacillus_anthracis_GCF_000697555\") has been submitted\n",
      "Your job 5485332 (\"merge_Bacillus_anthracis_GCF_000742875\") has been submitted\n",
      "Your job 5485333 (\"merge_Bacillus_anthracis_GCF_000783055\") has been submitted\n",
      "Your job 5485334 (\"merge_Bacillus_anthracis_GCF_001883895\") has been submitted\n",
      "Your job 5485335 (\"merge_Bacillus_anthracis_GCF_002233635\") has been submitted\n",
      "Your job 5485336 (\"merge_Bacillus_anthracis_GCF_002525695\") has been submitted\n",
      "Your job 5485337 (\"merge_Bacillus_anthracis_GCF_002525705\") has been submitted\n",
      "Your job 5485338 (\"merge_Bacillus_bombysepticus_GCF_000831065\") has been submitted\n",
      "Your job 5485339 (\"merge_Bacillus_cereus_GCF_000161315\") has been submitted\n",
      "Your job 5485340 (\"merge_Bacillus_cereus_GCF_000290795\") has been submitted\n",
      "Your job 5485341 (\"merge_Bacillus_cereus_GCF_000291075\") has been submitted\n",
      "Your job 5485342 (\"merge_Bacillus_cereus_GCF_001583875\") has been submitted\n",
      "Your job 5485343 (\"merge_Bacillus_cereus_GCF_001584025\") has been submitted\n",
      "Your job 5485344 (\"merge_Bacillus_cereus_GCF_001683695\") has been submitted\n",
      "Your job 5485345 (\"merge_Bacillus_cereus_GCF_002530565\") has been submitted\n",
      "Your job 5485346 (\"merge_Bacillus_cereus_GCF_002552015\") has been submitted\n",
      "Your job 5485347 (\"merge_Bacillus_cereus_GCF_002560715\") has been submitted\n",
      "Your job 5485348 (\"merge_Bacillus_cereus_GCF_002565765\") has been submitted\n",
      "Your job 5485349 (\"merge_Bacillus_cytotoxicus_GCF_000017425\") has been submitted\n",
      "Your job 5485350 (\"merge_Bacillus_cytotoxicus_GCF_001050335\") has been submitted\n",
      "Your job 5485351 (\"merge_Bacillus_cytotoxicus_GCF_002250885\") has been submitted\n",
      "Your job 5485352 (\"merge_Bacillus_cytotoxicus_GCF_002250905\") has been submitted\n",
      "Your job 5485353 (\"merge_Bacillus_cytotoxicus_GCF_002250925\") has been submitted\n",
      "Your job 5485354 (\"merge_Bacillus_cytotoxicus_GCF_002250945\") has been submitted\n",
      "Your job 5485355 (\"merge_Bacillus_cytotoxicus_GCF_002250965\") has been submitted\n",
      "Your job 5485356 (\"merge_Bacillus_cytotoxicus_GCF_002251005\") has been submitted\n",
      "Your job 5485357 (\"merge_Bacillus_cytotoxicus_GCF_002251025\") has been submitted\n",
      "Your job 5485358 (\"merge_Bacillus_cytotoxicus_GCF_002251045\") has been submitted\n",
      "Your job 5485359 (\"merge_Bacillus_gaemokensis_GCF_000712615\") has been submitted\n",
      "Your job 5485360 (\"merge_Bacillus_gaemokensis_GCF_001590835\") has been submitted\n",
      "Your job 5485361 (\"merge_Bacillus_luti_GCF_001884105\") has been submitted\n",
      "Your job 5485362 (\"merge_Bacillus_luti_GCF_008923665\") has been submitted\n",
      "Your job 5485363 (\"merge_Bacillus_luti_GCF_009739945\") has been submitted\n",
      "Your job 5485364 (\"merge_Bacillus_manliponensis_GCF_000712595\") has been submitted\n",
      "Your job 5485365 (\"merge_Bacillus_mobilis_GCF_001884045\") has been submitted\n",
      "Your job 5485366 (\"merge_Bacillus_mobilis_GCF_003612955\") has been submitted\n",
      "Your job 5485367 (\"merge_Bacillus_mobilis_GCF_007681185\") has been submitted\n",
      "Your job 5485368 (\"merge_Bacillus_mobilis_GCF_007681195\") has been submitted\n",
      "Your job 5485369 (\"merge_Bacillus_mobilis_GCF_007682195\") has been submitted\n",
      "Your job 5485370 (\"merge_Bacillus_mycoides_GCF_000018825\") has been submitted\n",
      "Your job 5485371 (\"merge_Bacillus_mycoides_GCF_000290695\") has been submitted\n",
      "Your job 5485372 (\"merge_Bacillus_mycoides_GCF_000517985\") has been submitted\n",
      "Your job 5485373 (\"merge_Bacillus_mycoides_GCF_001044935\") has been submitted\n",
      "Your job 5485374 (\"merge_Bacillus_mycoides_GCF_001541985\") has been submitted\n",
      "Your job 5485375 (\"merge_Bacillus_mycoides_GCF_001757825\") has been submitted\n",
      "Your job 5485376 (\"merge_Bacillus_mycoides_GCF_001757915\") has been submitted\n",
      "Your job 5485377 (\"merge_Bacillus_mycoides_GCF_001757985\") has been submitted\n",
      "Your job 5485378 (\"merge_Bacillus_mycoides_GCF_002014505\") has been submitted\n",
      "Your job 5485379 (\"merge_Bacillus_mycoides_GCF_002014695\") has been submitted\n",
      "Your job 5485380 (\"merge_Bacillus_nitratireducens_GCF_001884135\") has been submitted\n",
      "Your job 5485381 (\"merge_Bacillus_nitratireducens_GCF_007676595\") has been submitted\n",
      "Your job 5485382 (\"merge_Bacillus_nitratireducens_GCF_007676605\") has been submitted\n",
      "Your job 5485383 (\"merge_Bacillus_nitratireducens_GCF_007681065\") has been submitted\n",
      "Your job 5485384 (\"merge_Bacillus_nitratireducens_GCF_007681365\") has been submitted\n",
      "Your job 5485385 (\"merge_Bacillus_nitratireducens_GCF_007682355\") has been submitted\n",
      "Your job 5485386 (\"merge_Bacillus_pacificus_GCA_009884315\") has been submitted\n",
      "Your job 5485387 (\"merge_Bacillus_pacificus_GCF_001884025\") has been submitted\n",
      "Your job 5485388 (\"merge_Bacillus_pacificus_GCF_003858675\") has been submitted\n",
      "Your job 5485389 (\"merge_Bacillus_pacificus_GCF_006349595\") has been submitted\n",
      "Your job 5485390 (\"merge_Bacillus_paramycoides_GCF_001884235\") has been submitted\n",
      "Your job 5485391 (\"merge_Bacillus_paramycoides_GCF_007682005\") has been submitted\n",
      "Your job 5485392 (\"merge_Bacillus_paranthracis_GCA_009648955\") has been submitted\n",
      "Your job 5485393 (\"merge_Bacillus_paranthracis_GCA_009873395\") has been submitted\n",
      "Your job 5485394 (\"merge_Bacillus_paranthracis_GCA_009884335\") has been submitted\n",
      "Your job 5485395 (\"merge_Bacillus_paranthracis_GCF_001883995\") has been submitted\n",
      "Your job 5485396 (\"merge_Bacillus_paranthracis_GCF_004307985\") has been submitted\n",
      "Your job 5485397 (\"merge_Bacillus_paranthracis_GCF_005117095\") has been submitted\n",
      "Your job 5485398 (\"merge_Bacillus_paranthracis_GCF_005848985\") has been submitted\n",
      "Your job 5485399 (\"merge_Bacillus_paranthracis_GCF_007682135\") has been submitted\n",
      "Your job 5485400 (\"merge_Bacillus_paranthracis_GCF_007682155\") has been submitted\n",
      "Your job 5485401 (\"merge_Bacillus_paranthracis_GCF_009498675\") has been submitted\n",
      "Your job 5485402 (\"merge_Bacillus_paranthracis_GCF_009799965\") has been submitted\n",
      "Your job 5485403 (\"merge_Bacillus_proteolyticus_GCF_001884065\") has been submitted\n",
      "Your job 5485404 (\"merge_Bacillus_pseudomycoides_GCF_000161415\") has been submitted\n",
      "Your job 5485405 (\"merge_Bacillus_pseudomycoides_GCF_000161435\") has been submitted\n",
      "Your job 5485406 (\"merge_Bacillus_pseudomycoides_GCF_000746965\") has been submitted\n",
      "Your job 5485407 (\"merge_Bacillus_pseudomycoides_GCF_002550095\") has been submitted\n",
      "Your job 5485408 (\"merge_Bacillus_pseudomycoides_GCF_002555615\") has been submitted\n",
      "Your job 5485409 (\"merge_Bacillus_pseudomycoides_GCF_002555685\") has been submitted\n",
      "Your job 5485410 (\"merge_Bacillus_pseudomycoides_GCF_002555995\") has been submitted\n",
      "Your job 5485411 (\"merge_Bacillus_pseudomycoides_GCF_002556205\") has been submitted\n",
      "Your job 5485412 (\"merge_Bacillus_pseudomycoides_GCF_002557395\") has been submitted\n",
      "Your job 5485413 (\"merge_Bacillus_pseudomycoides_GCF_002569075\") has been submitted\n",
      "Your job 5485414 (\"merge_Bacillus_thuringiensis_GCF_000161735\") has been submitted\n",
      "Your job 5485415 (\"merge_Bacillus_thuringiensis_GCF_000710255\") has been submitted\n",
      "Your job 5485416 (\"merge_Bacillus_thuringiensis_GCF_001402735\") has been submitted\n",
      "Your job 5485417 (\"merge_Bacillus_thuringiensis_GCF_001675515\") has been submitted\n",
      "Your job 5485418 (\"merge_Bacillus_thuringiensis_GCF_002148135\") has been submitted\n",
      "Your job 5485419 (\"merge_Bacillus_thuringiensis_GCF_002551895\") has been submitted\n",
      "Your job 5485420 (\"merge_Bacillus_thuringiensis_GCF_002559825\") has been submitted\n",
      "Your job 5485421 (\"merge_Bacillus_thuringiensis_GCF_002562665\") has been submitted\n",
      "Your job 5485422 (\"merge_Bacillus_thuringiensis_GCF_002564505\") has been submitted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5485423 (\"merge_Bacillus_thuringiensis_GCF_002570925\") has been submitted\n",
      "Your job 5485424 (\"merge_Bacillus_toyonensis_GCA_009799785\") has been submitted\n",
      "Your job 5485425 (\"merge_Bacillus_toyonensis_GCF_002550925\") has been submitted\n",
      "Your job 5485426 (\"merge_Bacillus_toyonensis_GCF_002551965\") has been submitted\n",
      "Your job 5485427 (\"merge_Bacillus_toyonensis_GCF_002554865\") has been submitted\n",
      "Your job 5485428 (\"merge_Bacillus_toyonensis_GCF_002555905\") has been submitted\n",
      "Your job 5485429 (\"merge_Bacillus_toyonensis_GCF_002556675\") has been submitted\n",
      "Your job 5485430 (\"merge_Bacillus_toyonensis_GCF_002570155\") has been submitted\n",
      "Your job 5485431 (\"merge_Bacillus_toyonensis_GCF_002580355\") has been submitted\n",
      "Your job 5485432 (\"merge_Bacillus_toyonensis_GCF_002581325\") has been submitted\n",
      "Your job 5485433 (\"merge_Bacillus_toyonensis_GCF_002581955\") has been submitted\n",
      "Your job 5485434 (\"merge_Bacillus_tropicus_GCF_001884035\") has been submitted\n",
      "Your job 5485435 (\"merge_Bacillus_tropicus_GCF_006349625\") has been submitted\n",
      "Your job 5485436 (\"merge_Bacillus_tropicus_GCF_006349645\") has been submitted\n",
      "Your job 5485437 (\"merge_Bacillus_tropicus_GCF_006457285\") has been submitted\n",
      "Your job 5485438 (\"merge_Bacillus_tropicus_GCF_007672275\") has been submitted\n",
      "Your job 5485439 (\"merge_Bacillus_tropicus_GCF_007676425\") has been submitted\n",
      "Your job 5485440 (\"merge_Bacillus_tropicus_GCF_007682035\") has been submitted\n",
      "Your job 5485441 (\"merge_Bacillus_tropicus_GCF_007682405\") has been submitted\n",
      "Your job 5485442 (\"merge_Bacillus_wiedmannii_GCF_000160955\") has been submitted\n",
      "Your job 5485443 (\"merge_Bacillus_wiedmannii_GCF_001645395\") has been submitted\n",
      "Your job 5485444 (\"merge_Bacillus_wiedmannii_GCF_002014635\") has been submitted\n",
      "Your job 5485445 (\"merge_Bacillus_wiedmannii_GCF_002555555\") has been submitted\n",
      "Your job 5485446 (\"merge_Bacillus_wiedmannii_GCF_002555735\") has been submitted\n",
      "Your job 5485447 (\"merge_Bacillus_wiedmannii_GCF_002555855\") has been submitted\n",
      "Your job 5485448 (\"merge_Bacillus_wiedmannii_GCF_002555945\") has been submitted\n",
      "Your job 5485449 (\"merge_Bacillus_wiedmannii_GCF_002556175\") has been submitted\n",
      "Your job 5485450 (\"merge_Bacillus_wiedmannii_GCF_002556405\") has been submitted\n",
      "Your job 5485451 (\"merge_Bacillus_wiedmannii_GCF_002569595\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 3525 sequences from Bacillus_albus_GCF_001884185_merged.bed.  Filtered 566 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2959.\n",
      "Screened 3372 sequences from Bacillus_albus_GCF_003966295_merged.bed.  Filtered 562 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2810.\n",
      "Screened 3563 sequences from Bacillus_albus_GCF_004116025_merged.bed.  Filtered 584 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2979.\n",
      "Screened 3530 sequences from Bacillus_albus_GCF_004116035_merged.bed.  Filtered 580 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2950.\n",
      "Screened 3566 sequences from Bacillus_albus_GCF_004116065_merged.bed.  Filtered 574 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2992.\n",
      "Screened 3547 sequences from Bacillus_albus_GCF_004116085_merged.bed.  Filtered 575 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2972.\n",
      "Screened 3562 sequences from Bacillus_albus_GCF_004116415_merged.bed.  Filtered 594 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2968.\n",
      "Screened 5605 sequences from Bacillus_albus_GCF_004153665_merged.bed.  Filtered 1249 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4356.\n",
      "Screened 3573 sequences from Bacillus_albus_GCF_009740005_merged.bed.  Filtered 599 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2974.\n",
      "Screened 119 sequences from Bacillus_anthracis_GCF_000558965_merged.bed.  Filtered 5 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 114.\n",
      "Screened 265 sequences from Bacillus_anthracis_GCF_000559005_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 263.\n",
      "Screened 118 sequences from Bacillus_anthracis_GCF_000697555_merged.bed.  Filtered 7 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 87 sequences from Bacillus_anthracis_GCF_000742875_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 86.\n",
      "Screened 141 sequences from Bacillus_anthracis_GCF_000783055_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 127.\n",
      "Screened 1359 sequences from Bacillus_anthracis_GCF_001883895_merged.bed.  Filtered 155 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1204.\n",
      "Screened 89 sequences from Bacillus_anthracis_GCF_002233635_merged.bed.  Filtered 4 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 73 sequences from Bacillus_anthracis_GCF_002525695_merged.bed.  Filtered 1 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 72.\n",
      "Screened 82 sequences from Bacillus_anthracis_GCF_002525705_merged.bed.  Filtered 2 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 5602 sequences from Bacillus_bombysepticus_GCF_000831065_merged.bed.  Filtered 1248 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4354.\n",
      "Screened 5638 sequences from Bacillus_cereus_GCF_000161315_merged.bed.  Filtered 1201 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4437.\n",
      "Screened 5427 sequences from Bacillus_cereus_GCF_000290795_merged.bed.  Filtered 1115 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4312.\n",
      "Screened 7190 sequences from Bacillus_cereus_GCF_000291075_merged.bed.  Filtered 2005 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5185.\n",
      "Screened 5569 sequences from Bacillus_cereus_GCF_001583875_merged.bed.  Filtered 1213 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4356.\n",
      "Screened 6306 sequences from Bacillus_cereus_GCF_001584025_merged.bed.  Filtered 1472 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4834.\n",
      "Screened 2899 sequences from Bacillus_cereus_GCF_001683695_merged.bed.  Filtered 392 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2507.\n",
      "Screened 2363 sequences from Bacillus_cereus_GCF_002530565_merged.bed.  Filtered 325 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2038.\n",
      "Screened 5630 sequences from Bacillus_cereus_GCF_002552015_merged.bed.  Filtered 1210 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4420.\n",
      "Screened 5617 sequences from Bacillus_cereus_GCF_002560715_merged.bed.  Filtered 1186 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4431.\n",
      "Screened 3054 sequences from Bacillus_cereus_GCF_002565765_merged.bed.  Filtered 355 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2699.\n",
      "Screened 3339 sequences from Bacillus_cytotoxicus_GCF_000017425_merged.bed.  Filtered 1781 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1558.\n",
      "Screened 3339 sequences from Bacillus_cytotoxicus_GCF_001050335_merged.bed.  Filtered 1783 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1556.\n",
      "Screened 3314 sequences from Bacillus_cytotoxicus_GCF_002250885_merged.bed.  Filtered 1791 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1523.\n",
      "Screened 3379 sequences from Bacillus_cytotoxicus_GCF_002250905_merged.bed.  Filtered 1806 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1573.\n",
      "Screened 3347 sequences from Bacillus_cytotoxicus_GCF_002250925_merged.bed.  Filtered 1783 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1564.\n",
      "Screened 3328 sequences from Bacillus_cytotoxicus_GCF_002250945_merged.bed.  Filtered 1793 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1535.\n",
      "Screened 3305 sequences from Bacillus_cytotoxicus_GCF_002250965_merged.bed.  Filtered 1735 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1570.\n",
      "Screened 3343 sequences from Bacillus_cytotoxicus_GCF_002251005_merged.bed.  Filtered 1793 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1550.\n",
      "Screened 3305 sequences from Bacillus_cytotoxicus_GCF_002251025_merged.bed.  Filtered 1735 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1570.\n",
      "Screened 3316 sequences from Bacillus_cytotoxicus_GCF_002251045_merged.bed.  Filtered 1743 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1573.\n",
      "Screened 4383 sequences from Bacillus_gaemokensis_GCF_000712615_merged.bed.  Filtered 2260 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2123.\n",
      "Screened 4358 sequences from Bacillus_gaemokensis_GCF_001590835_merged.bed.  Filtered 2176 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2182.\n",
      "Screened 5491 sequences from Bacillus_luti_GCF_001884105_merged.bed.  Filtered 1201 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4290.\n",
      "Screened 5518 sequences from Bacillus_luti_GCF_008923665_merged.bed.  Filtered 1209 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4309.\n",
      "Screened 5496 sequences from Bacillus_luti_GCF_009739945_merged.bed.  Filtered 1243 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4253.\n",
      "Screened 1794 sequences from Bacillus_manliponensis_GCF_000712595_merged.bed.  Filtered 1124 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 670.\n",
      "Screened 3892 sequences from Bacillus_mobilis_GCF_001884045_merged.bed.  Filtered 638 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3254.\n",
      "Screened 3784 sequences from Bacillus_mobilis_GCF_003612955_merged.bed.  Filtered 617 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3167.\n",
      "Screened 3832 sequences from Bacillus_mobilis_GCF_007681185_merged.bed.  Filtered 622 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3210.\n",
      "Screened 3939 sequences from Bacillus_mobilis_GCF_007681195_merged.bed.  Filtered 665 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3274.\n",
      "Screened 3907 sequences from Bacillus_mobilis_GCF_007682195_merged.bed.  Filtered 675 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3232.\n",
      "Screened 7157 sequences from Bacillus_mycoides_GCF_000018825_merged.bed.  Filtered 1985 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5172.\n",
      "Screened 7213 sequences from Bacillus_mycoides_GCF_000290695_merged.bed.  Filtered 2008 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5205.\n",
      "Screened 7228 sequences from Bacillus_mycoides_GCF_000517985_merged.bed.  Filtered 2070 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5158.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 7190 sequences from Bacillus_mycoides_GCF_001044935_merged.bed.  Filtered 2061 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5129.\n",
      "Screened 7300 sequences from Bacillus_mycoides_GCF_001541985_merged.bed.  Filtered 2035 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5265.\n",
      "Screened 6945 sequences from Bacillus_mycoides_GCF_001757825_merged.bed.  Filtered 2067 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4878.\n",
      "Screened 6885 sequences from Bacillus_mycoides_GCF_001757915_merged.bed.  Filtered 1992 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4893.\n",
      "Screened 7014 sequences from Bacillus_mycoides_GCF_001757985_merged.bed.  Filtered 2080 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4934.\n",
      "Screened 7203 sequences from Bacillus_mycoides_GCF_002014505_merged.bed.  Filtered 2033 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5170.\n",
      "Screened 7195 sequences from Bacillus_mycoides_GCF_002014695_merged.bed.  Filtered 2024 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 5171.\n",
      "Screened 6620 sequences from Bacillus_nitratireducens_GCF_001884135_merged.bed.  Filtered 1785 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4835.\n",
      "Screened 6578 sequences from Bacillus_nitratireducens_GCF_007676595_merged.bed.  Filtered 1758 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4820.\n",
      "Screened 6558 sequences from Bacillus_nitratireducens_GCF_007676605_merged.bed.  Filtered 1756 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4802.\n",
      "Screened 6619 sequences from Bacillus_nitratireducens_GCF_007681065_merged.bed.  Filtered 1811 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4808.\n",
      "Screened 6601 sequences from Bacillus_nitratireducens_GCF_007681365_merged.bed.  Filtered 1776 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4825.\n",
      "Screened 6569 sequences from Bacillus_nitratireducens_GCF_007682355_merged.bed.  Filtered 1757 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4812.\n",
      "Screened 2713 sequences from Bacillus_pacificus_GCA_009884315_merged.bed.  Filtered 334 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2379.\n",
      "Screened 2767 sequences from Bacillus_pacificus_GCF_001884025_merged.bed.  Filtered 389 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2378.\n",
      "Screened 2762 sequences from Bacillus_pacificus_GCF_003858675_merged.bed.  Filtered 381 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2381.\n",
      "Screened 2774 sequences from Bacillus_pacificus_GCF_006349595_merged.bed.  Filtered 368 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2406.\n",
      "Screened 7149 sequences from Bacillus_paramycoides_GCF_001884235_merged.bed.  Filtered 2332 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4817.\n",
      "Screened 6998 sequences from Bacillus_paramycoides_GCF_007682005_merged.bed.  Filtered 2235 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4763.\n",
      "Screened 2395 sequences from Bacillus_paranthracis_GCA_009648955_merged.bed.  Filtered 332 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2063.\n",
      "Screened 2330 sequences from Bacillus_paranthracis_GCA_009873395_merged.bed.  Filtered 320 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2010.\n",
      "Screened 2258 sequences from Bacillus_paranthracis_GCA_009884335_merged.bed.  Filtered 282 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1976.\n",
      "Screened 2151 sequences from Bacillus_paranthracis_GCF_001883995_merged.bed.  Filtered 275 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1876.\n",
      "Screened 1807 sequences from Bacillus_paranthracis_GCF_004307985_merged.bed.  Filtered 260 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1547.\n",
      "Screened 2310 sequences from Bacillus_paranthracis_GCF_005117095_merged.bed.  Filtered 288 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2022.\n",
      "Screened 2508 sequences from Bacillus_paranthracis_GCF_005848985_merged.bed.  Filtered 317 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2191.\n",
      "Screened 2385 sequences from Bacillus_paranthracis_GCF_007682135_merged.bed.  Filtered 319 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2066.\n",
      "Screened 2331 sequences from Bacillus_paranthracis_GCF_007682155_merged.bed.  Filtered 282 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2049.\n",
      "Screened 2374 sequences from Bacillus_paranthracis_GCF_009498675_merged.bed.  Filtered 318 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2056.\n",
      "Screened 2344 sequences from Bacillus_paranthracis_GCF_009799965_merged.bed.  Filtered 328 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2016.\n",
      "Screened 6858 sequences from Bacillus_proteolyticus_GCF_001884065_merged.bed.  Filtered 1897 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4961.\n",
      "Screened 4179 sequences from Bacillus_pseudomycoides_GCF_000161415_merged.bed.  Filtered 2064 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2115.\n",
      "Screened 4205 sequences from Bacillus_pseudomycoides_GCF_000161435_merged.bed.  Filtered 2101 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2104.\n",
      "Screened 4224 sequences from Bacillus_pseudomycoides_GCF_000746965_merged.bed.  Filtered 2125 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2099.\n",
      "Screened 4236 sequences from Bacillus_pseudomycoides_GCF_002550095_merged.bed.  Filtered 2093 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2143.\n",
      "Screened 4264 sequences from Bacillus_pseudomycoides_GCF_002555615_merged.bed.  Filtered 2107 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2157.\n",
      "Screened 4212 sequences from Bacillus_pseudomycoides_GCF_002555685_merged.bed.  Filtered 2047 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2165.\n",
      "Screened 4209 sequences from Bacillus_pseudomycoides_GCF_002555995_merged.bed.  Filtered 2084 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2125.\n",
      "Screened 4203 sequences from Bacillus_pseudomycoides_GCF_002556205_merged.bed.  Filtered 2048 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2155.\n",
      "Screened 4162 sequences from Bacillus_pseudomycoides_GCF_002557395_merged.bed.  Filtered 2066 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2096.\n",
      "Screened 4225 sequences from Bacillus_pseudomycoides_GCF_002569075_merged.bed.  Filtered 2133 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2092.\n",
      "Screened 5721 sequences from Bacillus_thuringiensis_GCF_000161735_merged.bed.  Filtered 1169 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4552.\n",
      "Screened 5515 sequences from Bacillus_thuringiensis_GCF_000710255_merged.bed.  Filtered 1216 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4299.\n",
      "Screened 5695 sequences from Bacillus_thuringiensis_GCF_001402735_merged.bed.  Filtered 1220 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4475.\n",
      "Screened 5539 sequences from Bacillus_thuringiensis_GCF_001675515_merged.bed.  Filtered 1202 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4337.\n",
      "Screened 3801 sequences from Bacillus_thuringiensis_GCF_002148135_merged.bed.  Filtered 592 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3209.\n",
      "Screened 5695 sequences from Bacillus_thuringiensis_GCF_002551895_merged.bed.  Filtered 1221 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4474.\n",
      "Screened 5823 sequences from Bacillus_thuringiensis_GCF_002559825_merged.bed.  Filtered 1286 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4537.\n",
      "Screened 5720 sequences from Bacillus_thuringiensis_GCF_002562665_merged.bed.  Filtered 1195 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4525.\n",
      "Screened 5724 sequences from Bacillus_thuringiensis_GCF_002564505_merged.bed.  Filtered 1268 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4456.\n",
      "Screened 5685 sequences from Bacillus_thuringiensis_GCF_002570925_merged.bed.  Filtered 1252 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4433.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 6269 sequences from Bacillus_toyonensis_GCA_009799785_merged.bed.  Filtered 1456 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4813.\n",
      "Screened 6384 sequences from Bacillus_toyonensis_GCF_002550925_merged.bed.  Filtered 1485 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4899.\n",
      "Screened 6391 sequences from Bacillus_toyonensis_GCF_002551965_merged.bed.  Filtered 1489 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4902.\n",
      "Screened 6342 sequences from Bacillus_toyonensis_GCF_002554865_merged.bed.  Filtered 1513 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4829.\n",
      "Screened 6396 sequences from Bacillus_toyonensis_GCF_002555905_merged.bed.  Filtered 1491 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4905.\n",
      "Screened 6325 sequences from Bacillus_toyonensis_GCF_002556675_merged.bed.  Filtered 1448 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4877.\n",
      "Screened 6314 sequences from Bacillus_toyonensis_GCF_002570155_merged.bed.  Filtered 1462 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4852.\n",
      "Screened 6322 sequences from Bacillus_toyonensis_GCF_002580355_merged.bed.  Filtered 1446 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4876.\n",
      "Screened 6363 sequences from Bacillus_toyonensis_GCF_002581325_merged.bed.  Filtered 1473 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4890.\n",
      "Screened 6273 sequences from Bacillus_toyonensis_GCF_002581955_merged.bed.  Filtered 1427 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4846.\n",
      "Screened 2566 sequences from Bacillus_tropicus_GCF_001884035_merged.bed.  Filtered 324 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2242.\n",
      "Screened 2367 sequences from Bacillus_tropicus_GCF_006349625_merged.bed.  Filtered 301 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2066.\n",
      "Screened 2376 sequences from Bacillus_tropicus_GCF_006349645_merged.bed.  Filtered 323 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2053.\n",
      "Screened 2589 sequences from Bacillus_tropicus_GCF_006457285_merged.bed.  Filtered 313 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2276.\n",
      "Screened 2586 sequences from Bacillus_tropicus_GCF_007672275_merged.bed.  Filtered 350 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2236.\n",
      "Screened 2647 sequences from Bacillus_tropicus_GCF_007676425_merged.bed.  Filtered 361 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2286.\n",
      "Screened 2668 sequences from Bacillus_tropicus_GCF_007682035_merged.bed.  Filtered 384 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2284.\n",
      "Screened 2569 sequences from Bacillus_tropicus_GCF_007682405_merged.bed.  Filtered 345 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2224.\n",
      "Screened 3650 sequences from Bacillus_wiedmannii_GCF_000160955_merged.bed.  Filtered 507 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3143.\n",
      "Screened 3613 sequences from Bacillus_wiedmannii_GCF_001645395_merged.bed.  Filtered 549 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3064.\n",
      "Screened 3728 sequences from Bacillus_wiedmannii_GCF_002014635_merged.bed.  Filtered 553 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3175.\n",
      "Screened 3776 sequences from Bacillus_wiedmannii_GCF_002555555_merged.bed.  Filtered 623 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3153.\n",
      "Screened 3824 sequences from Bacillus_wiedmannii_GCF_002555735_merged.bed.  Filtered 631 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3193.\n",
      "Screened 3692 sequences from Bacillus_wiedmannii_GCF_002555855_merged.bed.  Filtered 553 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3139.\n",
      "Screened 3887 sequences from Bacillus_wiedmannii_GCF_002555945_merged.bed.  Filtered 613 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3274.\n",
      "Screened 3845 sequences from Bacillus_wiedmannii_GCF_002556175_merged.bed.  Filtered 616 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3229.\n",
      "Screened 3823 sequences from Bacillus_wiedmannii_GCF_002556405_merged.bed.  Filtered 678 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3145.\n",
      "Screened 3818 sequences from Bacillus_wiedmannii_GCF_002569595_merged.bed.  Filtered 620 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3198.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/bacillus/initial_intervals/Bacillus_wiedmannii_GCF_002569595_merged.bed --twobit results/phyluce/bacillus/cleaned_genomes/Bacillus_anthracis_GCF_000008165_formatted.2bit --output results/phyluce/bacillus/initial_intervals/Bacillus_wiedmannii_GCF_002569595_stripped.bed;\n",
      "bacillus_albus_gcf_001884185...\n",
      "bacillus_albus_gcf_003966295...\n",
      "bacillus_albus_gcf_004116025...\n",
      "bacillus_albus_gcf_004116035...\n",
      "bacillus_albus_gcf_004116065...\n",
      "bacillus_albus_gcf_004116085...\n",
      "bacillus_albus_gcf_004116415...\n",
      "bacillus_albus_gcf_004153665.....\n",
      "bacillus_albus_gcf_009740005...\n",
      "bacillus_anthracis_gcf_000558965.\n",
      "bacillus_anthracis_gcf_000559005.\n",
      "bacillus_anthracis_gcf_000697555.\n",
      "bacillus_anthracis_gcf_000742875.\n",
      "bacillus_anthracis_gcf_000783055.\n",
      "bacillus_anthracis_gcf_001883895..\n",
      "bacillus_anthracis_gcf_002233635.\n",
      "bacillus_anthracis_gcf_002525695.\n",
      "bacillus_anthracis_gcf_002525705.\n",
      "bacillus_bombysepticus_gcf_000831065.....\n",
      "bacillus_cereus_gcf_000161315.....\n",
      "bacillus_cereus_gcf_000290795.....\n",
      "bacillus_cereus_gcf_000291075......\n",
      "bacillus_cereus_gcf_001583875.....\n",
      "bacillus_cereus_gcf_001584025.....\n",
      "bacillus_cereus_gcf_001683695...\n",
      "bacillus_cereus_gcf_002530565...\n",
      "bacillus_cereus_gcf_002552015.....\n",
      "bacillus_cereus_gcf_002560715.....\n",
      "bacillus_cereus_gcf_002565765...\n",
      "bacillus_cytotoxicus_gcf_000017425..\n",
      "bacillus_cytotoxicus_gcf_001050335..\n",
      "bacillus_cytotoxicus_gcf_002250885..\n",
      "bacillus_cytotoxicus_gcf_002250905..\n",
      "bacillus_cytotoxicus_gcf_002250925..\n",
      "bacillus_cytotoxicus_gcf_002250945..\n",
      "bacillus_cytotoxicus_gcf_002250965..\n",
      "bacillus_cytotoxicus_gcf_002251005..\n",
      "bacillus_cytotoxicus_gcf_002251025..\n",
      "bacillus_cytotoxicus_gcf_002251045..\n",
      "bacillus_gaemokensis_gcf_000712615...\n",
      "bacillus_gaemokensis_gcf_001590835...\n",
      "bacillus_luti_gcf_001884105.....\n",
      "bacillus_luti_gcf_008923665.....\n",
      "bacillus_luti_gcf_009739945.....\n",
      "bacillus_manliponensis_gcf_000712595.\n",
      "bacillus_mobilis_gcf_001884045....\n",
      "bacillus_mobilis_gcf_003612955....\n",
      "bacillus_mobilis_gcf_007681185....\n",
      "bacillus_mobilis_gcf_007681195....\n",
      "bacillus_mobilis_gcf_007682195....\n",
      "bacillus_mycoides_gcf_000018825......\n",
      "bacillus_mycoides_gcf_000290695......\n",
      "bacillus_mycoides_gcf_000517985......\n",
      "bacillus_mycoides_gcf_001044935......\n",
      "bacillus_mycoides_gcf_001541985......\n",
      "bacillus_mycoides_gcf_001757825.....\n",
      "bacillus_mycoides_gcf_001757915.....\n",
      "bacillus_mycoides_gcf_001757985.....\n",
      "bacillus_mycoides_gcf_002014505......\n",
      "bacillus_mycoides_gcf_002014695......\n",
      "bacillus_nitratireducens_gcf_001884135.....\n",
      "bacillus_nitratireducens_gcf_007676595.....\n",
      "bacillus_nitratireducens_gcf_007676605.....\n",
      "bacillus_nitratireducens_gcf_007681065.....\n",
      "bacillus_nitratireducens_gcf_007681365.....\n",
      "bacillus_nitratireducens_gcf_007682355.....\n",
      "bacillus_pacificus_gca_009884315...\n",
      "bacillus_pacificus_gcf_001884025...\n",
      "bacillus_pacificus_gcf_003858675...\n",
      "bacillus_pacificus_gcf_006349595...\n",
      "bacillus_paramycoides_gcf_001884235.....\n",
      "bacillus_paramycoides_gcf_007682005.....\n",
      "bacillus_paranthracis_gca_009648955...\n",
      "bacillus_paranthracis_gca_009873395...\n",
      "bacillus_paranthracis_gca_009884335..\n",
      "bacillus_paranthracis_gcf_001883995..\n",
      "bacillus_paranthracis_gcf_004307985..\n",
      "bacillus_paranthracis_gcf_005117095...\n",
      "bacillus_paranthracis_gcf_005848985...\n",
      "bacillus_paranthracis_gcf_007682135...\n",
      "bacillus_paranthracis_gcf_007682155...\n",
      "bacillus_paranthracis_gcf_009498675...\n",
      "bacillus_paranthracis_gcf_009799965...\n",
      "bacillus_proteolyticus_gcf_001884065.....\n",
      "bacillus_pseudomycoides_gcf_000161415...\n",
      "bacillus_pseudomycoides_gcf_000161435...\n",
      "bacillus_pseudomycoides_gcf_000746965...\n",
      "bacillus_pseudomycoides_gcf_002550095...\n",
      "bacillus_pseudomycoides_gcf_002555615...\n",
      "bacillus_pseudomycoides_gcf_002555685...\n",
      "bacillus_pseudomycoides_gcf_002555995...\n",
      "bacillus_pseudomycoides_gcf_002556205...\n",
      "bacillus_pseudomycoides_gcf_002557395...\n",
      "bacillus_pseudomycoides_gcf_002569075...\n",
      "bacillus_thuringiensis_gcf_000161735.....\n",
      "bacillus_thuringiensis_gcf_000710255.....\n",
      "bacillus_thuringiensis_gcf_001402735.....\n",
      "bacillus_thuringiensis_gcf_001675515.....\n",
      "bacillus_thuringiensis_gcf_002148135....\n",
      "bacillus_thuringiensis_gcf_002551895.....\n",
      "bacillus_thuringiensis_gcf_002559825.....\n",
      "bacillus_thuringiensis_gcf_002562665.....\n",
      "bacillus_thuringiensis_gcf_002564505.....\n",
      "bacillus_thuringiensis_gcf_002570925.....\n",
      "bacillus_toyonensis_gca_009799785.....\n",
      "bacillus_toyonensis_gcf_002550925.....\n",
      "bacillus_toyonensis_gcf_002551965.....\n",
      "bacillus_toyonensis_gcf_002554865.....\n",
      "bacillus_toyonensis_gcf_002555905.....\n",
      "bacillus_toyonensis_gcf_002556675.....\n",
      "bacillus_toyonensis_gcf_002570155.....\n",
      "bacillus_toyonensis_gcf_002580355.....\n",
      "bacillus_toyonensis_gcf_002581325.....\n",
      "bacillus_toyonensis_gcf_002581955.....\n",
      "bacillus_tropicus_gcf_001884035...\n",
      "bacillus_tropicus_gcf_006349625...\n",
      "bacillus_tropicus_gcf_006349645...\n",
      "bacillus_tropicus_gcf_006457285...\n",
      "bacillus_tropicus_gcf_007672275...\n",
      "bacillus_tropicus_gcf_007676425...\n",
      "bacillus_tropicus_gcf_007682035...\n",
      "bacillus_tropicus_gcf_007682405...\n",
      "bacillus_wiedmannii_gcf_000160955....\n",
      "bacillus_wiedmannii_gcf_001645395....\n",
      "bacillus_wiedmannii_gcf_002014635....\n",
      "bacillus_wiedmannii_gcf_002555555....\n",
      "bacillus_wiedmannii_gcf_002555735....\n",
      "bacillus_wiedmannii_gcf_002555855....\n",
      "bacillus_wiedmannii_gcf_002555945....\n",
      "bacillus_wiedmannii_gcf_002556175....\n",
      "bacillus_wiedmannii_gcf_002556405....\n",
      "bacillus_wiedmannii_gcf_002569595....\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/bacillus/initial_intervals/Bacillus_wiedmannii_GCF_002569595_merged.bed --twobit results/phyluce/bacillus/cleaned_genomes/Bacillus_anthracis_GCF_000008165_formatted.2bit --output results/phyluce/bacillus/initial_intervals/Bacillus_wiedmannii_GCF_002569595_stripped.bed;\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 0 taxa:\t4,391.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 1 taxa:\t4,391.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 2 taxa:\t4,345.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 3 taxa:\t4,284.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 4 taxa:\t4,253.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 5 taxa:\t4,227.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 6 taxa:\t4,203.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 7 taxa:\t4,187.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 8 taxa:\t4,163.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 9 taxa:\t4,138.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 10 taxa:\t4,118.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 11 taxa:\t4,097.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 12 taxa:\t4,074.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 13 taxa:\t4,054.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 14 taxa:\t4,024.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 15 taxa:\t4,002.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 16 taxa:\t3,985.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 17 taxa:\t3,964.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 18 taxa:\t3,948.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 19 taxa:\t3,935.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 20 taxa:\t3,912.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 21 taxa:\t3,891.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 22 taxa:\t3,865.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 23 taxa:\t3,846.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 24 taxa:\t3,822.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 25 taxa:\t3,806.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 26 taxa:\t3,787.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 27 taxa:\t3,765.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 28 taxa:\t3,746.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 29 taxa:\t3,723.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 30 taxa:\t3,707.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 31 taxa:\t3,680.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 32 taxa:\t3,650.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 33 taxa:\t3,621.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 34 taxa:\t3,608.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 35 taxa:\t3,579.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 36 taxa:\t3,554.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 37 taxa:\t3,518.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 38 taxa:\t3,501.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 39 taxa:\t3,473.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 40 taxa:\t3,452.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 41 taxa:\t3,430.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 42 taxa:\t3,403.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 43 taxa:\t3,382.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 44 taxa:\t3,360.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 45 taxa:\t3,350.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 46 taxa:\t3,330.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 47 taxa:\t3,311.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 48 taxa:\t3,298.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 49 taxa:\t3,268.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 50 taxa:\t3,252.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 51 taxa:\t3,232.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 52 taxa:\t3,206.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 53 taxa:\t3,188.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 54 taxa:\t3,167.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 55 taxa:\t3,145.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 56 taxa:\t3,109.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 57 taxa:\t3,091.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 58 taxa:\t3,066.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 59 taxa:\t3,040.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 60 taxa:\t3,011.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 61 taxa:\t2,978.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 62 taxa:\t2,955.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 63 taxa:\t2,929.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 64 taxa:\t2,914.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 65 taxa:\t2,888.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 66 taxa:\t2,870.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 67 taxa:\t2,845.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 68 taxa:\t2,824.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 69 taxa:\t2,793.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 70 taxa:\t2,762.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 71 taxa:\t2,747.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 72 taxa:\t2,731.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 73 taxa:\t2,706.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 74 taxa:\t2,684.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 75 taxa:\t2,652.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 76 taxa:\t2,631.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 77 taxa:\t2,598.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 78 taxa:\t2,573.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 79 taxa:\t2,548.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 80 taxa:\t2,522.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 81 taxa:\t2,497.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 82 taxa:\t2,468.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 83 taxa:\t2,441.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 84 taxa:\t2,413.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 85 taxa:\t2,384.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 86 taxa:\t2,358.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 87 taxa:\t2,337.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 88 taxa:\t2,295.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 89 taxa:\t2,257.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 90 taxa:\t2,215.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 91 taxa:\t2,176.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 92 taxa:\t2,145.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 93 taxa:\t2,121.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 94 taxa:\t2,085.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 95 taxa:\t2,058.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 96 taxa:\t2,038.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 97 taxa:\t2,006.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 98 taxa:\t1,982.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 99 taxa:\t1,922.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 100 taxa:\t1,886.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 101 taxa:\t1,847.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 102 taxa:\t1,810.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 103 taxa:\t1,768.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 104 taxa:\t1,719.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 105 taxa:\t1,660.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 106 taxa:\t1,596.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 107 taxa:\t1,517.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 108 taxa:\t1,411.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 109 taxa:\t1,211.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 110 taxa:\t890.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 111 taxa:\t845.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 112 taxa:\t805.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 113 taxa:\t773.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 114 taxa:\t747.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 115 taxa:\t730.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 116 taxa:\t707.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 117 taxa:\t682.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 118 taxa:\t656.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 119 taxa:\t621.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 120 taxa:\t587.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 121 taxa:\t541.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 122 taxa:\t491.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 123 taxa:\t475.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 124 taxa:\t463.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 125 taxa:\t456.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 126 taxa:\t439.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 127 taxa:\t427.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 128 taxa:\t416.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 129 taxa:\t406.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 130 taxa:\t375.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 131 taxa:\t299.0\n",
      "Loci shared by Bacillus_anthracis_GCF_000008165 + 132 taxa:\t129.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/bacillus/initial_intervals/bacillus-to-Bacillus_anthracis_GCF_000008165.sqlite --base-taxon Bacillus_anthracis_GCF_000008165 --output results/phyluce/bacillus/initial_intervals/Bacillus_anthracis_GCF_000008165_+132.bed --specific-counts 132;\n",
      "Counter({'bacillus_cereus_gcf_000290795': 129, 'bacillus_albus_gcf_004116415': 129, 'bacillus_paranthracis_gca_009873395': 129, 'bacillus_pseudomycoides_gcf_002556205': 129, 'bacillus_mycoides_gcf_001757985': 129, 'bacillus_thuringiensis_gcf_002564505': 129, 'bacillus_anthracis_gcf_000697555': 129, 'bacillus_nitratireducens_gcf_007676605': 129, 'bacillus_paranthracis_gcf_009799965': 129, 'bacillus_albus_gcf_003966295': 129, 'bacillus_thuringiensis_gcf_000161735': 129, 'bacillus_cereus_gcf_002530565': 129, 'bacillus_nitratireducens_gcf_007681065': 129, 'bacillus_cereus_gcf_000161315': 129, 'bacillus_nitratireducens_gcf_007676595': 129, 'bacillus_luti_gcf_001884105': 129, 'bacillus_cereus_gcf_001584025': 129, 'bacillus_pseudomycoides_gcf_002555685': 129, 'bacillus_cytotoxicus_gcf_002250945': 129, 'bacillus_cereus_gcf_001583875': 129, 'bacillus_mobilis_gcf_007682195': 129, 'bacillus_mycoides_gcf_001541985': 129, 'bacillus_pacificus_gcf_001884025': 129, 'bacillus_toyonensis_gcf_002551965': 129, 'bacillus_albus_gcf_004116035': 129, 'bacillus_mycoides_gcf_001757915': 129, 'bacillus_paranthracis_gcf_005848985': 129, 'bacillus_pacificus_gca_009884315': 129, 'bacillus_albus_gcf_004153665': 129, 'bacillus_nitratireducens_gcf_007681365': 129, 'bacillus_thuringiensis_gcf_001402735': 129, 'bacillus_luti_gcf_008923665': 129, 'bacillus_cytotoxicus_gcf_001050335': 129, 'bacillus_cereus_gcf_002565765': 129, 'bacillus_wiedmannii_gcf_002556175': 129, 'bacillus_wiedmannii_gcf_001645395': 129, 'bacillus_cytotoxicus_gcf_002250925': 129, 'bacillus_pseudomycoides_gcf_002555615': 129, 'bacillus_albus_gcf_004116025': 129, 'bacillus_wiedmannii_gcf_002569595': 129, 'bacillus_cytotoxicus_gcf_002250885': 129, 'bacillus_mobilis_gcf_007681195': 129, 'bacillus_pacificus_gcf_006349595': 129, 'bacillus_toyonensis_gcf_002555905': 129, 'bacillus_anthracis_gcf_000558965': 129, 'bacillus_anthracis_gcf_000742875': 129, 'bacillus_toyonensis_gcf_002581955': 129, 'bacillus_luti_gcf_009739945': 129, 'bacillus_tropicus_gcf_007682405': 129, 'bacillus_thuringiensis_gcf_001675515': 129, 'bacillus_anthracis_gcf_000783055': 129, 'bacillus_toyonensis_gcf_002550925': 129, 'bacillus_wiedmannii_gcf_002014635': 129, 'bacillus_nitratireducens_gcf_007682355': 129, 'bacillus_mycoides_gcf_000290695': 129, 'bacillus_pseudomycoides_gcf_002555995': 129, 'bacillus_paranthracis_gcf_007682155': 129, 'bacillus_cytotoxicus_gcf_002251005': 129, 'bacillus_anthracis_gcf_001883895': 129, 'bacillus_mycoides_gcf_000517985': 129, 'bacillus_paranthracis_gca_009648955': 129, 'bacillus_mycoides_gcf_001044935': 129, 'bacillus_albus_gcf_004116085': 129, 'bacillus_mycoides_gcf_000018825': 129, 'bacillus_proteolyticus_gcf_001884065': 129, 'bacillus_cytotoxicus_gcf_002250905': 129, 'bacillus_paranthracis_gca_009884335': 129, 'bacillus_toyonensis_gcf_002581325': 129, 'bacillus_mycoides_gcf_001757825': 129, 'bacillus_thuringiensis_gcf_002559825': 129, 'bacillus_mobilis_gcf_007681185': 129, 'bacillus_pseudomycoides_gcf_002550095': 129, 'bacillus_thuringiensis_gcf_002148135': 129, 'bacillus_tropicus_gcf_006457285': 129, 'bacillus_cereus_gcf_000291075': 129, 'bacillus_toyonensis_gcf_002554865': 129, 'bacillus_thuringiensis_gcf_002551895': 129, 'bacillus_toyonensis_gcf_002570155': 129, 'bacillus_pseudomycoides_gcf_000746965': 129, 'bacillus_pseudomycoides_gcf_000161435': 129, 'bacillus_tropicus_gcf_007672275': 129, 'bacillus_tropicus_gcf_007676425': 129, 'bacillus_tropicus_gcf_007682035': 129, 'bacillus_cereus_gcf_002560715': 129, 'bacillus_pseudomycoides_gcf_002557395': 129, 'bacillus_anthracis_gcf_002525695': 129, 'bacillus_cereus_gcf_001683695': 129, 'bacillus_paranthracis_gcf_007682135': 129, 'bacillus_anthracis_gcf_002233635': 129, 'bacillus_cereus_gcf_002552015': 129, 'bacillus_anthracis_gcf_000559005': 129, 'bacillus_albus_gcf_009740005': 129, 'bacillus_tropicus_gcf_001884035': 129, 'bacillus_gaemokensis_gcf_000712615': 129, 'bacillus_gaemokensis_gcf_001590835': 129, 'bacillus_bombysepticus_gcf_000831065': 129, 'bacillus_cytotoxicus_gcf_002251025': 129, 'bacillus_wiedmannii_gcf_002555555': 129, 'bacillus_nitratireducens_gcf_001884135': 129, 'bacillus_toyonensis_gcf_002556675': 129, 'bacillus_paranthracis_gcf_005117095': 129, 'bacillus_albus_gcf_004116065': 129, 'bacillus_paramycoides_gcf_001884235': 129, 'bacillus_toyonensis_gcf_002580355': 129, 'bacillus_mobilis_gcf_003612955': 129, 'bacillus_thuringiensis_gcf_002570925': 129, 'bacillus_cytotoxicus_gcf_000017425': 129, 'bacillus_albus_gcf_001884185': 129, 'bacillus_tropicus_gcf_006349625': 129, 'bacillus_thuringiensis_gcf_000710255': 129, 'bacillus_thuringiensis_gcf_002562665': 129, 'bacillus_wiedmannii_gcf_002556405': 129, 'bacillus_mobilis_gcf_001884045': 129, 'bacillus_wiedmannii_gcf_002555945': 129, 'bacillus_mycoides_gcf_002014505': 129, 'bacillus_cytotoxicus_gcf_002251045': 129, 'bacillus_cytotoxicus_gcf_002250965': 129, 'bacillus_mycoides_gcf_002014695': 129, 'bacillus_toyonensis_gca_009799785': 129, 'bacillus_wiedmannii_gcf_000160955': 129, 'bacillus_wiedmannii_gcf_002555855': 129, 'bacillus_tropicus_gcf_006349645': 129, 'bacillus_paranthracis_gcf_001883995': 129, 'bacillus_pacificus_gcf_003858675': 129, 'bacillus_paranthracis_gcf_009498675': 129, 'bacillus_paranthracis_gcf_004307985': 129, 'bacillus_wiedmannii_gcf_002555735': 129, 'bacillus_pseudomycoides_gcf_000161415': 129, 'bacillus_pseudomycoides_gcf_002569075': 129, 'bacillus_anthracis_gcf_002525705': 129, 'bacillus_paramycoides_gcf_007682005': 129, 'bacillus_manliponensis_gcf_000712595': 129})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 132\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/bacillus/initial_intervals/Bacillus_anthracis_GCF_000008165_+132.bed --twobit results/phyluce/bacillus/cleaned_genomes/Bacillus_anthracis_GCF_000008165_formatted.2bit --buffer-to 160 --output results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132.fasta;\n",
      "Screened 129 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 129.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132.fasta --probe-prefix uce_bacillus_ --design bacillus_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 121\n",
      "Probe Count = 235\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas --query results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes_vself.lastz;\n",
      "Started:  Fri Feb 07, 2020  15:31:37\n",
      "Ended:  Fri Feb 07, 2020  15:31:37\n",
      "Time for execution:  0.00862280130386 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas                        --lastz results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes_vself.lastz                       --probe-prefix=uce_bacillus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 234 fasta sequences.  Filtered 0 duplicates. Kept 235.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/bacillus/validate_intervals/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas --scaffoldlist Bacillus_albus_GCF_001884185 Bacillus_albus_GCF_003966295 Bacillus_albus_GCF_004116025 Bacillus_albus_GCF_004116035 Bacillus_albus_GCF_004116065 Bacillus_albus_GCF_004116085 Bacillus_albus_GCF_004116415 Bacillus_albus_GCF_004153665 Bacillus_albus_GCF_009740005 Bacillus_anthracis_GCF_000558965 Bacillus_anthracis_GCF_000559005 Bacillus_anthracis_GCF_000697555 Bacillus_anthracis_GCF_000742875 Bacillus_anthracis_GCF_000783055 Bacillus_anthracis_GCF_001883895 Bacillus_anthracis_GCF_002233635 Bacillus_anthracis_GCF_002525695 Bacillus_anthracis_GCF_002525705 Bacillus_bombysepticus_GCF_000831065 Bacillus_cereus_GCF_000161315 Bacillus_cereus_GCF_000290795 Bacillus_cereus_GCF_000291075 Bacillus_cereus_GCF_001583875 Bacillus_cereus_GCF_001584025 Bacillus_cereus_GCF_001683695 Bacillus_cereus_GCF_002530565 Bacillus_cereus_GCF_002552015 Bacillus_cereus_GCF_002560715 Bacillus_cereus_GCF_002565765 Bacillus_cytotoxicus_GCF_000017425 Bacillus_cytotoxicus_GCF_001050335 Bacillus_cytotoxicus_GCF_002250885 Bacillus_cytotoxicus_GCF_002250905 Bacillus_cytotoxicus_GCF_002250925 Bacillus_cytotoxicus_GCF_002250945 Bacillus_cytotoxicus_GCF_002250965 Bacillus_cytotoxicus_GCF_002251005 Bacillus_cytotoxicus_GCF_002251025 Bacillus_cytotoxicus_GCF_002251045 Bacillus_gaemokensis_GCF_000712615 Bacillus_gaemokensis_GCF_001590835 Bacillus_luti_GCF_001884105 Bacillus_luti_GCF_008923665 Bacillus_luti_GCF_009739945 Bacillus_manliponensis_GCF_000712595 Bacillus_mobilis_GCF_001884045 Bacillus_mobilis_GCF_003612955 Bacillus_mobilis_GCF_007681185 Bacillus_mobilis_GCF_007681195 Bacillus_mobilis_GCF_007682195 Bacillus_mycoides_GCF_000018825 Bacillus_mycoides_GCF_000290695 Bacillus_mycoides_GCF_000517985 Bacillus_mycoides_GCF_001044935 Bacillus_mycoides_GCF_001541985 Bacillus_mycoides_GCF_001757825 Bacillus_mycoides_GCF_001757915 Bacillus_mycoides_GCF_001757985 Bacillus_mycoides_GCF_002014505 Bacillus_mycoides_GCF_002014695 Bacillus_nitratireducens_GCF_001884135 Bacillus_nitratireducens_GCF_007676595 Bacillus_nitratireducens_GCF_007676605 Bacillus_nitratireducens_GCF_007681065 Bacillus_nitratireducens_GCF_007681365 Bacillus_nitratireducens_GCF_007682355 Bacillus_pacificus_GCA_009884315 Bacillus_pacificus_GCF_001884025 Bacillus_pacificus_GCF_003858675 Bacillus_pacificus_GCF_006349595 Bacillus_paramycoides_GCF_001884235 Bacillus_paramycoides_GCF_007682005 Bacillus_paranthracis_GCA_009648955 Bacillus_paranthracis_GCA_009873395 Bacillus_paranthracis_GCA_009884335 Bacillus_paranthracis_GCF_001883995 Bacillus_paranthracis_GCF_004307985 Bacillus_paranthracis_GCF_005117095 Bacillus_paranthracis_GCF_005848985 Bacillus_paranthracis_GCF_007682135 Bacillus_paranthracis_GCF_007682155 Bacillus_paranthracis_GCF_009498675 Bacillus_paranthracis_GCF_009799965 Bacillus_proteolyticus_GCF_001884065 Bacillus_pseudomycoides_GCF_000161415 Bacillus_pseudomycoides_GCF_000161435 Bacillus_pseudomycoides_GCF_000746965 Bacillus_pseudomycoides_GCF_002550095 Bacillus_pseudomycoides_GCF_002555615 Bacillus_pseudomycoides_GCF_002555685 Bacillus_pseudomycoides_GCF_002555995 Bacillus_pseudomycoides_GCF_002556205 Bacillus_pseudomycoides_GCF_002557395 Bacillus_pseudomycoides_GCF_002569075 Bacillus_thuringiensis_GCF_000161735 Bacillus_thuringiensis_GCF_000710255 Bacillus_thuringiensis_GCF_001402735 Bacillus_thuringiensis_GCF_001675515 Bacillus_thuringiensis_GCF_002148135 Bacillus_thuringiensis_GCF_002551895 Bacillus_thuringiensis_GCF_002559825 Bacillus_thuringiensis_GCF_002562665 Bacillus_thuringiensis_GCF_002564505 Bacillus_thuringiensis_GCF_002570925 Bacillus_toyonensis_GCA_009799785 Bacillus_toyonensis_GCF_002550925 Bacillus_toyonensis_GCF_002551965 Bacillus_toyonensis_GCF_002554865 Bacillus_toyonensis_GCF_002555905 Bacillus_toyonensis_GCF_002556675 Bacillus_toyonensis_GCF_002570155 Bacillus_toyonensis_GCF_002580355 Bacillus_toyonensis_GCF_002581325 Bacillus_toyonensis_GCF_002581955 Bacillus_tropicus_GCF_001884035 Bacillus_tropicus_GCF_006349625 Bacillus_tropicus_GCF_006349645 Bacillus_tropicus_GCF_006457285 Bacillus_tropicus_GCF_007672275 Bacillus_tropicus_GCF_007676425 Bacillus_tropicus_GCF_007682035 Bacillus_tropicus_GCF_007682405 Bacillus_wiedmannii_GCF_000160955 Bacillus_wiedmannii_GCF_001645395 Bacillus_wiedmannii_GCF_002014635 Bacillus_wiedmannii_GCF_002555555 Bacillus_wiedmannii_GCF_002555735 Bacillus_wiedmannii_GCF_002555855 Bacillus_wiedmannii_GCF_002555945 Bacillus_wiedmannii_GCF_002556175 Bacillus_wiedmannii_GCF_002556405 Bacillus_wiedmannii_GCF_002569595 Bacillus_anthracis_GCF_000008165 --genome-base-path results/phyluce/bacillus/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/bacillus/validate_intervals/bacillus-to-Bacillus_anthracis_GCF_000008165.sqlite --output results/phyluce/bacillus/validate_intervals/lastz/;\n",
      "\n",
      "Running against Bacillus_albus_GCF_001884185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8dQzDp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwMDgDF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_001884185.lastz\n",
      "Creating Bacillus_albus_GCF_001884185 table\n",
      "Inserting data to Bacillus_albus_GCF_001884185 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_003966295.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNmu3K6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9Y2_kQ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_003966295.lastz\n",
      "Creating Bacillus_albus_GCF_003966295 table\n",
      "Inserting data to Bacillus_albus_GCF_003966295 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_004116025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZlJVD7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpasnWwO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_004116025.lastz\n",
      "Creating Bacillus_albus_GCF_004116025 table\n",
      "Inserting data to Bacillus_albus_GCF_004116025 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_004116035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpI9nWPv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxSW6qh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_004116035.lastz\n",
      "Creating Bacillus_albus_GCF_004116035 table\n",
      "Inserting data to Bacillus_albus_GCF_004116035 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_004116065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRVeCUk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwFrggZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_004116065.lastz\n",
      "Creating Bacillus_albus_GCF_004116065 table\n",
      "Inserting data to Bacillus_albus_GCF_004116065 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_004116085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpygNChO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpt37YMA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_004116085.lastz\n",
      "Creating Bacillus_albus_GCF_004116085 table\n",
      "Inserting data to Bacillus_albus_GCF_004116085 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_004116415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfb84Gn.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzwphgF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_004116415.lastz\n",
      "Creating Bacillus_albus_GCF_004116415 table\n",
      "Inserting data to Bacillus_albus_GCF_004116415 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_004153665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp12Yq1B.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp55cyeO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_004153665.lastz\n",
      "Creating Bacillus_albus_GCF_004153665 table\n",
      "Inserting data to Bacillus_albus_GCF_004153665 table\n",
      "\n",
      "Running against Bacillus_albus_GCF_009740005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpx3kuUL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAigR5K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_albus_GCF_009740005.lastz\n",
      "Creating Bacillus_albus_GCF_009740005 table\n",
      "Inserting data to Bacillus_albus_GCF_009740005 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_000558965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpg22m85.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuEBdMZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_000558965.lastz\n",
      "Creating Bacillus_anthracis_GCF_000558965 table\n",
      "Inserting data to Bacillus_anthracis_GCF_000558965 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_000559005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsr5ZYM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5hu1U_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_000559005.lastz\n",
      "Creating Bacillus_anthracis_GCF_000559005 table\n",
      "Inserting data to Bacillus_anthracis_GCF_000559005 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_000697555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpeV6bfv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpM4hrJw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_000697555.lastz\n",
      "Creating Bacillus_anthracis_GCF_000697555 table\n",
      "Inserting data to Bacillus_anthracis_GCF_000697555 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_000742875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpl7qTt2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3kRkF7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_000742875.lastz\n",
      "Creating Bacillus_anthracis_GCF_000742875 table\n",
      "Inserting data to Bacillus_anthracis_GCF_000742875 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_000783055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpq1N0Ts.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptHbJfO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_000783055.lastz\n",
      "Creating Bacillus_anthracis_GCF_000783055 table\n",
      "Inserting data to Bacillus_anthracis_GCF_000783055 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_001883895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2f8oni.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpc3BLS1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_001883895.lastz\n",
      "Creating Bacillus_anthracis_GCF_001883895 table\n",
      "Inserting data to Bacillus_anthracis_GCF_001883895 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_002233635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7KkwqN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1f1Ifo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_002233635.lastz\n",
      "Creating Bacillus_anthracis_GCF_002233635 table\n",
      "Inserting data to Bacillus_anthracis_GCF_002233635 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_002525695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEJmTMT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeeexm8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_002525695.lastz\n",
      "Creating Bacillus_anthracis_GCF_002525695 table\n",
      "Inserting data to Bacillus_anthracis_GCF_002525695 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_002525705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8Y08q9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmph7SaWn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_002525705.lastz\n",
      "Creating Bacillus_anthracis_GCF_002525705 table\n",
      "Inserting data to Bacillus_anthracis_GCF_002525705 table\n",
      "\n",
      "Running against Bacillus_bombysepticus_GCF_000831065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsCEnTd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoLYxIo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_bombysepticus_GCF_000831065.lastz\n",
      "Creating Bacillus_bombysepticus_GCF_000831065 table\n",
      "Inserting data to Bacillus_bombysepticus_GCF_000831065 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_000161315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpujJOOV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_jXgJ2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_000161315.lastz\n",
      "Creating Bacillus_cereus_GCF_000161315 table\n",
      "Inserting data to Bacillus_cereus_GCF_000161315 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_000290795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5lBnyE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGp_hs7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_000290795.lastz\n",
      "Creating Bacillus_cereus_GCF_000290795 table\n",
      "Inserting data to Bacillus_cereus_GCF_000290795 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_000291075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWUlXm5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpaEM49w.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_000291075.lastz\n",
      "Creating Bacillus_cereus_GCF_000291075 table\n",
      "Inserting data to Bacillus_cereus_GCF_000291075 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_001583875.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5Sp_Hu.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDTSQBX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_001583875.lastz\n",
      "Creating Bacillus_cereus_GCF_001583875 table\n",
      "Inserting data to Bacillus_cereus_GCF_001583875 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_001584025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBM3Q1T.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVYW2AK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_001584025.lastz\n",
      "Creating Bacillus_cereus_GCF_001584025 table\n",
      "Inserting data to Bacillus_cereus_GCF_001584025 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_001683695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFweMUH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpw2IMG0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_001683695.lastz\n",
      "Creating Bacillus_cereus_GCF_001683695 table\n",
      "Inserting data to Bacillus_cereus_GCF_001683695 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_002530565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkX136P.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBZkLM1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_002530565.lastz\n",
      "Creating Bacillus_cereus_GCF_002530565 table\n",
      "Inserting data to Bacillus_cereus_GCF_002530565 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_002552015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnu4r7Q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTsgLV6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_002552015.lastz\n",
      "Creating Bacillus_cereus_GCF_002552015 table\n",
      "Inserting data to Bacillus_cereus_GCF_002552015 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_002560715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjdw4g9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4EDwxE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_002560715.lastz\n",
      "Creating Bacillus_cereus_GCF_002560715 table\n",
      "Inserting data to Bacillus_cereus_GCF_002560715 table\n",
      "\n",
      "Running against Bacillus_cereus_GCF_002565765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRm6jXB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpf_fbPC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cereus_GCF_002565765.lastz\n",
      "Creating Bacillus_cereus_GCF_002565765 table\n",
      "Inserting data to Bacillus_cereus_GCF_002565765 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_000017425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwrBMVC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvaE3ht.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_000017425.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_000017425 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_000017425 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_001050335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRoPhlc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJIEF1D.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_001050335.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_001050335 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_001050335 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002250885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpSPpH16.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcek3Jj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002250885.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002250885 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002250885 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002250905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptA0ooZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVkZqIS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002250905.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002250905 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002250905 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002250925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp39X797.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5w7lVR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002250925.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002250925 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002250925 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002250945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTwjiWc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOXp4oq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002250945.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002250945 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002250945 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002250965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbwyvbY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprjgFax.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002250965.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002250965 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002250965 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002251005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNYMhGr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkDkC4q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002251005.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002251005 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002251005 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002251025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDhJ1UL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnx12b2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002251025.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002251025 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002251025 table\n",
      "\n",
      "Running against Bacillus_cytotoxicus_GCF_002251045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJ3RO82.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmponypy_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_cytotoxicus_GCF_002251045.lastz\n",
      "Creating Bacillus_cytotoxicus_GCF_002251045 table\n",
      "Inserting data to Bacillus_cytotoxicus_GCF_002251045 table\n",
      "\n",
      "Running against Bacillus_gaemokensis_GCF_000712615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpm2VHEe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7Q2jUC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_gaemokensis_GCF_000712615.lastz\n",
      "Creating Bacillus_gaemokensis_GCF_000712615 table\n",
      "Inserting data to Bacillus_gaemokensis_GCF_000712615 table\n",
      "\n",
      "Running against Bacillus_gaemokensis_GCF_001590835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVhRHSY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBfAhis.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_gaemokensis_GCF_001590835.lastz\n",
      "Creating Bacillus_gaemokensis_GCF_001590835 table\n",
      "Inserting data to Bacillus_gaemokensis_GCF_001590835 table\n",
      "\n",
      "Running against Bacillus_luti_GCF_001884105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfNZTwP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeFLnx3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_luti_GCF_001884105.lastz\n",
      "Creating Bacillus_luti_GCF_001884105 table\n",
      "Inserting data to Bacillus_luti_GCF_001884105 table\n",
      "\n",
      "Running against Bacillus_luti_GCF_008923665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmppUDBU3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbHzrf0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_luti_GCF_008923665.lastz\n",
      "Creating Bacillus_luti_GCF_008923665 table\n",
      "Inserting data to Bacillus_luti_GCF_008923665 table\n",
      "\n",
      "Running against Bacillus_luti_GCF_009739945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGm0zTH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2bVja8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_luti_GCF_009739945.lastz\n",
      "Creating Bacillus_luti_GCF_009739945 table\n",
      "Inserting data to Bacillus_luti_GCF_009739945 table\n",
      "\n",
      "Running against Bacillus_manliponensis_GCF_000712595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbiktEm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpO7Irlu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_manliponensis_GCF_000712595.lastz\n",
      "Creating Bacillus_manliponensis_GCF_000712595 table\n",
      "Inserting data to Bacillus_manliponensis_GCF_000712595 table\n",
      "\n",
      "Running against Bacillus_mobilis_GCF_001884045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmmrObT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplDBdGu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mobilis_GCF_001884045.lastz\n",
      "Creating Bacillus_mobilis_GCF_001884045 table\n",
      "Inserting data to Bacillus_mobilis_GCF_001884045 table\n",
      "\n",
      "Running against Bacillus_mobilis_GCF_003612955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTCpuM9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnN8prp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mobilis_GCF_003612955.lastz\n",
      "Creating Bacillus_mobilis_GCF_003612955 table\n",
      "Inserting data to Bacillus_mobilis_GCF_003612955 table\n",
      "\n",
      "Running against Bacillus_mobilis_GCF_007681185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvuAdBc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptc3qag.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mobilis_GCF_007681185.lastz\n",
      "Creating Bacillus_mobilis_GCF_007681185 table\n",
      "Inserting data to Bacillus_mobilis_GCF_007681185 table\n",
      "\n",
      "Running against Bacillus_mobilis_GCF_007681195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmhnwBR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprikbwV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mobilis_GCF_007681195.lastz\n",
      "Creating Bacillus_mobilis_GCF_007681195 table\n",
      "Inserting data to Bacillus_mobilis_GCF_007681195 table\n",
      "\n",
      "Running against Bacillus_mobilis_GCF_007682195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpS2RLsQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUVtBwd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mobilis_GCF_007682195.lastz\n",
      "Creating Bacillus_mobilis_GCF_007682195 table\n",
      "Inserting data to Bacillus_mobilis_GCF_007682195 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_000018825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpibEnNh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0igzau.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_000018825.lastz\n",
      "Creating Bacillus_mycoides_GCF_000018825 table\n",
      "Inserting data to Bacillus_mycoides_GCF_000018825 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Bacillus_mycoides_GCF_000290695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKoaAKO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpu0EuU0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_000290695.lastz\n",
      "Creating Bacillus_mycoides_GCF_000290695 table\n",
      "Inserting data to Bacillus_mycoides_GCF_000290695 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_000517985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqRX6Sv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbhIe7g.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_000517985.lastz\n",
      "Creating Bacillus_mycoides_GCF_000517985 table\n",
      "Inserting data to Bacillus_mycoides_GCF_000517985 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_001044935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpS0UskV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYnf8CJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_001044935.lastz\n",
      "Creating Bacillus_mycoides_GCF_001044935 table\n",
      "Inserting data to Bacillus_mycoides_GCF_001044935 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_001541985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdXFQsD.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnB6dOJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_001541985.lastz\n",
      "Creating Bacillus_mycoides_GCF_001541985 table\n",
      "Inserting data to Bacillus_mycoides_GCF_001541985 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_001757825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVx3AkC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQNAWdF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_001757825.lastz\n",
      "Creating Bacillus_mycoides_GCF_001757825 table\n",
      "Inserting data to Bacillus_mycoides_GCF_001757825 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_001757915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjy29W3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpixbB8s.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_001757915.lastz\n",
      "Creating Bacillus_mycoides_GCF_001757915 table\n",
      "Inserting data to Bacillus_mycoides_GCF_001757915 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_001757985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEs8Obc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9vDdfM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_001757985.lastz\n",
      "Creating Bacillus_mycoides_GCF_001757985 table\n",
      "Inserting data to Bacillus_mycoides_GCF_001757985 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_002014505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0Yu5Mg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZ6_O88.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_002014505.lastz\n",
      "Creating Bacillus_mycoides_GCF_002014505 table\n",
      "Inserting data to Bacillus_mycoides_GCF_002014505 table\n",
      "\n",
      "Running against Bacillus_mycoides_GCF_002014695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaNAlQN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpjwD6f8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_mycoides_GCF_002014695.lastz\n",
      "Creating Bacillus_mycoides_GCF_002014695 table\n",
      "Inserting data to Bacillus_mycoides_GCF_002014695 table\n",
      "\n",
      "Running against Bacillus_nitratireducens_GCF_001884135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzgFx2p.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpc4ykz6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_nitratireducens_GCF_001884135.lastz\n",
      "Creating Bacillus_nitratireducens_GCF_001884135 table\n",
      "Inserting data to Bacillus_nitratireducens_GCF_001884135 table\n",
      "\n",
      "Running against Bacillus_nitratireducens_GCF_007676595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqXaTLh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpeT2mn1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_nitratireducens_GCF_007676595.lastz\n",
      "Creating Bacillus_nitratireducens_GCF_007676595 table\n",
      "Inserting data to Bacillus_nitratireducens_GCF_007676595 table\n",
      "\n",
      "Running against Bacillus_nitratireducens_GCF_007676605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmcDB2i.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8nVit0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_nitratireducens_GCF_007676605.lastz\n",
      "Creating Bacillus_nitratireducens_GCF_007676605 table\n",
      "Inserting data to Bacillus_nitratireducens_GCF_007676605 table\n",
      "\n",
      "Running against Bacillus_nitratireducens_GCF_007681065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPi9BqH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpw01AOb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_nitratireducens_GCF_007681065.lastz\n",
      "Creating Bacillus_nitratireducens_GCF_007681065 table\n",
      "Inserting data to Bacillus_nitratireducens_GCF_007681065 table\n",
      "\n",
      "Running against Bacillus_nitratireducens_GCF_007681365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpT2eStm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRGylA_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_nitratireducens_GCF_007681365.lastz\n",
      "Creating Bacillus_nitratireducens_GCF_007681365 table\n",
      "Inserting data to Bacillus_nitratireducens_GCF_007681365 table\n",
      "\n",
      "Running against Bacillus_nitratireducens_GCF_007682355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp42bo9r.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp896Tub.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_nitratireducens_GCF_007682355.lastz\n",
      "Creating Bacillus_nitratireducens_GCF_007682355 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Bacillus_nitratireducens_GCF_007682355 table\n",
      "\n",
      "Running against Bacillus_pacificus_GCA_009884315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0zY48V.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5cvgIv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pacificus_GCA_009884315.lastz\n",
      "Creating Bacillus_pacificus_GCA_009884315 table\n",
      "Inserting data to Bacillus_pacificus_GCA_009884315 table\n",
      "\n",
      "Running against Bacillus_pacificus_GCF_001884025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4fTUjG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWae4Dw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pacificus_GCF_001884025.lastz\n",
      "Creating Bacillus_pacificus_GCF_001884025 table\n",
      "Inserting data to Bacillus_pacificus_GCF_001884025 table\n",
      "\n",
      "Running against Bacillus_pacificus_GCF_003858675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpinUtiT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpt5zx8X.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pacificus_GCF_003858675.lastz\n",
      "Creating Bacillus_pacificus_GCF_003858675 table\n",
      "Inserting data to Bacillus_pacificus_GCF_003858675 table\n",
      "\n",
      "Running against Bacillus_pacificus_GCF_006349595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkgYIFU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTUh2J_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pacificus_GCF_006349595.lastz\n",
      "Creating Bacillus_pacificus_GCF_006349595 table\n",
      "Inserting data to Bacillus_pacificus_GCF_006349595 table\n",
      "\n",
      "Running against Bacillus_paramycoides_GCF_001884235.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFYiDBq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpblmOMC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paramycoides_GCF_001884235.lastz\n",
      "Creating Bacillus_paramycoides_GCF_001884235 table\n",
      "Inserting data to Bacillus_paramycoides_GCF_001884235 table\n",
      "\n",
      "Running against Bacillus_paramycoides_GCF_007682005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpp_JlTF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSBQNga.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paramycoides_GCF_007682005.lastz\n",
      "Creating Bacillus_paramycoides_GCF_007682005 table\n",
      "Inserting data to Bacillus_paramycoides_GCF_007682005 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCA_009648955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjDIbRp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFX2aQI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCA_009648955.lastz\n",
      "Creating Bacillus_paranthracis_GCA_009648955 table\n",
      "Inserting data to Bacillus_paranthracis_GCA_009648955 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCA_009873395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpESEkFv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplZimC3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCA_009873395.lastz\n",
      "Creating Bacillus_paranthracis_GCA_009873395 table\n",
      "Inserting data to Bacillus_paranthracis_GCA_009873395 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCA_009884335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphgQioY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplgfxPI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCA_009884335.lastz\n",
      "Creating Bacillus_paranthracis_GCA_009884335 table\n",
      "Inserting data to Bacillus_paranthracis_GCA_009884335 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_001883995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJvTpvE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSIhToC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_001883995.lastz\n",
      "Creating Bacillus_paranthracis_GCF_001883995 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_001883995 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_004307985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpu8EswK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB4EmuW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_004307985.lastz\n",
      "Creating Bacillus_paranthracis_GCF_004307985 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_004307985 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_005117095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphye4bp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcZn95d.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_005117095.lastz\n",
      "Creating Bacillus_paranthracis_GCF_005117095 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_005117095 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_005848985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplu7dRb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8zg3LL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_005848985.lastz\n",
      "Creating Bacillus_paranthracis_GCF_005848985 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_005848985 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_007682135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMKfr54.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIvHCu2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_007682135.lastz\n",
      "Creating Bacillus_paranthracis_GCF_007682135 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_007682135 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_007682155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWIeqZj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpA2VkbB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_007682155.lastz\n",
      "Creating Bacillus_paranthracis_GCF_007682155 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Bacillus_paranthracis_GCF_007682155 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_009498675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5PXdbB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTnO4Ho.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_009498675.lastz\n",
      "Creating Bacillus_paranthracis_GCF_009498675 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_009498675 table\n",
      "\n",
      "Running against Bacillus_paranthracis_GCF_009799965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPFQRN7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9sE_VY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_paranthracis_GCF_009799965.lastz\n",
      "Creating Bacillus_paranthracis_GCF_009799965 table\n",
      "Inserting data to Bacillus_paranthracis_GCF_009799965 table\n",
      "\n",
      "Running against Bacillus_proteolyticus_GCF_001884065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprEzG1B.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvoaILR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_proteolyticus_GCF_001884065.lastz\n",
      "Creating Bacillus_proteolyticus_GCF_001884065 table\n",
      "Inserting data to Bacillus_proteolyticus_GCF_001884065 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_000161415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPuzQD3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptXPf9d.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_000161415.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_000161415 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_000161415 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_000161435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5rceBQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAV2XfM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_000161435.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_000161435 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_000161435 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_000746965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmps9QkvZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprr9Xza.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_000746965.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_000746965 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_000746965 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002550095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyFk9HS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptoVd39.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002550095.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002550095 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002550095 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002555615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptDjevz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpF_T7vj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002555615.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002555615 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002555615 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002555685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpY6sVMg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7wu5Wk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002555685.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002555685 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002555685 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002555995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpoVmDRp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTtIirS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002555995.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002555995 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002555995 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002556205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsUgLGJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZckrjz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002556205.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002556205 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002556205 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002557395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGsUHF9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqlMB9S.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002557395.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002557395 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002557395 table\n",
      "\n",
      "Running against Bacillus_pseudomycoides_GCF_002569075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGoIFlc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPKtpHa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_pseudomycoides_GCF_002569075.lastz\n",
      "Creating Bacillus_pseudomycoides_GCF_002569075 table\n",
      "Inserting data to Bacillus_pseudomycoides_GCF_002569075 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_000161735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5NMOaV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpW4xZnA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_000161735.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_000161735 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_000161735 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_000710255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvIqjmY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmiuwC1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_000710255.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_000710255 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_000710255 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_001402735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxYQ2Qj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOUvaXY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_001402735.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_001402735 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_001402735 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_001675515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9Dbo1n.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdd3z7z.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_001675515.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_001675515 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_001675515 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_002148135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIwftfL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSzcVAd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_002148135.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_002148135 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_002148135 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_002551895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpB6me7X.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPMEpXW.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_002551895.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_002551895 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_002551895 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_002559825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTWWPFv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxBO69d.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_002559825.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_002559825 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_002559825 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_002562665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKs0m7q.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKaKKRX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_002562665.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_002562665 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_002562665 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_002564505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmDD9GY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVnvkfZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_002564505.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_002564505 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_002564505 table\n",
      "\n",
      "Running against Bacillus_thuringiensis_GCF_002570925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOwQnJQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp66SBE5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_thuringiensis_GCF_002570925.lastz\n",
      "Creating Bacillus_thuringiensis_GCF_002570925 table\n",
      "Inserting data to Bacillus_thuringiensis_GCF_002570925 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCA_009799785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpasIeWe.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcmewEO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCA_009799785.lastz\n",
      "Creating Bacillus_toyonensis_GCA_009799785 table\n",
      "Inserting data to Bacillus_toyonensis_GCA_009799785 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002550925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHAgj7v.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpanRfTU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002550925.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002550925 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002550925 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002551965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZiJMn8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIycxnv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002551965.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002551965 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002551965 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002554865.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6mPgmi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCTsvN4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002554865.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002554865 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002554865 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002555905.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpp4KBBP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpu67uCf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002555905.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002555905 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002555905 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002556675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOt6oad.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEhxRb3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002556675.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002556675 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Bacillus_toyonensis_GCF_002556675 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002570155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpThaBk3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnWfwG9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002570155.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002570155 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002570155 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002580355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzsEMRg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplJTiRv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002580355.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002580355 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002580355 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002581325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp31JThy.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvKD85I.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002581325.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002581325 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002581325 table\n",
      "\n",
      "Running against Bacillus_toyonensis_GCF_002581955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMtyAhr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVxbQnL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_toyonensis_GCF_002581955.lastz\n",
      "Creating Bacillus_toyonensis_GCF_002581955 table\n",
      "Inserting data to Bacillus_toyonensis_GCF_002581955 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_001884035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmppN5wa6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppooJeA.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_001884035.lastz\n",
      "Creating Bacillus_tropicus_GCF_001884035 table\n",
      "Inserting data to Bacillus_tropicus_GCF_001884035 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_006349625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJNPv9i.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpu1Tt2o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_006349625.lastz\n",
      "Creating Bacillus_tropicus_GCF_006349625 table\n",
      "Inserting data to Bacillus_tropicus_GCF_006349625 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_006349645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4K2WZ2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpMKYdrO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_006349645.lastz\n",
      "Creating Bacillus_tropicus_GCF_006349645 table\n",
      "Inserting data to Bacillus_tropicus_GCF_006349645 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_006457285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7mNUZb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWKnj3X.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_006457285.lastz\n",
      "Creating Bacillus_tropicus_GCF_006457285 table\n",
      "Inserting data to Bacillus_tropicus_GCF_006457285 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_007672275.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3b9T_z.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVMj_pL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_007672275.lastz\n",
      "Creating Bacillus_tropicus_GCF_007672275 table\n",
      "Inserting data to Bacillus_tropicus_GCF_007672275 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_007676425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpslMtfO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnkCd1Y.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_007676425.lastz\n",
      "Creating Bacillus_tropicus_GCF_007676425 table\n",
      "Inserting data to Bacillus_tropicus_GCF_007676425 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_007682035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIzj_Ih.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0OjL11.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_007682035.lastz\n",
      "Creating Bacillus_tropicus_GCF_007682035 table\n",
      "Inserting data to Bacillus_tropicus_GCF_007682035 table\n",
      "\n",
      "Running against Bacillus_tropicus_GCF_007682405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkN4mWr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZWT2Gh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_tropicus_GCF_007682405.lastz\n",
      "Creating Bacillus_tropicus_GCF_007682405 table\n",
      "Inserting data to Bacillus_tropicus_GCF_007682405 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_000160955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpLlQkSB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0w5N5W.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_000160955.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_000160955 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_000160955 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_001645395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJJeoSW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr0AT9B.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_001645395.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_001645395 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_001645395 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002014635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6eSDxO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpL76d79.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002014635.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002014635 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002014635 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002555555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRWHajo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkoRYj1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002555555.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002555555 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002555555 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002555735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp95UrNL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2K6HHE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002555735.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002555735 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002555735 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002555855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWwgYXl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNwbSxj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002555855.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002555855 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002555855 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002555945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpSpJ6CR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0QfymD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002555945.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002555945 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002555945 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002556175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZ8MeYH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdbprtU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002556175.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002556175 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002556175 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002556405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBQCfgF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdHmZ7F.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002556405.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002556405 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002556405 table\n",
      "\n",
      "Running against Bacillus_wiedmannii_GCF_002569595.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpoEhHMR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXPITUS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_wiedmannii_GCF_002569595.lastz\n",
      "Creating Bacillus_wiedmannii_GCF_002569595 table\n",
      "Inserting data to Bacillus_wiedmannii_GCF_002569595 table\n",
      "\n",
      "Running against Bacillus_anthracis_GCF_000008165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp26RZx5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLim4Aa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bacillus/validate_intervals/lastz/Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_Bacillus_anthracis_GCF_000008165.lastz\n",
      "Creating Bacillus_anthracis_GCF_000008165 table\n",
      "Inserting data to Bacillus_anthracis_GCF_000008165 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/bacillus/extract_probes_from_group/bacillus_genome.conf --lastz results/phyluce/bacillus/validate_intervals/lastz --probes 120 --probe-prefix uce_bacillus_ --name-pattern \"Bacillus_anthracis_GCF_000008165_+132_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/bacillus/extract_probes_from_group/probe_fasta;\n",
      "2020-02-07 15:38:53,409 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-07 15:38:53,419 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_001884185 genome --------\n",
      "2020-02-07 15:38:53,419 - Phyluce - INFO - Reading Bacillus_albus_GCF_001884185 genome\n",
      "2020-02-07 15:38:53,957 - Phyluce - INFO - Bacillus_albus_GCF_001884185: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:38:53,958 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_003966295 genome --------\n",
      "2020-02-07 15:38:53,962 - Phyluce - INFO - Reading Bacillus_albus_GCF_003966295 genome\n",
      "2020-02-07 15:38:54,428 - Phyluce - INFO - Bacillus_albus_GCF_003966295: 120 uces, 0 dupes, 120 non-dupes, 7 orient drop, 5 length drop, 108 written\n",
      "2020-02-07 15:38:54,428 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_004116025 genome --------\n",
      "2020-02-07 15:38:54,437 - Phyluce - INFO - Reading Bacillus_albus_GCF_004116025 genome\n",
      "2020-02-07 15:38:54,873 - Phyluce - INFO - Bacillus_albus_GCF_004116025: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 2 length drop, 110 written\n",
      "2020-02-07 15:38:54,873 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_004116035 genome --------\n",
      "2020-02-07 15:38:54,874 - Phyluce - INFO - Reading Bacillus_albus_GCF_004116035 genome\n",
      "2020-02-07 15:38:55,307 - Phyluce - INFO - Bacillus_albus_GCF_004116035: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 2 length drop, 110 written\n",
      "2020-02-07 15:38:55,307 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_004116065 genome --------\n",
      "2020-02-07 15:38:55,308 - Phyluce - INFO - Reading Bacillus_albus_GCF_004116065 genome\n",
      "2020-02-07 15:38:55,753 - Phyluce - INFO - Bacillus_albus_GCF_004116065: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 2 length drop, 110 written\n",
      "2020-02-07 15:38:55,753 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_004116085 genome --------\n",
      "2020-02-07 15:38:55,754 - Phyluce - INFO - Reading Bacillus_albus_GCF_004116085 genome\n",
      "2020-02-07 15:38:56,187 - Phyluce - INFO - Bacillus_albus_GCF_004116085: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:38:56,187 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_004116415 genome --------\n",
      "2020-02-07 15:38:56,188 - Phyluce - INFO - Reading Bacillus_albus_GCF_004116415 genome\n",
      "2020-02-07 15:38:56,618 - Phyluce - INFO - Bacillus_albus_GCF_004116415: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 2 length drop, 110 written\n",
      "2020-02-07 15:38:56,619 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_004153665 genome --------\n",
      "2020-02-07 15:38:56,619 - Phyluce - INFO - Reading Bacillus_albus_GCF_004153665 genome\n",
      "2020-02-07 15:38:57,051 - Phyluce - INFO - Bacillus_albus_GCF_004153665: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 0 length drop, 112 written\n",
      "2020-02-07 15:38:57,051 - Phyluce - INFO - --------- Working on Bacillus_albus_GCF_009740005 genome --------\n",
      "2020-02-07 15:38:57,060 - Phyluce - INFO - Reading Bacillus_albus_GCF_009740005 genome\n",
      "2020-02-07 15:38:57,572 - Phyluce - INFO - Bacillus_albus_GCF_009740005: 121 uces, 2 dupes, 119 non-dupes, 5 orient drop, 4 length drop, 110 written\n",
      "2020-02-07 15:38:57,573 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_000558965 genome ------\n",
      "2020-02-07 15:38:57,573 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_000558965 genome\n",
      "2020-02-07 15:38:58,058 - Phyluce - INFO - Bacillus_anthracis_GCF_000558965: 121 uces, 0 dupes, 121 non-dupes, 8 orient drop, 6 length drop, 107 written\n",
      "2020-02-07 15:38:58,058 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_000559005 genome ------\n",
      "2020-02-07 15:38:58,062 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_000559005 genome\n",
      "2020-02-07 15:38:58,559 - Phyluce - INFO - Bacillus_anthracis_GCF_000559005: 121 uces, 0 dupes, 121 non-dupes, 8 orient drop, 6 length drop, 107 written\n",
      "2020-02-07 15:38:58,559 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_000697555 genome ------\n",
      "2020-02-07 15:38:58,568 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_000697555 genome\n",
      "2020-02-07 15:38:59,001 - Phyluce - INFO - Bacillus_anthracis_GCF_000697555: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:38:59,001 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_000742875 genome ------\n",
      "2020-02-07 15:38:59,002 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_000742875 genome\n",
      "2020-02-07 15:38:59,484 - Phyluce - INFO - Bacillus_anthracis_GCF_000742875: 121 uces, 0 dupes, 121 non-dupes, 8 orient drop, 6 length drop, 107 written\n",
      "2020-02-07 15:38:59,484 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_000783055 genome ------\n",
      "2020-02-07 15:38:59,485 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_000783055 genome\n",
      "2020-02-07 15:38:59,907 - Phyluce - INFO - Bacillus_anthracis_GCF_000783055: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 3 length drop, 109 written\n",
      "2020-02-07 15:38:59,907 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_001883895 genome ------\n",
      "2020-02-07 15:38:59,907 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_001883895 genome\n",
      "2020-02-07 15:39:00,339 - Phyluce - INFO - Bacillus_anthracis_GCF_001883895: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 0 length drop, 108 written\n",
      "2020-02-07 15:39:00,339 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_002233635 genome ------\n",
      "2020-02-07 15:39:00,340 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_002233635 genome\n",
      "2020-02-07 15:39:00,755 - Phyluce - INFO - Bacillus_anthracis_GCF_002233635: 120 uces, 10 dupes, 110 non-dupes, 0 orient drop, 2 length drop, 108 written\n",
      "2020-02-07 15:39:00,755 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_002525695 genome ------\n",
      "2020-02-07 15:39:00,764 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_002525695 genome\n",
      "2020-02-07 15:39:01,217 - Phyluce - INFO - Bacillus_anthracis_GCF_002525695: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 3 length drop, 108 written\n",
      "2020-02-07 15:39:01,217 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_002525705 genome ------\n",
      "2020-02-07 15:39:01,222 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_002525705 genome\n",
      "2020-02-07 15:39:01,657 - Phyluce - INFO - Bacillus_anthracis_GCF_002525705: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 3 length drop, 108 written\n",
      "2020-02-07 15:39:01,657 - Phyluce - INFO - ----- Working on Bacillus_bombysepticus_GCF_000831065 genome ----\n",
      "2020-02-07 15:39:01,664 - Phyluce - INFO - Reading Bacillus_bombysepticus_GCF_000831065 genome\n",
      "2020-02-07 15:39:02,167 - Phyluce - INFO - Bacillus_bombysepticus_GCF_000831065: 121 uces, 1 dupes, 120 non-dupes, 6 orient drop, 4 length drop, 110 written\n",
      "2020-02-07 15:39:02,167 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_000161315 genome --------\n",
      "2020-02-07 15:39:02,168 - Phyluce - INFO - Reading Bacillus_cereus_GCF_000161315 genome\n",
      "2020-02-07 15:39:02,605 - Phyluce - INFO - Bacillus_cereus_GCF_000161315: 121 uces, 0 dupes, 121 non-dupes, 6 orient drop, 4 length drop, 111 written\n",
      "2020-02-07 15:39:02,605 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_000290795 genome --------\n",
      "2020-02-07 15:39:02,623 - Phyluce - INFO - Reading Bacillus_cereus_GCF_000290795 genome\n",
      "2020-02-07 15:39:03,080 - Phyluce - INFO - Bacillus_cereus_GCF_000290795: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:03,080 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_000291075 genome --------\n",
      "2020-02-07 15:39:03,091 - Phyluce - INFO - Reading Bacillus_cereus_GCF_000291075 genome\n",
      "2020-02-07 15:39:03,574 - Phyluce - INFO - Bacillus_cereus_GCF_000291075: 121 uces, 8 dupes, 113 non-dupes, 2 orient drop, 5 length drop, 106 written\n",
      "2020-02-07 15:39:03,575 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_001583875 genome --------\n",
      "2020-02-07 15:39:03,575 - Phyluce - INFO - Reading Bacillus_cereus_GCF_001583875 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 15:39:04,053 - Phyluce - INFO - Bacillus_cereus_GCF_001583875: 121 uces, 8 dupes, 113 non-dupes, 0 orient drop, 1 length drop, 112 written\n",
      "2020-02-07 15:39:04,054 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_001584025 genome --------\n",
      "2020-02-07 15:39:04,087 - Phyluce - INFO - Reading Bacillus_cereus_GCF_001584025 genome\n",
      "2020-02-07 15:39:04,580 - Phyluce - INFO - Bacillus_cereus_GCF_001584025: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:04,580 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_001683695 genome --------\n",
      "2020-02-07 15:39:04,610 - Phyluce - INFO - Reading Bacillus_cereus_GCF_001683695 genome\n",
      "2020-02-07 15:39:05,080 - Phyluce - INFO - Bacillus_cereus_GCF_001683695: 120 uces, 11 dupes, 109 non-dupes, 0 orient drop, 0 length drop, 109 written\n",
      "2020-02-07 15:39:05,080 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_002530565 genome --------\n",
      "2020-02-07 15:39:05,081 - Phyluce - INFO - Reading Bacillus_cereus_GCF_002530565 genome\n",
      "2020-02-07 15:39:05,509 - Phyluce - INFO - Bacillus_cereus_GCF_002530565: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:05,510 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_002552015 genome --------\n",
      "2020-02-07 15:39:05,510 - Phyluce - INFO - Reading Bacillus_cereus_GCF_002552015 genome\n",
      "2020-02-07 15:39:05,950 - Phyluce - INFO - Bacillus_cereus_GCF_002552015: 121 uces, 8 dupes, 113 non-dupes, 0 orient drop, 1 length drop, 112 written\n",
      "2020-02-07 15:39:05,950 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_002560715 genome --------\n",
      "2020-02-07 15:39:05,958 - Phyluce - INFO - Reading Bacillus_cereus_GCF_002560715 genome\n",
      "2020-02-07 15:39:06,402 - Phyluce - INFO - Bacillus_cereus_GCF_002560715: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 0 length drop, 112 written\n",
      "2020-02-07 15:39:06,402 - Phyluce - INFO - -------- Working on Bacillus_cereus_GCF_002565765 genome --------\n",
      "2020-02-07 15:39:06,403 - Phyluce - INFO - Reading Bacillus_cereus_GCF_002565765 genome\n",
      "2020-02-07 15:39:06,853 - Phyluce - INFO - Bacillus_cereus_GCF_002565765: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 3 length drop, 109 written\n",
      "2020-02-07 15:39:06,853 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_000017425 genome -----\n",
      "2020-02-07 15:39:06,854 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_000017425 genome\n",
      "2020-02-07 15:39:07,326 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_000017425: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:07,326 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_001050335 genome -----\n",
      "2020-02-07 15:39:07,327 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_001050335 genome\n",
      "2020-02-07 15:39:07,752 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_001050335: 119 uces, 8 dupes, 111 non-dupes, 0 orient drop, 3 length drop, 108 written\n",
      "2020-02-07 15:39:07,752 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002250885 genome -----\n",
      "2020-02-07 15:39:07,759 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002250885 genome\n",
      "2020-02-07 15:39:08,244 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002250885: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:08,244 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002250905 genome -----\n",
      "2020-02-07 15:39:08,245 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002250905 genome\n",
      "2020-02-07 15:39:08,718 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002250905: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:08,718 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002250925 genome -----\n",
      "2020-02-07 15:39:08,719 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002250925 genome\n",
      "2020-02-07 15:39:09,190 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002250925: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:09,190 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002250945 genome -----\n",
      "2020-02-07 15:39:09,194 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002250945 genome\n",
      "2020-02-07 15:39:09,676 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002250945: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:09,676 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002250965 genome -----\n",
      "2020-02-07 15:39:09,685 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002250965 genome\n",
      "2020-02-07 15:39:10,251 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002250965: 119 uces, 0 dupes, 119 non-dupes, 6 orient drop, 5 length drop, 108 written\n",
      "2020-02-07 15:39:10,252 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002251005 genome -----\n",
      "2020-02-07 15:39:10,252 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002251005 genome\n",
      "2020-02-07 15:39:10,733 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002251005: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:10,734 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002251025 genome -----\n",
      "2020-02-07 15:39:10,734 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002251025 genome\n",
      "2020-02-07 15:39:11,212 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002251025: 119 uces, 0 dupes, 119 non-dupes, 6 orient drop, 5 length drop, 108 written\n",
      "2020-02-07 15:39:11,212 - Phyluce - INFO - ------ Working on Bacillus_cytotoxicus_GCF_002251045 genome -----\n",
      "2020-02-07 15:39:11,213 - Phyluce - INFO - Reading Bacillus_cytotoxicus_GCF_002251045 genome\n",
      "2020-02-07 15:39:11,688 - Phyluce - INFO - Bacillus_cytotoxicus_GCF_002251045: 119 uces, 0 dupes, 119 non-dupes, 7 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:11,688 - Phyluce - INFO - ------ Working on Bacillus_gaemokensis_GCF_000712615 genome -----\n",
      "2020-02-07 15:39:11,696 - Phyluce - INFO - Reading Bacillus_gaemokensis_GCF_000712615 genome\n",
      "2020-02-07 15:39:12,129 - Phyluce - INFO - Bacillus_gaemokensis_GCF_000712615: 119 uces, 12 dupes, 107 non-dupes, 0 orient drop, 2 length drop, 105 written\n",
      "2020-02-07 15:39:12,129 - Phyluce - INFO - ------ Working on Bacillus_gaemokensis_GCF_001590835 genome -----\n",
      "2020-02-07 15:39:12,130 - Phyluce - INFO - Reading Bacillus_gaemokensis_GCF_001590835 genome\n",
      "2020-02-07 15:39:12,558 - Phyluce - INFO - Bacillus_gaemokensis_GCF_001590835: 119 uces, 10 dupes, 109 non-dupes, 0 orient drop, 2 length drop, 107 written\n",
      "2020-02-07 15:39:12,558 - Phyluce - INFO - --------- Working on Bacillus_luti_GCF_001884105 genome ---------\n",
      "2020-02-07 15:39:12,558 - Phyluce - INFO - Reading Bacillus_luti_GCF_001884105 genome\n",
      "2020-02-07 15:39:12,988 - Phyluce - INFO - Bacillus_luti_GCF_001884105: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 0 length drop, 109 written\n",
      "2020-02-07 15:39:12,988 - Phyluce - INFO - --------- Working on Bacillus_luti_GCF_008923665 genome ---------\n",
      "2020-02-07 15:39:13,015 - Phyluce - INFO - Reading Bacillus_luti_GCF_008923665 genome\n",
      "2020-02-07 15:39:13,448 - Phyluce - INFO - Bacillus_luti_GCF_008923665: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:13,448 - Phyluce - INFO - --------- Working on Bacillus_luti_GCF_009739945 genome ---------\n",
      "2020-02-07 15:39:13,449 - Phyluce - INFO - Reading Bacillus_luti_GCF_009739945 genome\n",
      "2020-02-07 15:39:13,956 - Phyluce - INFO - Bacillus_luti_GCF_009739945: 121 uces, 0 dupes, 121 non-dupes, 7 orient drop, 6 length drop, 108 written\n",
      "2020-02-07 15:39:13,956 - Phyluce - INFO - ----- Working on Bacillus_manliponensis_GCF_000712595 genome ----\n",
      "2020-02-07 15:39:13,957 - Phyluce - INFO - Reading Bacillus_manliponensis_GCF_000712595 genome\n",
      "2020-02-07 15:39:14,334 - Phyluce - INFO - Bacillus_manliponensis_GCF_000712595: 115 uces, 7 dupes, 108 non-dupes, 0 orient drop, 2 length drop, 106 written\n",
      "2020-02-07 15:39:14,334 - Phyluce - INFO - -------- Working on Bacillus_mobilis_GCF_001884045 genome -------\n",
      "2020-02-07 15:39:14,334 - Phyluce - INFO - Reading Bacillus_mobilis_GCF_001884045 genome\n",
      "2020-02-07 15:39:14,768 - Phyluce - INFO - Bacillus_mobilis_GCF_001884045: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:14,768 - Phyluce - INFO - -------- Working on Bacillus_mobilis_GCF_003612955 genome -------\n",
      "2020-02-07 15:39:14,769 - Phyluce - INFO - Reading Bacillus_mobilis_GCF_003612955 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 15:39:15,285 - Phyluce - INFO - Bacillus_mobilis_GCF_003612955: 121 uces, 0 dupes, 121 non-dupes, 7 orient drop, 5 length drop, 109 written\n",
      "2020-02-07 15:39:15,285 - Phyluce - INFO - -------- Working on Bacillus_mobilis_GCF_007681185 genome -------\n",
      "2020-02-07 15:39:15,286 - Phyluce - INFO - Reading Bacillus_mobilis_GCF_007681185 genome\n",
      "2020-02-07 15:39:15,762 - Phyluce - INFO - Bacillus_mobilis_GCF_007681185: 120 uces, 9 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:15,763 - Phyluce - INFO - -------- Working on Bacillus_mobilis_GCF_007681195 genome -------\n",
      "2020-02-07 15:39:15,792 - Phyluce - INFO - Reading Bacillus_mobilis_GCF_007681195 genome\n",
      "2020-02-07 15:39:16,303 - Phyluce - INFO - Bacillus_mobilis_GCF_007681195: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 1 length drop, 107 written\n",
      "2020-02-07 15:39:16,303 - Phyluce - INFO - -------- Working on Bacillus_mobilis_GCF_007682195 genome -------\n",
      "2020-02-07 15:39:16,304 - Phyluce - INFO - Reading Bacillus_mobilis_GCF_007682195 genome\n",
      "2020-02-07 15:39:16,738 - Phyluce - INFO - Bacillus_mobilis_GCF_007682195: 120 uces, 12 dupes, 108 non-dupes, 0 orient drop, 0 length drop, 108 written\n",
      "2020-02-07 15:39:16,739 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_000018825 genome -------\n",
      "2020-02-07 15:39:16,739 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_000018825 genome\n",
      "2020-02-07 15:39:17,262 - Phyluce - INFO - Bacillus_mycoides_GCF_000018825: 121 uces, 2 dupes, 119 non-dupes, 8 orient drop, 4 length drop, 107 written\n",
      "2020-02-07 15:39:17,262 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_000290695 genome -------\n",
      "2020-02-07 15:39:17,263 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_000290695 genome\n",
      "2020-02-07 15:39:17,708 - Phyluce - INFO - Bacillus_mycoides_GCF_000290695: 121 uces, 6 dupes, 115 non-dupes, 3 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:17,708 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_000517985 genome -------\n",
      "2020-02-07 15:39:17,709 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_000517985 genome\n",
      "2020-02-07 15:39:18,154 - Phyluce - INFO - Bacillus_mycoides_GCF_000517985: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:18,154 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_001044935 genome -------\n",
      "2020-02-07 15:39:18,157 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_001044935 genome\n",
      "2020-02-07 15:39:18,679 - Phyluce - INFO - Bacillus_mycoides_GCF_001044935: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 1 length drop, 107 written\n",
      "2020-02-07 15:39:18,679 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_001541985 genome -------\n",
      "2020-02-07 15:39:18,685 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_001541985 genome\n",
      "2020-02-07 15:39:19,144 - Phyluce - INFO - Bacillus_mycoides_GCF_001541985: 121 uces, 14 dupes, 107 non-dupes, 0 orient drop, 0 length drop, 107 written\n",
      "2020-02-07 15:39:19,144 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_001757825 genome -------\n",
      "2020-02-07 15:39:19,149 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_001757825 genome\n",
      "2020-02-07 15:39:19,585 - Phyluce - INFO - Bacillus_mycoides_GCF_001757825: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:19,585 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_001757915 genome -------\n",
      "2020-02-07 15:39:19,586 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_001757915 genome\n",
      "2020-02-07 15:39:20,014 - Phyluce - INFO - Bacillus_mycoides_GCF_001757915: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-07 15:39:20,014 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_001757985 genome -------\n",
      "2020-02-07 15:39:20,015 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_001757985 genome\n",
      "2020-02-07 15:39:20,443 - Phyluce - INFO - Bacillus_mycoides_GCF_001757985: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 0 length drop, 109 written\n",
      "2020-02-07 15:39:20,444 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_002014505 genome -------\n",
      "2020-02-07 15:39:20,453 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_002014505 genome\n",
      "2020-02-07 15:39:20,980 - Phyluce - INFO - Bacillus_mycoides_GCF_002014505: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:20,980 - Phyluce - INFO - ------- Working on Bacillus_mycoides_GCF_002014695 genome -------\n",
      "2020-02-07 15:39:20,981 - Phyluce - INFO - Reading Bacillus_mycoides_GCF_002014695 genome\n",
      "2020-02-07 15:39:21,416 - Phyluce - INFO - Bacillus_mycoides_GCF_002014695: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:21,417 - Phyluce - INFO - ---- Working on Bacillus_nitratireducens_GCF_001884135 genome ---\n",
      "2020-02-07 15:39:21,417 - Phyluce - INFO - Reading Bacillus_nitratireducens_GCF_001884135 genome\n",
      "2020-02-07 15:39:21,835 - Phyluce - INFO - Bacillus_nitratireducens_GCF_001884135: 120 uces, 9 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:21,835 - Phyluce - INFO - ---- Working on Bacillus_nitratireducens_GCF_007676595 genome ---\n",
      "2020-02-07 15:39:21,836 - Phyluce - INFO - Reading Bacillus_nitratireducens_GCF_007676595 genome\n",
      "2020-02-07 15:39:22,257 - Phyluce - INFO - Bacillus_nitratireducens_GCF_007676595: 120 uces, 9 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:22,257 - Phyluce - INFO - ---- Working on Bacillus_nitratireducens_GCF_007676605 genome ---\n",
      "2020-02-07 15:39:22,258 - Phyluce - INFO - Reading Bacillus_nitratireducens_GCF_007676605 genome\n",
      "2020-02-07 15:39:22,688 - Phyluce - INFO - Bacillus_nitratireducens_GCF_007676605: 120 uces, 10 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:22,689 - Phyluce - INFO - ---- Working on Bacillus_nitratireducens_GCF_007681065 genome ---\n",
      "2020-02-07 15:39:22,694 - Phyluce - INFO - Reading Bacillus_nitratireducens_GCF_007681065 genome\n",
      "2020-02-07 15:39:23,123 - Phyluce - INFO - Bacillus_nitratireducens_GCF_007681065: 120 uces, 10 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:23,123 - Phyluce - INFO - ---- Working on Bacillus_nitratireducens_GCF_007681365 genome ---\n",
      "2020-02-07 15:39:23,124 - Phyluce - INFO - Reading Bacillus_nitratireducens_GCF_007681365 genome\n",
      "2020-02-07 15:39:23,541 - Phyluce - INFO - Bacillus_nitratireducens_GCF_007681365: 120 uces, 8 dupes, 112 non-dupes, 0 orient drop, 1 length drop, 111 written\n",
      "2020-02-07 15:39:23,541 - Phyluce - INFO - ---- Working on Bacillus_nitratireducens_GCF_007682355 genome ---\n",
      "2020-02-07 15:39:23,542 - Phyluce - INFO - Reading Bacillus_nitratireducens_GCF_007682355 genome\n",
      "2020-02-07 15:39:23,974 - Phyluce - INFO - Bacillus_nitratireducens_GCF_007682355: 120 uces, 7 dupes, 113 non-dupes, 0 orient drop, 2 length drop, 111 written\n",
      "2020-02-07 15:39:23,974 - Phyluce - INFO - ------- Working on Bacillus_pacificus_GCA_009884315 genome ------\n",
      "2020-02-07 15:39:23,975 - Phyluce - INFO - Reading Bacillus_pacificus_GCA_009884315 genome\n",
      "2020-02-07 15:39:24,482 - Phyluce - INFO - Bacillus_pacificus_GCA_009884315: 121 uces, 1 dupes, 120 non-dupes, 5 orient drop, 6 length drop, 109 written\n",
      "2020-02-07 15:39:24,482 - Phyluce - INFO - ------- Working on Bacillus_pacificus_GCF_001884025 genome ------\n",
      "2020-02-07 15:39:24,483 - Phyluce - INFO - Reading Bacillus_pacificus_GCF_001884025 genome\n",
      "2020-02-07 15:39:24,918 - Phyluce - INFO - Bacillus_pacificus_GCF_001884025: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 2 length drop, 109 written\n",
      "2020-02-07 15:39:24,919 - Phyluce - INFO - ------- Working on Bacillus_pacificus_GCF_003858675 genome ------\n",
      "2020-02-07 15:39:24,920 - Phyluce - INFO - Reading Bacillus_pacificus_GCF_003858675 genome\n",
      "2020-02-07 15:39:25,347 - Phyluce - INFO - Bacillus_pacificus_GCF_003858675: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 1 length drop, 111 written\n",
      "2020-02-07 15:39:25,348 - Phyluce - INFO - ------- Working on Bacillus_pacificus_GCF_006349595 genome ------\n",
      "2020-02-07 15:39:25,365 - Phyluce - INFO - Reading Bacillus_pacificus_GCF_006349595 genome\n",
      "2020-02-07 15:39:25,886 - Phyluce - INFO - Bacillus_pacificus_GCF_006349595: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 3 length drop, 109 written\n",
      "2020-02-07 15:39:25,886 - Phyluce - INFO - ----- Working on Bacillus_paramycoides_GCF_001884235 genome -----\n",
      "2020-02-07 15:39:25,887 - Phyluce - INFO - Reading Bacillus_paramycoides_GCF_001884235 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 15:39:26,331 - Phyluce - INFO - Bacillus_paramycoides_GCF_001884235: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-07 15:39:26,332 - Phyluce - INFO - ----- Working on Bacillus_paramycoides_GCF_007682005 genome -----\n",
      "2020-02-07 15:39:26,333 - Phyluce - INFO - Reading Bacillus_paramycoides_GCF_007682005 genome\n",
      "2020-02-07 15:39:26,810 - Phyluce - INFO - Bacillus_paramycoides_GCF_007682005: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:26,810 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCA_009648955 genome -----\n",
      "2020-02-07 15:39:26,814 - Phyluce - INFO - Reading Bacillus_paranthracis_GCA_009648955 genome\n",
      "2020-02-07 15:39:27,331 - Phyluce - INFO - Bacillus_paranthracis_GCA_009648955: 121 uces, 1 dupes, 120 non-dupes, 5 orient drop, 7 length drop, 108 written\n",
      "2020-02-07 15:39:27,331 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCA_009873395 genome -----\n",
      "2020-02-07 15:39:27,332 - Phyluce - INFO - Reading Bacillus_paranthracis_GCA_009873395 genome\n",
      "2020-02-07 15:39:27,835 - Phyluce - INFO - Bacillus_paranthracis_GCA_009873395: 120 uces, 0 dupes, 120 non-dupes, 6 orient drop, 7 length drop, 107 written\n",
      "2020-02-07 15:39:27,835 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCA_009884335 genome -----\n",
      "2020-02-07 15:39:27,836 - Phyluce - INFO - Reading Bacillus_paranthracis_GCA_009884335 genome\n",
      "2020-02-07 15:39:28,344 - Phyluce - INFO - Bacillus_paranthracis_GCA_009884335: 121 uces, 1 dupes, 120 non-dupes, 6 orient drop, 6 length drop, 108 written\n",
      "2020-02-07 15:39:28,344 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_001883995 genome -----\n",
      "2020-02-07 15:39:28,344 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_001883995 genome\n",
      "2020-02-07 15:39:28,774 - Phyluce - INFO - Bacillus_paranthracis_GCF_001883995: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:28,775 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_004307985 genome -----\n",
      "2020-02-07 15:39:28,787 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_004307985 genome\n",
      "2020-02-07 15:39:29,278 - Phyluce - INFO - Bacillus_paranthracis_GCF_004307985: 119 uces, 10 dupes, 109 non-dupes, 0 orient drop, 2 length drop, 107 written\n",
      "2020-02-07 15:39:29,278 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_005117095 genome -----\n",
      "2020-02-07 15:39:29,279 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_005117095 genome\n",
      "2020-02-07 15:39:29,714 - Phyluce - INFO - Bacillus_paranthracis_GCF_005117095: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-07 15:39:29,714 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_005848985 genome -----\n",
      "2020-02-07 15:39:29,725 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_005848985 genome\n",
      "2020-02-07 15:39:30,244 - Phyluce - INFO - Bacillus_paranthracis_GCF_005848985: 120 uces, 0 dupes, 120 non-dupes, 8 orient drop, 6 length drop, 106 written\n",
      "2020-02-07 15:39:30,244 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_007682135 genome -----\n",
      "2020-02-07 15:39:30,245 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_007682135 genome\n",
      "2020-02-07 15:39:30,747 - Phyluce - INFO - Bacillus_paranthracis_GCF_007682135: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:30,747 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_007682155 genome -----\n",
      "2020-02-07 15:39:30,748 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_007682155 genome\n",
      "2020-02-07 15:39:31,183 - Phyluce - INFO - Bacillus_paranthracis_GCF_007682155: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 1 length drop, 111 written\n",
      "2020-02-07 15:39:31,183 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_009498675 genome -----\n",
      "2020-02-07 15:39:31,184 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_009498675 genome\n",
      "2020-02-07 15:39:31,615 - Phyluce - INFO - Bacillus_paranthracis_GCF_009498675: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:31,616 - Phyluce - INFO - ----- Working on Bacillus_paranthracis_GCF_009799965 genome -----\n",
      "2020-02-07 15:39:31,627 - Phyluce - INFO - Reading Bacillus_paranthracis_GCF_009799965 genome\n",
      "2020-02-07 15:39:32,153 - Phyluce - INFO - Bacillus_paranthracis_GCF_009799965: 121 uces, 1 dupes, 120 non-dupes, 5 orient drop, 6 length drop, 109 written\n",
      "2020-02-07 15:39:32,153 - Phyluce - INFO - ----- Working on Bacillus_proteolyticus_GCF_001884065 genome ----\n",
      "2020-02-07 15:39:32,154 - Phyluce - INFO - Reading Bacillus_proteolyticus_GCF_001884065 genome\n",
      "2020-02-07 15:39:32,587 - Phyluce - INFO - Bacillus_proteolyticus_GCF_001884065: 120 uces, 12 dupes, 108 non-dupes, 0 orient drop, 1 length drop, 107 written\n",
      "2020-02-07 15:39:32,587 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_000161415 genome ----\n",
      "2020-02-07 15:39:32,596 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_000161415 genome\n",
      "2020-02-07 15:39:33,012 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_000161415: 118 uces, 0 dupes, 118 non-dupes, 7 orient drop, 3 length drop, 108 written\n",
      "2020-02-07 15:39:33,012 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_000161435 genome ----\n",
      "2020-02-07 15:39:33,012 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_000161435 genome\n",
      "2020-02-07 15:39:33,426 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_000161435: 119 uces, 0 dupes, 119 non-dupes, 9 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:33,427 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_000746965 genome ----\n",
      "2020-02-07 15:39:33,436 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_000746965 genome\n",
      "2020-02-07 15:39:33,858 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_000746965: 119 uces, 9 dupes, 110 non-dupes, 0 orient drop, 2 length drop, 108 written\n",
      "2020-02-07 15:39:33,859 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002550095 genome ----\n",
      "2020-02-07 15:39:33,859 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002550095 genome\n",
      "2020-02-07 15:39:34,276 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002550095: 119 uces, 9 dupes, 110 non-dupes, 0 orient drop, 2 length drop, 108 written\n",
      "2020-02-07 15:39:34,276 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002555615 genome ----\n",
      "2020-02-07 15:39:34,277 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002555615 genome\n",
      "2020-02-07 15:39:34,691 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002555615: 119 uces, 12 dupes, 107 non-dupes, 0 orient drop, 0 length drop, 107 written\n",
      "2020-02-07 15:39:34,691 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002555685 genome ----\n",
      "2020-02-07 15:39:34,692 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002555685 genome\n",
      "2020-02-07 15:39:35,124 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002555685: 119 uces, 8 dupes, 111 non-dupes, 0 orient drop, 2 length drop, 109 written\n",
      "2020-02-07 15:39:35,125 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002555995 genome ----\n",
      "2020-02-07 15:39:35,125 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002555995 genome\n",
      "2020-02-07 15:39:35,533 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002555995: 119 uces, 8 dupes, 111 non-dupes, 0 orient drop, 2 length drop, 109 written\n",
      "2020-02-07 15:39:35,533 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002556205 genome ----\n",
      "2020-02-07 15:39:35,542 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002556205 genome\n",
      "2020-02-07 15:39:36,017 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002556205: 119 uces, 11 dupes, 108 non-dupes, 0 orient drop, 0 length drop, 108 written\n",
      "2020-02-07 15:39:36,017 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002557395 genome ----\n",
      "2020-02-07 15:39:36,047 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002557395 genome\n",
      "2020-02-07 15:39:36,527 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002557395: 119 uces, 12 dupes, 107 non-dupes, 0 orient drop, 1 length drop, 106 written\n",
      "2020-02-07 15:39:36,528 - Phyluce - INFO - ---- Working on Bacillus_pseudomycoides_GCF_002569075 genome ----\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 15:39:36,540 - Phyluce - INFO - Reading Bacillus_pseudomycoides_GCF_002569075 genome\n",
      "2020-02-07 15:39:36,956 - Phyluce - INFO - Bacillus_pseudomycoides_GCF_002569075: 119 uces, 10 dupes, 109 non-dupes, 0 orient drop, 0 length drop, 109 written\n",
      "2020-02-07 15:39:36,956 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_000161735 genome ----\n",
      "2020-02-07 15:39:36,969 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_000161735 genome\n",
      "2020-02-07 15:39:37,422 - Phyluce - INFO - Bacillus_thuringiensis_GCF_000161735: 121 uces, 0 dupes, 121 non-dupes, 9 orient drop, 3 length drop, 109 written\n",
      "2020-02-07 15:39:37,422 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_000710255 genome ----\n",
      "2020-02-07 15:39:37,423 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_000710255 genome\n",
      "2020-02-07 15:39:37,876 - Phyluce - INFO - Bacillus_thuringiensis_GCF_000710255: 120 uces, 10 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-07 15:39:37,876 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_001402735 genome ----\n",
      "2020-02-07 15:39:37,879 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_001402735 genome\n",
      "2020-02-07 15:39:38,324 - Phyluce - INFO - Bacillus_thuringiensis_GCF_001402735: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 1 length drop, 107 written\n",
      "2020-02-07 15:39:38,325 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_001675515 genome ----\n",
      "2020-02-07 15:39:38,326 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_001675515 genome\n",
      "2020-02-07 15:39:38,846 - Phyluce - INFO - Bacillus_thuringiensis_GCF_001675515: 121 uces, 1 dupes, 120 non-dupes, 7 orient drop, 3 length drop, 110 written\n",
      "2020-02-07 15:39:38,846 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_002148135 genome ----\n",
      "2020-02-07 15:39:38,862 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_002148135 genome\n",
      "2020-02-07 15:39:39,327 - Phyluce - INFO - Bacillus_thuringiensis_GCF_002148135: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-07 15:39:39,327 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_002551895 genome ----\n",
      "2020-02-07 15:39:39,356 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_002551895 genome\n",
      "2020-02-07 15:39:39,804 - Phyluce - INFO - Bacillus_thuringiensis_GCF_002551895: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 0 length drop, 109 written\n",
      "2020-02-07 15:39:39,805 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_002559825 genome ----\n",
      "2020-02-07 15:39:39,805 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_002559825 genome\n",
      "2020-02-07 15:39:40,250 - Phyluce - INFO - Bacillus_thuringiensis_GCF_002559825: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 0 length drop, 108 written\n",
      "2020-02-07 15:39:40,250 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_002562665 genome ----\n",
      "2020-02-07 15:39:40,251 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_002562665 genome\n",
      "2020-02-07 15:39:40,695 - Phyluce - INFO - Bacillus_thuringiensis_GCF_002562665: 121 uces, 8 dupes, 113 non-dupes, 0 orient drop, 1 length drop, 112 written\n",
      "2020-02-07 15:39:40,695 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_002564505 genome ----\n",
      "2020-02-07 15:39:40,696 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_002564505 genome\n",
      "2020-02-07 15:39:41,201 - Phyluce - INFO - Bacillus_thuringiensis_GCF_002564505: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-07 15:39:41,201 - Phyluce - INFO - ----- Working on Bacillus_thuringiensis_GCF_002570925 genome ----\n",
      "2020-02-07 15:39:41,202 - Phyluce - INFO - Reading Bacillus_thuringiensis_GCF_002570925 genome\n",
      "2020-02-07 15:39:41,644 - Phyluce - INFO - Bacillus_thuringiensis_GCF_002570925: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:41,644 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCA_009799785 genome ------\n",
      "2020-02-07 15:39:41,645 - Phyluce - INFO - Reading Bacillus_toyonensis_GCA_009799785 genome\n",
      "2020-02-07 15:39:42,156 - Phyluce - INFO - Bacillus_toyonensis_GCA_009799785: 121 uces, 0 dupes, 121 non-dupes, 9 orient drop, 5 length drop, 107 written\n",
      "2020-02-07 15:39:42,156 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002550925 genome ------\n",
      "2020-02-07 15:39:42,157 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002550925 genome\n",
      "2020-02-07 15:39:42,591 - Phyluce - INFO - Bacillus_toyonensis_GCF_002550925: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-07 15:39:42,592 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002551965 genome ------\n",
      "2020-02-07 15:39:42,592 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002551965 genome\n",
      "2020-02-07 15:39:43,041 - Phyluce - INFO - Bacillus_toyonensis_GCF_002551965: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 2 length drop, 109 written\n",
      "2020-02-07 15:39:43,042 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002554865 genome ------\n",
      "2020-02-07 15:39:43,070 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002554865 genome\n",
      "2020-02-07 15:39:43,516 - Phyluce - INFO - Bacillus_toyonensis_GCF_002554865: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-07 15:39:43,516 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002555905 genome ------\n",
      "2020-02-07 15:39:43,530 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002555905 genome\n",
      "2020-02-07 15:39:43,975 - Phyluce - INFO - Bacillus_toyonensis_GCF_002555905: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-07 15:39:43,975 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002556675 genome ------\n",
      "2020-02-07 15:39:43,976 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002556675 genome\n",
      "2020-02-07 15:39:44,408 - Phyluce - INFO - Bacillus_toyonensis_GCF_002556675: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 1 length drop, 109 written\n",
      "2020-02-07 15:39:44,409 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002570155 genome ------\n",
      "2020-02-07 15:39:44,409 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002570155 genome\n",
      "2020-02-07 15:39:44,843 - Phyluce - INFO - Bacillus_toyonensis_GCF_002570155: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 0 length drop, 110 written\n",
      "2020-02-07 15:39:44,843 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002580355 genome ------\n",
      "2020-02-07 15:39:44,844 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002580355 genome\n",
      "2020-02-07 15:39:45,278 - Phyluce - INFO - Bacillus_toyonensis_GCF_002580355: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 1 length drop, 111 written\n",
      "2020-02-07 15:39:45,278 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002581325 genome ------\n",
      "2020-02-07 15:39:45,279 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002581325 genome\n",
      "2020-02-07 15:39:45,713 - Phyluce - INFO - Bacillus_toyonensis_GCF_002581325: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 1 length drop, 107 written\n",
      "2020-02-07 15:39:45,714 - Phyluce - INFO - ------ Working on Bacillus_toyonensis_GCF_002581955 genome ------\n",
      "2020-02-07 15:39:45,718 - Phyluce - INFO - Reading Bacillus_toyonensis_GCF_002581955 genome\n",
      "2020-02-07 15:39:46,173 - Phyluce - INFO - Bacillus_toyonensis_GCF_002581955: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 2 length drop, 109 written\n",
      "2020-02-07 15:39:46,173 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_001884035 genome -------\n",
      "2020-02-07 15:39:46,174 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_001884035 genome\n",
      "2020-02-07 15:39:46,610 - Phyluce - INFO - Bacillus_tropicus_GCF_001884035: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 2 length drop, 108 written\n",
      "2020-02-07 15:39:46,611 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_006349625 genome -------\n",
      "2020-02-07 15:39:46,611 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_006349625 genome\n",
      "2020-02-07 15:39:47,047 - Phyluce - INFO - Bacillus_tropicus_GCF_006349625: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 3 length drop, 107 written\n",
      "2020-02-07 15:39:47,048 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_006349645 genome -------\n",
      "2020-02-07 15:39:47,048 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_006349645 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-07 15:39:47,480 - Phyluce - INFO - Bacillus_tropicus_GCF_006349645: 121 uces, 11 dupes, 110 non-dupes, 0 orient drop, 2 length drop, 108 written\n",
      "2020-02-07 15:39:47,480 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_006457285 genome -------\n",
      "2020-02-07 15:39:47,481 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_006457285 genome\n",
      "2020-02-07 15:39:47,992 - Phyluce - INFO - Bacillus_tropicus_GCF_006457285: 121 uces, 0 dupes, 121 non-dupes, 10 orient drop, 4 length drop, 107 written\n",
      "2020-02-07 15:39:47,992 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_007672275 genome -------\n",
      "2020-02-07 15:39:47,993 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_007672275 genome\n",
      "2020-02-07 15:39:48,421 - Phyluce - INFO - Bacillus_tropicus_GCF_007672275: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:48,422 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_007676425 genome -------\n",
      "2020-02-07 15:39:48,422 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_007676425 genome\n",
      "2020-02-07 15:39:48,859 - Phyluce - INFO - Bacillus_tropicus_GCF_007676425: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 3 length drop, 108 written\n",
      "2020-02-07 15:39:48,860 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_007682035 genome -------\n",
      "2020-02-07 15:39:48,871 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_007682035 genome\n",
      "2020-02-07 15:39:49,324 - Phyluce - INFO - Bacillus_tropicus_GCF_007682035: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-07 15:39:49,324 - Phyluce - INFO - ------- Working on Bacillus_tropicus_GCF_007682405 genome -------\n",
      "2020-02-07 15:39:49,325 - Phyluce - INFO - Reading Bacillus_tropicus_GCF_007682405 genome\n",
      "2020-02-07 15:39:49,753 - Phyluce - INFO - Bacillus_tropicus_GCF_007682405: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:49,753 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_000160955 genome ------\n",
      "2020-02-07 15:39:49,754 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_000160955 genome\n",
      "2020-02-07 15:39:50,182 - Phyluce - INFO - Bacillus_wiedmannii_GCF_000160955: 120 uces, 0 dupes, 120 non-dupes, 8 orient drop, 4 length drop, 108 written\n",
      "2020-02-07 15:39:50,182 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_001645395 genome ------\n",
      "2020-02-07 15:39:50,190 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_001645395 genome\n",
      "2020-02-07 15:39:50,624 - Phyluce - INFO - Bacillus_wiedmannii_GCF_001645395: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:50,624 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002014635 genome ------\n",
      "2020-02-07 15:39:50,625 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002014635 genome\n",
      "2020-02-07 15:39:51,051 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002014635: 121 uces, 9 dupes, 112 non-dupes, 0 orient drop, 1 length drop, 111 written\n",
      "2020-02-07 15:39:51,052 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002555555 genome ------\n",
      "2020-02-07 15:39:51,052 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002555555 genome\n",
      "2020-02-07 15:39:51,483 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002555555: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 0 length drop, 111 written\n",
      "2020-02-07 15:39:51,483 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002555735 genome ------\n",
      "2020-02-07 15:39:51,484 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002555735 genome\n",
      "2020-02-07 15:39:51,935 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002555735: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 1 length drop, 108 written\n",
      "2020-02-07 15:39:51,935 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002555855 genome ------\n",
      "2020-02-07 15:39:51,940 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002555855 genome\n",
      "2020-02-07 15:39:52,377 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002555855: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:52,377 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002555945 genome ------\n",
      "2020-02-07 15:39:52,378 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002555945 genome\n",
      "2020-02-07 15:39:52,813 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002555945: 121 uces, 12 dupes, 109 non-dupes, 0 orient drop, 0 length drop, 109 written\n",
      "2020-02-07 15:39:52,813 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002556175 genome ------\n",
      "2020-02-07 15:39:52,814 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002556175 genome\n",
      "2020-02-07 15:39:53,244 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002556175: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:53,244 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002556405 genome ------\n",
      "2020-02-07 15:39:53,259 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002556405 genome\n",
      "2020-02-07 15:39:53,693 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002556405: 121 uces, 10 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-07 15:39:53,693 - Phyluce - INFO - ------ Working on Bacillus_wiedmannii_GCF_002569595 genome ------\n",
      "2020-02-07 15:39:53,694 - Phyluce - INFO - Reading Bacillus_wiedmannii_GCF_002569595 genome\n",
      "2020-02-07 15:39:54,134 - Phyluce - INFO - Bacillus_wiedmannii_GCF_002569595: 121 uces, 13 dupes, 108 non-dupes, 0 orient drop, 0 length drop, 108 written\n",
      "2020-02-07 15:39:54,134 - Phyluce - INFO - ------- Working on Bacillus_anthracis_GCF_000008165 genome ------\n",
      "2020-02-07 15:39:54,146 - Phyluce - INFO - Reading Bacillus_anthracis_GCF_000008165 genome\n",
      "2020-02-07 15:39:54,649 - Phyluce - INFO - Bacillus_anthracis_GCF_000008165: 121 uces, 0 dupes, 121 non-dupes, 8 orient drop, 6 length drop, 107 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/bacillus/extract_probes_from_group/probe_fasta --output results/phyluce/bacillus/extract_probes_from_group/multifastas.sqlite --base-taxon Bacillus_anthracis_GCF_000008165;\n",
      "bacillus_albus_gcf_001884185.\n",
      "bacillus_albus_gcf_003966295.\n",
      "bacillus_albus_gcf_004116025.\n",
      "bacillus_albus_gcf_004116035.\n",
      "bacillus_albus_gcf_004116065.\n",
      "bacillus_albus_gcf_004116085.\n",
      "bacillus_albus_gcf_004116415.\n",
      "bacillus_albus_gcf_004153665.\n",
      "bacillus_albus_gcf_009740005.\n",
      "bacillus_anthracis_gcf_000558965.\n",
      "bacillus_anthracis_gcf_000559005.\n",
      "bacillus_anthracis_gcf_000697555.\n",
      "bacillus_anthracis_gcf_000742875.\n",
      "bacillus_anthracis_gcf_000783055.\n",
      "bacillus_anthracis_gcf_001883895.\n",
      "bacillus_anthracis_gcf_002233635.\n",
      "bacillus_anthracis_gcf_002525695.\n",
      "bacillus_anthracis_gcf_002525705.\n",
      "bacillus_bombysepticus_gcf_000831065.\n",
      "bacillus_cereus_gcf_000161315.\n",
      "bacillus_cereus_gcf_000290795.\n",
      "bacillus_cereus_gcf_000291075.\n",
      "bacillus_cereus_gcf_001583875.\n",
      "bacillus_cereus_gcf_001584025.\n",
      "bacillus_cereus_gcf_001683695.\n",
      "bacillus_cereus_gcf_002530565.\n",
      "bacillus_cereus_gcf_002552015.\n",
      "bacillus_cereus_gcf_002560715.\n",
      "bacillus_cereus_gcf_002565765.\n",
      "bacillus_cytotoxicus_gcf_000017425.\n",
      "bacillus_cytotoxicus_gcf_001050335.\n",
      "bacillus_cytotoxicus_gcf_002250885.\n",
      "bacillus_cytotoxicus_gcf_002250905.\n",
      "bacillus_cytotoxicus_gcf_002250925.\n",
      "bacillus_cytotoxicus_gcf_002250945.\n",
      "bacillus_cytotoxicus_gcf_002250965.\n",
      "bacillus_cytotoxicus_gcf_002251005.\n",
      "bacillus_cytotoxicus_gcf_002251025.\n",
      "bacillus_cytotoxicus_gcf_002251045.\n",
      "bacillus_gaemokensis_gcf_000712615.\n",
      "bacillus_gaemokensis_gcf_001590835.\n",
      "bacillus_luti_gcf_001884105.\n",
      "bacillus_luti_gcf_008923665.\n",
      "bacillus_luti_gcf_009739945.\n",
      "bacillus_manliponensis_gcf_000712595.\n",
      "bacillus_mobilis_gcf_001884045.\n",
      "bacillus_mobilis_gcf_003612955.\n",
      "bacillus_mobilis_gcf_007681185.\n",
      "bacillus_mobilis_gcf_007681195.\n",
      "bacillus_mobilis_gcf_007682195.\n",
      "bacillus_mycoides_gcf_000018825.\n",
      "bacillus_mycoides_gcf_000290695.\n",
      "bacillus_mycoides_gcf_000517985.\n",
      "bacillus_mycoides_gcf_001044935.\n",
      "bacillus_mycoides_gcf_001541985.\n",
      "bacillus_mycoides_gcf_001757825.\n",
      "bacillus_mycoides_gcf_001757915.\n",
      "bacillus_mycoides_gcf_001757985.\n",
      "bacillus_mycoides_gcf_002014505.\n",
      "bacillus_mycoides_gcf_002014695.\n",
      "bacillus_nitratireducens_gcf_001884135.\n",
      "bacillus_nitratireducens_gcf_007676595.\n",
      "bacillus_nitratireducens_gcf_007676605.\n",
      "bacillus_nitratireducens_gcf_007681065.\n",
      "bacillus_nitratireducens_gcf_007681365.\n",
      "bacillus_nitratireducens_gcf_007682355.\n",
      "bacillus_pacificus_gca_009884315.\n",
      "bacillus_pacificus_gcf_001884025.\n",
      "bacillus_pacificus_gcf_003858675.\n",
      "bacillus_pacificus_gcf_006349595.\n",
      "bacillus_paramycoides_gcf_001884235.\n",
      "bacillus_paramycoides_gcf_007682005.\n",
      "bacillus_paranthracis_gca_009648955.\n",
      "bacillus_paranthracis_gca_009873395.\n",
      "bacillus_paranthracis_gca_009884335.\n",
      "bacillus_paranthracis_gcf_001883995.\n",
      "bacillus_paranthracis_gcf_004307985.\n",
      "bacillus_paranthracis_gcf_005117095.\n",
      "bacillus_paranthracis_gcf_005848985.\n",
      "bacillus_paranthracis_gcf_007682135.\n",
      "bacillus_paranthracis_gcf_007682155.\n",
      "bacillus_paranthracis_gcf_009498675.\n",
      "bacillus_paranthracis_gcf_009799965.\n",
      "bacillus_proteolyticus_gcf_001884065.\n",
      "bacillus_pseudomycoides_gcf_000161415.\n",
      "bacillus_pseudomycoides_gcf_000161435.\n",
      "bacillus_pseudomycoides_gcf_000746965.\n",
      "bacillus_pseudomycoides_gcf_002550095.\n",
      "bacillus_pseudomycoides_gcf_002555615.\n",
      "bacillus_pseudomycoides_gcf_002555685.\n",
      "bacillus_pseudomycoides_gcf_002555995.\n",
      "bacillus_pseudomycoides_gcf_002556205.\n",
      "bacillus_pseudomycoides_gcf_002557395.\n",
      "bacillus_pseudomycoides_gcf_002569075.\n",
      "bacillus_thuringiensis_gcf_000161735.\n",
      "bacillus_thuringiensis_gcf_000710255.\n",
      "bacillus_thuringiensis_gcf_001402735.\n",
      "bacillus_thuringiensis_gcf_001675515.\n",
      "bacillus_thuringiensis_gcf_002148135.\n",
      "bacillus_thuringiensis_gcf_002551895.\n",
      "bacillus_thuringiensis_gcf_002559825.\n",
      "bacillus_thuringiensis_gcf_002562665.\n",
      "bacillus_thuringiensis_gcf_002564505.\n",
      "bacillus_thuringiensis_gcf_002570925.\n",
      "bacillus_toyonensis_gca_009799785.\n",
      "bacillus_toyonensis_gcf_002550925.\n",
      "bacillus_toyonensis_gcf_002551965.\n",
      "bacillus_toyonensis_gcf_002554865.\n",
      "bacillus_toyonensis_gcf_002555905.\n",
      "bacillus_toyonensis_gcf_002556675.\n",
      "bacillus_toyonensis_gcf_002570155.\n",
      "bacillus_toyonensis_gcf_002580355.\n",
      "bacillus_toyonensis_gcf_002581325.\n",
      "bacillus_toyonensis_gcf_002581955.\n",
      "bacillus_tropicus_gcf_001884035.\n",
      "bacillus_tropicus_gcf_006349625.\n",
      "bacillus_tropicus_gcf_006349645.\n",
      "bacillus_tropicus_gcf_006457285.\n",
      "bacillus_tropicus_gcf_007672275.\n",
      "bacillus_tropicus_gcf_007676425.\n",
      "bacillus_tropicus_gcf_007682035.\n",
      "bacillus_tropicus_gcf_007682405.\n",
      "bacillus_wiedmannii_gcf_000160955.\n",
      "bacillus_wiedmannii_gcf_001645395.\n",
      "bacillus_wiedmannii_gcf_002014635.\n",
      "bacillus_wiedmannii_gcf_002555555.\n",
      "bacillus_wiedmannii_gcf_002555735.\n",
      "bacillus_wiedmannii_gcf_002555855.\n",
      "bacillus_wiedmannii_gcf_002555945.\n",
      "bacillus_wiedmannii_gcf_002556175.\n",
      "bacillus_wiedmannii_gcf_002556405.\n",
      "bacillus_wiedmannii_gcf_002569595.\n",
      "bacillus_anthracis_gcf_000008165.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/bacillus/extract_probes_from_group/multifastas.sqlite --base-taxon Bacillus_anthracis_GCF_000008165;\n",
      "Loci shared by 0 taxa:\t116.0\n",
      "Loci shared by 1 taxa:\t116.0\n",
      "Loci shared by 2 taxa:\t114.0\n",
      "Loci shared by 3 taxa:\t114.0\n",
      "Loci shared by 4 taxa:\t114.0\n",
      "Loci shared by 5 taxa:\t114.0\n",
      "Loci shared by 6 taxa:\t114.0\n",
      "Loci shared by 7 taxa:\t114.0\n",
      "Loci shared by 8 taxa:\t114.0\n",
      "Loci shared by 9 taxa:\t114.0\n",
      "Loci shared by 10 taxa:\t114.0\n",
      "Loci shared by 11 taxa:\t114.0\n",
      "Loci shared by 12 taxa:\t114.0\n",
      "Loci shared by 13 taxa:\t114.0\n",
      "Loci shared by 14 taxa:\t114.0\n",
      "Loci shared by 15 taxa:\t114.0\n",
      "Loci shared by 16 taxa:\t114.0\n",
      "Loci shared by 17 taxa:\t114.0\n",
      "Loci shared by 18 taxa:\t114.0\n",
      "Loci shared by 19 taxa:\t114.0\n",
      "Loci shared by 20 taxa:\t114.0\n",
      "Loci shared by 21 taxa:\t114.0\n",
      "Loci shared by 22 taxa:\t114.0\n",
      "Loci shared by 23 taxa:\t113.0\n",
      "Loci shared by 24 taxa:\t113.0\n",
      "Loci shared by 25 taxa:\t113.0\n",
      "Loci shared by 26 taxa:\t113.0\n",
      "Loci shared by 27 taxa:\t113.0\n",
      "Loci shared by 28 taxa:\t112.0\n",
      "Loci shared by 29 taxa:\t112.0\n",
      "Loci shared by 30 taxa:\t112.0\n",
      "Loci shared by 31 taxa:\t112.0\n",
      "Loci shared by 32 taxa:\t112.0\n",
      "Loci shared by 33 taxa:\t112.0\n",
      "Loci shared by 34 taxa:\t112.0\n",
      "Loci shared by 35 taxa:\t112.0\n",
      "Loci shared by 36 taxa:\t112.0\n",
      "Loci shared by 37 taxa:\t112.0\n",
      "Loci shared by 38 taxa:\t112.0\n",
      "Loci shared by 39 taxa:\t112.0\n",
      "Loci shared by 40 taxa:\t112.0\n",
      "Loci shared by 41 taxa:\t112.0\n",
      "Loci shared by 42 taxa:\t112.0\n",
      "Loci shared by 43 taxa:\t112.0\n",
      "Loci shared by 44 taxa:\t112.0\n",
      "Loci shared by 45 taxa:\t112.0\n",
      "Loci shared by 46 taxa:\t112.0\n",
      "Loci shared by 47 taxa:\t112.0\n",
      "Loci shared by 48 taxa:\t112.0\n",
      "Loci shared by 49 taxa:\t112.0\n",
      "Loci shared by 50 taxa:\t112.0\n",
      "Loci shared by 51 taxa:\t112.0\n",
      "Loci shared by 52 taxa:\t112.0\n",
      "Loci shared by 53 taxa:\t112.0\n",
      "Loci shared by 54 taxa:\t112.0\n",
      "Loci shared by 55 taxa:\t112.0\n",
      "Loci shared by 56 taxa:\t112.0\n",
      "Loci shared by 57 taxa:\t112.0\n",
      "Loci shared by 58 taxa:\t111.0\n",
      "Loci shared by 59 taxa:\t111.0\n",
      "Loci shared by 60 taxa:\t111.0\n",
      "Loci shared by 61 taxa:\t111.0\n",
      "Loci shared by 62 taxa:\t111.0\n",
      "Loci shared by 63 taxa:\t111.0\n",
      "Loci shared by 64 taxa:\t111.0\n",
      "Loci shared by 65 taxa:\t111.0\n",
      "Loci shared by 66 taxa:\t110.0\n",
      "Loci shared by 67 taxa:\t110.0\n",
      "Loci shared by 68 taxa:\t110.0\n",
      "Loci shared by 69 taxa:\t110.0\n",
      "Loci shared by 70 taxa:\t110.0\n",
      "Loci shared by 71 taxa:\t110.0\n",
      "Loci shared by 72 taxa:\t110.0\n",
      "Loci shared by 73 taxa:\t110.0\n",
      "Loci shared by 74 taxa:\t110.0\n",
      "Loci shared by 75 taxa:\t110.0\n",
      "Loci shared by 76 taxa:\t110.0\n",
      "Loci shared by 77 taxa:\t110.0\n",
      "Loci shared by 78 taxa:\t110.0\n",
      "Loci shared by 79 taxa:\t110.0\n",
      "Loci shared by 80 taxa:\t110.0\n",
      "Loci shared by 81 taxa:\t110.0\n",
      "Loci shared by 82 taxa:\t110.0\n",
      "Loci shared by 83 taxa:\t110.0\n",
      "Loci shared by 84 taxa:\t109.0\n",
      "Loci shared by 85 taxa:\t109.0\n",
      "Loci shared by 86 taxa:\t109.0\n",
      "Loci shared by 87 taxa:\t109.0\n",
      "Loci shared by 88 taxa:\t109.0\n",
      "Loci shared by 89 taxa:\t109.0\n",
      "Loci shared by 90 taxa:\t109.0\n",
      "Loci shared by 91 taxa:\t109.0\n",
      "Loci shared by 92 taxa:\t109.0\n",
      "Loci shared by 93 taxa:\t109.0\n",
      "Loci shared by 94 taxa:\t109.0\n",
      "Loci shared by 95 taxa:\t106.0\n",
      "Loci shared by 96 taxa:\t106.0\n",
      "Loci shared by 97 taxa:\t106.0\n",
      "Loci shared by 98 taxa:\t106.0\n",
      "Loci shared by 99 taxa:\t106.0\n",
      "Loci shared by 100 taxa:\t106.0\n",
      "Loci shared by 101 taxa:\t106.0\n",
      "Loci shared by 102 taxa:\t106.0\n",
      "Loci shared by 103 taxa:\t106.0\n",
      "Loci shared by 104 taxa:\t106.0\n",
      "Loci shared by 105 taxa:\t106.0\n",
      "Loci shared by 106 taxa:\t105.0\n",
      "Loci shared by 107 taxa:\t105.0\n",
      "Loci shared by 108 taxa:\t105.0\n",
      "Loci shared by 109 taxa:\t105.0\n",
      "Loci shared by 110 taxa:\t104.0\n",
      "Loci shared by 111 taxa:\t103.0\n",
      "Loci shared by 112 taxa:\t103.0\n",
      "Loci shared by 113 taxa:\t103.0\n",
      "Loci shared by 114 taxa:\t103.0\n",
      "Loci shared by 115 taxa:\t103.0\n",
      "Loci shared by 116 taxa:\t103.0\n",
      "Loci shared by 117 taxa:\t103.0\n",
      "Loci shared by 118 taxa:\t103.0\n",
      "Loci shared by 119 taxa:\t103.0\n",
      "Loci shared by 120 taxa:\t103.0\n",
      "Loci shared by 121 taxa:\t101.0\n",
      "Loci shared by 122 taxa:\t101.0\n",
      "Loci shared by 123 taxa:\t101.0\n",
      "Loci shared by 124 taxa:\t101.0\n",
      "Loci shared by 125 taxa:\t101.0\n",
      "Loci shared by 126 taxa:\t100.0\n",
      "Loci shared by 127 taxa:\t99.0\n",
      "Loci shared by 128 taxa:\t99.0\n",
      "Loci shared by 129 taxa:\t97.0\n",
      "Loci shared by 130 taxa:\t96.0\n",
      "Loci shared by 131 taxa:\t92.0\n",
      "Loci shared by 132 taxa:\t92.0\n",
      "Loci shared by 133 taxa:\t76.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/bacillus/extract_probes_from_group/multifastas.sqlite  --base-taxon Bacillus_anthracis_GCF_000008165 --output results/phyluce/bacillus/extract_probes_from_group/Bacillus_anthracis_GCF_000008165+132-back-to-133.conf --specific-counts 133;\n",
      "Counter({'bacillus_pseudomycoides_gcf_002555685': 76, 'bacillus_albus_gcf_004116415': 76, 'bacillus_paranthracis_gca_009873395': 76, 'bacillus_pseudomycoides_gcf_002556205': 76, 'bacillus_mycoides_gcf_001757985': 76, 'bacillus_thuringiensis_gcf_002564505': 76, 'bacillus_anthracis_gcf_000697555': 76, 'bacillus_nitratireducens_gcf_007676605': 76, 'bacillus_paranthracis_gcf_009799965': 76, 'bacillus_albus_gcf_003966295': 76, 'bacillus_thuringiensis_gcf_000161735': 76, 'bacillus_cereus_gcf_002530565': 76, 'bacillus_nitratireducens_gcf_007681065': 76, 'bacillus_cereus_gcf_000161315': 76, 'bacillus_nitratireducens_gcf_007676595': 76, 'bacillus_luti_gcf_001884105': 76, 'bacillus_cereus_gcf_001584025': 76, 'bacillus_cereus_gcf_000290795': 76, 'bacillus_cytotoxicus_gcf_002250945': 76, 'bacillus_cereus_gcf_001583875': 76, 'bacillus_mobilis_gcf_007682195': 76, 'bacillus_mycoides_gcf_001541985': 76, 'bacillus_pacificus_gcf_001884025': 76, 'bacillus_toyonensis_gcf_002551965': 76, 'bacillus_albus_gcf_004116035': 76, 'bacillus_mycoides_gcf_001757915': 76, 'bacillus_paranthracis_gcf_005848985': 76, 'bacillus_pacificus_gca_009884315': 76, 'bacillus_albus_gcf_004153665': 76, 'bacillus_nitratireducens_gcf_007681365': 76, 'bacillus_thuringiensis_gcf_001402735': 76, 'bacillus_luti_gcf_008923665': 76, 'bacillus_cytotoxicus_gcf_001050335': 76, 'bacillus_cereus_gcf_002565765': 76, 'bacillus_wiedmannii_gcf_002556175': 76, 'bacillus_wiedmannii_gcf_001645395': 76, 'bacillus_cytotoxicus_gcf_002250925': 76, 'bacillus_pseudomycoides_gcf_002555615': 76, 'bacillus_albus_gcf_004116025': 76, 'bacillus_wiedmannii_gcf_002569595': 76, 'bacillus_cytotoxicus_gcf_002250885': 76, 'bacillus_albus_gcf_009740005': 76, 'bacillus_pacificus_gcf_006349595': 76, 'bacillus_toyonensis_gcf_002555905': 76, 'bacillus_anthracis_gcf_000558965': 76, 'bacillus_anthracis_gcf_000742875': 76, 'bacillus_toyonensis_gcf_002581955': 76, 'bacillus_luti_gcf_009739945': 76, 'bacillus_tropicus_gcf_007682405': 76, 'bacillus_thuringiensis_gcf_001675515': 76, 'bacillus_anthracis_gcf_000783055': 76, 'bacillus_toyonensis_gcf_002550925': 76, 'bacillus_wiedmannii_gcf_002014635': 76, 'bacillus_nitratireducens_gcf_007682355': 76, 'bacillus_mycoides_gcf_000290695': 76, 'bacillus_pseudomycoides_gcf_002555995': 76, 'bacillus_paranthracis_gcf_007682155': 76, 'bacillus_cytotoxicus_gcf_002251005': 76, 'bacillus_anthracis_gcf_001883895': 76, 'bacillus_cytotoxicus_gcf_002250905': 76, 'bacillus_paranthracis_gca_009648955': 76, 'bacillus_mycoides_gcf_001044935': 76, 'bacillus_albus_gcf_004116085': 76, 'bacillus_mycoides_gcf_000018825': 76, 'bacillus_proteolyticus_gcf_001884065': 76, 'bacillus_mycoides_gcf_000517985': 76, 'bacillus_paranthracis_gca_009884335': 76, 'bacillus_toyonensis_gcf_002581325': 76, 'bacillus_mycoides_gcf_001757825': 76, 'bacillus_thuringiensis_gcf_002559825': 76, 'bacillus_mobilis_gcf_007681185': 76, 'bacillus_pseudomycoides_gcf_002550095': 76, 'bacillus_thuringiensis_gcf_002148135': 76, 'bacillus_cereus_gcf_000291075': 76, 'bacillus_toyonensis_gcf_002554865': 76, 'bacillus_toyonensis_gcf_002570155': 76, 'bacillus_pseudomycoides_gcf_000746965': 76, 'bacillus_pseudomycoides_gcf_000161435': 76, 'bacillus_tropicus_gcf_007672275': 76, 'bacillus_tropicus_gcf_007676425': 76, 'bacillus_tropicus_gcf_007682035': 76, 'bacillus_cereus_gcf_002560715': 76, 'bacillus_pseudomycoides_gcf_002557395': 76, 'bacillus_anthracis_gcf_002525695': 76, 'bacillus_cereus_gcf_001683695': 76, 'bacillus_paranthracis_gcf_007682135': 76, 'bacillus_thuringiensis_gcf_002551895': 76, 'bacillus_cereus_gcf_002552015': 76, 'bacillus_anthracis_gcf_000559005': 76, 'bacillus_mobilis_gcf_007681195': 76, 'bacillus_tropicus_gcf_001884035': 76, 'bacillus_gaemokensis_gcf_000712615': 76, 'bacillus_gaemokensis_gcf_001590835': 76, 'bacillus_bombysepticus_gcf_000831065': 76, 'bacillus_cytotoxicus_gcf_002251025': 76, 'bacillus_pacificus_gcf_003858675': 76, 'bacillus_wiedmannii_gcf_002555555': 76, 'bacillus_wiedmannii_gcf_002555855': 76, 'bacillus_nitratireducens_gcf_001884135': 76, 'bacillus_anthracis_gcf_000008165': 76, 'bacillus_toyonensis_gcf_002556675': 76, 'bacillus_paranthracis_gcf_005117095': 76, 'bacillus_albus_gcf_004116065': 76, 'bacillus_paramycoides_gcf_001884235': 76, 'bacillus_toyonensis_gcf_002580355': 76, 'bacillus_mobilis_gcf_003612955': 76, 'bacillus_thuringiensis_gcf_002570925': 76, 'bacillus_cytotoxicus_gcf_000017425': 76, 'bacillus_albus_gcf_001884185': 76, 'bacillus_tropicus_gcf_006349625': 76, 'bacillus_thuringiensis_gcf_000710255': 76, 'bacillus_thuringiensis_gcf_002562665': 76, 'bacillus_wiedmannii_gcf_002556405': 76, 'bacillus_mobilis_gcf_001884045': 76, 'bacillus_wiedmannii_gcf_002555945': 76, 'bacillus_mycoides_gcf_002014505': 76, 'bacillus_cytotoxicus_gcf_002251045': 76, 'bacillus_cytotoxicus_gcf_002250965': 76, 'bacillus_anthracis_gcf_002525705': 76, 'bacillus_toyonensis_gca_009799785': 76, 'bacillus_wiedmannii_gcf_000160955': 76, 'bacillus_tropicus_gcf_006457285': 76, 'bacillus_tropicus_gcf_006349645': 76, 'bacillus_paranthracis_gcf_001883995': 76, 'bacillus_anthracis_gcf_002233635': 76, 'bacillus_paranthracis_gcf_009498675': 76, 'bacillus_paranthracis_gcf_004307985': 76, 'bacillus_wiedmannii_gcf_002555735': 76, 'bacillus_pseudomycoides_gcf_000161415': 76, 'bacillus_pseudomycoides_gcf_002569075': 76, 'bacillus_mycoides_gcf_002014695': 76, 'bacillus_paramycoides_gcf_007682005': 76, 'bacillus_manliponensis_gcf_000712595': 76})\n",
      "Total loci = 76\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(133)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/bacillus/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/bacillus/extract_probes_from_group/Bacillus_anthracis_GCF_000008165+132-back-to-133.conf --probe-prefix uce_bacillus_ --designer rnplattii --design bacillus_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGNNNNGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 76\n",
      "Probe Count = 19200\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list.fasta --query results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Fri Feb 07, 2020  15:45:24\n",
      "Ended:  Fri Feb 07, 2020  15:49:31\n",
      "Time for execution:  4.11266578436 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list.fasta --lastz results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_bacillus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 19199 fasta sequences.  Filtered 0 duplicates. Kept 19200.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/bacillus/final_probe_design/bacillus_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Fri Feb  7 17:36:29 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 19200\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1536000\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 4M\n",
      "Buffer          : 4 X 12M = 49M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 91M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3932319\n",
      "Max number of word counting entries: 88516514\n",
      "\n",
      "# comparing sequences from          0  to       3200\n",
      "...---------- new table with      262 representatives\n",
      "# comparing sequences from       3200  to       5866\n",
      "----------   1620 remaining sequences to the next cycle\n",
      "---------- new table with      119 representatives\n",
      "# comparing sequences from       4246  to       6738\n",
      "----------   1318 remaining sequences to the next cycle\n",
      "---------- new table with      101 representatives\n",
      "# comparing sequences from       5420  to       7716\n",
      "----------   1119 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       6597  to       8697\n",
      "----------   1057 remaining sequences to the next cycle\n",
      "---------- new table with      126 representatives\n",
      "# comparing sequences from       7640  to       9566\n",
      "----------    127 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       9439  to      11065\n",
      "..........    10000  finished        874  clusters\n",
      "----------    711 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      10354  to      11828\n",
      "----------    655 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      11173  to      12510\n",
      "----------    355 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      12155  to      13329\n",
      "....................---------- new table with       97 representatives\n",
      "# comparing sequences from      13329  to      14307\n",
      "....................---------- new table with       53 representatives\n",
      "# comparing sequences from      14307  to      15122\n",
      ".....................---------- new table with       73 representatives\n",
      "# comparing sequences from      15122  to      15801\n",
      "....................---------- new table with       67 representatives\n",
      "# comparing sequences from      15801  to      16367\n",
      ".....................---------- new table with       73 representatives\n",
      "# comparing sequences from      16367  to      16839\n",
      "....................---------- new table with       54 representatives\n",
      "# comparing sequences from      16839  to      17232\n",
      ".....................---------- new table with       56 representatives\n",
      "# comparing sequences from      17232  to      17560\n",
      "...................---------- new table with       35 representatives\n",
      "# comparing sequences from      17560  to      17833\n",
      "...................---------- new table with       39 representatives\n",
      "# comparing sequences from      17833  to      18060\n",
      "100.0%---------- new table with       19 representatives\n",
      "# comparing sequences from      18060  to      18250\n",
      "99.9%---------- new table with       16 representatives\n",
      "# comparing sequences from      18250  to      19200\n",
      "....................---------- new table with       87 representatives\n",
      "\n",
      "    19200  finished       1777  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 92M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 2.95\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Borrelia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'borrelia'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Borrelia_anserina_GCF_001936255\",\n",
    "               \"Borrelia_coriaceae_GCF_000568755\",\n",
    "               \"Borrelia_crocidurae_GCF_000259345\",\n",
    "               \"Borrelia_duttonii_GCF_000019685\",\n",
    "               \"Borrelia_hermsii_GCF_001660005\",\n",
    "               \"Borrelia_hispnica_GCF_000500065\",\n",
    "               \"Borrelia_miyamotoi_GCF_000445425\",\n",
    "               \"Borrelia_parkeri_GCF_000512145\",\n",
    "               \"Borrelia_persica_GCF_000500045\",\n",
    "               \"Borrelia_recurrentis_GCF_000019705\",\n",
    "               \"Borrelia_turcica_GCF_003606285\",\n",
    "               \"Borrelia_turicatae_GCF_000012085\",\n",
    "               \"Borreliella_afzelii_GCA_000502195\",\n",
    "               \"Borreliella_garinii_GCF_000501735\" ]\n",
    "\n",
    "reference_taxon = \"Borreliella_burgdorferi_GCF_000502155\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001936255.1_ASM193625v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 297132 bytes  198116.00 bytes/sec\n",
      "total size is 296946  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000568755.1_ASM56875v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 488227 bytes  325512.67 bytes/sec\n",
      "total size is 488002  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000259345.1_ASM25934v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 411363 bytes  274270.00 bytes/sec\n",
      "total size is 411154  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000019685.1_ASM1968v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 437507 bytes  175019.60 bytes/sec\n",
      "total size is 437291  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001660005.1_ASM166000v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 392989 bytes  262020.67 bytes/sec\n",
      "total size is 392787  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000500065.1_AB2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 497029 bytes  331380.67 bytes/sec\n",
      "total size is 496803  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000445425.4_ASM44542v4_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 267309 bytes  178234.00 bytes/sec\n",
      "total size is 267132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000512145.2_ASM51214v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 293811 bytes  195902.00 bytes/sec\n",
      "total size is 293634  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000500045.1_AB6.5.1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 500481 bytes  333682.00 bytes/sec\n",
      "total size is 500251  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000019705.1_ASM1970v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 349522 bytes  233042.67 bytes/sec\n",
      "total size is 349330  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003606285.1_ASM360628v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 365396 bytes  243625.33 bytes/sec\n",
      "total size is 365194  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000012085.2_ASM1208v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 329927 bytes  659938.00 bytes/sec\n",
      "total size is 329735  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000502195.1_BorAfzIPT138_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 349542 bytes  699168.00 bytes/sec\n",
      "total size is 349347  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000502155.1_BorBurgIPT141_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 340226 bytes  226845.33 bytes/sec\n",
      "total size is 340030  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000501735.1_BorGarIPT108_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 352859 bytes  235267.33 bytes/sec\n",
      "total size is 352664  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481461 (\"sim_Borrelia_anserina_GCF_001936255\") has been submitted\n",
      "Your job 5481462 (\"sim_Borrelia_coriaceae_GCF_000568755\") has been submitted\n",
      "Your job 5481463 (\"sim_Borrelia_crocidurae_GCF_000259345\") has been submitted\n",
      "Your job 5481464 (\"sim_Borrelia_duttonii_GCF_000019685\") has been submitted\n",
      "Your job 5481465 (\"sim_Borrelia_hermsii_GCF_001660005\") has been submitted\n",
      "Your job 5481466 (\"sim_Borrelia_hispnica_GCF_000500065\") has been submitted\n",
      "Your job 5481467 (\"sim_Borrelia_miyamotoi_GCF_000445425\") has been submitted\n",
      "Your job 5481468 (\"sim_Borrelia_parkeri_GCF_000512145\") has been submitted\n",
      "Your job 5481469 (\"sim_Borrelia_persica_GCF_000500045\") has been submitted\n",
      "Your job 5481470 (\"sim_Borrelia_recurrentis_GCF_000019705\") has been submitted\n",
      "Your job 5481471 (\"sim_Borrelia_turcica_GCF_003606285\") has been submitted\n",
      "Your job 5481472 (\"sim_Borrelia_turicatae_GCF_000012085\") has been submitted\n",
      "Your job 5481473 (\"sim_Borreliella_afzelii_GCA_000502195\") has been submitted\n",
      "Your job 5481474 (\"sim_Borreliella_garinii_GCF_000501735\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/borrelia/cleaned_genomes/Borreliella_burgdorferi_GCF_000502155_formatted.fas path=results/phyluce/borrelia/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481475 (\"map_Borrelia_anserina_GCF_001936255\") has been submitted\n",
      "Your job 5481476 (\"map_Borrelia_coriaceae_GCF_000568755\") has been submitted\n",
      "Your job 5481477 (\"map_Borrelia_crocidurae_GCF_000259345\") has been submitted\n",
      "Your job 5481478 (\"map_Borrelia_duttonii_GCF_000019685\") has been submitted\n",
      "Your job 5481479 (\"map_Borrelia_hermsii_GCF_001660005\") has been submitted\n",
      "Your job 5481480 (\"map_Borrelia_hispnica_GCF_000500065\") has been submitted\n",
      "Your job 5481481 (\"map_Borrelia_miyamotoi_GCF_000445425\") has been submitted\n",
      "Your job 5481482 (\"map_Borrelia_parkeri_GCF_000512145\") has been submitted\n",
      "Your job 5481483 (\"map_Borrelia_persica_GCF_000500045\") has been submitted\n",
      "Your job 5481484 (\"map_Borrelia_recurrentis_GCF_000019705\") has been submitted\n",
      "Your job 5481485 (\"map_Borrelia_turcica_GCF_003606285\") has been submitted\n",
      "Your job 5481486 (\"map_Borrelia_turicatae_GCF_000012085\") has been submitted\n",
      "Your job 5481487 (\"map_Borreliella_afzelii_GCA_000502195\") has been submitted\n",
      "Your job 5481488 (\"map_Borreliella_garinii_GCF_000501735\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481489 (\"merge_Borrelia_anserina_GCF_001936255\") has been submitted\n",
      "Your job 5481490 (\"merge_Borrelia_coriaceae_GCF_000568755\") has been submitted\n",
      "Your job 5481491 (\"merge_Borrelia_crocidurae_GCF_000259345\") has been submitted\n",
      "Your job 5481492 (\"merge_Borrelia_duttonii_GCF_000019685\") has been submitted\n",
      "Your job 5481493 (\"merge_Borrelia_hermsii_GCF_001660005\") has been submitted\n",
      "Your job 5481494 (\"merge_Borrelia_hispnica_GCF_000500065\") has been submitted\n",
      "Your job 5481495 (\"merge_Borrelia_miyamotoi_GCF_000445425\") has been submitted\n",
      "Your job 5481496 (\"merge_Borrelia_parkeri_GCF_000512145\") has been submitted\n",
      "Your job 5481497 (\"merge_Borrelia_persica_GCF_000500045\") has been submitted\n",
      "Your job 5481498 (\"merge_Borrelia_recurrentis_GCF_000019705\") has been submitted\n",
      "Your job 5481499 (\"merge_Borrelia_turcica_GCF_003606285\") has been submitted\n",
      "Your job 5481500 (\"merge_Borrelia_turicatae_GCF_000012085\") has been submitted\n",
      "Your job 5481501 (\"merge_Borreliella_afzelii_GCA_000502195\") has been submitted\n",
      "Your job 5481502 (\"merge_Borreliella_garinii_GCF_000501735\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 176 sequences from Borrelia_anserina_GCF_001936255_merged.bed.  Filtered 142 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 34.\n",
      "Screened 187 sequences from Borrelia_coriaceae_GCF_000568755_merged.bed.  Filtered 159 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 28.\n",
      "Screened 216 sequences from Borrelia_crocidurae_GCF_000259345_merged.bed.  Filtered 166 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 50.\n",
      "Screened 220 sequences from Borrelia_duttonii_GCF_000019685_merged.bed.  Filtered 176 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 44.\n",
      "Screened 198 sequences from Borrelia_hermsii_GCF_001660005_merged.bed.  Filtered 160 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 38.\n",
      "Screened 227 sequences from Borrelia_hispnica_GCF_000500065_merged.bed.  Filtered 182 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 45.\n",
      "Screened 187 sequences from Borrelia_miyamotoi_GCF_000445425_merged.bed.  Filtered 145 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 42.\n",
      "Screened 233 sequences from Borrelia_parkeri_GCF_000512145_merged.bed.  Filtered 195 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 38.\n",
      "Screened 204 sequences from Borrelia_persica_GCF_000500045_merged.bed.  Filtered 168 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 36.\n",
      "Screened 220 sequences from Borrelia_recurrentis_GCF_000019705_merged.bed.  Filtered 168 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 201 sequences from Borrelia_turcica_GCF_003606285_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 202 sequences from Borrelia_turicatae_GCF_000012085_merged.bed.  Filtered 159 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 1216 sequences from Borreliella_afzelii_GCA_000502195_merged.bed.  Filtered 275 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 941.\n",
      "Screened 1221 sequences from Borreliella_garinii_GCF_000501735_merged.bed.  Filtered 274 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 947.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/borrelia/initial_intervals/Borreliella_garinii_GCF_000501735_merged.bed --twobit results/phyluce/borrelia/cleaned_genomes/Borreliella_burgdorferi_GCF_000502155_formatted.2bit --output results/phyluce/borrelia/initial_intervals/Borreliella_garinii_GCF_000501735_stripped.bed;\n",
      "borrelia_anserina_gcf_001936255.\n",
      "borrelia_coriaceae_gcf_000568755.\n",
      "borrelia_crocidurae_gcf_000259345.\n",
      "borrelia_duttonii_gcf_000019685.\n",
      "borrelia_hermsii_gcf_001660005.\n",
      "borrelia_hispnica_gcf_000500065.\n",
      "borrelia_miyamotoi_gcf_000445425.\n",
      "borrelia_parkeri_gcf_000512145.\n",
      "borrelia_persica_gcf_000500045.\n",
      "borrelia_recurrentis_gcf_000019705.\n",
      "borrelia_turcica_gcf_003606285.\n",
      "borrelia_turicatae_gcf_000012085.\n",
      "borreliella_afzelii_gca_000502195.\n",
      "borreliella_garinii_gcf_000501735.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/borrelia/initial_intervals/Borreliella_garinii_GCF_000501735_merged.bed --twobit results/phyluce/borrelia/cleaned_genomes/Borreliella_burgdorferi_GCF_000502155_formatted.2bit --output results/phyluce/borrelia/initial_intervals/Borreliella_garinii_GCF_000501735_stripped.bed;\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 0 taxa:\t1,077.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 1 taxa:\t1,077.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 2 taxa:\t928.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 3 taxa:\t128.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 4 taxa:\t86.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 5 taxa:\t68.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 6 taxa:\t50.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 7 taxa:\t39.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 8 taxa:\t30.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 9 taxa:\t21.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 10 taxa:\t15.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 11 taxa:\t11.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 12 taxa:\t10.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 13 taxa:\t7.0\n",
      "Loci shared by Borreliella_burgdorferi_GCF_000502155 + 14 taxa:\t3.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/borrelia/initial_intervals/borrelia-to-Borreliella_burgdorferi_GCF_000502155.sqlite --base-taxon Borreliella_burgdorferi_GCF_000502155 --output results/phyluce/borrelia/initial_intervals/Borreliella_burgdorferi_GCF_000502155_+4.bed --specific-counts 4;\n",
      "Counter({'borreliella_afzelii_gca_000502195': 85, 'borreliella_garinii_gcf_000501735': 83, 'borrelia_recurrentis_gcf_000019705': 47, 'borrelia_crocidurae_gcf_000259345': 46, 'borrelia_hispnica_gcf_000500065': 41, 'borrelia_turicatae_gcf_000012085': 40, 'borrelia_duttonii_gcf_000019685': 39, 'borrelia_miyamotoi_gcf_000445425': 37, 'borrelia_parkeri_gcf_000512145': 35, 'borrelia_hermsii_gcf_001660005': 32, 'borrelia_persica_gcf_000500045': 29, 'borrelia_turcica_gcf_003606285': 29, 'borrelia_anserina_gcf_001936255': 28, 'borrelia_coriaceae_gcf_000568755': 27})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 4\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/borrelia/initial_intervals/Borreliella_burgdorferi_GCF_000502155_+4.bed --twobit results/phyluce/borrelia/cleaned_genomes/Borreliella_burgdorferi_GCF_000502155_formatted.2bit --buffer-to 160 --output results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4.fasta;\n",
      "Screened 86 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 86.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4.fasta --probe-prefix uce_borrelia_ --design borrelia_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 68\n",
      "Probe Count = 125\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas --query results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  16:33:56\n",
      "Ended:  Thu Feb 06, 2020  16:33:57\n",
      "Time for execution:  0.0021835009257 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas                        --lastz results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes_vself.lastz                       --probe-prefix=uce_borrelia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 124 fasta sequences.  Filtered 0 duplicates. Kept 125.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/borrelia/validate_intervals/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas --scaffoldlist Borrelia_anserina_GCF_001936255 Borrelia_coriaceae_GCF_000568755 Borrelia_crocidurae_GCF_000259345 Borrelia_duttonii_GCF_000019685 Borrelia_hermsii_GCF_001660005 Borrelia_hispnica_GCF_000500065 Borrelia_miyamotoi_GCF_000445425 Borrelia_parkeri_GCF_000512145 Borrelia_persica_GCF_000500045 Borrelia_recurrentis_GCF_000019705 Borrelia_turcica_GCF_003606285 Borrelia_turicatae_GCF_000012085 Borreliella_afzelii_GCA_000502195 Borreliella_garinii_GCF_000501735 Borreliella_burgdorferi_GCF_000502155 --genome-base-path results/phyluce/borrelia/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/borrelia/validate_intervals/borrelia-to-Borreliella_burgdorferi_GCF_000502155.sqlite --output results/phyluce/borrelia/validate_intervals/lastz/;\n",
      "\n",
      "Running against Borrelia_anserina_GCF_001936255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkPLiH8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIEPLJv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_anserina_GCF_001936255.lastz\n",
      "Creating Borrelia_anserina_GCF_001936255 table\n",
      "Inserting data to Borrelia_anserina_GCF_001936255 table\n",
      "\n",
      "Running against Borrelia_coriaceae_GCF_000568755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWcT2CE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpYMOo3Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_coriaceae_GCF_000568755.lastz\n",
      "Creating Borrelia_coriaceae_GCF_000568755 table\n",
      "Inserting data to Borrelia_coriaceae_GCF_000568755 table\n",
      "\n",
      "Running against Borrelia_crocidurae_GCF_000259345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7lmhEk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpa4ciPd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_crocidurae_GCF_000259345.lastz\n",
      "Creating Borrelia_crocidurae_GCF_000259345 table\n",
      "Inserting data to Borrelia_crocidurae_GCF_000259345 table\n",
      "\n",
      "Running against Borrelia_duttonii_GCF_000019685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp90VrH5.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdgvMon.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_duttonii_GCF_000019685.lastz\n",
      "Creating Borrelia_duttonii_GCF_000019685 table\n",
      "Inserting data to Borrelia_duttonii_GCF_000019685 table\n",
      "\n",
      "Running against Borrelia_hermsii_GCF_001660005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpUNKEJz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBp_e1o.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_hermsii_GCF_001660005.lastz\n",
      "Creating Borrelia_hermsii_GCF_001660005 table\n",
      "Inserting data to Borrelia_hermsii_GCF_001660005 table\n",
      "\n",
      "Running against Borrelia_hispnica_GCF_000500065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprH3G5A.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfQAEMl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_hispnica_GCF_000500065.lastz\n",
      "Creating Borrelia_hispnica_GCF_000500065 table\n",
      "Inserting data to Borrelia_hispnica_GCF_000500065 table\n",
      "\n",
      "Running against Borrelia_miyamotoi_GCF_000445425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpD5RG8V.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpr9UMIq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_miyamotoi_GCF_000445425.lastz\n",
      "Creating Borrelia_miyamotoi_GCF_000445425 table\n",
      "Inserting data to Borrelia_miyamotoi_GCF_000445425 table\n",
      "\n",
      "Running against Borrelia_parkeri_GCF_000512145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphfqOoU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvsvjNX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_parkeri_GCF_000512145.lastz\n",
      "Creating Borrelia_parkeri_GCF_000512145 table\n",
      "Inserting data to Borrelia_parkeri_GCF_000512145 table\n",
      "\n",
      "Running against Borrelia_persica_GCF_000500045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsRvkKG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1WYBDt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_persica_GCF_000500045.lastz\n",
      "Creating Borrelia_persica_GCF_000500045 table\n",
      "Inserting data to Borrelia_persica_GCF_000500045 table\n",
      "\n",
      "Running against Borrelia_recurrentis_GCF_000019705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRSWG7j.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6qglBH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_recurrentis_GCF_000019705.lastz\n",
      "Creating Borrelia_recurrentis_GCF_000019705 table\n",
      "Inserting data to Borrelia_recurrentis_GCF_000019705 table\n",
      "\n",
      "Running against Borrelia_turcica_GCF_003606285.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjJojPl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXlLHub.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_turcica_GCF_003606285.lastz\n",
      "Creating Borrelia_turcica_GCF_003606285 table\n",
      "Inserting data to Borrelia_turcica_GCF_003606285 table\n",
      "\n",
      "Running against Borrelia_turicatae_GCF_000012085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbArzN_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8cwS4j.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borrelia_turicatae_GCF_000012085.lastz\n",
      "Creating Borrelia_turicatae_GCF_000012085 table\n",
      "Inserting data to Borrelia_turicatae_GCF_000012085 table\n",
      "\n",
      "Running against Borreliella_afzelii_GCA_000502195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpomeYFp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2ONyCm.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borreliella_afzelii_GCA_000502195.lastz\n",
      "Creating Borreliella_afzelii_GCA_000502195 table\n",
      "Inserting data to Borreliella_afzelii_GCA_000502195 table\n",
      "\n",
      "Running against Borreliella_garinii_GCF_000501735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpBpRSNI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoVuuS2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borreliella_garinii_GCF_000501735.lastz\n",
      "Creating Borreliella_garinii_GCF_000501735 table\n",
      "Inserting data to Borreliella_garinii_GCF_000501735 table\n",
      "\n",
      "Running against Borreliella_burgdorferi_GCF_000502155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIGkiX8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1LmNOt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/borrelia/validate_intervals/lastz/Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_Borreliella_burgdorferi_GCF_000502155.lastz\n",
      "Creating Borreliella_burgdorferi_GCF_000502155 table\n",
      "Inserting data to Borreliella_burgdorferi_GCF_000502155 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/borrelia/extract_probes_from_group/borrelia_genome.conf --lastz results/phyluce/borrelia/validate_intervals/lastz --probes 120 --probe-prefix uce_borrelia_ --name-pattern \"Borreliella_burgdorferi_GCF_000502155_+4_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/borrelia/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 16:34:09,000 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 16:34:09,003 - Phyluce - INFO - ------- Working on Borrelia_anserina_GCF_001936255 genome -------\n",
      "2020-02-06 16:34:09,018 - Phyluce - INFO - Reading Borrelia_anserina_GCF_001936255 genome\n",
      "2020-02-06 16:34:09,287 - Phyluce - INFO - Borrelia_anserina_GCF_001936255: 67 uces, 0 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:09,287 - Phyluce - INFO - ------- Working on Borrelia_coriaceae_GCF_000568755 genome ------\n",
      "2020-02-06 16:34:09,288 - Phyluce - INFO - Reading Borrelia_coriaceae_GCF_000568755 genome\n",
      "2020-02-06 16:34:09,497 - Phyluce - INFO - Borrelia_coriaceae_GCF_000568755: 68 uces, 0 dupes, 68 non-dupes, 0 orient drop, 3 length drop, 65 written\n",
      "2020-02-06 16:34:09,497 - Phyluce - INFO - ------ Working on Borrelia_crocidurae_GCF_000259345 genome ------\n",
      "2020-02-06 16:34:09,510 - Phyluce - INFO - Reading Borrelia_crocidurae_GCF_000259345 genome\n",
      "2020-02-06 16:34:09,730 - Phyluce - INFO - Borrelia_crocidurae_GCF_000259345: 68 uces, 0 dupes, 68 non-dupes, 1 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:09,730 - Phyluce - INFO - ------- Working on Borrelia_duttonii_GCF_000019685 genome -------\n",
      "2020-02-06 16:34:09,738 - Phyluce - INFO - Reading Borrelia_duttonii_GCF_000019685 genome\n",
      "2020-02-06 16:34:09,956 - Phyluce - INFO - Borrelia_duttonii_GCF_000019685: 68 uces, 1 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:09,956 - Phyluce - INFO - -------- Working on Borrelia_hermsii_GCF_001660005 genome -------\n",
      "2020-02-06 16:34:09,957 - Phyluce - INFO - Reading Borrelia_hermsii_GCF_001660005 genome\n",
      "2020-02-06 16:34:10,173 - Phyluce - INFO - Borrelia_hermsii_GCF_001660005: 68 uces, 1 dupes, 67 non-dupes, 1 orient drop, 2 length drop, 64 written\n",
      "2020-02-06 16:34:10,173 - Phyluce - INFO - ------- Working on Borrelia_hispnica_GCF_000500065 genome -------\n",
      "2020-02-06 16:34:10,174 - Phyluce - INFO - Reading Borrelia_hispnica_GCF_000500065 genome\n",
      "2020-02-06 16:34:10,388 - Phyluce - INFO - Borrelia_hispnica_GCF_000500065: 68 uces, 1 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:10,389 - Phyluce - INFO - ------- Working on Borrelia_miyamotoi_GCF_000445425 genome ------\n",
      "2020-02-06 16:34:10,389 - Phyluce - INFO - Reading Borrelia_miyamotoi_GCF_000445425 genome\n",
      "2020-02-06 16:34:10,595 - Phyluce - INFO - Borrelia_miyamotoi_GCF_000445425: 67 uces, 0 dupes, 67 non-dupes, 0 orient drop, 2 length drop, 65 written\n",
      "2020-02-06 16:34:10,595 - Phyluce - INFO - -------- Working on Borrelia_parkeri_GCF_000512145 genome -------\n",
      "2020-02-06 16:34:10,612 - Phyluce - INFO - Reading Borrelia_parkeri_GCF_000512145 genome\n",
      "2020-02-06 16:34:10,828 - Phyluce - INFO - Borrelia_parkeri_GCF_000512145: 67 uces, 0 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:10,828 - Phyluce - INFO - -------- Working on Borrelia_persica_GCF_000500045 genome -------\n",
      "2020-02-06 16:34:10,829 - Phyluce - INFO - Reading Borrelia_persica_GCF_000500045 genome\n",
      "2020-02-06 16:34:11,034 - Phyluce - INFO - Borrelia_persica_GCF_000500045: 67 uces, 0 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:11,035 - Phyluce - INFO - ------ Working on Borrelia_recurrentis_GCF_000019705 genome -----\n",
      "2020-02-06 16:34:11,045 - Phyluce - INFO - Reading Borrelia_recurrentis_GCF_000019705 genome\n",
      "2020-02-06 16:34:11,273 - Phyluce - INFO - Borrelia_recurrentis_GCF_000019705: 68 uces, 1 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:11,273 - Phyluce - INFO - -------- Working on Borrelia_turcica_GCF_003606285 genome -------\n",
      "2020-02-06 16:34:11,274 - Phyluce - INFO - Reading Borrelia_turcica_GCF_003606285 genome\n",
      "2020-02-06 16:34:11,477 - Phyluce - INFO - Borrelia_turcica_GCF_003606285: 67 uces, 0 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:11,477 - Phyluce - INFO - ------- Working on Borrelia_turicatae_GCF_000012085 genome ------\n",
      "2020-02-06 16:34:11,478 - Phyluce - INFO - Reading Borrelia_turicatae_GCF_000012085 genome\n",
      "2020-02-06 16:34:11,683 - Phyluce - INFO - Borrelia_turicatae_GCF_000012085: 67 uces, 0 dupes, 67 non-dupes, 0 orient drop, 1 length drop, 66 written\n",
      "2020-02-06 16:34:11,683 - Phyluce - INFO - ------ Working on Borreliella_afzelii_GCA_000502195 genome ------\n",
      "2020-02-06 16:34:11,684 - Phyluce - INFO - Reading Borreliella_afzelii_GCA_000502195 genome\n",
      "2020-02-06 16:34:11,904 - Phyluce - INFO - Borreliella_afzelii_GCA_000502195: 68 uces, 4 dupes, 64 non-dupes, 0 orient drop, 0 length drop, 64 written\n",
      "2020-02-06 16:34:11,904 - Phyluce - INFO - ------ Working on Borreliella_garinii_GCF_000501735 genome ------\n",
      "2020-02-06 16:34:11,905 - Phyluce - INFO - Reading Borreliella_garinii_GCF_000501735 genome\n",
      "2020-02-06 16:34:12,117 - Phyluce - INFO - Borreliella_garinii_GCF_000501735: 68 uces, 2 dupes, 66 non-dupes, 0 orient drop, 1 length drop, 65 written\n",
      "2020-02-06 16:34:12,117 - Phyluce - INFO - ---- Working on Borreliella_burgdorferi_GCF_000502155 genome ----\n",
      "2020-02-06 16:34:12,118 - Phyluce - INFO - Reading Borreliella_burgdorferi_GCF_000502155 genome\n",
      "2020-02-06 16:34:12,340 - Phyluce - INFO - Borreliella_burgdorferi_GCF_000502155: 68 uces, 1 dupes, 67 non-dupes, 0 orient drop, 2 length drop, 65 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/borrelia/extract_probes_from_group/probe_fasta --output results/phyluce/borrelia/extract_probes_from_group/multifastas.sqlite --base-taxon Borreliella_burgdorferi_GCF_000502155;\n",
      "borrelia_anserina_gcf_001936255.\n",
      "borrelia_coriaceae_gcf_000568755.\n",
      "borrelia_crocidurae_gcf_000259345.\n",
      "borrelia_duttonii_gcf_000019685.\n",
      "borrelia_hermsii_gcf_001660005.\n",
      "borrelia_hispnica_gcf_000500065.\n",
      "borrelia_miyamotoi_gcf_000445425.\n",
      "borrelia_parkeri_gcf_000512145.\n",
      "borrelia_persica_gcf_000500045.\n",
      "borrelia_recurrentis_gcf_000019705.\n",
      "borrelia_turcica_gcf_003606285.\n",
      "borrelia_turicatae_gcf_000012085.\n",
      "borreliella_afzelii_gca_000502195.\n",
      "borreliella_garinii_gcf_000501735.\n",
      "borreliella_burgdorferi_gcf_000502155.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/borrelia/extract_probes_from_group/multifastas.sqlite --base-taxon Borreliella_burgdorferi_GCF_000502155;\n",
      "Loci shared by 0 taxa:\t67.0\n",
      "Loci shared by 1 taxa:\t67.0\n",
      "Loci shared by 2 taxa:\t66.0\n",
      "Loci shared by 3 taxa:\t66.0\n",
      "Loci shared by 4 taxa:\t66.0\n",
      "Loci shared by 5 taxa:\t66.0\n",
      "Loci shared by 6 taxa:\t66.0\n",
      "Loci shared by 7 taxa:\t66.0\n",
      "Loci shared by 8 taxa:\t66.0\n",
      "Loci shared by 9 taxa:\t66.0\n",
      "Loci shared by 10 taxa:\t66.0\n",
      "Loci shared by 11 taxa:\t66.0\n",
      "Loci shared by 12 taxa:\t66.0\n",
      "Loci shared by 13 taxa:\t64.0\n",
      "Loci shared by 14 taxa:\t64.0\n",
      "Loci shared by 15 taxa:\t61.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/borrelia/extract_probes_from_group/multifastas.sqlite  --base-taxon Borreliella_burgdorferi_GCF_000502155 --output results/phyluce/borrelia/extract_probes_from_group/Borreliella_burgdorferi_GCF_000502155+4-back-to-1.conf --specific-counts 1;\n",
      "Counter({'borrelia_hispnica_gcf_000500065': 66, 'borrelia_turicatae_gcf_000012085': 66, 'borrelia_recurrentis_gcf_000019705': 66, 'borrelia_anserina_gcf_001936255': 66, 'borrelia_persica_gcf_000500045': 66, 'borrelia_parkeri_gcf_000512145': 66, 'borrelia_crocidurae_gcf_000259345': 66, 'borrelia_turcica_gcf_003606285': 66, 'borrelia_duttonii_gcf_000019685': 66, 'borrelia_coriaceae_gcf_000568755': 65, 'borrelia_miyamotoi_gcf_000445425': 65, 'borreliella_garinii_gcf_000501735': 65, 'borreliella_burgdorferi_gcf_000502155': 65, 'borreliella_afzelii_gca_000502195': 64, 'borrelia_hermsii_gcf_001660005': 64})\n",
      "Total loci = 67\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(1)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/borrelia/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/borrelia/extract_probes_from_group/Borreliella_burgdorferi_GCF_000502155+4-back-to-1.conf --probe-prefix uce_borrelia_ --designer rnplattii --design borrelia_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 67\n",
      "Probe Count = 1783\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list.fasta --query results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  16:35:06\n",
      "Ended:  Thu Feb 06, 2020  16:35:08\n",
      "Time for execution:  0.0482173323631 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list.fasta --lastz results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_borrelia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 1782 fasta sequences.  Filtered 0 duplicates. Kept 1783.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/borrelia/final_probe_design/borrelia_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 16:36:16 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 1783\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 142640\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 0M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 86M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3961964\n",
      "Max number of word counting entries: 89183821\n",
      "\n",
      "# comparing sequences from          0  to        297\n",
      "---------- new table with      193 representatives\n",
      "# comparing sequences from        297  to        544\n",
      "----------     23 remaining sequences to the next cycle\n",
      "---------- new table with      145 representatives\n",
      "# comparing sequences from        521  to        731\n",
      "----------     52 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from        679  to        863\n",
      "99.9%---------- new table with       93 representatives\n",
      "# comparing sequences from        863  to       1783\n",
      "....................---------- new table with      541 representatives\n",
      "\n",
      "     1783  finished       1072  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 86M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 0.55\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bartonella"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'bartonella'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Bartonella_alsatica_GCF_000280015\",\n",
    "               \"Bartonella_ancashensis_GCF_001281405\",\n",
    "               \"Bartonella_apis_GCF_002007565\",\n",
    "               \"Bartonella_australis_GCF_000341355\",\n",
    "               \"Bartonella_birtlesii_GCF_000273375\",\n",
    "               \"Bartonella_bovis_GCF_000384965\",\n",
    "               \"Bartonella_clarridgeiae_GCF_000253015\",\n",
    "               \"Bartonella_doshiae_GCF_000278155\",\n",
    "               \"Bartonella_elizabethae_GCF_900638615\",\n",
    "               \"Bartonella_florencae_GCF_000312525\",\n",
    "               \"Bartonella_grahamii_GCF_000022725\",\n",
    "               \"Bartonella_henselae_GCF_000046705\",\n",
    "               \"Bartonella_koehlerae_GCF_000706625\",\n",
    "               \"Bartonella_massiliensis_GCF_902150025\",\n",
    "               \"Bartonella_mastomydis_GCF_900185775\",\n",
    "               \"Bartonella_melophagi_GCF_000278255\",\n",
    "               \"Bartonella_queenslandensis_GCF_000312585\",\n",
    "               \"Bartonella_quintana_GCF_000046685\",\n",
    "               \"Bartonella_rattaustraliani_GCF_000312565\",\n",
    "               \"Bartonella_rattimassiliensis_GCF_000278215\",\n",
    "               \"Bartonella_rochalimae_GCF_000706645\",\n",
    "               \"Bartonella_schoenbuchensis_GCF_002022685\",\n",
    "               \"Bartonella_senegalensis_GCF_000312545\",\n",
    "               \"Bartonella_sp_GCF_003606325\",\n",
    "               \"Bartonella_tamiae_GCF_000279995\",\n",
    "               \"Bartonella_taylorii_GCF_000278295\",\n",
    "               \"Bartonella_tribocorum_GCF_000196435\",\n",
    "               \"Bartonella_vinsonii_GCF_000341385\",\n",
    "               \"Bartonella_washoeensis_GCF_000278135\" ]\n",
    "\n",
    "reference_taxon = \"Bartonella_bacilliformis_GCF_000015445\"\n",
    "\n",
    "all_taxa = group_taxa + [reference_taxon]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000280015.1_Bart_alsa_IBS_382_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 486645 bytes  194674.80 bytes/sec\n",
      "total size is 486410  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001281405.1_ASM128140v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 433571 bytes  289075.33 bytes/sec\n",
      "total size is 433353  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002007565.1_ASM200756v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 851659 bytes  567800.67 bytes/sec\n",
      "total size is 851345  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000341355.1_ASM34135v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 473351 bytes  315595.33 bytes/sec\n",
      "total size is 473126  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000015445.1_ASM1544v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 420669 bytes  280474.00 bytes/sec\n",
      "total size is 420461  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000273375.1_ASM27337v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 532526 bytes  355045.33 bytes/sec\n",
      "total size is 532285  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000384965.1_BBbMcIrM01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 468800 bytes  312561.33 bytes/sec\n",
      "total size is 468575  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000253015.1_ASM25301v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 443857 bytes  295932.67 bytes/sec\n",
      "total size is 443640  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000278155.1_Bart_dosh_NCTC_12862_R18_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 530843 bytes  353923.33 bytes/sec\n",
      "total size is 530585  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900638615.1_58174_F01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 585120 bytes  390108.00 bytes/sec\n",
      "total size is 584872  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000312525.1_ASM31252v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 590098 bytes  393426.67 bytes/sec\n",
      "total size is 589841  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000022725.1_ASM2272v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 690170 bytes  460141.33 bytes/sec\n",
      "total size is 689890  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000046705.1_ASM4670v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 561080 bytes  374081.33 bytes/sec\n",
      "total size is 560832  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000706625.1_Bart_koeh_C29_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 514703 bytes  343163.33 bytes/sec\n",
      "total size is 514464  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_902150025.1_PRJEB33160_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 671595 bytes  447758.00 bytes/sec\n",
      "total size is 671322  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900185775.1_PRJEB21083_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 594776 bytes  396545.33 bytes/sec\n",
      "total size is 594519  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000278255.1_Bart_melo_K-2C_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 449134 bytes  299450.67 bytes/sec\n",
      "total size is 448910  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000312585.1_ASM31258v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 674931 bytes  449982.00 bytes/sec\n",
      "total size is 674658  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000046685.1_ASM4668v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 466060 bytes  310734.67 bytes/sec\n",
      "total size is 465836  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000312565.2_ASM31256v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 637836 bytes  425252.00 bytes/sec\n",
      "total size is 637571  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000278215.1_Bart_ratt_15908_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 614857 bytes  409932.67 bytes/sec\n",
      "total size is 614592  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000706645.1_Bart_roch_BMGH_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 444131 bytes  296115.33 bytes/sec\n",
      "total size is 443907  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002022685.1_ASM202268v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 501791 bytes  334555.33 bytes/sec\n",
      "total size is 501557  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000312545.1_ASM31254v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 580058 bytes  386733.33 bytes/sec\n",
      "total size is 579809  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003606325.3_ASM360632v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 662586 bytes  441752.00 bytes/sec\n",
      "total size is 662312  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000279995.1_Bart_tami_Th307_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 661567 bytes  441072.67 bytes/sec\n",
      "total size is 661286  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000278295.1_Bart_tayl_8TBB_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 569313 bytes  379570.00 bytes/sec\n",
      "total size is 569057  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000196435.1_ASM19643v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 765261 bytes  510202.00 bytes/sec\n",
      "total size is 764964  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000341385.1_ASM34138v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 529480 bytes  353014.67 bytes/sec\n",
      "total size is 529239  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000278135.1_Bart_wash_Sb944nv_V1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 564512 bytes  376369.33 bytes/sec\n",
      "total size is 564253  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481503 (\"sim_Bartonella_alsatica_GCF_000280015\") has been submitted\n",
      "Your job 5481504 (\"sim_Bartonella_ancashensis_GCF_001281405\") has been submitted\n",
      "Your job 5481505 (\"sim_Bartonella_apis_GCF_002007565\") has been submitted\n",
      "Your job 5481506 (\"sim_Bartonella_australis_GCF_000341355\") has been submitted\n",
      "Your job 5481507 (\"sim_Bartonella_birtlesii_GCF_000273375\") has been submitted\n",
      "Your job 5481508 (\"sim_Bartonella_bovis_GCF_000384965\") has been submitted\n",
      "Your job 5481509 (\"sim_Bartonella_clarridgeiae_GCF_000253015\") has been submitted\n",
      "Your job 5481510 (\"sim_Bartonella_doshiae_GCF_000278155\") has been submitted\n",
      "Your job 5481511 (\"sim_Bartonella_elizabethae_GCF_900638615\") has been submitted\n",
      "Your job 5481512 (\"sim_Bartonella_florencae_GCF_000312525\") has been submitted\n",
      "Your job 5481513 (\"sim_Bartonella_grahamii_GCF_000022725\") has been submitted\n",
      "Your job 5481514 (\"sim_Bartonella_henselae_GCF_000046705\") has been submitted\n",
      "Your job 5481515 (\"sim_Bartonella_koehlerae_GCF_000706625\") has been submitted\n",
      "Your job 5481516 (\"sim_Bartonella_massiliensis_GCF_902150025\") has been submitted\n",
      "Your job 5481517 (\"sim_Bartonella_mastomydis_GCF_900185775\") has been submitted\n",
      "Your job 5481518 (\"sim_Bartonella_melophagi_GCF_000278255\") has been submitted\n",
      "Your job 5481519 (\"sim_Bartonella_queenslandensis_GCF_000312585\") has been submitted\n",
      "Your job 5481520 (\"sim_Bartonella_quintana_GCF_000046685\") has been submitted\n",
      "Your job 5481521 (\"sim_Bartonella_rattaustraliani_GCF_000312565\") has been submitted\n",
      "Your job 5481522 (\"sim_Bartonella_rattimassiliensis_GCF_000278215\") has been submitted\n",
      "Your job 5481523 (\"sim_Bartonella_rochalimae_GCF_000706645\") has been submitted\n",
      "Your job 5481524 (\"sim_Bartonella_schoenbuchensis_GCF_002022685\") has been submitted\n",
      "Your job 5481525 (\"sim_Bartonella_senegalensis_GCF_000312545\") has been submitted\n",
      "Your job 5481526 (\"sim_Bartonella_sp_GCF_003606325\") has been submitted\n",
      "Your job 5481527 (\"sim_Bartonella_tamiae_GCF_000279995\") has been submitted\n",
      "Your job 5481528 (\"sim_Bartonella_taylorii_GCF_000278295\") has been submitted\n",
      "Your job 5481529 (\"sim_Bartonella_tribocorum_GCF_000196435\") has been submitted\n",
      "Your job 5481530 (\"sim_Bartonella_vinsonii_GCF_000341385\") has been submitted\n",
      "Your job 5481531 (\"sim_Bartonella_washoeensis_GCF_000278135\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/bartonella/cleaned_genomes/Bartonella_bacilliformis_GCF_000015445_formatted.fas path=results/phyluce/bartonella/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481532 (\"map_Bartonella_alsatica_GCF_000280015\") has been submitted\n",
      "Your job 5481533 (\"map_Bartonella_ancashensis_GCF_001281405\") has been submitted\n",
      "Your job 5481534 (\"map_Bartonella_apis_GCF_002007565\") has been submitted\n",
      "Your job 5481535 (\"map_Bartonella_australis_GCF_000341355\") has been submitted\n",
      "Your job 5481536 (\"map_Bartonella_birtlesii_GCF_000273375\") has been submitted\n",
      "Your job 5481537 (\"map_Bartonella_bovis_GCF_000384965\") has been submitted\n",
      "Your job 5481538 (\"map_Bartonella_clarridgeiae_GCF_000253015\") has been submitted\n",
      "Your job 5481539 (\"map_Bartonella_doshiae_GCF_000278155\") has been submitted\n",
      "Your job 5481540 (\"map_Bartonella_elizabethae_GCF_900638615\") has been submitted\n",
      "Your job 5481541 (\"map_Bartonella_florencae_GCF_000312525\") has been submitted\n",
      "Your job 5481542 (\"map_Bartonella_grahamii_GCF_000022725\") has been submitted\n",
      "Your job 5481543 (\"map_Bartonella_henselae_GCF_000046705\") has been submitted\n",
      "Your job 5481544 (\"map_Bartonella_koehlerae_GCF_000706625\") has been submitted\n",
      "Your job 5481545 (\"map_Bartonella_massiliensis_GCF_902150025\") has been submitted\n",
      "Your job 5481546 (\"map_Bartonella_mastomydis_GCF_900185775\") has been submitted\n",
      "Your job 5481547 (\"map_Bartonella_melophagi_GCF_000278255\") has been submitted\n",
      "Your job 5481548 (\"map_Bartonella_queenslandensis_GCF_000312585\") has been submitted\n",
      "Your job 5481549 (\"map_Bartonella_quintana_GCF_000046685\") has been submitted\n",
      "Your job 5481550 (\"map_Bartonella_rattaustraliani_GCF_000312565\") has been submitted\n",
      "Your job 5481551 (\"map_Bartonella_rattimassiliensis_GCF_000278215\") has been submitted\n",
      "Your job 5481552 (\"map_Bartonella_rochalimae_GCF_000706645\") has been submitted\n",
      "Your job 5481553 (\"map_Bartonella_schoenbuchensis_GCF_002022685\") has been submitted\n",
      "Your job 5481554 (\"map_Bartonella_senegalensis_GCF_000312545\") has been submitted\n",
      "Your job 5481555 (\"map_Bartonella_sp_GCF_003606325\") has been submitted\n",
      "Your job 5481556 (\"map_Bartonella_tamiae_GCF_000279995\") has been submitted\n",
      "Your job 5481557 (\"map_Bartonella_taylorii_GCF_000278295\") has been submitted\n",
      "Your job 5481558 (\"map_Bartonella_tribocorum_GCF_000196435\") has been submitted\n",
      "Your job 5481559 (\"map_Bartonella_vinsonii_GCF_000341385\") has been submitted\n",
      "Your job 5481560 (\"map_Bartonella_washoeensis_GCF_000278135\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481561 (\"merge_Bartonella_alsatica_GCF_000280015\") has been submitted\n",
      "Your job 5481562 (\"merge_Bartonella_ancashensis_GCF_001281405\") has been submitted\n",
      "Your job 5481563 (\"merge_Bartonella_apis_GCF_002007565\") has been submitted\n",
      "Your job 5481564 (\"merge_Bartonella_australis_GCF_000341355\") has been submitted\n",
      "Your job 5481565 (\"merge_Bartonella_birtlesii_GCF_000273375\") has been submitted\n",
      "Your job 5481566 (\"merge_Bartonella_bovis_GCF_000384965\") has been submitted\n",
      "Your job 5481567 (\"merge_Bartonella_clarridgeiae_GCF_000253015\") has been submitted\n",
      "Your job 5481568 (\"merge_Bartonella_doshiae_GCF_000278155\") has been submitted\n",
      "Your job 5481569 (\"merge_Bartonella_elizabethae_GCF_900638615\") has been submitted\n",
      "Your job 5481570 (\"merge_Bartonella_florencae_GCF_000312525\") has been submitted\n",
      "Your job 5481571 (\"merge_Bartonella_grahamii_GCF_000022725\") has been submitted\n",
      "Your job 5481572 (\"merge_Bartonella_henselae_GCF_000046705\") has been submitted\n",
      "Your job 5481573 (\"merge_Bartonella_koehlerae_GCF_000706625\") has been submitted\n",
      "Your job 5481574 (\"merge_Bartonella_massiliensis_GCF_902150025\") has been submitted\n",
      "Your job 5481575 (\"merge_Bartonella_mastomydis_GCF_900185775\") has been submitted\n",
      "Your job 5481576 (\"merge_Bartonella_melophagi_GCF_000278255\") has been submitted\n",
      "Your job 5481577 (\"merge_Bartonella_queenslandensis_GCF_000312585\") has been submitted\n",
      "Your job 5481578 (\"merge_Bartonella_quintana_GCF_000046685\") has been submitted\n",
      "Your job 5481579 (\"merge_Bartonella_rattaustraliani_GCF_000312565\") has been submitted\n",
      "Your job 5481580 (\"merge_Bartonella_rattimassiliensis_GCF_000278215\") has been submitted\n",
      "Your job 5481581 (\"merge_Bartonella_rochalimae_GCF_000706645\") has been submitted\n",
      "Your job 5481582 (\"merge_Bartonella_schoenbuchensis_GCF_002022685\") has been submitted\n",
      "Your job 5481583 (\"merge_Bartonella_senegalensis_GCF_000312545\") has been submitted\n",
      "Your job 5481584 (\"merge_Bartonella_sp_GCF_003606325\") has been submitted\n",
      "Your job 5481585 (\"merge_Bartonella_tamiae_GCF_000279995\") has been submitted\n",
      "Your job 5481586 (\"merge_Bartonella_taylorii_GCF_000278295\") has been submitted\n",
      "Your job 5481587 (\"merge_Bartonella_tribocorum_GCF_000196435\") has been submitted\n",
      "Your job 5481588 (\"merge_Bartonella_vinsonii_GCF_000341385\") has been submitted\n",
      "Your job 5481589 (\"merge_Bartonella_washoeensis_GCF_000278135\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 720 sequences from Bartonella_alsatica_GCF_000280015_merged.bed.  Filtered 550 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 170.\n",
      "Screened 562 sequences from Bartonella_ancashensis_GCF_001281405_merged.bed.  Filtered 430 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 132.\n",
      "Screened 38 sequences from Bartonella_apis_GCF_002007565_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 6.\n",
      "Screened 400 sequences from Bartonella_australis_GCF_000341355_merged.bed.  Filtered 333 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 67.\n",
      "Screened 678 sequences from Bartonella_birtlesii_GCF_000273375_merged.bed.  Filtered 521 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 157.\n",
      "Screened 1124 sequences from Bartonella_bovis_GCF_000384965_merged.bed.  Filtered 784 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 340.\n",
      "Screened 897 sequences from Bartonella_clarridgeiae_GCF_000253015_merged.bed.  Filtered 651 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 246.\n",
      "Screened 719 sequences from Bartonella_doshiae_GCF_000278155_merged.bed.  Filtered 555 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 164.\n",
      "Screened 535 sequences from Bartonella_elizabethae_GCF_900638615_merged.bed.  Filtered 428 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 638 sequences from Bartonella_florencae_GCF_000312525_merged.bed.  Filtered 504 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 134.\n",
      "Screened 635 sequences from Bartonella_grahamii_GCF_000022725_merged.bed.  Filtered 512 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 123.\n",
      "Screened 756 sequences from Bartonella_henselae_GCF_000046705_merged.bed.  Filtered 561 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 195.\n",
      "Screened 701 sequences from Bartonella_koehlerae_GCF_000706625_merged.bed.  Filtered 522 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 179.\n",
      "Screened 548 sequences from Bartonella_massiliensis_GCF_902150025_merged.bed.  Filtered 442 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 106.\n",
      "Screened 536 sequences from Bartonella_mastomydis_GCF_900185775_merged.bed.  Filtered 429 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 1101 sequences from Bartonella_melophagi_GCF_000278255_merged.bed.  Filtered 783 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 318.\n",
      "Screened 525 sequences from Bartonella_queenslandensis_GCF_000312585_merged.bed.  Filtered 414 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 776 sequences from Bartonella_quintana_GCF_000046685_merged.bed.  Filtered 587 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 189.\n",
      "Screened 570 sequences from Bartonella_rattaustraliani_GCF_000312565_merged.bed.  Filtered 463 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 515 sequences from Bartonella_rattimassiliensis_GCF_000278215_merged.bed.  Filtered 408 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 794 sequences from Bartonella_rochalimae_GCF_000706645_merged.bed.  Filtered 605 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 189.\n",
      "Screened 1178 sequences from Bartonella_schoenbuchensis_GCF_002022685_merged.bed.  Filtered 821 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 357.\n",
      "Screened 705 sequences from Bartonella_senegalensis_GCF_000312545_merged.bed.  Filtered 548 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 157.\n",
      "Screened 540 sequences from Bartonella_sp_GCF_003606325_merged.bed.  Filtered 425 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 115.\n",
      "Screened 74 sequences from Bartonella_tamiae_GCF_000279995_merged.bed.  Filtered 63 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 796 sequences from Bartonella_taylorii_GCF_000278295_merged.bed.  Filtered 613 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 183.\n",
      "Screened 565 sequences from Bartonella_tribocorum_GCF_000196435_merged.bed.  Filtered 451 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 114.\n",
      "Screened 709 sequences from Bartonella_vinsonii_GCF_000341385_merged.bed.  Filtered 533 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 176.\n",
      "Screened 832 sequences from Bartonella_washoeensis_GCF_000278135_merged.bed.  Filtered 636 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 196.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/bartonella/initial_intervals/Bartonella_washoeensis_GCF_000278135_merged.bed --twobit results/phyluce/bartonella/cleaned_genomes/Bartonella_bacilliformis_GCF_000015445_formatted.2bit --output results/phyluce/bartonella/initial_intervals/Bartonella_washoeensis_GCF_000278135_stripped.bed;\n",
      "bartonella_alsatica_gcf_000280015.\n",
      "bartonella_ancashensis_gcf_001281405.\n",
      "bartonella_apis_gcf_002007565.\n",
      "bartonella_australis_gcf_000341355.\n",
      "bartonella_birtlesii_gcf_000273375.\n",
      "bartonella_bovis_gcf_000384965.\n",
      "bartonella_clarridgeiae_gcf_000253015.\n",
      "bartonella_doshiae_gcf_000278155.\n",
      "bartonella_elizabethae_gcf_900638615.\n",
      "bartonella_florencae_gcf_000312525.\n",
      "bartonella_grahamii_gcf_000022725.\n",
      "bartonella_henselae_gcf_000046705.\n",
      "bartonella_koehlerae_gcf_000706625.\n",
      "bartonella_massiliensis_gcf_902150025.\n",
      "bartonella_mastomydis_gcf_900185775.\n",
      "bartonella_melophagi_gcf_000278255.\n",
      "bartonella_queenslandensis_gcf_000312585.\n",
      "bartonella_quintana_gcf_000046685.\n",
      "bartonella_rattaustraliani_gcf_000312565.\n",
      "bartonella_rattimassiliensis_gcf_000278215.\n",
      "bartonella_rochalimae_gcf_000706645.\n",
      "bartonella_schoenbuchensis_gcf_002022685.\n",
      "bartonella_senegalensis_gcf_000312545.\n",
      "bartonella_sp_gcf_003606325.\n",
      "bartonella_tamiae_gcf_000279995.\n",
      "bartonella_taylorii_gcf_000278295.\n",
      "bartonella_tribocorum_gcf_000196435.\n",
      "bartonella_vinsonii_gcf_000341385.\n",
      "bartonella_washoeensis_gcf_000278135.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/bartonella/initial_intervals/Bartonella_washoeensis_GCF_000278135_merged.bed --twobit results/phyluce/bartonella/cleaned_genomes/Bartonella_bacilliformis_GCF_000015445_formatted.2bit --output results/phyluce/bartonella/initial_intervals/Bartonella_washoeensis_GCF_000278135_stripped.bed;\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 0 taxa:\t911.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 1 taxa:\t911.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 2 taxa:\t643.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 3 taxa:\t498.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 4 taxa:\t400.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 5 taxa:\t335.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 6 taxa:\t274.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 7 taxa:\t231.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 8 taxa:\t200.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 9 taxa:\t174.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 10 taxa:\t157.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 11 taxa:\t136.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 12 taxa:\t119.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 13 taxa:\t104.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 14 taxa:\t94.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 15 taxa:\t78.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 16 taxa:\t67.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 17 taxa:\t62.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 18 taxa:\t55.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 19 taxa:\t43.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 20 taxa:\t38.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 21 taxa:\t32.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 22 taxa:\t24.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 23 taxa:\t21.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 24 taxa:\t21.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 25 taxa:\t13.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 26 taxa:\t8.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 27 taxa:\t6.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 28 taxa:\t3.0\n",
      "Loci shared by Bartonella_bacilliformis_GCF_000015445 + 29 taxa:\t2.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/bartonella/initial_intervals/bartonella-to-Bartonella_bacilliformis_GCF_000015445.sqlite --base-taxon Bartonella_bacilliformis_GCF_000015445 --output results/phyluce/bartonella/initial_intervals/Bartonella_bacilliformis_GCF_000015445_+11.bed --specific-counts 11;\n",
      "Counter({'bartonella_melophagi_gcf_000278255': 113, 'bartonella_schoenbuchensis_gcf_002022685': 107, 'bartonella_bovis_gcf_000384965': 106, 'bartonella_vinsonii_gcf_000341385': 101, 'bartonella_koehlerae_gcf_000706625': 98, 'bartonella_clarridgeiae_gcf_000253015': 98, 'bartonella_quintana_gcf_000046685': 97, 'bartonella_taylorii_gcf_000278295': 96, 'bartonella_washoeensis_gcf_000278135': 96, 'bartonella_henselae_gcf_000046705': 95, 'bartonella_birtlesii_gcf_000273375': 88, 'bartonella_senegalensis_gcf_000312545': 86, 'bartonella_rochalimae_gcf_000706645': 85, 'bartonella_grahamii_gcf_000022725': 85, 'bartonella_alsatica_gcf_000280015': 84, 'bartonella_doshiae_gcf_000278155': 83, 'bartonella_florencae_gcf_000312525': 83, 'bartonella_elizabethae_gcf_900638615': 80, 'bartonella_tribocorum_gcf_000196435': 79, 'bartonella_queenslandensis_gcf_000312585': 77, 'bartonella_mastomydis_gcf_900185775': 74, 'bartonella_rattimassiliensis_gcf_000278215': 72, 'bartonella_sp_gcf_003606325': 70, 'bartonella_massiliensis_gcf_902150025': 67, 'bartonella_rattaustraliani_gcf_000312565': 64, 'bartonella_ancashensis_gcf_001281405': 52, 'bartonella_australis_gcf_000341355': 39, 'bartonella_tamiae_gcf_000279995': 8, 'bartonella_apis_gcf_002007565': 3})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 11\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/bartonella/initial_intervals/Bartonella_bacilliformis_GCF_000015445_+11.bed --twobit results/phyluce/bartonella/cleaned_genomes/Bartonella_bacilliformis_GCF_000015445_formatted.2bit --buffer-to 160 --output results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11.fasta;\n",
      "Screened 136 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 136.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11.fasta --probe-prefix uce_bartonella_ --design bartonella_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "G\n",
      "\n",
      "\n",
      "Conserved locus count = 136\n",
      "Probe Count = 271\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas --query results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  16:42:19\n",
      "Ended:  Thu Feb 06, 2020  16:42:19\n",
      "Time for execution:  0.00309858322144 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas                        --lastz results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes_vself.lastz                       --probe-prefix=uce_bartonella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 270 fasta sequences.  Filtered 6 duplicates. Kept 259.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/bartonella/validate_intervals/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas --scaffoldlist Bartonella_alsatica_GCF_000280015 Bartonella_ancashensis_GCF_001281405 Bartonella_apis_GCF_002007565 Bartonella_australis_GCF_000341355 Bartonella_birtlesii_GCF_000273375 Bartonella_bovis_GCF_000384965 Bartonella_clarridgeiae_GCF_000253015 Bartonella_doshiae_GCF_000278155 Bartonella_elizabethae_GCF_900638615 Bartonella_florencae_GCF_000312525 Bartonella_grahamii_GCF_000022725 Bartonella_henselae_GCF_000046705 Bartonella_koehlerae_GCF_000706625 Bartonella_massiliensis_GCF_902150025 Bartonella_mastomydis_GCF_900185775 Bartonella_melophagi_GCF_000278255 Bartonella_queenslandensis_GCF_000312585 Bartonella_quintana_GCF_000046685 Bartonella_rattaustraliani_GCF_000312565 Bartonella_rattimassiliensis_GCF_000278215 Bartonella_rochalimae_GCF_000706645 Bartonella_schoenbuchensis_GCF_002022685 Bartonella_senegalensis_GCF_000312545 Bartonella_sp_GCF_003606325 Bartonella_tamiae_GCF_000279995 Bartonella_taylorii_GCF_000278295 Bartonella_tribocorum_GCF_000196435 Bartonella_vinsonii_GCF_000341385 Bartonella_washoeensis_GCF_000278135 Bartonella_bacilliformis_GCF_000015445 --genome-base-path results/phyluce/bartonella/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/bartonella/validate_intervals/bartonella-to-Bartonella_bacilliformis_GCF_000015445.sqlite --output results/phyluce/bartonella/validate_intervals/lastz/;\n",
      "\n",
      "Running against Bartonella_alsatica_GCF_000280015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprjSiDs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplGGGRt.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_alsatica_GCF_000280015.lastz\n",
      "Creating Bartonella_alsatica_GCF_000280015 table\n",
      "Inserting data to Bartonella_alsatica_GCF_000280015 table\n",
      "\n",
      "Running against Bartonella_ancashensis_GCF_001281405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_bSRmM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppuTSm5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_ancashensis_GCF_001281405.lastz\n",
      "Creating Bartonella_ancashensis_GCF_001281405 table\n",
      "Inserting data to Bartonella_ancashensis_GCF_001281405 table\n",
      "\n",
      "Running against Bartonella_apis_GCF_002007565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpblmE6K.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXwgRQd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_apis_GCF_002007565.lastz\n",
      "Creating Bartonella_apis_GCF_002007565 table\n",
      "Inserting data to Bartonella_apis_GCF_002007565 table\n",
      "\n",
      "Running against Bartonella_australis_GCF_000341355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRUnvEO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQey7QF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_australis_GCF_000341355.lastz\n",
      "Creating Bartonella_australis_GCF_000341355 table\n",
      "Inserting data to Bartonella_australis_GCF_000341355 table\n",
      "\n",
      "Running against Bartonella_birtlesii_GCF_000273375.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDLWwlf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOYdO9k.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_birtlesii_GCF_000273375.lastz\n",
      "Creating Bartonella_birtlesii_GCF_000273375 table\n",
      "Inserting data to Bartonella_birtlesii_GCF_000273375 table\n",
      "\n",
      "Running against Bartonella_bovis_GCF_000384965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmporLGVa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp92XNzk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_bovis_GCF_000384965.lastz\n",
      "Creating Bartonella_bovis_GCF_000384965 table\n",
      "Inserting data to Bartonella_bovis_GCF_000384965 table\n",
      "\n",
      "Running against Bartonella_clarridgeiae_GCF_000253015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp97lgxg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp91G0R6.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_clarridgeiae_GCF_000253015.lastz\n",
      "Creating Bartonella_clarridgeiae_GCF_000253015 table\n",
      "Inserting data to Bartonella_clarridgeiae_GCF_000253015 table\n",
      "\n",
      "Running against Bartonella_doshiae_GCF_000278155.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpeA1JMK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpyeB3cK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_doshiae_GCF_000278155.lastz\n",
      "Creating Bartonella_doshiae_GCF_000278155 table\n",
      "Inserting data to Bartonella_doshiae_GCF_000278155 table\n",
      "\n",
      "Running against Bartonella_elizabethae_GCF_900638615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBaWF8p.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmphYZTZ8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_elizabethae_GCF_900638615.lastz\n",
      "Creating Bartonella_elizabethae_GCF_900638615 table\n",
      "Inserting data to Bartonella_elizabethae_GCF_900638615 table\n",
      "\n",
      "Running against Bartonella_florencae_GCF_000312525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpN9tXOO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdaWodq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_florencae_GCF_000312525.lastz\n",
      "Creating Bartonella_florencae_GCF_000312525 table\n",
      "Inserting data to Bartonella_florencae_GCF_000312525 table\n",
      "\n",
      "Running against Bartonella_grahamii_GCF_000022725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVqlWLh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJkbtxx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_grahamii_GCF_000022725.lastz\n",
      "Creating Bartonella_grahamii_GCF_000022725 table\n",
      "Inserting data to Bartonella_grahamii_GCF_000022725 table\n",
      "\n",
      "Running against Bartonella_henselae_GCF_000046705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5T5hiI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbLlNGf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_henselae_GCF_000046705.lastz\n",
      "Creating Bartonella_henselae_GCF_000046705 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Bartonella_henselae_GCF_000046705 table\n",
      "\n",
      "Running against Bartonella_koehlerae_GCF_000706625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxvhd4u.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptue79K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_koehlerae_GCF_000706625.lastz\n",
      "Creating Bartonella_koehlerae_GCF_000706625 table\n",
      "Inserting data to Bartonella_koehlerae_GCF_000706625 table\n",
      "\n",
      "Running against Bartonella_massiliensis_GCF_902150025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQAdrFi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcgXntp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_massiliensis_GCF_902150025.lastz\n",
      "Creating Bartonella_massiliensis_GCF_902150025 table\n",
      "Inserting data to Bartonella_massiliensis_GCF_902150025 table\n",
      "\n",
      "Running against Bartonella_mastomydis_GCF_900185775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQsPBjz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEXnScI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_mastomydis_GCF_900185775.lastz\n",
      "Creating Bartonella_mastomydis_GCF_900185775 table\n",
      "Inserting data to Bartonella_mastomydis_GCF_900185775 table\n",
      "\n",
      "Running against Bartonella_melophagi_GCF_000278255.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpN2ktVB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9NYpth.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_melophagi_GCF_000278255.lastz\n",
      "Creating Bartonella_melophagi_GCF_000278255 table\n",
      "Inserting data to Bartonella_melophagi_GCF_000278255 table\n",
      "\n",
      "Running against Bartonella_queenslandensis_GCF_000312585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDPXojd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpm4GeWM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_queenslandensis_GCF_000312585.lastz\n",
      "Creating Bartonella_queenslandensis_GCF_000312585 table\n",
      "Inserting data to Bartonella_queenslandensis_GCF_000312585 table\n",
      "\n",
      "Running against Bartonella_quintana_GCF_000046685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvYQh9x.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPK7ko0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_quintana_GCF_000046685.lastz\n",
      "Creating Bartonella_quintana_GCF_000046685 table\n",
      "Inserting data to Bartonella_quintana_GCF_000046685 table\n",
      "\n",
      "Running against Bartonella_rattaustraliani_GCF_000312565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpw4G04S.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNWR19s.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_rattaustraliani_GCF_000312565.lastz\n",
      "Creating Bartonella_rattaustraliani_GCF_000312565 table\n",
      "Inserting data to Bartonella_rattaustraliani_GCF_000312565 table\n",
      "\n",
      "Running against Bartonella_rattimassiliensis_GCF_000278215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgbMYdC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_ffxOZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_rattimassiliensis_GCF_000278215.lastz\n",
      "Creating Bartonella_rattimassiliensis_GCF_000278215 table\n",
      "Inserting data to Bartonella_rattimassiliensis_GCF_000278215 table\n",
      "\n",
      "Running against Bartonella_rochalimae_GCF_000706645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpeK8zVM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIbI1s0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_rochalimae_GCF_000706645.lastz\n",
      "Creating Bartonella_rochalimae_GCF_000706645 table\n",
      "Inserting data to Bartonella_rochalimae_GCF_000706645 table\n",
      "\n",
      "Running against Bartonella_schoenbuchensis_GCF_002022685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRUiuGJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDYKiJK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_schoenbuchensis_GCF_002022685.lastz\n",
      "Creating Bartonella_schoenbuchensis_GCF_002022685 table\n",
      "Inserting data to Bartonella_schoenbuchensis_GCF_002022685 table\n",
      "\n",
      "Running against Bartonella_senegalensis_GCF_000312545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1Tt7ap.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQH_NTu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_senegalensis_GCF_000312545.lastz\n",
      "Creating Bartonella_senegalensis_GCF_000312545 table\n",
      "Inserting data to Bartonella_senegalensis_GCF_000312545 table\n",
      "\n",
      "Running against Bartonella_sp_GCF_003606325.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQDkAPC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpuaVjrB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_sp_GCF_003606325.lastz\n",
      "Creating Bartonella_sp_GCF_003606325 table\n",
      "Inserting data to Bartonella_sp_GCF_003606325 table\n",
      "\n",
      "Running against Bartonella_tamiae_GCF_000279995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyKc8A3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQSSiOR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_tamiae_GCF_000279995.lastz\n",
      "Creating Bartonella_tamiae_GCF_000279995 table\n",
      "Inserting data to Bartonella_tamiae_GCF_000279995 table\n",
      "\n",
      "Running against Bartonella_taylorii_GCF_000278295.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzA4zKr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpy4L3IF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_taylorii_GCF_000278295.lastz\n",
      "Creating Bartonella_taylorii_GCF_000278295 table\n",
      "Inserting data to Bartonella_taylorii_GCF_000278295 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Bartonella_tribocorum_GCF_000196435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgGSIBS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprTXj4n.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_tribocorum_GCF_000196435.lastz\n",
      "Creating Bartonella_tribocorum_GCF_000196435 table\n",
      "Inserting data to Bartonella_tribocorum_GCF_000196435 table\n",
      "\n",
      "Running against Bartonella_vinsonii_GCF_000341385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4Tk4M2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmkv9rH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_vinsonii_GCF_000341385.lastz\n",
      "Creating Bartonella_vinsonii_GCF_000341385 table\n",
      "Inserting data to Bartonella_vinsonii_GCF_000341385 table\n",
      "\n",
      "Running against Bartonella_washoeensis_GCF_000278135.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgwyVGT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFGn8yZ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_washoeensis_GCF_000278135.lastz\n",
      "Creating Bartonella_washoeensis_GCF_000278135 table\n",
      "Inserting data to Bartonella_washoeensis_GCF_000278135 table\n",
      "\n",
      "Running against Bartonella_bacilliformis_GCF_000015445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpIC654c.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpS8SHLs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/bartonella/validate_intervals/lastz/Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_Bartonella_bacilliformis_GCF_000015445.lastz\n",
      "Creating Bartonella_bacilliformis_GCF_000015445 table\n",
      "Inserting data to Bartonella_bacilliformis_GCF_000015445 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/bartonella/extract_probes_from_group/bartonella_genome.conf --lastz results/phyluce/bartonella/validate_intervals/lastz --probes 120 --probe-prefix uce_bartonella_ --name-pattern \"Bartonella_bacilliformis_GCF_000015445_+11_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/bartonella/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 16:42:52,837 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 16:42:52,840 - Phyluce - INFO - ------ Working on Bartonella_alsatica_GCF_000280015 genome ------\n",
      "2020-02-06 16:42:52,841 - Phyluce - INFO - Reading Bartonella_alsatica_GCF_000280015 genome\n",
      "2020-02-06 16:42:53,399 - Phyluce - INFO - Bartonella_alsatica_GCF_000280015: 136 uces, 0 dupes, 136 non-dupes, 11 orient drop, 4 length drop, 121 written\n",
      "2020-02-06 16:42:53,399 - Phyluce - INFO - ----- Working on Bartonella_ancashensis_GCF_001281405 genome ----\n",
      "2020-02-06 16:42:53,411 - Phyluce - INFO - Reading Bartonella_ancashensis_GCF_001281405 genome\n",
      "2020-02-06 16:42:53,901 - Phyluce - INFO - Bartonella_ancashensis_GCF_001281405: 136 uces, 0 dupes, 136 non-dupes, 9 orient drop, 8 length drop, 119 written\n",
      "2020-02-06 16:42:53,901 - Phyluce - INFO - -------- Working on Bartonella_apis_GCF_002007565 genome --------\n",
      "2020-02-06 16:42:53,902 - Phyluce - INFO - Reading Bartonella_apis_GCF_002007565 genome\n",
      "2020-02-06 16:42:54,367 - Phyluce - INFO - Bartonella_apis_GCF_002007565: 134 uces, 0 dupes, 134 non-dupes, 7 orient drop, 9 length drop, 118 written\n",
      "2020-02-06 16:42:54,367 - Phyluce - INFO - ------ Working on Bartonella_australis_GCF_000341355 genome -----\n",
      "2020-02-06 16:42:54,368 - Phyluce - INFO - Reading Bartonella_australis_GCF_000341355 genome\n",
      "2020-02-06 16:42:54,859 - Phyluce - INFO - Bartonella_australis_GCF_000341355: 136 uces, 0 dupes, 136 non-dupes, 10 orient drop, 8 length drop, 118 written\n",
      "2020-02-06 16:42:54,860 - Phyluce - INFO - ------ Working on Bartonella_birtlesii_GCF_000273375 genome -----\n",
      "2020-02-06 16:42:54,860 - Phyluce - INFO - Reading Bartonella_birtlesii_GCF_000273375 genome\n",
      "2020-02-06 16:42:55,321 - Phyluce - INFO - Bartonella_birtlesii_GCF_000273375: 134 uces, 0 dupes, 134 non-dupes, 7 orient drop, 5 length drop, 122 written\n",
      "2020-02-06 16:42:55,321 - Phyluce - INFO - -------- Working on Bartonella_bovis_GCF_000384965 genome -------\n",
      "2020-02-06 16:42:55,321 - Phyluce - INFO - Reading Bartonella_bovis_GCF_000384965 genome\n",
      "2020-02-06 16:42:55,803 - Phyluce - INFO - Bartonella_bovis_GCF_000384965: 136 uces, 0 dupes, 136 non-dupes, 12 orient drop, 4 length drop, 120 written\n",
      "2020-02-06 16:42:55,803 - Phyluce - INFO - ---- Working on Bartonella_clarridgeiae_GCF_000253015 genome ----\n",
      "2020-02-06 16:42:55,804 - Phyluce - INFO - Reading Bartonella_clarridgeiae_GCF_000253015 genome\n",
      "2020-02-06 16:42:56,294 - Phyluce - INFO - Bartonella_clarridgeiae_GCF_000253015: 135 uces, 0 dupes, 135 non-dupes, 5 orient drop, 11 length drop, 119 written\n",
      "2020-02-06 16:42:56,294 - Phyluce - INFO - ------- Working on Bartonella_doshiae_GCF_000278155 genome ------\n",
      "2020-02-06 16:42:56,294 - Phyluce - INFO - Reading Bartonella_doshiae_GCF_000278155 genome\n",
      "2020-02-06 16:42:56,786 - Phyluce - INFO - Bartonella_doshiae_GCF_000278155: 136 uces, 15 dupes, 121 non-dupes, 1 orient drop, 1 length drop, 119 written\n",
      "2020-02-06 16:42:56,786 - Phyluce - INFO - ----- Working on Bartonella_elizabethae_GCF_900638615 genome ----\n",
      "2020-02-06 16:42:56,787 - Phyluce - INFO - Reading Bartonella_elizabethae_GCF_900638615 genome\n",
      "2020-02-06 16:42:57,273 - Phyluce - INFO - Bartonella_elizabethae_GCF_900638615: 136 uces, 0 dupes, 136 non-dupes, 7 orient drop, 10 length drop, 119 written\n",
      "2020-02-06 16:42:57,273 - Phyluce - INFO - ------ Working on Bartonella_florencae_GCF_000312525 genome -----\n",
      "2020-02-06 16:42:57,273 - Phyluce - INFO - Reading Bartonella_florencae_GCF_000312525 genome\n",
      "2020-02-06 16:42:57,742 - Phyluce - INFO - Bartonella_florencae_GCF_000312525: 136 uces, 7 dupes, 129 non-dupes, 1 orient drop, 1 length drop, 126 written\n",
      "2020-02-06 16:42:57,742 - Phyluce - INFO - ------ Working on Bartonella_grahamii_GCF_000022725 genome ------\n",
      "2020-02-06 16:42:57,743 - Phyluce - INFO - Reading Bartonella_grahamii_GCF_000022725 genome\n",
      "2020-02-06 16:42:58,228 - Phyluce - INFO - Bartonella_grahamii_GCF_000022725: 136 uces, 0 dupes, 136 non-dupes, 12 orient drop, 5 length drop, 119 written\n",
      "2020-02-06 16:42:58,228 - Phyluce - INFO - ------ Working on Bartonella_henselae_GCF_000046705 genome ------\n",
      "2020-02-06 16:42:58,229 - Phyluce - INFO - Reading Bartonella_henselae_GCF_000046705 genome\n",
      "2020-02-06 16:42:58,716 - Phyluce - INFO - Bartonella_henselae_GCF_000046705: 136 uces, 0 dupes, 136 non-dupes, 11 orient drop, 5 length drop, 120 written\n",
      "2020-02-06 16:42:58,716 - Phyluce - INFO - ------ Working on Bartonella_koehlerae_GCF_000706625 genome -----\n",
      "2020-02-06 16:42:58,736 - Phyluce - INFO - Reading Bartonella_koehlerae_GCF_000706625 genome\n",
      "2020-02-06 16:42:59,222 - Phyluce - INFO - Bartonella_koehlerae_GCF_000706625: 136 uces, 0 dupes, 136 non-dupes, 10 orient drop, 4 length drop, 122 written\n",
      "2020-02-06 16:42:59,223 - Phyluce - INFO - ---- Working on Bartonella_massiliensis_GCF_902150025 genome ----\n",
      "2020-02-06 16:42:59,223 - Phyluce - INFO - Reading Bartonella_massiliensis_GCF_902150025 genome\n",
      "2020-02-06 16:42:59,707 - Phyluce - INFO - Bartonella_massiliensis_GCF_902150025: 136 uces, 13 dupes, 123 non-dupes, 0 orient drop, 0 length drop, 123 written\n",
      "2020-02-06 16:42:59,708 - Phyluce - INFO - ----- Working on Bartonella_mastomydis_GCF_900185775 genome -----\n",
      "2020-02-06 16:42:59,721 - Phyluce - INFO - Reading Bartonella_mastomydis_GCF_900185775 genome\n",
      "2020-02-06 16:43:00,218 - Phyluce - INFO - Bartonella_mastomydis_GCF_900185775: 136 uces, 6 dupes, 130 non-dupes, 7 orient drop, 0 length drop, 123 written\n",
      "2020-02-06 16:43:00,218 - Phyluce - INFO - ------ Working on Bartonella_melophagi_GCF_000278255 genome -----\n",
      "2020-02-06 16:43:00,219 - Phyluce - INFO - Reading Bartonella_melophagi_GCF_000278255 genome\n",
      "2020-02-06 16:43:00,710 - Phyluce - INFO - Bartonella_melophagi_GCF_000278255: 136 uces, 12 dupes, 124 non-dupes, 4 orient drop, 0 length drop, 120 written\n",
      "2020-02-06 16:43:00,710 - Phyluce - INFO - --- Working on Bartonella_queenslandensis_GCF_000312585 genome --\n",
      "2020-02-06 16:43:00,711 - Phyluce - INFO - Reading Bartonella_queenslandensis_GCF_000312585 genome\n",
      "2020-02-06 16:43:01,171 - Phyluce - INFO - Bartonella_queenslandensis_GCF_000312585: 135 uces, 8 dupes, 127 non-dupes, 0 orient drop, 0 length drop, 127 written\n",
      "2020-02-06 16:43:01,172 - Phyluce - INFO - ------ Working on Bartonella_quintana_GCF_000046685 genome ------\n",
      "2020-02-06 16:43:01,172 - Phyluce - INFO - Reading Bartonella_quintana_GCF_000046685 genome\n",
      "2020-02-06 16:43:01,652 - Phyluce - INFO - Bartonella_quintana_GCF_000046685: 136 uces, 0 dupes, 136 non-dupes, 7 orient drop, 8 length drop, 121 written\n",
      "2020-02-06 16:43:01,652 - Phyluce - INFO - --- Working on Bartonella_rattaustraliani_GCF_000312565 genome --\n",
      "2020-02-06 16:43:01,683 - Phyluce - INFO - Reading Bartonella_rattaustraliani_GCF_000312565 genome\n",
      "2020-02-06 16:43:02,148 - Phyluce - INFO - Bartonella_rattaustraliani_GCF_000312565: 133 uces, 10 dupes, 123 non-dupes, 0 orient drop, 0 length drop, 123 written\n",
      "2020-02-06 16:43:02,148 - Phyluce - INFO - -- Working on Bartonella_rattimassiliensis_GCF_000278215 genome -\n",
      "2020-02-06 16:43:02,156 - Phyluce - INFO - Reading Bartonella_rattimassiliensis_GCF_000278215 genome\n",
      "2020-02-06 16:43:02,631 - Phyluce - INFO - Bartonella_rattimassiliensis_GCF_000278215: 135 uces, 9 dupes, 126 non-dupes, 1 orient drop, 2 length drop, 123 written\n",
      "2020-02-06 16:43:02,631 - Phyluce - INFO - ----- Working on Bartonella_rochalimae_GCF_000706645 genome -----\n",
      "2020-02-06 16:43:02,641 - Phyluce - INFO - Reading Bartonella_rochalimae_GCF_000706645 genome\n",
      "2020-02-06 16:43:03,148 - Phyluce - INFO - Bartonella_rochalimae_GCF_000706645: 135 uces, 7 dupes, 128 non-dupes, 2 orient drop, 6 length drop, 120 written\n",
      "2020-02-06 16:43:03,148 - Phyluce - INFO - --- Working on Bartonella_schoenbuchensis_GCF_002022685 genome --\n",
      "2020-02-06 16:43:03,150 - Phyluce - INFO - Reading Bartonella_schoenbuchensis_GCF_002022685 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 16:43:03,631 - Phyluce - INFO - Bartonella_schoenbuchensis_GCF_002022685: 136 uces, 0 dupes, 136 non-dupes, 12 orient drop, 4 length drop, 120 written\n",
      "2020-02-06 16:43:03,631 - Phyluce - INFO - ---- Working on Bartonella_senegalensis_GCF_000312545 genome ----\n",
      "2020-02-06 16:43:03,632 - Phyluce - INFO - Reading Bartonella_senegalensis_GCF_000312545 genome\n",
      "2020-02-06 16:43:04,092 - Phyluce - INFO - Bartonella_senegalensis_GCF_000312545: 136 uces, 5 dupes, 131 non-dupes, 2 orient drop, 1 length drop, 128 written\n",
      "2020-02-06 16:43:04,093 - Phyluce - INFO - --------- Working on Bartonella_sp_GCF_003606325 genome ---------\n",
      "2020-02-06 16:43:04,099 - Phyluce - INFO - Reading Bartonella_sp_GCF_003606325 genome\n",
      "2020-02-06 16:43:04,590 - Phyluce - INFO - Bartonella_sp_GCF_003606325: 136 uces, 0 dupes, 136 non-dupes, 7 orient drop, 10 length drop, 119 written\n",
      "2020-02-06 16:43:04,590 - Phyluce - INFO - ------- Working on Bartonella_tamiae_GCF_000279995 genome -------\n",
      "2020-02-06 16:43:04,591 - Phyluce - INFO - Reading Bartonella_tamiae_GCF_000279995 genome\n",
      "2020-02-06 16:43:05,069 - Phyluce - INFO - Bartonella_tamiae_GCF_000279995: 134 uces, 0 dupes, 134 non-dupes, 9 orient drop, 10 length drop, 115 written\n",
      "2020-02-06 16:43:05,069 - Phyluce - INFO - ------ Working on Bartonella_taylorii_GCF_000278295 genome ------\n",
      "2020-02-06 16:43:05,069 - Phyluce - INFO - Reading Bartonella_taylorii_GCF_000278295 genome\n",
      "2020-02-06 16:43:05,537 - Phyluce - INFO - Bartonella_taylorii_GCF_000278295: 136 uces, 10 dupes, 126 non-dupes, 1 orient drop, 0 length drop, 125 written\n",
      "2020-02-06 16:43:05,537 - Phyluce - INFO - ----- Working on Bartonella_tribocorum_GCF_000196435 genome -----\n",
      "2020-02-06 16:43:05,538 - Phyluce - INFO - Reading Bartonella_tribocorum_GCF_000196435 genome\n",
      "2020-02-06 16:43:06,031 - Phyluce - INFO - Bartonella_tribocorum_GCF_000196435: 136 uces, 0 dupes, 136 non-dupes, 9 orient drop, 8 length drop, 119 written\n",
      "2020-02-06 16:43:06,032 - Phyluce - INFO - ------ Working on Bartonella_vinsonii_GCF_000341385 genome ------\n",
      "2020-02-06 16:43:06,041 - Phyluce - INFO - Reading Bartonella_vinsonii_GCF_000341385 genome\n",
      "2020-02-06 16:43:06,543 - Phyluce - INFO - Bartonella_vinsonii_GCF_000341385: 136 uces, 0 dupes, 136 non-dupes, 11 orient drop, 7 length drop, 118 written\n",
      "2020-02-06 16:43:06,544 - Phyluce - INFO - ----- Working on Bartonella_washoeensis_GCF_000278135 genome ----\n",
      "2020-02-06 16:43:06,544 - Phyluce - INFO - Reading Bartonella_washoeensis_GCF_000278135 genome\n",
      "2020-02-06 16:43:07,018 - Phyluce - INFO - Bartonella_washoeensis_GCF_000278135: 136 uces, 11 dupes, 125 non-dupes, 0 orient drop, 0 length drop, 125 written\n",
      "2020-02-06 16:43:07,018 - Phyluce - INFO - ---- Working on Bartonella_bacilliformis_GCF_000015445 genome ---\n",
      "2020-02-06 16:43:07,018 - Phyluce - INFO - Reading Bartonella_bacilliformis_GCF_000015445 genome\n",
      "2020-02-06 16:43:07,565 - Phyluce - INFO - Bartonella_bacilliformis_GCF_000015445: 136 uces, 0 dupes, 136 non-dupes, 8 orient drop, 21 length drop, 107 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/bartonella/extract_probes_from_group/probe_fasta --output results/phyluce/bartonella/extract_probes_from_group/multifastas.sqlite --base-taxon Bartonella_bacilliformis_GCF_000015445;\n",
      "bartonella_alsatica_gcf_000280015.\n",
      "bartonella_ancashensis_gcf_001281405.\n",
      "bartonella_apis_gcf_002007565.\n",
      "bartonella_australis_gcf_000341355.\n",
      "bartonella_birtlesii_gcf_000273375.\n",
      "bartonella_bovis_gcf_000384965.\n",
      "bartonella_clarridgeiae_gcf_000253015.\n",
      "bartonella_doshiae_gcf_000278155.\n",
      "bartonella_elizabethae_gcf_900638615.\n",
      "bartonella_florencae_gcf_000312525.\n",
      "bartonella_grahamii_gcf_000022725.\n",
      "bartonella_henselae_gcf_000046705.\n",
      "bartonella_koehlerae_gcf_000706625.\n",
      "bartonella_massiliensis_gcf_902150025.\n",
      "bartonella_mastomydis_gcf_900185775.\n",
      "bartonella_melophagi_gcf_000278255.\n",
      "bartonella_queenslandensis_gcf_000312585.\n",
      "bartonella_quintana_gcf_000046685.\n",
      "bartonella_rattaustraliani_gcf_000312565.\n",
      "bartonella_rattimassiliensis_gcf_000278215.\n",
      "bartonella_rochalimae_gcf_000706645.\n",
      "bartonella_schoenbuchensis_gcf_002022685.\n",
      "bartonella_senegalensis_gcf_000312545.\n",
      "bartonella_sp_gcf_003606325.\n",
      "bartonella_tamiae_gcf_000279995.\n",
      "bartonella_taylorii_gcf_000278295.\n",
      "bartonella_tribocorum_gcf_000196435.\n",
      "bartonella_vinsonii_gcf_000341385.\n",
      "bartonella_washoeensis_gcf_000278135.\n",
      "bartonella_bacilliformis_gcf_000015445.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/bartonella/extract_probes_from_group/multifastas.sqlite --base-taxon Bartonella_bacilliformis_GCF_000015445;\n",
      "Loci shared by 0 taxa:\t131.0\n",
      "Loci shared by 1 taxa:\t131.0\n",
      "Loci shared by 2 taxa:\t130.0\n",
      "Loci shared by 3 taxa:\t130.0\n",
      "Loci shared by 4 taxa:\t129.0\n",
      "Loci shared by 5 taxa:\t127.0\n",
      "Loci shared by 6 taxa:\t126.0\n",
      "Loci shared by 7 taxa:\t126.0\n",
      "Loci shared by 8 taxa:\t126.0\n",
      "Loci shared by 9 taxa:\t126.0\n",
      "Loci shared by 10 taxa:\t123.0\n",
      "Loci shared by 11 taxa:\t120.0\n",
      "Loci shared by 12 taxa:\t120.0\n",
      "Loci shared by 13 taxa:\t120.0\n",
      "Loci shared by 14 taxa:\t120.0\n",
      "Loci shared by 15 taxa:\t120.0\n",
      "Loci shared by 16 taxa:\t120.0\n",
      "Loci shared by 17 taxa:\t120.0\n",
      "Loci shared by 18 taxa:\t119.0\n",
      "Loci shared by 19 taxa:\t118.0\n",
      "Loci shared by 20 taxa:\t118.0\n",
      "Loci shared by 21 taxa:\t118.0\n",
      "Loci shared by 22 taxa:\t118.0\n",
      "Loci shared by 23 taxa:\t118.0\n",
      "Loci shared by 24 taxa:\t118.0\n",
      "Loci shared by 25 taxa:\t118.0\n",
      "Loci shared by 26 taxa:\t118.0\n",
      "Loci shared by 27 taxa:\t117.0\n",
      "Loci shared by 28 taxa:\t116.0\n",
      "Loci shared by 29 taxa:\t112.0\n",
      "Loci shared by 30 taxa:\t96.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/bartonella/extract_probes_from_group/multifastas.sqlite  --base-taxon Bartonella_bacilliformis_GCF_000015445 --output results/phyluce/bartonella/extract_probes_from_group/Bartonella_bacilliformis_GCF_000015445+11-back-to-29.conf --specific-counts 29;\n",
      "Counter({'bartonella_bovis_gcf_000384965': 112, 'bartonella_rochalimae_gcf_000706645': 112, 'bartonella_tamiae_gcf_000279995': 112, 'bartonella_tribocorum_gcf_000196435': 112, 'bartonella_koehlerae_gcf_000706625': 112, 'bartonella_melophagi_gcf_000278255': 112, 'bartonella_doshiae_gcf_000278155': 112, 'bartonella_birtlesii_gcf_000273375': 112, 'bartonella_henselae_gcf_000046705': 112, 'bartonella_grahamii_gcf_000022725': 112, 'bartonella_schoenbuchensis_gcf_002022685': 112, 'bartonella_massiliensis_gcf_902150025': 112, 'bartonella_australis_gcf_000341355': 112, 'bartonella_elizabethae_gcf_900638615': 112, 'bartonella_ancashensis_gcf_001281405': 112, 'bartonella_rattaustraliani_gcf_000312565': 112, 'bartonella_taylorii_gcf_000278295': 112, 'bartonella_sp_gcf_003606325': 112, 'bartonella_washoeensis_gcf_000278135': 112, 'bartonella_alsatica_gcf_000280015': 112, 'bartonella_mastomydis_gcf_900185775': 112, 'bartonella_apis_gcf_002007565': 112, 'bartonella_florencae_gcf_000312525': 112, 'bartonella_senegalensis_gcf_000312545': 112, 'bartonella_quintana_gcf_000046685': 112, 'bartonella_vinsonii_gcf_000341385': 112, 'bartonella_queenslandensis_gcf_000312585': 111, 'bartonella_rattimassiliensis_gcf_000278215': 111, 'bartonella_clarridgeiae_gcf_000253015': 111, 'bartonella_bacilliformis_gcf_000015445': 99})\n",
      "Total loci = 112\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(29)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/bartonella/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/bartonella/extract_probes_from_group/Bartonella_bacilliformis_GCF_000015445+11-back-to-29.conf --probe-prefix uce_bartonella_ --designer rnplattii --design bartonella_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 112\n",
      "Probe Count = 6644\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list.fasta --query results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  16:45:53\n",
      "Ended:  Thu Feb 06, 2020  16:46:14\n",
      "Time for execution:  0.355904750029 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list.fasta --lastz results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_bartonella_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6643 fasta sequences.  Filtered 0 duplicates. Kept 6644.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/bartonella/final_probe_design/bartonella_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 16:54:31 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 6644\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 531520\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3953696\n",
      "Max number of word counting entries: 88997714\n",
      "\n",
      "# comparing sequences from          0  to       1107\n",
      ".---------- new table with      760 representatives\n",
      "# comparing sequences from       1107  to       2029\n",
      "----------    463 remaining sequences to the next cycle\n",
      "---------- new table with      332 representatives\n",
      "# comparing sequences from       1566  to       2412\n",
      "----------    484 remaining sequences to the next cycle\n",
      "---------- new table with      261 representatives\n",
      "# comparing sequences from       1928  to       2714\n",
      "----------    327 remaining sequences to the next cycle\n",
      "---------- new table with      347 representatives\n",
      "# comparing sequences from       2387  to       3096\n",
      "----------    372 remaining sequences to the next cycle\n",
      "---------- new table with      248 representatives\n",
      "# comparing sequences from       2724  to       3377\n",
      "----------    261 remaining sequences to the next cycle\n",
      "---------- new table with      279 representatives\n",
      "# comparing sequences from       3116  to       3704\n",
      "----------    218 remaining sequences to the next cycle\n",
      "---------- new table with      261 representatives\n",
      "# comparing sequences from       3486  to       4012\n",
      "----------    196 remaining sequences to the next cycle\n",
      "---------- new table with      254 representatives\n",
      "# comparing sequences from       3816  to       4287\n",
      "----------    153 remaining sequences to the next cycle\n",
      "---------- new table with      218 representatives\n",
      "# comparing sequences from       4134  to       4552\n",
      "----------    127 remaining sequences to the next cycle\n",
      "---------- new table with      210 representatives\n",
      "# comparing sequences from       4425  to       4794\n",
      "----------    138 remaining sequences to the next cycle\n",
      "---------- new table with      160 representatives\n",
      "# comparing sequences from       4656  to       4987\n",
      "----------    106 remaining sequences to the next cycle\n",
      "---------- new table with      148 representatives\n",
      "# comparing sequences from       4881  to       5174\n",
      "----------    156 remaining sequences to the next cycle\n",
      "---------- new table with      110 representatives\n",
      "# comparing sequences from       5018  to       5289\n",
      "----------     52 remaining sequences to the next cycle\n",
      "---------- new table with      172 representatives\n",
      "# comparing sequences from       5237  to       5471\n",
      "----------     96 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       5375  to       5586\n",
      "----------     62 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       5524  to       5710\n",
      "----------     46 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       5664  to       6644\n",
      "....................---------- new table with      701 representatives\n",
      "\n",
      "     6644  finished       4761  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.50\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Burkholderia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'burkholderia'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Burkholderia_agricolaris_GCF_009455635\",\n",
    "               \"Burkholderia_ambifaria_GCF_000959545\",\n",
    "               \"Burkholderia_anthina_GCF_001547525\",\n",
    "               \"Burkholderia_bonniea_GCF_009455625\",\n",
    "               \"Burkholderia_catarinensis_GCF_001883705\",\n",
    "               \"Burkholderia_cenocepacia_GCF_001999785\",\n",
    "               \"Burkholderia_cepacia_GCF_001411495\",\n",
    "               \"Burkholderia_contaminans_GCF_001029145\",\n",
    "               \"Burkholderia_diffusa_GCF_001718315\",\n",
    "               \"Burkholderia_dolosa_GCF_003813045\",\n",
    "               \"Burkholderia_gladioli_GCF_000959725\",\n",
    "               \"Burkholderia_glumae_GCF_000960995\",\n",
    "               \"Burkholderia_hayleyella_GCF_009455685\",\n",
    "               \"Burkholderia_humptydooensis_GCF_001462435\",\n",
    "               \"Burkholderia_insecticola_GCF_000402035\",\n",
    "               \"Burkholderia_lata_GCF_000012945\",\n",
    "               \"Burkholderia_latens_GCF_001718795\",\n",
    "               \"Burkholderia_metallica_GCF_001718555\",\n",
    "               \"Burkholderia_multivorans_GCF_000959525\",\n",
    "               \"Burkholderia_novacaledonica_GCF_900258035\",\n",
    "               \"Burkholderia_oklahomensis_GCF_000959365\",\n",
    "               \"Burkholderia_paludis_GCF_000732615\",\n",
    "               \"Burkholderia_plantarii_GCF_001411805\",\n",
    "               \"Burkholderia_pseudomallei_GCF_000011545\",\n",
    "               \"Burkholderia_pseudomultivorans_GCF_001718415\",\n",
    "               \"Burkholderia_puraquae_GCF_002099195\",\n",
    "               \"Burkholderia_pyrrocinia_GCF_001028665\",\n",
    "               \"Burkholderia_reimsis_GCF_003294055\",\n",
    "               \"Burkholderia_seminalis_GCF_001718535\",\n",
    "               \"Burkholderia_singularis_GCF_900176645\",\n",
    "               \"Burkholderia_sp_GCF_005280735\",\n",
    "               \"Burkholderia_stabilis_GCF_001742165\",\n",
    "               \"Burkholderia_stagnalis_GCF_001718955\",\n",
    "               \"Burkholderia_territorii_GCF_001718335\",\n",
    "               \"Burkholderia_thailandensis_GCF_003568605\",\n",
    "               \"Burkholderia_ubonensis_GCF_000959245\",\n",
    "               \"Burkholderia_vietnamiensis_GCF_000959445\" ]\n",
    "                    \n",
    "reference_taxon = \"Burkholderia_mallei_GCF_000011705\"\n",
    "\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009455635.1_ASM945563v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2550418 bytes  1020184.00 bytes/sec\n",
      "total size is 2549688  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000959545.1_ASM95954v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2128660 bytes  1419134.67 bytes/sec\n",
      "total size is 2128035  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001547525.1_ASM154752v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2048639 bytes  1365787.33 bytes/sec\n",
      "total size is 2048029  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009455625.1_ASM945562v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1195955 bytes  797331.33 bytes/sec\n",
      "total size is 1195553  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001883705.1_ASM188370v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2350524 bytes  1567044.00 bytes/sec\n",
      "total size is 2349842  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001999785.1_ASM199978v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2116861 bytes  1411268.67 bytes/sec\n",
      "total size is 2116235  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001411495.1_ASM141149v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2438912 bytes  1625969.33 bytes/sec\n",
      "total size is 2438206  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001029145.1_ASM102914v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2415498 bytes  966216.00 bytes/sec\n",
      "total size is 2414800  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718315.1_ASM171831v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1942717 bytes  1295172.67 bytes/sec\n",
      "total size is 1942131  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003813045.1_ASM381304v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1796801 bytes  1197895.33 bytes/sec\n",
      "total size is 1796255  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000959725.1_ASM95972v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2512846 bytes  1675258.67 bytes/sec\n",
      "total size is 2512125  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000960995.1_ASM96099v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1926199 bytes  1284160.67 bytes/sec\n",
      "total size is 1925622  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009455685.1_ASM945568v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1195008 bytes  796700.00 bytes/sec\n",
      "total size is 1194606  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001462435.1_ASM146243v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1892638 bytes  1261786.67 bytes/sec\n",
      "total size is 1892068  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000402035.1_ASM40203v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2003540 bytes  1335721.33 bytes/sec\n",
      "total size is 2002939  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000012945.1_ASM1294v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2463329 bytes  1642247.33 bytes/sec\n",
      "total size is 2462617  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718795.1_ASM171879v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1874364 bytes  1249604.00 bytes/sec\n",
      "total size is 1873794  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000011705.1_ASM1170v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1617616 bytes  1078438.67 bytes/sec\n",
      "total size is 1617112  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718555.1_ASM171855v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2093001 bytes  1395362.00 bytes/sec\n",
      "total size is 2092383  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000959525.1_ASM95952v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1771205 bytes  1180831.33 bytes/sec\n",
      "total size is 1770660  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900258035.1_LMG28615_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1951701 bytes  1301162.00 bytes/sec\n",
      "total size is 1951118  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000959365.1_ASM95936v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2003289 bytes  1335554.00 bytes/sec\n",
      "total size is 2002688  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000732615.1_ASM73261v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2453364 bytes  981362.40 bytes/sec\n",
      "total size is 2452659  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001411805.1_ASM141180v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2270870 bytes  1513941.33 bytes/sec\n",
      "total size is 2270204  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000011545.1_ASM1154v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2018348 bytes  1345593.33 bytes/sec\n",
      "total size is 2017748  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718415.1_ASM171841v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2238076 bytes  1492078.67 bytes/sec\n",
      "total size is 2237418  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002099195.1_ASM209919v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2301510 bytes  920620.80 bytes/sec\n",
      "total size is 2300836  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001028665.1_ASM102866v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2255233 bytes  1503516.67 bytes/sec\n",
      "total size is 2254575  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003294055.1_ASM329405v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2549455 bytes  1699664.67 bytes/sec\n",
      "total size is 2548725  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718535.1_ASM171853v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2154419 bytes  1436307.33 bytes/sec\n",
      "total size is 2153785  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900176645.1_LMG28154v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1600431 bytes  1066982.00 bytes/sec\n",
      "total size is 1599934  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_005280735.1_ASM528073v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2489990 bytes  1660021.33 bytes/sec\n",
      "total size is 2489276  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001742165.1_ASM174216v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2414315 bytes  1609571.33 bytes/sec\n",
      "total size is 2413617  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718955.1_ASM171895v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2129384 bytes  1419617.33 bytes/sec\n",
      "total size is 2128758  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001718335.1_ASM171833v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1948455 bytes  1298998.00 bytes/sec\n",
      "total size is 1947869  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003568605.1_ASM356860v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1880833 bytes  1253916.67 bytes/sec\n",
      "total size is 1880263  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000959245.1_ASM95924v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 2025877 bytes  1350612.67 bytes/sec\n",
      "total size is 2025276  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000959445.1_ASM95944v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 1958494 bytes  1305690.67 bytes/sec\n",
      "total size is 1957909  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481656 (\"sim_Burkholderia_agricolaris_GCF_009455635\") has been submitted\n",
      "Your job 5481657 (\"sim_Burkholderia_ambifaria_GCF_000959545\") has been submitted\n",
      "Your job 5481658 (\"sim_Burkholderia_anthina_GCF_001547525\") has been submitted\n",
      "Your job 5481659 (\"sim_Burkholderia_bonniea_GCF_009455625\") has been submitted\n",
      "Your job 5481660 (\"sim_Burkholderia_catarinensis_GCF_001883705\") has been submitted\n",
      "Your job 5481661 (\"sim_Burkholderia_cenocepacia_GCF_001999785\") has been submitted\n",
      "Your job 5481662 (\"sim_Burkholderia_cepacia_GCF_001411495\") has been submitted\n",
      "Your job 5481663 (\"sim_Burkholderia_contaminans_GCF_001029145\") has been submitted\n",
      "Your job 5481664 (\"sim_Burkholderia_diffusa_GCF_001718315\") has been submitted\n",
      "Your job 5481665 (\"sim_Burkholderia_dolosa_GCF_003813045\") has been submitted\n",
      "Your job 5481666 (\"sim_Burkholderia_gladioli_GCF_000959725\") has been submitted\n",
      "Your job 5481667 (\"sim_Burkholderia_glumae_GCF_000960995\") has been submitted\n",
      "Your job 5481668 (\"sim_Burkholderia_hayleyella_GCF_009455685\") has been submitted\n",
      "Your job 5481669 (\"sim_Burkholderia_humptydooensis_GCF_001462435\") has been submitted\n",
      "Your job 5481670 (\"sim_Burkholderia_insecticola_GCF_000402035\") has been submitted\n",
      "Your job 5481671 (\"sim_Burkholderia_lata_GCF_000012945\") has been submitted\n",
      "Your job 5481672 (\"sim_Burkholderia_latens_GCF_001718795\") has been submitted\n",
      "Your job 5481673 (\"sim_Burkholderia_metallica_GCF_001718555\") has been submitted\n",
      "Your job 5481674 (\"sim_Burkholderia_multivorans_GCF_000959525\") has been submitted\n",
      "Your job 5481675 (\"sim_Burkholderia_novacaledonica_GCF_900258035\") has been submitted\n",
      "Your job 5481676 (\"sim_Burkholderia_oklahomensis_GCF_000959365\") has been submitted\n",
      "Your job 5481677 (\"sim_Burkholderia_paludis_GCF_000732615\") has been submitted\n",
      "Your job 5481678 (\"sim_Burkholderia_plantarii_GCF_001411805\") has been submitted\n",
      "Your job 5481679 (\"sim_Burkholderia_pseudomallei_GCF_000011545\") has been submitted\n",
      "Your job 5481680 (\"sim_Burkholderia_pseudomultivorans_GCF_001718415\") has been submitted\n",
      "Your job 5481681 (\"sim_Burkholderia_puraquae_GCF_002099195\") has been submitted\n",
      "Your job 5481682 (\"sim_Burkholderia_pyrrocinia_GCF_001028665\") has been submitted\n",
      "Your job 5481683 (\"sim_Burkholderia_reimsis_GCF_003294055\") has been submitted\n",
      "Your job 5481684 (\"sim_Burkholderia_seminalis_GCF_001718535\") has been submitted\n",
      "Your job 5481685 (\"sim_Burkholderia_singularis_GCF_900176645\") has been submitted\n",
      "Your job 5481686 (\"sim_Burkholderia_sp_GCF_005280735\") has been submitted\n",
      "Your job 5481687 (\"sim_Burkholderia_stabilis_GCF_001742165\") has been submitted\n",
      "Your job 5481688 (\"sim_Burkholderia_stagnalis_GCF_001718955\") has been submitted\n",
      "Your job 5481689 (\"sim_Burkholderia_territorii_GCF_001718335\") has been submitted\n",
      "Your job 5481690 (\"sim_Burkholderia_thailandensis_GCF_003568605\") has been submitted\n",
      "Your job 5481691 (\"sim_Burkholderia_ubonensis_GCF_000959245\") has been submitted\n",
      "Your job 5481692 (\"sim_Burkholderia_vietnamiensis_GCF_000959445\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/burkholderia/cleaned_genomes/Burkholderia_mallei_GCF_000011705_formatted.fas path=results/phyluce/burkholderia/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481693 (\"map_Burkholderia_agricolaris_GCF_009455635\") has been submitted\n",
      "Your job 5481694 (\"map_Burkholderia_ambifaria_GCF_000959545\") has been submitted\n",
      "Your job 5481695 (\"map_Burkholderia_anthina_GCF_001547525\") has been submitted\n",
      "Your job 5481696 (\"map_Burkholderia_bonniea_GCF_009455625\") has been submitted\n",
      "Your job 5481697 (\"map_Burkholderia_catarinensis_GCF_001883705\") has been submitted\n",
      "Your job 5481698 (\"map_Burkholderia_cenocepacia_GCF_001999785\") has been submitted\n",
      "Your job 5481699 (\"map_Burkholderia_cepacia_GCF_001411495\") has been submitted\n",
      "Your job 5481700 (\"map_Burkholderia_contaminans_GCF_001029145\") has been submitted\n",
      "Your job 5481701 (\"map_Burkholderia_diffusa_GCF_001718315\") has been submitted\n",
      "Your job 5481702 (\"map_Burkholderia_dolosa_GCF_003813045\") has been submitted\n",
      "Your job 5481703 (\"map_Burkholderia_gladioli_GCF_000959725\") has been submitted\n",
      "Your job 5481704 (\"map_Burkholderia_glumae_GCF_000960995\") has been submitted\n",
      "Your job 5481705 (\"map_Burkholderia_hayleyella_GCF_009455685\") has been submitted\n",
      "Your job 5481706 (\"map_Burkholderia_humptydooensis_GCF_001462435\") has been submitted\n",
      "Your job 5481707 (\"map_Burkholderia_insecticola_GCF_000402035\") has been submitted\n",
      "Your job 5481708 (\"map_Burkholderia_lata_GCF_000012945\") has been submitted\n",
      "Your job 5481709 (\"map_Burkholderia_latens_GCF_001718795\") has been submitted\n",
      "Your job 5481710 (\"map_Burkholderia_metallica_GCF_001718555\") has been submitted\n",
      "Your job 5481711 (\"map_Burkholderia_multivorans_GCF_000959525\") has been submitted\n",
      "Your job 5481712 (\"map_Burkholderia_novacaledonica_GCF_900258035\") has been submitted\n",
      "Your job 5481713 (\"map_Burkholderia_oklahomensis_GCF_000959365\") has been submitted\n",
      "Your job 5481714 (\"map_Burkholderia_paludis_GCF_000732615\") has been submitted\n",
      "Your job 5481715 (\"map_Burkholderia_plantarii_GCF_001411805\") has been submitted\n",
      "Your job 5481716 (\"map_Burkholderia_pseudomallei_GCF_000011545\") has been submitted\n",
      "Your job 5481717 (\"map_Burkholderia_pseudomultivorans_GCF_001718415\") has been submitted\n",
      "Your job 5481718 (\"map_Burkholderia_puraquae_GCF_002099195\") has been submitted\n",
      "Your job 5481719 (\"map_Burkholderia_pyrrocinia_GCF_001028665\") has been submitted\n",
      "Your job 5481720 (\"map_Burkholderia_reimsis_GCF_003294055\") has been submitted\n",
      "Your job 5481721 (\"map_Burkholderia_seminalis_GCF_001718535\") has been submitted\n",
      "Your job 5481722 (\"map_Burkholderia_singularis_GCF_900176645\") has been submitted\n",
      "Your job 5481723 (\"map_Burkholderia_sp_GCF_005280735\") has been submitted\n",
      "Your job 5481724 (\"map_Burkholderia_stabilis_GCF_001742165\") has been submitted\n",
      "Your job 5481725 (\"map_Burkholderia_stagnalis_GCF_001718955\") has been submitted\n",
      "Your job 5481726 (\"map_Burkholderia_territorii_GCF_001718335\") has been submitted\n",
      "Your job 5481727 (\"map_Burkholderia_thailandensis_GCF_003568605\") has been submitted\n",
      "Your job 5481728 (\"map_Burkholderia_ubonensis_GCF_000959245\") has been submitted\n",
      "Your job 5481729 (\"map_Burkholderia_vietnamiensis_GCF_000959445\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481730 (\"merge_Burkholderia_agricolaris_GCF_009455635\") has been submitted\n",
      "Your job 5481731 (\"merge_Burkholderia_ambifaria_GCF_000959545\") has been submitted\n",
      "Your job 5481732 (\"merge_Burkholderia_anthina_GCF_001547525\") has been submitted\n",
      "Your job 5481733 (\"merge_Burkholderia_bonniea_GCF_009455625\") has been submitted\n",
      "Your job 5481734 (\"merge_Burkholderia_catarinensis_GCF_001883705\") has been submitted\n",
      "Your job 5481735 (\"merge_Burkholderia_cenocepacia_GCF_001999785\") has been submitted\n",
      "Your job 5481736 (\"merge_Burkholderia_cepacia_GCF_001411495\") has been submitted\n",
      "Your job 5481737 (\"merge_Burkholderia_contaminans_GCF_001029145\") has been submitted\n",
      "Your job 5481738 (\"merge_Burkholderia_diffusa_GCF_001718315\") has been submitted\n",
      "Your job 5481739 (\"merge_Burkholderia_dolosa_GCF_003813045\") has been submitted\n",
      "Your job 5481740 (\"merge_Burkholderia_gladioli_GCF_000959725\") has been submitted\n",
      "Your job 5481741 (\"merge_Burkholderia_glumae_GCF_000960995\") has been submitted\n",
      "Your job 5481742 (\"merge_Burkholderia_hayleyella_GCF_009455685\") has been submitted\n",
      "Your job 5481743 (\"merge_Burkholderia_humptydooensis_GCF_001462435\") has been submitted\n",
      "Your job 5481744 (\"merge_Burkholderia_insecticola_GCF_000402035\") has been submitted\n",
      "Your job 5481745 (\"merge_Burkholderia_lata_GCF_000012945\") has been submitted\n",
      "Your job 5481746 (\"merge_Burkholderia_latens_GCF_001718795\") has been submitted\n",
      "Your job 5481747 (\"merge_Burkholderia_metallica_GCF_001718555\") has been submitted\n",
      "Your job 5481748 (\"merge_Burkholderia_multivorans_GCF_000959525\") has been submitted\n",
      "Your job 5481749 (\"merge_Burkholderia_novacaledonica_GCF_900258035\") has been submitted\n",
      "Your job 5481750 (\"merge_Burkholderia_oklahomensis_GCF_000959365\") has been submitted\n",
      "Your job 5481751 (\"merge_Burkholderia_paludis_GCF_000732615\") has been submitted\n",
      "Your job 5481752 (\"merge_Burkholderia_plantarii_GCF_001411805\") has been submitted\n",
      "Your job 5481753 (\"merge_Burkholderia_pseudomallei_GCF_000011545\") has been submitted\n",
      "Your job 5481754 (\"merge_Burkholderia_pseudomultivorans_GCF_001718415\") has been submitted\n",
      "Your job 5481755 (\"merge_Burkholderia_puraquae_GCF_002099195\") has been submitted\n",
      "Your job 5481756 (\"merge_Burkholderia_pyrrocinia_GCF_001028665\") has been submitted\n",
      "Your job 5481757 (\"merge_Burkholderia_reimsis_GCF_003294055\") has been submitted\n",
      "Your job 5481758 (\"merge_Burkholderia_seminalis_GCF_001718535\") has been submitted\n",
      "Your job 5481759 (\"merge_Burkholderia_singularis_GCF_900176645\") has been submitted\n",
      "Your job 5481760 (\"merge_Burkholderia_sp_GCF_005280735\") has been submitted\n",
      "Your job 5481761 (\"merge_Burkholderia_stabilis_GCF_001742165\") has been submitted\n",
      "Your job 5481762 (\"merge_Burkholderia_stagnalis_GCF_001718955\") has been submitted\n",
      "Your job 5481763 (\"merge_Burkholderia_territorii_GCF_001718335\") has been submitted\n",
      "Your job 5481764 (\"merge_Burkholderia_thailandensis_GCF_003568605\") has been submitted\n",
      "Your job 5481765 (\"merge_Burkholderia_ubonensis_GCF_000959245\") has been submitted\n",
      "Your job 5481766 (\"merge_Burkholderia_vietnamiensis_GCF_000959445\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 2546 sequences from Burkholderia_agricolaris_GCF_009455635_merged.bed.  Filtered 1621 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 925.\n",
      "Screened 5940 sequences from Burkholderia_ambifaria_GCF_000959545_merged.bed.  Filtered 2660 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3280.\n",
      "Screened 5988 sequences from Burkholderia_anthina_GCF_001547525_merged.bed.  Filtered 2724 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3264.\n",
      "Screened 670 sequences from Burkholderia_bonniea_GCF_009455625_merged.bed.  Filtered 519 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 151.\n",
      "Screened 6137 sequences from Burkholderia_catarinensis_GCF_001883705_merged.bed.  Filtered 2805 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3332.\n",
      "Screened 6075 sequences from Burkholderia_cenocepacia_GCF_001999785_merged.bed.  Filtered 2766 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3309.\n",
      "Screened 6080 sequences from Burkholderia_cepacia_GCF_001411495_merged.bed.  Filtered 2721 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3359.\n",
      "Screened 6059 sequences from Burkholderia_contaminans_GCF_001029145_merged.bed.  Filtered 2771 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3288.\n",
      "Screened 5849 sequences from Burkholderia_diffusa_GCF_001718315_merged.bed.  Filtered 2681 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3168.\n",
      "Screened 5827 sequences from Burkholderia_dolosa_GCF_003813045_merged.bed.  Filtered 2579 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3248.\n",
      "Screened 4823 sequences from Burkholderia_gladioli_GCF_000959725_merged.bed.  Filtered 2603 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2220.\n",
      "Screened 4941 sequences from Burkholderia_glumae_GCF_000960995_merged.bed.  Filtered 2454 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2487.\n",
      "Screened 778 sequences from Burkholderia_hayleyella_GCF_009455685_merged.bed.  Filtered 589 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 189.\n",
      "Screened 4014 sequences from Burkholderia_humptydooensis_GCF_001462435_merged.bed.  Filtered 525 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3489.\n",
      "Screened 2927 sequences from Burkholderia_insecticola_GCF_000402035_merged.bed.  Filtered 1753 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1174.\n",
      "Screened 5972 sequences from Burkholderia_lata_GCF_000012945_merged.bed.  Filtered 2713 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3259.\n",
      "Screened 5728 sequences from Burkholderia_latens_GCF_001718795_merged.bed.  Filtered 2656 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3072.\n",
      "Screened 6003 sequences from Burkholderia_metallica_GCF_001718555_merged.bed.  Filtered 2689 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3314.\n",
      "Screened 5907 sequences from Burkholderia_multivorans_GCF_000959525_merged.bed.  Filtered 2676 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3231.\n",
      "Screened 3033 sequences from Burkholderia_novacaledonica_GCF_900258035_merged.bed.  Filtered 1814 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1219.\n",
      "Screened 6062 sequences from Burkholderia_oklahomensis_GCF_000959365_merged.bed.  Filtered 1215 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 4847.\n",
      "Screened 5933 sequences from Burkholderia_paludis_GCF_000732615_merged.bed.  Filtered 2662 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3271.\n",
      "Screened 5219 sequences from Burkholderia_plantarii_GCF_001411805_merged.bed.  Filtered 2573 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 2646.\n",
      "Screened 553 sequences from Burkholderia_pseudomallei_GCF_000011545_merged.bed.  Filtered 28 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 525.\n",
      "Screened 6117 sequences from Burkholderia_pseudomultivorans_GCF_001718415_merged.bed.  Filtered 2728 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3389.\n",
      "Screened 6082 sequences from Burkholderia_puraquae_GCF_002099195_merged.bed.  Filtered 2772 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3310.\n",
      "Screened 6068 sequences from Burkholderia_pyrrocinia_GCF_001028665_merged.bed.  Filtered 2672 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3396.\n",
      "Screened 6118 sequences from Burkholderia_reimsis_GCF_003294055_merged.bed.  Filtered 2807 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3311.\n",
      "Screened 6101 sequences from Burkholderia_seminalis_GCF_001718535_merged.bed.  Filtered 2765 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3336.\n",
      "Screened 6470 sequences from Burkholderia_singularis_GCF_900176645_merged.bed.  Filtered 2743 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3727.\n",
      "Screened 3750 sequences from Burkholderia_sp_GCF_005280735_merged.bed.  Filtered 1995 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1755.\n",
      "Screened 6168 sequences from Burkholderia_stabilis_GCF_001742165_merged.bed.  Filtered 2806 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3362.\n",
      "Screened 6281 sequences from Burkholderia_stagnalis_GCF_001718955_merged.bed.  Filtered 2706 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3575.\n",
      "Screened 5882 sequences from Burkholderia_territorii_GCF_001718335_merged.bed.  Filtered 2652 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3230.\n",
      "Screened 4278 sequences from Burkholderia_thailandensis_GCF_003568605_merged.bed.  Filtered 592 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3686.\n",
      "Screened 6363 sequences from Burkholderia_ubonensis_GCF_000959245_merged.bed.  Filtered 2696 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3667.\n",
      "Screened 5691 sequences from Burkholderia_vietnamiensis_GCF_000959445_merged.bed.  Filtered 2590 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 3101.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/burkholderia/initial_intervals/Burkholderia_vietnamiensis_GCF_000959445_merged.bed --twobit results/phyluce/burkholderia/cleaned_genomes/Burkholderia_mallei_GCF_000011705_formatted.2bit --output results/phyluce/burkholderia/initial_intervals/Burkholderia_vietnamiensis_GCF_000959445_stripped.bed;\n",
      "burkholderia_agricolaris_gcf_009455635.\n",
      "burkholderia_ambifaria_gcf_000959545....\n",
      "burkholderia_anthina_gcf_001547525....\n",
      "burkholderia_bonniea_gcf_009455625.\n",
      "burkholderia_catarinensis_gcf_001883705....\n",
      "burkholderia_cenocepacia_gcf_001999785....\n",
      "burkholderia_cepacia_gcf_001411495....\n",
      "burkholderia_contaminans_gcf_001029145....\n",
      "burkholderia_diffusa_gcf_001718315....\n",
      "burkholderia_dolosa_gcf_003813045....\n",
      "burkholderia_gladioli_gcf_000959725...\n",
      "burkholderia_glumae_gcf_000960995...\n",
      "burkholderia_hayleyella_gcf_009455685.\n",
      "burkholderia_humptydooensis_gcf_001462435....\n",
      "burkholderia_insecticola_gcf_000402035..\n",
      "burkholderia_lata_gcf_000012945....\n",
      "burkholderia_latens_gcf_001718795....\n",
      "burkholderia_metallica_gcf_001718555....\n",
      "burkholderia_multivorans_gcf_000959525....\n",
      "burkholderia_novacaledonica_gcf_900258035..\n",
      "burkholderia_oklahomensis_gcf_000959365.....\n",
      "burkholderia_paludis_gcf_000732615....\n",
      "burkholderia_plantarii_gcf_001411805...\n",
      "burkholderia_pseudomallei_gcf_000011545.\n",
      "burkholderia_pseudomultivorans_gcf_001718415....\n",
      "burkholderia_puraquae_gcf_002099195....\n",
      "burkholderia_pyrrocinia_gcf_001028665....\n",
      "burkholderia_reimsis_gcf_003294055....\n",
      "burkholderia_seminalis_gcf_001718535....\n",
      "burkholderia_singularis_gcf_900176645....\n",
      "burkholderia_sp_gcf_005280735..\n",
      "burkholderia_stabilis_gcf_001742165....\n",
      "burkholderia_stagnalis_gcf_001718955....\n",
      "burkholderia_territorii_gcf_001718335....\n",
      "burkholderia_thailandensis_gcf_003568605....\n",
      "burkholderia_ubonensis_gcf_000959245....\n",
      "burkholderia_vietnamiensis_gcf_000959445....\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/burkholderia/initial_intervals/Burkholderia_vietnamiensis_GCF_000959445_merged.bed --twobit results/phyluce/burkholderia/cleaned_genomes/Burkholderia_mallei_GCF_000011705_formatted.2bit --output results/phyluce/burkholderia/initial_intervals/Burkholderia_vietnamiensis_GCF_000959445_stripped.bed;\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 0 taxa:\t9,314.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 1 taxa:\t9,314.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 2 taxa:\t8,927.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 3 taxa:\t8,337.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 4 taxa:\t7,220.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 5 taxa:\t6,128.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 6 taxa:\t5,723.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 7 taxa:\t5,469.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 8 taxa:\t5,214.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 9 taxa:\t5,020.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 10 taxa:\t4,846.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 11 taxa:\t4,688.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 12 taxa:\t4,535.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 13 taxa:\t4,390.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 14 taxa:\t4,251.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 15 taxa:\t4,112.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 16 taxa:\t4,003.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 17 taxa:\t3,886.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 18 taxa:\t3,782.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 19 taxa:\t3,674.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 20 taxa:\t3,564.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 21 taxa:\t3,446.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 22 taxa:\t3,330.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 23 taxa:\t3,174.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 24 taxa:\t3,009.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 25 taxa:\t2,854.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 26 taxa:\t2,701.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 27 taxa:\t2,483.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 28 taxa:\t2,279.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 29 taxa:\t2,075.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 30 taxa:\t1,847.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 31 taxa:\t1,592.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 32 taxa:\t1,294.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 33 taxa:\t1,032.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 34 taxa:\t757.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 35 taxa:\t485.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 36 taxa:\t153.0\n",
      "Loci shared by Burkholderia_mallei_GCF_000011705 + 37 taxa:\t80.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/burkholderia/initial_intervals/burkholderia-to-Burkholderia_mallei_GCF_000011705.sqlite --base-taxon Burkholderia_mallei_GCF_000011705 --output results/phyluce/burkholderia/initial_intervals/Burkholderia_mallei_GCF_000011705_+36.bed --specific-counts 36;\n",
      "Counter({'burkholderia_ambifaria_gcf_000959545': 153, 'burkholderia_humptydooensis_gcf_001462435': 153, 'burkholderia_pyrrocinia_gcf_001028665': 153, 'burkholderia_dolosa_gcf_003813045': 153, 'burkholderia_contaminans_gcf_001029145': 153, 'burkholderia_cepacia_gcf_001411495': 153, 'burkholderia_plantarii_gcf_001411805': 153, 'burkholderia_latens_gcf_001718795': 153, 'burkholderia_cenocepacia_gcf_001999785': 153, 'burkholderia_vietnamiensis_gcf_000959445': 153, 'burkholderia_gladioli_gcf_000959725': 153, 'burkholderia_insecticola_gcf_000402035': 153, 'burkholderia_ubonensis_gcf_000959245': 153, 'burkholderia_territorii_gcf_001718335': 153, 'burkholderia_glumae_gcf_000960995': 153, 'burkholderia_catarinensis_gcf_001883705': 153, 'burkholderia_pseudomultivorans_gcf_001718415': 153, 'burkholderia_agricolaris_gcf_009455635': 153, 'burkholderia_lata_gcf_000012945': 153, 'burkholderia_puraquae_gcf_002099195': 153, 'burkholderia_multivorans_gcf_000959525': 153, 'burkholderia_diffusa_gcf_001718315': 153, 'burkholderia_metallica_gcf_001718555': 153, 'burkholderia_thailandensis_gcf_003568605': 153, 'burkholderia_pseudomallei_gcf_000011545': 153, 'burkholderia_seminalis_gcf_001718535': 153, 'burkholderia_stabilis_gcf_001742165': 153, 'burkholderia_stagnalis_gcf_001718955': 153, 'burkholderia_oklahomensis_gcf_000959365': 153, 'burkholderia_paludis_gcf_000732615': 153, 'burkholderia_reimsis_gcf_003294055': 153, 'burkholderia_anthina_gcf_001547525': 153, 'burkholderia_novacaledonica_gcf_900258035': 152, 'burkholderia_singularis_gcf_900176645': 152, 'burkholderia_sp_gcf_005280735': 152, 'burkholderia_hayleyella_gcf_009455685': 129, 'burkholderia_bonniea_gcf_009455625': 107})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 36\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/burkholderia/initial_intervals/Burkholderia_mallei_GCF_000011705_+36.bed --twobit results/phyluce/burkholderia/cleaned_genomes/Burkholderia_mallei_GCF_000011705_formatted.2bit --buffer-to 160 --output results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36.fasta;\n",
      "Screened 153 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 153.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36.fasta --probe-prefix uce_burkholderia_ --design burkholderia_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 149\n",
      "Probe Count = 292\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas --query results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  17:10:16\n",
      "Ended:  Thu Feb 06, 2020  17:10:16\n",
      "Time for execution:  0.00308758020401 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas                        --lastz results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes_vself.lastz                       --probe-prefix=uce_burkholderia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 291 fasta sequences.  Filtered 2 duplicates. Kept 288.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/burkholderia/validate_intervals/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas --scaffoldlist Burkholderia_agricolaris_GCF_009455635 Burkholderia_ambifaria_GCF_000959545 Burkholderia_anthina_GCF_001547525 Burkholderia_bonniea_GCF_009455625 Burkholderia_catarinensis_GCF_001883705 Burkholderia_cenocepacia_GCF_001999785 Burkholderia_cepacia_GCF_001411495 Burkholderia_contaminans_GCF_001029145 Burkholderia_diffusa_GCF_001718315 Burkholderia_dolosa_GCF_003813045 Burkholderia_gladioli_GCF_000959725 Burkholderia_glumae_GCF_000960995 Burkholderia_hayleyella_GCF_009455685 Burkholderia_humptydooensis_GCF_001462435 Burkholderia_insecticola_GCF_000402035 Burkholderia_lata_GCF_000012945 Burkholderia_latens_GCF_001718795 Burkholderia_metallica_GCF_001718555 Burkholderia_multivorans_GCF_000959525 Burkholderia_novacaledonica_GCF_900258035 Burkholderia_oklahomensis_GCF_000959365 Burkholderia_paludis_GCF_000732615 Burkholderia_plantarii_GCF_001411805 Burkholderia_pseudomallei_GCF_000011545 Burkholderia_pseudomultivorans_GCF_001718415 Burkholderia_puraquae_GCF_002099195 Burkholderia_pyrrocinia_GCF_001028665 Burkholderia_reimsis_GCF_003294055 Burkholderia_seminalis_GCF_001718535 Burkholderia_singularis_GCF_900176645 Burkholderia_sp_GCF_005280735 Burkholderia_stabilis_GCF_001742165 Burkholderia_stagnalis_GCF_001718955 Burkholderia_territorii_GCF_001718335 Burkholderia_thailandensis_GCF_003568605 Burkholderia_ubonensis_GCF_000959245 Burkholderia_vietnamiensis_GCF_000959445 Burkholderia_mallei_GCF_000011705 --genome-base-path results/phyluce/burkholderia/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/burkholderia/validate_intervals/burkholderia-to-Burkholderia_mallei_GCF_000011705.sqlite --output results/phyluce/burkholderia/validate_intervals/lastz/;\n",
      "\n",
      "Running against Burkholderia_agricolaris_GCF_009455635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_oISVp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplOEfMs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_agricolaris_GCF_009455635.lastz\n",
      "Creating Burkholderia_agricolaris_GCF_009455635 table\n",
      "Inserting data to Burkholderia_agricolaris_GCF_009455635 table\n",
      "\n",
      "Running against Burkholderia_ambifaria_GCF_000959545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp5ykFgN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptAYLXr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_ambifaria_GCF_000959545.lastz\n",
      "Creating Burkholderia_ambifaria_GCF_000959545 table\n",
      "Inserting data to Burkholderia_ambifaria_GCF_000959545 table\n",
      "\n",
      "Running against Burkholderia_anthina_GCF_001547525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpi3nx6o.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGnhOYO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_anthina_GCF_001547525.lastz\n",
      "Creating Burkholderia_anthina_GCF_001547525 table\n",
      "Inserting data to Burkholderia_anthina_GCF_001547525 table\n",
      "\n",
      "Running against Burkholderia_bonniea_GCF_009455625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDzOBHB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptOhhCR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_bonniea_GCF_009455625.lastz\n",
      "Creating Burkholderia_bonniea_GCF_009455625 table\n",
      "Inserting data to Burkholderia_bonniea_GCF_009455625 table\n",
      "\n",
      "Running against Burkholderia_catarinensis_GCF_001883705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYyTYsP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplWGI5t.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_catarinensis_GCF_001883705.lastz\n",
      "Creating Burkholderia_catarinensis_GCF_001883705 table\n",
      "Inserting data to Burkholderia_catarinensis_GCF_001883705 table\n",
      "\n",
      "Running against Burkholderia_cenocepacia_GCF_001999785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvRIi5K.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpB4mGZy.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_cenocepacia_GCF_001999785.lastz\n",
      "Creating Burkholderia_cenocepacia_GCF_001999785 table\n",
      "Inserting data to Burkholderia_cenocepacia_GCF_001999785 table\n",
      "\n",
      "Running against Burkholderia_cepacia_GCF_001411495.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplQBvUa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIQedqo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_cepacia_GCF_001411495.lastz\n",
      "Creating Burkholderia_cepacia_GCF_001411495 table\n",
      "Inserting data to Burkholderia_cepacia_GCF_001411495 table\n",
      "\n",
      "Running against Burkholderia_contaminans_GCF_001029145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYWtqwZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpvXT5S0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_contaminans_GCF_001029145.lastz\n",
      "Creating Burkholderia_contaminans_GCF_001029145 table\n",
      "Inserting data to Burkholderia_contaminans_GCF_001029145 table\n",
      "\n",
      "Running against Burkholderia_diffusa_GCF_001718315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCOoWP0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpsov9dj.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_diffusa_GCF_001718315.lastz\n",
      "Creating Burkholderia_diffusa_GCF_001718315 table\n",
      "Inserting data to Burkholderia_diffusa_GCF_001718315 table\n",
      "\n",
      "Running against Burkholderia_dolosa_GCF_003813045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWVG2Fa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgetePf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_dolosa_GCF_003813045.lastz\n",
      "Creating Burkholderia_dolosa_GCF_003813045 table\n",
      "Inserting data to Burkholderia_dolosa_GCF_003813045 table\n",
      "\n",
      "Running against Burkholderia_gladioli_GCF_000959725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpjI2f9C.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgcQftO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_gladioli_GCF_000959725.lastz\n",
      "Creating Burkholderia_gladioli_GCF_000959725 table\n",
      "Inserting data to Burkholderia_gladioli_GCF_000959725 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Burkholderia_glumae_GCF_000960995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp4wTLPw.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzM4ijp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_glumae_GCF_000960995.lastz\n",
      "Creating Burkholderia_glumae_GCF_000960995 table\n",
      "Inserting data to Burkholderia_glumae_GCF_000960995 table\n",
      "\n",
      "Running against Burkholderia_hayleyella_GCF_009455685.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZ3w3W3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl3Ggpx.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_hayleyella_GCF_009455685.lastz\n",
      "Creating Burkholderia_hayleyella_GCF_009455685 table\n",
      "Inserting data to Burkholderia_hayleyella_GCF_009455685 table\n",
      "\n",
      "Running against Burkholderia_humptydooensis_GCF_001462435.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp1PqaeG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPSxkv1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_humptydooensis_GCF_001462435.lastz\n",
      "Creating Burkholderia_humptydooensis_GCF_001462435 table\n",
      "Inserting data to Burkholderia_humptydooensis_GCF_001462435 table\n",
      "\n",
      "Running against Burkholderia_insecticola_GCF_000402035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmw8yFc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwPdSVn.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_insecticola_GCF_000402035.lastz\n",
      "Creating Burkholderia_insecticola_GCF_000402035 table\n",
      "Inserting data to Burkholderia_insecticola_GCF_000402035 table\n",
      "\n",
      "Running against Burkholderia_lata_GCF_000012945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpM7JM5f.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXit0BX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_lata_GCF_000012945.lastz\n",
      "Creating Burkholderia_lata_GCF_000012945 table\n",
      "Inserting data to Burkholderia_lata_GCF_000012945 table\n",
      "\n",
      "Running against Burkholderia_latens_GCF_001718795.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprIgdvB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8bYkGE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_latens_GCF_001718795.lastz\n",
      "Creating Burkholderia_latens_GCF_001718795 table\n",
      "Inserting data to Burkholderia_latens_GCF_001718795 table\n",
      "\n",
      "Running against Burkholderia_metallica_GCF_001718555.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpt7FDR3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpI_mLGp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_metallica_GCF_001718555.lastz\n",
      "Creating Burkholderia_metallica_GCF_001718555 table\n",
      "Inserting data to Burkholderia_metallica_GCF_001718555 table\n",
      "\n",
      "Running against Burkholderia_multivorans_GCF_000959525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPl0zqs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQZisIR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_multivorans_GCF_000959525.lastz\n",
      "Creating Burkholderia_multivorans_GCF_000959525 table\n",
      "Inserting data to Burkholderia_multivorans_GCF_000959525 table\n",
      "\n",
      "Running against Burkholderia_novacaledonica_GCF_900258035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnmVVvs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgNUbHG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_novacaledonica_GCF_900258035.lastz\n",
      "Creating Burkholderia_novacaledonica_GCF_900258035 table\n",
      "Inserting data to Burkholderia_novacaledonica_GCF_900258035 table\n",
      "\n",
      "Running against Burkholderia_oklahomensis_GCF_000959365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptiwBcf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRxt6RS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_oklahomensis_GCF_000959365.lastz\n",
      "Creating Burkholderia_oklahomensis_GCF_000959365 table\n",
      "Inserting data to Burkholderia_oklahomensis_GCF_000959365 table\n",
      "\n",
      "Running against Burkholderia_paludis_GCF_000732615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyJCrEo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp81Y4Su.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_paludis_GCF_000732615.lastz\n",
      "Creating Burkholderia_paludis_GCF_000732615 table\n",
      "Inserting data to Burkholderia_paludis_GCF_000732615 table\n",
      "\n",
      "Running against Burkholderia_plantarii_GCF_001411805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpDw5U1o.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_QqTdB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_plantarii_GCF_001411805.lastz\n",
      "Creating Burkholderia_plantarii_GCF_001411805 table\n",
      "Inserting data to Burkholderia_plantarii_GCF_001411805 table\n",
      "\n",
      "Running against Burkholderia_pseudomallei_GCF_000011545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_HuUWE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZ0DNTw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_pseudomallei_GCF_000011545.lastz\n",
      "Creating Burkholderia_pseudomallei_GCF_000011545 table\n",
      "Inserting data to Burkholderia_pseudomallei_GCF_000011545 table\n",
      "\n",
      "Running against Burkholderia_pseudomultivorans_GCF_001718415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaHAAtb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBJHOBg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_pseudomultivorans_GCF_001718415.lastz\n",
      "Creating Burkholderia_pseudomultivorans_GCF_001718415 table\n",
      "Inserting data to Burkholderia_pseudomultivorans_GCF_001718415 table\n",
      "\n",
      "Running against Burkholderia_puraquae_GCF_002099195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpzxCplW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkaxWlo.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_puraquae_GCF_002099195.lastz\n",
      "Creating Burkholderia_puraquae_GCF_002099195 table\n",
      "Inserting data to Burkholderia_puraquae_GCF_002099195 table\n",
      "\n",
      "Running against Burkholderia_pyrrocinia_GCF_001028665.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBqaoau.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5q6zIR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_pyrrocinia_GCF_001028665.lastz\n",
      "Creating Burkholderia_pyrrocinia_GCF_001028665 table\n",
      "Inserting data to Burkholderia_pyrrocinia_GCF_001028665 table\n",
      "\n",
      "Running against Burkholderia_reimsis_GCF_003294055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpng2lBK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnF7r79.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_reimsis_GCF_003294055.lastz\n",
      "Creating Burkholderia_reimsis_GCF_003294055 table\n",
      "Inserting data to Burkholderia_reimsis_GCF_003294055 table\n",
      "\n",
      "Running against Burkholderia_seminalis_GCF_001718535.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpy9dVkh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmplLtOl3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_seminalis_GCF_001718535.lastz\n",
      "Creating Burkholderia_seminalis_GCF_001718535 table\n",
      "Inserting data to Burkholderia_seminalis_GCF_001718535 table\n",
      "\n",
      "Running against Burkholderia_singularis_GCF_900176645.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpo1n2yP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEdTrRP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_singularis_GCF_900176645.lastz\n",
      "Creating Burkholderia_singularis_GCF_900176645 table\n",
      "Inserting data to Burkholderia_singularis_GCF_900176645 table\n",
      "\n",
      "Running against Burkholderia_sp_GCF_005280735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPHRtHi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxUSweh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_sp_GCF_005280735.lastz\n",
      "Creating Burkholderia_sp_GCF_005280735 table\n",
      "Inserting data to Burkholderia_sp_GCF_005280735 table\n",
      "\n",
      "Running against Burkholderia_stabilis_GCF_001742165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6srcFa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6vlTsE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_stabilis_GCF_001742165.lastz\n",
      "Creating Burkholderia_stabilis_GCF_001742165 table\n",
      "Inserting data to Burkholderia_stabilis_GCF_001742165 table\n",
      "\n",
      "Running against Burkholderia_stagnalis_GCF_001718955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGYxd7n.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpoUz0KT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_stagnalis_GCF_001718955.lastz\n",
      "Creating Burkholderia_stagnalis_GCF_001718955 table\n",
      "Inserting data to Burkholderia_stagnalis_GCF_001718955 table\n",
      "\n",
      "Running against Burkholderia_territorii_GCF_001718335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMz7c0y.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpn3BNqJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_territorii_GCF_001718335.lastz\n",
      "Creating Burkholderia_territorii_GCF_001718335 table\n",
      "Inserting data to Burkholderia_territorii_GCF_001718335 table\n",
      "\n",
      "Running against Burkholderia_thailandensis_GCF_003568605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplV4f8O.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmYx47m.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_thailandensis_GCF_003568605.lastz\n",
      "Creating Burkholderia_thailandensis_GCF_003568605 table\n",
      "Inserting data to Burkholderia_thailandensis_GCF_003568605 table\n",
      "\n",
      "Running against Burkholderia_ubonensis_GCF_000959245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprA93tO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpWQmeny.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_ubonensis_GCF_000959245.lastz\n",
      "Creating Burkholderia_ubonensis_GCF_000959245 table\n",
      "Inserting data to Burkholderia_ubonensis_GCF_000959245 table\n",
      "\n",
      "Running against Burkholderia_vietnamiensis_GCF_000959445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpww7K1M.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0CmB2Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_vietnamiensis_GCF_000959445.lastz\n",
      "Creating Burkholderia_vietnamiensis_GCF_000959445 table\n",
      "Inserting data to Burkholderia_vietnamiensis_GCF_000959445 table\n",
      "\n",
      "Running against Burkholderia_mallei_GCF_000011705.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQHhZFC.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpOGfg44.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/burkholderia/validate_intervals/lastz/Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_Burkholderia_mallei_GCF_000011705.lastz\n",
      "Creating Burkholderia_mallei_GCF_000011705 table\n",
      "Inserting data to Burkholderia_mallei_GCF_000011705 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/burkholderia/extract_probes_from_group/burkholderia_genome.conf --lastz results/phyluce/burkholderia/validate_intervals/lastz --probes 120 --probe-prefix uce_burkholderia_ --name-pattern \"Burkholderia_mallei_GCF_000011705_+36_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/burkholderia/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 17:12:20,797 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 17:12:20,800 - Phyluce - INFO - ---- Working on Burkholderia_agricolaris_GCF_009455635 genome ---\n",
      "2020-02-06 17:12:20,801 - Phyluce - INFO - Reading Burkholderia_agricolaris_GCF_009455635 genome\n",
      "2020-02-06 17:12:21,652 - Phyluce - INFO - Burkholderia_agricolaris_GCF_009455635: 149 uces, 27 dupes, 122 non-dupes, 11 orient drop, 8 length drop, 103 written\n",
      "2020-02-06 17:12:21,652 - Phyluce - INFO - ----- Working on Burkholderia_ambifaria_GCF_000959545 genome ----\n",
      "2020-02-06 17:12:21,662 - Phyluce - INFO - Reading Burkholderia_ambifaria_GCF_000959545 genome\n",
      "2020-02-06 17:12:22,427 - Phyluce - INFO - Burkholderia_ambifaria_GCF_000959545: 149 uces, 31 dupes, 118 non-dupes, 0 orient drop, 16 length drop, 102 written\n",
      "2020-02-06 17:12:22,427 - Phyluce - INFO - ------ Working on Burkholderia_anthina_GCF_001547525 genome -----\n",
      "2020-02-06 17:12:22,428 - Phyluce - INFO - Reading Burkholderia_anthina_GCF_001547525 genome\n",
      "2020-02-06 17:12:23,195 - Phyluce - INFO - Burkholderia_anthina_GCF_001547525: 149 uces, 28 dupes, 121 non-dupes, 10 orient drop, 8 length drop, 103 written\n",
      "2020-02-06 17:12:23,195 - Phyluce - INFO - ------ Working on Burkholderia_bonniea_GCF_009455625 genome -----\n",
      "2020-02-06 17:12:23,196 - Phyluce - INFO - Reading Burkholderia_bonniea_GCF_009455625 genome\n",
      "2020-02-06 17:12:23,790 - Phyluce - INFO - Burkholderia_bonniea_GCF_009455625: 148 uces, 13 dupes, 135 non-dupes, 3 orient drop, 7 length drop, 125 written\n",
      "2020-02-06 17:12:23,790 - Phyluce - INFO - --- Working on Burkholderia_catarinensis_GCF_001883705 genome ---\n",
      "2020-02-06 17:12:23,791 - Phyluce - INFO - Reading Burkholderia_catarinensis_GCF_001883705 genome\n",
      "2020-02-06 17:12:24,428 - Phyluce - INFO - Burkholderia_catarinensis_GCF_001883705: 149 uces, 38 dupes, 111 non-dupes, 0 orient drop, 1 length drop, 110 written\n",
      "2020-02-06 17:12:24,428 - Phyluce - INFO - ---- Working on Burkholderia_cenocepacia_GCF_001999785 genome ---\n",
      "2020-02-06 17:12:24,429 - Phyluce - INFO - Reading Burkholderia_cenocepacia_GCF_001999785 genome\n",
      "2020-02-06 17:12:25,175 - Phyluce - INFO - Burkholderia_cenocepacia_GCF_001999785: 149 uces, 0 dupes, 149 non-dupes, 31 orient drop, 13 length drop, 105 written\n",
      "2020-02-06 17:12:25,175 - Phyluce - INFO - ------ Working on Burkholderia_cepacia_GCF_001411495 genome -----\n",
      "2020-02-06 17:12:25,175 - Phyluce - INFO - Reading Burkholderia_cepacia_GCF_001411495 genome\n",
      "2020-02-06 17:12:25,974 - Phyluce - INFO - Burkholderia_cepacia_GCF_001411495: 149 uces, 36 dupes, 113 non-dupes, 2 orient drop, 15 length drop, 96 written\n",
      "2020-02-06 17:12:25,975 - Phyluce - INFO - ---- Working on Burkholderia_contaminans_GCF_001029145 genome ---\n",
      "2020-02-06 17:12:25,989 - Phyluce - INFO - Reading Burkholderia_contaminans_GCF_001029145 genome\n",
      "2020-02-06 17:12:26,734 - Phyluce - INFO - Burkholderia_contaminans_GCF_001029145: 149 uces, 32 dupes, 117 non-dupes, 6 orient drop, 8 length drop, 103 written\n",
      "2020-02-06 17:12:26,735 - Phyluce - INFO - ------ Working on Burkholderia_diffusa_GCF_001718315 genome -----\n",
      "2020-02-06 17:12:26,742 - Phyluce - INFO - Reading Burkholderia_diffusa_GCF_001718315 genome\n",
      "2020-02-06 17:12:27,499 - Phyluce - INFO - Burkholderia_diffusa_GCF_001718315: 149 uces, 27 dupes, 122 non-dupes, 2 orient drop, 13 length drop, 107 written\n",
      "2020-02-06 17:12:27,499 - Phyluce - INFO - ------ Working on Burkholderia_dolosa_GCF_003813045 genome ------\n",
      "2020-02-06 17:12:27,499 - Phyluce - INFO - Reading Burkholderia_dolosa_GCF_003813045 genome\n",
      "2020-02-06 17:12:28,225 - Phyluce - INFO - Burkholderia_dolosa_GCF_003813045: 149 uces, 26 dupes, 123 non-dupes, 1 orient drop, 14 length drop, 108 written\n",
      "2020-02-06 17:12:28,226 - Phyluce - INFO - ----- Working on Burkholderia_gladioli_GCF_000959725 genome -----\n",
      "2020-02-06 17:12:28,226 - Phyluce - INFO - Reading Burkholderia_gladioli_GCF_000959725 genome\n",
      "2020-02-06 17:12:28,929 - Phyluce - INFO - Burkholderia_gladioli_GCF_000959725: 149 uces, 29 dupes, 120 non-dupes, 4 orient drop, 12 length drop, 104 written\n",
      "2020-02-06 17:12:28,929 - Phyluce - INFO - ------ Working on Burkholderia_glumae_GCF_000960995 genome ------\n",
      "2020-02-06 17:12:28,942 - Phyluce - INFO - Reading Burkholderia_glumae_GCF_000960995 genome\n",
      "2020-02-06 17:12:29,675 - Phyluce - INFO - Burkholderia_glumae_GCF_000960995: 149 uces, 25 dupes, 124 non-dupes, 3 orient drop, 18 length drop, 103 written\n",
      "2020-02-06 17:12:29,675 - Phyluce - INFO - ---- Working on Burkholderia_hayleyella_GCF_009455685 genome ----\n",
      "2020-02-06 17:12:29,676 - Phyluce - INFO - Reading Burkholderia_hayleyella_GCF_009455685 genome\n",
      "2020-02-06 17:12:30,278 - Phyluce - INFO - Burkholderia_hayleyella_GCF_009455685: 149 uces, 16 dupes, 133 non-dupes, 2 orient drop, 6 length drop, 125 written\n",
      "2020-02-06 17:12:30,278 - Phyluce - INFO - -- Working on Burkholderia_humptydooensis_GCF_001462435 genome --\n",
      "2020-02-06 17:12:30,279 - Phyluce - INFO - Reading Burkholderia_humptydooensis_GCF_001462435 genome\n",
      "2020-02-06 17:12:30,982 - Phyluce - INFO - Burkholderia_humptydooensis_GCF_001462435: 149 uces, 29 dupes, 120 non-dupes, 6 orient drop, 8 length drop, 106 written\n",
      "2020-02-06 17:12:30,983 - Phyluce - INFO - ---- Working on Burkholderia_insecticola_GCF_000402035 genome ---\n",
      "2020-02-06 17:12:30,983 - Phyluce - INFO - Reading Burkholderia_insecticola_GCF_000402035 genome\n",
      "2020-02-06 17:12:31,704 - Phyluce - INFO - Burkholderia_insecticola_GCF_000402035: 149 uces, 31 dupes, 118 non-dupes, 6 orient drop, 5 length drop, 107 written\n",
      "2020-02-06 17:12:31,704 - Phyluce - INFO - ------- Working on Burkholderia_lata_GCF_000012945 genome -------\n",
      "2020-02-06 17:12:31,705 - Phyluce - INFO - Reading Burkholderia_lata_GCF_000012945 genome\n",
      "2020-02-06 17:12:32,475 - Phyluce - INFO - Burkholderia_lata_GCF_000012945: 149 uces, 33 dupes, 116 non-dupes, 2 orient drop, 16 length drop, 98 written\n",
      "2020-02-06 17:12:32,475 - Phyluce - INFO - ------ Working on Burkholderia_latens_GCF_001718795 genome ------\n",
      "2020-02-06 17:12:32,476 - Phyluce - INFO - Reading Burkholderia_latens_GCF_001718795 genome\n",
      "2020-02-06 17:12:33,218 - Phyluce - INFO - Burkholderia_latens_GCF_001718795: 149 uces, 27 dupes, 122 non-dupes, 6 orient drop, 10 length drop, 106 written\n",
      "2020-02-06 17:12:33,218 - Phyluce - INFO - ----- Working on Burkholderia_metallica_GCF_001718555 genome ----\n",
      "2020-02-06 17:12:33,219 - Phyluce - INFO - Reading Burkholderia_metallica_GCF_001718555 genome\n",
      "2020-02-06 17:12:33,972 - Phyluce - INFO - Burkholderia_metallica_GCF_001718555: 149 uces, 26 dupes, 123 non-dupes, 2 orient drop, 16 length drop, 105 written\n",
      "2020-02-06 17:12:33,972 - Phyluce - INFO - ---- Working on Burkholderia_multivorans_GCF_000959525 genome ---\n",
      "2020-02-06 17:12:33,972 - Phyluce - INFO - Reading Burkholderia_multivorans_GCF_000959525 genome\n",
      "2020-02-06 17:12:34,689 - Phyluce - INFO - Burkholderia_multivorans_GCF_000959525: 149 uces, 24 dupes, 125 non-dupes, 2 orient drop, 16 length drop, 107 written\n",
      "2020-02-06 17:12:34,689 - Phyluce - INFO - -- Working on Burkholderia_novacaledonica_GCF_900258035 genome --\n",
      "2020-02-06 17:12:34,707 - Phyluce - INFO - Reading Burkholderia_novacaledonica_GCF_900258035 genome\n",
      "2020-02-06 17:12:35,310 - Phyluce - INFO - Burkholderia_novacaledonica_GCF_900258035: 149 uces, 31 dupes, 118 non-dupes, 1 orient drop, 2 length drop, 115 written\n",
      "2020-02-06 17:12:35,311 - Phyluce - INFO - --- Working on Burkholderia_oklahomensis_GCF_000959365 genome ---\n",
      "2020-02-06 17:12:35,311 - Phyluce - INFO - Reading Burkholderia_oklahomensis_GCF_000959365 genome\n",
      "2020-02-06 17:12:36,002 - Phyluce - INFO - Burkholderia_oklahomensis_GCF_000959365: 149 uces, 27 dupes, 122 non-dupes, 8 orient drop, 8 length drop, 106 written\n",
      "2020-02-06 17:12:36,002 - Phyluce - INFO - ------ Working on Burkholderia_paludis_GCF_000732615 genome -----\n",
      "2020-02-06 17:12:36,018 - Phyluce - INFO - Reading Burkholderia_paludis_GCF_000732615 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 17:12:36,679 - Phyluce - INFO - Burkholderia_paludis_GCF_000732615: 149 uces, 38 dupes, 111 non-dupes, 0 orient drop, 2 length drop, 109 written\n",
      "2020-02-06 17:12:36,679 - Phyluce - INFO - ----- Working on Burkholderia_plantarii_GCF_001411805 genome ----\n",
      "2020-02-06 17:12:36,691 - Phyluce - INFO - Reading Burkholderia_plantarii_GCF_001411805 genome\n",
      "2020-02-06 17:12:37,427 - Phyluce - INFO - Burkholderia_plantarii_GCF_001411805: 149 uces, 28 dupes, 121 non-dupes, 8 orient drop, 12 length drop, 101 written\n",
      "2020-02-06 17:12:37,427 - Phyluce - INFO - --- Working on Burkholderia_pseudomallei_GCF_000011545 genome ---\n",
      "2020-02-06 17:12:37,427 - Phyluce - INFO - Reading Burkholderia_pseudomallei_GCF_000011545 genome\n",
      "2020-02-06 17:12:38,125 - Phyluce - INFO - Burkholderia_pseudomallei_GCF_000011545: 149 uces, 27 dupes, 122 non-dupes, 2 orient drop, 15 length drop, 105 written\n",
      "2020-02-06 17:12:38,125 - Phyluce - INFO - - Working on Burkholderia_pseudomultivorans_GCF_001718415 genome \n",
      "2020-02-06 17:12:38,126 - Phyluce - INFO - Reading Burkholderia_pseudomultivorans_GCF_001718415 genome\n",
      "2020-02-06 17:12:38,893 - Phyluce - INFO - Burkholderia_pseudomultivorans_GCF_001718415: 149 uces, 24 dupes, 125 non-dupes, 6 orient drop, 17 length drop, 102 written\n",
      "2020-02-06 17:12:38,893 - Phyluce - INFO - ----- Working on Burkholderia_puraquae_GCF_002099195 genome -----\n",
      "2020-02-06 17:12:38,894 - Phyluce - INFO - Reading Burkholderia_puraquae_GCF_002099195 genome\n",
      "2020-02-06 17:12:39,528 - Phyluce - INFO - Burkholderia_puraquae_GCF_002099195: 149 uces, 37 dupes, 112 non-dupes, 0 orient drop, 2 length drop, 110 written\n",
      "2020-02-06 17:12:39,529 - Phyluce - INFO - ---- Working on Burkholderia_pyrrocinia_GCF_001028665 genome ----\n",
      "2020-02-06 17:12:39,529 - Phyluce - INFO - Reading Burkholderia_pyrrocinia_GCF_001028665 genome\n",
      "2020-02-06 17:12:40,291 - Phyluce - INFO - Burkholderia_pyrrocinia_GCF_001028665: 149 uces, 33 dupes, 116 non-dupes, 2 orient drop, 16 length drop, 98 written\n",
      "2020-02-06 17:12:40,291 - Phyluce - INFO - ------ Working on Burkholderia_reimsis_GCF_003294055 genome -----\n",
      "2020-02-06 17:12:40,292 - Phyluce - INFO - Reading Burkholderia_reimsis_GCF_003294055 genome\n",
      "2020-02-06 17:12:40,952 - Phyluce - INFO - Burkholderia_reimsis_GCF_003294055: 149 uces, 41 dupes, 108 non-dupes, 0 orient drop, 2 length drop, 106 written\n",
      "2020-02-06 17:12:40,952 - Phyluce - INFO - ----- Working on Burkholderia_seminalis_GCF_001718535 genome ----\n",
      "2020-02-06 17:12:40,953 - Phyluce - INFO - Reading Burkholderia_seminalis_GCF_001718535 genome\n",
      "2020-02-06 17:12:41,728 - Phyluce - INFO - Burkholderia_seminalis_GCF_001718535: 149 uces, 32 dupes, 117 non-dupes, 2 orient drop, 14 length drop, 101 written\n",
      "2020-02-06 17:12:41,729 - Phyluce - INFO - ---- Working on Burkholderia_singularis_GCF_900176645 genome ----\n",
      "2020-02-06 17:12:41,735 - Phyluce - INFO - Reading Burkholderia_singularis_GCF_900176645 genome\n",
      "2020-02-06 17:12:42,282 - Phyluce - INFO - Burkholderia_singularis_GCF_900176645: 149 uces, 23 dupes, 126 non-dupes, 0 orient drop, 3 length drop, 123 written\n",
      "2020-02-06 17:12:42,282 - Phyluce - INFO - -------- Working on Burkholderia_sp_GCF_005280735 genome --------\n",
      "2020-02-06 17:12:42,282 - Phyluce - INFO - Reading Burkholderia_sp_GCF_005280735 genome\n",
      "2020-02-06 17:12:42,988 - Phyluce - INFO - Burkholderia_sp_GCF_005280735: 148 uces, 26 dupes, 122 non-dupes, 9 orient drop, 10 length drop, 103 written\n",
      "2020-02-06 17:12:42,988 - Phyluce - INFO - ----- Working on Burkholderia_stabilis_GCF_001742165 genome -----\n",
      "2020-02-06 17:12:42,988 - Phyluce - INFO - Reading Burkholderia_stabilis_GCF_001742165 genome\n",
      "2020-02-06 17:12:43,756 - Phyluce - INFO - Burkholderia_stabilis_GCF_001742165: 149 uces, 31 dupes, 118 non-dupes, 7 orient drop, 11 length drop, 100 written\n",
      "2020-02-06 17:12:43,757 - Phyluce - INFO - ----- Working on Burkholderia_stagnalis_GCF_001718955 genome ----\n",
      "2020-02-06 17:12:43,773 - Phyluce - INFO - Reading Burkholderia_stagnalis_GCF_001718955 genome\n",
      "2020-02-06 17:12:44,558 - Phyluce - INFO - Burkholderia_stagnalis_GCF_001718955: 149 uces, 30 dupes, 119 non-dupes, 2 orient drop, 19 length drop, 98 written\n",
      "2020-02-06 17:12:44,558 - Phyluce - INFO - ---- Working on Burkholderia_territorii_GCF_001718335 genome ----\n",
      "2020-02-06 17:12:44,559 - Phyluce - INFO - Reading Burkholderia_territorii_GCF_001718335 genome\n",
      "2020-02-06 17:12:45,319 - Phyluce - INFO - Burkholderia_territorii_GCF_001718335: 149 uces, 31 dupes, 118 non-dupes, 2 orient drop, 13 length drop, 103 written\n",
      "2020-02-06 17:12:45,319 - Phyluce - INFO - --- Working on Burkholderia_thailandensis_GCF_003568605 genome --\n",
      "2020-02-06 17:12:45,319 - Phyluce - INFO - Reading Burkholderia_thailandensis_GCF_003568605 genome\n",
      "2020-02-06 17:12:46,020 - Phyluce - INFO - Burkholderia_thailandensis_GCF_003568605: 149 uces, 28 dupes, 121 non-dupes, 13 orient drop, 4 length drop, 104 written\n",
      "2020-02-06 17:12:46,020 - Phyluce - INFO - ----- Working on Burkholderia_ubonensis_GCF_000959245 genome ----\n",
      "2020-02-06 17:12:46,021 - Phyluce - INFO - Reading Burkholderia_ubonensis_GCF_000959245 genome\n",
      "2020-02-06 17:12:46,788 - Phyluce - INFO - Burkholderia_ubonensis_GCF_000959245: 149 uces, 25 dupes, 124 non-dupes, 1 orient drop, 16 length drop, 107 written\n",
      "2020-02-06 17:12:46,789 - Phyluce - INFO - --- Working on Burkholderia_vietnamiensis_GCF_000959445 genome --\n",
      "2020-02-06 17:12:46,796 - Phyluce - INFO - Reading Burkholderia_vietnamiensis_GCF_000959445 genome\n",
      "2020-02-06 17:12:47,546 - Phyluce - INFO - Burkholderia_vietnamiensis_GCF_000959445: 149 uces, 26 dupes, 123 non-dupes, 4 orient drop, 15 length drop, 104 written\n",
      "2020-02-06 17:12:47,547 - Phyluce - INFO - ------ Working on Burkholderia_mallei_GCF_000011705 genome ------\n",
      "2020-02-06 17:12:47,547 - Phyluce - INFO - Reading Burkholderia_mallei_GCF_000011705 genome\n",
      "2020-02-06 17:12:48,202 - Phyluce - INFO - Burkholderia_mallei_GCF_000011705: 149 uces, 24 dupes, 125 non-dupes, 8 orient drop, 6 length drop, 111 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/burkholderia/extract_probes_from_group/probe_fasta --output results/phyluce/burkholderia/extract_probes_from_group/multifastas.sqlite --base-taxon Burkholderia_mallei_GCF_000011705;\n",
      "burkholderia_agricolaris_gcf_009455635.\n",
      "burkholderia_ambifaria_gcf_000959545.\n",
      "burkholderia_anthina_gcf_001547525.\n",
      "burkholderia_bonniea_gcf_009455625.\n",
      "burkholderia_catarinensis_gcf_001883705.\n",
      "burkholderia_cenocepacia_gcf_001999785.\n",
      "burkholderia_cepacia_gcf_001411495.\n",
      "burkholderia_contaminans_gcf_001029145.\n",
      "burkholderia_diffusa_gcf_001718315.\n",
      "burkholderia_dolosa_gcf_003813045.\n",
      "burkholderia_gladioli_gcf_000959725.\n",
      "burkholderia_glumae_gcf_000960995.\n",
      "burkholderia_hayleyella_gcf_009455685.\n",
      "burkholderia_humptydooensis_gcf_001462435.\n",
      "burkholderia_insecticola_gcf_000402035.\n",
      "burkholderia_lata_gcf_000012945.\n",
      "burkholderia_latens_gcf_001718795.\n",
      "burkholderia_metallica_gcf_001718555.\n",
      "burkholderia_multivorans_gcf_000959525.\n",
      "burkholderia_novacaledonica_gcf_900258035.\n",
      "burkholderia_oklahomensis_gcf_000959365.\n",
      "burkholderia_paludis_gcf_000732615.\n",
      "burkholderia_plantarii_gcf_001411805.\n",
      "burkholderia_pseudomallei_gcf_000011545.\n",
      "burkholderia_pseudomultivorans_gcf_001718415.\n",
      "burkholderia_puraquae_gcf_002099195.\n",
      "burkholderia_pyrrocinia_gcf_001028665.\n",
      "burkholderia_reimsis_gcf_003294055.\n",
      "burkholderia_seminalis_gcf_001718535.\n",
      "burkholderia_singularis_gcf_900176645.\n",
      "burkholderia_sp_gcf_005280735.\n",
      "burkholderia_stabilis_gcf_001742165.\n",
      "burkholderia_stagnalis_gcf_001718955.\n",
      "burkholderia_territorii_gcf_001718335.\n",
      "burkholderia_thailandensis_gcf_003568605.\n",
      "burkholderia_ubonensis_gcf_000959245.\n",
      "burkholderia_vietnamiensis_gcf_000959445.\n",
      "burkholderia_mallei_gcf_000011705.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/burkholderia/extract_probes_from_group/multifastas.sqlite --base-taxon Burkholderia_mallei_GCF_000011705;\n",
      "Loci shared by 0 taxa:\t139.0\n",
      "Loci shared by 1 taxa:\t139.0\n",
      "Loci shared by 2 taxa:\t138.0\n",
      "Loci shared by 3 taxa:\t135.0\n",
      "Loci shared by 4 taxa:\t129.0\n",
      "Loci shared by 5 taxa:\t127.0\n",
      "Loci shared by 6 taxa:\t126.0\n",
      "Loci shared by 7 taxa:\t114.0\n",
      "Loci shared by 8 taxa:\t112.0\n",
      "Loci shared by 9 taxa:\t112.0\n",
      "Loci shared by 10 taxa:\t112.0\n",
      "Loci shared by 11 taxa:\t112.0\n",
      "Loci shared by 12 taxa:\t111.0\n",
      "Loci shared by 13 taxa:\t110.0\n",
      "Loci shared by 14 taxa:\t108.0\n",
      "Loci shared by 15 taxa:\t106.0\n",
      "Loci shared by 16 taxa:\t105.0\n",
      "Loci shared by 17 taxa:\t105.0\n",
      "Loci shared by 18 taxa:\t105.0\n",
      "Loci shared by 19 taxa:\t104.0\n",
      "Loci shared by 20 taxa:\t103.0\n",
      "Loci shared by 21 taxa:\t103.0\n",
      "Loci shared by 22 taxa:\t103.0\n",
      "Loci shared by 23 taxa:\t102.0\n",
      "Loci shared by 24 taxa:\t102.0\n",
      "Loci shared by 25 taxa:\t102.0\n",
      "Loci shared by 26 taxa:\t101.0\n",
      "Loci shared by 27 taxa:\t101.0\n",
      "Loci shared by 28 taxa:\t99.0\n",
      "Loci shared by 29 taxa:\t99.0\n",
      "Loci shared by 30 taxa:\t99.0\n",
      "Loci shared by 31 taxa:\t98.0\n",
      "Loci shared by 32 taxa:\t96.0\n",
      "Loci shared by 33 taxa:\t95.0\n",
      "Loci shared by 34 taxa:\t94.0\n",
      "Loci shared by 35 taxa:\t91.0\n",
      "Loci shared by 36 taxa:\t85.0\n",
      "Loci shared by 37 taxa:\t80.0\n",
      "Loci shared by 38 taxa:\t66.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/burkholderia/extract_probes_from_group/multifastas.sqlite  --base-taxon Burkholderia_mallei_GCF_000011705 --output results/phyluce/burkholderia/extract_probes_from_group/Burkholderia_mallei_GCF_000011705+36-back-to-32.conf --specific-counts 32;\n",
      "Counter({'burkholderia_dolosa_gcf_003813045': 96, 'burkholderia_cenocepacia_gcf_001999785': 96, 'burkholderia_puraquae_gcf_002099195': 96, 'burkholderia_multivorans_gcf_000959525': 96, 'burkholderia_diffusa_gcf_001718315': 96, 'burkholderia_singularis_gcf_900176645': 96, 'burkholderia_contaminans_gcf_001029145': 96, 'burkholderia_hayleyella_gcf_009455685': 96, 'burkholderia_reimsis_gcf_003294055': 96, 'burkholderia_ambifaria_gcf_000959545': 95, 'burkholderia_mallei_gcf_000011705': 95, 'burkholderia_vietnamiensis_gcf_000959445': 95, 'burkholderia_catarinensis_gcf_001883705': 95, 'burkholderia_metallica_gcf_001718555': 95, 'burkholderia_seminalis_gcf_001718535': 95, 'burkholderia_stabilis_gcf_001742165': 95, 'burkholderia_bonniea_gcf_009455625': 95, 'burkholderia_cepacia_gcf_001411495': 94, 'burkholderia_latens_gcf_001718795': 94, 'burkholderia_ubonensis_gcf_000959245': 94, 'burkholderia_territorii_gcf_001718335': 94, 'burkholderia_pseudomultivorans_gcf_001718415': 94, 'burkholderia_agricolaris_gcf_009455635': 94, 'burkholderia_novacaledonica_gcf_900258035': 94, 'burkholderia_lata_gcf_000012945': 94, 'burkholderia_pseudomallei_gcf_000011545': 94, 'burkholderia_stagnalis_gcf_001718955': 94, 'burkholderia_paludis_gcf_000732615': 94, 'burkholderia_humptydooensis_gcf_001462435': 93, 'burkholderia_oklahomensis_gcf_000959365': 93, 'burkholderia_plantarii_gcf_001411805': 93, 'burkholderia_gladioli_gcf_000959725': 93, 'burkholderia_thailandensis_gcf_003568605': 93, 'burkholderia_anthina_gcf_001547525': 93, 'burkholderia_pyrrocinia_gcf_001028665': 92, 'burkholderia_insecticola_gcf_000402035': 92, 'burkholderia_glumae_gcf_000960995': 92, 'burkholderia_sp_gcf_005280735': 91})\n",
      "Total loci = 96\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(32)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/burkholderia/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/burkholderia/extract_probes_from_group/Burkholderia_mallei_GCF_000011705+36-back-to-32.conf --probe-prefix uce_burkholderia_ --designer rnplattii --design burkholderia_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGN\n",
      "\n",
      "\n",
      "Conserved locus count = 96\n",
      "Probe Count = 7005\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list.fasta --query results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  17:13:39\n",
      "Ended:  Thu Feb 06, 2020  17:14:07\n",
      "Time for execution:  0.463900315762 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list.fasta --lastz results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_burkholderia_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 7004 fasta sequences.  Filtered 0 duplicates. Kept 7005.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/burkholderia/final_probe_design/burkholderia_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 17:25:14 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 7005\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 560400\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 88M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3953038\n",
      "Max number of word counting entries: 88982893\n",
      "\n",
      "# comparing sequences from          0  to       1167\n",
      ".---------- new table with      340 representatives\n",
      "# comparing sequences from       1167  to       2140\n",
      "----------    499 remaining sequences to the next cycle\n",
      "---------- new table with      144 representatives\n",
      "# comparing sequences from       1641  to       2535\n",
      "----------    505 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       2030  to       2859\n",
      "----------    453 remaining sequences to the next cycle\n",
      "---------- new table with      114 representatives\n",
      "# comparing sequences from       2406  to       3172\n",
      "----------    288 remaining sequences to the next cycle\n",
      "---------- new table with      134 representatives\n",
      "# comparing sequences from       2884  to       3570\n",
      "----------    273 remaining sequences to the next cycle\n",
      "---------- new table with      101 representatives\n",
      "# comparing sequences from       3297  to       3915\n",
      "----------    182 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       3733  to       4278\n",
      "----------    172 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       4106  to       4589\n",
      "----------    113 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       4476  to       4897\n",
      "----------     61 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       4836  to       5197\n",
      "....................---------- new table with       95 representatives\n",
      "# comparing sequences from       5197  to       5498\n",
      "...................---------- new table with       62 representatives\n",
      "# comparing sequences from       5498  to       5749\n",
      "....................---------- new table with       82 representatives\n",
      "# comparing sequences from       5749  to       5958\n",
      "...................---------- new table with       49 representatives\n",
      "# comparing sequences from       5958  to       6132\n",
      ".....................---------- new table with       68 representatives\n",
      "# comparing sequences from       6132  to       7005\n",
      ".....................---------- new table with      198 representatives\n",
      "\n",
      "     7005  finished       1887  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.33\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Campylobacter"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "group = 'campylobacter'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "group_taxa = [ \"Campylobacter_armoricus_GCF_009036385\",\n",
    "               \"Campylobacter_avium_GCF_002238335\",\n",
    "               \"Campylobacter_blaseri_GCF_003015205\",\n",
    "               \"Campylobacter_coli_GCF_002024185\",\n",
    "               \"Campylobacter_concisus_GCF_001298465\",\n",
    "               \"Campylobacter_corcagiensis_GCF_000597805\",\n",
    "               \"Campylobacter_cuniculorum_GCF_002104335\",\n",
    "               \"Campylobacter_curvus_GCF_000017465\",\n",
    "               \"Campylobacter_fetus_GCF_008271385\",\n",
    "               \"Campylobacter_geochelonis_GCF_900065885\",\n",
    "               \"Campylobacter_gracilis_GCF_001190745\",\n",
    "               \"Campylobacter_helveticus_GCF_002080395\",\n",
    "               \"Campylobacter_hepaticus_GCF_001687475\",\n",
    "               \"Campylobacter_hominis_GCF_000017585\",\n",
    "               \"Campylobacter_hyointestinalis_GCF_001643955\",\n",
    "               \"Campylobacter_iguaniorum_GCF_000736415\",\n",
    "               \"Campylobacter_insulaenigrae_GCF_000816185\",\n",
    "               \"Campylobacter_lanienae_GCF_002139935\",\n",
    "               \"Campylobacter_lari_GCF_000019205\",\n",
    "               \"Campylobacter_ornithocola_GCF_001705345\",\n",
    "               \"Campylobacter_peloridis_GCF_000816785\",\n",
    "               \"Campylobacter_pinnipediorum_GCF_002021925\",\n",
    "               \"Campylobacter_rectus_GCF_000174175\",\n",
    "               \"Campylobacter_showae_GCF_000313615\",\n",
    "               \"Campylobacter_sp_GCF_002139895\",\n",
    "               \"Campylobacter_sptorum_GCF_002220735\",\n",
    "               \"Campylobacter_subantarcticus_GCF_000816305\",\n",
    "               \"Campylobacter_troglodytis_GCF_006864425\",\n",
    "               \"Campylobacter_upsaliensis_GCF_000620965\",\n",
    "               \"Campylobacter_ureolyticus_GCF_001190755\",\n",
    "               \"Campylobacter_volucris_GCF_000816345\" ]\n",
    "                    \n",
    "reference_taxon = \"Campylobacter_jejuni_GCF_000009085\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009036385.1_ASM903638v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 491694 bytes  327824.00 bytes/sec\n",
      "total size is 491468  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002238335.1_ASM223833v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 497481 bytes  331682.00 bytes/sec\n",
      "total size is 497247  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003015205.1_ASM301520v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 532065 bytes  354738.00 bytes/sec\n",
      "total size is 531823  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002024185.1_ASM202418v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 574796 bytes  383225.33 bytes/sec\n",
      "total size is 574546  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001298465.1_ASM129846v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 535450 bytes  356994.67 bytes/sec\n",
      "total size is 535208  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000597805.1_Ccorc1.0_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 480000 bytes  320028.00 bytes/sec\n",
      "total size is 479777  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002104335.1_ASM210433v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 549016 bytes  366038.67 bytes/sec\n",
      "total size is 548774  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017465.2_ASM1746v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 577239 bytes  384854.00 bytes/sec\n",
      "total size is 576991  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_008271385.1_ASM827138v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 552883 bytes  368616.67 bytes/sec\n",
      "total size is 552641  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900065885.1_16473_4_70_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 586361 bytes  390935.33 bytes/sec\n",
      "total size is 586112  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_001190745.1_ASM119074v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 664480 bytes  443014.67 bytes/sec\n",
      "total size is 664206  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002080395.1_ASM208039v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 534429 bytes  356314.00 bytes/sec\n",
      "total size is 534187  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001687475.2_ASM168747v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 427027 bytes  854138.00 bytes/sec\n",
      "total size is 426809  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017585.1_ASM1758v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 488617 bytes  325772.67 bytes/sec\n",
      "total size is 488393  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001643955.1_ASM164395v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 507394 bytes  338290.67 bytes/sec\n",
      "total size is 507160  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000736415.1_ASM73641v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 507382 bytes  338282.67 bytes/sec\n",
      "total size is 507149  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000816185.1_ASM81618v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 414516 bytes  276372.00 bytes/sec\n",
      "total size is 414307  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000009085.1_ASM908v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 465896 bytes  310625.33 bytes/sec\n",
      "total size is 465673  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002139935.1_ASM213993v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 458670 bytes  305808.00 bytes/sec\n",
      "total size is 458452  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000019205.1_ASM1920v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 443929 bytes  295980.67 bytes/sec\n",
      "total size is 443713  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001705345.1_ASM170534v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 477964 bytes  318670.67 bytes/sec\n",
      "total size is 477738  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000816785.1_ASM81678v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 480434 bytes  320317.33 bytes/sec\n",
      "total size is 480209  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002021925.1_ASM202192v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 507997 bytes  338692.67 bytes/sec\n",
      "total size is 507763  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000174175.1_ASM17417v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 735065 bytes  490071.33 bytes/sec\n",
      "total size is 734776  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000313615.1_Velvet_assembly_of_Campylobacter_showae_CSUNSWCD_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 623131 bytes  415448.67 bytes/sec\n",
      "total size is 622828  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002139895.1_ASM213989v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 501524 bytes  334377.33 bytes/sec\n",
      "total size is 501290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002220735.1_ASM222073v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 498255 bytes  332198.00 bytes/sec\n",
      "total size is 498021  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000816305.1_ASM81630v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 514811 bytes  343235.33 bytes/sec\n",
      "total size is 514578  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_006864425.1_ASM686442v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 842751 bytes  561862.00 bytes/sec\n",
      "total size is 842437  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000620965.1_ASM62096v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 466903 bytes  311296.67 bytes/sec\n",
      "total size is 466678  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001190755.1_ASM119075v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 462295 bytes  308224.67 bytes/sec\n",
      "total size is 462069  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000816345.1_ASM81634v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 426464 bytes  284337.33 bytes/sec\n",
      "total size is 426247  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481767 (\"sim_Campylobacter_armoricus_GCF_009036385\") has been submitted\n",
      "Your job 5481768 (\"sim_Campylobacter_avium_GCF_002238335\") has been submitted\n",
      "Your job 5481769 (\"sim_Campylobacter_blaseri_GCF_003015205\") has been submitted\n",
      "Your job 5481770 (\"sim_Campylobacter_coli_GCF_002024185\") has been submitted\n",
      "Your job 5481771 (\"sim_Campylobacter_concisus_GCF_001298465\") has been submitted\n",
      "Your job 5481772 (\"sim_Campylobacter_corcagiensis_GCF_000597805\") has been submitted\n",
      "Your job 5481773 (\"sim_Campylobacter_cuniculorum_GCF_002104335\") has been submitted\n",
      "Your job 5481774 (\"sim_Campylobacter_curvus_GCF_000017465\") has been submitted\n",
      "Your job 5481775 (\"sim_Campylobacter_fetus_GCF_008271385\") has been submitted\n",
      "Your job 5481776 (\"sim_Campylobacter_geochelonis_GCF_900065885\") has been submitted\n",
      "Your job 5481777 (\"sim_Campylobacter_gracilis_GCF_001190745\") has been submitted\n",
      "Your job 5481778 (\"sim_Campylobacter_helveticus_GCF_002080395\") has been submitted\n",
      "Your job 5481779 (\"sim_Campylobacter_hepaticus_GCF_001687475\") has been submitted\n",
      "Your job 5481780 (\"sim_Campylobacter_hominis_GCF_000017585\") has been submitted\n",
      "Your job 5481781 (\"sim_Campylobacter_hyointestinalis_GCF_001643955\") has been submitted\n",
      "Your job 5481782 (\"sim_Campylobacter_iguaniorum_GCF_000736415\") has been submitted\n",
      "Your job 5481783 (\"sim_Campylobacter_insulaenigrae_GCF_000816185\") has been submitted\n",
      "Your job 5481784 (\"sim_Campylobacter_lanienae_GCF_002139935\") has been submitted\n",
      "Your job 5481785 (\"sim_Campylobacter_lari_GCF_000019205\") has been submitted\n",
      "Your job 5481786 (\"sim_Campylobacter_ornithocola_GCF_001705345\") has been submitted\n",
      "Your job 5481787 (\"sim_Campylobacter_peloridis_GCF_000816785\") has been submitted\n",
      "Your job 5481788 (\"sim_Campylobacter_pinnipediorum_GCF_002021925\") has been submitted\n",
      "Your job 5481789 (\"sim_Campylobacter_rectus_GCF_000174175\") has been submitted\n",
      "Your job 5481790 (\"sim_Campylobacter_showae_GCF_000313615\") has been submitted\n",
      "Your job 5481791 (\"sim_Campylobacter_sp_GCF_002139895\") has been submitted\n",
      "Your job 5481792 (\"sim_Campylobacter_sptorum_GCF_002220735\") has been submitted\n",
      "Your job 5481793 (\"sim_Campylobacter_subantarcticus_GCF_000816305\") has been submitted\n",
      "Your job 5481794 (\"sim_Campylobacter_troglodytis_GCF_006864425\") has been submitted\n",
      "Your job 5481795 (\"sim_Campylobacter_upsaliensis_GCF_000620965\") has been submitted\n",
      "Your job 5481796 (\"sim_Campylobacter_ureolyticus_GCF_001190755\") has been submitted\n",
      "Your job 5481797 (\"sim_Campylobacter_volucris_GCF_000816345\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/campylobacter/cleaned_genomes/Campylobacter_jejuni_GCF_000009085_formatted.fas path=results/phyluce/campylobacter/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481798 (\"map_Campylobacter_armoricus_GCF_009036385\") has been submitted\n",
      "Your job 5481799 (\"map_Campylobacter_avium_GCF_002238335\") has been submitted\n",
      "Your job 5481800 (\"map_Campylobacter_blaseri_GCF_003015205\") has been submitted\n",
      "Your job 5481801 (\"map_Campylobacter_coli_GCF_002024185\") has been submitted\n",
      "Your job 5481802 (\"map_Campylobacter_concisus_GCF_001298465\") has been submitted\n",
      "Your job 5481803 (\"map_Campylobacter_corcagiensis_GCF_000597805\") has been submitted\n",
      "Your job 5481804 (\"map_Campylobacter_cuniculorum_GCF_002104335\") has been submitted\n",
      "Your job 5481805 (\"map_Campylobacter_curvus_GCF_000017465\") has been submitted\n",
      "Your job 5481806 (\"map_Campylobacter_fetus_GCF_008271385\") has been submitted\n",
      "Your job 5481807 (\"map_Campylobacter_geochelonis_GCF_900065885\") has been submitted\n",
      "Your job 5481808 (\"map_Campylobacter_gracilis_GCF_001190745\") has been submitted\n",
      "Your job 5481809 (\"map_Campylobacter_helveticus_GCF_002080395\") has been submitted\n",
      "Your job 5481810 (\"map_Campylobacter_hepaticus_GCF_001687475\") has been submitted\n",
      "Your job 5481811 (\"map_Campylobacter_hominis_GCF_000017585\") has been submitted\n",
      "Your job 5481812 (\"map_Campylobacter_hyointestinalis_GCF_001643955\") has been submitted\n",
      "Your job 5481813 (\"map_Campylobacter_iguaniorum_GCF_000736415\") has been submitted\n",
      "Your job 5481814 (\"map_Campylobacter_insulaenigrae_GCF_000816185\") has been submitted\n",
      "Your job 5481815 (\"map_Campylobacter_lanienae_GCF_002139935\") has been submitted\n",
      "Your job 5481816 (\"map_Campylobacter_lari_GCF_000019205\") has been submitted\n",
      "Your job 5481817 (\"map_Campylobacter_ornithocola_GCF_001705345\") has been submitted\n",
      "Your job 5481818 (\"map_Campylobacter_peloridis_GCF_000816785\") has been submitted\n",
      "Your job 5481819 (\"map_Campylobacter_pinnipediorum_GCF_002021925\") has been submitted\n",
      "Your job 5481820 (\"map_Campylobacter_rectus_GCF_000174175\") has been submitted\n",
      "Your job 5481821 (\"map_Campylobacter_showae_GCF_000313615\") has been submitted\n",
      "Your job 5481822 (\"map_Campylobacter_sp_GCF_002139895\") has been submitted\n",
      "Your job 5481823 (\"map_Campylobacter_sptorum_GCF_002220735\") has been submitted\n",
      "Your job 5481824 (\"map_Campylobacter_subantarcticus_GCF_000816305\") has been submitted\n",
      "Your job 5481825 (\"map_Campylobacter_troglodytis_GCF_006864425\") has been submitted\n",
      "Your job 5481826 (\"map_Campylobacter_upsaliensis_GCF_000620965\") has been submitted\n",
      "Your job 5481827 (\"map_Campylobacter_ureolyticus_GCF_001190755\") has been submitted\n",
      "Your job 5481828 (\"map_Campylobacter_volucris_GCF_000816345\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481829 (\"merge_Campylobacter_armoricus_GCF_009036385\") has been submitted\n",
      "Your job 5481830 (\"merge_Campylobacter_avium_GCF_002238335\") has been submitted\n",
      "Your job 5481831 (\"merge_Campylobacter_blaseri_GCF_003015205\") has been submitted\n",
      "Your job 5481832 (\"merge_Campylobacter_coli_GCF_002024185\") has been submitted\n",
      "Your job 5481833 (\"merge_Campylobacter_concisus_GCF_001298465\") has been submitted\n",
      "Your job 5481834 (\"merge_Campylobacter_corcagiensis_GCF_000597805\") has been submitted\n",
      "Your job 5481835 (\"merge_Campylobacter_cuniculorum_GCF_002104335\") has been submitted\n",
      "Your job 5481836 (\"merge_Campylobacter_curvus_GCF_000017465\") has been submitted\n",
      "Your job 5481837 (\"merge_Campylobacter_fetus_GCF_008271385\") has been submitted\n",
      "Your job 5481838 (\"merge_Campylobacter_geochelonis_GCF_900065885\") has been submitted\n",
      "Your job 5481839 (\"merge_Campylobacter_gracilis_GCF_001190745\") has been submitted\n",
      "Your job 5481840 (\"merge_Campylobacter_helveticus_GCF_002080395\") has been submitted\n",
      "Your job 5481841 (\"merge_Campylobacter_hepaticus_GCF_001687475\") has been submitted\n",
      "Your job 5481842 (\"merge_Campylobacter_hominis_GCF_000017585\") has been submitted\n",
      "Your job 5481843 (\"merge_Campylobacter_hyointestinalis_GCF_001643955\") has been submitted\n",
      "Your job 5481844 (\"merge_Campylobacter_iguaniorum_GCF_000736415\") has been submitted\n",
      "Your job 5481845 (\"merge_Campylobacter_insulaenigrae_GCF_000816185\") has been submitted\n",
      "Your job 5481846 (\"merge_Campylobacter_lanienae_GCF_002139935\") has been submitted\n",
      "Your job 5481847 (\"merge_Campylobacter_lari_GCF_000019205\") has been submitted\n",
      "Your job 5481848 (\"merge_Campylobacter_ornithocola_GCF_001705345\") has been submitted\n",
      "Your job 5481849 (\"merge_Campylobacter_peloridis_GCF_000816785\") has been submitted\n",
      "Your job 5481850 (\"merge_Campylobacter_pinnipediorum_GCF_002021925\") has been submitted\n",
      "Your job 5481851 (\"merge_Campylobacter_rectus_GCF_000174175\") has been submitted\n",
      "Your job 5481852 (\"merge_Campylobacter_showae_GCF_000313615\") has been submitted\n",
      "Your job 5481853 (\"merge_Campylobacter_sp_GCF_002139895\") has been submitted\n",
      "Your job 5481854 (\"merge_Campylobacter_sptorum_GCF_002220735\") has been submitted\n",
      "Your job 5481855 (\"merge_Campylobacter_subantarcticus_GCF_000816305\") has been submitted\n",
      "Your job 5481856 (\"merge_Campylobacter_troglodytis_GCF_006864425\") has been submitted\n",
      "Your job 5481857 (\"merge_Campylobacter_upsaliensis_GCF_000620965\") has been submitted\n",
      "Your job 5481858 (\"merge_Campylobacter_ureolyticus_GCF_001190755\") has been submitted\n",
      "Your job 5481859 (\"merge_Campylobacter_volucris_GCF_000816345\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "remove loci that were masked in the original genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 437 sequences from Campylobacter_armoricus_GCF_009036385_merged.bed.  Filtered 326 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 66 sequences from Campylobacter_avium_GCF_002238335_merged.bed.  Filtered 52 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 14.\n",
      "Screened 44 sequences from Campylobacter_blaseri_GCF_003015205_merged.bed.  Filtered 32 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 1953 sequences from Campylobacter_coli_GCF_002024185_merged.bed.  Filtered 1057 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 896.\n",
      "Screened 25 sequences from Campylobacter_concisus_GCF_001298465_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 30 sequences from Campylobacter_corcagiensis_GCF_000597805_merged.bed.  Filtered 21 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 439 sequences from Campylobacter_cuniculorum_GCF_002104335_merged.bed.  Filtered 341 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 98.\n",
      "Screened 23 sequences from Campylobacter_curvus_GCF_000017465_merged.bed.  Filtered 15 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 31 sequences from Campylobacter_fetus_GCF_008271385_merged.bed.  Filtered 19 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 41 sequences from Campylobacter_geochelonis_GCF_900065885_merged.bed.  Filtered 29 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 13 sequences from Campylobacter_gracilis_GCF_001190745_merged.bed.  Filtered 6 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 360 sequences from Campylobacter_helveticus_GCF_002080395_merged.bed.  Filtered 283 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 77.\n",
      "Screened 2336 sequences from Campylobacter_hepaticus_GCF_001687475_merged.bed.  Filtered 1306 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1030.\n",
      "Screened 22 sequences from Campylobacter_hominis_GCF_000017585_merged.bed.  Filtered 13 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 40 sequences from Campylobacter_hyointestinalis_GCF_001643955_merged.bed.  Filtered 33 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 7.\n",
      "Screened 40 sequences from Campylobacter_iguaniorum_GCF_000736415_merged.bed.  Filtered 31 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 343 sequences from Campylobacter_insulaenigrae_GCF_000816185_merged.bed.  Filtered 263 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 40 sequences from Campylobacter_lanienae_GCF_002139935_merged.bed.  Filtered 25 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 15.\n",
      "Screened 446 sequences from Campylobacter_lari_GCF_000019205_merged.bed.  Filtered 324 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 122.\n",
      "Screened 427 sequences from Campylobacter_ornithocola_GCF_001705345_merged.bed.  Filtered 316 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 458 sequences from Campylobacter_peloridis_GCF_000816785_merged.bed.  Filtered 337 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 121.\n",
      "Screened 40 sequences from Campylobacter_pinnipediorum_GCF_002021925_merged.bed.  Filtered 30 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 10.\n",
      "Screened 20 sequences from Campylobacter_rectus_GCF_000174175_merged.bed.  Filtered 9 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 11.\n",
      "Screened 22 sequences from Campylobacter_showae_GCF_000313615_merged.bed.  Filtered 14 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 44 sequences from Campylobacter_sp_GCF_002139895_merged.bed.  Filtered 35 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 45 sequences from Campylobacter_sptorum_GCF_002220735_merged.bed.  Filtered 33 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 12.\n",
      "Screened 394 sequences from Campylobacter_subantarcticus_GCF_000816305_merged.bed.  Filtered 291 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 103.\n",
      "Screened 30 sequences from Campylobacter_troglodytis_GCF_006864425_merged.bed.  Filtered 22 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 8.\n",
      "Screened 335 sequences from Campylobacter_upsaliensis_GCF_000620965_merged.bed.  Filtered 269 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 66.\n",
      "Screened 36 sequences from Campylobacter_ureolyticus_GCF_001190755_merged.bed.  Filtered 27 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 9.\n",
      "Screened 445 sequences from Campylobacter_volucris_GCF_000816345_merged.bed.  Filtered 333 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 112.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/campylobacter/initial_intervals/Campylobacter_volucris_GCF_000816345_merged.bed --twobit results/phyluce/campylobacter/cleaned_genomes/Campylobacter_jejuni_GCF_000009085_formatted.2bit --output results/phyluce/campylobacter/initial_intervals/Campylobacter_volucris_GCF_000816345_stripped.bed;\n",
      "campylobacter_armoricus_gcf_009036385.\n",
      "campylobacter_avium_gcf_002238335.\n",
      "campylobacter_blaseri_gcf_003015205.\n",
      "campylobacter_coli_gcf_002024185.\n",
      "campylobacter_concisus_gcf_001298465.\n",
      "campylobacter_corcagiensis_gcf_000597805.\n",
      "campylobacter_cuniculorum_gcf_002104335.\n",
      "campylobacter_curvus_gcf_000017465.\n",
      "campylobacter_fetus_gcf_008271385.\n",
      "campylobacter_geochelonis_gcf_900065885.\n",
      "campylobacter_gracilis_gcf_001190745.\n",
      "campylobacter_helveticus_gcf_002080395.\n",
      "campylobacter_hepaticus_gcf_001687475..\n",
      "campylobacter_hominis_gcf_000017585.\n",
      "campylobacter_hyointestinalis_gcf_001643955.\n",
      "campylobacter_iguaniorum_gcf_000736415.\n",
      "campylobacter_insulaenigrae_gcf_000816185.\n",
      "campylobacter_lanienae_gcf_002139935.\n",
      "campylobacter_lari_gcf_000019205.\n",
      "campylobacter_ornithocola_gcf_001705345.\n",
      "campylobacter_peloridis_gcf_000816785.\n",
      "campylobacter_pinnipediorum_gcf_002021925.\n",
      "campylobacter_rectus_gcf_000174175.\n",
      "campylobacter_showae_gcf_000313615.\n",
      "campylobacter_sp_gcf_002139895.\n",
      "campylobacter_sptorum_gcf_002220735.\n",
      "campylobacter_subantarcticus_gcf_000816305.\n",
      "campylobacter_troglodytis_gcf_006864425.\n",
      "campylobacter_upsaliensis_gcf_000620965.\n",
      "campylobacter_ureolyticus_gcf_001190755.\n",
      "campylobacter_volucris_gcf_000816345.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/campylobacter/initial_intervals/Campylobacter_volucris_GCF_000816345_merged.bed --twobit results/phyluce/campylobacter/cleaned_genomes/Campylobacter_jejuni_GCF_000009085_formatted.2bit --output results/phyluce/campylobacter/initial_intervals/Campylobacter_volucris_GCF_000816345_stripped.bed;\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 0 taxa:\t1,514.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 1 taxa:\t1,514.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 2 taxa:\t661.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 3 taxa:\t244.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 4 taxa:\t169.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 5 taxa:\t130.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 6 taxa:\t78.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 7 taxa:\t59.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 8 taxa:\t50.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 9 taxa:\t39.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 10 taxa:\t32.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 11 taxa:\t25.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 12 taxa:\t20.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 13 taxa:\t15.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 14 taxa:\t14.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 15 taxa:\t10.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 16 taxa:\t5.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 17 taxa:\t5.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 18 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 19 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 20 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 21 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 22 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 23 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 24 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 25 taxa:\t4.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 26 taxa:\t3.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 27 taxa:\t3.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 28 taxa:\t3.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 29 taxa:\t3.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 30 taxa:\t3.0\n",
      "Loci shared by Campylobacter_jejuni_GCF_000009085 + 31 taxa:\t3.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/campylobacter/initial_intervals/campylobacter-to-Campylobacter_jejuni_GCF_000009085.sqlite --base-taxon Campylobacter_jejuni_GCF_000009085 --output results/phyluce/campylobacter/initial_intervals/Campylobacter_jejuni_GCF_000009085_+4.bed --specific-counts 4;\n",
      "Counter({'campylobacter_hepaticus_gcf_001687475': 154, 'campylobacter_coli_gcf_002024185': 138, 'campylobacter_lari_gcf_000019205': 102, 'campylobacter_subantarcticus_gcf_000816305': 97, 'campylobacter_ornithocola_gcf_001705345': 93, 'campylobacter_peloridis_gcf_000816785': 92, 'campylobacter_armoricus_gcf_009036385': 91, 'campylobacter_volucris_gcf_000816345': 87, 'campylobacter_insulaenigrae_gcf_000816185': 66, 'campylobacter_cuniculorum_gcf_002104335': 58, 'campylobacter_helveticus_gcf_002080395': 57, 'campylobacter_upsaliensis_gcf_000620965': 48, 'campylobacter_avium_gcf_002238335': 11, 'campylobacter_geochelonis_gcf_900065885': 10, 'campylobacter_lanienae_gcf_002139935': 9, 'campylobacter_pinnipediorum_gcf_002021925': 9, 'campylobacter_fetus_gcf_008271385': 8, 'campylobacter_sp_gcf_002139895': 8, 'campylobacter_iguaniorum_gcf_000736415': 8, 'campylobacter_concisus_gcf_001298465': 7, 'campylobacter_ureolyticus_gcf_001190755': 6, 'campylobacter_blaseri_gcf_003015205': 6, 'campylobacter_rectus_gcf_000174175': 6, 'campylobacter_sptorum_gcf_002220735': 6, 'campylobacter_hyointestinalis_gcf_001643955': 6, 'campylobacter_curvus_gcf_000017465': 5, 'campylobacter_troglodytis_gcf_006864425': 5, 'campylobacter_corcagiensis_gcf_000597805': 5, 'campylobacter_hominis_gcf_000017585': 4, 'campylobacter_showae_gcf_000313615': 3, 'campylobacter_gracilis_gcf_001190745': 3})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 4\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/campylobacter/initial_intervals/Campylobacter_jejuni_GCF_000009085_+4.bed --twobit results/phyluce/campylobacter/cleaned_genomes/Campylobacter_jejuni_GCF_000009085_formatted.2bit --buffer-to 160 --output results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4.fasta;\n",
      "Screened 169 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 169.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4.fasta --probe-prefix uce_campylobacter_ --design campylobacter_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 153\n",
      "Probe Count = 296\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas --query results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  17:31:09\n",
      "Ended:  Thu Feb 06, 2020  17:31:09\n",
      "Time for execution:  0.00314044952393 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas                        --lastz results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes_vself.lastz                       --probe-prefix=uce_campylobacter_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 295 fasta sequences.  Filtered 3 duplicates. Kept 290.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/campylobacter/validate_intervals/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas --scaffoldlist Campylobacter_armoricus_GCF_009036385 Campylobacter_avium_GCF_002238335 Campylobacter_blaseri_GCF_003015205 Campylobacter_coli_GCF_002024185 Campylobacter_concisus_GCF_001298465 Campylobacter_corcagiensis_GCF_000597805 Campylobacter_cuniculorum_GCF_002104335 Campylobacter_curvus_GCF_000017465 Campylobacter_fetus_GCF_008271385 Campylobacter_geochelonis_GCF_900065885 Campylobacter_gracilis_GCF_001190745 Campylobacter_helveticus_GCF_002080395 Campylobacter_hepaticus_GCF_001687475 Campylobacter_hominis_GCF_000017585 Campylobacter_hyointestinalis_GCF_001643955 Campylobacter_iguaniorum_GCF_000736415 Campylobacter_insulaenigrae_GCF_000816185 Campylobacter_lanienae_GCF_002139935 Campylobacter_lari_GCF_000019205 Campylobacter_ornithocola_GCF_001705345 Campylobacter_peloridis_GCF_000816785 Campylobacter_pinnipediorum_GCF_002021925 Campylobacter_rectus_GCF_000174175 Campylobacter_showae_GCF_000313615 Campylobacter_sp_GCF_002139895 Campylobacter_sptorum_GCF_002220735 Campylobacter_subantarcticus_GCF_000816305 Campylobacter_troglodytis_GCF_006864425 Campylobacter_upsaliensis_GCF_000620965 Campylobacter_ureolyticus_GCF_001190755 Campylobacter_volucris_GCF_000816345 Campylobacter_jejuni_GCF_000009085 --genome-base-path results/phyluce/campylobacter/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/campylobacter/validate_intervals/campylobacter-to-Campylobacter_jejuni_GCF_000009085.sqlite --output results/phyluce/campylobacter/validate_intervals/lastz/;\n",
      "\n",
      "Running against Campylobacter_armoricus_GCF_009036385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpYWNwGN.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIaUU2u.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_armoricus_GCF_009036385.lastz\n",
      "Creating Campylobacter_armoricus_GCF_009036385 table\n",
      "Inserting data to Campylobacter_armoricus_GCF_009036385 table\n",
      "\n",
      "Running against Campylobacter_avium_GCF_002238335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpt6wZHf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6JKW3v.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_avium_GCF_002238335.lastz\n",
      "Creating Campylobacter_avium_GCF_002238335 table\n",
      "Inserting data to Campylobacter_avium_GCF_002238335 table\n",
      "\n",
      "Running against Campylobacter_blaseri_GCF_003015205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphxWUNg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzNEDrB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_blaseri_GCF_003015205.lastz\n",
      "Creating Campylobacter_blaseri_GCF_003015205 table\n",
      "Inserting data to Campylobacter_blaseri_GCF_003015205 table\n",
      "\n",
      "Running against Campylobacter_coli_GCF_002024185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphwYUfG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpD3dGuz.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_coli_GCF_002024185.lastz\n",
      "Creating Campylobacter_coli_GCF_002024185 table\n",
      "Inserting data to Campylobacter_coli_GCF_002024185 table\n",
      "\n",
      "Running against Campylobacter_concisus_GCF_001298465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqBvIBg.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9kNt6q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_concisus_GCF_001298465.lastz\n",
      "Creating Campylobacter_concisus_GCF_001298465 table\n",
      "Inserting data to Campylobacter_concisus_GCF_001298465 table\n",
      "\n",
      "Running against Campylobacter_corcagiensis_GCF_000597805.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJkELgH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIEp1Ff.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_corcagiensis_GCF_000597805.lastz\n",
      "Creating Campylobacter_corcagiensis_GCF_000597805 table\n",
      "Inserting data to Campylobacter_corcagiensis_GCF_000597805 table\n",
      "\n",
      "Running against Campylobacter_cuniculorum_GCF_002104335.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGAWbka.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0uNgK0.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_cuniculorum_GCF_002104335.lastz\n",
      "Creating Campylobacter_cuniculorum_GCF_002104335 table\n",
      "Inserting data to Campylobacter_cuniculorum_GCF_002104335 table\n",
      "\n",
      "Running against Campylobacter_curvus_GCF_000017465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7JX6xF.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpkShBxL.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_curvus_GCF_000017465.lastz\n",
      "Creating Campylobacter_curvus_GCF_000017465 table\n",
      "Inserting data to Campylobacter_curvus_GCF_000017465 table\n",
      "\n",
      "Running against Campylobacter_fetus_GCF_008271385.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6xx8yY.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPo5QUf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_fetus_GCF_008271385.lastz\n",
      "Creating Campylobacter_fetus_GCF_008271385 table\n",
      "Inserting data to Campylobacter_fetus_GCF_008271385 table\n",
      "\n",
      "Running against Campylobacter_geochelonis_GCF_900065885.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCQIapV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpClXJjb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_geochelonis_GCF_900065885.lastz\n",
      "Creating Campylobacter_geochelonis_GCF_900065885 table\n",
      "Inserting data to Campylobacter_geochelonis_GCF_900065885 table\n",
      "\n",
      "Running against Campylobacter_gracilis_GCF_001190745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpbmHV5j.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3Zfups.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_gracilis_GCF_001190745.lastz\n",
      "Creating Campylobacter_gracilis_GCF_001190745 table\n",
      "Inserting data to Campylobacter_gracilis_GCF_001190745 table\n",
      "\n",
      "Running against Campylobacter_helveticus_GCF_002080395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t/tmp/tmpgPDhyE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSGPfzE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_helveticus_GCF_002080395.lastz\n",
      "Creating Campylobacter_helveticus_GCF_002080395 table\n",
      "Inserting data to Campylobacter_helveticus_GCF_002080395 table\n",
      "\n",
      "Running against Campylobacter_hepaticus_GCF_001687475.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphluDLO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgTn_lB.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_hepaticus_GCF_001687475.lastz\n",
      "Creating Campylobacter_hepaticus_GCF_001687475 table\n",
      "Inserting data to Campylobacter_hepaticus_GCF_001687475 table\n",
      "\n",
      "Running against Campylobacter_hominis_GCF_000017585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpenvBjT.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpZr467Y.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_hominis_GCF_000017585.lastz\n",
      "Creating Campylobacter_hominis_GCF_000017585 table\n",
      "Inserting data to Campylobacter_hominis_GCF_000017585 table\n",
      "\n",
      "Running against Campylobacter_hyointestinalis_GCF_001643955.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpE7NIxW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbCWbKD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_hyointestinalis_GCF_001643955.lastz\n",
      "Creating Campylobacter_hyointestinalis_GCF_001643955 table\n",
      "Inserting data to Campylobacter_hyointestinalis_GCF_001643955 table\n",
      "\n",
      "Running against Campylobacter_iguaniorum_GCF_000736415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpEP25Z9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0hCezl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_iguaniorum_GCF_000736415.lastz\n",
      "Creating Campylobacter_iguaniorum_GCF_000736415 table\n",
      "Inserting data to Campylobacter_iguaniorum_GCF_000736415 table\n",
      "\n",
      "Running against Campylobacter_insulaenigrae_GCF_000816185.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpW3SolO.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5cEiSr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_insulaenigrae_GCF_000816185.lastz\n",
      "Creating Campylobacter_insulaenigrae_GCF_000816185 table\n",
      "Inserting data to Campylobacter_insulaenigrae_GCF_000816185 table\n",
      "\n",
      "Running against Campylobacter_lanienae_GCF_002139935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3pA469.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3wBWj1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_lanienae_GCF_002139935.lastz\n",
      "Creating Campylobacter_lanienae_GCF_002139935 table\n",
      "Inserting data to Campylobacter_lanienae_GCF_002139935 table\n",
      "\n",
      "Running against Campylobacter_lari_GCF_000019205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpp2G9iz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdfkwTg.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_lari_GCF_000019205.lastz\n",
      "Creating Campylobacter_lari_GCF_000019205 table\n",
      "Inserting data to Campylobacter_lari_GCF_000019205 table\n",
      "\n",
      "Running against Campylobacter_ornithocola_GCF_001705345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpMKdyY7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDKKMdR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_ornithocola_GCF_001705345.lastz\n",
      "Creating Campylobacter_ornithocola_GCF_001705345 table\n",
      "Inserting data to Campylobacter_ornithocola_GCF_001705345 table\n",
      "\n",
      "Running against Campylobacter_peloridis_GCF_000816785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnmtUcz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1ebeW_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_peloridis_GCF_000816785.lastz\n",
      "Creating Campylobacter_peloridis_GCF_000816785 table\n",
      "Inserting data to Campylobacter_peloridis_GCF_000816785 table\n",
      "\n",
      "Running against Campylobacter_pinnipediorum_GCF_002021925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpC9zeJP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpccRq3q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_pinnipediorum_GCF_002021925.lastz\n",
      "Creating Campylobacter_pinnipediorum_GCF_002021925 table\n",
      "Inserting data to Campylobacter_pinnipediorum_GCF_002021925 table\n",
      "\n",
      "Running against Campylobacter_rectus_GCF_000174175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpPW2Ijp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4sLtXJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_rectus_GCF_000174175.lastz\n",
      "Creating Campylobacter_rectus_GCF_000174175 table\n",
      "Inserting data to Campylobacter_rectus_GCF_000174175 table\n",
      "\n",
      "Running against Campylobacter_showae_GCF_000313615.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_X8y3G.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGItfbO.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_showae_GCF_000313615.lastz\n",
      "Creating Campylobacter_showae_GCF_000313615 table\n",
      "Inserting data to Campylobacter_showae_GCF_000313615 table\n",
      "\n",
      "Running against Campylobacter_sp_GCF_002139895.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdkG71L.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGt5aKp.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_sp_GCF_002139895.lastz\n",
      "Creating Campylobacter_sp_GCF_002139895 table\n",
      "Inserting data to Campylobacter_sp_GCF_002139895 table\n",
      "\n",
      "Running against Campylobacter_sptorum_GCF_002220735.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQF13Be.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpizW7V3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_sptorum_GCF_002220735.lastz\n",
      "Creating Campylobacter_sptorum_GCF_002220735 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Campylobacter_sptorum_GCF_002220735 table\n",
      "\n",
      "Running against Campylobacter_subantarcticus_GCF_000816305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6k_dBW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPWFHlG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_subantarcticus_GCF_000816305.lastz\n",
      "Creating Campylobacter_subantarcticus_GCF_000816305 table\n",
      "Inserting data to Campylobacter_subantarcticus_GCF_000816305 table\n",
      "\n",
      "Running against Campylobacter_troglodytis_GCF_006864425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpC_X02a.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpNYCl0J.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_troglodytis_GCF_006864425.lastz\n",
      "Creating Campylobacter_troglodytis_GCF_006864425 table\n",
      "Inserting data to Campylobacter_troglodytis_GCF_006864425 table\n",
      "\n",
      "Running against Campylobacter_upsaliensis_GCF_000620965.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpyio0XG.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpej2ruF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_upsaliensis_GCF_000620965.lastz\n",
      "Creating Campylobacter_upsaliensis_GCF_000620965 table\n",
      "Inserting data to Campylobacter_upsaliensis_GCF_000620965 table\n",
      "\n",
      "Running against Campylobacter_ureolyticus_GCF_001190755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdM0ayc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6sGVny.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_ureolyticus_GCF_001190755.lastz\n",
      "Creating Campylobacter_ureolyticus_GCF_001190755 table\n",
      "Inserting data to Campylobacter_ureolyticus_GCF_001190755 table\n",
      "\n",
      "Running against Campylobacter_volucris_GCF_000816345.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7QVUqV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpruRO2N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_volucris_GCF_000816345.lastz\n",
      "Creating Campylobacter_volucris_GCF_000816345 table\n",
      "Inserting data to Campylobacter_volucris_GCF_000816345 table\n",
      "\n",
      "Running against Campylobacter_jejuni_GCF_000009085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFp28jf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpgFrUMV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/campylobacter/validate_intervals/lastz/Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_Campylobacter_jejuni_GCF_000009085.lastz\n",
      "Creating Campylobacter_jejuni_GCF_000009085 table\n",
      "Inserting data to Campylobacter_jejuni_GCF_000009085 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/campylobacter/extract_probes_from_group/campylobacter_genome.conf --lastz results/phyluce/campylobacter/validate_intervals/lastz --probes 120 --probe-prefix uce_campylobacter_ --name-pattern \"Campylobacter_jejuni_GCF_000009085_+4_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/campylobacter/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 17:31:49,378 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 17:31:49,381 - Phyluce - INFO - ---- Working on Campylobacter_armoricus_GCF_009036385 genome ----\n",
      "2020-02-06 17:31:49,382 - Phyluce - INFO - Reading Campylobacter_armoricus_GCF_009036385 genome\n",
      "2020-02-06 17:31:49,960 - Phyluce - INFO - Campylobacter_armoricus_GCF_009036385: 148 uces, 0 dupes, 148 non-dupes, 11 orient drop, 4 length drop, 133 written\n",
      "2020-02-06 17:31:49,960 - Phyluce - INFO - ------ Working on Campylobacter_avium_GCF_002238335 genome ------\n",
      "2020-02-06 17:31:49,971 - Phyluce - INFO - Reading Campylobacter_avium_GCF_002238335 genome\n",
      "2020-02-06 17:31:50,461 - Phyluce - INFO - Campylobacter_avium_GCF_002238335: 145 uces, 0 dupes, 145 non-dupes, 2 orient drop, 9 length drop, 134 written\n",
      "2020-02-06 17:31:50,462 - Phyluce - INFO - ----- Working on Campylobacter_blaseri_GCF_003015205 genome -----\n",
      "2020-02-06 17:31:50,473 - Phyluce - INFO - Reading Campylobacter_blaseri_GCF_003015205 genome\n",
      "2020-02-06 17:31:50,914 - Phyluce - INFO - Campylobacter_blaseri_GCF_003015205: 137 uces, 4 dupes, 133 non-dupes, 1 orient drop, 2 length drop, 130 written\n",
      "2020-02-06 17:31:50,914 - Phyluce - INFO - ------- Working on Campylobacter_coli_GCF_002024185 genome ------\n",
      "2020-02-06 17:31:50,921 - Phyluce - INFO - Reading Campylobacter_coli_GCF_002024185 genome\n",
      "2020-02-06 17:31:51,472 - Phyluce - INFO - Campylobacter_coli_GCF_002024185: 153 uces, 0 dupes, 153 non-dupes, 9 orient drop, 3 length drop, 141 written\n",
      "2020-02-06 17:31:51,472 - Phyluce - INFO - ----- Working on Campylobacter_concisus_GCF_001298465 genome ----\n",
      "2020-02-06 17:31:51,473 - Phyluce - INFO - Reading Campylobacter_concisus_GCF_001298465 genome\n",
      "2020-02-06 17:31:51,930 - Phyluce - INFO - Campylobacter_concisus_GCF_001298465: 141 uces, 0 dupes, 141 non-dupes, 5 orient drop, 7 length drop, 129 written\n",
      "2020-02-06 17:31:51,930 - Phyluce - INFO - --- Working on Campylobacter_corcagiensis_GCF_000597805 genome --\n",
      "2020-02-06 17:31:51,931 - Phyluce - INFO - Reading Campylobacter_corcagiensis_GCF_000597805 genome\n",
      "2020-02-06 17:31:52,344 - Phyluce - INFO - Campylobacter_corcagiensis_GCF_000597805: 136 uces, 1 dupes, 135 non-dupes, 0 orient drop, 2 length drop, 133 written\n",
      "2020-02-06 17:31:52,345 - Phyluce - INFO - --- Working on Campylobacter_cuniculorum_GCF_002104335 genome ---\n",
      "2020-02-06 17:31:52,345 - Phyluce - INFO - Reading Campylobacter_cuniculorum_GCF_002104335 genome\n",
      "2020-02-06 17:31:52,853 - Phyluce - INFO - Campylobacter_cuniculorum_GCF_002104335: 151 uces, 0 dupes, 151 non-dupes, 2 orient drop, 12 length drop, 137 written\n",
      "2020-02-06 17:31:52,853 - Phyluce - INFO - ------ Working on Campylobacter_curvus_GCF_000017465 genome -----\n",
      "2020-02-06 17:31:52,854 - Phyluce - INFO - Reading Campylobacter_curvus_GCF_000017465 genome\n",
      "2020-02-06 17:31:53,302 - Phyluce - INFO - Campylobacter_curvus_GCF_000017465: 141 uces, 0 dupes, 141 non-dupes, 4 orient drop, 1 length drop, 136 written\n",
      "2020-02-06 17:31:53,302 - Phyluce - INFO - ------ Working on Campylobacter_fetus_GCF_008271385 genome ------\n",
      "2020-02-06 17:31:53,303 - Phyluce - INFO - Reading Campylobacter_fetus_GCF_008271385 genome\n",
      "2020-02-06 17:31:53,769 - Phyluce - INFO - Campylobacter_fetus_GCF_008271385: 143 uces, 0 dupes, 143 non-dupes, 2 orient drop, 5 length drop, 136 written\n",
      "2020-02-06 17:31:53,769 - Phyluce - INFO - --- Working on Campylobacter_geochelonis_GCF_900065885 genome ---\n",
      "2020-02-06 17:31:53,770 - Phyluce - INFO - Reading Campylobacter_geochelonis_GCF_900065885 genome\n",
      "2020-02-06 17:31:54,215 - Phyluce - INFO - Campylobacter_geochelonis_GCF_900065885: 139 uces, 1 dupes, 138 non-dupes, 0 orient drop, 2 length drop, 136 written\n",
      "2020-02-06 17:31:54,215 - Phyluce - INFO - ----- Working on Campylobacter_gracilis_GCF_001190745 genome ----\n",
      "2020-02-06 17:31:54,216 - Phyluce - INFO - Reading Campylobacter_gracilis_GCF_001190745 genome\n",
      "2020-02-06 17:31:54,597 - Phyluce - INFO - Campylobacter_gracilis_GCF_001190745: 121 uces, 0 dupes, 121 non-dupes, 5 orient drop, 0 length drop, 116 written\n",
      "2020-02-06 17:31:54,598 - Phyluce - INFO - ---- Working on Campylobacter_helveticus_GCF_002080395 genome ---\n",
      "2020-02-06 17:31:54,608 - Phyluce - INFO - Reading Campylobacter_helveticus_GCF_002080395 genome\n",
      "2020-02-06 17:31:55,132 - Phyluce - INFO - Campylobacter_helveticus_GCF_002080395: 148 uces, 0 dupes, 148 non-dupes, 7 orient drop, 5 length drop, 136 written\n",
      "2020-02-06 17:31:55,132 - Phyluce - INFO - ---- Working on Campylobacter_hepaticus_GCF_001687475 genome ----\n",
      "2020-02-06 17:31:55,133 - Phyluce - INFO - Reading Campylobacter_hepaticus_GCF_001687475 genome\n",
      "2020-02-06 17:31:55,680 - Phyluce - INFO - Campylobacter_hepaticus_GCF_001687475: 150 uces, 0 dupes, 150 non-dupes, 8 orient drop, 3 length drop, 139 written\n",
      "2020-02-06 17:31:55,680 - Phyluce - INFO - ----- Working on Campylobacter_hominis_GCF_000017585 genome -----\n",
      "2020-02-06 17:31:55,699 - Phyluce - INFO - Reading Campylobacter_hominis_GCF_000017585 genome\n",
      "2020-02-06 17:31:56,133 - Phyluce - INFO - Campylobacter_hominis_GCF_000017585: 132 uces, 0 dupes, 132 non-dupes, 4 orient drop, 3 length drop, 125 written\n",
      "2020-02-06 17:31:56,133 - Phyluce - INFO - - Working on Campylobacter_hyointestinalis_GCF_001643955 genome -\n",
      "2020-02-06 17:31:56,134 - Phyluce - INFO - Reading Campylobacter_hyointestinalis_GCF_001643955 genome\n",
      "2020-02-06 17:31:56,598 - Phyluce - INFO - Campylobacter_hyointestinalis_GCF_001643955: 140 uces, 0 dupes, 140 non-dupes, 5 orient drop, 2 length drop, 133 written\n",
      "2020-02-06 17:31:56,598 - Phyluce - INFO - ---- Working on Campylobacter_iguaniorum_GCF_000736415 genome ---\n",
      "2020-02-06 17:31:56,607 - Phyluce - INFO - Reading Campylobacter_iguaniorum_GCF_000736415 genome\n",
      "2020-02-06 17:31:57,090 - Phyluce - INFO - Campylobacter_iguaniorum_GCF_000736415: 140 uces, 0 dupes, 140 non-dupes, 2 orient drop, 4 length drop, 134 written\n",
      "2020-02-06 17:31:57,090 - Phyluce - INFO - -- Working on Campylobacter_insulaenigrae_GCF_000816185 genome --\n",
      "2020-02-06 17:31:57,091 - Phyluce - INFO - Reading Campylobacter_insulaenigrae_GCF_000816185 genome\n",
      "2020-02-06 17:31:57,617 - Phyluce - INFO - Campylobacter_insulaenigrae_GCF_000816185: 153 uces, 0 dupes, 153 non-dupes, 8 orient drop, 3 length drop, 142 written\n",
      "2020-02-06 17:31:57,618 - Phyluce - INFO - ----- Working on Campylobacter_lanienae_GCF_002139935 genome ----\n",
      "2020-02-06 17:31:57,618 - Phyluce - INFO - Reading Campylobacter_lanienae_GCF_002139935 genome\n",
      "2020-02-06 17:31:58,084 - Phyluce - INFO - Campylobacter_lanienae_GCF_002139935: 141 uces, 0 dupes, 141 non-dupes, 6 orient drop, 0 length drop, 135 written\n",
      "2020-02-06 17:31:58,084 - Phyluce - INFO - ------- Working on Campylobacter_lari_GCF_000019205 genome ------\n",
      "2020-02-06 17:31:58,085 - Phyluce - INFO - Reading Campylobacter_lari_GCF_000019205 genome\n",
      "2020-02-06 17:31:58,629 - Phyluce - INFO - Campylobacter_lari_GCF_000019205: 151 uces, 0 dupes, 151 non-dupes, 10 orient drop, 4 length drop, 137 written\n",
      "2020-02-06 17:31:58,629 - Phyluce - INFO - --- Working on Campylobacter_ornithocola_GCF_001705345 genome ---\n",
      "2020-02-06 17:31:58,630 - Phyluce - INFO - Reading Campylobacter_ornithocola_GCF_001705345 genome\n",
      "2020-02-06 17:31:59,145 - Phyluce - INFO - Campylobacter_ornithocola_GCF_001705345: 151 uces, 10 dupes, 141 non-dupes, 0 orient drop, 0 length drop, 141 written\n",
      "2020-02-06 17:31:59,145 - Phyluce - INFO - ---- Working on Campylobacter_peloridis_GCF_000816785 genome ----\n",
      "2020-02-06 17:31:59,146 - Phyluce - INFO - Reading Campylobacter_peloridis_GCF_000816785 genome\n",
      "2020-02-06 17:31:59,679 - Phyluce - INFO - Campylobacter_peloridis_GCF_000816785: 150 uces, 0 dupes, 150 non-dupes, 9 orient drop, 5 length drop, 136 written\n",
      "2020-02-06 17:31:59,680 - Phyluce - INFO - -- Working on Campylobacter_pinnipediorum_GCF_002021925 genome --\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 17:31:59,696 - Phyluce - INFO - Reading Campylobacter_pinnipediorum_GCF_002021925 genome\n",
      "2020-02-06 17:32:00,166 - Phyluce - INFO - Campylobacter_pinnipediorum_GCF_002021925: 129 uces, 0 dupes, 129 non-dupes, 4 orient drop, 9 length drop, 116 written\n",
      "2020-02-06 17:32:00,166 - Phyluce - INFO - ------ Working on Campylobacter_rectus_GCF_000174175 genome -----\n",
      "2020-02-06 17:32:00,167 - Phyluce - INFO - Reading Campylobacter_rectus_GCF_000174175 genome\n",
      "2020-02-06 17:32:00,553 - Phyluce - INFO - Campylobacter_rectus_GCF_000174175: 128 uces, 2 dupes, 126 non-dupes, 0 orient drop, 0 length drop, 126 written\n",
      "2020-02-06 17:32:00,553 - Phyluce - INFO - ------ Working on Campylobacter_showae_GCF_000313615 genome -----\n",
      "2020-02-06 17:32:00,553 - Phyluce - INFO - Reading Campylobacter_showae_GCF_000313615 genome\n",
      "2020-02-06 17:32:00,960 - Phyluce - INFO - Campylobacter_showae_GCF_000313615: 133 uces, 5 dupes, 128 non-dupes, 0 orient drop, 0 length drop, 128 written\n",
      "2020-02-06 17:32:00,960 - Phyluce - INFO - -------- Working on Campylobacter_sp_GCF_002139895 genome -------\n",
      "2020-02-06 17:32:00,968 - Phyluce - INFO - Reading Campylobacter_sp_GCF_002139895 genome\n",
      "2020-02-06 17:32:01,436 - Phyluce - INFO - Campylobacter_sp_GCF_002139895: 137 uces, 0 dupes, 137 non-dupes, 4 orient drop, 3 length drop, 130 written\n",
      "2020-02-06 17:32:01,436 - Phyluce - INFO - ----- Working on Campylobacter_sptorum_GCF_002220735 genome -----\n",
      "2020-02-06 17:32:01,443 - Phyluce - INFO - Reading Campylobacter_sptorum_GCF_002220735 genome\n",
      "2020-02-06 17:32:01,917 - Phyluce - INFO - Campylobacter_sptorum_GCF_002220735: 143 uces, 0 dupes, 143 non-dupes, 9 orient drop, 0 length drop, 134 written\n",
      "2020-02-06 17:32:01,917 - Phyluce - INFO - -- Working on Campylobacter_subantarcticus_GCF_000816305 genome -\n",
      "2020-02-06 17:32:01,918 - Phyluce - INFO - Reading Campylobacter_subantarcticus_GCF_000816305 genome\n",
      "2020-02-06 17:32:02,454 - Phyluce - INFO - Campylobacter_subantarcticus_GCF_000816305: 151 uces, 0 dupes, 151 non-dupes, 10 orient drop, 4 length drop, 137 written\n",
      "2020-02-06 17:32:02,454 - Phyluce - INFO - --- Working on Campylobacter_troglodytis_GCF_006864425 genome ---\n",
      "2020-02-06 17:32:02,455 - Phyluce - INFO - Reading Campylobacter_troglodytis_GCF_006864425 genome\n",
      "2020-02-06 17:32:02,912 - Phyluce - INFO - Campylobacter_troglodytis_GCF_006864425: 142 uces, 4 dupes, 138 non-dupes, 0 orient drop, 0 length drop, 138 written\n",
      "2020-02-06 17:32:02,912 - Phyluce - INFO - --- Working on Campylobacter_upsaliensis_GCF_000620965 genome ---\n",
      "2020-02-06 17:32:02,913 - Phyluce - INFO - Reading Campylobacter_upsaliensis_GCF_000620965 genome\n",
      "2020-02-06 17:32:03,398 - Phyluce - INFO - Campylobacter_upsaliensis_GCF_000620965: 147 uces, 6 dupes, 141 non-dupes, 0 orient drop, 0 length drop, 141 written\n",
      "2020-02-06 17:32:03,398 - Phyluce - INFO - --- Working on Campylobacter_ureolyticus_GCF_001190755 genome ---\n",
      "2020-02-06 17:32:03,409 - Phyluce - INFO - Reading Campylobacter_ureolyticus_GCF_001190755 genome\n",
      "2020-02-06 17:32:03,879 - Phyluce - INFO - Campylobacter_ureolyticus_GCF_001190755: 138 uces, 0 dupes, 138 non-dupes, 7 orient drop, 6 length drop, 125 written\n",
      "2020-02-06 17:32:03,879 - Phyluce - INFO - ----- Working on Campylobacter_volucris_GCF_000816345 genome ----\n",
      "2020-02-06 17:32:03,880 - Phyluce - INFO - Reading Campylobacter_volucris_GCF_000816345 genome\n",
      "2020-02-06 17:32:04,409 - Phyluce - INFO - Campylobacter_volucris_GCF_000816345: 153 uces, 0 dupes, 153 non-dupes, 8 orient drop, 4 length drop, 141 written\n",
      "2020-02-06 17:32:04,409 - Phyluce - INFO - ------ Working on Campylobacter_jejuni_GCF_000009085 genome -----\n",
      "2020-02-06 17:32:04,410 - Phyluce - INFO - Reading Campylobacter_jejuni_GCF_000009085 genome\n",
      "2020-02-06 17:32:04,949 - Phyluce - INFO - Campylobacter_jejuni_GCF_000009085: 153 uces, 0 dupes, 153 non-dupes, 4 orient drop, 10 length drop, 139 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/campylobacter/extract_probes_from_group/probe_fasta --output results/phyluce/campylobacter/extract_probes_from_group/multifastas.sqlite --base-taxon Campylobacter_jejuni_GCF_000009085;\n",
      "campylobacter_armoricus_gcf_009036385.\n",
      "campylobacter_avium_gcf_002238335.\n",
      "campylobacter_blaseri_gcf_003015205.\n",
      "campylobacter_coli_gcf_002024185.\n",
      "campylobacter_concisus_gcf_001298465.\n",
      "campylobacter_corcagiensis_gcf_000597805.\n",
      "campylobacter_cuniculorum_gcf_002104335.\n",
      "campylobacter_curvus_gcf_000017465.\n",
      "campylobacter_fetus_gcf_008271385.\n",
      "campylobacter_geochelonis_gcf_900065885.\n",
      "campylobacter_gracilis_gcf_001190745.\n",
      "campylobacter_helveticus_gcf_002080395.\n",
      "campylobacter_hepaticus_gcf_001687475.\n",
      "campylobacter_hominis_gcf_000017585.\n",
      "campylobacter_hyointestinalis_gcf_001643955.\n",
      "campylobacter_iguaniorum_gcf_000736415.\n",
      "campylobacter_insulaenigrae_gcf_000816185.\n",
      "campylobacter_lanienae_gcf_002139935.\n",
      "campylobacter_lari_gcf_000019205.\n",
      "campylobacter_ornithocola_gcf_001705345.\n",
      "campylobacter_peloridis_gcf_000816785.\n",
      "campylobacter_pinnipediorum_gcf_002021925.\n",
      "campylobacter_rectus_gcf_000174175.\n",
      "campylobacter_showae_gcf_000313615.\n",
      "campylobacter_sp_gcf_002139895.\n",
      "campylobacter_sptorum_gcf_002220735.\n",
      "campylobacter_subantarcticus_gcf_000816305.\n",
      "campylobacter_troglodytis_gcf_006864425.\n",
      "campylobacter_upsaliensis_gcf_000620965.\n",
      "campylobacter_ureolyticus_gcf_001190755.\n",
      "campylobacter_volucris_gcf_000816345.\n",
      "campylobacter_jejuni_gcf_000009085.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/campylobacter/extract_probes_from_group/multifastas.sqlite --base-taxon Campylobacter_jejuni_GCF_000009085;\n",
      "Loci shared by 0 taxa:\t153.0\n",
      "Loci shared by 1 taxa:\t153.0\n",
      "Loci shared by 2 taxa:\t153.0\n",
      "Loci shared by 3 taxa:\t153.0\n",
      "Loci shared by 4 taxa:\t153.0\n",
      "Loci shared by 5 taxa:\t152.0\n",
      "Loci shared by 6 taxa:\t152.0\n",
      "Loci shared by 7 taxa:\t152.0\n",
      "Loci shared by 8 taxa:\t150.0\n",
      "Loci shared by 9 taxa:\t148.0\n",
      "Loci shared by 10 taxa:\t145.0\n",
      "Loci shared by 11 taxa:\t145.0\n",
      "Loci shared by 12 taxa:\t144.0\n",
      "Loci shared by 13 taxa:\t144.0\n",
      "Loci shared by 14 taxa:\t144.0\n",
      "Loci shared by 15 taxa:\t142.0\n",
      "Loci shared by 16 taxa:\t142.0\n",
      "Loci shared by 17 taxa:\t140.0\n",
      "Loci shared by 18 taxa:\t137.0\n",
      "Loci shared by 19 taxa:\t133.0\n",
      "Loci shared by 20 taxa:\t129.0\n",
      "Loci shared by 21 taxa:\t126.0\n",
      "Loci shared by 22 taxa:\t126.0\n",
      "Loci shared by 23 taxa:\t125.0\n",
      "Loci shared by 24 taxa:\t122.0\n",
      "Loci shared by 25 taxa:\t121.0\n",
      "Loci shared by 26 taxa:\t121.0\n",
      "Loci shared by 27 taxa:\t117.0\n",
      "Loci shared by 28 taxa:\t114.0\n",
      "Loci shared by 29 taxa:\t110.0\n",
      "Loci shared by 30 taxa:\t102.0\n",
      "Loci shared by 31 taxa:\t93.0\n",
      "Loci shared by 32 taxa:\t86.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/campylobacter/extract_probes_from_group/multifastas.sqlite  --base-taxon Campylobacter_jejuni_GCF_000009085 --output results/phyluce/campylobacter/extract_probes_from_group/Campylobacter_jejuni_GCF_000009085+4-back-to-29.conf --specific-counts 29;\n",
      "Counter({'campylobacter_fetus_gcf_008271385': 110, 'campylobacter_curvus_gcf_000017465': 110, 'campylobacter_insulaenigrae_gcf_000816185': 110, 'campylobacter_jejuni_gcf_000009085': 110, 'campylobacter_peloridis_gcf_000816785': 110, 'campylobacter_lanienae_gcf_002139935': 110, 'campylobacter_cuniculorum_gcf_002104335': 110, 'campylobacter_hepaticus_gcf_001687475': 110, 'campylobacter_coli_gcf_002024185': 110, 'campylobacter_hyointestinalis_gcf_001643955': 110, 'campylobacter_volucris_gcf_000816345': 110, 'campylobacter_corcagiensis_gcf_000597805': 110, 'campylobacter_iguaniorum_gcf_000736415': 110, 'campylobacter_geochelonis_gcf_900065885': 110, 'campylobacter_avium_gcf_002238335': 110, 'campylobacter_ureolyticus_gcf_001190755': 109, 'campylobacter_subantarcticus_gcf_000816305': 109, 'campylobacter_concisus_gcf_001298465': 109, 'campylobacter_sptorum_gcf_002220735': 109, 'campylobacter_helveticus_gcf_002080395': 109, 'campylobacter_upsaliensis_gcf_000620965': 109, 'campylobacter_lari_gcf_000019205': 108, 'campylobacter_armoricus_gcf_009036385': 108, 'campylobacter_ornithocola_gcf_001705345': 108, 'campylobacter_hominis_gcf_000017585': 108, 'campylobacter_troglodytis_gcf_006864425': 107, 'campylobacter_sp_gcf_002139895': 107, 'campylobacter_showae_gcf_000313615': 106, 'campylobacter_blaseri_gcf_003015205': 105, 'campylobacter_pinnipediorum_gcf_002021925': 105, 'campylobacter_rectus_gcf_000174175': 104, 'campylobacter_gracilis_gcf_001190745': 101})\n",
      "Total loci = 110\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(29)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/campylobacter/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/campylobacter/extract_probes_from_group/Campylobacter_jejuni_GCF_000009085+4-back-to-29.conf --probe-prefix uce_campylobacter_ --designer rnplattii --design campylobacter_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 110\n",
      "Probe Count = 6312\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list.fasta --query results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  17:33:14\n",
      "Ended:  Thu Feb 06, 2020  17:33:30\n",
      "Time for execution:  0.282086964448 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list.fasta --lastz results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_campylobacter_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 6311 fasta sequences.  Filtered 0 duplicates. Kept 6312.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/campylobacter/final_probe_design/campylobacter_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 17:39:40 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 6312\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 504960\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 1M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 16M = 33M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 87M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3954246\n",
      "Max number of word counting entries: 89010081\n",
      "\n",
      "# comparing sequences from          0  to       1052\n",
      ".---------- new table with      936 representatives\n",
      "# comparing sequences from       1052  to       1928\n",
      "----------    135 remaining sequences to the next cycle\n",
      "---------- new table with      686 representatives\n",
      "# comparing sequences from       1793  to       2546\n",
      "----------    234 remaining sequences to the next cycle\n",
      "---------- new table with      457 representatives\n",
      "# comparing sequences from       2312  to       2978\n",
      "----------    219 remaining sequences to the next cycle\n",
      "---------- new table with      410 representatives\n",
      "# comparing sequences from       2759  to       3351\n",
      "----------    107 remaining sequences to the next cycle\n",
      "---------- new table with      466 representatives\n",
      "# comparing sequences from       3244  to       3755\n",
      "----------     80 remaining sequences to the next cycle\n",
      "---------- new table with      406 representatives\n",
      "# comparing sequences from       3675  to       4114\n",
      "----------    165 remaining sequences to the next cycle\n",
      "---------- new table with      256 representatives\n",
      "# comparing sequences from       3949  to       4342\n",
      "----------    165 remaining sequences to the next cycle\n",
      "---------- new table with      218 representatives\n",
      "# comparing sequences from       4177  to       4532\n",
      "----------    104 remaining sequences to the next cycle\n",
      "---------- new table with      223 representatives\n",
      "# comparing sequences from       4428  to       4742\n",
      "----------     34 remaining sequences to the next cycle\n",
      "---------- new table with      255 representatives\n",
      "# comparing sequences from       4708  to       4975\n",
      "----------     59 remaining sequences to the next cycle\n",
      "---------- new table with      176 representatives\n",
      "# comparing sequences from       4916  to       5148\n",
      "----------      1 remaining sequences to the next cycle\n",
      "---------- new table with      205 representatives\n",
      "# comparing sequences from       5147  to       5341\n",
      "----------     58 remaining sequences to the next cycle\n",
      "---------- new table with      115 representatives\n",
      "# comparing sequences from       5283  to       5454\n",
      "99.8%---------- new table with      155 representatives\n",
      "# comparing sequences from       5454  to       6312\n",
      ".....................---------- new table with      797 representatives\n",
      "\n",
      "     6312  finished       5761  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 88M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 1.29\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "186px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
