{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import shutil\n",
    "from shutil import copy\n",
    "import time\n",
    "from collections import defaultdict\n",
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "os.chdir(\"/master/nplatt/pathogen_probes/\")\n",
    "\n",
    "\n",
    "def wait_on_running_jobs():\n",
    "   \n",
    "    num_jobs = 1\n",
    "    \n",
    "    while num_jobs > 0:\n",
    "        num_jobs = len(subprocess.check_output('qstat', shell=True).split(\"\\n\")) - 2\n",
    "        time.sleep(60)\n",
    "        print(\".\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Streptococcus"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep directory and group info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a clean directory to work from and build a list of all the taxa to include.  Desired taxa should includ Genus, species, and assembly accession like so : \n",
    "\n",
    "```\n",
    "Genus_species_GCA_123456789\n",
    "```\n",
    "\n",
    "In addition, for each \"group\" (albiet genus, phylum, etc...) we need to designate a `reference_taxon`.  This is the species/genome that all the others will be compared to.  Ideally it will be from a high quality genome.  \n",
    "\n",
    "I used NCBI Genomes to identify all available __\"representative\"__ assemblies for each group.  I extracted the accesion numbers and genreated a csv file that looks like:\n",
    "\n",
    "```\n",
    "$> cat data/genomes/trematode_accessions.csv\n",
    "genus,species,accession\n",
    "Clonorchis,sinensis,GCA_003604175.1\n",
    "Dicrocoelium,dendriticum,GCA_000950715.1\n",
    "Echinostoma,caproni,GCA_900618425.1\n",
    "Fasciola,gigantica,GCA_006461475.1\n",
    "Fasciola,hepatica,GCA_900302435.1\n",
    "Fasciolopsis,buski,GCA_008360955.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group = 'streptococcus'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group)\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "group_taxa = [ \"Streptococcus_acidominimus_GCF_900187245\",\n",
    "               \"Streptococcus_agalactiae_GCF_000007265\",\n",
    "               \"Streptococcus_alactolyticus_GCF_009695625\",\n",
    "               \"Streptococcus_anginosus_GCF_000463505\",\n",
    "               \"Streptococcus_australis_GCF_900476055\",\n",
    "               \"Streptococcus_azizii_GCF_001984715\",\n",
    "               \"Streptococcus_bovimastitidis_GCF_001885095\",\n",
    "               \"Streptococcus_caballi_GCF_000379985\",\n",
    "               \"Streptococcus_canis_GCF_900636575\",\n",
    "               \"Streptococcus_castoreus_GCF_000425025\",\n",
    "               \"Streptococcus_constellatus_GCF_000463425\",\n",
    "               \"Streptococcus_criceti_GCF_000187975\",\n",
    "               \"Streptococcus_cristatus_GCF_000385925\",\n",
    "               \"Streptococcus_cuniculi_GCF_001921845\",\n",
    "               \"Streptococcus_danieliae_GCF_009767945\",\n",
    "               \"Streptococcus_devriesei_GCF_000423725\",\n",
    "               \"Streptococcus_didelphis_GCF_000380005\",\n",
    "               \"Streptococcus_downei_GCF_900459175\",\n",
    "               \"Streptococcus_dysgalactiae_GCF_000317855\",\n",
    "               \"Streptococcus_entericus_GCF_000380025\",\n",
    "               \"Streptococcus_equi_GCF_000026605\",\n",
    "               \"Streptococcus_equinus_GCF_000964315\",\n",
    "               \"Streptococcus_ferus_GCF_000372425\",\n",
    "               \"Streptococcus_gallolyticus_GCF_002000985\",\n",
    "               \"Streptococcus_gordonii_GCF_000017005\",\n",
    "               \"Streptococcus_halichoeri_GCF_009870755\",\n",
    "               \"Streptococcus_halitosis_GCF_003143695\",\n",
    "               \"Streptococcus_halotolerans_GCF_001598035\",\n",
    "               \"Streptococcus_henryi_GCF_000376985\",\n",
    "               \"Streptococcus_himalayensis_GCF_001708305\",\n",
    "               \"Streptococcus_hongkongensis_GCF_000785845\",\n",
    "               \"Streptococcus_hyointestinalis_GCF_900459405\",\n",
    "               \"Streptococcus_hyovaginalis_GCF_000420785\",\n",
    "               \"Streptococcus_ictaluri_GCF_000188015\",\n",
    "               \"Streptococcus_infantarius_GCF_000246835\",\n",
    "               \"Streptococcus_infantis_GCF_000187465\",\n",
    "               \"Streptococcus_iniae_GCF_000831485\",\n",
    "               \"Streptococcus_intermedius_GCF_000463355\",\n",
    "               \"Streptococcus_lutetiensis_GCF_900475675\",\n",
    "               \"Streptococcus_macacae_GCF_000187995\",\n",
    "               \"Streptococcus_macedonicus_GCF_000283635\",\n",
    "               \"Streptococcus_marimammalium_GCF_000380045\",\n",
    "               \"Streptococcus_marmotae_GCF_001623565\",\n",
    "               \"Streptococcus_massiliensis_GCF_000380065\",\n",
    "               \"Streptococcus_merionis_GCF_000380085\",\n",
    "               \"Streptococcus_milleri_GCF_900636715\",\n",
    "               \"Streptococcus_minor_GCF_000377005\",\n",
    "               \"Streptococcus_mitis_GCF_000027165\",\n",
    "               \"Streptococcus_mutans_GCF_000007465\",\n",
    "               \"Streptococcus_oralis_GCF_900637025\",\n",
    "               \"Streptococcus_orisasini_GCF_001431045\",\n",
    "               \"Streptococcus_orisratti_GCF_000380105\",\n",
    "               \"Streptococcus_ovis_GCF_000380125\",\n",
    "               \"Streptococcus_pantholopis_GCF_001642085\",\n",
    "               \"Streptococcus_parasanguinis_GCF_000164675\",\n",
    "               \"Streptococcus_parasuis_GCF_004283785\",\n",
    "               \"Streptococcus_parauberis_GCF_000213825\",\n",
    "               \"Streptococcus_pasteurianus_GCF_900478025\",\n",
    "               \"Streptococcus_penaeicida_GCF_002887775\",\n",
    "               \"Streptococcus_peroris_GCF_000187585\",\n",
    "               \"Streptococcus_pharyngis_GCF_007859195\",\n",
    "               \"Streptococcus_phocae_GCF_000772915\",\n",
    "               \"Streptococcus_pluranimalium_GCF_003352995\",\n",
    "               \"Streptococcus_plurextorum_GCF_000423745\",              \n",
    "               \"Streptococcus_porci_GCF_000423765\",\n",
    "               \"Streptococcus_porcinus_GCF_900475415\",\n",
    "               \"Streptococcus_pseudopneumoniae_GCF_000221985\",\n",
    "               \"Streptococcus_pseudoporcinus_GCF_000188035\",\n",
    "               \"Streptococcus_pyogenes_GCF_000006785\",\n",
    "               \"Streptococcus_ratti_GCF_000286075\",\n",
    "               \"Streptococcus_respraculi_GCF_003595525\",\n",
    "               \"Streptococcus_rubneri_GCF_004785935\",\n",
    "               \"Streptococcus_ruminantium_GCF_003609975\",\n",
    "               \"Streptococcus_salivarius_GCF_000785515\",\n",
    "               \"Streptococcus_sanguinis_GCF_000014205\",\n",
    "               \"Streptococcus_sinensis_GCF_000767835\",\n",
    "               \"Streptococcus_sobrinus_GCF_000686605\",\n",
    "               \"Streptococcus_sp_GCF_003086355\",\n",
    "               \"Streptococcus_suis_GCF_000026745\",\n",
    "               \"Streptococcus_thermophilus_GCF_000253395\",\n",
    "               \"Streptococcus_thoraltensis_GCF_000380145\",\n",
    "               \"Streptococcus_timonensis_GCF_900095845\",\n",
    "               \"Streptococcus_troglodytae_GCF_002355215\",\n",
    "               \"Streptococcus_uberis_GCF_000009545\",\n",
    "               \"Streptococcus_urinalis_GCF_000188055\",\n",
    "               \"Streptococcus_varani_GCF_001375655\",\n",
    "               \"Streptococcus_vestibularis_GCF_900636445\",\n",
    "               \"Streptococcus_viridans_GCF_900636365\" ]\n",
    "                    \n",
    "reference_taxon = \"Streptococcus_pneumoniae_GCF_000007045\"\n",
    "\n",
    "all_taxa = group_taxa + [ reference_taxon ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read from the accessions `data/genomes/*_accessions.csv` then find the ftp link, download, and format the genome file name in a standard format `Genus_species_GCA_123456789`\n",
    "\n",
    "For sanity's sake, I have saved all the ftp info in a csv file stored in `data/genomes/*group*`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900187245.1_51470_D02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 687125 bytes  196333.43 bytes/sec\n",
      "total size is 686853  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007265.1_ASM726v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 634787 bytes  423219.33 bytes/sec\n",
      "total size is 634524  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009695625.1_ASM969562v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 522123 bytes  348110.00 bytes/sec\n",
      "total size is 521889  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000463505.1_ASM46350v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 659465 bytes  439671.33 bytes/sec\n",
      "total size is 659192  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900476055.1_53750_F01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 636295 bytes  1272674.00 bytes/sec\n",
      "total size is 636031  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001984715.1_ASM198471v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 691194 bytes  460824.00 bytes/sec\n",
      "total size is 690912  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001885095.1_ASM188509v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 613156 bytes  408798.67 bytes/sec\n",
      "total size is 612898  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000379985.1_ASM37998v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 635318 bytes  423573.33 bytes/sec\n",
      "total size is 635053  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900636575.1_42912_C01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 613705 bytes  409164.67 bytes/sec\n",
      "total size is 613449  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000425025.1_ASM42502v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 561215 bytes  374171.33 bytes/sec\n",
      "total size is 560966  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000463425.1_ASM46342v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 588864 bytes  392604.00 bytes/sec\n",
      "total size is 588615  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187975.3_ASM18797v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 718440 bytes  478988.00 bytes/sec\n",
      "total size is 718159  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000385925.1_ASM38592v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 634315 bytes  422904.67 bytes/sec\n",
      "total size is 634050  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001921845.1_ASM192184v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 622464 bytes  415004.00 bytes/sec\n",
      "total size is 622206  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009767945.1_ASM976794v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 559301 bytes  372895.33 bytes/sec\n",
      "total size is 559051  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000423725.1_ASM42372v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 631010 bytes  420701.33 bytes/sec\n",
      "total size is 630745  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380005.1_ASM38000v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 552950 bytes  368661.33 bytes/sec\n",
      "total size is 552709  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900459175.1_52087_B01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 662540 bytes  441721.33 bytes/sec\n",
      "total size is 662268  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000317855.1_ASM31785v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 645532 bytes  430382.67 bytes/sec\n",
      "total size is 645267  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380025.1_ASM38002v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 609952 bytes  406662.67 bytes/sec\n",
      "total size is 609695  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000026605.1_ASM2660v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 634515 bytes  423038.00 bytes/sec\n",
      "total size is 634251  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000964315.1_ASM96431v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 565160 bytes  376801.33 bytes/sec\n",
      "total size is 564911  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000372425.1_ASM37242v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 551834 bytes  367917.33 bytes/sec\n",
      "total size is 551593  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002000985.1_ASM200098v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 731886 bytes  487952.00 bytes/sec\n",
      "total size is 731596  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000017005.1_ASM1700v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 650116 bytes  433438.67 bytes/sec\n",
      "total size is 649852  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_009870755.1_ASM987075v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 569389 bytes  379620.67 bytes/sec\n",
      "total size is 569139  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003143695.2_ASM314369v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 560341 bytes  373588.67 bytes/sec\n",
      "total size is 560091  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001598035.1_ASM159803v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 639476 bytes  426345.33 bytes/sec\n",
      "total size is 639210  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000376985.1_ASM37698v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 721784 bytes  481217.33 bytes/sec\n",
      "total size is 721495  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001708305.1_ASM170830v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 666402 bytes  444296.00 bytes/sec\n",
      "total size is 666128  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000785845.1_CAIM1895T_spades_scaffold_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 590058 bytes  393400.00 bytes/sec\n",
      "total size is 589794  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900459405.1_41965_B01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 746015 bytes  497371.33 bytes/sec\n",
      "total size is 745727  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000420785.1_ASM42078v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 621677 bytes  414479.33 bytes/sec\n",
      "total size is 621420  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000188015.3_ASM18801v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657139 bytes  438120.67 bytes/sec\n",
      "total size is 656866  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000246835.1_ASM24683v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 590879 bytes  393947.33 bytes/sec\n",
      "total size is 590622  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187465.1_ASM18746v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 558304 bytes  372230.67 bytes/sec\n",
      "total size is 558055  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000831485.1_ASM83148v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 613389 bytes  408954.00 bytes/sec\n",
      "total size is 613132  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000463355.1_ASM46335v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 589479 bytes  393014.00 bytes/sec\n",
      "total size is 589230  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900475675.1_45473_D02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 525723 bytes  350510.00 bytes/sec\n",
      "total size is 525483  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187995.3_ASM18799v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 564124 bytes  376110.67 bytes/sec\n",
      "total size is 563875  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000283635.1_ASM28363v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 625437 bytes  416986.00 bytes/sec\n",
      "total size is 625172  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380045.1_ASM38004v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 439459 bytes  293000.67 bytes/sec\n",
      "total size is 439242  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001623565.1_ASM162356v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 684540 bytes  456388.00 bytes/sec\n",
      "total size is 684266  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000380065.1_ASM38006v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 557536 bytes  371718.67 bytes/sec\n",
      "total size is 557287  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380085.1_ASM38008v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 705605 bytes  470431.33 bytes/sec\n",
      "total size is 705324  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900636715.1_43908_E01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 557131 bytes  371448.67 bytes/sec\n",
      "total size is 556891  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000377005.1_ASM37700v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 576037 bytes  384052.67 bytes/sec\n",
      "total size is 575788  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000027165.1_ASM2716v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 629208 bytes  419500.00 bytes/sec\n",
      "total size is 628944  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007465.2_ASM746v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 597171 bytes  398142.00 bytes/sec\n",
      "total size is 596916  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900637025.1_46338_H01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 571143 bytes  380790.00 bytes/sec\n",
      "total size is 570895  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001431045.1_ASM143104v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 685495 bytes  457024.67 bytes/sec\n",
      "total size is 685221  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380105.1_ASM38010v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 722804 bytes  481897.33 bytes/sec\n",
      "total size is 722515  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380125.1_ASM38012v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 703797 bytes  469226.00 bytes/sec\n",
      "total size is 703516  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001642085.1_ASM164208v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657218 bytes  438173.33 bytes/sec\n",
      "total size is 656944  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000164675.2_ASM16467v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 637082 bytes  424749.33 bytes/sec\n",
      "total size is 636817  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004283785.1_ASM428378v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 562323 bytes  374910.00 bytes/sec\n",
      "total size is 562073  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000213825.1_ASM21382v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 631468 bytes  421006.67 bytes/sec\n",
      "total size is 631203  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900478025.1_48128_C02_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 639261 bytes  426202.00 bytes/sec\n",
      "total size is 638997  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002887775.1_ASM288777v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 606631 bytes  404448.67 bytes/sec\n",
      "total size is 606373  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000187585.1_ASM18758v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 485318 bytes  970720.00 bytes/sec\n",
      "total size is 485093  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_007859195.1_ASM785919v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 449091 bytes  299422.00 bytes/sec\n",
      "total size is 448873  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000772915.1_ASM77291v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 497553 bytes  331730.00 bytes/sec\n",
      "total size is 497320  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003352995.1_ASM335299v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 657971 bytes  438675.33 bytes/sec\n",
      "total size is 657697  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000423745.1_ASM42374v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 627729 bytes  1255542.00 bytes/sec\n",
      "total size is 627464  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000007045.1_ASM704v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 602545 bytes  401724.67 bytes/sec\n",
      "total size is 602290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000423765.1_ASM42376v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 684563 bytes  456403.33 bytes/sec\n",
      "total size is 684290  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900475415.1_42206_H01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 600001 bytes  400028.67 bytes/sec\n",
      "total size is 599745  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000221985.1_ASM22198v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 649335 bytes  432918.00 bytes/sec\n",
      "total size is 649070  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000188035.3_ASM18803v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 629490 bytes  419688.00 bytes/sec\n",
      "total size is 629225  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000006785.2_ASM678v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 548272 bytes  365542.67 bytes/sec\n",
      "total size is 548033  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000286075.1_ASM28607v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 612045 bytes  1224174.00 bytes/sec\n",
      "total size is 611788  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003595525.1_ASM359552v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 612875 bytes  408611.33 bytes/sec\n",
      "total size is 612617  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_004785935.1_ASM478593v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 639521 bytes  426375.33 bytes/sec\n",
      "total size is 639255  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003609975.1_ASM360997v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 617986 bytes  412018.67 bytes/sec\n",
      "total size is 617728  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000785515.1_ASM78551v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 646181 bytes  430815.33 bytes/sec\n",
      "total size is 645916  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000014205.1_ASM1420v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 708385 bytes  472284.67 bytes/sec\n",
      "total size is 708105  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000767835.1_ASM76783v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 618228 bytes  412180.00 bytes/sec\n",
      "total size is 617971  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000686605.1_ASM68660v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 641205 bytes  427498.00 bytes/sec\n",
      "total size is 640940  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_003086355.2_ASM308635v2_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 720458 bytes  1441000.00 bytes/sec\n",
      "total size is 720176  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000026745.1_ASM2674v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 644781 bytes  429882.00 bytes/sec\n",
      "total size is 644517  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000253395.1_ASM25339v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 570272 bytes  380209.33 bytes/sec\n",
      "total size is 570023  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000380145.1_ASM38014v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 708641 bytes  472455.33 bytes/sec\n",
      "total size is 708360  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900095845.1_PRJEB15309_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 569860 bytes  379934.67 bytes/sec\n",
      "total size is 569611  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_002355215.1_ASM235521v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 617097 bytes  411426.00 bytes/sec\n",
      "total size is 616839  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000009545.1_ASM954v1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 546418 bytes  364306.67 bytes/sec\n",
      "total size is 546179  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_000188055.3_ASM18805v3_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 622696 bytes  415158.67 bytes/sec\n",
      "total size is 622439  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_001375655.1_PRJEB8936_assembly_1_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 728228 bytes  485513.33 bytes/sec\n",
      "total size is 727929  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GCA_900636445.1_41965_G01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 575065 bytes  383404.67 bytes/sec\n",
      "total size is 574817  speedup is 1.00\n",
      "\n",
      "Warning Notice!\n",
      "\n",
      "You are accessing a U.S. Government information system which includes this\n",
      "computer, network, and all attached devices. This system is for\n",
      "Government-authorized use only. Unauthorized use of this system may result in\n",
      "disciplinary action and civil and criminal penalties. System users have no\n",
      "expectation of privacy regarding any communications or data processed by this\n",
      "system. At any time, the government may monitor, record, or seize any\n",
      "communication or data transiting or stored on this information system.\n",
      "\n",
      "-------------------------------------------------------------------------------\n",
      "\n",
      "Welcome to the NCBI rsync server.\n",
      "\n",
      "\n",
      "GCA_900636365.1_41559_G01_genomic.fna.gz\n",
      "\n",
      "sent 42 bytes  received 592864 bytes  395270.67 bytes/sec\n",
      "total size is 592608  speedup is 1.00\n"
     ]
    }
   ],
   "source": [
    "#get all of the genomes and gffs for a single representative\n",
    "if not os.path.exists(\"data/genomes/\" + group):\n",
    "    os.makedirs(\"data/genomes/\" + group)\n",
    "    \n",
    "accession_df = pd.read_csv(\"data/\" + group +  \"_accessions.csv\")\n",
    "accession_w_ftp=[]\n",
    "\n",
    "for index, row in accession_df.iterrows():\n",
    "    \n",
    "    #get the ftp site from the accession\n",
    "    ftp=!(esearch -db assembly -query {row[\"accession\"]} | esummary | xtract -pattern DocumentSummary -element FtpPath_GenBank)\n",
    "    ftp=ftp[0].replace(\"ftp://\", \"\")\n",
    "    \n",
    "    #generate a fas and download\n",
    "    fas_ftp_link = ftp + \"/\" + ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "    fas_rsync_cmd=\"rsync --copy-links --times --verbose rsync://{} data/genomes/{}\".format(fas_ftp_link, group)\n",
    "   \n",
    "    !{fas_rsync_cmd}\n",
    "\n",
    "    #create \"clean/standardized\" name for gff and fas\n",
    "    #Genus_species_GCA_12345678\n",
    "    clean_fas_name = \"_\".join([ str(row[\"genus\"]), str(row[\"species\"]), row[\"accession\"].split(\".\")[0]]) + \".fas.gz\"\n",
    "\n",
    "    orig_fas_name = ftp.split(\"/\")[-1] + \"_genomic.fna.gz\"\n",
    "\n",
    "    os.rename(\"data/genomes/\" + group + \"/\" + orig_fas_name, \"data/genomes/\" + group + \"/\" + clean_fas_name)\n",
    "    \n",
    "    #finally make a table documenting changes\n",
    "    #accession, fas_ftp, gff_ftp, original file name, updated file name\n",
    "    xfer_info = [row[\"accession\"], ftp, orig_fas_name, clean_fas_name]\n",
    "    accession_w_ftp.append(xfer_info)\n",
    "    \n",
    "#convert the tableto a df\n",
    "ftp_df=pd.DataFrame(accession_w_ftp, columns = [\"accession\", \n",
    "                                                \"ftp\", \n",
    "                                                \"orig_fas_name\", \n",
    "                                                \"new_fas_name\"])\n",
    "\n",
    "#save to csv\n",
    "ftp_df.to_csv(\"data/genomes/\" + group + \"/ftp_info.csv\", index=False)\n",
    "\n",
    "#finally uncompress all of the gzipped files (will be easier to manage donw the line)\n",
    "!gunzip data/genomes/{group}/*gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clean up and format genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Further process the genomes so that the headers only contain the accession and then the entire genome is converted into 2bit.  These are reqs. for the `phyluce` pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/cleaned_genomes\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/cleaned_genomes\")\n",
    "\n",
    "for sample in all_taxa:\n",
    "        \n",
    "        i_genome_file = \"data/genomes/\" + group + \"/\" + sample + \".fas\"\n",
    "        o_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "        o_2bit_file   = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "        \n",
    "        with open(i_genome_file, \"rU\") as infile:\n",
    "            with open(o_genome_file, \"w\") as outfile:\n",
    "                for seq in SeqIO.parse(infile, 'fasta'):\n",
    "                    seq.name = \"\"\n",
    "                    seq.description = \"\"\n",
    "                    outfile.write(seq.format('fasta'))\n",
    "                    \n",
    "        f2bit_cmd = \"faToTwoBit {} {}\".format(i_genome_file, o_2bit_file)\n",
    "        subprocess.call(f2bit_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simluate and map reads at 10x coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use `art` to simulate 25x coverage of error free reads for each species/genome.  In the next step these reads will all be mapped against the reference taxon to find conserved regions were we can map reads from multiple different species.  \n",
    "\n",
    "A quick comment about coverage: In most cases 4-6x is acceptable. I have chosen 25x coverage because it should provide 4bp resolution on conserved orthologous regions and since my genomes are relatilvely small...I can trade compute time for resolution.\n",
    "\n",
    "__NOTE__: Here and in several other places throught the notebook(s) jobs are not run on the local machine, but instead sent to our cluster via PBS.  In these cases you cannot proceed to the next step in the pipeline/notebook untill all jobs from the current step have completed.  In some cases this mean waiting up to a few hours.  These steps are identifed in giant/commented block text. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481185 (\"sim_Streptococcus_acidominimus_GCF_900187245\") has been submitted\n",
      "Your job 5481186 (\"sim_Streptococcus_agalactiae_GCF_000007265\") has been submitted\n",
      "Your job 5481187 (\"sim_Streptococcus_alactolyticus_GCF_009695625\") has been submitted\n",
      "Your job 5481188 (\"sim_Streptococcus_anginosus_GCF_000463505\") has been submitted\n",
      "Your job 5481189 (\"sim_Streptococcus_australis_GCF_900476055\") has been submitted\n",
      "Your job 5481190 (\"sim_Streptococcus_azizii_GCF_001984715\") has been submitted\n",
      "Your job 5481191 (\"sim_Streptococcus_bovimastitidis_GCF_001885095\") has been submitted\n",
      "Your job 5481192 (\"sim_Streptococcus_caballi_GCF_000379985\") has been submitted\n",
      "Your job 5481193 (\"sim_Streptococcus_canis_GCF_900636575\") has been submitted\n",
      "Your job 5481194 (\"sim_Streptococcus_castoreus_GCF_000425025\") has been submitted\n",
      "Your job 5481195 (\"sim_Streptococcus_constellatus_GCF_000463425\") has been submitted\n",
      "Your job 5481196 (\"sim_Streptococcus_criceti_GCF_000187975\") has been submitted\n",
      "Your job 5481197 (\"sim_Streptococcus_cristatus_GCF_000385925\") has been submitted\n",
      "Your job 5481198 (\"sim_Streptococcus_cuniculi_GCF_001921845\") has been submitted\n",
      "Your job 5481199 (\"sim_Streptococcus_danieliae_GCF_009767945\") has been submitted\n",
      "Your job 5481200 (\"sim_Streptococcus_devriesei_GCF_000423725\") has been submitted\n",
      "Your job 5481201 (\"sim_Streptococcus_didelphis_GCF_000380005\") has been submitted\n",
      "Your job 5481202 (\"sim_Streptococcus_downei_GCF_900459175\") has been submitted\n",
      "Your job 5481203 (\"sim_Streptococcus_dysgalactiae_GCF_000317855\") has been submitted\n",
      "Your job 5481204 (\"sim_Streptococcus_entericus_GCF_000380025\") has been submitted\n",
      "Your job 5481205 (\"sim_Streptococcus_equi_GCF_000026605\") has been submitted\n",
      "Your job 5481206 (\"sim_Streptococcus_equinus_GCF_000964315\") has been submitted\n",
      "Your job 5481207 (\"sim_Streptococcus_ferus_GCF_000372425\") has been submitted\n",
      "Your job 5481208 (\"sim_Streptococcus_gallolyticus_GCF_002000985\") has been submitted\n",
      "Your job 5481209 (\"sim_Streptococcus_gordonii_GCF_000017005\") has been submitted\n",
      "Your job 5481210 (\"sim_Streptococcus_halichoeri_GCF_009870755\") has been submitted\n",
      "Your job 5481211 (\"sim_Streptococcus_halitosis_GCF_003143695\") has been submitted\n",
      "Your job 5481212 (\"sim_Streptococcus_halotolerans_GCF_001598035\") has been submitted\n",
      "Your job 5481213 (\"sim_Streptococcus_henryi_GCF_000376985\") has been submitted\n",
      "Your job 5481214 (\"sim_Streptococcus_himalayensis_GCF_001708305\") has been submitted\n",
      "Your job 5481215 (\"sim_Streptococcus_hongkongensis_GCF_000785845\") has been submitted\n",
      "Your job 5481216 (\"sim_Streptococcus_hyointestinalis_GCF_900459405\") has been submitted\n",
      "Your job 5481217 (\"sim_Streptococcus_hyovaginalis_GCF_000420785\") has been submitted\n",
      "Your job 5481218 (\"sim_Streptococcus_ictaluri_GCF_000188015\") has been submitted\n",
      "Your job 5481219 (\"sim_Streptococcus_infantarius_GCF_000246835\") has been submitted\n",
      "Your job 5481220 (\"sim_Streptococcus_infantis_GCF_000187465\") has been submitted\n",
      "Your job 5481221 (\"sim_Streptococcus_iniae_GCF_000831485\") has been submitted\n",
      "Your job 5481222 (\"sim_Streptococcus_intermedius_GCF_000463355\") has been submitted\n",
      "Your job 5481223 (\"sim_Streptococcus_lutetiensis_GCF_900475675\") has been submitted\n",
      "Your job 5481224 (\"sim_Streptococcus_macacae_GCF_000187995\") has been submitted\n",
      "Your job 5481225 (\"sim_Streptococcus_macedonicus_GCF_000283635\") has been submitted\n",
      "Your job 5481226 (\"sim_Streptococcus_marimammalium_GCF_000380045\") has been submitted\n",
      "Your job 5481227 (\"sim_Streptococcus_marmotae_GCF_001623565\") has been submitted\n",
      "Your job 5481228 (\"sim_Streptococcus_massiliensis_GCF_000380065\") has been submitted\n",
      "Your job 5481229 (\"sim_Streptococcus_merionis_GCF_000380085\") has been submitted\n",
      "Your job 5481230 (\"sim_Streptococcus_milleri_GCF_900636715\") has been submitted\n",
      "Your job 5481231 (\"sim_Streptococcus_minor_GCF_000377005\") has been submitted\n",
      "Your job 5481232 (\"sim_Streptococcus_mitis_GCF_000027165\") has been submitted\n",
      "Your job 5481233 (\"sim_Streptococcus_mutans_GCF_000007465\") has been submitted\n",
      "Your job 5481234 (\"sim_Streptococcus_oralis_GCF_900637025\") has been submitted\n",
      "Your job 5481235 (\"sim_Streptococcus_orisasini_GCF_001431045\") has been submitted\n",
      "Your job 5481236 (\"sim_Streptococcus_orisratti_GCF_000380105\") has been submitted\n",
      "Your job 5481237 (\"sim_Streptococcus_ovis_GCF_000380125\") has been submitted\n",
      "Your job 5481238 (\"sim_Streptococcus_pantholopis_GCF_001642085\") has been submitted\n",
      "Your job 5481239 (\"sim_Streptococcus_parasanguinis_GCF_000164675\") has been submitted\n",
      "Your job 5481240 (\"sim_Streptococcus_parasuis_GCF_004283785\") has been submitted\n",
      "Your job 5481241 (\"sim_Streptococcus_parauberis_GCF_000213825\") has been submitted\n",
      "Your job 5481242 (\"sim_Streptococcus_pasteurianus_GCF_900478025\") has been submitted\n",
      "Your job 5481243 (\"sim_Streptococcus_penaeicida_GCF_002887775\") has been submitted\n",
      "Your job 5481244 (\"sim_Streptococcus_peroris_GCF_000187585\") has been submitted\n",
      "Your job 5481245 (\"sim_Streptococcus_pharyngis_GCF_007859195\") has been submitted\n",
      "Your job 5481246 (\"sim_Streptococcus_phocae_GCF_000772915\") has been submitted\n",
      "Your job 5481247 (\"sim_Streptococcus_pluranimalium_GCF_003352995\") has been submitted\n",
      "Your job 5481248 (\"sim_Streptococcus_plurextorum_GCF_000423745\") has been submitted\n",
      "Your job 5481249 (\"sim_Streptococcus_porci_GCF_000423765\") has been submitted\n",
      "Your job 5481250 (\"sim_Streptococcus_porcinus_GCF_900475415\") has been submitted\n",
      "Your job 5481251 (\"sim_Streptococcus_pseudopneumoniae_GCF_000221985\") has been submitted\n",
      "Your job 5481252 (\"sim_Streptococcus_pseudoporcinus_GCF_000188035\") has been submitted\n",
      "Your job 5481253 (\"sim_Streptococcus_pyogenes_GCF_000006785\") has been submitted\n",
      "Your job 5481254 (\"sim_Streptococcus_ratti_GCF_000286075\") has been submitted\n",
      "Your job 5481255 (\"sim_Streptococcus_respraculi_GCF_003595525\") has been submitted\n",
      "Your job 5481256 (\"sim_Streptococcus_rubneri_GCF_004785935\") has been submitted\n",
      "Your job 5481257 (\"sim_Streptococcus_ruminantium_GCF_003609975\") has been submitted\n",
      "Your job 5481258 (\"sim_Streptococcus_salivarius_GCF_000785515\") has been submitted\n",
      "Your job 5481259 (\"sim_Streptococcus_sanguinis_GCF_000014205\") has been submitted\n",
      "Your job 5481260 (\"sim_Streptococcus_sinensis_GCF_000767835\") has been submitted\n",
      "Your job 5481261 (\"sim_Streptococcus_sobrinus_GCF_000686605\") has been submitted\n",
      "Your job 5481262 (\"sim_Streptococcus_sp_GCF_003086355\") has been submitted\n",
      "Your job 5481263 (\"sim_Streptococcus_suis_GCF_000026745\") has been submitted\n",
      "Your job 5481264 (\"sim_Streptococcus_thermophilus_GCF_000253395\") has been submitted\n",
      "Your job 5481265 (\"sim_Streptococcus_thoraltensis_GCF_000380145\") has been submitted\n",
      "Your job 5481266 (\"sim_Streptococcus_timonensis_GCF_900095845\") has been submitted\n",
      "Your job 5481267 (\"sim_Streptococcus_troglodytae_GCF_002355215\") has been submitted\n",
      "Your job 5481268 (\"sim_Streptococcus_uberis_GCF_000009545\") has been submitted\n",
      "Your job 5481269 (\"sim_Streptococcus_urinalis_GCF_000188055\") has been submitted\n",
      "Your job 5481270 (\"sim_Streptococcus_varani_GCF_001375655\") has been submitted\n",
      "Your job 5481271 (\"sim_Streptococcus_vestibularis_GCF_900636445\") has been submitted\n",
      "Your job 5481272 (\"sim_Streptococcus_viridans_GCF_900636365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/simulated_reads/logs\")\n",
    "\n",
    "for sample in group_taxa:\n",
    "    i_genome_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.fas\"\n",
    "    o_prefix      = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_\"\n",
    "        \n",
    "    art_cmd = str(\"art_illumina \"\n",
    "                      \"--paired \"\n",
    "                      \"--len 100 \"\n",
    "                      \"--fcov 10 \"\n",
    "                      \"--mflen 200 \"\n",
    "                      \"--sdev 150 \"\n",
    "                      \"-ir 0.0 \"\n",
    "                      \"-ir2 0.0 \"\n",
    "                      \"-dr 0.0 \"\n",
    "                      \"-dr2 0.0 \"\n",
    "                      \"-qs 100 \"\n",
    "                      \"-qs2 100 \"\n",
    "                      \"-na \"\n",
    "                      \"--in {} \"\n",
    "                      \"--out {};\").format(i_genome_file, o_prefix)\n",
    "\n",
    "    #now set up qsub\n",
    "    jid= \"sim_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/simulated_reads/logs/sim_\" + sample + \".log\"\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -pe smp 12\".format(jid, log)\n",
    "\n",
    "    #conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #and submit to the queue\n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + art_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before we can map the reads to the reference taxon genome we need to index it with `bbmap`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bbmap.sh ref=results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.fas path=results/phyluce/streptococcus/map_simulated_reads/\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/map_simulated_reads\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/map_simulated_reads\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/map_simulated_reads/logs\")\n",
    "\n",
    "#generate a bbmap index of the reference genome\n",
    "i_ref_genome = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.fas\"\n",
    "o_dir        = \"results/phyluce/\" + group + \"/map_simulated_reads/\"\n",
    "\n",
    "bbmap_cmd = \"bbmap.sh ref={} path={}\".format(i_ref_genome, o_dir)\n",
    "\n",
    "print(bbmap_cmd)\n",
    "subprocess.call(bbmap_cmd.split(\" \"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use bbmap to align reads from all species to the reference genome.  In this case we are allowing up to 15% divergence between the reference genome and simulated read.  Generally this number is smaller and may be adjusted for other groups, however, given the diversity of some groups a larger number may identify more orthologous regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481273 (\"map_Streptococcus_acidominimus_GCF_900187245\") has been submitted\n",
      "Your job 5481274 (\"map_Streptococcus_agalactiae_GCF_000007265\") has been submitted\n",
      "Your job 5481275 (\"map_Streptococcus_alactolyticus_GCF_009695625\") has been submitted\n",
      "Your job 5481276 (\"map_Streptococcus_anginosus_GCF_000463505\") has been submitted\n",
      "Your job 5481277 (\"map_Streptococcus_australis_GCF_900476055\") has been submitted\n",
      "Your job 5481278 (\"map_Streptococcus_azizii_GCF_001984715\") has been submitted\n",
      "Your job 5481279 (\"map_Streptococcus_bovimastitidis_GCF_001885095\") has been submitted\n",
      "Your job 5481280 (\"map_Streptococcus_caballi_GCF_000379985\") has been submitted\n",
      "Your job 5481281 (\"map_Streptococcus_canis_GCF_900636575\") has been submitted\n",
      "Your job 5481282 (\"map_Streptococcus_castoreus_GCF_000425025\") has been submitted\n",
      "Your job 5481283 (\"map_Streptococcus_constellatus_GCF_000463425\") has been submitted\n",
      "Your job 5481284 (\"map_Streptococcus_criceti_GCF_000187975\") has been submitted\n",
      "Your job 5481285 (\"map_Streptococcus_cristatus_GCF_000385925\") has been submitted\n",
      "Your job 5481286 (\"map_Streptococcus_cuniculi_GCF_001921845\") has been submitted\n",
      "Your job 5481287 (\"map_Streptococcus_danieliae_GCF_009767945\") has been submitted\n",
      "Your job 5481288 (\"map_Streptococcus_devriesei_GCF_000423725\") has been submitted\n",
      "Your job 5481289 (\"map_Streptococcus_didelphis_GCF_000380005\") has been submitted\n",
      "Your job 5481290 (\"map_Streptococcus_downei_GCF_900459175\") has been submitted\n",
      "Your job 5481291 (\"map_Streptococcus_dysgalactiae_GCF_000317855\") has been submitted\n",
      "Your job 5481292 (\"map_Streptococcus_entericus_GCF_000380025\") has been submitted\n",
      "Your job 5481293 (\"map_Streptococcus_equi_GCF_000026605\") has been submitted\n",
      "Your job 5481294 (\"map_Streptococcus_equinus_GCF_000964315\") has been submitted\n",
      "Your job 5481295 (\"map_Streptococcus_ferus_GCF_000372425\") has been submitted\n",
      "Your job 5481296 (\"map_Streptococcus_gallolyticus_GCF_002000985\") has been submitted\n",
      "Your job 5481297 (\"map_Streptococcus_gordonii_GCF_000017005\") has been submitted\n",
      "Your job 5481298 (\"map_Streptococcus_halichoeri_GCF_009870755\") has been submitted\n",
      "Your job 5481299 (\"map_Streptococcus_halitosis_GCF_003143695\") has been submitted\n",
      "Your job 5481300 (\"map_Streptococcus_halotolerans_GCF_001598035\") has been submitted\n",
      "Your job 5481301 (\"map_Streptococcus_henryi_GCF_000376985\") has been submitted\n",
      "Your job 5481302 (\"map_Streptococcus_himalayensis_GCF_001708305\") has been submitted\n",
      "Your job 5481303 (\"map_Streptococcus_hongkongensis_GCF_000785845\") has been submitted\n",
      "Your job 5481304 (\"map_Streptococcus_hyointestinalis_GCF_900459405\") has been submitted\n",
      "Your job 5481305 (\"map_Streptococcus_hyovaginalis_GCF_000420785\") has been submitted\n",
      "Your job 5481306 (\"map_Streptococcus_ictaluri_GCF_000188015\") has been submitted\n",
      "Your job 5481307 (\"map_Streptococcus_infantarius_GCF_000246835\") has been submitted\n",
      "Your job 5481308 (\"map_Streptococcus_infantis_GCF_000187465\") has been submitted\n",
      "Your job 5481309 (\"map_Streptococcus_iniae_GCF_000831485\") has been submitted\n",
      "Your job 5481310 (\"map_Streptococcus_intermedius_GCF_000463355\") has been submitted\n",
      "Your job 5481311 (\"map_Streptococcus_lutetiensis_GCF_900475675\") has been submitted\n",
      "Your job 5481312 (\"map_Streptococcus_macacae_GCF_000187995\") has been submitted\n",
      "Your job 5481313 (\"map_Streptococcus_macedonicus_GCF_000283635\") has been submitted\n",
      "Your job 5481314 (\"map_Streptococcus_marimammalium_GCF_000380045\") has been submitted\n",
      "Your job 5481315 (\"map_Streptococcus_marmotae_GCF_001623565\") has been submitted\n",
      "Your job 5481316 (\"map_Streptococcus_massiliensis_GCF_000380065\") has been submitted\n",
      "Your job 5481317 (\"map_Streptococcus_merionis_GCF_000380085\") has been submitted\n",
      "Your job 5481318 (\"map_Streptococcus_milleri_GCF_900636715\") has been submitted\n",
      "Your job 5481319 (\"map_Streptococcus_minor_GCF_000377005\") has been submitted\n",
      "Your job 5481320 (\"map_Streptococcus_mitis_GCF_000027165\") has been submitted\n",
      "Your job 5481321 (\"map_Streptococcus_mutans_GCF_000007465\") has been submitted\n",
      "Your job 5481322 (\"map_Streptococcus_oralis_GCF_900637025\") has been submitted\n",
      "Your job 5481323 (\"map_Streptococcus_orisasini_GCF_001431045\") has been submitted\n",
      "Your job 5481324 (\"map_Streptococcus_orisratti_GCF_000380105\") has been submitted\n",
      "Your job 5481325 (\"map_Streptococcus_ovis_GCF_000380125\") has been submitted\n",
      "Your job 5481326 (\"map_Streptococcus_pantholopis_GCF_001642085\") has been submitted\n",
      "Your job 5481327 (\"map_Streptococcus_parasanguinis_GCF_000164675\") has been submitted\n",
      "Your job 5481328 (\"map_Streptococcus_parasuis_GCF_004283785\") has been submitted\n",
      "Your job 5481329 (\"map_Streptococcus_parauberis_GCF_000213825\") has been submitted\n",
      "Your job 5481330 (\"map_Streptococcus_pasteurianus_GCF_900478025\") has been submitted\n",
      "Your job 5481331 (\"map_Streptococcus_penaeicida_GCF_002887775\") has been submitted\n",
      "Your job 5481332 (\"map_Streptococcus_peroris_GCF_000187585\") has been submitted\n",
      "Your job 5481333 (\"map_Streptococcus_pharyngis_GCF_007859195\") has been submitted\n",
      "Your job 5481334 (\"map_Streptococcus_phocae_GCF_000772915\") has been submitted\n",
      "Your job 5481335 (\"map_Streptococcus_pluranimalium_GCF_003352995\") has been submitted\n",
      "Your job 5481336 (\"map_Streptococcus_plurextorum_GCF_000423745\") has been submitted\n",
      "Your job 5481337 (\"map_Streptococcus_porci_GCF_000423765\") has been submitted\n",
      "Your job 5481338 (\"map_Streptococcus_porcinus_GCF_900475415\") has been submitted\n",
      "Your job 5481339 (\"map_Streptococcus_pseudopneumoniae_GCF_000221985\") has been submitted\n",
      "Your job 5481340 (\"map_Streptococcus_pseudoporcinus_GCF_000188035\") has been submitted\n",
      "Your job 5481341 (\"map_Streptococcus_pyogenes_GCF_000006785\") has been submitted\n",
      "Your job 5481342 (\"map_Streptococcus_ratti_GCF_000286075\") has been submitted\n",
      "Your job 5481343 (\"map_Streptococcus_respraculi_GCF_003595525\") has been submitted\n",
      "Your job 5481344 (\"map_Streptococcus_rubneri_GCF_004785935\") has been submitted\n",
      "Your job 5481345 (\"map_Streptococcus_ruminantium_GCF_003609975\") has been submitted\n",
      "Your job 5481346 (\"map_Streptococcus_salivarius_GCF_000785515\") has been submitted\n",
      "Your job 5481347 (\"map_Streptococcus_sanguinis_GCF_000014205\") has been submitted\n",
      "Your job 5481348 (\"map_Streptococcus_sinensis_GCF_000767835\") has been submitted\n",
      "Your job 5481349 (\"map_Streptococcus_sobrinus_GCF_000686605\") has been submitted\n",
      "Your job 5481350 (\"map_Streptococcus_sp_GCF_003086355\") has been submitted\n",
      "Your job 5481351 (\"map_Streptococcus_suis_GCF_000026745\") has been submitted\n",
      "Your job 5481352 (\"map_Streptococcus_thermophilus_GCF_000253395\") has been submitted\n",
      "Your job 5481353 (\"map_Streptococcus_thoraltensis_GCF_000380145\") has been submitted\n",
      "Your job 5481354 (\"map_Streptococcus_timonensis_GCF_900095845\") has been submitted\n",
      "Your job 5481355 (\"map_Streptococcus_troglodytae_GCF_002355215\") has been submitted\n",
      "Your job 5481356 (\"map_Streptococcus_uberis_GCF_000009545\") has been submitted\n",
      "Your job 5481357 (\"map_Streptococcus_urinalis_GCF_000188055\") has been submitted\n",
      "Your job 5481358 (\"map_Streptococcus_varani_GCF_001375655\") has been submitted\n",
      "Your job 5481359 (\"map_Streptococcus_vestibularis_GCF_900636445\") has been submitted\n",
      "Your job 5481360 (\"map_Streptococcus_viridans_GCF_900636365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "#map reads to reference genome   \n",
    "\n",
    "\"results/phyluce/\" + group + \"/map_simulated_reads/logs\"\n",
    "\n",
    "for sample in group_taxa:\n",
    "    #\n",
    "    #input and output files\n",
    "    i_sim_r1_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_1.fq\"\n",
    "    i_sim_r2_fas  = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \"_2.fq\"\n",
    "    i_ref_genome  = i_ref_genome\n",
    "    o_sim_r12_fas = \"results/phyluce/\" + group + \"/simulated_reads/\" + sample + \".fq\"\n",
    "    o_bbmap_sam   = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "\n",
    "    #merge r1 and r2 simulated reads\n",
    "    cat_cmd = \"cat {} {} >{}; \".format(i_sim_r1_fas, i_sim_r2_fas, o_sim_r12_fas)\n",
    "\n",
    "    #map with bbmap (via qsub)\n",
    "    bbmap_cmd = str(\"bbmap.sh t=12 \"\n",
    "                        \"in={} \"\n",
    "                        \"ref={} \"\n",
    "                        \"path=results/phyluce/\" + group + \"/map_simulated_reads/ \"\n",
    "                        \"out={} \"\n",
    "                        \"minid=0.90 \"\n",
    "                        \"mappedonly=t \"\n",
    "                        \"ignorebadquality=t; \").format(o_sim_r12_fas, \n",
    "                                                       i_ref_genome, \n",
    "                                                       o_bbmap_sam)\n",
    "\n",
    "    #set up qsub\n",
    "    jid= \"map_\" + sample\n",
    "    log= \"results/phyluce/\" + group + \"/map_simulated_reads/logs/map_\" + sample + \".log\"\n",
    "    hold = \"sim_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    #submit to the queue\n",
    "    combined_cmd ='echo \\\"' + conda_cmd + \" \" + cat_cmd + \"  \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "    #combined_cmd ='echo \\\"' + conda_cmd + \" \" + bbmap_cmd + '\\\" | ' + qsub_cmd\n",
    "\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use mapped reads to find orthologous loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bbmap` aligns all reads and stores the info in a `.sam` file.  Here we use a few filtering steps to remove unmapped reads, and then sort and merge mapped reads for each species/sample.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5481361 (\"merge_Streptococcus_acidominimus_GCF_900187245\") has been submitted\n",
      "Your job 5481362 (\"merge_Streptococcus_agalactiae_GCF_000007265\") has been submitted\n",
      "Your job 5481363 (\"merge_Streptococcus_alactolyticus_GCF_009695625\") has been submitted\n",
      "Your job 5481364 (\"merge_Streptococcus_anginosus_GCF_000463505\") has been submitted\n",
      "Your job 5481365 (\"merge_Streptococcus_australis_GCF_900476055\") has been submitted\n",
      "Your job 5481366 (\"merge_Streptococcus_azizii_GCF_001984715\") has been submitted\n",
      "Your job 5481367 (\"merge_Streptococcus_bovimastitidis_GCF_001885095\") has been submitted\n",
      "Your job 5481368 (\"merge_Streptococcus_caballi_GCF_000379985\") has been submitted\n",
      "Your job 5481369 (\"merge_Streptococcus_canis_GCF_900636575\") has been submitted\n",
      "Your job 5481370 (\"merge_Streptococcus_castoreus_GCF_000425025\") has been submitted\n",
      "Your job 5481371 (\"merge_Streptococcus_constellatus_GCF_000463425\") has been submitted\n",
      "Your job 5481372 (\"merge_Streptococcus_criceti_GCF_000187975\") has been submitted\n",
      "Your job 5481373 (\"merge_Streptococcus_cristatus_GCF_000385925\") has been submitted\n",
      "Your job 5481374 (\"merge_Streptococcus_cuniculi_GCF_001921845\") has been submitted\n",
      "Your job 5481375 (\"merge_Streptococcus_danieliae_GCF_009767945\") has been submitted\n",
      "Your job 5481376 (\"merge_Streptococcus_devriesei_GCF_000423725\") has been submitted\n",
      "Your job 5481377 (\"merge_Streptococcus_didelphis_GCF_000380005\") has been submitted\n",
      "Your job 5481378 (\"merge_Streptococcus_downei_GCF_900459175\") has been submitted\n",
      "Your job 5481379 (\"merge_Streptococcus_dysgalactiae_GCF_000317855\") has been submitted\n",
      "Your job 5481380 (\"merge_Streptococcus_entericus_GCF_000380025\") has been submitted\n",
      "Your job 5481381 (\"merge_Streptococcus_equi_GCF_000026605\") has been submitted\n",
      "Your job 5481382 (\"merge_Streptococcus_equinus_GCF_000964315\") has been submitted\n",
      "Your job 5481383 (\"merge_Streptococcus_ferus_GCF_000372425\") has been submitted\n",
      "Your job 5481384 (\"merge_Streptococcus_gallolyticus_GCF_002000985\") has been submitted\n",
      "Your job 5481385 (\"merge_Streptococcus_gordonii_GCF_000017005\") has been submitted\n",
      "Your job 5481386 (\"merge_Streptococcus_halichoeri_GCF_009870755\") has been submitted\n",
      "Your job 5481387 (\"merge_Streptococcus_halitosis_GCF_003143695\") has been submitted\n",
      "Your job 5481388 (\"merge_Streptococcus_halotolerans_GCF_001598035\") has been submitted\n",
      "Your job 5481389 (\"merge_Streptococcus_henryi_GCF_000376985\") has been submitted\n",
      "Your job 5481390 (\"merge_Streptococcus_himalayensis_GCF_001708305\") has been submitted\n",
      "Your job 5481391 (\"merge_Streptococcus_hongkongensis_GCF_000785845\") has been submitted\n",
      "Your job 5481392 (\"merge_Streptococcus_hyointestinalis_GCF_900459405\") has been submitted\n",
      "Your job 5481393 (\"merge_Streptococcus_hyovaginalis_GCF_000420785\") has been submitted\n",
      "Your job 5481394 (\"merge_Streptococcus_ictaluri_GCF_000188015\") has been submitted\n",
      "Your job 5481395 (\"merge_Streptococcus_infantarius_GCF_000246835\") has been submitted\n",
      "Your job 5481396 (\"merge_Streptococcus_infantis_GCF_000187465\") has been submitted\n",
      "Your job 5481397 (\"merge_Streptococcus_iniae_GCF_000831485\") has been submitted\n",
      "Your job 5481398 (\"merge_Streptococcus_intermedius_GCF_000463355\") has been submitted\n",
      "Your job 5481399 (\"merge_Streptococcus_lutetiensis_GCF_900475675\") has been submitted\n",
      "Your job 5481400 (\"merge_Streptococcus_macacae_GCF_000187995\") has been submitted\n",
      "Your job 5481401 (\"merge_Streptococcus_macedonicus_GCF_000283635\") has been submitted\n",
      "Your job 5481402 (\"merge_Streptococcus_marimammalium_GCF_000380045\") has been submitted\n",
      "Your job 5481403 (\"merge_Streptococcus_marmotae_GCF_001623565\") has been submitted\n",
      "Your job 5481404 (\"merge_Streptococcus_massiliensis_GCF_000380065\") has been submitted\n",
      "Your job 5481405 (\"merge_Streptococcus_merionis_GCF_000380085\") has been submitted\n",
      "Your job 5481406 (\"merge_Streptococcus_milleri_GCF_900636715\") has been submitted\n",
      "Your job 5481407 (\"merge_Streptococcus_minor_GCF_000377005\") has been submitted\n",
      "Your job 5481408 (\"merge_Streptococcus_mitis_GCF_000027165\") has been submitted\n",
      "Your job 5481409 (\"merge_Streptococcus_mutans_GCF_000007465\") has been submitted\n",
      "Your job 5481410 (\"merge_Streptococcus_oralis_GCF_900637025\") has been submitted\n",
      "Your job 5481411 (\"merge_Streptococcus_orisasini_GCF_001431045\") has been submitted\n",
      "Your job 5481412 (\"merge_Streptococcus_orisratti_GCF_000380105\") has been submitted\n",
      "Your job 5481413 (\"merge_Streptococcus_ovis_GCF_000380125\") has been submitted\n",
      "Your job 5481414 (\"merge_Streptococcus_pantholopis_GCF_001642085\") has been submitted\n",
      "Your job 5481415 (\"merge_Streptococcus_parasanguinis_GCF_000164675\") has been submitted\n",
      "Your job 5481416 (\"merge_Streptococcus_parasuis_GCF_004283785\") has been submitted\n",
      "Your job 5481417 (\"merge_Streptococcus_parauberis_GCF_000213825\") has been submitted\n",
      "Your job 5481418 (\"merge_Streptococcus_pasteurianus_GCF_900478025\") has been submitted\n",
      "Your job 5481419 (\"merge_Streptococcus_penaeicida_GCF_002887775\") has been submitted\n",
      "Your job 5481420 (\"merge_Streptococcus_peroris_GCF_000187585\") has been submitted\n",
      "Your job 5481421 (\"merge_Streptococcus_pharyngis_GCF_007859195\") has been submitted\n",
      "Your job 5481422 (\"merge_Streptococcus_phocae_GCF_000772915\") has been submitted\n",
      "Your job 5481423 (\"merge_Streptococcus_pluranimalium_GCF_003352995\") has been submitted\n",
      "Your job 5481424 (\"merge_Streptococcus_plurextorum_GCF_000423745\") has been submitted\n",
      "Your job 5481425 (\"merge_Streptococcus_porci_GCF_000423765\") has been submitted\n",
      "Your job 5481426 (\"merge_Streptococcus_porcinus_GCF_900475415\") has been submitted\n",
      "Your job 5481427 (\"merge_Streptococcus_pseudopneumoniae_GCF_000221985\") has been submitted\n",
      "Your job 5481428 (\"merge_Streptococcus_pseudoporcinus_GCF_000188035\") has been submitted\n",
      "Your job 5481429 (\"merge_Streptococcus_pyogenes_GCF_000006785\") has been submitted\n",
      "Your job 5481430 (\"merge_Streptococcus_ratti_GCF_000286075\") has been submitted\n",
      "Your job 5481431 (\"merge_Streptococcus_respraculi_GCF_003595525\") has been submitted\n",
      "Your job 5481432 (\"merge_Streptococcus_rubneri_GCF_004785935\") has been submitted\n",
      "Your job 5481433 (\"merge_Streptococcus_ruminantium_GCF_003609975\") has been submitted\n",
      "Your job 5481434 (\"merge_Streptococcus_salivarius_GCF_000785515\") has been submitted\n",
      "Your job 5481435 (\"merge_Streptococcus_sanguinis_GCF_000014205\") has been submitted\n",
      "Your job 5481436 (\"merge_Streptococcus_sinensis_GCF_000767835\") has been submitted\n",
      "Your job 5481437 (\"merge_Streptococcus_sobrinus_GCF_000686605\") has been submitted\n",
      "Your job 5481438 (\"merge_Streptococcus_sp_GCF_003086355\") has been submitted\n",
      "Your job 5481439 (\"merge_Streptococcus_suis_GCF_000026745\") has been submitted\n",
      "Your job 5481440 (\"merge_Streptococcus_thermophilus_GCF_000253395\") has been submitted\n",
      "Your job 5481441 (\"merge_Streptococcus_thoraltensis_GCF_000380145\") has been submitted\n",
      "Your job 5481442 (\"merge_Streptococcus_timonensis_GCF_900095845\") has been submitted\n",
      "Your job 5481443 (\"merge_Streptococcus_troglodytae_GCF_002355215\") has been submitted\n",
      "Your job 5481444 (\"merge_Streptococcus_uberis_GCF_000009545\") has been submitted\n",
      "Your job 5481445 (\"merge_Streptococcus_urinalis_GCF_000188055\") has been submitted\n",
      "Your job 5481446 (\"merge_Streptococcus_varani_GCF_001375655\") has been submitted\n",
      "Your job 5481447 (\"merge_Streptococcus_vestibularis_GCF_900636445\") has been submitted\n",
      "Your job 5481448 (\"merge_Streptococcus_viridans_GCF_900636365\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "if os.path.exists(\"results/phyluce/\" + group + \"/initial_intervals\"):\n",
    "    shutil.rmtree(\"results/phyluce/\" + group + \"/initial_intervals\")\n",
    "\n",
    "os.makedirs(\"results/phyluce/\" + group + \"/initial_intervals/logs\")\n",
    "\n",
    "#map reads to reference genome   \n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "    \n",
    "    #sam to (filtered) bam\n",
    "    i_sam = \"results/phyluce/\" + group + \"/map_simulated_reads/\" + sample + \".sam\"\n",
    "    o_filt_bam = o_dir + sample + \".bam\"\n",
    "    \n",
    "    sam_cmd = \"samtools view -b -F 4 {} >{};\".format(i_sam, o_filt_bam)\n",
    "  \n",
    "    #bam to bed\n",
    "    i_filt_bam = o_filt_bam\n",
    "    o_bed      = o_dir + sample + \".bed\"\n",
    "    \n",
    "    bed_to_bam_cmd = \"bedtools bamtobed -i {} >{};\".format(i_filt_bam, o_bed)\n",
    "    \n",
    "    #sort bed\n",
    "    i_bed        = o_bed\n",
    "    o_sorted_bed = o_dir + sample +  \"_sorted.bed\"\n",
    "    \n",
    "    sort_bed_cmd = \"bedtools sort -i {} >{};\".format(i_bed, o_sorted_bed)\n",
    "    \n",
    "    #merge bed\n",
    "    i_sorted_bed   = o_sorted_bed\n",
    "    o_interval_bed = o_dir + sample +  \"_merged.bed\"\n",
    "    \n",
    "    merge_cmd = \"bedtools merge -i {} >{};\".format(i_sorted_bed, o_interval_bed)\n",
    "    \n",
    "    #set up qsub\n",
    "    jid= \"merge_\" + sample\n",
    "    log= o_dir + \"logs/merge_\" + sample + \".log\"\n",
    "    hold = \"map_\" + sample\n",
    "    qsub_cmd = \"qsub -V -cwd -S /bin/bash -q all.q -j y -N {} -o {} -hold_jid {} -pe smp 12\".format(jid, \n",
    "                                                                                                    log, \n",
    "                                                                                                    hold)\n",
    "\n",
    "    #and conda\n",
    "    conda_cmd = \"conda activate pathogen_probes-phyluce; \"\n",
    "\n",
    "    bed_cmd = sam_cmd + bed_to_bam_cmd + sort_bed_cmd + merge_cmd\n",
    "    \n",
    "    combined_cmd =\"echo \\\"\" + conda_cmd + bed_cmd + \"\\\" | \" + qsub_cmd\n",
    "    \n",
    "    #print(combined_cmd)\n",
    "    !{combined_cmd}\n",
    "    \n",
    "#################################################################################################\n",
    "#  __          __     _____ _______   ______ ____  _____     ____  _    _ ______ _    _ ______  #\n",
    "#  \\ \\        / /\\   |_   _|__   __| |  ____/ __ \\|  __ \\   / __ \\| |  | |  ____| |  | |  ____| #\n",
    "#   \\ \\  /\\  / /  \\    | |    | |    | |__ | |  | | |__) | | |  | | |  | | |__  | |  | | |__    #\n",
    "#    \\ \\/  \\/ / /\\ \\   | |    | |    |  __|| |  | |  _  /  | |  | | |  | |  __| | |  | |  __|   #\n",
    "#     \\  /\\  / ____ \\ _| |_   | |    | |   | |__| | | \\ \\  | |__| | |__| | |____| |__| | |____  #\n",
    "#      \\/  \\/_/    \\_\\_____|  |_|    |_|    \\____/|_|  \\_\\  \\___\\_\\\\____/|______|\\____/|______| #\n",
    "#                                                                                               #\n",
    "#################################################################################################\n",
    "#wait_on_running_jobs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 205 sequences from Streptococcus_acidominimus_GCF_900187245_merged.bed.  Filtered 117 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 88.\n",
      "Screened 203 sequences from Streptococcus_agalactiae_GCF_000007265_merged.bed.  Filtered 114 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 89.\n",
      "Screened 211 sequences from Streptococcus_alactolyticus_GCF_009695625_merged.bed.  Filtered 113 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 98.\n",
      "Screened 275 sequences from Streptococcus_anginosus_GCF_000463505_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 578 sequences from Streptococcus_australis_GCF_900476055_merged.bed.  Filtered 298 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 280.\n",
      "Screened 162 sequences from Streptococcus_azizii_GCF_001984715_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 74.\n",
      "Screened 174 sequences from Streptococcus_bovimastitidis_GCF_001885095_merged.bed.  Filtered 98 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 76.\n",
      "Screened 220 sequences from Streptococcus_caballi_GCF_000379985_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 100.\n",
      "Screened 193 sequences from Streptococcus_canis_GCF_900636575_merged.bed.  Filtered 102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 91.\n",
      "Screened 168 sequences from Streptococcus_castoreus_GCF_000425025_merged.bed.  Filtered 99 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 285 sequences from Streptococcus_constellatus_GCF_000463425_merged.bed.  Filtered 167 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 118.\n",
      "Screened 134 sequences from Streptococcus_criceti_GCF_000187975_merged.bed.  Filtered 86 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 48.\n",
      "Screened 497 sequences from Streptococcus_cristatus_GCF_000385925_merged.bed.  Filtered 241 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 256.\n",
      "Screened 217 sequences from Streptococcus_cuniculi_GCF_001921845_merged.bed.  Filtered 123 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 178 sequences from Streptococcus_danieliae_GCF_009767945_merged.bed.  Filtered 99 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 111 sequences from Streptococcus_devriesei_GCF_000423725_merged.bed.  Filtered 62 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 49.\n",
      "Screened 147 sequences from Streptococcus_didelphis_GCF_000380005_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 59.\n",
      "Screened 152 sequences from Streptococcus_downei_GCF_900459175_merged.bed.  Filtered 95 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 57.\n",
      "Screened 223 sequences from Streptococcus_dysgalactiae_GCF_000317855_merged.bed.  Filtered 129 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 134 sequences from Streptococcus_entericus_GCF_000380025_merged.bed.  Filtered 83 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 51.\n",
      "Screened 155 sequences from Streptococcus_equi_GCF_000026605_merged.bed.  Filtered 82 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 73.\n",
      "Screened 235 sequences from Streptococcus_equinus_GCF_000964315_merged.bed.  Filtered 125 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 135 sequences from Streptococcus_ferus_GCF_000372425_merged.bed.  Filtered 89 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 46.\n",
      "Screened 236 sequences from Streptococcus_gallolyticus_GCF_002000985_merged.bed.  Filtered 123 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 435 sequences from Streptococcus_gordonii_GCF_000017005_merged.bed.  Filtered 220 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 215.\n",
      "Screened 132 sequences from Streptococcus_halichoeri_GCF_009870755_merged.bed.  Filtered 80 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 52.\n",
      "Screened 2296 sequences from Streptococcus_halitosis_GCF_003143695_merged.bed.  Filtered 860 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1436.\n",
      "Screened 170 sequences from Streptococcus_halotolerans_GCF_001598035_merged.bed.  Filtered 85 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 270 sequences from Streptococcus_henryi_GCF_000376985_merged.bed.  Filtered 168 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 102.\n",
      "Screened 249 sequences from Streptococcus_himalayensis_GCF_001708305_merged.bed.  Filtered 140 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 109.\n",
      "Screened 163 sequences from Streptococcus_hongkongensis_GCF_000785845_merged.bed.  Filtered 94 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 200 sequences from Streptococcus_hyointestinalis_GCF_900459405_merged.bed.  Filtered 102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 98.\n",
      "Screened 182 sequences from Streptococcus_hyovaginalis_GCF_000420785_merged.bed.  Filtered 103 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 165 sequences from Streptococcus_ictaluri_GCF_000188015_merged.bed.  Filtered 86 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 242 sequences from Streptococcus_infantarius_GCF_000246835_merged.bed.  Filtered 137 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 105.\n",
      "Screened 1375 sequences from Streptococcus_infantis_GCF_000187465_merged.bed.  Filtered 707 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 668.\n",
      "Screened 169 sequences from Streptococcus_iniae_GCF_000831485_merged.bed.  Filtered 90 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 79.\n",
      "Screened 253 sequences from Streptococcus_intermedius_GCF_000463355_merged.bed.  Filtered 133 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 120.\n",
      "Screened 240 sequences from Streptococcus_lutetiensis_GCF_900475675_merged.bed.  Filtered 129 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 111.\n",
      "Screened 98 sequences from Streptococcus_macacae_GCF_000187995_merged.bed.  Filtered 55 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 43.\n",
      "Screened 238 sequences from Streptococcus_macedonicus_GCF_000283635_merged.bed.  Filtered 125 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 113.\n",
      "Screened 98 sequences from Streptococcus_marimammalium_GCF_000380045_merged.bed.  Filtered 54 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 44.\n",
      "Screened 217 sequences from Streptococcus_marmotae_GCF_001623565_merged.bed.  Filtered 116 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 101.\n",
      "Screened 225 sequences from Streptococcus_massiliensis_GCF_000380065_merged.bed.  Filtered 115 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 110.\n",
      "Screened 178 sequences from Streptococcus_merionis_GCF_000380085_merged.bed.  Filtered 109 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 291 sequences from Streptococcus_milleri_GCF_900636715_merged.bed.  Filtered 160 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 166 sequences from Streptococcus_minor_GCF_000377005_merged.bed.  Filtered 101 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 65.\n",
      "Screened 1639 sequences from Streptococcus_mitis_GCF_000027165_merged.bed.  Filtered 304 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1335.\n",
      "Screened 159 sequences from Streptococcus_mutans_GCF_000007465_merged.bed.  Filtered 98 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 61.\n",
      "Screened 2276 sequences from Streptococcus_oralis_GCF_900637025_merged.bed.  Filtered 812 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1464.\n",
      "Screened 161 sequences from Streptococcus_orisasini_GCF_001431045_merged.bed.  Filtered 100 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 61.\n",
      "Screened 221 sequences from Streptococcus_orisratti_GCF_000380105_merged.bed.  Filtered 131 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 90.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Screened 233 sequences from Streptococcus_ovis_GCF_000380125_merged.bed.  Filtered 127 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 106.\n",
      "Screened 70 sequences from Streptococcus_pantholopis_GCF_001642085_merged.bed.  Filtered 39 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 31.\n",
      "Screened 505 sequences from Streptococcus_parasanguinis_GCF_000164675_merged.bed.  Filtered 260 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 245.\n",
      "Screened 245 sequences from Streptococcus_parasuis_GCF_004283785_merged.bed.  Filtered 138 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 107.\n",
      "Screened 157 sequences from Streptococcus_parauberis_GCF_000213825_merged.bed.  Filtered 88 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 245 sequences from Streptococcus_pasteurianus_GCF_900478025_merged.bed.  Filtered 130 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 115.\n",
      "Screened 186 sequences from Streptococcus_penaeicida_GCF_002887775_merged.bed.  Filtered 111 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 1261 sequences from Streptococcus_peroris_GCF_000187585_merged.bed.  Filtered 645 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 616.\n",
      "Screened 160 sequences from Streptococcus_pharyngis_GCF_007859195_merged.bed.  Filtered 91 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 145 sequences from Streptococcus_phocae_GCF_000772915_merged.bed.  Filtered 73 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 72.\n",
      "Screened 211 sequences from Streptococcus_pluranimalium_GCF_003352995_merged.bed.  Filtered 112 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 99.\n",
      "Screened 208 sequences from Streptococcus_plurextorum_GCF_000423745_merged.bed.  Filtered 128 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 80.\n",
      "Screened 219 sequences from Streptococcus_porci_GCF_000423765_merged.bed.  Filtered 134 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 143 sequences from Streptococcus_porcinus_GCF_900475415_merged.bed.  Filtered 74 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 69.\n",
      "Screened 1299 sequences from Streptococcus_pseudopneumoniae_GCF_000221985_merged.bed.  Filtered 225 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 1074.\n",
      "Screened 160 sequences from Streptococcus_pseudoporcinus_GCF_000188035_merged.bed.  Filtered 92 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 68.\n",
      "Screened 213 sequences from Streptococcus_pyogenes_GCF_000006785_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 93.\n",
      "Screened 148 sequences from Streptococcus_ratti_GCF_000286075_merged.bed.  Filtered 86 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 62.\n",
      "Screened 224 sequences from Streptococcus_respraculi_GCF_003595525_merged.bed.  Filtered 130 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 94.\n",
      "Screened 581 sequences from Streptococcus_rubneri_GCF_004785935_merged.bed.  Filtered 296 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 285.\n",
      "Screened 197 sequences from Streptococcus_ruminantium_GCF_003609975_merged.bed.  Filtered 120 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 77.\n",
      "Screened 313 sequences from Streptococcus_salivarius_GCF_000785515_merged.bed.  Filtered 154 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 159.\n",
      "Screened 363 sequences from Streptococcus_sanguinis_GCF_000014205_merged.bed.  Filtered 199 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 164.\n",
      "Screened 434 sequences from Streptococcus_sinensis_GCF_000767835_merged.bed.  Filtered 236 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 198.\n",
      "Screened 124 sequences from Streptococcus_sobrinus_GCF_000686605_merged.bed.  Filtered 74 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 50.\n",
      "Screened 77 sequences from Streptococcus_sp_GCF_003086355_merged.bed.  Filtered 43 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 34.\n",
      "Screened 252 sequences from Streptococcus_suis_GCF_000026745_merged.bed.  Filtered 135 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 117.\n",
      "Screened 289 sequences from Streptococcus_thermophilus_GCF_000253395_merged.bed.  Filtered 158 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 131.\n",
      "Screened 187 sequences from Streptococcus_thoraltensis_GCF_000380145_merged.bed.  Filtered 102 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 85.\n",
      "Screened 1252 sequences from Streptococcus_timonensis_GCF_900095845_merged.bed.  Filtered 644 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 608.\n",
      "Screened 143 sequences from Streptococcus_troglodytae_GCF_002355215_merged.bed.  Filtered 87 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 56.\n",
      "Screened 158 sequences from Streptococcus_uberis_GCF_000009545_merged.bed.  Filtered 80 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 78.\n",
      "Screened 166 sequences from Streptococcus_urinalis_GCF_000188055_merged.bed.  Filtered 91 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 75.\n",
      "Screened 211 sequences from Streptococcus_varani_GCF_001375655_merged.bed.  Filtered 133 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 78.\n",
      "Screened 346 sequences from Streptococcus_vestibularis_GCF_900636445_merged.bed.  Filtered 181 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 165.\n",
      "Screened 510 sequences from Streptococcus_viridans_GCF_900636365_merged.bed.  Filtered 254 with > 25.0% masked bases or > 0 N-bases or < 160 length. Kept 256.\n"
     ]
    }
   ],
   "source": [
    "#strip repeats\n",
    "for sample in group_taxa:\n",
    "    \n",
    "    o_dir = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "    #input and output files\n",
    "    i_merged_bed   = o_dir + sample + \"_merged.bed\"\n",
    "    i_genome_2bit  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "    o_stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "    #repeat strip w/ phyluce\n",
    "    phyluce_cmd = str(\"phyluce_probe_strip_masked_loci_from_set \"\n",
    "                          \"--filter-mask 0.25 \"\n",
    "                          \"--min-length 160 \"\n",
    "                          \"--bed {} \"\n",
    "                          \"--twobit {} \" \n",
    "                          \"--output {};\").format(i_merged_bed,\n",
    "                                                 i_genome_2bit,\n",
    "                                                 o_stripped_bed)\n",
    "    !{phyluce_cmd}\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find conserved loci across taxa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "We used phyluce to find regions that are consistently mapped (\"conserved\") across various taxa.\n",
    "\n",
    "First, make ```conf``` file that indicates where the stripped bedfiles (from above) are stored. \n",
    "\n",
    "```\n",
    "[beds]\n",
    "agrPla1:agrPla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "anoGla1:anoGla1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "denPon1:denPon1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "lepDec1:lepDec1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "ontTau1:ontTau1-to-triCas1-MAPPING.bam.sort.merge.strip.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "o_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "with open(o_bed_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[beds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in group_taxa:\n",
    "        stripped_bed = o_dir + sample + \"_stripped.bed\"\n",
    "\n",
    "        outfile.write(sample + \":\" + stripped_bed + \"\\n\")     \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Then merge all of the data into a single ```sqlite``` db."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_merged.bed --twobit results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.2bit --output results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_stripped.bed;\n",
      "streptococcus_acidominimus_gcf_900187245.\n",
      "streptococcus_agalactiae_gcf_000007265.\n",
      "streptococcus_alactolyticus_gcf_009695625.\n",
      "streptococcus_anginosus_gcf_000463505.\n",
      "streptococcus_australis_gcf_900476055.\n",
      "streptococcus_azizii_gcf_001984715.\n",
      "streptococcus_bovimastitidis_gcf_001885095.\n",
      "streptococcus_caballi_gcf_000379985.\n",
      "streptococcus_canis_gcf_900636575.\n",
      "streptococcus_castoreus_gcf_000425025.\n",
      "streptococcus_constellatus_gcf_000463425.\n",
      "streptococcus_criceti_gcf_000187975.\n",
      "streptococcus_cristatus_gcf_000385925.\n",
      "streptococcus_cuniculi_gcf_001921845.\n",
      "streptococcus_danieliae_gcf_009767945.\n",
      "streptococcus_devriesei_gcf_000423725.\n",
      "streptococcus_didelphis_gcf_000380005.\n",
      "streptococcus_downei_gcf_900459175.\n",
      "streptococcus_dysgalactiae_gcf_000317855.\n",
      "streptococcus_entericus_gcf_000380025.\n",
      "streptococcus_equi_gcf_000026605.\n",
      "streptococcus_equinus_gcf_000964315.\n",
      "streptococcus_ferus_gcf_000372425.\n",
      "streptococcus_gallolyticus_gcf_002000985.\n",
      "streptococcus_gordonii_gcf_000017005.\n",
      "streptococcus_halichoeri_gcf_009870755.\n",
      "streptococcus_halitosis_gcf_003143695..\n",
      "streptococcus_halotolerans_gcf_001598035.\n",
      "streptococcus_henryi_gcf_000376985.\n",
      "streptococcus_himalayensis_gcf_001708305.\n",
      "streptococcus_hongkongensis_gcf_000785845.\n",
      "streptococcus_hyointestinalis_gcf_900459405.\n",
      "streptococcus_hyovaginalis_gcf_000420785.\n",
      "streptococcus_ictaluri_gcf_000188015.\n",
      "streptococcus_infantarius_gcf_000246835.\n",
      "streptococcus_infantis_gcf_000187465.\n",
      "streptococcus_iniae_gcf_000831485.\n",
      "streptococcus_intermedius_gcf_000463355.\n",
      "streptococcus_lutetiensis_gcf_900475675.\n",
      "streptococcus_macacae_gcf_000187995.\n",
      "streptococcus_macedonicus_gcf_000283635.\n",
      "streptococcus_marimammalium_gcf_000380045.\n",
      "streptococcus_marmotae_gcf_001623565.\n",
      "streptococcus_massiliensis_gcf_000380065.\n",
      "streptococcus_merionis_gcf_000380085.\n",
      "streptococcus_milleri_gcf_900636715.\n",
      "streptococcus_minor_gcf_000377005.\n",
      "streptococcus_mitis_gcf_000027165..\n",
      "streptococcus_mutans_gcf_000007465.\n",
      "streptococcus_oralis_gcf_900637025..\n",
      "streptococcus_orisasini_gcf_001431045.\n",
      "streptococcus_orisratti_gcf_000380105.\n",
      "streptococcus_ovis_gcf_000380125.\n",
      "streptococcus_pantholopis_gcf_001642085.\n",
      "streptococcus_parasanguinis_gcf_000164675.\n",
      "streptococcus_parasuis_gcf_004283785.\n",
      "streptococcus_parauberis_gcf_000213825.\n",
      "streptococcus_pasteurianus_gcf_900478025.\n",
      "streptococcus_penaeicida_gcf_002887775.\n",
      "streptococcus_peroris_gcf_000187585.\n",
      "streptococcus_pharyngis_gcf_007859195.\n",
      "streptococcus_phocae_gcf_000772915.\n",
      "streptococcus_pluranimalium_gcf_003352995.\n",
      "streptococcus_plurextorum_gcf_000423745.\n",
      "streptococcus_porci_gcf_000423765.\n",
      "streptococcus_porcinus_gcf_900475415.\n",
      "streptococcus_pseudopneumoniae_gcf_000221985..\n",
      "streptococcus_pseudoporcinus_gcf_000188035.\n",
      "streptococcus_pyogenes_gcf_000006785.\n",
      "streptococcus_ratti_gcf_000286075.\n",
      "streptococcus_respraculi_gcf_003595525.\n",
      "streptococcus_rubneri_gcf_004785935.\n",
      "streptococcus_ruminantium_gcf_003609975.\n",
      "streptococcus_salivarius_gcf_000785515.\n",
      "streptococcus_sanguinis_gcf_000014205.\n",
      "streptococcus_sinensis_gcf_000767835.\n",
      "streptococcus_sobrinus_gcf_000686605.\n",
      "streptococcus_sp_gcf_003086355.\n",
      "streptococcus_suis_gcf_000026745.\n",
      "streptococcus_thermophilus_gcf_000253395.\n",
      "streptococcus_thoraltensis_gcf_000380145.\n",
      "streptococcus_timonensis_gcf_900095845.\n",
      "streptococcus_troglodytae_gcf_002355215.\n",
      "streptococcus_uberis_gcf_000009545.\n",
      "streptococcus_urinalis_gcf_000188055.\n",
      "streptococcus_varani_gcf_001375655.\n",
      "streptococcus_vestibularis_gcf_900636445.\n",
      "streptococcus_viridans_gcf_900636365.\n",
      "Creating database\n",
      "Inserting results\n"
     ]
    }
   ],
   "source": [
    "#make output file\n",
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "i_bed_conf = o_dir + group + \"_bed-files.conf\"\n",
    "\n",
    "#now run phyluce to get table of orthologous sequeneces across the genus\n",
    "o_sqlite_db = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "\n",
    "phyluce_merge_cmd=str(\"phyluce_probe_get_multi_merge_table \"\n",
    "                          \"--conf {} \"\n",
    "                          \"--base-taxon {} \"\n",
    "                          \"--output {};\").format(o_bed_conf,\n",
    "                                                 reference_taxon,\n",
    "                                                 o_sqlite_db)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_merge_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Quantify probes and the number of targeted taxa for each."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_strip_masked_loci_from_set --filter-mask 0.25 --min-length 160 --bed results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_merged.bed --twobit results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.2bit --output results/phyluce/streptococcus/initial_intervals/Streptococcus_viridans_GCF_900636365_stripped.bed;\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 0 taxa:\t2,656.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 1 taxa:\t2,656.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 2 taxa:\t2,359.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 3 taxa:\t1,812.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 4 taxa:\t1,506.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 5 taxa:\t908.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 6 taxa:\t723.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 7 taxa:\t583.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 8 taxa:\t424.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 9 taxa:\t373.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 10 taxa:\t336.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 11 taxa:\t292.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 12 taxa:\t264.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 13 taxa:\t242.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 14 taxa:\t224.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 15 taxa:\t213.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 16 taxa:\t203.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 17 taxa:\t196.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 18 taxa:\t190.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 19 taxa:\t177.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 20 taxa:\t168.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 21 taxa:\t166.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 22 taxa:\t161.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 23 taxa:\t154.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 24 taxa:\t149.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 25 taxa:\t146.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 26 taxa:\t142.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 27 taxa:\t137.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 28 taxa:\t135.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 29 taxa:\t132.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 30 taxa:\t127.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 31 taxa:\t127.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 32 taxa:\t125.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 33 taxa:\t123.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 34 taxa:\t120.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 35 taxa:\t115.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 36 taxa:\t114.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 37 taxa:\t113.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 38 taxa:\t110.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 39 taxa:\t108.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 40 taxa:\t106.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 41 taxa:\t100.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 42 taxa:\t99.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 43 taxa:\t97.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 44 taxa:\t92.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 45 taxa:\t90.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 46 taxa:\t89.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 47 taxa:\t88.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 48 taxa:\t84.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 49 taxa:\t82.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 50 taxa:\t82.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 51 taxa:\t78.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 52 taxa:\t76.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 53 taxa:\t73.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 54 taxa:\t72.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 55 taxa:\t69.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 56 taxa:\t69.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 57 taxa:\t65.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 58 taxa:\t63.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 59 taxa:\t61.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 60 taxa:\t59.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 61 taxa:\t57.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 62 taxa:\t55.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 63 taxa:\t54.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 64 taxa:\t53.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 65 taxa:\t51.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 66 taxa:\t51.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 67 taxa:\t49.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 68 taxa:\t48.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 69 taxa:\t44.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 70 taxa:\t42.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 71 taxa:\t40.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 72 taxa:\t39.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 73 taxa:\t36.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 74 taxa:\t34.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 75 taxa:\t31.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 76 taxa:\t31.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 77 taxa:\t30.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 78 taxa:\t27.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 79 taxa:\t24.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 80 taxa:\t22.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 81 taxa:\t18.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 82 taxa:\t17.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 83 taxa:\t17.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 84 taxa:\t16.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 85 taxa:\t14.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 86 taxa:\t9.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 87 taxa:\t6.0\n",
      "Loci shared by Streptococcus_pneumoniae_GCF_000007045 + 88 taxa:\t3.0\n"
     ]
    }
   ],
   "source": [
    "#get genus and species\n",
    "i_sqlite_db = o_sqlite_db\n",
    "\n",
    "phyluce_query_cmd = \"phyluce_probe_query_multi_merge_table --db {} --base-taxon {}\".format(i_sqlite_db,\n",
    "                                                                                           reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_query_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_merge_table --db results/phyluce/streptococcus/initial_intervals/streptococcus-to-Streptococcus_pneumoniae_GCF_000007045.sqlite --base-taxon Streptococcus_pneumoniae_GCF_000007045 --output results/phyluce/streptococcus/initial_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.bed --specific-counts 41;\n",
      "Counter({'streptococcus_oralis_gcf_900637025': 100, 'streptococcus_pseudopneumoniae_gcf_000221985': 100, 'streptococcus_mitis_gcf_000027165': 100, 'streptococcus_timonensis_gcf_900095845': 99, 'streptococcus_halitosis_gcf_003143695': 99, 'streptococcus_parasanguinis_gcf_000164675': 99, 'streptococcus_infantis_gcf_000187465': 98, 'streptococcus_gordonii_gcf_000017005': 98, 'streptococcus_viridans_gcf_900636365': 97, 'streptococcus_cristatus_gcf_000385925': 96, 'streptococcus_salivarius_gcf_000785515': 95, 'streptococcus_peroris_gcf_000187585': 95, 'streptococcus_gallolyticus_gcf_002000985': 94, 'streptococcus_infantarius_gcf_000246835': 94, 'streptococcus_sanguinis_gcf_000014205': 94, 'streptococcus_australis_gcf_900476055': 93, 'streptococcus_rubneri_gcf_004785935': 93, 'streptococcus_sinensis_gcf_000767835': 92, 'streptococcus_thermophilus_gcf_000253395': 92, 'streptococcus_vestibularis_gcf_900636445': 91, 'streptococcus_macedonicus_gcf_000283635': 91, 'streptococcus_himalayensis_gcf_001708305': 89, 'streptococcus_equinus_gcf_000964315': 88, 'streptococcus_suis_gcf_000026745': 88, 'streptococcus_marmotae_gcf_001623565': 87, 'streptococcus_milleri_gcf_900636715': 86, 'streptococcus_pasteurianus_gcf_900478025': 86, 'streptococcus_lutetiensis_gcf_900475675': 86, 'streptococcus_anginosus_gcf_000463505': 85, 'streptococcus_constellatus_gcf_000463425': 85, 'streptococcus_parasuis_gcf_004283785': 84, 'streptococcus_pyogenes_gcf_000006785': 83, 'streptococcus_ovis_gcf_000380125': 82, 'streptococcus_intermedius_gcf_000463355': 82, 'streptococcus_dysgalactiae_gcf_000317855': 82, 'streptococcus_pluranimalium_gcf_003352995': 82, 'streptococcus_canis_gcf_900636575': 82, 'streptococcus_respraculi_gcf_003595525': 81, 'streptococcus_hyointestinalis_gcf_900459405': 80, 'streptococcus_henryi_gcf_000376985': 80, 'streptococcus_cuniculi_gcf_001921845': 79, 'streptococcus_caballi_gcf_000379985': 78, 'streptococcus_ictaluri_gcf_000188015': 77, 'streptococcus_massiliensis_gcf_000380065': 76, 'streptococcus_orisratti_gcf_000380105': 76, 'streptococcus_plurextorum_gcf_000423745': 75, 'streptococcus_porci_gcf_000423765': 74, 'streptococcus_uberis_gcf_000009545': 72, 'streptococcus_iniae_gcf_000831485': 72, 'streptococcus_ruminantium_gcf_003609975': 71, 'streptococcus_alactolyticus_gcf_009695625': 71, 'streptococcus_halotolerans_gcf_001598035': 71, 'streptococcus_agalactiae_gcf_000007265': 70, 'streptococcus_acidominimus_gcf_900187245': 70, 'streptococcus_urinalis_gcf_000188055': 68, 'streptococcus_hyovaginalis_gcf_000420785': 68, 'streptococcus_varani_gcf_001375655': 67, 'streptococcus_phocae_gcf_000772915': 67, 'streptococcus_penaeicida_gcf_002887775': 67, 'streptococcus_pharyngis_gcf_007859195': 66, 'streptococcus_thoraltensis_gcf_000380145': 65, 'streptococcus_bovimastitidis_gcf_001885095': 65, 'streptococcus_danieliae_gcf_009767945': 64, 'streptococcus_azizii_gcf_001984715': 63, 'streptococcus_castoreus_gcf_000425025': 63, 'streptococcus_parauberis_gcf_000213825': 61, 'streptococcus_pseudoporcinus_gcf_000188035': 61, 'streptococcus_hongkongensis_gcf_000785845': 61, 'streptococcus_porcinus_gcf_900475415': 61, 'streptococcus_equi_gcf_000026605': 60, 'streptococcus_minor_gcf_000377005': 60, 'streptococcus_merionis_gcf_000380085': 59, 'streptococcus_didelphis_gcf_000380005': 58, 'streptococcus_orisasini_gcf_001431045': 56, 'streptococcus_sobrinus_gcf_000686605': 54, 'streptococcus_downei_gcf_900459175': 52, 'streptococcus_mutans_gcf_000007465': 51, 'streptococcus_devriesei_gcf_000423725': 51, 'streptococcus_troglodytae_gcf_002355215': 48, 'streptococcus_ratti_gcf_000286075': 48, 'streptococcus_halichoeri_gcf_009870755': 47, 'streptococcus_entericus_gcf_000380025': 46, 'streptococcus_ferus_gcf_000372425': 45, 'streptococcus_criceti_gcf_000187975': 43, 'streptococcus_marimammalium_gcf_000380045': 41, 'streptococcus_macacae_gcf_000187995': 33, 'streptococcus_sp_gcf_003086355': 25, 'streptococcus_pantholopis_gcf_001642085': 22})\n"
     ]
    }
   ],
   "source": [
    "o_dir      = \"results/phyluce/\" + group + \"/initial_intervals/\"\n",
    "\n",
    "#set up input and output files\n",
    "num_taxa = 41\n",
    "i_sqlite_db = o_sqlite_db\n",
    "out_bed = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_merge_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db,\n",
    "                                                      reference_taxon,\n",
    "                                                      out_bed,\n",
    "                                                      num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Design temp set of baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_genome_sequences_from_bed --bed results/phyluce/streptococcus/initial_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.bed --twobit results/phyluce/streptococcus/cleaned_genomes/Streptococcus_pneumoniae_GCF_000007045_formatted.2bit --buffer-to 160 --output results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.fasta;\n",
      "Screened 100 sequences.  Filtered 0 < 160 bp or with > 25.0% masked bases or > 0 N-bases. Kept 100.\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/validate_intervals/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.makedirs(o_dir)\n",
    "\n",
    "\n",
    "i_2bit_genome  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + reference_taxon + \"_formatted.2bit\"\n",
    "i_interval_bed = \"results/phyluce/\" + group + \"/initial_intervals/\" + reference_taxon + \"_+\" + str(num_taxa) + \".bed\"\n",
    "i_bp_buffer    = 160\n",
    "o_interval_fas = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \".fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_genome_sequences_from_bed \"\n",
    "                      \"--bed {} \"\n",
    "                      \"--twobit {} \"\n",
    "                      \"--buffer-to {} \"\n",
    "                      \"--output {};\").format(i_interval_bed,\n",
    "                                             i_2bit_genome,\n",
    "                                             i_bp_buffer,\n",
    "                                             o_interval_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "design the baits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probes --input results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41.fasta --probe-prefix uce_streptococcus_ --design streptococcus_v1 --designer rnplattii --tiling-density 3 --two-probes --overlap middle --masking 0.25 --remove-gc --output results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas\n",
      "Probes removed for masking (.) / low GC % (G) / ambiguous bases (N):\n",
      "\n",
      "\n",
      "\n",
      "Conserved locus count = 100\n",
      "Probe Count = 200\n"
     ]
    }
   ],
   "source": [
    "i_interval_fas   = o_interval_fas\n",
    "i_probe_prefix   = \"uce_\" + group + \"_\"\n",
    "i_design_txt     = group + \"_v1\"\n",
    "i_designer_txt   = \"rnplattii\"\n",
    "i_tiling_density = \"3\"\n",
    "o_probes_fas     = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas\"\n",
    "\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probes \"\n",
    "                      \"--input {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--tiling-density {} \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--output {}\").format(i_interval_fas, \n",
    "                                            i_probe_prefix, \n",
    "                                            i_design_txt, \n",
    "                                            i_designer_txt, \n",
    "                                            i_tiling_density,\n",
    "                                            o_probes_fas)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find duplicate baited regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas --query results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas --identity 50 --coverage 50 --output results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes_vself.lastz;\n",
      "Started:  Thu Feb 06, 2020  15:38:46\n",
      "Ended:  Thu Feb 06, 2020  15:38:46\n",
      "Time for execution:  0.00281981627146 minutes\n",
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz                       --fasta results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas                        --lastz results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes_vself.lastz                       --probe-prefix=uce_streptococcus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 199 fasta sequences.  Filtered 9 duplicates. Kept 182.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#find duplicated probes within the probes set by a self v self lastz search\n",
    "#set up input and output files\n",
    "i_probes_fas   = o_probes_fas\n",
    "i_identity_int = str(50)\n",
    "i_coverage_int = str(50)\n",
    "\n",
    "o_probe_lastz  = o_dir + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes_vself.lastz\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--coverage {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_probes_fas, \n",
    "                                             i_identity_int, \n",
    "                                             i_coverage_int, \n",
    "                                             o_probe_lastz)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "\n",
    "#remove duplicated probes\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \\\n",
    "                      --fasta {}  \\\n",
    "                      --lastz {} \\\n",
    "                      --probe-prefix={};\").format(i_probes_fas, \n",
    "                                                  o_probe_lastz,\n",
    "                                                  i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Find orthologous seqs in other taxa (via lastz)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Create a modified genome dir structure to work with phyluce"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#need to set up soft links to make this work properly\n",
    "for sample in all_taxa:\n",
    "    \n",
    "    src_file = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"_formatted.2bit\"\n",
    "    \n",
    "    dst_dir  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\"\n",
    "    dst_file = dst_dir + sample + \".2bit\"\n",
    "    \n",
    "    if os.path.exists(dst_dir):\n",
    "        shutil.rmtree(dst_dir)\n",
    "        \n",
    "    os.mkdir(dst_dir)\n",
    "    copy(src_file, dst_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Use phyluce to run lastz runs against each genome in group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_run_multiple_lastzs_sqlite --probefile results/phyluce/streptococcus/validate_intervals/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas --scaffoldlist Streptococcus_acidominimus_GCF_900187245 Streptococcus_agalactiae_GCF_000007265 Streptococcus_alactolyticus_GCF_009695625 Streptococcus_anginosus_GCF_000463505 Streptococcus_australis_GCF_900476055 Streptococcus_azizii_GCF_001984715 Streptococcus_bovimastitidis_GCF_001885095 Streptococcus_caballi_GCF_000379985 Streptococcus_canis_GCF_900636575 Streptococcus_castoreus_GCF_000425025 Streptococcus_constellatus_GCF_000463425 Streptococcus_criceti_GCF_000187975 Streptococcus_cristatus_GCF_000385925 Streptococcus_cuniculi_GCF_001921845 Streptococcus_danieliae_GCF_009767945 Streptococcus_devriesei_GCF_000423725 Streptococcus_didelphis_GCF_000380005 Streptococcus_downei_GCF_900459175 Streptococcus_dysgalactiae_GCF_000317855 Streptococcus_entericus_GCF_000380025 Streptococcus_equi_GCF_000026605 Streptococcus_equinus_GCF_000964315 Streptococcus_ferus_GCF_000372425 Streptococcus_gallolyticus_GCF_002000985 Streptococcus_gordonii_GCF_000017005 Streptococcus_halichoeri_GCF_009870755 Streptococcus_halitosis_GCF_003143695 Streptococcus_halotolerans_GCF_001598035 Streptococcus_henryi_GCF_000376985 Streptococcus_himalayensis_GCF_001708305 Streptococcus_hongkongensis_GCF_000785845 Streptococcus_hyointestinalis_GCF_900459405 Streptococcus_hyovaginalis_GCF_000420785 Streptococcus_ictaluri_GCF_000188015 Streptococcus_infantarius_GCF_000246835 Streptococcus_infantis_GCF_000187465 Streptococcus_iniae_GCF_000831485 Streptococcus_intermedius_GCF_000463355 Streptococcus_lutetiensis_GCF_900475675 Streptococcus_macacae_GCF_000187995 Streptococcus_macedonicus_GCF_000283635 Streptococcus_marimammalium_GCF_000380045 Streptococcus_marmotae_GCF_001623565 Streptococcus_massiliensis_GCF_000380065 Streptococcus_merionis_GCF_000380085 Streptococcus_milleri_GCF_900636715 Streptococcus_minor_GCF_000377005 Streptococcus_mitis_GCF_000027165 Streptococcus_mutans_GCF_000007465 Streptococcus_oralis_GCF_900637025 Streptococcus_orisasini_GCF_001431045 Streptococcus_orisratti_GCF_000380105 Streptococcus_ovis_GCF_000380125 Streptococcus_pantholopis_GCF_001642085 Streptococcus_parasanguinis_GCF_000164675 Streptococcus_parasuis_GCF_004283785 Streptococcus_parauberis_GCF_000213825 Streptococcus_pasteurianus_GCF_900478025 Streptococcus_penaeicida_GCF_002887775 Streptococcus_peroris_GCF_000187585 Streptococcus_pharyngis_GCF_007859195 Streptococcus_phocae_GCF_000772915 Streptococcus_pluranimalium_GCF_003352995 Streptococcus_plurextorum_GCF_000423745 Streptococcus_porci_GCF_000423765 Streptococcus_porcinus_GCF_900475415 Streptococcus_pseudopneumoniae_GCF_000221985 Streptococcus_pseudoporcinus_GCF_000188035 Streptococcus_pyogenes_GCF_000006785 Streptococcus_ratti_GCF_000286075 Streptococcus_respraculi_GCF_003595525 Streptococcus_rubneri_GCF_004785935 Streptococcus_ruminantium_GCF_003609975 Streptococcus_salivarius_GCF_000785515 Streptococcus_sanguinis_GCF_000014205 Streptococcus_sinensis_GCF_000767835 Streptococcus_sobrinus_GCF_000686605 Streptococcus_sp_GCF_003086355 Streptococcus_suis_GCF_000026745 Streptococcus_thermophilus_GCF_000253395 Streptococcus_thoraltensis_GCF_000380145 Streptococcus_timonensis_GCF_900095845 Streptococcus_troglodytae_GCF_002355215 Streptococcus_uberis_GCF_000009545 Streptococcus_urinalis_GCF_000188055 Streptococcus_varani_GCF_001375655 Streptococcus_vestibularis_GCF_900636445 Streptococcus_viridans_GCF_900636365 Streptococcus_pneumoniae_GCF_000007045 --genome-base-path results/phyluce/streptococcus/cleaned_genomes --identity 30 --cores 4 --db results/phyluce/streptococcus/validate_intervals/streptococcus-to-Streptococcus_pneumoniae_GCF_000007045.sqlite --output results/phyluce/streptococcus/validate_intervals/lastz/;\n",
      "\n",
      "Running against Streptococcus_acidominimus_GCF_900187245.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpNzOKsa.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7DKdqk.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_acidominimus_GCF_900187245.lastz\n",
      "Creating Streptococcus_acidominimus_GCF_900187245 table\n",
      "Inserting data to Streptococcus_acidominimus_GCF_900187245 table\n",
      "\n",
      "Running against Streptococcus_agalactiae_GCF_000007265.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpVKB9xh.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbz9Kiv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_agalactiae_GCF_000007265.lastz\n",
      "Creating Streptococcus_agalactiae_GCF_000007265 table\n",
      "Inserting data to Streptococcus_agalactiae_GCF_000007265 table\n",
      "\n",
      "Running against Streptococcus_alactolyticus_GCF_009695625.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmptwi_Q4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmph5A5bG.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_alactolyticus_GCF_009695625.lastz\n",
      "Creating Streptococcus_alactolyticus_GCF_009695625 table\n",
      "Inserting data to Streptococcus_alactolyticus_GCF_009695625 table\n",
      "\n",
      "Running against Streptococcus_anginosus_GCF_000463505.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpA7ZXrc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp1NW4IF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_anginosus_GCF_000463505.lastz\n",
      "Creating Streptococcus_anginosus_GCF_000463505 table\n",
      "Inserting data to Streptococcus_anginosus_GCF_000463505 table\n",
      "\n",
      "Running against Streptococcus_australis_GCF_900476055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaS_uFs.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpEUYUM7.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_australis_GCF_900476055.lastz\n",
      "Creating Streptococcus_australis_GCF_900476055 table\n",
      "Inserting data to Streptococcus_australis_GCF_900476055 table\n",
      "\n",
      "Running against Streptococcus_azizii_GCF_001984715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2vtEsx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdp4in_.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_azizii_GCF_001984715.lastz\n",
      "Creating Streptococcus_azizii_GCF_001984715 table\n",
      "Inserting data to Streptococcus_azizii_GCF_001984715 table\n",
      "\n",
      "Running against Streptococcus_bovimastitidis_GCF_001885095.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7jVi6Y.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9hKhnl.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_bovimastitidis_GCF_001885095.lastz\n",
      "Creating Streptococcus_bovimastitidis_GCF_001885095 table\n",
      "Inserting data to Streptococcus_bovimastitidis_GCF_001885095 table\n",
      "\n",
      "Running against Streptococcus_caballi_GCF_000379985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpD6wm2v.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdn8uJC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_caballi_GCF_000379985.lastz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating Streptococcus_caballi_GCF_000379985 table\n",
      "Inserting data to Streptococcus_caballi_GCF_000379985 table\n",
      "\n",
      "Running against Streptococcus_canis_GCF_900636575.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpz6O8Vl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmppVZ5hF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_canis_GCF_900636575.lastz\n",
      "Creating Streptococcus_canis_GCF_900636575 table\n",
      "Inserting data to Streptococcus_canis_GCF_900636575 table\n",
      "\n",
      "Running against Streptococcus_castoreus_GCF_000425025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZxyIro.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8ERmAr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_castoreus_GCF_000425025.lastz\n",
      "Creating Streptococcus_castoreus_GCF_000425025 table\n",
      "Inserting data to Streptococcus_castoreus_GCF_000425025 table\n",
      "\n",
      "Running against Streptococcus_constellatus_GCF_000463425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpq6ZcN3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVqXCUI.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_constellatus_GCF_000463425.lastz\n",
      "Creating Streptococcus_constellatus_GCF_000463425 table\n",
      "Inserting data to Streptococcus_constellatus_GCF_000463425 table\n",
      "\n",
      "Running against Streptococcus_criceti_GCF_000187975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpy_7qW_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpRR8rhC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_criceti_GCF_000187975.lastz\n",
      "Creating Streptococcus_criceti_GCF_000187975 table\n",
      "Inserting data to Streptococcus_criceti_GCF_000187975 table\n",
      "\n",
      "Running against Streptococcus_cristatus_GCF_000385925.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3zPxUf.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4XwGIN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_cristatus_GCF_000385925.lastz\n",
      "Creating Streptococcus_cristatus_GCF_000385925 table\n",
      "Inserting data to Streptococcus_cristatus_GCF_000385925 table\n",
      "\n",
      "Running against Streptococcus_cuniculi_GCF_001921845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwKJZEQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpboS5Qu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_cuniculi_GCF_001921845.lastz\n",
      "Creating Streptococcus_cuniculi_GCF_001921845 table\n",
      "Inserting data to Streptococcus_cuniculi_GCF_001921845 table\n",
      "\n",
      "Running against Streptococcus_danieliae_GCF_009767945.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6FbUFL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnUf2S1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_danieliae_GCF_009767945.lastz\n",
      "Creating Streptococcus_danieliae_GCF_009767945 table\n",
      "Inserting data to Streptococcus_danieliae_GCF_009767945 table\n",
      "\n",
      "Running against Streptococcus_devriesei_GCF_000423725.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQLCnYR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfwrlU1.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_devriesei_GCF_000423725.lastz\n",
      "Creating Streptococcus_devriesei_GCF_000423725 table\n",
      "Inserting data to Streptococcus_devriesei_GCF_000423725 table\n",
      "\n",
      "Running against Streptococcus_didelphis_GCF_000380005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxvXgwU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGNRjYN.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_didelphis_GCF_000380005.lastz\n",
      "Creating Streptococcus_didelphis_GCF_000380005 table\n",
      "Inserting data to Streptococcus_didelphis_GCF_000380005 table\n",
      "\n",
      "Running against Streptococcus_downei_GCF_900459175.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwtLGZx.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdNVIww.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_downei_GCF_900459175.lastz\n",
      "Creating Streptococcus_downei_GCF_900459175 table\n",
      "Inserting data to Streptococcus_downei_GCF_900459175 table\n",
      "\n",
      "Running against Streptococcus_dysgalactiae_GCF_000317855.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaHMb6K.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp75i_NE.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_dysgalactiae_GCF_000317855.lastz\n",
      "Creating Streptococcus_dysgalactiae_GCF_000317855 table\n",
      "Inserting data to Streptococcus_dysgalactiae_GCF_000317855 table\n",
      "\n",
      "Running against Streptococcus_entericus_GCF_000380025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpuB8QIB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp4HepAP.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_entericus_GCF_000380025.lastz\n",
      "Creating Streptococcus_entericus_GCF_000380025 table\n",
      "Inserting data to Streptococcus_entericus_GCF_000380025 table\n",
      "\n",
      "Running against Streptococcus_equi_GCF_000026605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwZEbKm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpnqtrPf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_equi_GCF_000026605.lastz\n",
      "Creating Streptococcus_equi_GCF_000026605 table\n",
      "Inserting data to Streptococcus_equi_GCF_000026605 table\n",
      "\n",
      "Running against Streptococcus_equinus_GCF_000964315.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOfv_P2.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHKdws3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_equinus_GCF_000964315.lastz\n",
      "Creating Streptococcus_equinus_GCF_000964315 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Streptococcus_equinus_GCF_000964315 table\n",
      "\n",
      "Running against Streptococcus_ferus_GCF_000372425.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp9s3guk.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmprpK1wS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ferus_GCF_000372425.lastz\n",
      "Creating Streptococcus_ferus_GCF_000372425 table\n",
      "Inserting data to Streptococcus_ferus_GCF_000372425 table\n",
      "\n",
      "Running against Streptococcus_gallolyticus_GCF_002000985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpN6PSFq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp_tCpfa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_gallolyticus_GCF_002000985.lastz\n",
      "Creating Streptococcus_gallolyticus_GCF_002000985 table\n",
      "Inserting data to Streptococcus_gallolyticus_GCF_002000985 table\n",
      "\n",
      "Running against Streptococcus_gordonii_GCF_000017005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0tANUP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpBAIJy9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_gordonii_GCF_000017005.lastz\n",
      "Creating Streptococcus_gordonii_GCF_000017005 table\n",
      "Inserting data to Streptococcus_gordonii_GCF_000017005 table\n",
      "\n",
      "Running against Streptococcus_halichoeri_GCF_009870755.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpeVXY9M.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3B2lHF.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_halichoeri_GCF_009870755.lastz\n",
      "Creating Streptococcus_halichoeri_GCF_009870755 table\n",
      "Inserting data to Streptococcus_halichoeri_GCF_009870755 table\n",
      "\n",
      "Running against Streptococcus_halitosis_GCF_003143695.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpT77s0k.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpVPqXZb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_halitosis_GCF_003143695.lastz\n",
      "Creating Streptococcus_halitosis_GCF_003143695 table\n",
      "Inserting data to Streptococcus_halitosis_GCF_003143695 table\n",
      "\n",
      "Running against Streptococcus_halotolerans_GCF_001598035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7MuMl8.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpwpW2AH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_halotolerans_GCF_001598035.lastz\n",
      "Creating Streptococcus_halotolerans_GCF_001598035 table\n",
      "Inserting data to Streptococcus_halotolerans_GCF_001598035 table\n",
      "\n",
      "Running against Streptococcus_henryi_GCF_000376985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpC4ILJS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHl3w5K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_henryi_GCF_000376985.lastz\n",
      "Creating Streptococcus_henryi_GCF_000376985 table\n",
      "Inserting data to Streptococcus_henryi_GCF_000376985 table\n",
      "\n",
      "Running against Streptococcus_himalayensis_GCF_001708305.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWSNU6O.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpdgT8G4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_himalayensis_GCF_001708305.lastz\n",
      "Creating Streptococcus_himalayensis_GCF_001708305 table\n",
      "Inserting data to Streptococcus_himalayensis_GCF_001708305 table\n",
      "\n",
      "Running against Streptococcus_hongkongensis_GCF_000785845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpd75k59.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpX93aYd.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_hongkongensis_GCF_000785845.lastz\n",
      "Creating Streptococcus_hongkongensis_GCF_000785845 table\n",
      "Inserting data to Streptococcus_hongkongensis_GCF_000785845 table\n",
      "\n",
      "Running against Streptococcus_hyointestinalis_GCF_900459405.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBT8kZp.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpplm72N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_hyointestinalis_GCF_900459405.lastz\n",
      "Creating Streptococcus_hyointestinalis_GCF_900459405 table\n",
      "Inserting data to Streptococcus_hyointestinalis_GCF_900459405 table\n",
      "\n",
      "Running against Streptococcus_hyovaginalis_GCF_000420785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0FRcSZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpP3CisV.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_hyovaginalis_GCF_000420785.lastz\n",
      "Creating Streptococcus_hyovaginalis_GCF_000420785 table\n",
      "Inserting data to Streptococcus_hyovaginalis_GCF_000420785 table\n",
      "\n",
      "Running against Streptococcus_ictaluri_GCF_000188015.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpGApkus.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9AOqXT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ictaluri_GCF_000188015.lastz\n",
      "Creating Streptococcus_ictaluri_GCF_000188015 table\n",
      "Inserting data to Streptococcus_ictaluri_GCF_000188015 table\n",
      "\n",
      "Running against Streptococcus_infantarius_GCF_000246835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmVGpg4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp27GAZu.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_infantarius_GCF_000246835.lastz\n",
      "Creating Streptococcus_infantarius_GCF_000246835 table\n",
      "Inserting data to Streptococcus_infantarius_GCF_000246835 table\n",
      "\n",
      "Running against Streptococcus_infantis_GCF_000187465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp2MzRWQ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpXJKlyJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_infantis_GCF_000187465.lastz\n",
      "Creating Streptococcus_infantis_GCF_000187465 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inserting data to Streptococcus_infantis_GCF_000187465 table\n",
      "\n",
      "Running against Streptococcus_iniae_GCF_000831485.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJLsAGi.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJb2aaC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_iniae_GCF_000831485.lastz\n",
      "Creating Streptococcus_iniae_GCF_000831485 table\n",
      "Inserting data to Streptococcus_iniae_GCF_000831485 table\n",
      "\n",
      "Running against Streptococcus_intermedius_GCF_000463355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp81yovK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpzE3xC3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_intermedius_GCF_000463355.lastz\n",
      "Creating Streptococcus_intermedius_GCF_000463355 table\n",
      "Inserting data to Streptococcus_intermedius_GCF_000463355 table\n",
      "\n",
      "Running against Streptococcus_lutetiensis_GCF_900475675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0hSFEV.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpY9d9Sf.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_lutetiensis_GCF_900475675.lastz\n",
      "Creating Streptococcus_lutetiensis_GCF_900475675 table\n",
      "Inserting data to Streptococcus_lutetiensis_GCF_900475675 table\n",
      "\n",
      "Running against Streptococcus_macacae_GCF_000187995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprLPlb_.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAXlZDM.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_macacae_GCF_000187995.lastz\n",
      "Creating Streptococcus_macacae_GCF_000187995 table\n",
      "Inserting data to Streptococcus_macacae_GCF_000187995 table\n",
      "\n",
      "Running against Streptococcus_macedonicus_GCF_000283635.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpcht_vS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpDWfIR3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_macedonicus_GCF_000283635.lastz\n",
      "Creating Streptococcus_macedonicus_GCF_000283635 table\n",
      "Inserting data to Streptococcus_macedonicus_GCF_000283635 table\n",
      "\n",
      "Running against Streptococcus_marimammalium_GCF_000380045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpFW7m9y.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpl7uQ8Q.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_marimammalium_GCF_000380045.lastz\n",
      "Creating Streptococcus_marimammalium_GCF_000380045 table\n",
      "Inserting data to Streptococcus_marimammalium_GCF_000380045 table\n",
      "\n",
      "Running against Streptococcus_marmotae_GCF_001623565.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp0E5Y5L.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfred0D.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_marmotae_GCF_001623565.lastz\n",
      "Creating Streptococcus_marmotae_GCF_001623565 table\n",
      "Inserting data to Streptococcus_marmotae_GCF_001623565 table\n",
      "\n",
      "Running against Streptococcus_massiliensis_GCF_000380065.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJkfsv7.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp9gLuGX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_massiliensis_GCF_000380065.lastz\n",
      "Creating Streptococcus_massiliensis_GCF_000380065 table\n",
      "Inserting data to Streptococcus_massiliensis_GCF_000380065 table\n",
      "\n",
      "Running against Streptococcus_merionis_GCF_000380085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWFKgqd.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpo3jV5W.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_merionis_GCF_000380085.lastz\n",
      "Creating Streptococcus_merionis_GCF_000380085 table\n",
      "Inserting data to Streptococcus_merionis_GCF_000380085 table\n",
      "\n",
      "Running against Streptococcus_milleri_GCF_900636715.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpdGgH4g.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpImEctX.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_milleri_GCF_900636715.lastz\n",
      "Creating Streptococcus_milleri_GCF_900636715 table\n",
      "Inserting data to Streptococcus_milleri_GCF_900636715 table\n",
      "\n",
      "Running against Streptococcus_minor_GCF_000377005.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpADUrS9.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpFe2CT5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_minor_GCF_000377005.lastz\n",
      "Creating Streptococcus_minor_GCF_000377005 table\n",
      "Inserting data to Streptococcus_minor_GCF_000377005 table\n",
      "\n",
      "Running against Streptococcus_mitis_GCF_000027165.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpKyR3bS.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp7DuTQ5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_mitis_GCF_000027165.lastz\n",
      "Creating Streptococcus_mitis_GCF_000027165 table\n",
      "Inserting data to Streptococcus_mitis_GCF_000027165 table\n",
      "\n",
      "Running against Streptococcus_mutans_GCF_000007465.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpgcxmhZ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpE1tRja.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_mutans_GCF_000007465.lastz\n",
      "Creating Streptococcus_mutans_GCF_000007465 table\n",
      "Inserting data to Streptococcus_mutans_GCF_000007465 table\n",
      "\n",
      "Running against Streptococcus_oralis_GCF_900637025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZAJ2Qm.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpx0FX5d.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_oralis_GCF_900637025.lastz\n",
      "Creating Streptococcus_oralis_GCF_900637025 table\n",
      "Inserting data to Streptococcus_oralis_GCF_900637025 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Streptococcus_orisasini_GCF_001431045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpCxQQss.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpxcLn4K.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_orisasini_GCF_001431045.lastz\n",
      "Creating Streptococcus_orisasini_GCF_001431045 table\n",
      "Inserting data to Streptococcus_orisasini_GCF_001431045 table\n",
      "\n",
      "Running against Streptococcus_orisratti_GCF_000380105.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp09d6mv.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpAcPUrC.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_orisratti_GCF_000380105.lastz\n",
      "Creating Streptococcus_orisratti_GCF_000380105 table\n",
      "Inserting data to Streptococcus_orisratti_GCF_000380105 table\n",
      "\n",
      "Running against Streptococcus_ovis_GCF_000380125.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmphTkQu6.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpPkzfTb.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ovis_GCF_000380125.lastz\n",
      "Creating Streptococcus_ovis_GCF_000380125 table\n",
      "Inserting data to Streptococcus_ovis_GCF_000380125 table\n",
      "\n",
      "Running against Streptococcus_pantholopis_GCF_001642085.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpqE0YUX.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp0pRspH.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pantholopis_GCF_001642085.lastz\n",
      "Creating Streptococcus_pantholopis_GCF_001642085 table\n",
      "Inserting data to Streptococcus_pantholopis_GCF_001642085 table\n",
      "\n",
      "Running against Streptococcus_parasanguinis_GCF_000164675.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpfnJ8vB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpQx_SxU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_parasanguinis_GCF_000164675.lastz\n",
      "Creating Streptococcus_parasanguinis_GCF_000164675 table\n",
      "Inserting data to Streptococcus_parasanguinis_GCF_000164675 table\n",
      "\n",
      "Running against Streptococcus_parasuis_GCF_004283785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpOuGFhK.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6uDiHY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_parasuis_GCF_004283785.lastz\n",
      "Creating Streptococcus_parasuis_GCF_004283785 table\n",
      "Inserting data to Streptococcus_parasuis_GCF_004283785 table\n",
      "\n",
      "Running against Streptococcus_parauberis_GCF_000213825.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpkOF43V.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpw7exCr.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_parauberis_GCF_000213825.lastz\n",
      "Creating Streptococcus_parauberis_GCF_000213825 table\n",
      "Inserting data to Streptococcus_parauberis_GCF_000213825 table\n",
      "\n",
      "Running against Streptococcus_pasteurianus_GCF_900478025.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpRoeSw4.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpg5c8E8.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pasteurianus_GCF_900478025.lastz\n",
      "Creating Streptococcus_pasteurianus_GCF_900478025 table\n",
      "Inserting data to Streptococcus_pasteurianus_GCF_900478025 table\n",
      "\n",
      "Running against Streptococcus_penaeicida_GCF_002887775.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpBd8P8E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcaFefw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_penaeicida_GCF_002887775.lastz\n",
      "Creating Streptococcus_penaeicida_GCF_002887775 table\n",
      "Inserting data to Streptococcus_penaeicida_GCF_002887775 table\n",
      "\n",
      "Running against Streptococcus_peroris_GCF_000187585.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpHHcIqP.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTgFuDJ.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_peroris_GCF_000187585.lastz\n",
      "Creating Streptococcus_peroris_GCF_000187585 table\n",
      "Inserting data to Streptococcus_peroris_GCF_000187585 table\n",
      "\n",
      "Running against Streptococcus_pharyngis_GCF_007859195.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6GqUnU.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp3vnlZY.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pharyngis_GCF_007859195.lastz\n",
      "Creating Streptococcus_pharyngis_GCF_007859195 table\n",
      "Inserting data to Streptococcus_pharyngis_GCF_007859195 table\n",
      "\n",
      "Running against Streptococcus_phocae_GCF_000772915.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpneZN_E.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpcflk93.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_phocae_GCF_000772915.lastz\n",
      "Creating Streptococcus_phocae_GCF_000772915 table\n",
      "Inserting data to Streptococcus_phocae_GCF_000772915 table\n",
      "\n",
      "Running against Streptococcus_pluranimalium_GCF_003352995.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpw4zukz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpSHbDMS.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pluranimalium_GCF_003352995.lastz\n",
      "Creating Streptococcus_pluranimalium_GCF_003352995 table\n",
      "Inserting data to Streptococcus_pluranimalium_GCF_003352995 table\n",
      "\n",
      "Running against Streptococcus_plurextorum_GCF_000423745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpC2QTrj.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpd1W95L.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_plurextorum_GCF_000423745.lastz\n",
      "Creating Streptococcus_plurextorum_GCF_000423745 table\n",
      "Inserting data to Streptococcus_plurextorum_GCF_000423745 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Streptococcus_porci_GCF_000423765.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpZPMOyW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6Getaa.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_porci_GCF_000423765.lastz\n",
      "Creating Streptococcus_porci_GCF_000423765 table\n",
      "Inserting data to Streptococcus_porci_GCF_000423765 table\n",
      "\n",
      "Running against Streptococcus_porcinus_GCF_900475415.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpwRfBqW.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp2bBfpT.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_porcinus_GCF_900475415.lastz\n",
      "Creating Streptococcus_porcinus_GCF_900475415 table\n",
      "Inserting data to Streptococcus_porcinus_GCF_900475415 table\n",
      "\n",
      "Running against Streptococcus_pseudopneumoniae_GCF_000221985.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpm7w8hr.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGHuGdq.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pseudopneumoniae_GCF_000221985.lastz\n",
      "Creating Streptococcus_pseudopneumoniae_GCF_000221985 table\n",
      "Inserting data to Streptococcus_pseudopneumoniae_GCF_000221985 table\n",
      "\n",
      "Running against Streptococcus_pseudoporcinus_GCF_000188035.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJaurvE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp6Nogec.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pseudoporcinus_GCF_000188035.lastz\n",
      "Creating Streptococcus_pseudoporcinus_GCF_000188035 table\n",
      "Inserting data to Streptococcus_pseudoporcinus_GCF_000188035 table\n",
      "\n",
      "Running against Streptococcus_pyogenes_GCF_000006785.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp7ibKoz.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmptobWDU.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pyogenes_GCF_000006785.lastz\n",
      "Creating Streptococcus_pyogenes_GCF_000006785 table\n",
      "Inserting data to Streptococcus_pyogenes_GCF_000006785 table\n",
      "\n",
      "Running against Streptococcus_ratti_GCF_000286075.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxG50tH.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKxk9ek.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ratti_GCF_000286075.lastz\n",
      "Creating Streptococcus_ratti_GCF_000286075 table\n",
      "Inserting data to Streptococcus_ratti_GCF_000286075 table\n",
      "\n",
      "Running against Streptococcus_respraculi_GCF_003595525.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpJu95zM.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmps5YHP9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_respraculi_GCF_003595525.lastz\n",
      "Creating Streptococcus_respraculi_GCF_003595525 table\n",
      "Inserting data to Streptococcus_respraculi_GCF_003595525 table\n",
      "\n",
      "Running against Streptococcus_rubneri_GCF_004785935.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp8i2YFR.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpREVpeh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_rubneri_GCF_004785935.lastz\n",
      "Creating Streptococcus_rubneri_GCF_004785935 table\n",
      "Inserting data to Streptococcus_rubneri_GCF_004785935 table\n",
      "\n",
      "Running against Streptococcus_ruminantium_GCF_003609975.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmprzEeZo.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpCYkYxR.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_ruminantium_GCF_003609975.lastz\n",
      "Creating Streptococcus_ruminantium_GCF_003609975 table\n",
      "Inserting data to Streptococcus_ruminantium_GCF_003609975 table\n",
      "\n",
      "Running against Streptococcus_salivarius_GCF_000785515.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpsaDi2A.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpLBZP5f.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_salivarius_GCF_000785515.lastz\n",
      "Creating Streptococcus_salivarius_GCF_000785515 table\n",
      "Inserting data to Streptococcus_salivarius_GCF_000785515 table\n",
      "\n",
      "Running against Streptococcus_sanguinis_GCF_000014205.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmplYdboE.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpIQjj7C.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sanguinis_GCF_000014205.lastz\n",
      "Creating Streptococcus_sanguinis_GCF_000014205 table\n",
      "Inserting data to Streptococcus_sanguinis_GCF_000014205 table\n",
      "\n",
      "Running against Streptococcus_sinensis_GCF_000767835.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp_mgLyq.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpy2Yhvs.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sinensis_GCF_000767835.lastz\n",
      "Creating Streptococcus_sinensis_GCF_000767835 table\n",
      "Inserting data to Streptococcus_sinensis_GCF_000767835 table\n",
      "\n",
      "Running against Streptococcus_sobrinus_GCF_000686605.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpWfXxeB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqeaCzK.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sobrinus_GCF_000686605.lastz\n",
      "Creating Streptococcus_sobrinus_GCF_000686605 table\n",
      "Inserting data to Streptococcus_sobrinus_GCF_000686605 table\n",
      "\n",
      "Running against Streptococcus_sp_GCF_003086355.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpxnAnPb.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp8stMo9.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_sp_GCF_003086355.lastz\n",
      "Creating Streptococcus_sp_GCF_003086355 table\n",
      "Inserting data to Streptococcus_sp_GCF_003086355 table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running against Streptococcus_suis_GCF_000026745.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpvukvvc.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpbRqZo3.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_suis_GCF_000026745.lastz\n",
      "Creating Streptococcus_suis_GCF_000026745 table\n",
      "Inserting data to Streptococcus_suis_GCF_000026745 table\n",
      "\n",
      "Running against Streptococcus_thermophilus_GCF_000253395.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpakGhf3.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpmRYu_r.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_thermophilus_GCF_000253395.lastz\n",
      "Creating Streptococcus_thermophilus_GCF_000253395 table\n",
      "Inserting data to Streptococcus_thermophilus_GCF_000253395 table\n",
      "\n",
      "Running against Streptococcus_thoraltensis_GCF_000380145.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp3nAiXI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpTttoCw.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_thoraltensis_GCF_000380145.lastz\n",
      "Creating Streptococcus_thoraltensis_GCF_000380145 table\n",
      "Inserting data to Streptococcus_thoraltensis_GCF_000380145 table\n",
      "\n",
      "Running against Streptococcus_timonensis_GCF_900095845.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpTfEaLI.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpHOLqH2.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_timonensis_GCF_900095845.lastz\n",
      "Creating Streptococcus_timonensis_GCF_900095845 table\n",
      "Inserting data to Streptococcus_timonensis_GCF_900095845 table\n",
      "\n",
      "Running against Streptococcus_troglodytae_GCF_002355215.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmp6plCxl.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmp5gLQwD.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_troglodytae_GCF_002355215.lastz\n",
      "Creating Streptococcus_troglodytae_GCF_002355215 table\n",
      "Inserting data to Streptococcus_troglodytae_GCF_002355215 table\n",
      "\n",
      "Running against Streptococcus_uberis_GCF_000009545.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpaC0TID.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpJ4zn02.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_uberis_GCF_000009545.lastz\n",
      "Creating Streptococcus_uberis_GCF_000009545 table\n",
      "Inserting data to Streptococcus_uberis_GCF_000009545 table\n",
      "\n",
      "Running against Streptococcus_urinalis_GCF_000188055.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpmzXlA0.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpqN8Uiv.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_urinalis_GCF_000188055.lastz\n",
      "Creating Streptococcus_urinalis_GCF_000188055 table\n",
      "Inserting data to Streptococcus_urinalis_GCF_000188055 table\n",
      "\n",
      "Running against Streptococcus_varani_GCF_001375655.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQEkcuJ.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpGSaf6N.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_varani_GCF_001375655.lastz\n",
      "Creating Streptococcus_varani_GCF_001375655 table\n",
      "Inserting data to Streptococcus_varani_GCF_001375655 table\n",
      "\n",
      "Running against Streptococcus_vestibularis_GCF_900636445.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnN8F57.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpKsbtK5.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_vestibularis_GCF_900636445.lastz\n",
      "Creating Streptococcus_vestibularis_GCF_900636445 table\n",
      "Inserting data to Streptococcus_vestibularis_GCF_900636445 table\n",
      "\n",
      "Running against Streptococcus_viridans_GCF_900636365.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpnuhvtL.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpfOoUFh.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_viridans_GCF_900636365.lastz\n",
      "Creating Streptococcus_viridans_GCF_900636365 table\n",
      "Inserting data to Streptococcus_viridans_GCF_900636365 table\n",
      "\n",
      "Running against Streptococcus_pneumoniae_GCF_000007045.2bit\n",
      "Running with the --huge option.  Chunking files into 10000000 bp...\n",
      "Running the targets against 1 queries...\n",
      "\t/tmp/tmpQaOQwB.fasta\n",
      "\n",
      "Writing the results file...\n",
      "\t/tmp/tmpUUiNM4.lastz\n",
      "Cleaning up the chunked files...\n",
      "Cleaning /master/nplatt/pathogen_probes/results/phyluce/streptococcus/validate_intervals/lastz/Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_Streptococcus_pneumoniae_GCF_000007045.lastz\n",
      "Creating Streptococcus_pneumoniae_GCF_000007045 table\n",
      "Inserting data to Streptococcus_pneumoniae_GCF_000007045 table\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas    = i_probes_fas\n",
    "i_genome_list   = \" \".join(all_taxa)\n",
    "i_genome_dir    = \"results/phyluce/\" + group + \"/cleaned_genomes\"\n",
    "i_percident_int = str(30)\n",
    "i_ncores_int    = str(4)\n",
    "o_sqlite_db     = o_dir + group + \"-to-\" + reference_taxon + \".sqlite\"\n",
    "o_lastz_dir     = o_dir + \"lastz/\"\n",
    "#triCas1+5+menMol1.sqlite\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_run_multiple_lastzs_sqlite \"\n",
    "                      \"--probefile {} \"\n",
    "                      \"--scaffoldlist {} \"\n",
    "                      \"--genome-base-path {} \"\n",
    "                      \"--identity {} \"\n",
    "                      \"--cores {} \"\n",
    "                      \"--db {} \"\n",
    "                      \"--output {};\").format(i_probes_fas, \n",
    "                                             i_genome_list, \n",
    "                                             i_genome_dir,\n",
    "                                             i_percident_int,\n",
    "                                             i_ncores_int,\n",
    "                                             o_sqlite_db,\n",
    "                                             o_lastz_dir)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Extract seqeunces from group genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "make a conf file with the following format\n",
    "\n",
    "```\n",
    "[scaffolds]\n",
    "menMol1:/path/to/uce-coleoptera/genomes/menMol1/menMol1.2bit\n",
    "agrPla1:/path/to/uce-coleoptera/genomes/agrPla1/agrPla1.2bit\n",
    "anoGla1:/path/to/uce-coleoptera/genomes/anoGla1/anoGla1.2bit\n",
    "denPon1:/path/to/uce-coleoptera/genomes/denPon1/denPon1.2bit\n",
    "lepDec1:/path/to/uce-coleoptera/genomes/lepDec1/lepDec1.2bit\n",
    "ontTau1:/path/to/uce-coleoptera/genomes/ontTau1/ontTau1.2bit\n",
    "triCas1:/path/to/uce-coleoptera/genomes/triCas1/triCas1.2bit\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_slice_sequence_from_genomes --conf results/phyluce/streptococcus/extract_probes_from_group/streptococcus_genome.conf --lastz results/phyluce/streptococcus/validate_intervals/lastz --probes 120 --probe-prefix uce_streptococcus_ --name-pattern \"Streptococcus_pneumoniae_GCF_000007045_+41_temp_probes.fas_v_{}.lastz.clean\" --output results/phyluce/streptococcus/extract_probes_from_group/probe_fasta;\n",
      "2020-02-06 15:40:38,165 - Phyluce - INFO - =================== Starting Phyluce: Slice Sequence ===================\n",
      "2020-02-06 15:40:38,174 - Phyluce - INFO - --- Working on Streptococcus_acidominimus_GCF_900187245 genome --\n",
      "2020-02-06 15:40:38,174 - Phyluce - INFO - Reading Streptococcus_acidominimus_GCF_900187245 genome\n",
      "2020-02-06 15:40:39,054 - Phyluce - INFO - Streptococcus_acidominimus_GCF_900187245: 100 uces, 0 dupes, 100 non-dupes, 18 orient drop, 1 length drop, 81 written\n",
      "2020-02-06 15:40:39,054 - Phyluce - INFO - ---- Working on Streptococcus_agalactiae_GCF_000007265 genome ---\n",
      "2020-02-06 15:40:39,055 - Phyluce - INFO - Reading Streptococcus_agalactiae_GCF_000007265 genome\n",
      "2020-02-06 15:40:39,641 - Phyluce - INFO - Streptococcus_agalactiae_GCF_000007265: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 19 length drop, 81 written\n",
      "2020-02-06 15:40:39,641 - Phyluce - INFO - -- Working on Streptococcus_alactolyticus_GCF_009695625 genome --\n",
      "2020-02-06 15:40:39,642 - Phyluce - INFO - Reading Streptococcus_alactolyticus_GCF_009695625 genome\n",
      "2020-02-06 15:40:39,968 - Phyluce - INFO - Streptococcus_alactolyticus_GCF_009695625: 98 uces, 2 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:40:39,968 - Phyluce - INFO - ---- Working on Streptococcus_anginosus_GCF_000463505 genome ----\n",
      "2020-02-06 15:40:39,968 - Phyluce - INFO - Reading Streptococcus_anginosus_GCF_000463505 genome\n",
      "2020-02-06 15:40:40,422 - Phyluce - INFO - Streptococcus_anginosus_GCF_000463505: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:40,422 - Phyluce - INFO - ---- Working on Streptococcus_australis_GCF_900476055 genome ----\n",
      "2020-02-06 15:40:40,438 - Phyluce - INFO - Reading Streptococcus_australis_GCF_900476055 genome\n",
      "2020-02-06 15:40:40,901 - Phyluce - INFO - Streptococcus_australis_GCF_900476055: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:40,901 - Phyluce - INFO - ------ Working on Streptococcus_azizii_GCF_001984715 genome -----\n",
      "2020-02-06 15:40:40,908 - Phyluce - INFO - Reading Streptococcus_azizii_GCF_001984715 genome\n",
      "2020-02-06 15:40:41,252 - Phyluce - INFO - Streptococcus_azizii_GCF_001984715: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:41,252 - Phyluce - INFO - -- Working on Streptococcus_bovimastitidis_GCF_001885095 genome -\n",
      "2020-02-06 15:40:41,252 - Phyluce - INFO - Reading Streptococcus_bovimastitidis_GCF_001885095 genome\n",
      "2020-02-06 15:40:41,605 - Phyluce - INFO - Streptococcus_bovimastitidis_GCF_001885095: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 2 length drop, 91 written\n",
      "2020-02-06 15:40:41,605 - Phyluce - INFO - ----- Working on Streptococcus_caballi_GCF_000379985 genome -----\n",
      "2020-02-06 15:40:41,606 - Phyluce - INFO - Reading Streptococcus_caballi_GCF_000379985 genome\n",
      "2020-02-06 15:40:41,930 - Phyluce - INFO - Streptococcus_caballi_GCF_000379985: 98 uces, 1 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-06 15:40:41,930 - Phyluce - INFO - ------ Working on Streptococcus_canis_GCF_900636575 genome ------\n",
      "2020-02-06 15:40:41,931 - Phyluce - INFO - Reading Streptococcus_canis_GCF_900636575 genome\n",
      "2020-02-06 15:40:42,473 - Phyluce - INFO - Streptococcus_canis_GCF_900636575: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:42,474 - Phyluce - INFO - ---- Working on Streptococcus_castoreus_GCF_000425025 genome ----\n",
      "2020-02-06 15:40:42,474 - Phyluce - INFO - Reading Streptococcus_castoreus_GCF_000425025 genome\n",
      "2020-02-06 15:40:42,778 - Phyluce - INFO - Streptococcus_castoreus_GCF_000425025: 93 uces, 1 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-06 15:40:42,778 - Phyluce - INFO - --- Working on Streptococcus_constellatus_GCF_000463425 genome --\n",
      "2020-02-06 15:40:42,779 - Phyluce - INFO - Reading Streptococcus_constellatus_GCF_000463425 genome\n",
      "2020-02-06 15:40:43,240 - Phyluce - INFO - Streptococcus_constellatus_GCF_000463425: 100 uces, 0 dupes, 100 non-dupes, 17 orient drop, 1 length drop, 82 written\n",
      "2020-02-06 15:40:43,240 - Phyluce - INFO - ----- Working on Streptococcus_criceti_GCF_000187975 genome -----\n",
      "2020-02-06 15:40:43,241 - Phyluce - INFO - Reading Streptococcus_criceti_GCF_000187975 genome\n",
      "2020-02-06 15:40:43,731 - Phyluce - INFO - Streptococcus_criceti_GCF_000187975: 99 uces, 1 dupes, 98 non-dupes, 12 orient drop, 6 length drop, 80 written\n",
      "2020-02-06 15:40:43,731 - Phyluce - INFO - ---- Working on Streptococcus_cristatus_GCF_000385925 genome ----\n",
      "2020-02-06 15:40:43,746 - Phyluce - INFO - Reading Streptococcus_cristatus_GCF_000385925 genome\n",
      "2020-02-06 15:40:44,199 - Phyluce - INFO - Streptococcus_cristatus_GCF_000385925: 100 uces, 0 dupes, 100 non-dupes, 14 orient drop, 3 length drop, 83 written\n",
      "2020-02-06 15:40:44,199 - Phyluce - INFO - ----- Working on Streptococcus_cuniculi_GCF_001921845 genome ----\n",
      "2020-02-06 15:40:44,212 - Phyluce - INFO - Reading Streptococcus_cuniculi_GCF_001921845 genome\n",
      "2020-02-06 15:40:44,549 - Phyluce - INFO - Streptococcus_cuniculi_GCF_001921845: 98 uces, 4 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-06 15:40:44,549 - Phyluce - INFO - ---- Working on Streptococcus_danieliae_GCF_009767945 genome ----\n",
      "2020-02-06 15:40:44,550 - Phyluce - INFO - Reading Streptococcus_danieliae_GCF_009767945 genome\n",
      "2020-02-06 15:40:44,885 - Phyluce - INFO - Streptococcus_danieliae_GCF_009767945: 100 uces, 2 dupes, 98 non-dupes, 0 orient drop, 0 length drop, 98 written\n",
      "2020-02-06 15:40:44,885 - Phyluce - INFO - ---- Working on Streptococcus_devriesei_GCF_000423725 genome ----\n",
      "2020-02-06 15:40:44,885 - Phyluce - INFO - Reading Streptococcus_devriesei_GCF_000423725 genome\n",
      "2020-02-06 15:40:45,187 - Phyluce - INFO - Streptococcus_devriesei_GCF_000423725: 91 uces, 4 dupes, 87 non-dupes, 0 orient drop, 0 length drop, 87 written\n",
      "2020-02-06 15:40:45,188 - Phyluce - INFO - ---- Working on Streptococcus_didelphis_GCF_000380005 genome ----\n",
      "2020-02-06 15:40:45,209 - Phyluce - INFO - Reading Streptococcus_didelphis_GCF_000380005 genome\n",
      "2020-02-06 15:40:45,561 - Phyluce - INFO - Streptococcus_didelphis_GCF_000380005: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:45,561 - Phyluce - INFO - ------ Working on Streptococcus_downei_GCF_900459175 genome -----\n",
      "2020-02-06 15:40:45,562 - Phyluce - INFO - Reading Streptococcus_downei_GCF_900459175 genome\n",
      "2020-02-06 15:40:46,127 - Phyluce - INFO - Streptococcus_downei_GCF_900459175: 99 uces, 0 dupes, 99 non-dupes, 16 orient drop, 5 length drop, 78 written\n",
      "2020-02-06 15:40:46,127 - Phyluce - INFO - --- Working on Streptococcus_dysgalactiae_GCF_000317855 genome --\n",
      "2020-02-06 15:40:46,143 - Phyluce - INFO - Reading Streptococcus_dysgalactiae_GCF_000317855 genome\n",
      "2020-02-06 15:40:46,648 - Phyluce - INFO - Streptococcus_dysgalactiae_GCF_000317855: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:46,648 - Phyluce - INFO - ---- Working on Streptococcus_entericus_GCF_000380025 genome ----\n",
      "2020-02-06 15:40:46,648 - Phyluce - INFO - Reading Streptococcus_entericus_GCF_000380025 genome\n",
      "2020-02-06 15:40:46,968 - Phyluce - INFO - Streptococcus_entericus_GCF_000380025: 96 uces, 3 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:40:46,968 - Phyluce - INFO - ------- Working on Streptococcus_equi_GCF_000026605 genome ------\n",
      "2020-02-06 15:40:46,968 - Phyluce - INFO - Reading Streptococcus_equi_GCF_000026605 genome\n",
      "2020-02-06 15:40:47,463 - Phyluce - INFO - Streptococcus_equi_GCF_000026605: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:47,463 - Phyluce - INFO - ----- Working on Streptococcus_equinus_GCF_000964315 genome -----\n",
      "2020-02-06 15:40:47,473 - Phyluce - INFO - Reading Streptococcus_equinus_GCF_000964315 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 15:40:48,061 - Phyluce - INFO - Streptococcus_equinus_GCF_000964315: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:48,061 - Phyluce - INFO - ------ Working on Streptococcus_ferus_GCF_000372425 genome ------\n",
      "2020-02-06 15:40:48,061 - Phyluce - INFO - Reading Streptococcus_ferus_GCF_000372425 genome\n",
      "2020-02-06 15:40:48,393 - Phyluce - INFO - Streptococcus_ferus_GCF_000372425: 99 uces, 3 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:40:48,394 - Phyluce - INFO - --- Working on Streptococcus_gallolyticus_GCF_002000985 genome --\n",
      "2020-02-06 15:40:48,394 - Phyluce - INFO - Reading Streptococcus_gallolyticus_GCF_002000985 genome\n",
      "2020-02-06 15:40:48,942 - Phyluce - INFO - Streptococcus_gallolyticus_GCF_002000985: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:48,943 - Phyluce - INFO - ----- Working on Streptococcus_gordonii_GCF_000017005 genome ----\n",
      "2020-02-06 15:40:48,948 - Phyluce - INFO - Reading Streptococcus_gordonii_GCF_000017005 genome\n",
      "2020-02-06 15:40:49,414 - Phyluce - INFO - Streptococcus_gordonii_GCF_000017005: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:40:49,414 - Phyluce - INFO - ---- Working on Streptococcus_halichoeri_GCF_009870755 genome ---\n",
      "2020-02-06 15:40:49,415 - Phyluce - INFO - Reading Streptococcus_halichoeri_GCF_009870755 genome\n",
      "2020-02-06 15:40:49,745 - Phyluce - INFO - Streptococcus_halichoeri_GCF_009870755: 99 uces, 3 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:40:49,745 - Phyluce - INFO - ---- Working on Streptococcus_halitosis_GCF_003143695 genome ----\n",
      "2020-02-06 15:40:49,745 - Phyluce - INFO - Reading Streptococcus_halitosis_GCF_003143695 genome\n",
      "2020-02-06 15:40:50,079 - Phyluce - INFO - Streptococcus_halitosis_GCF_003143695: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:50,079 - Phyluce - INFO - --- Working on Streptococcus_halotolerans_GCF_001598035 genome --\n",
      "2020-02-06 15:40:50,079 - Phyluce - INFO - Reading Streptococcus_halotolerans_GCF_001598035 genome\n",
      "2020-02-06 15:40:50,613 - Phyluce - INFO - Streptococcus_halotolerans_GCF_001598035: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:50,613 - Phyluce - INFO - ------ Working on Streptococcus_henryi_GCF_000376985 genome -----\n",
      "2020-02-06 15:40:50,632 - Phyluce - INFO - Reading Streptococcus_henryi_GCF_000376985 genome\n",
      "2020-02-06 15:40:50,961 - Phyluce - INFO - Streptococcus_henryi_GCF_000376985: 97 uces, 0 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-06 15:40:50,961 - Phyluce - INFO - --- Working on Streptococcus_himalayensis_GCF_001708305 genome --\n",
      "2020-02-06 15:40:50,962 - Phyluce - INFO - Reading Streptococcus_himalayensis_GCF_001708305 genome\n",
      "2020-02-06 15:40:51,419 - Phyluce - INFO - Streptococcus_himalayensis_GCF_001708305: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n",
      "2020-02-06 15:40:51,419 - Phyluce - INFO - -- Working on Streptococcus_hongkongensis_GCF_000785845 genome --\n",
      "2020-02-06 15:40:51,420 - Phyluce - INFO - Reading Streptococcus_hongkongensis_GCF_000785845 genome\n",
      "2020-02-06 15:40:51,788 - Phyluce - INFO - Streptococcus_hongkongensis_GCF_000785845: 100 uces, 9 dupes, 91 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-06 15:40:51,788 - Phyluce - INFO - - Working on Streptococcus_hyointestinalis_GCF_900459405 genome -\n",
      "2020-02-06 15:40:51,804 - Phyluce - INFO - Reading Streptococcus_hyointestinalis_GCF_900459405 genome\n",
      "2020-02-06 15:40:52,356 - Phyluce - INFO - Streptococcus_hyointestinalis_GCF_900459405: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 4 length drop, 81 written\n",
      "2020-02-06 15:40:52,357 - Phyluce - INFO - --- Working on Streptococcus_hyovaginalis_GCF_000420785 genome --\n",
      "2020-02-06 15:40:52,364 - Phyluce - INFO - Reading Streptococcus_hyovaginalis_GCF_000420785 genome\n",
      "2020-02-06 15:40:52,696 - Phyluce - INFO - Streptococcus_hyovaginalis_GCF_000420785: 97 uces, 4 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:40:52,696 - Phyluce - INFO - ----- Working on Streptococcus_ictaluri_GCF_000188015 genome ----\n",
      "2020-02-06 15:40:52,697 - Phyluce - INFO - Reading Streptococcus_ictaluri_GCF_000188015 genome\n",
      "2020-02-06 15:40:53,109 - Phyluce - INFO - Streptococcus_ictaluri_GCF_000188015: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 16 length drop, 84 written\n",
      "2020-02-06 15:40:53,109 - Phyluce - INFO - --- Working on Streptococcus_infantarius_GCF_000246835 genome ---\n",
      "2020-02-06 15:40:53,109 - Phyluce - INFO - Reading Streptococcus_infantarius_GCF_000246835 genome\n",
      "2020-02-06 15:40:53,643 - Phyluce - INFO - Streptococcus_infantarius_GCF_000246835: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:40:53,643 - Phyluce - INFO - ----- Working on Streptococcus_infantis_GCF_000187465 genome ----\n",
      "2020-02-06 15:40:53,643 - Phyluce - INFO - Reading Streptococcus_infantis_GCF_000187465 genome\n",
      "2020-02-06 15:40:53,975 - Phyluce - INFO - Streptococcus_infantis_GCF_000187465: 99 uces, 1 dupes, 98 non-dupes, 3 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:40:53,976 - Phyluce - INFO - ------ Working on Streptococcus_iniae_GCF_000831485 genome ------\n",
      "2020-02-06 15:40:53,996 - Phyluce - INFO - Reading Streptococcus_iniae_GCF_000831485 genome\n",
      "2020-02-06 15:40:54,499 - Phyluce - INFO - Streptococcus_iniae_GCF_000831485: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 5 length drop, 82 written\n",
      "2020-02-06 15:40:54,500 - Phyluce - INFO - --- Working on Streptococcus_intermedius_GCF_000463355 genome ---\n",
      "2020-02-06 15:40:54,500 - Phyluce - INFO - Reading Streptococcus_intermedius_GCF_000463355 genome\n",
      "2020-02-06 15:40:54,954 - Phyluce - INFO - Streptococcus_intermedius_GCF_000463355: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:40:54,954 - Phyluce - INFO - --- Working on Streptococcus_lutetiensis_GCF_900475675 genome ---\n",
      "2020-02-06 15:40:54,963 - Phyluce - INFO - Reading Streptococcus_lutetiensis_GCF_900475675 genome\n",
      "2020-02-06 15:40:55,547 - Phyluce - INFO - Streptococcus_lutetiensis_GCF_900475675: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 19 length drop, 81 written\n",
      "2020-02-06 15:40:55,548 - Phyluce - INFO - ----- Working on Streptococcus_macacae_GCF_000187995 genome -----\n",
      "2020-02-06 15:40:55,548 - Phyluce - INFO - Reading Streptococcus_macacae_GCF_000187995 genome\n",
      "2020-02-06 15:40:56,059 - Phyluce - INFO - Streptococcus_macacae_GCF_000187995: 100 uces, 0 dupes, 100 non-dupes, 18 orient drop, 3 length drop, 79 written\n",
      "2020-02-06 15:40:56,059 - Phyluce - INFO - --- Working on Streptococcus_macedonicus_GCF_000283635 genome ---\n",
      "2020-02-06 15:40:56,060 - Phyluce - INFO - Reading Streptococcus_macedonicus_GCF_000283635 genome\n",
      "2020-02-06 15:40:56,601 - Phyluce - INFO - Streptococcus_macedonicus_GCF_000283635: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 19 length drop, 81 written\n",
      "2020-02-06 15:40:56,601 - Phyluce - INFO - -- Working on Streptococcus_marimammalium_GCF_000380045 genome --\n",
      "2020-02-06 15:40:56,613 - Phyluce - INFO - Reading Streptococcus_marimammalium_GCF_000380045 genome\n",
      "2020-02-06 15:40:56,976 - Phyluce - INFO - Streptococcus_marimammalium_GCF_000380045: 99 uces, 1 dupes, 98 non-dupes, 0 orient drop, 3 length drop, 95 written\n",
      "2020-02-06 15:40:56,977 - Phyluce - INFO - ----- Working on Streptococcus_marmotae_GCF_001623565 genome ----\n",
      "2020-02-06 15:40:56,977 - Phyluce - INFO - Reading Streptococcus_marmotae_GCF_001623565 genome\n",
      "2020-02-06 15:40:57,441 - Phyluce - INFO - Streptococcus_marmotae_GCF_001623565: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 2 length drop, 83 written\n",
      "2020-02-06 15:40:57,441 - Phyluce - INFO - --- Working on Streptococcus_massiliensis_GCF_000380065 genome --\n",
      "2020-02-06 15:40:57,441 - Phyluce - INFO - Reading Streptococcus_massiliensis_GCF_000380065 genome\n",
      "2020-02-06 15:40:57,743 - Phyluce - INFO - Streptococcus_massiliensis_GCF_000380065: 93 uces, 1 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:40:57,743 - Phyluce - INFO - ----- Working on Streptococcus_merionis_GCF_000380085 genome ----\n",
      "2020-02-06 15:40:57,744 - Phyluce - INFO - Reading Streptococcus_merionis_GCF_000380085 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 15:40:58,080 - Phyluce - INFO - Streptococcus_merionis_GCF_000380085: 99 uces, 5 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-06 15:40:58,080 - Phyluce - INFO - ----- Working on Streptococcus_milleri_GCF_900636715 genome -----\n",
      "2020-02-06 15:40:58,081 - Phyluce - INFO - Reading Streptococcus_milleri_GCF_900636715 genome\n",
      "2020-02-06 15:40:58,537 - Phyluce - INFO - Streptococcus_milleri_GCF_900636715: 100 uces, 0 dupes, 100 non-dupes, 17 orient drop, 1 length drop, 82 written\n",
      "2020-02-06 15:40:58,537 - Phyluce - INFO - ------ Working on Streptococcus_minor_GCF_000377005 genome ------\n",
      "2020-02-06 15:40:58,538 - Phyluce - INFO - Reading Streptococcus_minor_GCF_000377005 genome\n",
      "2020-02-06 15:40:58,868 - Phyluce - INFO - Streptococcus_minor_GCF_000377005: 99 uces, 5 dupes, 94 non-dupes, 0 orient drop, 0 length drop, 94 written\n",
      "2020-02-06 15:40:58,868 - Phyluce - INFO - ------ Working on Streptococcus_mitis_GCF_000027165 genome ------\n",
      "2020-02-06 15:40:58,868 - Phyluce - INFO - Reading Streptococcus_mitis_GCF_000027165 genome\n",
      "2020-02-06 15:40:59,328 - Phyluce - INFO - Streptococcus_mitis_GCF_000027165: 100 uces, 0 dupes, 100 non-dupes, 2 orient drop, 17 length drop, 81 written\n",
      "2020-02-06 15:40:59,328 - Phyluce - INFO - ------ Working on Streptococcus_mutans_GCF_000007465 genome -----\n",
      "2020-02-06 15:40:59,329 - Phyluce - INFO - Reading Streptococcus_mutans_GCF_000007465 genome\n",
      "2020-02-06 15:40:59,833 - Phyluce - INFO - Streptococcus_mutans_GCF_000007465: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:40:59,833 - Phyluce - INFO - ------ Working on Streptococcus_oralis_GCF_900637025 genome -----\n",
      "2020-02-06 15:40:59,839 - Phyluce - INFO - Reading Streptococcus_oralis_GCF_900637025 genome\n",
      "2020-02-06 15:41:00,315 - Phyluce - INFO - Streptococcus_oralis_GCF_900637025: 100 uces, 0 dupes, 100 non-dupes, 19 orient drop, 0 length drop, 81 written\n",
      "2020-02-06 15:41:00,315 - Phyluce - INFO - ---- Working on Streptococcus_orisasini_GCF_001431045 genome ----\n",
      "2020-02-06 15:41:00,316 - Phyluce - INFO - Reading Streptococcus_orisasini_GCF_001431045 genome\n",
      "2020-02-06 15:41:00,628 - Phyluce - INFO - Streptococcus_orisasini_GCF_001431045: 97 uces, 1 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:41:00,628 - Phyluce - INFO - ---- Working on Streptococcus_orisratti_GCF_000380105 genome ----\n",
      "2020-02-06 15:41:00,654 - Phyluce - INFO - Reading Streptococcus_orisratti_GCF_000380105 genome\n",
      "2020-02-06 15:41:00,982 - Phyluce - INFO - Streptococcus_orisratti_GCF_000380105: 98 uces, 2 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:41:00,982 - Phyluce - INFO - ------- Working on Streptococcus_ovis_GCF_000380125 genome ------\n",
      "2020-02-06 15:41:00,982 - Phyluce - INFO - Reading Streptococcus_ovis_GCF_000380125 genome\n",
      "2020-02-06 15:41:01,337 - Phyluce - INFO - Streptococcus_ovis_GCF_000380125: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 2 length drop, 91 written\n",
      "2020-02-06 15:41:01,337 - Phyluce - INFO - --- Working on Streptococcus_pantholopis_GCF_001642085 genome ---\n",
      "2020-02-06 15:41:01,337 - Phyluce - INFO - Reading Streptococcus_pantholopis_GCF_001642085 genome\n",
      "2020-02-06 15:41:01,834 - Phyluce - INFO - Streptococcus_pantholopis_GCF_001642085: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:41:01,835 - Phyluce - INFO - -- Working on Streptococcus_parasanguinis_GCF_000164675 genome --\n",
      "2020-02-06 15:41:01,845 - Phyluce - INFO - Reading Streptococcus_parasanguinis_GCF_000164675 genome\n",
      "2020-02-06 15:41:02,310 - Phyluce - INFO - Streptococcus_parasanguinis_GCF_000164675: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:41:02,310 - Phyluce - INFO - ----- Working on Streptococcus_parasuis_GCF_004283785 genome ----\n",
      "2020-02-06 15:41:02,345 - Phyluce - INFO - Reading Streptococcus_parasuis_GCF_004283785 genome\n",
      "2020-02-06 15:41:02,688 - Phyluce - INFO - Streptococcus_parasuis_GCF_004283785: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:02,688 - Phyluce - INFO - ---- Working on Streptococcus_parauberis_GCF_000213825 genome ---\n",
      "2020-02-06 15:41:02,688 - Phyluce - INFO - Reading Streptococcus_parauberis_GCF_000213825 genome\n",
      "2020-02-06 15:41:03,188 - Phyluce - INFO - Streptococcus_parauberis_GCF_000213825: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 2 length drop, 83 written\n",
      "2020-02-06 15:41:03,188 - Phyluce - INFO - --- Working on Streptococcus_pasteurianus_GCF_900478025 genome --\n",
      "2020-02-06 15:41:03,189 - Phyluce - INFO - Reading Streptococcus_pasteurianus_GCF_900478025 genome\n",
      "2020-02-06 15:41:03,733 - Phyluce - INFO - Streptococcus_pasteurianus_GCF_900478025: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 20 length drop, 80 written\n",
      "2020-02-06 15:41:03,734 - Phyluce - INFO - ---- Working on Streptococcus_penaeicida_GCF_002887775 genome ---\n",
      "2020-02-06 15:41:03,734 - Phyluce - INFO - Reading Streptococcus_penaeicida_GCF_002887775 genome\n",
      "2020-02-06 15:41:04,131 - Phyluce - INFO - Streptococcus_penaeicida_GCF_002887775: 100 uces, 7 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:04,132 - Phyluce - INFO - ----- Working on Streptococcus_peroris_GCF_000187585 genome -----\n",
      "2020-02-06 15:41:04,132 - Phyluce - INFO - Reading Streptococcus_peroris_GCF_000187585 genome\n",
      "2020-02-06 15:41:04,453 - Phyluce - INFO - Streptococcus_peroris_GCF_000187585: 96 uces, 2 dupes, 94 non-dupes, 1 orient drop, 2 length drop, 91 written\n",
      "2020-02-06 15:41:04,453 - Phyluce - INFO - ---- Working on Streptococcus_pharyngis_GCF_007859195 genome ----\n",
      "2020-02-06 15:41:04,453 - Phyluce - INFO - Reading Streptococcus_pharyngis_GCF_007859195 genome\n",
      "2020-02-06 15:41:04,798 - Phyluce - INFO - Streptococcus_pharyngis_GCF_007859195: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:41:04,798 - Phyluce - INFO - ------ Working on Streptococcus_phocae_GCF_000772915 genome -----\n",
      "2020-02-06 15:41:04,799 - Phyluce - INFO - Reading Streptococcus_phocae_GCF_000772915 genome\n",
      "2020-02-06 15:41:05,140 - Phyluce - INFO - Streptococcus_phocae_GCF_000772915: 99 uces, 6 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:05,141 - Phyluce - INFO - -- Working on Streptococcus_pluranimalium_GCF_003352995 genome --\n",
      "2020-02-06 15:41:05,142 - Phyluce - INFO - Reading Streptococcus_pluranimalium_GCF_003352995 genome\n",
      "2020-02-06 15:41:05,691 - Phyluce - INFO - Streptococcus_pluranimalium_GCF_003352995: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:41:05,691 - Phyluce - INFO - --- Working on Streptococcus_plurextorum_GCF_000423745 genome ---\n",
      "2020-02-06 15:41:05,691 - Phyluce - INFO - Reading Streptococcus_plurextorum_GCF_000423745 genome\n",
      "2020-02-06 15:41:06,011 - Phyluce - INFO - Streptococcus_plurextorum_GCF_000423745: 96 uces, 4 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:41:06,011 - Phyluce - INFO - ------ Working on Streptococcus_porci_GCF_000423765 genome ------\n",
      "2020-02-06 15:41:06,012 - Phyluce - INFO - Reading Streptococcus_porci_GCF_000423765 genome\n",
      "2020-02-06 15:41:06,322 - Phyluce - INFO - Streptococcus_porci_GCF_000423765: 94 uces, 1 dupes, 93 non-dupes, 0 orient drop, 0 length drop, 93 written\n",
      "2020-02-06 15:41:06,322 - Phyluce - INFO - ----- Working on Streptococcus_porcinus_GCF_900475415 genome ----\n",
      "2020-02-06 15:41:06,330 - Phyluce - INFO - Reading Streptococcus_porcinus_GCF_900475415 genome\n",
      "2020-02-06 15:41:06,879 - Phyluce - INFO - Streptococcus_porcinus_GCF_900475415: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n",
      "2020-02-06 15:41:06,879 - Phyluce - INFO - - Working on Streptococcus_pseudopneumoniae_GCF_000221985 genome \n",
      "2020-02-06 15:41:06,891 - Phyluce - INFO - Reading Streptococcus_pseudopneumoniae_GCF_000221985 genome\n",
      "2020-02-06 15:41:07,232 - Phyluce - INFO - Streptococcus_pseudopneumoniae_GCF_000221985: 99 uces, 0 dupes, 99 non-dupes, 0 orient drop, 4 length drop, 95 written\n",
      "2020-02-06 15:41:07,232 - Phyluce - INFO - -- Working on Streptococcus_pseudoporcinus_GCF_000188035 genome -\n",
      "2020-02-06 15:41:07,233 - Phyluce - INFO - Reading Streptococcus_pseudoporcinus_GCF_000188035 genome\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-02-06 15:41:07,723 - Phyluce - INFO - Streptococcus_pseudoporcinus_GCF_000188035: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n",
      "2020-02-06 15:41:07,723 - Phyluce - INFO - ----- Working on Streptococcus_pyogenes_GCF_000006785 genome ----\n",
      "2020-02-06 15:41:07,724 - Phyluce - INFO - Reading Streptococcus_pyogenes_GCF_000006785 genome\n",
      "2020-02-06 15:41:08,250 - Phyluce - INFO - Streptococcus_pyogenes_GCF_000006785: 100 uces, 0 dupes, 100 non-dupes, 14 orient drop, 4 length drop, 82 written\n",
      "2020-02-06 15:41:08,250 - Phyluce - INFO - ------ Working on Streptococcus_ratti_GCF_000286075 genome ------\n",
      "2020-02-06 15:41:08,251 - Phyluce - INFO - Reading Streptococcus_ratti_GCF_000286075 genome\n",
      "2020-02-06 15:41:08,565 - Phyluce - INFO - Streptococcus_ratti_GCF_000286075: 95 uces, 0 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:41:08,565 - Phyluce - INFO - ---- Working on Streptococcus_respraculi_GCF_003595525 genome ---\n",
      "2020-02-06 15:41:08,565 - Phyluce - INFO - Reading Streptococcus_respraculi_GCF_003595525 genome\n",
      "2020-02-06 15:41:09,018 - Phyluce - INFO - Streptococcus_respraculi_GCF_003595525: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 5 length drop, 82 written\n",
      "2020-02-06 15:41:09,018 - Phyluce - INFO - ----- Working on Streptococcus_rubneri_GCF_004785935 genome -----\n",
      "2020-02-06 15:41:09,019 - Phyluce - INFO - Reading Streptococcus_rubneri_GCF_004785935 genome\n",
      "2020-02-06 15:41:09,345 - Phyluce - INFO - Streptococcus_rubneri_GCF_004785935: 99 uces, 2 dupes, 97 non-dupes, 0 orient drop, 0 length drop, 97 written\n",
      "2020-02-06 15:41:09,346 - Phyluce - INFO - --- Working on Streptococcus_ruminantium_GCF_003609975 genome ---\n",
      "2020-02-06 15:41:09,346 - Phyluce - INFO - Reading Streptococcus_ruminantium_GCF_003609975 genome\n",
      "2020-02-06 15:41:09,800 - Phyluce - INFO - Streptococcus_ruminantium_GCF_003609975: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:41:09,800 - Phyluce - INFO - ---- Working on Streptococcus_salivarius_GCF_000785515 genome ---\n",
      "2020-02-06 15:41:09,800 - Phyluce - INFO - Reading Streptococcus_salivarius_GCF_000785515 genome\n",
      "2020-02-06 15:41:10,338 - Phyluce - INFO - Streptococcus_salivarius_GCF_000785515: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 6 length drop, 81 written\n",
      "2020-02-06 15:41:10,338 - Phyluce - INFO - ---- Working on Streptococcus_sanguinis_GCF_000014205 genome ----\n",
      "2020-02-06 15:41:10,346 - Phyluce - INFO - Reading Streptococcus_sanguinis_GCF_000014205 genome\n",
      "2020-02-06 15:41:10,806 - Phyluce - INFO - Streptococcus_sanguinis_GCF_000014205: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:41:10,806 - Phyluce - INFO - ----- Working on Streptococcus_sinensis_GCF_000767835 genome ----\n",
      "2020-02-06 15:41:10,814 - Phyluce - INFO - Reading Streptococcus_sinensis_GCF_000767835 genome\n",
      "2020-02-06 15:41:11,169 - Phyluce - INFO - Streptococcus_sinensis_GCF_000767835: 100 uces, 4 dupes, 96 non-dupes, 0 orient drop, 0 length drop, 96 written\n",
      "2020-02-06 15:41:11,169 - Phyluce - INFO - ----- Working on Streptococcus_sobrinus_GCF_000686605 genome ----\n",
      "2020-02-06 15:41:11,169 - Phyluce - INFO - Reading Streptococcus_sobrinus_GCF_000686605 genome\n",
      "2020-02-06 15:41:11,497 - Phyluce - INFO - Streptococcus_sobrinus_GCF_000686605: 96 uces, 4 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 92 written\n",
      "2020-02-06 15:41:11,497 - Phyluce - INFO - -------- Working on Streptococcus_sp_GCF_003086355 genome -------\n",
      "2020-02-06 15:41:11,510 - Phyluce - INFO - Reading Streptococcus_sp_GCF_003086355 genome\n",
      "2020-02-06 15:41:12,019 - Phyluce - INFO - Streptococcus_sp_GCF_003086355: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 2 length drop, 82 written\n",
      "2020-02-06 15:41:12,019 - Phyluce - INFO - ------- Working on Streptococcus_suis_GCF_000026745 genome ------\n",
      "2020-02-06 15:41:12,020 - Phyluce - INFO - Reading Streptococcus_suis_GCF_000026745 genome\n",
      "2020-02-06 15:41:12,482 - Phyluce - INFO - Streptococcus_suis_GCF_000026745: 100 uces, 0 dupes, 100 non-dupes, 1 orient drop, 18 length drop, 81 written\n",
      "2020-02-06 15:41:12,482 - Phyluce - INFO - --- Working on Streptococcus_thermophilus_GCF_000253395 genome --\n",
      "2020-02-06 15:41:12,483 - Phyluce - INFO - Reading Streptococcus_thermophilus_GCF_000253395 genome\n",
      "2020-02-06 15:41:13,022 - Phyluce - INFO - Streptococcus_thermophilus_GCF_000253395: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 6 length drop, 81 written\n",
      "2020-02-06 15:41:13,023 - Phyluce - INFO - --- Working on Streptococcus_thoraltensis_GCF_000380145 genome --\n",
      "2020-02-06 15:41:13,023 - Phyluce - INFO - Reading Streptococcus_thoraltensis_GCF_000380145 genome\n",
      "2020-02-06 15:41:13,344 - Phyluce - INFO - Streptococcus_thoraltensis_GCF_000380145: 95 uces, 3 dupes, 92 non-dupes, 0 orient drop, 0 length drop, 91 written\n",
      "2020-02-06 15:41:13,344 - Phyluce - INFO - ---- Working on Streptococcus_timonensis_GCF_900095845 genome ---\n",
      "2020-02-06 15:41:13,345 - Phyluce - INFO - Reading Streptococcus_timonensis_GCF_900095845 genome\n",
      "2020-02-06 15:41:13,878 - Phyluce - INFO - Streptococcus_timonensis_GCF_900095845: 100 uces, 15 dupes, 85 non-dupes, 3 orient drop, 0 length drop, 82 written\n",
      "2020-02-06 15:41:13,878 - Phyluce - INFO - --- Working on Streptococcus_troglodytae_GCF_002355215 genome ---\n",
      "2020-02-06 15:41:13,879 - Phyluce - INFO - Reading Streptococcus_troglodytae_GCF_002355215 genome\n",
      "2020-02-06 15:41:14,389 - Phyluce - INFO - Streptococcus_troglodytae_GCF_002355215: 100 uces, 0 dupes, 100 non-dupes, 18 orient drop, 0 length drop, 82 written\n",
      "2020-02-06 15:41:14,389 - Phyluce - INFO - ------ Working on Streptococcus_uberis_GCF_000009545 genome -----\n",
      "2020-02-06 15:41:14,390 - Phyluce - INFO - Reading Streptococcus_uberis_GCF_000009545 genome\n",
      "2020-02-06 15:41:14,883 - Phyluce - INFO - Streptococcus_uberis_GCF_000009545: 100 uces, 0 dupes, 100 non-dupes, 15 orient drop, 3 length drop, 82 written\n",
      "2020-02-06 15:41:14,884 - Phyluce - INFO - ----- Working on Streptococcus_urinalis_GCF_000188055 genome ----\n",
      "2020-02-06 15:41:14,884 - Phyluce - INFO - Reading Streptococcus_urinalis_GCF_000188055 genome\n",
      "2020-02-06 15:41:15,375 - Phyluce - INFO - Streptococcus_urinalis_GCF_000188055: 100 uces, 0 dupes, 100 non-dupes, 16 orient drop, 1 length drop, 83 written\n",
      "2020-02-06 15:41:15,375 - Phyluce - INFO - ------ Working on Streptococcus_varani_GCF_001375655 genome -----\n",
      "2020-02-06 15:41:15,375 - Phyluce - INFO - Reading Streptococcus_varani_GCF_001375655 genome\n",
      "2020-02-06 15:41:15,715 - Phyluce - INFO - Streptococcus_varani_GCF_001375655: 100 uces, 5 dupes, 95 non-dupes, 0 orient drop, 0 length drop, 95 written\n",
      "2020-02-06 15:41:15,715 - Phyluce - INFO - --- Working on Streptococcus_vestibularis_GCF_900636445 genome --\n",
      "2020-02-06 15:41:15,716 - Phyluce - INFO - Reading Streptococcus_vestibularis_GCF_900636445 genome\n",
      "2020-02-06 15:41:16,261 - Phyluce - INFO - Streptococcus_vestibularis_GCF_900636445: 100 uces, 0 dupes, 100 non-dupes, 13 orient drop, 7 length drop, 80 written\n",
      "2020-02-06 15:41:16,261 - Phyluce - INFO - ----- Working on Streptococcus_viridans_GCF_900636365 genome ----\n",
      "2020-02-06 15:41:16,262 - Phyluce - INFO - Reading Streptococcus_viridans_GCF_900636365 genome\n",
      "2020-02-06 15:41:16,724 - Phyluce - INFO - Streptococcus_viridans_GCF_900636365: 100 uces, 0 dupes, 100 non-dupes, 0 orient drop, 18 length drop, 82 written\n",
      "2020-02-06 15:41:16,724 - Phyluce - INFO - ---- Working on Streptococcus_pneumoniae_GCF_000007045 genome ---\n",
      "2020-02-06 15:41:16,725 - Phyluce - INFO - Reading Streptococcus_pneumoniae_GCF_000007045 genome\n",
      "2020-02-06 15:41:17,179 - Phyluce - INFO - Streptococcus_pneumoniae_GCF_000007045: 100 uces, 0 dupes, 100 non-dupes, 12 orient drop, 5 length drop, 83 written\n"
     ]
    }
   ],
   "source": [
    "o_dir = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "o_scaff_conf = o_dir + group + \"_genome.conf\"\n",
    "\n",
    "in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "with open(o_scaff_conf, \"w\") as outfile:\n",
    "\n",
    "    #print the header line\n",
    "    outfile.write(\"[scaffolds]\\n\")\n",
    "\n",
    "    #print bedinfo for each species for the genus\n",
    "    for sample in all_taxa:\n",
    "        in_2bit_dest  = \"results/phyluce/\" + group + \"/cleaned_genomes/\" + sample + \"/\" + sample + \".2bit\"\n",
    "\n",
    "        outfile.write(sample + \":\" + in_2bit_dest + \"\\n\")     \n",
    "\n",
    "\n",
    "i_scaff_conf   = o_scaff_conf\n",
    "i_lastz        = \"results/phyluce/\" + group + \"/validate_intervals/lastz\"\n",
    "i_bp_int       = str(120)\n",
    "i_name_pattern = \"\\\"\" + reference_taxon + \"_+\" + str(num_taxa) + \"_temp_probes.fas_v_{}.lastz.clean\\\"\"\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "o_probe_dir    = o_dir + \"probe_fasta\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_slice_sequence_from_genomes \"\n",
    "                      \"--conf {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probes {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--name-pattern {} \"\n",
    "                      \"--output {};\").format(i_scaff_conf, \n",
    "                                             i_lastz, \n",
    "                                             i_bp_int,\n",
    "                                             i_probe_prefix,\n",
    "                                             i_name_pattern, \n",
    "                                             o_probe_dir)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_multi_fasta_table --fastas results/phyluce/streptococcus/extract_probes_from_group/probe_fasta --output results/phyluce/streptococcus/extract_probes_from_group/multifastas.sqlite --base-taxon Streptococcus_pneumoniae_GCF_000007045;\n",
      "streptococcus_acidominimus_gcf_900187245.\n",
      "streptococcus_agalactiae_gcf_000007265.\n",
      "streptococcus_alactolyticus_gcf_009695625.\n",
      "streptococcus_anginosus_gcf_000463505.\n",
      "streptococcus_australis_gcf_900476055.\n",
      "streptococcus_azizii_gcf_001984715.\n",
      "streptococcus_bovimastitidis_gcf_001885095.\n",
      "streptococcus_caballi_gcf_000379985.\n",
      "streptococcus_canis_gcf_900636575.\n",
      "streptococcus_castoreus_gcf_000425025.\n",
      "streptococcus_constellatus_gcf_000463425.\n",
      "streptococcus_criceti_gcf_000187975.\n",
      "streptococcus_cristatus_gcf_000385925.\n",
      "streptococcus_cuniculi_gcf_001921845.\n",
      "streptococcus_danieliae_gcf_009767945.\n",
      "streptococcus_devriesei_gcf_000423725.\n",
      "streptococcus_didelphis_gcf_000380005.\n",
      "streptococcus_downei_gcf_900459175.\n",
      "streptococcus_dysgalactiae_gcf_000317855.\n",
      "streptococcus_entericus_gcf_000380025.\n",
      "streptococcus_equi_gcf_000026605.\n",
      "streptococcus_equinus_gcf_000964315.\n",
      "streptococcus_ferus_gcf_000372425.\n",
      "streptococcus_gallolyticus_gcf_002000985.\n",
      "streptococcus_gordonii_gcf_000017005.\n",
      "streptococcus_halichoeri_gcf_009870755.\n",
      "streptococcus_halitosis_gcf_003143695.\n",
      "streptococcus_halotolerans_gcf_001598035.\n",
      "streptococcus_henryi_gcf_000376985.\n",
      "streptococcus_himalayensis_gcf_001708305.\n",
      "streptococcus_hongkongensis_gcf_000785845.\n",
      "streptococcus_hyointestinalis_gcf_900459405.\n",
      "streptococcus_hyovaginalis_gcf_000420785.\n",
      "streptococcus_ictaluri_gcf_000188015.\n",
      "streptococcus_infantarius_gcf_000246835.\n",
      "streptococcus_infantis_gcf_000187465.\n",
      "streptococcus_iniae_gcf_000831485.\n",
      "streptococcus_intermedius_gcf_000463355.\n",
      "streptococcus_lutetiensis_gcf_900475675.\n",
      "streptococcus_macacae_gcf_000187995.\n",
      "streptococcus_macedonicus_gcf_000283635.\n",
      "streptococcus_marimammalium_gcf_000380045.\n",
      "streptococcus_marmotae_gcf_001623565.\n",
      "streptococcus_massiliensis_gcf_000380065.\n",
      "streptococcus_merionis_gcf_000380085.\n",
      "streptococcus_milleri_gcf_900636715.\n",
      "streptococcus_minor_gcf_000377005.\n",
      "streptococcus_mitis_gcf_000027165.\n",
      "streptococcus_mutans_gcf_000007465.\n",
      "streptococcus_oralis_gcf_900637025.\n",
      "streptococcus_orisasini_gcf_001431045.\n",
      "streptococcus_orisratti_gcf_000380105.\n",
      "streptococcus_ovis_gcf_000380125.\n",
      "streptococcus_pantholopis_gcf_001642085.\n",
      "streptococcus_parasanguinis_gcf_000164675.\n",
      "streptococcus_parasuis_gcf_004283785.\n",
      "streptococcus_parauberis_gcf_000213825.\n",
      "streptococcus_pasteurianus_gcf_900478025.\n",
      "streptococcus_penaeicida_gcf_002887775.\n",
      "streptococcus_peroris_gcf_000187585.\n",
      "streptococcus_pharyngis_gcf_007859195.\n",
      "streptococcus_phocae_gcf_000772915.\n",
      "streptococcus_pluranimalium_gcf_003352995.\n",
      "streptococcus_plurextorum_gcf_000423745.\n",
      "streptococcus_porci_gcf_000423765.\n",
      "streptococcus_porcinus_gcf_900475415.\n",
      "streptococcus_pseudopneumoniae_gcf_000221985.\n",
      "streptococcus_pseudoporcinus_gcf_000188035.\n",
      "streptococcus_pyogenes_gcf_000006785.\n",
      "streptococcus_ratti_gcf_000286075.\n",
      "streptococcus_respraculi_gcf_003595525.\n",
      "streptococcus_rubneri_gcf_004785935.\n",
      "streptococcus_ruminantium_gcf_003609975.\n",
      "streptococcus_salivarius_gcf_000785515.\n",
      "streptococcus_sanguinis_gcf_000014205.\n",
      "streptococcus_sinensis_gcf_000767835.\n",
      "streptococcus_sobrinus_gcf_000686605.\n",
      "streptococcus_sp_gcf_003086355.\n",
      "streptococcus_suis_gcf_000026745.\n",
      "streptococcus_thermophilus_gcf_000253395.\n",
      "streptococcus_thoraltensis_gcf_000380145.\n",
      "streptococcus_timonensis_gcf_900095845.\n",
      "streptococcus_troglodytae_gcf_002355215.\n",
      "streptococcus_uberis_gcf_000009545.\n",
      "streptococcus_urinalis_gcf_000188055.\n",
      "streptococcus_varani_gcf_001375655.\n",
      "streptococcus_vestibularis_gcf_900636445.\n",
      "streptococcus_viridans_gcf_900636365.\n",
      "streptococcus_pneumoniae_gcf_000007045.\n",
      "Creating database\n",
      "Inserting results\n",
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/streptococcus/extract_probes_from_group/multifastas.sqlite --base-taxon Streptococcus_pneumoniae_GCF_000007045;\n",
      "Loci shared by 0 taxa:\t100.0\n",
      "Loci shared by 1 taxa:\t100.0\n",
      "Loci shared by 2 taxa:\t100.0\n",
      "Loci shared by 3 taxa:\t100.0\n",
      "Loci shared by 4 taxa:\t100.0\n",
      "Loci shared by 5 taxa:\t100.0\n",
      "Loci shared by 6 taxa:\t100.0\n",
      "Loci shared by 7 taxa:\t100.0\n",
      "Loci shared by 8 taxa:\t100.0\n",
      "Loci shared by 9 taxa:\t99.0\n",
      "Loci shared by 10 taxa:\t98.0\n",
      "Loci shared by 11 taxa:\t97.0\n",
      "Loci shared by 12 taxa:\t97.0\n",
      "Loci shared by 13 taxa:\t97.0\n",
      "Loci shared by 14 taxa:\t97.0\n",
      "Loci shared by 15 taxa:\t97.0\n",
      "Loci shared by 16 taxa:\t96.0\n",
      "Loci shared by 17 taxa:\t95.0\n",
      "Loci shared by 18 taxa:\t95.0\n",
      "Loci shared by 19 taxa:\t95.0\n",
      "Loci shared by 20 taxa:\t95.0\n",
      "Loci shared by 21 taxa:\t95.0\n",
      "Loci shared by 22 taxa:\t95.0\n",
      "Loci shared by 23 taxa:\t95.0\n",
      "Loci shared by 24 taxa:\t95.0\n",
      "Loci shared by 25 taxa:\t94.0\n",
      "Loci shared by 26 taxa:\t94.0\n",
      "Loci shared by 27 taxa:\t94.0\n",
      "Loci shared by 28 taxa:\t94.0\n",
      "Loci shared by 29 taxa:\t94.0\n",
      "Loci shared by 30 taxa:\t94.0\n",
      "Loci shared by 31 taxa:\t94.0\n",
      "Loci shared by 32 taxa:\t92.0\n",
      "Loci shared by 33 taxa:\t92.0\n",
      "Loci shared by 34 taxa:\t91.0\n",
      "Loci shared by 35 taxa:\t91.0\n",
      "Loci shared by 36 taxa:\t89.0\n",
      "Loci shared by 37 taxa:\t86.0\n",
      "Loci shared by 38 taxa:\t83.0\n",
      "Loci shared by 39 taxa:\t83.0\n",
      "Loci shared by 40 taxa:\t82.0\n",
      "Loci shared by 41 taxa:\t82.0\n",
      "Loci shared by 42 taxa:\t82.0\n",
      "Loci shared by 43 taxa:\t82.0\n",
      "Loci shared by 44 taxa:\t82.0\n",
      "Loci shared by 45 taxa:\t82.0\n",
      "Loci shared by 46 taxa:\t82.0\n",
      "Loci shared by 47 taxa:\t82.0\n",
      "Loci shared by 48 taxa:\t82.0\n",
      "Loci shared by 49 taxa:\t82.0\n",
      "Loci shared by 50 taxa:\t82.0\n",
      "Loci shared by 51 taxa:\t82.0\n",
      "Loci shared by 52 taxa:\t82.0\n",
      "Loci shared by 53 taxa:\t82.0\n",
      "Loci shared by 54 taxa:\t82.0\n",
      "Loci shared by 55 taxa:\t82.0\n",
      "Loci shared by 56 taxa:\t82.0\n",
      "Loci shared by 57 taxa:\t82.0\n",
      "Loci shared by 58 taxa:\t82.0\n",
      "Loci shared by 59 taxa:\t82.0\n",
      "Loci shared by 60 taxa:\t82.0\n",
      "Loci shared by 61 taxa:\t82.0\n",
      "Loci shared by 62 taxa:\t82.0\n",
      "Loci shared by 63 taxa:\t81.0\n",
      "Loci shared by 64 taxa:\t81.0\n",
      "Loci shared by 65 taxa:\t81.0\n",
      "Loci shared by 66 taxa:\t81.0\n",
      "Loci shared by 67 taxa:\t81.0\n",
      "Loci shared by 68 taxa:\t81.0\n",
      "Loci shared by 69 taxa:\t81.0\n",
      "Loci shared by 70 taxa:\t81.0\n",
      "Loci shared by 71 taxa:\t81.0\n",
      "Loci shared by 72 taxa:\t81.0\n",
      "Loci shared by 73 taxa:\t81.0\n",
      "Loci shared by 74 taxa:\t81.0\n",
      "Loci shared by 75 taxa:\t81.0\n",
      "Loci shared by 76 taxa:\t81.0\n",
      "Loci shared by 77 taxa:\t81.0\n",
      "Loci shared by 78 taxa:\t80.0\n",
      "Loci shared by 79 taxa:\t80.0\n",
      "Loci shared by 80 taxa:\t80.0\n",
      "Loci shared by 81 taxa:\t80.0\n",
      "Loci shared by 82 taxa:\t80.0\n",
      "Loci shared by 83 taxa:\t80.0\n",
      "Loci shared by 84 taxa:\t80.0\n",
      "Loci shared by 85 taxa:\t78.0\n",
      "Loci shared by 86 taxa:\t76.0\n",
      "Loci shared by 87 taxa:\t76.0\n",
      "Loci shared by 88 taxa:\t76.0\n",
      "Loci shared by 89 taxa:\t67.0\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/extract_probes_from_group/\"\n",
    "o_probe_dir = o_probe_dir\n",
    "o_sqlite_db = o_dir + \"multifastas.sqlite\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_multi_fasta_table \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--base-taxon {};\").format(o_probe_dir, \n",
    "                                                 o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {} \"\n",
    "                      \"--base-taxon {};\").format(o_sqlite_db, \n",
    "                                                 reference_taxon)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_query_multi_fasta_table --db results/phyluce/streptococcus/extract_probes_from_group/multifastas.sqlite  --base-taxon Streptococcus_pneumoniae_GCF_000007045 --output results/phyluce/streptococcus/extract_probes_from_group/Streptococcus_pneumoniae_GCF_000007045+41-back-to-77.conf --specific-counts 77;\n",
      "Counter({'streptococcus_varani_gcf_001375655': 81, 'streptococcus_australis_gcf_900476055': 81, 'streptococcus_anginosus_gcf_000463505': 81, 'streptococcus_timonensis_gcf_900095845': 81, 'streptococcus_gallolyticus_gcf_002000985': 81, 'streptococcus_mutans_gcf_000007465': 81, 'streptococcus_intermedius_gcf_000463355': 81, 'streptococcus_infantarius_gcf_000246835': 81, 'streptococcus_equi_gcf_000026605': 81, 'streptococcus_respraculi_gcf_003595525': 81, 'streptococcus_pluranimalium_gcf_003352995': 81, 'streptococcus_uberis_gcf_000009545': 81, 'streptococcus_danieliae_gcf_009767945': 81, 'streptococcus_constellatus_gcf_000463425': 81, 'streptococcus_parauberis_gcf_000213825': 81, 'streptococcus_sinensis_gcf_000767835': 81, 'streptococcus_halichoeri_gcf_009870755': 81, 'streptococcus_infantis_gcf_000187465': 81, 'streptococcus_dysgalactiae_gcf_000317855': 81, 'streptococcus_gordonii_gcf_000017005': 81, 'streptococcus_troglodytae_gcf_002355215': 81, 'streptococcus_ruminantium_gcf_003609975': 81, 'streptococcus_viridans_gcf_900636365': 81, 'streptococcus_equinus_gcf_000964315': 81, 'streptococcus_minor_gcf_000377005': 81, 'streptococcus_rubneri_gcf_004785935': 81, 'streptococcus_milleri_gcf_900636715': 81, 'streptococcus_cuniculi_gcf_001921845': 81, 'streptococcus_thermophilus_gcf_000253395': 81, 'streptococcus_pyogenes_gcf_000006785': 81, 'streptococcus_pseudopneumoniae_gcf_000221985': 81, 'streptococcus_penaeicida_gcf_002887775': 81, 'streptococcus_azizii_gcf_001984715': 81, 'streptococcus_marmotae_gcf_001623565': 81, 'streptococcus_sp_gcf_003086355': 81, 'streptococcus_urinalis_gcf_000188055': 81, 'streptococcus_ictaluri_gcf_000188015': 81, 'streptococcus_himalayensis_gcf_001708305': 81, 'streptococcus_sanguinis_gcf_000014205': 81, 'streptococcus_ratti_gcf_000286075': 81, 'streptococcus_ferus_gcf_000372425': 81, 'streptococcus_didelphis_gcf_000380005': 81, 'streptococcus_entericus_gcf_000380025': 81, 'streptococcus_castoreus_gcf_000425025': 81, 'streptococcus_pneumoniae_gcf_000007045': 81, 'streptococcus_halitosis_gcf_003143695': 81, 'streptococcus_salivarius_gcf_000785515': 81, 'streptococcus_caballi_gcf_000379985': 81, 'streptococcus_plurextorum_gcf_000423745': 81, 'streptococcus_pseudoporcinus_gcf_000188035': 81, 'streptococcus_suis_gcf_000026745': 81, 'streptococcus_canis_gcf_900636575': 81, 'streptococcus_cristatus_gcf_000385925': 81, 'streptococcus_pantholopis_gcf_001642085': 81, 'streptococcus_iniae_gcf_000831485': 81, 'streptococcus_halotolerans_gcf_001598035': 81, 'streptococcus_parasanguinis_gcf_000164675': 81, 'streptococcus_marimammalium_gcf_000380045': 81, 'streptococcus_orisasini_gcf_001431045': 81, 'streptococcus_pharyngis_gcf_007859195': 81, 'streptococcus_devriesei_gcf_000423725': 81, 'streptococcus_porcinus_gcf_900475415': 81, 'streptococcus_parasuis_gcf_004283785': 81, 'streptococcus_porci_gcf_000423765': 81, 'streptococcus_agalactiae_gcf_000007265': 80, 'streptococcus_oralis_gcf_900637025': 80, 'streptococcus_alactolyticus_gcf_009695625': 80, 'streptococcus_merionis_gcf_000380085': 80, 'streptococcus_acidominimus_gcf_900187245': 80, 'streptococcus_hyovaginalis_gcf_000420785': 80, 'streptococcus_hyointestinalis_gcf_900459405': 80, 'streptococcus_mitis_gcf_000027165': 80, 'streptococcus_orisratti_gcf_000380105': 80, 'streptococcus_henryi_gcf_000376985': 80, 'streptococcus_pasteurianus_gcf_900478025': 80, 'streptococcus_lutetiensis_gcf_900475675': 80, 'streptococcus_bovimastitidis_gcf_001885095': 80, 'streptococcus_macedonicus_gcf_000283635': 80, 'streptococcus_ovis_gcf_000380125': 79, 'streptococcus_massiliensis_gcf_000380065': 79, 'streptococcus_vestibularis_gcf_900636445': 79, 'streptococcus_macacae_gcf_000187995': 79, 'streptococcus_sobrinus_gcf_000686605': 79, 'streptococcus_phocae_gcf_000772915': 79, 'streptococcus_hongkongensis_gcf_000785845': 79, 'streptococcus_criceti_gcf_000187975': 79, 'streptococcus_downei_gcf_900459175': 78, 'streptococcus_thoraltensis_gcf_000380145': 78, 'streptococcus_peroris_gcf_000187585': 78})\n",
      "Total loci = 81\n"
     ]
    }
   ],
   "source": [
    "i_sqlite_db     = o_sqlite_db\n",
    "target_num_taxa = str(77)\n",
    "o_conf          = \"\".join([o_dir, reference_taxon, \"+\", str(num_taxa), \"-back-to-\", target_num_taxa, \".conf\"])\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_query_multi_fasta_table \"\n",
    "                      \"--db {}  \"\n",
    "                      \"--base-taxon {} \"\n",
    "                      \"--output {} \"\n",
    "                      \"--specific-counts {};\").format(i_sqlite_db, \n",
    "                                                      reference_taxon, \n",
    "                                                      o_conf, \n",
    "                                                      target_num_taxa)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Final group specific bait design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_get_tiled_probe_from_multiple_inputs --fastas results/phyluce/streptococcus/extract_probes_from_group/probe_fasta --multi-fasta-output results/phyluce/streptococcus/extract_probes_from_group/Streptococcus_pneumoniae_GCF_000007045+41-back-to-77.conf --probe-prefix uce_streptococcus_ --designer rnplattii --design streptococcus_v1 --probe-length 80 --tiling-density 3 --overlap middle --masking 0.25 --remove-gc --two-probes --output results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta;\n",
      "GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG\n",
      "\n",
      "\n",
      "Conserved locus count = 81\n",
      "Probe Count = 14298\n"
     ]
    }
   ],
   "source": [
    "o_dir       = \"results/phyluce/\" + group + \"/final_probe_design/\"\n",
    "\n",
    "if os.path.exists(o_dir):\n",
    "    shutil.rmtree(o_dir)\n",
    "\n",
    "os.mkdir(o_dir)\n",
    "\n",
    "i_probe_dir = o_probe_dir\n",
    "i_conf = o_conf\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "i_designer_txt = \"rnplattii\"\n",
    "i_design_txt   = group + \"_v1\"\n",
    "i_tiling_density = \"3\"\n",
    "\n",
    "o_probes_fas     = o_dir + i_design_txt + \"-master_probe_list.fasta\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_get_tiled_probe_from_multiple_inputs \"\n",
    "                      \"--fastas {} \"\n",
    "                      \"--multi-fasta-output {} \"\n",
    "                      \"--probe-prefix {} \"\n",
    "                      \"--designer {} \"\n",
    "                      \"--design {} \"\n",
    "                      \"--probe-length 80 \"\n",
    "                      \"--tiling-density 3 \"\n",
    "                      \"--overlap middle \"\n",
    "                      \"--masking 0.25 \"\n",
    "                      \"--remove-gc \"\n",
    "                      \"--two-probes \"\n",
    "                      \"--output {};\").format(i_probe_dir, \n",
    "                                             i_conf, i_probe_prefix, \n",
    "                                             i_designer_txt, \n",
    "                                             i_design_txt, \n",
    "                                             o_probes_fas)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_easy_lastz --target results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta --query results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta --identity 50 --coverage 50 --output results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list-TO-SELF-PROBES.lastz;\n",
      "Started:  Thu Feb 06, 2020  15:42:14\n",
      "Ended:  Thu Feb 06, 2020  15:44:20\n",
      "Time for execution:  2.11134323279 minutes\n"
     ]
    }
   ],
   "source": [
    "i_probes_fas   = o_probes_fas\n",
    "o_lastz_tbl    = o_dir + i_design_txt + \"-master_probe_list-TO-SELF-PROBES.lastz\"\n",
    "\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_easy_lastz \"\n",
    "                      \"--target {} \"\n",
    "                      \"--query {} \"\n",
    "                      \"--identity 50 \"\n",
    "                      \"--coverage 50 \"\n",
    "                      \"--output {};\").format(o_probes_fas, \n",
    "                                             o_probes_fas, \n",
    "                                             o_lastz_tbl)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phyluce_probe_remove_duplicate_hits_from_probes_using_lastz --fasta results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta --lastz results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list-TO-SELF-PROBES.lastz --probe-prefix=uce_streptococcus_;\n",
      "Parsing lastz file...\n",
      "Screening results...\n",
      "Screened 14297 fasta sequences.  Filtered 0 duplicates. Kept 14298.\n"
     ]
    }
   ],
   "source": [
    "i_lastz_tbl = o_lastz_tbl\n",
    "i_probe_prefix = \"uce_\" + group + \"_\"\n",
    "\n",
    "phyluce_cmd = str(\"phyluce_probe_remove_duplicate_hits_from_probes_using_lastz \"\n",
    "                      \"--fasta {} \"\n",
    "                      \"--lastz {} \"\n",
    "                      \"--probe-prefix={};\").format(i_probes_fas, \n",
    "                                                   i_lastz_tbl, \n",
    "                                                   i_probe_prefix)\n",
    "\n",
    "print(phyluce_cmd)\n",
    "!{phyluce_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CDhit to reduce numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "Program: CD-HIT, V4.8.1 (+OpenMP), Oct 26 2019, 14:51:47\n",
      "Command: cd-hit-est -c 0.95 -G 1 -T 4 -i\n",
      "         results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.fasta\n",
      "         -o\n",
      "         results/phyluce/streptococcus/final_probe_design/streptococcus_v1-master_probe_list.95P_cdhit\n",
      "\n",
      "Started: Thu Feb  6 16:37:39 2020\n",
      "================================================================\n",
      "                            Output                              \n",
      "----------------------------------------------------------------\n",
      "total seq: 14298\n",
      "longest and shortest : 80 and 80\n",
      "Total letters: 1143840\n",
      "Sequences have been sorted\n",
      "\n",
      "Approximated minimal memory consumption:\n",
      "Sequence        : 3M\n",
      "Buffer          : 4 X 12M = 48M\n",
      "Table           : 2 X 17M = 34M\n",
      "Miscellaneous   : 4M\n",
      "Total           : 90M\n",
      "\n",
      "Table limit with the given memory limit:\n",
      "Max number of representatives: 3940576\n",
      "Max number of word counting entries: 88702384\n",
      "\n",
      "# comparing sequences from          0  to       2383\n",
      "..---------- new table with     1206 representatives\n",
      "# comparing sequences from       2383  to       4368\n",
      "----------   1283 remaining sequences to the next cycle\n",
      "---------- new table with      293 representatives\n",
      "# comparing sequences from       3085  to       4953\n",
      "----------   1166 remaining sequences to the next cycle\n",
      "---------- new table with      303 representatives\n",
      "# comparing sequences from       3787  to       5538\n",
      "----------   1151 remaining sequences to the next cycle\n",
      "---------- new table with      322 representatives\n",
      "# comparing sequences from       4387  to       6038\n",
      "----------   1028 remaining sequences to the next cycle\n",
      "---------- new table with      378 representatives\n",
      "# comparing sequences from       5010  to       6558\n",
      "----------    996 remaining sequences to the next cycle\n",
      "---------- new table with      328 representatives\n",
      "# comparing sequences from       5562  to       7018\n",
      "----------    851 remaining sequences to the next cycle\n",
      "---------- new table with      424 representatives\n",
      "# comparing sequences from       6167  to       7522\n",
      "----------    973 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from       6549  to       7840\n",
      "----------    798 remaining sequences to the next cycle\n",
      "---------- new table with      334 representatives\n",
      "# comparing sequences from       7042  to       8251\n",
      "----------    823 remaining sequences to the next cycle\n",
      "---------- new table with      136 representatives\n",
      "# comparing sequences from       7428  to       8573\n",
      "----------    644 remaining sequences to the next cycle\n",
      "---------- new table with      294 representatives\n",
      "# comparing sequences from       7929  to       8990\n",
      "----------    715 remaining sequences to the next cycle\n",
      "---------- new table with      209 representatives\n",
      "# comparing sequences from       8275  to       9278\n",
      "----------    708 remaining sequences to the next cycle\n",
      "---------- new table with      108 representatives\n",
      "# comparing sequences from       8570  to       9524\n",
      "----------    521 remaining sequences to the next cycle\n",
      "---------- new table with      307 representatives\n",
      "# comparing sequences from       9003  to       9885\n",
      "----------    404 remaining sequences to the next cycle\n",
      "---------- new table with      165 representatives\n",
      "# comparing sequences from       9481  to      10283\n",
      "----------    530 remaining sequences to the next cycle\n",
      "---------- new table with      163 representatives\n",
      "# comparing sequences from       9753  to      10510\n",
      "..........    10000  finished       5197  clusters\n",
      "----------    359 remaining sequences to the next cycle\n",
      "---------- new table with      199 representatives\n",
      "# comparing sequences from      10151  to      10842\n",
      "----------    423 remaining sequences to the next cycle\n",
      "---------- new table with      130 representatives\n",
      "# comparing sequences from      10419  to      11065\n",
      "----------    373 remaining sequences to the next cycle\n",
      "---------- new table with      122 representatives\n",
      "# comparing sequences from      10692  to      11293\n",
      "----------    271 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      11022  to      11568\n",
      "----------    250 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      11318  to      11814\n",
      "----------    205 remaining sequences to the next cycle\n",
      "---------- new table with      151 representatives\n",
      "# comparing sequences from      11609  to      12057\n",
      "----------    226 remaining sequences to the next cycle\n",
      "---------- new table with      139 representatives\n",
      "# comparing sequences from      11831  to      12242\n",
      "----------    159 remaining sequences to the next cycle\n",
      "---------- new table with      100 representatives\n",
      "# comparing sequences from      12083  to      12452\n",
      "----------     72 remaining sequences to the next cycle\n",
      "---------- new table with      163 representatives\n",
      "# comparing sequences from      12380  to      12699\n",
      "----------     97 remaining sequences to the next cycle\n",
      "---------- new table with      160 representatives\n",
      "# comparing sequences from      12602  to      12884\n",
      "----------     21 remaining sequences to the next cycle\n",
      "---------- new table with      182 representatives\n",
      "# comparing sequences from      12863  to      13102\n",
      ".....................---------- new table with       92 representatives\n",
      "# comparing sequences from      13102  to      13301\n",
      "....................---------- new table with       61 representatives\n",
      "# comparing sequences from      13301  to      14298\n",
      "....................---------- new table with      502 representatives\n",
      "\n",
      "    14298  finished       7271  clusters\n",
      "\n",
      "Approximated maximum memory consumption: 91M\n",
      "writing new database\n",
      "writing clustering information\n",
      "program completed !\n",
      "\n",
      "Total CPU time 3.16\n"
     ]
    }
   ],
   "source": [
    "i_probe_fasta = o_probes_fas \n",
    "o_clust_prfx = o_probes_fas.replace(\"fasta\", \"95P_cdhit\")\n",
    "\n",
    "cdhit_cmd = \"cd-hit-est -c 0.95 -G 1 -T 4 -i {} -o {}\".format(i_probe_fasta, o_clust_prfx)\n",
    "\n",
    "!{cdhit_cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "319px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
